<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Example | Statistical Analysis in Metabolomics</title>
  <meta name="description" content="A template for Statistical Analysis in Metabolomics" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Example | Statistical Analysis in Metabolomics" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A template for Statistical Analysis in Metabolomics" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Example | Statistical Analysis in Metabolomics" />
  
  <meta name="twitter:description" content="A template for Statistical Analysis in Metabolomics" />
  

<meta name="author" content="Hua Zou" />


<meta name="date" content="2022-08-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="network-analysis.html"/>
<link rel="next" href="references.html"/>
<script src="libs/header-attrs-2.13/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="assets/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Statistical Analysis in Metabolomics</a></li>

<li class="divider"></li>
<li><a href="index.html#introduction" id="toc-introduction"><span class="toc-section-number">1</span> Introduction<span></span></a>
<ul>
<li><a href="index.html#what-is-metabolomics" id="toc-what-is-metabolomics"><span class="toc-section-number">1.1</span> What is metabolomics?<span></span></a></li>
<li><a href="index.html#what-are-the-procedures-in-this-template" id="toc-what-are-the-procedures-in-this-template"><span class="toc-section-number">1.2</span> What are the procedures in this template?<span></span></a></li>
<li><a href="index.html#software" id="toc-software"><span class="toc-section-number">1.3</span> Software<span></span></a></li>
<li><a href="index.html#reference" id="toc-reference"><span class="toc-section-number">1.4</span> Reference<span></span></a></li>
</ul></li>
<li><a href="data-processing.html#data-processing" id="toc-data-processing"><span class="toc-section-number">2</span> Data Processing<span></span></a>
<ul>
<li><a href="data-processing.html#loading-packages" id="toc-loading-packages"><span class="toc-section-number">2.1</span> Loading packages<span></span></a></li>
<li><a href="data-processing.html#importing-data" id="toc-importing-data"><span class="toc-section-number">2.2</span> Importing data<span></span></a></li>
<li><a href="data-processing.html#data-checking" id="toc-data-checking"><span class="toc-section-number">2.3</span> Data Checking<span></span></a></li>
<li><a href="data-processing.html#data-filtering" id="toc-data-filtering"><span class="toc-section-number">2.4</span> Data Filtering<span></span></a></li>
<li><a href="data-processing.html#missing-value-imputation" id="toc-missing-value-imputation"><span class="toc-section-number">2.5</span> Missing Value Imputation<span></span></a></li>
<li><a href="data-processing.html#data-normalization" id="toc-data-normalization"><span class="toc-section-number">2.6</span> Data Normalization<span></span></a>
<ul>
<li><a href="data-processing.html#normalization-by-normalizedata-function" id="toc-normalization-by-normalizedata-function"><span class="toc-section-number">2.6.1</span> Normalization by <strong>NormalizeData</strong> function<span></span></a></li>
<li><a href="data-processing.html#normalization-by-poma-r-package" id="toc-normalization-by-poma-r-package"><span class="toc-section-number">2.6.2</span> Normalization by <strong>POMA</strong> R package<span></span></a></li>
<li><a href="data-processing.html#comparison-of-unnormalized-and-normalized-dataset" id="toc-comparison-of-unnormalized-and-normalized-dataset"><span class="toc-section-number">2.6.3</span> Comparison of unnormalized and normalized dataset<span></span></a></li>
</ul></li>
<li><a href="data-processing.html#removing-outliers" id="toc-removing-outliers"><span class="toc-section-number">2.7</span> Removing outliers<span></span></a></li>
<li><a href="data-processing.html#saving-datasets-into-rds-files" id="toc-saving-datasets-into-rds-files"><span class="toc-section-number">2.8</span> Saving datasets into RDS files<span></span></a></li>
<li><a href="data-processing.html#systematic-information" id="toc-systematic-information"><span class="toc-section-number">2.9</span> Systematic Information<span></span></a></li>
</ul></li>
<li><a href="cluster-analysis.html#cluster-analysis" id="toc-cluster-analysis"><span class="toc-section-number">3</span> Cluster Analysis<span></span></a>
<ul>
<li><a href="cluster-analysis.html#loading-packages-1" id="toc-loading-packages-1"><span class="toc-section-number">3.1</span> Loading packages<span></span></a></li>
<li><a href="cluster-analysis.html#importing-data-1" id="toc-importing-data-1"><span class="toc-section-number">3.2</span> Importing data<span></span></a></li>
<li><a href="cluster-analysis.html#hierarchical-clustering" id="toc-hierarchical-clustering"><span class="toc-section-number">3.3</span> Hierarchical Clustering<span></span></a>
<ul>
<li><a href="cluster-analysis.html#agglomerative-hierarchical-clustering" id="toc-agglomerative-hierarchical-clustering"><span class="toc-section-number">3.3.1</span> Agglomerative Hierarchical Clustering<span></span></a></li>
<li><a href="cluster-analysis.html#divisive-hierarchical-clustering" id="toc-divisive-hierarchical-clustering"><span class="toc-section-number">3.3.2</span> Divisive Hierarchical Clustering<span></span></a></li>
<li><a href="cluster-analysis.html#comparison" id="toc-comparison"><span class="toc-section-number">3.3.3</span> Comparison<span></span></a></li>
<li><a href="cluster-analysis.html#determining-optimal-clusters" id="toc-determining-optimal-clusters"><span class="toc-section-number">3.3.4</span> Determining Optimal Clusters<span></span></a></li>
</ul></li>
<li><a href="cluster-analysis.html#partitional-clustering" id="toc-partitional-clustering"><span class="toc-section-number">3.4</span> Partitional Clustering<span></span></a></li>
<li><a href="cluster-analysis.html#systematic-information-1" id="toc-systematic-information-1"><span class="toc-section-number">3.5</span> Systematic Information<span></span></a></li>
</ul></li>
<li><a href="chemometrics-analysis.html#chemometrics-analysis" id="toc-chemometrics-analysis"><span class="toc-section-number">4</span> Chemometrics Analysis<span></span></a>
<ul>
<li><a href="chemometrics-analysis.html#loading-packages-2" id="toc-loading-packages-2"><span class="toc-section-number">4.1</span> Loading packages<span></span></a></li>
<li><a href="chemometrics-analysis.html#importing-data-2" id="toc-importing-data-2"><span class="toc-section-number">4.2</span> Importing data<span></span></a></li>
<li><a href="chemometrics-analysis.html#principal-component-analysis-pca" id="toc-principal-component-analysis-pca"><span class="toc-section-number">4.3</span> Principal Component Analysis (PCA)<span></span></a></li>
<li><a href="chemometrics-analysis.html#partial-least-squares-discriminant-analysis-pls-da" id="toc-partial-least-squares-discriminant-analysis-pls-da"><span class="toc-section-number">4.4</span> Partial Least Squares-Discriminant Analysis (PLS-DA)<span></span></a></li>
<li><a href="chemometrics-analysis.html#sparse-partial-least-squares-discriminant-analysis-spls-da" id="toc-sparse-partial-least-squares-discriminant-analysis-spls-da"><span class="toc-section-number">4.5</span> Sparse Partial Least Squares-Discriminant Analysis (sPLS-DA)<span></span></a></li>
<li><a href="chemometrics-analysis.html#orthogonal-partial-least-squares-discriminant-analysis-orthopls-da" id="toc-orthogonal-partial-least-squares-discriminant-analysis-orthopls-da"><span class="toc-section-number">4.6</span> Orthogonal Partial Least Squares-Discriminant Analysis (orthoPLS-DA)<span></span></a></li>
<li><a href="chemometrics-analysis.html#systematic-information-2" id="toc-systematic-information-2"><span class="toc-section-number">4.7</span> Systematic Information<span></span></a></li>
</ul></li>
<li><a href="univariate-analysis.html#univariate-analysis" id="toc-univariate-analysis"><span class="toc-section-number">5</span> Univariate Analysis<span></span></a>
<ul>
<li><a href="univariate-analysis.html#loading-packages-3" id="toc-loading-packages-3"><span class="toc-section-number">5.1</span> Loading packages<span></span></a></li>
<li><a href="univariate-analysis.html#importing-data-3" id="toc-importing-data-3"><span class="toc-section-number">5.2</span> Importing data<span></span></a></li>
<li><a href="univariate-analysis.html#fold-change-analysis" id="toc-fold-change-analysis"><span class="toc-section-number">5.3</span> Fold Change Analysis<span></span></a></li>
<li><a href="univariate-analysis.html#t-test" id="toc-t-test"><span class="toc-section-number">5.4</span> T Test<span></span></a></li>
<li><a href="univariate-analysis.html#wilcoxon-test" id="toc-wilcoxon-test"><span class="toc-section-number">5.5</span> Wilcoxon Test<span></span></a></li>
<li><a href="univariate-analysis.html#limma-test" id="toc-limma-test"><span class="toc-section-number">5.6</span> Limma Test<span></span></a></li>
<li><a href="univariate-analysis.html#volcano-plot" id="toc-volcano-plot"><span class="toc-section-number">5.7</span> Volcano plot<span></span></a></li>
<li><a href="univariate-analysis.html#correlation-heatmaps" id="toc-correlation-heatmaps"><span class="toc-section-number">5.8</span> Correlation Heatmaps<span></span></a></li>
<li><a href="univariate-analysis.html#glasso-this-function-will-compute-a-gaussian-graphical-model-using-the-glmnet-package" id="toc-glasso-this-function-will-compute-a-gaussian-graphical-model-using-the-glmnet-package"><span class="toc-section-number">5.9</span> glasso: this function will compute a Gaussian Graphical Model using the glmnet package<span></span></a></li>
<li><a href="univariate-analysis.html#systematic-information-3" id="toc-systematic-information-3"><span class="toc-section-number">5.10</span> Systematic Information<span></span></a></li>
</ul></li>
<li><a href="feature-selection.html#feature-selection" id="toc-feature-selection"><span class="toc-section-number">6</span> Feature Selection<span></span></a>
<ul>
<li><a href="feature-selection.html#importing-data-4" id="toc-importing-data-4"><span class="toc-section-number">6.1</span> Importing data<span></span></a></li>
<li><a href="feature-selection.html#data-curation" id="toc-data-curation"><span class="toc-section-number">6.2</span> Data curation<span></span></a></li>
<li><a href="feature-selection.html#regularized-generalized-linear-models" id="toc-regularized-generalized-linear-models"><span class="toc-section-number">6.3</span> Regularized Generalized Linear Models<span></span></a>
<ul>
<li><a href="feature-selection.html#lasso-alpha-1" id="toc-lasso-alpha-1"><span class="toc-section-number">6.3.1</span> Lasso: alpha = 1<span></span></a></li>
<li><a href="feature-selection.html#ridge-alpha-0" id="toc-ridge-alpha-0"><span class="toc-section-number">6.3.2</span> Ridge: alpha = 0<span></span></a></li>
<li><a href="feature-selection.html#elasticnet-0-alpha-1" id="toc-elasticnet-0-alpha-1"><span class="toc-section-number">6.3.3</span> Elasticnet: 0 &lt; alpha &lt; 1<span></span></a></li>
</ul></li>
<li><a href="feature-selection.html#classification" id="toc-classification"><span class="toc-section-number">6.4</span> Classification<span></span></a>
<ul>
<li><a href="feature-selection.html#random-forest" id="toc-random-forest"><span class="toc-section-number">6.4.1</span> Random Forest<span></span></a></li>
<li><a href="feature-selection.html#support-vector-machine-svm" id="toc-support-vector-machine-svm"><span class="toc-section-number">6.4.2</span> Support Vector Machine (SVM)<span></span></a></li>
</ul></li>
<li><a href="feature-selection.html#systematic-information-4" id="toc-systematic-information-4"><span class="toc-section-number">6.5</span> Systematic Information<span></span></a></li>
</ul></li>
<li><a href="network-analysis.html#network-analysis" id="toc-network-analysis"><span class="toc-section-number">7</span> Network Analysis<span></span></a>
<ul>
<li><a href="network-analysis.html#introduction-1" id="toc-introduction-1"><span class="toc-section-number">7.1</span> Introduction<span></span></a></li>
<li><a href="network-analysis.html#loading-packages-4" id="toc-loading-packages-4"><span class="toc-section-number">7.2</span> Loading packages<span></span></a></li>
<li><a href="network-analysis.html#importing-data-5" id="toc-importing-data-5"><span class="toc-section-number">7.3</span> Importing data<span></span></a></li>
<li><a href="network-analysis.html#data-curation-1" id="toc-data-curation-1"><span class="toc-section-number">7.4</span> Data curation<span></span></a></li>
<li><a href="network-analysis.html#single-network-with-spring-as-association-measure" id="toc-single-network-with-spring-as-association-measure"><span class="toc-section-number">7.5</span> Single network with SPRING as association measure<span></span></a>
<ul>
<li><a href="network-analysis.html#building-network-model" id="toc-building-network-model"><span class="toc-section-number">7.5.1</span> Building network model<span></span></a></li>
<li><a href="network-analysis.html#analyzing-the-constructed-network" id="toc-analyzing-the-constructed-network"><span class="toc-section-number">7.5.2</span> Analyzing the constructed network<span></span></a></li>
<li><a href="network-analysis.html#visualizing-the-network" id="toc-visualizing-the-network"><span class="toc-section-number">7.5.3</span> Visualizing the network<span></span></a></li>
</ul></li>
<li><a href="network-analysis.html#single-network-with-pearson-correlation-as-association-measure" id="toc-single-network-with-pearson-correlation-as-association-measure"><span class="toc-section-number">7.6</span> Single network with Pearson correlation as association measure<span></span></a>
<ul>
<li><a href="network-analysis.html#building-network-model-1" id="toc-building-network-model-1"><span class="toc-section-number">7.6.1</span> Building network model<span></span></a></li>
<li><a href="network-analysis.html#visualizing-the-network-1" id="toc-visualizing-the-network-1"><span class="toc-section-number">7.6.2</span> Visualizing the network<span></span></a></li>
</ul></li>
<li><a href="network-analysis.html#single-network-with-spieceasi-correlation-as-association-measure" id="toc-single-network-with-spieceasi-correlation-as-association-measure"><span class="toc-section-number">7.7</span> Single network with SpiecEasi correlation as association measure<span></span></a>
<ul>
<li><a href="network-analysis.html#building-network-model-2" id="toc-building-network-model-2"><span class="toc-section-number">7.7.1</span> Building network model<span></span></a></li>
<li><a href="network-analysis.html#visualizing-the-network-2" id="toc-visualizing-the-network-2"><span class="toc-section-number">7.7.2</span> Visualizing the network<span></span></a></li>
</ul></li>
<li><a href="network-analysis.html#single-network-with-wgcna-bicor-as-association-measure" id="toc-single-network-with-wgcna-bicor-as-association-measure"><span class="toc-section-number">7.8</span> Single network with WGCNA (bicor) as association measure<span></span></a>
<ul>
<li><a href="network-analysis.html#building-network-model-3" id="toc-building-network-model-3"><span class="toc-section-number">7.8.1</span> Building network model<span></span></a></li>
<li><a href="network-analysis.html#visualizing-the-network-3" id="toc-visualizing-the-network-3"><span class="toc-section-number">7.8.2</span> Visualizing the network<span></span></a></li>
</ul></li>
<li><a href="network-analysis.html#single-network-with-sparcc-correlation-as-association-measure" id="toc-single-network-with-sparcc-correlation-as-association-measure"><span class="toc-section-number">7.9</span> Single network with sparcc correlation as association measure<span></span></a>
<ul>
<li><a href="network-analysis.html#building-network-model-4" id="toc-building-network-model-4"><span class="toc-section-number">7.9.1</span> Building network model<span></span></a></li>
<li><a href="network-analysis.html#visualizing-the-network-4" id="toc-visualizing-the-network-4"><span class="toc-section-number">7.9.2</span> Visualizing the network<span></span></a></li>
</ul></li>
<li><a href="network-analysis.html#dissimilarity-based-networks" id="toc-dissimilarity-based-networks"><span class="toc-section-number">7.10</span> Dissimilarity-based Networks<span></span></a>
<ul>
<li><a href="network-analysis.html#visualizing-the-network-5" id="toc-visualizing-the-network-5"><span class="toc-section-number">7.10.1</span> Visualizing the network<span></span></a></li>
</ul></li>
<li><a href="network-analysis.html#network-comparison" id="toc-network-comparison"><span class="toc-section-number">7.11</span> Network comparison<span></span></a>
<ul>
<li><a href="network-analysis.html#data-preparing" id="toc-data-preparing"><span class="toc-section-number">7.11.1</span> Data preparing<span></span></a></li>
<li><a href="network-analysis.html#building-network-model-5" id="toc-building-network-model-5"><span class="toc-section-number">7.11.2</span> Building network model<span></span></a></li>
<li><a href="network-analysis.html#network-analysis-1" id="toc-network-analysis-1"><span class="toc-section-number">7.11.3</span> Network analysis<span></span></a></li>
<li><a href="network-analysis.html#visualizing-the-network-6" id="toc-visualizing-the-network-6"><span class="toc-section-number">7.11.4</span> Visualizing the network<span></span></a></li>
<li><a href="network-analysis.html#quantitative-network-comparison" id="toc-quantitative-network-comparison"><span class="toc-section-number">7.11.5</span> Quantitative network comparison<span></span></a></li>
</ul></li>
<li><a href="network-analysis.html#systematic-information-5" id="toc-systematic-information-5"><span class="toc-section-number">7.12</span> Systematic Information<span></span></a></li>
</ul></li>
<li><a href="example.html#example" id="toc-example"><span class="toc-section-number">8</span> Example<span></span></a>
<ul>
<li><a href="example.html#loading-packages-5" id="toc-loading-packages-5"><span class="toc-section-number">8.1</span> Loading packages<span></span></a></li>
<li><a href="example.html#importing-data-6" id="toc-importing-data-6"><span class="toc-section-number">8.2</span> Importing data<span></span></a></li>
<li><a href="example.html#data-processing-1" id="toc-data-processing-1"><span class="toc-section-number">8.3</span> Data Processing<span></span></a>
<ul>
<li><a href="example.html#data-checking-1" id="toc-data-checking-1"><span class="toc-section-number">8.3.1</span> Data Checking<span></span></a></li>
<li><a href="example.html#data-filtering-1" id="toc-data-filtering-1"><span class="toc-section-number">8.3.2</span> Data Filtering<span></span></a></li>
<li><a href="example.html#missing-value-imputation-1" id="toc-missing-value-imputation-1"><span class="toc-section-number">8.3.3</span> Missing Value Imputation<span></span></a></li>
<li><a href="example.html#data-normalization-1" id="toc-data-normalization-1"><span class="toc-section-number">8.3.4</span> Data Normalization<span></span></a></li>
</ul></li>
<li><a href="example.html#cluster-analysis-1" id="toc-cluster-analysis-1"><span class="toc-section-number">8.4</span> Cluster Analysis<span></span></a>
<ul>
<li><a href="example.html#hierarchical-clustering-1" id="toc-hierarchical-clustering-1"><span class="toc-section-number">8.4.1</span> Hierarchical Clustering<span></span></a></li>
</ul></li>
<li><a href="example.html#chemometrics-analysis-1" id="toc-chemometrics-analysis-1"><span class="toc-section-number">8.5</span> Chemometrics Analysis<span></span></a>
<ul>
<li><a href="example.html#partial-least-squares-discriminant-analysis-pls-da-1" id="toc-partial-least-squares-discriminant-analysis-pls-da-1"><span class="toc-section-number">8.5.1</span> Partial Least Squares-Discriminant Analysis (PLS-DA)<span></span></a></li>
<li><a href="example.html#sparse-partial-least-squares-discriminant-analysis-spls-da-1" id="toc-sparse-partial-least-squares-discriminant-analysis-spls-da-1"><span class="toc-section-number">8.5.2</span> Sparse Partial Least Squares-Discriminant Analysis (sPLS-DA)<span></span></a></li>
</ul></li>
<li><a href="example.html#univariate-analysis-1" id="toc-univariate-analysis-1"><span class="toc-section-number">8.6</span> Univariate Analysis<span></span></a>
<ul>
<li><a href="example.html#fold-change-analysis-1" id="toc-fold-change-analysis-1"><span class="toc-section-number">8.6.1</span> Fold Change Analysis<span></span></a></li>
<li><a href="example.html#t-test-1" id="toc-t-test-1"><span class="toc-section-number">8.6.2</span> T Test<span></span></a></li>
<li><a href="example.html#volcano-plot-1" id="toc-volcano-plot-1"><span class="toc-section-number">8.6.3</span> Volcano plot<span></span></a></li>
</ul></li>
<li><a href="example.html#feature-selection-1" id="toc-feature-selection-1"><span class="toc-section-number">8.7</span> Feature Selection<span></span></a>
<ul>
<li><a href="example.html#regularized-generalized-linear-models-lasso-alpha-1" id="toc-regularized-generalized-linear-models-lasso-alpha-1"><span class="toc-section-number">8.7.1</span> Regularized Generalized Linear Models (Lasso: alpha = 1)<span></span></a></li>
<li><a href="example.html#classification-random-forest" id="toc-classification-random-forest"><span class="toc-section-number">8.7.2</span> Classification (Random Forest)<span></span></a></li>
</ul></li>
<li><a href="example.html#network-analysis-2" id="toc-network-analysis-2"><span class="toc-section-number">8.8</span> Network Analysis<span></span></a>
<ul>
<li><a href="example.html#data-curation-2" id="toc-data-curation-2"><span class="toc-section-number">8.8.1</span> Data curation<span></span></a></li>
<li><a href="example.html#building-network-model-6" id="toc-building-network-model-6"><span class="toc-section-number">8.8.2</span> Building network model<span></span></a></li>
<li><a href="example.html#visualizing-the-network-7" id="toc-visualizing-the-network-7"><span class="toc-section-number">8.8.3</span> Visualizing the network<span></span></a></li>
</ul></li>
<li><a href="example.html#network-analysis-on-wgcna" id="toc-network-analysis-on-wgcna"><span class="toc-section-number">8.9</span> Network Analysis on WGCNA<span></span></a>
<ul>
<li><a href="example.html#data-curation-3" id="toc-data-curation-3"><span class="toc-section-number">8.9.1</span> Data curation<span></span></a></li>
<li><a href="example.html#tuning-soft-thresholds" id="toc-tuning-soft-thresholds"><span class="toc-section-number">8.9.2</span> Tuning soft thresholds<span></span></a></li>
<li><a href="example.html#create-the-network-using-the-blockwisemodules" id="toc-create-the-network-using-the-blockwisemodules"><span class="toc-section-number">8.9.3</span> Create the network using the <code>blockwiseModules</code><span></span></a></li>
<li><a href="example.html#plot-modules" id="toc-plot-modules"><span class="toc-section-number">8.9.4</span> plot modules<span></span></a></li>
<li><a href="example.html#relationships-among-modules" id="toc-relationships-among-modules"><span class="toc-section-number">8.9.5</span> Relationships among modules<span></span></a></li>
<li><a href="example.html#module-eigengene-correlation" id="toc-module-eigengene-correlation"><span class="toc-section-number">8.9.6</span> Module (Eigengene) correlation<span></span></a></li>
<li><a href="example.html#relate-module-cluster-assignments-to-groups" id="toc-relate-module-cluster-assignments-to-groups"><span class="toc-section-number">8.9.7</span> Relate Module (cluster) Assignments to Groups<span></span></a></li>
<li><a href="example.html#generate-and-export-networks" id="toc-generate-and-export-networks"><span class="toc-section-number">8.9.8</span> Generate and Export Networks<span></span></a></li>
<li><a href="example.html#network-visualization" id="toc-network-visualization"><span class="toc-section-number">8.9.9</span> Network visualization<span></span></a></li>
</ul></li>
<li><a href="example.html#systematic-information-6" id="toc-systematic-information-6"><span class="toc-section-number">8.10</span> Systematic Information<span></span></a></li>
</ul></li>
<li><a href="references.html#references" id="toc-references">References<span></span></a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Statistical Analysis in Metabolomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="example" class="section level1 hasAnchor" number="8">
<h1><span class="header-section-number">Chapter 8</span> Example<a href="example.html#example" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>We perform all the data analysis on our own metabolomic data in this chapter. There are several datasets:</p>
<ul>
<li><p><strong>GvHD stool metabolites TM wide-target sequencing</strong>: /home/xuxiaomin/project/NanFangHospitalGvHD/00.data/metabolites_TM/MWY-20-522-01_2021-03-25_17-34-06/1.Data_assess/ALL_sample_data.xlsx</p></li>
<li><p><strong>GvHD stool metabolites SCFA</strong>: /home/xuxiaomin/project/NanFangHospitalGvHD/00.data/metabolites_scfa/SCFA.levels.xlsx</p></li>
<li><p><strong>PD-1 mice trial (round 2) serum metabolites TM wide-target sequencing</strong>: /home/xuxiaomin/project/pd1_mice/Round_2/00.data/MetaboliteSerum/MWY-20-049/1.Data_assess/all_group/ALL_sample_data.xlsx</p></li>
</ul>
<p>and the metadata:</p>
<ul>
<li><strong>GvHD metadata</strong>: /home/xuxiaomin/project/NanFangHospitalGvHD/00.data/metadata/metadata_v4.txt</li>
</ul>
<p>We transform their names as following:</p>
<ul>
<li><p>GvHD_stool_metabolites_TM.xlsx</p></li>
<li><p>GvHD_stool_metabolites_SCFA.xlsx</p></li>
<li><p>PD1_mice_serum_metabolites_TM.xlsx</p></li>
<li><p>GvHD_metadata.txt</p></li>
</ul>
<p>Here, we use the <code>GvHD_stool_metabolites_TM.xlsx</code> to practice our template.</p>
<div id="loading-packages-5" class="section level2 hasAnchor" number="8.1">
<h2><span class="header-section-number">8.1</span> Loading packages<a href="example.html#loading-packages-5" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="example.html#cb131-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">warning =</span> F)</span>
<span id="cb131-2"><a href="example.html#cb131-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-3"><a href="example.html#cb131-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb131-4"><a href="example.html#cb131-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb131-5"><a href="example.html#cb131-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(POMA)</span>
<span id="cb131-6"><a href="example.html#cb131-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb131-7"><a href="example.html#cb131-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggraph)</span>
<span id="cb131-8"><a href="example.html#cb131-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb131-9"><a href="example.html#cb131-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb131-10"><a href="example.html#cb131-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb131-11"><a href="example.html#cb131-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-12"><a href="example.html#cb131-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cluster)    <span class="co"># clustering algorithms</span></span>
<span id="cb131-13"><a href="example.html#cb131-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(factoextra) <span class="co"># clustering visualization</span></span>
<span id="cb131-14"><a href="example.html#cb131-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dendextend) <span class="co"># for comparing two dendrograms</span></span>
<span id="cb131-15"><a href="example.html#cb131-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-16"><a href="example.html#cb131-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(NetCoMi)</span>
<span id="cb131-17"><a href="example.html#cb131-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SPRING)</span>
<span id="cb131-18"><a href="example.html#cb131-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SpiecEasi)</span>
<span id="cb131-19"><a href="example.html#cb131-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-20"><a href="example.html#cb131-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(WGCNA)</span>
<span id="cb131-21"><a href="example.html#cb131-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-22"><a href="example.html#cb131-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-23"><a href="example.html#cb131-23" aria-hidden="true" tabindex="-1"></a><span class="co"># rm(list = ls())</span></span>
<span id="cb131-24"><a href="example.html#cb131-24" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb131-25"><a href="example.html#cb131-25" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">future.globals.maxSize =</span> <span class="dv">1000</span> <span class="sc">*</span> <span class="dv">1024</span><span class="sc">^</span><span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="importing-data-6" class="section level2 hasAnchor" number="8.2">
<h2><span class="header-section-number">8.2</span> Importing data<a href="example.html#importing-data-6" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>features table</li>
</ul>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="example.html#cb132-1" aria-hidden="true" tabindex="-1"></a>profile <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(<span class="st">&quot;./dataset/GvHD_stool_metabolites_TM.xlsx&quot;</span>, <span class="at">sheet =</span> <span class="dv">1</span>)</span>
<span id="cb132-2"><a href="example.html#cb132-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(profile)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 78
##   Index     Compounds    物质  `Class I` 物质一级分类 `Class II` 物质二级分类 `Q1 (Da)` `Molecular Wei…` `Ionization mo…`
##   &lt;chr&gt;     &lt;chr&gt;        &lt;chr&gt; &lt;chr&gt;     &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt;            &lt;dbl&gt;            &lt;dbl&gt; &lt;chr&gt;           
## 1 MEDL00066 2-Hydroxy-2… -     -         -            -          -                 117.             118. [M-H]-          
## 2 MEDL00356 Phe-Asp      -     -         -            -          -                 245.             348. [M+H-2H2O]+     
## 3 MEDL00369 Glu-Phe      -     -         -            -          -                 293.             294. [M-H]-          
## 4 MEDL00375 Glu-Trp      -     -         -            -          -                 334.             333. [M+H]+          
## 5 MEDL00392 1-Oleoyl-sn… -     -         -            -          -                 525.             521. [M+H]+          
## 6 MEDL00401 Confertifol… -     -         -            -          -                 233.             234. [M-H]-          
## # … with 68 more variables: Formula &lt;chr&gt;, `KEGG ID` &lt;chr&gt;, HMDB &lt;chr&gt;, `Pubchem CID` &lt;chr&gt;, CAS &lt;chr&gt;, ChEBI &lt;chr&gt;,
## #   Metlin &lt;chr&gt;, mix01 &lt;dbl&gt;, mix02 &lt;dbl&gt;, mix03 &lt;dbl&gt;, mix04 &lt;dbl&gt;, mix05 &lt;dbl&gt;, mix06 &lt;dbl&gt;, mix07 &lt;dbl&gt;,
## #   `TZW-V0` &lt;dbl&gt;, `LBC-V0` &lt;dbl&gt;, `HXZ-V0` &lt;dbl&gt;, `LJY-V0` &lt;dbl&gt;, `CJY-V0` &lt;dbl&gt;, `WKM-V0` &lt;dbl&gt;, `WJH-V0` &lt;dbl&gt;,
## #   `ZTX2-V0` &lt;dbl&gt;, `DGL-V0` &lt;dbl&gt;, `ZHP-V0` &lt;dbl&gt;, `MZH-V0` &lt;dbl&gt;, `WL-V0` &lt;dbl&gt;, `LWJ2-V0` &lt;dbl&gt;, `SZP-V0` &lt;dbl&gt;,
## #   `LPQ-V0` &lt;dbl&gt;, `LHR-V0` &lt;dbl&gt;, `XQ-V0` &lt;dbl&gt;, `LJD-V0` &lt;dbl&gt;, `LXH-V0` &lt;dbl&gt;, `YF-V0` &lt;dbl&gt;, `ZJW-V0` &lt;dbl&gt;,
## #   `LHC-V0` &lt;dbl&gt;, `LQB-V0` &lt;dbl&gt;, `LWL-V0` &lt;dbl&gt;, `XD-V0` &lt;dbl&gt;, `ZCY-V0` &lt;dbl&gt;, `ZTX1-V0` &lt;dbl&gt;, `TZW-V1` &lt;dbl&gt;,
## #   `HXZ-V1` &lt;dbl&gt;, `LJY-V1` &lt;dbl&gt;, `CJY-V1` &lt;dbl&gt;, `WKM-V1` &lt;dbl&gt;, `WJH-V1` &lt;dbl&gt;, `ZTX2-V1` &lt;dbl&gt;, `DGL-V1` &lt;dbl&gt;, …</code></pre>
<ul>
<li>metadata table</li>
</ul>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="example.html#cb134-1" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;./dataset/GvHD_metadata.txt&quot;</span>)</span>
<span id="cb134-2"><a href="example.html#cb134-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(metadata)</span></code></pre></div>
<pre><code>##    seq_id           SampleName FMT_status        SampleType Group V1_outcome GVHD_type GVHD_type_phynotype Donor
## 1:   4371                  CJY         V0             stool   kid         CR     cGVHD            diarrhea   DO1
## 2:   4389                  CJY         V1             stool   kid         CR     cGVHD            diarrhea   DO1
## 3:   4426 CJY_WKM_LXH_FMTdonor            bacteria_solution donor                                               
## 4:   4375                  DGL         V0             stool adult         CR     cGVHD               bowel   DO2
## 5:   4393                  DGL         V1             stool adult         CR     cGVHD               bowel   DO2
## 6:   4435         DGL_FMTdonor            bacteria_solution donor                                               
##    baseline_merge baseline_aGVHD_sainai baseline_aGVHD_sainai_detail baseline_aGVHD_glucksberg
## 1:                                                                                            
## 2:                                                                                            
## 3:                                                                                            
## 4:         medium                                                                             
## 5:                                                                                            
## 6:                                                                                            
##    baseline_aGVHD_glucksberg_detail baseline_cGVHD_NIH baseline_cGVHD_NIH_detail Baseline_infection
## 1:                                                                                                 
## 2:                                                                                                 
## 3:                                                                                                 
## 4:                                                中度          肠道2级，皮肤1级                   
## 5:                                                                                                 
## 6:                                                                                                 
##    antibiotics_48h_pre_or_post_FMT GVHDdrug_antiMicrobiomeDrug_2weeksPreV0 GVHDdrug_antiMicrobiomeDrug_V0toV1
## 1:                                                                                                           
## 2:                                                                                                           
## 3:                                                                                                           
## 4:                                            他克莫司、麦考酚酯、伊曲康唑       他克莫司、麦考酚酯、伊曲康唑
## 5:                                            他克莫司、麦考酚酯、伊曲康唑       他克莫司、麦考酚酯、伊曲康唑
## 6:                                                                                                           
##          FMT_method gender age Diagnosis HSCT_DLI_date                                          Pretreatment
## 1: nasojejunal tube   male   3       AML     2020/5/28 Ara-c+CY+IVBu+ATG+Flu+PTCy+PTCy（Ara+Cy+Bu+Flu+PTCy）
## 2: nasojejunal tube   male   3       AML     2020/5/28 Ara-c+CY+IVBu+ATG+Flu+PTCy+PTCy（Ara+Cy+Bu+Flu+PTCy）
## 3:                          NA                                                                              
## 4:      colonoscope   male  36       AML     2019/12/1                                                  BUCY
## 5:      colonoscope   male  36       AML     2019/12/1                                                  BUCY
## 6:                          NA                                                                              
##    Pretreatment_type      GVHD_prevention death relapse
## 1:  marrow cleansing PTCY+PTFLU+MMF+FK506              
## 2:  marrow cleansing PTCY+PTFLU+MMF+FK506              
## 3:                                                     
## 4:  marrow cleansing      CSA+MTX+MMF+ATG              
## 5:  marrow cleansing      CSA+MTX+MMF+ATG              
## 6:</code></pre>
<ul>
<li><p>Data Preparation: SummarizedExperiment object</p></li>
<li><p>Renaming column</p></li>
<li><p>Replace 9 by NA</p></li>
<li><p>column with <code>mix</code> prefix are regards as <strong>QC</strong> samples</p></li>
</ul>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="example.html#cb136-1" aria-hidden="true" tabindex="-1"></a>getSEobject <span class="ot">&lt;-</span> <span class="cf">function</span>(x, y) {</span>
<span id="cb136-2"><a href="example.html#cb136-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb136-3"><a href="example.html#cb136-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># x = metadata</span></span>
<span id="cb136-4"><a href="example.html#cb136-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># y = profile</span></span>
<span id="cb136-5"><a href="example.html#cb136-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-6"><a href="example.html#cb136-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># mix : qc samples</span></span>
<span id="cb136-7"><a href="example.html#cb136-7" aria-hidden="true" tabindex="-1"></a>  qc_samples <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">&quot;mix&quot;</span>, <span class="fu">colnames</span>(profile), <span class="at">value =</span> T)</span>
<span id="cb136-8"><a href="example.html#cb136-8" aria-hidden="true" tabindex="-1"></a>  qc_groups <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">SampleID =</span> qc_samples,</span>
<span id="cb136-9"><a href="example.html#cb136-9" aria-hidden="true" tabindex="-1"></a>                          <span class="at">V1_outcome =</span> <span class="fu">rep</span>(<span class="st">&quot;QC&quot;</span>, <span class="fu">length</span>(qc_samples)),</span>
<span id="cb136-10"><a href="example.html#cb136-10" aria-hidden="true" tabindex="-1"></a>                          <span class="at">SampleName =</span> <span class="fu">rep</span>(<span class="st">&quot;SampleName&quot;</span>, <span class="fu">length</span>(qc_samples)),</span>
<span id="cb136-11"><a href="example.html#cb136-11" aria-hidden="true" tabindex="-1"></a>                          <span class="at">FMT_status =</span> <span class="fu">rep</span>(<span class="st">&quot;FMT_status&quot;</span>, <span class="fu">length</span>(qc_samples)),</span>
<span id="cb136-12"><a href="example.html#cb136-12" aria-hidden="true" tabindex="-1"></a>                          <span class="at">gender =</span> <span class="fu">rep</span>(<span class="st">&quot;gender&quot;</span>, <span class="fu">length</span>(qc_samples)),</span>
<span id="cb136-13"><a href="example.html#cb136-13" aria-hidden="true" tabindex="-1"></a>                          <span class="at">age =</span> <span class="fu">rep</span>(<span class="st">&quot;age&quot;</span>, <span class="fu">length</span>(qc_samples)))</span>
<span id="cb136-14"><a href="example.html#cb136-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># target table</span></span>
<span id="cb136-15"><a href="example.html#cb136-15" aria-hidden="true" tabindex="-1"></a>  target <span class="ot">&lt;-</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb136-16"><a href="example.html#cb136-16" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(FMT_status <span class="sc">!=</span> <span class="st">&quot;&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb136-17"><a href="example.html#cb136-17" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">SampleID =</span> <span class="fu">paste</span>(SampleName, FMT_status, <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb136-18"><a href="example.html#cb136-18" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(SampleID, V1_outcome, SampleName, FMT_status, gender, age) <span class="sc">%&gt;%</span></span>
<span id="cb136-19"><a href="example.html#cb136-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rbind</span>(qc_groups)</span>
<span id="cb136-20"><a href="example.html#cb136-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb136-21"><a href="example.html#cb136-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># profile column </span></span>
<span id="cb136-22"><a href="example.html#cb136-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(y) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>, <span class="st">&quot;_&quot;</span>, <span class="fu">colnames</span>(y))</span>
<span id="cb136-23"><a href="example.html#cb136-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-24"><a href="example.html#cb136-24" aria-hidden="true" tabindex="-1"></a>  sid <span class="ot">&lt;-</span> <span class="fu">intersect</span>(target<span class="sc">$</span>SampleID, <span class="fu">colnames</span>(y))</span>
<span id="cb136-25"><a href="example.html#cb136-25" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb136-26"><a href="example.html#cb136-26" aria-hidden="true" tabindex="-1"></a>  features <span class="ot">&lt;-</span> y <span class="sc">%&gt;%</span> </span>
<span id="cb136-27"><a href="example.html#cb136-27" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(sid)) <span class="sc">%&gt;%</span></span>
<span id="cb136-28"><a href="example.html#cb136-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">t</span>()</span>
<span id="cb136-29"><a href="example.html#cb136-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(features) <span class="ot">&lt;-</span> y<span class="sc">$</span>Index</span>
<span id="cb136-30"><a href="example.html#cb136-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb136-31"><a href="example.html#cb136-31" aria-hidden="true" tabindex="-1"></a>  <span class="co"># replace 9 by NA </span></span>
<span id="cb136-32"><a href="example.html#cb136-32" aria-hidden="true" tabindex="-1"></a>  features[features <span class="sc">==</span> <span class="dv">9</span>] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb136-33"><a href="example.html#cb136-33" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb136-34"><a href="example.html#cb136-34" aria-hidden="true" tabindex="-1"></a>  target <span class="ot">&lt;-</span> target[<span class="fu">pmatch</span>(sid, target<span class="sc">$</span>SampleID), , F]</span>
<span id="cb136-35"><a href="example.html#cb136-35" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb136-36"><a href="example.html#cb136-36" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb136-37"><a href="example.html#cb136-37" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">features =</span> features)</span>
<span id="cb136-38"><a href="example.html#cb136-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(res)  </span>
<span id="cb136-39"><a href="example.html#cb136-39" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb136-40"><a href="example.html#cb136-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-41"><a href="example.html#cb136-41" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">getSEobject</span>(metadata, profile)</span>
<span id="cb136-42"><a href="example.html#cb136-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb136-43"><a href="example.html#cb136-43" aria-hidden="true" tabindex="-1"></a>se</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 1235 61 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(1235): MEDL00066 MEDL00356 ... MW0169477 MW0169569
## rowData names(0):
## colnames(61): CJY_V0 CJY_V1 ... mix06 mix07
## colData names(5): group SampleName FMT_status gender age</code></pre>
</div>
<div id="data-processing-1" class="section level2 hasAnchor" number="8.3">
<h2><span class="header-section-number">8.3</span> Data Processing<a href="example.html#data-processing-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="data-checking-1" class="section level3 hasAnchor" number="8.3.1">
<h3><span class="header-section-number">8.3.1</span> Data Checking<a href="example.html#data-checking-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Features in PomaSummarizedExperiment object must have the following criterion:</p>
<ul>
<li><p>All data values are numeric.</p></li>
<li><p>A total of 0 (0%) missing values were detected.</p></li>
</ul>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="example.html#cb138-1" aria-hidden="true" tabindex="-1"></a>CheckData <span class="ot">&lt;-</span> <span class="cf">function</span>(object) {</span>
<span id="cb138-2"><a href="example.html#cb138-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb138-3"><a href="example.html#cb138-3" aria-hidden="true" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object)</span>
<span id="cb138-4"><a href="example.html#cb138-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb138-5"><a href="example.html#cb138-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># numeric &amp; missing values</span></span>
<span id="cb138-6"><a href="example.html#cb138-6" aria-hidden="true" tabindex="-1"></a>  int_mat <span class="ot">&lt;-</span> features_tab</span>
<span id="cb138-7"><a href="example.html#cb138-7" aria-hidden="true" tabindex="-1"></a>  rowNms <span class="ot">&lt;-</span> <span class="fu">rownames</span>(int_mat)</span>
<span id="cb138-8"><a href="example.html#cb138-8" aria-hidden="true" tabindex="-1"></a>  colNms <span class="ot">&lt;-</span> <span class="fu">colnames</span>(int_mat)</span>
<span id="cb138-9"><a href="example.html#cb138-9" aria-hidden="true" tabindex="-1"></a>  naNms <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">is.na</span>(int_mat))</span>
<span id="cb138-10"><a href="example.html#cb138-10" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(int_mat)) {</span>
<span id="cb138-11"><a href="example.html#cb138-11" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">class</span>(int_mat[, i]) <span class="sc">==</span> <span class="st">&quot;integer64&quot;</span>) {</span>
<span id="cb138-12"><a href="example.html#cb138-12" aria-hidden="true" tabindex="-1"></a>      int_mat[, i] <span class="ot">&lt;-</span> <span class="fu">as.double</span>(int_mat[, i])</span>
<span id="cb138-13"><a href="example.html#cb138-13" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb138-14"><a href="example.html#cb138-14" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb138-15"><a href="example.html#cb138-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb138-16"><a href="example.html#cb138-16" aria-hidden="true" tabindex="-1"></a>  num_mat <span class="ot">&lt;-</span> <span class="fu">apply</span>(int_mat, <span class="dv">2</span>, as.numeric)</span>
<span id="cb138-17"><a href="example.html#cb138-17" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">sum</span>(<span class="fu">is.na</span>(num_mat)) <span class="sc">&gt;</span> naNms) {</span>
<span id="cb138-18"><a href="example.html#cb138-18" aria-hidden="true" tabindex="-1"></a>    num_mat <span class="ot">&lt;-</span> <span class="fu">apply</span>(int_mat, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;,&quot;</span>,  <span class="st">&quot;&quot;</span>, x)))</span>
<span id="cb138-19"><a href="example.html#cb138-19" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">sum</span>(<span class="fu">is.na</span>(num_mat)) <span class="sc">&gt;</span> naNms) {</span>
<span id="cb138-20"><a href="example.html#cb138-20" aria-hidden="true" tabindex="-1"></a>      <span class="fu">message</span>(<span class="st">&quot;&lt;font color=</span><span class="sc">\&quot;</span><span class="st">red</span><span class="sc">\&quot;</span><span class="st">&gt;Non-numeric values were found and replaced by NA.&lt;/font&gt;&quot;</span>)</span>
<span id="cb138-21"><a href="example.html#cb138-21" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb138-22"><a href="example.html#cb138-22" aria-hidden="true" tabindex="-1"></a>      <span class="fu">message</span>(<span class="st">&quot;All data values are numeric.&quot;</span>)</span>
<span id="cb138-23"><a href="example.html#cb138-23" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb138-24"><a href="example.html#cb138-24" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb138-25"><a href="example.html#cb138-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">&quot;All data values are numeric.&quot;</span>)</span>
<span id="cb138-26"><a href="example.html#cb138-26" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb138-27"><a href="example.html#cb138-27" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb138-28"><a href="example.html#cb138-28" aria-hidden="true" tabindex="-1"></a>  int_mat <span class="ot">&lt;-</span> num_mat</span>
<span id="cb138-29"><a href="example.html#cb138-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(int_mat) <span class="ot">&lt;-</span> rowNms</span>
<span id="cb138-30"><a href="example.html#cb138-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(int_mat) <span class="ot">&lt;-</span> colNms</span>
<span id="cb138-31"><a href="example.html#cb138-31" aria-hidden="true" tabindex="-1"></a>  varCol <span class="ot">&lt;-</span> <span class="fu">apply</span>(int_mat, <span class="dv">2</span>, var, <span class="at">na.rm =</span> T)</span>
<span id="cb138-32"><a href="example.html#cb138-32" aria-hidden="true" tabindex="-1"></a>  constCol <span class="ot">&lt;-</span> (varCol <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> <span class="fu">is.na</span>(varCol))</span>
<span id="cb138-33"><a href="example.html#cb138-33" aria-hidden="true" tabindex="-1"></a>  constNum <span class="ot">&lt;-</span> <span class="fu">sum</span>(constCol, <span class="at">na.rm =</span> T)</span>
<span id="cb138-34"><a href="example.html#cb138-34" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (constNum <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb138-35"><a href="example.html#cb138-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="fu">paste</span>(<span class="st">&quot;&lt;font color=</span><span class="sc">\&quot;</span><span class="st">red</span><span class="sc">\&quot;</span><span class="st">&gt;&quot;</span>, constNum, </span>
<span id="cb138-36"><a href="example.html#cb138-36" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;features with a constant or single value across samples were found and deleted.&lt;/font&gt;&quot;</span>))</span>
<span id="cb138-37"><a href="example.html#cb138-37" aria-hidden="true" tabindex="-1"></a>    int_mat <span class="ot">&lt;-</span> int_mat[, <span class="sc">!</span>constCol, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb138-38"><a href="example.html#cb138-38" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb138-39"><a href="example.html#cb138-39" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb138-40"><a href="example.html#cb138-40" aria-hidden="true" tabindex="-1"></a>  totalCount <span class="ot">&lt;-</span> <span class="fu">nrow</span>(int_mat) <span class="sc">*</span> <span class="fu">ncol</span>(int_mat)</span>
<span id="cb138-41"><a href="example.html#cb138-41" aria-hidden="true" tabindex="-1"></a>  naCount <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">is.na</span>(int_mat))</span>
<span id="cb138-42"><a href="example.html#cb138-42" aria-hidden="true" tabindex="-1"></a>  naPercent <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> naCount<span class="sc">/</span>totalCount, <span class="dv">1</span>)</span>
<span id="cb138-43"><a href="example.html#cb138-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-44"><a href="example.html#cb138-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="fu">paste</span>(<span class="st">&quot;A total of &quot;</span>, naCount, <span class="st">&quot; (&quot;</span>, naPercent, </span>
<span id="cb138-45"><a href="example.html#cb138-45" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;%) missing values were detected.&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))  </span>
<span id="cb138-46"><a href="example.html#cb138-46" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb138-47"><a href="example.html#cb138-47" aria-hidden="true" tabindex="-1"></a>  <span class="co"># save int_mat into se object </span></span>
<span id="cb138-48"><a href="example.html#cb138-48" aria-hidden="true" tabindex="-1"></a>  target <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object) <span class="sc">%&gt;%</span></span>
<span id="cb138-49"><a href="example.html#cb138-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb138-50"><a href="example.html#cb138-50" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb138-51"><a href="example.html#cb138-51" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb138-52"><a href="example.html#cb138-52" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">features =</span> <span class="fu">t</span>(int_mat))</span>
<span id="cb138-53"><a href="example.html#cb138-53" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb138-54"><a href="example.html#cb138-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb138-55"><a href="example.html#cb138-55" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb138-56"><a href="example.html#cb138-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-57"><a href="example.html#cb138-57" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">CheckData</span>(<span class="at">object =</span> se)</span>
<span id="cb138-58"><a href="example.html#cb138-58" aria-hidden="true" tabindex="-1"></a>se</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 1235 61 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(1235): MEDL00066 MEDL00356 ... MW0169477 MW0169569
## rowData names(0):
## colnames(61): CJY_V0 CJY_V1 ... mix06 mix07
## colData names(5): group SampleName FMT_status gender age</code></pre>
</div>
<div id="data-filtering-1" class="section level3 hasAnchor" number="8.3.2">
<h3><span class="header-section-number">8.3.2</span> Data Filtering<a href="example.html#data-filtering-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="example.html#cb140-1" aria-hidden="true" tabindex="-1"></a>FilterFeature <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb140-2"><a href="example.html#cb140-2" aria-hidden="true" tabindex="-1"></a>                       object,</span>
<span id="cb140-3"><a href="example.html#cb140-3" aria-hidden="true" tabindex="-1"></a>                       qc_label,</span>
<span id="cb140-4"><a href="example.html#cb140-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="fu">c</span>(<span class="st">&quot;none&quot;</span>, <span class="st">&quot;iqr&quot;</span>, <span class="st">&quot;rsd&quot;</span>, </span>
<span id="cb140-5"><a href="example.html#cb140-5" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;nrsd&quot;</span>, <span class="st">&quot;mean&quot;</span>, <span class="st">&quot;sd&quot;</span>,</span>
<span id="cb140-6"><a href="example.html#cb140-6" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;mad&quot;</span>, <span class="st">&quot;median&quot;</span>),</span>
<span id="cb140-7"><a href="example.html#cb140-7" aria-hidden="true" tabindex="-1"></a>                       <span class="at">rsd_cutoff =</span> <span class="dv">25</span>) {</span>
<span id="cb140-8"><a href="example.html#cb140-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb140-9"><a href="example.html#cb140-9" aria-hidden="true" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) </span>
<span id="cb140-10"><a href="example.html#cb140-10" aria-hidden="true" tabindex="-1"></a>  metadata_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object) </span>
<span id="cb140-11"><a href="example.html#cb140-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb140-12"><a href="example.html#cb140-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># QC samples</span></span>
<span id="cb140-13"><a href="example.html#cb140-13" aria-hidden="true" tabindex="-1"></a>  qc_samples <span class="ot">&lt;-</span> metadata_tab <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb140-14"><a href="example.html#cb140-14" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(group <span class="sc">==</span> qc_label)</span>
<span id="cb140-15"><a href="example.html#cb140-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">dim</span>(qc_samples)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb140-16"><a href="example.html#cb140-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;No qc samples have been chosen, please check your input&quot;</span>)</span>
<span id="cb140-17"><a href="example.html#cb140-17" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb140-18"><a href="example.html#cb140-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb140-19"><a href="example.html#cb140-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># QC samples&#39; feature table</span></span>
<span id="cb140-20"><a href="example.html#cb140-20" aria-hidden="true" tabindex="-1"></a>  qc_feature <span class="ot">&lt;-</span> features_tab[, <span class="fu">colnames</span>(features_tab) <span class="sc">%in%</span> <span class="fu">rownames</span>(qc_samples)] <span class="sc">%&gt;%</span></span>
<span id="cb140-21"><a href="example.html#cb140-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">t</span>()</span>
<span id="cb140-22"><a href="example.html#cb140-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb140-23"><a href="example.html#cb140-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># filter features by QC RSD</span></span>
<span id="cb140-24"><a href="example.html#cb140-24" aria-hidden="true" tabindex="-1"></a>  rsd <span class="ot">&lt;-</span> rsd_cutoff <span class="sc">/</span> <span class="dv">100</span></span>
<span id="cb140-25"><a href="example.html#cb140-25" aria-hidden="true" tabindex="-1"></a>  sds <span class="ot">&lt;-</span> <span class="fu">apply</span>(qc_feature, <span class="dv">2</span>, sd, <span class="at">na.rm =</span> T)</span>
<span id="cb140-26"><a href="example.html#cb140-26" aria-hidden="true" tabindex="-1"></a>  mns <span class="ot">&lt;-</span> <span class="fu">apply</span>(qc_feature, <span class="dv">2</span>, mean, <span class="at">na.rm =</span> T)</span>
<span id="cb140-27"><a href="example.html#cb140-27" aria-hidden="true" tabindex="-1"></a>  rsd_vals <span class="ot">&lt;-</span> <span class="fu">abs</span>(sds<span class="sc">/</span>mns) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb140-28"><a href="example.html#cb140-28" aria-hidden="true" tabindex="-1"></a>  gd_inx <span class="ot">&lt;-</span> rsd_vals <span class="sc">&lt;</span> rsd</span>
<span id="cb140-29"><a href="example.html#cb140-29" aria-hidden="true" tabindex="-1"></a>  int_mat <span class="ot">&lt;-</span> features_tab[gd_inx, ]</span>
<span id="cb140-30"><a href="example.html#cb140-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">&quot;Removed &quot;</span>, (<span class="fu">dim</span>(qc_feature)[<span class="dv">2</span>] <span class="sc">-</span> <span class="fu">dim</span>(int_mat)[<span class="dv">1</span>]), </span>
<span id="cb140-31"><a href="example.html#cb140-31" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot; features based on QC RSD values. QC samples are excluded from downstream functional analysis.&quot;</span>)</span>
<span id="cb140-32"><a href="example.html#cb140-32" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb140-33"><a href="example.html#cb140-33" aria-hidden="true" tabindex="-1"></a>  <span class="co"># whether to filter features by percentage according to the number</span></span>
<span id="cb140-34"><a href="example.html#cb140-34" aria-hidden="true" tabindex="-1"></a>  PerformFeatureFilter <span class="ot">&lt;-</span> <span class="cf">function</span>(datMatrix, </span>
<span id="cb140-35"><a href="example.html#cb140-35" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">qc_method =</span> method,</span>
<span id="cb140-36"><a href="example.html#cb140-36" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">remain_num =</span> <span class="cn">NULL</span>) {</span>
<span id="cb140-37"><a href="example.html#cb140-37" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb140-38"><a href="example.html#cb140-38" aria-hidden="true" tabindex="-1"></a>    dat <span class="ot">&lt;-</span> datMatrix</span>
<span id="cb140-39"><a href="example.html#cb140-39" aria-hidden="true" tabindex="-1"></a>    feat_num <span class="ot">&lt;-</span> <span class="fu">ncol</span>(dat)</span>
<span id="cb140-40"><a href="example.html#cb140-40" aria-hidden="true" tabindex="-1"></a>    feat_nms <span class="ot">&lt;-</span> <span class="fu">colnames</span>(dat)</span>
<span id="cb140-41"><a href="example.html#cb140-41" aria-hidden="true" tabindex="-1"></a>    nm <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb140-42"><a href="example.html#cb140-42" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;none&quot;</span> <span class="sc">&amp;&amp;</span> feat_num <span class="sc">&lt;</span> <span class="dv">5000</span>) { <span class="co"># only allow for less than 4000</span></span>
<span id="cb140-43"><a href="example.html#cb140-43" aria-hidden="true" tabindex="-1"></a>      remain <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">TRUE</span>, feat_num)</span>
<span id="cb140-44"><a href="example.html#cb140-44" aria-hidden="true" tabindex="-1"></a>      nm <span class="ot">&lt;-</span> <span class="st">&quot;No filtering was applied&quot;</span></span>
<span id="cb140-45"><a href="example.html#cb140-45" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb140-46"><a href="example.html#cb140-46" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;rsd&quot;</span>){</span>
<span id="cb140-47"><a href="example.html#cb140-47" aria-hidden="true" tabindex="-1"></a>        sds <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, sd, <span class="at">na.rm =</span> T)</span>
<span id="cb140-48"><a href="example.html#cb140-48" aria-hidden="true" tabindex="-1"></a>        mns <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, mean, <span class="at">na.rm =</span> T)</span>
<span id="cb140-49"><a href="example.html#cb140-49" aria-hidden="true" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">abs</span>(sds<span class="sc">/</span>mns)</span>
<span id="cb140-50"><a href="example.html#cb140-50" aria-hidden="true" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;Relative standard deviation&quot;</span></span>
<span id="cb140-51"><a href="example.html#cb140-51" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;nrsd&quot;</span> ) {</span>
<span id="cb140-52"><a href="example.html#cb140-52" aria-hidden="true" tabindex="-1"></a>        mads <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, mad, <span class="at">na.rm =</span> T)</span>
<span id="cb140-53"><a href="example.html#cb140-53" aria-hidden="true" tabindex="-1"></a>        meds <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, median, <span class="at">na.rm =</span> T)</span>
<span id="cb140-54"><a href="example.html#cb140-54" aria-hidden="true" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">abs</span>(mads<span class="sc">/</span>meds)</span>
<span id="cb140-55"><a href="example.html#cb140-55" aria-hidden="true" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;Non-paramatric relative standard deviation&quot;</span></span>
<span id="cb140-56"><a href="example.html#cb140-56" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;mean&quot;</span>) {</span>
<span id="cb140-57"><a href="example.html#cb140-57" aria-hidden="true" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, mean, <span class="at">na.rm =</span> T)</span>
<span id="cb140-58"><a href="example.html#cb140-58" aria-hidden="true" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;mean&quot;</span></span>
<span id="cb140-59"><a href="example.html#cb140-59" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;sd&quot;</span>) {</span>
<span id="cb140-60"><a href="example.html#cb140-60" aria-hidden="true" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, sd, <span class="at">na.rm =</span> T)</span>
<span id="cb140-61"><a href="example.html#cb140-61" aria-hidden="true" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;standard deviation&quot;</span></span>
<span id="cb140-62"><a href="example.html#cb140-62" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;mad&quot;</span>) {</span>
<span id="cb140-63"><a href="example.html#cb140-63" aria-hidden="true" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, mad, <span class="at">na.rm =</span> T)</span>
<span id="cb140-64"><a href="example.html#cb140-64" aria-hidden="true" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;Median absolute deviation&quot;</span></span>
<span id="cb140-65"><a href="example.html#cb140-65" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;median&quot;</span>) {</span>
<span id="cb140-66"><a href="example.html#cb140-66" aria-hidden="true" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, median, <span class="at">na.rm =</span> T)</span>
<span id="cb140-67"><a href="example.html#cb140-67" aria-hidden="true" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;median&quot;</span></span>
<span id="cb140-68"><a href="example.html#cb140-68" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;iqr&quot;</span>) { <span class="co"># iqr</span></span>
<span id="cb140-69"><a href="example.html#cb140-69" aria-hidden="true" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, IQR, <span class="at">na.rm =</span> T)</span>
<span id="cb140-70"><a href="example.html#cb140-70" aria-hidden="true" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;Interquantile Range&quot;</span></span>
<span id="cb140-71"><a href="example.html#cb140-71" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb140-72"><a href="example.html#cb140-72" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb140-73"><a href="example.html#cb140-73" aria-hidden="true" tabindex="-1"></a>      <span class="co"># get the rank of the filtered variables</span></span>
<span id="cb140-74"><a href="example.html#cb140-74" aria-hidden="true" tabindex="-1"></a>      rk <span class="ot">&lt;-</span> <span class="fu">rank</span>(<span class="sc">-</span>filter_val, <span class="at">ties.method =</span> <span class="st">&quot;random&quot;</span>)</span>
<span id="cb140-75"><a href="example.html#cb140-75" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb140-76"><a href="example.html#cb140-76" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">is.null</span>(remain_num)) { <span class="co"># apply empirical filtering based on data size</span></span>
<span id="cb140-77"><a href="example.html#cb140-77" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span> (feat_num <span class="sc">&lt;</span> <span class="dv">250</span>) { <span class="co"># reduce 5%</span></span>
<span id="cb140-78"><a href="example.html#cb140-78" aria-hidden="true" tabindex="-1"></a>            remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> feat_num <span class="sc">*</span> <span class="fl">0.95</span></span>
<span id="cb140-79"><a href="example.html#cb140-79" aria-hidden="true" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Further feature filtering based on &quot;</span>, nm)</span>
<span id="cb140-80"><a href="example.html#cb140-80" aria-hidden="true" tabindex="-1"></a>          } <span class="cf">else</span> <span class="cf">if</span> (feat_num <span class="sc">&lt;</span> <span class="dv">500</span>) { <span class="co"># reduce 10%</span></span>
<span id="cb140-81"><a href="example.html#cb140-81" aria-hidden="true" tabindex="-1"></a>            remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> feat_num <span class="sc">*</span> <span class="fl">0.9</span></span>
<span id="cb140-82"><a href="example.html#cb140-82" aria-hidden="true" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Further feature filtering based on &quot;</span>, nm)</span>
<span id="cb140-83"><a href="example.html#cb140-83" aria-hidden="true" tabindex="-1"></a>          } <span class="cf">else</span> <span class="cf">if</span> (feat_num <span class="sc">&lt;</span> <span class="dv">1000</span>) { <span class="co"># reduce 25%</span></span>
<span id="cb140-84"><a href="example.html#cb140-84" aria-hidden="true" tabindex="-1"></a>            remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> feat_num <span class="sc">*</span> <span class="fl">0.75</span></span>
<span id="cb140-85"><a href="example.html#cb140-85" aria-hidden="true" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Further feature filtering based on &quot;</span>, nm)</span>
<span id="cb140-86"><a href="example.html#cb140-86" aria-hidden="true" tabindex="-1"></a>          } <span class="cf">else</span> { <span class="co"># reduce 40%, if still over 5000, then only use top 5000</span></span>
<span id="cb140-87"><a href="example.html#cb140-87" aria-hidden="true" tabindex="-1"></a>            remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> feat_num <span class="sc">*</span> <span class="fl">0.6</span></span>
<span id="cb140-88"><a href="example.html#cb140-88" aria-hidden="true" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Further feature filtering based on &quot;</span>, nm)</span>
<span id="cb140-89"><a href="example.html#cb140-89" aria-hidden="true" tabindex="-1"></a>          }</span>
<span id="cb140-90"><a href="example.html#cb140-90" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb140-91"><a href="example.html#cb140-91" aria-hidden="true" tabindex="-1"></a>        remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> remain_num</span>
<span id="cb140-92"><a href="example.html#cb140-92" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb140-93"><a href="example.html#cb140-93" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb140-94"><a href="example.html#cb140-94" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb140-95"><a href="example.html#cb140-95" aria-hidden="true" tabindex="-1"></a>    res <span class="ot">&lt;-</span> datMatrix[, remain]</span>
<span id="cb140-96"><a href="example.html#cb140-96" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb140-97"><a href="example.html#cb140-97" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(res)</span>
<span id="cb140-98"><a href="example.html#cb140-98" aria-hidden="true" tabindex="-1"></a>  }  </span>
<span id="cb140-99"><a href="example.html#cb140-99" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb140-100"><a href="example.html#cb140-100" aria-hidden="true" tabindex="-1"></a>  feature_res <span class="ot">&lt;-</span> <span class="fu">PerformFeatureFilter</span>(<span class="fu">t</span>(int_mat))</span>
<span id="cb140-101"><a href="example.html#cb140-101" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb140-102"><a href="example.html#cb140-102" aria-hidden="true" tabindex="-1"></a>  <span class="co"># remove QC samples </span></span>
<span id="cb140-103"><a href="example.html#cb140-103" aria-hidden="true" tabindex="-1"></a>  feature_final <span class="ot">&lt;-</span> feature_res[<span class="sc">!</span><span class="fu">rownames</span>(feature_res) <span class="sc">%in%</span> <span class="fu">rownames</span>(qc_samples), ]</span>
<span id="cb140-104"><a href="example.html#cb140-104" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb140-105"><a href="example.html#cb140-105" aria-hidden="true" tabindex="-1"></a>  <span class="co"># save int_mat into se object </span></span>
<span id="cb140-106"><a href="example.html#cb140-106" aria-hidden="true" tabindex="-1"></a>  target <span class="ot">&lt;-</span> metadata_tab <span class="sc">%&gt;%</span> </span>
<span id="cb140-107"><a href="example.html#cb140-107" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb140-108"><a href="example.html#cb140-108" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb140-109"><a href="example.html#cb140-109" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(SampleID <span class="sc">%in%</span> <span class="fu">rownames</span>(feature_final))</span>
<span id="cb140-110"><a href="example.html#cb140-110" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb140-111"><a href="example.html#cb140-111" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">features =</span> feature_final)</span>
<span id="cb140-112"><a href="example.html#cb140-112" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb140-113"><a href="example.html#cb140-113" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(res) </span>
<span id="cb140-114"><a href="example.html#cb140-114" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb140-115"><a href="example.html#cb140-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb140-116"><a href="example.html#cb140-116" aria-hidden="true" tabindex="-1"></a>se_filter <span class="ot">&lt;-</span> <span class="fu">FilterFeature</span>(<span class="at">object =</span> se,</span>
<span id="cb140-117"><a href="example.html#cb140-117" aria-hidden="true" tabindex="-1"></a>                           <span class="at">qc_label =</span> <span class="st">&quot;QC&quot;</span>,</span>
<span id="cb140-118"><a href="example.html#cb140-118" aria-hidden="true" tabindex="-1"></a>                           <span class="at">method =</span> <span class="st">&quot;iqr&quot;</span>)</span>
<span id="cb140-119"><a href="example.html#cb140-119" aria-hidden="true" tabindex="-1"></a>se_filter</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 715 54 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(715): MEDL00066 MEDL00375 ... MW0168764 MW0169569
## rowData names(0):
## colnames(54): CJY_V0 CJY_V1 ... ZTX2_V0 ZTX2_V1
## colData names(5): group SampleName FMT_status gender age</code></pre>
</div>
<div id="missing-value-imputation-1" class="section level3 hasAnchor" number="8.3.3">
<h3><span class="header-section-number">8.3.3</span> Missing Value Imputation<a href="example.html#missing-value-imputation-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="example.html#cb142-1" aria-hidden="true" tabindex="-1"></a>se_impute <span class="ot">&lt;-</span> <span class="fu">PomaImpute</span>(</span>
<span id="cb142-2"><a href="example.html#cb142-2" aria-hidden="true" tabindex="-1"></a>                    se_filter, </span>
<span id="cb142-3"><a href="example.html#cb142-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ZerosAsNA =</span> <span class="cn">TRUE</span>, </span>
<span id="cb142-4"><a href="example.html#cb142-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">RemoveNA =</span> <span class="cn">TRUE</span>, </span>
<span id="cb142-5"><a href="example.html#cb142-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">cutoff =</span> <span class="dv">20</span>, </span>
<span id="cb142-6"><a href="example.html#cb142-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;knn&quot;</span>)</span>
<span id="cb142-7"><a href="example.html#cb142-7" aria-hidden="true" tabindex="-1"></a>se_impute</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 540 54 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(540): MEDL00066 MEDL00375 ... MW0168675 MW0169569
## rowData names(0):
## colnames(54): CJY_V0 CJY_V1 ... ZTX2_V0 ZTX2_V1
## colData names(5): group SampleName FMT_status gender age</code></pre>
</div>
<div id="data-normalization-1" class="section level3 hasAnchor" number="8.3.4">
<h3><span class="header-section-number">8.3.4</span> Data Normalization<a href="example.html#data-normalization-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="normalization-by-normalizedata-function-1" class="section level4 hasAnchor" number="8.3.4.1">
<h4><span class="header-section-number">8.3.4.1</span> Normalization by <strong>NormalizeData</strong> function<a href="example.html#normalization-by-normalizedata-function-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="example.html#cb144-1" aria-hidden="true" tabindex="-1"></a>NormalizeData <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb144-2"><a href="example.html#cb144-2" aria-hidden="true" tabindex="-1"></a>                      object,</span>
<span id="cb144-3"><a href="example.html#cb144-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">rowNorm =</span> <span class="fu">c</span>(<span class="st">&quot;Quantile&quot;</span>, <span class="st">&quot;GroupPQN&quot;</span>, <span class="st">&quot;SamplePQN&quot;</span>, </span>
<span id="cb144-4"><a href="example.html#cb144-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;CompNorm&quot;</span>, <span class="st">&quot;SumNorm&quot;</span>, <span class="st">&quot;MedianNorm&quot;</span>,</span>
<span id="cb144-5"><a href="example.html#cb144-5" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;SpecNorm&quot;</span>, <span class="st">&quot;None&quot;</span>),</span>
<span id="cb144-6"><a href="example.html#cb144-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">transNorm =</span> <span class="fu">c</span>(<span class="st">&quot;LogNorm&quot;</span>, <span class="st">&quot;SrNorm&quot;</span>, <span class="st">&quot;CrNorm&quot;</span>, <span class="st">&quot;None&quot;</span>),</span>
<span id="cb144-7"><a href="example.html#cb144-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">scaleNorm =</span> <span class="fu">c</span>(<span class="st">&quot;MeanCenter&quot;</span>, <span class="st">&quot;AutoNorm&quot;</span>, <span class="st">&quot;ParetoNorm&quot;</span>, </span>
<span id="cb144-8"><a href="example.html#cb144-8" aria-hidden="true" tabindex="-1"></a>                                    <span class="st">&quot;RangeNorm&quot;</span>, <span class="st">&quot;None&quot;</span>),</span>
<span id="cb144-9"><a href="example.html#cb144-9" aria-hidden="true" tabindex="-1"></a>                      <span class="at">ref =</span> <span class="cn">NULL</span>,</span>
<span id="cb144-10"><a href="example.html#cb144-10" aria-hidden="true" tabindex="-1"></a>                      <span class="at">SpeWeight =</span> <span class="dv">1</span>) {</span>
<span id="cb144-11"><a href="example.html#cb144-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-12"><a href="example.html#cb144-12" aria-hidden="true" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) </span>
<span id="cb144-13"><a href="example.html#cb144-13" aria-hidden="true" tabindex="-1"></a>  metadata_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object)   </span>
<span id="cb144-14"><a href="example.html#cb144-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-15"><a href="example.html#cb144-15" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">t</span>(features_tab)</span>
<span id="cb144-16"><a href="example.html#cb144-16" aria-hidden="true" tabindex="-1"></a>  colNames <span class="ot">&lt;-</span> <span class="fu">colnames</span>(data)</span>
<span id="cb144-17"><a href="example.html#cb144-17" aria-hidden="true" tabindex="-1"></a>  rowNames <span class="ot">&lt;-</span> <span class="fu">rownames</span>(data)</span>
<span id="cb144-18"><a href="example.html#cb144-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-19"><a href="example.html#cb144-19" aria-hidden="true" tabindex="-1"></a>  <span class="do">#############################################</span></span>
<span id="cb144-20"><a href="example.html#cb144-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Sample normalization</span></span>
<span id="cb144-21"><a href="example.html#cb144-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># perform quantile normalization on the raw data (can be log transformed later by user)</span></span>
<span id="cb144-22"><a href="example.html#cb144-22" aria-hidden="true" tabindex="-1"></a>  QuantileNormalize <span class="ot">&lt;-</span> <span class="cf">function</span>(data) {</span>
<span id="cb144-23"><a href="example.html#cb144-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">t</span>(preprocessCore<span class="sc">::</span><span class="fu">normalize.quantiles</span>(<span class="fu">t</span>(data), <span class="at">copy=</span><span class="cn">FALSE</span>)));</span>
<span id="cb144-24"><a href="example.html#cb144-24" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb144-25"><a href="example.html#cb144-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># normalize by a reference sample (probability quotient normalization)</span></span>
<span id="cb144-26"><a href="example.html#cb144-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ref should be the name of the reference sample</span></span>
<span id="cb144-27"><a href="example.html#cb144-27" aria-hidden="true" tabindex="-1"></a>  ProbNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x, ref_smpl) {</span>
<span id="cb144-28"><a href="example.html#cb144-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(x<span class="sc">/</span><span class="fu">median</span>(<span class="fu">as.numeric</span>(x<span class="sc">/</span>ref_smpl), <span class="at">na.rm =</span> T))</span>
<span id="cb144-29"><a href="example.html#cb144-29" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb144-30"><a href="example.html#cb144-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-31"><a href="example.html#cb144-31" aria-hidden="true" tabindex="-1"></a>  <span class="co"># normalize by a reference reference (i.e. creatinine)</span></span>
<span id="cb144-32"><a href="example.html#cb144-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ref should be the name of the cmpd</span></span>
<span id="cb144-33"><a href="example.html#cb144-33" aria-hidden="true" tabindex="-1"></a>  CompNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x, ref) {</span>
<span id="cb144-34"><a href="example.html#cb144-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="dv">1000</span><span class="sc">*</span>x<span class="sc">/</span>x[ref])</span>
<span id="cb144-35"><a href="example.html#cb144-35" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb144-36"><a href="example.html#cb144-36" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-37"><a href="example.html#cb144-37" aria-hidden="true" tabindex="-1"></a>  SumNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb144-38"><a href="example.html#cb144-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="dv">1000</span><span class="sc">*</span>x<span class="sc">/</span><span class="fu">sum</span>(x, <span class="at">na.rm =</span> T))</span>
<span id="cb144-39"><a href="example.html#cb144-39" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb144-40"><a href="example.html#cb144-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-41"><a href="example.html#cb144-41" aria-hidden="true" tabindex="-1"></a>  <span class="co"># normalize by median</span></span>
<span id="cb144-42"><a href="example.html#cb144-42" aria-hidden="true" tabindex="-1"></a>  MedianNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb144-43"><a href="example.html#cb144-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(x<span class="sc">/</span><span class="fu">median</span>(x, <span class="at">na.rm =</span> T))</span>
<span id="cb144-44"><a href="example.html#cb144-44" aria-hidden="true" tabindex="-1"></a>  }  </span>
<span id="cb144-45"><a href="example.html#cb144-45" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-46"><a href="example.html#cb144-46" aria-hidden="true" tabindex="-1"></a>  <span class="co"># row-wise normalization</span></span>
<span id="cb144-47"><a href="example.html#cb144-47" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;Quantile&quot;</span>) {</span>
<span id="cb144-48"><a href="example.html#cb144-48" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">QuantileNormalize</span>(data)</span>
<span id="cb144-49"><a href="example.html#cb144-49" aria-hidden="true" tabindex="-1"></a>    <span class="co"># this can introduce constant variables if a variable is </span></span>
<span id="cb144-50"><a href="example.html#cb144-50" aria-hidden="true" tabindex="-1"></a>    <span class="co"># at the same rank across all samples (replaced by its average across all)</span></span>
<span id="cb144-51"><a href="example.html#cb144-51" aria-hidden="true" tabindex="-1"></a>    varCol <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, var, <span class="at">na.rm =</span> T)</span>
<span id="cb144-52"><a href="example.html#cb144-52" aria-hidden="true" tabindex="-1"></a>    constCol <span class="ot">&lt;-</span> (varCol <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> <span class="fu">is.na</span>(varCol))</span>
<span id="cb144-53"><a href="example.html#cb144-53" aria-hidden="true" tabindex="-1"></a>    constNum <span class="ot">&lt;-</span> <span class="fu">sum</span>(constCol, <span class="at">na.rm =</span> T)</span>
<span id="cb144-54"><a href="example.html#cb144-54" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (constNum <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb144-55"><a href="example.html#cb144-55" aria-hidden="true" tabindex="-1"></a>      <span class="fu">message</span>(<span class="fu">paste</span>(<span class="st">&quot;After quantile normalization&quot;</span>, constNum, </span>
<span id="cb144-56"><a href="example.html#cb144-56" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;features with a constant value were found and deleted.&quot;</span>))</span>
<span id="cb144-57"><a href="example.html#cb144-57" aria-hidden="true" tabindex="-1"></a>      data <span class="ot">&lt;-</span> data[, <span class="sc">!</span>constCol, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb144-58"><a href="example.html#cb144-58" aria-hidden="true" tabindex="-1"></a>      colNames <span class="ot">&lt;-</span> <span class="fu">colnames</span>(data)</span>
<span id="cb144-59"><a href="example.html#cb144-59" aria-hidden="true" tabindex="-1"></a>      rowNames <span class="ot">&lt;-</span> <span class="fu">rownames</span>(data)</span>
<span id="cb144-60"><a href="example.html#cb144-60" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb144-61"><a href="example.html#cb144-61" aria-hidden="true" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Quantile Normalization&quot;</span></span>
<span id="cb144-62"><a href="example.html#cb144-62" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;GroupPQN&quot;</span>) {</span>
<span id="cb144-63"><a href="example.html#cb144-63" aria-hidden="true" tabindex="-1"></a>    grp_inx <span class="ot">&lt;-</span> metadata_tab<span class="sc">$</span>group <span class="sc">==</span> ref</span>
<span id="cb144-64"><a href="example.html#cb144-64" aria-hidden="true" tabindex="-1"></a>    ref.smpl <span class="ot">&lt;-</span> <span class="fu">apply</span>(data[grp_inx, , <span class="at">drop =</span> <span class="cn">FALSE</span>], <span class="dv">2</span>, mean)</span>
<span id="cb144-65"><a href="example.html#cb144-65" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(data, <span class="dv">1</span>, ProbNorm, ref.smpl))</span>
<span id="cb144-66"><a href="example.html#cb144-66" aria-hidden="true" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Probabilistic Quotient Normalization by a reference group&quot;</span></span>
<span id="cb144-67"><a href="example.html#cb144-67" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;SamplePQN&quot;</span>) {</span>
<span id="cb144-68"><a href="example.html#cb144-68" aria-hidden="true" tabindex="-1"></a>    ref.smpl <span class="ot">&lt;-</span> data[ref, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb144-69"><a href="example.html#cb144-69" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(data, <span class="dv">1</span>, ProbNorm, ref.smpl))</span>
<span id="cb144-70"><a href="example.html#cb144-70" aria-hidden="true" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Probabilistic Quotient Normalization by a reference sample&quot;</span></span>
<span id="cb144-71"><a href="example.html#cb144-71" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;CompNorm&quot;</span>) {</span>
<span id="cb144-72"><a href="example.html#cb144-72" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(<span class="fu">t</span>(data), <span class="dv">1</span>, CompNorm, ref))</span>
<span id="cb144-73"><a href="example.html#cb144-73" aria-hidden="true" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Normalization by a reference feature&quot;</span>;</span>
<span id="cb144-74"><a href="example.html#cb144-74" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;SumNorm&quot;</span>) {</span>
<span id="cb144-75"><a href="example.html#cb144-75" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(data, <span class="dv">1</span>, SumNorm))</span>
<span id="cb144-76"><a href="example.html#cb144-76" aria-hidden="true" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Normalization to constant sum&quot;</span></span>
<span id="cb144-77"><a href="example.html#cb144-77" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;MedianNorm&quot;</span>) {</span>
<span id="cb144-78"><a href="example.html#cb144-78" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(data, <span class="dv">1</span>, MedianNorm))</span>
<span id="cb144-79"><a href="example.html#cb144-79" aria-hidden="true" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Normalization to sample median&quot;</span></span>
<span id="cb144-80"><a href="example.html#cb144-80" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span>(rowNorm <span class="sc">==</span> <span class="st">&quot;SpecNorm&quot;</span>) {</span>
<span id="cb144-81"><a href="example.html#cb144-81" aria-hidden="true" tabindex="-1"></a>    norm.vec <span class="ot">&lt;-</span> <span class="fu">rep</span>(SpeWeight, <span class="fu">nrow</span>(data)) <span class="co"># default all same weight vec to prevent error</span></span>
<span id="cb144-82"><a href="example.html#cb144-82" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> data <span class="sc">/</span> norm.vec</span>
<span id="cb144-83"><a href="example.html#cb144-83" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">&quot;No sample specific information were given, all set to 1.0&quot;</span>)</span>
<span id="cb144-84"><a href="example.html#cb144-84" aria-hidden="true" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Normalization by sample-specific factor&quot;</span></span>
<span id="cb144-85"><a href="example.html#cb144-85" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb144-86"><a href="example.html#cb144-86" aria-hidden="true" tabindex="-1"></a>    <span class="co"># nothing to do</span></span>
<span id="cb144-87"><a href="example.html#cb144-87" aria-hidden="true" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;N/A&quot;</span></span>
<span id="cb144-88"><a href="example.html#cb144-88" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb144-89"><a href="example.html#cb144-89" aria-hidden="true" tabindex="-1"></a>  <span class="do">################################################ </span></span>
<span id="cb144-90"><a href="example.html#cb144-90" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-91"><a href="example.html#cb144-91" aria-hidden="true" tabindex="-1"></a>  <span class="co"># use apply will lose dimension info (i.e. row names and colnames)</span></span>
<span id="cb144-92"><a href="example.html#cb144-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(data) <span class="ot">&lt;-</span> rowNames</span>
<span id="cb144-93"><a href="example.html#cb144-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(data) <span class="ot">&lt;-</span> colNames</span>
<span id="cb144-94"><a href="example.html#cb144-94" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-95"><a href="example.html#cb144-95" aria-hidden="true" tabindex="-1"></a>  <span class="co"># if the reference by feature, the feature column should be removed, since it is all 1</span></span>
<span id="cb144-96"><a href="example.html#cb144-96" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(rowNorm <span class="sc">==</span> <span class="st">&quot;CompNorm&quot;</span> <span class="sc">&amp;&amp;</span> <span class="sc">!</span><span class="fu">is.null</span>(ref)){</span>
<span id="cb144-97"><a href="example.html#cb144-97" aria-hidden="true" tabindex="-1"></a>    inx <span class="ot">&lt;-</span> <span class="fu">match</span>(ref, <span class="fu">colnames</span>(data))</span>
<span id="cb144-98"><a href="example.html#cb144-98" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> data[, <span class="sc">-</span>inx, drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb144-99"><a href="example.html#cb144-99" aria-hidden="true" tabindex="-1"></a>    colNames <span class="ot">&lt;-</span> colNames[<span class="sc">-</span>inx]</span>
<span id="cb144-100"><a href="example.html#cb144-100" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb144-101"><a href="example.html#cb144-101" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-102"><a href="example.html#cb144-102" aria-hidden="true" tabindex="-1"></a>  <span class="do">#############################################</span></span>
<span id="cb144-103"><a href="example.html#cb144-103" aria-hidden="true" tabindex="-1"></a>  <span class="co">#  Data transformation</span></span>
<span id="cb144-104"><a href="example.html#cb144-104" aria-hidden="true" tabindex="-1"></a>  <span class="co"># generalize log, tolerant to 0 and negative values</span></span>
<span id="cb144-105"><a href="example.html#cb144-105" aria-hidden="true" tabindex="-1"></a>  LogNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x, min.val) {</span>
<span id="cb144-106"><a href="example.html#cb144-106" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">log10</span>((x <span class="sc">+</span> <span class="fu">sqrt</span>(x<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> min.val<span class="sc">^</span><span class="dv">2</span>))<span class="sc">/</span><span class="dv">2</span>))</span>
<span id="cb144-107"><a href="example.html#cb144-107" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb144-108"><a href="example.html#cb144-108" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-109"><a href="example.html#cb144-109" aria-hidden="true" tabindex="-1"></a>  <span class="co"># square root, tolerant to negative values</span></span>
<span id="cb144-110"><a href="example.html#cb144-110" aria-hidden="true" tabindex="-1"></a>  SquareRootNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x, min.val) {</span>
<span id="cb144-111"><a href="example.html#cb144-111" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(((x <span class="sc">+</span> <span class="fu">sqrt</span>(x<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> min.val<span class="sc">^</span><span class="dv">2</span>))<span class="sc">/</span><span class="dv">2</span>)<span class="sc">^</span>(<span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>))</span>
<span id="cb144-112"><a href="example.html#cb144-112" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb144-113"><a href="example.html#cb144-113" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-114"><a href="example.html#cb144-114" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (transNorm <span class="sc">==</span> <span class="st">&quot;LogNorm&quot;</span>) {</span>
<span id="cb144-115"><a href="example.html#cb144-115" aria-hidden="true" tabindex="-1"></a>    min.val <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">abs</span>(data[data <span class="sc">!=</span> <span class="dv">0</span>]))<span class="sc">/</span><span class="dv">10</span></span>
<span id="cb144-116"><a href="example.html#cb144-116" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, LogNorm, min.val)</span>
<span id="cb144-117"><a href="example.html#cb144-117" aria-hidden="true" tabindex="-1"></a>    transnm <span class="ot">&lt;-</span> <span class="st">&quot;Log10 Normalization&quot;</span></span>
<span id="cb144-118"><a href="example.html#cb144-118" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (transNorm <span class="sc">==</span> <span class="st">&quot;SrNorm&quot;</span>) {</span>
<span id="cb144-119"><a href="example.html#cb144-119" aria-hidden="true" tabindex="-1"></a>    min.val <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">abs</span>(data[data <span class="sc">!=</span> <span class="dv">0</span>]))<span class="sc">/</span><span class="dv">10</span></span>
<span id="cb144-120"><a href="example.html#cb144-120" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, SquareRootNorm, min.val)</span>
<span id="cb144-121"><a href="example.html#cb144-121" aria-hidden="true" tabindex="-1"></a>    transnm <span class="ot">&lt;-</span> <span class="st">&quot;Square Root Transformation&quot;</span></span>
<span id="cb144-122"><a href="example.html#cb144-122" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (transNorm <span class="sc">==</span> <span class="st">&quot;CrNorm&quot;</span>) {</span>
<span id="cb144-123"><a href="example.html#cb144-123" aria-hidden="true" tabindex="-1"></a>    norm.data <span class="ot">&lt;-</span> <span class="fu">abs</span>(data)<span class="sc">^</span>(<span class="dv">1</span><span class="sc">/</span><span class="dv">3</span>)</span>
<span id="cb144-124"><a href="example.html#cb144-124" aria-hidden="true" tabindex="-1"></a>    norm.data[data <span class="sc">&lt;</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="sc">-</span>norm.data[data <span class="sc">&lt;</span> <span class="dv">0</span>]</span>
<span id="cb144-125"><a href="example.html#cb144-125" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> norm.data</span>
<span id="cb144-126"><a href="example.html#cb144-126" aria-hidden="true" tabindex="-1"></a>    transnm <span class="ot">&lt;-</span> <span class="st">&quot;Cubic Root Transformation&quot;</span></span>
<span id="cb144-127"><a href="example.html#cb144-127" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb144-128"><a href="example.html#cb144-128" aria-hidden="true" tabindex="-1"></a>    transnm <span class="ot">&lt;-</span> <span class="st">&quot;N/A&quot;</span></span>
<span id="cb144-129"><a href="example.html#cb144-129" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb144-130"><a href="example.html#cb144-130" aria-hidden="true" tabindex="-1"></a>  <span class="do">#############################################</span></span>
<span id="cb144-131"><a href="example.html#cb144-131" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-132"><a href="example.html#cb144-132" aria-hidden="true" tabindex="-1"></a>  <span class="do">#############################################</span></span>
<span id="cb144-133"><a href="example.html#cb144-133" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Data scaling</span></span>
<span id="cb144-134"><a href="example.html#cb144-134" aria-hidden="true" tabindex="-1"></a>  <span class="co"># normalize to zero mean and unit variance</span></span>
<span id="cb144-135"><a href="example.html#cb144-135" aria-hidden="true" tabindex="-1"></a>  AutoNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb144-136"><a href="example.html#cb144-136" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>((x <span class="sc">-</span> <span class="fu">mean</span>(x))<span class="sc">/</span><span class="fu">sd</span>(x, <span class="at">na.rm =</span> T))</span>
<span id="cb144-137"><a href="example.html#cb144-137" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb144-138"><a href="example.html#cb144-138" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-139"><a href="example.html#cb144-139" aria-hidden="true" tabindex="-1"></a>  <span class="co"># normalize to zero mean but variance/SE</span></span>
<span id="cb144-140"><a href="example.html#cb144-140" aria-hidden="true" tabindex="-1"></a>  ParetoNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb144-141"><a href="example.html#cb144-141" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>((x <span class="sc">-</span> <span class="fu">mean</span>(x))<span class="sc">/</span><span class="fu">sqrt</span>(<span class="fu">sd</span>(x, <span class="at">na.rm =</span> T)))</span>
<span id="cb144-142"><a href="example.html#cb144-142" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb144-143"><a href="example.html#cb144-143" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-144"><a href="example.html#cb144-144" aria-hidden="true" tabindex="-1"></a>  <span class="co"># normalize to zero mean but variance/SE</span></span>
<span id="cb144-145"><a href="example.html#cb144-145" aria-hidden="true" tabindex="-1"></a>  MeanCenter <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb144-146"><a href="example.html#cb144-146" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(x <span class="sc">-</span> <span class="fu">mean</span>(x))</span>
<span id="cb144-147"><a href="example.html#cb144-147" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb144-148"><a href="example.html#cb144-148" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-149"><a href="example.html#cb144-149" aria-hidden="true" tabindex="-1"></a>  <span class="co"># normalize to zero mean but variance/SE</span></span>
<span id="cb144-150"><a href="example.html#cb144-150" aria-hidden="true" tabindex="-1"></a>  RangeNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb144-151"><a href="example.html#cb144-151" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">max</span>(x) <span class="sc">==</span> <span class="fu">min</span>(x)) {</span>
<span id="cb144-152"><a href="example.html#cb144-152" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(x)</span>
<span id="cb144-153"><a href="example.html#cb144-153" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb144-154"><a href="example.html#cb144-154" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>((x <span class="sc">-</span> <span class="fu">mean</span>(x))<span class="sc">/</span>(<span class="fu">max</span>(x) <span class="sc">-</span> <span class="fu">min</span>(x)))</span>
<span id="cb144-155"><a href="example.html#cb144-155" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb144-156"><a href="example.html#cb144-156" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb144-157"><a href="example.html#cb144-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb144-158"><a href="example.html#cb144-158" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (scaleNorm <span class="sc">==</span> <span class="st">&quot;MeanCenter&quot;</span>) {</span>
<span id="cb144-159"><a href="example.html#cb144-159" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, MeanCenter)</span>
<span id="cb144-160"><a href="example.html#cb144-160" aria-hidden="true" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;Mean Centering&quot;</span></span>
<span id="cb144-161"><a href="example.html#cb144-161" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (scaleNorm <span class="sc">==</span> <span class="st">&quot;AutoNorm&quot;</span>) {</span>
<span id="cb144-162"><a href="example.html#cb144-162" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, AutoNorm)</span>
<span id="cb144-163"><a href="example.html#cb144-163" aria-hidden="true" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;Autoscaling&quot;</span></span>
<span id="cb144-164"><a href="example.html#cb144-164" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (scaleNorm <span class="sc">==</span> <span class="st">&quot;ParetoNorm&quot;</span>) {</span>
<span id="cb144-165"><a href="example.html#cb144-165" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, ParetoNorm)</span>
<span id="cb144-166"><a href="example.html#cb144-166" aria-hidden="true" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;Pareto Scaling&quot;</span></span>
<span id="cb144-167"><a href="example.html#cb144-167" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (scaleNorm <span class="sc">==</span> <span class="st">&quot;RangeNorm&quot;</span>) {</span>
<span id="cb144-168"><a href="example.html#cb144-168" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, RangeNorm)</span>
<span id="cb144-169"><a href="example.html#cb144-169" aria-hidden="true" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;Range Scaling&quot;</span></span>
<span id="cb144-170"><a href="example.html#cb144-170" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb144-171"><a href="example.html#cb144-171" aria-hidden="true" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;N/A&quot;</span></span>
<span id="cb144-172"><a href="example.html#cb144-172" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb144-173"><a href="example.html#cb144-173" aria-hidden="true" tabindex="-1"></a>  <span class="do">############################################# </span></span>
<span id="cb144-174"><a href="example.html#cb144-174" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-175"><a href="example.html#cb144-175" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">&quot;Row norm: &quot;</span>, rownm, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>,</span>
<span id="cb144-176"><a href="example.html#cb144-176" aria-hidden="true" tabindex="-1"></a>          <span class="st">&quot;Data Transformation norm: &quot;</span>, transnm, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>,</span>
<span id="cb144-177"><a href="example.html#cb144-177" aria-hidden="true" tabindex="-1"></a>          <span class="st">&quot;Data Scaling norm: &quot;</span>, scalenm, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)  </span>
<span id="cb144-178"><a href="example.html#cb144-178" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-179"><a href="example.html#cb144-179" aria-hidden="true" tabindex="-1"></a>  <span class="co"># note after using &quot;apply&quot; function, all the attribute lost, need to add back</span></span>
<span id="cb144-180"><a href="example.html#cb144-180" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(data) <span class="ot">&lt;-</span> rowNames</span>
<span id="cb144-181"><a href="example.html#cb144-181" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(data) <span class="ot">&lt;-</span> colNames</span>
<span id="cb144-182"><a href="example.html#cb144-182" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-183"><a href="example.html#cb144-183" aria-hidden="true" tabindex="-1"></a>  target <span class="ot">&lt;-</span> metadata_tab <span class="sc">%&gt;%</span> </span>
<span id="cb144-184"><a href="example.html#cb144-184" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb144-185"><a href="example.html#cb144-185" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb144-186"><a href="example.html#cb144-186" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(SampleID<span class="sc">%in%</span><span class="fu">rownames</span>(data))</span>
<span id="cb144-187"><a href="example.html#cb144-187" aria-hidden="true" tabindex="-1"></a>  se <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb144-188"><a href="example.html#cb144-188" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">features =</span> data)</span>
<span id="cb144-189"><a href="example.html#cb144-189" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-190"><a href="example.html#cb144-190" aria-hidden="true" tabindex="-1"></a>  <span class="co"># need to do some sanity check, for log there may be Inf values introduced</span></span>
<span id="cb144-191"><a href="example.html#cb144-191" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">CheckData</span>(se)</span>
<span id="cb144-192"><a href="example.html#cb144-192" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb144-193"><a href="example.html#cb144-193" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb144-194"><a href="example.html#cb144-194" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb144-195"><a href="example.html#cb144-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb144-196"><a href="example.html#cb144-196" aria-hidden="true" tabindex="-1"></a>se_normalize <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(</span>
<span id="cb144-197"><a href="example.html#cb144-197" aria-hidden="true" tabindex="-1"></a>                      <span class="at">object =</span> se_impute,</span>
<span id="cb144-198"><a href="example.html#cb144-198" aria-hidden="true" tabindex="-1"></a>                      <span class="at">rowNorm =</span> <span class="st">&quot;None&quot;</span>,</span>
<span id="cb144-199"><a href="example.html#cb144-199" aria-hidden="true" tabindex="-1"></a>                      <span class="at">transNorm =</span> <span class="st">&quot;LogNorm&quot;</span>,</span>
<span id="cb144-200"><a href="example.html#cb144-200" aria-hidden="true" tabindex="-1"></a>                      <span class="at">scaleNorm =</span> <span class="st">&quot;ParetoNorm&quot;</span>)</span>
<span id="cb144-201"><a href="example.html#cb144-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb144-202"><a href="example.html#cb144-202" aria-hidden="true" tabindex="-1"></a>se_normalize</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 540 54 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(540): MEDL00066 MEDL00375 ... MW0168675 MW0169569
## rowData names(0):
## colnames(54): CJY_V0 CJY_V1 ... ZTX2_V0 ZTX2_V1
## colData names(5): group SampleName FMT_status gender age</code></pre>
</div>
<div id="normalization-by-poma-r-package-1" class="section level4 hasAnchor" number="8.3.4.2">
<h4><span class="header-section-number">8.3.4.2</span> Normalization by <strong>POMA</strong> R package<a href="example.html#normalization-by-poma-r-package-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="example.html#cb146-1" aria-hidden="true" tabindex="-1"></a>se_normalize_v2 <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_impute, <span class="at">method =</span> <span class="st">&quot;log_pareto&quot;</span>)</span>
<span id="cb146-2"><a href="example.html#cb146-2" aria-hidden="true" tabindex="-1"></a>se_normalize_v2</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 540 54 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(540): MEDL00066 MEDL00375 ... MW0168675 MW0169569
## rowData names(0):
## colnames(54): CJY_V0 CJY_V1 ... ZTX2_V0 ZTX2_V1
## colData names(5): group SampleName FMT_status gender age</code></pre>
</div>
<div id="comparison-of-unnormalized-and-normalized-dataset-1" class="section level4 hasAnchor" number="8.3.4.3">
<h4><span class="header-section-number">8.3.4.3</span> Comparison of unnormalized and normalized dataset<a href="example.html#comparison-of-unnormalized-and-normalized-dataset-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>boxplot</li>
</ul>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="example.html#cb148-1" aria-hidden="true" tabindex="-1"></a>pl_unnor <span class="ot">&lt;-</span> <span class="fu">PomaBoxplots</span>(se_impute, <span class="at">group =</span> <span class="st">&quot;samples&quot;</span>, <span class="at">jitter =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb148-2"><a href="example.html#cb148-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Not Normalized&quot;</span>) <span class="sc">+</span></span>
<span id="cb148-3"><a href="example.html#cb148-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="co"># data before normalization</span></span>
<span id="cb148-4"><a href="example.html#cb148-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-5"><a href="example.html#cb148-5" aria-hidden="true" tabindex="-1"></a>pl_nor <span class="ot">&lt;-</span> <span class="fu">PomaBoxplots</span>(se_normalize, <span class="at">group =</span> <span class="st">&quot;samples&quot;</span>, <span class="at">jitter =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb148-6"><a href="example.html#cb148-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Normalized&quot;</span>) <span class="co"># data after normalization</span></span>
<span id="cb148-7"><a href="example.html#cb148-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-8"><a href="example.html#cb148-8" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(pl_unnor, pl_nor, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">align =</span> <span class="st">&quot;v&quot;</span>)</span></code></pre></div>
<p><img src="Statistical-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-94-1.png" width="1344" /></p>
<ul>
<li>density</li>
</ul>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="example.html#cb149-1" aria-hidden="true" tabindex="-1"></a>pl_unnor <span class="ot">&lt;-</span> <span class="fu">PomaDensity</span>(se_impute, <span class="at">group =</span> <span class="st">&quot;features&quot;</span>) <span class="sc">+</span></span>
<span id="cb149-2"><a href="example.html#cb149-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Not Normalized&quot;</span>) <span class="sc">+</span></span>
<span id="cb149-3"><a href="example.html#cb149-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="co"># data before normalization</span></span>
<span id="cb149-4"><a href="example.html#cb149-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb149-5"><a href="example.html#cb149-5" aria-hidden="true" tabindex="-1"></a>pl_nor <span class="ot">&lt;-</span> <span class="fu">PomaDensity</span>(se_normalize, <span class="at">group =</span> <span class="st">&quot;features&quot;</span>) <span class="sc">+</span></span>
<span id="cb149-6"><a href="example.html#cb149-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Normalized&quot;</span>) <span class="co"># data after normalization</span></span>
<span id="cb149-7"><a href="example.html#cb149-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb149-8"><a href="example.html#cb149-8" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(pl_unnor, pl_nor, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">align =</span> <span class="st">&quot;v&quot;</span>)</span></code></pre></div>
<p><img src="Statistical-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-95-1.png" width="960" /></p>
</div>
</div>
</div>
<div id="cluster-analysis-1" class="section level2 hasAnchor" number="8.4">
<h2><span class="header-section-number">8.4</span> Cluster Analysis<a href="example.html#cluster-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="hierarchical-clustering-1" class="section level3 hasAnchor" number="8.4.1">
<h3><span class="header-section-number">8.4.1</span> Hierarchical Clustering<a href="example.html#hierarchical-clustering-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb150"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb150-1"><a href="example.html#cb150-1" aria-hidden="true" tabindex="-1"></a>HieraCluster <span class="ot">&lt;-</span> <span class="cf">function</span>(object,</span>
<span id="cb150-2"><a href="example.html#cb150-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">method_dis =</span> <span class="fu">c</span>(<span class="st">&quot;euclidean&quot;</span>, <span class="st">&quot;bray&quot;</span>),</span>
<span id="cb150-3"><a href="example.html#cb150-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">method_cluster =</span> <span class="fu">c</span>(<span class="st">&quot;average&quot;</span>, <span class="st">&quot;single&quot;</span>, <span class="st">&quot;complete&quot;</span>, <span class="st">&quot;ward&quot;</span>, <span class="st">&quot;ward.D2&quot;</span>),</span>
<span id="cb150-4"><a href="example.html#cb150-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">cluster_type =</span> <span class="fu">c</span>(<span class="st">&quot;Agglomerative&quot;</span>, <span class="st">&quot;Divisive&quot;</span>),</span>
<span id="cb150-5"><a href="example.html#cb150-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">tree_num =</span> <span class="dv">4</span>) {</span>
<span id="cb150-6"><a href="example.html#cb150-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb150-7"><a href="example.html#cb150-7" aria-hidden="true" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) </span>
<span id="cb150-8"><a href="example.html#cb150-8" aria-hidden="true" tabindex="-1"></a>  metadata_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object)</span>
<span id="cb150-9"><a href="example.html#cb150-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb150-10"><a href="example.html#cb150-10" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">t</span>(features_tab)</span>
<span id="cb150-11"><a href="example.html#cb150-11" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (cluster_type <span class="sc">==</span> <span class="st">&quot;Agglomerative&quot;</span>) {</span>
<span id="cb150-12"><a href="example.html#cb150-12" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Agglomerative Hierarchical Clustering </span></span>
<span id="cb150-13"><a href="example.html#cb150-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Dissimilarity matrix</span></span>
<span id="cb150-14"><a href="example.html#cb150-14" aria-hidden="true" tabindex="-1"></a>    d <span class="ot">&lt;-</span> <span class="fu">dist</span>(df, <span class="at">method =</span> method_dis)</span>
<span id="cb150-15"><a href="example.html#cb150-15" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb150-16"><a href="example.html#cb150-16" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Hierarchical clustering using Linkage method</span></span>
<span id="cb150-17"><a href="example.html#cb150-17" aria-hidden="true" tabindex="-1"></a>    hc <span class="ot">&lt;-</span> <span class="fu">hclust</span>(d, <span class="at">method =</span> method_cluster)</span>
<span id="cb150-18"><a href="example.html#cb150-18" aria-hidden="true" tabindex="-1"></a>    <span class="co"># hc &lt;- agnes(df, method = method_cluster)</span></span>
<span id="cb150-19"><a href="example.html#cb150-19" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb150-20"><a href="example.html#cb150-20" aria-hidden="true" tabindex="-1"></a>    <span class="do">####### identifying the strongest clustering structure ################</span></span>
<span id="cb150-21"><a href="example.html#cb150-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># # methods to assess</span></span>
<span id="cb150-22"><a href="example.html#cb150-22" aria-hidden="true" tabindex="-1"></a>    <span class="co"># m &lt;- c( &quot;average&quot;, &quot;single&quot;, &quot;complete&quot;, &quot;ward&quot;)</span></span>
<span id="cb150-23"><a href="example.html#cb150-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># names(m) &lt;- c( &quot;average&quot;, &quot;single&quot;, &quot;complete&quot;, &quot;ward&quot;)</span></span>
<span id="cb150-24"><a href="example.html#cb150-24" aria-hidden="true" tabindex="-1"></a>    <span class="co"># </span></span>
<span id="cb150-25"><a href="example.html#cb150-25" aria-hidden="true" tabindex="-1"></a>    <span class="co"># # function to compute coefficient</span></span>
<span id="cb150-26"><a href="example.html#cb150-26" aria-hidden="true" tabindex="-1"></a>    <span class="co"># ac &lt;- function(x) {</span></span>
<span id="cb150-27"><a href="example.html#cb150-27" aria-hidden="true" tabindex="-1"></a>    <span class="co">#   agnes(df, method = x)$ac</span></span>
<span id="cb150-28"><a href="example.html#cb150-28" aria-hidden="true" tabindex="-1"></a>    <span class="co"># }</span></span>
<span id="cb150-29"><a href="example.html#cb150-29" aria-hidden="true" tabindex="-1"></a>    <span class="co"># </span></span>
<span id="cb150-30"><a href="example.html#cb150-30" aria-hidden="true" tabindex="-1"></a>    <span class="co"># map_dbl(m, ac)     </span></span>
<span id="cb150-31"><a href="example.html#cb150-31" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (cluster_type <span class="sc">==</span> <span class="st">&quot;Divisive&quot;</span>) {</span>
<span id="cb150-32"><a href="example.html#cb150-32" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Divisive Hierarchical Clustering </span></span>
<span id="cb150-33"><a href="example.html#cb150-33" aria-hidden="true" tabindex="-1"></a>    hc <span class="ot">&lt;-</span> <span class="fu">diana</span>(df, <span class="at">metric =</span> method_dis)</span>
<span id="cb150-34"><a href="example.html#cb150-34" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb150-35"><a href="example.html#cb150-35" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb150-36"><a href="example.html#cb150-36" aria-hidden="true" tabindex="-1"></a>  hc_res <span class="ot">&lt;-</span> <span class="fu">as.hclust</span>(hc)</span>
<span id="cb150-37"><a href="example.html#cb150-37" aria-hidden="true" tabindex="-1"></a>  sub_grp <span class="ot">&lt;-</span> <span class="fu">cutree</span>(hc_res, <span class="at">k =</span> tree_num)</span>
<span id="cb150-38"><a href="example.html#cb150-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb150-39"><a href="example.html#cb150-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(hc_res, <span class="at">cex =</span> <span class="fl">0.6</span>)</span>
<span id="cb150-40"><a href="example.html#cb150-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rect.hclust</span>(hc_res, <span class="at">k =</span> tree_num, <span class="at">border =</span> <span class="dv">2</span><span class="sc">:</span>(tree_num<span class="sc">+</span><span class="dv">1</span>)) </span>
<span id="cb150-41"><a href="example.html#cb150-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb150-42"><a href="example.html#cb150-42" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">data=</span>df,</span>
<span id="cb150-43"><a href="example.html#cb150-43" aria-hidden="true" tabindex="-1"></a>              <span class="at">cluster=</span>sub_grp,</span>
<span id="cb150-44"><a href="example.html#cb150-44" aria-hidden="true" tabindex="-1"></a>              <span class="at">hc=</span>hc_res)</span>
<span id="cb150-45"><a href="example.html#cb150-45" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb150-46"><a href="example.html#cb150-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb150-47"><a href="example.html#cb150-47" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="example.html#cb151-1" aria-hidden="true" tabindex="-1"></a>Agg_hc_res <span class="ot">&lt;-</span> <span class="fu">HieraCluster</span>(</span>
<span id="cb151-2"><a href="example.html#cb151-2" aria-hidden="true" tabindex="-1"></a>      <span class="at">object =</span> se_normalize,</span>
<span id="cb151-3"><a href="example.html#cb151-3" aria-hidden="true" tabindex="-1"></a>      <span class="at">method_dis =</span> <span class="st">&quot;euclidean&quot;</span>,</span>
<span id="cb151-4"><a href="example.html#cb151-4" aria-hidden="true" tabindex="-1"></a>      <span class="at">method_cluster =</span> <span class="st">&quot;ward.D2&quot;</span>,</span>
<span id="cb151-5"><a href="example.html#cb151-5" aria-hidden="true" tabindex="-1"></a>      <span class="at">cluster_type =</span> <span class="st">&quot;Agglomerative&quot;</span>,</span>
<span id="cb151-6"><a href="example.html#cb151-6" aria-hidden="true" tabindex="-1"></a>      <span class="at">tree_num =</span> <span class="dv">3</span>)</span></code></pre></div>
<p><img src="Statistical-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-97-1.png" width="768" /></p>
<ul>
<li>Visualization: visualize the result in a scatter plot</li>
</ul>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="example.html#cb152-1" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_cluster</span>(<span class="fu">list</span>(<span class="at">data =</span> Agg_hc_res<span class="sc">$</span>data, </span>
<span id="cb152-2"><a href="example.html#cb152-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> Agg_hc_res<span class="sc">$</span>cluster))</span></code></pre></div>
<p><img src="Statistical-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-98-1.png" width="768" /></p>
</div>
</div>
<div id="chemometrics-analysis-1" class="section level2 hasAnchor" number="8.5">
<h2><span class="header-section-number">8.5</span> Chemometrics Analysis<a href="example.html#chemometrics-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="partial-least-squares-discriminant-analysis-pls-da-1" class="section level3 hasAnchor" number="8.5.1">
<h3><span class="header-section-number">8.5.1</span> Partial Least Squares-Discriminant Analysis (PLS-DA)<a href="example.html#partial-least-squares-discriminant-analysis-pls-da-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="example.html#cb153-1" aria-hidden="true" tabindex="-1"></a>poma_plsda <span class="ot">&lt;-</span> <span class="fu">PomaMultivariate</span>(se_normalize, <span class="at">method =</span> <span class="st">&quot;plsda&quot;</span>)</span></code></pre></div>
<ul>
<li>scatter plot</li>
</ul>
<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb154-1"><a href="example.html#cb154-1" aria-hidden="true" tabindex="-1"></a>poma_plsda<span class="sc">$</span>scoresplot <span class="sc">+</span></span>
<span id="cb154-2"><a href="example.html#cb154-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Scores Plot (plsda)&quot;</span>)</span></code></pre></div>
<p><img src="Statistical-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-100-1.png" width="768" /></p>
<ul>
<li>errors plot</li>
</ul>
<div class="sourceCode" id="cb155"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb155-1"><a href="example.html#cb155-1" aria-hidden="true" tabindex="-1"></a>poma_plsda<span class="sc">$</span>errors_plsda_plot <span class="sc">+</span></span>
<span id="cb155-2"><a href="example.html#cb155-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Error Plot (plsda)&quot;</span>)</span></code></pre></div>
<p><img src="Statistical-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-101-1.png" width="768" /></p>
</div>
<div id="sparse-partial-least-squares-discriminant-analysis-spls-da-1" class="section level3 hasAnchor" number="8.5.2">
<h3><span class="header-section-number">8.5.2</span> Sparse Partial Least Squares-Discriminant Analysis (sPLS-DA)<a href="example.html#sparse-partial-least-squares-discriminant-analysis-spls-da-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<blockquote>
<p>Even though PLS is highly efficient in a high dimensional context, the interpretability of PLS needed to be improved. sPLS has been recently developed by our team to perform simultaneous variable selection in both data sets X and Y data sets, by including LASSO penalizations in PLS on each pair of loading vectors</p>
</blockquote>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="example.html#cb156-1" aria-hidden="true" tabindex="-1"></a>poma_splsda <span class="ot">&lt;-</span> <span class="fu">PomaMultivariate</span>(se_normalize, <span class="at">method =</span> <span class="st">&quot;splsda&quot;</span>)</span></code></pre></div>
<ul>
<li>scatter plot</li>
</ul>
<div class="sourceCode" id="cb157"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb157-1"><a href="example.html#cb157-1" aria-hidden="true" tabindex="-1"></a>poma_splsda<span class="sc">$</span>scoresplot <span class="sc">+</span></span>
<span id="cb157-2"><a href="example.html#cb157-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Scores Plot (splsda)&quot;</span>)</span></code></pre></div>
<p><img src="Statistical-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-103-1.png" width="768" /></p>
</div>
</div>
<div id="univariate-analysis-1" class="section level2 hasAnchor" number="8.6">
<h2><span class="header-section-number">8.6</span> Univariate Analysis<a href="example.html#univariate-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="fold-change-analysis-1" class="section level3 hasAnchor" number="8.6.1">
<h3><span class="header-section-number">8.6.1</span> Fold Change Analysis<a href="example.html#fold-change-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="example.html#cb158-1" aria-hidden="true" tabindex="-1"></a>FoldChange <span class="ot">&lt;-</span> <span class="cf">function</span>(object,</span>
<span id="cb158-2"><a href="example.html#cb158-2" aria-hidden="true" tabindex="-1"></a>                       group_names,</span>
<span id="cb158-3"><a href="example.html#cb158-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">fc_cutoff =</span> <span class="dv">2</span>,</span>
<span id="cb158-4"><a href="example.html#cb158-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">cmp_type =</span> <span class="dv">0</span>, </span>
<span id="cb158-5"><a href="example.html#cb158-5" aria-hidden="true" tabindex="-1"></a>                       <span class="at">paired =</span> <span class="cn">FALSE</span>) {</span>
<span id="cb158-6"><a href="example.html#cb158-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb158-7"><a href="example.html#cb158-7" aria-hidden="true" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) </span>
<span id="cb158-8"><a href="example.html#cb158-8" aria-hidden="true" tabindex="-1"></a>  metadata_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object)   </span>
<span id="cb158-9"><a href="example.html#cb158-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb158-10"><a href="example.html#cb158-10" aria-hidden="true" tabindex="-1"></a>  metadata <span class="ot">&lt;-</span> metadata_tab <span class="sc">%&gt;%</span></span>
<span id="cb158-11"><a href="example.html#cb158-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb158-12"><a href="example.html#cb158-12" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(group <span class="sc">%in%</span> group_names)</span>
<span id="cb158-13"><a href="example.html#cb158-13" aria-hidden="true" tabindex="-1"></a>  profile <span class="ot">&lt;-</span> features_tab[, <span class="fu">pmatch</span>(<span class="fu">rownames</span>(metadata), <span class="fu">colnames</span>(features_tab))]</span>
<span id="cb158-14"><a href="example.html#cb158-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb158-15"><a href="example.html#cb158-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make sure threshold is above 1</span></span>
<span id="cb158-16"><a href="example.html#cb158-16" aria-hidden="true" tabindex="-1"></a>  fc_cutoff <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(fc_cutoff <span class="sc">&gt;</span> <span class="dv">1</span>, fc_cutoff, <span class="dv">1</span><span class="sc">/</span>fc_cutoff)</span>
<span id="cb158-17"><a href="example.html#cb158-17" aria-hidden="true" tabindex="-1"></a>  max_cutoff <span class="ot">&lt;-</span> fc_cutoff</span>
<span id="cb158-18"><a href="example.html#cb158-18" aria-hidden="true" tabindex="-1"></a>  min_cutoff <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">/</span> fc_cutoff</span>
<span id="cb158-19"><a href="example.html#cb158-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb158-20"><a href="example.html#cb158-20" aria-hidden="true" tabindex="-1"></a>  GetFC <span class="ot">&lt;-</span> <span class="cf">function</span>(x,</span>
<span id="cb158-21"><a href="example.html#cb158-21" aria-hidden="true" tabindex="-1"></a>                    y,</span>
<span id="cb158-22"><a href="example.html#cb158-22" aria-hidden="true" tabindex="-1"></a>                    cmpType, </span>
<span id="cb158-23"><a href="example.html#cb158-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">paired =</span> <span class="cn">FALSE</span>) {</span>
<span id="cb158-24"><a href="example.html#cb158-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-25"><a href="example.html#cb158-25" aria-hidden="true" tabindex="-1"></a>    <span class="co"># x = profile</span></span>
<span id="cb158-26"><a href="example.html#cb158-26" aria-hidden="true" tabindex="-1"></a>    <span class="co"># y = metadata</span></span>
<span id="cb158-27"><a href="example.html#cb158-27" aria-hidden="true" tabindex="-1"></a>    <span class="co"># cmpType = cmp_type</span></span>
<span id="cb158-28"><a href="example.html#cb158-28" aria-hidden="true" tabindex="-1"></a>    <span class="co"># paired = paired    </span></span>
<span id="cb158-29"><a href="example.html#cb158-29" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb158-30"><a href="example.html#cb158-30" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> <span class="fu">t</span>(x)</span>
<span id="cb158-31"><a href="example.html#cb158-31" aria-hidden="true" tabindex="-1"></a>    colNames <span class="ot">&lt;-</span> <span class="fu">colnames</span>(x)</span>
<span id="cb158-32"><a href="example.html#cb158-32" aria-hidden="true" tabindex="-1"></a>    rowNames <span class="ot">&lt;-</span> <span class="fu">rownames</span>(x)</span>
<span id="cb158-33"><a href="example.html#cb158-33" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb158-34"><a href="example.html#cb158-34" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (paired) { </span>
<span id="cb158-35"><a href="example.html#cb158-35" aria-hidden="true" tabindex="-1"></a>      <span class="co"># compute the average of paired FC (unit is pair)</span></span>
<span id="cb158-36"><a href="example.html#cb158-36" aria-hidden="true" tabindex="-1"></a>      G1 <span class="ot">&lt;-</span> x[<span class="fu">which</span>(y<span class="sc">$</span>group <span class="sc">==</span> group_names[<span class="dv">1</span>]), ]</span>
<span id="cb158-37"><a href="example.html#cb158-37" aria-hidden="true" tabindex="-1"></a>      G2 <span class="ot">&lt;-</span> x[<span class="fu">which</span>(y<span class="sc">$</span>group <span class="sc">==</span> group_names[<span class="dv">2</span>]), ]</span>
<span id="cb158-38"><a href="example.html#cb158-38" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb158-39"><a href="example.html#cb158-39" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (cmpType <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb158-40"><a href="example.html#cb158-40" aria-hidden="true" tabindex="-1"></a>        fc.mat <span class="ot">&lt;-</span> G1 <span class="sc">-</span> G2</span>
<span id="cb158-41"><a href="example.html#cb158-41" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb158-42"><a href="example.html#cb158-42" aria-hidden="true" tabindex="-1"></a>        fc.mat <span class="ot">&lt;-</span> G2 <span class="sc">-</span> G1</span>
<span id="cb158-43"><a href="example.html#cb158-43" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb158-44"><a href="example.html#cb158-44" aria-hidden="true" tabindex="-1"></a>      fc.log <span class="ot">&lt;-</span> <span class="fu">apply</span>(fc.mat, <span class="dv">2</span>, mean)</span>
<span id="cb158-45"><a href="example.html#cb158-45" aria-hidden="true" tabindex="-1"></a>      fc.all <span class="ot">&lt;-</span> <span class="fu">signif</span>(<span class="dv">2</span><span class="sc">^</span>fc.log, <span class="dv">5</span>)</span>
<span id="cb158-46"><a href="example.html#cb158-46" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb158-47"><a href="example.html#cb158-47" aria-hidden="true" tabindex="-1"></a>      m1 <span class="ot">&lt;-</span> <span class="fu">colMeans</span>(x[<span class="fu">which</span>(y<span class="sc">$</span>group <span class="sc">==</span> group_names[<span class="dv">1</span>]), ])</span>
<span id="cb158-48"><a href="example.html#cb158-48" aria-hidden="true" tabindex="-1"></a>      m2 <span class="ot">&lt;-</span> <span class="fu">colMeans</span>(x[<span class="fu">which</span>(y<span class="sc">$</span>group <span class="sc">==</span> group_names[<span class="dv">2</span>]), ])</span>
<span id="cb158-49"><a href="example.html#cb158-49" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb158-50"><a href="example.html#cb158-50" aria-hidden="true" tabindex="-1"></a>      <span class="co"># create a named matrix of sig vars for display</span></span>
<span id="cb158-51"><a href="example.html#cb158-51" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (cmpType <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb158-52"><a href="example.html#cb158-52" aria-hidden="true" tabindex="-1"></a>          ratio <span class="ot">&lt;-</span> m1<span class="sc">/</span>m2</span>
<span id="cb158-53"><a href="example.html#cb158-53" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb158-54"><a href="example.html#cb158-54" aria-hidden="true" tabindex="-1"></a>          ratio <span class="ot">&lt;-</span> m2<span class="sc">/</span>m1</span>
<span id="cb158-55"><a href="example.html#cb158-55" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb158-56"><a href="example.html#cb158-56" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb158-57"><a href="example.html#cb158-57" aria-hidden="true" tabindex="-1"></a>      fc.all <span class="ot">&lt;-</span> <span class="fu">signif</span>(ratio, <span class="dv">5</span>)</span>
<span id="cb158-58"><a href="example.html#cb158-58" aria-hidden="true" tabindex="-1"></a>      ratio[ratio <span class="sc">&lt;</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb158-59"><a href="example.html#cb158-59" aria-hidden="true" tabindex="-1"></a>      fc.log <span class="ot">&lt;-</span> <span class="fu">signif</span>(<span class="fu">log2</span>(ratio), <span class="dv">5</span>)</span>
<span id="cb158-60"><a href="example.html#cb158-60" aria-hidden="true" tabindex="-1"></a>      fc.log[<span class="fu">is.infinite</span>(fc.log) <span class="sc">&amp;</span> fc.log <span class="sc">&lt;</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="sc">-</span><span class="dv">99</span></span>
<span id="cb158-61"><a href="example.html#cb158-61" aria-hidden="true" tabindex="-1"></a>      fc.log[<span class="fu">is.infinite</span>(fc.log) <span class="sc">&amp;</span> fc.log <span class="sc">&gt;</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="dv">99</span></span>
<span id="cb158-62"><a href="example.html#cb158-62" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb158-63"><a href="example.html#cb158-63" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb158-64"><a href="example.html#cb158-64" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(fc.all) <span class="ot">&lt;-</span> <span class="fu">names</span>(fc.log) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(x)</span>
<span id="cb158-65"><a href="example.html#cb158-65" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb158-66"><a href="example.html#cb158-66" aria-hidden="true" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">fc.all =</span> fc.all, <span class="at">fc.log =</span> fc.log)</span>
<span id="cb158-67"><a href="example.html#cb158-67" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(res)</span>
<span id="cb158-68"><a href="example.html#cb158-68" aria-hidden="true" tabindex="-1"></a>  }  </span>
<span id="cb158-69"><a href="example.html#cb158-69" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb158-70"><a href="example.html#cb158-70" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">GetFC</span>(profile, metadata, cmp_type, paired)</span>
<span id="cb158-71"><a href="example.html#cb158-71" aria-hidden="true" tabindex="-1"></a>  fc.all <span class="ot">&lt;-</span> res<span class="sc">$</span>fc.all</span>
<span id="cb158-72"><a href="example.html#cb158-72" aria-hidden="true" tabindex="-1"></a>  fc.log <span class="ot">&lt;-</span> res<span class="sc">$</span>fc.log</span>
<span id="cb158-73"><a href="example.html#cb158-73" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb158-74"><a href="example.html#cb158-74" aria-hidden="true" tabindex="-1"></a>  inx.up <span class="ot">&lt;-</span> fc.all <span class="sc">&gt;</span> max_cutoff</span>
<span id="cb158-75"><a href="example.html#cb158-75" aria-hidden="true" tabindex="-1"></a>  inx.down <span class="ot">&lt;-</span> fc.all <span class="sc">&lt;</span> min_cutoff</span>
<span id="cb158-76"><a href="example.html#cb158-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(inx.up) <span class="ot">&lt;-</span> <span class="fu">names</span>(inx.down) <span class="ot">&lt;-</span> <span class="fu">names</span>(fc.all)</span>
<span id="cb158-77"><a href="example.html#cb158-77" aria-hidden="true" tabindex="-1"></a>  imp.inx <span class="ot">&lt;-</span> inx.up <span class="sc">|</span> inx.down</span>
<span id="cb158-78"><a href="example.html#cb158-78" aria-hidden="true" tabindex="-1"></a>  sig.mat <span class="ot">&lt;-</span> <span class="fu">cbind</span>(fc.all[imp.inx, <span class="at">drop =</span> F], fc.log[imp.inx, <span class="at">drop =</span> F])</span>
<span id="cb158-79"><a href="example.html#cb158-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(sig.mat) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Fold Change&quot;</span>, <span class="st">&quot;log2(FC)&quot;</span>)</span>
<span id="cb158-80"><a href="example.html#cb158-80" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb158-81"><a href="example.html#cb158-81" aria-hidden="true" tabindex="-1"></a>  <span class="co"># order by absolute log value (since symmetrical in pos and neg)</span></span>
<span id="cb158-82"><a href="example.html#cb158-82" aria-hidden="true" tabindex="-1"></a>  inx.ord <span class="ot">&lt;-</span> <span class="fu">order</span>(<span class="fu">abs</span>(sig.mat[, <span class="dv">2</span>]), <span class="at">decreasing =</span> T)</span>
<span id="cb158-83"><a href="example.html#cb158-83" aria-hidden="true" tabindex="-1"></a>  sig.mat <span class="ot">&lt;-</span> sig.mat[inx.ord, , drop <span class="ot">=</span> F]</span>
<span id="cb158-84"><a href="example.html#cb158-84" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb158-85"><a href="example.html#cb158-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(sig.mat)</span>
<span id="cb158-86"><a href="example.html#cb158-86" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb158-87"><a href="example.html#cb158-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-88"><a href="example.html#cb158-88" aria-hidden="true" tabindex="-1"></a>FC_res <span class="ot">&lt;-</span> <span class="fu">FoldChange</span>(</span>
<span id="cb158-89"><a href="example.html#cb158-89" aria-hidden="true" tabindex="-1"></a>           <span class="at">object =</span> se_normalize, </span>
<span id="cb158-90"><a href="example.html#cb158-90" aria-hidden="true" tabindex="-1"></a>           <span class="at">fc_cutoff =</span> <span class="dv">2</span>,</span>
<span id="cb158-91"><a href="example.html#cb158-91" aria-hidden="true" tabindex="-1"></a>           <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;NR&quot;</span>, <span class="st">&quot;PR&quot;</span>),</span>
<span id="cb158-92"><a href="example.html#cb158-92" aria-hidden="true" tabindex="-1"></a>           <span class="at">cmp_type =</span> <span class="dv">0</span>, </span>
<span id="cb158-93"><a href="example.html#cb158-93" aria-hidden="true" tabindex="-1"></a>           <span class="at">paired =</span> <span class="cn">FALSE</span>)</span>
<span id="cb158-94"><a href="example.html#cb158-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-95"><a href="example.html#cb158-95" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(FC_res)</span></code></pre></div>
<pre><code>##           Fold Change log2(FC)
## MEDL00066    -8.31480      -99
## MEDL00375    -0.34779      -99
## MEDL01844    -0.40162      -99
## MEDL01922    -0.18899      -99
## MEDL01994    -0.54180      -99
## MEDL02018    -0.34413      -99</code></pre>
</div>
<div id="t-test-1" class="section level3 hasAnchor" number="8.6.2">
<h3><span class="header-section-number">8.6.2</span> T Test<a href="example.html#t-test-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb160"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb160-1"><a href="example.html#cb160-1" aria-hidden="true" tabindex="-1"></a>group_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;NR&quot;</span>, <span class="st">&quot;PR&quot;</span>)</span>
<span id="cb160-2"><a href="example.html#cb160-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb160-3"><a href="example.html#cb160-3" aria-hidden="true" tabindex="-1"></a>se_normalize_subset <span class="ot">&lt;-</span> se_normalize[, se_normalize<span class="sc">$</span>group <span class="sc">%in%</span> group_names]</span>
<span id="cb160-4"><a href="example.html#cb160-4" aria-hidden="true" tabindex="-1"></a>se_normalize_subset<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">as.character</span>(se_normalize_subset<span class="sc">$</span>group))</span>
<span id="cb160-5"><a href="example.html#cb160-5" aria-hidden="true" tabindex="-1"></a>ttest_res <span class="ot">&lt;-</span> <span class="fu">PomaUnivariate</span>(se_normalize_subset, <span class="at">method =</span> <span class="st">&quot;ttest&quot;</span>)</span>
<span id="cb160-6"><a href="example.html#cb160-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ttest_res)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 9
##   feature      FC diff_means pvalue pvalueAdj mean_NR mean_PR sd_NR sd_PR
##   &lt;chr&gt;     &lt;dbl&gt;      &lt;dbl&gt;  &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;
## 1 MEDL00066 -0.12      0.174 0.506      0.833 -0.155   0.0187 0.700 0.991
## 2 MEDL00375 -2.88      0.149 0.503      0.833 -0.0384  0.111  0.827 0.620
## 3 MEDL00587  1.40      0.014 0.937      0.980  0.0342  0.0480 0.571 0.573
## 4 MEDL01844 -2.49     -0.311 0.232      0.662  0.0891 -0.222  0.850 0.849
## 5 MEDL01922 -5.29      0.069 0.812      0.931 -0.0110  0.0584 0.751 1.13 
## 6 MEDL01994 -1.85      0.117 0.0795     0.434 -0.0412  0.0760 0.182 0.245</code></pre>
</div>
<div id="volcano-plot-1" class="section level3 hasAnchor" number="8.6.3">
<h3><span class="header-section-number">8.6.3</span> Volcano plot<a href="example.html#volcano-plot-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb162"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb162-1"><a href="example.html#cb162-1" aria-hidden="true" tabindex="-1"></a>se_impute_subset <span class="ot">&lt;-</span> se_impute[, se_impute<span class="sc">$</span>group <span class="sc">%in%</span> group_names]</span>
<span id="cb162-2"><a href="example.html#cb162-2" aria-hidden="true" tabindex="-1"></a>se_impute_subset<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">as.character</span>(se_impute_subset<span class="sc">$</span>group))</span>
<span id="cb162-3"><a href="example.html#cb162-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb162-4"><a href="example.html#cb162-4" aria-hidden="true" tabindex="-1"></a><span class="fu">PomaVolcano</span>(se_impute_subset, </span>
<span id="cb162-5"><a href="example.html#cb162-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">pval =</span> <span class="st">&quot;raw&quot;</span>,</span>
<span id="cb162-6"><a href="example.html#cb162-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">pval_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb162-7"><a href="example.html#cb162-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">log2FC =</span> <span class="fl">0.5</span>,</span>
<span id="cb162-8"><a href="example.html#cb162-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">xlim =</span> <span class="dv">3</span>,</span>
<span id="cb162-9"><a href="example.html#cb162-9" aria-hidden="true" tabindex="-1"></a>            <span class="at">labels =</span> <span class="cn">TRUE</span>,</span>
<span id="cb162-10"><a href="example.html#cb162-10" aria-hidden="true" tabindex="-1"></a>            <span class="at">plot_title =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Statistical-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-106-1.png" width="576" /></p>
</div>
</div>
<div id="feature-selection-1" class="section level2 hasAnchor" number="8.7">
<h2><span class="header-section-number">8.7</span> Feature Selection<a href="example.html#feature-selection-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="regularized-generalized-linear-models-lasso-alpha-1" class="section level3 hasAnchor" number="8.7.1">
<h3><span class="header-section-number">8.7.1</span> Regularized Generalized Linear Models (Lasso: alpha = 1)<a href="example.html#regularized-generalized-linear-models-lasso-alpha-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb163"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb163-1"><a href="example.html#cb163-1" aria-hidden="true" tabindex="-1"></a>lasso_res <span class="ot">&lt;-</span> <span class="fu">PomaLasso</span>(se_normalize_subset, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">labels =</span> <span class="cn">TRUE</span>)</span>
<span id="cb163-2"><a href="example.html#cb163-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb163-3"><a href="example.html#cb163-3" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(lasso_res<span class="sc">$</span>cvLassoPlot,</span>
<span id="cb163-4"><a href="example.html#cb163-4" aria-hidden="true" tabindex="-1"></a>                   lasso_res<span class="sc">$</span>coefficientPlot,</span>
<span id="cb163-5"><a href="example.html#cb163-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">align =</span> <span class="st">&quot;h&quot;</span>)</span></code></pre></div>
<p><img src="Statistical-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-107-1.png" width="1152" /></p>
<div class="sourceCode" id="cb164"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb164-1"><a href="example.html#cb164-1" aria-hidden="true" tabindex="-1"></a>lasso_res<span class="sc">$</span>coefficients</span></code></pre></div>
<pre><code>## # A tibble: 21 × 2
##    feature     coefficient
##    &lt;chr&gt;             &lt;dbl&gt;
##  1 (Intercept)     -0.207 
##  2 MEDL01844       -0.0577
##  3 MEDN0366        -0.478 
##  4 MEDN1135         0.542 
##  5 MEDN1285        -0.152 
##  6 MEDP0006         0.208 
##  7 MEDP0089        -0.228 
##  8 MEDP0151         0.329 
##  9 MEDP0295        -0.134 
## 10 MEDP0401        -0.220 
## # … with 11 more rows</code></pre>
</div>
<div id="classification-random-forest" class="section level3 hasAnchor" number="8.7.2">
<h3><span class="header-section-number">8.7.2</span> Classification (Random Forest)<a href="example.html#classification-random-forest" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb166"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb166-1"><a href="example.html#cb166-1" aria-hidden="true" tabindex="-1"></a>poma_rf <span class="ot">&lt;-</span> <span class="fu">PomaRandForest</span>(se_normalize_subset, <span class="at">ntest =</span> <span class="dv">10</span>, <span class="at">nvar =</span> <span class="dv">10</span>)</span>
<span id="cb166-2"><a href="example.html#cb166-2" aria-hidden="true" tabindex="-1"></a>poma_rf<span class="sc">$</span>error_tree</span></code></pre></div>
<p><img src="Statistical-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-108-1.png" width="576" /></p>
<ul>
<li>table</li>
</ul>
<div class="sourceCode" id="cb167"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb167-1"><a href="example.html#cb167-1" aria-hidden="true" tabindex="-1"></a>poma_rf<span class="sc">$</span>confusionMatrix<span class="sc">$</span>table</span></code></pre></div>
<pre><code>##           Reference
## Prediction 1 2
##          1 1 1
##          2 0 2</code></pre>
<ul>
<li>Important features</li>
</ul>
<div class="sourceCode" id="cb169"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb169-1"><a href="example.html#cb169-1" aria-hidden="true" tabindex="-1"></a>poma_rf<span class="sc">$</span>MeanDecreaseGini_plot</span></code></pre></div>
<p><img src="Statistical-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-110-1.png" width="576" /></p>
</div>
</div>
<div id="network-analysis-2" class="section level2 hasAnchor" number="8.8">
<h2><span class="header-section-number">8.8</span> Network Analysis<a href="example.html#network-analysis-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="data-curation-2" class="section level3 hasAnchor" number="8.8.1">
<h3><span class="header-section-number">8.8.1</span> Data curation<a href="example.html#data-curation-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb170"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb170-1"><a href="example.html#cb170-1" aria-hidden="true" tabindex="-1"></a>features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(se_filter) <span class="sc">%&gt;%</span> </span>
<span id="cb170-2"><a href="example.html#cb170-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>()</span>
<span id="cb170-3"><a href="example.html#cb170-3" aria-hidden="true" tabindex="-1"></a>features_tab[<span class="fu">is.na</span>(features_tab)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb170-4"><a href="example.html#cb170-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-5"><a href="example.html#cb170-5" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(features_tab[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>])</span></code></pre></div>
<pre><code>##        MEDL00066 MEDL00375 MEDL00587 MEDL01799 MEDL01801 MEDL01844 MEDL01922 MEDL01994 MEDL02002 MEDL02018
## CJY_V0     10447     31087    618200         0         0    127250         0   2590400         0    485040
## CJY_V1   4011900     88306    563130     51267   1046200   4172100    107560   3772200    133920   1400100
## DGL_V0    560560    449150   1079300    304040    147100  19462000     39149   2950700   3216700    684140
## DGL_V1    149120    263050   2047500     24342    357050  14373000    964600   3328000   2512900   1659500
## HXZ_V0    442540    216830   1399700   1481800   1976000   7830000    269570   2588200   3719000   3433700
## HXZ_V1    181010     37280   1171700         0         0   5171400    196770   3097700         0   3269900</code></pre>
</div>
<div id="building-network-model-6" class="section level3 hasAnchor" number="8.8.2">
<h3><span class="header-section-number">8.8.2</span> Building network model<a href="example.html#building-network-model-6" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb172"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb172-1"><a href="example.html#cb172-1" aria-hidden="true" tabindex="-1"></a>net_single <span class="ot">&lt;-</span> <span class="fu">netConstruct</span>(features_tab, </span>
<span id="cb172-2"><a href="example.html#cb172-2" aria-hidden="true" tabindex="-1"></a>                           <span class="at">measure =</span> <span class="st">&quot;sparcc&quot;</span>,</span>
<span id="cb172-3"><a href="example.html#cb172-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">measurePar =</span> <span class="fu">list</span>(<span class="at">iter =</span> <span class="dv">20</span>,</span>
<span id="cb172-4"><a href="example.html#cb172-4" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">inner_iter =</span> <span class="dv">10</span>,</span>
<span id="cb172-5"><a href="example.html#cb172-5" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">th =</span> <span class="fl">0.1</span>),</span>
<span id="cb172-6"><a href="example.html#cb172-6" aria-hidden="true" tabindex="-1"></a>                           <span class="at">filtTax =</span> <span class="st">&quot;highestVar&quot;</span>,</span>
<span id="cb172-7"><a href="example.html#cb172-7" aria-hidden="true" tabindex="-1"></a>                           <span class="at">filtTaxPar =</span> <span class="fu">list</span>(<span class="at">highestVar =</span> <span class="dv">50</span>),</span>
<span id="cb172-8"><a href="example.html#cb172-8" aria-hidden="true" tabindex="-1"></a>                           <span class="at">filtSamp =</span> <span class="st">&quot;totalReads&quot;</span>,</span>
<span id="cb172-9"><a href="example.html#cb172-9" aria-hidden="true" tabindex="-1"></a>                           <span class="at">filtSampPar =</span> <span class="fu">list</span>(<span class="at">totalReads =</span> <span class="dv">100</span>),</span>
<span id="cb172-10"><a href="example.html#cb172-10" aria-hidden="true" tabindex="-1"></a>                           <span class="at">verbose =</span> <span class="dv">3</span>,</span>
<span id="cb172-11"><a href="example.html#cb172-11" aria-hidden="true" tabindex="-1"></a>                           <span class="at">seed =</span> <span class="dv">123</span>)</span></code></pre></div>
</div>
<div id="visualizing-the-network-7" class="section level3 hasAnchor" number="8.8.3">
<h3><span class="header-section-number">8.8.3</span> Visualizing the network<a href="example.html#visualizing-the-network-7" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb173"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb173-1"><a href="example.html#cb173-1" aria-hidden="true" tabindex="-1"></a>props_single <span class="ot">&lt;-</span> <span class="fu">netAnalyze</span>(net_single, <span class="at">clustMethod =</span> <span class="st">&quot;cluster_fast_greedy&quot;</span>)</span>
<span id="cb173-2"><a href="example.html#cb173-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb173-3"><a href="example.html#cb173-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(props_single, </span>
<span id="cb173-4"><a href="example.html#cb173-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeColor =</span> <span class="st">&quot;cluster&quot;</span>, </span>
<span id="cb173-5"><a href="example.html#cb173-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeSize =</span> <span class="st">&quot;eigenvector&quot;</span>,</span>
<span id="cb173-6"><a href="example.html#cb173-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">repulsion =</span> <span class="fl">0.8</span>,</span>
<span id="cb173-7"><a href="example.html#cb173-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">rmSingles =</span> <span class="cn">TRUE</span>,</span>
<span id="cb173-8"><a href="example.html#cb173-8" aria-hidden="true" tabindex="-1"></a>     <span class="at">labelScale =</span> <span class="cn">FALSE</span>,</span>
<span id="cb173-9"><a href="example.html#cb173-9" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexLabels =</span> <span class="fl">1.6</span>,</span>
<span id="cb173-10"><a href="example.html#cb173-10" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeSizeSpread =</span> <span class="dv">3</span>,</span>
<span id="cb173-11"><a href="example.html#cb173-11" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexNodes =</span> <span class="dv">2</span>,</span>
<span id="cb173-12"><a href="example.html#cb173-12" aria-hidden="true" tabindex="-1"></a>     <span class="at">title1 =</span> <span class="st">&quot;Network on metabolomics with Pearson correlations&quot;</span>, </span>
<span id="cb173-13"><a href="example.html#cb173-13" aria-hidden="true" tabindex="-1"></a>     <span class="at">showTitle =</span> <span class="cn">TRUE</span>,</span>
<span id="cb173-14"><a href="example.html#cb173-14" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexTitle =</span> <span class="fl">1.5</span>)</span>
<span id="cb173-15"><a href="example.html#cb173-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb173-16"><a href="example.html#cb173-16" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="fl">0.7</span>, <span class="fl">1.1</span>, <span class="at">cex =</span> <span class="fl">1.2</span>, <span class="at">title =</span> <span class="st">&quot;estimated correlation:&quot;</span>,</span>
<span id="cb173-17"><a href="example.html#cb173-17" aria-hidden="true" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;+&quot;</span>,<span class="st">&quot;-&quot;</span>), <span class="at">lty =</span> <span class="dv">1</span>, <span class="at">lwd =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;#009900&quot;</span>,<span class="st">&quot;red&quot;</span>),</span>
<span id="cb173-18"><a href="example.html#cb173-18" aria-hidden="true" tabindex="-1"></a>       <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">horiz =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Statistical-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-113-1.png" width="1152" /></p>
</div>
</div>
<div id="network-analysis-on-wgcna" class="section level2 hasAnchor" number="8.9">
<h2><span class="header-section-number">8.9</span> Network Analysis on WGCNA<a href="example.html#network-analysis-on-wgcna" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Performing Network Analysis step by step through <strong>WGCNA</strong> R package.</p>
<div id="data-curation-3" class="section level3 hasAnchor" number="8.9.1">
<h3><span class="header-section-number">8.9.1</span> Data curation<a href="example.html#data-curation-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><p>Data Matrix</p></li>
<li><p>Row -&gt; metabolites</p></li>
<li><p>Column -&gt; samples</p></li>
</ul>
<div class="sourceCode" id="cb174"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb174-1"><a href="example.html#cb174-1" aria-hidden="true" tabindex="-1"></a>features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(se_impute)</span>
<span id="cb174-2"><a href="example.html#cb174-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(features_tab[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>])</span></code></pre></div>
<pre><code>##            CJY_V0  CJY_V1   DGL_V0   DGL_V1  HXZ_V0  HXZ_V1  LBC_V0  LBC_V1    LHC_V1     LHC_V0
## MEDL00066   10447 4011900   560560   149120  442540  181010   12620  173400   97848.0   25072.83
## MEDL00375   31087   88306   449150   263050  216830   37280  302150  607850   87101.0    7595.70
## MEDL00587  618200  563130  1079300  2047500 1399700 1171700 1120700 1109300 1025300.0  533040.00
## MEDL01844  127250 4172100 19462000 14373000 7830000 5171400   57063 1946700  100670.0  143560.00
## MEDL01922  184698  107560    39149   964600  269570  196770   28797   72528  140508.2  182958.02
## MEDL01994 2590400 3772200  2950700  3328000 2588200 3097700 3373500 3283000 3427700.0 3223700.00</code></pre>
<ul>
<li>Data normalization</li>
</ul>
<div class="sourceCode" id="cb176"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb176-1"><a href="example.html#cb176-1" aria-hidden="true" tabindex="-1"></a><span class="co"># TSS</span></span>
<span id="cb176-2"><a href="example.html#cb176-2" aria-hidden="true" tabindex="-1"></a>features_tab_norm <span class="ot">&lt;-</span> XMAS2<span class="sc">::</span><span class="fu">norm_tss</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(features_tab, <span class="at">taxa_are_rows =</span> T)) <span class="sc">%&gt;%</span> </span>
<span id="cb176-3"><a href="example.html#cb176-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">t</span>()</span>
<span id="cb176-4"><a href="example.html#cb176-4" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(features_tab_norm[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>])</span></code></pre></div>
<pre><code>##           MEDL00066    MEDL00375    MEDL00587    MEDL01844    MEDL01922    MEDL01994    MEDL02002    MEDL02018
## CJY_V0 7.266924e-06 2.162409e-05 0.0004300194 8.851499e-05 1.284758e-04 0.0018018800 9.688354e-04 0.0003373934
## CJY_V1 2.018886e-03 4.443774e-05 0.0002833808 2.099503e-03 5.412682e-05 0.0018982634 6.739182e-05 0.0007045646
## DGL_V0 1.534502e-04 1.229523e-04 0.0002954523 5.327614e-03 1.071682e-05 0.0008077376 8.805537e-04 0.0001872795
## DGL_V1 3.721535e-05 6.564846e-05 0.0005109874 3.587019e-03 2.407318e-04 0.0008305572 6.271356e-04 0.0004141556
## HXZ_V0 1.100101e-04 5.390131e-05 0.0003479485 1.946443e-03 6.701183e-05 0.0006433951 9.244983e-04 0.0008535762
## HXZ_V1 6.310899e-05 1.299764e-05 0.0004085122 1.803004e-03 6.860370e-05 0.0010800105 8.873314e-04 0.0011400479
##           MEDL02174    MEDL02519
## CJY_V0 0.0008276238 5.364045e-04
## CJY_V1 0.0053125408 9.121950e-04
## DGL_V0 0.0013967809 6.574793e-04
## DGL_V1 0.0060542331 1.840278e-04
## HXZ_V0 0.0018885718 1.518648e-04
## HXZ_V1 0.0009249314 5.116427e-06</code></pre>
<div class="sourceCode" id="cb178"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb178-1"><a href="example.html#cb178-1" aria-hidden="true" tabindex="-1"></a><span class="co"># # CSS</span></span>
<span id="cb178-2"><a href="example.html#cb178-2" aria-hidden="true" tabindex="-1"></a><span class="co"># features_tab_norm &lt;- XMAS2::norm_css(phyloseq::otu_table(features_tab, taxa_are_rows = T)) %&gt;% </span></span>
<span id="cb178-3"><a href="example.html#cb178-3" aria-hidden="true" tabindex="-1"></a><span class="co">#   data.frame() %&gt;% t()</span></span>
<span id="cb178-4"><a href="example.html#cb178-4" aria-hidden="true" tabindex="-1"></a><span class="co"># print(features_tab_norm[1:6, 1:10])</span></span></code></pre></div>
</div>
<div id="tuning-soft-thresholds" class="section level3 hasAnchor" number="8.9.2">
<h3><span class="header-section-number">8.9.2</span> Tuning soft thresholds<a href="example.html#tuning-soft-thresholds" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Picking a threshhold value (if correlation is below threshold, remove the edge). WGCNA will try a range of soft thresholds and create a diagnostic plot.</p>
<ul>
<li>Choose a set of soft-thresholding powers</li>
</ul>
<div class="sourceCode" id="cb179"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb179-1"><a href="example.html#cb179-1" aria-hidden="true" tabindex="-1"></a>powers <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>), <span class="fu">seq</span>(<span class="dv">12</span>, <span class="dv">20</span>, <span class="dv">2</span>))</span></code></pre></div>
<ul>
<li><p>Call the network topology analysis function</p></li>
<li><p>Row -&gt; samples</p></li>
<li><p>Column -&gt; metabolites</p></li>
</ul>
<div class="sourceCode" id="cb180"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb180-1"><a href="example.html#cb180-1" aria-hidden="true" tabindex="-1"></a>sft <span class="ot">&lt;-</span> <span class="fu">pickSoftThreshold</span>(</span>
<span id="cb180-2"><a href="example.html#cb180-2" aria-hidden="true" tabindex="-1"></a>  features_tab_norm,</span>
<span id="cb180-3"><a href="example.html#cb180-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">powerVector =</span> powers,</span>
<span id="cb180-4"><a href="example.html#cb180-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">networkType =</span> <span class="st">&quot;unsigned&quot;</span>,</span>
<span id="cb180-5"><a href="example.html#cb180-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="dv">2</span>)</span></code></pre></div>
<pre><code>## pickSoftThreshold: will use block size 540.
##  pickSoftThreshold: calculating connectivity for given powers...
##    ..working on genes 1 through 540 of 540
##    Power SFT.R.sq  slope truncated.R.sq mean.k. median.k. max.k.
## 1      1   0.1930 -1.720          0.956  76.600    76.600 125.00
## 2      2   0.0573 -0.241          0.516  22.200    19.800  44.10
## 3      3   0.8500 -1.090          0.860  10.600     7.420  32.90
## 4      4   0.9330 -1.200          0.923   6.690     3.780  27.80
## 5      5   0.9750 -1.170          0.968   4.900     2.430  24.20
## 6      6   0.9680 -1.190          0.959   3.870     1.750  21.40
## 7      7   0.9400 -1.160          0.923   3.200     1.350  19.20
## 8      8   0.9570 -1.160          0.947   2.720     1.140  17.40
## 9      9   0.9500 -1.180          0.939   2.370     1.040  15.80
## 10    10   0.9480 -1.200          0.936   2.090     0.944  14.60
## 11    12   0.9420 -1.200          0.930   1.700     0.686  12.60
## 12    14   0.8850 -1.260          0.867   1.430     0.501  11.00
## 13    16   0.8260 -1.220          0.846   1.240     0.326   9.87
## 14    18   0.8220 -1.180          0.859   1.100     0.236   8.94
## 15    20   0.8910 -1.140          0.942   0.989     0.191   8.19</code></pre>
<ul>
<li>the optimal power value</li>
</ul>
<div class="sourceCode" id="cb182"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb182-1"><a href="example.html#cb182-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb182-2"><a href="example.html#cb182-2" aria-hidden="true" tabindex="-1"></a>cex1 <span class="ot">=</span> <span class="fl">1.2</span></span>
<span id="cb182-3"><a href="example.html#cb182-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb182-4"><a href="example.html#cb182-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(sft<span class="sc">$</span>fitIndices[, <span class="dv">1</span>],</span>
<span id="cb182-5"><a href="example.html#cb182-5" aria-hidden="true" tabindex="-1"></a>     <span class="sc">-</span><span class="fu">sign</span>(sft<span class="sc">$</span>fitIndices[, <span class="dv">3</span>]) <span class="sc">*</span> sft<span class="sc">$</span>fitIndices[, <span class="dv">2</span>],</span>
<span id="cb182-6"><a href="example.html#cb182-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">&quot;Soft Threshold (power)&quot;</span>,</span>
<span id="cb182-7"><a href="example.html#cb182-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">&quot;Scale Free Topology Model Fit, signed R^2&quot;</span>,</span>
<span id="cb182-8"><a href="example.html#cb182-8" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">&quot;Scale independence&quot;</span>)</span>
<span id="cb182-9"><a href="example.html#cb182-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb182-10"><a href="example.html#cb182-10" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(sft<span class="sc">$</span>fitIndices[, <span class="dv">1</span>],</span>
<span id="cb182-11"><a href="example.html#cb182-11" aria-hidden="true" tabindex="-1"></a>     <span class="sc">-</span><span class="fu">sign</span>(sft<span class="sc">$</span>fitIndices[, <span class="dv">3</span>]) <span class="sc">*</span> sft<span class="sc">$</span>fitIndices[, <span class="dv">2</span>],</span>
<span id="cb182-12"><a href="example.html#cb182-12" aria-hidden="true" tabindex="-1"></a>     <span class="at">labels =</span> powers, <span class="at">cex =</span> cex1, <span class="at">col =</span> <span class="st">&quot;red&quot;</span></span>
<span id="cb182-13"><a href="example.html#cb182-13" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb182-14"><a href="example.html#cb182-14" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="fl">0.90</span>, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>)</span>
<span id="cb182-15"><a href="example.html#cb182-15" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(sft<span class="sc">$</span>fitIndices[, <span class="dv">1</span>],</span>
<span id="cb182-16"><a href="example.html#cb182-16" aria-hidden="true" tabindex="-1"></a>     sft<span class="sc">$</span>fitIndices[, <span class="dv">5</span>],</span>
<span id="cb182-17"><a href="example.html#cb182-17" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">&quot;Soft Threshold (power)&quot;</span>,</span>
<span id="cb182-18"><a href="example.html#cb182-18" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">&quot;Mean Connectivity&quot;</span>,</span>
<span id="cb182-19"><a href="example.html#cb182-19" aria-hidden="true" tabindex="-1"></a>     <span class="at">type =</span> <span class="st">&quot;n&quot;</span>,</span>
<span id="cb182-20"><a href="example.html#cb182-20" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">&quot;Mean connectivity&quot;</span>)</span>
<span id="cb182-21"><a href="example.html#cb182-21" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb182-22"><a href="example.html#cb182-22" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(sft<span class="sc">$</span>fitIndices[, <span class="dv">1</span>],</span>
<span id="cb182-23"><a href="example.html#cb182-23" aria-hidden="true" tabindex="-1"></a>     sft<span class="sc">$</span>fitIndices[, <span class="dv">5</span>],</span>
<span id="cb182-24"><a href="example.html#cb182-24" aria-hidden="true" tabindex="-1"></a>     <span class="at">labels =</span> powers,</span>
<span id="cb182-25"><a href="example.html#cb182-25" aria-hidden="true" tabindex="-1"></a>     <span class="at">cex =</span> cex1, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="Statistical-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-118-1.png" width="768" /></p>
<p><strong>Notice</strong>: We’ pick <strong>5</strong> but feel free to experiment with other powers to see how it affects your results.</p>
</div>
<div id="create-the-network-using-the-blockwisemodules" class="section level3 hasAnchor" number="8.9.3">
<h3><span class="header-section-number">8.9.3</span> Create the network using the <code>blockwiseModules</code><a href="example.html#create-the-network-using-the-blockwisemodules" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>building network</li>
</ul>
<div class="sourceCode" id="cb183"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb183-1"><a href="example.html#cb183-1" aria-hidden="true" tabindex="-1"></a>picked_power <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb183-2"><a href="example.html#cb183-2" aria-hidden="true" tabindex="-1"></a><span class="co"># temp_cor &lt;- cor       </span></span>
<span id="cb183-3"><a href="example.html#cb183-3" aria-hidden="true" tabindex="-1"></a><span class="co"># cor &lt;- WGCNA::cor </span></span>
<span id="cb183-4"><a href="example.html#cb183-4" aria-hidden="true" tabindex="-1"></a>netwk <span class="ot">&lt;-</span> <span class="fu">blockwiseModules</span>(features_tab_norm,</span>
<span id="cb183-5"><a href="example.html#cb183-5" aria-hidden="true" tabindex="-1"></a>                          </span>
<span id="cb183-6"><a href="example.html#cb183-6" aria-hidden="true" tabindex="-1"></a>                          <span class="co"># == Adjacency Function ==</span></span>
<span id="cb183-7"><a href="example.html#cb183-7" aria-hidden="true" tabindex="-1"></a>                          <span class="at">power =</span> picked_power,                <span class="co"># &lt;= power here</span></span>
<span id="cb183-8"><a href="example.html#cb183-8" aria-hidden="true" tabindex="-1"></a>                          <span class="at">networkType =</span> <span class="st">&quot;signed&quot;</span>,</span>
<span id="cb183-9"><a href="example.html#cb183-9" aria-hidden="true" tabindex="-1"></a>                          </span>
<span id="cb183-10"><a href="example.html#cb183-10" aria-hidden="true" tabindex="-1"></a>                          <span class="co"># == Network construction arguments: correlation options</span></span>
<span id="cb183-11"><a href="example.html#cb183-11" aria-hidden="true" tabindex="-1"></a>                          <span class="at">corType =</span> <span class="st">&quot;bicor&quot;</span>,</span>
<span id="cb183-12"><a href="example.html#cb183-12" aria-hidden="true" tabindex="-1"></a>                          <span class="at">maxPOutliers =</span> <span class="fl">0.05</span>,</span>
<span id="cb183-13"><a href="example.html#cb183-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-14"><a href="example.html#cb183-14" aria-hidden="true" tabindex="-1"></a>                          <span class="co"># == Tree and Block Options ==</span></span>
<span id="cb183-15"><a href="example.html#cb183-15" aria-hidden="true" tabindex="-1"></a>                          <span class="at">deepSplit =</span> <span class="dv">2</span>,</span>
<span id="cb183-16"><a href="example.html#cb183-16" aria-hidden="true" tabindex="-1"></a>                          <span class="at">pamRespectsDendro =</span> F,</span>
<span id="cb183-17"><a href="example.html#cb183-17" aria-hidden="true" tabindex="-1"></a>                          <span class="at">minModuleSize =</span> <span class="dv">20</span>,</span>
<span id="cb183-18"><a href="example.html#cb183-18" aria-hidden="true" tabindex="-1"></a>                          <span class="at">maxBlockSize =</span> <span class="dv">4000</span>,</span>
<span id="cb183-19"><a href="example.html#cb183-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-20"><a href="example.html#cb183-20" aria-hidden="true" tabindex="-1"></a>                          <span class="co"># == Module Adjustments ==</span></span>
<span id="cb183-21"><a href="example.html#cb183-21" aria-hidden="true" tabindex="-1"></a>                          <span class="at">reassignThreshold =</span> <span class="dv">0</span>,</span>
<span id="cb183-22"><a href="example.html#cb183-22" aria-hidden="true" tabindex="-1"></a>                          <span class="at">mergeCutHeight =</span> <span class="fl">0.25</span>,</span>
<span id="cb183-23"><a href="example.html#cb183-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-24"><a href="example.html#cb183-24" aria-hidden="true" tabindex="-1"></a>                          <span class="co"># == TOM == Archive the run results in TOM file (saves time)</span></span>
<span id="cb183-25"><a href="example.html#cb183-25" aria-hidden="true" tabindex="-1"></a>                          <span class="at">saveTOMs =</span> T,</span>
<span id="cb183-26"><a href="example.html#cb183-26" aria-hidden="true" tabindex="-1"></a>                          <span class="at">saveTOMFileBase =</span> <span class="fu">paste0</span>(<span class="st">&quot;./dataset/&quot;</span>, <span class="st">&quot;GvHD&quot;</span>),</span>
<span id="cb183-27"><a href="example.html#cb183-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb183-28"><a href="example.html#cb183-28" aria-hidden="true" tabindex="-1"></a>                          <span class="co"># == Output Options</span></span>
<span id="cb183-29"><a href="example.html#cb183-29" aria-hidden="true" tabindex="-1"></a>                          <span class="at">numericLabels =</span> T,</span>
<span id="cb183-30"><a href="example.html#cb183-30" aria-hidden="true" tabindex="-1"></a>                          <span class="at">verbose =</span> <span class="dv">3</span>,</span>
<span id="cb183-31"><a href="example.html#cb183-31" aria-hidden="true" tabindex="-1"></a>                          <span class="at">randomSeed =</span> <span class="dv">123</span>)</span></code></pre></div>
<pre><code>##  Calculating module eigengenes block-wise from all genes
##    Flagging genes and samples with too many missing values...
##     ..step 1
##  ..Working on block 1 .
##     TOM calculation: adjacency..
##     ..will not use multithreading.
##      Fraction of slow calculations: 0.000000
##     ..connectivity..
##     ..matrix multiplication (system BLAS)..
##     ..normalization..
##     ..done.
##    ..saving TOM for block 1 into file ./dataset/GvHD-block.1.RData
##  ....clustering..
##  ....detecting modules..
##  ....calculating module eigengenes..
##  ....checking kME in modules..
##      ..removing 22 genes from module 1 because their KME is too low.
##      ..removing 19 genes from module 2 because their KME is too low.
##      ..removing 36 genes from module 3 because their KME is too low.
##      ..removing 15 genes from module 4 because their KME is too low.
##      ..removing 2 genes from module 5 because their KME is too low.
##      ..removing 2 genes from module 6 because their KME is too low.
##  ..merging modules that are too close..
##      mergeCloseModules: Merging modules whose distance is less than 0.25
##        Calculating new MEs...</code></pre>
<ul>
<li>Modules’ number</li>
</ul>
<div class="sourceCode" id="cb185"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb185-1"><a href="example.html#cb185-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(netwk<span class="sc">$</span>colors)</span></code></pre></div>
<pre><code>## 
##   0   1   2   3   4   5   6   7 
##  96 128  89  68  55  44  33  27</code></pre>
<ul>
<li>hubs</li>
</ul>
<div class="sourceCode" id="cb187"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb187-1"><a href="example.html#cb187-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(netwk<span class="sc">$</span>MEs) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(features_tab_norm)</span>
<span id="cb187-2"><a href="example.html#cb187-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(netwk<span class="sc">$</span>colors) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(features_tab_norm)</span>
<span id="cb187-3"><a href="example.html#cb187-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(netwk<span class="sc">$</span>unmergedColors) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(features_tab_norm)</span>
<span id="cb187-4"><a href="example.html#cb187-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-5"><a href="example.html#cb187-5" aria-hidden="true" tabindex="-1"></a>hubs <span class="ot">&lt;-</span> <span class="fu">chooseTopHubInEachModule</span>(features_tab_norm, netwk<span class="sc">$</span>colors)</span>
<span id="cb187-6"><a href="example.html#cb187-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb187-7"><a href="example.html#cb187-7" aria-hidden="true" tabindex="-1"></a>hubs</span></code></pre></div>
<pre><code>##           0           1           2           3           4           5           6           7 
##  &quot;MEDN1108&quot;  &quot;MEDP1255&quot;  &quot;MEDP1033&quot;  &quot;MEDP0434&quot; &quot;MW0107573&quot;  &quot;MEDP0065&quot;  &quot;MEDP1386&quot;  &quot;MEDP0389&quot;</code></pre>
<ul>
<li>The number of features per module</li>
</ul>
<div class="sourceCode" id="cb189"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb189-1"><a href="example.html#cb189-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(netwk<span class="sc">$</span>colors) <span class="sc">%&gt;%</span> </span>
<span id="cb189-2"><a href="example.html#cb189-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb189-3"><a href="example.html#cb189-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Module =</span> Var1, <span class="at">Number =</span> Freq) <span class="sc">%&gt;%</span> </span>
<span id="cb189-4"><a href="example.html#cb189-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Module_color =</span> <span class="fu">labels2colors</span>(<span class="fu">as.numeric</span>(<span class="fu">as.character</span>(Module)))) <span class="sc">%&gt;%</span> </span>
<span id="cb189-5"><a href="example.html#cb189-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Module, <span class="at">y =</span> Number, <span class="at">fill =</span> Module)) <span class="sc">+</span></span>
<span id="cb189-6"><a href="example.html#cb189-6" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> Number), <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.18</span>, <span class="at">size =</span> <span class="fl">3.5</span>) <span class="sc">+</span></span>
<span id="cb189-7"><a href="example.html#cb189-7" aria-hidden="true" tabindex="-1"></a>      <span class="fu">geom_col</span>(<span class="at">color =</span> <span class="st">&quot;#000000&quot;</span>) <span class="sc">+</span></span>
<span id="cb189-8"><a href="example.html#cb189-8" aria-hidden="true" tabindex="-1"></a>      <span class="fu">ggtitle</span>(<span class="st">&quot;Number of features per module&quot;</span>) <span class="sc">+</span></span>
<span id="cb189-9"><a href="example.html#cb189-9" aria-hidden="true" tabindex="-1"></a>      <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb189-10"><a href="example.html#cb189-10" aria-hidden="true" tabindex="-1"></a>      <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb189-11"><a href="example.html#cb189-11" aria-hidden="true" tabindex="-1"></a>      <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb189-12"><a href="example.html#cb189-12" aria-hidden="true" tabindex="-1"></a>      <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="st">&quot;pt&quot;</span>),</span>
<span id="cb189-13"><a href="example.html#cb189-13" aria-hidden="true" tabindex="-1"></a>            <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>),</span>
<span id="cb189-14"><a href="example.html#cb189-14" aria-hidden="true" tabindex="-1"></a>            <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<p><img src="Statistical-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-122-1.png" width="768" /></p>
</div>
<div id="plot-modules" class="section level3 hasAnchor" number="8.9.4">
<h3><span class="header-section-number">8.9.4</span> plot modules<a href="example.html#plot-modules" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb190"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb190-1"><a href="example.html#cb190-1" aria-hidden="true" tabindex="-1"></a>mergedColors <span class="ot">&lt;-</span> <span class="fu">labels2colors</span>(netwk<span class="sc">$</span>colors)</span>
<span id="cb190-2"><a href="example.html#cb190-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb190-3"><a href="example.html#cb190-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(</span>
<span id="cb190-4"><a href="example.html#cb190-4" aria-hidden="true" tabindex="-1"></a>  netwk<span class="sc">$</span>dendrograms[[<span class="dv">1</span>]],</span>
<span id="cb190-5"><a href="example.html#cb190-5" aria-hidden="true" tabindex="-1"></a>  mergedColors[netwk<span class="sc">$</span>blockGenes[[<span class="dv">1</span>]]],</span>
<span id="cb190-6"><a href="example.html#cb190-6" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;Module colors&quot;</span>,</span>
<span id="cb190-7"><a href="example.html#cb190-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">dendroLabels =</span> <span class="cn">FALSE</span>,</span>
<span id="cb190-8"><a href="example.html#cb190-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">hang =</span> <span class="fl">0.03</span>,</span>
<span id="cb190-9"><a href="example.html#cb190-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">addGuide =</span> <span class="cn">TRUE</span>,</span>
<span id="cb190-10"><a href="example.html#cb190-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">guideHang =</span> <span class="fl">0.05</span> )</span></code></pre></div>
<p><img src="Statistical-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-123-1.png" width="768" /></p>
</div>
<div id="relationships-among-modules" class="section level3 hasAnchor" number="8.9.5">
<h3><span class="header-section-number">8.9.5</span> Relationships among modules<a href="example.html#relationships-among-modules" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb191"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb191-1"><a href="example.html#cb191-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotEigengeneNetworks</span>(netwk<span class="sc">$</span>MEs, </span>
<span id="cb191-2"><a href="example.html#cb191-2" aria-hidden="true" tabindex="-1"></a>                      <span class="st">&quot;Eigengene adjacency heatmap&quot;</span>,</span>
<span id="cb191-3"><a href="example.html#cb191-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">marDendro =</span> <span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">2</span>, <span class="dv">4</span>),</span>
<span id="cb191-4"><a href="example.html#cb191-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">marHeatmap =</span> <span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">2</span>, <span class="dv">2</span>), </span>
<span id="cb191-5"><a href="example.html#cb191-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">plotDendrograms =</span> T,</span>
<span id="cb191-6"><a href="example.html#cb191-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">xLabelsAngle =</span> <span class="dv">90</span>)</span></code></pre></div>
<p><img src="Statistical-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-124-1.png" width="768" /></p>
</div>
<div id="module-eigengene-correlation" class="section level3 hasAnchor" number="8.9.6">
<h3><span class="header-section-number">8.9.6</span> Module (Eigengene) correlation<a href="example.html#module-eigengene-correlation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb192"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb192-1"><a href="example.html#cb192-1" aria-hidden="true" tabindex="-1"></a>MEs <span class="ot">&lt;-</span> netwk<span class="sc">$</span>MEs</span>
<span id="cb192-2"><a href="example.html#cb192-2" aria-hidden="true" tabindex="-1"></a>MEs_R <span class="ot">&lt;-</span> <span class="fu">bicor</span>(MEs, MEs, <span class="at">maxPOutliers =</span> <span class="fl">0.05</span>)</span>
<span id="cb192-3"><a href="example.html#cb192-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-4"><a href="example.html#cb192-4" aria-hidden="true" tabindex="-1"></a>idx.r <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">rownames</span>(MEs_R) <span class="sc">==</span> <span class="st">&quot;ME0&quot;</span>)</span>
<span id="cb192-5"><a href="example.html#cb192-5" aria-hidden="true" tabindex="-1"></a>idx.c <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">colnames</span>(MEs_R) <span class="sc">==</span> <span class="st">&quot;ME0&quot;</span>)</span>
<span id="cb192-6"><a href="example.html#cb192-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-7"><a href="example.html#cb192-7" aria-hidden="true" tabindex="-1"></a>MEs_R_noME0 <span class="ot">&lt;-</span> MEs_R[<span class="sc">-</span>idx.r, <span class="sc">-</span>idx.c]</span>
<span id="cb192-8"><a href="example.html#cb192-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-9"><a href="example.html#cb192-9" aria-hidden="true" tabindex="-1"></a>MEs_R_density <span class="ot">&lt;-</span> MEs_R[<span class="fu">upper.tri</span>(MEs_R_noME0)] <span class="sc">%&gt;%</span> </span>
<span id="cb192-10"><a href="example.html#cb192-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb192-11"><a href="example.html#cb192-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;correlation&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb192-12"><a href="example.html#cb192-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>correlation)) <span class="sc">+</span> </span>
<span id="cb192-13"><a href="example.html#cb192-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>() <span class="sc">+</span> </span>
<span id="cb192-14"><a href="example.html#cb192-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(<span class="st">&quot;ME correlation density</span><span class="sc">\n</span><span class="st"> without &quot;</span>, <span class="st">&quot;ME0&quot;</span>))</span>
<span id="cb192-15"><a href="example.html#cb192-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-16"><a href="example.html#cb192-16" aria-hidden="true" tabindex="-1"></a>MEs_R_Corr <span class="ot">&lt;-</span> pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>(MEs_R, <span class="at">color =</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">&quot;Blue&quot;</span>, <span class="st">&quot;White&quot;</span>, <span class="st">&quot;Red&quot;</span>))(<span class="dv">100</span>),</span>
<span id="cb192-17"><a href="example.html#cb192-17" aria-hidden="true" tabindex="-1"></a>                   <span class="at">silent =</span> T, </span>
<span id="cb192-18"><a href="example.html#cb192-18" aria-hidden="true" tabindex="-1"></a>                   <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>,<span class="at">length.out =</span> <span class="dv">101</span>),</span>
<span id="cb192-19"><a href="example.html#cb192-19" aria-hidden="true" tabindex="-1"></a>                   <span class="at">treeheight_row =</span> <span class="dv">5</span>, </span>
<span id="cb192-20"><a href="example.html#cb192-20" aria-hidden="true" tabindex="-1"></a>                   <span class="at">treeheight_col =</span> <span class="dv">5</span>,</span>
<span id="cb192-21"><a href="example.html#cb192-21" aria-hidden="true" tabindex="-1"></a>                   <span class="at">main =</span> <span class="fu">paste0</span>(<span class="st">&quot;ME correlation heatmap&quot;</span>),</span>
<span id="cb192-22"><a href="example.html#cb192-22" aria-hidden="true" tabindex="-1"></a>                   <span class="at">labels_row =</span> <span class="fu">rownames</span>(MEs_R),</span>
<span id="cb192-23"><a href="example.html#cb192-23" aria-hidden="true" tabindex="-1"></a>                   <span class="at">labels_col =</span>  <span class="fu">colnames</span>(MEs_R)) </span>
<span id="cb192-24"><a href="example.html#cb192-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb192-25"><a href="example.html#cb192-25" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(MEs_R_density, MEs_R_Corr<span class="sc">$</span>gtable, </span>
<span id="cb192-26"><a href="example.html#cb192-26" aria-hidden="true" tabindex="-1"></a>                   <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>), </span>
<span id="cb192-27"><a href="example.html#cb192-27" aria-hidden="true" tabindex="-1"></a>                   <span class="at">label_size =</span> <span class="dv">15</span>, </span>
<span id="cb192-28"><a href="example.html#cb192-28" aria-hidden="true" tabindex="-1"></a>                   <span class="at">rel_widths =</span> <span class="fu">c</span>(<span class="fl">0.6</span>, <span class="dv">1</span>),</span>
<span id="cb192-29"><a href="example.html#cb192-29" aria-hidden="true" tabindex="-1"></a>                   <span class="at">align =</span> <span class="st">&quot;h&quot;</span>) </span></code></pre></div>
<p><img src="Statistical-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-125-1.png" width="576" /></p>
</div>
<div id="relate-module-cluster-assignments-to-groups" class="section level3 hasAnchor" number="8.9.7">
<h3><span class="header-section-number">8.9.7</span> Relate Module (cluster) Assignments to Groups<a href="example.html#relate-module-cluster-assignments-to-groups" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb193"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb193-1"><a href="example.html#cb193-1" aria-hidden="true" tabindex="-1"></a>module_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb193-2"><a href="example.html#cb193-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">featureID =</span> <span class="fu">names</span>(netwk<span class="sc">$</span>colors),</span>
<span id="cb193-3"><a href="example.html#cb193-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">colors =</span> <span class="fu">labels2colors</span>(netwk<span class="sc">$</span>colors)</span>
<span id="cb193-4"><a href="example.html#cb193-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb193-5"><a href="example.html#cb193-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-6"><a href="example.html#cb193-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Get Module Eigengenes per cluster</span></span>
<span id="cb193-7"><a href="example.html#cb193-7" aria-hidden="true" tabindex="-1"></a>MEs0 <span class="ot">&lt;-</span> <span class="fu">moduleEigengenes</span>(features_tab_norm, mergedColors)<span class="sc">$</span>eigengenes</span>
<span id="cb193-8"><a href="example.html#cb193-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-9"><a href="example.html#cb193-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Reorder modules so similar modules are next to each other</span></span>
<span id="cb193-10"><a href="example.html#cb193-10" aria-hidden="true" tabindex="-1"></a>MEs0 <span class="ot">&lt;-</span> <span class="fu">orderMEs</span>(MEs0)</span>
<span id="cb193-11"><a href="example.html#cb193-11" aria-hidden="true" tabindex="-1"></a>module_order <span class="ot">&lt;-</span> <span class="fu">names</span>(MEs0) <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&quot;ME&quot;</span>,<span class="st">&quot;&quot;</span>, .)</span>
<span id="cb193-12"><a href="example.html#cb193-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-13"><a href="example.html#cb193-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Add group names</span></span>
<span id="cb193-14"><a href="example.html#cb193-14" aria-hidden="true" tabindex="-1"></a>MEs0<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">paste0</span>(se_impute<span class="sc">$</span>group, <span class="fu">rownames</span>(<span class="fu">colData</span>(se_impute)))  <span class="co"># row.names(MEs0) == rownames(colData(se_impute))</span></span>
<span id="cb193-15"><a href="example.html#cb193-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-16"><a href="example.html#cb193-16" aria-hidden="true" tabindex="-1"></a><span class="co"># tidy &amp; plot data</span></span>
<span id="cb193-17"><a href="example.html#cb193-17" aria-hidden="true" tabindex="-1"></a>mME <span class="ot">&lt;-</span> MEs0 <span class="sc">%&gt;%</span></span>
<span id="cb193-18"><a href="example.html#cb193-18" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(<span class="sc">-</span>group) <span class="sc">%&gt;%</span></span>
<span id="cb193-19"><a href="example.html#cb193-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb193-20"><a href="example.html#cb193-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="fu">gsub</span>(<span class="st">&quot;ME&quot;</span>, <span class="st">&quot;&quot;</span>, name),</span>
<span id="cb193-21"><a href="example.html#cb193-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="fu">factor</span>(name, <span class="at">levels =</span> module_order)</span>
<span id="cb193-22"><a href="example.html#cb193-22" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb193-23"><a href="example.html#cb193-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb193-24"><a href="example.html#cb193-24" aria-hidden="true" tabindex="-1"></a>mME <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x=</span>group, <span class="at">y=</span>name, <span class="at">fill=</span>value)) <span class="sc">+</span></span>
<span id="cb193-25"><a href="example.html#cb193-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb193-26"><a href="example.html#cb193-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Module-samples Relationships&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Modules&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;corr&quot;</span>) <span class="sc">+</span>  </span>
<span id="cb193-27"><a href="example.html#cb193-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient2</span>(</span>
<span id="cb193-28"><a href="example.html#cb193-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">low =</span> <span class="st">&quot;blue&quot;</span>,</span>
<span id="cb193-29"><a href="example.html#cb193-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">high =</span> <span class="st">&quot;red&quot;</span>,</span>
<span id="cb193-30"><a href="example.html#cb193-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">mid =</span> <span class="st">&quot;white&quot;</span>,</span>
<span id="cb193-31"><a href="example.html#cb193-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">midpoint =</span> <span class="dv">0</span>,</span>
<span id="cb193-32"><a href="example.html#cb193-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">limit =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb193-33"><a href="example.html#cb193-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb193-34"><a href="example.html#cb193-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>))</span></code></pre></div>
<p><img src="Statistical-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-126-1.png" width="768" /></p>
<p><strong>Result</strong>:</p>
<ol style="list-style-type: decimal">
<li>the <em>black</em> modules seems negatively associated (red shading) with the PR groups.</li>
</ol>
</div>
<div id="generate-and-export-networks" class="section level3 hasAnchor" number="8.9.8">
<h3><span class="header-section-number">8.9.8</span> Generate and Export Networks<a href="example.html#generate-and-export-networks" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb194"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb194-1"><a href="example.html#cb194-1" aria-hidden="true" tabindex="-1"></a><span class="co"># modules_of_interest &lt;- c(&quot;green&quot;, &quot;brown&quot;, &quot;black&quot;)</span></span>
<span id="cb194-2"><a href="example.html#cb194-2" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb194-3"><a href="example.html#cb194-3" aria-hidden="true" tabindex="-1"></a><span class="co"># genes_of_interest &lt;- module_df %&gt;%</span></span>
<span id="cb194-4"><a href="example.html#cb194-4" aria-hidden="true" tabindex="-1"></a><span class="co">#   subset(colors %in% modules_of_interest)</span></span>
<span id="cb194-5"><a href="example.html#cb194-5" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb194-6"><a href="example.html#cb194-6" aria-hidden="true" tabindex="-1"></a><span class="co"># expr_of_interest &lt;- features_tab_norm[, genes_of_interest$featureID]</span></span>
<span id="cb194-7"><a href="example.html#cb194-7" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb194-8"><a href="example.html#cb194-8" aria-hidden="true" tabindex="-1"></a><span class="co"># # Only recalculate TOM for modules of interest </span></span>
<span id="cb194-9"><a href="example.html#cb194-9" aria-hidden="true" tabindex="-1"></a><span class="co"># TOM &lt;- TOMsimilarityFromExpr(expr_of_interest,</span></span>
<span id="cb194-10"><a href="example.html#cb194-10" aria-hidden="true" tabindex="-1"></a><span class="co">#                              power = picked_power)</span></span>
<span id="cb194-11"><a href="example.html#cb194-11" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb194-12"><a href="example.html#cb194-12" aria-hidden="true" tabindex="-1"></a><span class="co"># # Add feature id to row and columns</span></span>
<span id="cb194-13"><a href="example.html#cb194-13" aria-hidden="true" tabindex="-1"></a><span class="co"># rownames(TOM) &lt;- colnames(expr_of_interest)</span></span>
<span id="cb194-14"><a href="example.html#cb194-14" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(TOM) &lt;- colnames(expr_of_interest)</span></span>
<span id="cb194-15"><a href="example.html#cb194-15" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb194-16"><a href="example.html#cb194-16" aria-hidden="true" tabindex="-1"></a><span class="co"># edge_list &lt;- data.frame(TOM) %&gt;%</span></span>
<span id="cb194-17"><a href="example.html#cb194-17" aria-hidden="true" tabindex="-1"></a><span class="co">#   tibble::rownames_to_column(&quot;featureID&quot;) %&gt;%</span></span>
<span id="cb194-18"><a href="example.html#cb194-18" aria-hidden="true" tabindex="-1"></a><span class="co">#   tidyr::pivot_longer(-featureID) %&gt;%</span></span>
<span id="cb194-19"><a href="example.html#cb194-19" aria-hidden="true" tabindex="-1"></a><span class="co">#   dplyr::rename(featureID2 = name, correlation = value) %&gt;%</span></span>
<span id="cb194-20"><a href="example.html#cb194-20" aria-hidden="true" tabindex="-1"></a><span class="co">#   unique() %&gt;%</span></span>
<span id="cb194-21"><a href="example.html#cb194-21" aria-hidden="true" tabindex="-1"></a><span class="co">#   subset(!(featureID == featureID2)) %&gt;%</span></span>
<span id="cb194-22"><a href="example.html#cb194-22" aria-hidden="true" tabindex="-1"></a><span class="co">#   dplyr::mutate(</span></span>
<span id="cb194-23"><a href="example.html#cb194-23" aria-hidden="true" tabindex="-1"></a><span class="co">#     module1 = module_df[featureID, ]$colors,</span></span>
<span id="cb194-24"><a href="example.html#cb194-24" aria-hidden="true" tabindex="-1"></a><span class="co">#     module2 = module_df[featureID2, ]$colors)</span></span>
<span id="cb194-25"><a href="example.html#cb194-25" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb194-26"><a href="example.html#cb194-26" aria-hidden="true" tabindex="-1"></a><span class="co"># head(edge_list)</span></span></code></pre></div>
</div>
<div id="network-visualization" class="section level3 hasAnchor" number="8.9.9">
<h3><span class="header-section-number">8.9.9</span> Network visualization<a href="example.html#network-visualization" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb195"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb195-1"><a href="example.html#cb195-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(igraph)</span>
<span id="cb195-2"><a href="example.html#cb195-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb195-3"><a href="example.html#cb195-3" aria-hidden="true" tabindex="-1"></a>strength_adjust <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb195-4"><a href="example.html#cb195-4" aria-hidden="true" tabindex="-1"></a>TOM <span class="ot">&lt;-</span> <span class="fu">TOMsimilarityFromExpr</span>(features_tab_norm,</span>
<span id="cb195-5"><a href="example.html#cb195-5" aria-hidden="true" tabindex="-1"></a>                             <span class="at">power =</span> picked_power)</span></code></pre></div>
<pre><code>## TOM calculation: adjacency..
## ..will not use multithreading.
##  Fraction of slow calculations: 0.000000
## ..connectivity..
## ..matrix multiplication (system BLAS)..
## ..normalization..
## ..done.</code></pre>
<div class="sourceCode" id="cb197"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb197-1"><a href="example.html#cb197-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add feature id to row and columns</span></span>
<span id="cb197-2"><a href="example.html#cb197-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(TOM) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(features_tab_norm)</span>
<span id="cb197-3"><a href="example.html#cb197-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(TOM) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(features_tab_norm)</span>
<span id="cb197-4"><a href="example.html#cb197-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb197-5"><a href="example.html#cb197-5" aria-hidden="true" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">graph.adjacency</span>(TOM, <span class="at">mode=</span><span class="st">&quot;undirected&quot;</span>, <span class="at">weighted=</span> <span class="cn">TRUE</span>)</span>
<span id="cb197-6"><a href="example.html#cb197-6" aria-hidden="true" tabindex="-1"></a><span class="fu">delete.edges</span>(g, <span class="fu">which</span>(<span class="fu">E</span>(g)<span class="sc">$</span>weight <span class="sc">&lt;</span><span class="dv">1</span>))</span></code></pre></div>
<pre><code>## IGRAPH d1521b3 UNW- 540 540 -- 
## + attr: name (v/c), weight (e/n)
## + edges from d1521b3 (vertex names):
##  [1] MEDL00066--MEDL00066 MEDL00375--MEDL00375 MEDL00587--MEDL00587 MEDL01844--MEDL01844 MEDL01922--MEDL01922
##  [6] MEDL01994--MEDL01994 MEDL02002--MEDL02002 MEDL02018--MEDL02018 MEDL02174--MEDL02174 MEDL02519--MEDL02519
## [11] MEDL02630--MEDL02630 MEDL02681--MEDL02681 MEDN0005 --MEDN0005  MEDN0006 --MEDN0006  MEDN0007 --MEDN0007 
## [16] MEDN0010 --MEDN0010  MEDN0015 --MEDN0015  MEDN0018 --MEDN0018  MEDN0025 --MEDN0025  MEDN0032 --MEDN0032 
## [21] MEDN0036 --MEDN0036  MEDN0042 --MEDN0042  MEDN0046 --MEDN0046  MEDN0047 --MEDN0047  MEDN0049 --MEDN0049 
## [26] MEDN0052 --MEDN0052  MEDN0053 --MEDN0053  MEDN0056 --MEDN0056  MEDN0059 --MEDN0059  MEDN0075 --MEDN0075 
## [31] MEDN0089 --MEDN0089  MEDN0098 --MEDN0098  MEDN0100 --MEDN0100  MEDN0110 --MEDN0110  MEDN0113 --MEDN0113 
## [36] MEDN0120 --MEDN0120  MEDN0140 --MEDN0140  MEDN0141 --MEDN0141  MEDN0144 --MEDN0144  MEDN0147 --MEDN0147 
## + ... omitted several edges</code></pre>
<div class="sourceCode" id="cb199"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb199-1"><a href="example.html#cb199-1" aria-hidden="true" tabindex="-1"></a><span class="fu">E</span>(g)<span class="sc">$</span>width <span class="ot">&lt;-</span> <span class="fu">E</span>(g)<span class="sc">$</span>weight<span class="sc">*</span>strength_adjust <span class="sc">+</span> <span class="fu">min</span>(<span class="fu">E</span>(g)<span class="sc">$</span>weight)</span>
<span id="cb199-2"><a href="example.html#cb199-2" aria-hidden="true" tabindex="-1"></a><span class="fu">E</span>(g)<span class="sc">$</span>color <span class="ot">&lt;-</span> <span class="st">&quot;red&quot;</span></span>
<span id="cb199-3"><a href="example.html#cb199-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb199-4"><a href="example.html#cb199-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(g)</span></code></pre></div>
<p><img src="Statistical-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-128-1.png" width="768" /></p>
</div>
</div>
<div id="systematic-information-6" class="section level2 hasAnchor" number="8.10">
<h2><span class="header-section-number">8.10</span> Systematic Information<a href="example.html#systematic-information-6" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb200"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb200-1"><a href="example.html#cb200-1" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ────────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 4.1.2 (2021-11-01)
##  os       macOS Monterey 12.2.1
##  system   x86_64, darwin17.0
##  ui       RStudio
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       Asia/Shanghai
##  date     2022-08-03
##  rstudio  2022.07.1+554 Spotted Wakerobin (desktop)
##  pandoc   2.18 @ /Applications/RStudio.app/Contents/MacOS/quarto/bin/tools/ (via rmarkdown)
## 
## ─ Packages ────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package              * version    date (UTC) lib source
##  abind                  1.4-5      2016-07-21 [1] CRAN (R 4.1.0)
##  ade4                   1.7-18     2021-09-16 [1] CRAN (R 4.1.0)
##  annotate               1.72.0     2021-10-26 [1] Bioconductor
##  AnnotationDbi          1.56.2     2021-11-09 [1] Bioconductor
##  ape                    5.6-2      2022-03-02 [1] CRAN (R 4.1.2)
##  assertthat             0.2.1      2019-03-21 [1] CRAN (R 4.1.0)
##  backports              1.4.1      2021-12-13 [1] CRAN (R 4.1.0)
##  base64enc              0.1-3      2015-07-28 [1] CRAN (R 4.1.0)
##  Biobase              * 2.54.0     2021-10-26 [1] Bioconductor
##  BiocGenerics         * 0.40.0     2021-10-26 [1] Bioconductor
##  BiocParallel           1.28.3     2021-12-09 [1] Bioconductor
##  biomformat             1.22.0     2021-10-26 [1] Bioconductor
##  Biostrings             2.62.0     2021-10-26 [1] Bioconductor
##  bit                    4.0.4      2020-08-04 [1] CRAN (R 4.1.0)
##  bit64                  4.0.5      2020-08-30 [1] CRAN (R 4.1.0)
##  bitops                 1.0-7      2021-04-24 [1] CRAN (R 4.1.0)
##  blob                   1.2.2      2021-07-23 [1] CRAN (R 4.1.0)
##  bookdown               0.24       2021-09-02 [1] CRAN (R 4.1.0)
##  brio                   1.1.3      2021-11-30 [1] CRAN (R 4.1.0)
##  broom                  0.7.12     2022-01-28 [1] CRAN (R 4.1.2)
##  bslib                  0.3.1      2021-10-06 [1] CRAN (R 4.1.0)
##  cachem                 1.0.6      2021-08-19 [1] CRAN (R 4.1.0)
##  callr                  3.7.0      2021-04-20 [1] CRAN (R 4.1.0)
##  car                    3.0-12     2021-11-06 [1] CRAN (R 4.1.0)
##  carData                3.0-5      2022-01-06 [1] CRAN (R 4.1.2)
##  caret                  6.0-92     2022-04-19 [1] CRAN (R 4.1.2)
##  caTools                1.18.2     2021-03-28 [1] CRAN (R 4.1.0)
##  cccd                   1.6        2022-04-08 [1] CRAN (R 4.1.2)
##  cellranger             1.1.0      2016-07-27 [1] CRAN (R 4.1.0)
##  checkmate              2.0.0      2020-02-06 [1] CRAN (R 4.1.0)
##  circlize               0.4.14     2022-02-11 [1] CRAN (R 4.1.2)
##  class                  7.3-20     2022-01-13 [1] CRAN (R 4.1.2)
##  cli                    3.3.0      2022-04-25 [1] CRAN (R 4.1.2)
##  clue                   0.3-60     2021-10-11 [1] CRAN (R 4.1.0)
##  cluster              * 2.1.2      2021-04-17 [1] CRAN (R 4.1.2)
##  codetools              0.2-18     2020-11-04 [1] CRAN (R 4.1.2)
##  coin                   1.4-2      2021-10-08 [1] CRAN (R 4.1.0)
##  colorspace             2.0-3      2022-02-21 [1] CRAN (R 4.1.2)
##  ComplexHeatmap         2.10.0     2021-10-26 [1] Bioconductor
##  corpcor                1.6.10     2021-09-16 [1] CRAN (R 4.1.0)
##  cowplot                1.1.1      2020-12-30 [1] CRAN (R 4.1.0)
##  crayon                 1.5.0      2022-02-14 [1] CRAN (R 4.1.2)
##  data.table             1.14.2     2021-09-27 [1] CRAN (R 4.1.0)
##  DBI                    1.1.2      2021-12-20 [1] CRAN (R 4.1.0)
##  DelayedArray           0.20.0     2021-10-26 [1] Bioconductor
##  deldir                 1.0-6      2021-10-23 [1] CRAN (R 4.1.0)
##  dendextend           * 1.15.2     2021-10-28 [1] CRAN (R 4.1.0)
##  desc                   1.4.1      2022-03-06 [1] CRAN (R 4.1.2)
##  DESeq2                 1.34.0     2021-10-26 [1] Bioconductor
##  devtools               2.4.3      2021-11-30 [1] CRAN (R 4.1.0)
##  digest                 0.6.29     2021-12-01 [1] CRAN (R 4.1.0)
##  doParallel             1.0.17     2022-02-07 [1] CRAN (R 4.1.2)
##  doSNOW                 1.0.20     2022-02-04 [1] CRAN (R 4.1.2)
##  dplyr                * 1.0.8      2022-02-08 [1] CRAN (R 4.1.2)
##  dynamicTreeCut       * 1.63-1     2016-03-11 [1] CRAN (R 4.1.0)
##  e1071                  1.7-9      2021-09-16 [1] CRAN (R 4.1.0)
##  edgeR                  3.36.0     2021-10-26 [1] Bioconductor
##  ellipse                0.4.2      2020-05-27 [1] CRAN (R 4.1.0)
##  ellipsis               0.3.2      2021-04-29 [1] CRAN (R 4.1.0)
##  evaluate               0.15       2022-02-18 [1] CRAN (R 4.1.2)
##  factoextra           * 1.0.7      2020-04-01 [1] CRAN (R 4.1.0)
##  fansi                  1.0.2      2022-01-14 [1] CRAN (R 4.1.2)
##  farver                 2.1.0      2021-02-28 [1] CRAN (R 4.1.0)
##  fastcluster          * 1.2.3      2021-05-24 [1] CRAN (R 4.1.0)
##  fastmap                1.1.0      2021-01-25 [1] CRAN (R 4.1.0)
##  fdrtool                1.2.17     2021-11-13 [1] CRAN (R 4.1.0)
##  filematrix             1.3        2018-02-27 [1] CRAN (R 4.1.0)
##  FNN                    1.1.3      2019-02-15 [1] CRAN (R 4.1.0)
##  foreach                1.5.2      2022-02-02 [1] CRAN (R 4.1.2)
##  foreign                0.8-82     2022-01-13 [1] CRAN (R 4.1.2)
##  Formula                1.2-4      2020-10-16 [1] CRAN (R 4.1.0)
##  fs                     1.5.2      2021-12-08 [1] CRAN (R 4.1.0)
##  future                 1.24.0     2022-02-19 [1] CRAN (R 4.1.2)
##  future.apply           1.8.1      2021-08-10 [1] CRAN (R 4.1.0)
##  genefilter             1.76.0     2021-10-26 [1] Bioconductor
##  geneplotter            1.72.0     2021-10-26 [1] Bioconductor
##  generics               0.1.2      2022-01-31 [1] CRAN (R 4.1.2)
##  GenomeInfoDb         * 1.30.1     2022-01-30 [1] Bioconductor
##  GenomeInfoDbData       1.2.7      2022-03-09 [1] Bioconductor
##  GenomicRanges        * 1.46.1     2021-11-18 [1] Bioconductor
##  GetoptLong             1.0.5      2020-12-15 [1] CRAN (R 4.1.0)
##  ggforce                0.3.3      2021-03-05 [1] CRAN (R 4.1.0)
##  ggplot2              * 3.3.5      2021-06-25 [1] CRAN (R 4.1.0)
##  ggpubr                 0.4.0      2020-06-27 [1] CRAN (R 4.1.0)
##  ggraph               * 2.0.5      2021-02-23 [1] CRAN (R 4.1.0)
##  ggrepel                0.9.1      2021-01-15 [1] CRAN (R 4.1.0)
##  ggsignif               0.6.3      2021-09-09 [1] CRAN (R 4.1.0)
##  glasso                 1.11       2019-10-01 [1] CRAN (R 4.1.0)
##  glmnet               * 4.1-3      2021-11-02 [1] CRAN (R 4.1.0)
##  GlobalOptions          0.1.2      2020-06-10 [1] CRAN (R 4.1.0)
##  globals                0.14.0     2020-11-22 [1] CRAN (R 4.1.0)
##  glue                   1.6.2      2022-02-24 [1] CRAN (R 4.1.2)
##  gmp                    0.6-5      2022-03-17 [1] CRAN (R 4.1.2)
##  GO.db                  3.14.0     2022-04-11 [1] Bioconductor
##  gower                  1.0.0      2022-02-03 [1] CRAN (R 4.1.2)
##  gplots                 3.1.1      2020-11-28 [1] CRAN (R 4.1.0)
##  graphlayouts           0.8.0      2022-01-03 [1] CRAN (R 4.1.2)
##  gridExtra              2.3        2017-09-09 [1] CRAN (R 4.1.0)
##  gtable                 0.3.0      2019-03-25 [1] CRAN (R 4.1.0)
##  gtools                 3.9.2      2021-06-06 [1] CRAN (R 4.1.0)
##  hardhat                0.2.0      2022-01-24 [1] CRAN (R 4.1.2)
##  highr                  0.9        2021-04-16 [1] CRAN (R 4.1.0)
##  Hmisc                  4.6-0      2021-10-07 [1] CRAN (R 4.1.0)
##  htmlTable              2.4.0      2022-01-04 [1] CRAN (R 4.1.2)
##  htmltools              0.5.2      2021-08-25 [1] CRAN (R 4.1.0)
##  htmlwidgets            1.5.4      2021-09-08 [1] CRAN (R 4.1.0)
##  httr                   1.4.2      2020-07-20 [1] CRAN (R 4.1.0)
##  huge                   1.3.5      2021-06-30 [1] CRAN (R 4.1.0)
##  igraph               * 1.2.11     2022-01-04 [1] CRAN (R 4.1.2)
##  impute                 1.68.0     2021-10-26 [1] Bioconductor
##  ipred                  0.9-12     2021-09-15 [1] CRAN (R 4.1.0)
##  IRanges              * 2.28.0     2021-10-26 [1] Bioconductor
##  irlba                  2.3.5      2021-12-06 [1] CRAN (R 4.1.0)
##  iterators              1.0.14     2022-02-05 [1] CRAN (R 4.1.2)
##  jpeg                   0.1-9      2021-07-24 [1] CRAN (R 4.1.0)
##  jquerylib              0.1.4      2021-04-26 [1] CRAN (R 4.1.0)
##  jsonlite               1.8.0      2022-02-22 [1] CRAN (R 4.1.2)
##  KEGGREST               1.34.0     2021-10-26 [1] Bioconductor
##  KernSmooth             2.23-20    2021-05-03 [1] CRAN (R 4.1.2)
##  knitr                  1.37       2021-12-16 [1] CRAN (R 4.1.0)
##  labeling               0.4.2      2020-10-20 [1] CRAN (R 4.1.0)
##  lattice                0.20-45    2021-09-22 [1] CRAN (R 4.1.2)
##  latticeExtra           0.6-29     2019-12-19 [1] CRAN (R 4.1.0)
##  lava                   1.6.10     2021-09-02 [1] CRAN (R 4.1.0)
##  lavaan                 0.6-11     2022-03-31 [1] CRAN (R 4.1.2)
##  lazyeval               0.2.2      2019-03-15 [1] CRAN (R 4.1.0)
##  libcoin                1.0-9      2021-09-27 [1] CRAN (R 4.1.0)
##  lifecycle              1.0.1      2021-09-24 [1] CRAN (R 4.1.0)
##  limma                  3.50.1     2022-02-17 [1] Bioconductor
##  listenv                0.8.0      2019-12-05 [1] CRAN (R 4.1.0)
##  locfit                 1.5-9.5    2022-03-03 [1] CRAN (R 4.1.2)
##  lubridate              1.8.0      2021-10-07 [1] CRAN (R 4.1.0)
##  magrittr               2.0.2      2022-01-26 [1] CRAN (R 4.1.2)
##  MASS                   7.3-55     2022-01-13 [1] CRAN (R 4.1.2)
##  Matrix               * 1.4-0      2021-12-08 [1] CRAN (R 4.1.0)
##  MatrixGenerics       * 1.6.0      2021-10-26 [1] Bioconductor
##  matrixStats          * 0.61.0     2021-09-17 [1] CRAN (R 4.1.0)
##  memoise                2.0.1      2021-11-26 [1] CRAN (R 4.1.0)
##  metagenomeSeq          1.36.0     2021-10-26 [1] Bioconductor
##  mgcv                   1.8-39     2022-02-24 [1] CRAN (R 4.1.2)
##  mixedCCA               1.5.2      2022-07-14 [1] Github (irinagain/mixedCCA@c6d41a3)
##  mixOmics               6.18.1     2021-11-18 [1] Bioconductor (R 4.1.2)
##  mnormt                 2.0.2      2020-09-01 [1] CRAN (R 4.1.0)
##  ModelMetrics           1.2.2.2    2020-03-17 [1] CRAN (R 4.1.0)
##  modeltools             0.2-23     2020-03-05 [1] CRAN (R 4.1.0)
##  multcomp               1.4-18     2022-01-04 [1] CRAN (R 4.1.2)
##  multtest               2.50.0     2021-10-26 [1] Bioconductor
##  munsell                0.5.0      2018-06-12 [1] CRAN (R 4.1.0)
##  mvtnorm                1.1-3      2021-10-08 [1] CRAN (R 4.1.0)
##  NADA                   1.6-1.1    2020-03-22 [1] CRAN (R 4.1.0)
##  NetCoMi              * 1.0.3      2022-07-14 [1] Github (stefpeschel/NetCoMi@d4d80d3)
##  nlme                   3.1-155    2022-01-13 [1] CRAN (R 4.1.2)
##  nnet                   7.3-17     2022-01-13 [1] CRAN (R 4.1.2)
##  parallelly             1.31.0     2022-04-07 [1] CRAN (R 4.1.2)
##  pbapply                1.5-0      2021-09-16 [1] CRAN (R 4.1.0)
##  pbivnorm               0.6.0      2015-01-23 [1] CRAN (R 4.1.0)
##  pcaPP                  1.9-74     2021-04-23 [1] CRAN (R 4.1.0)
##  permute                0.9-7      2022-01-27 [1] CRAN (R 4.1.2)
##  pheatmap               1.0.12     2019-01-04 [1] CRAN (R 4.1.0)
##  phyloseq               1.38.0     2021-10-26 [1] Bioconductor
##  pillar                 1.7.0      2022-02-01 [1] CRAN (R 4.1.2)
##  pkgbuild               1.3.1      2021-12-20 [1] CRAN (R 4.1.0)
##  pkgconfig              2.0.3      2019-09-22 [1] CRAN (R 4.1.0)
##  pkgload                1.2.4      2021-11-30 [1] CRAN (R 4.1.0)
##  plotly               * 4.10.0     2021-10-09 [1] CRAN (R 4.1.0)
##  plyr                   1.8.6      2020-03-03 [1] CRAN (R 4.1.0)
##  png                    0.1-7      2013-12-03 [1] CRAN (R 4.1.0)
##  polyclip               1.10-0     2019-03-14 [1] CRAN (R 4.1.0)
##  POMA                 * 1.7.2      2022-07-26 [1] Github (pcastellanoescuder/POMA@bc8a972)
##  preprocessCore         1.56.0     2021-10-26 [1] Bioconductor
##  prettyunits            1.1.1      2020-01-24 [1] CRAN (R 4.1.0)
##  pROC                   1.18.0     2021-09-03 [1] CRAN (R 4.1.0)
##  processx               3.5.2      2021-04-30 [1] CRAN (R 4.1.0)
##  prodlim                2019.11.13 2019-11-17 [1] CRAN (R 4.1.0)
##  proxy                  0.4-26     2021-06-07 [1] CRAN (R 4.1.0)
##  ps                     1.6.0      2021-02-28 [1] CRAN (R 4.1.0)
##  psych                  2.2.5      2022-05-10 [1] CRAN (R 4.1.2)
##  pulsar                 0.3.7      2020-08-07 [1] CRAN (R 4.1.0)
##  purrr                  0.3.4      2020-04-17 [1] CRAN (R 4.1.0)
##  qgraph                 1.9.2      2022-03-04 [1] CRAN (R 4.1.2)
##  R6                     2.5.1      2021-08-19 [1] CRAN (R 4.1.0)
##  randomForest           4.7-1      2022-02-03 [1] CRAN (R 4.1.2)
##  RankProd               3.20.0     2021-10-26 [1] Bioconductor
##  rARPACK                0.11-0     2016-03-10 [1] CRAN (R 4.1.0)
##  rbibutils              2.2.7      2021-12-07 [1] CRAN (R 4.1.0)
##  RColorBrewer           1.1-2      2014-12-07 [1] CRAN (R 4.1.0)
##  Rcpp                   1.0.8.2    2022-03-11 [1] CRAN (R 4.1.2)
##  RCurl                  1.98-1.6   2022-02-08 [1] CRAN (R 4.1.2)
##  Rdpack                 2.2        2022-03-19 [1] CRAN (R 4.1.2)
##  readxl               * 1.4.0      2022-03-28 [1] CRAN (R 4.1.2)
##  recipes                0.2.0      2022-02-18 [1] CRAN (R 4.1.2)
##  remotes                2.4.2      2021-11-30 [1] CRAN (R 4.1.0)
##  reshape2               1.4.4      2020-04-09 [1] CRAN (R 4.1.0)
##  rhdf5                  2.38.1     2022-03-10 [1] Bioconductor
##  rhdf5filters           1.6.0      2021-10-26 [1] Bioconductor
##  Rhdf5lib               1.16.0     2021-10-26 [1] Bioconductor
##  rjson                  0.2.21     2022-01-09 [1] CRAN (R 4.1.2)
##  rlang                  1.0.2      2022-03-04 [1] CRAN (R 4.1.2)
##  rmarkdown              2.13       2022-03-10 [1] CRAN (R 4.1.2)
##  Rmpfr                  0.8-9      2022-06-03 [1] CRAN (R 4.1.2)
##  rootSolve              1.8.2.3    2021-09-29 [1] CRAN (R 4.1.0)
##  rpart                  4.1.16     2022-01-24 [1] CRAN (R 4.1.2)
##  rprojroot              2.0.2      2020-11-15 [1] CRAN (R 4.1.0)
##  RSpectra               0.16-0     2019-12-01 [1] CRAN (R 4.1.0)
##  RSQLite                2.2.10     2022-02-17 [1] CRAN (R 4.1.2)
##  rstatix                0.7.0      2021-02-13 [1] CRAN (R 4.1.0)
##  rstudioapi             0.13       2020-11-12 [1] CRAN (R 4.1.0)
##  S4Vectors            * 0.32.3     2021-11-21 [1] Bioconductor
##  sandwich               3.0-1      2021-05-18 [1] CRAN (R 4.1.0)
##  sass                   0.4.0      2021-05-12 [1] CRAN (R 4.1.0)
##  scales                 1.1.1      2020-05-11 [1] CRAN (R 4.1.0)
##  sessioninfo            1.2.2      2021-12-06 [1] CRAN (R 4.1.0)
##  shape                  1.4.6      2021-05-19 [1] CRAN (R 4.1.0)
##  snow                   0.4-4      2021-10-27 [1] CRAN (R 4.1.0)
##  SpiecEasi            * 1.1.2      2022-07-14 [1] Github (zdk123/SpiecEasi@c463727)
##  SPRING               * 1.0.4      2022-08-03 [1] Github (GraceYoon/SPRING@3d641a4)
##  stringi                1.7.6      2021-11-29 [1] CRAN (R 4.1.0)
##  stringr                1.4.0      2019-02-10 [1] CRAN (R 4.1.0)
##  SummarizedExperiment * 1.24.0     2021-10-26 [1] Bioconductor
##  survival               3.3-1      2022-03-03 [1] CRAN (R 4.1.2)
##  testthat               3.1.2      2022-01-20 [1] CRAN (R 4.1.2)
##  TH.data                1.1-0      2021-09-27 [1] CRAN (R 4.1.0)
##  tibble               * 3.1.6      2021-11-07 [1] CRAN (R 4.1.0)
##  tidygraph              1.2.1      2022-04-05 [1] CRAN (R 4.1.2)
##  tidyr                  1.2.0      2022-02-01 [1] CRAN (R 4.1.2)
##  tidyselect             1.1.2      2022-02-21 [1] CRAN (R 4.1.2)
##  timeDate               3043.102   2018-02-21 [1] CRAN (R 4.1.0)
##  tmvnsim                1.0-2      2016-12-15 [1] CRAN (R 4.1.0)
##  truncnorm              1.0-8      2018-02-27 [1] CRAN (R 4.1.0)
##  tweenr                 1.0.2      2021-03-23 [1] CRAN (R 4.1.0)
##  usethis                2.1.5      2021-12-09 [1] CRAN (R 4.1.0)
##  utf8                   1.2.2      2021-07-24 [1] CRAN (R 4.1.0)
##  vctrs                  0.3.8      2021-04-29 [1] CRAN (R 4.1.0)
##  vegan                  2.5-7      2020-11-28 [1] CRAN (R 4.1.0)
##  VGAM                   1.1-6      2022-02-14 [1] CRAN (R 4.1.2)
##  viridis                0.6.2      2021-10-13 [1] CRAN (R 4.1.0)
##  viridisLite            0.4.0      2021-04-13 [1] CRAN (R 4.1.0)
##  WGCNA                * 1.71       2022-04-22 [1] CRAN (R 4.1.2)
##  withr                  2.5.0      2022-03-03 [1] CRAN (R 4.1.2)
##  Wrench                 1.12.0     2021-10-26 [1] Bioconductor
##  xfun                   0.30       2022-03-02 [1] CRAN (R 4.1.2)
##  XMAS2                  2.1.7.2    2022-07-29 [1] local
##  XML                    3.99-0.9   2022-02-24 [1] CRAN (R 4.1.2)
##  xtable                 1.8-4      2019-04-21 [1] CRAN (R 4.1.0)
##  XVector                0.34.0     2021-10-26 [1] Bioconductor
##  yaml                   2.3.5      2022-02-21 [1] CRAN (R 4.1.2)
##  zCompositions          1.4.0      2022-01-13 [1] CRAN (R 4.1.2)
##  zlibbioc               1.40.0     2021-10-26 [1] Bioconductor
##  zoo                    1.8-9      2021-03-09 [1] CRAN (R 4.1.0)
## 
##  [1] /Library/Frameworks/R.framework/Versions/4.1/Resources/library
## 
## ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="network-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/bookdown_demo/blob/main/19-Example.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Statistical Analysis in Metabolomics.pdf", "Statistical Analysis in Metabolomics.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
