<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Data Processing | Data Analysis in Metabolomics</title>
  <meta name="description" content="A template for Data Analysis in Metabolomics" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Data Processing | Data Analysis in Metabolomics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A template for Data Analysis in Metabolomics" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Data Processing | Data Analysis in Metabolomics" />
  
  <meta name="twitter:description" content="A template for Data Analysis in Metabolomics" />
  

<meta name="author" content="Hua Zou and Bangzhuo Tong" />


<meta name="date" content="2023-12-07" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="statistical-analysis-1.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="assets/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Data Analysis in Metabolomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#what-is-metabolomics"><i class="fa fa-check"></i><b>1.1</b> What is metabolomics?</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#workflow-of-metabolomics-data-analysis"><i class="fa fa-check"></i><b>1.2</b> Workflow of metabolomics data analysis</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#statistical-analysis"><i class="fa fa-check"></i><b>1.2.1</b> Statistical Analysis</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#functional-analysis"><i class="fa fa-check"></i><b>1.2.2</b> Functional Analysis</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#software"><i class="fa fa-check"></i><b>1.3</b> Software</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#reference"><i class="fa fa-check"></i><b>1.4</b> Reference</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-processing.html"><a href="data-processing.html"><i class="fa fa-check"></i><b>2</b> Data Processing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-processing.html"><a href="data-processing.html#loading-packages"><i class="fa fa-check"></i><b>2.1</b> Loading packages</a></li>
<li class="chapter" data-level="2.2" data-path="data-processing.html"><a href="data-processing.html#importing-data"><i class="fa fa-check"></i><b>2.2</b> Importing data</a></li>
<li class="chapter" data-level="2.3" data-path="data-processing.html"><a href="data-processing.html#data-checking"><i class="fa fa-check"></i><b>2.3</b> Data Checking</a></li>
<li class="chapter" data-level="2.4" data-path="data-processing.html"><a href="data-processing.html#data-trimming"><i class="fa fa-check"></i><b>2.4</b> Data Trimming</a></li>
<li class="chapter" data-level="2.5" data-path="data-processing.html"><a href="data-processing.html#missing-value-imputation"><i class="fa fa-check"></i><b>2.5</b> Missing value imputation</a></li>
<li class="chapter" data-level="2.6" data-path="data-processing.html"><a href="data-processing.html#data-filtering"><i class="fa fa-check"></i><b>2.6</b> Data Filtering</a></li>
<li class="chapter" data-level="2.7" data-path="data-processing.html"><a href="data-processing.html#data-normalization"><i class="fa fa-check"></i><b>2.7</b> Data Normalization</a>
<ul>
<li class="chapter" data-level="2.7.1" data-path="data-processing.html"><a href="data-processing.html#normalization-by-normalizedata-function"><i class="fa fa-check"></i><b>2.7.1</b> Normalization by <strong>NormalizeData</strong> function</a></li>
<li class="chapter" data-level="2.7.2" data-path="data-processing.html"><a href="data-processing.html#normalization-by-poma-r-package"><i class="fa fa-check"></i><b>2.7.2</b> Normalization by <strong>POMA</strong> R package</a></li>
<li class="chapter" data-level="2.7.3" data-path="data-processing.html"><a href="data-processing.html#comparison-of-unnormalized-and-normalized-dataset"><i class="fa fa-check"></i><b>2.7.3</b> Comparison of unnormalized and normalized dataset</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="data-processing.html"><a href="data-processing.html#removing-outliers"><i class="fa fa-check"></i><b>2.8</b> Removing outliers</a></li>
<li class="chapter" data-level="2.9" data-path="data-processing.html"><a href="data-processing.html#data-distribution"><i class="fa fa-check"></i><b>2.9</b> Data distribution</a></li>
<li class="chapter" data-level="2.10" data-path="data-processing.html"><a href="data-processing.html#saving-datasets-into-rds-files"><i class="fa fa-check"></i><b>2.10</b> Saving datasets into RDS files</a></li>
<li class="chapter" data-level="2.11" data-path="data-processing.html"><a href="data-processing.html#systematic-information"><i class="fa fa-check"></i><b>2.11</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html"><i class="fa fa-check"></i><b>3</b> Statistical Analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-preprocessing"><i class="fa fa-check"></i><b>3.1</b> Data Preprocessing</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#environment-setup"><i class="fa fa-check"></i><b>3.1.1</b> Environment setup</a></li>
<li class="chapter" data-level="3.1.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-data"><i class="fa fa-check"></i><b>3.1.2</b> Loading data</a></li>
<li class="chapter" data-level="3.1.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#object-preparation"><i class="fa fa-check"></i><b>3.1.3</b> Object Preparation</a></li>
<li class="chapter" data-level="3.1.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-checking-1"><i class="fa fa-check"></i><b>3.1.4</b> Data Checking</a></li>
<li class="chapter" data-level="3.1.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#missing-value-imputation-1"><i class="fa fa-check"></i><b>3.1.5</b> Missing value imputation</a></li>
<li class="chapter" data-level="3.1.6" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-filtering-1"><i class="fa fa-check"></i><b>3.1.6</b> Data Filtering</a></li>
<li class="chapter" data-level="3.1.7" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-normalization-1"><i class="fa fa-check"></i><b>3.1.7</b> Data Normalization</a></li>
<li class="chapter" data-level="3.1.8" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#removing-outliers-1"><i class="fa fa-check"></i><b>3.1.8</b> Removing outliers</a></li>
<li class="chapter" data-level="3.1.9" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#saving-datasets-into-rds-files-1"><i class="fa fa-check"></i><b>3.1.9</b> Saving datasets into RDS files</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#cluster-analysis"><i class="fa fa-check"></i><b>3.2</b> Cluster Analysis</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-packages-1"><i class="fa fa-check"></i><b>3.2.1</b> Loading packages</a></li>
<li class="chapter" data-level="3.2.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#importing-data-1"><i class="fa fa-check"></i><b>3.2.2</b> Importing data</a></li>
<li class="chapter" data-level="3.2.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#hierarchical-clustering"><i class="fa fa-check"></i><b>3.2.3</b> Hierarchical Clustering</a></li>
<li class="chapter" data-level="3.2.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#partitional-clustering"><i class="fa fa-check"></i><b>3.2.4</b> Partitional Clustering</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#chemometrics-analysis"><i class="fa fa-check"></i><b>3.3</b> Chemometrics Analysis</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-packages-2"><i class="fa fa-check"></i><b>3.3.1</b> Loading packages</a></li>
<li class="chapter" data-level="3.3.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#importing-data-2"><i class="fa fa-check"></i><b>3.3.2</b> Importing data</a></li>
<li class="chapter" data-level="3.3.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>3.3.3</b> Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="3.3.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#partial-least-squares-discriminant-analysis-pls-da"><i class="fa fa-check"></i><b>3.3.4</b> Partial Least Squares-Discriminant Analysis (PLS-DA)</a></li>
<li class="chapter" data-level="3.3.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#sparse-partial-least-squares-discriminant-analysis-spls-da"><i class="fa fa-check"></i><b>3.3.5</b> Sparse Partial Least Squares-Discriminant Analysis (sPLS-DA)</a></li>
<li class="chapter" data-level="3.3.6" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#orthogonal-partial-least-squares-discriminant-analysis-orthopls-da"><i class="fa fa-check"></i><b>3.3.6</b> Orthogonal Partial Least Squares-Discriminant Analysis (orthoPLS-DA)</a></li>
<li class="chapter" data-level="3.3.7" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#chemometrics-analysis-by-own-scripts"><i class="fa fa-check"></i><b>3.3.7</b> Chemometrics Analysis by own scripts</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#univariate-analysis"><i class="fa fa-check"></i><b>3.4</b> Univariate Analysis</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-packages-3"><i class="fa fa-check"></i><b>3.4.1</b> Loading packages</a></li>
<li class="chapter" data-level="3.4.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#importing-data-3"><i class="fa fa-check"></i><b>3.4.2</b> Importing data</a></li>
<li class="chapter" data-level="3.4.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#fold-change-analysis"><i class="fa fa-check"></i><b>3.4.3</b> Fold Change Analysis</a></li>
<li class="chapter" data-level="3.4.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#t-test"><i class="fa fa-check"></i><b>3.4.4</b> T Test</a></li>
<li class="chapter" data-level="3.4.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#wilcoxon-test"><i class="fa fa-check"></i><b>3.4.5</b> Wilcoxon Test</a></li>
<li class="chapter" data-level="3.4.6" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#limma-test"><i class="fa fa-check"></i><b>3.4.6</b> Limma Test</a></li>
<li class="chapter" data-level="3.4.7" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#volcano-plot"><i class="fa fa-check"></i><b>3.4.7</b> Volcano plot</a></li>
<li class="chapter" data-level="3.4.8" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#vip-variable-influence-on-projection-coefficient"><i class="fa fa-check"></i><b>3.4.8</b> VIP (Variable influence on projection &amp; coefficient)</a></li>
<li class="chapter" data-level="3.4.9" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#t-test-by-local-codes"><i class="fa fa-check"></i><b>3.4.9</b> T-test by local codes</a></li>
<li class="chapter" data-level="3.4.10" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#merging-result"><i class="fa fa-check"></i><b>3.4.10</b> Merging result</a></li>
<li class="chapter" data-level="3.4.11" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#volcano-of-merged-results"><i class="fa fa-check"></i><b>3.4.11</b> Volcano of Merged Results</a></li>
<li class="chapter" data-level="3.4.12" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#adjusting-for-confounding-variables"><i class="fa fa-check"></i><b>3.4.12</b> Adjusting for confounding variables</a></li>
<li class="chapter" data-level="3.4.13" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#correlation-heatmaps"><i class="fa fa-check"></i><b>3.4.13</b> Correlation Heatmaps</a></li>
<li class="chapter" data-level="3.4.14" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#glasso-this-function-will-compute-a-gaussian-graphical-model-using-the-glmnet-package"><i class="fa fa-check"></i><b>3.4.14</b> glasso: this function will compute a Gaussian Graphical Model using the glmnet package</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#multivariate-analysis"><i class="fa fa-check"></i><b>3.5</b> Multivariate analysis</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-packages-4"><i class="fa fa-check"></i><b>3.5.1</b> Loading packages</a></li>
<li class="chapter" data-level="3.5.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#importing-data-4"><i class="fa fa-check"></i><b>3.5.2</b> Importing data</a></li>
<li class="chapter" data-level="3.5.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-curation"><i class="fa fa-check"></i><b>3.5.3</b> Data curation</a></li>
<li class="chapter" data-level="3.5.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#regularized-generalized-linear-models"><i class="fa fa-check"></i><b>3.5.4</b> Regularized Generalized Linear Models</a></li>
<li class="chapter" data-level="3.5.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#classification"><i class="fa fa-check"></i><b>3.5.5</b> Classification</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#network-analysis"><i class="fa fa-check"></i><b>3.6</b> Network Analysis</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#introduction-1"><i class="fa fa-check"></i><b>3.6.1</b> Introduction</a></li>
<li class="chapter" data-level="3.6.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-packages-5"><i class="fa fa-check"></i><b>3.6.2</b> Loading packages</a></li>
<li class="chapter" data-level="3.6.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#importing-data-5"><i class="fa fa-check"></i><b>3.6.3</b> Importing data</a></li>
<li class="chapter" data-level="3.6.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-curation-1"><i class="fa fa-check"></i><b>3.6.4</b> Data curation</a></li>
<li class="chapter" data-level="3.6.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#associations-among-features"><i class="fa fa-check"></i><b>3.6.5</b> Associations Among Features</a></li>
<li class="chapter" data-level="3.6.6" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#network-comparison"><i class="fa fa-check"></i><b>3.6.6</b> Network comparison</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#systematic-information-1"><i class="fa fa-check"></i><b>3.7</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html"><i class="fa fa-check"></i><b>4</b> Functional Analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#enrichment-analysis"><i class="fa fa-check"></i><b>4.1</b> Enrichment Analysis</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#enviroment-setup"><i class="fa fa-check"></i><b>4.1.1</b> Enviroment Setup</a></li>
<li class="chapter" data-level="4.1.2" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#over-representation-analysis-ora"><i class="fa fa-check"></i><b>4.1.2</b> Over representation analysis (ORA)</a></li>
<li class="chapter" data-level="4.1.3" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#clean-environment-1"><i class="fa fa-check"></i><b>4.1.3</b> Clean environment 1</a></li>
<li class="chapter" data-level="4.1.4" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#single-sample-profiling"><i class="fa fa-check"></i><b>4.1.4</b> Single Sample Profiling</a></li>
<li class="chapter" data-level="4.1.5" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#clean-environment-2"><i class="fa fa-check"></i><b>4.1.5</b> Clean environment 2</a></li>
<li class="chapter" data-level="4.1.6" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#quantitative-enrichment-analysis-qea"><i class="fa fa-check"></i><b>4.1.6</b> Quantitative Enrichment Analysis (QEA)</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#pathway-analysis"><i class="fa fa-check"></i><b>4.2</b> Pathway Analysis</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#enviroment-set-up"><i class="fa fa-check"></i><b>4.2.1</b> Enviroment Set up</a></li>
<li class="chapter" data-level="4.2.2" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#over-representation-analysis"><i class="fa fa-check"></i><b>4.2.2</b> Over representation analysis</a></li>
<li class="chapter" data-level="4.2.3" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#clean-environment-3"><i class="fa fa-check"></i><b>4.2.3</b> Clean environment 3</a></li>
<li class="chapter" data-level="4.2.4" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#concentration-table-qea"><i class="fa fa-check"></i><b>4.2.4</b> Concentration Table (QEA)</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#functional-analysis-by-r-package"><i class="fa fa-check"></i><b>4.3</b> Functional Analysis by R package</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#loading-r-packages"><i class="fa fa-check"></i><b>4.3.1</b> Loading R packages</a></li>
<li class="chapter" data-level="4.3.2" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#importing-data-6"><i class="fa fa-check"></i><b>4.3.2</b> Importing Data</a></li>
<li class="chapter" data-level="4.3.3" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#differential-analysis"><i class="fa fa-check"></i><b>4.3.3</b> Differential Analysis</a></li>
<li class="chapter" data-level="4.3.4" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#metabolites-id-for-merged-results"><i class="fa fa-check"></i><b>4.3.4</b> Metabolites ID for Merged Results</a></li>
<li class="chapter" data-level="4.3.5" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#kegg-pathway-compound"><i class="fa fa-check"></i><b>4.3.5</b> KEGG PATHWAY COMPOUND</a></li>
<li class="chapter" data-level="4.3.6" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#functions-related-to-analysis"><i class="fa fa-check"></i><b>4.3.6</b> Functions related to Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#session-info"><i class="fa fa-check"></i><b>4.4</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html"><i class="fa fa-check"></i><b>5</b> MetOrigin Analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#loading-packages-6"><i class="fa fa-check"></i><b>5.1</b> Loading packages</a></li>
<li class="chapter" data-level="5.2" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#importing-data-7"><i class="fa fa-check"></i><b>5.2</b> Importing data</a></li>
<li class="chapter" data-level="5.3" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#differential-analysis-1"><i class="fa fa-check"></i><b>5.3</b> Differential Analysis</a></li>
<li class="chapter" data-level="5.4" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#de-metabolites"><i class="fa fa-check"></i><b>5.4</b> DE metabolites</a></li>
<li class="chapter" data-level="5.5" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#obtain-inputs-for-metorigin"><i class="fa fa-check"></i><b>5.5</b> Obtain inputs for Metorigin</a></li>
<li class="chapter" data-level="5.6" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#save-result"><i class="fa fa-check"></i><b>5.6</b> Save result</a></li>
<li class="chapter" data-level="5.7" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#metorigin-user-tutorial"><i class="fa fa-check"></i><b>5.7</b> MetOrigin User Tutorial</a></li>
<li class="chapter" data-level="5.8" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#session-info-1"><i class="fa fa-check"></i><b>5.8</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="example.html"><a href="example.html"><i class="fa fa-check"></i><b>6</b> Example</a>
<ul>
<li class="chapter" data-level="6.1" data-path="example.html"><a href="example.html#loading-packages-7"><i class="fa fa-check"></i><b>6.1</b> Loading packages</a></li>
<li class="chapter" data-level="6.2" data-path="example.html"><a href="example.html#importing-data-8"><i class="fa fa-check"></i><b>6.2</b> Importing data</a></li>
<li class="chapter" data-level="6.3" data-path="example.html"><a href="example.html#data-processing-1"><i class="fa fa-check"></i><b>6.3</b> Data Processing</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="example.html"><a href="example.html#data-checking-2"><i class="fa fa-check"></i><b>6.3.1</b> Data Checking</a></li>
<li class="chapter" data-level="6.3.2" data-path="example.html"><a href="example.html#missing-value-imputation-2"><i class="fa fa-check"></i><b>6.3.2</b> Missing value imputation</a></li>
<li class="chapter" data-level="6.3.3" data-path="example.html"><a href="example.html#data-filtering-2"><i class="fa fa-check"></i><b>6.3.3</b> Data Filtering</a></li>
<li class="chapter" data-level="6.3.4" data-path="example.html"><a href="example.html#data-normalization-2"><i class="fa fa-check"></i><b>6.3.4</b> Data Normalization</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="example.html"><a href="example.html#cluster-analysis-1"><i class="fa fa-check"></i><b>6.4</b> Cluster Analysis</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="example.html"><a href="example.html#hierarchical-clustering-1"><i class="fa fa-check"></i><b>6.4.1</b> Hierarchical Clustering</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="example.html"><a href="example.html#chemometrics-analysis-1"><i class="fa fa-check"></i><b>6.5</b> Chemometrics Analysis</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="example.html"><a href="example.html#partial-least-squares-discriminant-analysis-pls-da-1"><i class="fa fa-check"></i><b>6.5.1</b> Partial Least Squares-Discriminant Analysis (PLS-DA)</a></li>
<li class="chapter" data-level="6.5.2" data-path="example.html"><a href="example.html#sparse-partial-least-squares-discriminant-analysis-spls-da-1"><i class="fa fa-check"></i><b>6.5.2</b> Sparse Partial Least Squares-Discriminant Analysis (sPLS-DA)</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="example.html"><a href="example.html#univariate-analysis-1"><i class="fa fa-check"></i><b>6.6</b> Univariate Analysis</a>
<ul>
<li class="chapter" data-level="6.6.1" data-path="example.html"><a href="example.html#fold-change-analysis-1"><i class="fa fa-check"></i><b>6.6.1</b> Fold Change Analysis</a></li>
<li class="chapter" data-level="6.6.2" data-path="example.html"><a href="example.html#vip-variable-influence-on-projection-coefficient-1"><i class="fa fa-check"></i><b>6.6.2</b> VIP (Variable influence on projection &amp; coefficient)</a></li>
<li class="chapter" data-level="6.6.3" data-path="example.html"><a href="example.html#t-test-by-local-codes-1"><i class="fa fa-check"></i><b>6.6.3</b> T-test by local codes</a></li>
<li class="chapter" data-level="6.6.4" data-path="example.html"><a href="example.html#merging-result-1"><i class="fa fa-check"></i><b>6.6.4</b> Merging result</a></li>
<li class="chapter" data-level="6.6.5" data-path="example.html"><a href="example.html#volcano-of-merged-results-1"><i class="fa fa-check"></i><b>6.6.5</b> Volcano of Merged Results</a></li>
<li class="chapter" data-level="6.6.6" data-path="example.html"><a href="example.html#t-test-1"><i class="fa fa-check"></i><b>6.6.6</b> T Test</a></li>
<li class="chapter" data-level="6.6.7" data-path="example.html"><a href="example.html#volcano-plot-1"><i class="fa fa-check"></i><b>6.6.7</b> Volcano plot</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="example.html"><a href="example.html#feature-selection"><i class="fa fa-check"></i><b>6.7</b> Feature Selection</a>
<ul>
<li class="chapter" data-level="6.7.1" data-path="example.html"><a href="example.html#regularized-generalized-linear-models-lasso-alpha-1"><i class="fa fa-check"></i><b>6.7.1</b> Regularized Generalized Linear Models (Lasso: alpha = 1)</a></li>
<li class="chapter" data-level="6.7.2" data-path="example.html"><a href="example.html#classification-random-forest"><i class="fa fa-check"></i><b>6.7.2</b> Classification (Random Forest)</a></li>
</ul></li>
<li class="chapter" data-level="6.8" data-path="example.html"><a href="example.html#network-analysis-2"><i class="fa fa-check"></i><b>6.8</b> Network Analysis</a>
<ul>
<li class="chapter" data-level="6.8.1" data-path="example.html"><a href="example.html#data-curation-2"><i class="fa fa-check"></i><b>6.8.1</b> Data curation</a></li>
<li class="chapter" data-level="6.8.2" data-path="example.html"><a href="example.html#building-network-model-4"><i class="fa fa-check"></i><b>6.8.2</b> Building network model</a></li>
<li class="chapter" data-level="6.8.3" data-path="example.html"><a href="example.html#visualizing-the-network-4"><i class="fa fa-check"></i><b>6.8.3</b> Visualizing the network</a></li>
</ul></li>
<li class="chapter" data-level="6.9" data-path="example.html"><a href="example.html#network-analysis-by-wgcna"><i class="fa fa-check"></i><b>6.9</b> Network Analysis by WGCNA</a>
<ul>
<li class="chapter" data-level="6.9.1" data-path="example.html"><a href="example.html#data-curation-3"><i class="fa fa-check"></i><b>6.9.1</b> Data curation</a></li>
<li class="chapter" data-level="6.9.2" data-path="example.html"><a href="example.html#tuning-soft-thresholds"><i class="fa fa-check"></i><b>6.9.2</b> Tuning soft thresholds</a></li>
<li class="chapter" data-level="6.9.3" data-path="example.html"><a href="example.html#create-the-network-using-the-blockwisemodules"><i class="fa fa-check"></i><b>6.9.3</b> Create the network using the <code>blockwiseModules</code></a></li>
<li class="chapter" data-level="6.9.4" data-path="example.html"><a href="example.html#plot-modules"><i class="fa fa-check"></i><b>6.9.4</b> plot modules</a></li>
<li class="chapter" data-level="6.9.5" data-path="example.html"><a href="example.html#relationships-among-modules"><i class="fa fa-check"></i><b>6.9.5</b> Relationships among modules</a></li>
<li class="chapter" data-level="6.9.6" data-path="example.html"><a href="example.html#module-eigengene-correlation"><i class="fa fa-check"></i><b>6.9.6</b> Module (Eigengene) correlation</a></li>
<li class="chapter" data-level="6.9.7" data-path="example.html"><a href="example.html#relate-module-cluster-assignments-to-groups"><i class="fa fa-check"></i><b>6.9.7</b> Relate Module (cluster) Assignments to Groups</a></li>
<li class="chapter" data-level="6.9.8" data-path="example.html"><a href="example.html#generate-and-export-networks"><i class="fa fa-check"></i><b>6.9.8</b> Generate and Export Networks</a></li>
<li class="chapter" data-level="6.9.9" data-path="example.html"><a href="example.html#network-visualization"><i class="fa fa-check"></i><b>6.9.9</b> Network visualization</a></li>
</ul></li>
<li class="chapter" data-level="6.10" data-path="example.html"><a href="example.html#systematic-information-2"><i class="fa fa-check"></i><b>6.10</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data Analysis in Metabolomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="data-processing" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Data Processing<a href="data-processing.html#data-processing" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Although the horrible experience of data analysis by using <strong>MetaboAnalystR</strong> R package <span class="citation">(<a href="#ref-pang2020metaboanalystr">Pang et al. 2020</a>)</span>, its thought of data processing are very useful. Therefore, this template is based on the workflow from <strong>MetaboAnalystR</strong>.</p>
<p>We integrated R packages and our own scripts to build the data analysis template on metabolomic data. Particularly, we thanks very much for <strong>POMA</strong> R package <span class="citation">(<a href="#ref-castellano2021pomashiny">Castellano-Escuder et al. 2021</a>)</span>. <strong>POMA</strong> is a flexible data cleaning and statistical analysis processes in one comprehensible and user-friendly R package.</p>
<div id="loading-packages" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Loading packages<a href="data-processing.html#loading-packages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="data-processing.html#cb1-1" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">warning =</span> F)</span>
<span id="cb1-2"><a href="data-processing.html#cb1-2" tabindex="-1"></a></span>
<span id="cb1-3"><a href="data-processing.html#cb1-3" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-4"><a href="data-processing.html#cb1-4" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb1-5"><a href="data-processing.html#cb1-5" tabindex="-1"></a><span class="fu">library</span>(Biobase)</span>
<span id="cb1-6"><a href="data-processing.html#cb1-6" tabindex="-1"></a><span class="fu">library</span>(POMA)</span>
<span id="cb1-7"><a href="data-processing.html#cb1-7" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-8"><a href="data-processing.html#cb1-8" tabindex="-1"></a><span class="fu">library</span>(ggraph)</span>
<span id="cb1-9"><a href="data-processing.html#cb1-9" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb1-10"><a href="data-processing.html#cb1-10" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb1-11"><a href="data-processing.html#cb1-11" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb1-12"><a href="data-processing.html#cb1-12" tabindex="-1"></a></span>
<span id="cb1-13"><a href="data-processing.html#cb1-13" tabindex="-1"></a><span class="co"># rm(list = ls())</span></span>
<span id="cb1-14"><a href="data-processing.html#cb1-14" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb1-15"><a href="data-processing.html#cb1-15" tabindex="-1"></a><span class="fu">options</span>(<span class="at">future.globals.maxSize =</span> <span class="dv">1000</span> <span class="sc">*</span> <span class="dv">1024</span><span class="sc">^</span><span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="importing-data" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Importing data<a href="data-processing.html#importing-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The dataset is from the Zeybel-2022 published paper <span class="citation">(<a href="#ref-zeybel2022multiomics">Zeybel et al. 2022</a>)</span>.</p>
<ul>
<li>features table</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="data-processing.html#cb2-1" tabindex="-1"></a>profile <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(<span class="st">&quot;./dataset/OmicsDataSet-Zeybel-2022.xlsx&quot;</span>, <span class="at">sheet =</span> <span class="dv">6</span>)</span>
<span id="cb2-2"><a href="data-processing.html#cb2-2" tabindex="-1"></a><span class="fu">head</span>(profile)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 67
##   BIOCHEMICAL `SUPER PATHWAY` `SUB PATHWAY` `COMP ID` PLATFORM `CHEMICAL ID`    RI  MASS PUBCHEM CAS   KEGG  `SampleID HMDBID` P101001 P101012 P101030
##   &lt;chr&gt;       &lt;chr&gt;           &lt;chr&gt;             &lt;dbl&gt; &lt;chr&gt;            &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;               &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;
## 1 (14 or 15)… Lipid           Fatty Acid, …     38768 LC/MS N…     100002945  5695  269. 8181;1… &lt;NA&gt;  C169… HMDB0061859        5.11e7  5.12e7  3.84e7
## 2 (16 or 17)… Lipid           Fatty Acid, …     38296 LC/MS N…     100002356  5993  297. 3083779 2724… &lt;NA&gt;  HMDB0037397        5.11e6  6.00e6  2.86e6
## 3 (2 or 3)-d… Lipid           Medium Chain…     63436 LC/MS N…     100021502  4990  169. &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;  &lt;NA&gt;               7.57e5  5.98e5  3.67e5
## 4 (2,4 or 2,… Xenobiotics     Food Compone…     62533 LC/MS N…     100020519  3474  201. &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;  &lt;NA&gt;              NA      NA       5.64e5
## 5 (N(1) + N(… Amino Acid      Polyamine Me…     57814 LC/MS P…     100016038  3080  188. 123689… &lt;NA&gt;  C006… HMDB0001276,HMDB…  2.82e5  2.49e5  2.31e5
## 6 (R)-3-hydr… Lipid           Fatty Acid M…     43264 LC/MS P…     100003926  2400  248. 534816… &lt;NA&gt;  &lt;NA&gt;  HMDB0013127       NA      NA      NA     
## # ℹ 52 more variables: P101031 &lt;dbl&gt;, P101050 &lt;dbl&gt;, P101059 &lt;dbl&gt;, P101071 &lt;dbl&gt;, P101072 &lt;dbl&gt;, P101084 &lt;dbl&gt;, P101003 &lt;dbl&gt;, P101004 &lt;dbl&gt;,
## #   P101013 &lt;dbl&gt;, P101016 &lt;dbl&gt;, P101017 &lt;dbl&gt;, P101038 &lt;dbl&gt;, P101051 &lt;dbl&gt;, P101061 &lt;dbl&gt;, P101062 &lt;dbl&gt;, P101074 &lt;dbl&gt;, P101075 &lt;dbl&gt;,
## #   P101076 &lt;dbl&gt;, P101085 &lt;dbl&gt;, P101088 &lt;dbl&gt;, P101007 &lt;dbl&gt;, P101018 &lt;dbl&gt;, P101019 &lt;dbl&gt;, P101041 &lt;dbl&gt;, P101052 &lt;dbl&gt;, P101064 &lt;dbl&gt;,
## #   P101065 &lt;dbl&gt;, P101077 &lt;dbl&gt;, P101090 &lt;dbl&gt;, P101094 &lt;dbl&gt;, P101009 &lt;dbl&gt;, P101010 &lt;dbl&gt;, P101021 &lt;dbl&gt;, P101022 &lt;dbl&gt;, P101042 &lt;dbl&gt;,
## #   P101054 &lt;dbl&gt;, P101056 &lt;dbl&gt;, P101067 &lt;dbl&gt;, P101068 &lt;dbl&gt;, P101079 &lt;dbl&gt;, P101095 &lt;dbl&gt;, P101096 &lt;dbl&gt;, P101011 &lt;dbl&gt;, P101024 &lt;dbl&gt;,
## #   P101025 &lt;dbl&gt;, P101027 &lt;dbl&gt;, P101047 &lt;dbl&gt;, P101057 &lt;dbl&gt;, P101069 &lt;dbl&gt;, P101080 &lt;dbl&gt;, P101081 &lt;dbl&gt;, P101082 &lt;dbl&gt;</code></pre>
<ul>
<li>metadata table</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="data-processing.html#cb4-1" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(<span class="st">&quot;./dataset/OmicsDataSet-Zeybel-2022.xlsx&quot;</span>, <span class="at">sheet =</span> <span class="dv">2</span>)</span>
<span id="cb4-2"><a href="data-processing.html#cb4-2" tabindex="-1"></a><span class="fu">head</span>(metadata)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 11
##   PatientID Stage  Metabolomics Proteomics GutMetagenomics OralMetagenomics LiverFatClass Gender AlcoholConsumption Smoker   Age
##   &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt;           &lt;chr&gt;            &lt;chr&gt;         &lt;chr&gt;  &lt;chr&gt;              &lt;chr&gt;  &lt;dbl&gt;
## 1 P101001   Before Send         Send       Send            Send             Severe        Male   No                 No        52
## 2 P101003   Before Send         Send       Send            Send             None          Female No                 No        31
## 3 P101004   Before Send         Send       Send            Send             Moderate      Male   Yes                No        43
## 4 P101007   Before Send         Send       Send            Send             Severe        Female No                 No        61
## 5 P101009   Before Send         Send       Send            Send             Moderate      Male   No                 Yes       51
## 6 P101010   Before Send         Send       Send            Send             Mild          Male   Yes                No        27</code></pre>
<ul>
<li>Data Preparation: ExpressionSet object</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="data-processing.html#cb6-1" tabindex="-1"></a>get_ExpressionSet <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb6-2"><a href="data-processing.html#cb6-2" tabindex="-1"></a>    x,</span>
<span id="cb6-3"><a href="data-processing.html#cb6-3" tabindex="-1"></a>    y) {</span>
<span id="cb6-4"><a href="data-processing.html#cb6-4" tabindex="-1"></a>  </span>
<span id="cb6-5"><a href="data-processing.html#cb6-5" tabindex="-1"></a>  <span class="co"># x = metadata</span></span>
<span id="cb6-6"><a href="data-processing.html#cb6-6" tabindex="-1"></a>  <span class="co"># y = profile</span></span>
<span id="cb6-7"><a href="data-processing.html#cb6-7" tabindex="-1"></a> </span>
<span id="cb6-8"><a href="data-processing.html#cb6-8" tabindex="-1"></a>  phen <span class="ot">&lt;-</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb6-9"><a href="data-processing.html#cb6-9" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(Metabolomics <span class="sc">==</span> <span class="st">&quot;Send&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-10"><a href="data-processing.html#cb6-10" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(PatientID, LiverFatClass, Gender, Smoker, Age, AlcoholConsumption)</span>
<span id="cb6-11"><a href="data-processing.html#cb6-11" tabindex="-1"></a></span>
<span id="cb6-12"><a href="data-processing.html#cb6-12" tabindex="-1"></a>  sid <span class="ot">&lt;-</span> <span class="fu">intersect</span>(phen<span class="sc">$</span>PatientID, <span class="fu">colnames</span>(y))</span>
<span id="cb6-13"><a href="data-processing.html#cb6-13" tabindex="-1"></a>  </span>
<span id="cb6-14"><a href="data-processing.html#cb6-14" tabindex="-1"></a>  prof <span class="ot">&lt;-</span> y <span class="sc">%&gt;%</span> </span>
<span id="cb6-15"><a href="data-processing.html#cb6-15" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(sid)) <span class="sc">%&gt;%</span></span>
<span id="cb6-16"><a href="data-processing.html#cb6-16" tabindex="-1"></a>    <span class="fu">data.frame</span>()</span>
<span id="cb6-17"><a href="data-processing.html#cb6-17" tabindex="-1"></a>  <span class="fu">rownames</span>(prof) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;M_&quot;</span>, y<span class="sc">$</span><span class="st">`</span><span class="at">COMP ID</span><span class="st">`</span>)</span>
<span id="cb6-18"><a href="data-processing.html#cb6-18" tabindex="-1"></a>  </span>
<span id="cb6-19"><a href="data-processing.html#cb6-19" tabindex="-1"></a>  phen <span class="ot">&lt;-</span> phen[<span class="fu">pmatch</span>(sid, phen<span class="sc">$</span>PatientID), , F] <span class="sc">%&gt;%</span></span>
<span id="cb6-20"><a href="data-processing.html#cb6-20" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;PatientID&quot;</span>)</span>
<span id="cb6-21"><a href="data-processing.html#cb6-21" tabindex="-1"></a>  </span>
<span id="cb6-22"><a href="data-processing.html#cb6-22" tabindex="-1"></a>  feat <span class="ot">&lt;-</span> y <span class="sc">%&gt;%</span> </span>
<span id="cb6-23"><a href="data-processing.html#cb6-23" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">12</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-24"><a href="data-processing.html#cb6-24" tabindex="-1"></a>    <span class="fu">as.data.frame</span>()</span>
<span id="cb6-25"><a href="data-processing.html#cb6-25" tabindex="-1"></a>  <span class="fu">rownames</span>(feat) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;M_&quot;</span>, y<span class="sc">$</span><span class="st">`</span><span class="at">COMP ID</span><span class="st">`</span>)</span>
<span id="cb6-26"><a href="data-processing.html#cb6-26" tabindex="-1"></a>  </span>
<span id="cb6-27"><a href="data-processing.html#cb6-27" tabindex="-1"></a>  <span class="co"># expressionSet</span></span>
<span id="cb6-28"><a href="data-processing.html#cb6-28" tabindex="-1"></a>  phen_ADF <span class="ot">&lt;-</span>  <span class="fu">new</span>(<span class="st">&quot;AnnotatedDataFrame&quot;</span>, <span class="at">data=</span>phen)</span>
<span id="cb6-29"><a href="data-processing.html#cb6-29" tabindex="-1"></a>  feature_ADF <span class="ot">&lt;-</span>  <span class="fu">new</span>(<span class="st">&quot;AnnotatedDataFrame&quot;</span>, <span class="at">data=</span>feat)</span>
<span id="cb6-30"><a href="data-processing.html#cb6-30" tabindex="-1"></a>  experimentData <span class="ot">&lt;-</span> <span class="fu">new</span>(</span>
<span id="cb6-31"><a href="data-processing.html#cb6-31" tabindex="-1"></a>    <span class="st">&quot;MIAME&quot;</span>,</span>
<span id="cb6-32"><a href="data-processing.html#cb6-32" tabindex="-1"></a>    <span class="at">name=</span><span class="st">&quot;Hua&quot;</span>, </span>
<span id="cb6-33"><a href="data-processing.html#cb6-33" tabindex="-1"></a>    <span class="at">lab=</span><span class="st">&quot;Xbiome Company&quot;</span>,</span>
<span id="cb6-34"><a href="data-processing.html#cb6-34" tabindex="-1"></a>    <span class="at">contact=</span><span class="st">&quot;Hua@xbiome.com&quot;</span>,</span>
<span id="cb6-35"><a href="data-processing.html#cb6-35" tabindex="-1"></a>    <span class="at">title=</span><span class="st">&quot;Metabolomics&quot;</span>,</span>
<span id="cb6-36"><a href="data-processing.html#cb6-36" tabindex="-1"></a>    <span class="at">abstract=</span><span class="st">&quot;The Mass Spectrometry ExpressionSet without imputation value&quot;</span>,</span>
<span id="cb6-37"><a href="data-processing.html#cb6-37" tabindex="-1"></a>    <span class="at">url=</span><span class="st">&quot;www.xbiome.cn&quot;</span>,</span>
<span id="cb6-38"><a href="data-processing.html#cb6-38" tabindex="-1"></a>    <span class="at">other=</span><span class="fu">list</span>(<span class="at">notes=</span><span class="st">&quot;Metabolomics&quot;</span>))</span>
<span id="cb6-39"><a href="data-processing.html#cb6-39" tabindex="-1"></a>  </span>
<span id="cb6-40"><a href="data-processing.html#cb6-40" tabindex="-1"></a>  expressionSet <span class="ot">&lt;-</span> <span class="fu">new</span>(</span>
<span id="cb6-41"><a href="data-processing.html#cb6-41" tabindex="-1"></a>    <span class="st">&quot;ExpressionSet&quot;</span>,</span>
<span id="cb6-42"><a href="data-processing.html#cb6-42" tabindex="-1"></a>    <span class="at">exprs=</span>prof,</span>
<span id="cb6-43"><a href="data-processing.html#cb6-43" tabindex="-1"></a>    <span class="at">phenoData=</span>phen_ADF,</span>
<span id="cb6-44"><a href="data-processing.html#cb6-44" tabindex="-1"></a>    <span class="at">featureData=</span>feature_ADF,</span>
<span id="cb6-45"><a href="data-processing.html#cb6-45" tabindex="-1"></a>    <span class="at">experimentData=</span>experimentData)</span>
<span id="cb6-46"><a href="data-processing.html#cb6-46" tabindex="-1"></a>  </span>
<span id="cb6-47"><a href="data-processing.html#cb6-47" tabindex="-1"></a>  <span class="fu">return</span>(expressionSet)</span>
<span id="cb6-48"><a href="data-processing.html#cb6-48" tabindex="-1"></a>}</span>
<span id="cb6-49"><a href="data-processing.html#cb6-49" tabindex="-1"></a></span>
<span id="cb6-50"><a href="data-processing.html#cb6-50" tabindex="-1"></a></span>
<span id="cb6-51"><a href="data-processing.html#cb6-51" tabindex="-1"></a>ExprSet <span class="ot">&lt;-</span> <span class="fu">get_ExpressionSet</span>(<span class="at">x =</span> metadata, <span class="at">y =</span> profile)</span>
<span id="cb6-52"><a href="data-processing.html#cb6-52" tabindex="-1"></a></span>
<span id="cb6-53"><a href="data-processing.html#cb6-53" tabindex="-1"></a>ExprSet</span></code></pre></div>
<pre><code>## ExpressionSet (storageMode: lockedEnvironment)
## assayData: 1032 features, 55 samples 
##   element names: exprs 
## protocolData: none
## phenoData
##   sampleNames: P101001 P101003 ... P101096 (55 total)
##   varLabels: LiverFatClass Gender ... AlcoholConsumption (5 total)
##   varMetadata: labelDescription
## featureData
##   featureNames: M_38768 M_38296 ... M_15581 (1032 total)
##   fvarLabels: BIOCHEMICAL SUPER PATHWAY ... SampleID HMDBID (12 total)
##   fvarMetadata: labelDescription
## experimentData: use &#39;experimentData(object)&#39;
## Annotation:</code></pre>
<ul>
<li>Data Preparation: SummarizedExperiment object</li>
</ul>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="data-processing.html#cb8-1" tabindex="-1"></a>getSEobject <span class="ot">&lt;-</span> <span class="cf">function</span>(x, y) {</span>
<span id="cb8-2"><a href="data-processing.html#cb8-2" tabindex="-1"></a></span>
<span id="cb8-3"><a href="data-processing.html#cb8-3" tabindex="-1"></a>  target <span class="ot">&lt;-</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb8-4"><a href="data-processing.html#cb8-4" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(Metabolomics <span class="sc">==</span> <span class="st">&quot;Send&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-5"><a href="data-processing.html#cb8-5" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(PatientID, LiverFatClass, Gender, Smoker, Age, AlcoholConsumption)</span>
<span id="cb8-6"><a href="data-processing.html#cb8-6" tabindex="-1"></a></span>
<span id="cb8-7"><a href="data-processing.html#cb8-7" tabindex="-1"></a>  sid <span class="ot">&lt;-</span> <span class="fu">intersect</span>(target<span class="sc">$</span>PatientID, <span class="fu">colnames</span>(y))</span>
<span id="cb8-8"><a href="data-processing.html#cb8-8" tabindex="-1"></a>  </span>
<span id="cb8-9"><a href="data-processing.html#cb8-9" tabindex="-1"></a>  features <span class="ot">&lt;-</span> y <span class="sc">%&gt;%</span> </span>
<span id="cb8-10"><a href="data-processing.html#cb8-10" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(sid)) <span class="sc">%&gt;%</span></span>
<span id="cb8-11"><a href="data-processing.html#cb8-11" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">t</span>()</span>
<span id="cb8-12"><a href="data-processing.html#cb8-12" tabindex="-1"></a>  <span class="fu">colnames</span>(features) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;M_&quot;</span>, y<span class="sc">$</span><span class="st">`</span><span class="at">COMP ID</span><span class="st">`</span>)</span>
<span id="cb8-13"><a href="data-processing.html#cb8-13" tabindex="-1"></a>  </span>
<span id="cb8-14"><a href="data-processing.html#cb8-14" tabindex="-1"></a>  target <span class="ot">&lt;-</span> target[<span class="fu">pmatch</span>(sid, target<span class="sc">$</span>PatientID), , F]</span>
<span id="cb8-15"><a href="data-processing.html#cb8-15" tabindex="-1"></a>  </span>
<span id="cb8-16"><a href="data-processing.html#cb8-16" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb8-17"><a href="data-processing.html#cb8-17" tabindex="-1"></a>                                  <span class="at">features =</span> features)</span>
<span id="cb8-18"><a href="data-processing.html#cb8-18" tabindex="-1"></a>  <span class="fu">return</span>(res)  </span>
<span id="cb8-19"><a href="data-processing.html#cb8-19" tabindex="-1"></a>}</span>
<span id="cb8-20"><a href="data-processing.html#cb8-20" tabindex="-1"></a></span>
<span id="cb8-21"><a href="data-processing.html#cb8-21" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">getSEobject</span>(metadata, profile)</span>
<span id="cb8-22"><a href="data-processing.html#cb8-22" tabindex="-1"></a></span>
<span id="cb8-23"><a href="data-processing.html#cb8-23" tabindex="-1"></a>se</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 1032 55 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(1032): M_38768 M_38296 ... M_57517 M_15581
## rowData names(0):
## colnames(55): P101001 P101003 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
<ul>
<li>Extract data for test dataset</li>
</ul>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="data-processing.html#cb10-1" tabindex="-1"></a>get_testData <span class="ot">&lt;-</span> <span class="cf">function</span>(object, <span class="at">num =</span> <span class="dv">200</span>) {</span>
<span id="cb10-2"><a href="data-processing.html#cb10-2" tabindex="-1"></a></span>
<span id="cb10-3"><a href="data-processing.html#cb10-3" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) <span class="sc">%&gt;%</span></span>
<span id="cb10-4"><a href="data-processing.html#cb10-4" tabindex="-1"></a>    <span class="fu">t</span>()</span>
<span id="cb10-5"><a href="data-processing.html#cb10-5" tabindex="-1"></a>  metadata_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object) <span class="sc">%&gt;%</span></span>
<span id="cb10-6"><a href="data-processing.html#cb10-6" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb10-7"><a href="data-processing.html#cb10-7" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;ID&quot;</span>)</span>
<span id="cb10-8"><a href="data-processing.html#cb10-8" tabindex="-1"></a>  </span>
<span id="cb10-9"><a href="data-processing.html#cb10-9" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> metadata_tab, </span>
<span id="cb10-10"><a href="data-processing.html#cb10-10" tabindex="-1"></a>                                  <span class="at">features =</span> features_tab[, <span class="dv">1</span><span class="sc">:</span>num])</span>
<span id="cb10-11"><a href="data-processing.html#cb10-11" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb10-12"><a href="data-processing.html#cb10-12" tabindex="-1"></a>}</span>
<span id="cb10-13"><a href="data-processing.html#cb10-13" tabindex="-1"></a></span>
<span id="cb10-14"><a href="data-processing.html#cb10-14" tabindex="-1"></a>se_raw <span class="ot">&lt;-</span> <span class="fu">get_testData</span>(<span class="at">object =</span> se)</span>
<span id="cb10-15"><a href="data-processing.html#cb10-15" tabindex="-1"></a>se_raw</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 200 55 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(200): M_38768 M_38296 ... M_31787 M_63361
## rowData names(0):
## colnames(55): P101001 P101003 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="data-checking" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Data Checking<a href="data-processing.html#data-checking" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Features in PomaSummarizedExperiment object must have the following criterion:</p>
<ul>
<li><p>All data values are numeric.</p></li>
<li><p>A total of 0 (0%) missing values were detected.</p></li>
</ul>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="data-processing.html#cb12-1" tabindex="-1"></a>CheckData <span class="ot">&lt;-</span> <span class="cf">function</span>(object) {</span>
<span id="cb12-2"><a href="data-processing.html#cb12-2" tabindex="-1"></a>  </span>
<span id="cb12-3"><a href="data-processing.html#cb12-3" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object)</span>
<span id="cb12-4"><a href="data-processing.html#cb12-4" tabindex="-1"></a>  </span>
<span id="cb12-5"><a href="data-processing.html#cb12-5" tabindex="-1"></a>  <span class="co"># numeric &amp; missing values</span></span>
<span id="cb12-6"><a href="data-processing.html#cb12-6" tabindex="-1"></a>  int_mat <span class="ot">&lt;-</span> features_tab</span>
<span id="cb12-7"><a href="data-processing.html#cb12-7" tabindex="-1"></a>  rowNms <span class="ot">&lt;-</span> <span class="fu">rownames</span>(int_mat)</span>
<span id="cb12-8"><a href="data-processing.html#cb12-8" tabindex="-1"></a>  colNms <span class="ot">&lt;-</span> <span class="fu">colnames</span>(int_mat)</span>
<span id="cb12-9"><a href="data-processing.html#cb12-9" tabindex="-1"></a>  naNms <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">is.na</span>(int_mat))</span>
<span id="cb12-10"><a href="data-processing.html#cb12-10" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(int_mat)) {</span>
<span id="cb12-11"><a href="data-processing.html#cb12-11" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">class</span>(int_mat[, i]) <span class="sc">==</span> <span class="st">&quot;integer64&quot;</span>) {</span>
<span id="cb12-12"><a href="data-processing.html#cb12-12" tabindex="-1"></a>      int_mat[, i] <span class="ot">&lt;-</span> <span class="fu">as.double</span>(int_mat[, i])</span>
<span id="cb12-13"><a href="data-processing.html#cb12-13" tabindex="-1"></a>    }</span>
<span id="cb12-14"><a href="data-processing.html#cb12-14" tabindex="-1"></a>  }</span>
<span id="cb12-15"><a href="data-processing.html#cb12-15" tabindex="-1"></a>  </span>
<span id="cb12-16"><a href="data-processing.html#cb12-16" tabindex="-1"></a>  num_mat <span class="ot">&lt;-</span> <span class="fu">apply</span>(int_mat, <span class="dv">2</span>, as.numeric)</span>
<span id="cb12-17"><a href="data-processing.html#cb12-17" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">sum</span>(<span class="fu">is.na</span>(num_mat)) <span class="sc">&gt;</span> naNms) {</span>
<span id="cb12-18"><a href="data-processing.html#cb12-18" tabindex="-1"></a>    num_mat <span class="ot">&lt;-</span> <span class="fu">apply</span>(int_mat, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;,&quot;</span>,  <span class="st">&quot;&quot;</span>, x)))</span>
<span id="cb12-19"><a href="data-processing.html#cb12-19" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">sum</span>(<span class="fu">is.na</span>(num_mat)) <span class="sc">&gt;</span> naNms) {</span>
<span id="cb12-20"><a href="data-processing.html#cb12-20" tabindex="-1"></a>      <span class="fu">message</span>(<span class="st">&quot;&lt;font color=</span><span class="sc">\&quot;</span><span class="st">red</span><span class="sc">\&quot;</span><span class="st">&gt;Non-numeric values were found and replaced by NA.&lt;/font&gt;&quot;</span>)</span>
<span id="cb12-21"><a href="data-processing.html#cb12-21" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb12-22"><a href="data-processing.html#cb12-22" tabindex="-1"></a>      <span class="fu">message</span>(<span class="st">&quot;All data values are numeric.&quot;</span>)</span>
<span id="cb12-23"><a href="data-processing.html#cb12-23" tabindex="-1"></a>    }</span>
<span id="cb12-24"><a href="data-processing.html#cb12-24" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb12-25"><a href="data-processing.html#cb12-25" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">&quot;All data values are numeric.&quot;</span>)</span>
<span id="cb12-26"><a href="data-processing.html#cb12-26" tabindex="-1"></a>  }</span>
<span id="cb12-27"><a href="data-processing.html#cb12-27" tabindex="-1"></a>  </span>
<span id="cb12-28"><a href="data-processing.html#cb12-28" tabindex="-1"></a>  int_mat <span class="ot">&lt;-</span> num_mat</span>
<span id="cb12-29"><a href="data-processing.html#cb12-29" tabindex="-1"></a>  <span class="fu">rownames</span>(int_mat) <span class="ot">&lt;-</span> rowNms</span>
<span id="cb12-30"><a href="data-processing.html#cb12-30" tabindex="-1"></a>  <span class="fu">colnames</span>(int_mat) <span class="ot">&lt;-</span> colNms</span>
<span id="cb12-31"><a href="data-processing.html#cb12-31" tabindex="-1"></a>  varCol <span class="ot">&lt;-</span> <span class="fu">apply</span>(int_mat, <span class="dv">2</span>, var, <span class="at">na.rm =</span> T)</span>
<span id="cb12-32"><a href="data-processing.html#cb12-32" tabindex="-1"></a>  constCol <span class="ot">&lt;-</span> (varCol <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> <span class="fu">is.na</span>(varCol))</span>
<span id="cb12-33"><a href="data-processing.html#cb12-33" tabindex="-1"></a>  constNum <span class="ot">&lt;-</span> <span class="fu">sum</span>(constCol, <span class="at">na.rm =</span> T)</span>
<span id="cb12-34"><a href="data-processing.html#cb12-34" tabindex="-1"></a>  <span class="cf">if</span> (constNum <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb12-35"><a href="data-processing.html#cb12-35" tabindex="-1"></a>    <span class="fu">message</span>(<span class="fu">paste</span>(<span class="st">&quot;&lt;font color=</span><span class="sc">\&quot;</span><span class="st">red</span><span class="sc">\&quot;</span><span class="st">&gt;&quot;</span>, constNum, </span>
<span id="cb12-36"><a href="data-processing.html#cb12-36" tabindex="-1"></a>      <span class="st">&quot;features with a constant or single value across samples were found and deleted.&lt;/font&gt;&quot;</span>))</span>
<span id="cb12-37"><a href="data-processing.html#cb12-37" tabindex="-1"></a>    int_mat <span class="ot">&lt;-</span> int_mat[, <span class="sc">!</span>constCol, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb12-38"><a href="data-processing.html#cb12-38" tabindex="-1"></a>  }</span>
<span id="cb12-39"><a href="data-processing.html#cb12-39" tabindex="-1"></a>  </span>
<span id="cb12-40"><a href="data-processing.html#cb12-40" tabindex="-1"></a>  totalCount <span class="ot">&lt;-</span> <span class="fu">nrow</span>(int_mat) <span class="sc">*</span> <span class="fu">ncol</span>(int_mat)</span>
<span id="cb12-41"><a href="data-processing.html#cb12-41" tabindex="-1"></a>  naCount <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">is.na</span>(int_mat))</span>
<span id="cb12-42"><a href="data-processing.html#cb12-42" tabindex="-1"></a>  naPercent <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> naCount<span class="sc">/</span>totalCount, <span class="dv">1</span>)</span>
<span id="cb12-43"><a href="data-processing.html#cb12-43" tabindex="-1"></a></span>
<span id="cb12-44"><a href="data-processing.html#cb12-44" tabindex="-1"></a>  <span class="fu">message</span>(<span class="fu">paste</span>(<span class="st">&quot;A total of &quot;</span>, naCount, <span class="st">&quot; (&quot;</span>, naPercent, </span>
<span id="cb12-45"><a href="data-processing.html#cb12-45" tabindex="-1"></a>    <span class="st">&quot;%) missing values were detected.&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))  </span>
<span id="cb12-46"><a href="data-processing.html#cb12-46" tabindex="-1"></a>  </span>
<span id="cb12-47"><a href="data-processing.html#cb12-47" tabindex="-1"></a>  <span class="co"># save int_mat into se object </span></span>
<span id="cb12-48"><a href="data-processing.html#cb12-48" tabindex="-1"></a>  target <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object) <span class="sc">%&gt;%</span></span>
<span id="cb12-49"><a href="data-processing.html#cb12-49" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb12-50"><a href="data-processing.html#cb12-50" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb12-51"><a href="data-processing.html#cb12-51" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb12-52"><a href="data-processing.html#cb12-52" tabindex="-1"></a>                                  <span class="at">features =</span> <span class="fu">t</span>(int_mat))</span>
<span id="cb12-53"><a href="data-processing.html#cb12-53" tabindex="-1"></a>  </span>
<span id="cb12-54"><a href="data-processing.html#cb12-54" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb12-55"><a href="data-processing.html#cb12-55" tabindex="-1"></a>}</span>
<span id="cb12-56"><a href="data-processing.html#cb12-56" tabindex="-1"></a></span>
<span id="cb12-57"><a href="data-processing.html#cb12-57" tabindex="-1"></a>se_check <span class="ot">&lt;-</span> <span class="fu">CheckData</span>(<span class="at">object =</span> se_raw)</span></code></pre></div>
<pre><code>## All data values are numeric.</code></pre>
<pre><code>## A total of 1146 (10.4%) missing values were detected.</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="data-processing.html#cb15-1" tabindex="-1"></a>se_check</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 200 55 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(200): M_38768 M_38296 ... M_31787 M_63361
## rowData names(0):
## colnames(55): P101001 P101003 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="data-trimming" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Data Trimming<a href="data-processing.html#data-trimming" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Trimming features whose prevalence is less than the threshold (default: 80%) before missing value imputation. Firstly, dropping features per group beyond cutoff. Then, combining all the features from each group into the remained features.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="data-processing.html#cb17-1" tabindex="-1"></a>trim_feature <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb17-2"><a href="data-processing.html#cb17-2" tabindex="-1"></a>    object, </span>
<span id="cb17-3"><a href="data-processing.html#cb17-3" tabindex="-1"></a>    group,</span>
<span id="cb17-4"><a href="data-processing.html#cb17-4" tabindex="-1"></a>    <span class="at">prevalence =</span> <span class="fl">0.8</span>) {</span>
<span id="cb17-5"><a href="data-processing.html#cb17-5" tabindex="-1"></a>  </span>
<span id="cb17-6"><a href="data-processing.html#cb17-6" tabindex="-1"></a>  <span class="co"># object = se_check</span></span>
<span id="cb17-7"><a href="data-processing.html#cb17-7" tabindex="-1"></a>  <span class="co"># group = &quot;group&quot;</span></span>
<span id="cb17-8"><a href="data-processing.html#cb17-8" tabindex="-1"></a>  <span class="co"># prevalence = 0.8</span></span>
<span id="cb17-9"><a href="data-processing.html#cb17-9" tabindex="-1"></a></span>
<span id="cb17-10"><a href="data-processing.html#cb17-10" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) <span class="sc">%&gt;%</span></span>
<span id="cb17-11"><a href="data-processing.html#cb17-11" tabindex="-1"></a>    <span class="fu">t</span>() <span class="sc">%&gt;%</span></span>
<span id="cb17-12"><a href="data-processing.html#cb17-12" tabindex="-1"></a>    <span class="fu">as.data.frame</span>()</span>
<span id="cb17-13"><a href="data-processing.html#cb17-13" tabindex="-1"></a>  metadata_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object) <span class="sc">%&gt;%</span></span>
<span id="cb17-14"><a href="data-processing.html#cb17-14" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb17-15"><a href="data-processing.html#cb17-15" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;ID&quot;</span>)</span>
<span id="cb17-16"><a href="data-processing.html#cb17-16" tabindex="-1"></a>  </span>
<span id="cb17-17"><a href="data-processing.html#cb17-17" tabindex="-1"></a>  trim_FeatureOrSample <span class="ot">&lt;-</span> <span class="cf">function</span>(x, nRow, threshold) {</span>
<span id="cb17-18"><a href="data-processing.html#cb17-18" tabindex="-1"></a>  </span>
<span id="cb17-19"><a href="data-processing.html#cb17-19" tabindex="-1"></a>    <span class="co"># x = features_tab</span></span>
<span id="cb17-20"><a href="data-processing.html#cb17-20" tabindex="-1"></a>    <span class="co"># nRow = 2</span></span>
<span id="cb17-21"><a href="data-processing.html#cb17-21" tabindex="-1"></a>    <span class="co"># threshold = prevalence</span></span>
<span id="cb17-22"><a href="data-processing.html#cb17-22" tabindex="-1"></a>    </span>
<span id="cb17-23"><a href="data-processing.html#cb17-23" tabindex="-1"></a>    df_occ <span class="ot">&lt;-</span> <span class="fu">apply</span>(x, nRow, <span class="cf">function</span>(x) {</span>
<span id="cb17-24"><a href="data-processing.html#cb17-24" tabindex="-1"></a>      <span class="fu">length</span>(x[<span class="fu">c</span>(<span class="fu">which</span>(<span class="sc">!</span><span class="fu">is.na</span>(x) <span class="sc">&amp;</span> x <span class="sc">!=</span> <span class="dv">0</span>))]) <span class="sc">/</span> <span class="fu">length</span>(x)</span>
<span id="cb17-25"><a href="data-processing.html#cb17-25" tabindex="-1"></a>    }) <span class="sc">%&gt;%</span></span>
<span id="cb17-26"><a href="data-processing.html#cb17-26" tabindex="-1"></a>      <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb17-27"><a href="data-processing.html#cb17-27" tabindex="-1"></a>      stats<span class="sc">::</span><span class="fu">setNames</span>(<span class="st">&quot;Occ&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-28"><a href="data-processing.html#cb17-28" tabindex="-1"></a>      tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;type&quot;</span>)</span>
<span id="cb17-29"><a href="data-processing.html#cb17-29" tabindex="-1"></a>    <span class="cf">if</span>(nRow <span class="sc">==</span> <span class="dv">1</span>){</span>
<span id="cb17-30"><a href="data-processing.html#cb17-30" tabindex="-1"></a>      <span class="fu">rownames</span>(df_occ) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(x)</span>
<span id="cb17-31"><a href="data-processing.html#cb17-31" tabindex="-1"></a>    }<span class="cf">else</span>{</span>
<span id="cb17-32"><a href="data-processing.html#cb17-32" tabindex="-1"></a>      <span class="fu">rownames</span>(df_occ) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(x)</span>
<span id="cb17-33"><a href="data-processing.html#cb17-33" tabindex="-1"></a>    }</span>
<span id="cb17-34"><a href="data-processing.html#cb17-34" tabindex="-1"></a>    df_KEEP <span class="ot">&lt;-</span> <span class="fu">apply</span>(df_occ <span class="sc">&gt;</span> threshold, <span class="dv">1</span>, all) <span class="sc">%&gt;%</span></span>
<span id="cb17-35"><a href="data-processing.html#cb17-35" tabindex="-1"></a>      <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb17-36"><a href="data-processing.html#cb17-36" tabindex="-1"></a>      stats<span class="sc">::</span><span class="fu">setNames</span>(<span class="st">&quot;Status&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-37"><a href="data-processing.html#cb17-37" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">filter</span>(Status)</span>
<span id="cb17-38"><a href="data-processing.html#cb17-38" tabindex="-1"></a>  </span>
<span id="cb17-39"><a href="data-processing.html#cb17-39" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">rownames</span>(df_KEEP))</span>
<span id="cb17-40"><a href="data-processing.html#cb17-40" tabindex="-1"></a>  }</span>
<span id="cb17-41"><a href="data-processing.html#cb17-41" tabindex="-1"></a>  </span>
<span id="cb17-42"><a href="data-processing.html#cb17-42" tabindex="-1"></a>  <span class="fu">colnames</span>(metadata_tab)[<span class="fu">colnames</span>(metadata_tab) <span class="sc">==</span> group] <span class="ot">&lt;-</span> <span class="st">&quot;TempGroup&quot;</span></span>
<span id="cb17-43"><a href="data-processing.html#cb17-43" tabindex="-1"></a>  </span>
<span id="cb17-44"><a href="data-processing.html#cb17-44" tabindex="-1"></a>  group_levels <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">unique</span>(metadata_tab<span class="sc">$</span>TempGroup))</span>
<span id="cb17-45"><a href="data-processing.html#cb17-45" tabindex="-1"></a>  </span>
<span id="cb17-46"><a href="data-processing.html#cb17-46" tabindex="-1"></a>  remain_features <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb17-47"><a href="data-processing.html#cb17-47" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(group_levels)) {</span>
<span id="cb17-48"><a href="data-processing.html#cb17-48" tabindex="-1"></a>    temp_SampleID <span class="ot">&lt;-</span> metadata_tab[metadata_tab<span class="sc">$</span>TempGroup <span class="sc">==</span> group_levels[i], , ]</span>
<span id="cb17-49"><a href="data-processing.html#cb17-49" tabindex="-1"></a>    temp_feature_tab <span class="ot">&lt;-</span> features_tab[<span class="fu">pmatch</span>(temp_SampleID<span class="sc">$</span>ID, <span class="fu">rownames</span>(features_tab)), ,]</span>
<span id="cb17-50"><a href="data-processing.html#cb17-50" tabindex="-1"></a>    temp_feature <span class="ot">&lt;-</span> <span class="fu">trim_FeatureOrSample</span>(temp_feature_tab, <span class="dv">2</span>, prevalence)</span>
<span id="cb17-51"><a href="data-processing.html#cb17-51" tabindex="-1"></a>    remain_features <span class="ot">&lt;-</span> <span class="fu">c</span>(remain_features, temp_feature)</span>
<span id="cb17-52"><a href="data-processing.html#cb17-52" tabindex="-1"></a>  }</span>
<span id="cb17-53"><a href="data-processing.html#cb17-53" tabindex="-1"></a>  </span>
<span id="cb17-54"><a href="data-processing.html#cb17-54" tabindex="-1"></a>  features_tab_final <span class="ot">&lt;-</span> features_tab[, <span class="fu">pmatch</span>(<span class="fu">unique</span>(remain_features), <span class="fu">colnames</span>(features_tab)), ] <span class="sc">%&gt;%</span></span>
<span id="cb17-55"><a href="data-processing.html#cb17-55" tabindex="-1"></a>    <span class="fu">as.matrix</span>()</span>
<span id="cb17-56"><a href="data-processing.html#cb17-56" tabindex="-1"></a>  <span class="fu">colnames</span>(metadata_tab)[<span class="fu">colnames</span>(metadata_tab) <span class="sc">==</span> <span class="st">&quot;TempGroup&quot;</span>] <span class="ot">&lt;-</span> group</span>
<span id="cb17-57"><a href="data-processing.html#cb17-57" tabindex="-1"></a>  </span>
<span id="cb17-58"><a href="data-processing.html#cb17-58" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> metadata_tab, </span>
<span id="cb17-59"><a href="data-processing.html#cb17-59" tabindex="-1"></a>                                  <span class="at">features =</span> features_tab_final)</span>
<span id="cb17-60"><a href="data-processing.html#cb17-60" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb17-61"><a href="data-processing.html#cb17-61" tabindex="-1"></a>}</span>
<span id="cb17-62"><a href="data-processing.html#cb17-62" tabindex="-1"></a></span>
<span id="cb17-63"><a href="data-processing.html#cb17-63" tabindex="-1"></a>se_trim <span class="ot">&lt;-</span> <span class="fu">trim_feature</span>(<span class="at">object =</span> se_check, <span class="at">group =</span> <span class="st">&quot;group&quot;</span>)</span>
<span id="cb17-64"><a href="data-processing.html#cb17-64" tabindex="-1"></a>se_trim</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 173 55 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(173): M_38768 M_38296 ... M_38276 M_63361
## rowData names(0):
## colnames(55): P101001 P101003 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="missing-value-imputation" class="section level2 hasAnchor" number="2.5">
<h2><span class="header-section-number">2.5</span> Missing value imputation<a href="data-processing.html#missing-value-imputation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>“none”: all missing values will be replaced by zero.</p>
<p>“LOD”: specific Limit Of Detection which provides by user.</p>
<p>“half_min”: half minimal values across samples except zero.</p>
<p>“median”: median values across samples except zero.</p>
<p>“mean”: mean values across samples except zero.</p>
<p>“min”: minimal values across samples except zero.</p>
<p>“knn”: k-nearest neighbors samples.</p>
<p>“rf”: nonparametric missing value imputation using Random Forest.</p>
<p>“QRILC”: missing values imputation based quantile regression. (default: “none”).</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="data-processing.html#cb19-1" tabindex="-1"></a>impute_abundance <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb19-2"><a href="data-processing.html#cb19-2" tabindex="-1"></a>    object,</span>
<span id="cb19-3"><a href="data-processing.html#cb19-3" tabindex="-1"></a>    group,</span>
<span id="cb19-4"><a href="data-processing.html#cb19-4" tabindex="-1"></a>    <span class="at">ZerosAsNA =</span> <span class="cn">FALSE</span>,</span>
<span id="cb19-5"><a href="data-processing.html#cb19-5" tabindex="-1"></a>    <span class="at">RemoveNA =</span> <span class="cn">TRUE</span>,</span>
<span id="cb19-6"><a href="data-processing.html#cb19-6" tabindex="-1"></a>    <span class="at">prevalence =</span> <span class="fl">0.5</span>,</span>
<span id="cb19-7"><a href="data-processing.html#cb19-7" tabindex="-1"></a>    <span class="at">method =</span> <span class="fu">c</span>(<span class="st">&quot;none&quot;</span>, <span class="st">&quot;LOD&quot;</span>, <span class="st">&quot;half_min&quot;</span>, <span class="st">&quot;median&quot;</span>,</span>
<span id="cb19-8"><a href="data-processing.html#cb19-8" tabindex="-1"></a>               <span class="st">&quot;mean&quot;</span>, <span class="st">&quot;min&quot;</span>, <span class="st">&quot;knn&quot;</span>, <span class="st">&quot;rf&quot;</span>, <span class="st">&quot;QRILC&quot;</span>),</span>
<span id="cb19-9"><a href="data-processing.html#cb19-9" tabindex="-1"></a>    <span class="at">LOD =</span> <span class="cn">NULL</span>) {</span>
<span id="cb19-10"><a href="data-processing.html#cb19-10" tabindex="-1"></a></span>
<span id="cb19-11"><a href="data-processing.html#cb19-11" tabindex="-1"></a>  <span class="co"># object = se_check</span></span>
<span id="cb19-12"><a href="data-processing.html#cb19-12" tabindex="-1"></a>  <span class="co"># group = &quot;group&quot;</span></span>
<span id="cb19-13"><a href="data-processing.html#cb19-13" tabindex="-1"></a>  <span class="co"># ZerosAsNA = TRUE</span></span>
<span id="cb19-14"><a href="data-processing.html#cb19-14" tabindex="-1"></a>  <span class="co"># RemoveNA = TRUE</span></span>
<span id="cb19-15"><a href="data-processing.html#cb19-15" tabindex="-1"></a>  <span class="co"># prevalence = 0.5</span></span>
<span id="cb19-16"><a href="data-processing.html#cb19-16" tabindex="-1"></a>  <span class="co"># method = &quot;knn&quot;</span></span>
<span id="cb19-17"><a href="data-processing.html#cb19-17" tabindex="-1"></a></span>
<span id="cb19-18"><a href="data-processing.html#cb19-18" tabindex="-1"></a>  <span class="cf">if</span> (base<span class="sc">::</span><span class="fu">missing</span>(object)) {</span>
<span id="cb19-19"><a href="data-processing.html#cb19-19" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;object argument is empty!&quot;</span>)</span>
<span id="cb19-20"><a href="data-processing.html#cb19-20" tabindex="-1"></a>  }</span>
<span id="cb19-21"><a href="data-processing.html#cb19-21" tabindex="-1"></a></span>
<span id="cb19-22"><a href="data-processing.html#cb19-22" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span>methods<span class="sc">::</span><span class="fu">is</span>(object, <span class="st">&quot;SummarizedExperiment&quot;</span>)) {</span>
<span id="cb19-23"><a href="data-processing.html#cb19-23" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;object is not either a phyloseq or SummarizedExperiment object.&quot;</span>)</span>
<span id="cb19-24"><a href="data-processing.html#cb19-24" tabindex="-1"></a>  }</span>
<span id="cb19-25"><a href="data-processing.html#cb19-25" tabindex="-1"></a></span>
<span id="cb19-26"><a href="data-processing.html#cb19-26" tabindex="-1"></a>  method <span class="ot">&lt;-</span> <span class="fu">match.arg</span>(</span>
<span id="cb19-27"><a href="data-processing.html#cb19-27" tabindex="-1"></a>    method, <span class="fu">c</span>(<span class="st">&quot;none&quot;</span>, <span class="st">&quot;LOD&quot;</span>, <span class="st">&quot;half_min&quot;</span>, <span class="st">&quot;median&quot;</span>,</span>
<span id="cb19-28"><a href="data-processing.html#cb19-28" tabindex="-1"></a>              <span class="st">&quot;mean&quot;</span>, <span class="st">&quot;min&quot;</span>, <span class="st">&quot;knn&quot;</span>, <span class="st">&quot;rf&quot;</span>, <span class="st">&quot;QRILC&quot;</span>)</span>
<span id="cb19-29"><a href="data-processing.html#cb19-29" tabindex="-1"></a>  )</span>
<span id="cb19-30"><a href="data-processing.html#cb19-30" tabindex="-1"></a></span>
<span id="cb19-31"><a href="data-processing.html#cb19-31" tabindex="-1"></a>  <span class="cf">if</span> (base<span class="sc">::</span><span class="fu">missing</span>(method)) {</span>
<span id="cb19-32"><a href="data-processing.html#cb19-32" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">&quot;method argument is empty! KNN will be used&quot;</span>)</span>
<span id="cb19-33"><a href="data-processing.html#cb19-33" tabindex="-1"></a>  }</span>
<span id="cb19-34"><a href="data-processing.html#cb19-34" tabindex="-1"></a></span>
<span id="cb19-35"><a href="data-processing.html#cb19-35" tabindex="-1"></a>  <span class="co"># profile: row-&gt;samples; col-&gt;features</span></span>
<span id="cb19-36"><a href="data-processing.html#cb19-36" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">all</span>(<span class="sc">!</span><span class="fu">is.null</span>(object), <span class="fu">inherits</span>(object, <span class="st">&quot;SummarizedExperiment&quot;</span>))) {</span>
<span id="cb19-37"><a href="data-processing.html#cb19-37" tabindex="-1"></a>    <span class="co"># sample table &amp; profile table</span></span>
<span id="cb19-38"><a href="data-processing.html#cb19-38" tabindex="-1"></a>    sam_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object) <span class="sc">%&gt;%</span></span>
<span id="cb19-39"><a href="data-processing.html#cb19-39" tabindex="-1"></a>      <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-40"><a href="data-processing.html#cb19-40" tabindex="-1"></a>      tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;TempRowNames&quot;</span>)</span>
<span id="cb19-41"><a href="data-processing.html#cb19-41" tabindex="-1"></a>    prf_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) <span class="sc">%&gt;%</span></span>
<span id="cb19-42"><a href="data-processing.html#cb19-42" tabindex="-1"></a>      <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-43"><a href="data-processing.html#cb19-43" tabindex="-1"></a>      <span class="fu">t</span>()</span>
<span id="cb19-44"><a href="data-processing.html#cb19-44" tabindex="-1"></a>  }</span>
<span id="cb19-45"><a href="data-processing.html#cb19-45" tabindex="-1"></a></span>
<span id="cb19-46"><a href="data-processing.html#cb19-46" tabindex="-1"></a>  group_index <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">colnames</span>(sam_tab) <span class="sc">==</span> group)</span>
<span id="cb19-47"><a href="data-processing.html#cb19-47" tabindex="-1"></a>  samples_groups <span class="ot">&lt;-</span> sam_tab[, group_index]</span>
<span id="cb19-48"><a href="data-processing.html#cb19-48" tabindex="-1"></a>  to_imp_data <span class="ot">&lt;-</span> prf_tab <span class="sc">%&gt;%</span> <span class="fu">as.matrix</span>()</span>
<span id="cb19-49"><a href="data-processing.html#cb19-49" tabindex="-1"></a></span>
<span id="cb19-50"><a href="data-processing.html#cb19-50" tabindex="-1"></a>  <span class="cf">if</span> (ZerosAsNA) {</span>
<span id="cb19-51"><a href="data-processing.html#cb19-51" tabindex="-1"></a>    to_imp_data[to_imp_data <span class="sc">==</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb19-52"><a href="data-processing.html#cb19-52" tabindex="-1"></a>    to_imp_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">cbind</span>(<span class="at">Group =</span> samples_groups, to_imp_data))</span>
<span id="cb19-53"><a href="data-processing.html#cb19-53" tabindex="-1"></a>    <span class="fu">colnames</span>(to_imp_data)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(to_imp_data)] <span class="ot">&lt;-</span> <span class="fu">colnames</span>(prf_tab)</span>
<span id="cb19-54"><a href="data-processing.html#cb19-54" tabindex="-1"></a></span>
<span id="cb19-55"><a href="data-processing.html#cb19-55" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb19-56"><a href="data-processing.html#cb19-56" tabindex="-1"></a>    to_imp_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">cbind</span>(<span class="at">Group =</span> samples_groups, to_imp_data))</span>
<span id="cb19-57"><a href="data-processing.html#cb19-57" tabindex="-1"></a>    <span class="fu">colnames</span>(to_imp_data)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(to_imp_data)] <span class="ot">&lt;-</span> <span class="fu">colnames</span>(prf_tab)</span>
<span id="cb19-58"><a href="data-processing.html#cb19-58" tabindex="-1"></a>  }</span>
<span id="cb19-59"><a href="data-processing.html#cb19-59" tabindex="-1"></a></span>
<span id="cb19-60"><a href="data-processing.html#cb19-60" tabindex="-1"></a>  percent_na <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">is.na</span>(to_imp_data))</span>
<span id="cb19-61"><a href="data-processing.html#cb19-61" tabindex="-1"></a>  <span class="cf">if</span> (percent_na <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb19-62"><a href="data-processing.html#cb19-62" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">&quot;No missing values detected in your data&quot;</span>)</span>
<span id="cb19-63"><a href="data-processing.html#cb19-63" tabindex="-1"></a>    <span class="cf">if</span> (method <span class="sc">!=</span> <span class="st">&quot;none&quot;</span>) {</span>
<span id="cb19-64"><a href="data-processing.html#cb19-64" tabindex="-1"></a>      method <span class="ot">&lt;-</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb19-65"><a href="data-processing.html#cb19-65" tabindex="-1"></a>    }</span>
<span id="cb19-66"><a href="data-processing.html#cb19-66" tabindex="-1"></a>  }</span>
<span id="cb19-67"><a href="data-processing.html#cb19-67" tabindex="-1"></a></span>
<span id="cb19-68"><a href="data-processing.html#cb19-68" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">isTRUE</span>(RemoveNA)) {</span>
<span id="cb19-69"><a href="data-processing.html#cb19-69" tabindex="-1"></a>    count_NA <span class="ot">&lt;-</span> stats<span class="sc">::</span><span class="fu">aggregate</span>(</span>
<span id="cb19-70"><a href="data-processing.html#cb19-70" tabindex="-1"></a>          . <span class="sc">~</span> Group,</span>
<span id="cb19-71"><a href="data-processing.html#cb19-71" tabindex="-1"></a>          <span class="at">data =</span> to_imp_data,</span>
<span id="cb19-72"><a href="data-processing.html#cb19-72" tabindex="-1"></a>          <span class="cf">function</span>(x) {(<span class="fu">sum</span>(<span class="fu">is.na</span>(x)) <span class="sc">/</span> (<span class="fu">sum</span>(<span class="fu">is.na</span>(x)) <span class="sc">+</span> <span class="fu">sum</span>(<span class="sc">!</span><span class="fu">is.na</span>(x))) ) },</span>
<span id="cb19-73"><a href="data-processing.html#cb19-73" tabindex="-1"></a>          <span class="at">na.action =</span> <span class="cn">NULL</span>)</span>
<span id="cb19-74"><a href="data-processing.html#cb19-74" tabindex="-1"></a>    count_NA <span class="ot">&lt;-</span> count_NA <span class="sc">%&gt;%</span></span>
<span id="cb19-75"><a href="data-processing.html#cb19-75" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>Group)</span>
<span id="cb19-76"><a href="data-processing.html#cb19-76" tabindex="-1"></a>    correct_names <span class="ot">&lt;-</span> <span class="fu">names</span>(count_NA)</span>
<span id="cb19-77"><a href="data-processing.html#cb19-77" tabindex="-1"></a>    supress <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">as.data.frame</span>(<span class="fu">lapply</span>(count_NA, <span class="cf">function</span>(x) <span class="fu">any</span>(x <span class="sc">&gt;</span> prevalence))))</span>
<span id="cb19-78"><a href="data-processing.html#cb19-78" tabindex="-1"></a>    <span class="fu">names</span>(supress) <span class="ot">&lt;-</span> correct_names</span>
<span id="cb19-79"><a href="data-processing.html#cb19-79" tabindex="-1"></a>    correct_names <span class="ot">&lt;-</span> <span class="fu">names</span>(supress[supress <span class="sc">==</span> <span class="st">&quot;FALSE&quot;</span>])</span>
<span id="cb19-80"><a href="data-processing.html#cb19-80" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> to_imp_data[, <span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(to_imp_data)][<span class="sc">!</span>supress]</span>
<span id="cb19-81"><a href="data-processing.html#cb19-81" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">sapply</span>(depurdata, <span class="cf">function</span>(x) <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(x)))</span>
<span id="cb19-82"><a href="data-processing.html#cb19-82" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb19-83"><a href="data-processing.html#cb19-83" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> to_imp_data[, <span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(to_imp_data)]</span>
<span id="cb19-84"><a href="data-processing.html#cb19-84" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">sapply</span>(depurdata, <span class="cf">function</span>(x) <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(x)))</span>
<span id="cb19-85"><a href="data-processing.html#cb19-85" tabindex="-1"></a>    correct_names <span class="ot">&lt;-</span> <span class="fu">colnames</span>(prf_tab)</span>
<span id="cb19-86"><a href="data-processing.html#cb19-86" tabindex="-1"></a>  }</span>
<span id="cb19-87"><a href="data-processing.html#cb19-87" tabindex="-1"></a></span>
<span id="cb19-88"><a href="data-processing.html#cb19-88" tabindex="-1"></a>  <span class="co"># Row-&gt;feature;Col-&gt;sample</span></span>
<span id="cb19-89"><a href="data-processing.html#cb19-89" tabindex="-1"></a>  <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;none&quot;</span>) {</span>
<span id="cb19-90"><a href="data-processing.html#cb19-90" tabindex="-1"></a>    depurdata[<span class="fu">is.na</span>(depurdata)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb19-91"><a href="data-processing.html#cb19-91" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;LOD&quot;</span>) {</span>
<span id="cb19-92"><a href="data-processing.html#cb19-92" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.null</span>(LOD)) {</span>
<span id="cb19-93"><a href="data-processing.html#cb19-93" tabindex="-1"></a>      <span class="fu">message</span>(<span class="st">&quot;No LOD provided, regard one-tenth mininal value as LOD&quot;</span>)</span>
<span id="cb19-94"><a href="data-processing.html#cb19-94" tabindex="-1"></a>      depurdata_withoutNA <span class="ot">&lt;-</span> depurdata[<span class="sc">!</span><span class="fu">is.na</span>(depurdata)]</span>
<span id="cb19-95"><a href="data-processing.html#cb19-95" tabindex="-1"></a>      LOD <span class="ot">&lt;-</span> <span class="fu">min</span>(depurdata_withoutNA[depurdata_withoutNA <span class="sc">!=</span> <span class="dv">0</span>]) <span class="sc">/</span> <span class="dv">10</span></span>
<span id="cb19-96"><a href="data-processing.html#cb19-96" tabindex="-1"></a>    }</span>
<span id="cb19-97"><a href="data-processing.html#cb19-97" tabindex="-1"></a>    depurdata[<span class="fu">is.na</span>(depurdata)] <span class="ot">&lt;-</span> LOD</span>
<span id="cb19-98"><a href="data-processing.html#cb19-98" tabindex="-1"></a>    depurdata[depurdata <span class="sc">==</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> LOD</span>
<span id="cb19-99"><a href="data-processing.html#cb19-99" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;half_min&quot;</span>) {</span>
<span id="cb19-100"><a href="data-processing.html#cb19-100" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">apply</span>(depurdata, <span class="dv">2</span>, <span class="cf">function</span>(x) {</span>
<span id="cb19-101"><a href="data-processing.html#cb19-101" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">is.numeric</span>(x)) <span class="fu">ifelse</span>(<span class="fu">is.na</span>(x), <span class="fu">min</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)<span class="sc">/</span><span class="dv">2</span>, x) <span class="cf">else</span> x})</span>
<span id="cb19-102"><a href="data-processing.html#cb19-102" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;median&quot;</span>) {</span>
<span id="cb19-103"><a href="data-processing.html#cb19-103" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">apply</span>(depurdata, <span class="dv">2</span>, <span class="cf">function</span>(x) {</span>
<span id="cb19-104"><a href="data-processing.html#cb19-104" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">is.numeric</span>(x)) <span class="fu">ifelse</span>(<span class="fu">is.na</span>(x), <span class="fu">median</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), x) <span class="cf">else</span> x})</span>
<span id="cb19-105"><a href="data-processing.html#cb19-105" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;mean&quot;</span>) {</span>
<span id="cb19-106"><a href="data-processing.html#cb19-106" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">apply</span>(depurdata, <span class="dv">2</span>, <span class="cf">function</span>(x) {</span>
<span id="cb19-107"><a href="data-processing.html#cb19-107" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">is.numeric</span>(x)) <span class="fu">ifelse</span>(<span class="fu">is.na</span>(x), <span class="fu">mean</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), x) <span class="cf">else</span> x})</span>
<span id="cb19-108"><a href="data-processing.html#cb19-108" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;min&quot;</span>) {</span>
<span id="cb19-109"><a href="data-processing.html#cb19-109" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">apply</span>(depurdata, <span class="dv">2</span>, <span class="cf">function</span>(x) {</span>
<span id="cb19-110"><a href="data-processing.html#cb19-110" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">is.numeric</span>(x)) <span class="fu">ifelse</span>(<span class="fu">is.na</span>(x), <span class="fu">min</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), x) <span class="cf">else</span> x})</span>
<span id="cb19-111"><a href="data-processing.html#cb19-111" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;knn&quot;</span>) {</span>
<span id="cb19-112"><a href="data-processing.html#cb19-112" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">t</span>(depurdata)</span>
<span id="cb19-113"><a href="data-processing.html#cb19-113" tabindex="-1"></a>    datai <span class="ot">&lt;-</span> impute<span class="sc">::</span><span class="fu">impute.knn</span>(depurdata, <span class="at">k =</span> <span class="dv">20</span>)</span>
<span id="cb19-114"><a href="data-processing.html#cb19-114" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">t</span>(datai<span class="sc">$</span>data)</span>
<span id="cb19-115"><a href="data-processing.html#cb19-115" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;rf&quot;</span>) {</span>
<span id="cb19-116"><a href="data-processing.html#cb19-116" tabindex="-1"></a>    fit <span class="ot">&lt;-</span> missForest<span class="sc">::</span><span class="fu">missForest</span>(<span class="fu">t</span>(depurdata))</span>
<span id="cb19-117"><a href="data-processing.html#cb19-117" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> fit<span class="sc">$</span>ximp <span class="sc">%&gt;%</span></span>
<span id="cb19-118"><a href="data-processing.html#cb19-118" tabindex="-1"></a>      <span class="fu">t</span>()</span>
<span id="cb19-119"><a href="data-processing.html#cb19-119" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;QRILC&quot;</span>) {</span>
<span id="cb19-120"><a href="data-processing.html#cb19-120" tabindex="-1"></a>    fit <span class="ot">&lt;-</span> <span class="fu">log</span>(<span class="fu">t</span>(depurdata)) <span class="sc">%&gt;%</span></span>
<span id="cb19-121"><a href="data-processing.html#cb19-121" tabindex="-1"></a>      imputeLCMD<span class="sc">::</span><span class="fu">impute.QRILC</span>()</span>
<span id="cb19-122"><a href="data-processing.html#cb19-122" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">t</span>(fit[[<span class="dv">1</span>]])</span>
<span id="cb19-123"><a href="data-processing.html#cb19-123" tabindex="-1"></a>  }</span>
<span id="cb19-124"><a href="data-processing.html#cb19-124" tabindex="-1"></a></span>
<span id="cb19-125"><a href="data-processing.html#cb19-125" tabindex="-1"></a>  <span class="fu">colnames</span>(depurdata) <span class="ot">&lt;-</span> correct_names</span>
<span id="cb19-126"><a href="data-processing.html#cb19-126" tabindex="-1"></a>  <span class="fu">rownames</span>(depurdata) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(prf_tab)</span>
<span id="cb19-127"><a href="data-processing.html#cb19-127" tabindex="-1"></a></span>
<span id="cb19-128"><a href="data-processing.html#cb19-128" tabindex="-1"></a>  <span class="cf">if</span> (methods<span class="sc">::</span><span class="fu">is</span>(object, <span class="st">&quot;SummarizedExperiment&quot;</span>)) {</span>
<span id="cb19-129"><a href="data-processing.html#cb19-129" tabindex="-1"></a>    target <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object) <span class="sc">%&gt;%</span></span>
<span id="cb19-130"><a href="data-processing.html#cb19-130" tabindex="-1"></a>      <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb19-131"><a href="data-processing.html#cb19-131" tabindex="-1"></a>      <span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb19-132"><a href="data-processing.html#cb19-132" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb19-133"><a href="data-processing.html#cb19-133" tabindex="-1"></a>                                    <span class="at">features =</span> depurdata)</span>
<span id="cb19-134"><a href="data-processing.html#cb19-134" tabindex="-1"></a>    </span>
<span id="cb19-135"><a href="data-processing.html#cb19-135" tabindex="-1"></a>  }</span>
<span id="cb19-136"><a href="data-processing.html#cb19-136" tabindex="-1"></a></span>
<span id="cb19-137"><a href="data-processing.html#cb19-137" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb19-138"><a href="data-processing.html#cb19-138" tabindex="-1"></a>}</span>
<span id="cb19-139"><a href="data-processing.html#cb19-139" tabindex="-1"></a></span>
<span id="cb19-140"><a href="data-processing.html#cb19-140" tabindex="-1"></a>se_impute <span class="ot">&lt;-</span> <span class="fu">impute_abundance</span>(</span>
<span id="cb19-141"><a href="data-processing.html#cb19-141" tabindex="-1"></a>                    se_check, </span>
<span id="cb19-142"><a href="data-processing.html#cb19-142" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;group&quot;</span>,</span>
<span id="cb19-143"><a href="data-processing.html#cb19-143" tabindex="-1"></a>                    <span class="at">ZerosAsNA =</span> <span class="cn">TRUE</span>, </span>
<span id="cb19-144"><a href="data-processing.html#cb19-144" tabindex="-1"></a>                    <span class="at">RemoveNA =</span> <span class="cn">TRUE</span>, </span>
<span id="cb19-145"><a href="data-processing.html#cb19-145" tabindex="-1"></a>                    <span class="at">prevalence =</span> <span class="fl">0.5</span>, </span>
<span id="cb19-146"><a href="data-processing.html#cb19-146" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;knn&quot;</span>)</span>
<span id="cb19-147"><a href="data-processing.html#cb19-147" tabindex="-1"></a>se_impute</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 180 55 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(180): M_38768 M_38296 ... M_31787 M_63361
## rowData names(0):
## colnames(55): P101001 P101003 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="data-processing.html#cb21-1" tabindex="-1"></a><span class="co"># profile</span></span>
<span id="cb21-2"><a href="data-processing.html#cb21-2" tabindex="-1"></a><span class="fu">head</span>(SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(se_impute))</span></code></pre></div>
<pre><code>##            P101001    P101003     P101004    P101007     P101009    P101010    P101011     P101012    P101013    P101016     P101017     P101018
## M_38768 51127588.0 42040432.0 34940596.00 58518636.0 51118832.00 83783688.0 29017984.0 51222064.00 77550128.0 30949554.0 26923596.00 56720032.00
## M_38296  5105020.5  4006120.2  3885477.00  4285129.5  6665653.50  9057441.0  2802655.2  5996555.00 11367511.0  3874736.8  2817151.00  8029728.00
## M_63436   756686.2   983889.2   851026.50   726593.9   232959.52   650261.1   541954.8   598491.00   438885.6  1625844.8   566466.94   427850.62
## M_57814   281502.0   175893.0   297304.38   319016.7   242172.20   200135.8   242804.9   248842.02   377212.6   183718.3   232514.53   279062.56
## M_52984   125465.8   154494.6   128320.37   176100.9    77345.62   122282.8   131924.3    80226.58   102010.7   129011.9    97377.79    98456.04
## M_48762   559069.1   596473.9    53293.76   627140.5  7016015.00  1914246.9  2762589.2   140576.45   723530.9   295995.2  3321584.25  3489639.25
##            P101019     P101021    P101022    P101024    P101025    P101027    P101030    P101031    P101038     P101041     P101042     P101047
## M_38768 27956064.0 48723600.00 16282054.0 77028824.0 32022342.0 22589448.0 38449788.0 59134052.0 32038030.0 20833830.00 33809080.00 18637508.00
## M_38296  3766663.8  5174967.00  1746182.9  5519105.5  2557365.0  1882902.6  2860324.2  4721201.0  4011627.5  2938779.00  3017260.50  1935144.12
## M_63436   519559.0  1301591.25  1474247.4   970475.8   628680.1   635516.6   367246.8   512037.9   852000.1   634488.56  1680135.75   326005.62
## M_57814   195083.9   290545.53   514479.0   420295.6   181825.7   196115.8   231290.3   315392.4   248820.8   214474.08   295862.56   397441.94
## M_52984   361975.5   119374.53   169170.4   101209.9   105117.3   104063.2   153193.4   181312.4   235948.1    68496.68   112239.87    94186.81
## M_48762   417727.6    16078.62 24787876.0  2287562.2  4724330.0  2960643.8  1503947.2   338791.1  1028211.6 16129281.00    87286.19  2211343.75
##            P101050    P101051     P101052    P101054    P101056    P101057    P101059     P101061    P101062    P101064     P101065     P101067
## M_38768 21978476.0 24265162.0 52203780.00 12836384.0 18546636.0 32301820.0 22645984.0 23683254.00 29027646.0 32629048.0 22950806.00 33555116.00
## M_38296  2897211.0  2476279.5  5928454.00  1685760.6  1650011.0  3419157.8  2196044.2  3217499.25  4060367.8  3031529.5  2467147.25  3567913.25
## M_63436   316650.2   737202.9   459385.94   346176.6   585470.2   417958.5   734586.3   337035.72   982299.4  1255148.0   637699.06   284516.12
## M_57814   253910.2   295948.6   246857.86   330762.7   209301.7   265086.1   181746.3   256601.55   187952.7   165441.1   203872.41   238433.59
## M_52984   112969.0   118040.4   115115.18   142644.3   103307.2   113114.2    89261.8    56381.96   186628.3   179940.6    82205.24    56987.89
## M_48762    98800.9  1933066.2    54885.02  4003695.2  1786579.6  2497416.7 10217042.0  7272486.00  4477679.0   395835.7   460618.81  8156659.00
##            P101068    P101069    P101071    P101072    P101074     P101075    P101076    P101077    P101079    P101080    P101081    P101082
## M_38768 44283972.0 52685972.0 32415040.0 34170948.0 22550616.0 22058076.00 24455466.0 25225170.0 15718590.0 29120336.0 65904836.0 22908578.0
## M_38296  6525382.0  3984333.5  3001414.5  4679519.0  2529255.5  2583265.50  3515218.2  3272875.0  2449462.5  2695001.5  6474709.5  2110243.8
## M_63436   664800.1   684813.6   596846.1   316855.0   646136.8   198381.73   255897.7   547243.4   508791.6  1256550.2   339909.3   596292.2
## M_57814   220632.9   209510.0   412294.9   335871.3   229869.7   225894.31   201601.9   285970.2   191453.5   264148.1   212220.1   335886.6
## M_52984   144673.7   151451.9   131184.4   127776.5   102868.8    64535.59   105523.9   130333.8   127343.7   153657.1   125355.2   107572.9
## M_48762  3966085.0  2887422.5   190455.1 46928844.0  3240584.5    91241.58  3088418.0  6992567.5  1325582.1   600080.8  3410091.7  1303324.6
##            P101084    P101085     P101088     P101090     P101094    P101095     P101096
## M_38768 29140440.0 20427124.0 29199012.00 24042020.00 36910084.00 35662068.0 66402192.00
## M_38296  3648091.2  3253531.8  4154170.75  2396959.75  4759584.50  3452283.2  6374383.00
## M_63436   497300.8   309859.3   601515.12   794206.00   414972.84  3606340.5  1077637.50
## M_57814   228471.4   345303.3   333549.22   321148.53   313197.78   500135.6   226660.25
## M_52984   151915.6   106491.1    89181.83   147634.67    91856.74   340070.0   137341.48
## M_48762   684199.1  2319273.2   854781.06    92700.81  1132143.00 31216882.0    34001.17</code></pre>
</div>
<div id="data-filtering" class="section level2 hasAnchor" number="2.6">
<h2><span class="header-section-number">2.6</span> Data Filtering<a href="data-processing.html#data-filtering" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The purpose of the data filtering is to identify and remove variables that are unlikely to be of use when modeling the data. No phenotype information are used in the filtering process, so the result can be used with any downstream analysis. This step is strongly recommended for untargeted metabolomics datasets (i.e. spectral binning data, peak lists) with large number of variables, many of them are from baseline noises. Filtering can usually improve the results. For details, please refer to the paper by Hackstadt, et al.</p>
<p>Non-informative variables can be characterized in three groups: 1) variables of very small values (close to baseline or detection limit) - these variables can be detected using mean or median; 2) variables that are near-constant values throughout the experiment conditions (housekeeping or homeostasis) - these variables can be detected using standard deviation (SD); or the robust estimate such as interquantile range (IQR); and 3) variables that show low repeatability - this can be measured using QC samples using the relative standard deviation(RSD = SD/mean). Features with high percent RSD should be removed from the subsequent analysis (the suggested threshold is 20% for LC-MS and 30% for GC-MS). For data filtering based on the first two categories, the following empirical rules are applied during data filtering:</p>
<ul>
<li>Less than 250 variables: 5% will be filtered;</li>
<li>Between 250 - 500 variables: 10% will be filtered;</li>
<li>Between 500 - 1000 variables: 25% will be filtered;</li>
<li>Over 1000 variables: 40% will be filtered;</li>
</ul>
<p>Filtering features if their RSDs are &gt; 25% in QC samples</p>
<ul>
<li><p>Interquantile range (IQR)</p></li>
<li><p>Standard deviation (SD)</p></li>
<li><p>Median absolute deviation (MAD)</p></li>
<li><p>Relative standard deviation (RSD = SD/mean)</p></li>
<li><p>Non-parametric relative standard deviation (MAD/median)</p></li>
<li><p>Mean intensity value</p></li>
<li><p>Median intensity value</p></li>
</ul>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="data-processing.html#cb23-1" tabindex="-1"></a>FilterFeature <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb23-2"><a href="data-processing.html#cb23-2" tabindex="-1"></a>    object,</span>
<span id="cb23-3"><a href="data-processing.html#cb23-3" tabindex="-1"></a>    qc_label,</span>
<span id="cb23-4"><a href="data-processing.html#cb23-4" tabindex="-1"></a>    <span class="at">method =</span> <span class="fu">c</span>(<span class="st">&quot;none&quot;</span>, <span class="st">&quot;iqr&quot;</span>, <span class="st">&quot;rsd&quot;</span>,</span>
<span id="cb23-5"><a href="data-processing.html#cb23-5" tabindex="-1"></a>               <span class="st">&quot;nrsd&quot;</span>, <span class="st">&quot;mean&quot;</span>, <span class="st">&quot;sd&quot;</span>,</span>
<span id="cb23-6"><a href="data-processing.html#cb23-6" tabindex="-1"></a>               <span class="st">&quot;mad&quot;</span>, <span class="st">&quot;median&quot;</span>),</span>
<span id="cb23-7"><a href="data-processing.html#cb23-7" tabindex="-1"></a>    <span class="at">rsd_cutoff =</span> <span class="dv">25</span>) {</span>
<span id="cb23-8"><a href="data-processing.html#cb23-8" tabindex="-1"></a>  </span>
<span id="cb23-9"><a href="data-processing.html#cb23-9" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) </span>
<span id="cb23-10"><a href="data-processing.html#cb23-10" tabindex="-1"></a>  metadata_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object) </span>
<span id="cb23-11"><a href="data-processing.html#cb23-11" tabindex="-1"></a>  </span>
<span id="cb23-12"><a href="data-processing.html#cb23-12" tabindex="-1"></a>  <span class="co"># QC samples</span></span>
<span id="cb23-13"><a href="data-processing.html#cb23-13" tabindex="-1"></a>  qc_samples <span class="ot">&lt;-</span> metadata_tab <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb23-14"><a href="data-processing.html#cb23-14" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(group <span class="sc">==</span> qc_label)</span>
<span id="cb23-15"><a href="data-processing.html#cb23-15" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">dim</span>(qc_samples)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb23-16"><a href="data-processing.html#cb23-16" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;No qc samples have been chosen, please check your input&quot;</span>)</span>
<span id="cb23-17"><a href="data-processing.html#cb23-17" tabindex="-1"></a>  }</span>
<span id="cb23-18"><a href="data-processing.html#cb23-18" tabindex="-1"></a>  </span>
<span id="cb23-19"><a href="data-processing.html#cb23-19" tabindex="-1"></a>  <span class="co"># QC samples&#39; feature table</span></span>
<span id="cb23-20"><a href="data-processing.html#cb23-20" tabindex="-1"></a>  qc_feature <span class="ot">&lt;-</span> features_tab[, <span class="fu">colnames</span>(features_tab)<span class="sc">%in%</span><span class="fu">rownames</span>(qc_samples)] <span class="sc">%&gt;%</span></span>
<span id="cb23-21"><a href="data-processing.html#cb23-21" tabindex="-1"></a>    <span class="fu">t</span>()</span>
<span id="cb23-22"><a href="data-processing.html#cb23-22" tabindex="-1"></a>  </span>
<span id="cb23-23"><a href="data-processing.html#cb23-23" tabindex="-1"></a>  <span class="co"># filter features by QC RSD</span></span>
<span id="cb23-24"><a href="data-processing.html#cb23-24" tabindex="-1"></a>  rsd <span class="ot">&lt;-</span> rsd_cutoff <span class="sc">/</span> <span class="dv">100</span></span>
<span id="cb23-25"><a href="data-processing.html#cb23-25" tabindex="-1"></a>  sds <span class="ot">&lt;-</span> <span class="fu">apply</span>(qc_feature, <span class="dv">2</span>, sd, <span class="at">na.rm =</span> T)</span>
<span id="cb23-26"><a href="data-processing.html#cb23-26" tabindex="-1"></a>  mns <span class="ot">&lt;-</span> <span class="fu">apply</span>(qc_feature, <span class="dv">2</span>, mean, <span class="at">na.rm =</span> T)</span>
<span id="cb23-27"><a href="data-processing.html#cb23-27" tabindex="-1"></a>  rsd_vals <span class="ot">&lt;-</span> <span class="fu">abs</span>(sds<span class="sc">/</span>mns) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb23-28"><a href="data-processing.html#cb23-28" tabindex="-1"></a>  gd_inx <span class="ot">&lt;-</span> rsd_vals <span class="sc">&lt;</span> rsd</span>
<span id="cb23-29"><a href="data-processing.html#cb23-29" tabindex="-1"></a>  int_mat <span class="ot">&lt;-</span> features_tab[gd_inx, ]</span>
<span id="cb23-30"><a href="data-processing.html#cb23-30" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">&quot;Removed &quot;</span>, (<span class="fu">dim</span>(qc_feature)[<span class="dv">2</span>] <span class="sc">-</span> <span class="fu">dim</span>(int_mat)[<span class="dv">1</span>]), </span>
<span id="cb23-31"><a href="data-processing.html#cb23-31" tabindex="-1"></a>  <span class="st">&quot; features based on QC RSD values. QC samples are excluded from downstream functional analysis.&quot;</span>)</span>
<span id="cb23-32"><a href="data-processing.html#cb23-32" tabindex="-1"></a>  </span>
<span id="cb23-33"><a href="data-processing.html#cb23-33" tabindex="-1"></a>  <span class="co"># whether to filter features by percentage according to the number</span></span>
<span id="cb23-34"><a href="data-processing.html#cb23-34" tabindex="-1"></a>  PerformFeatureFilter <span class="ot">&lt;-</span> <span class="cf">function</span>(datMatrix, </span>
<span id="cb23-35"><a href="data-processing.html#cb23-35" tabindex="-1"></a>                                   <span class="at">qc_method =</span> method,</span>
<span id="cb23-36"><a href="data-processing.html#cb23-36" tabindex="-1"></a>                                   <span class="at">remain_num =</span> <span class="cn">NULL</span>) {</span>
<span id="cb23-37"><a href="data-processing.html#cb23-37" tabindex="-1"></a>    </span>
<span id="cb23-38"><a href="data-processing.html#cb23-38" tabindex="-1"></a>    dat <span class="ot">&lt;-</span> datMatrix</span>
<span id="cb23-39"><a href="data-processing.html#cb23-39" tabindex="-1"></a>    feat_num <span class="ot">&lt;-</span> <span class="fu">ncol</span>(dat)</span>
<span id="cb23-40"><a href="data-processing.html#cb23-40" tabindex="-1"></a>    feat_nms <span class="ot">&lt;-</span> <span class="fu">colnames</span>(dat)</span>
<span id="cb23-41"><a href="data-processing.html#cb23-41" tabindex="-1"></a>    nm <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb23-42"><a href="data-processing.html#cb23-42" tabindex="-1"></a>    <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;none&quot;</span> <span class="sc">&amp;&amp;</span> feat_num <span class="sc">&lt;</span> <span class="dv">5000</span>) { <span class="co"># only allow for less than 4000</span></span>
<span id="cb23-43"><a href="data-processing.html#cb23-43" tabindex="-1"></a>      remain <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">TRUE</span>, feat_num)</span>
<span id="cb23-44"><a href="data-processing.html#cb23-44" tabindex="-1"></a>      nm <span class="ot">&lt;-</span> <span class="st">&quot;No filtering was applied&quot;</span></span>
<span id="cb23-45"><a href="data-processing.html#cb23-45" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb23-46"><a href="data-processing.html#cb23-46" tabindex="-1"></a>      <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;rsd&quot;</span>){</span>
<span id="cb23-47"><a href="data-processing.html#cb23-47" tabindex="-1"></a>        sds <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, sd, <span class="at">na.rm =</span> T)</span>
<span id="cb23-48"><a href="data-processing.html#cb23-48" tabindex="-1"></a>        mns <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, mean, <span class="at">na.rm =</span> T)</span>
<span id="cb23-49"><a href="data-processing.html#cb23-49" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">abs</span>(sds<span class="sc">/</span>mns)</span>
<span id="cb23-50"><a href="data-processing.html#cb23-50" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;Relative standard deviation&quot;</span></span>
<span id="cb23-51"><a href="data-processing.html#cb23-51" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;nrsd&quot;</span> ) {</span>
<span id="cb23-52"><a href="data-processing.html#cb23-52" tabindex="-1"></a>        mads <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, mad, <span class="at">na.rm =</span> T)</span>
<span id="cb23-53"><a href="data-processing.html#cb23-53" tabindex="-1"></a>        meds <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, median, <span class="at">na.rm =</span> T)</span>
<span id="cb23-54"><a href="data-processing.html#cb23-54" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">abs</span>(mads<span class="sc">/</span>meds)</span>
<span id="cb23-55"><a href="data-processing.html#cb23-55" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;Non-paramatric relative standard deviation&quot;</span></span>
<span id="cb23-56"><a href="data-processing.html#cb23-56" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;mean&quot;</span>) {</span>
<span id="cb23-57"><a href="data-processing.html#cb23-57" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, mean, <span class="at">na.rm =</span> T)</span>
<span id="cb23-58"><a href="data-processing.html#cb23-58" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;mean&quot;</span></span>
<span id="cb23-59"><a href="data-processing.html#cb23-59" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;sd&quot;</span>) {</span>
<span id="cb23-60"><a href="data-processing.html#cb23-60" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, sd, <span class="at">na.rm =</span> T)</span>
<span id="cb23-61"><a href="data-processing.html#cb23-61" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;standard deviation&quot;</span></span>
<span id="cb23-62"><a href="data-processing.html#cb23-62" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;mad&quot;</span>) {</span>
<span id="cb23-63"><a href="data-processing.html#cb23-63" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, mad, <span class="at">na.rm =</span> T)</span>
<span id="cb23-64"><a href="data-processing.html#cb23-64" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;Median absolute deviation&quot;</span></span>
<span id="cb23-65"><a href="data-processing.html#cb23-65" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;median&quot;</span>) {</span>
<span id="cb23-66"><a href="data-processing.html#cb23-66" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, median, <span class="at">na.rm =</span> T)</span>
<span id="cb23-67"><a href="data-processing.html#cb23-67" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;median&quot;</span></span>
<span id="cb23-68"><a href="data-processing.html#cb23-68" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;iqr&quot;</span>) { <span class="co"># iqr</span></span>
<span id="cb23-69"><a href="data-processing.html#cb23-69" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, IQR, <span class="at">na.rm =</span> T)</span>
<span id="cb23-70"><a href="data-processing.html#cb23-70" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;Interquantile Range&quot;</span></span>
<span id="cb23-71"><a href="data-processing.html#cb23-71" tabindex="-1"></a>      }</span>
<span id="cb23-72"><a href="data-processing.html#cb23-72" tabindex="-1"></a>      </span>
<span id="cb23-73"><a href="data-processing.html#cb23-73" tabindex="-1"></a>      <span class="co"># get the rank of the filtered variables</span></span>
<span id="cb23-74"><a href="data-processing.html#cb23-74" tabindex="-1"></a>      rk <span class="ot">&lt;-</span> <span class="fu">rank</span>(<span class="sc">-</span>filter_val, <span class="at">ties.method =</span> <span class="st">&quot;random&quot;</span>)</span>
<span id="cb23-75"><a href="data-processing.html#cb23-75" tabindex="-1"></a>      </span>
<span id="cb23-76"><a href="data-processing.html#cb23-76" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">is.null</span>(remain_num)) { <span class="co"># apply empirical filtering based on data size</span></span>
<span id="cb23-77"><a href="data-processing.html#cb23-77" tabindex="-1"></a>          <span class="cf">if</span> (feat_num <span class="sc">&lt;</span> <span class="dv">250</span>) { <span class="co"># reduce 5%</span></span>
<span id="cb23-78"><a href="data-processing.html#cb23-78" tabindex="-1"></a>            remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> feat_num <span class="sc">*</span> <span class="fl">0.95</span></span>
<span id="cb23-79"><a href="data-processing.html#cb23-79" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Further feature filtering based on &quot;</span>, nm)</span>
<span id="cb23-80"><a href="data-processing.html#cb23-80" tabindex="-1"></a>          } <span class="cf">else</span> <span class="cf">if</span> (feat_num <span class="sc">&lt;</span> <span class="dv">500</span>) { <span class="co"># reduce 10%</span></span>
<span id="cb23-81"><a href="data-processing.html#cb23-81" tabindex="-1"></a>            remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> feat_num <span class="sc">*</span> <span class="fl">0.9</span></span>
<span id="cb23-82"><a href="data-processing.html#cb23-82" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Further feature filtering based on &quot;</span>, nm)</span>
<span id="cb23-83"><a href="data-processing.html#cb23-83" tabindex="-1"></a>          } <span class="cf">else</span> <span class="cf">if</span> (feat_num <span class="sc">&lt;</span> <span class="dv">1000</span>) { <span class="co"># reduce 25%</span></span>
<span id="cb23-84"><a href="data-processing.html#cb23-84" tabindex="-1"></a>            remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> feat_num <span class="sc">*</span> <span class="fl">0.75</span></span>
<span id="cb23-85"><a href="data-processing.html#cb23-85" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Further feature filtering based on &quot;</span>, nm)</span>
<span id="cb23-86"><a href="data-processing.html#cb23-86" tabindex="-1"></a>          } <span class="cf">else</span> { <span class="co"># reduce 40%, if still over 5000, then only use top 5000</span></span>
<span id="cb23-87"><a href="data-processing.html#cb23-87" tabindex="-1"></a>            remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> feat_num <span class="sc">*</span> <span class="fl">0.6</span></span>
<span id="cb23-88"><a href="data-processing.html#cb23-88" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Further feature filtering based on &quot;</span>, nm)</span>
<span id="cb23-89"><a href="data-processing.html#cb23-89" tabindex="-1"></a>          }</span>
<span id="cb23-90"><a href="data-processing.html#cb23-90" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb23-91"><a href="data-processing.html#cb23-91" tabindex="-1"></a>        remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> remain_num</span>
<span id="cb23-92"><a href="data-processing.html#cb23-92" tabindex="-1"></a>      }</span>
<span id="cb23-93"><a href="data-processing.html#cb23-93" tabindex="-1"></a>    }</span>
<span id="cb23-94"><a href="data-processing.html#cb23-94" tabindex="-1"></a>    </span>
<span id="cb23-95"><a href="data-processing.html#cb23-95" tabindex="-1"></a>    res <span class="ot">&lt;-</span> datMatrix[, remain]</span>
<span id="cb23-96"><a href="data-processing.html#cb23-96" tabindex="-1"></a>    </span>
<span id="cb23-97"><a href="data-processing.html#cb23-97" tabindex="-1"></a>    <span class="fu">return</span>(res)</span>
<span id="cb23-98"><a href="data-processing.html#cb23-98" tabindex="-1"></a>  }  </span>
<span id="cb23-99"><a href="data-processing.html#cb23-99" tabindex="-1"></a>  </span>
<span id="cb23-100"><a href="data-processing.html#cb23-100" tabindex="-1"></a>  feature_res <span class="ot">&lt;-</span> <span class="fu">PerformFeatureFilter</span>(<span class="fu">t</span>(int_mat))</span>
<span id="cb23-101"><a href="data-processing.html#cb23-101" tabindex="-1"></a>  </span>
<span id="cb23-102"><a href="data-processing.html#cb23-102" tabindex="-1"></a>  <span class="co"># remove QC samples </span></span>
<span id="cb23-103"><a href="data-processing.html#cb23-103" tabindex="-1"></a>  feature_final <span class="ot">&lt;-</span> feature_res[<span class="sc">!</span><span class="fu">rownames</span>(feature_res) <span class="sc">%in%</span> <span class="fu">rownames</span>(qc_samples), ]</span>
<span id="cb23-104"><a href="data-processing.html#cb23-104" tabindex="-1"></a>  </span>
<span id="cb23-105"><a href="data-processing.html#cb23-105" tabindex="-1"></a>  <span class="co"># save int_mat into se object </span></span>
<span id="cb23-106"><a href="data-processing.html#cb23-106" tabindex="-1"></a>  target <span class="ot">&lt;-</span> metadata_tab <span class="sc">%&gt;%</span> </span>
<span id="cb23-107"><a href="data-processing.html#cb23-107" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb23-108"><a href="data-processing.html#cb23-108" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb23-109"><a href="data-processing.html#cb23-109" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(SampleID <span class="sc">%in%</span> <span class="fu">rownames</span>(feature_final))</span>
<span id="cb23-110"><a href="data-processing.html#cb23-110" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb23-111"><a href="data-processing.html#cb23-111" tabindex="-1"></a>                                  <span class="at">features =</span> feature_final)</span>
<span id="cb23-112"><a href="data-processing.html#cb23-112" tabindex="-1"></a>  </span>
<span id="cb23-113"><a href="data-processing.html#cb23-113" tabindex="-1"></a>  <span class="fu">return</span>(res) </span>
<span id="cb23-114"><a href="data-processing.html#cb23-114" tabindex="-1"></a>}</span>
<span id="cb23-115"><a href="data-processing.html#cb23-115" tabindex="-1"></a></span>
<span id="cb23-116"><a href="data-processing.html#cb23-116" tabindex="-1"></a>se_filter <span class="ot">&lt;-</span> <span class="fu">FilterFeature</span>(</span>
<span id="cb23-117"><a href="data-processing.html#cb23-117" tabindex="-1"></a>  <span class="at">object =</span> se_impute,</span>
<span id="cb23-118"><a href="data-processing.html#cb23-118" tabindex="-1"></a>  <span class="at">qc_label =</span> <span class="st">&quot;None&quot;</span>,</span>
<span id="cb23-119"><a href="data-processing.html#cb23-119" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">&quot;iqr&quot;</span>)</span></code></pre></div>
<pre><code>## Removed 133 features based on QC RSD values. QC samples are excluded from downstream functional analysis.</code></pre>
<pre><code>## Further feature filtering based on Interquantile Range</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="data-processing.html#cb26-1" tabindex="-1"></a>se_filter</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 44 45 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(44): M_52603 M_19130 ... M_62952 M_32197
## rowData names(0):
## colnames(45): P101001 P101004 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="data-normalization" class="section level2 hasAnchor" number="2.7">
<h2><span class="header-section-number">2.7</span> Data Normalization<a href="data-processing.html#data-normalization" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The normalization procedures are grouped into three categories. You can use one or combine them to achieve better results.</p>
<ul>
<li><p>Sample normalization is for general-purpose adjustment for systematic differences among samples;</p>
<ul>
<li><p>Sample-specific normalization (i.e. weight, volume)</p></li>
<li><p>Normalization by sum</p></li>
<li><p>Normalization by median</p></li>
<li><p>Normalization by a reference sample (PQN)</p></li>
<li><p>Normalization by a pooled sample from group (group PQN)</p></li>
<li><p>Normalization by reference feature</p></li>
<li><p>Quantile normalization (suggested only for &gt; 1000 features)</p></li>
</ul></li>
<li><p>Data transformation applies a mathematical transformation on individual values themselves. A simple mathematical approach is used to deal with negative values in log and square root.</p>
<ul>
<li><p>Log transformation (base 10)</p></li>
<li><p>Square root transformation (square root of data values)</p></li>
<li><p>Cube root transformation (cube root of data values)</p></li>
</ul></li>
<li><p>Data scaling adjusts each variable/feature by a scaling factor computed based on the dispersion of the variable.</p>
<ul>
<li><p>Mean centering (mean-centered only)</p></li>
<li><p>Auto scaling (mean-centered and divided by the standard deviation of each variable)</p></li>
<li><p>Pareto scaling (mean-centered and divided by the square root of the standard deviation of each variable)</p></li>
<li><p>Range scaling (mean-centered and divided by the range of each variable)</p></li>
</ul></li>
</ul>
<div id="normalization-by-normalizedata-function" class="section level3 hasAnchor" number="2.7.1">
<h3><span class="header-section-number">2.7.1</span> Normalization by <strong>NormalizeData</strong> function<a href="data-processing.html#normalization-by-normalizedata-function" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="data-processing.html#cb28-1" tabindex="-1"></a>NormalizeData <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb28-2"><a href="data-processing.html#cb28-2" tabindex="-1"></a>                      object,</span>
<span id="cb28-3"><a href="data-processing.html#cb28-3" tabindex="-1"></a>                      <span class="at">rowNorm =</span> <span class="fu">c</span>(<span class="st">&quot;Quantile&quot;</span>, <span class="st">&quot;GroupPQN&quot;</span>, <span class="st">&quot;SamplePQN&quot;</span>, </span>
<span id="cb28-4"><a href="data-processing.html#cb28-4" tabindex="-1"></a>                                  <span class="st">&quot;CompNorm&quot;</span>, <span class="st">&quot;SumNorm&quot;</span>, <span class="st">&quot;MedianNorm&quot;</span>,</span>
<span id="cb28-5"><a href="data-processing.html#cb28-5" tabindex="-1"></a>                                  <span class="st">&quot;SpecNorm&quot;</span>, <span class="st">&quot;None&quot;</span>),</span>
<span id="cb28-6"><a href="data-processing.html#cb28-6" tabindex="-1"></a>                      <span class="at">transNorm =</span> <span class="fu">c</span>(<span class="st">&quot;LogNorm&quot;</span>, <span class="st">&quot;SrNorm&quot;</span>, <span class="st">&quot;CrNorm&quot;</span>, <span class="st">&quot;None&quot;</span>),</span>
<span id="cb28-7"><a href="data-processing.html#cb28-7" tabindex="-1"></a>                      <span class="at">scaleNorm =</span> <span class="fu">c</span>(<span class="st">&quot;MeanCenter&quot;</span>, <span class="st">&quot;AutoNorm&quot;</span>, <span class="st">&quot;ParetoNorm&quot;</span>, </span>
<span id="cb28-8"><a href="data-processing.html#cb28-8" tabindex="-1"></a>                                    <span class="st">&quot;RangeNorm&quot;</span>, <span class="st">&quot;None&quot;</span>),</span>
<span id="cb28-9"><a href="data-processing.html#cb28-9" tabindex="-1"></a>                      <span class="at">ref =</span> <span class="cn">NULL</span>,</span>
<span id="cb28-10"><a href="data-processing.html#cb28-10" tabindex="-1"></a>                      <span class="at">SpeWeight =</span> <span class="dv">1</span>) {</span>
<span id="cb28-11"><a href="data-processing.html#cb28-11" tabindex="-1"></a>  </span>
<span id="cb28-12"><a href="data-processing.html#cb28-12" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) </span>
<span id="cb28-13"><a href="data-processing.html#cb28-13" tabindex="-1"></a>  metadata_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object)   </span>
<span id="cb28-14"><a href="data-processing.html#cb28-14" tabindex="-1"></a>  </span>
<span id="cb28-15"><a href="data-processing.html#cb28-15" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">t</span>(features_tab)</span>
<span id="cb28-16"><a href="data-processing.html#cb28-16" tabindex="-1"></a>  colNames <span class="ot">&lt;-</span> <span class="fu">colnames</span>(data)</span>
<span id="cb28-17"><a href="data-processing.html#cb28-17" tabindex="-1"></a>  rowNames <span class="ot">&lt;-</span> <span class="fu">rownames</span>(data)</span>
<span id="cb28-18"><a href="data-processing.html#cb28-18" tabindex="-1"></a>  </span>
<span id="cb28-19"><a href="data-processing.html#cb28-19" tabindex="-1"></a>  <span class="do">#############################################</span></span>
<span id="cb28-20"><a href="data-processing.html#cb28-20" tabindex="-1"></a>  <span class="co"># Sample normalization</span></span>
<span id="cb28-21"><a href="data-processing.html#cb28-21" tabindex="-1"></a>  <span class="co"># perform quantile normalization on the raw data (can be log transformed later by user)</span></span>
<span id="cb28-22"><a href="data-processing.html#cb28-22" tabindex="-1"></a>  QuantileNormalize <span class="ot">&lt;-</span> <span class="cf">function</span>(data) {</span>
<span id="cb28-23"><a href="data-processing.html#cb28-23" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">t</span>(preprocessCore<span class="sc">::</span><span class="fu">normalize.quantiles</span>(<span class="fu">t</span>(data), <span class="at">copy=</span><span class="cn">FALSE</span>)));</span>
<span id="cb28-24"><a href="data-processing.html#cb28-24" tabindex="-1"></a>  }</span>
<span id="cb28-25"><a href="data-processing.html#cb28-25" tabindex="-1"></a>  <span class="co"># normalize by a reference sample (probability quotient normalization)</span></span>
<span id="cb28-26"><a href="data-processing.html#cb28-26" tabindex="-1"></a>  <span class="co"># ref should be the name of the reference sample</span></span>
<span id="cb28-27"><a href="data-processing.html#cb28-27" tabindex="-1"></a>  ProbNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x, ref_smpl) {</span>
<span id="cb28-28"><a href="data-processing.html#cb28-28" tabindex="-1"></a>    <span class="fu">return</span>(x<span class="sc">/</span><span class="fu">median</span>(<span class="fu">as.numeric</span>(x<span class="sc">/</span>ref_smpl), <span class="at">na.rm =</span> T))</span>
<span id="cb28-29"><a href="data-processing.html#cb28-29" tabindex="-1"></a>  }</span>
<span id="cb28-30"><a href="data-processing.html#cb28-30" tabindex="-1"></a>  </span>
<span id="cb28-31"><a href="data-processing.html#cb28-31" tabindex="-1"></a>  <span class="co"># normalize by a reference reference (i.e. creatinine)</span></span>
<span id="cb28-32"><a href="data-processing.html#cb28-32" tabindex="-1"></a>  <span class="co"># ref should be the name of the cmpd</span></span>
<span id="cb28-33"><a href="data-processing.html#cb28-33" tabindex="-1"></a>  CompNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x, ref) {</span>
<span id="cb28-34"><a href="data-processing.html#cb28-34" tabindex="-1"></a>    <span class="fu">return</span>(<span class="dv">1000</span><span class="sc">*</span>x<span class="sc">/</span>x[ref])</span>
<span id="cb28-35"><a href="data-processing.html#cb28-35" tabindex="-1"></a>  }</span>
<span id="cb28-36"><a href="data-processing.html#cb28-36" tabindex="-1"></a>  </span>
<span id="cb28-37"><a href="data-processing.html#cb28-37" tabindex="-1"></a>  SumNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb28-38"><a href="data-processing.html#cb28-38" tabindex="-1"></a>    <span class="fu">return</span>(<span class="dv">1000</span><span class="sc">*</span>x<span class="sc">/</span><span class="fu">sum</span>(x, <span class="at">na.rm =</span> T))</span>
<span id="cb28-39"><a href="data-processing.html#cb28-39" tabindex="-1"></a>  }</span>
<span id="cb28-40"><a href="data-processing.html#cb28-40" tabindex="-1"></a>  </span>
<span id="cb28-41"><a href="data-processing.html#cb28-41" tabindex="-1"></a>  <span class="co"># normalize by median</span></span>
<span id="cb28-42"><a href="data-processing.html#cb28-42" tabindex="-1"></a>  MedianNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb28-43"><a href="data-processing.html#cb28-43" tabindex="-1"></a>    <span class="fu">return</span>(x<span class="sc">/</span><span class="fu">median</span>(x, <span class="at">na.rm =</span> T))</span>
<span id="cb28-44"><a href="data-processing.html#cb28-44" tabindex="-1"></a>  }  </span>
<span id="cb28-45"><a href="data-processing.html#cb28-45" tabindex="-1"></a>  </span>
<span id="cb28-46"><a href="data-processing.html#cb28-46" tabindex="-1"></a>  <span class="co"># row-wise normalization</span></span>
<span id="cb28-47"><a href="data-processing.html#cb28-47" tabindex="-1"></a>  <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;Quantile&quot;</span>) {</span>
<span id="cb28-48"><a href="data-processing.html#cb28-48" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">QuantileNormalize</span>(data)</span>
<span id="cb28-49"><a href="data-processing.html#cb28-49" tabindex="-1"></a>    <span class="co"># this can introduce constant variables if a variable is </span></span>
<span id="cb28-50"><a href="data-processing.html#cb28-50" tabindex="-1"></a>    <span class="co"># at the same rank across all samples (replaced by its average across all)</span></span>
<span id="cb28-51"><a href="data-processing.html#cb28-51" tabindex="-1"></a>    varCol <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, var, <span class="at">na.rm =</span> T)</span>
<span id="cb28-52"><a href="data-processing.html#cb28-52" tabindex="-1"></a>    constCol <span class="ot">&lt;-</span> (varCol <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> <span class="fu">is.na</span>(varCol))</span>
<span id="cb28-53"><a href="data-processing.html#cb28-53" tabindex="-1"></a>    constNum <span class="ot">&lt;-</span> <span class="fu">sum</span>(constCol, <span class="at">na.rm =</span> T)</span>
<span id="cb28-54"><a href="data-processing.html#cb28-54" tabindex="-1"></a>    <span class="cf">if</span> (constNum <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb28-55"><a href="data-processing.html#cb28-55" tabindex="-1"></a>      <span class="fu">message</span>(<span class="fu">paste</span>(<span class="st">&quot;After quantile normalization&quot;</span>, constNum, </span>
<span id="cb28-56"><a href="data-processing.html#cb28-56" tabindex="-1"></a>                    <span class="st">&quot;features with a constant value were found and deleted.&quot;</span>))</span>
<span id="cb28-57"><a href="data-processing.html#cb28-57" tabindex="-1"></a>      data <span class="ot">&lt;-</span> data[, <span class="sc">!</span>constCol, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb28-58"><a href="data-processing.html#cb28-58" tabindex="-1"></a>      colNames <span class="ot">&lt;-</span> <span class="fu">colnames</span>(data)</span>
<span id="cb28-59"><a href="data-processing.html#cb28-59" tabindex="-1"></a>      rowNames <span class="ot">&lt;-</span> <span class="fu">rownames</span>(data)</span>
<span id="cb28-60"><a href="data-processing.html#cb28-60" tabindex="-1"></a>    }</span>
<span id="cb28-61"><a href="data-processing.html#cb28-61" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Quantile Normalization&quot;</span></span>
<span id="cb28-62"><a href="data-processing.html#cb28-62" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;GroupPQN&quot;</span>) {</span>
<span id="cb28-63"><a href="data-processing.html#cb28-63" tabindex="-1"></a>    grp_inx <span class="ot">&lt;-</span> metadata_tab<span class="sc">$</span>group <span class="sc">==</span> ref</span>
<span id="cb28-64"><a href="data-processing.html#cb28-64" tabindex="-1"></a>    ref.smpl <span class="ot">&lt;-</span> <span class="fu">apply</span>(data[grp_inx, , <span class="at">drop =</span> <span class="cn">FALSE</span>], <span class="dv">2</span>, mean)</span>
<span id="cb28-65"><a href="data-processing.html#cb28-65" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(data, <span class="dv">1</span>, ProbNorm, ref.smpl))</span>
<span id="cb28-66"><a href="data-processing.html#cb28-66" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Probabilistic Quotient Normalization by a reference group&quot;</span></span>
<span id="cb28-67"><a href="data-processing.html#cb28-67" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;SamplePQN&quot;</span>) {</span>
<span id="cb28-68"><a href="data-processing.html#cb28-68" tabindex="-1"></a>    ref.smpl <span class="ot">&lt;-</span> data[ref, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb28-69"><a href="data-processing.html#cb28-69" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(data, <span class="dv">1</span>, ProbNorm, ref.smpl))</span>
<span id="cb28-70"><a href="data-processing.html#cb28-70" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Probabilistic Quotient Normalization by a reference sample&quot;</span></span>
<span id="cb28-71"><a href="data-processing.html#cb28-71" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;CompNorm&quot;</span>) {</span>
<span id="cb28-72"><a href="data-processing.html#cb28-72" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(<span class="fu">t</span>(data), <span class="dv">1</span>, CompNorm, ref))</span>
<span id="cb28-73"><a href="data-processing.html#cb28-73" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Normalization by a reference feature&quot;</span>;</span>
<span id="cb28-74"><a href="data-processing.html#cb28-74" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;SumNorm&quot;</span>) {</span>
<span id="cb28-75"><a href="data-processing.html#cb28-75" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(data, <span class="dv">1</span>, SumNorm))</span>
<span id="cb28-76"><a href="data-processing.html#cb28-76" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Normalization to constant sum&quot;</span></span>
<span id="cb28-77"><a href="data-processing.html#cb28-77" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;MedianNorm&quot;</span>) {</span>
<span id="cb28-78"><a href="data-processing.html#cb28-78" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(data, <span class="dv">1</span>, MedianNorm))</span>
<span id="cb28-79"><a href="data-processing.html#cb28-79" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Normalization to sample median&quot;</span></span>
<span id="cb28-80"><a href="data-processing.html#cb28-80" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span>(rowNorm <span class="sc">==</span> <span class="st">&quot;SpecNorm&quot;</span>) {</span>
<span id="cb28-81"><a href="data-processing.html#cb28-81" tabindex="-1"></a>    norm.vec <span class="ot">&lt;-</span> <span class="fu">rep</span>(SpeWeight, <span class="fu">nrow</span>(data)) <span class="co"># default all same weight vec to prevent error</span></span>
<span id="cb28-82"><a href="data-processing.html#cb28-82" tabindex="-1"></a>    data <span class="ot">&lt;-</span> data <span class="sc">/</span> norm.vec</span>
<span id="cb28-83"><a href="data-processing.html#cb28-83" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">&quot;No sample specific information were given, all set to 1.0&quot;</span>)</span>
<span id="cb28-84"><a href="data-processing.html#cb28-84" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Normalization by sample-specific factor&quot;</span></span>
<span id="cb28-85"><a href="data-processing.html#cb28-85" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb28-86"><a href="data-processing.html#cb28-86" tabindex="-1"></a>    <span class="co"># nothing to do</span></span>
<span id="cb28-87"><a href="data-processing.html#cb28-87" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;N/A&quot;</span></span>
<span id="cb28-88"><a href="data-processing.html#cb28-88" tabindex="-1"></a>  }</span>
<span id="cb28-89"><a href="data-processing.html#cb28-89" tabindex="-1"></a>  <span class="do">################################################ </span></span>
<span id="cb28-90"><a href="data-processing.html#cb28-90" tabindex="-1"></a>  </span>
<span id="cb28-91"><a href="data-processing.html#cb28-91" tabindex="-1"></a>  <span class="co"># use apply will lose dimension info (i.e. row names and colnames)</span></span>
<span id="cb28-92"><a href="data-processing.html#cb28-92" tabindex="-1"></a>  <span class="fu">rownames</span>(data) <span class="ot">&lt;-</span> rowNames</span>
<span id="cb28-93"><a href="data-processing.html#cb28-93" tabindex="-1"></a>  <span class="fu">colnames</span>(data) <span class="ot">&lt;-</span> colNames</span>
<span id="cb28-94"><a href="data-processing.html#cb28-94" tabindex="-1"></a>  </span>
<span id="cb28-95"><a href="data-processing.html#cb28-95" tabindex="-1"></a>  <span class="co"># if the reference by feature, the feature column should be removed, since it is all 1</span></span>
<span id="cb28-96"><a href="data-processing.html#cb28-96" tabindex="-1"></a>  <span class="cf">if</span>(rowNorm <span class="sc">==</span> <span class="st">&quot;CompNorm&quot;</span> <span class="sc">&amp;&amp;</span> <span class="sc">!</span><span class="fu">is.null</span>(ref)){</span>
<span id="cb28-97"><a href="data-processing.html#cb28-97" tabindex="-1"></a>    inx <span class="ot">&lt;-</span> <span class="fu">match</span>(ref, <span class="fu">colnames</span>(data))</span>
<span id="cb28-98"><a href="data-processing.html#cb28-98" tabindex="-1"></a>    data <span class="ot">&lt;-</span> data[, <span class="sc">-</span>inx, drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb28-99"><a href="data-processing.html#cb28-99" tabindex="-1"></a>    colNames <span class="ot">&lt;-</span> colNames[<span class="sc">-</span>inx]</span>
<span id="cb28-100"><a href="data-processing.html#cb28-100" tabindex="-1"></a>  }</span>
<span id="cb28-101"><a href="data-processing.html#cb28-101" tabindex="-1"></a>  </span>
<span id="cb28-102"><a href="data-processing.html#cb28-102" tabindex="-1"></a>  <span class="do">#############################################</span></span>
<span id="cb28-103"><a href="data-processing.html#cb28-103" tabindex="-1"></a>  <span class="co">#  Data transformation</span></span>
<span id="cb28-104"><a href="data-processing.html#cb28-104" tabindex="-1"></a>  <span class="co"># generalize log, tolerant to 0 and negative values</span></span>
<span id="cb28-105"><a href="data-processing.html#cb28-105" tabindex="-1"></a>  LogNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x, min.val) {</span>
<span id="cb28-106"><a href="data-processing.html#cb28-106" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">log10</span>((x <span class="sc">+</span> <span class="fu">sqrt</span>(x<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> min.val<span class="sc">^</span><span class="dv">2</span>))<span class="sc">/</span><span class="dv">2</span>))</span>
<span id="cb28-107"><a href="data-processing.html#cb28-107" tabindex="-1"></a>  }</span>
<span id="cb28-108"><a href="data-processing.html#cb28-108" tabindex="-1"></a>  </span>
<span id="cb28-109"><a href="data-processing.html#cb28-109" tabindex="-1"></a>  <span class="co"># square root, tolerant to negative values</span></span>
<span id="cb28-110"><a href="data-processing.html#cb28-110" tabindex="-1"></a>  SquareRootNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x, min.val) {</span>
<span id="cb28-111"><a href="data-processing.html#cb28-111" tabindex="-1"></a>    <span class="fu">return</span>(((x <span class="sc">+</span> <span class="fu">sqrt</span>(x<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> min.val<span class="sc">^</span><span class="dv">2</span>))<span class="sc">/</span><span class="dv">2</span>)<span class="sc">^</span>(<span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>))</span>
<span id="cb28-112"><a href="data-processing.html#cb28-112" tabindex="-1"></a>  }</span>
<span id="cb28-113"><a href="data-processing.html#cb28-113" tabindex="-1"></a>  </span>
<span id="cb28-114"><a href="data-processing.html#cb28-114" tabindex="-1"></a>  <span class="cf">if</span> (transNorm <span class="sc">==</span> <span class="st">&quot;LogNorm&quot;</span>) {</span>
<span id="cb28-115"><a href="data-processing.html#cb28-115" tabindex="-1"></a>    min.val <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">abs</span>(data[data <span class="sc">!=</span> <span class="dv">0</span>]))<span class="sc">/</span><span class="dv">10</span></span>
<span id="cb28-116"><a href="data-processing.html#cb28-116" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, LogNorm, min.val)</span>
<span id="cb28-117"><a href="data-processing.html#cb28-117" tabindex="-1"></a>    transnm <span class="ot">&lt;-</span> <span class="st">&quot;Log10 Normalization&quot;</span></span>
<span id="cb28-118"><a href="data-processing.html#cb28-118" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (transNorm <span class="sc">==</span> <span class="st">&quot;SrNorm&quot;</span>) {</span>
<span id="cb28-119"><a href="data-processing.html#cb28-119" tabindex="-1"></a>    min.val <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">abs</span>(data[data <span class="sc">!=</span> <span class="dv">0</span>]))<span class="sc">/</span><span class="dv">10</span></span>
<span id="cb28-120"><a href="data-processing.html#cb28-120" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, SquareRootNorm, min.val)</span>
<span id="cb28-121"><a href="data-processing.html#cb28-121" tabindex="-1"></a>    transnm <span class="ot">&lt;-</span> <span class="st">&quot;Square Root Transformation&quot;</span></span>
<span id="cb28-122"><a href="data-processing.html#cb28-122" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (transNorm <span class="sc">==</span> <span class="st">&quot;CrNorm&quot;</span>) {</span>
<span id="cb28-123"><a href="data-processing.html#cb28-123" tabindex="-1"></a>    norm.data <span class="ot">&lt;-</span> <span class="fu">abs</span>(data)<span class="sc">^</span>(<span class="dv">1</span><span class="sc">/</span><span class="dv">3</span>)</span>
<span id="cb28-124"><a href="data-processing.html#cb28-124" tabindex="-1"></a>    norm.data[data <span class="sc">&lt;</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="sc">-</span>norm.data[data <span class="sc">&lt;</span> <span class="dv">0</span>]</span>
<span id="cb28-125"><a href="data-processing.html#cb28-125" tabindex="-1"></a>    data <span class="ot">&lt;-</span> norm.data</span>
<span id="cb28-126"><a href="data-processing.html#cb28-126" tabindex="-1"></a>    transnm <span class="ot">&lt;-</span> <span class="st">&quot;Cubic Root Transformation&quot;</span></span>
<span id="cb28-127"><a href="data-processing.html#cb28-127" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb28-128"><a href="data-processing.html#cb28-128" tabindex="-1"></a>    transnm <span class="ot">&lt;-</span> <span class="st">&quot;N/A&quot;</span></span>
<span id="cb28-129"><a href="data-processing.html#cb28-129" tabindex="-1"></a>  }</span>
<span id="cb28-130"><a href="data-processing.html#cb28-130" tabindex="-1"></a>  <span class="do">#############################################</span></span>
<span id="cb28-131"><a href="data-processing.html#cb28-131" tabindex="-1"></a>  </span>
<span id="cb28-132"><a href="data-processing.html#cb28-132" tabindex="-1"></a>  <span class="do">#############################################</span></span>
<span id="cb28-133"><a href="data-processing.html#cb28-133" tabindex="-1"></a>  <span class="co"># Data scaling</span></span>
<span id="cb28-134"><a href="data-processing.html#cb28-134" tabindex="-1"></a>  <span class="co"># normalize to zero mean and unit variance</span></span>
<span id="cb28-135"><a href="data-processing.html#cb28-135" tabindex="-1"></a>  AutoNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb28-136"><a href="data-processing.html#cb28-136" tabindex="-1"></a>    <span class="fu">return</span>((x <span class="sc">-</span> <span class="fu">mean</span>(x))<span class="sc">/</span><span class="fu">sd</span>(x, <span class="at">na.rm =</span> T))</span>
<span id="cb28-137"><a href="data-processing.html#cb28-137" tabindex="-1"></a>  }</span>
<span id="cb28-138"><a href="data-processing.html#cb28-138" tabindex="-1"></a>  </span>
<span id="cb28-139"><a href="data-processing.html#cb28-139" tabindex="-1"></a>  <span class="co"># normalize to zero mean but variance/SE</span></span>
<span id="cb28-140"><a href="data-processing.html#cb28-140" tabindex="-1"></a>  ParetoNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb28-141"><a href="data-processing.html#cb28-141" tabindex="-1"></a>    <span class="fu">return</span>((x <span class="sc">-</span> <span class="fu">mean</span>(x))<span class="sc">/</span><span class="fu">sqrt</span>(<span class="fu">sd</span>(x, <span class="at">na.rm =</span> T)))</span>
<span id="cb28-142"><a href="data-processing.html#cb28-142" tabindex="-1"></a>  }</span>
<span id="cb28-143"><a href="data-processing.html#cb28-143" tabindex="-1"></a>  </span>
<span id="cb28-144"><a href="data-processing.html#cb28-144" tabindex="-1"></a>  <span class="co"># normalize to zero mean but variance/SE</span></span>
<span id="cb28-145"><a href="data-processing.html#cb28-145" tabindex="-1"></a>  MeanCenter <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb28-146"><a href="data-processing.html#cb28-146" tabindex="-1"></a>    <span class="fu">return</span>(x <span class="sc">-</span> <span class="fu">mean</span>(x))</span>
<span id="cb28-147"><a href="data-processing.html#cb28-147" tabindex="-1"></a>  }</span>
<span id="cb28-148"><a href="data-processing.html#cb28-148" tabindex="-1"></a>  </span>
<span id="cb28-149"><a href="data-processing.html#cb28-149" tabindex="-1"></a>  <span class="co"># normalize to zero mean but variance/SE</span></span>
<span id="cb28-150"><a href="data-processing.html#cb28-150" tabindex="-1"></a>  RangeNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb28-151"><a href="data-processing.html#cb28-151" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">max</span>(x) <span class="sc">==</span> <span class="fu">min</span>(x)) {</span>
<span id="cb28-152"><a href="data-processing.html#cb28-152" tabindex="-1"></a>      <span class="fu">return</span>(x)</span>
<span id="cb28-153"><a href="data-processing.html#cb28-153" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb28-154"><a href="data-processing.html#cb28-154" tabindex="-1"></a>      <span class="fu">return</span>((x <span class="sc">-</span> <span class="fu">mean</span>(x))<span class="sc">/</span>(<span class="fu">max</span>(x) <span class="sc">-</span> <span class="fu">min</span>(x)))</span>
<span id="cb28-155"><a href="data-processing.html#cb28-155" tabindex="-1"></a>    }</span>
<span id="cb28-156"><a href="data-processing.html#cb28-156" tabindex="-1"></a>  }</span>
<span id="cb28-157"><a href="data-processing.html#cb28-157" tabindex="-1"></a></span>
<span id="cb28-158"><a href="data-processing.html#cb28-158" tabindex="-1"></a>  <span class="cf">if</span> (scaleNorm <span class="sc">==</span> <span class="st">&quot;MeanCenter&quot;</span>) {</span>
<span id="cb28-159"><a href="data-processing.html#cb28-159" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, MeanCenter)</span>
<span id="cb28-160"><a href="data-processing.html#cb28-160" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;Mean Centering&quot;</span></span>
<span id="cb28-161"><a href="data-processing.html#cb28-161" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (scaleNorm <span class="sc">==</span> <span class="st">&quot;AutoNorm&quot;</span>) {</span>
<span id="cb28-162"><a href="data-processing.html#cb28-162" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, AutoNorm)</span>
<span id="cb28-163"><a href="data-processing.html#cb28-163" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;Autoscaling&quot;</span></span>
<span id="cb28-164"><a href="data-processing.html#cb28-164" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (scaleNorm <span class="sc">==</span> <span class="st">&quot;ParetoNorm&quot;</span>) {</span>
<span id="cb28-165"><a href="data-processing.html#cb28-165" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, ParetoNorm)</span>
<span id="cb28-166"><a href="data-processing.html#cb28-166" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;Pareto Scaling&quot;</span></span>
<span id="cb28-167"><a href="data-processing.html#cb28-167" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (scaleNorm <span class="sc">==</span> <span class="st">&quot;RangeNorm&quot;</span>) {</span>
<span id="cb28-168"><a href="data-processing.html#cb28-168" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, RangeNorm)</span>
<span id="cb28-169"><a href="data-processing.html#cb28-169" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;Range Scaling&quot;</span></span>
<span id="cb28-170"><a href="data-processing.html#cb28-170" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb28-171"><a href="data-processing.html#cb28-171" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;N/A&quot;</span></span>
<span id="cb28-172"><a href="data-processing.html#cb28-172" tabindex="-1"></a>  }</span>
<span id="cb28-173"><a href="data-processing.html#cb28-173" tabindex="-1"></a>  <span class="do">############################################# </span></span>
<span id="cb28-174"><a href="data-processing.html#cb28-174" tabindex="-1"></a>  </span>
<span id="cb28-175"><a href="data-processing.html#cb28-175" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">&quot;Row norm: &quot;</span>, rownm, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>,</span>
<span id="cb28-176"><a href="data-processing.html#cb28-176" tabindex="-1"></a>          <span class="st">&quot;Data Transformation norm: &quot;</span>, transnm, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>,</span>
<span id="cb28-177"><a href="data-processing.html#cb28-177" tabindex="-1"></a>          <span class="st">&quot;Data Scaling norm: &quot;</span>, scalenm, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)  </span>
<span id="cb28-178"><a href="data-processing.html#cb28-178" tabindex="-1"></a>  </span>
<span id="cb28-179"><a href="data-processing.html#cb28-179" tabindex="-1"></a>  <span class="co"># note after using &quot;apply&quot; function, all the attribute lost, need to add back</span></span>
<span id="cb28-180"><a href="data-processing.html#cb28-180" tabindex="-1"></a>  <span class="fu">rownames</span>(data) <span class="ot">&lt;-</span> rowNames</span>
<span id="cb28-181"><a href="data-processing.html#cb28-181" tabindex="-1"></a>  <span class="fu">colnames</span>(data) <span class="ot">&lt;-</span> colNames</span>
<span id="cb28-182"><a href="data-processing.html#cb28-182" tabindex="-1"></a>  </span>
<span id="cb28-183"><a href="data-processing.html#cb28-183" tabindex="-1"></a>  target <span class="ot">&lt;-</span> metadata_tab <span class="sc">%&gt;%</span> </span>
<span id="cb28-184"><a href="data-processing.html#cb28-184" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb28-185"><a href="data-processing.html#cb28-185" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb28-186"><a href="data-processing.html#cb28-186" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(SampleID<span class="sc">%in%</span><span class="fu">rownames</span>(data))</span>
<span id="cb28-187"><a href="data-processing.html#cb28-187" tabindex="-1"></a>  se <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb28-188"><a href="data-processing.html#cb28-188" tabindex="-1"></a>                                 <span class="at">features =</span> data)</span>
<span id="cb28-189"><a href="data-processing.html#cb28-189" tabindex="-1"></a>  </span>
<span id="cb28-190"><a href="data-processing.html#cb28-190" tabindex="-1"></a>  <span class="co"># need to do some sanity check, for log there may be Inf values introduced</span></span>
<span id="cb28-191"><a href="data-processing.html#cb28-191" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">CheckData</span>(se)</span>
<span id="cb28-192"><a href="data-processing.html#cb28-192" tabindex="-1"></a>  </span>
<span id="cb28-193"><a href="data-processing.html#cb28-193" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb28-194"><a href="data-processing.html#cb28-194" tabindex="-1"></a>}</span>
<span id="cb28-195"><a href="data-processing.html#cb28-195" tabindex="-1"></a></span>
<span id="cb28-196"><a href="data-processing.html#cb28-196" tabindex="-1"></a>se_normalize <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(</span>
<span id="cb28-197"><a href="data-processing.html#cb28-197" tabindex="-1"></a>                      <span class="at">object =</span> se_impute,</span>
<span id="cb28-198"><a href="data-processing.html#cb28-198" tabindex="-1"></a>                      <span class="at">rowNorm =</span> <span class="st">&quot;None&quot;</span>,</span>
<span id="cb28-199"><a href="data-processing.html#cb28-199" tabindex="-1"></a>                      <span class="at">transNorm =</span> <span class="st">&quot;LogNorm&quot;</span>,</span>
<span id="cb28-200"><a href="data-processing.html#cb28-200" tabindex="-1"></a>                      <span class="at">scaleNorm =</span> <span class="st">&quot;ParetoNorm&quot;</span>)</span></code></pre></div>
<pre><code>## Row norm: N/A
## Data Transformation norm: Log10 Normalization
## Data Scaling norm: Pareto Scaling</code></pre>
<pre><code>## All data values are numeric.</code></pre>
<pre><code>## A total of 0 (0%) missing values were detected.</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="data-processing.html#cb32-1" tabindex="-1"></a>se_normalize</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 180 55 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(180): M_38768 M_38296 ... M_31787 M_63361
## rowData names(0):
## colnames(55): P101001 P101003 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="normalization-by-poma-r-package" class="section level3 hasAnchor" number="2.7.2">
<h3><span class="header-section-number">2.7.2</span> Normalization by <strong>POMA</strong> R package<a href="data-processing.html#normalization-by-poma-r-package" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="data-processing.html#cb34-1" tabindex="-1"></a>none <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_impute, <span class="at">method =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb34-2"><a href="data-processing.html#cb34-2" tabindex="-1"></a>auto_scaling <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_impute, <span class="at">method =</span> <span class="st">&quot;auto_scaling&quot;</span>)</span>
<span id="cb34-3"><a href="data-processing.html#cb34-3" tabindex="-1"></a>evel_scaling <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_impute, <span class="at">method =</span> <span class="st">&quot;level_scaling&quot;</span>)</span>
<span id="cb34-4"><a href="data-processing.html#cb34-4" tabindex="-1"></a>log_scaling <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_impute, <span class="at">method =</span> <span class="st">&quot;log_scaling&quot;</span>)</span>
<span id="cb34-5"><a href="data-processing.html#cb34-5" tabindex="-1"></a>log_transformation <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_impute, <span class="at">method =</span> <span class="st">&quot;log_transformation&quot;</span>)</span>
<span id="cb34-6"><a href="data-processing.html#cb34-6" tabindex="-1"></a>vast_scaling <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_impute, <span class="at">method =</span> <span class="st">&quot;vast_scaling&quot;</span>)</span>
<span id="cb34-7"><a href="data-processing.html#cb34-7" tabindex="-1"></a>se_normalize_v2 <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_impute, <span class="at">method =</span> <span class="st">&quot;log_pareto&quot;</span>)</span>
<span id="cb34-8"><a href="data-processing.html#cb34-8" tabindex="-1"></a>se_normalize_v2</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 180 55 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(180): M_38768 M_38296 ... M_31787 M_63361
## rowData names(0):
## colnames(55): P101001 P101003 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="comparison-of-unnormalized-and-normalized-dataset" class="section level3 hasAnchor" number="2.7.3">
<h3><span class="header-section-number">2.7.3</span> Comparison of unnormalized and normalized dataset<a href="data-processing.html#comparison-of-unnormalized-and-normalized-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>boxplot</li>
</ul>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="data-processing.html#cb36-1" tabindex="-1"></a>pl_unnor <span class="ot">&lt;-</span> <span class="fu">PomaBoxplots</span>(se_impute, <span class="at">group =</span> <span class="st">&quot;samples&quot;</span>, <span class="at">jitter =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb36-2"><a href="data-processing.html#cb36-2" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Not Normalized&quot;</span>) <span class="sc">+</span></span>
<span id="cb36-3"><a href="data-processing.html#cb36-3" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="co"># data before normalization</span></span>
<span id="cb36-4"><a href="data-processing.html#cb36-4" tabindex="-1"></a></span>
<span id="cb36-5"><a href="data-processing.html#cb36-5" tabindex="-1"></a>pl_nor <span class="ot">&lt;-</span> <span class="fu">PomaBoxplots</span>(se_normalize, <span class="at">group =</span> <span class="st">&quot;samples&quot;</span>, <span class="at">jitter =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb36-6"><a href="data-processing.html#cb36-6" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Normalized&quot;</span>) <span class="co"># data after normalization</span></span>
<span id="cb36-7"><a href="data-processing.html#cb36-7" tabindex="-1"></a></span>
<span id="cb36-8"><a href="data-processing.html#cb36-8" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(pl_unnor, pl_nor, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">align =</span> <span class="st">&quot;v&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-88-1.png" width="1344" /></p>
<ul>
<li>density</li>
</ul>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="data-processing.html#cb37-1" tabindex="-1"></a>pl_unnor <span class="ot">&lt;-</span> <span class="fu">PomaDensity</span>(se_impute, <span class="at">group =</span> <span class="st">&quot;features&quot;</span>) <span class="sc">+</span></span>
<span id="cb37-2"><a href="data-processing.html#cb37-2" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Not Normalized&quot;</span>) <span class="sc">+</span></span>
<span id="cb37-3"><a href="data-processing.html#cb37-3" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="co"># data before normalization</span></span>
<span id="cb37-4"><a href="data-processing.html#cb37-4" tabindex="-1"></a></span>
<span id="cb37-5"><a href="data-processing.html#cb37-5" tabindex="-1"></a>pl_nor <span class="ot">&lt;-</span> <span class="fu">PomaDensity</span>(se_normalize, <span class="at">group =</span> <span class="st">&quot;features&quot;</span>) <span class="sc">+</span></span>
<span id="cb37-6"><a href="data-processing.html#cb37-6" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Normalized&quot;</span>) <span class="co"># data after normalization</span></span>
<span id="cb37-7"><a href="data-processing.html#cb37-7" tabindex="-1"></a></span>
<span id="cb37-8"><a href="data-processing.html#cb37-8" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(pl_unnor, pl_nor, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">align =</span> <span class="st">&quot;v&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-89-1.png" width="960" /></p>
</div>
</div>
<div id="removing-outliers" class="section level2 hasAnchor" number="2.8">
<h2><span class="header-section-number">2.8</span> Removing outliers<a href="data-processing.html#removing-outliers" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Calculating the pairwise distance (distance measured by “euclidean”, “maximum”, “manhattan”, “canberra” or “minkowski”), and using 1.5 in Q3 + 1.5*IQR formula to detect outliers. More details to see <code>?POMA::PomaOutliers</code>.</p>
<p><strong>Caution</strong>: this step will remove samples regarded as outliers, please pay attention to your own data.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="data-processing.html#cb38-1" tabindex="-1"></a><span class="fu">PomaOutliers</span>(se_normalize, <span class="at">do =</span> <span class="st">&quot;analyze&quot;</span>)<span class="sc">$</span>polygon_plot <span class="co"># to explore</span></span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-90-1.png" width="960" /></p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="data-processing.html#cb39-1" tabindex="-1"></a>se_processed <span class="ot">&lt;-</span> <span class="fu">PomaOutliers</span>(se_normalize, <span class="at">do =</span> <span class="st">&quot;clean&quot;</span>) <span class="co"># to remove outliers</span></span>
<span id="cb39-2"><a href="data-processing.html#cb39-2" tabindex="-1"></a>se_processed</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 180 53 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(180): M_38768 M_38296 ... M_31787 M_63361
## rowData names(0):
## colnames(53): P101001 P101003 ... P101094 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="data-distribution" class="section level2 hasAnchor" number="2.9">
<h2><span class="header-section-number">2.9</span> Data distribution<a href="data-processing.html#data-distribution" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="data-processing.html#cb41-1" tabindex="-1"></a>get_distribution <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb41-2"><a href="data-processing.html#cb41-2" tabindex="-1"></a>    datset,</span>
<span id="cb41-3"><a href="data-processing.html#cb41-3" tabindex="-1"></a>    <span class="at">Type =</span> <span class="fu">c</span>(<span class="st">&quot;raw&quot;</span>, <span class="st">&quot;check&quot;</span>, <span class="st">&quot;filter&quot;</span>, </span>
<span id="cb41-4"><a href="data-processing.html#cb41-4" tabindex="-1"></a>             <span class="st">&quot;impute&quot;</span>, <span class="st">&quot;norm_relative&quot;</span>, <span class="st">&quot;norm_log10&quot;</span>)){</span>
<span id="cb41-5"><a href="data-processing.html#cb41-5" tabindex="-1"></a>  </span>
<span id="cb41-6"><a href="data-processing.html#cb41-6" tabindex="-1"></a>  <span class="co"># datset = se</span></span>
<span id="cb41-7"><a href="data-processing.html#cb41-7" tabindex="-1"></a>  <span class="co"># Type = &quot;raw&quot;</span></span>
<span id="cb41-8"><a href="data-processing.html#cb41-8" tabindex="-1"></a>  </span>
<span id="cb41-9"><a href="data-processing.html#cb41-9" tabindex="-1"></a>  dat <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(datset) <span class="sc">%&gt;%</span> </span>
<span id="cb41-10"><a href="data-processing.html#cb41-10" tabindex="-1"></a>      <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb41-11"><a href="data-processing.html#cb41-11" tabindex="-1"></a>      <span class="fu">rownames_to_column</span>(<span class="st">&quot;name&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb41-12"><a href="data-processing.html#cb41-12" tabindex="-1"></a>      tidyr<span class="sc">::</span><span class="fu">gather</span>(<span class="at">key=</span><span class="st">&quot;sample&quot;</span>, <span class="at">value=</span><span class="st">&quot;value&quot;</span>, <span class="sc">-</span>name)    </span>
<span id="cb41-13"><a href="data-processing.html#cb41-13" tabindex="-1"></a>  </span>
<span id="cb41-14"><a href="data-processing.html#cb41-14" tabindex="-1"></a>  <span class="cf">if</span> (Type <span class="sc">==</span> <span class="st">&quot;raw&quot;</span>) {</span>
<span id="cb41-15"><a href="data-processing.html#cb41-15" tabindex="-1"></a>    pl <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(dat, <span class="fu">aes</span>(<span class="at">x=</span>value)) <span class="sc">+</span> </span>
<span id="cb41-16"><a href="data-processing.html#cb41-16" tabindex="-1"></a>            <span class="fu">geom_histogram</span>(<span class="at">colour=</span><span class="st">&quot;black&quot;</span>, <span class="at">fill=</span><span class="st">&quot;white&quot;</span>)<span class="sc">+</span></span>
<span id="cb41-17"><a href="data-processing.html#cb41-17" tabindex="-1"></a>            <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Distribution of Raw </span><span class="sc">\n</span><span class="st"> Metabolites Intensity&quot;</span>, </span>
<span id="cb41-18"><a href="data-processing.html#cb41-18" tabindex="-1"></a>                 <span class="at">x=</span><span class="st">&quot;Raw Intensity&quot;</span>, <span class="at">y=</span><span class="st">&quot;Frequency&quot;</span>)<span class="sc">+</span></span>
<span id="cb41-19"><a href="data-processing.html#cb41-19" tabindex="-1"></a>            <span class="fu">theme_bw</span>()    </span>
<span id="cb41-20"><a href="data-processing.html#cb41-20" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (Type <span class="sc">==</span> <span class="st">&quot;check&quot;</span>) {</span>
<span id="cb41-21"><a href="data-processing.html#cb41-21" tabindex="-1"></a>    pl <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(dat, <span class="fu">aes</span>(<span class="at">x=</span>value)) <span class="sc">+</span> </span>
<span id="cb41-22"><a href="data-processing.html#cb41-22" tabindex="-1"></a>            <span class="fu">geom_histogram</span>(<span class="at">colour=</span><span class="st">&quot;black&quot;</span>, <span class="at">fill=</span><span class="st">&quot;white&quot;</span>)<span class="sc">+</span></span>
<span id="cb41-23"><a href="data-processing.html#cb41-23" tabindex="-1"></a>            <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Distribution of check </span><span class="sc">\n</span><span class="st"> Metabolites Intensity&quot;</span>, </span>
<span id="cb41-24"><a href="data-processing.html#cb41-24" tabindex="-1"></a>                 <span class="at">x=</span><span class="st">&quot;checked Intensity&quot;</span>, <span class="at">y=</span><span class="st">&quot;Frequency&quot;</span>)<span class="sc">+</span></span>
<span id="cb41-25"><a href="data-processing.html#cb41-25" tabindex="-1"></a>            <span class="fu">theme_bw</span>()    </span>
<span id="cb41-26"><a href="data-processing.html#cb41-26" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (Type <span class="sc">==</span> <span class="st">&quot;filter&quot;</span>) {</span>
<span id="cb41-27"><a href="data-processing.html#cb41-27" tabindex="-1"></a>    pl <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(dat, <span class="fu">aes</span>(<span class="at">x=</span>value)) <span class="sc">+</span> </span>
<span id="cb41-28"><a href="data-processing.html#cb41-28" tabindex="-1"></a>            <span class="fu">geom_histogram</span>(<span class="at">colour=</span><span class="st">&quot;black&quot;</span>, <span class="at">fill=</span><span class="st">&quot;white&quot;</span>)<span class="sc">+</span></span>
<span id="cb41-29"><a href="data-processing.html#cb41-29" tabindex="-1"></a>            <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Distribution of filter</span><span class="sc">\n</span><span class="st"> Metabolites Intensity&quot;</span>, </span>
<span id="cb41-30"><a href="data-processing.html#cb41-30" tabindex="-1"></a>                 <span class="at">x=</span><span class="st">&quot;filtered Intensity&quot;</span>, <span class="at">y=</span><span class="st">&quot;Frequency&quot;</span>)<span class="sc">+</span></span>
<span id="cb41-31"><a href="data-processing.html#cb41-31" tabindex="-1"></a>            <span class="fu">theme_bw</span>()    </span>
<span id="cb41-32"><a href="data-processing.html#cb41-32" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (Type <span class="sc">==</span> <span class="st">&quot;impute&quot;</span>) {</span>
<span id="cb41-33"><a href="data-processing.html#cb41-33" tabindex="-1"></a>    pl <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(dat, <span class="fu">aes</span>(<span class="at">x=</span>value)) <span class="sc">+</span> </span>
<span id="cb41-34"><a href="data-processing.html#cb41-34" tabindex="-1"></a>            <span class="fu">geom_histogram</span>(<span class="at">colour=</span><span class="st">&quot;black&quot;</span>, <span class="at">fill=</span><span class="st">&quot;white&quot;</span>)<span class="sc">+</span></span>
<span id="cb41-35"><a href="data-processing.html#cb41-35" tabindex="-1"></a>            <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Distribution of impute</span><span class="sc">\n</span><span class="st"> Metabolites Intensity&quot;</span>, </span>
<span id="cb41-36"><a href="data-processing.html#cb41-36" tabindex="-1"></a>                 <span class="at">x=</span><span class="st">&quot;imputed Intensity&quot;</span>, <span class="at">y=</span><span class="st">&quot;Frequency&quot;</span>)<span class="sc">+</span></span>
<span id="cb41-37"><a href="data-processing.html#cb41-37" tabindex="-1"></a>            <span class="fu">theme_bw</span>()    </span>
<span id="cb41-38"><a href="data-processing.html#cb41-38" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (Type <span class="sc">==</span> <span class="st">&quot;norm&quot;</span>) {</span>
<span id="cb41-39"><a href="data-processing.html#cb41-39" tabindex="-1"></a>    pl <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(dat, <span class="fu">aes</span>(<span class="at">x=</span>value)) <span class="sc">+</span> </span>
<span id="cb41-40"><a href="data-processing.html#cb41-40" tabindex="-1"></a>            <span class="fu">geom_histogram</span>(<span class="at">colour=</span><span class="st">&quot;black&quot;</span>, <span class="at">fill=</span><span class="st">&quot;white&quot;</span>)<span class="sc">+</span></span>
<span id="cb41-41"><a href="data-processing.html#cb41-41" tabindex="-1"></a>            <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Distribution of norm</span><span class="sc">\n</span><span class="st"> Metabolites Intensity&quot;</span>, </span>
<span id="cb41-42"><a href="data-processing.html#cb41-42" tabindex="-1"></a>                 <span class="at">x=</span><span class="st">&quot;norm&quot;</span>, <span class="at">y=</span><span class="st">&quot;Frequency&quot;</span>)<span class="sc">+</span></span>
<span id="cb41-43"><a href="data-processing.html#cb41-43" tabindex="-1"></a>            <span class="fu">theme_bw</span>()    </span>
<span id="cb41-44"><a href="data-processing.html#cb41-44" tabindex="-1"></a>  }</span>
<span id="cb41-45"><a href="data-processing.html#cb41-45" tabindex="-1"></a>  </span>
<span id="cb41-46"><a href="data-processing.html#cb41-46" tabindex="-1"></a>  <span class="fu">return</span>(pl)</span>
<span id="cb41-47"><a href="data-processing.html#cb41-47" tabindex="-1"></a>}</span>
<span id="cb41-48"><a href="data-processing.html#cb41-48" tabindex="-1"></a></span>
<span id="cb41-49"><a href="data-processing.html#cb41-49" tabindex="-1"></a>raw_pl <span class="ot">&lt;-</span> <span class="fu">get_distribution</span>(<span class="at">datset=</span>se_raw, <span class="at">Type=</span><span class="st">&quot;raw&quot;</span>)</span>
<span id="cb41-50"><a href="data-processing.html#cb41-50" tabindex="-1"></a>check_pl <span class="ot">&lt;-</span> <span class="fu">get_distribution</span>(<span class="at">datset=</span>se_check, <span class="at">Type=</span><span class="st">&quot;check&quot;</span>)</span>
<span id="cb41-51"><a href="data-processing.html#cb41-51" tabindex="-1"></a>filter_pl <span class="ot">&lt;-</span> <span class="fu">get_distribution</span>(<span class="at">datset=</span>se_filter, <span class="at">Type=</span><span class="st">&quot;filter&quot;</span>)</span>
<span id="cb41-52"><a href="data-processing.html#cb41-52" tabindex="-1"></a>impute_pl <span class="ot">&lt;-</span> <span class="fu">get_distribution</span>(<span class="at">datset=</span>se_impute, <span class="at">Type=</span><span class="st">&quot;impute&quot;</span>)</span>
<span id="cb41-53"><a href="data-processing.html#cb41-53" tabindex="-1"></a>norm_pl <span class="ot">&lt;-</span> <span class="fu">get_distribution</span>(<span class="at">datset=</span>se_normalize_v2, <span class="at">Type=</span><span class="st">&quot;norm&quot;</span>)</span>
<span id="cb41-54"><a href="data-processing.html#cb41-54" tabindex="-1"></a></span>
<span id="cb41-55"><a href="data-processing.html#cb41-55" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(raw_pl, check_pl, </span>
<span id="cb41-56"><a href="data-processing.html#cb41-56" tabindex="-1"></a>                   filter_pl, impute_pl,</span>
<span id="cb41-57"><a href="data-processing.html#cb41-57" tabindex="-1"></a>                   norm_pl,</span>
<span id="cb41-58"><a href="data-processing.html#cb41-58" tabindex="-1"></a>                   <span class="at">align =</span> <span class="st">&quot;hv&quot;</span>, <span class="at">nrow =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-91-1.png" width="960" /></p>
</div>
<div id="saving-datasets-into-rds-files" class="section level2 hasAnchor" number="2.10">
<h2><span class="header-section-number">2.10</span> Saving datasets into RDS files<a href="data-processing.html#saving-datasets-into-rds-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="data-processing.html#cb42-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="st">&quot;./dataset/POMA/&quot;</span>)) {</span>
<span id="cb42-2"><a href="data-processing.html#cb42-2" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="st">&quot;./dataset/POMA/&quot;</span>)</span>
<span id="cb42-3"><a href="data-processing.html#cb42-3" tabindex="-1"></a>}</span>
<span id="cb42-4"><a href="data-processing.html#cb42-4" tabindex="-1"></a></span>
<span id="cb42-5"><a href="data-processing.html#cb42-5" tabindex="-1"></a><span class="fu">saveRDS</span>(ExprSet, <span class="st">&quot;./dataset/POMA/ExprSet_raw.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span>
<span id="cb42-6"><a href="data-processing.html#cb42-6" tabindex="-1"></a><span class="fu">saveRDS</span>(se_raw, <span class="st">&quot;./dataset/POMA/se_raw.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span>
<span id="cb42-7"><a href="data-processing.html#cb42-7" tabindex="-1"></a><span class="fu">saveRDS</span>(se_check, <span class="st">&quot;./dataset/POMA/se_check.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span>
<span id="cb42-8"><a href="data-processing.html#cb42-8" tabindex="-1"></a><span class="fu">saveRDS</span>(se_filter, <span class="st">&quot;./dataset/POMA/se_filter.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span>
<span id="cb42-9"><a href="data-processing.html#cb42-9" tabindex="-1"></a><span class="fu">saveRDS</span>(se_impute, <span class="st">&quot;./dataset/POMA/se_impute.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span>
<span id="cb42-10"><a href="data-processing.html#cb42-10" tabindex="-1"></a><span class="fu">saveRDS</span>(se_normalize, <span class="st">&quot;./dataset/POMA/se_normalize.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span>
<span id="cb42-11"><a href="data-processing.html#cb42-11" tabindex="-1"></a><span class="fu">saveRDS</span>(se_processed, <span class="st">&quot;./dataset/POMA/se_processed.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="systematic-information" class="section level2 hasAnchor" number="2.11">
<h2><span class="header-section-number">2.11</span> Systematic Information<a href="data-processing.html#systematic-information" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="data-processing.html#cb43-1" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 4.1.3 (2022-03-10)
##  os       macOS Monterey 12.2.1
##  system   x86_64, darwin17.0
##  ui       RStudio
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       Asia/Shanghai
##  date     2023-12-07
##  rstudio  2023.09.0+463 Desert Sunflower (desktop)
##  pandoc   3.1.1 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/ (via rmarkdown)
## 
## ─ Packages ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package              * version    date (UTC) lib source
##  abind                  1.4-5      2016-07-21 [2] CRAN (R 4.1.0)
##  backports              1.4.1      2021-12-13 [2] CRAN (R 4.1.0)
##  base64enc              0.1-3      2015-07-28 [2] CRAN (R 4.1.0)
##  Biobase              * 2.54.0     2021-10-26 [2] Bioconductor
##  BiocGenerics         * 0.40.0     2021-10-26 [2] Bioconductor
##  BiocParallel           1.28.3     2021-12-09 [2] Bioconductor
##  bitops                 1.0-7      2021-04-24 [2] CRAN (R 4.1.0)
##  bookdown               0.34       2023-05-09 [2] CRAN (R 4.1.2)
##  broom                  1.0.5      2023-06-09 [2] CRAN (R 4.1.3)
##  bslib                  0.6.0      2023-11-21 [1] CRAN (R 4.1.3)
##  cachem                 1.0.8      2023-05-01 [2] CRAN (R 4.1.2)
##  callr                  3.7.3      2022-11-02 [2] CRAN (R 4.1.2)
##  caret                  6.0-94     2023-03-21 [2] CRAN (R 4.1.2)
##  cellranger             1.1.0      2016-07-27 [2] CRAN (R 4.1.0)
##  checkmate              2.2.0      2023-04-27 [2] CRAN (R 4.1.2)
##  circlize               0.4.15     2022-05-10 [2] CRAN (R 4.1.2)
##  class                  7.3-22     2023-05-03 [2] CRAN (R 4.1.2)
##  cli                    3.6.1      2023-03-23 [2] CRAN (R 4.1.2)
##  clue                   0.3-64     2023-01-31 [2] CRAN (R 4.1.2)
##  cluster                2.1.4      2022-08-22 [2] CRAN (R 4.1.2)
##  codetools              0.2-19     2023-02-01 [2] CRAN (R 4.1.2)
##  colorspace             2.1-0      2023-01-23 [2] CRAN (R 4.1.2)
##  ComplexHeatmap         2.10.0     2021-10-26 [2] Bioconductor
##  corpcor                1.6.10     2021-09-16 [2] CRAN (R 4.1.0)
##  cowplot                1.1.1      2020-12-30 [2] CRAN (R 4.1.0)
##  crayon                 1.5.2      2022-09-29 [2] CRAN (R 4.1.2)
##  data.table             1.14.8     2023-02-17 [2] CRAN (R 4.1.2)
##  DelayedArray           0.20.0     2021-10-26 [2] Bioconductor
##  devtools               2.4.5      2022-10-11 [2] CRAN (R 4.1.2)
##  digest                 0.6.33     2023-07-07 [1] CRAN (R 4.1.3)
##  doParallel             1.0.17     2022-02-07 [2] CRAN (R 4.1.2)
##  doRNG                  1.8.6      2023-01-16 [2] CRAN (R 4.1.2)
##  dplyr                * 1.1.2      2023-04-20 [2] CRAN (R 4.1.2)
##  e1071                  1.7-13     2023-02-01 [2] CRAN (R 4.1.2)
##  ellipse                0.4.5      2023-04-05 [2] CRAN (R 4.1.2)
##  ellipsis               0.3.2      2021-04-29 [2] CRAN (R 4.1.0)
##  evaluate               0.21       2023-05-05 [2] CRAN (R 4.1.2)
##  fansi                  1.0.4      2023-01-22 [2] CRAN (R 4.1.2)
##  farver                 2.1.1      2022-07-06 [2] CRAN (R 4.1.2)
##  fastmap                1.1.1      2023-02-24 [2] CRAN (R 4.1.2)
##  foreach                1.5.2      2022-02-02 [2] CRAN (R 4.1.2)
##  foreign                0.8-84     2022-12-06 [2] CRAN (R 4.1.2)
##  forestplot             3.1.1      2022-12-06 [2] CRAN (R 4.1.2)
##  Formula                1.2-5      2023-02-24 [2] CRAN (R 4.1.2)
##  fs                     1.6.2      2023-04-25 [2] CRAN (R 4.1.2)
##  future                 1.33.0     2023-07-01 [2] CRAN (R 4.1.3)
##  future.apply           1.11.0     2023-05-21 [2] CRAN (R 4.1.3)
##  generics               0.1.3      2022-07-05 [2] CRAN (R 4.1.2)
##  GenomeInfoDb         * 1.30.1     2022-01-30 [2] Bioconductor
##  GenomeInfoDbData       1.2.7      2022-03-09 [2] Bioconductor
##  GenomicRanges        * 1.46.1     2021-11-18 [2] Bioconductor
##  GetoptLong             1.0.5      2020-12-15 [2] CRAN (R 4.1.0)
##  ggforce                0.4.1      2022-10-04 [2] CRAN (R 4.1.2)
##  ggplot2              * 3.4.2      2023-04-03 [2] CRAN (R 4.1.2)
##  ggraph               * 2.1.0.9000 2023-07-11 [1] Github (thomasp85/ggraph@febab71)
##  ggrepel                0.9.3      2023-02-03 [2] CRAN (R 4.1.2)
##  glasso                 1.11       2019-10-01 [2] CRAN (R 4.1.0)
##  glmnet                 4.1-7      2023-03-23 [2] CRAN (R 4.1.2)
##  GlobalOptions          0.1.2      2020-06-10 [2] CRAN (R 4.1.0)
##  globals                0.16.2     2022-11-21 [2] CRAN (R 4.1.2)
##  glue                 * 1.6.2      2022-02-24 [2] CRAN (R 4.1.2)
##  Gmisc                * 3.0.2      2023-03-13 [2] CRAN (R 4.1.2)
##  gmm                    1.8        2023-06-06 [2] CRAN (R 4.1.3)
##  gmp                    0.7-1      2023-02-07 [2] CRAN (R 4.1.2)
##  gower                  1.0.1      2022-12-22 [2] CRAN (R 4.1.2)
##  graphlayouts           1.0.0      2023-05-01 [2] CRAN (R 4.1.2)
##  gridExtra              2.3        2017-09-09 [2] CRAN (R 4.1.0)
##  gtable                 0.3.3      2023-03-21 [2] CRAN (R 4.1.2)
##  hardhat                1.3.0      2023-03-30 [2] CRAN (R 4.1.2)
##  highr                  0.10       2022-12-22 [2] CRAN (R 4.1.2)
##  Hmisc                  5.1-0      2023-05-08 [2] CRAN (R 4.1.2)
##  htmlTable            * 2.4.1      2022-07-07 [2] CRAN (R 4.1.2)
##  htmltools              0.5.7      2023-11-03 [1] CRAN (R 4.1.3)
##  htmlwidgets            1.6.2      2023-03-17 [2] CRAN (R 4.1.2)
##  httpuv                 1.6.11     2023-05-11 [2] CRAN (R 4.1.3)
##  httr                   1.4.6      2023-05-08 [2] CRAN (R 4.1.2)
##  igraph                 1.5.0      2023-06-16 [1] CRAN (R 4.1.3)
##  impute                 1.68.0     2021-10-26 [2] Bioconductor
##  imputeLCMD             2.1        2022-06-10 [2] CRAN (R 4.1.2)
##  ipred                  0.9-14     2023-03-09 [2] CRAN (R 4.1.2)
##  IRanges              * 2.28.0     2021-10-26 [2] Bioconductor
##  iterators              1.0.14     2022-02-05 [2] CRAN (R 4.1.2)
##  itertools              0.1-3      2014-03-12 [2] CRAN (R 4.1.0)
##  jquerylib              0.1.4      2021-04-26 [2] CRAN (R 4.1.0)
##  jsonlite               1.8.7      2023-06-29 [2] CRAN (R 4.1.3)
##  knitr                  1.43       2023-05-25 [2] CRAN (R 4.1.3)
##  labeling               0.4.2      2020-10-20 [2] CRAN (R 4.1.0)
##  later                  1.3.1      2023-05-02 [2] CRAN (R 4.1.2)
##  lattice                0.21-8     2023-04-05 [2] CRAN (R 4.1.2)
##  lava                   1.7.2.1    2023-02-27 [2] CRAN (R 4.1.2)
##  lazyeval               0.2.2      2019-03-15 [2] CRAN (R 4.1.0)
##  lifecycle              1.0.3      2022-10-07 [2] CRAN (R 4.1.2)
##  limma                  3.50.3     2022-04-07 [2] Bioconductor
##  listenv                0.9.0      2022-12-16 [2] CRAN (R 4.1.2)
##  lubridate              1.9.2      2023-02-10 [2] CRAN (R 4.1.2)
##  magrittr             * 2.0.3      2022-03-30 [2] CRAN (R 4.1.2)
##  MASS                   7.3-60     2023-05-04 [2] CRAN (R 4.1.2)
##  Matrix                 1.6-0      2023-07-08 [2] CRAN (R 4.1.3)
##  MatrixGenerics       * 1.6.0      2021-10-26 [2] Bioconductor
##  matrixStats          * 1.0.0      2023-06-02 [2] CRAN (R 4.1.3)
##  memoise                2.0.1      2021-11-26 [2] CRAN (R 4.1.0)
##  mgcv                   1.8-42     2023-03-02 [2] CRAN (R 4.1.2)
##  mime                   0.12       2021-09-28 [2] CRAN (R 4.1.0)
##  miniUI                 0.1.1.1    2018-05-18 [2] CRAN (R 4.1.0)
##  missForest             1.5        2022-04-14 [2] CRAN (R 4.1.2)
##  mixOmics               6.18.1     2021-11-18 [2] Bioconductor (R 4.1.2)
##  ModelMetrics           1.2.2.2    2020-03-17 [2] CRAN (R 4.1.0)
##  munsell                0.5.0      2018-06-12 [2] CRAN (R 4.1.0)
##  mvtnorm                1.2-2      2023-06-08 [2] CRAN (R 4.1.3)
##  nlme                   3.1-162    2023-01-31 [2] CRAN (R 4.1.2)
##  nnet                   7.3-19     2023-05-03 [2] CRAN (R 4.1.2)
##  norm                   1.0-11.1   2023-06-18 [2] CRAN (R 4.1.3)
##  parallelly             1.36.0     2023-05-26 [2] CRAN (R 4.1.3)
##  pcaMethods             1.86.0     2021-10-26 [2] Bioconductor
##  permute                0.9-7      2022-01-27 [2] CRAN (R 4.1.2)
##  pillar                 1.9.0      2023-03-22 [2] CRAN (R 4.1.2)
##  pkgbuild               1.4.2      2023-06-26 [2] CRAN (R 4.1.3)
##  pkgconfig              2.0.3      2019-09-22 [2] CRAN (R 4.1.0)
##  pkgload                1.3.2.1    2023-07-08 [2] CRAN (R 4.1.3)
##  plotly               * 4.10.2     2023-06-03 [2] CRAN (R 4.1.3)
##  plyr                   1.8.8      2022-11-11 [2] CRAN (R 4.1.2)
##  png                    0.1-8      2022-11-29 [2] CRAN (R 4.1.2)
##  polyclip               1.10-4     2022-10-20 [2] CRAN (R 4.1.2)
##  POMA                 * 1.7.2      2022-07-26 [2] Github (pcastellanoescuder/POMA@bc8a972)
##  preprocessCore         1.56.0     2021-10-26 [2] Bioconductor
##  prettyunits            1.1.1      2020-01-24 [2] CRAN (R 4.1.0)
##  pROC                   1.18.4     2023-07-06 [2] CRAN (R 4.1.3)
##  processx               3.8.2      2023-06-30 [2] CRAN (R 4.1.3)
##  prodlim                2023.03.31 2023-04-02 [2] CRAN (R 4.1.2)
##  profvis                0.3.8      2023-05-02 [2] CRAN (R 4.1.2)
##  promises               1.2.0.1    2021-02-11 [2] CRAN (R 4.1.0)
##  proxy                  0.4-27     2022-06-09 [2] CRAN (R 4.1.2)
##  ps                     1.7.5      2023-04-18 [2] CRAN (R 4.1.2)
##  purrr                  1.0.1      2023-01-10 [2] CRAN (R 4.1.2)
##  R6                     2.5.1      2021-08-19 [2] CRAN (R 4.1.0)
##  randomForest           4.7-1.1    2022-05-23 [2] CRAN (R 4.1.2)
##  RankProd               3.20.0     2021-10-26 [2] Bioconductor
##  rARPACK                0.11-0     2016-03-10 [2] CRAN (R 4.1.0)
##  RColorBrewer           1.1-3      2022-04-03 [2] CRAN (R 4.1.2)
##  Rcpp                 * 1.0.11     2023-07-06 [1] CRAN (R 4.1.3)
##  RCurl                  1.98-1.12  2023-03-27 [2] CRAN (R 4.1.2)
##  readxl               * 1.4.3      2023-07-06 [2] CRAN (R 4.1.3)
##  recipes                1.0.6      2023-04-25 [2] CRAN (R 4.1.2)
##  remotes                2.4.2      2021-11-30 [2] CRAN (R 4.1.0)
##  reshape2               1.4.4      2020-04-09 [2] CRAN (R 4.1.0)
##  rjson                  0.2.21     2022-01-09 [2] CRAN (R 4.1.2)
##  rlang                  1.1.1      2023-04-28 [1] CRAN (R 4.1.2)
##  rmarkdown              2.23       2023-07-01 [2] CRAN (R 4.1.3)
##  Rmpfr                  0.9-2      2023-04-22 [2] CRAN (R 4.1.2)
##  rngtools               1.5.2      2021-09-20 [2] CRAN (R 4.1.0)
##  ropls                  1.26.4     2022-01-11 [2] Bioconductor
##  rpart                  4.1.19     2022-10-21 [2] CRAN (R 4.1.2)
##  RSpectra               0.16-1     2022-04-24 [2] CRAN (R 4.1.2)
##  rstudioapi             0.15.0     2023-07-07 [2] CRAN (R 4.1.3)
##  S4Vectors            * 0.32.4     2022-03-29 [2] Bioconductor
##  sandwich               3.0-2      2022-06-15 [2] CRAN (R 4.1.2)
##  sass                   0.4.6      2023-05-03 [2] CRAN (R 4.1.2)
##  scales                 1.2.1      2022-08-20 [2] CRAN (R 4.1.2)
##  sessioninfo            1.2.2      2021-12-06 [2] CRAN (R 4.1.0)
##  shape                  1.4.6      2021-05-19 [2] CRAN (R 4.1.0)
##  shiny                  1.7.4.1    2023-07-06 [2] CRAN (R 4.1.3)
##  stringi                1.7.12     2023-01-11 [2] CRAN (R 4.1.2)
##  stringr                1.5.0      2022-12-02 [2] CRAN (R 4.1.2)
##  SummarizedExperiment * 1.24.0     2021-10-26 [2] Bioconductor
##  survival               3.5-5      2023-03-12 [2] CRAN (R 4.1.2)
##  tibble               * 3.2.1      2023-03-20 [2] CRAN (R 4.1.2)
##  tidygraph              1.2.3      2023-02-01 [2] CRAN (R 4.1.2)
##  tidyr                  1.3.0      2023-01-24 [2] CRAN (R 4.1.2)
##  tidyselect             1.2.0      2022-10-10 [2] CRAN (R 4.1.2)
##  timechange             0.2.0      2023-01-11 [2] CRAN (R 4.1.2)
##  timeDate               4022.108   2023-01-07 [2] CRAN (R 4.1.2)
##  tmvtnorm               1.5        2022-03-22 [2] CRAN (R 4.1.2)
##  tweenr                 2.0.2      2022-09-06 [2] CRAN (R 4.1.2)
##  urlchecker             1.0.1      2021-11-30 [2] CRAN (R 4.1.0)
##  usethis                2.2.2      2023-07-06 [2] CRAN (R 4.1.3)
##  utf8                   1.2.3      2023-01-31 [2] CRAN (R 4.1.2)
##  vctrs                  0.6.3      2023-06-14 [1] CRAN (R 4.1.3)
##  vegan                  2.6-4      2022-10-11 [2] CRAN (R 4.1.2)
##  viridis                0.6.3      2023-05-03 [2] CRAN (R 4.1.2)
##  viridisLite            0.4.2      2023-05-02 [2] CRAN (R 4.1.2)
##  withr                  2.5.0      2022-03-03 [2] CRAN (R 4.1.2)
##  xfun                   0.40       2023-08-09 [1] CRAN (R 4.1.3)
##  XML                    3.99-0.14  2023-03-19 [2] CRAN (R 4.1.2)
##  xtable                 1.8-4      2019-04-21 [2] CRAN (R 4.1.0)
##  XVector                0.34.0     2021-10-26 [2] Bioconductor
##  yaml                   2.3.7      2023-01-23 [2] CRAN (R 4.1.2)
##  zlibbioc               1.40.0     2021-10-26 [2] Bioconductor
##  zoo                    1.8-12     2023-04-13 [2] CRAN (R 4.1.2)
## 
##  [1] /Users/zouhua/Library/R/x86_64/4.1/library
##  [2] /Library/Frameworks/R.framework/Versions/4.1/Resources/library
## 
## ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-castellano2021pomashiny" class="csl-entry">
Castellano-Escuder, Pol, Raúl González-Domı́nguez, Francesc Carmona-Pontaque, Cristina Andrés-Lacueva, and Alex Sánchez-Pla. 2021. <span>“POMAShiny: A User-Friendly Web-Based Workflow for Metabolomics and Proteomics Data Analysis.”</span> <em>PLoS Computational Biology</em> 17 (7): e1009148.
</div>
<div id="ref-pang2020metaboanalystr" class="csl-entry">
Pang, Zhiqiang, Jasmine Chong, Shuzhao Li, and Jianguo Xia. 2020. <span>“MetaboAnalystR 3.0: Toward an Optimized Workflow for Global Metabolomics.”</span> <em>Metabolites</em> 10 (5): 186.
</div>
<div id="ref-zeybel2022multiomics" class="csl-entry">
Zeybel, Mujdat, Muhammad Arif, Xiangyu Li, Ozlem Altay, Hong Yang, Mengnan Shi, Murat Akyildiz, et al. 2022. <span>“Multiomics Analysis Reveals the Impact of Microbiota on Host Metabolism in Hepatic Steatosis.”</span> <em>Advanced Science</em> 9 (11): 2104373.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="statistical-analysis-1.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/Metabolomics_Aanlysis/blob/main/01-Data-Processing.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true,
"toc_depth": 3
}
});
});
</script>

</body>

</html>
