<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Data Processing | Data Analysis in Metabolomics</title>
  <meta name="description" content="A template for Data Analysis in Metabolomics" />
  <meta name="generator" content="bookdown 0.27 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Data Processing | Data Analysis in Metabolomics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A template for Data Analysis in Metabolomics" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Data Processing | Data Analysis in Metabolomics" />
  
  <meta name="twitter:description" content="A template for Data Analysis in Metabolomics" />
  

<meta name="author" content="Hua Zou and Bangzhuo Tong" />


<meta name="date" content="2022-08-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="cluster-analysis.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="assets/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Data Analysis in Metabolomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#what-is-metabolomics"><i class="fa fa-check"></i><b>1.1</b> What is metabolomics?</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#what-are-the-procedures-in-this-template"><i class="fa fa-check"></i><b>1.2</b> What are the procedures in this template?</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#statistical-analysis"><i class="fa fa-check"></i><b>1.2.1</b> Statistical Analysis</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#functional-analysis"><i class="fa fa-check"></i><b>1.2.2</b> Functional Analysis</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#software"><i class="fa fa-check"></i><b>1.3</b> Software</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#reference"><i class="fa fa-check"></i><b>1.4</b> Reference</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-processing.html"><a href="data-processing.html"><i class="fa fa-check"></i><b>2</b> Data Processing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-processing.html"><a href="data-processing.html#loading-packages"><i class="fa fa-check"></i><b>2.1</b> Loading packages</a></li>
<li class="chapter" data-level="2.2" data-path="data-processing.html"><a href="data-processing.html#importing-data"><i class="fa fa-check"></i><b>2.2</b> Importing data</a></li>
<li class="chapter" data-level="2.3" data-path="data-processing.html"><a href="data-processing.html#data-checking"><i class="fa fa-check"></i><b>2.3</b> Data Checking</a></li>
<li class="chapter" data-level="2.4" data-path="data-processing.html"><a href="data-processing.html#data-filtering"><i class="fa fa-check"></i><b>2.4</b> Data Filtering</a></li>
<li class="chapter" data-level="2.5" data-path="data-processing.html"><a href="data-processing.html#missing-value-imputation"><i class="fa fa-check"></i><b>2.5</b> Missing Value Imputation</a></li>
<li class="chapter" data-level="2.6" data-path="data-processing.html"><a href="data-processing.html#data-normalization"><i class="fa fa-check"></i><b>2.6</b> Data Normalization</a>
<ul>
<li class="chapter" data-level="2.6.1" data-path="data-processing.html"><a href="data-processing.html#normalization-by-normalizedata-function"><i class="fa fa-check"></i><b>2.6.1</b> Normalization by <strong>NormalizeData</strong> function</a></li>
<li class="chapter" data-level="2.6.2" data-path="data-processing.html"><a href="data-processing.html#normalization-by-poma-r-package"><i class="fa fa-check"></i><b>2.6.2</b> Normalization by <strong>POMA</strong> R package</a></li>
<li class="chapter" data-level="2.6.3" data-path="data-processing.html"><a href="data-processing.html#comparison-of-unnormalized-and-normalized-dataset"><i class="fa fa-check"></i><b>2.6.3</b> Comparison of unnormalized and normalized dataset</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="data-processing.html"><a href="data-processing.html#removing-outliers"><i class="fa fa-check"></i><b>2.7</b> Removing outliers</a></li>
<li class="chapter" data-level="2.8" data-path="data-processing.html"><a href="data-processing.html#saving-datasets-into-rds-files"><i class="fa fa-check"></i><b>2.8</b> Saving datasets into RDS files</a></li>
<li class="chapter" data-level="2.9" data-path="data-processing.html"><a href="data-processing.html#systematic-information"><i class="fa fa-check"></i><b>2.9</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cluster-analysis.html"><a href="cluster-analysis.html"><i class="fa fa-check"></i><b>3</b> Cluster Analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="cluster-analysis.html"><a href="cluster-analysis.html#loading-packages-1"><i class="fa fa-check"></i><b>3.1</b> Loading packages</a></li>
<li class="chapter" data-level="3.2" data-path="cluster-analysis.html"><a href="cluster-analysis.html#importing-data-1"><i class="fa fa-check"></i><b>3.2</b> Importing data</a></li>
<li class="chapter" data-level="3.3" data-path="cluster-analysis.html"><a href="cluster-analysis.html#hierarchical-clustering"><i class="fa fa-check"></i><b>3.3</b> Hierarchical Clustering</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="cluster-analysis.html"><a href="cluster-analysis.html#agglomerative-hierarchical-clustering"><i class="fa fa-check"></i><b>3.3.1</b> Agglomerative Hierarchical Clustering</a></li>
<li class="chapter" data-level="3.3.2" data-path="cluster-analysis.html"><a href="cluster-analysis.html#divisive-hierarchical-clustering"><i class="fa fa-check"></i><b>3.3.2</b> Divisive Hierarchical Clustering</a></li>
<li class="chapter" data-level="3.3.3" data-path="cluster-analysis.html"><a href="cluster-analysis.html#comparison"><i class="fa fa-check"></i><b>3.3.3</b> Comparison</a></li>
<li class="chapter" data-level="3.3.4" data-path="cluster-analysis.html"><a href="cluster-analysis.html#determining-optimal-clusters"><i class="fa fa-check"></i><b>3.3.4</b> Determining Optimal Clusters</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="cluster-analysis.html"><a href="cluster-analysis.html#partitional-clustering"><i class="fa fa-check"></i><b>3.4</b> Partitional Clustering</a></li>
<li class="chapter" data-level="3.5" data-path="cluster-analysis.html"><a href="cluster-analysis.html#systematic-information-1"><i class="fa fa-check"></i><b>3.5</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="chemometrics-analysis.html"><a href="chemometrics-analysis.html"><i class="fa fa-check"></i><b>4</b> Chemometrics Analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="chemometrics-analysis.html"><a href="chemometrics-analysis.html#loading-packages-2"><i class="fa fa-check"></i><b>4.1</b> Loading packages</a></li>
<li class="chapter" data-level="4.2" data-path="chemometrics-analysis.html"><a href="chemometrics-analysis.html#importing-data-2"><i class="fa fa-check"></i><b>4.2</b> Importing data</a></li>
<li class="chapter" data-level="4.3" data-path="chemometrics-analysis.html"><a href="chemometrics-analysis.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>4.3</b> Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="4.4" data-path="chemometrics-analysis.html"><a href="chemometrics-analysis.html#partial-least-squares-discriminant-analysis-pls-da"><i class="fa fa-check"></i><b>4.4</b> Partial Least Squares-Discriminant Analysis (PLS-DA)</a></li>
<li class="chapter" data-level="4.5" data-path="chemometrics-analysis.html"><a href="chemometrics-analysis.html#sparse-partial-least-squares-discriminant-analysis-spls-da"><i class="fa fa-check"></i><b>4.5</b> Sparse Partial Least Squares-Discriminant Analysis (sPLS-DA)</a></li>
<li class="chapter" data-level="4.6" data-path="chemometrics-analysis.html"><a href="chemometrics-analysis.html#orthogonal-partial-least-squares-discriminant-analysis-orthopls-da"><i class="fa fa-check"></i><b>4.6</b> Orthogonal Partial Least Squares-Discriminant Analysis (orthoPLS-DA)</a></li>
<li class="chapter" data-level="4.7" data-path="chemometrics-analysis.html"><a href="chemometrics-analysis.html#systematic-information-2"><i class="fa fa-check"></i><b>4.7</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="univariate-analysis.html"><a href="univariate-analysis.html"><i class="fa fa-check"></i><b>5</b> Univariate Analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="univariate-analysis.html"><a href="univariate-analysis.html#loading-packages-3"><i class="fa fa-check"></i><b>5.1</b> Loading packages</a></li>
<li class="chapter" data-level="5.2" data-path="univariate-analysis.html"><a href="univariate-analysis.html#importing-data-3"><i class="fa fa-check"></i><b>5.2</b> Importing data</a></li>
<li class="chapter" data-level="5.3" data-path="univariate-analysis.html"><a href="univariate-analysis.html#fold-change-analysis"><i class="fa fa-check"></i><b>5.3</b> Fold Change Analysis</a></li>
<li class="chapter" data-level="5.4" data-path="univariate-analysis.html"><a href="univariate-analysis.html#t-test"><i class="fa fa-check"></i><b>5.4</b> T Test</a></li>
<li class="chapter" data-level="5.5" data-path="univariate-analysis.html"><a href="univariate-analysis.html#wilcoxon-test"><i class="fa fa-check"></i><b>5.5</b> Wilcoxon Test</a></li>
<li class="chapter" data-level="5.6" data-path="univariate-analysis.html"><a href="univariate-analysis.html#limma-test"><i class="fa fa-check"></i><b>5.6</b> Limma Test</a></li>
<li class="chapter" data-level="5.7" data-path="univariate-analysis.html"><a href="univariate-analysis.html#volcano-plot"><i class="fa fa-check"></i><b>5.7</b> Volcano plot</a></li>
<li class="chapter" data-level="5.8" data-path="univariate-analysis.html"><a href="univariate-analysis.html#correlation-heatmaps"><i class="fa fa-check"></i><b>5.8</b> Correlation Heatmaps</a></li>
<li class="chapter" data-level="5.9" data-path="univariate-analysis.html"><a href="univariate-analysis.html#glasso-this-function-will-compute-a-gaussian-graphical-model-using-the-glmnet-package"><i class="fa fa-check"></i><b>5.9</b> glasso: this function will compute a Gaussian Graphical Model using the glmnet package</a></li>
<li class="chapter" data-level="5.10" data-path="univariate-analysis.html"><a href="univariate-analysis.html#systematic-information-3"><i class="fa fa-check"></i><b>5.10</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html"><i class="fa fa-check"></i><b>6</b> Multivariate analysis</a>
<ul>
<li class="chapter" data-level="6.1" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#importing-data-4"><i class="fa fa-check"></i><b>6.1</b> Importing data</a></li>
<li class="chapter" data-level="6.2" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#data-curation"><i class="fa fa-check"></i><b>6.2</b> Data curation</a></li>
<li class="chapter" data-level="6.3" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#regularized-generalized-linear-models"><i class="fa fa-check"></i><b>6.3</b> Regularized Generalized Linear Models</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#lasso-alpha-1"><i class="fa fa-check"></i><b>6.3.1</b> Lasso: alpha = 1</a></li>
<li class="chapter" data-level="6.3.2" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#ridge-alpha-0"><i class="fa fa-check"></i><b>6.3.2</b> Ridge: alpha = 0</a></li>
<li class="chapter" data-level="6.3.3" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#elasticnet-0-alpha-1"><i class="fa fa-check"></i><b>6.3.3</b> Elasticnet: 0 &lt; alpha &lt; 1</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#classification"><i class="fa fa-check"></i><b>6.4</b> Classification</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#random-forest"><i class="fa fa-check"></i><b>6.4.1</b> Random Forest</a></li>
<li class="chapter" data-level="6.4.2" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#support-vector-machine-svm"><i class="fa fa-check"></i><b>6.4.2</b> Support Vector Machine (SVM)</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#systematic-information-4"><i class="fa fa-check"></i><b>6.5</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="network-analysis.html"><a href="network-analysis.html"><i class="fa fa-check"></i><b>7</b> Network Analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="network-analysis.html"><a href="network-analysis.html#introduction-1"><i class="fa fa-check"></i><b>7.1</b> Introduction</a></li>
<li class="chapter" data-level="7.2" data-path="network-analysis.html"><a href="network-analysis.html#loading-packages-4"><i class="fa fa-check"></i><b>7.2</b> Loading packages</a></li>
<li class="chapter" data-level="7.3" data-path="network-analysis.html"><a href="network-analysis.html#importing-data-5"><i class="fa fa-check"></i><b>7.3</b> Importing data</a></li>
<li class="chapter" data-level="7.4" data-path="network-analysis.html"><a href="network-analysis.html#data-curation-1"><i class="fa fa-check"></i><b>7.4</b> Data curation</a></li>
<li class="chapter" data-level="7.5" data-path="network-analysis.html"><a href="network-analysis.html#associations-among-features"><i class="fa fa-check"></i><b>7.5</b> Associations Among Features</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="network-analysis.html"><a href="network-analysis.html#single-network-with-pearson-correlation-as-association-measure"><i class="fa fa-check"></i><b>7.5.1</b> Single network with Pearson correlation as association measure</a></li>
<li class="chapter" data-level="7.5.2" data-path="network-analysis.html"><a href="network-analysis.html#single-network-with-spearman-correlation-as-association-measure"><i class="fa fa-check"></i><b>7.5.2</b> Single network with spearman correlation as association measure</a></li>
<li class="chapter" data-level="7.5.3" data-path="network-analysis.html"><a href="network-analysis.html#single-network-with-wgcna-bicor-as-association-measure"><i class="fa fa-check"></i><b>7.5.3</b> Single network with WGCNA (bicor) as association measure</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="network-analysis.html"><a href="network-analysis.html#network-comparison"><i class="fa fa-check"></i><b>7.6</b> Network comparison</a>
<ul>
<li class="chapter" data-level="7.6.1" data-path="network-analysis.html"><a href="network-analysis.html#data-preparing"><i class="fa fa-check"></i><b>7.6.1</b> Data preparing</a></li>
<li class="chapter" data-level="7.6.2" data-path="network-analysis.html"><a href="network-analysis.html#building-network-model-3"><i class="fa fa-check"></i><b>7.6.2</b> Building network model</a></li>
<li class="chapter" data-level="7.6.3" data-path="network-analysis.html"><a href="network-analysis.html#network-analysis-1"><i class="fa fa-check"></i><b>7.6.3</b> Network analysis</a></li>
<li class="chapter" data-level="7.6.4" data-path="network-analysis.html"><a href="network-analysis.html#visualizing-the-network-3"><i class="fa fa-check"></i><b>7.6.4</b> Visualizing the network</a></li>
<li class="chapter" data-level="7.6.5" data-path="network-analysis.html"><a href="network-analysis.html#quantitative-network-comparison"><i class="fa fa-check"></i><b>7.6.5</b> Quantitative network comparison</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="network-analysis.html"><a href="network-analysis.html#systematic-information-5"><i class="fa fa-check"></i><b>7.7</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html"><i class="fa fa-check"></i><b>8</b> Enrichment Analysis</a>
<ul>
<li class="chapter" data-level="8.1" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#enviroment-set-up"><i class="fa fa-check"></i><b>8.1</b> Enviroment Set up</a></li>
<li class="chapter" data-level="8.2" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#over-representation-analysis-ora"><i class="fa fa-check"></i><b>8.2</b> Over representation analysis (ORA)</a></li>
<li class="chapter" data-level="8.3" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#clean-environment-1"><i class="fa fa-check"></i><b>8.3</b> Clean environment 1</a></li>
<li class="chapter" data-level="8.4" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#single-sample-profiling"><i class="fa fa-check"></i><b>8.4</b> Single Sample Profiling</a></li>
<li class="chapter" data-level="8.5" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#clean-environment-2"><i class="fa fa-check"></i><b>8.5</b> Clean environment 2</a></li>
<li class="chapter" data-level="8.6" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#quantitative-enrichment-analysis-qea"><i class="fa fa-check"></i><b>8.6</b> Quantitative Enrichment Analysis (QEA)</a></li>
<li class="chapter" data-level="8.7" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#systematic-information-6"><i class="fa fa-check"></i><b>8.7</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="pathway-analysis.html"><a href="pathway-analysis.html"><i class="fa fa-check"></i><b>9</b> Pathway Analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="pathway-analysis.html"><a href="pathway-analysis.html#enviroment-set-up-1"><i class="fa fa-check"></i><b>9.1</b> Enviroment Set up</a></li>
<li class="chapter" data-level="9.2" data-path="pathway-analysis.html"><a href="pathway-analysis.html#over-representation-analysis"><i class="fa fa-check"></i><b>9.2</b> Over representation analysis</a></li>
<li class="chapter" data-level="9.3" data-path="pathway-analysis.html"><a href="pathway-analysis.html#clean-environment-1-1"><i class="fa fa-check"></i><b>9.3</b> Clean environment 1</a></li>
<li class="chapter" data-level="9.4" data-path="pathway-analysis.html"><a href="pathway-analysis.html#concentration-table-qea"><i class="fa fa-check"></i><b>9.4</b> Concentration Table (QEA)</a></li>
<li class="chapter" data-level="9.5" data-path="pathway-analysis.html"><a href="pathway-analysis.html#systematic-information-7"><i class="fa fa-check"></i><b>9.5</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="example.html"><a href="example.html"><i class="fa fa-check"></i><b>10</b> Example</a>
<ul>
<li class="chapter" data-level="10.1" data-path="example.html"><a href="example.html#loading-packages-5"><i class="fa fa-check"></i><b>10.1</b> Loading packages</a></li>
<li class="chapter" data-level="10.2" data-path="example.html"><a href="example.html#importing-data-6"><i class="fa fa-check"></i><b>10.2</b> Importing data</a></li>
<li class="chapter" data-level="10.3" data-path="example.html"><a href="example.html#data-processing-1"><i class="fa fa-check"></i><b>10.3</b> Data Processing</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="example.html"><a href="example.html#data-checking-1"><i class="fa fa-check"></i><b>10.3.1</b> Data Checking</a></li>
<li class="chapter" data-level="10.3.2" data-path="example.html"><a href="example.html#data-filtering-1"><i class="fa fa-check"></i><b>10.3.2</b> Data Filtering</a></li>
<li class="chapter" data-level="10.3.3" data-path="example.html"><a href="example.html#missing-value-imputation-1"><i class="fa fa-check"></i><b>10.3.3</b> Missing Value Imputation</a></li>
<li class="chapter" data-level="10.3.4" data-path="example.html"><a href="example.html#data-normalization-1"><i class="fa fa-check"></i><b>10.3.4</b> Data Normalization</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="example.html"><a href="example.html#cluster-analysis-1"><i class="fa fa-check"></i><b>10.4</b> Cluster Analysis</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="example.html"><a href="example.html#hierarchical-clustering-1"><i class="fa fa-check"></i><b>10.4.1</b> Hierarchical Clustering</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="example.html"><a href="example.html#chemometrics-analysis-1"><i class="fa fa-check"></i><b>10.5</b> Chemometrics Analysis</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="example.html"><a href="example.html#partial-least-squares-discriminant-analysis-pls-da-1"><i class="fa fa-check"></i><b>10.5.1</b> Partial Least Squares-Discriminant Analysis (PLS-DA)</a></li>
<li class="chapter" data-level="10.5.2" data-path="example.html"><a href="example.html#sparse-partial-least-squares-discriminant-analysis-spls-da-1"><i class="fa fa-check"></i><b>10.5.2</b> Sparse Partial Least Squares-Discriminant Analysis (sPLS-DA)</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="example.html"><a href="example.html#univariate-analysis-1"><i class="fa fa-check"></i><b>10.6</b> Univariate Analysis</a>
<ul>
<li class="chapter" data-level="10.6.1" data-path="example.html"><a href="example.html#fold-change-analysis-1"><i class="fa fa-check"></i><b>10.6.1</b> Fold Change Analysis</a></li>
<li class="chapter" data-level="10.6.2" data-path="example.html"><a href="example.html#t-test-1"><i class="fa fa-check"></i><b>10.6.2</b> T Test</a></li>
<li class="chapter" data-level="10.6.3" data-path="example.html"><a href="example.html#volcano-plot-1"><i class="fa fa-check"></i><b>10.6.3</b> Volcano plot</a></li>
</ul></li>
<li class="chapter" data-level="10.7" data-path="example.html"><a href="example.html#feature-selection"><i class="fa fa-check"></i><b>10.7</b> Feature Selection</a>
<ul>
<li class="chapter" data-level="10.7.1" data-path="example.html"><a href="example.html#regularized-generalized-linear-models-lasso-alpha-1"><i class="fa fa-check"></i><b>10.7.1</b> Regularized Generalized Linear Models (Lasso: alpha = 1)</a></li>
<li class="chapter" data-level="10.7.2" data-path="example.html"><a href="example.html#classification-random-forest"><i class="fa fa-check"></i><b>10.7.2</b> Classification (Random Forest)</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="example.html"><a href="example.html#network-analysis-2"><i class="fa fa-check"></i><b>10.8</b> Network Analysis</a>
<ul>
<li class="chapter" data-level="10.8.1" data-path="example.html"><a href="example.html#data-curation-2"><i class="fa fa-check"></i><b>10.8.1</b> Data curation</a></li>
<li class="chapter" data-level="10.8.2" data-path="example.html"><a href="example.html#building-network-model-4"><i class="fa fa-check"></i><b>10.8.2</b> Building network model</a></li>
<li class="chapter" data-level="10.8.3" data-path="example.html"><a href="example.html#visualizing-the-network-4"><i class="fa fa-check"></i><b>10.8.3</b> Visualizing the network</a></li>
</ul></li>
<li class="chapter" data-level="10.9" data-path="example.html"><a href="example.html#network-analysis-by-wgcna"><i class="fa fa-check"></i><b>10.9</b> Network Analysis by WGCNA</a>
<ul>
<li class="chapter" data-level="10.9.1" data-path="example.html"><a href="example.html#data-curation-3"><i class="fa fa-check"></i><b>10.9.1</b> Data curation</a></li>
<li class="chapter" data-level="10.9.2" data-path="example.html"><a href="example.html#tuning-soft-thresholds"><i class="fa fa-check"></i><b>10.9.2</b> Tuning soft thresholds</a></li>
<li class="chapter" data-level="10.9.3" data-path="example.html"><a href="example.html#create-the-network-using-the-blockwisemodules"><i class="fa fa-check"></i><b>10.9.3</b> Create the network using the <code>blockwiseModules</code></a></li>
<li class="chapter" data-level="10.9.4" data-path="example.html"><a href="example.html#plot-modules"><i class="fa fa-check"></i><b>10.9.4</b> plot modules</a></li>
<li class="chapter" data-level="10.9.5" data-path="example.html"><a href="example.html#relationships-among-modules"><i class="fa fa-check"></i><b>10.9.5</b> Relationships among modules</a></li>
<li class="chapter" data-level="10.9.6" data-path="example.html"><a href="example.html#module-eigengene-correlation"><i class="fa fa-check"></i><b>10.9.6</b> Module (Eigengene) correlation</a></li>
<li class="chapter" data-level="10.9.7" data-path="example.html"><a href="example.html#relate-module-cluster-assignments-to-groups"><i class="fa fa-check"></i><b>10.9.7</b> Relate Module (cluster) Assignments to Groups</a></li>
<li class="chapter" data-level="10.9.8" data-path="example.html"><a href="example.html#generate-and-export-networks"><i class="fa fa-check"></i><b>10.9.8</b> Generate and Export Networks</a></li>
<li class="chapter" data-level="10.9.9" data-path="example.html"><a href="example.html#network-visualization"><i class="fa fa-check"></i><b>10.9.9</b> Network visualization</a></li>
</ul></li>
<li class="chapter" data-level="10.10" data-path="example.html"><a href="example.html#systematic-information-8"><i class="fa fa-check"></i><b>10.10</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data Analysis in Metabolomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="data-processing" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Data Processing<a href="data-processing.html#data-processing" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Although the horrible experience of data analysis by using <strong>MetaboAnalystR</strong> R package <span class="citation">(<a href="#ref-pang2020metaboanalystr" role="doc-biblioref">Pang et al. 2020</a>)</span>, its thought of data processing are very useful. Therefore, this template is based on the workflow from <strong>MetaboAnalystR</strong>.</p>
<p>We integrated R packages and our own scripts to build the data analysis template on metabolomic data. Particularly, we thanks very much for <strong>POMA</strong> R package <span class="citation">(<a href="#ref-castellano2021pomashiny" role="doc-biblioref">Castellano-Escuder et al. 2021</a>)</span>. <strong>POMA</strong> is a flexible data cleaning and statistical analysis processes in one comprehensible and user-friendly R package.</p>
<div id="loading-packages" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Loading packages<a href="data-processing.html#loading-packages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="data-processing.html#cb1-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">warning =</span> F)</span>
<span id="cb1-2"><a href="data-processing.html#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="data-processing.html#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code></pre></div>
<pre><code>## 
## Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="data-processing.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb5-2"><a href="data-processing.html#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(POMA)</span></code></pre></div>
<pre><code>## Welcome to POMA!
## Version 1.7.2
## POMAShiny app: https://github.com/pcastellanoescuder/POMAShiny
## For more detailed package information please visit https://pcastellanoescuder.github.io/POMA/</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="data-processing.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb7-2"><a href="data-processing.html#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggraph)</span>
<span id="cb7-3"><a href="data-processing.html#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plotly)</span></code></pre></div>
<pre><code>## 
## Attaching package: &#39;plotly&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:ggplot2&#39;:
## 
##     last_plot</code></pre>
<pre><code>## The following object is masked from &#39;package:stats&#39;:
## 
##     filter</code></pre>
<pre><code>## The following object is masked from &#39;package:graphics&#39;:
## 
##     layout</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="data-processing.html#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb12-2"><a href="data-processing.html#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span></code></pre></div>
<pre><code>## Loading required package: MatrixGenerics</code></pre>
<pre><code>## Loading required package: matrixStats</code></pre>
<pre><code>## 
## Attaching package: &#39;matrixStats&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:dplyr&#39;:
## 
##     count</code></pre>
<pre><code>## 
## Attaching package: &#39;MatrixGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:matrixStats&#39;:
## 
##     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse, colCounts, colCummaxs, colCummins,
##     colCumprods, colCumsums, colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs, colMads, colMaxs,
##     colMeans2, colMedians, colMins, colOrderStats, colProds, colQuantiles, colRanges, colRanks, colSdDiffs,
##     colSds, colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads, colWeightedMeans,
##     colWeightedMedians, colWeightedSds, colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
##     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods, rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs,
##     rowLogSumExps, rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins, rowOrderStats, rowProds,
##     rowQuantiles, rowRanges, rowRanks, rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
##     rowWeightedMads, rowWeightedMeans, rowWeightedMedians, rowWeightedSds, rowWeightedVars</code></pre>
<pre><code>## Loading required package: GenomicRanges</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## 
## Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     combine, intersect, setdiff, union</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colnames, dirname, do.call, duplicated, eval,
##     evalq, Filter, Find, get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget, order,
##     paste, pmax, pmax.int, pmin, pmin.int, Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,
##     table, tapply, union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## 
## Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:plotly&#39;:
## 
##     rename</code></pre>
<pre><code>## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     first, rename</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     expand.grid, I, unname</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## 
## Attaching package: &#39;IRanges&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:plotly&#39;:
## 
##     slice</code></pre>
<pre><code>## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     collapse, desc, slice</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with &#39;browseVignettes()&#39;. To cite Bioconductor, see
##     &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>## 
## Attaching package: &#39;Biobase&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:MatrixGenerics&#39;:
## 
##     rowMedians</code></pre>
<pre><code>## The following objects are masked from &#39;package:matrixStats&#39;:
## 
##     anyMissing, rowMedians</code></pre>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="data-processing.html#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># rm(list = ls())</span></span>
<span id="cb41-2"><a href="data-processing.html#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb41-3"><a href="data-processing.html#cb41-3" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">future.globals.maxSize =</span> <span class="dv">1000</span> <span class="sc">*</span> <span class="dv">1024</span><span class="sc">^</span><span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="importing-data" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Importing data<a href="data-processing.html#importing-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The dataset is from the Zeybel-2022 published paper <span class="citation">(<a href="#ref-zeybel2022multiomics" role="doc-biblioref">Zeybel et al. 2022</a>)</span>.</p>
<ul>
<li>features table</li>
</ul>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="data-processing.html#cb42-1" aria-hidden="true" tabindex="-1"></a>profile <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(<span class="st">&quot;./dataset/OmicsDataSet-Zeybel-2022.xlsx&quot;</span>, <span class="at">sheet =</span> <span class="dv">6</span>)</span>
<span id="cb42-2"><a href="data-processing.html#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(profile)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 67
##   BIOCHEMICAL               `SUPER PATHWAY` `SUB PATHWAY` `COMP ID` PLATFORM `CHEMICAL ID`    RI  MASS PUBCHEM CAS   KEGG 
##   &lt;chr&gt;                     &lt;chr&gt;           &lt;chr&gt;             &lt;dbl&gt; &lt;chr&gt;            &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt;
## 1 (14 or 15)-methylpalmita… Lipid           Fatty Acid, …     38768 LC/MS N…     100002945  5695  269. 8181;1… &lt;NA&gt;  C169…
## 2 (16 or 17)-methylstearat… Lipid           Fatty Acid, …     38296 LC/MS N…     100002356  5993  297. 3083779 2724… &lt;NA&gt; 
## 3 (2 or 3)-decenoate (10:1… Lipid           Medium Chain…     63436 LC/MS N…     100021502  4990  169. &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; 
## 4 (2,4 or 2,5)-dimethylphe… Xenobiotics     Food Compone…     62533 LC/MS N…     100020519  3474  201. &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt; 
## 5 (N(1) + N(8))-acetylsper… Amino Acid      Polyamine Me…     57814 LC/MS P…     100016038  3080  188. 123689… &lt;NA&gt;  C006…
## 6 (R)-3-hydroxybutyrylcarn… Lipid           Fatty Acid M…     43264 LC/MS P…     100003926  2400  248. 534816… &lt;NA&gt;  &lt;NA&gt; 
## # … with 56 more variables: `SampleID HMDBID` &lt;chr&gt;, P101001 &lt;dbl&gt;, P101012 &lt;dbl&gt;, P101030 &lt;dbl&gt;, P101031 &lt;dbl&gt;,
## #   P101050 &lt;dbl&gt;, P101059 &lt;dbl&gt;, P101071 &lt;dbl&gt;, P101072 &lt;dbl&gt;, P101084 &lt;dbl&gt;, P101003 &lt;dbl&gt;, P101004 &lt;dbl&gt;,
## #   P101013 &lt;dbl&gt;, P101016 &lt;dbl&gt;, P101017 &lt;dbl&gt;, P101038 &lt;dbl&gt;, P101051 &lt;dbl&gt;, P101061 &lt;dbl&gt;, P101062 &lt;dbl&gt;,
## #   P101074 &lt;dbl&gt;, P101075 &lt;dbl&gt;, P101076 &lt;dbl&gt;, P101085 &lt;dbl&gt;, P101088 &lt;dbl&gt;, P101007 &lt;dbl&gt;, P101018 &lt;dbl&gt;,
## #   P101019 &lt;dbl&gt;, P101041 &lt;dbl&gt;, P101052 &lt;dbl&gt;, P101064 &lt;dbl&gt;, P101065 &lt;dbl&gt;, P101077 &lt;dbl&gt;, P101090 &lt;dbl&gt;,
## #   P101094 &lt;dbl&gt;, P101009 &lt;dbl&gt;, P101010 &lt;dbl&gt;, P101021 &lt;dbl&gt;, P101022 &lt;dbl&gt;, P101042 &lt;dbl&gt;, P101054 &lt;dbl&gt;,
## #   P101056 &lt;dbl&gt;, P101067 &lt;dbl&gt;, P101068 &lt;dbl&gt;, P101079 &lt;dbl&gt;, P101095 &lt;dbl&gt;, P101096 &lt;dbl&gt;, P101011 &lt;dbl&gt;, …</code></pre>
<ul>
<li>metadata table</li>
</ul>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="data-processing.html#cb44-1" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(<span class="st">&quot;./dataset/OmicsDataSet-Zeybel-2022.xlsx&quot;</span>, <span class="at">sheet =</span> <span class="dv">2</span>)</span>
<span id="cb44-2"><a href="data-processing.html#cb44-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(metadata)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 11
##   PatientID Stage  Metabolomics Proteomics GutMetagenomics OralMetagenomics LiverFatClass Gender AlcoholConsumption Smoker
##   &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt;           &lt;chr&gt;            &lt;chr&gt;         &lt;chr&gt;  &lt;chr&gt;              &lt;chr&gt; 
## 1 P101001   Before Send         Send       Send            Send             Severe        Male   No                 No    
## 2 P101003   Before Send         Send       Send            Send             None          Female No                 No    
## 3 P101004   Before Send         Send       Send            Send             Moderate      Male   Yes                No    
## 4 P101007   Before Send         Send       Send            Send             Severe        Female No                 No    
## 5 P101009   Before Send         Send       Send            Send             Moderate      Male   No                 Yes   
## 6 P101010   Before Send         Send       Send            Send             Mild          Male   Yes                No    
## # … with 1 more variable: Age &lt;dbl&gt;</code></pre>
<ul>
<li>Data Preparation: SummarizedExperiment object</li>
</ul>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="data-processing.html#cb46-1" aria-hidden="true" tabindex="-1"></a>getSEobject <span class="ot">&lt;-</span> <span class="cf">function</span>(x, y) {</span>
<span id="cb46-2"><a href="data-processing.html#cb46-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-3"><a href="data-processing.html#cb46-3" aria-hidden="true" tabindex="-1"></a>  target <span class="ot">&lt;-</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb46-4"><a href="data-processing.html#cb46-4" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(Metabolomics <span class="sc">==</span> <span class="st">&quot;Send&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb46-5"><a href="data-processing.html#cb46-5" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(PatientID, LiverFatClass, Gender, Smoker, Age, AlcoholConsumption)</span>
<span id="cb46-6"><a href="data-processing.html#cb46-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-7"><a href="data-processing.html#cb46-7" aria-hidden="true" tabindex="-1"></a>  sid <span class="ot">&lt;-</span> <span class="fu">intersect</span>(target<span class="sc">$</span>PatientID, <span class="fu">colnames</span>(profile))</span>
<span id="cb46-8"><a href="data-processing.html#cb46-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-9"><a href="data-processing.html#cb46-9" aria-hidden="true" tabindex="-1"></a>  features <span class="ot">&lt;-</span> y <span class="sc">%&gt;%</span> </span>
<span id="cb46-10"><a href="data-processing.html#cb46-10" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(sid)) <span class="sc">%&gt;%</span></span>
<span id="cb46-11"><a href="data-processing.html#cb46-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">t</span>()</span>
<span id="cb46-12"><a href="data-processing.html#cb46-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(features) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;M_&quot;</span>, profile<span class="sc">$</span><span class="st">`</span><span class="at">COMP ID</span><span class="st">`</span>)</span>
<span id="cb46-13"><a href="data-processing.html#cb46-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-14"><a href="data-processing.html#cb46-14" aria-hidden="true" tabindex="-1"></a>  target <span class="ot">&lt;-</span> target[<span class="fu">pmatch</span>(sid, target<span class="sc">$</span>PatientID), , F]</span>
<span id="cb46-15"><a href="data-processing.html#cb46-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb46-16"><a href="data-processing.html#cb46-16" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb46-17"><a href="data-processing.html#cb46-17" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">features =</span> features)</span>
<span id="cb46-18"><a href="data-processing.html#cb46-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(res)  </span>
<span id="cb46-19"><a href="data-processing.html#cb46-19" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb46-20"><a href="data-processing.html#cb46-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-21"><a href="data-processing.html#cb46-21" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">getSEobject</span>(metadata, profile)</span>
<span id="cb46-22"><a href="data-processing.html#cb46-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-23"><a href="data-processing.html#cb46-23" aria-hidden="true" tabindex="-1"></a>se</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 1032 55 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(1032): M_38768 M_38296 ... M_57517 M_15581
## rowData names(0):
## colnames(55): P101001 P101003 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
<ul>
<li>Extract data for test dataset</li>
</ul>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="data-processing.html#cb48-1" aria-hidden="true" tabindex="-1"></a>get_testData <span class="ot">&lt;-</span> <span class="cf">function</span>(object, <span class="at">num =</span> <span class="dv">200</span>) {</span>
<span id="cb48-2"><a href="data-processing.html#cb48-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-3"><a href="data-processing.html#cb48-3" aria-hidden="true" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) <span class="sc">%&gt;%</span></span>
<span id="cb48-4"><a href="data-processing.html#cb48-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">t</span>()</span>
<span id="cb48-5"><a href="data-processing.html#cb48-5" aria-hidden="true" tabindex="-1"></a>  metadata_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object) <span class="sc">%&gt;%</span></span>
<span id="cb48-6"><a href="data-processing.html#cb48-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb48-7"><a href="data-processing.html#cb48-7" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;ID&quot;</span>)</span>
<span id="cb48-8"><a href="data-processing.html#cb48-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb48-9"><a href="data-processing.html#cb48-9" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> metadata_tab, </span>
<span id="cb48-10"><a href="data-processing.html#cb48-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">features =</span> features_tab[, <span class="dv">1</span><span class="sc">:</span>num])</span>
<span id="cb48-11"><a href="data-processing.html#cb48-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb48-12"><a href="data-processing.html#cb48-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb48-13"><a href="data-processing.html#cb48-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-14"><a href="data-processing.html#cb48-14" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">get_testData</span>(<span class="at">object =</span> se)</span>
<span id="cb48-15"><a href="data-processing.html#cb48-15" aria-hidden="true" tabindex="-1"></a>se</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 200 55 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(200): M_38768 M_38296 ... M_31787 M_63361
## rowData names(0):
## colnames(55): P101001 P101003 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="data-checking" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Data Checking<a href="data-processing.html#data-checking" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Features in PomaSummarizedExperiment object must have the following criterion:</p>
<ul>
<li><p>All data values are numeric.</p></li>
<li><p>A total of 0 (0%) missing values were detected.</p></li>
</ul>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="data-processing.html#cb50-1" aria-hidden="true" tabindex="-1"></a>CheckData <span class="ot">&lt;-</span> <span class="cf">function</span>(object) {</span>
<span id="cb50-2"><a href="data-processing.html#cb50-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-3"><a href="data-processing.html#cb50-3" aria-hidden="true" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object)</span>
<span id="cb50-4"><a href="data-processing.html#cb50-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-5"><a href="data-processing.html#cb50-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># numeric &amp; missing values</span></span>
<span id="cb50-6"><a href="data-processing.html#cb50-6" aria-hidden="true" tabindex="-1"></a>  int_mat <span class="ot">&lt;-</span> features_tab</span>
<span id="cb50-7"><a href="data-processing.html#cb50-7" aria-hidden="true" tabindex="-1"></a>  rowNms <span class="ot">&lt;-</span> <span class="fu">rownames</span>(int_mat)</span>
<span id="cb50-8"><a href="data-processing.html#cb50-8" aria-hidden="true" tabindex="-1"></a>  colNms <span class="ot">&lt;-</span> <span class="fu">colnames</span>(int_mat)</span>
<span id="cb50-9"><a href="data-processing.html#cb50-9" aria-hidden="true" tabindex="-1"></a>  naNms <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">is.na</span>(int_mat))</span>
<span id="cb50-10"><a href="data-processing.html#cb50-10" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(int_mat)) {</span>
<span id="cb50-11"><a href="data-processing.html#cb50-11" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">class</span>(int_mat[, i]) <span class="sc">==</span> <span class="st">&quot;integer64&quot;</span>) {</span>
<span id="cb50-12"><a href="data-processing.html#cb50-12" aria-hidden="true" tabindex="-1"></a>      int_mat[, i] <span class="ot">&lt;-</span> <span class="fu">as.double</span>(int_mat[, i])</span>
<span id="cb50-13"><a href="data-processing.html#cb50-13" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb50-14"><a href="data-processing.html#cb50-14" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb50-15"><a href="data-processing.html#cb50-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-16"><a href="data-processing.html#cb50-16" aria-hidden="true" tabindex="-1"></a>  num_mat <span class="ot">&lt;-</span> <span class="fu">apply</span>(int_mat, <span class="dv">2</span>, as.numeric)</span>
<span id="cb50-17"><a href="data-processing.html#cb50-17" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">sum</span>(<span class="fu">is.na</span>(num_mat)) <span class="sc">&gt;</span> naNms) {</span>
<span id="cb50-18"><a href="data-processing.html#cb50-18" aria-hidden="true" tabindex="-1"></a>    num_mat <span class="ot">&lt;-</span> <span class="fu">apply</span>(int_mat, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;,&quot;</span>,  <span class="st">&quot;&quot;</span>, x)))</span>
<span id="cb50-19"><a href="data-processing.html#cb50-19" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">sum</span>(<span class="fu">is.na</span>(num_mat)) <span class="sc">&gt;</span> naNms) {</span>
<span id="cb50-20"><a href="data-processing.html#cb50-20" aria-hidden="true" tabindex="-1"></a>      <span class="fu">message</span>(<span class="st">&quot;&lt;font color=</span><span class="sc">\&quot;</span><span class="st">red</span><span class="sc">\&quot;</span><span class="st">&gt;Non-numeric values were found and replaced by NA.&lt;/font&gt;&quot;</span>)</span>
<span id="cb50-21"><a href="data-processing.html#cb50-21" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb50-22"><a href="data-processing.html#cb50-22" aria-hidden="true" tabindex="-1"></a>      <span class="fu">message</span>(<span class="st">&quot;All data values are numeric.&quot;</span>)</span>
<span id="cb50-23"><a href="data-processing.html#cb50-23" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb50-24"><a href="data-processing.html#cb50-24" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb50-25"><a href="data-processing.html#cb50-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">&quot;All data values are numeric.&quot;</span>)</span>
<span id="cb50-26"><a href="data-processing.html#cb50-26" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb50-27"><a href="data-processing.html#cb50-27" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-28"><a href="data-processing.html#cb50-28" aria-hidden="true" tabindex="-1"></a>  int_mat <span class="ot">&lt;-</span> num_mat</span>
<span id="cb50-29"><a href="data-processing.html#cb50-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(int_mat) <span class="ot">&lt;-</span> rowNms</span>
<span id="cb50-30"><a href="data-processing.html#cb50-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(int_mat) <span class="ot">&lt;-</span> colNms</span>
<span id="cb50-31"><a href="data-processing.html#cb50-31" aria-hidden="true" tabindex="-1"></a>  varCol <span class="ot">&lt;-</span> <span class="fu">apply</span>(int_mat, <span class="dv">2</span>, var, <span class="at">na.rm =</span> T)</span>
<span id="cb50-32"><a href="data-processing.html#cb50-32" aria-hidden="true" tabindex="-1"></a>  constCol <span class="ot">&lt;-</span> (varCol <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> <span class="fu">is.na</span>(varCol))</span>
<span id="cb50-33"><a href="data-processing.html#cb50-33" aria-hidden="true" tabindex="-1"></a>  constNum <span class="ot">&lt;-</span> <span class="fu">sum</span>(constCol, <span class="at">na.rm =</span> T)</span>
<span id="cb50-34"><a href="data-processing.html#cb50-34" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (constNum <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb50-35"><a href="data-processing.html#cb50-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="fu">paste</span>(<span class="st">&quot;&lt;font color=</span><span class="sc">\&quot;</span><span class="st">red</span><span class="sc">\&quot;</span><span class="st">&gt;&quot;</span>, constNum, </span>
<span id="cb50-36"><a href="data-processing.html#cb50-36" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;features with a constant or single value across samples were found and deleted.&lt;/font&gt;&quot;</span>))</span>
<span id="cb50-37"><a href="data-processing.html#cb50-37" aria-hidden="true" tabindex="-1"></a>    int_mat <span class="ot">&lt;-</span> int_mat[, <span class="sc">!</span>constCol, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb50-38"><a href="data-processing.html#cb50-38" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb50-39"><a href="data-processing.html#cb50-39" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-40"><a href="data-processing.html#cb50-40" aria-hidden="true" tabindex="-1"></a>  totalCount <span class="ot">&lt;-</span> <span class="fu">nrow</span>(int_mat) <span class="sc">*</span> <span class="fu">ncol</span>(int_mat)</span>
<span id="cb50-41"><a href="data-processing.html#cb50-41" aria-hidden="true" tabindex="-1"></a>  naCount <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">is.na</span>(int_mat))</span>
<span id="cb50-42"><a href="data-processing.html#cb50-42" aria-hidden="true" tabindex="-1"></a>  naPercent <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> naCount<span class="sc">/</span>totalCount, <span class="dv">1</span>)</span>
<span id="cb50-43"><a href="data-processing.html#cb50-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-44"><a href="data-processing.html#cb50-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="fu">paste</span>(<span class="st">&quot;A total of &quot;</span>, naCount, <span class="st">&quot; (&quot;</span>, naPercent, </span>
<span id="cb50-45"><a href="data-processing.html#cb50-45" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;%) missing values were detected.&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))  </span>
<span id="cb50-46"><a href="data-processing.html#cb50-46" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-47"><a href="data-processing.html#cb50-47" aria-hidden="true" tabindex="-1"></a>  <span class="co"># save int_mat into se object </span></span>
<span id="cb50-48"><a href="data-processing.html#cb50-48" aria-hidden="true" tabindex="-1"></a>  target <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object) <span class="sc">%&gt;%</span></span>
<span id="cb50-49"><a href="data-processing.html#cb50-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb50-50"><a href="data-processing.html#cb50-50" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb50-51"><a href="data-processing.html#cb50-51" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb50-52"><a href="data-processing.html#cb50-52" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">features =</span> <span class="fu">t</span>(int_mat))</span>
<span id="cb50-53"><a href="data-processing.html#cb50-53" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-54"><a href="data-processing.html#cb50-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb50-55"><a href="data-processing.html#cb50-55" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb50-56"><a href="data-processing.html#cb50-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-57"><a href="data-processing.html#cb50-57" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">CheckData</span>(<span class="at">object =</span> se)</span></code></pre></div>
<pre><code>## All data values are numeric.</code></pre>
<pre><code>## A total of 1146 (10.4%) missing values were detected.</code></pre>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="data-processing.html#cb53-1" aria-hidden="true" tabindex="-1"></a>se</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 200 55 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(200): M_38768 M_38296 ... M_31787 M_63361
## rowData names(0):
## colnames(55): P101001 P101003 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="data-filtering" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Data Filtering<a href="data-processing.html#data-filtering" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The purpose of the data filtering is to identify and remove variables that are unlikely to be of use when modeling the data. No phenotype information are used in the filtering process, so the result can be used with any downstream analysis. This step is strongly recommended for untargeted metabolomics datasets (i.e. spectral binning data, peak lists) with large number of variables, many of them are from baseline noises. Filtering can usually improve the results. For details, please refer to the paper by Hackstadt, et al.</p>
<p>Non-informative variables can be characterized in three groups: 1) variables of very small values (close to baseline or detection limit) - these variables can be detected using mean or median; 2) variables that are near-constant values throughout the experiment conditions (housekeeping or homeostasis) - these variables can be detected using standard deviation (SD); or the robust estimate such as interquantile range (IQR); and 3) variables that show low repeatability - this can be measured using QC samples using the relative standard deviation(RSD = SD/mean). Features with high percent RSD should be removed from the subsequent analysis (the suggested threshold is 20% for LC-MS and 30% for GC-MS). For data filtering based on the first two categories, the following empirical rules are applied during data filtering:</p>
<ul>
<li>Less than 250 variables: 5% will be filtered;</li>
<li>Between 250 - 500 variables: 10% will be filtered;</li>
<li>Between 500 - 1000 variables: 25% will be filtered;</li>
<li>Over 1000 variables: 40% will be filtered;</li>
</ul>
<p>Filtering features if their RSDs are &gt; 25% in QC samples</p>
<ul>
<li><p>Interquantile range (IQR)</p></li>
<li><p>Standard deviation (SD)</p></li>
<li><p>Median absolute deviation (MAD)</p></li>
<li><p>Relative standard deviation (RSD = SD/mean)</p></li>
<li><p>Non-parametric relative standard deviation (MAD/median)</p></li>
<li><p>Mean intensity value</p></li>
<li><p>Median intensity value</p></li>
</ul>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="data-processing.html#cb55-1" aria-hidden="true" tabindex="-1"></a>FilterFeature <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb55-2"><a href="data-processing.html#cb55-2" aria-hidden="true" tabindex="-1"></a>                       object,</span>
<span id="cb55-3"><a href="data-processing.html#cb55-3" aria-hidden="true" tabindex="-1"></a>                       qc_label,</span>
<span id="cb55-4"><a href="data-processing.html#cb55-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="fu">c</span>(<span class="st">&quot;none&quot;</span>, <span class="st">&quot;iqr&quot;</span>, <span class="st">&quot;rsd&quot;</span>, </span>
<span id="cb55-5"><a href="data-processing.html#cb55-5" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;nrsd&quot;</span>, <span class="st">&quot;mean&quot;</span>, <span class="st">&quot;sd&quot;</span>,</span>
<span id="cb55-6"><a href="data-processing.html#cb55-6" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;mad&quot;</span>, <span class="st">&quot;median&quot;</span>),</span>
<span id="cb55-7"><a href="data-processing.html#cb55-7" aria-hidden="true" tabindex="-1"></a>                       <span class="at">rsd_cutoff =</span> <span class="dv">25</span>) {</span>
<span id="cb55-8"><a href="data-processing.html#cb55-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb55-9"><a href="data-processing.html#cb55-9" aria-hidden="true" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) </span>
<span id="cb55-10"><a href="data-processing.html#cb55-10" aria-hidden="true" tabindex="-1"></a>  metadata_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object) </span>
<span id="cb55-11"><a href="data-processing.html#cb55-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb55-12"><a href="data-processing.html#cb55-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># QC samples</span></span>
<span id="cb55-13"><a href="data-processing.html#cb55-13" aria-hidden="true" tabindex="-1"></a>  qc_samples <span class="ot">&lt;-</span> metadata_tab <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb55-14"><a href="data-processing.html#cb55-14" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(group <span class="sc">==</span> qc_label)</span>
<span id="cb55-15"><a href="data-processing.html#cb55-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">dim</span>(qc_samples)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb55-16"><a href="data-processing.html#cb55-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;No qc samples have been chosen, please check your input&quot;</span>)</span>
<span id="cb55-17"><a href="data-processing.html#cb55-17" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb55-18"><a href="data-processing.html#cb55-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb55-19"><a href="data-processing.html#cb55-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># QC samples&#39; feature table</span></span>
<span id="cb55-20"><a href="data-processing.html#cb55-20" aria-hidden="true" tabindex="-1"></a>  qc_feature <span class="ot">&lt;-</span> features_tab[, <span class="fu">colnames</span>(features_tab)<span class="sc">%in%</span><span class="fu">rownames</span>(qc_samples)] <span class="sc">%&gt;%</span></span>
<span id="cb55-21"><a href="data-processing.html#cb55-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">t</span>()</span>
<span id="cb55-22"><a href="data-processing.html#cb55-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb55-23"><a href="data-processing.html#cb55-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># filter features by QC RSD</span></span>
<span id="cb55-24"><a href="data-processing.html#cb55-24" aria-hidden="true" tabindex="-1"></a>  rsd <span class="ot">&lt;-</span> rsd_cutoff <span class="sc">/</span> <span class="dv">100</span></span>
<span id="cb55-25"><a href="data-processing.html#cb55-25" aria-hidden="true" tabindex="-1"></a>  sds <span class="ot">&lt;-</span> <span class="fu">apply</span>(qc_feature, <span class="dv">2</span>, sd, <span class="at">na.rm =</span> T)</span>
<span id="cb55-26"><a href="data-processing.html#cb55-26" aria-hidden="true" tabindex="-1"></a>  mns <span class="ot">&lt;-</span> <span class="fu">apply</span>(qc_feature, <span class="dv">2</span>, mean, <span class="at">na.rm =</span> T)</span>
<span id="cb55-27"><a href="data-processing.html#cb55-27" aria-hidden="true" tabindex="-1"></a>  rsd_vals <span class="ot">&lt;-</span> <span class="fu">abs</span>(sds<span class="sc">/</span>mns) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb55-28"><a href="data-processing.html#cb55-28" aria-hidden="true" tabindex="-1"></a>  gd_inx <span class="ot">&lt;-</span> rsd_vals <span class="sc">&lt;</span> rsd</span>
<span id="cb55-29"><a href="data-processing.html#cb55-29" aria-hidden="true" tabindex="-1"></a>  int_mat <span class="ot">&lt;-</span> features_tab[gd_inx, ]</span>
<span id="cb55-30"><a href="data-processing.html#cb55-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">&quot;Removed &quot;</span>, (<span class="fu">dim</span>(qc_feature)[<span class="dv">2</span>] <span class="sc">-</span> <span class="fu">dim</span>(int_mat)[<span class="dv">1</span>]), </span>
<span id="cb55-31"><a href="data-processing.html#cb55-31" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot; features based on QC RSD values. QC samples are excluded from downstream functional analysis.&quot;</span>)</span>
<span id="cb55-32"><a href="data-processing.html#cb55-32" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb55-33"><a href="data-processing.html#cb55-33" aria-hidden="true" tabindex="-1"></a>  <span class="co"># whether to filter features by percentage according to the number</span></span>
<span id="cb55-34"><a href="data-processing.html#cb55-34" aria-hidden="true" tabindex="-1"></a>  PerformFeatureFilter <span class="ot">&lt;-</span> <span class="cf">function</span>(datMatrix, </span>
<span id="cb55-35"><a href="data-processing.html#cb55-35" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">qc_method =</span> method,</span>
<span id="cb55-36"><a href="data-processing.html#cb55-36" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">remain_num =</span> <span class="cn">NULL</span>) {</span>
<span id="cb55-37"><a href="data-processing.html#cb55-37" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb55-38"><a href="data-processing.html#cb55-38" aria-hidden="true" tabindex="-1"></a>    dat <span class="ot">&lt;-</span> datMatrix</span>
<span id="cb55-39"><a href="data-processing.html#cb55-39" aria-hidden="true" tabindex="-1"></a>    feat_num <span class="ot">&lt;-</span> <span class="fu">ncol</span>(dat)</span>
<span id="cb55-40"><a href="data-processing.html#cb55-40" aria-hidden="true" tabindex="-1"></a>    feat_nms <span class="ot">&lt;-</span> <span class="fu">colnames</span>(dat)</span>
<span id="cb55-41"><a href="data-processing.html#cb55-41" aria-hidden="true" tabindex="-1"></a>    nm <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb55-42"><a href="data-processing.html#cb55-42" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;none&quot;</span> <span class="sc">&amp;&amp;</span> feat_num <span class="sc">&lt;</span> <span class="dv">5000</span>) { <span class="co"># only allow for less than 4000</span></span>
<span id="cb55-43"><a href="data-processing.html#cb55-43" aria-hidden="true" tabindex="-1"></a>      remain <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">TRUE</span>, feat_num)</span>
<span id="cb55-44"><a href="data-processing.html#cb55-44" aria-hidden="true" tabindex="-1"></a>      nm <span class="ot">&lt;-</span> <span class="st">&quot;No filtering was applied&quot;</span></span>
<span id="cb55-45"><a href="data-processing.html#cb55-45" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb55-46"><a href="data-processing.html#cb55-46" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;rsd&quot;</span>){</span>
<span id="cb55-47"><a href="data-processing.html#cb55-47" aria-hidden="true" tabindex="-1"></a>        sds <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, sd, <span class="at">na.rm =</span> T)</span>
<span id="cb55-48"><a href="data-processing.html#cb55-48" aria-hidden="true" tabindex="-1"></a>        mns <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, mean, <span class="at">na.rm =</span> T)</span>
<span id="cb55-49"><a href="data-processing.html#cb55-49" aria-hidden="true" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">abs</span>(sds<span class="sc">/</span>mns)</span>
<span id="cb55-50"><a href="data-processing.html#cb55-50" aria-hidden="true" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;Relative standard deviation&quot;</span></span>
<span id="cb55-51"><a href="data-processing.html#cb55-51" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;nrsd&quot;</span> ) {</span>
<span id="cb55-52"><a href="data-processing.html#cb55-52" aria-hidden="true" tabindex="-1"></a>        mads <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, mad, <span class="at">na.rm =</span> T)</span>
<span id="cb55-53"><a href="data-processing.html#cb55-53" aria-hidden="true" tabindex="-1"></a>        meds <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, median, <span class="at">na.rm =</span> T)</span>
<span id="cb55-54"><a href="data-processing.html#cb55-54" aria-hidden="true" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">abs</span>(mads<span class="sc">/</span>meds)</span>
<span id="cb55-55"><a href="data-processing.html#cb55-55" aria-hidden="true" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;Non-paramatric relative standard deviation&quot;</span></span>
<span id="cb55-56"><a href="data-processing.html#cb55-56" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;mean&quot;</span>) {</span>
<span id="cb55-57"><a href="data-processing.html#cb55-57" aria-hidden="true" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, mean, <span class="at">na.rm =</span> T)</span>
<span id="cb55-58"><a href="data-processing.html#cb55-58" aria-hidden="true" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;mean&quot;</span></span>
<span id="cb55-59"><a href="data-processing.html#cb55-59" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;sd&quot;</span>) {</span>
<span id="cb55-60"><a href="data-processing.html#cb55-60" aria-hidden="true" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, sd, <span class="at">na.rm =</span> T)</span>
<span id="cb55-61"><a href="data-processing.html#cb55-61" aria-hidden="true" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;standard deviation&quot;</span></span>
<span id="cb55-62"><a href="data-processing.html#cb55-62" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;mad&quot;</span>) {</span>
<span id="cb55-63"><a href="data-processing.html#cb55-63" aria-hidden="true" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, mad, <span class="at">na.rm =</span> T)</span>
<span id="cb55-64"><a href="data-processing.html#cb55-64" aria-hidden="true" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;Median absolute deviation&quot;</span></span>
<span id="cb55-65"><a href="data-processing.html#cb55-65" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;median&quot;</span>) {</span>
<span id="cb55-66"><a href="data-processing.html#cb55-66" aria-hidden="true" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, median, <span class="at">na.rm =</span> T)</span>
<span id="cb55-67"><a href="data-processing.html#cb55-67" aria-hidden="true" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;median&quot;</span></span>
<span id="cb55-68"><a href="data-processing.html#cb55-68" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;iqr&quot;</span>) { <span class="co"># iqr</span></span>
<span id="cb55-69"><a href="data-processing.html#cb55-69" aria-hidden="true" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, IQR, <span class="at">na.rm =</span> T)</span>
<span id="cb55-70"><a href="data-processing.html#cb55-70" aria-hidden="true" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;Interquantile Range&quot;</span></span>
<span id="cb55-71"><a href="data-processing.html#cb55-71" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb55-72"><a href="data-processing.html#cb55-72" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb55-73"><a href="data-processing.html#cb55-73" aria-hidden="true" tabindex="-1"></a>      <span class="co"># get the rank of the filtered variables</span></span>
<span id="cb55-74"><a href="data-processing.html#cb55-74" aria-hidden="true" tabindex="-1"></a>      rk <span class="ot">&lt;-</span> <span class="fu">rank</span>(<span class="sc">-</span>filter_val, <span class="at">ties.method =</span> <span class="st">&quot;random&quot;</span>)</span>
<span id="cb55-75"><a href="data-processing.html#cb55-75" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb55-76"><a href="data-processing.html#cb55-76" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">is.null</span>(remain_num)) { <span class="co"># apply empirical filtering based on data size</span></span>
<span id="cb55-77"><a href="data-processing.html#cb55-77" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span> (feat_num <span class="sc">&lt;</span> <span class="dv">250</span>) { <span class="co"># reduce 5%</span></span>
<span id="cb55-78"><a href="data-processing.html#cb55-78" aria-hidden="true" tabindex="-1"></a>            remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> feat_num <span class="sc">*</span> <span class="fl">0.95</span></span>
<span id="cb55-79"><a href="data-processing.html#cb55-79" aria-hidden="true" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Further feature filtering based on &quot;</span>, nm)</span>
<span id="cb55-80"><a href="data-processing.html#cb55-80" aria-hidden="true" tabindex="-1"></a>          } <span class="cf">else</span> <span class="cf">if</span> (feat_num <span class="sc">&lt;</span> <span class="dv">500</span>) { <span class="co"># reduce 10%</span></span>
<span id="cb55-81"><a href="data-processing.html#cb55-81" aria-hidden="true" tabindex="-1"></a>            remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> feat_num <span class="sc">*</span> <span class="fl">0.9</span></span>
<span id="cb55-82"><a href="data-processing.html#cb55-82" aria-hidden="true" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Further feature filtering based on &quot;</span>, nm)</span>
<span id="cb55-83"><a href="data-processing.html#cb55-83" aria-hidden="true" tabindex="-1"></a>          } <span class="cf">else</span> <span class="cf">if</span> (feat_num <span class="sc">&lt;</span> <span class="dv">1000</span>) { <span class="co"># reduce 25%</span></span>
<span id="cb55-84"><a href="data-processing.html#cb55-84" aria-hidden="true" tabindex="-1"></a>            remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> feat_num <span class="sc">*</span> <span class="fl">0.75</span></span>
<span id="cb55-85"><a href="data-processing.html#cb55-85" aria-hidden="true" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Further feature filtering based on &quot;</span>, nm)</span>
<span id="cb55-86"><a href="data-processing.html#cb55-86" aria-hidden="true" tabindex="-1"></a>          } <span class="cf">else</span> { <span class="co"># reduce 40%, if still over 5000, then only use top 5000</span></span>
<span id="cb55-87"><a href="data-processing.html#cb55-87" aria-hidden="true" tabindex="-1"></a>            remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> feat_num <span class="sc">*</span> <span class="fl">0.6</span></span>
<span id="cb55-88"><a href="data-processing.html#cb55-88" aria-hidden="true" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Further feature filtering based on &quot;</span>, nm)</span>
<span id="cb55-89"><a href="data-processing.html#cb55-89" aria-hidden="true" tabindex="-1"></a>          }</span>
<span id="cb55-90"><a href="data-processing.html#cb55-90" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb55-91"><a href="data-processing.html#cb55-91" aria-hidden="true" tabindex="-1"></a>        remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> remain_num</span>
<span id="cb55-92"><a href="data-processing.html#cb55-92" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb55-93"><a href="data-processing.html#cb55-93" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb55-94"><a href="data-processing.html#cb55-94" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb55-95"><a href="data-processing.html#cb55-95" aria-hidden="true" tabindex="-1"></a>    res <span class="ot">&lt;-</span> datMatrix[, remain]</span>
<span id="cb55-96"><a href="data-processing.html#cb55-96" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb55-97"><a href="data-processing.html#cb55-97" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(res)</span>
<span id="cb55-98"><a href="data-processing.html#cb55-98" aria-hidden="true" tabindex="-1"></a>  }  </span>
<span id="cb55-99"><a href="data-processing.html#cb55-99" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb55-100"><a href="data-processing.html#cb55-100" aria-hidden="true" tabindex="-1"></a>  feature_res <span class="ot">&lt;-</span> <span class="fu">PerformFeatureFilter</span>(<span class="fu">t</span>(int_mat))</span>
<span id="cb55-101"><a href="data-processing.html#cb55-101" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb55-102"><a href="data-processing.html#cb55-102" aria-hidden="true" tabindex="-1"></a>  <span class="co"># remove QC samples </span></span>
<span id="cb55-103"><a href="data-processing.html#cb55-103" aria-hidden="true" tabindex="-1"></a>  feature_final <span class="ot">&lt;-</span> feature_res[<span class="sc">!</span><span class="fu">rownames</span>(feature_res) <span class="sc">%in%</span> <span class="fu">rownames</span>(qc_samples), ]</span>
<span id="cb55-104"><a href="data-processing.html#cb55-104" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb55-105"><a href="data-processing.html#cb55-105" aria-hidden="true" tabindex="-1"></a>  <span class="co"># save int_mat into se object </span></span>
<span id="cb55-106"><a href="data-processing.html#cb55-106" aria-hidden="true" tabindex="-1"></a>  target <span class="ot">&lt;-</span> metadata_tab <span class="sc">%&gt;%</span> </span>
<span id="cb55-107"><a href="data-processing.html#cb55-107" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb55-108"><a href="data-processing.html#cb55-108" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb55-109"><a href="data-processing.html#cb55-109" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(SampleID <span class="sc">%in%</span> <span class="fu">rownames</span>(feature_final))</span>
<span id="cb55-110"><a href="data-processing.html#cb55-110" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb55-111"><a href="data-processing.html#cb55-111" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">features =</span> feature_final)</span>
<span id="cb55-112"><a href="data-processing.html#cb55-112" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb55-113"><a href="data-processing.html#cb55-113" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(res) </span>
<span id="cb55-114"><a href="data-processing.html#cb55-114" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb55-115"><a href="data-processing.html#cb55-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-116"><a href="data-processing.html#cb55-116" aria-hidden="true" tabindex="-1"></a>se_filter <span class="ot">&lt;-</span> <span class="fu">FilterFeature</span>(<span class="at">object =</span> se,</span>
<span id="cb55-117"><a href="data-processing.html#cb55-117" aria-hidden="true" tabindex="-1"></a>                           <span class="at">qc_label =</span> <span class="st">&quot;None&quot;</span>,</span>
<span id="cb55-118"><a href="data-processing.html#cb55-118" aria-hidden="true" tabindex="-1"></a>                           <span class="at">method =</span> <span class="st">&quot;iqr&quot;</span>)</span></code></pre></div>
<pre><code>## Removed 149 features based on QC RSD values. QC samples are excluded from downstream functional analysis.</code></pre>
<pre><code>## Further feature filtering based on Interquantile Range</code></pre>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="data-processing.html#cb58-1" aria-hidden="true" tabindex="-1"></a>se_filter</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 48 45 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(48): M_52603 M_19130 ... M_63681 M_63361
## rowData names(0):
## colnames(45): P101001 P101004 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="missing-value-imputation" class="section level2 hasAnchor" number="2.5">
<h2><span class="header-section-number">2.5</span> Missing Value Imputation<a href="data-processing.html#missing-value-imputation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="data-processing.html#cb60-1" aria-hidden="true" tabindex="-1"></a>se_impute <span class="ot">&lt;-</span> <span class="fu">PomaImpute</span>(</span>
<span id="cb60-2"><a href="data-processing.html#cb60-2" aria-hidden="true" tabindex="-1"></a>                    se_filter, </span>
<span id="cb60-3"><a href="data-processing.html#cb60-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ZerosAsNA =</span> <span class="cn">TRUE</span>, </span>
<span id="cb60-4"><a href="data-processing.html#cb60-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">RemoveNA =</span> <span class="cn">TRUE</span>, </span>
<span id="cb60-5"><a href="data-processing.html#cb60-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">cutoff =</span> <span class="dv">20</span>, </span>
<span id="cb60-6"><a href="data-processing.html#cb60-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;knn&quot;</span>)</span>
<span id="cb60-7"><a href="data-processing.html#cb60-7" aria-hidden="true" tabindex="-1"></a>se_impute</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 48 45 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(48): M_52603 M_19130 ... M_63681 M_63361
## rowData names(0):
## colnames(45): P101001 P101004 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="data-normalization" class="section level2 hasAnchor" number="2.6">
<h2><span class="header-section-number">2.6</span> Data Normalization<a href="data-processing.html#data-normalization" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The normalization procedures are grouped into three categories. You can use one or combine them to achieve better results.</p>
<ul>
<li><p>Sample normalization is for general-purpose adjustment for systematic differences among samples;</p>
<ul>
<li><p>Sample-specific normalization (i.e. weight, volume)</p></li>
<li><p>Normalization by sum</p></li>
<li><p>Normalization by median</p></li>
<li><p>Normalization by a reference sample (PQN)</p></li>
<li><p>Normalization by a pooled sample from group (group PQN)</p></li>
<li><p>Normalization by reference feature</p></li>
<li><p>Quantile normalization (suggested only for &gt; 1000 features)</p></li>
</ul></li>
<li><p>Data transformation applies a mathematical transformation on individual values themselves. A simple mathematical approach is used to deal with negative values in log and square root.</p>
<ul>
<li><p>Log transformation (base 10)</p></li>
<li><p>Square root transformation (square root of data values)</p></li>
<li><p>Cube root transformation (cube root of data values)</p></li>
</ul></li>
<li><p>Data scaling adjusts each variable/feature by a scaling factor computed based on the dispersion of the variable.</p>
<ul>
<li><p>Mean centering (mean-centered only)</p></li>
<li><p>Auto scaling (mean-centered and divided by the standard deviation of each variable)</p></li>
<li><p>Pareto scaling (mean-centered and divided by the square root of the standard deviation of each variable)</p></li>
<li><p>Range scaling (mean-centered and divided by the range of each variable)</p></li>
</ul></li>
</ul>
<div id="normalization-by-normalizedata-function" class="section level3 hasAnchor" number="2.6.1">
<h3><span class="header-section-number">2.6.1</span> Normalization by <strong>NormalizeData</strong> function<a href="data-processing.html#normalization-by-normalizedata-function" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="data-processing.html#cb62-1" aria-hidden="true" tabindex="-1"></a>NormalizeData <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb62-2"><a href="data-processing.html#cb62-2" aria-hidden="true" tabindex="-1"></a>                      object,</span>
<span id="cb62-3"><a href="data-processing.html#cb62-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">rowNorm =</span> <span class="fu">c</span>(<span class="st">&quot;Quantile&quot;</span>, <span class="st">&quot;GroupPQN&quot;</span>, <span class="st">&quot;SamplePQN&quot;</span>, </span>
<span id="cb62-4"><a href="data-processing.html#cb62-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;CompNorm&quot;</span>, <span class="st">&quot;SumNorm&quot;</span>, <span class="st">&quot;MedianNorm&quot;</span>,</span>
<span id="cb62-5"><a href="data-processing.html#cb62-5" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;SpecNorm&quot;</span>, <span class="st">&quot;None&quot;</span>),</span>
<span id="cb62-6"><a href="data-processing.html#cb62-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">transNorm =</span> <span class="fu">c</span>(<span class="st">&quot;LogNorm&quot;</span>, <span class="st">&quot;SrNorm&quot;</span>, <span class="st">&quot;CrNorm&quot;</span>, <span class="st">&quot;None&quot;</span>),</span>
<span id="cb62-7"><a href="data-processing.html#cb62-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">scaleNorm =</span> <span class="fu">c</span>(<span class="st">&quot;MeanCenter&quot;</span>, <span class="st">&quot;AutoNorm&quot;</span>, <span class="st">&quot;ParetoNorm&quot;</span>, </span>
<span id="cb62-8"><a href="data-processing.html#cb62-8" aria-hidden="true" tabindex="-1"></a>                                    <span class="st">&quot;RangeNorm&quot;</span>, <span class="st">&quot;None&quot;</span>),</span>
<span id="cb62-9"><a href="data-processing.html#cb62-9" aria-hidden="true" tabindex="-1"></a>                      <span class="at">ref =</span> <span class="cn">NULL</span>,</span>
<span id="cb62-10"><a href="data-processing.html#cb62-10" aria-hidden="true" tabindex="-1"></a>                      <span class="at">SpeWeight =</span> <span class="dv">1</span>) {</span>
<span id="cb62-11"><a href="data-processing.html#cb62-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-12"><a href="data-processing.html#cb62-12" aria-hidden="true" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) </span>
<span id="cb62-13"><a href="data-processing.html#cb62-13" aria-hidden="true" tabindex="-1"></a>  metadata_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object)   </span>
<span id="cb62-14"><a href="data-processing.html#cb62-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-15"><a href="data-processing.html#cb62-15" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">t</span>(features_tab)</span>
<span id="cb62-16"><a href="data-processing.html#cb62-16" aria-hidden="true" tabindex="-1"></a>  colNames <span class="ot">&lt;-</span> <span class="fu">colnames</span>(data)</span>
<span id="cb62-17"><a href="data-processing.html#cb62-17" aria-hidden="true" tabindex="-1"></a>  rowNames <span class="ot">&lt;-</span> <span class="fu">rownames</span>(data)</span>
<span id="cb62-18"><a href="data-processing.html#cb62-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-19"><a href="data-processing.html#cb62-19" aria-hidden="true" tabindex="-1"></a>  <span class="do">#############################################</span></span>
<span id="cb62-20"><a href="data-processing.html#cb62-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Sample normalization</span></span>
<span id="cb62-21"><a href="data-processing.html#cb62-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># perform quantile normalization on the raw data (can be log transformed later by user)</span></span>
<span id="cb62-22"><a href="data-processing.html#cb62-22" aria-hidden="true" tabindex="-1"></a>  QuantileNormalize <span class="ot">&lt;-</span> <span class="cf">function</span>(data) {</span>
<span id="cb62-23"><a href="data-processing.html#cb62-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">t</span>(preprocessCore<span class="sc">::</span><span class="fu">normalize.quantiles</span>(<span class="fu">t</span>(data), <span class="at">copy=</span><span class="cn">FALSE</span>)));</span>
<span id="cb62-24"><a href="data-processing.html#cb62-24" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb62-25"><a href="data-processing.html#cb62-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># normalize by a reference sample (probability quotient normalization)</span></span>
<span id="cb62-26"><a href="data-processing.html#cb62-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ref should be the name of the reference sample</span></span>
<span id="cb62-27"><a href="data-processing.html#cb62-27" aria-hidden="true" tabindex="-1"></a>  ProbNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x, ref_smpl) {</span>
<span id="cb62-28"><a href="data-processing.html#cb62-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(x<span class="sc">/</span><span class="fu">median</span>(<span class="fu">as.numeric</span>(x<span class="sc">/</span>ref_smpl), <span class="at">na.rm =</span> T))</span>
<span id="cb62-29"><a href="data-processing.html#cb62-29" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb62-30"><a href="data-processing.html#cb62-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-31"><a href="data-processing.html#cb62-31" aria-hidden="true" tabindex="-1"></a>  <span class="co"># normalize by a reference reference (i.e. creatinine)</span></span>
<span id="cb62-32"><a href="data-processing.html#cb62-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ref should be the name of the cmpd</span></span>
<span id="cb62-33"><a href="data-processing.html#cb62-33" aria-hidden="true" tabindex="-1"></a>  CompNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x, ref) {</span>
<span id="cb62-34"><a href="data-processing.html#cb62-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="dv">1000</span><span class="sc">*</span>x<span class="sc">/</span>x[ref])</span>
<span id="cb62-35"><a href="data-processing.html#cb62-35" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb62-36"><a href="data-processing.html#cb62-36" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-37"><a href="data-processing.html#cb62-37" aria-hidden="true" tabindex="-1"></a>  SumNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb62-38"><a href="data-processing.html#cb62-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="dv">1000</span><span class="sc">*</span>x<span class="sc">/</span><span class="fu">sum</span>(x, <span class="at">na.rm =</span> T))</span>
<span id="cb62-39"><a href="data-processing.html#cb62-39" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb62-40"><a href="data-processing.html#cb62-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-41"><a href="data-processing.html#cb62-41" aria-hidden="true" tabindex="-1"></a>  <span class="co"># normalize by median</span></span>
<span id="cb62-42"><a href="data-processing.html#cb62-42" aria-hidden="true" tabindex="-1"></a>  MedianNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb62-43"><a href="data-processing.html#cb62-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(x<span class="sc">/</span><span class="fu">median</span>(x, <span class="at">na.rm =</span> T))</span>
<span id="cb62-44"><a href="data-processing.html#cb62-44" aria-hidden="true" tabindex="-1"></a>  }  </span>
<span id="cb62-45"><a href="data-processing.html#cb62-45" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-46"><a href="data-processing.html#cb62-46" aria-hidden="true" tabindex="-1"></a>  <span class="co"># row-wise normalization</span></span>
<span id="cb62-47"><a href="data-processing.html#cb62-47" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;Quantile&quot;</span>) {</span>
<span id="cb62-48"><a href="data-processing.html#cb62-48" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">QuantileNormalize</span>(data)</span>
<span id="cb62-49"><a href="data-processing.html#cb62-49" aria-hidden="true" tabindex="-1"></a>    <span class="co"># this can introduce constant variables if a variable is </span></span>
<span id="cb62-50"><a href="data-processing.html#cb62-50" aria-hidden="true" tabindex="-1"></a>    <span class="co"># at the same rank across all samples (replaced by its average across all)</span></span>
<span id="cb62-51"><a href="data-processing.html#cb62-51" aria-hidden="true" tabindex="-1"></a>    varCol <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, var, <span class="at">na.rm =</span> T)</span>
<span id="cb62-52"><a href="data-processing.html#cb62-52" aria-hidden="true" tabindex="-1"></a>    constCol <span class="ot">&lt;-</span> (varCol <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> <span class="fu">is.na</span>(varCol))</span>
<span id="cb62-53"><a href="data-processing.html#cb62-53" aria-hidden="true" tabindex="-1"></a>    constNum <span class="ot">&lt;-</span> <span class="fu">sum</span>(constCol, <span class="at">na.rm =</span> T)</span>
<span id="cb62-54"><a href="data-processing.html#cb62-54" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (constNum <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb62-55"><a href="data-processing.html#cb62-55" aria-hidden="true" tabindex="-1"></a>      <span class="fu">message</span>(<span class="fu">paste</span>(<span class="st">&quot;After quantile normalization&quot;</span>, constNum, </span>
<span id="cb62-56"><a href="data-processing.html#cb62-56" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;features with a constant value were found and deleted.&quot;</span>))</span>
<span id="cb62-57"><a href="data-processing.html#cb62-57" aria-hidden="true" tabindex="-1"></a>      data <span class="ot">&lt;-</span> data[, <span class="sc">!</span>constCol, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb62-58"><a href="data-processing.html#cb62-58" aria-hidden="true" tabindex="-1"></a>      colNames <span class="ot">&lt;-</span> <span class="fu">colnames</span>(data)</span>
<span id="cb62-59"><a href="data-processing.html#cb62-59" aria-hidden="true" tabindex="-1"></a>      rowNames <span class="ot">&lt;-</span> <span class="fu">rownames</span>(data)</span>
<span id="cb62-60"><a href="data-processing.html#cb62-60" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb62-61"><a href="data-processing.html#cb62-61" aria-hidden="true" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Quantile Normalization&quot;</span></span>
<span id="cb62-62"><a href="data-processing.html#cb62-62" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;GroupPQN&quot;</span>) {</span>
<span id="cb62-63"><a href="data-processing.html#cb62-63" aria-hidden="true" tabindex="-1"></a>    grp_inx <span class="ot">&lt;-</span> metadata_tab<span class="sc">$</span>group <span class="sc">==</span> ref</span>
<span id="cb62-64"><a href="data-processing.html#cb62-64" aria-hidden="true" tabindex="-1"></a>    ref.smpl <span class="ot">&lt;-</span> <span class="fu">apply</span>(data[grp_inx, , <span class="at">drop =</span> <span class="cn">FALSE</span>], <span class="dv">2</span>, mean)</span>
<span id="cb62-65"><a href="data-processing.html#cb62-65" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(data, <span class="dv">1</span>, ProbNorm, ref.smpl))</span>
<span id="cb62-66"><a href="data-processing.html#cb62-66" aria-hidden="true" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Probabilistic Quotient Normalization by a reference group&quot;</span></span>
<span id="cb62-67"><a href="data-processing.html#cb62-67" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;SamplePQN&quot;</span>) {</span>
<span id="cb62-68"><a href="data-processing.html#cb62-68" aria-hidden="true" tabindex="-1"></a>    ref.smpl <span class="ot">&lt;-</span> data[ref, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb62-69"><a href="data-processing.html#cb62-69" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(data, <span class="dv">1</span>, ProbNorm, ref.smpl))</span>
<span id="cb62-70"><a href="data-processing.html#cb62-70" aria-hidden="true" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Probabilistic Quotient Normalization by a reference sample&quot;</span></span>
<span id="cb62-71"><a href="data-processing.html#cb62-71" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;CompNorm&quot;</span>) {</span>
<span id="cb62-72"><a href="data-processing.html#cb62-72" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(<span class="fu">t</span>(data), <span class="dv">1</span>, CompNorm, ref))</span>
<span id="cb62-73"><a href="data-processing.html#cb62-73" aria-hidden="true" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Normalization by a reference feature&quot;</span>;</span>
<span id="cb62-74"><a href="data-processing.html#cb62-74" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;SumNorm&quot;</span>) {</span>
<span id="cb62-75"><a href="data-processing.html#cb62-75" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(data, <span class="dv">1</span>, SumNorm))</span>
<span id="cb62-76"><a href="data-processing.html#cb62-76" aria-hidden="true" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Normalization to constant sum&quot;</span></span>
<span id="cb62-77"><a href="data-processing.html#cb62-77" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;MedianNorm&quot;</span>) {</span>
<span id="cb62-78"><a href="data-processing.html#cb62-78" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(data, <span class="dv">1</span>, MedianNorm))</span>
<span id="cb62-79"><a href="data-processing.html#cb62-79" aria-hidden="true" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Normalization to sample median&quot;</span></span>
<span id="cb62-80"><a href="data-processing.html#cb62-80" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span>(rowNorm <span class="sc">==</span> <span class="st">&quot;SpecNorm&quot;</span>) {</span>
<span id="cb62-81"><a href="data-processing.html#cb62-81" aria-hidden="true" tabindex="-1"></a>    norm.vec <span class="ot">&lt;-</span> <span class="fu">rep</span>(SpeWeight, <span class="fu">nrow</span>(data)) <span class="co"># default all same weight vec to prevent error</span></span>
<span id="cb62-82"><a href="data-processing.html#cb62-82" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> data <span class="sc">/</span> norm.vec</span>
<span id="cb62-83"><a href="data-processing.html#cb62-83" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">&quot;No sample specific information were given, all set to 1.0&quot;</span>)</span>
<span id="cb62-84"><a href="data-processing.html#cb62-84" aria-hidden="true" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Normalization by sample-specific factor&quot;</span></span>
<span id="cb62-85"><a href="data-processing.html#cb62-85" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb62-86"><a href="data-processing.html#cb62-86" aria-hidden="true" tabindex="-1"></a>    <span class="co"># nothing to do</span></span>
<span id="cb62-87"><a href="data-processing.html#cb62-87" aria-hidden="true" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;N/A&quot;</span></span>
<span id="cb62-88"><a href="data-processing.html#cb62-88" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb62-89"><a href="data-processing.html#cb62-89" aria-hidden="true" tabindex="-1"></a>  <span class="do">################################################ </span></span>
<span id="cb62-90"><a href="data-processing.html#cb62-90" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-91"><a href="data-processing.html#cb62-91" aria-hidden="true" tabindex="-1"></a>  <span class="co"># use apply will lose dimension info (i.e. row names and colnames)</span></span>
<span id="cb62-92"><a href="data-processing.html#cb62-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(data) <span class="ot">&lt;-</span> rowNames</span>
<span id="cb62-93"><a href="data-processing.html#cb62-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(data) <span class="ot">&lt;-</span> colNames</span>
<span id="cb62-94"><a href="data-processing.html#cb62-94" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-95"><a href="data-processing.html#cb62-95" aria-hidden="true" tabindex="-1"></a>  <span class="co"># if the reference by feature, the feature column should be removed, since it is all 1</span></span>
<span id="cb62-96"><a href="data-processing.html#cb62-96" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(rowNorm <span class="sc">==</span> <span class="st">&quot;CompNorm&quot;</span> <span class="sc">&amp;&amp;</span> <span class="sc">!</span><span class="fu">is.null</span>(ref)){</span>
<span id="cb62-97"><a href="data-processing.html#cb62-97" aria-hidden="true" tabindex="-1"></a>    inx <span class="ot">&lt;-</span> <span class="fu">match</span>(ref, <span class="fu">colnames</span>(data))</span>
<span id="cb62-98"><a href="data-processing.html#cb62-98" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> data[, <span class="sc">-</span>inx, drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb62-99"><a href="data-processing.html#cb62-99" aria-hidden="true" tabindex="-1"></a>    colNames <span class="ot">&lt;-</span> colNames[<span class="sc">-</span>inx]</span>
<span id="cb62-100"><a href="data-processing.html#cb62-100" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb62-101"><a href="data-processing.html#cb62-101" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-102"><a href="data-processing.html#cb62-102" aria-hidden="true" tabindex="-1"></a>  <span class="do">#############################################</span></span>
<span id="cb62-103"><a href="data-processing.html#cb62-103" aria-hidden="true" tabindex="-1"></a>  <span class="co">#  Data transformation</span></span>
<span id="cb62-104"><a href="data-processing.html#cb62-104" aria-hidden="true" tabindex="-1"></a>  <span class="co"># generalize log, tolerant to 0 and negative values</span></span>
<span id="cb62-105"><a href="data-processing.html#cb62-105" aria-hidden="true" tabindex="-1"></a>  LogNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x, min.val) {</span>
<span id="cb62-106"><a href="data-processing.html#cb62-106" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">log10</span>((x <span class="sc">+</span> <span class="fu">sqrt</span>(x<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> min.val<span class="sc">^</span><span class="dv">2</span>))<span class="sc">/</span><span class="dv">2</span>))</span>
<span id="cb62-107"><a href="data-processing.html#cb62-107" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb62-108"><a href="data-processing.html#cb62-108" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-109"><a href="data-processing.html#cb62-109" aria-hidden="true" tabindex="-1"></a>  <span class="co"># square root, tolerant to negative values</span></span>
<span id="cb62-110"><a href="data-processing.html#cb62-110" aria-hidden="true" tabindex="-1"></a>  SquareRootNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x, min.val) {</span>
<span id="cb62-111"><a href="data-processing.html#cb62-111" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(((x <span class="sc">+</span> <span class="fu">sqrt</span>(x<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> min.val<span class="sc">^</span><span class="dv">2</span>))<span class="sc">/</span><span class="dv">2</span>)<span class="sc">^</span>(<span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>))</span>
<span id="cb62-112"><a href="data-processing.html#cb62-112" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb62-113"><a href="data-processing.html#cb62-113" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-114"><a href="data-processing.html#cb62-114" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (transNorm <span class="sc">==</span> <span class="st">&quot;LogNorm&quot;</span>) {</span>
<span id="cb62-115"><a href="data-processing.html#cb62-115" aria-hidden="true" tabindex="-1"></a>    min.val <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">abs</span>(data[data <span class="sc">!=</span> <span class="dv">0</span>]))<span class="sc">/</span><span class="dv">10</span></span>
<span id="cb62-116"><a href="data-processing.html#cb62-116" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, LogNorm, min.val)</span>
<span id="cb62-117"><a href="data-processing.html#cb62-117" aria-hidden="true" tabindex="-1"></a>    transnm <span class="ot">&lt;-</span> <span class="st">&quot;Log10 Normalization&quot;</span></span>
<span id="cb62-118"><a href="data-processing.html#cb62-118" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (transNorm <span class="sc">==</span> <span class="st">&quot;SrNorm&quot;</span>) {</span>
<span id="cb62-119"><a href="data-processing.html#cb62-119" aria-hidden="true" tabindex="-1"></a>    min.val <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">abs</span>(data[data <span class="sc">!=</span> <span class="dv">0</span>]))<span class="sc">/</span><span class="dv">10</span></span>
<span id="cb62-120"><a href="data-processing.html#cb62-120" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, SquareRootNorm, min.val)</span>
<span id="cb62-121"><a href="data-processing.html#cb62-121" aria-hidden="true" tabindex="-1"></a>    transnm <span class="ot">&lt;-</span> <span class="st">&quot;Square Root Transformation&quot;</span></span>
<span id="cb62-122"><a href="data-processing.html#cb62-122" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (transNorm <span class="sc">==</span> <span class="st">&quot;CrNorm&quot;</span>) {</span>
<span id="cb62-123"><a href="data-processing.html#cb62-123" aria-hidden="true" tabindex="-1"></a>    norm.data <span class="ot">&lt;-</span> <span class="fu">abs</span>(data)<span class="sc">^</span>(<span class="dv">1</span><span class="sc">/</span><span class="dv">3</span>)</span>
<span id="cb62-124"><a href="data-processing.html#cb62-124" aria-hidden="true" tabindex="-1"></a>    norm.data[data <span class="sc">&lt;</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="sc">-</span>norm.data[data <span class="sc">&lt;</span> <span class="dv">0</span>]</span>
<span id="cb62-125"><a href="data-processing.html#cb62-125" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> norm.data</span>
<span id="cb62-126"><a href="data-processing.html#cb62-126" aria-hidden="true" tabindex="-1"></a>    transnm <span class="ot">&lt;-</span> <span class="st">&quot;Cubic Root Transformation&quot;</span></span>
<span id="cb62-127"><a href="data-processing.html#cb62-127" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb62-128"><a href="data-processing.html#cb62-128" aria-hidden="true" tabindex="-1"></a>    transnm <span class="ot">&lt;-</span> <span class="st">&quot;N/A&quot;</span></span>
<span id="cb62-129"><a href="data-processing.html#cb62-129" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb62-130"><a href="data-processing.html#cb62-130" aria-hidden="true" tabindex="-1"></a>  <span class="do">#############################################</span></span>
<span id="cb62-131"><a href="data-processing.html#cb62-131" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-132"><a href="data-processing.html#cb62-132" aria-hidden="true" tabindex="-1"></a>  <span class="do">#############################################</span></span>
<span id="cb62-133"><a href="data-processing.html#cb62-133" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Data scaling</span></span>
<span id="cb62-134"><a href="data-processing.html#cb62-134" aria-hidden="true" tabindex="-1"></a>  <span class="co"># normalize to zero mean and unit variance</span></span>
<span id="cb62-135"><a href="data-processing.html#cb62-135" aria-hidden="true" tabindex="-1"></a>  AutoNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb62-136"><a href="data-processing.html#cb62-136" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>((x <span class="sc">-</span> <span class="fu">mean</span>(x))<span class="sc">/</span><span class="fu">sd</span>(x, <span class="at">na.rm =</span> T))</span>
<span id="cb62-137"><a href="data-processing.html#cb62-137" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb62-138"><a href="data-processing.html#cb62-138" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-139"><a href="data-processing.html#cb62-139" aria-hidden="true" tabindex="-1"></a>  <span class="co"># normalize to zero mean but variance/SE</span></span>
<span id="cb62-140"><a href="data-processing.html#cb62-140" aria-hidden="true" tabindex="-1"></a>  ParetoNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb62-141"><a href="data-processing.html#cb62-141" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>((x <span class="sc">-</span> <span class="fu">mean</span>(x))<span class="sc">/</span><span class="fu">sqrt</span>(<span class="fu">sd</span>(x, <span class="at">na.rm =</span> T)))</span>
<span id="cb62-142"><a href="data-processing.html#cb62-142" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb62-143"><a href="data-processing.html#cb62-143" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-144"><a href="data-processing.html#cb62-144" aria-hidden="true" tabindex="-1"></a>  <span class="co"># normalize to zero mean but variance/SE</span></span>
<span id="cb62-145"><a href="data-processing.html#cb62-145" aria-hidden="true" tabindex="-1"></a>  MeanCenter <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb62-146"><a href="data-processing.html#cb62-146" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(x <span class="sc">-</span> <span class="fu">mean</span>(x))</span>
<span id="cb62-147"><a href="data-processing.html#cb62-147" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb62-148"><a href="data-processing.html#cb62-148" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-149"><a href="data-processing.html#cb62-149" aria-hidden="true" tabindex="-1"></a>  <span class="co"># normalize to zero mean but variance/SE</span></span>
<span id="cb62-150"><a href="data-processing.html#cb62-150" aria-hidden="true" tabindex="-1"></a>  RangeNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb62-151"><a href="data-processing.html#cb62-151" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">max</span>(x) <span class="sc">==</span> <span class="fu">min</span>(x)) {</span>
<span id="cb62-152"><a href="data-processing.html#cb62-152" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(x)</span>
<span id="cb62-153"><a href="data-processing.html#cb62-153" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb62-154"><a href="data-processing.html#cb62-154" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>((x <span class="sc">-</span> <span class="fu">mean</span>(x))<span class="sc">/</span>(<span class="fu">max</span>(x) <span class="sc">-</span> <span class="fu">min</span>(x)))</span>
<span id="cb62-155"><a href="data-processing.html#cb62-155" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb62-156"><a href="data-processing.html#cb62-156" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb62-157"><a href="data-processing.html#cb62-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-158"><a href="data-processing.html#cb62-158" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (scaleNorm <span class="sc">==</span> <span class="st">&quot;MeanCenter&quot;</span>) {</span>
<span id="cb62-159"><a href="data-processing.html#cb62-159" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, MeanCenter)</span>
<span id="cb62-160"><a href="data-processing.html#cb62-160" aria-hidden="true" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;Mean Centering&quot;</span></span>
<span id="cb62-161"><a href="data-processing.html#cb62-161" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (scaleNorm <span class="sc">==</span> <span class="st">&quot;AutoNorm&quot;</span>) {</span>
<span id="cb62-162"><a href="data-processing.html#cb62-162" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, AutoNorm)</span>
<span id="cb62-163"><a href="data-processing.html#cb62-163" aria-hidden="true" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;Autoscaling&quot;</span></span>
<span id="cb62-164"><a href="data-processing.html#cb62-164" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (scaleNorm <span class="sc">==</span> <span class="st">&quot;ParetoNorm&quot;</span>) {</span>
<span id="cb62-165"><a href="data-processing.html#cb62-165" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, ParetoNorm)</span>
<span id="cb62-166"><a href="data-processing.html#cb62-166" aria-hidden="true" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;Pareto Scaling&quot;</span></span>
<span id="cb62-167"><a href="data-processing.html#cb62-167" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (scaleNorm <span class="sc">==</span> <span class="st">&quot;RangeNorm&quot;</span>) {</span>
<span id="cb62-168"><a href="data-processing.html#cb62-168" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, RangeNorm)</span>
<span id="cb62-169"><a href="data-processing.html#cb62-169" aria-hidden="true" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;Range Scaling&quot;</span></span>
<span id="cb62-170"><a href="data-processing.html#cb62-170" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb62-171"><a href="data-processing.html#cb62-171" aria-hidden="true" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;N/A&quot;</span></span>
<span id="cb62-172"><a href="data-processing.html#cb62-172" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb62-173"><a href="data-processing.html#cb62-173" aria-hidden="true" tabindex="-1"></a>  <span class="do">############################################# </span></span>
<span id="cb62-174"><a href="data-processing.html#cb62-174" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-175"><a href="data-processing.html#cb62-175" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">&quot;Row norm: &quot;</span>, rownm, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>,</span>
<span id="cb62-176"><a href="data-processing.html#cb62-176" aria-hidden="true" tabindex="-1"></a>          <span class="st">&quot;Data Transformation norm: &quot;</span>, transnm, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>,</span>
<span id="cb62-177"><a href="data-processing.html#cb62-177" aria-hidden="true" tabindex="-1"></a>          <span class="st">&quot;Data Scaling norm: &quot;</span>, scalenm, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)  </span>
<span id="cb62-178"><a href="data-processing.html#cb62-178" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-179"><a href="data-processing.html#cb62-179" aria-hidden="true" tabindex="-1"></a>  <span class="co"># note after using &quot;apply&quot; function, all the attribute lost, need to add back</span></span>
<span id="cb62-180"><a href="data-processing.html#cb62-180" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(data) <span class="ot">&lt;-</span> rowNames</span>
<span id="cb62-181"><a href="data-processing.html#cb62-181" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(data) <span class="ot">&lt;-</span> colNames</span>
<span id="cb62-182"><a href="data-processing.html#cb62-182" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-183"><a href="data-processing.html#cb62-183" aria-hidden="true" tabindex="-1"></a>  target <span class="ot">&lt;-</span> metadata_tab <span class="sc">%&gt;%</span> </span>
<span id="cb62-184"><a href="data-processing.html#cb62-184" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb62-185"><a href="data-processing.html#cb62-185" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb62-186"><a href="data-processing.html#cb62-186" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(SampleID<span class="sc">%in%</span><span class="fu">rownames</span>(data))</span>
<span id="cb62-187"><a href="data-processing.html#cb62-187" aria-hidden="true" tabindex="-1"></a>  se <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb62-188"><a href="data-processing.html#cb62-188" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">features =</span> data)</span>
<span id="cb62-189"><a href="data-processing.html#cb62-189" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-190"><a href="data-processing.html#cb62-190" aria-hidden="true" tabindex="-1"></a>  <span class="co"># need to do some sanity check, for log there may be Inf values introduced</span></span>
<span id="cb62-191"><a href="data-processing.html#cb62-191" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">CheckData</span>(se)</span>
<span id="cb62-192"><a href="data-processing.html#cb62-192" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-193"><a href="data-processing.html#cb62-193" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb62-194"><a href="data-processing.html#cb62-194" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb62-195"><a href="data-processing.html#cb62-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-196"><a href="data-processing.html#cb62-196" aria-hidden="true" tabindex="-1"></a>se_normalize <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(</span>
<span id="cb62-197"><a href="data-processing.html#cb62-197" aria-hidden="true" tabindex="-1"></a>                      <span class="at">object =</span> se_impute,</span>
<span id="cb62-198"><a href="data-processing.html#cb62-198" aria-hidden="true" tabindex="-1"></a>                      <span class="at">rowNorm =</span> <span class="st">&quot;None&quot;</span>,</span>
<span id="cb62-199"><a href="data-processing.html#cb62-199" aria-hidden="true" tabindex="-1"></a>                      <span class="at">transNorm =</span> <span class="st">&quot;LogNorm&quot;</span>,</span>
<span id="cb62-200"><a href="data-processing.html#cb62-200" aria-hidden="true" tabindex="-1"></a>                      <span class="at">scaleNorm =</span> <span class="st">&quot;ParetoNorm&quot;</span>)</span></code></pre></div>
<pre><code>## Row norm: N/A
## Data Transformation norm: Log10 Normalization
## Data Scaling norm: Pareto Scaling</code></pre>
<pre><code>## All data values are numeric.</code></pre>
<pre><code>## A total of 0 (0%) missing values were detected.</code></pre>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="data-processing.html#cb66-1" aria-hidden="true" tabindex="-1"></a>se_normalize</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 48 45 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(48): M_52603 M_19130 ... M_63681 M_63361
## rowData names(0):
## colnames(45): P101001 P101004 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="normalization-by-poma-r-package" class="section level3 hasAnchor" number="2.6.2">
<h3><span class="header-section-number">2.6.2</span> Normalization by <strong>POMA</strong> R package<a href="data-processing.html#normalization-by-poma-r-package" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="data-processing.html#cb68-1" aria-hidden="true" tabindex="-1"></a>none <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_impute, <span class="at">method =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb68-2"><a href="data-processing.html#cb68-2" aria-hidden="true" tabindex="-1"></a>auto_scaling <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_impute, <span class="at">method =</span> <span class="st">&quot;auto_scaling&quot;</span>)</span>
<span id="cb68-3"><a href="data-processing.html#cb68-3" aria-hidden="true" tabindex="-1"></a>evel_scaling <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_impute, <span class="at">method =</span> <span class="st">&quot;level_scaling&quot;</span>)</span>
<span id="cb68-4"><a href="data-processing.html#cb68-4" aria-hidden="true" tabindex="-1"></a>log_scaling <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_impute, <span class="at">method =</span> <span class="st">&quot;log_scaling&quot;</span>)</span>
<span id="cb68-5"><a href="data-processing.html#cb68-5" aria-hidden="true" tabindex="-1"></a>log_transformation <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_impute, <span class="at">method =</span> <span class="st">&quot;log_transformation&quot;</span>)</span>
<span id="cb68-6"><a href="data-processing.html#cb68-6" aria-hidden="true" tabindex="-1"></a>vast_scaling <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_impute, <span class="at">method =</span> <span class="st">&quot;vast_scaling&quot;</span>)</span>
<span id="cb68-7"><a href="data-processing.html#cb68-7" aria-hidden="true" tabindex="-1"></a>se_normalize_v2 <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_impute, <span class="at">method =</span> <span class="st">&quot;log_pareto&quot;</span>)</span>
<span id="cb68-8"><a href="data-processing.html#cb68-8" aria-hidden="true" tabindex="-1"></a>se_normalize_v2</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 48 45 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(48): M_52603 M_19130 ... M_63681 M_63361
## rowData names(0):
## colnames(45): P101001 P101004 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="comparison-of-unnormalized-and-normalized-dataset" class="section level3 hasAnchor" number="2.6.3">
<h3><span class="header-section-number">2.6.3</span> Comparison of unnormalized and normalized dataset<a href="data-processing.html#comparison-of-unnormalized-and-normalized-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>boxplot</li>
</ul>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="data-processing.html#cb70-1" aria-hidden="true" tabindex="-1"></a>pl_unnor <span class="ot">&lt;-</span> <span class="fu">PomaBoxplots</span>(se_impute, <span class="at">group =</span> <span class="st">&quot;samples&quot;</span>, <span class="at">jitter =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb70-2"><a href="data-processing.html#cb70-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Not Normalized&quot;</span>) <span class="sc">+</span></span>
<span id="cb70-3"><a href="data-processing.html#cb70-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="co"># data before normalization</span></span>
<span id="cb70-4"><a href="data-processing.html#cb70-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-5"><a href="data-processing.html#cb70-5" aria-hidden="true" tabindex="-1"></a>pl_nor <span class="ot">&lt;-</span> <span class="fu">PomaBoxplots</span>(se_normalize, <span class="at">group =</span> <span class="st">&quot;samples&quot;</span>, <span class="at">jitter =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb70-6"><a href="data-processing.html#cb70-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Normalized&quot;</span>) <span class="co"># data after normalization</span></span>
<span id="cb70-7"><a href="data-processing.html#cb70-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-8"><a href="data-processing.html#cb70-8" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(pl_unnor, pl_nor, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">align =</span> <span class="st">&quot;v&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-12-1.png" width="1344" /></p>
<ul>
<li>density</li>
</ul>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="data-processing.html#cb71-1" aria-hidden="true" tabindex="-1"></a>pl_unnor <span class="ot">&lt;-</span> <span class="fu">PomaDensity</span>(se_impute, <span class="at">group =</span> <span class="st">&quot;features&quot;</span>) <span class="sc">+</span></span>
<span id="cb71-2"><a href="data-processing.html#cb71-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Not Normalized&quot;</span>) <span class="sc">+</span></span>
<span id="cb71-3"><a href="data-processing.html#cb71-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="co"># data before normalization</span></span>
<span id="cb71-4"><a href="data-processing.html#cb71-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-5"><a href="data-processing.html#cb71-5" aria-hidden="true" tabindex="-1"></a>pl_nor <span class="ot">&lt;-</span> <span class="fu">PomaDensity</span>(se_normalize, <span class="at">group =</span> <span class="st">&quot;features&quot;</span>) <span class="sc">+</span></span>
<span id="cb71-6"><a href="data-processing.html#cb71-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Normalized&quot;</span>) <span class="co"># data after normalization</span></span>
<span id="cb71-7"><a href="data-processing.html#cb71-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-8"><a href="data-processing.html#cb71-8" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(pl_unnor, pl_nor, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">align =</span> <span class="st">&quot;v&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-13-1.png" width="960" /></p>
</div>
</div>
<div id="removing-outliers" class="section level2 hasAnchor" number="2.7">
<h2><span class="header-section-number">2.7</span> Removing outliers<a href="data-processing.html#removing-outliers" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="data-processing.html#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">PomaOutliers</span>(se_normalize, <span class="at">do =</span> <span class="st">&quot;analyze&quot;</span>)<span class="sc">$</span>polygon_plot <span class="co"># to explore</span></span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-14-1.png" width="960" /></p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="data-processing.html#cb73-1" aria-hidden="true" tabindex="-1"></a>se_processed <span class="ot">&lt;-</span> <span class="fu">PomaOutliers</span>(se_normalize, <span class="at">do =</span> <span class="st">&quot;clean&quot;</span>) <span class="co"># to remove outliers</span></span>
<span id="cb73-2"><a href="data-processing.html#cb73-2" aria-hidden="true" tabindex="-1"></a>se_processed</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 48 43 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(48): M_52603 M_19130 ... M_63681 M_63361
## rowData names(0):
## colnames(43): P101001 P101004 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="saving-datasets-into-rds-files" class="section level2 hasAnchor" number="2.8">
<h2><span class="header-section-number">2.8</span> Saving datasets into RDS files<a href="data-processing.html#saving-datasets-into-rds-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="data-processing.html#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="st">&quot;./dataset/POMA/&quot;</span>)) {</span>
<span id="cb75-2"><a href="data-processing.html#cb75-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="st">&quot;./dataset/POMA/&quot;</span>)</span>
<span id="cb75-3"><a href="data-processing.html#cb75-3" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb75-4"><a href="data-processing.html#cb75-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-5"><a href="data-processing.html#cb75-5" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(se_filter, <span class="st">&quot;./dataset/POMA/se_filter.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span>
<span id="cb75-6"><a href="data-processing.html#cb75-6" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(se_impute, <span class="st">&quot;./dataset/POMA/se_impute.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span>
<span id="cb75-7"><a href="data-processing.html#cb75-7" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(se_normalize, <span class="st">&quot;./dataset/POMA/se_normalize.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span>
<span id="cb75-8"><a href="data-processing.html#cb75-8" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(se_processed, <span class="st">&quot;./dataset/POMA/se_processed.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="systematic-information" class="section level2 hasAnchor" number="2.9">
<h2><span class="header-section-number">2.9</span> Systematic Information<a href="data-processing.html#systematic-information" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="data-processing.html#cb76-1" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ─────────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 4.1.2 (2021-11-01)
##  os       macOS Monterey 12.2.1
##  system   x86_64, darwin17.0
##  ui       RStudio
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       Asia/Shanghai
##  date     2022-08-09
##  rstudio  2022.07.1+554 Spotted Wakerobin (desktop)
##  pandoc   2.18 @ /Applications/RStudio.app/Contents/MacOS/quarto/bin/tools/ (via rmarkdown)
## 
## ─ Packages ─────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package              * version    date (UTC) lib source
##  assertthat             0.2.1      2019-03-21 [1] CRAN (R 4.1.0)
##  backports              1.4.1      2021-12-13 [1] CRAN (R 4.1.0)
##  Biobase              * 2.54.0     2021-10-26 [1] Bioconductor
##  BiocGenerics         * 0.40.0     2021-10-26 [1] Bioconductor
##  BiocParallel           1.28.3     2021-12-09 [1] Bioconductor
##  bitops                 1.0-7      2021-04-24 [1] CRAN (R 4.1.0)
##  bookdown               0.27       2022-06-14 [1] CRAN (R 4.1.2)
##  brio                   1.1.3      2021-11-30 [1] CRAN (R 4.1.0)
##  broom                  0.7.12     2022-01-28 [1] CRAN (R 4.1.2)
##  bslib                  0.3.1      2021-10-06 [1] CRAN (R 4.1.0)
##  cachem                 1.0.6      2021-08-19 [1] CRAN (R 4.1.0)
##  callr                  3.7.0      2021-04-20 [1] CRAN (R 4.1.0)
##  caret                  6.0-92     2022-04-19 [1] CRAN (R 4.1.2)
##  cellranger             1.1.0      2016-07-27 [1] CRAN (R 4.1.0)
##  circlize               0.4.14     2022-02-11 [1] CRAN (R 4.1.2)
##  class                  7.3-20     2022-01-13 [1] CRAN (R 4.1.2)
##  cli                    3.3.0      2022-04-25 [1] CRAN (R 4.1.2)
##  clue                   0.3-60     2021-10-11 [1] CRAN (R 4.1.0)
##  cluster                2.1.2      2021-04-17 [1] CRAN (R 4.1.2)
##  codetools              0.2-18     2020-11-04 [1] CRAN (R 4.1.2)
##  colorspace             2.0-3      2022-02-21 [1] CRAN (R 4.1.2)
##  ComplexHeatmap         2.10.0     2021-10-26 [1] Bioconductor
##  corpcor                1.6.10     2021-09-16 [1] CRAN (R 4.1.0)
##  cowplot                1.1.1      2020-12-30 [1] CRAN (R 4.1.0)
##  crayon                 1.5.0      2022-02-14 [1] CRAN (R 4.1.2)
##  data.table             1.14.2     2021-09-27 [1] CRAN (R 4.1.0)
##  DBI                    1.1.2      2021-12-20 [1] CRAN (R 4.1.0)
##  DelayedArray           0.20.0     2021-10-26 [1] Bioconductor
##  desc                   1.4.1      2022-03-06 [1] CRAN (R 4.1.2)
##  devtools               2.4.3      2021-11-30 [1] CRAN (R 4.1.0)
##  digest                 0.6.29     2021-12-01 [1] CRAN (R 4.1.0)
##  doParallel             1.0.17     2022-02-07 [1] CRAN (R 4.1.2)
##  dplyr                * 1.0.8      2022-02-08 [1] CRAN (R 4.1.2)
##  e1071                  1.7-9      2021-09-16 [1] CRAN (R 4.1.0)
##  ellipse                0.4.2      2020-05-27 [1] CRAN (R 4.1.0)
##  ellipsis               0.3.2      2021-04-29 [1] CRAN (R 4.1.0)
##  evaluate               0.15       2022-02-18 [1] CRAN (R 4.1.2)
##  fansi                  1.0.2      2022-01-14 [1] CRAN (R 4.1.2)
##  farver                 2.1.0      2021-02-28 [1] CRAN (R 4.1.0)
##  fastmap                1.1.0      2021-01-25 [1] CRAN (R 4.1.0)
##  foreach                1.5.2      2022-02-02 [1] CRAN (R 4.1.2)
##  fs                     1.5.2      2021-12-08 [1] CRAN (R 4.1.0)
##  future                 1.24.0     2022-02-19 [1] CRAN (R 4.1.2)
##  future.apply           1.8.1      2021-08-10 [1] CRAN (R 4.1.0)
##  generics               0.1.2      2022-01-31 [1] CRAN (R 4.1.2)
##  GenomeInfoDb         * 1.30.1     2022-01-30 [1] Bioconductor
##  GenomeInfoDbData       1.2.7      2022-03-09 [1] Bioconductor
##  GenomicRanges        * 1.46.1     2021-11-18 [1] Bioconductor
##  GetoptLong             1.0.5      2020-12-15 [1] CRAN (R 4.1.0)
##  ggforce                0.3.3      2021-03-05 [1] CRAN (R 4.1.0)
##  ggplot2              * 3.3.5      2021-06-25 [1] CRAN (R 4.1.0)
##  ggraph               * 2.0.5      2021-02-23 [1] CRAN (R 4.1.0)
##  ggrepel                0.9.1      2021-01-15 [1] CRAN (R 4.1.0)
##  glasso                 1.11       2019-10-01 [1] CRAN (R 4.1.0)
##  glmnet                 4.1-3      2021-11-02 [1] CRAN (R 4.1.0)
##  GlobalOptions          0.1.2      2020-06-10 [1] CRAN (R 4.1.0)
##  globals                0.14.0     2020-11-22 [1] CRAN (R 4.1.0)
##  glue                   1.6.2      2022-02-24 [1] CRAN (R 4.1.2)
##  gmp                    0.6-5      2022-03-17 [1] CRAN (R 4.1.2)
##  gower                  1.0.0      2022-02-03 [1] CRAN (R 4.1.2)
##  graphlayouts           0.8.0      2022-01-03 [1] CRAN (R 4.1.2)
##  gridExtra              2.3        2017-09-09 [1] CRAN (R 4.1.0)
##  gtable                 0.3.0      2019-03-25 [1] CRAN (R 4.1.0)
##  hardhat                0.2.0      2022-01-24 [1] CRAN (R 4.1.2)
##  highr                  0.9        2021-04-16 [1] CRAN (R 4.1.0)
##  htmltools              0.5.2      2021-08-25 [1] CRAN (R 4.1.0)
##  htmlwidgets            1.5.4      2021-09-08 [1] CRAN (R 4.1.0)
##  httr                   1.4.2      2020-07-20 [1] CRAN (R 4.1.0)
##  igraph                 1.2.11     2022-01-04 [1] CRAN (R 4.1.2)
##  impute                 1.68.0     2021-10-26 [1] Bioconductor
##  ipred                  0.9-12     2021-09-15 [1] CRAN (R 4.1.0)
##  IRanges              * 2.28.0     2021-10-26 [1] Bioconductor
##  iterators              1.0.14     2022-02-05 [1] CRAN (R 4.1.2)
##  jquerylib              0.1.4      2021-04-26 [1] CRAN (R 4.1.0)
##  jsonlite               1.8.0      2022-02-22 [1] CRAN (R 4.1.2)
##  knitr                  1.39       2022-04-26 [1] CRAN (R 4.1.2)
##  labeling               0.4.2      2020-10-20 [1] CRAN (R 4.1.0)
##  lattice                0.20-45    2021-09-22 [1] CRAN (R 4.1.2)
##  lava                   1.6.10     2021-09-02 [1] CRAN (R 4.1.0)
##  lazyeval               0.2.2      2019-03-15 [1] CRAN (R 4.1.0)
##  lifecycle              1.0.1      2021-09-24 [1] CRAN (R 4.1.0)
##  limma                  3.50.1     2022-02-17 [1] Bioconductor
##  listenv                0.8.0      2019-12-05 [1] CRAN (R 4.1.0)
##  lubridate              1.8.0      2021-10-07 [1] CRAN (R 4.1.0)
##  magrittr               2.0.2      2022-01-26 [1] CRAN (R 4.1.2)
##  MASS                   7.3-55     2022-01-13 [1] CRAN (R 4.1.2)
##  Matrix                 1.4-0      2021-12-08 [1] CRAN (R 4.1.0)
##  MatrixGenerics       * 1.6.0      2021-10-26 [1] Bioconductor
##  matrixStats          * 0.61.0     2021-09-17 [1] CRAN (R 4.1.0)
##  memoise                2.0.1      2021-11-26 [1] CRAN (R 4.1.0)
##  mgcv                   1.8-39     2022-02-24 [1] CRAN (R 4.1.2)
##  mixOmics               6.18.1     2021-11-18 [1] Bioconductor (R 4.1.2)
##  ModelMetrics           1.2.2.2    2020-03-17 [1] CRAN (R 4.1.0)
##  munsell                0.5.0      2018-06-12 [1] CRAN (R 4.1.0)
##  nlme                   3.1-155    2022-01-13 [1] CRAN (R 4.1.2)
##  nnet                   7.3-17     2022-01-13 [1] CRAN (R 4.1.2)
##  parallelly             1.31.0     2022-04-07 [1] CRAN (R 4.1.2)
##  permute                0.9-7      2022-01-27 [1] CRAN (R 4.1.2)
##  pillar                 1.7.0      2022-02-01 [1] CRAN (R 4.1.2)
##  pkgbuild               1.3.1      2021-12-20 [1] CRAN (R 4.1.0)
##  pkgconfig              2.0.3      2019-09-22 [1] CRAN (R 4.1.0)
##  pkgload                1.2.4      2021-11-30 [1] CRAN (R 4.1.0)
##  plotly               * 4.10.0     2021-10-09 [1] CRAN (R 4.1.0)
##  plyr                   1.8.6      2020-03-03 [1] CRAN (R 4.1.0)
##  png                    0.1-7      2013-12-03 [1] CRAN (R 4.1.0)
##  polyclip               1.10-0     2019-03-14 [1] CRAN (R 4.1.0)
##  POMA                 * 1.7.2      2022-07-26 [1] Github (pcastellanoescuder/POMA@bc8a972)
##  prettyunits            1.1.1      2020-01-24 [1] CRAN (R 4.1.0)
##  pROC                   1.18.0     2021-09-03 [1] CRAN (R 4.1.0)
##  processx               3.5.2      2021-04-30 [1] CRAN (R 4.1.0)
##  prodlim                2019.11.13 2019-11-17 [1] CRAN (R 4.1.0)
##  proxy                  0.4-26     2021-06-07 [1] CRAN (R 4.1.0)
##  ps                     1.6.0      2021-02-28 [1] CRAN (R 4.1.0)
##  purrr                  0.3.4      2020-04-17 [1] CRAN (R 4.1.0)
##  R6                     2.5.1      2021-08-19 [1] CRAN (R 4.1.0)
##  randomForest           4.7-1      2022-02-03 [1] CRAN (R 4.1.2)
##  RankProd               3.20.0     2021-10-26 [1] Bioconductor
##  rARPACK                0.11-0     2016-03-10 [1] CRAN (R 4.1.0)
##  RColorBrewer           1.1-2      2014-12-07 [1] CRAN (R 4.1.0)
##  Rcpp                   1.0.8.2    2022-03-11 [1] CRAN (R 4.1.2)
##  RCurl                  1.98-1.6   2022-02-08 [1] CRAN (R 4.1.2)
##  readxl               * 1.4.0      2022-03-28 [1] CRAN (R 4.1.2)
##  recipes                0.2.0      2022-02-18 [1] CRAN (R 4.1.2)
##  remotes                2.4.2      2021-11-30 [1] CRAN (R 4.1.0)
##  reshape2               1.4.4      2020-04-09 [1] CRAN (R 4.1.0)
##  rjson                  0.2.21     2022-01-09 [1] CRAN (R 4.1.2)
##  rlang                  1.0.2      2022-03-04 [1] CRAN (R 4.1.2)
##  rmarkdown              2.14       2022-04-25 [1] CRAN (R 4.1.2)
##  Rmpfr                  0.8-9      2022-06-03 [1] CRAN (R 4.1.2)
##  rpart                  4.1.16     2022-01-24 [1] CRAN (R 4.1.2)
##  rprojroot              2.0.2      2020-11-15 [1] CRAN (R 4.1.0)
##  RSpectra               0.16-0     2019-12-01 [1] CRAN (R 4.1.0)
##  rstudioapi             0.13       2020-11-12 [1] CRAN (R 4.1.0)
##  S4Vectors            * 0.32.3     2021-11-21 [1] Bioconductor
##  sass                   0.4.0      2021-05-12 [1] CRAN (R 4.1.0)
##  scales                 1.1.1      2020-05-11 [1] CRAN (R 4.1.0)
##  sessioninfo            1.2.2      2021-12-06 [1] CRAN (R 4.1.0)
##  shape                  1.4.6      2021-05-19 [1] CRAN (R 4.1.0)
##  stringi                1.7.6      2021-11-29 [1] CRAN (R 4.1.0)
##  stringr                1.4.0      2019-02-10 [1] CRAN (R 4.1.0)
##  SummarizedExperiment * 1.24.0     2021-10-26 [1] Bioconductor
##  survival               3.3-1      2022-03-03 [1] CRAN (R 4.1.2)
##  testthat               3.1.2      2022-01-20 [1] CRAN (R 4.1.2)
##  tibble               * 3.1.6      2021-11-07 [1] CRAN (R 4.1.0)
##  tidygraph              1.2.1      2022-04-05 [1] CRAN (R 4.1.2)
##  tidyr                  1.2.0      2022-02-01 [1] CRAN (R 4.1.2)
##  tidyselect             1.1.2      2022-02-21 [1] CRAN (R 4.1.2)
##  timeDate               3043.102   2018-02-21 [1] CRAN (R 4.1.0)
##  tweenr                 1.0.2      2021-03-23 [1] CRAN (R 4.1.0)
##  usethis                2.1.5      2021-12-09 [1] CRAN (R 4.1.0)
##  utf8                   1.2.2      2021-07-24 [1] CRAN (R 4.1.0)
##  vctrs                  0.3.8      2021-04-29 [1] CRAN (R 4.1.0)
##  vegan                  2.5-7      2020-11-28 [1] CRAN (R 4.1.0)
##  viridis                0.6.2      2021-10-13 [1] CRAN (R 4.1.0)
##  viridisLite            0.4.0      2021-04-13 [1] CRAN (R 4.1.0)
##  withr                  2.5.0      2022-03-03 [1] CRAN (R 4.1.2)
##  xfun                   0.30       2022-03-02 [1] CRAN (R 4.1.2)
##  XVector                0.34.0     2021-10-26 [1] Bioconductor
##  yaml                   2.3.5      2022-02-21 [1] CRAN (R 4.1.2)
##  zlibbioc               1.40.0     2021-10-26 [1] Bioconductor
## 
##  [1] /Library/Frameworks/R.framework/Versions/4.1/Resources/library
## 
## ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-castellano2021pomashiny" class="csl-entry">
Castellano-Escuder, Pol, Raúl González-Domı́nguez, Francesc Carmona-Pontaque, Cristina Andrés-Lacueva, and Alex Sánchez-Pla. 2021. <span>“POMAShiny: A User-Friendly Web-Based Workflow for Metabolomics and Proteomics Data Analysis.”</span> <em>PLoS Computational Biology</em> 17 (7): e1009148.
</div>
<div id="ref-pang2020metaboanalystr" class="csl-entry">
Pang, Zhiqiang, Jasmine Chong, Shuzhao Li, and Jianguo Xia. 2020. <span>“MetaboAnalystR 3.0: Toward an Optimized Workflow for Global Metabolomics.”</span> <em>Metabolites</em> 10 (5): 186.
</div>
<div id="ref-zeybel2022multiomics" class="csl-entry">
Zeybel, Mujdat, Muhammad Arif, Xiangyu Li, Ozlem Altay, Hong Yang, Mengnan Shi, Murat Akyildiz, et al. 2022. <span>“Multiomics Analysis Reveals the Impact of Microbiota on Host Metabolism in Hepatic Steatosis.”</span> <em>Advanced Science</em> 9 (11): 2104373.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="cluster-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/Metabolomics_Aanlysis/blob/main/01-Data-Processing.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection",
"scroll_highlight": true,
"toc_depth": 3
}
});
});
</script>

</body>

</html>
