<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 10 Differential Analysis | XMAS 2.0 Tutorial</title>
  <meta name="description" content="A tutorial of XMAS 2.0 package for in-house training" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 10 Differential Analysis | XMAS 2.0 Tutorial" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 10 Differential Analysis | XMAS 2.0 Tutorial" />
  
  <meta name="twitter:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

<meta name="author" content="Hua Zou" />


<meta name="date" content="2023-11-30" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="core-microbiota.html"/>
<link rel="next" href="visualization-5.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.28/datatables.js"></script>
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="assets/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">XMAS 2.0 Tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#overview-of-the-design-of-xmas"><i class="fa fa-check"></i><b>1.1</b> Overview of the design of <strong>XMAS</strong></a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#input-data"><i class="fa fa-check"></i><b>1.2</b> Input data</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#methods"><i class="fa fa-check"></i><b>1.3</b> Methods</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#outline-of-this-tutorial"><i class="fa fa-check"></i><b>1.4</b> Outline of this Tutorial</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#key-publications"><i class="fa fa-check"></i><b>1.5</b> Key publications</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#citation"><i class="fa fa-check"></i><b>1.6</b> Citation</a></li>
<li class="chapter" data-level="1.7" data-path="index.html"><a href="index.html#change-log"><i class="fa fa-check"></i><b>1.7</b> Change log</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html"><i class="fa fa-check"></i><b>2</b> How to install XMAS</a>
<ul>
<li class="chapter" data-level="2.1" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#description"><i class="fa fa-check"></i><b>2.1</b> Description</a></li>
<li class="chapter" data-level="2.2" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#installation-of-dependencies"><i class="fa fa-check"></i><b>2.2</b> Installation of dependencies</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#packages-in-cran-bioconductor"><i class="fa fa-check"></i><b>2.2.1</b> Packages in CRAN &amp; Bioconductor</a></li>
<li class="chapter" data-level="2.2.2" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#packages-in-github"><i class="fa fa-check"></i><b>2.2.2</b> Packages in github</a></li>
<li class="chapter" data-level="2.2.3" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#packages-from-website"><i class="fa fa-check"></i><b>2.2.3</b> Packages from website</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#installation"><i class="fa fa-check"></i><b>2.3</b> Installation</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#installation-in-linux-or-mac-os-through-command-line"><i class="fa fa-check"></i><b>2.3.1</b> Installation in linux or Mac os through command line</a></li>
<li class="chapter" data-level="2.3.2" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#installation-by-r-from-gitlab"><i class="fa fa-check"></i><b>2.3.2</b> Installation by R from gitlab</a></li>
<li class="chapter" data-level="2.3.3" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#installation-by-r-from-local-xmas.tar.gz-file"><i class="fa fa-check"></i><b>2.3.3</b> Installation by R from local XMAS*.tar.gz file</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#vignette"><i class="fa fa-check"></i><b>2.4</b> Vignette</a></li>
<li class="chapter" data-level="2.5" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#authors"><i class="fa fa-check"></i><b>2.5</b> Authors</a></li>
<li class="chapter" data-level="2.6" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#future-plan"><i class="fa fa-check"></i><b>2.6</b> Future plan</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html"><i class="fa fa-check"></i><b>3</b> Convert inputdata into phyloseq object</a>
<ul>
<li class="chapter" data-level="3.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#loading-packages"><i class="fa fa-check"></i><b>3.1</b> Loading Packages</a></li>
<li class="chapter" data-level="3.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-data"><i class="fa fa-check"></i><b>3.2</b> Importing Data</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#result-from-dada2"><i class="fa fa-check"></i><b>3.2.1</b> Result from dada2</a></li>
<li class="chapter" data-level="3.2.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#result-from-metaphlan"><i class="fa fa-check"></i><b>3.2.2</b> Result from Metaphlan</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#systematic-information"><i class="fa fa-check"></i><b>3.3</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="quality-evaluation.html"><a href="quality-evaluation.html"><i class="fa fa-check"></i><b>4</b> Quality Evaluation</a>
<ul>
<li class="chapter" data-level="4.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#loading-packages-1"><i class="fa fa-check"></i><b>4.1</b> Loading Packages</a></li>
<li class="chapter" data-level="4.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#importing-data-1"><i class="fa fa-check"></i><b>4.2</b> Importing Data</a></li>
<li class="chapter" data-level="4.3" data-path="quality-evaluation.html"><a href="quality-evaluation.html#reads-track"><i class="fa fa-check"></i><b>4.3</b> Reads track</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#plotting-dad2"><i class="fa fa-check"></i><b>4.3.1</b> Plotting DAD2</a></li>
<li class="chapter" data-level="4.3.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#plotting-metaphlan"><i class="fa fa-check"></i><b>4.3.2</b> Plotting Metaphlan</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="quality-evaluation.html"><a href="quality-evaluation.html#spike-in-brs-sample-assessment"><i class="fa fa-check"></i><b>4.4</b> Spike-in (BRS) sample assessment</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#s-data"><i class="fa fa-check"></i><b>4.4.1</b> 16s data</a></li>
<li class="chapter" data-level="4.4.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#metagenomic-data"><i class="fa fa-check"></i><b>4.4.2</b> Metagenomic data</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="quality-evaluation.html"><a href="quality-evaluation.html#removing-brs"><i class="fa fa-check"></i><b>4.5</b> Removing BRS</a></li>
<li class="chapter" data-level="4.6" data-path="quality-evaluation.html"><a href="quality-evaluation.html#rarefaction-curves"><i class="fa fa-check"></i><b>4.6</b> Rarefaction Curves</a></li>
<li class="chapter" data-level="4.7" data-path="quality-evaluation.html"><a href="quality-evaluation.html#summarize-phyloseq-class-object"><i class="fa fa-check"></i><b>4.7</b> Summarize phyloseq-class object</a></li>
<li class="chapter" data-level="4.8" data-path="quality-evaluation.html"><a href="quality-evaluation.html#variability"><i class="fa fa-check"></i><b>4.8</b> Variability</a>
<ul>
<li class="chapter" data-level="4.8.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#coefficient-of-variation"><i class="fa fa-check"></i><b>4.8.1</b> Coefficient of variation</a></li>
<li class="chapter" data-level="4.8.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#distribution-of-number-of-sequences"><i class="fa fa-check"></i><b>4.8.2</b> Distribution of number of sequences</a></li>
</ul></li>
<li class="chapter" data-level="4.9" data-path="quality-evaluation.html"><a href="quality-evaluation.html#systematic-information-1"><i class="fa fa-check"></i><b>4.9</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="pre-processing.html"><a href="pre-processing.html"><i class="fa fa-check"></i><b>5</b> Pre-processing</a>
<ul>
<li class="chapter" data-level="5.1" data-path="pre-processing.html"><a href="pre-processing.html#loading-packages-2"><i class="fa fa-check"></i><b>5.1</b> Loading Packages</a></li>
<li class="chapter" data-level="5.2" data-path="pre-processing.html"><a href="pre-processing.html#importing-data-2"><i class="fa fa-check"></i><b>5.2</b> Importing Data</a></li>
<li class="chapter" data-level="5.3" data-path="pre-processing.html"><a href="pre-processing.html#transformation"><i class="fa fa-check"></i><b>5.3</b> Transformation</a></li>
<li class="chapter" data-level="5.4" data-path="pre-processing.html"><a href="pre-processing.html#normalization"><i class="fa fa-check"></i><b>5.4</b> Normalization</a></li>
<li class="chapter" data-level="5.5" data-path="pre-processing.html"><a href="pre-processing.html#filtering"><i class="fa fa-check"></i><b>5.5</b> Filtering</a></li>
<li class="chapter" data-level="5.6" data-path="pre-processing.html"><a href="pre-processing.html#trimming"><i class="fa fa-check"></i><b>5.6</b> Trimming</a></li>
<li class="chapter" data-level="5.7" data-path="pre-processing.html"><a href="pre-processing.html#imputation"><i class="fa fa-check"></i><b>5.7</b> Imputation</a></li>
<li class="chapter" data-level="5.8" data-path="pre-processing.html"><a href="pre-processing.html#extracting-specific-taxa-phyloseq-class-object"><i class="fa fa-check"></i><b>5.8</b> Extracting specific taxa phyloseq-class object</a></li>
<li class="chapter" data-level="5.9" data-path="pre-processing.html"><a href="pre-processing.html#aggregating-low-relative-abundance-or-unclassified-taxa-into-others"><i class="fa fa-check"></i><b>5.9</b> Aggregating low relative abundance or unclassified taxa into others</a></li>
<li class="chapter" data-level="5.10" data-path="pre-processing.html"><a href="pre-processing.html#transform-the-abundance-of-taxa-whose-relative-abundance-under-limit-of-detection-lod-into-zeros-only-in-metaphlan23"><i class="fa fa-check"></i><b>5.10</b> Transform the abundance of taxa whose relative abundance under Limit Of Detection (LOD) into Zeros (only in metaphlan2/3)</a></li>
<li class="chapter" data-level="5.11" data-path="pre-processing.html"><a href="pre-processing.html#systematic-information-2"><i class="fa fa-check"></i><b>5.11</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="aplha-diversity.html"><a href="aplha-diversity.html"><i class="fa fa-check"></i><b>6</b> Aplha diversity</a>
<ul>
<li class="chapter" data-level="6.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#loading-packages-3"><i class="fa fa-check"></i><b>6.1</b> Loading Packages</a></li>
<li class="chapter" data-level="6.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#importing-data-3"><i class="fa fa-check"></i><b>6.2</b> Importing Data</a></li>
<li class="chapter" data-level="6.3" data-path="aplha-diversity.html"><a href="aplha-diversity.html#equal-sample-sums"><i class="fa fa-check"></i><b>6.3</b> Equal sample sums</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#dada2_ps-dataset"><i class="fa fa-check"></i><b>6.3.1</b> dada2_ps dataset</a></li>
<li class="chapter" data-level="6.3.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#amplicon_ps-dataset"><i class="fa fa-check"></i><b>6.3.2</b> amplicon_ps dataset</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="aplha-diversity.html"><a href="aplha-diversity.html#non-phylogenetic-diversities"><i class="fa fa-check"></i><b>6.4</b> Non-phylogenetic diversities</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#xmas2-r-package"><i class="fa fa-check"></i><b>6.4.1</b> XMAS2 R package</a></li>
<li class="chapter" data-level="6.4.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#microbiome-r-package"><i class="fa fa-check"></i><b>6.4.2</b> microbiome R package</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="aplha-diversity.html"><a href="aplha-diversity.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>6.5</b> Phylogenetic diversity</a></li>
<li class="chapter" data-level="6.6" data-path="aplha-diversity.html"><a href="aplha-diversity.html#rank-abundance"><i class="fa fa-check"></i><b>6.6</b> Rank Abundance</a></li>
<li class="chapter" data-level="6.7" data-path="aplha-diversity.html"><a href="aplha-diversity.html#evenness"><i class="fa fa-check"></i><b>6.7</b> Evenness</a></li>
<li class="chapter" data-level="6.8" data-path="aplha-diversity.html"><a href="aplha-diversity.html#systematic-information-3"><i class="fa fa-check"></i><b>6.8</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="beta-diversity.html"><a href="beta-diversity.html"><i class="fa fa-check"></i><b>7</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="beta-diversity.html"><a href="beta-diversity.html#loading-packages-4"><i class="fa fa-check"></i><b>7.1</b> Loading Packages</a></li>
<li class="chapter" data-level="7.2" data-path="beta-diversity.html"><a href="beta-diversity.html#importing-data-4"><i class="fa fa-check"></i><b>7.2</b> Importing Data</a></li>
<li class="chapter" data-level="7.3" data-path="beta-diversity.html"><a href="beta-diversity.html#phylogenetic-beta-diversity-metrics"><i class="fa fa-check"></i><b>7.3</b> Phylogenetic beta-diversity metrics</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="beta-diversity.html"><a href="beta-diversity.html#unweighted-unifrac"><i class="fa fa-check"></i><b>7.3.1</b> Unweighted Unifrac</a></li>
<li class="chapter" data-level="7.3.2" data-path="beta-diversity.html"><a href="beta-diversity.html#weighted-unifrac"><i class="fa fa-check"></i><b>7.3.2</b> Weighted Unifrac</a></li>
<li class="chapter" data-level="7.3.3" data-path="beta-diversity.html"><a href="beta-diversity.html#generalized-unifrac-distance-metrics"><i class="fa fa-check"></i><b>7.3.3</b> Generalized UniFrac Distance Metrics</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="beta-diversity.html"><a href="beta-diversity.html#distance-dissimilarity-coefficients-bray-curtis-index"><i class="fa fa-check"></i><b>7.4</b> Distance (Dissimilarity) Coefficients: Bray-Curtis Index</a></li>
<li class="chapter" data-level="7.5" data-path="beta-diversity.html"><a href="beta-diversity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>7.5</b> Checking the homogeneity condition</a></li>
<li class="chapter" data-level="7.6" data-path="beta-diversity.html"><a href="beta-diversity.html#permutational-multivariate-analysis-of-variance-permanova"><i class="fa fa-check"></i><b>7.6</b> Permutational Multivariate Analysis of Variance (PERMANOVA)</a></li>
<li class="chapter" data-level="7.7" data-path="beta-diversity.html"><a href="beta-diversity.html#mantel-test"><i class="fa fa-check"></i><b>7.7</b> Mantel Test</a></li>
<li class="chapter" data-level="7.8" data-path="beta-diversity.html"><a href="beta-diversity.html#analysis-of-similarity-anosim"><i class="fa fa-check"></i><b>7.8</b> Analysis of Similarity (ANOSIM)</a></li>
<li class="chapter" data-level="7.9" data-path="beta-diversity.html"><a href="beta-diversity.html#multi-response-permutation-procedures-mrpp"><i class="fa fa-check"></i><b>7.9</b> Multi-response permutation procedures (MRPP)</a></li>
<li class="chapter" data-level="7.10" data-path="beta-diversity.html"><a href="beta-diversity.html#ordination"><i class="fa fa-check"></i><b>7.10</b> Ordination</a>
<ul>
<li class="chapter" data-level="7.10.1" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>7.10.1</b> Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="7.10.2" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-coordinate-analysis-pcoa"><i class="fa fa-check"></i><b>7.10.2</b> Principal Coordinate Analysis (PCoA)</a></li>
<li class="chapter" data-level="7.10.3" data-path="beta-diversity.html"><a href="beta-diversity.html#t-distributed-stochastic-neighbor-embedding-t-sne"><i class="fa fa-check"></i><b>7.10.3</b> t-distributed stochastic neighbor embedding (t-SNE)</a></li>
<li class="chapter" data-level="7.10.4" data-path="beta-diversity.html"><a href="beta-diversity.html#uniform-manifold-approximation-and-projection-for-dimension-reduction-umap"><i class="fa fa-check"></i><b>7.10.4</b> Uniform Manifold Approximation and Projection for Dimension Reduction (UMAP)</a></li>
</ul></li>
<li class="chapter" data-level="7.11" data-path="beta-diversity.html"><a href="beta-diversity.html#systematic-information-4"><i class="fa fa-check"></i><b>7.11</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="microbial-composition.html"><a href="microbial-composition.html"><i class="fa fa-check"></i><b>8</b> Microbial Composition</a>
<ul>
<li class="chapter" data-level="8.1" data-path="microbial-composition.html"><a href="microbial-composition.html#loading-packages-5"><i class="fa fa-check"></i><b>8.1</b> Loading Packages</a></li>
<li class="chapter" data-level="8.2" data-path="microbial-composition.html"><a href="microbial-composition.html#importing-data-5"><i class="fa fa-check"></i><b>8.2</b> Importing Data</a></li>
<li class="chapter" data-level="8.3" data-path="microbial-composition.html"><a href="microbial-composition.html#plot_stacked_bar-from-xviz"><i class="fa fa-check"></i><b>8.3</b> plot_stacked_bar from XVIZ</a></li>
<li class="chapter" data-level="8.4" data-path="microbial-composition.html"><a href="microbial-composition.html#plot_stackbarplot-from-xams2"><i class="fa fa-check"></i><b>8.4</b> plot_StackBarPlot from XAMS2</a></li>
<li class="chapter" data-level="8.5" data-path="microbial-composition.html"><a href="microbial-composition.html#heatmaps"><i class="fa fa-check"></i><b>8.5</b> Heatmaps</a></li>
<li class="chapter" data-level="8.6" data-path="microbial-composition.html"><a href="microbial-composition.html#systematic-information-5"><i class="fa fa-check"></i><b>8.6</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="core-microbiota.html"><a href="core-microbiota.html"><i class="fa fa-check"></i><b>9</b> Core microbiota</a>
<ul>
<li class="chapter" data-level="9.1" data-path="core-microbiota.html"><a href="core-microbiota.html#loading-packages-6"><i class="fa fa-check"></i><b>9.1</b> Loading Packages</a></li>
<li class="chapter" data-level="9.2" data-path="core-microbiota.html"><a href="core-microbiota.html#importing-data-6"><i class="fa fa-check"></i><b>9.2</b> Importing Data</a></li>
<li class="chapter" data-level="9.3" data-path="core-microbiota.html"><a href="core-microbiota.html#obtaining-core-microbiota"><i class="fa fa-check"></i><b>9.3</b> Obtaining core microbiota</a></li>
<li class="chapter" data-level="9.4" data-path="core-microbiota.html"><a href="core-microbiota.html#showing-core-abundance-and-diversity"><i class="fa fa-check"></i><b>9.4</b> Showing core abundance and diversity</a></li>
<li class="chapter" data-level="9.5" data-path="core-microbiota.html"><a href="core-microbiota.html#visualizing-core-taxa"><i class="fa fa-check"></i><b>9.5</b> Visualizing core taxa</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="core-microbiota.html"><a href="core-microbiota.html#heatmap"><i class="fa fa-check"></i><b>9.5.1</b> heatmap</a></li>
<li class="chapter" data-level="9.5.2" data-path="core-microbiota.html"><a href="core-microbiota.html#linechart"><i class="fa fa-check"></i><b>9.5.2</b> linechart</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="core-microbiota.html"><a href="core-microbiota.html#systematic-information-6"><i class="fa fa-check"></i><b>9.6</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="differential-analysis.html"><a href="differential-analysis.html"><i class="fa fa-check"></i><b>10</b> Differential Analysis</a>
<ul>
<li class="chapter" data-level="10.1" data-path="differential-analysis.html"><a href="differential-analysis.html#loading-packages-7"><i class="fa fa-check"></i><b>10.1</b> Loading Packages</a></li>
<li class="chapter" data-level="10.2" data-path="differential-analysis.html"><a href="differential-analysis.html#importing-data-7"><i class="fa fa-check"></i><b>10.2</b> Importing Data</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="differential-analysis.html"><a href="differential-analysis.html#s-data-1"><i class="fa fa-check"></i><b>10.2.1</b> 16s data</a></li>
<li class="chapter" data-level="10.2.2" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomic-data-1"><i class="fa fa-check"></i><b>10.2.2</b> Metagenomic data</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="differential-analysis.html"><a href="differential-analysis.html#amplicon-sequencing-microbial-data-16s"><i class="fa fa-check"></i><b>10.3</b> Amplicon sequencing microbial data (16s)</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="differential-analysis.html"><a href="differential-analysis.html#aldex2"><i class="fa fa-check"></i><b>10.3.1</b> ALDEx2</a></li>
<li class="chapter" data-level="10.3.2" data-path="differential-analysis.html"><a href="differential-analysis.html#limma_voom"><i class="fa fa-check"></i><b>10.3.2</b> limma_voom</a></li>
<li class="chapter" data-level="10.3.3" data-path="differential-analysis.html"><a href="differential-analysis.html#mbzinb"><i class="fa fa-check"></i><b>10.3.3</b> mbzinb</a></li>
<li class="chapter" data-level="10.3.4" data-path="differential-analysis.html"><a href="differential-analysis.html#omnibus"><i class="fa fa-check"></i><b>10.3.4</b> omnibus</a></li>
<li class="chapter" data-level="10.3.5" data-path="differential-analysis.html"><a href="differential-analysis.html#raida"><i class="fa fa-check"></i><b>10.3.5</b> RAIDA</a></li>
<li class="chapter" data-level="10.3.6" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests"><i class="fa fa-check"></i><b>10.3.6</b> Wilcoxon Rank Sum and Signed Rank Tests</a></li>
<li class="chapter" data-level="10.3.7" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse"><i class="fa fa-check"></i><b>10.3.7</b> Liner discriminant analysis (LDA) effect size (LEfSe)</a></li>
<li class="chapter" data-level="10.3.8" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test"><i class="fa fa-check"></i><b>10.3.8</b> t-test</a></li>
<li class="chapter" data-level="10.3.9" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomeseq"><i class="fa fa-check"></i><b>10.3.9</b> MetagenomeSeq</a></li>
<li class="chapter" data-level="10.3.10" data-path="differential-analysis.html"><a href="differential-analysis.html#deseq2"><i class="fa fa-check"></i><b>10.3.10</b> DESeq2</a></li>
<li class="chapter" data-level="10.3.11" data-path="differential-analysis.html"><a href="differential-analysis.html#edger"><i class="fa fa-check"></i><b>10.3.11</b> EdgeR</a></li>
<li class="chapter" data-level="10.3.12" data-path="differential-analysis.html"><a href="differential-analysis.html#ancom"><i class="fa fa-check"></i><b>10.3.12</b> ANCOM</a></li>
<li class="chapter" data-level="10.3.13" data-path="differential-analysis.html"><a href="differential-analysis.html#corncob"><i class="fa fa-check"></i><b>10.3.13</b> Corncob</a></li>
<li class="chapter" data-level="10.3.14" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models"><i class="fa fa-check"></i><b>10.3.14</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)</a></li>
<li class="chapter" data-level="10.3.15" data-path="differential-analysis.html"><a href="differential-analysis.html#locom"><i class="fa fa-check"></i><b>10.3.15</b> LOCOM</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomic-sequencing-microbial-data-metaphlan23"><i class="fa fa-check"></i><b>10.4</b> Metagenomic sequencing microbial data (metaphlan2/3)</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests-1"><i class="fa fa-check"></i><b>10.4.1</b> Wilcoxon Rank Sum and Signed Rank Tests</a></li>
<li class="chapter" data-level="10.4.2" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse-1"><i class="fa fa-check"></i><b>10.4.2</b> Liner discriminant analysis (LDA) effect size (LEfSe)</a></li>
<li class="chapter" data-level="10.4.3" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test-1"><i class="fa fa-check"></i><b>10.4.3</b> t-test</a></li>
<li class="chapter" data-level="10.4.4" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models-1"><i class="fa fa-check"></i><b>10.4.4</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="differential-analysis.html"><a href="differential-analysis.html#visualization-4"><i class="fa fa-check"></i><b>10.5</b> Visualization</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="differential-analysis.html"><a href="differential-analysis.html#volcano-plot"><i class="fa fa-check"></i><b>10.5.1</b> Volcano plot</a></li>
<li class="chapter" data-level="10.5.2" data-path="differential-analysis.html"><a href="differential-analysis.html#barplot-in-lefse"><i class="fa fa-check"></i><b>10.5.2</b> barplot in lefse</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="differential-analysis.html"><a href="differential-analysis.html#dominant-taxa"><i class="fa fa-check"></i><b>10.6</b> Dominant taxa</a></li>
<li class="chapter" data-level="10.7" data-path="differential-analysis.html"><a href="differential-analysis.html#multiple-differential-analysis-by-one-function"><i class="fa fa-check"></i><b>10.7</b> Multiple differential analysis by one function</a></li>
<li class="chapter" data-level="10.8" data-path="differential-analysis.html"><a href="differential-analysis.html#systematic-information-7"><i class="fa fa-check"></i><b>10.8</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="visualization-5.html"><a href="visualization-5.html"><i class="fa fa-check"></i><b>11</b> Visualization</a>
<ul>
<li class="chapter" data-level="11.1" data-path="visualization-5.html"><a href="visualization-5.html#loading-packages-8"><i class="fa fa-check"></i><b>11.1</b> Loading Packages</a></li>
<li class="chapter" data-level="11.2" data-path="visualization-5.html"><a href="visualization-5.html#importing-data-8"><i class="fa fa-check"></i><b>11.2</b> Importing Data</a></li>
<li class="chapter" data-level="11.3" data-path="visualization-5.html"><a href="visualization-5.html#plot_boxplot"><i class="fa fa-check"></i><b>11.3</b> plot_boxplot</a></li>
<li class="chapter" data-level="11.4" data-path="visualization-5.html"><a href="visualization-5.html#plot_barplot"><i class="fa fa-check"></i><b>11.4</b> plot_barplot</a></li>
<li class="chapter" data-level="11.5" data-path="visualization-5.html"><a href="visualization-5.html#plot_dotplot"><i class="fa fa-check"></i><b>11.5</b> plot_dotplot</a></li>
<li class="chapter" data-level="11.6" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_boxplot"><i class="fa fa-check"></i><b>11.6</b> plot_correlation_boxplot</a></li>
<li class="chapter" data-level="11.7" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_density"><i class="fa fa-check"></i><b>11.7</b> plot_correlation_density</a></li>
<li class="chapter" data-level="11.8" data-path="visualization-5.html"><a href="visualization-5.html#plot_ordination"><i class="fa fa-check"></i><b>11.8</b> plot_Ordination</a></li>
<li class="chapter" data-level="11.9" data-path="visualization-5.html"><a href="visualization-5.html#plot_ggbiplot"><i class="fa fa-check"></i><b>11.9</b> plot_ggbiplot</a></li>
<li class="chapter" data-level="11.10" data-path="visualization-5.html"><a href="visualization-5.html#plot_corrplot"><i class="fa fa-check"></i><b>11.10</b> plot_corrplot</a></li>
<li class="chapter" data-level="11.11" data-path="visualization-5.html"><a href="visualization-5.html#plot_2da_venn"><i class="fa fa-check"></i><b>11.11</b> plot_2DA_venn</a></li>
<li class="chapter" data-level="11.12" data-path="visualization-5.html"><a href="visualization-5.html#plot-the-da-results-from-the-significant-taxa-by-double-barplot"><i class="fa fa-check"></i><b>11.12</b> plot the DA results from the significant taxa by double barplot</a></li>
<li class="chapter" data-level="11.13" data-path="visualization-5.html"><a href="visualization-5.html#plot_stacked_bar_xivz"><i class="fa fa-check"></i><b>11.13</b> plot_stacked_bar_XIVZ</a></li>
<li class="chapter" data-level="11.14" data-path="visualization-5.html"><a href="visualization-5.html#plot_stackbarplot"><i class="fa fa-check"></i><b>11.14</b> plot_StackBarPlot</a></li>
<li class="chapter" data-level="11.15" data-path="visualization-5.html"><a href="visualization-5.html#color-palettes"><i class="fa fa-check"></i><b>11.15</b> Color Palettes</a>
<ul>
<li class="chapter" data-level="11.15.1" data-path="visualization-5.html"><a href="visualization-5.html#wes-anderson-palettes"><i class="fa fa-check"></i><b>11.15.1</b> Wes Anderson Palettes</a></li>
<li class="chapter" data-level="11.15.2" data-path="visualization-5.html"><a href="visualization-5.html#usemycol"><i class="fa fa-check"></i><b>11.15.2</b> useMyCol</a></li>
</ul></li>
<li class="chapter" data-level="11.16" data-path="visualization-5.html"><a href="visualization-5.html#ordination-plots-with-ggplot2"><i class="fa fa-check"></i><b>11.16</b> Ordination plots with ggplot2</a>
<ul>
<li class="chapter" data-level="11.16.1" data-path="visualization-5.html"><a href="visualization-5.html#principal-components-analysis-with-the-iris-data-set"><i class="fa fa-check"></i><b>11.16.1</b> principal components analysis with the iris data set</a></li>
<li class="chapter" data-level="11.16.2" data-path="visualization-5.html"><a href="visualization-5.html#multiple-correspondence-analysis-with-the-tea-dataset"><i class="fa fa-check"></i><b>11.16.2</b> multiple correspondence analysis with the tea dataset</a></li>
<li class="chapter" data-level="11.16.3" data-path="visualization-5.html"><a href="visualization-5.html#linear-discriminant-analysis"><i class="fa fa-check"></i><b>11.16.3</b> linear discriminant analysis</a></li>
</ul></li>
<li class="chapter" data-level="11.17" data-path="visualization-5.html"><a href="visualization-5.html#systematic-information-8"><i class="fa fa-check"></i><b>11.17</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="association-analysis.html"><a href="association-analysis.html"><i class="fa fa-check"></i><b>12</b> Association Analysis</a>
<ul>
<li class="chapter" data-level="12.1" data-path="association-analysis.html"><a href="association-analysis.html#loading-packages-9"><i class="fa fa-check"></i><b>12.1</b> Loading Packages</a></li>
<li class="chapter" data-level="12.2" data-path="association-analysis.html"><a href="association-analysis.html#importing-data-9"><i class="fa fa-check"></i><b>12.2</b> Importing Data</a></li>
<li class="chapter" data-level="12.3" data-path="association-analysis.html"><a href="association-analysis.html#spearman-correlation-analysis"><i class="fa fa-check"></i><b>12.3</b> Spearman Correlation Analysis</a></li>
<li class="chapter" data-level="12.4" data-path="association-analysis.html"><a href="association-analysis.html#partial-correlation-analysis"><i class="fa fa-check"></i><b>12.4</b> Partial Correlation Analysis</a></li>
<li class="chapter" data-level="12.5" data-path="association-analysis.html"><a href="association-analysis.html#systematic-information-9"><i class="fa fa-check"></i><b>12.5</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html"><i class="fa fa-check"></i><b>13</b> Principals of Differential Analysis methods</a>
<ul>
<li class="chapter" data-level="13.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#aldex2-1"><i class="fa fa-check"></i><b>13.1</b> ALDEx2</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-1"><i class="fa fa-check"></i><b>13.1.1</b> Vignette</a></li>
<li class="chapter" data-level="13.1.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type"><i class="fa fa-check"></i><b>13.1.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.1.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments"><i class="fa fa-check"></i><b>13.1.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.1.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures"><i class="fa fa-check"></i><b>13.1.4</b> Procedures</a></li>
<li class="chapter" data-level="13.1.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output"><i class="fa fa-check"></i><b>13.1.5</b> Key output</a></li>
<li class="chapter" data-level="13.1.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax"><i class="fa fa-check"></i><b>13.1.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.1.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax"><i class="fa fa-check"></i><b>13.1.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="13.1.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes"><i class="fa fa-check"></i><b>13.1.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#ancom-ii"><i class="fa fa-check"></i><b>13.2</b> ANCOM-II</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-1"><i class="fa fa-check"></i><b>13.2.1</b> Input Data Type</a></li>
<li class="chapter" data-level="13.2.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-1"><i class="fa fa-check"></i><b>13.2.2</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.2.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-1"><i class="fa fa-check"></i><b>13.2.3</b> Procedures</a></li>
<li class="chapter" data-level="13.2.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-1"><i class="fa fa-check"></i><b>13.2.4</b> Key output</a></li>
<li class="chapter" data-level="13.2.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-1"><i class="fa fa-check"></i><b>13.2.5</b> Original Syntax</a></li>
<li class="chapter" data-level="13.2.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-1"><i class="fa fa-check"></i><b>13.2.6</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="13.2.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-1"><i class="fa fa-check"></i><b>13.2.7</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#corncob-1"><i class="fa fa-check"></i><b>13.3</b> corncob</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-2"><i class="fa fa-check"></i><b>13.3.1</b> Vignette</a></li>
<li class="chapter" data-level="13.3.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-2"><i class="fa fa-check"></i><b>13.3.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.3.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-2"><i class="fa fa-check"></i><b>13.3.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.3.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-2"><i class="fa fa-check"></i><b>13.3.4</b> Procedures</a></li>
<li class="chapter" data-level="13.3.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-2"><i class="fa fa-check"></i><b>13.3.5</b> Key output</a></li>
<li class="chapter" data-level="13.3.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-2"><i class="fa fa-check"></i><b>13.3.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.3.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-2"><i class="fa fa-check"></i><b>13.3.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="13.3.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-2"><i class="fa fa-check"></i><b>13.3.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#deseq2-1"><i class="fa fa-check"></i><b>13.4</b> DESeq2</a>
<ul>
<li class="chapter" data-level="13.4.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-3"><i class="fa fa-check"></i><b>13.4.1</b> Vignette</a></li>
<li class="chapter" data-level="13.4.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-3"><i class="fa fa-check"></i><b>13.4.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.4.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-3"><i class="fa fa-check"></i><b>13.4.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.4.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-3"><i class="fa fa-check"></i><b>13.4.4</b> Procedures</a></li>
<li class="chapter" data-level="13.4.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-3"><i class="fa fa-check"></i><b>13.4.5</b> Key output</a></li>
<li class="chapter" data-level="13.4.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-3"><i class="fa fa-check"></i><b>13.4.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.4.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-3"><i class="fa fa-check"></i><b>13.4.7</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="13.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#edger-1"><i class="fa fa-check"></i><b>13.5</b> edgeR</a>
<ul>
<li class="chapter" data-level="13.5.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-4"><i class="fa fa-check"></i><b>13.5.1</b> Vignette</a></li>
<li class="chapter" data-level="13.5.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-4"><i class="fa fa-check"></i><b>13.5.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.5.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-4"><i class="fa fa-check"></i><b>13.5.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.5.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-4"><i class="fa fa-check"></i><b>13.5.4</b> Procedures</a></li>
<li class="chapter" data-level="13.5.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-4"><i class="fa fa-check"></i><b>13.5.5</b> Key output</a></li>
<li class="chapter" data-level="13.5.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-4"><i class="fa fa-check"></i><b>13.5.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.5.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-4"><i class="fa fa-check"></i><b>13.5.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="13.5.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-3"><i class="fa fa-check"></i><b>13.5.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#lefse"><i class="fa fa-check"></i><b>13.6</b> LEfSe</a>
<ul>
<li class="chapter" data-level="13.6.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-5"><i class="fa fa-check"></i><b>13.6.1</b> Vignette</a></li>
<li class="chapter" data-level="13.6.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-5"><i class="fa fa-check"></i><b>13.6.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.6.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-5"><i class="fa fa-check"></i><b>13.6.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.6.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-5"><i class="fa fa-check"></i><b>13.6.4</b> Procedures</a></li>
<li class="chapter" data-level="13.6.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-5"><i class="fa fa-check"></i><b>13.6.5</b> Key output</a></li>
<li class="chapter" data-level="13.6.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-5"><i class="fa fa-check"></i><b>13.6.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.6.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-5"><i class="fa fa-check"></i><b>13.6.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="13.6.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-4"><i class="fa fa-check"></i><b>13.6.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="13.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#limma-voom"><i class="fa fa-check"></i><b>13.7</b> limma voom</a>
<ul>
<li class="chapter" data-level="13.7.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-6"><i class="fa fa-check"></i><b>13.7.1</b> Vignette</a></li>
<li class="chapter" data-level="13.7.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-6"><i class="fa fa-check"></i><b>13.7.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.7.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-6"><i class="fa fa-check"></i><b>13.7.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.7.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-6"><i class="fa fa-check"></i><b>13.7.4</b> Procedures</a></li>
<li class="chapter" data-level="13.7.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-6"><i class="fa fa-check"></i><b>13.7.5</b> Key output</a></li>
<li class="chapter" data-level="13.7.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-6"><i class="fa fa-check"></i><b>13.7.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.7.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-6"><i class="fa fa-check"></i><b>13.7.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="13.7.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-5"><i class="fa fa-check"></i><b>13.7.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="13.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#maaslin2"><i class="fa fa-check"></i><b>13.8</b> MaAsLin2</a>
<ul>
<li class="chapter" data-level="13.8.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-7"><i class="fa fa-check"></i><b>13.8.1</b> Vignette</a></li>
<li class="chapter" data-level="13.8.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-7"><i class="fa fa-check"></i><b>13.8.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.8.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-7"><i class="fa fa-check"></i><b>13.8.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.8.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-7"><i class="fa fa-check"></i><b>13.8.4</b> Procedures</a></li>
<li class="chapter" data-level="13.8.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-7"><i class="fa fa-check"></i><b>13.8.5</b> Key output</a></li>
<li class="chapter" data-level="13.8.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-7"><i class="fa fa-check"></i><b>13.8.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.8.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-7"><i class="fa fa-check"></i><b>13.8.7</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="13.9" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#metagenomeseq-1"><i class="fa fa-check"></i><b>13.9</b> metagenomeSeq</a>
<ul>
<li class="chapter" data-level="13.9.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-8"><i class="fa fa-check"></i><b>13.9.1</b> Vignette</a></li>
<li class="chapter" data-level="13.9.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-8"><i class="fa fa-check"></i><b>13.9.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.9.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-8"><i class="fa fa-check"></i><b>13.9.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.9.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-8"><i class="fa fa-check"></i><b>13.9.4</b> Procedures</a></li>
<li class="chapter" data-level="13.9.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-8"><i class="fa fa-check"></i><b>13.9.5</b> Key output</a></li>
<li class="chapter" data-level="13.9.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-8"><i class="fa fa-check"></i><b>13.9.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.9.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-8"><i class="fa fa-check"></i><b>13.9.7</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="13.10" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#t-test-2"><i class="fa fa-check"></i><b>13.10</b> t-test</a>
<ul>
<li class="chapter" data-level="13.10.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-9"><i class="fa fa-check"></i><b>13.10.1</b> Input Data Type</a></li>
<li class="chapter" data-level="13.10.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-9"><i class="fa fa-check"></i><b>13.10.2</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.10.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-9"><i class="fa fa-check"></i><b>13.10.3</b> Procedures</a></li>
<li class="chapter" data-level="13.10.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-9"><i class="fa fa-check"></i><b>13.10.4</b> Key output</a></li>
<li class="chapter" data-level="13.10.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-9"><i class="fa fa-check"></i><b>13.10.5</b> Original Syntax</a></li>
<li class="chapter" data-level="13.10.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-9"><i class="fa fa-check"></i><b>13.10.6</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="13.11" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#wilcoxon-test"><i class="fa fa-check"></i><b>13.11</b> Wilcoxon test</a>
<ul>
<li class="chapter" data-level="13.11.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-10"><i class="fa fa-check"></i><b>13.11.1</b> Input Data Type</a></li>
<li class="chapter" data-level="13.11.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-10"><i class="fa fa-check"></i><b>13.11.2</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.11.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-10"><i class="fa fa-check"></i><b>13.11.3</b> Procedures</a></li>
<li class="chapter" data-level="13.11.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-10"><i class="fa fa-check"></i><b>13.11.4</b> Key output</a></li>
<li class="chapter" data-level="13.11.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-10"><i class="fa fa-check"></i><b>13.11.5</b> Original Syntax</a></li>
<li class="chapter" data-level="13.11.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-10"><i class="fa fa-check"></i><b>13.11.6</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="13.12" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#raida-1"><i class="fa fa-check"></i><b>13.12</b> RAIDA</a>
<ul>
<li class="chapter" data-level="13.12.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-9"><i class="fa fa-check"></i><b>13.12.1</b> Vignette</a></li>
<li class="chapter" data-level="13.12.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-11"><i class="fa fa-check"></i><b>13.12.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.12.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-11"><i class="fa fa-check"></i><b>13.12.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.12.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-11"><i class="fa fa-check"></i><b>13.12.4</b> Procedures</a></li>
<li class="chapter" data-level="13.12.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-11"><i class="fa fa-check"></i><b>13.12.5</b> Key output</a></li>
<li class="chapter" data-level="13.12.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-11"><i class="fa fa-check"></i><b>13.12.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.12.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-11"><i class="fa fa-check"></i><b>13.12.7</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="13.13" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#summary"><i class="fa fa-check"></i><b>13.13</b> Summary</a></li>
<li class="chapter" data-level="13.14" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#advantages-and-disadvantages"><i class="fa fa-check"></i><b>13.14</b> Advantages and disadvantages</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="examples.html"><a href="examples.html"><i class="fa fa-check"></i><b>14</b> Examples</a>
<ul>
<li class="chapter" data-level="14.1" data-path="examples.html"><a href="examples.html#loading-packages-10"><i class="fa fa-check"></i><b>14.1</b> Loading Packages</a></li>
<li class="chapter" data-level="14.2" data-path="examples.html"><a href="examples.html#workflow-description"><i class="fa fa-check"></i><b>14.2</b> Workflow description</a></li>
<li class="chapter" data-level="14.3" data-path="examples.html"><a href="examples.html#amplicon-sequencing-16s"><i class="fa fa-check"></i><b>14.3</b> Amplicon sequencing (16s)</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object"><i class="fa fa-check"></i><b>14.3.1</b> Converting inputs into phyloseq-class object</a></li>
<li class="chapter" data-level="14.3.2" data-path="examples.html"><a href="examples.html#quality-control"><i class="fa fa-check"></i><b>14.3.2</b> Quality Control</a></li>
<li class="chapter" data-level="14.3.3" data-path="examples.html"><a href="examples.html#data-processing"><i class="fa fa-check"></i><b>14.3.3</b> Data processing</a></li>
<li class="chapter" data-level="14.3.4" data-path="examples.html"><a href="examples.html#diversity-analysis"><i class="fa fa-check"></i><b>14.3.4</b> Diversity analysis</a></li>
<li class="chapter" data-level="14.3.5" data-path="examples.html"><a href="examples.html#permanova-ordination"><i class="fa fa-check"></i><b>14.3.5</b> PERMANOVA + Ordination</a></li>
<li class="chapter" data-level="14.3.6" data-path="examples.html"><a href="examples.html#microbial-composition-1"><i class="fa fa-check"></i><b>14.3.6</b> Microbial composition</a></li>
<li class="chapter" data-level="14.3.7" data-path="examples.html"><a href="examples.html#differential-analysis-1"><i class="fa fa-check"></i><b>14.3.7</b> Differential Analysis</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="examples.html"><a href="examples.html#metagenomics-mgs"><i class="fa fa-check"></i><b>14.4</b> Metagenomics (MGS)</a>
<ul>
<li class="chapter" data-level="14.4.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object-1"><i class="fa fa-check"></i><b>14.4.1</b> Converting inputs into phyloseq-class object</a></li>
<li class="chapter" data-level="14.4.2" data-path="examples.html"><a href="examples.html#quality-control-1"><i class="fa fa-check"></i><b>14.4.2</b> Quality Control</a></li>
<li class="chapter" data-level="14.4.3" data-path="examples.html"><a href="examples.html#data-processing-1"><i class="fa fa-check"></i><b>14.4.3</b> Data processing</a></li>
<li class="chapter" data-level="14.4.4" data-path="examples.html"><a href="examples.html#diversity-analysis-1"><i class="fa fa-check"></i><b>14.4.4</b> Diversity analysis</a></li>
<li class="chapter" data-level="14.4.5" data-path="examples.html"><a href="examples.html#permanova-ordination-1"><i class="fa fa-check"></i><b>14.4.5</b> PERMANOVA + Ordination</a></li>
<li class="chapter" data-level="14.4.6" data-path="examples.html"><a href="examples.html#microbial-composition-2"><i class="fa fa-check"></i><b>14.4.6</b> Microbial composition</a></li>
<li class="chapter" data-level="14.4.7" data-path="examples.html"><a href="examples.html#differential-analysis-2"><i class="fa fa-check"></i><b>14.4.7</b> Differential Analysis</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="examples.html"><a href="examples.html#systematic-information-10"><i class="fa fa-check"></i><b>14.5</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">XMAS 2.0 Tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="differential-analysis" class="section level1 hasAnchor" number="10">
<h1><span class="header-section-number">Chapter 10</span> Differential Analysis<a href="differential-analysis.html#differential-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Identifying the significant taxa between case and control group is necessary. There are too many differential analysis (DA) approaches to choose <span class="citation">(<a href="#ref-nearing2022microbiome">Nearing et al. 2022</a>)</span>, we provide some of them which focusing on microbial data, including:</p>
<div style="border: 0px;overflow-x: scroll; width:100%; ">
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:unnamed-chunk-139">Table 10.1: </span>Differential analysis tools in XMAS.
</caption>
<thead>
<tr>
<th style="text-align:left;">
Tool.version.
</th>
<th style="text-align:left;">
Input
</th>
<th style="text-align:left;">
Normalization
</th>
<th style="text-align:left;">
Transformation
</th>
<th style="text-align:left;">
Distribution
</th>
<th style="text-align:left;">
MicrobialData
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
ALDEx2 (1.26.0)
</td>
<td style="text-align:left;">
Counts
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
CLR
</td>
<td style="text-align:left;">
Dirichlet-multinormial
</td>
<td style="text-align:left;">
16s
</td>
</tr>
<tr>
<td style="text-align:left;">
limma voom (3.50.1)
</td>
<td style="text-align:left;">
Counts/Relative
</td>
<td style="text-align:left;">
None/TMM
</td>
<td style="text-align:left;">
Log; Precision weighting
</td>
<td style="text-align:left;">
Normal
</td>
<td style="text-align:left;">
16s/MGS
</td>
</tr>
<tr>
<td style="text-align:left;">
mbzinb (0.2)
</td>
<td style="text-align:left;">
Counts
</td>
<td style="text-align:left;">
RLE
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
Zero-inflated negative binomial
</td>
<td style="text-align:left;">
16s
</td>
</tr>
<tr>
<td style="text-align:left;">
omnibus (0.2)
</td>
<td style="text-align:left;">
Counts
</td>
<td style="text-align:left;">
GMPR(Geometric Mean of Pairwise Ratios)
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
Zero-inflated negative binomial
</td>
<td style="text-align:left;">
16s
</td>
</tr>
<tr>
<td style="text-align:left;">
RAIDA (1.0)
</td>
<td style="text-align:left;">
Counts
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
zero-inflated Log
</td>
<td style="text-align:left;">
Modified t-test
</td>
<td style="text-align:left;">
16s
</td>
</tr>
<tr>
<td style="text-align:left;">
Wilcox(rare/CLR)
</td>
<td style="text-align:left;">
Counts/Relative
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None/CLR
</td>
<td style="text-align:left;">
Non-parametric
</td>
<td style="text-align:left;">
16s/MGS
</td>
</tr>
<tr>
<td style="text-align:left;">
LEfSe
</td>
<td style="text-align:left;">
Rarefied Counts/Relative
</td>
<td style="text-align:left;">
TSS
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
Non-parametric
</td>
<td style="text-align:left;">
16s/MGS
</td>
</tr>
<tr>
<td style="text-align:left;">
t-test (rare)
</td>
<td style="text-align:left;">
Counts/Relative
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
Normal
</td>
<td style="text-align:left;">
16s/MGS
</td>
</tr>
<tr>
<td style="text-align:left;">
metagenomeSeq (1.36.0)
</td>
<td style="text-align:left;">
Counts
</td>
<td style="text-align:left;">
CSS
</td>
<td style="text-align:left;">
Log
</td>
<td style="text-align:left;">
Zero-inflated (log) Normal
</td>
<td style="text-align:left;">
16s
</td>
</tr>
<tr>
<td style="text-align:left;">
DESeq2 (1.34.0)
</td>
<td style="text-align:left;">
Counts
</td>
<td style="text-align:left;">
RLE
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
Negative binomial
</td>
<td style="text-align:left;">
16s
</td>
</tr>
<tr>
<td style="text-align:left;">
edgeR (3.36.0)
</td>
<td style="text-align:left;">
Counts
</td>
<td style="text-align:left;">
RLE/TMM
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
Negative binomial
</td>
<td style="text-align:left;">
16s
</td>
</tr>
<tr>
<td style="text-align:left;">
ANCOM-II (2.1)
</td>
<td style="text-align:left;">
Counts/Relative
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
ALR
</td>
<td style="text-align:left;">
Non-parametric
</td>
<td style="text-align:left;">
16s/MGS
</td>
</tr>
<tr>
<td style="text-align:left;">
Corncob (0.2.0)
</td>
<td style="text-align:left;">
Counts
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
None
</td>
<td style="text-align:left;">
Beta-binomial
</td>
<td style="text-align:left;">
16s
</td>
</tr>
<tr>
<td style="text-align:left;">
MaAslin2 (1.8.0)
</td>
<td style="text-align:left;">
Counts/Relative
</td>
<td style="text-align:left;">
None/TSS
</td>
<td style="text-align:left;">
AST
</td>
<td style="text-align:left;">
Normal
</td>
<td style="text-align:left;">
16s/MGS
</td>
</tr>
</tbody>
</table>
</div>
<p>Outline of this Chapter:</p>
<ul>
<li><p><a href="convert-inputdata-into-phyloseq-object.html#loading-packages">Loading Packages</a></p></li>
<li><p><a href="convert-inputdata-into-phyloseq-object.html#importing-data">Importing Data</a></p></li>
<li><p><a href="differential-analysis.html#amplicon-sequencing-microbial-data-16s">Amplicon sequencing microbial data (16s)</a></p></li>
<li><p><a href="differential-analysis.html#metagenomic-sequencing-microbial-data-metaphlan23">Metagenomic sequencing microbial data (metaphlan2/3)</a></p></li>
<li><p><a href="beta-diversity.html#visualization">Visualization</a></p></li>
<li><p><a href="differential-analysis.html#dominant-taxa">Dominant taxa</a></p></li>
<li><p><a href="differential-analysis.html#multiple-differential-analysis-by-one-function">Multiple differential analysis by one function</a></p></li>
</ul>
<div id="loading-packages-7" class="section level2 hasAnchor" number="10.1">
<h2><span class="header-section-number">10.1</span> Loading Packages<a href="differential-analysis.html#loading-packages-7" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb250"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb250-1"><a href="differential-analysis.html#cb250-1" tabindex="-1"></a><span class="fu">library</span>(XMAS2)</span>
<span id="cb250-2"><a href="differential-analysis.html#cb250-2" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb250-3"><a href="differential-analysis.html#cb250-3" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb250-4"><a href="differential-analysis.html#cb250-4" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb250-5"><a href="differential-analysis.html#cb250-5" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb250-6"><a href="differential-analysis.html#cb250-6" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span></code></pre></div>
</div>
<div id="importing-data-7" class="section level2 hasAnchor" number="10.2">
<h2><span class="header-section-number">10.2</span> Importing Data<a href="differential-analysis.html#importing-data-7" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="s-data-1" class="section level3 hasAnchor" number="10.2.1">
<h3><span class="header-section-number">10.2.1</span> 16s data<a href="differential-analysis.html#s-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb251"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb251-1"><a href="differential-analysis.html#cb251-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;dada2_ps&quot;</span>)</span>
<span id="cb251-2"><a href="differential-analysis.html#cb251-2" tabindex="-1"></a></span>
<span id="cb251-3"><a href="differential-analysis.html#cb251-3" tabindex="-1"></a><span class="co"># step1: Removing samples of specific group in phyloseq-class object</span></span>
<span id="cb251-4"><a href="differential-analysis.html#cb251-4" tabindex="-1"></a>dada2_ps_remove_BRS <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb251-5"><a href="differential-analysis.html#cb251-5" tabindex="-1"></a>                     <span class="at">ps =</span> dada2_ps,</span>
<span id="cb251-6"><a href="differential-analysis.html#cb251-6" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb251-7"><a href="differential-analysis.html#cb251-7" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>,</span>
<span id="cb251-8"><a href="differential-analysis.html#cb251-8" tabindex="-1"></a>                     <span class="at">discard =</span> <span class="cn">TRUE</span>)</span>
<span id="cb251-9"><a href="differential-analysis.html#cb251-9" tabindex="-1"></a></span>
<span id="cb251-10"><a href="differential-analysis.html#cb251-10" tabindex="-1"></a><span class="co"># step2: Rarefying counts in phyloseq-class object</span></span>
<span id="cb251-11"><a href="differential-analysis.html#cb251-11" tabindex="-1"></a>dada2_ps_rarefy <span class="ot">&lt;-</span> <span class="fu">norm_rarefy</span>(<span class="at">object =</span> dada2_ps_remove_BRS, </span>
<span id="cb251-12"><a href="differential-analysis.html#cb251-12" tabindex="-1"></a>                               <span class="at">size =</span> <span class="dv">51181</span>)</span>
<span id="cb251-13"><a href="differential-analysis.html#cb251-13" tabindex="-1"></a></span>
<span id="cb251-14"><a href="differential-analysis.html#cb251-14" tabindex="-1"></a><span class="co"># step3: Extracting specific taxa phyloseq-class object </span></span>
<span id="cb251-15"><a href="differential-analysis.html#cb251-15" tabindex="-1"></a>dada2_ps_rare_genus <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(<span class="at">ps =</span> dada2_ps_rarefy, </span>
<span id="cb251-16"><a href="differential-analysis.html#cb251-16" tabindex="-1"></a>                                      <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>, </span>
<span id="cb251-17"><a href="differential-analysis.html#cb251-17" tabindex="-1"></a>                                      <span class="at">absolute =</span> <span class="cn">TRUE</span>)</span>
<span id="cb251-18"><a href="differential-analysis.html#cb251-18" tabindex="-1"></a></span>
<span id="cb251-19"><a href="differential-analysis.html#cb251-19" tabindex="-1"></a><span class="co"># step4: Aggregating low relative abundance or unclassified taxa into others</span></span>
<span id="cb251-20"><a href="differential-analysis.html#cb251-20" tabindex="-1"></a><span class="co"># dada2_ps_genus_LRA &lt;- summarize_LowAbundance_taxa(ps = dada2_ps_rare_genus, </span></span>
<span id="cb251-21"><a href="differential-analysis.html#cb251-21" tabindex="-1"></a><span class="co">#                                                   cutoff = 10, </span></span>
<span id="cb251-22"><a href="differential-analysis.html#cb251-22" tabindex="-1"></a><span class="co">#                                                   unclass = TRUE)</span></span>
<span id="cb251-23"><a href="differential-analysis.html#cb251-23" tabindex="-1"></a></span>
<span id="cb251-24"><a href="differential-analysis.html#cb251-24" tabindex="-1"></a><span class="co"># step4: Filtering the low relative abundance or unclassified taxa by the threshold</span></span>
<span id="cb251-25"><a href="differential-analysis.html#cb251-25" tabindex="-1"></a>dada2_ps_genus_filter <span class="ot">&lt;-</span> <span class="fu">run_filter</span>(<span class="at">ps =</span> dada2_ps_rare_genus, </span>
<span id="cb251-26"><a href="differential-analysis.html#cb251-26" tabindex="-1"></a>                                    <span class="at">cutoff =</span> <span class="dv">10</span>, </span>
<span id="cb251-27"><a href="differential-analysis.html#cb251-27" tabindex="-1"></a>                                    <span class="at">unclass =</span> <span class="cn">TRUE</span>)</span>
<span id="cb251-28"><a href="differential-analysis.html#cb251-28" tabindex="-1"></a></span>
<span id="cb251-29"><a href="differential-analysis.html#cb251-29" tabindex="-1"></a><span class="co"># step5: Trimming the taxa with low occurrence less than threshold</span></span>
<span id="cb251-30"><a href="differential-analysis.html#cb251-30" tabindex="-1"></a>dada2_ps_genus_filter_trim <span class="ot">&lt;-</span> <span class="fu">run_trim</span>(<span class="at">object =</span> dada2_ps_genus_filter, </span>
<span id="cb251-31"><a href="differential-analysis.html#cb251-31" tabindex="-1"></a>                                       <span class="at">cutoff =</span> <span class="fl">0.2</span>, </span>
<span id="cb251-32"><a href="differential-analysis.html#cb251-32" tabindex="-1"></a>                                       <span class="at">trim =</span> <span class="st">&quot;feature&quot;</span>)</span>
<span id="cb251-33"><a href="differential-analysis.html#cb251-33" tabindex="-1"></a>dada2_ps_genus_filter_trim</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 83 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 83 taxa by 6 taxonomic ranks ]</code></pre>
</div>
<div id="metagenomic-data-1" class="section level3 hasAnchor" number="10.2.2">
<h3><span class="header-section-number">10.2.2</span> Metagenomic data<a href="differential-analysis.html#metagenomic-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb253"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb253-1"><a href="differential-analysis.html#cb253-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;metaphlan2_ps&quot;</span>)</span>
<span id="cb253-2"><a href="differential-analysis.html#cb253-2" tabindex="-1"></a></span>
<span id="cb253-3"><a href="differential-analysis.html#cb253-3" tabindex="-1"></a><span class="co"># step1: Removing samples of specific group in phyloseq-class object</span></span>
<span id="cb253-4"><a href="differential-analysis.html#cb253-4" tabindex="-1"></a>metaphlan2_ps_remove_BRS <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb253-5"><a href="differential-analysis.html#cb253-5" tabindex="-1"></a>                     <span class="at">ps =</span> metaphlan2_ps,</span>
<span id="cb253-6"><a href="differential-analysis.html#cb253-6" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb253-7"><a href="differential-analysis.html#cb253-7" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>,</span>
<span id="cb253-8"><a href="differential-analysis.html#cb253-8" tabindex="-1"></a>                     <span class="at">discard =</span> <span class="cn">TRUE</span>)</span>
<span id="cb253-9"><a href="differential-analysis.html#cb253-9" tabindex="-1"></a></span>
<span id="cb253-10"><a href="differential-analysis.html#cb253-10" tabindex="-1"></a><span class="co"># step2: Extracting specific taxa phyloseq-class object </span></span>
<span id="cb253-11"><a href="differential-analysis.html#cb253-11" tabindex="-1"></a>metaphlan2_ps_genus <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(<span class="at">ps =</span> metaphlan2_ps_remove_BRS, </span>
<span id="cb253-12"><a href="differential-analysis.html#cb253-12" tabindex="-1"></a>                                      <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>)</span>
<span id="cb253-13"><a href="differential-analysis.html#cb253-13" tabindex="-1"></a></span>
<span id="cb253-14"><a href="differential-analysis.html#cb253-14" tabindex="-1"></a><span class="co"># step3: Aggregating low relative abundance or unclassified taxa into others</span></span>
<span id="cb253-15"><a href="differential-analysis.html#cb253-15" tabindex="-1"></a><span class="co"># dada2_ps_genus_LRA &lt;- summarize_LowAbundance_taxa(ps = dada2_ps_genus, </span></span>
<span id="cb253-16"><a href="differential-analysis.html#cb253-16" tabindex="-1"></a><span class="co">#                                                   cutoff = 10, </span></span>
<span id="cb253-17"><a href="differential-analysis.html#cb253-17" tabindex="-1"></a><span class="co">#                                                   unclass = TRUE)</span></span>
<span id="cb253-18"><a href="differential-analysis.html#cb253-18" tabindex="-1"></a></span>
<span id="cb253-19"><a href="differential-analysis.html#cb253-19" tabindex="-1"></a><span class="co"># step4: Filtering the low relative abundance or unclassified taxa by the threshold</span></span>
<span id="cb253-20"><a href="differential-analysis.html#cb253-20" tabindex="-1"></a>metaphlan2_ps_genus_filter <span class="ot">&lt;-</span> <span class="fu">run_filter</span>(<span class="at">ps =</span> metaphlan2_ps_genus, </span>
<span id="cb253-21"><a href="differential-analysis.html#cb253-21" tabindex="-1"></a>                                         <span class="at">cutoff =</span> <span class="fl">1e-4</span>, </span>
<span id="cb253-22"><a href="differential-analysis.html#cb253-22" tabindex="-1"></a>                                         <span class="at">unclass =</span> <span class="cn">TRUE</span>)</span>
<span id="cb253-23"><a href="differential-analysis.html#cb253-23" tabindex="-1"></a></span>
<span id="cb253-24"><a href="differential-analysis.html#cb253-24" tabindex="-1"></a><span class="co"># step5: Trimming the taxa with low occurrence less than threshold</span></span>
<span id="cb253-25"><a href="differential-analysis.html#cb253-25" tabindex="-1"></a>metaphlan2_ps_genus_filter_trim <span class="ot">&lt;-</span> <span class="fu">run_trim</span>(<span class="at">object =</span> metaphlan2_ps_genus_filter, </span>
<span id="cb253-26"><a href="differential-analysis.html#cb253-26" tabindex="-1"></a>                                            <span class="at">cutoff =</span> <span class="fl">0.2</span>, </span>
<span id="cb253-27"><a href="differential-analysis.html#cb253-27" tabindex="-1"></a>                                            <span class="at">trim =</span> <span class="st">&quot;feature&quot;</span>)</span>
<span id="cb253-28"><a href="differential-analysis.html#cb253-28" tabindex="-1"></a>metaphlan2_ps_genus_filter_trim</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 46 taxa and 22 samples ]
## sample_data() Sample Data:       [ 22 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 46 taxa by 6 taxonomic ranks ]</code></pre>
</div>
</div>
<div id="amplicon-sequencing-microbial-data-16s" class="section level2 hasAnchor" number="10.3">
<h2><span class="header-section-number">10.3</span> Amplicon sequencing microbial data (16s)<a href="differential-analysis.html#amplicon-sequencing-microbial-data-16s" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="aldex2" class="section level3 hasAnchor" number="10.3.1">
<h3><span class="header-section-number">10.3.1</span> ALDEx2<a href="differential-analysis.html#aldex2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>ALDEx2</strong> package is from <em>Unifying the analysis of high-throughput sequencing datasets: characterizing RNA-seq, 16S rRNA gene sequencing and selective growth experiments by compositional data analysis</em> <span class="citation">(<a href="#ref-fernandes2014unifying">Fernandes et al. 2014</a>)</span>, and its principle is using log-ratio transformation and statistical testing to find the significant Taxa. (Caution: the <strong>otu_table</strong> must be integers).</p>
<p><code>run_aldex</code> provides 11 parameters. For instance, <em>norm</em> and <em>transform</em> are used to normalization and transformation input data. More details to see <code>help(run_aldex)</code>.</p>
<div class="sourceCode" id="cb255"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb255-1"><a href="differential-analysis.html#cb255-1" tabindex="-1"></a>DA_ALDEx2 <span class="ot">&lt;-</span> <span class="fu">run_aldex</span>(</span>
<span id="cb255-2"><a href="differential-analysis.html#cb255-2" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb255-3"><a href="differential-analysis.html#cb255-3" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb255-4"><a href="differential-analysis.html#cb255-4" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb255-5"><a href="differential-analysis.html#cb255-5" tabindex="-1"></a>                <span class="at">method =</span> <span class="st">&quot;t.test&quot;</span>)</span></code></pre></div>
<pre><code>## |------------(25%)----------(50%)----------(75%)----------|</code></pre>
<div class="sourceCode" id="cb257"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb257-1"><a href="differential-analysis.html#cb257-1" tabindex="-1"></a><span class="fu">colnames</span>(DA_ALDEx2)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;Median CLR \n(All)&quot;                &quot;Median CLR\nAA&quot;                    &quot;Median CLR\nBB&quot;                   
## [10] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [13] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [16] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [19] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb259"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb259-1"><a href="differential-analysis.html#cb259-1" tabindex="-1"></a><span class="fu">head</span>(DA_ALDEx2)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment  EffectSize    Pvalue AdjustedPvalue Median CLR \n(All) Median CLR\nAA Median CLR\nBB
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif  0.04861173 0.7039407      0.9107298          -4.007167      -4.276093     -3.8635463
## 2     g__Actinomyces 9_AA vs 14_BB  Nonsignif  0.08736536 0.7184127      0.9295344           2.038553       1.371707      2.3622339
## 3   g__Adlercreutzia 9_AA vs 14_BB  Nonsignif  0.30086313 0.3806117      0.7748127          -1.131394      -2.418139     -0.4730359
## 4    g__Agathobacter 9_AA vs 14_BB  Nonsignif -0.11018078 0.6594997      0.9065660           4.716196       5.534265      4.5602352
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif -0.15038796 0.5438251      0.8336895          -3.480155      -2.939025     -3.7592053
## 6       g__Alistipes 9_AA vs 14_BB  Nonsignif -0.14967229 0.5818133      0.8866180           3.097487       3.917980      2.8090005
##   Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB
## 1                                NA                       0                    0                  0.0                       1.7051442
## 2                        -0.9577718                      30                   26                 50.5                      -1.3107676
## 3                                NA                       0                    0                  9.0                      -2.4683647
## 4                         0.1524904                     316                  329                296.0                       1.7488297
## 5                                NA                       0                    0                  0.0                      -2.1676332
## 6                        -0.1085245                      64                   64                 69.0                       0.2000385
##   Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB
## 1              58.82609         101.777778           31.21429                    21.74                 22.22                 21.43
## 2              50.91304          26.777778           66.42857                    82.61                 88.89                 78.57
## 3              19.21739           5.111111           28.28571                    43.48                 33.33                 50.00
## 4             996.26087        1740.444444          517.85714                    65.22                 55.56                 71.43
## 5              93.78261          30.000000          134.78571                    21.74                 22.22                 21.43
## 6             163.86957         177.888889          154.85714                    73.91                 77.78                 71.43
##   Odds Ratio (95% CI)
## 1    0.67 (-0.11;1.5)
## 2       4.1 (6.8;1.3)
## 3        8.6 (13;4.3)
## 4     0.39 (-1.4;2.2)
## 5      1.5 (2.3;0.71)
## 6     0.88 (0.64;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <strong>ALDEx2</strong>;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue by <strong>ALDEx2</strong>;</p></li>
<li><p><strong>Median CLR (All)/(group AA)/(group BB)</strong>: Median CLR (normalization by <strong>ALDEx2</strong>) in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<p>Please open the below buttons, if you want to see other options for differential analysis in <strong>ALDEx2</strong>.</p>
<details>
<summary>
<strong>run_da() pattern</strong>
</summary>
<p>We also provide another function <code>run_da</code> to run ALDEx2 differential analysis.</p>
<div class="sourceCode" id="cb261"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb261-1"><a href="differential-analysis.html#cb261-1" tabindex="-1"></a>DA_ALDEx2 <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb261-2"><a href="differential-analysis.html#cb261-2" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb261-3"><a href="differential-analysis.html#cb261-3" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb261-4"><a href="differential-analysis.html#cb261-4" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb261-5"><a href="differential-analysis.html#cb261-5" tabindex="-1"></a>                <span class="at">da_method =</span> <span class="st">&quot;aldex&quot;</span>,</span>
<span id="cb261-6"><a href="differential-analysis.html#cb261-6" tabindex="-1"></a>                <span class="at">method =</span> <span class="st">&quot;t.test&quot;</span>)</span></code></pre></div>
<pre><code>## |------------(25%)----------(50%)----------(75%)----------|</code></pre>
<div class="sourceCode" id="cb263"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb263-1"><a href="differential-analysis.html#cb263-1" tabindex="-1"></a><span class="fu">colnames</span>(DA_ALDEx2)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;Median CLR \n(All)&quot;                &quot;Median CLR\nAA&quot;                    &quot;Median CLR\nBB&quot;                   
## [10] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [13] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [16] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [19] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
</details>
<details>
<summary>
<strong>otu_table or sample_table as inputdata</strong>
</summary>
<p>We also provide <em>data_otu</em> and <em>data_sam</em> as input data to run <code>run_aldex</code>.</p>
<div class="sourceCode" id="cb265"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb265-1"><a href="differential-analysis.html#cb265-1" tabindex="-1"></a>DA_ALDEx2 <span class="ot">&lt;-</span> <span class="fu">run_aldex</span>(</span>
<span id="cb265-2"><a href="differential-analysis.html#cb265-2" tabindex="-1"></a>                <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb265-3"><a href="differential-analysis.html#cb265-3" tabindex="-1"></a>                <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb265-4"><a href="differential-analysis.html#cb265-4" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb265-5"><a href="differential-analysis.html#cb265-5" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb265-6"><a href="differential-analysis.html#cb265-6" tabindex="-1"></a>                <span class="at">method =</span> <span class="st">&quot;t.test&quot;</span>)</span></code></pre></div>
<pre><code>## |------------(25%)----------(50%)----------(75%)----------|</code></pre>
<div class="sourceCode" id="cb267"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb267-1"><a href="differential-analysis.html#cb267-1" tabindex="-1"></a><span class="fu">colnames</span>(DA_ALDEx2)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;Median CLR \n(All)&quot;                &quot;Median CLR\nAA&quot;                    &quot;Median CLR\nBB&quot;                   
## [10] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [13] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [16] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [19] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
</details>
<details>
<summary>
<strong>taxa_level option</strong>
</summary>
<p>We also provide <em>taxa_level</em> for choosing the specific taxonomic level to run <code>run_aldex</code>.</p>
<div class="sourceCode" id="cb269"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb269-1"><a href="differential-analysis.html#cb269-1" tabindex="-1"></a>DA_ALDEx2 <span class="ot">&lt;-</span> <span class="fu">run_aldex</span>(</span>
<span id="cb269-2"><a href="differential-analysis.html#cb269-2" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps,</span>
<span id="cb269-3"><a href="differential-analysis.html#cb269-3" tabindex="-1"></a>                <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb269-4"><a href="differential-analysis.html#cb269-4" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb269-5"><a href="differential-analysis.html#cb269-5" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb269-6"><a href="differential-analysis.html#cb269-6" tabindex="-1"></a>                <span class="at">method =</span> <span class="st">&quot;t.test&quot;</span>)</span></code></pre></div>
<pre><code>## |------------(25%)----------(50%)----------(75%)----------|</code></pre>
<div class="sourceCode" id="cb271"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb271-1"><a href="differential-analysis.html#cb271-1" tabindex="-1"></a><span class="fu">colnames</span>(DA_ALDEx2)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;Median CLR \n(All)&quot;                &quot;Median CLR\nAA&quot;                    &quot;Median CLR\nBB&quot;                   
## [10] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [13] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [16] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [19] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
</details>
</div>
<div id="limma_voom" class="section level3 hasAnchor" number="10.3.2">
<h3><span class="header-section-number">10.3.2</span> limma_voom<a href="differential-analysis.html#limma_voom" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>limma</strong> package is from <em>voom: Precision weights unlock linear model analysis tools for RNA-seq read counts</em> <span class="citation">(<a href="#ref-law2014voom">Law et al. 2014</a>)</span>. Firstly, transforming count data to log2-counts per million (logCPM), estimate the mean-variance relationship and use this to compute appropriate observation-level weights. Secondly, fitting multiple linear models by weighted or generalized least squares. Finally, performing empirical bayes statistics for differential expression.</p>
<p><code>run_limma_voom</code> provides 11 parameters. For instance, <em>norm</em> and <em>transform</em> are used to normalization and transformation input data. More details to see <code>help(run_limma_voom)</code>.</p>
<div class="sourceCode" id="cb273"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb273-1"><a href="differential-analysis.html#cb273-1" tabindex="-1"></a>DA_limma_voom <span class="ot">&lt;-</span> <span class="fu">run_limma_voom</span>(</span>
<span id="cb273-2"><a href="differential-analysis.html#cb273-2" tabindex="-1"></a>                    <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb273-3"><a href="differential-analysis.html#cb273-3" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb273-4"><a href="differential-analysis.html#cb273-4" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb273-5"><a href="differential-analysis.html#cb273-5" tabindex="-1"></a></span>
<span id="cb273-6"><a href="differential-analysis.html#cb273-6" tabindex="-1"></a><span class="fu">colnames</span>(DA_limma_voom)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;logFC&quot;                             &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb275"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb275-1"><a href="differential-analysis.html#cb275-1" tabindex="-1"></a><span class="fu">head</span>(DA_limma_voom)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment EffectSize      logFC    Pvalue AdjustedPvalue Log2FoldChange (Median)\nAA_vs_BB
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif  0.1446687  0.1446687 0.9189011      0.9778050                                NA
## 2     g__Actinomyces 9_AA vs 14_BB  Nonsignif  0.4249473  0.4249473 0.7808748      0.9673523                        -0.9577718
## 3   g__Adlercreutzia 9_AA vs 14_BB  Nonsignif  1.7084410  1.7084410 0.2079463      0.7298206                                NA
## 4    g__Agathobacter 9_AA vs 14_BB  Nonsignif  0.2706159  0.2706159 0.8939813      0.9763217                         0.1524904
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif -0.2499025 -0.2499025 0.8518227      0.9763217                                NA
## 6       g__Alistipes 9_AA vs 14_BB  Nonsignif -0.4066430 -0.4066430 0.8167254      0.9736248                        -0.1085245
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All)
## 1                       0                    0                  0.0                       1.7051442              58.82609
## 2                      30                   26                 50.5                      -1.3107676              50.91304
## 3                       0                    0                  9.0                      -2.4683647              19.21739
## 4                     316                  329                296.0                       1.7488297             996.26087
## 5                       0                    0                  0.0                      -2.1676332              93.78261
## 6                      64                   64                 69.0                       0.2000385             163.86957
##   Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1         101.777778           31.21429                    21.74                 22.22                 21.43    0.67 (-0.11;1.5)
## 2          26.777778           66.42857                    82.61                 88.89                 78.57       4.1 (6.8;1.3)
## 3           5.111111           28.28571                    43.48                 33.33                 50.00        8.6 (13;4.3)
## 4        1740.444444          517.85714                    65.22                 55.56                 71.43     0.39 (-1.4;2.2)
## 5          30.000000          134.78571                    21.74                 22.22                 21.43      1.5 (2.3;0.71)
## 6         177.888889          154.85714                    73.91                 77.78                 71.43     0.88 (0.64;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <strong>limma</strong>;</p></li>
<li><p><strong>logFC</strong>: LogFC from groups’ coefficient by <strong>limma</strong>;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue by <strong>limma</strong>;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<p>Please open the below buttons, if you want to see other options for differential analysis in <strong>limma</strong>.</p>
<details>
<summary>
<strong>other options for limma-voom</strong>
</summary>
<div class="sourceCode" id="cb277"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb277-1"><a href="differential-analysis.html#cb277-1" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb277-2"><a href="differential-analysis.html#cb277-2" tabindex="-1"></a>DA_limma_voom <span class="ot">&lt;-</span> <span class="fu">run_limma_voom</span>(</span>
<span id="cb277-3"><a href="differential-analysis.html#cb277-3" tabindex="-1"></a>                    <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb277-4"><a href="differential-analysis.html#cb277-4" tabindex="-1"></a>                    <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb277-5"><a href="differential-analysis.html#cb277-5" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb277-6"><a href="differential-analysis.html#cb277-6" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb277-7"><a href="differential-analysis.html#cb277-7" tabindex="-1"></a></span>
<span id="cb277-8"><a href="differential-analysis.html#cb277-8" tabindex="-1"></a>DA_limma_voom <span class="ot">&lt;-</span> <span class="fu">run_limma_voom</span>(</span>
<span id="cb277-9"><a href="differential-analysis.html#cb277-9" tabindex="-1"></a>                    <span class="at">ps =</span> dada2_ps,</span>
<span id="cb277-10"><a href="differential-analysis.html#cb277-10" tabindex="-1"></a>                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb277-11"><a href="differential-analysis.html#cb277-11" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb277-12"><a href="differential-analysis.html#cb277-12" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb277-13"><a href="differential-analysis.html#cb277-13" tabindex="-1"></a></span>
<span id="cb277-14"><a href="differential-analysis.html#cb277-14" tabindex="-1"></a>DA_limma_voom <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb277-15"><a href="differential-analysis.html#cb277-15" tabindex="-1"></a>                    <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb277-16"><a href="differential-analysis.html#cb277-16" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb277-17"><a href="differential-analysis.html#cb277-17" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb277-18"><a href="differential-analysis.html#cb277-18" tabindex="-1"></a></span>
<span id="cb277-19"><a href="differential-analysis.html#cb277-19" tabindex="-1"></a><span class="fu">head</span>(DA_limma_voom)</span>
<span id="cb277-20"><a href="differential-analysis.html#cb277-20" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="mbzinb" class="section level3 hasAnchor" number="10.3.3">
<h3><span class="header-section-number">10.3.3</span> mbzinb<a href="differential-analysis.html#mbzinb" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>mbzinb</strong> package is from <em>An omnibus test for differential distribution analysis of microbiome sequencing data</em> <span class="citation">(<a href="#ref-chen2018omnibus">J. Chen et al. 2018</a>)</span>. It uses zeroinflated negative binomial model to investigate the significant taxa. (Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb278"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb278-1"><a href="differential-analysis.html#cb278-1" tabindex="-1"></a>DA_mbzinb <span class="ot">&lt;-</span> <span class="fu">run_mbzinb</span>(</span>
<span id="cb278-2"><a href="differential-analysis.html#cb278-2" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb278-3"><a href="differential-analysis.html#cb278-3" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb278-4"><a href="differential-analysis.html#cb278-4" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb278-5"><a href="differential-analysis.html#cb278-5" tabindex="-1"></a><span class="fu">colnames</span>(DA_mbzinb)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                    &quot;EffectSize&quot;                       
##  [7] &quot;base.mean&quot;                         &quot;mean.LFC&quot;                          &quot;base.abund&quot;                       
## [10] &quot;abund.LFC&quot;                         &quot;base.prev&quot;                         &quot;prev.change&quot;                      
## [13] &quot;base.disp&quot;                         &quot;disp.LFC&quot;                          &quot;statistic&quot;                        
## [16] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [19] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [22] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [25] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb280"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb280-1"><a href="differential-analysis.html#cb280-1" tabindex="-1"></a><span class="fu">head</span>(DA_mbzinb)</span></code></pre></div>
<pre><code>##                           TaxaID         Block Enrichment      Pvalue AdjustedPvalue EffectSize  base.mean    mean.LFC base.abund
## 1               g__Adlercreutzia 9_AA vs 14_BB  Nonsignif 0.033645697      0.2327161   23.17460   3.882556  2.80815942   11.64430
## 2              g__Butyricicoccus 9_AA vs 14_BB  Nonsignif 0.069088466      0.3290282   61.00000 225.449255 -0.03994743  289.82492
## 3 g__Clostridium_sensu_stricto_1 9_AA vs 14_BB  Nonsignif 0.006807056      0.1129971  399.57143   9.932788  4.56998285   43.25825
## 4                g__Enterococcus 9_AA vs 14_BB  Nonsignif 0.023035097      0.2057337   60.74603  31.348586  1.47586623   94.04742
## 5             g__Intestinibacter 9_AA vs 14_BB  Nonsignif 0.013689277      0.1420263   93.11111  19.172445  2.12197852   74.66317
## 6     g__Lachnospiraceae_UCG_001 9_AA vs 14_BB  Nonsignif 0.024787192      0.2057337   34.82540  13.541928 -0.36555648  121.90116
##    abund.LFC base.prev prev.change    base.disp  disp.LFC statistic Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All)
## 1  2.2189131 0.3334299   0.1682022 7.637695e-02  2.969534  8.694185                                NA                       0
## 2  0.2351311 0.7778809  -0.1350360 7.033369e-02  2.816332  7.089849                          2.174498                     282
## 3  2.4472798 0.2296160   0.7703828 1.167768e+00  1.501529 12.175016                                NA                      11
## 4  0.3642082 0.3333275   0.3869727 4.418562e-02  4.539508  9.528024                                NA                      36
## 5  0.3698263 0.2567858   0.6082248 2.699653e+00 -1.572935 10.663905                                NA                      14
## 6 -1.7286933 0.1110894   0.1746813 7.536909e-10 28.083675  9.367176                                NA                       0
##   Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB
## 1                    0                  9.0                      -2.4683647              19.21739           5.111111          28.285714
## 2                  316                 70.0                       0.3633225             236.86957         274.000000         213.000000
## 3                    0                 39.5                      -4.8846378             257.21739          14.000000         413.571429
## 4                    0                 51.5                      -1.4234156              73.08696          36.111111          96.857143
## 5                    0                 58.0                      -2.1579524              83.56522          26.888889         120.000000
## 6                    0                  0.0                       2.3793618              21.91304          43.111111           8.285714
##   Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                    43.48                 33.33                 50.00        8.6 (13;4.3)
## 2                    69.57                 77.78                 64.29      0.76 (0.2;1.3)
## 3                    60.87                 22.22                 85.71 11000 (11000;11000)
## 4                    56.52                 33.33                 71.43      1.9 (3.2;0.64)
## 5                    60.87                 22.22                 85.71         4.2 (7;1.4)
## 6                    21.74                 11.11                 28.57    0.48 (-0.93;1.9)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <strong>mbzinb</strong>;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue by <strong>mbzinb</strong>;</p></li>
<li><p><strong>base.mean</strong>: fitted mean abundance parameter times the fitted prevalence in baseline group;</p></li>
<li><p><strong>mean.LFC</strong>: log2-fold change in fitted mean between other group and baseline;</p></li>
<li><p><strong>base.abund</strong>: fitted mean abundance parameter in baseline group;</p></li>
<li><p><strong>abund.LFC</strong>: log2-fold change in fitted mean abundance parameter between other group and baseline;</p></li>
<li><p><strong>base.prev</strong>: fitted prevalence in baseline group;</p></li>
<li><p><strong>prev.change</strong>: (linear) difference in prevalence between baseline group and other group (other-baseline);</p></li>
<li><p><strong>base.disp</strong>: fitted dispersion parameter in baseline group;</p></li>
<li><p><strong>disp.LFC</strong>: log2-fold change in fitted dispersion parameter between other group and baseline;</p></li>
<li><p><strong>statistic</strong>: value of likelihood ratio test statistic;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<p>Please open the below buttons, if you want to see other options for differential analysis in <strong>mbzinb</strong>.</p>
<details>
<summary>
<strong>other options for mbzinb</strong>
</summary>
<div class="sourceCode" id="cb282"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb282-1"><a href="differential-analysis.html#cb282-1" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb282-2"><a href="differential-analysis.html#cb282-2" tabindex="-1"></a>DA_mbzinb <span class="ot">&lt;-</span> <span class="fu">run_mbzinb</span>(</span>
<span id="cb282-3"><a href="differential-analysis.html#cb282-3" tabindex="-1"></a>                <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb282-4"><a href="differential-analysis.html#cb282-4" tabindex="-1"></a>                <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb282-5"><a href="differential-analysis.html#cb282-5" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb282-6"><a href="differential-analysis.html#cb282-6" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb282-7"><a href="differential-analysis.html#cb282-7" tabindex="-1"></a></span>
<span id="cb282-8"><a href="differential-analysis.html#cb282-8" tabindex="-1"></a>DA_mbzinb <span class="ot">&lt;-</span> <span class="fu">run_mbzinb</span>(</span>
<span id="cb282-9"><a href="differential-analysis.html#cb282-9" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps,</span>
<span id="cb282-10"><a href="differential-analysis.html#cb282-10" tabindex="-1"></a>                <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb282-11"><a href="differential-analysis.html#cb282-11" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb282-12"><a href="differential-analysis.html#cb282-12" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb282-13"><a href="differential-analysis.html#cb282-13" tabindex="-1"></a></span>
<span id="cb282-14"><a href="differential-analysis.html#cb282-14" tabindex="-1"></a>DA_mbzinb <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb282-15"><a href="differential-analysis.html#cb282-15" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb282-16"><a href="differential-analysis.html#cb282-16" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb282-17"><a href="differential-analysis.html#cb282-17" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb282-18"><a href="differential-analysis.html#cb282-18" tabindex="-1"></a></span>
<span id="cb282-19"><a href="differential-analysis.html#cb282-19" tabindex="-1"></a><span class="fu">head</span>(DA_mbzinb)</span>
<span id="cb282-20"><a href="differential-analysis.html#cb282-20" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="omnibus" class="section level3 hasAnchor" number="10.3.4">
<h3><span class="header-section-number">10.3.4</span> omnibus<a href="differential-analysis.html#omnibus" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This approach is also from <strong>mbzinb</strong> <span class="citation">(<a href="#ref-chen2018omnibus">J. Chen et al. 2018</a>)</span> package. it uses GMPR (Geometric Mean of Pairwise Ratios) <span class="citation">(<a href="#ref-chen2018gmpr">L. Chen et al. 2018</a>)</span> to get the size factors.</p>
<blockquote>
<p>where we specify models for count (abundance), zero (prevalence) and dispersion part. We also provide likelihood ratio test (zinb.lrt) for different models.</p>
</blockquote>
<p>(Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb283"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb283-1"><a href="differential-analysis.html#cb283-1" tabindex="-1"></a>DA_omnibus <span class="ot">&lt;-</span> <span class="fu">run_omnibus</span>(</span>
<span id="cb283-2"><a href="differential-analysis.html#cb283-2" tabindex="-1"></a>                  <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb283-3"><a href="differential-analysis.html#cb283-3" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb283-4"><a href="differential-analysis.html#cb283-4" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb283-5"><a href="differential-analysis.html#cb283-5" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;omnibus&quot;</span>)</span></code></pre></div>
<pre><code>## Start GMPR normalization ...
## Start Winsorization ...
## Perform filtering ...
## --A total of  82  taxa will be tested with a sample size of 23 !
## --Omnibus test is selected!
## --Dispersion is treated as a parameter of interest!
## Start testing ...
## 10 %
## 20 %
## 30 %
## 40 %
## 50 %
## 60 %
## 70 %
## 80 %
## 90 %
## 100%!
## Handle failed taxa using permutation test!
## Permutation test ....
## Completed!</code></pre>
<div class="sourceCode" id="cb285"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb285-1"><a href="differential-analysis.html#cb285-1" tabindex="-1"></a><span class="fu">colnames</span>(DA_omnibus)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                    &quot;EffectSize&quot;                       
##  [7] &quot;chi.stat&quot;                          &quot;df&quot;                                &quot;abund.baseline&quot;                   
## [10] &quot;prev.baseline&quot;                     &quot;dispersion.baseline&quot;               &quot;abund.LFC.CompvarBB.est&quot;          
## [13] &quot;abund.LFC.CompvarBB.se&quot;            &quot;prev.LOD.CompvarBB.est&quot;            &quot;prev.LOD.CompvarBB.se&quot;            
## [16] &quot;dispersion.LFC.CompvarBB.est&quot;      &quot;dispersion.LFC.CompvarBB.se&quot;       &quot;method&quot;                           
## [19] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [22] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [25] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [28] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb287"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb287-1"><a href="differential-analysis.html#cb287-1" tabindex="-1"></a><span class="fu">head</span>(DA_omnibus)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment    Pvalue AdjustedPvalue EffectSize  chi.stat df abund.baseline prev.baseline
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif 0.2502358      0.5862666   70.56349 4.1060717  3   0.0051993264     0.2909212
## 2     g__Actinomyces 9_AA vs 14_BB  Nonsignif 0.1648826      0.4991395   39.65079 5.0962611  3   0.0006960207     0.8897527
## 3   g__Adlercreutzia 9_AA vs 14_BB  Nonsignif 0.0519902      0.3401811   23.17460 7.7275906  3   0.0003591401     0.3333635
## 4    g__Agathobacter 9_AA vs 14_BB  Nonsignif 0.1357775      0.4991395 1222.58730 5.5483110  3   0.0542267672     0.5555637
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif 0.8592253      0.9366882  104.78571 0.7590890  3   0.0030796788     0.2352914
## 6       g__Alistipes 9_AA vs 14_BB  Nonsignif 0.9750394      0.9783714   23.03175 0.2155583  3   0.0048440374     0.7818935
##   dispersion.baseline abund.LFC.CompvarBB.est abund.LFC.CompvarBB.se prev.LOD.CompvarBB.est prev.LOD.CompvarBB.se
## 1           0.2032144             -0.36486632              1.4055480             -0.4083678             0.9812349
## 2           2.8889814              0.72441843              0.3103895             -0.7762894             1.2490723
## 3          12.8780293              1.35189939              0.3640153              0.6970149             0.8863839
## 4           1.7061795             -1.10299087              0.4729012              0.6972936             0.8947909
## 5           0.5226139             -0.79545547              1.7405424              2.3810260             1.0097634
## 6           0.9949699              0.05944083              0.4766123             -0.3566236             1.0010846
##   dispersion.LFC.CompvarBB.est dispersion.LFC.CompvarBB.se  method Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All)
## 1                    2.8889332                   1.1380088 omnibus                                NA                       0
## 2                   -0.4153735                   0.6702301 omnibus                        -0.9577718                      30
## 3                   -2.0028316                   1.5731232 omnibus                                NA                       0
## 4                   -0.5953579                   0.7021925 omnibus                         0.1524904                     316
## 5                   -2.4791796                   1.0588301 omnibus                                NA                       0
## 6                    0.1881385                   0.6267815 omnibus                        -0.1085245                      64
##   Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB
## 1                    0                  0.0                       1.7051442              58.82609         101.777778           31.21429
## 2                   26                 50.5                      -1.3107676              50.91304          26.777778           66.42857
## 3                    0                  9.0                      -2.4683647              19.21739           5.111111           28.28571
## 4                  329                296.0                       1.7488297             996.26087        1740.444444          517.85714
## 5                    0                  0.0                      -2.1676332              93.78261          30.000000          134.78571
## 6                   64                 69.0                       0.2000385             163.86957         177.888889          154.85714
##   Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                    21.74                 22.22                 21.43    0.67 (-0.11;1.5)
## 2                    82.61                 88.89                 78.57       4.1 (6.8;1.3)
## 3                    43.48                 33.33                 50.00        8.6 (13;4.3)
## 4                    65.22                 55.56                 71.43     0.39 (-1.4;2.2)
## 5                    21.74                 22.22                 21.43      1.5 (2.3;0.71)
## 6                    73.91                 77.78                 71.43     0.88 (0.64;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <em>glm</em> function to assess the pvalue effect;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue by <strong>omnibus</strong>;</p></li>
<li><p><strong>chi.stat</strong>: chisquare statistics;</p></li>
<li><p><strong>df</strong>: degree freedom;</p></li>
<li><p><strong>abund.baseline</strong>: mean abundance in baseline group;</p></li>
<li><p><strong>prev.baseline</strong>: prevalence in baseline group;</p></li>
<li><p><strong>dispersion.baseline</strong>: dispersion in baseline group;</p></li>
<li><p><strong>abund.LFC.CompvarBB.est</strong>: log2-fold change in abundance between group BB and other group;</p></li>
<li><p><strong>abund.LFC.CompvarBB.se</strong>: log2-fold change of standard errors in abundance between group BB and other group;</p></li>
<li><p><strong>prev.LOD.CompvarBB.est</strong>: prevalence of low of detect value in group BB;</p></li>
<li><p><strong>prev.LOD.CompvarBB.se</strong>: prevalence’s standard errors of low of detect value in group BB;</p></li>
<li><p><strong>dispersion.LFC.CompvarBB.est</strong>: log2-fold change in dispersion in group BB;</p></li>
<li><p><strong>dispersion.LFC.CompvarBB.se</strong>: log2-fold change in dispersion’s standard errors in group BB;</p></li>
<li><p><strong>method</strong>: methods of test;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<p>Please open the below buttons, if you want to see other options for differential analysis in <strong>omnibus</strong>.</p>
<details>
<summary>
<strong>other options for omnibus</strong>
</summary>
<div class="sourceCode" id="cb289"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb289-1"><a href="differential-analysis.html#cb289-1" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb289-2"><a href="differential-analysis.html#cb289-2" tabindex="-1"></a>DA_omnibus <span class="ot">&lt;-</span> <span class="fu">run_omnibus</span>(</span>
<span id="cb289-3"><a href="differential-analysis.html#cb289-3" tabindex="-1"></a>                  <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb289-4"><a href="differential-analysis.html#cb289-4" tabindex="-1"></a>                  <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb289-5"><a href="differential-analysis.html#cb289-5" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb289-6"><a href="differential-analysis.html#cb289-6" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb289-7"><a href="differential-analysis.html#cb289-7" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;omnibus&quot;</span>)</span>
<span id="cb289-8"><a href="differential-analysis.html#cb289-8" tabindex="-1"></a></span>
<span id="cb289-9"><a href="differential-analysis.html#cb289-9" tabindex="-1"></a></span>
<span id="cb289-10"><a href="differential-analysis.html#cb289-10" tabindex="-1"></a>DA_omnibus <span class="ot">&lt;-</span> <span class="fu">run_omnibus</span>(</span>
<span id="cb289-11"><a href="differential-analysis.html#cb289-11" tabindex="-1"></a>                  <span class="at">ps =</span> dada2_ps,</span>
<span id="cb289-12"><a href="differential-analysis.html#cb289-12" tabindex="-1"></a>                  <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb289-13"><a href="differential-analysis.html#cb289-13" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb289-14"><a href="differential-analysis.html#cb289-14" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb289-15"><a href="differential-analysis.html#cb289-15" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;omnibus&quot;</span>)</span>
<span id="cb289-16"><a href="differential-analysis.html#cb289-16" tabindex="-1"></a></span>
<span id="cb289-17"><a href="differential-analysis.html#cb289-17" tabindex="-1"></a>DA_omnibus <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb289-18"><a href="differential-analysis.html#cb289-18" tabindex="-1"></a>                  <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb289-19"><a href="differential-analysis.html#cb289-19" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb289-20"><a href="differential-analysis.html#cb289-20" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb289-21"><a href="differential-analysis.html#cb289-21" tabindex="-1"></a>                  <span class="at">da_method =</span> <span class="st">&quot;omnibus&quot;</span>,</span>
<span id="cb289-22"><a href="differential-analysis.html#cb289-22" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;omnibus&quot;</span>)</span>
<span id="cb289-23"><a href="differential-analysis.html#cb289-23" tabindex="-1"></a></span>
<span id="cb289-24"><a href="differential-analysis.html#cb289-24" tabindex="-1"></a><span class="fu">head</span>(DA_omnibus)</span>
<span id="cb289-25"><a href="differential-analysis.html#cb289-25" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="raida" class="section level3 hasAnchor" number="10.3.5">
<h3><span class="header-section-number">10.3.5</span> RAIDA<a href="differential-analysis.html#raida" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>RAIDA</strong> package is from <em>A robust approach for identifying differentially abundant features in metagenomic samples</em> <span class="citation">(<a href="#ref-sohn2015robust">Sohn, Du, and An 2015</a>)</span>. It uses Ratio Approach for Identifying Differential Abundance (RAIDA). (Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb290"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb290-1"><a href="differential-analysis.html#cb290-1" tabindex="-1"></a>DA_RAIDA <span class="ot">&lt;-</span> <span class="fu">run_raida</span>(</span>
<span id="cb290-2"><a href="differential-analysis.html#cb290-2" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb290-3"><a href="differential-analysis.html#cb290-3" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb290-4"><a href="differential-analysis.html#cb290-4" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb290-5"><a href="differential-analysis.html#cb290-5" tabindex="-1"></a></span>
<span id="cb290-6"><a href="differential-analysis.html#cb290-6" tabindex="-1"></a><span class="fu">colnames</span>(DA_RAIDA)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;eta_AA&quot;                            &quot;mean_AA&quot;                           &quot;sd_AA&quot;                            
## [10] &quot;eta_BB&quot;                            &quot;mean_BB&quot;                           &quot;sd_BB&quot;                            
## [13] &quot;mod.pool.var&quot;                      &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [16] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [19] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [22] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [25] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb292"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb292-1"><a href="differential-analysis.html#cb292-1" tabindex="-1"></a><span class="fu">head</span>(DA_RAIDA)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment  EffectSize    Pvalue AdjustedPvalue    eta_AA    mean_AA     sd_AA    eta_BB    mean_BB
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif   70.563492 0.3554134      0.7515687 0.7777778 -1.1787353 3.4581703 0.7857143 -2.5738907
## 2     g__Actinomyces 9_AA vs 14_BB  Nonsignif   39.650794 0.3605048      0.7515687 0.1111111 -3.5858713 1.3594755 0.2142857 -2.9198320
## 3   g__Adlercreutzia 9_AA vs 14_BB  Nonsignif   23.174603 0.1340594      0.7013530 0.6666667 -4.4427957 0.8570999 0.5000000 -3.0988711
## 4    g__Agathobacter 9_AA vs 14_BB  Nonsignif 1222.587302 0.1441136      0.7013530 0.4444444  0.1696923 1.1949839 0.2857143 -0.8127871
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif  104.785714 0.6189981      0.8235732 0.7777778 -3.7046260 2.3018074 0.7857143 -4.4749687
## 6     g__Allisonella 9_AA vs 14_BB  Nonsignif    3.603175 0.8618432      0.9608650 0.7777778 -3.0106395 0.5628666 0.7142857 -2.8074387
##       sd_BB mod.pool.var Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB
## 1 1.3069089     2.560562                                NA                       0                    0                  0.0
## 2 1.8200871     2.378199                        -0.9577718                      30                   26                 50.5
## 3 1.0974531     1.550430                                NA                       0                    0                  9.0
## 4 0.9057351     1.412385                         0.1524904                     316                  329                296.0
## 5 2.5368159     2.756144                                NA                       0                    0                  0.0
## 6 1.3346198     1.757571                                NA                       0                    0                  0.0
##   Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All)
## 1                       1.7051442              58.82609         101.777778           31.21429                    21.74
## 2                      -1.3107676              50.91304          26.777778           66.42857                    82.61
## 3                      -2.4683647              19.21739           5.111111           28.28571                    43.48
## 4                       1.7488297             996.26087        1740.444444          517.85714                    65.22
## 5                      -2.1676332              93.78261          30.000000          134.78571                    21.74
## 6                      -0.4052144              13.30435          11.111111           14.71429                    26.09
##   Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                 22.22                 21.43    0.67 (-0.11;1.5)
## 2                 88.89                 78.57       4.1 (6.8;1.3)
## 3                 33.33                 50.00        8.6 (13;4.3)
## 4                 55.56                 71.43     0.39 (-1.4;2.2)
## 5                 22.22                 21.43      1.5 (2.3;0.71)
## 6                 22.22                 28.57      1.1 (1.4;0.89)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <em>glm</em> function to assess the pvalue effect;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue by <strong>RAIDA</strong>;</p></li>
<li><p><strong>eta_AA</strong>: vector containing estimated probabilities of the false zero state for group AA;</p></li>
<li><p><strong>mean_AA</strong>: vector containing estimated means of log ratios for group AA;</p></li>
<li><p><strong>sd_AA</strong>: vector containing estimated standard deviations of log ratios for group AA;</p></li>
<li><p><strong>eta_BB</strong>: vector containing estimated probabilities of the false zero state for group BB;</p></li>
<li><p><strong>mean_BB</strong>: vector containing estimated means of log ratios for group BB;</p></li>
<li><p><strong>sd_BB</strong>: vector containing estimated standard deviations of log ratios for group BB;</p></li>
<li><p><strong>mod.pool.var</strong>: vector containing estimated posterior variances of log ratios;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<p>Please open the below buttons, if you want to see other options for differential analysis in <strong>RAIDA</strong>.</p>
<details>
<summary>
<strong>other options for RAIDA</strong>
</summary>
<div class="sourceCode" id="cb294"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb294-1"><a href="differential-analysis.html#cb294-1" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb294-2"><a href="differential-analysis.html#cb294-2" tabindex="-1"></a>DA_RAIDA <span class="ot">&lt;-</span> <span class="fu">run_raida</span>(</span>
<span id="cb294-3"><a href="differential-analysis.html#cb294-3" tabindex="-1"></a>                <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb294-4"><a href="differential-analysis.html#cb294-4" tabindex="-1"></a>                <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb294-5"><a href="differential-analysis.html#cb294-5" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb294-6"><a href="differential-analysis.html#cb294-6" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb294-7"><a href="differential-analysis.html#cb294-7" tabindex="-1"></a></span>
<span id="cb294-8"><a href="differential-analysis.html#cb294-8" tabindex="-1"></a>DA_RAIDA <span class="ot">&lt;-</span> <span class="fu">run_raida</span>(</span>
<span id="cb294-9"><a href="differential-analysis.html#cb294-9" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps,</span>
<span id="cb294-10"><a href="differential-analysis.html#cb294-10" tabindex="-1"></a>                <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb294-11"><a href="differential-analysis.html#cb294-11" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb294-12"><a href="differential-analysis.html#cb294-12" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb294-13"><a href="differential-analysis.html#cb294-13" tabindex="-1"></a></span>
<span id="cb294-14"><a href="differential-analysis.html#cb294-14" tabindex="-1"></a>DA_RAIDA <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb294-15"><a href="differential-analysis.html#cb294-15" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb294-16"><a href="differential-analysis.html#cb294-16" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb294-17"><a href="differential-analysis.html#cb294-17" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb294-18"><a href="differential-analysis.html#cb294-18" tabindex="-1"></a>                <span class="at">da_method =</span> <span class="st">&quot;raida&quot;</span>)</span>
<span id="cb294-19"><a href="differential-analysis.html#cb294-19" tabindex="-1"></a></span>
<span id="cb294-20"><a href="differential-analysis.html#cb294-20" tabindex="-1"></a><span class="fu">head</span>(DA_RAIDA)</span>
<span id="cb294-21"><a href="differential-analysis.html#cb294-21" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="wilcoxon-rank-sum-and-signed-rank-tests" class="section level3 hasAnchor" number="10.3.6">
<h3><span class="header-section-number">10.3.6</span> Wilcoxon Rank Sum and Signed Rank Tests<a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Wilcoxon Rank Sum and Signed Rank Tests, which are nonparametric test methods, use the rank of taxa abundance to find the significant taxa.</p>
<ul>
<li>Ordinary pattern</li>
</ul>
<div class="sourceCode" id="cb295"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb295-1"><a href="differential-analysis.html#cb295-1" tabindex="-1"></a>DA_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb295-2"><a href="differential-analysis.html#cb295-2" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb295-3"><a href="differential-analysis.html#cb295-3" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb295-4"><a href="differential-analysis.html#cb295-4" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb295-5"><a href="differential-analysis.html#cb295-5" tabindex="-1"></a></span>
<span id="cb295-6"><a href="differential-analysis.html#cb295-6" tabindex="-1"></a><span class="fu">colnames</span>(DA_wilcox)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Statistic&quot;                         &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Rank)\nAA_vs_BB&quot;  
## [13] &quot;Mean Rank Abundance\nAA&quot;           &quot;Mean Rank Abundance\nBB&quot;           &quot;Occurrence (100%)\n(All)&quot;         
## [16] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb297"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb297-1"><a href="differential-analysis.html#cb297-1" tabindex="-1"></a><span class="fu">head</span>(DA_wilcox)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment EffectSize Statistic    Pvalue AdjustedPvalue Log2FoldChange (Median)\nAA_vs_BB
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif   70.56349      63.5 1.0000000      1.0000000                                NA
## 2     g__Actinomyces 9_AA vs 14_BB  Nonsignif   39.65079      44.5 0.2556616      0.6753999                        -0.9577718
## 3   g__Adlercreutzia 9_AA vs 14_BB  Nonsignif   23.17460      44.0 0.1980169      0.6753999                                NA
## 4    g__Agathobacter 9_AA vs 14_BB  Nonsignif 1222.58730      71.0 0.6293971      0.9164905                         0.1524904
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif  104.78571      64.5 0.9304707      0.9775831                                NA
## 6       g__Alistipes 9_AA vs 14_BB  Nonsignif   23.03175      67.0 0.8239942      0.9497378                        -0.1085245
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Rank)\nAA_vs_BB Mean Rank Abundance\nAA
## 1                       0                    0                  0.0                      0.01201252                   12.06
## 2                      30                   26                 50.5                     -0.42227633                    9.94
## 3                       0                    0                  9.0                     -0.43387758                    9.89
## 4                     316                  329                296.0                      0.17342686                   12.89
## 5                       0                    0                  0.0                      0.03358045                   12.17
## 6                      64                   64                 69.0                      0.08724541                   12.44
##   Mean Rank Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                   11.96                    21.74                 22.22                 21.43    0.67 (-0.11;1.5)
## 2                   13.32                    82.61                 88.89                 78.57       4.1 (6.8;1.3)
## 3                   13.36                    43.48                 33.33                 50.00        8.6 (13;4.3)
## 4                   11.43                    65.22                 55.56                 71.43     0.39 (-1.4;2.2)
## 5                   11.89                    21.74                 22.22                 21.43      1.5 (2.3;0.71)
## 6                   11.71                    73.91                 77.78                 71.43     0.88 (0.64;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <em>glm</em> function to assess the pvalue effect;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean Rank)_vs_BB**: Log2FoldChange (Mean Rank Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Rank Abundance (All)/(group AA)/(group BB)</strong>: Mean Rank Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for wilcox</strong>
</summary>
<div class="sourceCode" id="cb299"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb299-1"><a href="differential-analysis.html#cb299-1" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb299-2"><a href="differential-analysis.html#cb299-2" tabindex="-1"></a>DA_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb299-3"><a href="differential-analysis.html#cb299-3" tabindex="-1"></a>                <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb299-4"><a href="differential-analysis.html#cb299-4" tabindex="-1"></a>                <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb299-5"><a href="differential-analysis.html#cb299-5" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb299-6"><a href="differential-analysis.html#cb299-6" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb299-7"><a href="differential-analysis.html#cb299-7" tabindex="-1"></a></span>
<span id="cb299-8"><a href="differential-analysis.html#cb299-8" tabindex="-1"></a>DA_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb299-9"><a href="differential-analysis.html#cb299-9" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps,</span>
<span id="cb299-10"><a href="differential-analysis.html#cb299-10" tabindex="-1"></a>                <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb299-11"><a href="differential-analysis.html#cb299-11" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb299-12"><a href="differential-analysis.html#cb299-12" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb299-13"><a href="differential-analysis.html#cb299-13" tabindex="-1"></a></span>
<span id="cb299-14"><a href="differential-analysis.html#cb299-14" tabindex="-1"></a>DA_wilcox <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb299-15"><a href="differential-analysis.html#cb299-15" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb299-16"><a href="differential-analysis.html#cb299-16" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb299-17"><a href="differential-analysis.html#cb299-17" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb299-18"><a href="differential-analysis.html#cb299-18" tabindex="-1"></a>                <span class="at">da_method =</span> <span class="st">&quot;wilcox&quot;</span>)</span>
<span id="cb299-19"><a href="differential-analysis.html#cb299-19" tabindex="-1"></a></span>
<span id="cb299-20"><a href="differential-analysis.html#cb299-20" tabindex="-1"></a><span class="fu">head</span>(DA_wilcox)</span>
<span id="cb299-21"><a href="differential-analysis.html#cb299-21" tabindex="-1"></a>}</span></code></pre></div>
</details>
<ul>
<li><strong>wilcox_rarefy</strong>: random subsampling counts to the smallest library size in the data set.</li>
</ul>
<div class="sourceCode" id="cb300"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb300-1"><a href="differential-analysis.html#cb300-1" tabindex="-1"></a><span class="co"># summary</span></span>
<span id="cb300-2"><a href="differential-analysis.html#cb300-2" tabindex="-1"></a><span class="fu">summarize_phyloseq</span>(<span class="at">ps =</span> dada2_ps_genus_filter_trim)</span></code></pre></div>
<pre><code>## [[1]]
## [1] &quot;1] Min. number of reads = 32334&quot;
## 
## [[2]]
## [1] &quot;2] Max. number of reads = 49308&quot;
## 
## [[3]]
## [1] &quot;3] Total number of reads = 1006186&quot;
## 
## [[4]]
## [1] &quot;4] Average number of reads = 43747.2173913043&quot;
## 
## [[5]]
## [1] &quot;5] Median number of reads = 45549&quot;
## 
## [[6]]
## [1] &quot;7] Sparsity = 0.462022001047669&quot;
## 
## [[7]]
## [1] &quot;6] Any OTU sum to 1 or less? NO&quot;
## 
## [[8]]
## [1] &quot;8] Number of singletons = 0&quot;
## 
## [[9]]
## [1] &quot;9] Percent of OTUs that are singletons\n        (i.e. exactly one read detected across all samples)0&quot;
## 
## [[10]]
## [1] &quot;10] Number of sample variables are: 1&quot;
## 
## [[11]]
## [1] &quot;Group&quot;</code></pre>
<div class="sourceCode" id="cb302"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb302-1"><a href="differential-analysis.html#cb302-1" tabindex="-1"></a><span class="co"># run</span></span>
<span id="cb302-2"><a href="differential-analysis.html#cb302-2" tabindex="-1"></a>DA_wilcox_rarefy <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb302-3"><a href="differential-analysis.html#cb302-3" tabindex="-1"></a>                        <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb302-4"><a href="differential-analysis.html#cb302-4" tabindex="-1"></a>                        <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb302-5"><a href="differential-analysis.html#cb302-5" tabindex="-1"></a>                        <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb302-6"><a href="differential-analysis.html#cb302-6" tabindex="-1"></a>                        <span class="at">norm =</span> <span class="st">&quot;rarefy&quot;</span>)</span>
<span id="cb302-7"><a href="differential-analysis.html#cb302-7" tabindex="-1"></a><span class="fu">colnames</span>(DA_wilcox_rarefy)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Statistic&quot;                         &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Rank)\nAA_vs_BB&quot;  
## [13] &quot;Mean Rank Abundance\nAA&quot;           &quot;Mean Rank Abundance\nBB&quot;           &quot;Occurrence (100%)\n(All)&quot;         
## [16] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<p>The output is the same as the previous results of ordinary pattern.</p>
<details>
<summary>
<strong>other options for wilcox_rarefy</strong>
</summary>
<div class="sourceCode" id="cb304"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb304-1"><a href="differential-analysis.html#cb304-1" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb304-2"><a href="differential-analysis.html#cb304-2" tabindex="-1"></a>DA_wilcox_rarefy <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb304-3"><a href="differential-analysis.html#cb304-3" tabindex="-1"></a>                        <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb304-4"><a href="differential-analysis.html#cb304-4" tabindex="-1"></a>                        <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb304-5"><a href="differential-analysis.html#cb304-5" tabindex="-1"></a>                        <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb304-6"><a href="differential-analysis.html#cb304-6" tabindex="-1"></a>                        <span class="at">da_method =</span> <span class="st">&quot;wilcox&quot;</span>,</span>
<span id="cb304-7"><a href="differential-analysis.html#cb304-7" tabindex="-1"></a>                        <span class="at">norm =</span> <span class="st">&quot;rarefy&quot;</span>)</span>
<span id="cb304-8"><a href="differential-analysis.html#cb304-8" tabindex="-1"></a></span>
<span id="cb304-9"><a href="differential-analysis.html#cb304-9" tabindex="-1"></a><span class="fu">head</span>(DA_wilcox_rarefy)</span>
<span id="cb304-10"><a href="differential-analysis.html#cb304-10" tabindex="-1"></a>}</span></code></pre></div>
</details>
<ul>
<li><strong>wilcox_CLR</strong>: centered log-ratio normalization.</li>
</ul>
<div class="sourceCode" id="cb305"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb305-1"><a href="differential-analysis.html#cb305-1" tabindex="-1"></a>DA_wilcox_CLR <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb305-2"><a href="differential-analysis.html#cb305-2" tabindex="-1"></a>                    <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb305-3"><a href="differential-analysis.html#cb305-3" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb305-4"><a href="differential-analysis.html#cb305-4" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb305-5"><a href="differential-analysis.html#cb305-5" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CLR&quot;</span>)</span>
<span id="cb305-6"><a href="differential-analysis.html#cb305-6" tabindex="-1"></a></span>
<span id="cb305-7"><a href="differential-analysis.html#cb305-7" tabindex="-1"></a><span class="fu">colnames</span>(DA_wilcox_CLR)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Statistic&quot;                         &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Rank)\nAA_vs_BB&quot;  
## [13] &quot;Mean Rank Abundance\nAA&quot;           &quot;Mean Rank Abundance\nBB&quot;           &quot;Occurrence (100%)\n(All)&quot;         
## [16] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<p>The output is the same as the previous results of ordinary pattern.</p>
<details>
<summary>
<strong>other options for wilcox_CLR</strong>
</summary>
<div class="sourceCode" id="cb307"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb307-1"><a href="differential-analysis.html#cb307-1" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb307-2"><a href="differential-analysis.html#cb307-2" tabindex="-1"></a>DA_wilcox_CLR <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb307-3"><a href="differential-analysis.html#cb307-3" tabindex="-1"></a>                    <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb307-4"><a href="differential-analysis.html#cb307-4" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb307-5"><a href="differential-analysis.html#cb307-5" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb307-6"><a href="differential-analysis.html#cb307-6" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;wilcox&quot;</span>,</span>
<span id="cb307-7"><a href="differential-analysis.html#cb307-7" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CLR&quot;</span>)</span>
<span id="cb307-8"><a href="differential-analysis.html#cb307-8" tabindex="-1"></a></span>
<span id="cb307-9"><a href="differential-analysis.html#cb307-9" tabindex="-1"></a><span class="fu">head</span>(DA_wilcox_CLR)</span>
<span id="cb307-10"><a href="differential-analysis.html#cb307-10" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="liner-discriminant-analysis-lda-effect-size-lefse" class="section level3 hasAnchor" number="10.3.7">
<h3><span class="header-section-number">10.3.7</span> Liner discriminant analysis (LDA) effect size (LEfSe)<a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>LEfSe</strong> method is from <em>Metagenomic biomarker discovery and explanation</em> <span class="citation">(<a href="#ref-segata2011metagenomic">Segata et al. 2011</a>)</span>. It uses Liner discriminant analysis model to identify Differential Taxa.</p>
<ul>
<li><em><code>run_lefse</code> function was depcrecated</em></li>
</ul>
<div class="sourceCode" id="cb308"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb308-1"><a href="differential-analysis.html#cb308-1" tabindex="-1"></a><span class="co"># DA_lefse &lt;- run_lefse(</span></span>
<span id="cb308-2"><a href="differential-analysis.html#cb308-2" tabindex="-1"></a><span class="co">#                     ps = dada2_ps_genus_filter_trim,</span></span>
<span id="cb308-3"><a href="differential-analysis.html#cb308-3" tabindex="-1"></a><span class="co">#                     group = &quot;Group&quot;,</span></span>
<span id="cb308-4"><a href="differential-analysis.html#cb308-4" tabindex="-1"></a><span class="co">#                     group_names = c(&quot;AA&quot;, &quot;BB&quot;),</span></span>
<span id="cb308-5"><a href="differential-analysis.html#cb308-5" tabindex="-1"></a><span class="co">#                     norm = &quot;CPM&quot;,</span></span>
<span id="cb308-6"><a href="differential-analysis.html#cb308-6" tabindex="-1"></a><span class="co">#                     Lda = 2)</span></span>
<span id="cb308-7"><a href="differential-analysis.html#cb308-7" tabindex="-1"></a></span>
<span id="cb308-8"><a href="differential-analysis.html#cb308-8" tabindex="-1"></a>DA_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse2</span>(</span>
<span id="cb308-9"><a href="differential-analysis.html#cb308-9" tabindex="-1"></a>                    <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb308-10"><a href="differential-analysis.html#cb308-10" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb308-11"><a href="differential-analysis.html#cb308-11" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb308-12"><a href="differential-analysis.html#cb308-12" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CPM&quot;</span>,</span>
<span id="cb308-13"><a href="differential-analysis.html#cb308-13" tabindex="-1"></a>                    <span class="at">lda_cutoff =</span> <span class="dv">2</span>)</span>
<span id="cb308-14"><a href="differential-analysis.html#cb308-14" tabindex="-1"></a></span>
<span id="cb308-15"><a href="differential-analysis.html#cb308-15" tabindex="-1"></a><span class="fu">colnames</span>(DA_lefse)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;LDA_Score&quot;                        
##  [4] &quot;Enrichment&quot;                        &quot;EffectSize&quot;                        &quot;Pvalue&quot;                           
##  [7] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [10] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [13] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [16] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb310"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb310-1"><a href="differential-analysis.html#cb310-1" tabindex="-1"></a><span class="fu">head</span>(DA_lefse)</span></code></pre></div>
<pre><code>##                           TaxaID         Block LDA_Score Enrichment EffectSize      Pvalue Log2FoldChange (Median)\nAA_vs_BB
## 1 g__Clostridium_sensu_stricto_1 9_AA vs 14_BB  3.919548         BB   2.393458 0.005225385                                NA
## 2             g__Intestinibacter 9_AA vs 14_BB  3.359688         BB   2.095183 0.007746971                                NA
## 3               g__Lactobacillus 9_AA vs 14_BB  4.369476         BB   2.351692 0.037446557                         -3.335507
## 4                 g__Odoribacter 9_AA vs 14_BB -3.033350         AA   1.743137 0.032745426                                NA
## 5              g__Parasutterella 9_AA vs 14_BB -3.911302         AA   2.091653 0.036513630                          4.400349
## 6                  g__Romboutsia 9_AA vs 14_BB  3.953705         BB   2.771988 0.007839137                         -6.859316
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All)
## 1               225.71974              0.00000           1177.85428                       -4.745276             5870.1925
## 2               383.70882              0.00000           1402.64816                       -1.963798             1886.7924
## 3              1286.16050            521.13869           5260.66779                       -5.172830            19970.3868
## 4                21.95438            605.54943              0.00000                        2.099245              579.0073
## 5               173.39662           1081.33826             51.20642                        4.481898             3909.0175
## 6              2195.63745             50.38418           5849.97508                       -3.333827             7104.3451
##   Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1           351.1518          9418.1472                    60.87                 22.22                 85.71    4900 (4900;4900)
## 2           682.1993          2661.1738                    60.87                 22.22                 85.71         3.4 (5.8;1)
## 3           893.5890         32234.0425                    86.96                 77.78                 92.86 1e+08 (1e+08;1e+08)
## 4          1085.5805           253.3531                    52.17                 77.78                 35.71       0.31 (-2;2.6)
## 5          9339.5437           417.9649                    65.22                 88.89                 50.00     0.0025 (-12;12)
## 6          1088.1764         10971.8822                    78.26                 55.56                 92.86          76 (84;67)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>LDA score</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <em>lefse</em>;</p></li>
<li><p><strong>LDA_Score</strong>: LDA score;</p></li>
<li><p><strong>Pvalue</strong>: significant level of Pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for lefse</strong>
</summary>
<div class="sourceCode" id="cb312"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb312-1"><a href="differential-analysis.html#cb312-1" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb312-2"><a href="differential-analysis.html#cb312-2" tabindex="-1"></a><span class="co"># DA_lefse &lt;- run_lefse(</span></span>
<span id="cb312-3"><a href="differential-analysis.html#cb312-3" tabindex="-1"></a><span class="co">#                 data_otu = phyloseq::otu_table(dada2_ps_genus_filter_trim),</span></span>
<span id="cb312-4"><a href="differential-analysis.html#cb312-4" tabindex="-1"></a><span class="co">#                 data_sam = phyloseq::sample_data(dada2_ps_genus_filter_trim),</span></span>
<span id="cb312-5"><a href="differential-analysis.html#cb312-5" tabindex="-1"></a><span class="co">#                 group = &quot;Group&quot;,</span></span>
<span id="cb312-6"><a href="differential-analysis.html#cb312-6" tabindex="-1"></a><span class="co">#                 group_names = c(&quot;AA&quot;, &quot;BB&quot;),</span></span>
<span id="cb312-7"><a href="differential-analysis.html#cb312-7" tabindex="-1"></a><span class="co">#                 norm = &quot;CPM&quot;,</span></span>
<span id="cb312-8"><a href="differential-analysis.html#cb312-8" tabindex="-1"></a><span class="co">#                 Lda = 0)</span></span>
<span id="cb312-9"><a href="differential-analysis.html#cb312-9" tabindex="-1"></a></span>
<span id="cb312-10"><a href="differential-analysis.html#cb312-10" tabindex="-1"></a>DA_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse2</span>(</span>
<span id="cb312-11"><a href="differential-analysis.html#cb312-11" tabindex="-1"></a>                <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb312-12"><a href="differential-analysis.html#cb312-12" tabindex="-1"></a>                <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb312-13"><a href="differential-analysis.html#cb312-13" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb312-14"><a href="differential-analysis.html#cb312-14" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb312-15"><a href="differential-analysis.html#cb312-15" tabindex="-1"></a>                <span class="at">norm =</span> <span class="st">&quot;CPM&quot;</span>,</span>
<span id="cb312-16"><a href="differential-analysis.html#cb312-16" tabindex="-1"></a>                <span class="at">lda_cutoff =</span> <span class="dv">0</span>)</span>
<span id="cb312-17"><a href="differential-analysis.html#cb312-17" tabindex="-1"></a></span>
<span id="cb312-18"><a href="differential-analysis.html#cb312-18" tabindex="-1"></a>DA_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse2</span>(</span>
<span id="cb312-19"><a href="differential-analysis.html#cb312-19" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps,</span>
<span id="cb312-20"><a href="differential-analysis.html#cb312-20" tabindex="-1"></a>                <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb312-21"><a href="differential-analysis.html#cb312-21" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb312-22"><a href="differential-analysis.html#cb312-22" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb312-23"><a href="differential-analysis.html#cb312-23" tabindex="-1"></a>                <span class="at">norm =</span> <span class="st">&quot;CPM&quot;</span>,</span>
<span id="cb312-24"><a href="differential-analysis.html#cb312-24" tabindex="-1"></a>                <span class="at">lda_cutoff =</span> <span class="dv">0</span>)</span>
<span id="cb312-25"><a href="differential-analysis.html#cb312-25" tabindex="-1"></a></span>
<span id="cb312-26"><a href="differential-analysis.html#cb312-26" tabindex="-1"></a>DA_lefse <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb312-27"><a href="differential-analysis.html#cb312-27" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb312-28"><a href="differential-analysis.html#cb312-28" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb312-29"><a href="differential-analysis.html#cb312-29" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb312-30"><a href="differential-analysis.html#cb312-30" tabindex="-1"></a>                <span class="at">da_method =</span> <span class="st">&quot;lefse&quot;</span>,</span>
<span id="cb312-31"><a href="differential-analysis.html#cb312-31" tabindex="-1"></a>                <span class="at">norm =</span> <span class="st">&quot;CPM&quot;</span>,</span>
<span id="cb312-32"><a href="differential-analysis.html#cb312-32" tabindex="-1"></a>                <span class="at">lda_cutoff =</span> <span class="dv">0</span>)</span>
<span id="cb312-33"><a href="differential-analysis.html#cb312-33" tabindex="-1"></a></span>
<span id="cb312-34"><a href="differential-analysis.html#cb312-34" tabindex="-1"></a><span class="fu">head</span>(DA_lefse)</span>
<span id="cb312-35"><a href="differential-analysis.html#cb312-35" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="t-test" class="section level3 hasAnchor" number="10.3.8">
<h3><span class="header-section-number">10.3.8</span> t-test<a href="differential-analysis.html#t-test" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>T test, a parametric test method, identifies the significant taxa.</p>
<ul>
<li>Ordinary pattern</li>
</ul>
<div class="sourceCode" id="cb313"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb313-1"><a href="differential-analysis.html#cb313-1" tabindex="-1"></a>DA_ttest <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb313-2"><a href="differential-analysis.html#cb313-2" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb313-3"><a href="differential-analysis.html#cb313-3" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb313-4"><a href="differential-analysis.html#cb313-4" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb313-5"><a href="differential-analysis.html#cb313-5" tabindex="-1"></a></span>
<span id="cb313-6"><a href="differential-analysis.html#cb313-6" tabindex="-1"></a><span class="fu">colnames</span>(DA_ttest)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Statistic&quot;                         &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb315"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb315-1"><a href="differential-analysis.html#cb315-1" tabindex="-1"></a><span class="fu">head</span>(DA_ttest)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment EffectSize  Statistic     Pvalue AdjustedPvalue Log2FoldChange (Median)\nAA_vs_BB
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif   70.56349  0.6845254 0.51170983      0.7198630                                NA
## 2     g__Actinomyces 9_AA vs 14_BB  Nonsignif   39.65079 -1.9079866 0.07478702      0.5997144                        -0.9577718
## 3   g__Adlercreutzia 9_AA vs 14_BB  Nonsignif   23.17460 -2.1601919 0.04743155      0.5997144                                NA
## 4    g__Agathobacter 9_AA vs 14_BB  Nonsignif 1222.58730  1.4034718 0.19417159      0.6194454                         0.1524904
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif  104.78571 -0.7632086 0.45788140      0.7000410                                NA
## 6       g__Alistipes 9_AA vs 14_BB  Nonsignif   23.03175  0.2691871 0.79125424      0.9043946                        -0.1085245
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All)
## 1                       0                    0                  0.0                       1.7051442              58.82609
## 2                      30                   26                 50.5                      -1.3107676              50.91304
## 3                       0                    0                  9.0                      -2.4683647              19.21739
## 4                     316                  329                296.0                       1.7488297             996.26087
## 5                       0                    0                  0.0                      -2.1676332              93.78261
## 6                      64                   64                 69.0                       0.2000385             163.86957
##   Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1         101.777778           31.21429                    21.74                 22.22                 21.43    0.67 (-0.11;1.5)
## 2          26.777778           66.42857                    82.61                 88.89                 78.57       4.1 (6.8;1.3)
## 3           5.111111           28.28571                    43.48                 33.33                 50.00        8.6 (13;4.3)
## 4        1740.444444          517.85714                    65.22                 55.56                 71.43     0.39 (-1.4;2.2)
## 5          30.000000          134.78571                    21.74                 22.22                 21.43      1.5 (2.3;0.71)
## 6         177.888889          154.85714                    73.91                 77.78                 71.43     0.88 (0.64;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size of taxa by <em>glm</em> function to assess the pvalue effect;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (GeometricMean)_vs_BB**: Log2FoldChange (GeometricMean Abundance) between group AA and group BB;</p></li>
<li><p><strong>GeometricMean Abundance (All)/(group AA)/(group BB)</strong>: GeometricMean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for t-test</strong>
</summary>
<div class="sourceCode" id="cb317"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb317-1"><a href="differential-analysis.html#cb317-1" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb317-2"><a href="differential-analysis.html#cb317-2" tabindex="-1"></a>DA_ttest <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb317-3"><a href="differential-analysis.html#cb317-3" tabindex="-1"></a>                <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb317-4"><a href="differential-analysis.html#cb317-4" tabindex="-1"></a>                <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb317-5"><a href="differential-analysis.html#cb317-5" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb317-6"><a href="differential-analysis.html#cb317-6" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb317-7"><a href="differential-analysis.html#cb317-7" tabindex="-1"></a></span>
<span id="cb317-8"><a href="differential-analysis.html#cb317-8" tabindex="-1"></a>DA_ttest <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb317-9"><a href="differential-analysis.html#cb317-9" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps,</span>
<span id="cb317-10"><a href="differential-analysis.html#cb317-10" tabindex="-1"></a>                <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb317-11"><a href="differential-analysis.html#cb317-11" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb317-12"><a href="differential-analysis.html#cb317-12" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb317-13"><a href="differential-analysis.html#cb317-13" tabindex="-1"></a></span>
<span id="cb317-14"><a href="differential-analysis.html#cb317-14" tabindex="-1"></a>DA_ttest <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb317-15"><a href="differential-analysis.html#cb317-15" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb317-16"><a href="differential-analysis.html#cb317-16" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb317-17"><a href="differential-analysis.html#cb317-17" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb317-18"><a href="differential-analysis.html#cb317-18" tabindex="-1"></a>                <span class="at">da_method =</span> <span class="st">&quot;ttest&quot;</span>)</span>
<span id="cb317-19"><a href="differential-analysis.html#cb317-19" tabindex="-1"></a></span>
<span id="cb317-20"><a href="differential-analysis.html#cb317-20" tabindex="-1"></a><span class="fu">head</span>(DA_ttest)</span>
<span id="cb317-21"><a href="differential-analysis.html#cb317-21" tabindex="-1"></a>}</span></code></pre></div>
</details>
<ul>
<li><strong>t-test_rarefy</strong>: random subsampling counts to the smallest library size in the data set.</li>
</ul>
<div class="sourceCode" id="cb318"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb318-1"><a href="differential-analysis.html#cb318-1" tabindex="-1"></a>DA_ttest_rarefy <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb318-2"><a href="differential-analysis.html#cb318-2" tabindex="-1"></a>                      <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb318-3"><a href="differential-analysis.html#cb318-3" tabindex="-1"></a>                      <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb318-4"><a href="differential-analysis.html#cb318-4" tabindex="-1"></a>                      <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb318-5"><a href="differential-analysis.html#cb318-5" tabindex="-1"></a>                      <span class="at">norm =</span> <span class="st">&quot;rarefy&quot;</span>)</span>
<span id="cb318-6"><a href="differential-analysis.html#cb318-6" tabindex="-1"></a><span class="fu">colnames</span>(DA_ttest_rarefy)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Statistic&quot;                         &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<p>The output is the same as the previous results of ordinary pattern.</p>
<details>
<summary>
<strong>other options for ttest_rarefy</strong>
</summary>
<div class="sourceCode" id="cb320"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb320-1"><a href="differential-analysis.html#cb320-1" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb320-2"><a href="differential-analysis.html#cb320-2" tabindex="-1"></a>DA_ttest_rarefy <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb320-3"><a href="differential-analysis.html#cb320-3" tabindex="-1"></a>                      <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb320-4"><a href="differential-analysis.html#cb320-4" tabindex="-1"></a>                      <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb320-5"><a href="differential-analysis.html#cb320-5" tabindex="-1"></a>                      <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb320-6"><a href="differential-analysis.html#cb320-6" tabindex="-1"></a>                      <span class="at">da_method =</span> <span class="st">&quot;ttest&quot;</span>,</span>
<span id="cb320-7"><a href="differential-analysis.html#cb320-7" tabindex="-1"></a>                      <span class="at">norm =</span> <span class="st">&quot;rarefy&quot;</span>)</span>
<span id="cb320-8"><a href="differential-analysis.html#cb320-8" tabindex="-1"></a></span>
<span id="cb320-9"><a href="differential-analysis.html#cb320-9" tabindex="-1"></a><span class="fu">head</span>(DA_ttest_rarefy)</span>
<span id="cb320-10"><a href="differential-analysis.html#cb320-10" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="metagenomeseq" class="section level3 hasAnchor" number="10.3.9">
<h3><span class="header-section-number">10.3.9</span> MetagenomeSeq<a href="differential-analysis.html#metagenomeseq" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>MetagenomeSeq</strong> package is from <em>Differential abundance analysis for microbial marker-gene surveys</em> <span class="citation">(<a href="#ref-paulson2013differential">Paulson et al. 2013</a>)</span>. It uses zero-inflated Log-Normal mixture model or Zero-inflated Gaussian mixture model to identify the significant taxa between groups. (Caution: the <strong>otu_table</strong> should be integers).</p>
<div class="sourceCode" id="cb321"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb321-1"><a href="differential-analysis.html#cb321-1" tabindex="-1"></a>DA_metagenomeseq <span class="ot">&lt;-</span> <span class="fu">run_metagenomeseq</span>(</span>
<span id="cb321-2"><a href="differential-analysis.html#cb321-2" tabindex="-1"></a>                        <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb321-3"><a href="differential-analysis.html#cb321-3" tabindex="-1"></a>                        <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb321-4"><a href="differential-analysis.html#cb321-4" tabindex="-1"></a>                        <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb321-5"><a href="differential-analysis.html#cb321-5" tabindex="-1"></a>                        <span class="at">norm =</span> <span class="st">&quot;CSS&quot;</span>,</span>
<span id="cb321-6"><a href="differential-analysis.html#cb321-6" tabindex="-1"></a>                        <span class="at">method =</span> <span class="st">&quot;ZILN&quot;</span>)</span>
<span id="cb321-7"><a href="differential-analysis.html#cb321-7" tabindex="-1"></a></span>
<span id="cb321-8"><a href="differential-analysis.html#cb321-8" tabindex="-1"></a><span class="fu">colnames</span>(DA_metagenomeseq)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;logFC&quot;                             &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                   
##  [7] &quot;se&quot;                                &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb323"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb323-1"><a href="differential-analysis.html#cb323-1" tabindex="-1"></a><span class="fu">head</span>(DA_metagenomeseq)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment      logFC Pvalue AdjustedPvalue se Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All)
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif -1.1859956     NA             NA NA                                NA                 0.00000
## 2     g__Actinomyces 9_AA vs 14_BB       &lt;NA&gt;  0.5530981     NA             NA NA                        -1.0127644                31.76044
## 3   g__Adlercreutzia 9_AA vs 14_BB       &lt;NA&gt;  0.6655629     NA             NA NA                                NA                 0.00000
## 4    g__Agathobacter 9_AA vs 14_BB       &lt;NA&gt; -0.1747794     NA             NA NA                        -0.1568579               246.20874
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif -0.9353889     NA             NA NA                                NA                 0.00000
## 6       g__Alistipes 9_AA vs 14_BB       &lt;NA&gt;  0.2351433     NA             NA NA                         1.1457419                72.72727
##   Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB
## 1              0.00000             0.000000                       3.5123295             149.16614         335.470085           29.39932
## 2             22.73804            45.880218                      -0.5319720              60.23329          47.374821           68.49945
## 3              0.00000             7.142857                      -2.6476104              20.23937           4.812521           30.15663
## 4            222.89973           248.501196                       1.1036073             708.68746        1050.589421          488.89334
## 5              0.00000             0.000000                      -2.6539257              62.16641          14.723359           92.66551
## 6            136.75214            61.806114                       0.1758504             153.73588         165.280832          146.31413
##   Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                    21.74                 22.22                 21.43    0.48 (-0.98;1.9)
## 2                    82.61                 88.89                 78.57      1.4 (2.1;0.73)
## 3                    43.48                 33.33                 50.00          8.1 (12;4)
## 4                    65.22                 55.56                 71.43     0.55 (-0.6;1.7)
## 5                    21.74                 22.22                 21.43      1.6 (2.6;0.67)
## 6                    73.91                 77.78                 71.43      0.9 (0.69;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>logFC</strong>: fitted coefficient represents the fold-change for group AA and group BB;</p></li>
<li><p><strong>se</strong>: standard error;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for metagenomeseq</strong>
</summary>
<div class="sourceCode" id="cb325"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb325-1"><a href="differential-analysis.html#cb325-1" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb325-2"><a href="differential-analysis.html#cb325-2" tabindex="-1"></a>DA_metagenomeseq <span class="ot">&lt;-</span> <span class="fu">run_metagenomeseq</span>(</span>
<span id="cb325-3"><a href="differential-analysis.html#cb325-3" tabindex="-1"></a>                        <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb325-4"><a href="differential-analysis.html#cb325-4" tabindex="-1"></a>                        <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb325-5"><a href="differential-analysis.html#cb325-5" tabindex="-1"></a>                        <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb325-6"><a href="differential-analysis.html#cb325-6" tabindex="-1"></a>                        <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb325-7"><a href="differential-analysis.html#cb325-7" tabindex="-1"></a>                        <span class="at">norm =</span> <span class="st">&quot;CSS&quot;</span>,</span>
<span id="cb325-8"><a href="differential-analysis.html#cb325-8" tabindex="-1"></a>                        <span class="at">method =</span> <span class="st">&quot;ZILN&quot;</span>)</span>
<span id="cb325-9"><a href="differential-analysis.html#cb325-9" tabindex="-1"></a></span>
<span id="cb325-10"><a href="differential-analysis.html#cb325-10" tabindex="-1"></a>DA_metagenomeseq <span class="ot">&lt;-</span> <span class="fu">run_metagenomeseq</span>(</span>
<span id="cb325-11"><a href="differential-analysis.html#cb325-11" tabindex="-1"></a>                        <span class="at">ps =</span> dada2_ps,</span>
<span id="cb325-12"><a href="differential-analysis.html#cb325-12" tabindex="-1"></a>                        <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb325-13"><a href="differential-analysis.html#cb325-13" tabindex="-1"></a>                        <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb325-14"><a href="differential-analysis.html#cb325-14" tabindex="-1"></a>                        <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb325-15"><a href="differential-analysis.html#cb325-15" tabindex="-1"></a>                        <span class="at">norm =</span> <span class="st">&quot;CSS&quot;</span>,</span>
<span id="cb325-16"><a href="differential-analysis.html#cb325-16" tabindex="-1"></a>                        <span class="at">method =</span> <span class="st">&quot;ZILN&quot;</span>)</span>
<span id="cb325-17"><a href="differential-analysis.html#cb325-17" tabindex="-1"></a></span>
<span id="cb325-18"><a href="differential-analysis.html#cb325-18" tabindex="-1"></a>DA_metagenomeseq<span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb325-19"><a href="differential-analysis.html#cb325-19" tabindex="-1"></a>                      <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb325-20"><a href="differential-analysis.html#cb325-20" tabindex="-1"></a>                      <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb325-21"><a href="differential-analysis.html#cb325-21" tabindex="-1"></a>                      <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb325-22"><a href="differential-analysis.html#cb325-22" tabindex="-1"></a>                      <span class="at">da_method =</span> <span class="st">&quot;metagenomeseq&quot;</span>,</span>
<span id="cb325-23"><a href="differential-analysis.html#cb325-23" tabindex="-1"></a>                      <span class="at">norm =</span> <span class="st">&quot;CSS&quot;</span>,</span>
<span id="cb325-24"><a href="differential-analysis.html#cb325-24" tabindex="-1"></a>                      <span class="at">method =</span> <span class="st">&quot;ZILN&quot;</span>)</span>
<span id="cb325-25"><a href="differential-analysis.html#cb325-25" tabindex="-1"></a></span>
<span id="cb325-26"><a href="differential-analysis.html#cb325-26" tabindex="-1"></a><span class="fu">head</span>(DA_metagenomeseq)</span>
<span id="cb325-27"><a href="differential-analysis.html#cb325-27" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="deseq2" class="section level3 hasAnchor" number="10.3.10">
<h3><span class="header-section-number">10.3.10</span> DESeq2<a href="differential-analysis.html#deseq2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>DESeq2</strong> package is from <em>Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2</em> <span class="citation">(<a href="#ref-love2014moderated">Love, Huber, and Anders 2014</a>)</span>. Differential expression analysis based on the Negative Binomial (a.k.a. Gamma-Poisson) distribution.(Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb326"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb326-1"><a href="differential-analysis.html#cb326-1" tabindex="-1"></a>DA_deseq2 <span class="ot">&lt;-</span> <span class="fu">run_deseq2</span>(</span>
<span id="cb326-2"><a href="differential-analysis.html#cb326-2" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb326-3"><a href="differential-analysis.html#cb326-3" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb326-4"><a href="differential-analysis.html#cb326-4" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb326-5"><a href="differential-analysis.html#cb326-5" tabindex="-1"></a></span>
<span id="cb326-6"><a href="differential-analysis.html#cb326-6" tabindex="-1"></a><span class="fu">colnames</span>(DA_deseq2)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                    &quot;logFC&quot;                            
##  [7] &quot;Statistic&quot;                         &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb328"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb328-1"><a href="differential-analysis.html#cb328-1" tabindex="-1"></a><span class="fu">head</span>(DA_deseq2)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment      Pvalue AdjustedPvalue      logFC  Statistic Log2FoldChange (Median)\nAA_vs_BB
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif 0.009874105     0.04595673 -2.7857993 -2.5802072                                NA
## 2     g__Actinomyces 9_AA vs 14_BB  Nonsignif 0.170034415     0.36386636 -1.3037457 -1.3720932                        -0.9577718
## 3   g__Adlercreutzia 9_AA vs 14_BB         BB 0.010212606     0.04595673 -2.4201226 -2.5685463                                NA
## 4    g__Agathobacter 9_AA vs 14_BB  Nonsignif 0.394299794     0.57032649  0.9416658  0.8518456                         0.1524904
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif 0.416991678             NA  0.5802090  0.8116514                                NA
## 6       g__Alistipes 9_AA vs 14_BB  Nonsignif 0.539361523     0.68948490  0.5812285  0.6137788                        -0.1085245
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All)
## 1                       0                    0                  0.0                       1.7051442              58.82609
## 2                      30                   26                 50.5                      -1.3107676              50.91304
## 3                       0                    0                  9.0                      -2.4683647              19.21739
## 4                     316                  329                296.0                       1.7488297             996.26087
## 5                       0                    0                  0.0                      -2.1676332              93.78261
## 6                      64                   64                 69.0                       0.2000385             163.86957
##   Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1         101.777778           31.21429                    21.74                 22.22                 21.43    0.67 (-0.11;1.5)
## 2          26.777778           66.42857                    82.61                 88.89                 78.57       4.1 (6.8;1.3)
## 3           5.111111           28.28571                    43.48                 33.33                 50.00        8.6 (13;4.3)
## 4        1740.444444          517.85714                    65.22                 55.56                 71.43     0.39 (-1.4;2.2)
## 5          30.000000          134.78571                    21.74                 22.22                 21.43      1.5 (2.3;0.71)
## 6         177.888889          154.85714                    73.91                 77.78                 71.43     0.88 (0.64;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>logFC</strong>: fitted coefficient represents the fold-change for group AA and group BB;</p></li>
<li><p><strong>Statistic</strong>: test statistic (negative binomial model);</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for deseq2</strong>
</summary>
<div class="sourceCode" id="cb330"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb330-1"><a href="differential-analysis.html#cb330-1" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb330-2"><a href="differential-analysis.html#cb330-2" tabindex="-1"></a>DA_deseq2 <span class="ot">&lt;-</span> <span class="fu">run_deseq2</span>(</span>
<span id="cb330-3"><a href="differential-analysis.html#cb330-3" tabindex="-1"></a>                <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb330-4"><a href="differential-analysis.html#cb330-4" tabindex="-1"></a>                <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb330-5"><a href="differential-analysis.html#cb330-5" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb330-6"><a href="differential-analysis.html#cb330-6" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb330-7"><a href="differential-analysis.html#cb330-7" tabindex="-1"></a></span>
<span id="cb330-8"><a href="differential-analysis.html#cb330-8" tabindex="-1"></a>DA_deseq2 <span class="ot">&lt;-</span> <span class="fu">run_deseq2</span>(</span>
<span id="cb330-9"><a href="differential-analysis.html#cb330-9" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps,</span>
<span id="cb330-10"><a href="differential-analysis.html#cb330-10" tabindex="-1"></a>                <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb330-11"><a href="differential-analysis.html#cb330-11" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb330-12"><a href="differential-analysis.html#cb330-12" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb330-13"><a href="differential-analysis.html#cb330-13" tabindex="-1"></a></span>
<span id="cb330-14"><a href="differential-analysis.html#cb330-14" tabindex="-1"></a>DA_deseq2 <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb330-15"><a href="differential-analysis.html#cb330-15" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb330-16"><a href="differential-analysis.html#cb330-16" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb330-17"><a href="differential-analysis.html#cb330-17" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb330-18"><a href="differential-analysis.html#cb330-18" tabindex="-1"></a>                <span class="at">da_method =</span> <span class="st">&quot;deseq2&quot;</span>)</span>
<span id="cb330-19"><a href="differential-analysis.html#cb330-19" tabindex="-1"></a></span>
<span id="cb330-20"><a href="differential-analysis.html#cb330-20" tabindex="-1"></a><span class="fu">head</span>(DA_deseq2)</span>
<span id="cb330-21"><a href="differential-analysis.html#cb330-21" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="edger" class="section level3 hasAnchor" number="10.3.11">
<h3><span class="header-section-number">10.3.11</span> EdgeR<a href="differential-analysis.html#edger" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>EdgeR</strong> package is from <em>A scaling normalization method for differential expression analysis of RNA-seq data</em> <span class="citation">(<a href="#ref-robinson2010scaling">Robinson and Oshlack 2010</a>)</span>. Differential expression analysis based on the Negative Binomial (a.k.a. Gamma-Poisson) distribution. (Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb331"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb331-1"><a href="differential-analysis.html#cb331-1" tabindex="-1"></a>DA_edger <span class="ot">&lt;-</span> <span class="fu">run_edger</span>(</span>
<span id="cb331-2"><a href="differential-analysis.html#cb331-2" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb331-3"><a href="differential-analysis.html#cb331-3" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb331-4"><a href="differential-analysis.html#cb331-4" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-171"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-171-1.png" alt="EdgeR (BVC distance)" width="480" />
<p class="caption">
Figure 10.1: EdgeR (BVC distance)
</p>
</div>
<div class="sourceCode" id="cb332"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb332-1"><a href="differential-analysis.html#cb332-1" tabindex="-1"></a><span class="fu">colnames</span>(DA_edger)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;Pvalue&quot;                            &quot;AdjustedPvalue&quot;                    &quot;logFC&quot;                            
##  [7] &quot;logCPM&quot;                            &quot;LR&quot;                                &quot;Log2FoldChange (Median)\nAA_vs_BB&quot;
## [10] &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;             
## [13] &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;               
## [16] &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;            
## [19] &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb334"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb334-1"><a href="differential-analysis.html#cb334-1" tabindex="-1"></a><span class="fu">head</span>(DA_edger)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment      Pvalue AdjustedPvalue      logFC    logCPM        LR Log2FoldChange (Median)\nAA_vs_BB
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif 0.006331806     0.06339772 -3.6537692 11.624194 7.4533622                                NA
## 2     g__Actinomyces 9_AA vs 14_BB  Nonsignif 0.068349670     0.19562147  1.8820292 11.004313 3.3222153                        -0.9577718
## 3   g__Adlercreutzia 9_AA vs 14_BB  Nonsignif 0.018855166     0.09205757  2.3817310  8.875209 5.5148986                                NA
## 4    g__Agathobacter 9_AA vs 14_BB  Nonsignif 0.299740178     0.51830072 -1.1273701 13.563968 1.0753498                         0.1524904
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif 0.062219098     0.19562147  2.5474093 10.225497 3.4772223                                NA
## 6       g__Alistipes 9_AA vs 14_BB  Nonsignif 0.603600741     0.79522002 -0.4839035 11.355143 0.2695992                        -0.1085245
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All)
## 1                       0                    0                  0.0                       1.7051442              58.82609
## 2                      30                   26                 50.5                      -1.3107676              50.91304
## 3                       0                    0                  9.0                      -2.4683647              19.21739
## 4                     316                  329                296.0                       1.7488297             996.26087
## 5                       0                    0                  0.0                      -2.1676332              93.78261
## 6                      64                   64                 69.0                       0.2000385             163.86957
##   Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1         101.777778           31.21429                    21.74                 22.22                 21.43    0.67 (-0.11;1.5)
## 2          26.777778           66.42857                    82.61                 88.89                 78.57       4.1 (6.8;1.3)
## 3           5.111111           28.28571                    43.48                 33.33                 50.00        8.6 (13;4.3)
## 4        1740.444444          517.85714                    65.22                 55.56                 71.43     0.39 (-1.4;2.2)
## 5          30.000000          134.78571                    21.74                 22.22                 21.43      1.5 (2.3;0.71)
## 6         177.888889          154.85714                    73.91                 77.78                 71.43     0.88 (0.64;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>logFC</strong>: fitted coefficient represents the fold-change for group AA and group BB;</p></li>
<li><p><strong>logCPM</strong>: is the average expression of all samples for that particular gene across all samples on the log-scale expressed in counts per million (cpm, as calculated by edgeR after normalization);</p></li>
<li><p><strong>LR</strong>: the signed likelihood ratio test statistic;</p></li>
<li><p><strong>Pvalue and AdjustedPvalue</strong>: significant level of Pvalue and Adjusted-pvalue;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for EdgeR</strong>
</summary>
<div class="sourceCode" id="cb336"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb336-1"><a href="differential-analysis.html#cb336-1" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb336-2"><a href="differential-analysis.html#cb336-2" tabindex="-1"></a>DA_edger <span class="ot">&lt;-</span> <span class="fu">run_edger</span>(</span>
<span id="cb336-3"><a href="differential-analysis.html#cb336-3" tabindex="-1"></a>                <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb336-4"><a href="differential-analysis.html#cb336-4" tabindex="-1"></a>                <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb336-5"><a href="differential-analysis.html#cb336-5" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb336-6"><a href="differential-analysis.html#cb336-6" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb336-7"><a href="differential-analysis.html#cb336-7" tabindex="-1"></a></span>
<span id="cb336-8"><a href="differential-analysis.html#cb336-8" tabindex="-1"></a>DA_edger <span class="ot">&lt;-</span> <span class="fu">run_edger</span>(</span>
<span id="cb336-9"><a href="differential-analysis.html#cb336-9" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps,</span>
<span id="cb336-10"><a href="differential-analysis.html#cb336-10" tabindex="-1"></a>                <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb336-11"><a href="differential-analysis.html#cb336-11" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb336-12"><a href="differential-analysis.html#cb336-12" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb336-13"><a href="differential-analysis.html#cb336-13" tabindex="-1"></a></span>
<span id="cb336-14"><a href="differential-analysis.html#cb336-14" tabindex="-1"></a>DA_edger <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb336-15"><a href="differential-analysis.html#cb336-15" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb336-16"><a href="differential-analysis.html#cb336-16" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb336-17"><a href="differential-analysis.html#cb336-17" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb336-18"><a href="differential-analysis.html#cb336-18" tabindex="-1"></a>                <span class="at">da_method =</span> <span class="st">&quot;edger&quot;</span>)</span>
<span id="cb336-19"><a href="differential-analysis.html#cb336-19" tabindex="-1"></a></span>
<span id="cb336-20"><a href="differential-analysis.html#cb336-20" tabindex="-1"></a><span class="fu">head</span>(DA_edger)</span>
<span id="cb336-21"><a href="differential-analysis.html#cb336-21" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="ancom" class="section level3 hasAnchor" number="10.3.12">
<h3><span class="header-section-number">10.3.12</span> ANCOM<a href="differential-analysis.html#ancom" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>ANCOM</strong> (Analysis of composition of microbiomes) is from <em>Analysis of composition of microbiomes: a novel method for studying microbial composition”, Microbial Ecology in Health</em> <span class="citation">(<a href="#ref-mandal2015analysis">Mandal et al. 2015</a>)</span>. ANCOM makes no distributional assumptions and can be implemented in a linear model framework. (Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb337"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb337-1"><a href="differential-analysis.html#cb337-1" tabindex="-1"></a>DA_ancom <span class="ot">&lt;-</span> <span class="fu">run_ancom</span>(</span>
<span id="cb337-2"><a href="differential-analysis.html#cb337-2" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb337-3"><a href="differential-analysis.html#cb337-3" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb337-4"><a href="differential-analysis.html#cb337-4" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-173"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-173-1.png" alt="ANCOM (Structure Zero)" width="480" />
<p class="caption">
Figure 10.2: ANCOM (Structure Zero)
</p>
</div>
<div class="sourceCode" id="cb338"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb338-1"><a href="differential-analysis.html#cb338-1" tabindex="-1"></a><span class="fu">colnames</span>(DA_ancom)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;(W)q-values &lt; alpha&quot;               &quot;W_ratio&quot;                          
##  [7] &quot;detected_0.7&quot;                      &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb340"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb340-1"><a href="differential-analysis.html#cb340-1" tabindex="-1"></a><span class="fu">head</span>(DA_ancom)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment  EffectSize (W)q-values &lt; alpha W_ratio detected_0.7 Log2FoldChange (Median)\nAA_vs_BB
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif  0.05598359                   0       0        FALSE                                NA
## 2     g__Actinomyces 9_AA vs 14_BB  Nonsignif  0.77737041                   0       0        FALSE                        -0.9577718
## 3   g__Adlercreutzia 9_AA vs 14_BB  Nonsignif  0.24671418                   0       0        FALSE                                NA
## 4    g__Agathobacter 9_AA vs 14_BB  Nonsignif -0.51435011                   0       0        FALSE                         0.1524904
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif -0.07133173                   0       0        FALSE                                NA
## 6       g__Alistipes 9_AA vs 14_BB  Nonsignif  0.09952748                   0       0        FALSE                        -0.1085245
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All)
## 1                       0                    0                  0.0                       1.7051442              58.82609
## 2                      30                   26                 50.5                      -1.3107676              50.91304
## 3                       0                    0                  9.0                      -2.4683647              19.21739
## 4                     316                  329                296.0                       1.7488297             996.26087
## 5                       0                    0                  0.0                      -2.1676332              93.78261
## 6                      64                   64                 69.0                       0.2000385             163.86957
##   Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1         101.777778           31.21429                    21.74                 22.22                 21.43    0.67 (-0.11;1.5)
## 2          26.777778           66.42857                    82.61                 88.89                 78.57       4.1 (6.8;1.3)
## 3           5.111111           28.28571                    43.48                 33.33                 50.00        8.6 (13;4.3)
## 4        1740.444444          517.85714                    65.22                 55.56                 71.43     0.39 (-1.4;2.2)
## 5          30.000000          134.78571                    21.74                 22.22                 21.43      1.5 (2.3;0.71)
## 6         177.888889          154.85714                    73.91                 77.78                 71.43     0.88 (0.64;1.1)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>Enrichment</strong>: enriched direction based on <em>Median Abundance</em> and <em>AdjustedPvalue</em>;</p></li>
<li><p><strong>EffectSize</strong>: effect size by <strong>ANCOM</strong>;</p></li>
<li><p><strong>(W)q-values &lt; alpha</strong>: q-values less than alpha;</p></li>
<li><p><strong>W_ratio</strong>: the ratio of W values;</p></li>
<li><p><strong>detected_0.7</strong>: W_ratio more than 0.7;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
<details>
<summary>
<strong>other options for ANCOM</strong>
</summary>
<div class="sourceCode" id="cb342"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb342-1"><a href="differential-analysis.html#cb342-1" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb342-2"><a href="differential-analysis.html#cb342-2" tabindex="-1"></a>DA_ancom <span class="ot">&lt;-</span> <span class="fu">run_ancom</span>(</span>
<span id="cb342-3"><a href="differential-analysis.html#cb342-3" tabindex="-1"></a>                <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb342-4"><a href="differential-analysis.html#cb342-4" tabindex="-1"></a>                <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb342-5"><a href="differential-analysis.html#cb342-5" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb342-6"><a href="differential-analysis.html#cb342-6" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb342-7"><a href="differential-analysis.html#cb342-7" tabindex="-1"></a></span>
<span id="cb342-8"><a href="differential-analysis.html#cb342-8" tabindex="-1"></a>DA_ancom <span class="ot">&lt;-</span> <span class="fu">run_ancom</span>(</span>
<span id="cb342-9"><a href="differential-analysis.html#cb342-9" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps,</span>
<span id="cb342-10"><a href="differential-analysis.html#cb342-10" tabindex="-1"></a>                <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb342-11"><a href="differential-analysis.html#cb342-11" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb342-12"><a href="differential-analysis.html#cb342-12" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb342-13"><a href="differential-analysis.html#cb342-13" tabindex="-1"></a></span>
<span id="cb342-14"><a href="differential-analysis.html#cb342-14" tabindex="-1"></a>DA_ancom <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb342-15"><a href="differential-analysis.html#cb342-15" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb342-16"><a href="differential-analysis.html#cb342-16" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb342-17"><a href="differential-analysis.html#cb342-17" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb342-18"><a href="differential-analysis.html#cb342-18" tabindex="-1"></a>                <span class="at">da_method =</span> <span class="st">&quot;ancom&quot;</span>)</span>
<span id="cb342-19"><a href="differential-analysis.html#cb342-19" tabindex="-1"></a></span>
<span id="cb342-20"><a href="differential-analysis.html#cb342-20" tabindex="-1"></a><span class="fu">head</span>(DA_ancom)</span>
<span id="cb342-21"><a href="differential-analysis.html#cb342-21" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="corncob" class="section level3 hasAnchor" number="10.3.13">
<h3><span class="header-section-number">10.3.13</span> Corncob<a href="differential-analysis.html#corncob" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Corncob</strong> package is from <em>Modeling microbial abundances and dysbiosis with beta-binomial regression”, Microbial Ecology in Health</em> <span class="citation">(<a href="#ref-martin2020modeling">Martin, Witten, and Willis 2020</a>)</span>. Corncob is based on beta-binomial regression. (Caution: the <strong>otu_table</strong> must be integers).</p>
<div class="sourceCode" id="cb343"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb343-1"><a href="differential-analysis.html#cb343-1" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb343-2"><a href="differential-analysis.html#cb343-2" tabindex="-1"></a>DA_corncob <span class="ot">&lt;-</span> <span class="fu">run_corncob</span>(</span>
<span id="cb343-3"><a href="differential-analysis.html#cb343-3" tabindex="-1"></a>                  <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb343-4"><a href="differential-analysis.html#cb343-4" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb343-5"><a href="differential-analysis.html#cb343-5" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb343-6"><a href="differential-analysis.html#cb343-6" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;Wald&quot;</span>)</span>
<span id="cb343-7"><a href="differential-analysis.html#cb343-7" tabindex="-1"></a></span>
<span id="cb343-8"><a href="differential-analysis.html#cb343-8" tabindex="-1"></a><span class="fu">colnames</span>(DA_ancom)</span>
<span id="cb343-9"><a href="differential-analysis.html#cb343-9" tabindex="-1"></a><span class="fu">head</span>(DA_ancom)</span>
<span id="cb343-10"><a href="differential-analysis.html#cb343-10" tabindex="-1"></a>}</span></code></pre></div>
<details>
<summary>
<strong>other options for Corncob</strong>
</summary>
<div class="sourceCode" id="cb344"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb344-1"><a href="differential-analysis.html#cb344-1" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb344-2"><a href="differential-analysis.html#cb344-2" tabindex="-1"></a>DA_corncob <span class="ot">&lt;-</span> <span class="fu">run_corncob</span>(</span>
<span id="cb344-3"><a href="differential-analysis.html#cb344-3" tabindex="-1"></a>                  <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb344-4"><a href="differential-analysis.html#cb344-4" tabindex="-1"></a>                  <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb344-5"><a href="differential-analysis.html#cb344-5" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb344-6"><a href="differential-analysis.html#cb344-6" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb344-7"><a href="differential-analysis.html#cb344-7" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;Wald&quot;</span>)</span>
<span id="cb344-8"><a href="differential-analysis.html#cb344-8" tabindex="-1"></a></span>
<span id="cb344-9"><a href="differential-analysis.html#cb344-9" tabindex="-1"></a>DA_corncob <span class="ot">&lt;-</span> <span class="fu">run_corncob</span>(</span>
<span id="cb344-10"><a href="differential-analysis.html#cb344-10" tabindex="-1"></a>                  <span class="at">ps =</span> dada2_ps,</span>
<span id="cb344-11"><a href="differential-analysis.html#cb344-11" tabindex="-1"></a>                  <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb344-12"><a href="differential-analysis.html#cb344-12" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb344-13"><a href="differential-analysis.html#cb344-13" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb344-14"><a href="differential-analysis.html#cb344-14" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;Wald&quot;</span>)</span>
<span id="cb344-15"><a href="differential-analysis.html#cb344-15" tabindex="-1"></a></span>
<span id="cb344-16"><a href="differential-analysis.html#cb344-16" tabindex="-1"></a>DA_corncob <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb344-17"><a href="differential-analysis.html#cb344-17" tabindex="-1"></a>                  <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb344-18"><a href="differential-analysis.html#cb344-18" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb344-19"><a href="differential-analysis.html#cb344-19" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb344-20"><a href="differential-analysis.html#cb344-20" tabindex="-1"></a>                  <span class="at">da_method =</span> <span class="st">&quot;corncob&quot;</span>,</span>
<span id="cb344-21"><a href="differential-analysis.html#cb344-21" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;Wald&quot;</span>)</span>
<span id="cb344-22"><a href="differential-analysis.html#cb344-22" tabindex="-1"></a></span>
<span id="cb344-23"><a href="differential-analysis.html#cb344-23" tabindex="-1"></a><span class="fu">head</span>(DA_corncob)</span>
<span id="cb344-24"><a href="differential-analysis.html#cb344-24" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="maaslin2-microbiome-multivariable-association-with-linear-models" class="section level3 hasAnchor" number="10.3.14">
<h3><span class="header-section-number">10.3.14</span> Maaslin2 (Microbiome Multivariable Association with Linear Models)<a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Maaslin2</strong> package is from <em>Multivariable association discovery in population-scale meta-omics studies</em> <span class="citation">(<a href="#ref-mallick2021multivariable">Mallick et al. 2021</a>)</span>. Maaslin2 relies on general linear models to accommodate most modern epidemiological study designs, including cross-sectional and longitudinal, along with a variety of filtering, normalization, and transform methods.</p>
<div class="sourceCode" id="cb345"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb345-1"><a href="differential-analysis.html#cb345-1" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb345-2"><a href="differential-analysis.html#cb345-2" tabindex="-1"></a>DA_maaslin2 <span class="ot">&lt;-</span> <span class="fu">run_maaslin2</span>(</span>
<span id="cb345-3"><a href="differential-analysis.html#cb345-3" tabindex="-1"></a>                  <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb345-4"><a href="differential-analysis.html#cb345-4" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb345-5"><a href="differential-analysis.html#cb345-5" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb345-6"><a href="differential-analysis.html#cb345-6" tabindex="-1"></a>                  <span class="at">transform =</span> <span class="st">&quot;LOG&quot;</span>,</span>
<span id="cb345-7"><a href="differential-analysis.html#cb345-7" tabindex="-1"></a>                  <span class="at">norm =</span> <span class="st">&quot;TMM&quot;</span>,</span>
<span id="cb345-8"><a href="differential-analysis.html#cb345-8" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb345-9"><a href="differential-analysis.html#cb345-9" tabindex="-1"></a>                  <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb345-10"><a href="differential-analysis.html#cb345-10" tabindex="-1"></a></span>
<span id="cb345-11"><a href="differential-analysis.html#cb345-11" tabindex="-1"></a>DA_maaslin2 <span class="ot">&lt;-</span> <span class="fu">run_maaslin2</span>(</span>
<span id="cb345-12"><a href="differential-analysis.html#cb345-12" tabindex="-1"></a>                  <span class="at">ps =</span> dada2_ps,</span>
<span id="cb345-13"><a href="differential-analysis.html#cb345-13" tabindex="-1"></a>                  <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>,</span>
<span id="cb345-14"><a href="differential-analysis.html#cb345-14" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb345-15"><a href="differential-analysis.html#cb345-15" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb345-16"><a href="differential-analysis.html#cb345-16" tabindex="-1"></a>                  <span class="at">transform =</span> <span class="st">&quot;LOG&quot;</span>,</span>
<span id="cb345-17"><a href="differential-analysis.html#cb345-17" tabindex="-1"></a>                  <span class="at">norm =</span> <span class="st">&quot;TMM&quot;</span>,</span>
<span id="cb345-18"><a href="differential-analysis.html#cb345-18" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb345-19"><a href="differential-analysis.html#cb345-19" tabindex="-1"></a>                  <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb345-20"><a href="differential-analysis.html#cb345-20" tabindex="-1"></a></span>
<span id="cb345-21"><a href="differential-analysis.html#cb345-21" tabindex="-1"></a>DA_maaslin2 <span class="ot">&lt;-</span> <span class="fu">run_maaslin2</span>(</span>
<span id="cb345-22"><a href="differential-analysis.html#cb345-22" tabindex="-1"></a>                  <span class="at">data_otu =</span> phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb345-23"><a href="differential-analysis.html#cb345-23" tabindex="-1"></a>                  <span class="at">data_sam =</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_genus_filter_trim),</span>
<span id="cb345-24"><a href="differential-analysis.html#cb345-24" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb345-25"><a href="differential-analysis.html#cb345-25" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb345-26"><a href="differential-analysis.html#cb345-26" tabindex="-1"></a>                  <span class="at">transform =</span> <span class="st">&quot;LOG&quot;</span>,</span>
<span id="cb345-27"><a href="differential-analysis.html#cb345-27" tabindex="-1"></a>                  <span class="at">norm =</span> <span class="st">&quot;TMM&quot;</span>,</span>
<span id="cb345-28"><a href="differential-analysis.html#cb345-28" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb345-29"><a href="differential-analysis.html#cb345-29" tabindex="-1"></a>                  <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb345-30"><a href="differential-analysis.html#cb345-30" tabindex="-1"></a></span>
<span id="cb345-31"><a href="differential-analysis.html#cb345-31" tabindex="-1"></a>DA_maaslin2 <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb345-32"><a href="differential-analysis.html#cb345-32" tabindex="-1"></a>                  <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb345-33"><a href="differential-analysis.html#cb345-33" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb345-34"><a href="differential-analysis.html#cb345-34" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb345-35"><a href="differential-analysis.html#cb345-35" tabindex="-1"></a>                  <span class="at">da_method =</span> <span class="st">&quot;maaslin2&quot;</span>,</span>
<span id="cb345-36"><a href="differential-analysis.html#cb345-36" tabindex="-1"></a>                  <span class="at">transform =</span> <span class="st">&quot;LOG&quot;</span>,</span>
<span id="cb345-37"><a href="differential-analysis.html#cb345-37" tabindex="-1"></a>                  <span class="at">norm =</span> <span class="st">&quot;TMM&quot;</span>,</span>
<span id="cb345-38"><a href="differential-analysis.html#cb345-38" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb345-39"><a href="differential-analysis.html#cb345-39" tabindex="-1"></a>                  <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb345-40"><a href="differential-analysis.html#cb345-40" tabindex="-1"></a></span>
<span id="cb345-41"><a href="differential-analysis.html#cb345-41" tabindex="-1"></a><span class="fu">head</span>(DA_maaslin2)</span>
<span id="cb345-42"><a href="differential-analysis.html#cb345-42" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="locom" class="section level3 hasAnchor" number="10.3.15">
<h3><span class="header-section-number">10.3.15</span> LOCOM<a href="differential-analysis.html#locom" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>LOCOM: A logistic regression model for testing differential abundance in compositional microbiome data
with false discovery rate control <span class="citation">(<a href="#ref-hu2022locom">Hu, Satten, and Hu 2022</a>)</span>.</p>
<p><strong>Cautions</strong>: It reports an error information which the matrix is <strong>singularError</strong> when user uses low <em>prev.cut</em>.</p>
<div class="sourceCode" id="cb346"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb346-1"><a href="differential-analysis.html#cb346-1" tabindex="-1"></a><span class="fu">library</span>(permute)</span>
<span id="cb346-2"><a href="differential-analysis.html#cb346-2" tabindex="-1"></a>DA_locom <span class="ot">&lt;-</span> <span class="fu">run_locom</span>(<span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb346-3"><a href="differential-analysis.html#cb346-3" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb346-4"><a href="differential-analysis.html#cb346-4" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb346-5"><a href="differential-analysis.html#cb346-5" tabindex="-1"></a>                    <span class="at">fdr.nominal =</span> <span class="fl">0.2</span>,</span>
<span id="cb346-6"><a href="differential-analysis.html#cb346-6" tabindex="-1"></a>                    <span class="at">prev.cut =</span> <span class="fl">0.8</span>)</span></code></pre></div>
<pre><code>## Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred

## Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred

## Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</code></pre>
<pre><code>## 71 OTU(s) with fewer than 18.4 in all samples are removed 
## permutations: 1</code></pre>
<div class="sourceCode" id="cb349"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb349-1"><a href="differential-analysis.html#cb349-1" tabindex="-1"></a>DA_locom</span></code></pre></div>
<pre><code>##                 TaxaID         Block EffectSize Pvalue AdjustedPvalue Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All)
## 1       g__Bacteroides 9_AA vs 14_BB      1.030  0.024           0.12                         1.5367266                    3807
## 2           g__Blautia 9_AA vs 14_BB      0.570  0.034           0.12                         0.2641697                    8199
## 3 g__Lachnoclostridium 9_AA vs 14_BB      0.672  0.040           0.12                         0.7874146                     422
## 4     g__Lactobacillus 9_AA vs 14_BB     -3.180  0.028           0.12                        -3.0334230                      56
##   Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB
## 1                 9103               3137.5                       0.9841180             5745.5652         8219.44444          4155.2143
## 2                 8645               7198.5                       0.3178904             9145.9565        10397.55556          8341.3571
## 3                  485                281.0                       0.4658394              417.9565          502.33333           363.7143
## 4                   24                196.5                      -5.0905713              791.2609           37.44444          1275.8571
##   Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB       Odds Ratio (95% CI)
## 1                   100.00                100.00                100.00             0.45 (-1.1;2)
## 2                   100.00                100.00                100.00           0.62 (-0.3;1.5)
## 3                    95.65                100.00                 92.86          0.63 (-0.27;1.5)
## 4                    86.96                 77.78                 92.86 1.8e+08 (1.8e+08;1.8e+08)</code></pre>
<p><strong>Results</strong>:</p>
<p>The results comprises more than 10 columns, and the details as follow:</p>
<ol style="list-style-type: decimal">
<li><p><strong>TaxaID</strong>: taxa name;</p></li>
<li><p><strong>Block</strong>: groups’ names and numbers;</p></li>
<li><p><strong>EffectSize</strong>: effect size by <strong>ANCOM</strong>;</p></li>
<li><p><strong>Pvalue</strong>: p-values for OTU-specific tests;</p></li>
<li><p><strong>AdjustedPvalue </strong>: q-values (adjusted p-values by the BH procedure) for OTU-specific tests;</p></li>
<li><p>**Log2FoldChange (Median)_vs_BB**: Log2FoldChange (Median Abundance) between group AA and group BB;</p></li>
<li><p><strong>Median Abundance (All)/(group AA)/(group BB)</strong>: Median Abundance in all, group AA and group BB, respectively;</p></li>
<li><p>**Log2FoldChange (Mean)_vs_BB**: Log2FoldChange (Mean Abundance) between group AA and group BB;</p></li>
<li><p><strong>Mean Abundance (All)/(group AA)/(group BB)</strong>: Mean Abundance in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Occurrence (All)/(group AA)/(group BB)</strong>: Occurrence in all, group AA and group BB, respectively;</p></li>
<li><p><strong>Odds Ratio (95% CI)</strong>: 95% confidence interval odds ratio between group AA and group BB.</p></li>
</ol>
</div>
</div>
<div id="metagenomic-sequencing-microbial-data-metaphlan23" class="section level2 hasAnchor" number="10.4">
<h2><span class="header-section-number">10.4</span> Metagenomic sequencing microbial data (metaphlan2/3)<a href="differential-analysis.html#metagenomic-sequencing-microbial-data-metaphlan23" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="wilcoxon-rank-sum-and-signed-rank-tests-1" class="section level3 hasAnchor" number="10.4.1">
<h3><span class="header-section-number">10.4.1</span> Wilcoxon Rank Sum and Signed Rank Tests<a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Wilcoxon Rank Sum and Signed Rank Tests, which are nonparameter test methods, use the rank of taxa abundance to find the significant taxa.</p>
<div class="sourceCode" id="cb351"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb351-1"><a href="differential-analysis.html#cb351-1" tabindex="-1"></a>DA_wilcox_mgs <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb351-2"><a href="differential-analysis.html#cb351-2" tabindex="-1"></a>                    <span class="at">ps =</span> metaphlan2_ps_genus_filter_trim,</span>
<span id="cb351-3"><a href="differential-analysis.html#cb351-3" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb351-4"><a href="differential-analysis.html#cb351-4" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb351-5"><a href="differential-analysis.html#cb351-5" tabindex="-1"></a><span class="fu">colnames</span>(DA_wilcox_mgs)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Statistic&quot;                         &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Rank)\nAA_vs_BB&quot;  
## [13] &quot;Mean Rank Abundance\nAA&quot;           &quot;Mean Rank Abundance\nBB&quot;           &quot;Occurrence (100%)\n(All)&quot;         
## [16] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb353"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb353-1"><a href="differential-analysis.html#cb353-1" tabindex="-1"></a><span class="fu">head</span>(DA_wilcox_mgs)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment EffectSize Statistic     Pvalue AdjustedPvalue Log2FoldChange (Median)\nAA_vs_BB
## 1 g__Acidaminococcus 7_AA vs 15_BB  Nonsignif  0.9966151      36.0 0.19074499     0.52578545                                NA
## 2   g__Adlercreutzia 7_AA vs 15_BB  Nonsignif  0.5416258      28.0 0.05769072     0.20413638                                NA
## 3       g__Alistipes 7_AA vs 15_BB  Nonsignif  1.6269577      71.0 0.20193465     0.52578545                          2.061604
## 4    g__Anaerostipes 7_AA vs 15_BB  Nonsignif  0.3278059      31.0 0.13834560     0.42425985                         -4.964322
## 5   g__Anaerotruncus 7_AA vs 15_BB  Nonsignif  0.8349750      47.5 0.71288708     0.81982014                                NA
## 6     g__Bacteroides 7_AA vs 15_BB  Nonsignif  0.9956964      85.0 0.02128483     0.09495645                          1.229057
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Rank)\nAA_vs_BB Mean Rank Abundance\nAA
## 1              0.00000000            0.0000000            0.0000000                      -0.4631577                    9.14
## 2              0.00000000            0.0000000            0.0000558                      -0.7147950                    8.00
## 3              0.00711780            0.0236665            0.0056693                       0.4613459                   14.14
## 4              0.00031435            0.0000229            0.0007149                      -0.6171177                    8.43
## 5              0.00000000            0.0000000            0.0000000                      -0.1327552                   10.79
## 6              0.24894650            0.4568864            0.1949060                       0.7906916                   16.14
##   Mean Rank Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                   12.60                    36.36                 14.29                 46.67    8700 (8700;8700)
## 2                   13.13                    40.91                 14.29                 53.33       Inf (Inf;Inf)
## 3                   10.27                    77.27                 71.43                 80.00     0.22 (-2.7;3.2)
## 4                   12.93                    86.36                 71.43                 93.33         6 (9.5;2.5)
## 5                   11.83                    36.36                 28.57                 40.00      1.3 (1.8;0.78)
## 6                    9.33                   100.00                100.00                100.00       0.31 (-2;2.6)</code></pre>
<details>
<summary>
<strong>wilcox Metagenomic sequencing in run_da</strong>
</summary>
<div class="sourceCode" id="cb355"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb355-1"><a href="differential-analysis.html#cb355-1" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb355-2"><a href="differential-analysis.html#cb355-2" tabindex="-1"></a>DA_wilcox_mgs <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb355-3"><a href="differential-analysis.html#cb355-3" tabindex="-1"></a>                    <span class="at">ps =</span> metaphlan2_ps_genus_filter_trim,</span>
<span id="cb355-4"><a href="differential-analysis.html#cb355-4" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb355-5"><a href="differential-analysis.html#cb355-5" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb355-6"><a href="differential-analysis.html#cb355-6" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;wilcox&quot;</span>)</span>
<span id="cb355-7"><a href="differential-analysis.html#cb355-7" tabindex="-1"></a></span>
<span id="cb355-8"><a href="differential-analysis.html#cb355-8" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="liner-discriminant-analysis-lda-effect-size-lefse-1" class="section level3 hasAnchor" number="10.4.2">
<h3><span class="header-section-number">10.4.2</span> Liner discriminant analysis (LDA) effect size (LEfSe)<a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>LEfSe</strong> method is from <em>Metagenomic biomarker discovery and explanation</em> <span class="citation">(<a href="#ref-segata2011metagenomic">Segata et al. 2011</a>)</span>. It uses Liner discriminant analysis model to identify Differential Taxa.</p>
<div class="sourceCode" id="cb356"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb356-1"><a href="differential-analysis.html#cb356-1" tabindex="-1"></a><span class="co"># DA_lefse_mgs &lt;- run_lefse(</span></span>
<span id="cb356-2"><a href="differential-analysis.html#cb356-2" tabindex="-1"></a><span class="co">#                     ps = metaphlan2_ps_genus_filter_trim,</span></span>
<span id="cb356-3"><a href="differential-analysis.html#cb356-3" tabindex="-1"></a><span class="co">#                     group = &quot;Group&quot;,</span></span>
<span id="cb356-4"><a href="differential-analysis.html#cb356-4" tabindex="-1"></a><span class="co">#                     group_names = c(&quot;AA&quot;, &quot;BB&quot;),</span></span>
<span id="cb356-5"><a href="differential-analysis.html#cb356-5" tabindex="-1"></a><span class="co">#                     norm = &quot;CPM&quot;,</span></span>
<span id="cb356-6"><a href="differential-analysis.html#cb356-6" tabindex="-1"></a><span class="co">#                     Lda = 2) </span></span>
<span id="cb356-7"><a href="differential-analysis.html#cb356-7" tabindex="-1"></a></span>
<span id="cb356-8"><a href="differential-analysis.html#cb356-8" tabindex="-1"></a>DA_lefse_mgs <span class="ot">&lt;-</span> <span class="fu">run_lefse2</span>(</span>
<span id="cb356-9"><a href="differential-analysis.html#cb356-9" tabindex="-1"></a>                    <span class="at">ps =</span> metaphlan2_ps_genus_filter_trim,</span>
<span id="cb356-10"><a href="differential-analysis.html#cb356-10" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb356-11"><a href="differential-analysis.html#cb356-11" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb356-12"><a href="differential-analysis.html#cb356-12" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CPM&quot;</span>,</span>
<span id="cb356-13"><a href="differential-analysis.html#cb356-13" tabindex="-1"></a>                    <span class="at">lda_cutoff =</span> <span class="dv">2</span>) </span>
<span id="cb356-14"><a href="differential-analysis.html#cb356-14" tabindex="-1"></a></span>
<span id="cb356-15"><a href="differential-analysis.html#cb356-15" tabindex="-1"></a><span class="fu">colnames</span>(DA_lefse_mgs)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;LDA_Score&quot;                        
##  [4] &quot;Enrichment&quot;                        &quot;EffectSize&quot;                        &quot;Pvalue&quot;                           
##  [7] &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;           &quot;Median Abundance\nAA&quot;             
## [10] &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;   &quot;Mean Abundance\n(All)&quot;            
## [13] &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;                &quot;Occurrence (100%)\n(All)&quot;         
## [16] &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;             &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb358"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb358-1"><a href="differential-analysis.html#cb358-1" tabindex="-1"></a><span class="fu">head</span>(DA_lefse_mgs)</span></code></pre></div>
<pre><code>##               TaxaID         Block LDA_Score Enrichment EffectSize      Pvalue Log2FoldChange (Median)\nAA_vs_BB Median Abundance\n(All)
## 1     g__Bacteroides 7_AA vs 15_BB -5.379223         AA  2.5356524 0.021966365                          1.190538             254267.6272
## 2 g__Bifidobacterium 7_AA vs 15_BB  5.170414         BB  2.8199931 0.003440517                         -6.574737              39083.6006
## 3         g__Blautia 7_AA vs 15_BB  4.496247         BB  0.5696647 0.001046123                         -3.203712              21823.6865
## 4     g__Collinsella 7_AA vs 15_BB  4.229420         BB  2.4346418 0.019909355                                NA               7739.0203
## 5           g__Dorea 7_AA vs 15_BB  4.208815         BB  2.0503963 0.004294461                         -5.725635               8556.6787
## 6     g__Eggerthella 7_AA vs 15_BB  3.659350         BB  1.2390028 0.047014415                                NA                591.5818
##   Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All) Mean Abundance\nAA Mean Abundance\nBB
## 1          464614.8499          203566.1197                        1.141008            291728.672        465019.4054         210859.663
## 2             917.9094           87496.9227                       -3.459750            114237.628         14608.6929         160731.132
## 3            3704.9909           34135.0398                       -2.335372             25770.580          6855.3562          34597.684
## 4               0.0000           11644.4661                       -2.928408             12502.243          2269.5616          17277.494
## 5             270.5248           14315.1145                       -3.454703             11345.173          1455.6962          15960.262
## 6               0.0000             771.5754                       -3.764261              2232.472           232.9684           3165.574
##   Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                   100.00                100.00                100.00     0.32 (-1.9;2.5)
## 2                   100.00                100.00                100.00       150 (160;140)
## 3                   100.00                100.00                100.00         92 (100;83)
## 4                    68.18                 42.86                 80.00          20 (25;14)
## 5                    90.91                 71.43                100.00          28 (34;21)
## 6                    63.64                 42.86                 73.33       430 (450;420)</code></pre>
<details>
<summary>
<strong>lefse Metagenomic sequencing in run_da</strong>
</summary>
<div class="sourceCode" id="cb360"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb360-1"><a href="differential-analysis.html#cb360-1" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb360-2"><a href="differential-analysis.html#cb360-2" tabindex="-1"></a>DA_lefse_mgs <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb360-3"><a href="differential-analysis.html#cb360-3" tabindex="-1"></a>                    <span class="at">ps =</span> metaphlan2_ps_genus_filter_trim,</span>
<span id="cb360-4"><a href="differential-analysis.html#cb360-4" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb360-5"><a href="differential-analysis.html#cb360-5" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb360-6"><a href="differential-analysis.html#cb360-6" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;lefse&quot;</span>,</span>
<span id="cb360-7"><a href="differential-analysis.html#cb360-7" tabindex="-1"></a>                    <span class="at">norm =</span> <span class="st">&quot;CPM&quot;</span>,</span>
<span id="cb360-8"><a href="differential-analysis.html#cb360-8" tabindex="-1"></a>                    <span class="at">Lda =</span> <span class="dv">0</span>)</span>
<span id="cb360-9"><a href="differential-analysis.html#cb360-9" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="t-test-1" class="section level3 hasAnchor" number="10.4.3">
<h3><span class="header-section-number">10.4.3</span> t-test<a href="differential-analysis.html#t-test-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>T test, a parametric test method, identifies the significant taxa.</p>
<div class="sourceCode" id="cb361"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb361-1"><a href="differential-analysis.html#cb361-1" tabindex="-1"></a>DA_ttest_mgs <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb361-2"><a href="differential-analysis.html#cb361-2" tabindex="-1"></a>                    <span class="at">ps =</span> metaphlan2_ps_genus_filter_trim,</span>
<span id="cb361-3"><a href="differential-analysis.html#cb361-3" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb361-4"><a href="differential-analysis.html#cb361-4" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb361-5"><a href="differential-analysis.html#cb361-5" tabindex="-1"></a><span class="fu">colnames</span>(DA_ttest_mgs)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;Statistic&quot;                         &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb363"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb363-1"><a href="differential-analysis.html#cb363-1" tabindex="-1"></a><span class="fu">head</span>(DA_ttest_mgs)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment EffectSize  Statistic     Pvalue AdjustedPvalue Log2FoldChange (Median)\nAA_vs_BB
## 1 g__Acidaminococcus 7_AA vs 15_BB  Nonsignif  0.9966151 -1.0515590 0.31074638      0.6151313                                NA
## 2   g__Adlercreutzia 7_AA vs 15_BB  Nonsignif  0.5416258 -2.1929042 0.04570343      0.2102358                                NA
## 3       g__Alistipes 7_AA vs 15_BB  Nonsignif  1.6269577  1.5963681 0.15951558      0.5119489                          2.061604
## 4    g__Anaerostipes 7_AA vs 15_BB  Nonsignif  0.3278059 -1.6970563 0.10937430      0.4192682                         -4.964322
## 5   g__Anaerotruncus 7_AA vs 15_BB  Nonsignif  0.8349750 -0.5871152 0.56369794      0.6823712                                NA
## 6     g__Bacteroides 7_AA vs 15_BB  Nonsignif  0.9956964  2.2882319 0.04545820      0.2102358                          1.229057
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All)
## 1              0.00000000            0.0000000            0.0000000                      -4.6280567          0.0095489955
## 2              0.00000000            0.0000000            0.0000558                      -9.2633214          0.0008266409
## 3              0.00711780            0.0236665            0.0056693                       2.1761109          0.0199576000
## 4              0.00031435            0.0000229            0.0007149                      -2.2827694          0.0016651773
## 5              0.00000000            0.0000000            0.0000000                      -0.9025366          0.0002137000
## 6              0.24894650            0.4568864            0.1949060                       1.1807675          0.2848799364
##   Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1       5.558857e-04       0.0137457800                    36.36                 14.29                 46.67    8700 (8700;8700)
## 2       1.971429e-06       0.0012114867                    40.91                 14.29                 53.33       Inf (Inf;Inf)
## 3       4.254910e-02       0.0094149000                    77.27                 71.43                 80.00     0.22 (-2.7;3.2)
## 4       4.579714e-04       0.0022285400                    86.36                 71.43                 93.33         6 (9.5;2.5)
## 5       1.341714e-04       0.0002508133                    36.36                 28.57                 40.00      1.3 (1.8;0.78)
## 6       4.602683e-01       0.2030320467                   100.00                100.00                100.00       0.31 (-2;2.6)</code></pre>
<details>
<summary>
<strong>ttest Metagenomic sequencing in run_da</strong>
</summary>
<div class="sourceCode" id="cb365"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb365-1"><a href="differential-analysis.html#cb365-1" tabindex="-1"></a><span class="cf">if</span>(<span class="dv">0</span>) {</span>
<span id="cb365-2"><a href="differential-analysis.html#cb365-2" tabindex="-1"></a>DA_ttest_mgs <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb365-3"><a href="differential-analysis.html#cb365-3" tabindex="-1"></a>                    <span class="at">ps =</span> metaphlan2_ps_genus_filter_trim,</span>
<span id="cb365-4"><a href="differential-analysis.html#cb365-4" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb365-5"><a href="differential-analysis.html#cb365-5" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb365-6"><a href="differential-analysis.html#cb365-6" tabindex="-1"></a>                    <span class="at">da_method =</span> <span class="st">&quot;ttest&quot;</span>)</span>
<span id="cb365-7"><a href="differential-analysis.html#cb365-7" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
<div id="maaslin2-microbiome-multivariable-association-with-linear-models-1" class="section level3 hasAnchor" number="10.4.4">
<h3><span class="header-section-number">10.4.4</span> Maaslin2 (Microbiome Multivariable Association with Linear Models)<a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Maaslin2</strong> package is from <em>Multivariable association discovery in population-scale meta-omics studies</em> <span class="citation">(<a href="#ref-mallick2021multivariable">Mallick et al. 2021</a>)</span>. Maaslin2 relies on general linear models to accommodate most modern epidemiological study designs, including cross-sectional and longitudinal, along with a variety of filtering, normalization, and transform methods.</p>
<div class="sourceCode" id="cb366"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb366-1"><a href="differential-analysis.html#cb366-1" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb366-2"><a href="differential-analysis.html#cb366-2" tabindex="-1"></a>DA_maaslin2_mgs <span class="ot">&lt;-</span> <span class="fu">run_maaslin2</span>(</span>
<span id="cb366-3"><a href="differential-analysis.html#cb366-3" tabindex="-1"></a>                      <span class="at">ps =</span> metaphlan2_ps_genus_filter_trim,</span>
<span id="cb366-4"><a href="differential-analysis.html#cb366-4" tabindex="-1"></a>                      <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb366-5"><a href="differential-analysis.html#cb366-5" tabindex="-1"></a>                      <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb366-6"><a href="differential-analysis.html#cb366-6" tabindex="-1"></a>                      <span class="at">transform =</span> <span class="st">&quot;NONE&quot;</span>,</span>
<span id="cb366-7"><a href="differential-analysis.html#cb366-7" tabindex="-1"></a>                      <span class="at">norm =</span> <span class="st">&quot;NONE&quot;</span>,</span>
<span id="cb366-8"><a href="differential-analysis.html#cb366-8" tabindex="-1"></a>                      <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb366-9"><a href="differential-analysis.html#cb366-9" tabindex="-1"></a>                      <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb366-10"><a href="differential-analysis.html#cb366-10" tabindex="-1"></a></span>
<span id="cb366-11"><a href="differential-analysis.html#cb366-11" tabindex="-1"></a><span class="fu">head</span>(DA_maaslin2_mgs)</span>
<span id="cb366-12"><a href="differential-analysis.html#cb366-12" tabindex="-1"></a>}</span></code></pre></div>
<details>
<summary>
<strong>maaslin2 Metagenomic sequencing in run_da</strong>
</summary>
<div class="sourceCode" id="cb367"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb367-1"><a href="differential-analysis.html#cb367-1" tabindex="-1"></a><span class="cf">if</span> (<span class="dv">0</span>) {</span>
<span id="cb367-2"><a href="differential-analysis.html#cb367-2" tabindex="-1"></a>DA_maaslin2_mgs <span class="ot">&lt;-</span> <span class="fu">run_da</span>(</span>
<span id="cb367-3"><a href="differential-analysis.html#cb367-3" tabindex="-1"></a>                      <span class="at">ps =</span> metaphlan2_ps_genus_filter_trim,</span>
<span id="cb367-4"><a href="differential-analysis.html#cb367-4" tabindex="-1"></a>                      <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb367-5"><a href="differential-analysis.html#cb367-5" tabindex="-1"></a>                      <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb367-6"><a href="differential-analysis.html#cb367-6" tabindex="-1"></a>                      <span class="at">da_method =</span> <span class="st">&quot;maaslin2&quot;</span>,</span>
<span id="cb367-7"><a href="differential-analysis.html#cb367-7" tabindex="-1"></a>                      <span class="at">transform =</span> <span class="st">&quot;NONE&quot;</span>,</span>
<span id="cb367-8"><a href="differential-analysis.html#cb367-8" tabindex="-1"></a>                      <span class="at">norm =</span> <span class="st">&quot;NONE&quot;</span>,</span>
<span id="cb367-9"><a href="differential-analysis.html#cb367-9" tabindex="-1"></a>                      <span class="at">method =</span> <span class="st">&quot;LM&quot;</span>,</span>
<span id="cb367-10"><a href="differential-analysis.html#cb367-10" tabindex="-1"></a>                      <span class="at">outdir =</span> <span class="st">&quot;./demo_output&quot;</span>)</span>
<span id="cb367-11"><a href="differential-analysis.html#cb367-11" tabindex="-1"></a>}</span></code></pre></div>
</details>
</div>
</div>
<div id="visualization-4" class="section level2 hasAnchor" number="10.5">
<h2><span class="header-section-number">10.5</span> Visualization<a href="differential-analysis.html#visualization-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The Volcano plot is used to display differential analysis. <code>plot_volcano</code> provides multiple parameters for plotting volcano. More details to see <code>help(plot_volcano)</code>.</p>
<p>The barplot is used to display the lefse results.</p>
<div id="volcano-plot" class="section level3 hasAnchor" number="10.5.1">
<h3><span class="header-section-number">10.5.1</span> Volcano plot<a href="differential-analysis.html#volcano-plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The X and Y coordinate axis are flexible to choose for Volcano. Here, we choose <em>logFC</em> and <em>AdjustedPvalue</em> to visualize the results</p>
<div class="sourceCode" id="cb368"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb368-1"><a href="differential-analysis.html#cb368-1" tabindex="-1"></a>DA_limma_voom <span class="ot">&lt;-</span> <span class="fu">run_limma_voom</span>(</span>
<span id="cb368-2"><a href="differential-analysis.html#cb368-2" tabindex="-1"></a>                    <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb368-3"><a href="differential-analysis.html#cb368-3" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb368-4"><a href="differential-analysis.html#cb368-4" tabindex="-1"></a>                    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb368-5"><a href="differential-analysis.html#cb368-5" tabindex="-1"></a></span>
<span id="cb368-6"><a href="differential-analysis.html#cb368-6" tabindex="-1"></a><span class="fu">colnames</span>(DA_limma_voom)</span></code></pre></div>
<pre><code>##  [1] &quot;TaxaID&quot;                            &quot;Block&quot;                             &quot;Enrichment&quot;                       
##  [4] &quot;EffectSize&quot;                        &quot;logFC&quot;                             &quot;Pvalue&quot;                           
##  [7] &quot;AdjustedPvalue&quot;                    &quot;Log2FoldChange (Median)\nAA_vs_BB&quot; &quot;Median Abundance\n(All)&quot;          
## [10] &quot;Median Abundance\nAA&quot;              &quot;Median Abundance\nBB&quot;              &quot;Log2FoldChange (Mean)\nAA_vs_BB&quot;  
## [13] &quot;Mean Abundance\n(All)&quot;             &quot;Mean Abundance\nAA&quot;                &quot;Mean Abundance\nBB&quot;               
## [16] &quot;Occurrence (100%)\n(All)&quot;          &quot;Occurrence (100%)\nAA&quot;             &quot;Occurrence (100%)\nBB&quot;            
## [19] &quot;Odds Ratio (95% CI)&quot;</code></pre>
<div class="sourceCode" id="cb370"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb370-1"><a href="differential-analysis.html#cb370-1" tabindex="-1"></a>DA_limma_voom_volcano <span class="ot">&lt;-</span> <span class="fu">plot_volcano</span>(</span>
<span id="cb370-2"><a href="differential-analysis.html#cb370-2" tabindex="-1"></a>                          <span class="at">da_res =</span> DA_limma_voom,</span>
<span id="cb370-3"><a href="differential-analysis.html#cb370-3" tabindex="-1"></a>                          <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb370-4"><a href="differential-analysis.html#cb370-4" tabindex="-1"></a>                          <span class="at">x_index =</span> <span class="st">&quot;logFC&quot;</span>,</span>
<span id="cb370-5"><a href="differential-analysis.html#cb370-5" tabindex="-1"></a>                          <span class="at">x_index_cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb370-6"><a href="differential-analysis.html#cb370-6" tabindex="-1"></a>                          <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb370-7"><a href="differential-analysis.html#cb370-7" tabindex="-1"></a>                          <span class="at">y_index_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb370-8"><a href="differential-analysis.html#cb370-8" tabindex="-1"></a>                          <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb370-9"><a href="differential-analysis.html#cb370-9" tabindex="-1"></a>                          <span class="at">topN =</span> <span class="dv">5</span>)</span>
<span id="cb370-10"><a href="differential-analysis.html#cb370-10" tabindex="-1"></a></span>
<span id="cb370-11"><a href="differential-analysis.html#cb370-11" tabindex="-1"></a>DA_limma_voom_volcano</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-187"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-187-1.png" alt="Volcano (limma-voom: logFC and AdjustedPvalue)" width="672" />
<p class="caption">
Figure 10.3: Volcano (limma-voom: logFC and AdjustedPvalue)
</p>
</div>
<div class="sourceCode" id="cb371"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb371-1"><a href="differential-analysis.html#cb371-1" tabindex="-1"></a><span class="fu">plot_volcano</span>(</span>
<span id="cb371-2"><a href="differential-analysis.html#cb371-2" tabindex="-1"></a>  <span class="at">da_res =</span> DA_limma_voom,</span>
<span id="cb371-3"><a href="differential-analysis.html#cb371-3" tabindex="-1"></a>  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb371-4"><a href="differential-analysis.html#cb371-4" tabindex="-1"></a>  <span class="at">x_index =</span> <span class="st">&quot;logFC&quot;</span>,</span>
<span id="cb371-5"><a href="differential-analysis.html#cb371-5" tabindex="-1"></a>  <span class="at">x_index_cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb371-6"><a href="differential-analysis.html#cb371-6" tabindex="-1"></a>  <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb371-7"><a href="differential-analysis.html#cb371-7" tabindex="-1"></a>  <span class="at">y_index_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb371-8"><a href="differential-analysis.html#cb371-8" tabindex="-1"></a>  <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb371-9"><a href="differential-analysis.html#cb371-9" tabindex="-1"></a>  <span class="at">topN =</span> <span class="dv">5</span>,</span>
<span id="cb371-10"><a href="differential-analysis.html#cb371-10" tabindex="-1"></a>  <span class="at">add_enrich_arrow =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-188"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-188-1.png" alt="Volcano with arrow (limma-voom: logFC and AdjustedPvalue)" width="672" />
<p class="caption">
Figure 10.4: Volcano with arrow (limma-voom: logFC and AdjustedPvalue)
</p>
</div>
<details>
<summary>
<strong>DA_limma_voom_volcano effectsize</strong>
</summary>
<div class="sourceCode" id="cb372"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb372-1"><a href="differential-analysis.html#cb372-1" tabindex="-1"></a><span class="fu">plot_volcano</span>(</span>
<span id="cb372-2"><a href="differential-analysis.html#cb372-2" tabindex="-1"></a>    <span class="at">da_res =</span> DA_limma_voom,</span>
<span id="cb372-3"><a href="differential-analysis.html#cb372-3" tabindex="-1"></a>    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb372-4"><a href="differential-analysis.html#cb372-4" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;EffectSize&quot;</span>,</span>
<span id="cb372-5"><a href="differential-analysis.html#cb372-5" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="dv">1</span>,</span>
<span id="cb372-6"><a href="differential-analysis.html#cb372-6" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb372-7"><a href="differential-analysis.html#cb372-7" tabindex="-1"></a>    <span class="at">y_index_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb372-8"><a href="differential-analysis.html#cb372-8" tabindex="-1"></a>    <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb372-9"><a href="differential-analysis.html#cb372-9" tabindex="-1"></a>    <span class="at">topN =</span> <span class="dv">8</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-189"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-189-1.png" alt="Volcano (limma-voom: EffectSize and Pvalue)" width="672" />
<p class="caption">
Figure 10.5: Volcano (limma-voom: EffectSize and Pvalue)
</p>
</div>
<p>The logFC and EffectSize are the same values in limma-voom.</p>
</details>
</div>
<div id="barplot-in-lefse" class="section level3 hasAnchor" number="10.5.2">
<h3><span class="header-section-number">10.5.2</span> barplot in lefse<a href="differential-analysis.html#barplot-in-lefse" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb373"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb373-1"><a href="differential-analysis.html#cb373-1" tabindex="-1"></a><span class="co"># DA_lefse &lt;- run_lefse(</span></span>
<span id="cb373-2"><a href="differential-analysis.html#cb373-2" tabindex="-1"></a><span class="co">#               ps = dada2_ps_genus_filter_trim,</span></span>
<span id="cb373-3"><a href="differential-analysis.html#cb373-3" tabindex="-1"></a><span class="co">#               group = &quot;Group&quot;,</span></span>
<span id="cb373-4"><a href="differential-analysis.html#cb373-4" tabindex="-1"></a><span class="co">#               group_names = c(&quot;AA&quot;, &quot;BB&quot;),</span></span>
<span id="cb373-5"><a href="differential-analysis.html#cb373-5" tabindex="-1"></a><span class="co">#               norm = &quot;CPM&quot;,</span></span>
<span id="cb373-6"><a href="differential-analysis.html#cb373-6" tabindex="-1"></a><span class="co">#               Lda = 2)</span></span>
<span id="cb373-7"><a href="differential-analysis.html#cb373-7" tabindex="-1"></a></span>
<span id="cb373-8"><a href="differential-analysis.html#cb373-8" tabindex="-1"></a>DA_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse2</span>(</span>
<span id="cb373-9"><a href="differential-analysis.html#cb373-9" tabindex="-1"></a>              <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb373-10"><a href="differential-analysis.html#cb373-10" tabindex="-1"></a>              <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb373-11"><a href="differential-analysis.html#cb373-11" tabindex="-1"></a>              <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb373-12"><a href="differential-analysis.html#cb373-12" tabindex="-1"></a>              <span class="at">norm =</span> <span class="st">&quot;CPM&quot;</span>,</span>
<span id="cb373-13"><a href="differential-analysis.html#cb373-13" tabindex="-1"></a>              <span class="at">lda_cutoff =</span> <span class="dv">2</span>)</span>
<span id="cb373-14"><a href="differential-analysis.html#cb373-14" tabindex="-1"></a></span>
<span id="cb373-15"><a href="differential-analysis.html#cb373-15" tabindex="-1"></a><span class="co"># # don&#39;t run this code when you do lefse in reality</span></span>
<span id="cb373-16"><a href="differential-analysis.html#cb373-16" tabindex="-1"></a><span class="co"># DA_lefse$LDA_Score &lt;- DA_lefse$LDA_Score * 1000</span></span>
<span id="cb373-17"><a href="differential-analysis.html#cb373-17" tabindex="-1"></a></span>
<span id="cb373-18"><a href="differential-analysis.html#cb373-18" tabindex="-1"></a><span class="fu">plot_lefse</span>(</span>
<span id="cb373-19"><a href="differential-analysis.html#cb373-19" tabindex="-1"></a>  <span class="at">da_res =</span> DA_lefse,</span>
<span id="cb373-20"><a href="differential-analysis.html#cb373-20" tabindex="-1"></a>  <span class="at">x_index =</span> <span class="st">&quot;LDA_Score&quot;</span>,</span>
<span id="cb373-21"><a href="differential-analysis.html#cb373-21" tabindex="-1"></a>  <span class="at">x_index_cutoff =</span> <span class="dv">1</span>,</span>
<span id="cb373-22"><a href="differential-analysis.html#cb373-22" tabindex="-1"></a>  <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>)) </span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-190"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-190-1.png" alt="Barplot (Lefse)" width="576" />
<p class="caption">
Figure 10.6: Barplot (Lefse)
</p>
</div>
</div>
</div>
<div id="dominant-taxa" class="section level2 hasAnchor" number="10.6">
<h2><span class="header-section-number">10.6</span> Dominant taxa<a href="differential-analysis.html#dominant-taxa" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Display the significant taxa with selection using boxplot.</p>
<div class="sourceCode" id="cb374"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb374-1"><a href="differential-analysis.html#cb374-1" tabindex="-1"></a>DA_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb374-2"><a href="differential-analysis.html#cb374-2" tabindex="-1"></a>               <span class="at">ps =</span> metaphlan2_ps_genus_filter_trim,</span>
<span id="cb374-3"><a href="differential-analysis.html#cb374-3" tabindex="-1"></a>               <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb374-4"><a href="differential-analysis.html#cb374-4" tabindex="-1"></a>               <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb374-5"><a href="differential-analysis.html#cb374-5" tabindex="-1"></a></span>
<span id="cb374-6"><a href="differential-analysis.html#cb374-6" tabindex="-1"></a><span class="fu">plot_topN_boxplot</span>(</span>
<span id="cb374-7"><a href="differential-analysis.html#cb374-7" tabindex="-1"></a>    <span class="at">ps =</span> metaphlan2_ps_genus_filter_trim,</span>
<span id="cb374-8"><a href="differential-analysis.html#cb374-8" tabindex="-1"></a>    <span class="at">da_res =</span> DA_wilcox,</span>
<span id="cb374-9"><a href="differential-analysis.html#cb374-9" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;Log2FoldChange (Median)</span><span class="sc">\n</span><span class="st">AA_vs_BB&quot;</span>,</span>
<span id="cb374-10"><a href="differential-analysis.html#cb374-10" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="fl">0.2</span>,</span>
<span id="cb374-11"><a href="differential-analysis.html#cb374-11" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb374-12"><a href="differential-analysis.html#cb374-12" tabindex="-1"></a>    <span class="at">y_index_cutoff =</span> <span class="fl">0.3</span>,</span>
<span id="cb374-13"><a href="differential-analysis.html#cb374-13" tabindex="-1"></a>    <span class="at">topN =</span> <span class="dv">3</span>,</span>
<span id="cb374-14"><a href="differential-analysis.html#cb374-14" tabindex="-1"></a>    <span class="at">taxa_name =</span> <span class="st">&quot;s__Ruminococcus_torques&quot;</span>,</span>
<span id="cb374-15"><a href="differential-analysis.html#cb374-15" tabindex="-1"></a>    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-191"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-191-1.png" alt="Dominant Taxa" width="768" />
<p class="caption">
Figure 10.7: Dominant Taxa
</p>
</div>
</div>
<div id="multiple-differential-analysis-by-one-function" class="section level2 hasAnchor" number="10.7">
<h2><span class="header-section-number">10.7</span> Multiple differential analysis by one function<a href="differential-analysis.html#multiple-differential-analysis-by-one-function" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>here, we provide the <code>run_multiple_da</code> for obtaining the results list from multiple differential analysis methods.</p>
<div class="sourceCode" id="cb375"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb375-1"><a href="differential-analysis.html#cb375-1" tabindex="-1"></a>multiple_res <span class="ot">&lt;-</span> <span class="fu">run_multiple_da</span>(</span>
<span id="cb375-2"><a href="differential-analysis.html#cb375-2" tabindex="-1"></a>                  <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb375-3"><a href="differential-analysis.html#cb375-3" tabindex="-1"></a>                  <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb375-4"><a href="differential-analysis.html#cb375-4" tabindex="-1"></a>                  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb375-5"><a href="differential-analysis.html#cb375-5" tabindex="-1"></a>                  <span class="at">da_method =</span> <span class="fu">c</span>(<span class="st">&quot;aldex&quot;</span>, <span class="st">&quot;limma_voom&quot;</span>, <span class="st">&quot;mbzinb&quot;</span>, <span class="st">&quot;omnibus&quot;</span>),</span>
<span id="cb375-6"><a href="differential-analysis.html#cb375-6" tabindex="-1"></a>                  <span class="at">p_adjust =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<pre><code>## |------------(25%)----------(50%)----------(75%)----------|
## Start GMPR normalization ...
## Start Winsorization ...
## Perform filtering ...
## --A total of  82  taxa will be tested with a sample size of 23 !
## --Omnibus test is selected!
## --Dispersion is treated as a parameter of interest!
## Start testing ...
## 10 %
## 20 %
## 30 %
## 40 %
## 50 %
## 60 %
## 70 %
## 80 %
## 90 %
## 100%!
## Handle failed taxa using permutation test!
## Permutation test ....
## Completed!</code></pre>
<div class="sourceCode" id="cb377"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb377-1"><a href="differential-analysis.html#cb377-1" tabindex="-1"></a><span class="fu">names</span>(multiple_res)</span></code></pre></div>
<pre><code>## [1] &quot;aldex&quot;      &quot;limma_voom&quot; &quot;mbzinb&quot;     &quot;omnibus&quot;</code></pre>
<ul>
<li>plot results</li>
</ul>
<div class="sourceCode" id="cb379"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb379-1"><a href="differential-analysis.html#cb379-1" tabindex="-1"></a><span class="fu">plot_multiple_DA</span>(</span>
<span id="cb379-2"><a href="differential-analysis.html#cb379-2" tabindex="-1"></a>    <span class="at">Multip_DA_res =</span> multiple_res,</span>
<span id="cb379-3"><a href="differential-analysis.html#cb379-3" tabindex="-1"></a>    <span class="at">x_index_list =</span> <span class="fu">c</span>(<span class="st">&quot;EffectSize&quot;</span>, <span class="st">&quot;logFC&quot;</span>, <span class="st">&quot;mean.LFC&quot;</span>, <span class="st">&quot;abund.LFC.CompvarBB.est&quot;</span>),</span>
<span id="cb379-4"><a href="differential-analysis.html#cb379-4" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="dv">0</span>,</span>
<span id="cb379-5"><a href="differential-analysis.html#cb379-5" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;AdjustedPvalue&quot;</span>,</span>
<span id="cb379-6"><a href="differential-analysis.html#cb379-6" tabindex="-1"></a>    <span class="at">y_index_cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb379-7"><a href="differential-analysis.html#cb379-7" tabindex="-1"></a>    <span class="at">cellwidth =</span> <span class="dv">50</span>,</span>
<span id="cb379-8"><a href="differential-analysis.html#cb379-8" tabindex="-1"></a>    <span class="at">cellheight =</span> <span class="dv">10</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-193"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-193-1.png" alt="Multiple DA results" width="768" />
<p class="caption">
Figure 10.8: Multiple DA results
</p>
</div>
</div>
<div id="systematic-information-7" class="section level2 hasAnchor" number="10.8">
<h2><span class="header-section-number">10.8</span> Systematic Information<a href="differential-analysis.html#systematic-information-7" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb380"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb380-1"><a href="differential-analysis.html#cb380-1" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 4.1.3 (2022-03-10)
##  os       macOS Monterey 12.2.1
##  system   x86_64, darwin17.0
##  ui       RStudio
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       Asia/Shanghai
##  date     2023-11-30
##  rstudio  2023.09.0+463 Desert Sunflower (desktop)
##  pandoc   3.1.1 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/ (via rmarkdown)
## 
## ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package              * version   date (UTC) lib source
##  abind                  1.4-5     2016-07-21 [2] CRAN (R 4.1.0)
##  ade4                   1.7-22    2023-02-06 [2] CRAN (R 4.1.2)
##  ALDEx2                 1.30.0    2022-11-01 [2] Bioconductor
##  annotate               1.72.0    2021-10-26 [2] Bioconductor
##  AnnotationDbi          1.60.2    2023-03-10 [2] Bioconductor
##  ape                  * 5.7-1     2023-03-13 [2] CRAN (R 4.1.2)
##  askpass                1.1       2019-01-13 [2] CRAN (R 4.1.0)
##  backports              1.4.1     2021-12-13 [2] CRAN (R 4.1.0)
##  base64enc              0.1-3     2015-07-28 [2] CRAN (R 4.1.0)
##  bayesm                 3.1-5     2022-12-02 [2] CRAN (R 4.1.2)
##  Biobase                2.54.0    2021-10-26 [2] Bioconductor
##  BiocGenerics           0.40.0    2021-10-26 [2] Bioconductor
##  BiocParallel           1.28.3    2021-12-09 [2] Bioconductor
##  biomformat             1.22.0    2021-10-26 [2] Bioconductor
##  Biostrings             2.62.0    2021-10-26 [2] Bioconductor
##  bit                    4.0.5     2022-11-15 [2] CRAN (R 4.1.2)
##  bit64                  4.0.5     2020-08-30 [2] CRAN (R 4.1.0)
##  bitops                 1.0-7     2021-04-24 [2] CRAN (R 4.1.0)
##  blob                   1.2.4     2023-03-17 [2] CRAN (R 4.1.2)
##  bookdown               0.34      2023-05-09 [2] CRAN (R 4.1.2)
##  broom                  1.0.5     2023-06-09 [2] CRAN (R 4.1.3)
##  bslib                  0.6.0     2023-11-21 [1] CRAN (R 4.1.3)
##  cachem                 1.0.8     2023-05-01 [2] CRAN (R 4.1.2)
##  callr                  3.7.3     2022-11-02 [2] CRAN (R 4.1.2)
##  car                    3.1-2     2023-03-30 [2] CRAN (R 4.1.2)
##  carData                3.0-5     2022-01-06 [2] CRAN (R 4.1.2)
##  caTools                1.18.2    2021-03-28 [2] CRAN (R 4.1.0)
##  checkmate              2.2.0     2023-04-27 [2] CRAN (R 4.1.2)
##  cli                    3.6.1     2023-03-23 [2] CRAN (R 4.1.2)
##  cluster                2.1.4     2022-08-22 [2] CRAN (R 4.1.2)
##  coda                   0.19-4    2020-09-30 [2] CRAN (R 4.1.0)
##  codetools              0.2-19    2023-02-01 [2] CRAN (R 4.1.2)
##  coin                   1.4-2     2021-10-08 [2] CRAN (R 4.1.0)
##  colorspace             2.1-0     2023-01-23 [2] CRAN (R 4.1.2)
##  compositions           2.0-6     2023-04-13 [2] CRAN (R 4.1.2)
##  conflicted           * 1.2.0     2023-02-01 [2] CRAN (R 4.1.2)
##  corrplot               0.92      2021-11-18 [2] CRAN (R 4.1.0)
##  cowplot                1.1.1     2020-12-30 [2] CRAN (R 4.1.0)
##  crayon                 1.5.2     2022-09-29 [2] CRAN (R 4.1.2)
##  crosstalk              1.2.0     2021-11-04 [2] CRAN (R 4.1.0)
##  data.table             1.14.8    2023-02-17 [2] CRAN (R 4.1.2)
##  DBI                    1.1.3     2022-06-18 [2] CRAN (R 4.1.2)
##  DelayedArray           0.20.0    2021-10-26 [2] Bioconductor
##  DEoptimR               1.0-14    2023-06-09 [2] CRAN (R 4.1.3)
##  DESeq2                 1.34.0    2021-10-26 [2] Bioconductor
##  devtools               2.4.5     2022-10-11 [2] CRAN (R 4.1.2)
##  digest                 0.6.33    2023-07-07 [1] CRAN (R 4.1.3)
##  dplyr                * 1.1.2     2023-04-20 [2] CRAN (R 4.1.2)
##  DT                     0.28      2023-05-18 [2] CRAN (R 4.1.3)
##  edgeR                  3.36.0    2021-10-26 [2] Bioconductor
##  ellipsis               0.3.2     2021-04-29 [2] CRAN (R 4.1.0)
##  emmeans                1.8.7     2023-06-23 [1] CRAN (R 4.1.3)
##  estimability           1.4.1     2022-08-05 [2] CRAN (R 4.1.2)
##  evaluate               0.21      2023-05-05 [2] CRAN (R 4.1.2)
##  FactoMineR             2.8       2023-03-27 [2] CRAN (R 4.1.2)
##  fansi                  1.0.4     2023-01-22 [2] CRAN (R 4.1.2)
##  farver                 2.1.1     2022-07-06 [2] CRAN (R 4.1.2)
##  fastmap                1.1.1     2023-02-24 [2] CRAN (R 4.1.2)
##  flashClust             1.01-2    2012-08-21 [2] CRAN (R 4.1.0)
##  foreach                1.5.2     2022-02-02 [2] CRAN (R 4.1.2)
##  foreign                0.8-84    2022-12-06 [2] CRAN (R 4.1.2)
##  Formula                1.2-5     2023-02-24 [2] CRAN (R 4.1.2)
##  fs                     1.6.2     2023-04-25 [2] CRAN (R 4.1.2)
##  genefilter             1.76.0    2021-10-26 [2] Bioconductor
##  geneplotter            1.72.0    2021-10-26 [2] Bioconductor
##  generics               0.1.3     2022-07-05 [2] CRAN (R 4.1.2)
##  GenomeInfoDb           1.30.1    2022-01-30 [2] Bioconductor
##  GenomeInfoDbData       1.2.7     2022-03-09 [2] Bioconductor
##  GenomicRanges          1.46.1    2021-11-18 [2] Bioconductor
##  ggiraph                0.8.7     2023-03-17 [2] CRAN (R 4.1.2)
##  ggiraphExtra           0.3.0     2020-10-06 [2] CRAN (R 4.1.2)
##  ggplot2              * 3.4.2     2023-04-03 [2] CRAN (R 4.1.2)
##  ggpubr               * 0.6.0     2023-02-10 [2] CRAN (R 4.1.2)
##  ggrepel                0.9.3     2023-02-03 [2] CRAN (R 4.1.2)
##  ggsci                  3.0.0     2023-03-08 [2] CRAN (R 4.1.2)
##  ggsignif               0.6.4     2022-10-13 [2] CRAN (R 4.1.2)
##  glmnet                 4.1-7     2023-03-23 [2] CRAN (R 4.1.2)
##  glue                   1.6.2     2022-02-24 [2] CRAN (R 4.1.2)
##  gplots                 3.1.3     2022-04-25 [2] CRAN (R 4.1.2)
##  gridExtra              2.3       2017-09-09 [2] CRAN (R 4.1.0)
##  gtable                 0.3.3     2023-03-21 [2] CRAN (R 4.1.2)
##  gtools                 3.9.4     2022-11-27 [2] CRAN (R 4.1.2)
##  highr                  0.10      2022-12-22 [2] CRAN (R 4.1.2)
##  Hmisc                  5.1-0     2023-05-08 [2] CRAN (R 4.1.2)
##  htmlTable              2.4.1     2022-07-07 [2] CRAN (R 4.1.2)
##  htmltools              0.5.7     2023-11-03 [1] CRAN (R 4.1.3)
##  htmlwidgets            1.6.2     2023-03-17 [2] CRAN (R 4.1.2)
##  httpuv                 1.6.11    2023-05-11 [2] CRAN (R 4.1.3)
##  httr                   1.4.6     2023-05-08 [2] CRAN (R 4.1.2)
##  igraph                 1.5.0     2023-06-16 [1] CRAN (R 4.1.3)
##  insight                0.19.3    2023-06-29 [2] CRAN (R 4.1.3)
##  IRanges                2.28.0    2021-10-26 [2] Bioconductor
##  iterators              1.0.14    2022-02-05 [2] CRAN (R 4.1.2)
##  jquerylib              0.1.4     2021-04-26 [2] CRAN (R 4.1.0)
##  jsonlite               1.8.7     2023-06-29 [2] CRAN (R 4.1.3)
##  kableExtra             1.3.4     2021-02-20 [2] CRAN (R 4.1.2)
##  KEGGREST               1.34.0    2021-10-26 [2] Bioconductor
##  KernSmooth             2.23-22   2023-07-10 [2] CRAN (R 4.1.3)
##  knitr                  1.43      2023-05-25 [2] CRAN (R 4.1.3)
##  labeling               0.4.2     2020-10-20 [2] CRAN (R 4.1.0)
##  later                  1.3.1     2023-05-02 [2] CRAN (R 4.1.2)
##  lattice              * 0.21-8    2023-04-05 [2] CRAN (R 4.1.2)
##  leaps                  3.1       2020-01-16 [2] CRAN (R 4.1.0)
##  libcoin                1.0-9     2021-09-27 [2] CRAN (R 4.1.0)
##  lifecycle              1.0.3     2022-10-07 [2] CRAN (R 4.1.2)
##  limma                  3.50.3    2022-04-07 [2] Bioconductor
##  locfit                 1.5-9.8   2023-06-11 [2] CRAN (R 4.1.3)
##  LOCOM                  1.1       2022-08-05 [2] Github (yijuanhu/LOCOM@c181e0f)
##  magrittr               2.0.3     2022-03-30 [2] CRAN (R 4.1.2)
##  MASS                   7.3-60    2023-05-04 [2] CRAN (R 4.1.2)
##  Matrix                 1.6-0     2023-07-08 [2] CRAN (R 4.1.3)
##  MatrixGenerics         1.6.0     2021-10-26 [2] Bioconductor
##  matrixStats            1.0.0     2023-06-02 [2] CRAN (R 4.1.3)
##  mbzinb                 0.2       2022-03-16 [2] local
##  memoise                2.0.1     2021-11-26 [2] CRAN (R 4.1.0)
##  metagenomeSeq          1.36.0    2021-10-26 [2] Bioconductor
##  mgcv                   1.8-42    2023-03-02 [2] CRAN (R 4.1.2)
##  microbiome             1.16.0    2021-10-26 [2] Bioconductor
##  mime                   0.12      2021-09-28 [2] CRAN (R 4.1.0)
##  miniUI                 0.1.1.1   2018-05-18 [2] CRAN (R 4.1.0)
##  modeltools             0.2-23    2020-03-05 [2] CRAN (R 4.1.0)
##  multcomp               1.4-25    2023-06-20 [2] CRAN (R 4.1.3)
##  multcompView           0.1-9     2023-04-09 [2] CRAN (R 4.1.2)
##  multtest               2.50.0    2021-10-26 [2] Bioconductor
##  munsell                0.5.0     2018-06-12 [2] CRAN (R 4.1.0)
##  mvtnorm                1.2-2     2023-06-08 [2] CRAN (R 4.1.3)
##  mycor                  0.1.1     2018-04-10 [2] CRAN (R 4.1.0)
##  NADA                   1.6-1.1   2020-03-22 [2] CRAN (R 4.1.0)
##  nlme                 * 3.1-162   2023-01-31 [2] CRAN (R 4.1.2)
##  nnet                   7.3-19    2023-05-03 [2] CRAN (R 4.1.2)
##  openssl                2.0.6     2023-03-09 [2] CRAN (R 4.1.2)
##  permute              * 0.9-7     2022-01-27 [2] CRAN (R 4.1.2)
##  pheatmap               1.0.12    2019-01-04 [2] CRAN (R 4.1.0)
##  phyloseq             * 1.38.0    2021-10-26 [2] Bioconductor
##  picante              * 1.8.2     2020-06-10 [2] CRAN (R 4.1.0)
##  pillar                 1.9.0     2023-03-22 [2] CRAN (R 4.1.2)
##  pkgbuild               1.4.2     2023-06-26 [2] CRAN (R 4.1.3)
##  pkgconfig              2.0.3     2019-09-22 [2] CRAN (R 4.1.0)
##  pkgload                1.3.2.1   2023-07-08 [2] CRAN (R 4.1.3)
##  plyr                   1.8.8     2022-11-11 [2] CRAN (R 4.1.2)
##  png                    0.1-8     2022-11-29 [2] CRAN (R 4.1.2)
##  ppcor                  1.1       2015-12-03 [2] CRAN (R 4.1.0)
##  prettyunits            1.1.1     2020-01-24 [2] CRAN (R 4.1.0)
##  processx               3.8.2     2023-06-30 [2] CRAN (R 4.1.3)
##  profvis                0.3.8     2023-05-02 [2] CRAN (R 4.1.2)
##  promises               1.2.0.1   2021-02-11 [2] CRAN (R 4.1.0)
##  protoclust             1.6.4     2022-04-01 [2] CRAN (R 4.1.2)
##  ps                     1.7.5     2023-04-18 [2] CRAN (R 4.1.2)
##  pscl                   1.5.5.1   2023-05-10 [2] CRAN (R 4.1.2)
##  purrr                  1.0.1     2023-01-10 [2] CRAN (R 4.1.2)
##  qvalue                 2.26.0    2021-10-26 [2] Bioconductor
##  R6                     2.5.1     2021-08-19 [2] CRAN (R 4.1.0)
##  RAIDA                  1.0       2022-03-14 [2] local
##  RColorBrewer         * 1.1-3     2022-04-03 [2] CRAN (R 4.1.2)
##  Rcpp                   1.0.11    2023-07-06 [1] CRAN (R 4.1.3)
##  RcppZiggurat           0.1.6     2020-10-20 [2] CRAN (R 4.1.0)
##  RCurl                  1.98-1.12 2023-03-27 [2] CRAN (R 4.1.2)
##  remotes                2.4.2     2021-11-30 [2] CRAN (R 4.1.0)
##  reshape2               1.4.4     2020-04-09 [2] CRAN (R 4.1.0)
##  reticulate             1.30      2023-06-09 [2] CRAN (R 4.1.3)
##  Rfast                  2.0.8     2023-07-03 [2] CRAN (R 4.1.3)
##  rhdf5                  2.38.1    2022-03-10 [2] Bioconductor
##  rhdf5filters           1.6.0     2021-10-26 [2] Bioconductor
##  Rhdf5lib               1.16.0    2021-10-26 [2] Bioconductor
##  rlang                  1.1.1     2023-04-28 [1] CRAN (R 4.1.2)
##  rmarkdown              2.23      2023-07-01 [2] CRAN (R 4.1.3)
##  robustbase             0.99-0    2023-06-16 [2] CRAN (R 4.1.3)
##  rpart                  4.1.19    2022-10-21 [2] CRAN (R 4.1.2)
##  RSpectra               0.16-1    2022-04-24 [2] CRAN (R 4.1.2)
##  RSQLite                2.3.1     2023-04-03 [2] CRAN (R 4.1.2)
##  rstatix                0.7.2     2023-02-01 [2] CRAN (R 4.1.2)
##  rstudioapi             0.15.0    2023-07-07 [2] CRAN (R 4.1.3)
##  Rtsne                  0.16      2022-04-17 [2] CRAN (R 4.1.2)
##  rvest                  1.0.3     2022-08-19 [2] CRAN (R 4.1.2)
##  S4Vectors              0.32.4    2022-03-29 [2] Bioconductor
##  sandwich               3.0-2     2022-06-15 [2] CRAN (R 4.1.2)
##  sass                   0.4.6     2023-05-03 [2] CRAN (R 4.1.2)
##  scales                 1.2.1     2022-08-20 [2] CRAN (R 4.1.2)
##  scatterplot3d          0.3-44    2023-05-05 [2] CRAN (R 4.1.2)
##  sessioninfo            1.2.2     2021-12-06 [2] CRAN (R 4.1.0)
##  shape                  1.4.6     2021-05-19 [2] CRAN (R 4.1.0)
##  shiny                  1.7.4.1   2023-07-06 [2] CRAN (R 4.1.3)
##  sjlabelled             1.2.0     2022-04-10 [2] CRAN (R 4.1.2)
##  sjmisc                 2.8.9     2021-12-03 [2] CRAN (R 4.1.0)
##  stringi                1.7.12    2023-01-11 [2] CRAN (R 4.1.2)
##  stringr                1.5.0     2022-12-02 [2] CRAN (R 4.1.2)
##  SummarizedExperiment   1.24.0    2021-10-26 [2] Bioconductor
##  survival               3.5-5     2023-03-12 [2] CRAN (R 4.1.2)
##  svglite                2.1.1     2023-01-10 [2] CRAN (R 4.1.2)
##  systemfonts            1.0.4     2022-02-11 [2] CRAN (R 4.1.2)
##  tensorA                0.36.2    2020-11-19 [2] CRAN (R 4.1.0)
##  TH.data                1.1-2     2023-04-17 [2] CRAN (R 4.1.2)
##  tibble               * 3.2.1     2023-03-20 [2] CRAN (R 4.1.2)
##  tidyr                  1.3.0     2023-01-24 [2] CRAN (R 4.1.2)
##  tidyselect             1.2.0     2022-10-10 [2] CRAN (R 4.1.2)
##  truncnorm              1.0-9     2023-03-20 [2] CRAN (R 4.1.2)
##  umap                   0.2.10.0  2023-02-01 [2] CRAN (R 4.1.2)
##  urlchecker             1.0.1     2021-11-30 [2] CRAN (R 4.1.0)
##  usethis                2.2.2     2023-07-06 [2] CRAN (R 4.1.3)
##  utf8                   1.2.3     2023-01-31 [2] CRAN (R 4.1.2)
##  uuid                   1.1-0     2022-04-19 [2] CRAN (R 4.1.2)
##  vctrs                  0.6.3     2023-06-14 [1] CRAN (R 4.1.3)
##  vegan                * 2.6-4     2022-10-11 [2] CRAN (R 4.1.2)
##  viridis              * 0.6.3     2023-05-03 [2] CRAN (R 4.1.2)
##  viridisLite          * 0.4.2     2023-05-02 [2] CRAN (R 4.1.2)
##  webshot                0.5.5     2023-06-26 [2] CRAN (R 4.1.3)
##  withr                  2.5.0     2022-03-03 [2] CRAN (R 4.1.2)
##  Wrench                 1.12.0    2021-10-26 [2] Bioconductor
##  xfun                   0.40      2023-08-09 [1] CRAN (R 4.1.3)
##  XMAS2                * 2.2.0     2023-11-30 [1] local
##  XML                    3.99-0.14 2023-03-19 [2] CRAN (R 4.1.2)
##  xml2                   1.3.5     2023-07-06 [2] CRAN (R 4.1.3)
##  xtable                 1.8-4     2019-04-21 [2] CRAN (R 4.1.0)
##  XVector                0.34.0    2021-10-26 [2] Bioconductor
##  yaml                   2.3.7     2023-01-23 [2] CRAN (R 4.1.2)
##  zCompositions          1.4.0-1   2022-03-26 [2] CRAN (R 4.1.2)
##  zlibbioc               1.40.0    2021-10-26 [2] Bioconductor
##  zoo                    1.8-12    2023-04-13 [2] CRAN (R 4.1.2)
## 
##  [1] /Users/zouhua/Library/R/x86_64/4.1/library
##  [2] /Library/Frameworks/R.framework/Versions/4.1/Resources/library
## 
## ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-chen2018omnibus" class="csl-entry">
Chen, Jun, Emily King, Rebecca Deek, Zhi Wei, Yue Yu, Diane Grill, and Karla Ballman. 2018. <span>“An Omnibus Test for Differential Distribution Analysis of Microbiome Sequencing Data.”</span> <em>Bioinformatics</em> 34 (4): 643–51.
</div>
<div id="ref-chen2018gmpr" class="csl-entry">
Chen, Li, James Reeve, Lujun Zhang, Shengbing Huang, Xuefeng Wang, and Jun Chen. 2018. <span>“GMPR: A Robust Normalization Method for Zero-Inflated Count Data with Application to Microbiome Sequencing Data.”</span> <em>PeerJ</em> 6: e4600.
</div>
<div id="ref-fernandes2014unifying" class="csl-entry">
Fernandes, Andrew D, Jennifer Ns Reid, Jean M Macklaim, Thomas A McMurrough, David R Edgell, and Gregory B Gloor. 2014. <span>“Unifying the Analysis of High-Throughput Sequencing Datasets: Characterizing RNA-Seq, 16S rRNA Gene Sequencing and Selective Growth Experiments by Compositional Data Analysis.”</span> <em>Microbiome</em> 2 (1): 1–13.
</div>
<div id="ref-hu2022locom" class="csl-entry">
Hu, Yingtian, Glen A Satten, and Yi-Juan Hu. 2022. <span>“LOCOM: A Logistic Regression Model for Testing Differential Abundance in Compositional Microbiome Data with False Discovery Rate Control.”</span> <em>Proceedings of the National Academy of Sciences</em> 119 (30): e2122788119.
</div>
<div id="ref-law2014voom" class="csl-entry">
Law, Charity W, Yunshun Chen, Wei Shi, and Gordon K Smyth. 2014. <span>“Voom: Precision Weights Unlock Linear Model Analysis Tools for RNA-Seq Read Counts.”</span> <em>Genome Biology</em> 15 (2): 1–17.
</div>
<div id="ref-love2014moderated" class="csl-entry">
Love, Michael I, Wolfgang Huber, and Simon Anders. 2014. <span>“Moderated Estimation of Fold Change and Dispersion for RNA-Seq Data with DESeq2.”</span> <em>Genome Biology</em> 15 (12): 1–21.
</div>
<div id="ref-mallick2021multivariable" class="csl-entry">
Mallick, Himel, Ali Rahnavard, Lauren J McIver, Siyuan Ma, Yancong Zhang, Long H Nguyen, Timothy L Tickle, et al. 2021. <span>“Multivariable Association Discovery in Population-Scale Meta-Omics Studies.”</span> <em>PLoS Computational Biology</em> 17 (11): e1009442.
</div>
<div id="ref-mandal2015analysis" class="csl-entry">
Mandal, Siddhartha, Will Van Treuren, Richard A White, Merete Eggesbø, Rob Knight, and Shyamal D Peddada. 2015. <span>“Analysis of Composition of Microbiomes: A Novel Method for Studying Microbial Composition.”</span> <em>Microbial Ecology in Health and Disease</em> 26 (1): 27663.
</div>
<div id="ref-martin2020modeling" class="csl-entry">
Martin, Bryan D, Daniela Witten, and Amy D Willis. 2020. <span>“Modeling Microbial Abundances and Dysbiosis with Beta-Binomial Regression.”</span> <em>The Annals of Applied Statistics</em> 14 (1): 94.
</div>
<div id="ref-nearing2022microbiome" class="csl-entry">
Nearing, Jacob T, Gavin M Douglas, Molly G Hayes, Jocelyn MacDonald, Dhwani K Desai, Nicole Allward, Casey Jones, et al. 2022. <span>“Microbiome Differential Abundance Methods Produce Different Results Across 38 Datasets.”</span> <em>Nature Communications</em> 13 (1): 1–16.
</div>
<div id="ref-paulson2013differential" class="csl-entry">
Paulson, Joseph N, O Colin Stine, Héctor Corrada Bravo, and Mihai Pop. 2013. <span>“Differential Abundance Analysis for Microbial Marker-Gene Surveys.”</span> <em>Nature Methods</em> 10 (12): 1200–1202.
</div>
<div id="ref-robinson2010scaling" class="csl-entry">
Robinson, Mark D, and Alicia Oshlack. 2010. <span>“A Scaling Normalization Method for Differential Expression Analysis of RNA-Seq Data.”</span> <em>Genome Biology</em> 11 (3): 1–9.
</div>
<div id="ref-segata2011metagenomic" class="csl-entry">
Segata, Nicola, Jacques Izard, Levi Waldron, Dirk Gevers, Larisa Miropolsky, Wendy S Garrett, and Curtis Huttenhower. 2011. <span>“Metagenomic Biomarker Discovery and Explanation.”</span> <em>Genome Biology</em> 12 (6): 1–18.
</div>
<div id="ref-sohn2015robust" class="csl-entry">
Sohn, Michael B, Ruofei Du, and Lingling An. 2015. <span>“A Robust Approach for Identifying Differentially Abundant Features in Metagenomic Samples.”</span> <em>Bioinformatics</em> 31 (14): 2269–75.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="core-microbiota.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="visualization-5.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/XMAS_tutorial/blob/main/09-Differentail-Analysis.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["XMAS 2.0 Tutorial.pdf", "XMAS 2.0 Tutorial.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true,
"toc_depth": 3
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
