<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 14 Examples | XMAS 2.0 Tutorial</title>
  <meta name="description" content="A tutorial of XMAS 2.0 package for in-house training" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 14 Examples | XMAS 2.0 Tutorial" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 14 Examples | XMAS 2.0 Tutorial" />
  
  <meta name="twitter:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

<meta name="author" content="Hua Zou" />


<meta name="date" content="2023-11-30" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="principals-of-differential-analysis-methods.html"/>
<link rel="next" href="references.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.28/datatables.js"></script>
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="assets/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">XMAS 2.0 Tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#overview-of-the-design-of-xmas"><i class="fa fa-check"></i><b>1.1</b> Overview of the design of <strong>XMAS</strong></a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#input-data"><i class="fa fa-check"></i><b>1.2</b> Input data</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#methods"><i class="fa fa-check"></i><b>1.3</b> Methods</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#outline-of-this-tutorial"><i class="fa fa-check"></i><b>1.4</b> Outline of this Tutorial</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#key-publications"><i class="fa fa-check"></i><b>1.5</b> Key publications</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#citation"><i class="fa fa-check"></i><b>1.6</b> Citation</a></li>
<li class="chapter" data-level="1.7" data-path="index.html"><a href="index.html#change-log"><i class="fa fa-check"></i><b>1.7</b> Change log</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html"><i class="fa fa-check"></i><b>2</b> How to install XMAS</a>
<ul>
<li class="chapter" data-level="2.1" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#description"><i class="fa fa-check"></i><b>2.1</b> Description</a></li>
<li class="chapter" data-level="2.2" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#installation-of-dependencies"><i class="fa fa-check"></i><b>2.2</b> Installation of dependencies</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#packages-in-cran-bioconductor"><i class="fa fa-check"></i><b>2.2.1</b> Packages in CRAN &amp; Bioconductor</a></li>
<li class="chapter" data-level="2.2.2" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#packages-in-github"><i class="fa fa-check"></i><b>2.2.2</b> Packages in github</a></li>
<li class="chapter" data-level="2.2.3" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#packages-from-website"><i class="fa fa-check"></i><b>2.2.3</b> Packages from website</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#installation"><i class="fa fa-check"></i><b>2.3</b> Installation</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#installation-in-linux-or-mac-os-through-command-line"><i class="fa fa-check"></i><b>2.3.1</b> Installation in linux or Mac os through command line</a></li>
<li class="chapter" data-level="2.3.2" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#installation-by-r-from-gitlab"><i class="fa fa-check"></i><b>2.3.2</b> Installation by R from gitlab</a></li>
<li class="chapter" data-level="2.3.3" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#installation-by-r-from-local-xmas.tar.gz-file"><i class="fa fa-check"></i><b>2.3.3</b> Installation by R from local XMAS*.tar.gz file</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#vignette"><i class="fa fa-check"></i><b>2.4</b> Vignette</a></li>
<li class="chapter" data-level="2.5" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#authors"><i class="fa fa-check"></i><b>2.5</b> Authors</a></li>
<li class="chapter" data-level="2.6" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#future-plan"><i class="fa fa-check"></i><b>2.6</b> Future plan</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html"><i class="fa fa-check"></i><b>3</b> Convert inputdata into phyloseq object</a>
<ul>
<li class="chapter" data-level="3.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#loading-packages"><i class="fa fa-check"></i><b>3.1</b> Loading Packages</a></li>
<li class="chapter" data-level="3.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-data"><i class="fa fa-check"></i><b>3.2</b> Importing Data</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#result-from-dada2"><i class="fa fa-check"></i><b>3.2.1</b> Result from dada2</a></li>
<li class="chapter" data-level="3.2.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#result-from-metaphlan"><i class="fa fa-check"></i><b>3.2.2</b> Result from Metaphlan</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#systematic-information"><i class="fa fa-check"></i><b>3.3</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="quality-evaluation.html"><a href="quality-evaluation.html"><i class="fa fa-check"></i><b>4</b> Quality Evaluation</a>
<ul>
<li class="chapter" data-level="4.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#loading-packages-1"><i class="fa fa-check"></i><b>4.1</b> Loading Packages</a></li>
<li class="chapter" data-level="4.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#importing-data-1"><i class="fa fa-check"></i><b>4.2</b> Importing Data</a></li>
<li class="chapter" data-level="4.3" data-path="quality-evaluation.html"><a href="quality-evaluation.html#reads-track"><i class="fa fa-check"></i><b>4.3</b> Reads track</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#plotting-dad2"><i class="fa fa-check"></i><b>4.3.1</b> Plotting DAD2</a></li>
<li class="chapter" data-level="4.3.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#plotting-metaphlan"><i class="fa fa-check"></i><b>4.3.2</b> Plotting Metaphlan</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="quality-evaluation.html"><a href="quality-evaluation.html#spike-in-brs-sample-assessment"><i class="fa fa-check"></i><b>4.4</b> Spike-in (BRS) sample assessment</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#s-data"><i class="fa fa-check"></i><b>4.4.1</b> 16s data</a></li>
<li class="chapter" data-level="4.4.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#metagenomic-data"><i class="fa fa-check"></i><b>4.4.2</b> Metagenomic data</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="quality-evaluation.html"><a href="quality-evaluation.html#removing-brs"><i class="fa fa-check"></i><b>4.5</b> Removing BRS</a></li>
<li class="chapter" data-level="4.6" data-path="quality-evaluation.html"><a href="quality-evaluation.html#rarefaction-curves"><i class="fa fa-check"></i><b>4.6</b> Rarefaction Curves</a></li>
<li class="chapter" data-level="4.7" data-path="quality-evaluation.html"><a href="quality-evaluation.html#summarize-phyloseq-class-object"><i class="fa fa-check"></i><b>4.7</b> Summarize phyloseq-class object</a></li>
<li class="chapter" data-level="4.8" data-path="quality-evaluation.html"><a href="quality-evaluation.html#variability"><i class="fa fa-check"></i><b>4.8</b> Variability</a>
<ul>
<li class="chapter" data-level="4.8.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#coefficient-of-variation"><i class="fa fa-check"></i><b>4.8.1</b> Coefficient of variation</a></li>
<li class="chapter" data-level="4.8.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#distribution-of-number-of-sequences"><i class="fa fa-check"></i><b>4.8.2</b> Distribution of number of sequences</a></li>
</ul></li>
<li class="chapter" data-level="4.9" data-path="quality-evaluation.html"><a href="quality-evaluation.html#systematic-information-1"><i class="fa fa-check"></i><b>4.9</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="pre-processing.html"><a href="pre-processing.html"><i class="fa fa-check"></i><b>5</b> Pre-processing</a>
<ul>
<li class="chapter" data-level="5.1" data-path="pre-processing.html"><a href="pre-processing.html#loading-packages-2"><i class="fa fa-check"></i><b>5.1</b> Loading Packages</a></li>
<li class="chapter" data-level="5.2" data-path="pre-processing.html"><a href="pre-processing.html#importing-data-2"><i class="fa fa-check"></i><b>5.2</b> Importing Data</a></li>
<li class="chapter" data-level="5.3" data-path="pre-processing.html"><a href="pre-processing.html#transformation"><i class="fa fa-check"></i><b>5.3</b> Transformation</a></li>
<li class="chapter" data-level="5.4" data-path="pre-processing.html"><a href="pre-processing.html#normalization"><i class="fa fa-check"></i><b>5.4</b> Normalization</a></li>
<li class="chapter" data-level="5.5" data-path="pre-processing.html"><a href="pre-processing.html#filtering"><i class="fa fa-check"></i><b>5.5</b> Filtering</a></li>
<li class="chapter" data-level="5.6" data-path="pre-processing.html"><a href="pre-processing.html#trimming"><i class="fa fa-check"></i><b>5.6</b> Trimming</a></li>
<li class="chapter" data-level="5.7" data-path="pre-processing.html"><a href="pre-processing.html#imputation"><i class="fa fa-check"></i><b>5.7</b> Imputation</a></li>
<li class="chapter" data-level="5.8" data-path="pre-processing.html"><a href="pre-processing.html#extracting-specific-taxa-phyloseq-class-object"><i class="fa fa-check"></i><b>5.8</b> Extracting specific taxa phyloseq-class object</a></li>
<li class="chapter" data-level="5.9" data-path="pre-processing.html"><a href="pre-processing.html#aggregating-low-relative-abundance-or-unclassified-taxa-into-others"><i class="fa fa-check"></i><b>5.9</b> Aggregating low relative abundance or unclassified taxa into others</a></li>
<li class="chapter" data-level="5.10" data-path="pre-processing.html"><a href="pre-processing.html#transform-the-abundance-of-taxa-whose-relative-abundance-under-limit-of-detection-lod-into-zeros-only-in-metaphlan23"><i class="fa fa-check"></i><b>5.10</b> Transform the abundance of taxa whose relative abundance under Limit Of Detection (LOD) into Zeros (only in metaphlan2/3)</a></li>
<li class="chapter" data-level="5.11" data-path="pre-processing.html"><a href="pre-processing.html#systematic-information-2"><i class="fa fa-check"></i><b>5.11</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="aplha-diversity.html"><a href="aplha-diversity.html"><i class="fa fa-check"></i><b>6</b> Aplha diversity</a>
<ul>
<li class="chapter" data-level="6.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#loading-packages-3"><i class="fa fa-check"></i><b>6.1</b> Loading Packages</a></li>
<li class="chapter" data-level="6.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#importing-data-3"><i class="fa fa-check"></i><b>6.2</b> Importing Data</a></li>
<li class="chapter" data-level="6.3" data-path="aplha-diversity.html"><a href="aplha-diversity.html#equal-sample-sums"><i class="fa fa-check"></i><b>6.3</b> Equal sample sums</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#dada2_ps-dataset"><i class="fa fa-check"></i><b>6.3.1</b> dada2_ps dataset</a></li>
<li class="chapter" data-level="6.3.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#amplicon_ps-dataset"><i class="fa fa-check"></i><b>6.3.2</b> amplicon_ps dataset</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="aplha-diversity.html"><a href="aplha-diversity.html#non-phylogenetic-diversities"><i class="fa fa-check"></i><b>6.4</b> Non-phylogenetic diversities</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#xmas2-r-package"><i class="fa fa-check"></i><b>6.4.1</b> XMAS2 R package</a></li>
<li class="chapter" data-level="6.4.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#microbiome-r-package"><i class="fa fa-check"></i><b>6.4.2</b> microbiome R package</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="aplha-diversity.html"><a href="aplha-diversity.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>6.5</b> Phylogenetic diversity</a></li>
<li class="chapter" data-level="6.6" data-path="aplha-diversity.html"><a href="aplha-diversity.html#rank-abundance"><i class="fa fa-check"></i><b>6.6</b> Rank Abundance</a></li>
<li class="chapter" data-level="6.7" data-path="aplha-diversity.html"><a href="aplha-diversity.html#evenness"><i class="fa fa-check"></i><b>6.7</b> Evenness</a></li>
<li class="chapter" data-level="6.8" data-path="aplha-diversity.html"><a href="aplha-diversity.html#systematic-information-3"><i class="fa fa-check"></i><b>6.8</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="beta-diversity.html"><a href="beta-diversity.html"><i class="fa fa-check"></i><b>7</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="beta-diversity.html"><a href="beta-diversity.html#loading-packages-4"><i class="fa fa-check"></i><b>7.1</b> Loading Packages</a></li>
<li class="chapter" data-level="7.2" data-path="beta-diversity.html"><a href="beta-diversity.html#importing-data-4"><i class="fa fa-check"></i><b>7.2</b> Importing Data</a></li>
<li class="chapter" data-level="7.3" data-path="beta-diversity.html"><a href="beta-diversity.html#phylogenetic-beta-diversity-metrics"><i class="fa fa-check"></i><b>7.3</b> Phylogenetic beta-diversity metrics</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="beta-diversity.html"><a href="beta-diversity.html#unweighted-unifrac"><i class="fa fa-check"></i><b>7.3.1</b> Unweighted Unifrac</a></li>
<li class="chapter" data-level="7.3.2" data-path="beta-diversity.html"><a href="beta-diversity.html#weighted-unifrac"><i class="fa fa-check"></i><b>7.3.2</b> Weighted Unifrac</a></li>
<li class="chapter" data-level="7.3.3" data-path="beta-diversity.html"><a href="beta-diversity.html#generalized-unifrac-distance-metrics"><i class="fa fa-check"></i><b>7.3.3</b> Generalized UniFrac Distance Metrics</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="beta-diversity.html"><a href="beta-diversity.html#distance-dissimilarity-coefficients-bray-curtis-index"><i class="fa fa-check"></i><b>7.4</b> Distance (Dissimilarity) Coefficients: Bray-Curtis Index</a></li>
<li class="chapter" data-level="7.5" data-path="beta-diversity.html"><a href="beta-diversity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>7.5</b> Checking the homogeneity condition</a></li>
<li class="chapter" data-level="7.6" data-path="beta-diversity.html"><a href="beta-diversity.html#permutational-multivariate-analysis-of-variance-permanova"><i class="fa fa-check"></i><b>7.6</b> Permutational Multivariate Analysis of Variance (PERMANOVA)</a></li>
<li class="chapter" data-level="7.7" data-path="beta-diversity.html"><a href="beta-diversity.html#mantel-test"><i class="fa fa-check"></i><b>7.7</b> Mantel Test</a></li>
<li class="chapter" data-level="7.8" data-path="beta-diversity.html"><a href="beta-diversity.html#analysis-of-similarity-anosim"><i class="fa fa-check"></i><b>7.8</b> Analysis of Similarity (ANOSIM)</a></li>
<li class="chapter" data-level="7.9" data-path="beta-diversity.html"><a href="beta-diversity.html#multi-response-permutation-procedures-mrpp"><i class="fa fa-check"></i><b>7.9</b> Multi-response permutation procedures (MRPP)</a></li>
<li class="chapter" data-level="7.10" data-path="beta-diversity.html"><a href="beta-diversity.html#ordination"><i class="fa fa-check"></i><b>7.10</b> Ordination</a>
<ul>
<li class="chapter" data-level="7.10.1" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>7.10.1</b> Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="7.10.2" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-coordinate-analysis-pcoa"><i class="fa fa-check"></i><b>7.10.2</b> Principal Coordinate Analysis (PCoA)</a></li>
<li class="chapter" data-level="7.10.3" data-path="beta-diversity.html"><a href="beta-diversity.html#t-distributed-stochastic-neighbor-embedding-t-sne"><i class="fa fa-check"></i><b>7.10.3</b> t-distributed stochastic neighbor embedding (t-SNE)</a></li>
<li class="chapter" data-level="7.10.4" data-path="beta-diversity.html"><a href="beta-diversity.html#uniform-manifold-approximation-and-projection-for-dimension-reduction-umap"><i class="fa fa-check"></i><b>7.10.4</b> Uniform Manifold Approximation and Projection for Dimension Reduction (UMAP)</a></li>
</ul></li>
<li class="chapter" data-level="7.11" data-path="beta-diversity.html"><a href="beta-diversity.html#systematic-information-4"><i class="fa fa-check"></i><b>7.11</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="microbial-composition.html"><a href="microbial-composition.html"><i class="fa fa-check"></i><b>8</b> Microbial Composition</a>
<ul>
<li class="chapter" data-level="8.1" data-path="microbial-composition.html"><a href="microbial-composition.html#loading-packages-5"><i class="fa fa-check"></i><b>8.1</b> Loading Packages</a></li>
<li class="chapter" data-level="8.2" data-path="microbial-composition.html"><a href="microbial-composition.html#importing-data-5"><i class="fa fa-check"></i><b>8.2</b> Importing Data</a></li>
<li class="chapter" data-level="8.3" data-path="microbial-composition.html"><a href="microbial-composition.html#plot_stacked_bar-from-xviz"><i class="fa fa-check"></i><b>8.3</b> plot_stacked_bar from XVIZ</a></li>
<li class="chapter" data-level="8.4" data-path="microbial-composition.html"><a href="microbial-composition.html#plot_stackbarplot-from-xams2"><i class="fa fa-check"></i><b>8.4</b> plot_StackBarPlot from XAMS2</a></li>
<li class="chapter" data-level="8.5" data-path="microbial-composition.html"><a href="microbial-composition.html#heatmaps"><i class="fa fa-check"></i><b>8.5</b> Heatmaps</a></li>
<li class="chapter" data-level="8.6" data-path="microbial-composition.html"><a href="microbial-composition.html#systematic-information-5"><i class="fa fa-check"></i><b>8.6</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="core-microbiota.html"><a href="core-microbiota.html"><i class="fa fa-check"></i><b>9</b> Core microbiota</a>
<ul>
<li class="chapter" data-level="9.1" data-path="core-microbiota.html"><a href="core-microbiota.html#loading-packages-6"><i class="fa fa-check"></i><b>9.1</b> Loading Packages</a></li>
<li class="chapter" data-level="9.2" data-path="core-microbiota.html"><a href="core-microbiota.html#importing-data-6"><i class="fa fa-check"></i><b>9.2</b> Importing Data</a></li>
<li class="chapter" data-level="9.3" data-path="core-microbiota.html"><a href="core-microbiota.html#obtaining-core-microbiota"><i class="fa fa-check"></i><b>9.3</b> Obtaining core microbiota</a></li>
<li class="chapter" data-level="9.4" data-path="core-microbiota.html"><a href="core-microbiota.html#showing-core-abundance-and-diversity"><i class="fa fa-check"></i><b>9.4</b> Showing core abundance and diversity</a></li>
<li class="chapter" data-level="9.5" data-path="core-microbiota.html"><a href="core-microbiota.html#visualizing-core-taxa"><i class="fa fa-check"></i><b>9.5</b> Visualizing core taxa</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="core-microbiota.html"><a href="core-microbiota.html#heatmap"><i class="fa fa-check"></i><b>9.5.1</b> heatmap</a></li>
<li class="chapter" data-level="9.5.2" data-path="core-microbiota.html"><a href="core-microbiota.html#linechart"><i class="fa fa-check"></i><b>9.5.2</b> linechart</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="core-microbiota.html"><a href="core-microbiota.html#systematic-information-6"><i class="fa fa-check"></i><b>9.6</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="differential-analysis.html"><a href="differential-analysis.html"><i class="fa fa-check"></i><b>10</b> Differential Analysis</a>
<ul>
<li class="chapter" data-level="10.1" data-path="differential-analysis.html"><a href="differential-analysis.html#loading-packages-7"><i class="fa fa-check"></i><b>10.1</b> Loading Packages</a></li>
<li class="chapter" data-level="10.2" data-path="differential-analysis.html"><a href="differential-analysis.html#importing-data-7"><i class="fa fa-check"></i><b>10.2</b> Importing Data</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="differential-analysis.html"><a href="differential-analysis.html#s-data-1"><i class="fa fa-check"></i><b>10.2.1</b> 16s data</a></li>
<li class="chapter" data-level="10.2.2" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomic-data-1"><i class="fa fa-check"></i><b>10.2.2</b> Metagenomic data</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="differential-analysis.html"><a href="differential-analysis.html#amplicon-sequencing-microbial-data-16s"><i class="fa fa-check"></i><b>10.3</b> Amplicon sequencing microbial data (16s)</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="differential-analysis.html"><a href="differential-analysis.html#aldex2"><i class="fa fa-check"></i><b>10.3.1</b> ALDEx2</a></li>
<li class="chapter" data-level="10.3.2" data-path="differential-analysis.html"><a href="differential-analysis.html#limma_voom"><i class="fa fa-check"></i><b>10.3.2</b> limma_voom</a></li>
<li class="chapter" data-level="10.3.3" data-path="differential-analysis.html"><a href="differential-analysis.html#mbzinb"><i class="fa fa-check"></i><b>10.3.3</b> mbzinb</a></li>
<li class="chapter" data-level="10.3.4" data-path="differential-analysis.html"><a href="differential-analysis.html#omnibus"><i class="fa fa-check"></i><b>10.3.4</b> omnibus</a></li>
<li class="chapter" data-level="10.3.5" data-path="differential-analysis.html"><a href="differential-analysis.html#raida"><i class="fa fa-check"></i><b>10.3.5</b> RAIDA</a></li>
<li class="chapter" data-level="10.3.6" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests"><i class="fa fa-check"></i><b>10.3.6</b> Wilcoxon Rank Sum and Signed Rank Tests</a></li>
<li class="chapter" data-level="10.3.7" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse"><i class="fa fa-check"></i><b>10.3.7</b> Liner discriminant analysis (LDA) effect size (LEfSe)</a></li>
<li class="chapter" data-level="10.3.8" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test"><i class="fa fa-check"></i><b>10.3.8</b> t-test</a></li>
<li class="chapter" data-level="10.3.9" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomeseq"><i class="fa fa-check"></i><b>10.3.9</b> MetagenomeSeq</a></li>
<li class="chapter" data-level="10.3.10" data-path="differential-analysis.html"><a href="differential-analysis.html#deseq2"><i class="fa fa-check"></i><b>10.3.10</b> DESeq2</a></li>
<li class="chapter" data-level="10.3.11" data-path="differential-analysis.html"><a href="differential-analysis.html#edger"><i class="fa fa-check"></i><b>10.3.11</b> EdgeR</a></li>
<li class="chapter" data-level="10.3.12" data-path="differential-analysis.html"><a href="differential-analysis.html#ancom"><i class="fa fa-check"></i><b>10.3.12</b> ANCOM</a></li>
<li class="chapter" data-level="10.3.13" data-path="differential-analysis.html"><a href="differential-analysis.html#corncob"><i class="fa fa-check"></i><b>10.3.13</b> Corncob</a></li>
<li class="chapter" data-level="10.3.14" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models"><i class="fa fa-check"></i><b>10.3.14</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)</a></li>
<li class="chapter" data-level="10.3.15" data-path="differential-analysis.html"><a href="differential-analysis.html#locom"><i class="fa fa-check"></i><b>10.3.15</b> LOCOM</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomic-sequencing-microbial-data-metaphlan23"><i class="fa fa-check"></i><b>10.4</b> Metagenomic sequencing microbial data (metaphlan2/3)</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests-1"><i class="fa fa-check"></i><b>10.4.1</b> Wilcoxon Rank Sum and Signed Rank Tests</a></li>
<li class="chapter" data-level="10.4.2" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse-1"><i class="fa fa-check"></i><b>10.4.2</b> Liner discriminant analysis (LDA) effect size (LEfSe)</a></li>
<li class="chapter" data-level="10.4.3" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test-1"><i class="fa fa-check"></i><b>10.4.3</b> t-test</a></li>
<li class="chapter" data-level="10.4.4" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models-1"><i class="fa fa-check"></i><b>10.4.4</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="differential-analysis.html"><a href="differential-analysis.html#visualization-4"><i class="fa fa-check"></i><b>10.5</b> Visualization</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="differential-analysis.html"><a href="differential-analysis.html#volcano-plot"><i class="fa fa-check"></i><b>10.5.1</b> Volcano plot</a></li>
<li class="chapter" data-level="10.5.2" data-path="differential-analysis.html"><a href="differential-analysis.html#barplot-in-lefse"><i class="fa fa-check"></i><b>10.5.2</b> barplot in lefse</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="differential-analysis.html"><a href="differential-analysis.html#dominant-taxa"><i class="fa fa-check"></i><b>10.6</b> Dominant taxa</a></li>
<li class="chapter" data-level="10.7" data-path="differential-analysis.html"><a href="differential-analysis.html#multiple-differential-analysis-by-one-function"><i class="fa fa-check"></i><b>10.7</b> Multiple differential analysis by one function</a></li>
<li class="chapter" data-level="10.8" data-path="differential-analysis.html"><a href="differential-analysis.html#systematic-information-7"><i class="fa fa-check"></i><b>10.8</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="visualization-5.html"><a href="visualization-5.html"><i class="fa fa-check"></i><b>11</b> Visualization</a>
<ul>
<li class="chapter" data-level="11.1" data-path="visualization-5.html"><a href="visualization-5.html#loading-packages-8"><i class="fa fa-check"></i><b>11.1</b> Loading Packages</a></li>
<li class="chapter" data-level="11.2" data-path="visualization-5.html"><a href="visualization-5.html#importing-data-8"><i class="fa fa-check"></i><b>11.2</b> Importing Data</a></li>
<li class="chapter" data-level="11.3" data-path="visualization-5.html"><a href="visualization-5.html#plot_boxplot"><i class="fa fa-check"></i><b>11.3</b> plot_boxplot</a></li>
<li class="chapter" data-level="11.4" data-path="visualization-5.html"><a href="visualization-5.html#plot_barplot"><i class="fa fa-check"></i><b>11.4</b> plot_barplot</a></li>
<li class="chapter" data-level="11.5" data-path="visualization-5.html"><a href="visualization-5.html#plot_dotplot"><i class="fa fa-check"></i><b>11.5</b> plot_dotplot</a></li>
<li class="chapter" data-level="11.6" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_boxplot"><i class="fa fa-check"></i><b>11.6</b> plot_correlation_boxplot</a></li>
<li class="chapter" data-level="11.7" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_density"><i class="fa fa-check"></i><b>11.7</b> plot_correlation_density</a></li>
<li class="chapter" data-level="11.8" data-path="visualization-5.html"><a href="visualization-5.html#plot_ordination"><i class="fa fa-check"></i><b>11.8</b> plot_Ordination</a></li>
<li class="chapter" data-level="11.9" data-path="visualization-5.html"><a href="visualization-5.html#plot_ggbiplot"><i class="fa fa-check"></i><b>11.9</b> plot_ggbiplot</a></li>
<li class="chapter" data-level="11.10" data-path="visualization-5.html"><a href="visualization-5.html#plot_corrplot"><i class="fa fa-check"></i><b>11.10</b> plot_corrplot</a></li>
<li class="chapter" data-level="11.11" data-path="visualization-5.html"><a href="visualization-5.html#plot_2da_venn"><i class="fa fa-check"></i><b>11.11</b> plot_2DA_venn</a></li>
<li class="chapter" data-level="11.12" data-path="visualization-5.html"><a href="visualization-5.html#plot-the-da-results-from-the-significant-taxa-by-double-barplot"><i class="fa fa-check"></i><b>11.12</b> plot the DA results from the significant taxa by double barplot</a></li>
<li class="chapter" data-level="11.13" data-path="visualization-5.html"><a href="visualization-5.html#plot_stacked_bar_xivz"><i class="fa fa-check"></i><b>11.13</b> plot_stacked_bar_XIVZ</a></li>
<li class="chapter" data-level="11.14" data-path="visualization-5.html"><a href="visualization-5.html#plot_stackbarplot"><i class="fa fa-check"></i><b>11.14</b> plot_StackBarPlot</a></li>
<li class="chapter" data-level="11.15" data-path="visualization-5.html"><a href="visualization-5.html#color-palettes"><i class="fa fa-check"></i><b>11.15</b> Color Palettes</a>
<ul>
<li class="chapter" data-level="11.15.1" data-path="visualization-5.html"><a href="visualization-5.html#wes-anderson-palettes"><i class="fa fa-check"></i><b>11.15.1</b> Wes Anderson Palettes</a></li>
<li class="chapter" data-level="11.15.2" data-path="visualization-5.html"><a href="visualization-5.html#usemycol"><i class="fa fa-check"></i><b>11.15.2</b> useMyCol</a></li>
</ul></li>
<li class="chapter" data-level="11.16" data-path="visualization-5.html"><a href="visualization-5.html#ordination-plots-with-ggplot2"><i class="fa fa-check"></i><b>11.16</b> Ordination plots with ggplot2</a>
<ul>
<li class="chapter" data-level="11.16.1" data-path="visualization-5.html"><a href="visualization-5.html#principal-components-analysis-with-the-iris-data-set"><i class="fa fa-check"></i><b>11.16.1</b> principal components analysis with the iris data set</a></li>
<li class="chapter" data-level="11.16.2" data-path="visualization-5.html"><a href="visualization-5.html#multiple-correspondence-analysis-with-the-tea-dataset"><i class="fa fa-check"></i><b>11.16.2</b> multiple correspondence analysis with the tea dataset</a></li>
<li class="chapter" data-level="11.16.3" data-path="visualization-5.html"><a href="visualization-5.html#linear-discriminant-analysis"><i class="fa fa-check"></i><b>11.16.3</b> linear discriminant analysis</a></li>
</ul></li>
<li class="chapter" data-level="11.17" data-path="visualization-5.html"><a href="visualization-5.html#systematic-information-8"><i class="fa fa-check"></i><b>11.17</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="association-analysis.html"><a href="association-analysis.html"><i class="fa fa-check"></i><b>12</b> Association Analysis</a>
<ul>
<li class="chapter" data-level="12.1" data-path="association-analysis.html"><a href="association-analysis.html#loading-packages-9"><i class="fa fa-check"></i><b>12.1</b> Loading Packages</a></li>
<li class="chapter" data-level="12.2" data-path="association-analysis.html"><a href="association-analysis.html#importing-data-9"><i class="fa fa-check"></i><b>12.2</b> Importing Data</a></li>
<li class="chapter" data-level="12.3" data-path="association-analysis.html"><a href="association-analysis.html#spearman-correlation-analysis"><i class="fa fa-check"></i><b>12.3</b> Spearman Correlation Analysis</a></li>
<li class="chapter" data-level="12.4" data-path="association-analysis.html"><a href="association-analysis.html#partial-correlation-analysis"><i class="fa fa-check"></i><b>12.4</b> Partial Correlation Analysis</a></li>
<li class="chapter" data-level="12.5" data-path="association-analysis.html"><a href="association-analysis.html#systematic-information-9"><i class="fa fa-check"></i><b>12.5</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html"><i class="fa fa-check"></i><b>13</b> Principals of Differential Analysis methods</a>
<ul>
<li class="chapter" data-level="13.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#aldex2-1"><i class="fa fa-check"></i><b>13.1</b> ALDEx2</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-1"><i class="fa fa-check"></i><b>13.1.1</b> Vignette</a></li>
<li class="chapter" data-level="13.1.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type"><i class="fa fa-check"></i><b>13.1.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.1.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments"><i class="fa fa-check"></i><b>13.1.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.1.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures"><i class="fa fa-check"></i><b>13.1.4</b> Procedures</a></li>
<li class="chapter" data-level="13.1.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output"><i class="fa fa-check"></i><b>13.1.5</b> Key output</a></li>
<li class="chapter" data-level="13.1.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax"><i class="fa fa-check"></i><b>13.1.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.1.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax"><i class="fa fa-check"></i><b>13.1.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="13.1.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes"><i class="fa fa-check"></i><b>13.1.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#ancom-ii"><i class="fa fa-check"></i><b>13.2</b> ANCOM-II</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-1"><i class="fa fa-check"></i><b>13.2.1</b> Input Data Type</a></li>
<li class="chapter" data-level="13.2.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-1"><i class="fa fa-check"></i><b>13.2.2</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.2.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-1"><i class="fa fa-check"></i><b>13.2.3</b> Procedures</a></li>
<li class="chapter" data-level="13.2.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-1"><i class="fa fa-check"></i><b>13.2.4</b> Key output</a></li>
<li class="chapter" data-level="13.2.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-1"><i class="fa fa-check"></i><b>13.2.5</b> Original Syntax</a></li>
<li class="chapter" data-level="13.2.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-1"><i class="fa fa-check"></i><b>13.2.6</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="13.2.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-1"><i class="fa fa-check"></i><b>13.2.7</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#corncob-1"><i class="fa fa-check"></i><b>13.3</b> corncob</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-2"><i class="fa fa-check"></i><b>13.3.1</b> Vignette</a></li>
<li class="chapter" data-level="13.3.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-2"><i class="fa fa-check"></i><b>13.3.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.3.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-2"><i class="fa fa-check"></i><b>13.3.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.3.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-2"><i class="fa fa-check"></i><b>13.3.4</b> Procedures</a></li>
<li class="chapter" data-level="13.3.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-2"><i class="fa fa-check"></i><b>13.3.5</b> Key output</a></li>
<li class="chapter" data-level="13.3.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-2"><i class="fa fa-check"></i><b>13.3.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.3.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-2"><i class="fa fa-check"></i><b>13.3.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="13.3.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-2"><i class="fa fa-check"></i><b>13.3.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#deseq2-1"><i class="fa fa-check"></i><b>13.4</b> DESeq2</a>
<ul>
<li class="chapter" data-level="13.4.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-3"><i class="fa fa-check"></i><b>13.4.1</b> Vignette</a></li>
<li class="chapter" data-level="13.4.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-3"><i class="fa fa-check"></i><b>13.4.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.4.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-3"><i class="fa fa-check"></i><b>13.4.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.4.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-3"><i class="fa fa-check"></i><b>13.4.4</b> Procedures</a></li>
<li class="chapter" data-level="13.4.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-3"><i class="fa fa-check"></i><b>13.4.5</b> Key output</a></li>
<li class="chapter" data-level="13.4.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-3"><i class="fa fa-check"></i><b>13.4.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.4.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-3"><i class="fa fa-check"></i><b>13.4.7</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="13.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#edger-1"><i class="fa fa-check"></i><b>13.5</b> edgeR</a>
<ul>
<li class="chapter" data-level="13.5.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-4"><i class="fa fa-check"></i><b>13.5.1</b> Vignette</a></li>
<li class="chapter" data-level="13.5.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-4"><i class="fa fa-check"></i><b>13.5.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.5.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-4"><i class="fa fa-check"></i><b>13.5.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.5.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-4"><i class="fa fa-check"></i><b>13.5.4</b> Procedures</a></li>
<li class="chapter" data-level="13.5.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-4"><i class="fa fa-check"></i><b>13.5.5</b> Key output</a></li>
<li class="chapter" data-level="13.5.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-4"><i class="fa fa-check"></i><b>13.5.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.5.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-4"><i class="fa fa-check"></i><b>13.5.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="13.5.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-3"><i class="fa fa-check"></i><b>13.5.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#lefse"><i class="fa fa-check"></i><b>13.6</b> LEfSe</a>
<ul>
<li class="chapter" data-level="13.6.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-5"><i class="fa fa-check"></i><b>13.6.1</b> Vignette</a></li>
<li class="chapter" data-level="13.6.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-5"><i class="fa fa-check"></i><b>13.6.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.6.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-5"><i class="fa fa-check"></i><b>13.6.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.6.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-5"><i class="fa fa-check"></i><b>13.6.4</b> Procedures</a></li>
<li class="chapter" data-level="13.6.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-5"><i class="fa fa-check"></i><b>13.6.5</b> Key output</a></li>
<li class="chapter" data-level="13.6.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-5"><i class="fa fa-check"></i><b>13.6.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.6.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-5"><i class="fa fa-check"></i><b>13.6.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="13.6.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-4"><i class="fa fa-check"></i><b>13.6.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="13.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#limma-voom"><i class="fa fa-check"></i><b>13.7</b> limma voom</a>
<ul>
<li class="chapter" data-level="13.7.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-6"><i class="fa fa-check"></i><b>13.7.1</b> Vignette</a></li>
<li class="chapter" data-level="13.7.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-6"><i class="fa fa-check"></i><b>13.7.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.7.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-6"><i class="fa fa-check"></i><b>13.7.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.7.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-6"><i class="fa fa-check"></i><b>13.7.4</b> Procedures</a></li>
<li class="chapter" data-level="13.7.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-6"><i class="fa fa-check"></i><b>13.7.5</b> Key output</a></li>
<li class="chapter" data-level="13.7.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-6"><i class="fa fa-check"></i><b>13.7.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.7.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-6"><i class="fa fa-check"></i><b>13.7.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="13.7.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-5"><i class="fa fa-check"></i><b>13.7.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="13.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#maaslin2"><i class="fa fa-check"></i><b>13.8</b> MaAsLin2</a>
<ul>
<li class="chapter" data-level="13.8.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-7"><i class="fa fa-check"></i><b>13.8.1</b> Vignette</a></li>
<li class="chapter" data-level="13.8.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-7"><i class="fa fa-check"></i><b>13.8.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.8.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-7"><i class="fa fa-check"></i><b>13.8.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.8.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-7"><i class="fa fa-check"></i><b>13.8.4</b> Procedures</a></li>
<li class="chapter" data-level="13.8.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-7"><i class="fa fa-check"></i><b>13.8.5</b> Key output</a></li>
<li class="chapter" data-level="13.8.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-7"><i class="fa fa-check"></i><b>13.8.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.8.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-7"><i class="fa fa-check"></i><b>13.8.7</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="13.9" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#metagenomeseq-1"><i class="fa fa-check"></i><b>13.9</b> metagenomeSeq</a>
<ul>
<li class="chapter" data-level="13.9.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-8"><i class="fa fa-check"></i><b>13.9.1</b> Vignette</a></li>
<li class="chapter" data-level="13.9.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-8"><i class="fa fa-check"></i><b>13.9.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.9.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-8"><i class="fa fa-check"></i><b>13.9.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.9.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-8"><i class="fa fa-check"></i><b>13.9.4</b> Procedures</a></li>
<li class="chapter" data-level="13.9.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-8"><i class="fa fa-check"></i><b>13.9.5</b> Key output</a></li>
<li class="chapter" data-level="13.9.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-8"><i class="fa fa-check"></i><b>13.9.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.9.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-8"><i class="fa fa-check"></i><b>13.9.7</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="13.10" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#t-test-2"><i class="fa fa-check"></i><b>13.10</b> t-test</a>
<ul>
<li class="chapter" data-level="13.10.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-9"><i class="fa fa-check"></i><b>13.10.1</b> Input Data Type</a></li>
<li class="chapter" data-level="13.10.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-9"><i class="fa fa-check"></i><b>13.10.2</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.10.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-9"><i class="fa fa-check"></i><b>13.10.3</b> Procedures</a></li>
<li class="chapter" data-level="13.10.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-9"><i class="fa fa-check"></i><b>13.10.4</b> Key output</a></li>
<li class="chapter" data-level="13.10.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-9"><i class="fa fa-check"></i><b>13.10.5</b> Original Syntax</a></li>
<li class="chapter" data-level="13.10.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-9"><i class="fa fa-check"></i><b>13.10.6</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="13.11" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#wilcoxon-test"><i class="fa fa-check"></i><b>13.11</b> Wilcoxon test</a>
<ul>
<li class="chapter" data-level="13.11.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-10"><i class="fa fa-check"></i><b>13.11.1</b> Input Data Type</a></li>
<li class="chapter" data-level="13.11.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-10"><i class="fa fa-check"></i><b>13.11.2</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.11.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-10"><i class="fa fa-check"></i><b>13.11.3</b> Procedures</a></li>
<li class="chapter" data-level="13.11.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-10"><i class="fa fa-check"></i><b>13.11.4</b> Key output</a></li>
<li class="chapter" data-level="13.11.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-10"><i class="fa fa-check"></i><b>13.11.5</b> Original Syntax</a></li>
<li class="chapter" data-level="13.11.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-10"><i class="fa fa-check"></i><b>13.11.6</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="13.12" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#raida-1"><i class="fa fa-check"></i><b>13.12</b> RAIDA</a>
<ul>
<li class="chapter" data-level="13.12.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-9"><i class="fa fa-check"></i><b>13.12.1</b> Vignette</a></li>
<li class="chapter" data-level="13.12.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-11"><i class="fa fa-check"></i><b>13.12.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.12.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-11"><i class="fa fa-check"></i><b>13.12.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.12.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-11"><i class="fa fa-check"></i><b>13.12.4</b> Procedures</a></li>
<li class="chapter" data-level="13.12.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-11"><i class="fa fa-check"></i><b>13.12.5</b> Key output</a></li>
<li class="chapter" data-level="13.12.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-11"><i class="fa fa-check"></i><b>13.12.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.12.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-11"><i class="fa fa-check"></i><b>13.12.7</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="13.13" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#summary"><i class="fa fa-check"></i><b>13.13</b> Summary</a></li>
<li class="chapter" data-level="13.14" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#advantages-and-disadvantages"><i class="fa fa-check"></i><b>13.14</b> Advantages and disadvantages</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="examples.html"><a href="examples.html"><i class="fa fa-check"></i><b>14</b> Examples</a>
<ul>
<li class="chapter" data-level="14.1" data-path="examples.html"><a href="examples.html#loading-packages-10"><i class="fa fa-check"></i><b>14.1</b> Loading Packages</a></li>
<li class="chapter" data-level="14.2" data-path="examples.html"><a href="examples.html#workflow-description"><i class="fa fa-check"></i><b>14.2</b> Workflow description</a></li>
<li class="chapter" data-level="14.3" data-path="examples.html"><a href="examples.html#amplicon-sequencing-16s"><i class="fa fa-check"></i><b>14.3</b> Amplicon sequencing (16s)</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object"><i class="fa fa-check"></i><b>14.3.1</b> Converting inputs into phyloseq-class object</a></li>
<li class="chapter" data-level="14.3.2" data-path="examples.html"><a href="examples.html#quality-control"><i class="fa fa-check"></i><b>14.3.2</b> Quality Control</a></li>
<li class="chapter" data-level="14.3.3" data-path="examples.html"><a href="examples.html#data-processing"><i class="fa fa-check"></i><b>14.3.3</b> Data processing</a></li>
<li class="chapter" data-level="14.3.4" data-path="examples.html"><a href="examples.html#diversity-analysis"><i class="fa fa-check"></i><b>14.3.4</b> Diversity analysis</a></li>
<li class="chapter" data-level="14.3.5" data-path="examples.html"><a href="examples.html#permanova-ordination"><i class="fa fa-check"></i><b>14.3.5</b> PERMANOVA + Ordination</a></li>
<li class="chapter" data-level="14.3.6" data-path="examples.html"><a href="examples.html#microbial-composition-1"><i class="fa fa-check"></i><b>14.3.6</b> Microbial composition</a></li>
<li class="chapter" data-level="14.3.7" data-path="examples.html"><a href="examples.html#differential-analysis-1"><i class="fa fa-check"></i><b>14.3.7</b> Differential Analysis</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="examples.html"><a href="examples.html#metagenomics-mgs"><i class="fa fa-check"></i><b>14.4</b> Metagenomics (MGS)</a>
<ul>
<li class="chapter" data-level="14.4.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object-1"><i class="fa fa-check"></i><b>14.4.1</b> Converting inputs into phyloseq-class object</a></li>
<li class="chapter" data-level="14.4.2" data-path="examples.html"><a href="examples.html#quality-control-1"><i class="fa fa-check"></i><b>14.4.2</b> Quality Control</a></li>
<li class="chapter" data-level="14.4.3" data-path="examples.html"><a href="examples.html#data-processing-1"><i class="fa fa-check"></i><b>14.4.3</b> Data processing</a></li>
<li class="chapter" data-level="14.4.4" data-path="examples.html"><a href="examples.html#diversity-analysis-1"><i class="fa fa-check"></i><b>14.4.4</b> Diversity analysis</a></li>
<li class="chapter" data-level="14.4.5" data-path="examples.html"><a href="examples.html#permanova-ordination-1"><i class="fa fa-check"></i><b>14.4.5</b> PERMANOVA + Ordination</a></li>
<li class="chapter" data-level="14.4.6" data-path="examples.html"><a href="examples.html#microbial-composition-2"><i class="fa fa-check"></i><b>14.4.6</b> Microbial composition</a></li>
<li class="chapter" data-level="14.4.7" data-path="examples.html"><a href="examples.html#differential-analysis-2"><i class="fa fa-check"></i><b>14.4.7</b> Differential Analysis</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="examples.html"><a href="examples.html#systematic-information-10"><i class="fa fa-check"></i><b>14.5</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">XMAS 2.0 Tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="examples" class="section level1 hasAnchor" number="14">
<h1><span class="header-section-number">Chapter 14</span> Examples<a href="examples.html#examples" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Here, we give users two examples to practice the data analysis workflow by <strong>XMAS 2.0</strong>. By the way, we also recommend users handling your own microbiota data in a reasonable manner when you utilize this package or workflow. Pay attention to whether your data fit the methods this package provided.</p>
<div id="loading-packages-10" class="section level2 hasAnchor" number="14.1">
<h2><span class="header-section-number">14.1</span> Loading Packages<a href="examples.html#loading-packages-10" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb505"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb505-1"><a href="examples.html#cb505-1" tabindex="-1"></a><span class="fu">library</span>(XMAS2)</span>
<span id="cb505-2"><a href="examples.html#cb505-2" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb505-3"><a href="examples.html#cb505-3" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb505-4"><a href="examples.html#cb505-4" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb505-5"><a href="examples.html#cb505-5" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb505-6"><a href="examples.html#cb505-6" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb505-7"><a href="examples.html#cb505-7" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span></code></pre></div>
</div>
<div id="workflow-description" class="section level2 hasAnchor" number="14.2">
<h2><span class="header-section-number">14.2</span> Workflow description<a href="examples.html#workflow-description" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The standard data analysis of 16S and MGS data by <strong>XMAS 2.0</strong>.</p>
<ul>
<li>16S
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-264"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-264-1.png" alt="Functions of XMAS 2.0 in 16s" width="768" />
<p class="caption">
Figure 14.1: Functions of XMAS 2.0 in 16s
</p>
</div></li>
<li>MGS
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-265"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-265-1.png" alt="Functions of XMAS 2.0 in MGS" width="768" />
<p class="caption">
Figure 14.2: Functions of XMAS 2.0 in MGS
</p>
</div></li>
</ul>
</div>
<div id="amplicon-sequencing-16s" class="section level2 hasAnchor" number="14.3">
<h2><span class="header-section-number">14.3</span> Amplicon sequencing (16s)<a href="examples.html#amplicon-sequencing-16s" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The upstream process is performed by in-house pipeline. and this example just shows how to perform downstream data analysis. In briefly, the example comprises the following steps:</p>
<ol style="list-style-type: decimal">
<li><p>Converting inputs into phyloseq object;</p></li>
<li><p>Quality Evaluation;</p></li>
<li><p>Pre-Processing Data;</p></li>
<li><p>Diversity analysis;</p></li>
<li><p>Ordination analysis;</p></li>
<li><p>Composition analysis;</p></li>
<li><p>Differential analysis.</p></li>
</ol>
<div id="converting-inputs-into-phyloseq-class-object" class="section level3 hasAnchor" number="14.3.1">
<h3><span class="header-section-number">14.3.1</span> Converting inputs into phyloseq-class object<a href="examples.html#converting-inputs-into-phyloseq-class-object" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>dada2</strong> result from standardized_analytics_workflow_R_function.</p>
<ol style="list-style-type: decimal">
<li><p>/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/16S/process/xdada2/dada2_res.rds</p></li>
<li><p>/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/16S/process/fasta2tree/tree.nwk</p></li>
<li><p>/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/16S/metadata.txt</p></li>
</ol>
<div class="sourceCode" id="cb506"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb506-1"><a href="examples.html#cb506-1" tabindex="-1"></a><span class="co"># dada2 results from in-house 16s pipeline</span></span>
<span id="cb506-2"><a href="examples.html#cb506-2" tabindex="-1"></a>dada2_res <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(</span>
<span id="cb506-3"><a href="examples.html#cb506-3" tabindex="-1"></a>  <span class="fu">system.file</span>(</span>
<span id="cb506-4"><a href="examples.html#cb506-4" tabindex="-1"></a>    <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;dada2_res.rds&quot;</span>,</span>
<span id="cb506-5"><a href="examples.html#cb506-5" tabindex="-1"></a>    <span class="at">package =</span> <span class="st">&quot;XMAS2&quot;</span>    </span>
<span id="cb506-6"><a href="examples.html#cb506-6" tabindex="-1"></a>    )</span>
<span id="cb506-7"><a href="examples.html#cb506-7" tabindex="-1"></a>)</span>
<span id="cb506-8"><a href="examples.html#cb506-8" tabindex="-1"></a></span>
<span id="cb506-9"><a href="examples.html#cb506-9" tabindex="-1"></a><span class="co"># the metadata matches to dada2 result</span></span>
<span id="cb506-10"><a href="examples.html#cb506-10" tabindex="-1"></a>sam_tab <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb506-11"><a href="examples.html#cb506-11" tabindex="-1"></a>    <span class="fu">system.file</span>(</span>
<span id="cb506-12"><a href="examples.html#cb506-12" tabindex="-1"></a>        <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;dada2_metadata.tsv&quot;</span>,</span>
<span id="cb506-13"><a href="examples.html#cb506-13" tabindex="-1"></a>        <span class="at">package =</span> <span class="st">&quot;XMAS2&quot;</span></span>
<span id="cb506-14"><a href="examples.html#cb506-14" tabindex="-1"></a>    ),</span>
<span id="cb506-15"><a href="examples.html#cb506-15" tabindex="-1"></a>    <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb506-16"><a href="examples.html#cb506-16" tabindex="-1"></a>    <span class="at">header =</span> <span class="cn">TRUE</span>,</span>
<span id="cb506-17"><a href="examples.html#cb506-17" tabindex="-1"></a>    <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span></span>
<span id="cb506-18"><a href="examples.html#cb506-18" tabindex="-1"></a>)</span>
<span id="cb506-19"><a href="examples.html#cb506-19" tabindex="-1"></a></span>
<span id="cb506-20"><a href="examples.html#cb506-20" tabindex="-1"></a><span class="co"># tree file from dada2 reference data silva</span></span>
<span id="cb506-21"><a href="examples.html#cb506-21" tabindex="-1"></a>tree <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">read_tree</span>(</span>
<span id="cb506-22"><a href="examples.html#cb506-22" tabindex="-1"></a>  <span class="fu">system.file</span>(</span>
<span id="cb506-23"><a href="examples.html#cb506-23" tabindex="-1"></a>    <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;tree.nwk&quot;</span>,</span>
<span id="cb506-24"><a href="examples.html#cb506-24" tabindex="-1"></a>    <span class="at">package =</span> <span class="st">&quot;XMAS2&quot;</span>    </span>
<span id="cb506-25"><a href="examples.html#cb506-25" tabindex="-1"></a>    )</span>
<span id="cb506-26"><a href="examples.html#cb506-26" tabindex="-1"></a>)</span>
<span id="cb506-27"><a href="examples.html#cb506-27" tabindex="-1"></a></span>
<span id="cb506-28"><a href="examples.html#cb506-28" tabindex="-1"></a>tax_tab <span class="ot">&lt;-</span> <span class="fu">import_dada2_taxa</span>(<span class="at">dada2_taxa =</span> dada2_res<span class="sc">$</span>tax_tab)</span>
<span id="cb506-29"><a href="examples.html#cb506-29" tabindex="-1"></a>otu_tab <span class="ot">&lt;-</span> dada2_res<span class="sc">$</span>seq_tab</span>
<span id="cb506-30"><a href="examples.html#cb506-30" tabindex="-1"></a>sam_tab <span class="ot">&lt;-</span> sam_tab <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;seqID&quot;</span>)</span>
<span id="cb506-31"><a href="examples.html#cb506-31" tabindex="-1"></a></span>
<span id="cb506-32"><a href="examples.html#cb506-32" tabindex="-1"></a><span class="co"># Shouldn&#39;t use the Total Number as SampleID (wrong: 123456; right: X123456)</span></span>
<span id="cb506-33"><a href="examples.html#cb506-33" tabindex="-1"></a><span class="fu">rownames</span>(otu_tab) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;S&quot;</span>, <span class="fu">rownames</span>(otu_tab))</span>
<span id="cb506-34"><a href="examples.html#cb506-34" tabindex="-1"></a><span class="fu">rownames</span>(sam_tab) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;S&quot;</span>, <span class="fu">rownames</span>(sam_tab))</span>
<span id="cb506-35"><a href="examples.html#cb506-35" tabindex="-1"></a></span>
<span id="cb506-36"><a href="examples.html#cb506-36" tabindex="-1"></a>dada2_ps <span class="ot">&lt;-</span> <span class="fu">get_dada2_phyloseq</span>(</span>
<span id="cb506-37"><a href="examples.html#cb506-37" tabindex="-1"></a>                <span class="at">seq_tab =</span> otu_tab, </span>
<span id="cb506-38"><a href="examples.html#cb506-38" tabindex="-1"></a>                <span class="at">tax_tab =</span> tax_tab, </span>
<span id="cb506-39"><a href="examples.html#cb506-39" tabindex="-1"></a>                <span class="at">sam_tab =</span> sam_tab, </span>
<span id="cb506-40"><a href="examples.html#cb506-40" tabindex="-1"></a>                <span class="at">phy_tree =</span> tree)</span>
<span id="cb506-41"><a href="examples.html#cb506-41" tabindex="-1"></a>dada2_ps</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 896 taxa and 24 samples ]
## sample_data() Sample Data:       [ 24 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 896 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 896 tips and 893 internal nodes ]
## refseq()      DNAStringSet:      [ 896 reference sequences ]</code></pre>
<p>Here, the phyloseq object comprises five components (<em>OTU Table</em>, <em>Sample Data</em>, <em>Taxonomy Table</em>, <em>Phylogenetic Tree</em> and <em>DNAStringSet</em>).</p>
<div id="summarize-phyloseq-class-object-1" class="section level4 hasAnchor" number="14.3.1.1">
<h4><span class="header-section-number">14.3.1.1</span> Summarize phyloseq-class object<a href="examples.html#summarize-phyloseq-class-object-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb508"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb508-1"><a href="examples.html#cb508-1" tabindex="-1"></a><span class="fu">summarize_phyloseq</span>(<span class="at">ps =</span> dada2_ps)</span>
<span id="cb508-2"><a href="examples.html#cb508-2" tabindex="-1"></a><span class="do">## Compositional = NO2</span></span>
<span id="cb508-3"><a href="examples.html#cb508-3" tabindex="-1"></a><span class="do">## 1] Min. number of reads = 511812] Max. number of reads = 936223] Total number of reads = 15025374] Average number of reads = 62605.70833333335] Median number of reads = 619157] Sparsity = 0.8653738839285716] Any OTU sum to 1 or less? NO8] Number of singletons = 09] Percent of OTUs that are singletons</span></span>
<span id="cb508-4"><a href="examples.html#cb508-4" tabindex="-1"></a><span class="do">##         (i.e. exactly one read detected across all samples)010] Number of sample variables are: 1Group2</span></span>
<span id="cb508-5"><a href="examples.html#cb508-5" tabindex="-1"></a><span class="do">## [[1]]</span></span>
<span id="cb508-6"><a href="examples.html#cb508-6" tabindex="-1"></a><span class="do">## [1] &quot;1] Min. number of reads = 51181&quot;</span></span>
<span id="cb508-7"><a href="examples.html#cb508-7" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb508-8"><a href="examples.html#cb508-8" tabindex="-1"></a><span class="do">## [[2]]</span></span>
<span id="cb508-9"><a href="examples.html#cb508-9" tabindex="-1"></a><span class="do">## [1] &quot;2] Max. number of reads = 93622&quot;</span></span>
<span id="cb508-10"><a href="examples.html#cb508-10" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb508-11"><a href="examples.html#cb508-11" tabindex="-1"></a><span class="do">## [[3]]</span></span>
<span id="cb508-12"><a href="examples.html#cb508-12" tabindex="-1"></a><span class="do">## [1] &quot;3] Total number of reads = 1502537&quot;</span></span>
<span id="cb508-13"><a href="examples.html#cb508-13" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb508-14"><a href="examples.html#cb508-14" tabindex="-1"></a><span class="do">## [[4]]</span></span>
<span id="cb508-15"><a href="examples.html#cb508-15" tabindex="-1"></a><span class="do">## [1] &quot;4] Average number of reads = 62605.7083333333&quot;</span></span>
<span id="cb508-16"><a href="examples.html#cb508-16" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb508-17"><a href="examples.html#cb508-17" tabindex="-1"></a><span class="do">## [[5]]</span></span>
<span id="cb508-18"><a href="examples.html#cb508-18" tabindex="-1"></a><span class="do">## [1] &quot;5] Median number of reads = 61915&quot;</span></span>
<span id="cb508-19"><a href="examples.html#cb508-19" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb508-20"><a href="examples.html#cb508-20" tabindex="-1"></a><span class="do">## [[6]]</span></span>
<span id="cb508-21"><a href="examples.html#cb508-21" tabindex="-1"></a><span class="do">## [1] &quot;7] Sparsity = 0.865373883928571&quot;</span></span>
<span id="cb508-22"><a href="examples.html#cb508-22" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb508-23"><a href="examples.html#cb508-23" tabindex="-1"></a><span class="do">## [[7]]</span></span>
<span id="cb508-24"><a href="examples.html#cb508-24" tabindex="-1"></a><span class="do">## [1] &quot;6] Any OTU sum to 1 or less? NO&quot;</span></span>
<span id="cb508-25"><a href="examples.html#cb508-25" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb508-26"><a href="examples.html#cb508-26" tabindex="-1"></a><span class="do">## [[8]]</span></span>
<span id="cb508-27"><a href="examples.html#cb508-27" tabindex="-1"></a><span class="do">## [1] &quot;8] Number of singletons = 0&quot;</span></span>
<span id="cb508-28"><a href="examples.html#cb508-28" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb508-29"><a href="examples.html#cb508-29" tabindex="-1"></a><span class="do">## [[9]]</span></span>
<span id="cb508-30"><a href="examples.html#cb508-30" tabindex="-1"></a><span class="do">## [1] &quot;9] Percent of OTUs that are singletons\n        (i.e. exactly one read detected across all samples)0&quot;</span></span>
<span id="cb508-31"><a href="examples.html#cb508-31" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb508-32"><a href="examples.html#cb508-32" tabindex="-1"></a><span class="do">## [[10]]</span></span>
<span id="cb508-33"><a href="examples.html#cb508-33" tabindex="-1"></a><span class="do">## [1] &quot;10] Number of sample variables are: 1&quot;</span></span>
<span id="cb508-34"><a href="examples.html#cb508-34" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb508-35"><a href="examples.html#cb508-35" tabindex="-1"></a><span class="do">## [[11]]</span></span>
<span id="cb508-36"><a href="examples.html#cb508-36" tabindex="-1"></a><span class="do">## [1] &quot;Group&quot;</span></span></code></pre></div>
<p>The minus account of the OTU counts is <em>51181</em> in the phyloseq object, and we can use it as the threshold to rarefy.</p>
<p>Notice the <strong>Sparsity (0.865)</strong>, indicating the data has many zeros and pay attention to the downstream data analysis. A common property of amplicon based microbiota data generated by sequencing.</p>
</div>
</div>
<div id="quality-control" class="section level3 hasAnchor" number="14.3.2">
<h3><span class="header-section-number">14.3.2</span> Quality Control<a href="examples.html#quality-control" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Quality control of DADA2 results will help us have more rational determinations on the further data analysis.</p>
<div id="reads-track-by-dada2" class="section level4 hasAnchor" number="14.3.2.1">
<h4><span class="header-section-number">14.3.2.1</span> Reads’ track by DADA2<a href="examples.html#reads-track-by-dada2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb509"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb509-1"><a href="examples.html#cb509-1" tabindex="-1"></a><span class="fu">plot_Dada2Track</span>(<span class="at">data =</span> dada2_res<span class="sc">$</span>reads_track)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-268"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-268-1.png" alt="Reads' track by DADA2 (16s example)" width="576" />
<p class="caption">
Figure 14.3: Reads’ track by DADA2 (16s example)
</p>
</div>
<p>The percentage of the final remained read counts approximate <strong>70%</strong>, indicating that we should consider the sequence depth for analysis when we build the sequence library.</p>
</div>
<div id="spike-in-sample-brs-assessment" class="section level4 hasAnchor" number="14.3.2.2">
<h4><span class="header-section-number">14.3.2.2</span> Spike-in sample (BRS) assessment<a href="examples.html#spike-in-sample-brs-assessment" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Extract the genus level phyloseq and getting the BRS_ID</li>
</ul>
<div class="sourceCode" id="cb510"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb510-1"><a href="examples.html#cb510-1" tabindex="-1"></a>dada2_ps_genus <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(<span class="at">ps =</span> dada2_ps, </span>
<span id="cb510-2"><a href="examples.html#cb510-2" tabindex="-1"></a>                                 <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>)</span>
<span id="cb510-3"><a href="examples.html#cb510-3" tabindex="-1"></a><span class="fu">sample_data</span>(dada2_ps_genus)</span></code></pre></div>
<pre><code>##       Group
## S6030    BB
## S6032    BB
## S6033    BB
## S6035    AA
## S6036    BB
## S6037    AA
## S6040    BB
## S6043    AA
## S6045    BB
## S6046    BB
## S6048    BB
## S6049    AA
## S6050    BB
## S6054    BB
## S6055    BB
## S6058    BB
## S6059    AA
## S6060    AA
## S6061    AA
## S6063    BB
## S6065    AA
## S6066    AA
## S6068    BB
## S8005    QC</code></pre>
<p><strong>The BRS_ID is <em>S8005</em> </strong>.</p>
<ul>
<li>Run <code>run_RefCheck2</code></li>
</ul>
<div class="sourceCode" id="cb512"><pre class="sourceCode markdown"><code class="sourceCode markdown"><span id="cb512-1"><a href="examples.html#cb512-1" tabindex="-1"></a>run_RefCheck2(</span>
<span id="cb512-2"><a href="examples.html#cb512-2" tabindex="-1"></a>    ps = dada2_ps_genus,</span>
<span id="cb512-3"><a href="examples.html#cb512-3" tabindex="-1"></a>    BRS_ID = &quot;S8005&quot;,</span>
<span id="cb512-4"><a href="examples.html#cb512-4" tabindex="-1"></a>    Ref_type = &quot;16s&quot;)</span></code></pre></div>
<pre><code>## Noting: the Reference Matrix is for 16s
## S8005 is in the Reference Matrix&#39;s samples and remove it to run
## 
## ############Matched baterica of the BRS sample#############
## The number of BRS&#39; bacteria matched the Reference Matrix is [15]
## g__Bifidobacterium
## g__Bacteroides
## g__Faecalibacterium
## g__Lactobacillus
## g__Parabacteroides
## g__Collinsella
## g__Coprococcus_3
## g__Dorea
## g__Streptococcus
## g__Roseburia
## g__Anaerostipes
## g__Escherichia_Shigella
## g__Enterococcus
## g__Prevotella_9
## g__Eggerthella
## 
## The number of the additional bacteria compared to Reference Matrix is [1]
## ###########################################################
## 
## ##################Status of the BRS sample##################
## Whether the BRS has the all bateria of Reference Matrix: TRUE
## Correlation Coefficient of the BRS is: 0.9714
## Bray Curtis of the BRS is: 0.07607
## Impurity of Max additional genus (g__Cutibacterium) of the BRS is: 0.06409
## ###########################################################
## #####Final Evaluation Results of the BRS #######
## The BRS of sequencing dataset passed the cutoff of the Reference Matrix 
## Cutoff of Coefficient is 0.8946
## Cutoff of BrayCurtis is 0.1425
## Cutoff of Impurity is 0.1565
## ###########################################################
## $Matrix
##                             8002        8003        8004       8006        8007        8008        8009        8005        mean
## Bifidobacterium      31.11079015 30.88310969 32.31232692 18.4930259 20.20409870 17.96225391 18.03588291 27.22437034 24.52823232
## Bacteroides          20.44753484 14.46581958 24.57151411 26.7370147 25.85863655 27.51353663 26.99272343 24.23896093 23.85321759
## Faecalibacterium      0.79850615  0.62937893  1.05531023  1.7487249  1.64282727  1.96346413  1.81219797  1.04035376  1.33634542
## Lactobacillus         2.61732573  3.36856272  3.44379163  5.9292703  5.78000836  5.78189064  6.32672332  3.87088505  4.63980722
## Parabacteroides       7.11124408  7.45952579  5.36075144  8.7149995  8.01840234  8.74899584  8.62634005  5.61833757  7.45732457
## Collinsella           0.12792605  0.88271385  0.55665744  1.2764130  0.67921372  1.89356397  1.26367828  0.45502126  0.89189845
## Coprococcus_3         1.00380683  0.97969362  0.80270938  1.6693557  1.56419908  1.67969035  1.71221463  0.87586251  1.28594151
## Dorea                 2.80715148  3.45564660  2.46613277  3.9684605  3.99163530  3.81529666  3.69382881  2.34880690  3.31836988
## Streptococcus         2.91960260  3.43387563  2.59149764  3.4818362  3.31409452  3.51378702  3.37721491  2.68740253  3.16491388
## Roseburia             0.03404484  0.04750030  0.02806676  0.0338295  0.03178586  0.02503886  0.02499583  0.03311188  0.03229673
## Anaerostipes          0.32291011  0.43245062  0.31528329  0.5386697  0.44500209  0.53416240  0.48047548  0.32471000  0.42420796
## Escherichia_Shigella 15.27581475 16.00265210 12.36527954 10.8423545 13.43203680 10.15743185 11.59945565 14.03516268 12.96377349
## Enterococcus         14.51444842 14.66472707 11.04239952 13.1674820 12.07360937 12.88771113 12.84924735 11.61906390 12.85233610
## Prevotella_9          0.77374627  3.07465463  2.75709154  2.9145415  2.52446675  3.04952478  2.84952508  5.40257632  2.91826586
## Eggerthella           0.04951976  0.15437596  0.27131203  0.4840221  0.43998327  0.47365181  0.35549631  0.16128688  0.29870601
## Impurity_level        0.08562792  0.06531291  0.05987576  0.0000000  0.00000000  0.00000000  0.00000000  0.06409000  0.03436332
## 
## $Assess
##               Gold_Cutoff     BRS
## Coef               0.8946 0.97140
## Bray               0.1425 0.07607
## Impurity(max)      1.0000 0.06409</code></pre>
</div>
<div id="spike-in-samples-remove" class="section level4 hasAnchor" number="14.3.2.3">
<h4><span class="header-section-number">14.3.2.3</span> Spike-in sample’s remove<a href="examples.html#spike-in-samples-remove" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb514"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb514-1"><a href="examples.html#cb514-1" tabindex="-1"></a>dada2_ps_remove_BRS <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb514-2"><a href="examples.html#cb514-2" tabindex="-1"></a>                         <span class="at">ps =</span> dada2_ps,</span>
<span id="cb514-3"><a href="examples.html#cb514-3" tabindex="-1"></a>                         <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb514-4"><a href="examples.html#cb514-4" tabindex="-1"></a>                         <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>,</span>
<span id="cb514-5"><a href="examples.html#cb514-5" tabindex="-1"></a>                         <span class="at">discard =</span> <span class="cn">TRUE</span>)</span>
<span id="cb514-6"><a href="examples.html#cb514-6" tabindex="-1"></a>dada2_ps_remove_BRS</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 896 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 896 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 896 tips and 893 internal nodes ]
## refseq()      DNAStringSet:      [ 896 reference sequences ]</code></pre>
</div>
<div id="rarefaction-curves-1" class="section level4 hasAnchor" number="14.3.2.4">
<h4><span class="header-section-number">14.3.2.4</span> Rarefaction curves<a href="examples.html#rarefaction-curves-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb516"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb516-1"><a href="examples.html#cb516-1" tabindex="-1"></a><span class="fu">plot_RarefCurve</span>(<span class="at">ps =</span> dada2_ps_remove_BRS,</span>
<span id="cb516-2"><a href="examples.html#cb516-2" tabindex="-1"></a>                <span class="at">taxa_level =</span> <span class="st">&quot;OTU&quot;</span>,</span>
<span id="cb516-3"><a href="examples.html#cb516-3" tabindex="-1"></a>                <span class="at">step =</span> <span class="dv">400</span>,</span>
<span id="cb516-4"><a href="examples.html#cb516-4" tabindex="-1"></a>                <span class="at">label =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb516-5"><a href="examples.html#cb516-5" tabindex="-1"></a>                <span class="at">color =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<pre><code>## rarefying sample S6030
## rarefying sample S6032
## rarefying sample S6033
## rarefying sample S6035
## rarefying sample S6036
## rarefying sample S6037
## rarefying sample S6040
## rarefying sample S6043
## rarefying sample S6045
## rarefying sample S6046
## rarefying sample S6048
## rarefying sample S6049
## rarefying sample S6050
## rarefying sample S6054
## rarefying sample S6055
## rarefying sample S6058
## rarefying sample S6059
## rarefying sample S6060
## rarefying sample S6061
## rarefying sample S6063
## rarefying sample S6065
## rarefying sample S6066
## rarefying sample S6068</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-272"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-272-1.png" alt="Rarefaction curves (16s example)" width="576" />
<p class="caption">
Figure 14.4: Rarefaction curves (16s example)
</p>
</div>
<p>The result showed that all the samples had <strong>different sequence depth</strong> but had the full sample richness.</p>
</div>
</div>
<div id="data-processing" class="section level3 hasAnchor" number="14.3.3">
<h3><span class="header-section-number">14.3.3</span> Data processing<a href="examples.html#data-processing" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This part has too may procedures and we only choose some of them. Please go to <strong>Chapter 6</strong> to see more approaches and details for being familiar with this part.</p>
<div id="rarefy-otu-counts" class="section level4 hasAnchor" number="14.3.3.1">
<h4><span class="header-section-number">14.3.3.1</span> Rarefy otu counts<a href="examples.html#rarefy-otu-counts" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>From previous results of quality evaluation, the sequence depth of samples are different which have effects on the downstream analysis. Here, choosing the <strong>rarefy</strong> (Normaliztion method: random subsampling counts to the smallest library size) to get the equal sample sums.</p>
<div class="sourceCode" id="cb518"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb518-1"><a href="examples.html#cb518-1" tabindex="-1"></a>dada2_ps_rarefy <span class="ot">&lt;-</span> <span class="fu">norm_rarefy</span>(<span class="at">object =</span> dada2_ps_remove_BRS, </span>
<span id="cb518-2"><a href="examples.html#cb518-2" tabindex="-1"></a>                               <span class="at">size =</span> <span class="dv">51181</span>)</span>
<span id="cb518-3"><a href="examples.html#cb518-3" tabindex="-1"></a>dada2_ps_rarefy</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 891 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 891 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 891 tips and 888 internal nodes ]
## refseq()      DNAStringSet:      [ 891 reference sequences ]</code></pre>
<p>In addition, we could also perform some other normalization methods on the rarefied phyloseq object. By the way, we didn’t transform the data by using log algorithm because the count matrix is required by the following data analysis methods.</p>
</div>
<div id="extracting-specific-taxonomic-level" class="section level4 hasAnchor" number="14.3.3.2">
<h4><span class="header-section-number">14.3.3.2</span> Extracting specific taxonomic level<a href="examples.html#extracting-specific-taxonomic-level" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb520"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb520-1"><a href="examples.html#cb520-1" tabindex="-1"></a>dada2_ps_rare_genus <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(<span class="at">ps =</span> dada2_ps_rarefy, </span>
<span id="cb520-2"><a href="examples.html#cb520-2" tabindex="-1"></a>                                      <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>)</span>
<span id="cb520-3"><a href="examples.html#cb520-3" tabindex="-1"></a>dada2_ps_rare_genus</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 198 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 198 taxa by 6 taxonomic ranks ]</code></pre>
</div>
<div id="filtering-the-low-relative-abundance-or-unclassified-taxa-by-the-threshold-total-counts-10" class="section level4 hasAnchor" number="14.3.3.3">
<h4><span class="header-section-number">14.3.3.3</span> Filtering the low relative abundance or unclassified taxa by the threshold (total counts &lt; 10)<a href="examples.html#filtering-the-low-relative-abundance-or-unclassified-taxa-by-the-threshold-total-counts-10" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb522"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb522-1"><a href="examples.html#cb522-1" tabindex="-1"></a>dada2_ps_rare_genus_filter <span class="ot">&lt;-</span> <span class="fu">run_filter</span>(<span class="at">ps =</span> dada2_ps_rare_genus, </span>
<span id="cb522-2"><a href="examples.html#cb522-2" tabindex="-1"></a>                                         <span class="at">cutoff =</span> <span class="dv">10</span>, </span>
<span id="cb522-3"><a href="examples.html#cb522-3" tabindex="-1"></a>                                         <span class="at">unclass =</span> <span class="cn">TRUE</span>)</span>
<span id="cb522-4"><a href="examples.html#cb522-4" tabindex="-1"></a>dada2_ps_rare_genus_filter </span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 98 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 98 taxa by 6 taxonomic ranks ]</code></pre>
</div>
<div id="trimming-the-taxa-with-low-occurrence-less-than-threshold" class="section level4 hasAnchor" number="14.3.3.4">
<h4><span class="header-section-number">14.3.3.4</span> Trimming the taxa with low occurrence less than threshold<a href="examples.html#trimming-the-taxa-with-low-occurrence-less-than-threshold" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb524"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb524-1"><a href="examples.html#cb524-1" tabindex="-1"></a>dada2_ps_rare_genus_filter_trim <span class="ot">&lt;-</span> <span class="fu">run_trim</span>(<span class="at">object =</span> dada2_ps_rare_genus_filter, </span>
<span id="cb524-2"><a href="examples.html#cb524-2" tabindex="-1"></a>                                            <span class="at">cutoff =</span> <span class="fl">0.2</span>, </span>
<span id="cb524-3"><a href="examples.html#cb524-3" tabindex="-1"></a>                                            <span class="at">trim =</span> <span class="st">&quot;feature&quot;</span>)</span>
<span id="cb524-4"><a href="examples.html#cb524-4" tabindex="-1"></a>dada2_ps_rare_genus_filter_trim</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 83 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 83 taxa by 6 taxonomic ranks ]</code></pre>
<p>Finally, we obtained the final phyloseq-class object <em>dada2_ps_rare_genus_filter_trim</em> and changed its name.</p>
</div>
</div>
<div id="diversity-analysis" class="section level3 hasAnchor" number="14.3.4">
<h3><span class="header-section-number">14.3.4</span> Diversity analysis<a href="examples.html#diversity-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="alpha-diveristy" class="section level4 hasAnchor" number="14.3.4.1">
<h4><span class="header-section-number">14.3.4.1</span> Alpha diveristy<a href="examples.html#alpha-diveristy" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Calculate the alpha diversity</li>
</ul>
<p><strong>Notes:</strong> the otu table must be counts matrix (<em>rarefied but not trim counts matrix</em>) when you choose <em>Observed</em> etc measures.</p>
<div class="sourceCode" id="cb526"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb526-1"><a href="examples.html#cb526-1" tabindex="-1"></a>dada_ps_rare_genus_alpha <span class="ot">&lt;-</span> <span class="fu">run_alpha_diversity</span>(<span class="at">ps =</span> dada2_ps_rare_genus, </span>
<span id="cb526-2"><a href="examples.html#cb526-2" tabindex="-1"></a>                                                <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Chao1&quot;</span>, <span class="st">&quot;Observed&quot;</span>))</span>
<span id="cb526-3"><a href="examples.html#cb526-3" tabindex="-1"></a><span class="fu">print</span>(dada_ps_rare_genus_alpha)</span></code></pre></div>
<pre><code>##    TempRowNames Group Observed    Chao1  se.chao1  Shannon
## 1         S6030    BB       74 74.00000 0.0000000 2.622800
## 2         S6032    BB       38 38.00000 0.0000000 1.711068
## 3         S6033    BB       80 80.16667 0.5431879 2.719495
## 4         S6035    AA       35 35.00000 0.1232013 1.953807
## 5         S6036    BB       95 95.00000 0.0000000 2.942657
## 6         S6037    AA       35 35.00000 0.0000000 1.933007
## 7         S6040    BB       58 58.00000 0.0000000 2.490198
## 8         S6043    AA       65 65.33333 0.9246628 2.098003
## 9         S6045    BB       60 60.00000 0.0000000 2.920363
## 10        S6046    BB       71 71.00000 0.1241166 2.264433
## 11        S6048    BB       66 66.00000 0.0000000 2.630421
## 12        S6049    AA       60 60.00000 0.0000000 2.506885
## 13        S6050    BB       53 53.00000 0.0000000 2.431334
## 14        S6054    BB       84 84.00000 0.0000000 2.769973
## 15        S6055    BB       75 75.00000 0.0000000 2.297388
## 16        S6058    BB       37 37.00000 0.2465985 2.262059
## 17        S6059    AA       71 71.00000 0.0000000 2.682960
## 18        S6060    AA       48 48.00000 0.0000000 1.883809
## 19        S6061    AA       80 80.00000 0.0000000 3.088651
## 20        S6063    BB       38 38.00000 0.0000000 1.543267
## 21        S6065    AA       77 77.00000 0.0000000 2.877568
## 22        S6066    AA       67 67.00000 0.0000000 2.638564
## 23        S6068    BB       40 40.00000 0.0000000 2.100363</code></pre>
<ul>
<li>visualization</li>
</ul>
<div class="sourceCode" id="cb528"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb528-1"><a href="examples.html#cb528-1" tabindex="-1"></a><span class="fu">plot_boxplot</span>(<span class="at">data =</span> dada_ps_rare_genus_alpha,</span>
<span id="cb528-2"><a href="examples.html#cb528-2" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Chao1&quot;</span>, <span class="st">&quot;Observed&quot;</span>),</span>
<span id="cb528-3"><a href="examples.html#cb528-3" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb528-4"><a href="examples.html#cb528-4" tabindex="-1"></a>             <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb528-5"><a href="examples.html#cb528-5" tabindex="-1"></a>             <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb528-6"><a href="examples.html#cb528-6" tabindex="-1"></a>             <span class="at">method =</span> <span class="st">&quot;wilcox.test&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-278"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-278-1.png" alt="Alpha diversity (16s example)" width="576" />
<p class="caption">
Figure 14.5: Alpha diversity (16s example)
</p>
</div>
</div>
<div id="beta-diversity-1" class="section level4 hasAnchor" number="14.3.4.2">
<h4><span class="header-section-number">14.3.4.2</span> Beta diversity<a href="examples.html#beta-diversity-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>beta dipersion</li>
</ul>
<div class="sourceCode" id="cb529"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb529-1"><a href="examples.html#cb529-1" tabindex="-1"></a><span class="co"># dada2_ps_beta &lt;- run_beta_(ps = dada2_ps_rare_genus_filter_trim, </span></span>
<span id="cb529-2"><a href="examples.html#cb529-2" tabindex="-1"></a><span class="co">#                                     method = &quot;bray&quot;, </span></span>
<span id="cb529-3"><a href="examples.html#cb529-3" tabindex="-1"></a><span class="co">#                                     group = &quot;Group&quot;)</span></span>
<span id="cb529-4"><a href="examples.html#cb529-4" tabindex="-1"></a><span class="co"># dada2_ps_beta$BetaDispersion</span></span>
<span id="cb529-5"><a href="examples.html#cb529-5" tabindex="-1"></a></span>
<span id="cb529-6"><a href="examples.html#cb529-6" tabindex="-1"></a><span class="fu">run_beta_dispersion</span>(<span class="at">ps =</span> dada2_ps_rare_genus_filter_trim, </span>
<span id="cb529-7"><a href="examples.html#cb529-7" tabindex="-1"></a>          <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>, </span>
<span id="cb529-8"><a href="examples.html#cb529-8" tabindex="-1"></a>          <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<pre><code>## 
## Permutation test for homogeneity of multivariate dispersions
## Permutation: free
## Number of permutations: 999
## 
## Response: Distances
##           Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
## Groups     1 0.014137 0.0141374 1.6657    999  0.202
## Residuals 21 0.178236 0.0084874                     
## 
## Pairwise comparisons:
## (Observed p-value below diagonal, permuted p-value above diagonal)
##         AA   BB
## AA         0.21
## BB 0.21087</code></pre>
<pre><code>##        Df    Sum Sq   Mean Sq        F N.Perm Pr(&gt;F)
## Groups  1 0.0141374 0.0141374 1.665691    999  0.202</code></pre>
</div>
</div>
<div id="permanova-ordination" class="section level3 hasAnchor" number="14.3.5">
<h3><span class="header-section-number">14.3.5</span> PERMANOVA + Ordination<a href="examples.html#permanova-ordination" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="permanova" class="section level4 hasAnchor" number="14.3.5.1">
<h4><span class="header-section-number">14.3.5.1</span> PERMANOVA<a href="examples.html#permanova" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb532"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb532-1"><a href="examples.html#cb532-1" tabindex="-1"></a>dada2_ps_per <span class="ot">&lt;-</span> <span class="fu">run_permanova</span>(<span class="at">ps =</span> dada2_ps_rare_genus_filter_trim, </span>
<span id="cb532-2"><a href="examples.html#cb532-2" tabindex="-1"></a>                              <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>, </span>
<span id="cb532-3"><a href="examples.html#cb532-3" tabindex="-1"></a>                              <span class="at">columns =</span> <span class="st">&quot;Group&quot;</span>)</span>
<span id="cb532-4"><a href="examples.html#cb532-4" tabindex="-1"></a><span class="fu">print</span>(dada2_ps_per)</span></code></pre></div>
<pre><code>##       SumsOfSample Df SumsOfSqs   MeanSqs  F.Model         R2 Pr(&gt;F) AdjustedPvalue
## Group           23  1 0.2289051 0.2289051 1.333088 0.05969115  0.223          0.223</code></pre>
<p>The PERMANOVA result of the Group (<code>Pr(&gt;F)</code> &gt; 0.05) revealed that the two groups had not the distinct patterns of microbial community.</p>
</div>
<div id="ordination-1" class="section level4 hasAnchor" number="14.3.5.2">
<h4><span class="header-section-number">14.3.5.2</span> Ordination<a href="examples.html#ordination-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We performed ordination by using Principal Coordinate Analysis (PCoA). If you want to try other methods please go to see <strong>Chapter 8</strong> for more details.</p>
<div class="sourceCode" id="cb534"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb534-1"><a href="examples.html#cb534-1" tabindex="-1"></a>dada2_ps_ordination <span class="ot">&lt;-</span> <span class="fu">run_ordination</span>(</span>
<span id="cb534-2"><a href="examples.html#cb534-2" tabindex="-1"></a>                         <span class="at">ps =</span> dada2_ps_rare_genus_filter_trim,</span>
<span id="cb534-3"><a href="examples.html#cb534-3" tabindex="-1"></a>                         <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb534-4"><a href="examples.html#cb534-4" tabindex="-1"></a>                         <span class="at">method =</span> <span class="st">&quot;PCoA&quot;</span>)</span>
<span id="cb534-5"><a href="examples.html#cb534-5" tabindex="-1"></a></span>
<span id="cb534-6"><a href="examples.html#cb534-6" tabindex="-1"></a><span class="fu">plot_Ordination</span>(<span class="at">ResultList =</span> dada2_ps_ordination, </span>
<span id="cb534-7"><a href="examples.html#cb534-7" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>, </span>
<span id="cb534-8"><a href="examples.html#cb534-8" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb534-9"><a href="examples.html#cb534-9" tabindex="-1"></a>                <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-281"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-281-1.png" alt="PCoA (16s example)" width="768" />
<p class="caption">
Figure 14.6: PCoA (16s example)
</p>
</div>
</div>
</div>
<div id="microbial-composition-1" class="section level3 hasAnchor" number="14.3.6">
<h3><span class="header-section-number">14.3.6</span> Microbial composition<a href="examples.html#microbial-composition-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A whole picture of the microbial composition.</p>
<div id="stacked-barplot" class="section level4 hasAnchor" number="14.3.6.1">
<h4><span class="header-section-number">14.3.6.1</span> Stacked barplot<a href="examples.html#stacked-barplot" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li><strong>XMAS</strong> package</li>
</ul>
<div class="sourceCode" id="cb535"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb535-1"><a href="examples.html#cb535-1" tabindex="-1"></a><span class="fu">plot_StackBarPlot</span>(</span>
<span id="cb535-2"><a href="examples.html#cb535-2" tabindex="-1"></a>        <span class="at">ps =</span> dada2_ps_rarefy,</span>
<span id="cb535-3"><a href="examples.html#cb535-3" tabindex="-1"></a>        <span class="at">taxa_level =</span> <span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb535-4"><a href="examples.html#cb535-4" tabindex="-1"></a>        <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb535-5"><a href="examples.html#cb535-5" tabindex="-1"></a>        <span class="at">cluster =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## [1] &quot;This palatte have 20 colors!&quot;</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-282"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-282-1.png" alt="Microbial composition (16s example)" width="768" />
<p class="caption">
Figure 14.7: Microbial composition (16s example)
</p>
</div>
<ul>
<li><strong>XVIZ</strong> package</li>
</ul>
<div class="sourceCode" id="cb537"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb537-1"><a href="examples.html#cb537-1" tabindex="-1"></a><span class="fu">plot_stacked_bar_XIVZ</span>(</span>
<span id="cb537-2"><a href="examples.html#cb537-2" tabindex="-1"></a>        <span class="at">phyloseq =</span> dada2_ps_rarefy,</span>
<span id="cb537-3"><a href="examples.html#cb537-3" tabindex="-1"></a>        <span class="at">level =</span> <span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb537-4"><a href="examples.html#cb537-4" tabindex="-1"></a>        <span class="at">feature =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-283"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-283-1.png" alt="Microbial composition (16s example) XVIZ" width="768" />
<p class="caption">
Figure 14.8: Microbial composition (16s example) XVIZ
</p>
</div>
</div>
<div id="core-microbiota-1" class="section level4 hasAnchor" number="14.3.6.2">
<h4><span class="header-section-number">14.3.6.2</span> Core microbiota<a href="examples.html#core-microbiota-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>convert absolute abundance into relative abundance</li>
</ul>
<div class="sourceCode" id="cb538"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb538-1"><a href="examples.html#cb538-1" tabindex="-1"></a>dada2_ps_rare_genus_filter_trim_rb <span class="ot">&lt;-</span> XMAS2<span class="sc">::</span><span class="fu">normalize</span>(<span class="at">object =</span> dada2_ps_rare_genus_filter_trim, </span>
<span id="cb538-2"><a href="examples.html#cb538-2" tabindex="-1"></a>                                                       <span class="at">method =</span> <span class="st">&quot;TSS&quot;</span>)</span>
<span id="cb538-3"><a href="examples.html#cb538-3" tabindex="-1"></a>dada2_ps_rare_genus_filter_trim_rb</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 83 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 83 taxa by 6 taxonomic ranks ]</code></pre>
<ul>
<li>visualization</li>
</ul>
<div class="sourceCode" id="cb540"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb540-1"><a href="examples.html#cb540-1" tabindex="-1"></a>prevalences <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fl">0.05</span>, <span class="dv">1</span>, <span class="fl">0.05</span>)</span>
<span id="cb540-2"><a href="examples.html#cb540-2" tabindex="-1"></a>detections <span class="ot">&lt;-</span> <span class="dv">10</span><span class="sc">^</span><span class="fu">seq</span>(<span class="fu">log10</span>(<span class="fl">1e-3</span>), <span class="fu">log10</span>(<span class="fl">0.2</span>), <span class="at">length =</span> <span class="dv">10</span>)</span>
<span id="cb540-3"><a href="examples.html#cb540-3" tabindex="-1"></a></span>
<span id="cb540-4"><a href="examples.html#cb540-4" tabindex="-1"></a>pl_core <span class="ot">&lt;-</span> <span class="fu">plot_core_taxa</span>(dada2_ps_rare_genus_filter_trim_rb, </span>
<span id="cb540-5"><a href="examples.html#cb540-5" tabindex="-1"></a>                    <span class="at">plot.type =</span> <span class="st">&quot;heatmap&quot;</span>, </span>
<span id="cb540-6"><a href="examples.html#cb540-6" tabindex="-1"></a>                    <span class="at">colours =</span> <span class="fu">gray</span>(<span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="at">length=</span><span class="dv">5</span>)),</span>
<span id="cb540-7"><a href="examples.html#cb540-7" tabindex="-1"></a>                    <span class="at">prevalences =</span> prevalences, </span>
<span id="cb540-8"><a href="examples.html#cb540-8" tabindex="-1"></a>                    <span class="at">detections =</span> detections, </span>
<span id="cb540-9"><a href="examples.html#cb540-9" tabindex="-1"></a>                    <span class="at">min.prevalence =</span> <span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb540-10"><a href="examples.html#cb540-10" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">&quot;Detection Threshold (Relative Abundance (%))&quot;</span>)</span>
<span id="cb540-11"><a href="examples.html#cb540-11" tabindex="-1"></a></span>
<span id="cb540-12"><a href="examples.html#cb540-12" tabindex="-1"></a>pl_core</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-285"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-285-1.png" alt="Core taxa (16s example)" width="768" />
<p class="caption">
Figure 14.9: Core taxa (16s example)
</p>
</div>
<p>The degree of color indicates the size of abundance and prevalence.</p>
<ul>
<li>Use <code>core_members</code> to obtain the core taxa. detection for abundance and prevalence for occurrence.</li>
</ul>
<div class="sourceCode" id="cb541"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb541-1"><a href="examples.html#cb541-1" tabindex="-1"></a>core_taxa_name <span class="ot">&lt;-</span> <span class="fu">core_members</span>(dada2_ps_rare_genus_filter_trim_rb, <span class="at">detection =</span> <span class="fl">0.01</span>, <span class="at">prevalence =</span> <span class="fl">0.8</span>)</span>
<span id="cb541-2"><a href="examples.html#cb541-2" tabindex="-1"></a><span class="fu">print</span>(core_taxa_name)</span></code></pre></div>
<pre><code>## [1] &quot;g__Bifidobacterium&quot; &quot;g__Blautia&quot;</code></pre>
<p><strong>Result</strong>:</p>
<p>Only 2 genera (g__Bifidobacterium and g__Blautia) passed the threshold of detection and prevalence which we choose.</p>
</div>
</div>
<div id="differential-analysis-1" class="section level3 hasAnchor" number="14.3.7">
<h3><span class="header-section-number">14.3.7</span> Differential Analysis<a href="examples.html#differential-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>There are more than 10 approaches to perform differential analysis. Here, we choose two of them and recommend users going to <strong>Chapter 10</strong> to see more detials.</p>
<div id="liner-discriminant-analysis-lda-effect-size-lefse-2" class="section level4 hasAnchor" number="14.3.7.1">
<h4><span class="header-section-number">14.3.7.1</span> Liner discriminant analysis (LDA) effect size (LEfSe)<a href="examples.html#liner-discriminant-analysis-lda-effect-size-lefse-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb543"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb543-1"><a href="examples.html#cb543-1" tabindex="-1"></a><span class="co"># dada2_ps_lefse &lt;- run_lefse(</span></span>
<span id="cb543-2"><a href="examples.html#cb543-2" tabindex="-1"></a><span class="co">#                       ps = dada2_ps_rare_genus_filter_trim,</span></span>
<span id="cb543-3"><a href="examples.html#cb543-3" tabindex="-1"></a><span class="co">#                       group = &quot;Group&quot;,</span></span>
<span id="cb543-4"><a href="examples.html#cb543-4" tabindex="-1"></a><span class="co">#                       group_names = c(&quot;AA&quot;, &quot;BB&quot;),</span></span>
<span id="cb543-5"><a href="examples.html#cb543-5" tabindex="-1"></a><span class="co">#                       norm = &quot;CPM&quot;,</span></span>
<span id="cb543-6"><a href="examples.html#cb543-6" tabindex="-1"></a><span class="co">#                       Lda = 2)</span></span>
<span id="cb543-7"><a href="examples.html#cb543-7" tabindex="-1"></a></span>
<span id="cb543-8"><a href="examples.html#cb543-8" tabindex="-1"></a>dada2_ps_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse2</span>(</span>
<span id="cb543-9"><a href="examples.html#cb543-9" tabindex="-1"></a>                      <span class="at">ps =</span> dada2_ps_rare_genus_filter_trim,</span>
<span id="cb543-10"><a href="examples.html#cb543-10" tabindex="-1"></a>                      <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb543-11"><a href="examples.html#cb543-11" tabindex="-1"></a>                      <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb543-12"><a href="examples.html#cb543-12" tabindex="-1"></a>                      <span class="at">norm =</span> <span class="st">&quot;CPM&quot;</span>,</span>
<span id="cb543-13"><a href="examples.html#cb543-13" tabindex="-1"></a>                      <span class="at">lda_cutoff =</span> <span class="dv">2</span>)</span>
<span id="cb543-14"><a href="examples.html#cb543-14" tabindex="-1"></a><span class="fu">head</span>(dada2_ps_lefse)</span></code></pre></div>
<pre><code>##                           TaxaID         Block LDA_Score Enrichment EffectSize      Pvalue Log2FoldChange (Median)\nAA_vs_BB
## 1 g__Clostridium_sensu_stricto_1 9_AA vs 14_BB  3.989367         BB   2.393458 0.005225385                                NA
## 2             g__Intestinibacter 9_AA vs 14_BB  3.401850         BB   2.095183 0.007746971                                NA
## 3               g__Lactobacillus 9_AA vs 14_BB  4.541419         BB   2.351692 0.037446557                         -3.335507
## 4                 g__Odoribacter 9_AA vs 14_BB -2.956501         AA   1.743137 0.032745426                                NA
## 5              g__Parasutterella 9_AA vs 14_BB -3.997841         AA   2.091653 0.036513630                          4.400349
## 6                  g__Romboutsia 9_AA vs 14_BB  4.019602         BB   2.771988 0.007839137                         -6.859316
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All)
## 1               225.71974              0.00000           1177.85428                       -4.745276             5870.1925
## 2               383.70882              0.00000           1402.64816                       -1.963798             1886.7924
## 3              1286.16050            521.13869           5260.66779                       -5.172830            19970.3868
## 4                21.95438            605.54943              0.00000                        2.099245              579.0073
## 5               173.39662           1081.33826             51.20642                        4.481898             3909.0175
## 6              2195.63745             50.38418           5849.97508                       -3.333827             7104.3451
##   Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1           351.1518          9418.1472                    60.87                 22.22                 85.71    4900 (4900;4900)
## 2           682.1993          2661.1738                    60.87                 22.22                 85.71         3.4 (5.8;1)
## 3           893.5890         32234.0425                    86.96                 77.78                 92.86 1e+08 (1e+08;1e+08)
## 4          1085.5805           253.3531                    52.17                 77.78                 35.71       0.31 (-2;2.6)
## 5          9339.5437           417.9649                    65.22                 88.89                 50.00     0.0025 (-12;12)
## 6          1088.1764         10971.8822                    78.26                 55.56                 92.86          76 (84;67)</code></pre>
<ul>
<li>Visualization</li>
</ul>
<div class="sourceCode" id="cb545"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb545-1"><a href="examples.html#cb545-1" tabindex="-1"></a><span class="co"># # don&#39;t run this code when you do lefse in reality</span></span>
<span id="cb545-2"><a href="examples.html#cb545-2" tabindex="-1"></a><span class="co"># dada2_ps_lefse$LDA_Score &lt;- dada2_ps_lefse$LDA_Score * 1000</span></span>
<span id="cb545-3"><a href="examples.html#cb545-3" tabindex="-1"></a></span>
<span id="cb545-4"><a href="examples.html#cb545-4" tabindex="-1"></a><span class="fu">plot_lefse</span>(</span>
<span id="cb545-5"><a href="examples.html#cb545-5" tabindex="-1"></a>  <span class="at">da_res =</span> dada2_ps_lefse,</span>
<span id="cb545-6"><a href="examples.html#cb545-6" tabindex="-1"></a>  <span class="at">x_index =</span> <span class="st">&quot;LDA_Score&quot;</span>,</span>
<span id="cb545-7"><a href="examples.html#cb545-7" tabindex="-1"></a>  <span class="at">x_index_cutoff =</span> <span class="dv">1</span>,</span>
<span id="cb545-8"><a href="examples.html#cb545-8" tabindex="-1"></a>  <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-288"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-288-1.png" alt="Lefse analysis (16s example)" width="576" />
<p class="caption">
Figure 14.10: Lefse analysis (16s example)
</p>
</div>
</div>
<div id="wilcoxon-rank-sum-test" class="section level4 hasAnchor" number="14.3.7.2">
<h4><span class="header-section-number">14.3.7.2</span> Wilcoxon Rank-Sum test<a href="examples.html#wilcoxon-rank-sum-test" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb546"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb546-1"><a href="examples.html#cb546-1" tabindex="-1"></a>dada2_ps_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb546-2"><a href="examples.html#cb546-2" tabindex="-1"></a>                      <span class="at">ps =</span> dada2_ps_rare_genus_filter_trim,</span>
<span id="cb546-3"><a href="examples.html#cb546-3" tabindex="-1"></a>                      <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb546-4"><a href="examples.html#cb546-4" tabindex="-1"></a>                      <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb546-5"><a href="examples.html#cb546-5" tabindex="-1"></a></span>
<span id="cb546-6"><a href="examples.html#cb546-6" tabindex="-1"></a><span class="fu">head</span>(dada2_ps_wilcox)</span></code></pre></div>
<pre><code>##               TaxaID         Block Enrichment EffectSize Statistic    Pvalue AdjustedPvalue Log2FoldChange (Median)\nAA_vs_BB
## 1 g__Acidaminococcus 9_AA vs 14_BB  Nonsignif   70.56349      63.5 1.0000000      1.0000000                                NA
## 2     g__Actinomyces 9_AA vs 14_BB  Nonsignif   39.65079      44.5 0.2556616      0.6753999                        -0.9577718
## 3   g__Adlercreutzia 9_AA vs 14_BB  Nonsignif   23.17460      44.0 0.1980169      0.6753999                                NA
## 4    g__Agathobacter 9_AA vs 14_BB  Nonsignif 1222.58730      71.0 0.6293971      0.9164905                         0.1524904
## 5     g__Akkermansia 9_AA vs 14_BB  Nonsignif  104.78571      64.5 0.9304707      0.9775831                                NA
## 6       g__Alistipes 9_AA vs 14_BB  Nonsignif   23.03175      67.0 0.8239942      0.9497378                        -0.1085245
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Rank)\nAA_vs_BB Mean Rank Abundance\nAA
## 1                       0                    0                  0.0                      0.01201252                   12.06
## 2                      30                   26                 50.5                     -0.42227633                    9.94
## 3                       0                    0                  9.0                     -0.43387758                    9.89
## 4                     316                  329                296.0                      0.17342686                   12.89
## 5                       0                    0                  0.0                      0.03358045                   12.17
## 6                      64                   64                 69.0                      0.08724541                   12.44
##   Mean Rank Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                   11.96                    21.74                 22.22                 21.43    0.67 (-0.11;1.5)
## 2                   13.32                    82.61                 88.89                 78.57       4.1 (6.8;1.3)
## 3                   13.36                    43.48                 33.33                 50.00        8.6 (13;4.3)
## 4                   11.43                    65.22                 55.56                 71.43     0.39 (-1.4;2.2)
## 5                   11.89                    21.74                 22.22                 21.43      1.5 (2.3;0.71)
## 6                   11.71                    73.91                 77.78                 71.43     0.88 (0.64;1.1)</code></pre>
<ul>
<li>Volcano</li>
</ul>
<div class="sourceCode" id="cb548"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb548-1"><a href="examples.html#cb548-1" tabindex="-1"></a><span class="fu">plot_volcano</span>(</span>
<span id="cb548-2"><a href="examples.html#cb548-2" tabindex="-1"></a>    <span class="at">da_res =</span> dada2_ps_wilcox,</span>
<span id="cb548-3"><a href="examples.html#cb548-3" tabindex="-1"></a>    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb548-4"><a href="examples.html#cb548-4" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;Log2FoldChange (Rank)</span><span class="sc">\n</span><span class="st">AA_vs_BB&quot;</span>,</span>
<span id="cb548-5"><a href="examples.html#cb548-5" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb548-6"><a href="examples.html#cb548-6" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb548-7"><a href="examples.html#cb548-7" tabindex="-1"></a>    <span class="at">y_index_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb548-8"><a href="examples.html#cb548-8" tabindex="-1"></a>    <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb548-9"><a href="examples.html#cb548-9" tabindex="-1"></a>    <span class="at">topN =</span> <span class="dv">5</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-290"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-290-1.png" alt="Wilcoxon Rank-Sum test (16s example)" width="768" />
<p class="caption">
Figure 14.11: Wilcoxon Rank-Sum test (16s example)
</p>
</div>
</div>
</div>
</div>
<div id="metagenomics-mgs" class="section level2 hasAnchor" number="14.4">
<h2><span class="header-section-number">14.4</span> Metagenomics (MGS)<a href="examples.html#metagenomics-mgs" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The metagenomic data analysis pipeline is just the same as 16s. In briefly, the example comprises the following steps:</p>
<ol style="list-style-type: decimal">
<li><p>Converting inputs into phyloseq object;</p></li>
<li><p>Quality Evaluation;</p></li>
<li><p>Pre-Processing Data;</p></li>
<li><p>Diversity analysis;</p></li>
<li><p>Ordination analysis;</p></li>
<li><p>Composition analysis;</p></li>
</ol>
<div id="converting-inputs-into-phyloseq-class-object-1" class="section level3 hasAnchor" number="14.4.1">
<h3><span class="header-section-number">14.4.1</span> Converting inputs into phyloseq-class object<a href="examples.html#converting-inputs-into-phyloseq-class-object-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The result of the in-house <strong><a href="https://huttenhower.sph.harvard.edu/metaphlan2">Metaphlan2/3</a></strong> pipeline:</p>
<ol style="list-style-type: decimal">
<li><p>/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/MGS/metaphlan2_merged.tsv</p></li>
<li><p>/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/MGS/metadata.txt</p></li>
</ol>
<div class="sourceCode" id="cb549"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb549-1"><a href="examples.html#cb549-1" tabindex="-1"></a>metaphlan2_res <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb549-2"><a href="examples.html#cb549-2" tabindex="-1"></a>    <span class="fu">system.file</span>(</span>
<span id="cb549-3"><a href="examples.html#cb549-3" tabindex="-1"></a>        <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;metaphlan2_merged.tsv&quot;</span>,</span>
<span id="cb549-4"><a href="examples.html#cb549-4" tabindex="-1"></a>        <span class="at">package =</span> <span class="st">&quot;XMAS2&quot;</span></span>
<span id="cb549-5"><a href="examples.html#cb549-5" tabindex="-1"></a>    ),</span>
<span id="cb549-6"><a href="examples.html#cb549-6" tabindex="-1"></a>    <span class="at">header =</span> <span class="cn">TRUE</span>, </span>
<span id="cb549-7"><a href="examples.html#cb549-7" tabindex="-1"></a>    <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span></span>
<span id="cb549-8"><a href="examples.html#cb549-8" tabindex="-1"></a>)</span>
<span id="cb549-9"><a href="examples.html#cb549-9" tabindex="-1"></a>metaphlan2_sam <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb549-10"><a href="examples.html#cb549-10" tabindex="-1"></a>    <span class="fu">system.file</span>(</span>
<span id="cb549-11"><a href="examples.html#cb549-11" tabindex="-1"></a>        <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;metaphlan2_metadata.tsv&quot;</span>,</span>
<span id="cb549-12"><a href="examples.html#cb549-12" tabindex="-1"></a>        <span class="at">package =</span> <span class="st">&quot;XMAS2&quot;</span></span>
<span id="cb549-13"><a href="examples.html#cb549-13" tabindex="-1"></a>    ),</span>
<span id="cb549-14"><a href="examples.html#cb549-14" tabindex="-1"></a>    <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb549-15"><a href="examples.html#cb549-15" tabindex="-1"></a>    <span class="at">header =</span> <span class="cn">TRUE</span>,</span>
<span id="cb549-16"><a href="examples.html#cb549-16" tabindex="-1"></a>    <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span></span>
<span id="cb549-17"><a href="examples.html#cb549-17" tabindex="-1"></a>)</span>
<span id="cb549-18"><a href="examples.html#cb549-18" tabindex="-1"></a></span>
<span id="cb549-19"><a href="examples.html#cb549-19" tabindex="-1"></a>metaphlan2_res_list <span class="ot">&lt;-</span> <span class="fu">import_metaphlan_taxa</span>(<span class="at">data_metaphlan2 =</span> metaphlan2_res, </span>
<span id="cb549-20"><a href="examples.html#cb549-20" tabindex="-1"></a>                                             <span class="at">taxa_level =</span> <span class="st">&quot;Species&quot;</span>)</span>
<span id="cb549-21"><a href="examples.html#cb549-21" tabindex="-1"></a>otu_tab <span class="ot">&lt;-</span> metaphlan2_res_list<span class="sc">$</span>abu_tab</span>
<span id="cb549-22"><a href="examples.html#cb549-22" tabindex="-1"></a>tax_tab <span class="ot">&lt;-</span> metaphlan2_res_list<span class="sc">$</span>tax_tab</span>
<span id="cb549-23"><a href="examples.html#cb549-23" tabindex="-1"></a>sam_tab <span class="ot">&lt;-</span> metaphlan2_sam <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb549-24"><a href="examples.html#cb549-24" tabindex="-1"></a></span>
<span id="cb549-25"><a href="examples.html#cb549-25" tabindex="-1"></a>metaphlan2_ps <span class="ot">&lt;-</span> <span class="fu">get_metaphlan_phyloseq</span>(</span>
<span id="cb549-26"><a href="examples.html#cb549-26" tabindex="-1"></a>                    <span class="at">otu_tab =</span> otu_tab, </span>
<span id="cb549-27"><a href="examples.html#cb549-27" tabindex="-1"></a>                    <span class="at">sam_tab =</span> sam_tab,</span>
<span id="cb549-28"><a href="examples.html#cb549-28" tabindex="-1"></a>                    <span class="at">tax_tab =</span> tax_tab)</span>
<span id="cb549-29"><a href="examples.html#cb549-29" tabindex="-1"></a>metaphlan2_ps</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 322 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 322 taxa by 7 taxonomic ranks ]</code></pre>
<p>Here, the phyloseq object comprises three components (<em>OTU Table</em>, <em>Sample Data</em> and <em>Taxonomy Table</em>).</p>
<div id="summarize-phyloseq-class-object-2" class="section level4 hasAnchor" number="14.4.1.1">
<h4><span class="header-section-number">14.4.1.1</span> Summarize phyloseq-class object<a href="examples.html#summarize-phyloseq-class-object-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb551"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb551-1"><a href="examples.html#cb551-1" tabindex="-1"></a><span class="fu">summarize_phyloseq</span>(<span class="at">ps =</span> metaphlan2_ps)</span></code></pre></div>
<pre><code>## Compositional = NO2</code></pre>
<pre><code>## 1] Min. number of reads = 0.97918492] Max. number of reads = 1.00000023] Total number of reads = 22.92652624] Average number of reads = 0.9968054869565225] Median number of reads = 0.99980247] Sparsity = 0.7103699702943566] Any OTU sum to 1 or less? YES8] Number of singletons = 3199] Percent of OTUs that are singletons
##         (i.e. exactly one read detected across all samples)010] Number of sample variables are: 2Groupphynotype2</code></pre>
<pre><code>## [[1]]
## [1] &quot;1] Min. number of reads = 0.9791849&quot;
## 
## [[2]]
## [1] &quot;2] Max. number of reads = 1.0000002&quot;
## 
## [[3]]
## [1] &quot;3] Total number of reads = 22.9265262&quot;
## 
## [[4]]
## [1] &quot;4] Average number of reads = 0.996805486956522&quot;
## 
## [[5]]
## [1] &quot;5] Median number of reads = 0.9998024&quot;
## 
## [[6]]
## [1] &quot;7] Sparsity = 0.710369970294356&quot;
## 
## [[7]]
## [1] &quot;6] Any OTU sum to 1 or less? YES&quot;
## 
## [[8]]
## [1] &quot;8] Number of singletons = 319&quot;
## 
## [[9]]
## [1] &quot;9] Percent of OTUs that are singletons\n        (i.e. exactly one read detected across all samples)0&quot;
## 
## [[10]]
## [1] &quot;10] Number of sample variables are: 2&quot;
## 
## [[11]]
## [1] &quot;Group&quot;     &quot;phynotype&quot;</code></pre>
<p>Notice the <strong>Sparsity (0.713)</strong>, indicating the data has many zeros and pay attention to the downstream data analysis.</p>
</div>
</div>
<div id="quality-control-1" class="section level3 hasAnchor" number="14.4.2">
<h3><span class="header-section-number">14.4.2</span> Quality Control<a href="examples.html#quality-control-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="spike-in-sample-brs-assessment-1" class="section level4 hasAnchor" number="14.4.2.1">
<h4><span class="header-section-number">14.4.2.1</span> Spike-in sample (BRS) assessment<a href="examples.html#spike-in-sample-brs-assessment-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Extract the species level phyloseq and obtain the BRS_ID</li>
</ul>
<div class="sourceCode" id="cb555"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb555-1"><a href="examples.html#cb555-1" tabindex="-1"></a>metaphlan2_ps_species <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(<span class="at">ps =</span> metaphlan2_ps, </span>
<span id="cb555-2"><a href="examples.html#cb555-2" tabindex="-1"></a>                                        <span class="at">taxa_level =</span> <span class="st">&quot;Species&quot;</span>)</span>
<span id="cb555-3"><a href="examples.html#cb555-3" tabindex="-1"></a>metaphlan2_ps_species<span class="sc">@</span>sam_data</span></code></pre></div>
<pre><code>##      Group phynotype
## s1      BB      0.00
## s2      AA      2.50
## s3      BB      0.00
## s4      AA      1.25
## s5      AA     30.00
## s6      AA     15.00
## s7      BB      8.75
## s8      BB      0.00
## s9      BB      3.75
## s10     BB      2.50
## s11     BB     15.00
## s12     BB      2.50
## s13     BB      2.50
## s14     BB      0.00
## s15     BB      1.07
## s16     BB      2.50
## s17     AA      5.00
## s18     BB     35.00
## s19     BB      7.50
## s20     BB     15.00
## s21     AA      3.75
## s22     AA      3.75
## refE    QC        NA</code></pre>
<ul>
<li>Run <code>run_RefCheck</code></li>
</ul>
<div class="sourceCode" id="cb557"><pre class="sourceCode markdown"><code class="sourceCode markdown"><span id="cb557-1"><a href="examples.html#cb557-1" tabindex="-1"></a>run_RefCheck2(</span>
<span id="cb557-2"><a href="examples.html#cb557-2" tabindex="-1"></a>    ps = metaphlan2_ps_species,</span>
<span id="cb557-3"><a href="examples.html#cb557-3" tabindex="-1"></a>    BRS_ID = &quot;refE&quot;,</span>
<span id="cb557-4"><a href="examples.html#cb557-4" tabindex="-1"></a>    Ref_type = &quot;MGS&quot;)</span></code></pre></div>
<pre><code>## Noting: the Reference Matrix is for MGS
## 
## ############Matched baterica of the BRS sample#############
## The number of BRS&#39; bacteria matched the Reference Matrix is [15]
## s__Bifidobacterium_adolescentis
## s__Bifidobacterium_longum
## s__Bifidobacterium_pseudocatenulatum
## s__Collinsella_aerofaciens
## s__Bacteroides_ovatus
## s__Bacteroides_thetaiotaomicron
## s__Bacteroides_uniformis
## s__Bacteroides_vulgatus
## s__Bacteroides_xylanisolvens
## s__Prevotella_copri
## s__Streptococcus_salivarius
## s__Coprococcus_comes
## s__Dorea_formicigenerans
## s__Roseburia_hominis
## s__Faecalibacterium_prausnitzii
## The number of bacteria unmatched the Reference Matrix is [11]
## s__Propionibacterium_acnes
## s__Bifidobacterium_bifidum
## s__Eggerthella_unclassified
## s__Bacteroides_fragilis
## s__Bacteroides_intestinalis
## s__Parabacteroides_goldsteinii
## s__Enterococcus_faecalis
## s__Enterococcus_faecium
## s__Lactobacillus_pentosus
## s__Lactobacillus_salivarius
## s__Escherichia_coli
## The number of the additional bacteria compared to the Reference Matrix is [57]
## ###########################################################
## 
## ##################Status of the BRS sample##################
## Whether the BRS has the all bateria of Reference Matrix: FALSE
## Correlation Coefficient of the BRS is: -0.1179
## Bray Curtis of the BRS is: 0.8761
## Impurity of the BRS is: 32.76
## ###########################################################
## #####Final Evaluation Results of the BRS #######
## The BRS of sequencing dataset didn&#39;t pass the cutoff of the Reference Matrix
## ###########################################################
## $Matrix
##                                       7682     7683     7684     7685     7842     7843     7844     7845     refE        mean
## Propionibacterium_acnes            0.28649  0.22355  0.32199  0.28276  0.37568  0.44201  0.41135  0.39489       NA  0.30430222
## Bifidobacterium_adolescentis       7.05611  6.28460  6.57297  6.25448  4.69357  4.80628  4.94943  4.84278  0.04649  5.05630111
## Bifidobacterium_bifidum            1.32803  1.19346  1.26883  1.05154  1.34640  1.40471  1.43721  1.48097       NA  1.16790556
## Bifidobacterium_longum            10.31832  9.25812  9.69184  7.76031 11.03311 11.61484 12.29030 11.69019  0.01646  9.29705444
## Bifidobacterium_pseudocatenulatum  6.89760  6.36177  7.39605  5.76804  6.19464  6.47094  7.41615  6.27837  0.15023  5.88153222
## Collinsella_aerofaciens            0.56189  0.53249  0.60476  0.47934  0.65513  0.68833  0.76063  0.66896  0.07251  0.55822667
## Eggerthella_unclassified           1.21416  1.06143  1.19122  0.85743  1.36030  1.48457  1.61204  1.50700       NA  1.14312778
## Bacteroides_fragilis               6.56752  7.19828  7.27658  7.31998  7.10174  6.70626  6.60219  6.11088       NA  6.09815889
## Bacteroides_intestinalis           0.09448  0.09550  0.10216  0.09326  0.08448  0.09556  0.08852  0.10469       NA  0.08429444
## Bacteroides_ovatus                 3.08552  3.27226  3.10904  3.24565  3.51376  3.50063  3.37872  3.42300  0.25782  2.97626667
## Bacteroides_thetaiotaomicron       3.24207  3.31897  3.22418  3.43809  3.35611  3.38323  3.29098  3.30355  1.47422  3.11460000
## Bacteroides_uniformis              2.24271  2.24035  1.92015  2.18435  2.43230  2.38180  2.13830  2.41437  0.21061  2.01832667
## Bacteroides_vulgatus               3.06672  3.20369  3.14979  3.15352  3.24822  3.09280  3.13038  3.06113  2.14684  3.02812111
## Bacteroides_xylanisolvens          1.55687  1.84824  1.91166  1.85273  1.75220  1.74002  1.69811  1.67676  0.32466  1.59569444
## Parabacteroides_goldsteinii        5.92564  5.95499  6.04638  5.85282  6.85811  6.50800  6.35758  6.96646       NA  5.60777556
## Prevotella_copri                   2.03757  1.99619  1.92504  2.15422  1.57638  1.60584  1.57913  1.59224 60.84109  8.36752222
## Enterococcus_faecalis             11.46695 12.46200 11.92117 11.95884 13.71474 13.67805 13.21782 13.27297       NA 11.29917111
## Enterococcus_faecium               4.57147  4.64943  4.52463  4.76407  5.45590  5.17641  5.15276  5.24896       NA  4.39373667
## Lactobacillus_pentosus             0.75844  0.75524  0.72958  0.80796  0.91548  0.89940  0.81308  0.94831       NA  0.73638778
## Lactobacillus_salivarius           4.89005  5.18462  5.12745  7.79194  2.72469  2.25673  2.10356  2.02202       NA  3.56678444
## Streptococcus_salivarius           3.54220  3.74119  3.62036  4.01546  2.90216  2.70193  2.61760  2.57348  0.02846  2.86031556
## Coprococcus_comes                  2.28009  2.42978  2.25227  2.82904  1.21527  1.16485  1.07160  1.07834  0.01444  1.59285333
## Dorea_formicigenerans              4.83413  5.02149  5.18268  5.56891  3.34720  3.09877  3.08677  2.70732  0.02775  3.65278000
## Roseburia_hominis                  0.04384  0.04183  0.04107  0.02304  0.03853  0.03464  0.03532  0.03597  0.01307  0.03414556
## Faecalibacterium_prausnitzii       0.65572  0.60153  0.60112  0.62079  0.54147  0.55383  0.58806  0.54655  1.61939  0.70316222
## Escherichia_coli                   8.76957  8.33337  7.70416  8.11247 10.44118 10.33823  9.96261 11.06988       NA  8.30349667
## Impurity_level                     2.70584  2.73563  2.58287  1.75896  3.12125  4.17134  4.20980  4.97996 32.76000  6.55840556
##                                                                              Evaluation
## Propionibacterium_acnes           refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Bifidobacterium_adolescentis      refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Bifidobacterium_bifidum           refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Bifidobacterium_longum            refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Bifidobacterium_pseudocatenulatum refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Collinsella_aerofaciens           refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Eggerthella_unclassified          refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Bacteroides_fragilis              refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Bacteroides_intestinalis          refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Bacteroides_ovatus                refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Bacteroides_thetaiotaomicron      refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Bacteroides_uniformis             refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Bacteroides_vulgatus              refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Bacteroides_xylanisolvens         refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Parabacteroides_goldsteinii       refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Prevotella_copri                  refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Enterococcus_faecalis             refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Enterococcus_faecium              refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Lactobacillus_pentosus            refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Lactobacillus_salivarius          refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Streptococcus_salivarius          refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Coprococcus_comes                 refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Dorea_formicigenerans             refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Roseburia_hominis                 refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Faecalibacterium_prausnitzii      refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Escherichia_coli                  refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## Impurity_level                    refE didn&#39;t pass the threshold (2023-11-30 17:04:24).
## 
## $Assess
##          Gold_Cutoff     BRS
## Coef          0.8726 -0.1179
## Bray          0.2064  0.8761
## Impurity      6.4400 32.7600</code></pre>
<p>The spike-in samples didn’t pass the cutoff and failed to add the the Reference Matrix.</p>
</div>
<div id="spike-in-samples-remove-1" class="section level4 hasAnchor" number="14.4.2.2">
<h4><span class="header-section-number">14.4.2.2</span> Spike-in sample’s remove<a href="examples.html#spike-in-samples-remove-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb559"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb559-1"><a href="examples.html#cb559-1" tabindex="-1"></a>metaphlan2_ps_remove_BRS <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb559-2"><a href="examples.html#cb559-2" tabindex="-1"></a>                               <span class="at">ps =</span> metaphlan2_ps,</span>
<span id="cb559-3"><a href="examples.html#cb559-3" tabindex="-1"></a>                               <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb559-4"><a href="examples.html#cb559-4" tabindex="-1"></a>                               <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>,</span>
<span id="cb559-5"><a href="examples.html#cb559-5" tabindex="-1"></a>                               <span class="at">discard =</span> <span class="cn">TRUE</span>)</span>
<span id="cb559-6"><a href="examples.html#cb559-6" tabindex="-1"></a>metaphlan2_ps_remove_BRS</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 322 taxa and 22 samples ]
## sample_data() Sample Data:       [ 22 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 322 taxa by 7 taxonomic ranks ]</code></pre>
</div>
</div>
<div id="data-processing-1" class="section level3 hasAnchor" number="14.4.3">
<h3><span class="header-section-number">14.4.3</span> Data processing<a href="examples.html#data-processing-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This part has too may procedures and we only choose some of them. Please go to <strong>Chapter 6</strong> to see more approaches and details for being familiar with this part.</p>
<div id="extracting-specific-taxonomic-level-1" class="section level4 hasAnchor" number="14.4.3.1">
<h4><span class="header-section-number">14.4.3.1</span> Extracting specific taxonomic level<a href="examples.html#extracting-specific-taxonomic-level-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb561"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb561-1"><a href="examples.html#cb561-1" tabindex="-1"></a>metaphlan2_ps_species <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(<span class="at">ps =</span> metaphlan2_ps_remove_BRS, </span>
<span id="cb561-2"><a href="examples.html#cb561-2" tabindex="-1"></a>                                        <span class="at">taxa_level =</span> <span class="st">&quot;Species&quot;</span>)</span>
<span id="cb561-3"><a href="examples.html#cb561-3" tabindex="-1"></a>metaphlan2_ps_species</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 321 taxa and 22 samples ]
## sample_data() Sample Data:       [ 22 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 321 taxa by 7 taxonomic ranks ]</code></pre>
</div>
<div id="filtering-the-low-relative-abundance-or-unclassified-taxa-by-the-threshold-total-counts-1e-4" class="section level4 hasAnchor" number="14.4.3.2">
<h4><span class="header-section-number">14.4.3.2</span> Filtering the low relative abundance or unclassified taxa by the threshold (total counts &lt; 1e-4)<a href="examples.html#filtering-the-low-relative-abundance-or-unclassified-taxa-by-the-threshold-total-counts-1e-4" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The condition to filter low relative abundance is according to this article <span class="citation">(<a href="#ref-thingholm2019obese">Thingholm et al. 2019</a>)</span>.</p>
<blockquote>
<p>Species from taxonomic profiles were retained for further analysis if their mean relative abundance exceeded 0.005
(0.5%) across the dataset with a minimum abundance of 0.05 (5%) in at least one sample and non-zero abundance in at least
60% of samples.</p>
</blockquote>
<p>There are three conditions</p>
<ol style="list-style-type: decimal">
<li><p>Mean relative abundance: 0.005;</p></li>
<li><p>Minimum relative abundance: 0.05;</p></li>
<li><p>Occurrence: 60%.</p></li>
</ol>
<p>Here, we use 0.01 (the 1e-4 regarded as 0.01 compared to the Referece because Metaphlan2 data had been divided 100)</p>
<div class="sourceCode" id="cb563"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb563-1"><a href="examples.html#cb563-1" tabindex="-1"></a>metaphlan2_ps_species_filter <span class="ot">&lt;-</span> <span class="fu">run_filter</span>(<span class="at">ps =</span> metaphlan2_ps_species, </span>
<span id="cb563-2"><a href="examples.html#cb563-2" tabindex="-1"></a>                                           <span class="at">cutoff =</span> <span class="fl">1e-4</span>, </span>
<span id="cb563-3"><a href="examples.html#cb563-3" tabindex="-1"></a>                                           <span class="at">unclass =</span> <span class="cn">TRUE</span>)</span>
<span id="cb563-4"><a href="examples.html#cb563-4" tabindex="-1"></a>metaphlan2_ps_species_filter </span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 158 taxa and 22 samples ]
## sample_data() Sample Data:       [ 22 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 158 taxa by 7 taxonomic ranks ]</code></pre>
</div>
<div id="trimming-the-taxa-with-low-occurrence-less-than-threshold-1" class="section level4 hasAnchor" number="14.4.3.3">
<h4><span class="header-section-number">14.4.3.3</span> Trimming the taxa with low occurrence less than threshold<a href="examples.html#trimming-the-taxa-with-low-occurrence-less-than-threshold-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb565"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb565-1"><a href="examples.html#cb565-1" tabindex="-1"></a>metaphlan2_ps_species_filter_trim <span class="ot">&lt;-</span> <span class="fu">run_trim</span>(<span class="at">object =</span> metaphlan2_ps_species_filter, </span>
<span id="cb565-2"><a href="examples.html#cb565-2" tabindex="-1"></a>                                              <span class="at">cutoff =</span> <span class="fl">0.1</span>, </span>
<span id="cb565-3"><a href="examples.html#cb565-3" tabindex="-1"></a>                                              <span class="at">trim =</span> <span class="st">&quot;feature&quot;</span>)</span>
<span id="cb565-4"><a href="examples.html#cb565-4" tabindex="-1"></a>metaphlan2_ps_species_filter_trim</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 140 taxa and 22 samples ]
## sample_data() Sample Data:       [ 22 samples by 2 sample variables ]
## tax_table()   Taxonomy Table:    [ 140 taxa by 7 taxonomic ranks ]</code></pre>
<p>Finally, we obtained the final phyloseq-class object <em>metaphlan2_ps_species_filter_trim</em> and changed its name.</p>
</div>
</div>
<div id="diversity-analysis-1" class="section level3 hasAnchor" number="14.4.4">
<h3><span class="header-section-number">14.4.4</span> Diversity analysis<a href="examples.html#diversity-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="alpha-diveristy-1" class="section level4 hasAnchor" number="14.4.4.1">
<h4><span class="header-section-number">14.4.4.1</span> Alpha diveristy<a href="examples.html#alpha-diveristy-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Calculate the alpha diversity</li>
</ul>
<p><strong>Notes:</strong> choosing the measures (<em>Shannon, Simpson and InvSimpson</em>) only for relative abundance.</p>
<div class="sourceCode" id="cb567"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb567-1"><a href="examples.html#cb567-1" tabindex="-1"></a>metaphlan2_ps_genus_alpha <span class="ot">&lt;-</span> <span class="fu">run_alpha_diversity</span>(<span class="at">ps =</span> metaphlan2_ps_remove_BRS, </span>
<span id="cb567-2"><a href="examples.html#cb567-2" tabindex="-1"></a>                                                 <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Simpson&quot;</span>, <span class="st">&quot;InvSimpson&quot;</span>))</span>
<span id="cb567-3"><a href="examples.html#cb567-3" tabindex="-1"></a><span class="fu">head</span>(metaphlan2_ps_genus_alpha)</span></code></pre></div>
<pre><code>##   TempRowNames Group phynotype  Shannon   Simpson InvSimpson
## 1           s1    BB      0.00 2.876002 0.9108857  11.221549
## 2           s2    AA      2.50 2.045392 0.8105742   5.279111
## 3           s3    BB      0.00 3.441176 0.9439371  17.837114
## 4           s4    AA      1.25 2.746155 0.8290417   5.849380
## 5           s5    AA     30.00 1.450722 0.6412178   2.787207
## 6           s6    AA     15.00 2.619951 0.8950369   9.527154</code></pre>
<ul>
<li>visualization</li>
</ul>
<div class="sourceCode" id="cb569"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb569-1"><a href="examples.html#cb569-1" tabindex="-1"></a><span class="fu">plot_boxplot</span>(<span class="at">data =</span> metaphlan2_ps_genus_alpha,</span>
<span id="cb569-2"><a href="examples.html#cb569-2" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Simpson&quot;</span>, <span class="st">&quot;InvSimpson&quot;</span>),</span>
<span id="cb569-3"><a href="examples.html#cb569-3" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb569-4"><a href="examples.html#cb569-4" tabindex="-1"></a>             <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb569-5"><a href="examples.html#cb569-5" tabindex="-1"></a>             <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb569-6"><a href="examples.html#cb569-6" tabindex="-1"></a>             <span class="at">method =</span> <span class="st">&quot;wilcox.test&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-300"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-300-1.png" alt="Alpha diversity (MGS example)" width="576" />
<p class="caption">
Figure 14.12: Alpha diversity (MGS example)
</p>
</div>
</div>
<div id="beta-diversity-2" class="section level4 hasAnchor" number="14.4.4.2">
<h4><span class="header-section-number">14.4.4.2</span> Beta diversity<a href="examples.html#beta-diversity-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>beta dipersion</li>
</ul>
<div class="sourceCode" id="cb570"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb570-1"><a href="examples.html#cb570-1" tabindex="-1"></a><span class="co"># metaphlan2_ps_beta &lt;- run_beta_diversity(ps = metaphlan2_ps_species_filter_trim, </span></span>
<span id="cb570-2"><a href="examples.html#cb570-2" tabindex="-1"></a><span class="co">#                                          method = &quot;bray&quot;, </span></span>
<span id="cb570-3"><a href="examples.html#cb570-3" tabindex="-1"></a><span class="co">#                                          group = &quot;Group&quot;)</span></span>
<span id="cb570-4"><a href="examples.html#cb570-4" tabindex="-1"></a><span class="co"># metaphlan2_ps_beta$BetaDispersion</span></span>
<span id="cb570-5"><a href="examples.html#cb570-5" tabindex="-1"></a></span>
<span id="cb570-6"><a href="examples.html#cb570-6" tabindex="-1"></a></span>
<span id="cb570-7"><a href="examples.html#cb570-7" tabindex="-1"></a><span class="fu">run_beta_dispersion</span>(<span class="at">ps =</span> metaphlan2_ps_species_filter_trim, </span>
<span id="cb570-8"><a href="examples.html#cb570-8" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>, </span>
<span id="cb570-9"><a href="examples.html#cb570-9" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<pre><code>## 
## Permutation test for homogeneity of multivariate dispersions
## Permutation: free
## Number of permutations: 999
## 
## Response: Distances
##           Df   Sum Sq   Mean Sq      F N.Perm Pr(&gt;F)
## Groups     1 0.006014 0.0060141 0.7954    999  0.371
## Residuals 20 0.151230 0.0075615                     
## 
## Pairwise comparisons:
## (Observed p-value below diagonal, permuted p-value above diagonal)
##         AA    BB
## AA         0.365
## BB 0.38309</code></pre>
<pre><code>##        Df      Sum Sq     Mean Sq         F N.Perm Pr(&gt;F)
## Groups  1 0.006014143 0.006014143 0.7953615    999  0.371</code></pre>
</div>
</div>
<div id="permanova-ordination-1" class="section level3 hasAnchor" number="14.4.5">
<h3><span class="header-section-number">14.4.5</span> PERMANOVA + Ordination<a href="examples.html#permanova-ordination-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="permanova-1" class="section level4 hasAnchor" number="14.4.5.1">
<h4><span class="header-section-number">14.4.5.1</span> PERMANOVA<a href="examples.html#permanova-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb573"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb573-1"><a href="examples.html#cb573-1" tabindex="-1"></a>metaphlan2_ps_per <span class="ot">&lt;-</span> <span class="fu">run_permanova</span>(<span class="at">ps =</span> metaphlan2_ps_species_filter_trim, </span>
<span id="cb573-2"><a href="examples.html#cb573-2" tabindex="-1"></a>                                   <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>, </span>
<span id="cb573-3"><a href="examples.html#cb573-3" tabindex="-1"></a>                                   <span class="at">columns =</span> <span class="st">&quot;Group&quot;</span>)</span>
<span id="cb573-4"><a href="examples.html#cb573-4" tabindex="-1"></a><span class="fu">head</span>(metaphlan2_ps_per)</span></code></pre></div>
<pre><code>##       SumsOfSample Df SumsOfSqs  MeanSqs F.Model        R2 Pr(&gt;F) AdjustedPvalue
## Group           22  1  0.845706 0.845706 2.73086 0.1201389  0.002          0.002</code></pre>
<p>The PERMANOVA result of the Group (AdjustedPvalue &lt; 0.05) revealed that the two groups had the distinct patterns of microbial community.</p>
</div>
<div id="ordination-2" class="section level4 hasAnchor" number="14.4.5.2">
<h4><span class="header-section-number">14.4.5.2</span> Ordination<a href="examples.html#ordination-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We performed ordination by using Uniform Manifold Approximation and Projection for Dimension Reduction (UMAP). If you want to try other methods please go to see <strong>Chapter 8</strong> for more details.</p>
<div class="sourceCode" id="cb575"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb575-1"><a href="examples.html#cb575-1" tabindex="-1"></a>metaphlan2_ps_ordination <span class="ot">&lt;-</span> <span class="fu">run_ordination</span>(</span>
<span id="cb575-2"><a href="examples.html#cb575-2" tabindex="-1"></a>                       <span class="at">ps =</span> metaphlan2_ps_species_filter_trim,</span>
<span id="cb575-3"><a href="examples.html#cb575-3" tabindex="-1"></a>                       <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb575-4"><a href="examples.html#cb575-4" tabindex="-1"></a>                       <span class="at">method =</span> <span class="st">&quot;UMAP&quot;</span>)</span>
<span id="cb575-5"><a href="examples.html#cb575-5" tabindex="-1"></a></span>
<span id="cb575-6"><a href="examples.html#cb575-6" tabindex="-1"></a><span class="fu">plot_Ordination</span>(<span class="at">ResultList =</span> metaphlan2_ps_ordination, </span>
<span id="cb575-7"><a href="examples.html#cb575-7" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>, </span>
<span id="cb575-8"><a href="examples.html#cb575-8" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb575-9"><a href="examples.html#cb575-9" tabindex="-1"></a>                <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>),</span>
<span id="cb575-10"><a href="examples.html#cb575-10" tabindex="-1"></a>                <span class="at">sample =</span> <span class="cn">TRUE</span>,</span>
<span id="cb575-11"><a href="examples.html#cb575-11" tabindex="-1"></a>                <span class="at">sidelinechart =</span> <span class="cn">FALSE</span>,</span>
<span id="cb575-12"><a href="examples.html#cb575-12" tabindex="-1"></a>                <span class="at">circle_type =</span> <span class="st">&quot;ellipse_line&quot;</span>,</span>
<span id="cb575-13"><a href="examples.html#cb575-13" tabindex="-1"></a>                <span class="at">sideboxplot =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-303"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-303-1.png" alt="PCoA (MGS example)" width="480" />
<p class="caption">
Figure 14.13: PCoA (MGS example)
</p>
</div>
</div>
</div>
<div id="microbial-composition-2" class="section level3 hasAnchor" number="14.4.6">
<h3><span class="header-section-number">14.4.6</span> Microbial composition<a href="examples.html#microbial-composition-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="stacked-barplot-1" class="section level4 hasAnchor" number="14.4.6.1">
<h4><span class="header-section-number">14.4.6.1</span> Stacked barplot<a href="examples.html#stacked-barplot-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li><strong>XVIZ</strong> package</li>
</ul>
<div class="sourceCode" id="cb576"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb576-1"><a href="examples.html#cb576-1" tabindex="-1"></a><span class="fu">plot_stacked_bar_XIVZ</span>(</span>
<span id="cb576-2"><a href="examples.html#cb576-2" tabindex="-1"></a>        <span class="at">phyloseq =</span> metaphlan2_ps_species,</span>
<span id="cb576-3"><a href="examples.html#cb576-3" tabindex="-1"></a>        <span class="at">level =</span> <span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb576-4"><a href="examples.html#cb576-4" tabindex="-1"></a>        <span class="at">feature =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-304"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-304-1.png" alt="Microbial composition (MGS) XVIZ" width="768" />
<p class="caption">
Figure 14.14: Microbial composition (MGS) XVIZ
</p>
</div>
<ul>
<li><strong>XMAS</strong> package</li>
</ul>
<div class="sourceCode" id="cb577"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb577-1"><a href="examples.html#cb577-1" tabindex="-1"></a><span class="fu">plot_StackBarPlot</span>(</span>
<span id="cb577-2"><a href="examples.html#cb577-2" tabindex="-1"></a>        <span class="at">ps =</span> metaphlan2_ps_species,</span>
<span id="cb577-3"><a href="examples.html#cb577-3" tabindex="-1"></a>        <span class="at">taxa_level =</span> <span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb577-4"><a href="examples.html#cb577-4" tabindex="-1"></a>        <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb577-5"><a href="examples.html#cb577-5" tabindex="-1"></a>        <span class="at">cluster =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## [1] &quot;This palatte have 20 colors!&quot;</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-305"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-305-1.png" alt="Microbial composition (MGS example)" width="768" />
<p class="caption">
Figure 14.15: Microbial composition (MGS example)
</p>
</div>
</div>
<div id="core-microbiota-2" class="section level4 hasAnchor" number="14.4.6.2">
<h4><span class="header-section-number">14.4.6.2</span> Core microbiota<a href="examples.html#core-microbiota-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>visualization</li>
</ul>
<div class="sourceCode" id="cb579"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb579-1"><a href="examples.html#cb579-1" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb579-2"><a href="examples.html#cb579-2" tabindex="-1"></a></span>
<span id="cb579-3"><a href="examples.html#cb579-3" tabindex="-1"></a>prevalences <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fl">0.05</span>, <span class="dv">1</span>, <span class="fl">0.05</span>)</span>
<span id="cb579-4"><a href="examples.html#cb579-4" tabindex="-1"></a>detections <span class="ot">&lt;-</span> <span class="dv">10</span><span class="sc">^</span><span class="fu">seq</span>(<span class="fu">log10</span>(<span class="fl">1e-3</span>), <span class="fu">log10</span>(.<span class="dv">2</span>), <span class="at">length =</span> <span class="dv">10</span>)</span>
<span id="cb579-5"><a href="examples.html#cb579-5" tabindex="-1"></a></span>
<span id="cb579-6"><a href="examples.html#cb579-6" tabindex="-1"></a><span class="fu">plot_core_taxa</span>(metaphlan2_ps_species_filter_trim, </span>
<span id="cb579-7"><a href="examples.html#cb579-7" tabindex="-1"></a>               <span class="at">plot.type =</span> <span class="st">&quot;heatmap&quot;</span>, </span>
<span id="cb579-8"><a href="examples.html#cb579-8" tabindex="-1"></a>               <span class="at">colours =</span> <span class="fu">rev</span>(<span class="fu">brewer.pal</span>(<span class="dv">5</span>, <span class="st">&quot;Spectral&quot;</span>)),</span>
<span id="cb579-9"><a href="examples.html#cb579-9" tabindex="-1"></a>               <span class="at">prevalences =</span> prevalences, </span>
<span id="cb579-10"><a href="examples.html#cb579-10" tabindex="-1"></a>               <span class="at">detections =</span> detections, </span>
<span id="cb579-11"><a href="examples.html#cb579-11" tabindex="-1"></a>               <span class="at">min.prevalence =</span> <span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb579-12"><a href="examples.html#cb579-12" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">&quot;Detection Threshold (Relative Abundance (%))&quot;</span>) <span class="sc">+</span></span>
<span id="cb579-13"><a href="examples.html#cb579-13" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">&quot;italic&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-306"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-306-1.png" alt="Core taxa (MGS example)" width="768" />
<p class="caption">
Figure 14.16: Core taxa (MGS example)
</p>
</div>
<p>The degree of color indicates the size of abundance and prevalence.</p>
<ul>
<li>Use <code>core_members</code> to obtain the core taxa. detection for abundance and prevalence for occurrence.</li>
</ul>
<div class="sourceCode" id="cb580"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb580-1"><a href="examples.html#cb580-1" tabindex="-1"></a>core_taxa_name <span class="ot">&lt;-</span> <span class="fu">core_members</span>(metaphlan2_ps_species_filter_trim, </span>
<span id="cb580-2"><a href="examples.html#cb580-2" tabindex="-1"></a>                               <span class="at">detection =</span> <span class="fl">0.001</span>, </span>
<span id="cb580-3"><a href="examples.html#cb580-3" tabindex="-1"></a>                               <span class="at">prevalence =</span> <span class="fl">0.5</span>)</span>
<span id="cb580-4"><a href="examples.html#cb580-4" tabindex="-1"></a><span class="fu">print</span>(core_taxa_name)</span></code></pre></div>
<pre><code>##  [1] &quot;s__Bacteroides_ovatus&quot;                &quot;s__Bacteroides_thetaiotaomicron&quot;      &quot;s__Bacteroides_uniformis&quot;            
##  [4] &quot;s__Bacteroides_vulgatus&quot;              &quot;s__Bifidobacterium_longum&quot;            &quot;s__Bifidobacterium_pseudocatenulatum&quot;
##  [7] &quot;s__Collinsella_aerofaciens&quot;           &quot;s__Escherichia_coli&quot;                  &quot;s__Eubacterium_eligens&quot;              
## [10] &quot;s__Eubacterium_hallii&quot;                &quot;s__Faecalibacterium_prausnitzii&quot;      &quot;s__Roseburia_inulinivorans&quot;          
## [13] &quot;s__Ruminococcus_gnavus&quot;               &quot;s__Ruminococcus_obeum&quot;                &quot;s__Ruminococcus_sp_5_1_39BFAA&quot;       
## [16] &quot;s__Ruminococcus_torques&quot;              &quot;s__Streptococcus_salivarius&quot;</code></pre>
<p><strong>Result</strong>:</p>
<p>17 species passed the threshold of detection and prevalence which we choose.</p>
</div>
</div>
<div id="differential-analysis-2" class="section level3 hasAnchor" number="14.4.7">
<h3><span class="header-section-number">14.4.7</span> Differential Analysis<a href="examples.html#differential-analysis-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>There are more than 10 approaches to perform differential analysis. Here, we choose two of them and recommend users going to <strong>Chapter 10</strong> to see more detials.</p>
<div id="liner-discriminant-analysis-lda-effect-size-lefse-3" class="section level4 hasAnchor" number="14.4.7.1">
<h4><span class="header-section-number">14.4.7.1</span> Liner discriminant analysis (LDA) effect size (LEfSe)<a href="examples.html#liner-discriminant-analysis-lda-effect-size-lefse-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb582"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb582-1"><a href="examples.html#cb582-1" tabindex="-1"></a><span class="co"># metaphlan2_ps_lefse &lt;- run_lefse(</span></span>
<span id="cb582-2"><a href="examples.html#cb582-2" tabindex="-1"></a><span class="co">#                           ps = metaphlan2_ps_species_filter_trim,</span></span>
<span id="cb582-3"><a href="examples.html#cb582-3" tabindex="-1"></a><span class="co">#                           group = &quot;Group&quot;,</span></span>
<span id="cb582-4"><a href="examples.html#cb582-4" tabindex="-1"></a><span class="co">#                           group_names = c(&quot;AA&quot;, &quot;BB&quot;),</span></span>
<span id="cb582-5"><a href="examples.html#cb582-5" tabindex="-1"></a><span class="co">#                           norm = &quot;CPM&quot;,</span></span>
<span id="cb582-6"><a href="examples.html#cb582-6" tabindex="-1"></a><span class="co">#                           Lda = 2)</span></span>
<span id="cb582-7"><a href="examples.html#cb582-7" tabindex="-1"></a></span>
<span id="cb582-8"><a href="examples.html#cb582-8" tabindex="-1"></a>metaphlan2_ps_lefse <span class="ot">&lt;-</span> <span class="fu">run_lefse2</span>(</span>
<span id="cb582-9"><a href="examples.html#cb582-9" tabindex="-1"></a>                          <span class="at">ps =</span> metaphlan2_ps_species_filter_trim,</span>
<span id="cb582-10"><a href="examples.html#cb582-10" tabindex="-1"></a>                          <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb582-11"><a href="examples.html#cb582-11" tabindex="-1"></a>                          <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb582-12"><a href="examples.html#cb582-12" tabindex="-1"></a>                          <span class="at">norm =</span> <span class="st">&quot;CPM&quot;</span>,</span>
<span id="cb582-13"><a href="examples.html#cb582-13" tabindex="-1"></a>                          <span class="at">lda_cutoff =</span> <span class="dv">2</span>)</span>
<span id="cb582-14"><a href="examples.html#cb582-14" tabindex="-1"></a><span class="fu">head</span>(metaphlan2_ps_lefse)</span></code></pre></div>
<pre><code>##                            TaxaID         Block LDA_Score Enrichment EffectSize      Pvalue Log2FoldChange (Median)\nAA_vs_BB
## 1 s__Bacteroides_thetaiotaomicron 7_AA vs 15_BB -4.826720         AA   4.278888 0.005349957                          5.418919
## 2 s__Bifidobacterium_adolescentis 7_AA vs 15_BB  4.697569         BB   3.845065 0.012732015                                NA
## 3       s__Bifidobacterium_longum 7_AA vs 15_BB  4.717255         BB   1.968233 0.031365896                         -4.652818
## 4       s__Clostridium_bartlettii 7_AA vs 15_BB  3.177558         BB   1.533233 0.048850983                                NA
## 5      s__Collinsella_aerofaciens 7_AA vs 15_BB  4.175147         BB   3.121105 0.033120537                                NA
## 6            s__Dorea_longicatena 7_AA vs 15_BB  4.003730         BB   3.263339 0.018910418                                NA
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Mean)\nAA_vs_BB Mean Abundance\n(All)
## 1               2880.6116            46644.493            1090.2931                        3.196812             24133.533
## 2                  0.0000                0.000            3852.9342                              NA             36085.659
## 3              11059.7586             1204.497           30300.0596                       -3.715992             41261.284
## 4                287.8303                0.000             460.9082                       -3.713334              1296.564
## 5               7694.5538                0.000           13242.2170                       -3.239003             12659.440
## 6                828.5380                0.000            8760.0060                       -7.366158              7116.615
##   Mean Abundance\nAA Mean Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB       Odds Ratio (95% CI)
## 1        61480.34642           6705.020                    90.91                100.00                 86.67          0.064 (-5.3;5.5)
## 2            0.00000          52925.633                    40.91                  0.00                 60.00                      &lt;NA&gt;
## 3         4447.27254          58441.156                    86.36                 85.71                 86.67             270 (280;260)
## 4          139.99665           1836.295                    68.18                 42.86                 80.00             460 (470;440)
## 5         1873.94282          17692.672                    63.64                 42.86                 73.33                28 (35;22)
## 6           63.08757          10408.261                    63.64                 42.86                 73.33 3.8e+13 (3.8e+13;3.8e+13)</code></pre>
<ul>
<li>Visualization</li>
</ul>
<div class="sourceCode" id="cb584"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb584-1"><a href="examples.html#cb584-1" tabindex="-1"></a><span class="co"># # don&#39;t run this code when you do lefse in reality</span></span>
<span id="cb584-2"><a href="examples.html#cb584-2" tabindex="-1"></a><span class="co"># metaphlan2_ps_lefse$LDA_Score &lt;- metaphlan2_ps_lefse$LDA_Score * 100</span></span>
<span id="cb584-3"><a href="examples.html#cb584-3" tabindex="-1"></a></span>
<span id="cb584-4"><a href="examples.html#cb584-4" tabindex="-1"></a><span class="fu">plot_lefse</span>(</span>
<span id="cb584-5"><a href="examples.html#cb584-5" tabindex="-1"></a>  <span class="at">da_res =</span> metaphlan2_ps_lefse,</span>
<span id="cb584-6"><a href="examples.html#cb584-6" tabindex="-1"></a>  <span class="at">x_index =</span> <span class="st">&quot;LDA_Score&quot;</span>,</span>
<span id="cb584-7"><a href="examples.html#cb584-7" tabindex="-1"></a>  <span class="at">x_index_cutoff =</span> <span class="dv">2</span>,</span>
<span id="cb584-8"><a href="examples.html#cb584-8" tabindex="-1"></a>  <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;green&quot;</span>, <span class="st">&quot;red&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-309"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-309-1.png" alt="Lefse analysis (MGS example)" width="768" />
<p class="caption">
Figure 14.17: Lefse analysis (MGS example)
</p>
</div>
</div>
<div id="wilcoxon-rank-sum-test-1" class="section level4 hasAnchor" number="14.4.7.2">
<h4><span class="header-section-number">14.4.7.2</span> Wilcoxon Rank-Sum test<a href="examples.html#wilcoxon-rank-sum-test-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb585"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb585-1"><a href="examples.html#cb585-1" tabindex="-1"></a>metaphlan2_ps_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb585-2"><a href="examples.html#cb585-2" tabindex="-1"></a>                          <span class="at">ps =</span> metaphlan2_ps_species_filter_trim,</span>
<span id="cb585-3"><a href="examples.html#cb585-3" tabindex="-1"></a>                          <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb585-4"><a href="examples.html#cb585-4" tabindex="-1"></a>                          <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb585-5"><a href="examples.html#cb585-5" tabindex="-1"></a></span>
<span id="cb585-6"><a href="examples.html#cb585-6" tabindex="-1"></a><span class="fu">head</span>(metaphlan2_ps_wilcox)</span></code></pre></div>
<pre><code>##                           TaxaID         Block Enrichment EffectSize Statistic     Pvalue AdjustedPvalue Log2FoldChange (Median)\nAA_vs_BB
## 1  s__Acidaminococcus_fermentans 7_AA vs 15_BB  Nonsignif  0.1123778      49.0 0.75333110      0.8862719                                NA
## 2   s__Acidaminococcus_intestini 7_AA vs 15_BB  Nonsignif  0.7508683      51.0 0.91657416      0.9786929                                NA
## 3 s__Adlercreutzia_equolifaciens 7_AA vs 15_BB  Nonsignif  0.6300047      28.0 0.05769072      0.3938692                                NA
## 4     s__Akkermansia_muciniphila 7_AA vs 15_BB  Nonsignif  1.3630629      63.0 0.23762783      0.5638626                                NA
## 5        s__Alistipes_finegoldii 7_AA vs 15_BB  Nonsignif  0.7590404      73.5 0.11454477      0.4635865                                NA
## 6      s__Alistipes_indistinctus 7_AA vs 15_BB  Nonsignif  0.3706772      66.0 0.26778154      0.5982091                                NA
##   Median Abundance\n(All) Median Abundance\nAA Median Abundance\nBB Log2FoldChange (Rank)\nAA_vs_BB Mean Rank Abundance\nAA
## 1                       0            0.0000000             0.00e+00                     -0.09269949                   11.00
## 2                       0            0.0000000             0.00e+00                     -0.03907932                   11.29
## 3                       0            0.0000000             5.58e-05                     -0.71479501                    8.00
## 4                       0            0.0000000             0.00e+00                      0.26748031                   13.00
## 5                       0            0.0027362             0.00e+00                      0.52169761                   14.50
## 6                       0            0.0000000             0.00e+00                      0.34139504                   13.43
##   Mean Rank Abundance\nBB Occurrence (100%)\n(All) Occurrence (100%)\nAA Occurrence (100%)\nBB Odds Ratio (95% CI)
## 1                   11.73                    18.18                 14.29                 20.00      1.5 (2.4;0.69)
## 2                   11.60                    18.18                 14.29                 20.00     0.77 (0.26;1.3)
## 3                   13.13                    40.91                 14.29                 53.33       Inf (Inf;Inf)
## 4                   10.80                    13.64                 28.57                  6.67     0.83 (0.48;1.2)
## 5                   10.10                    45.45                 57.14                 40.00     0.0028 (-12;12)
## 6                   10.60                    31.82                 42.86                 26.67    0.067 (-5.2;5.4)</code></pre>
<ul>
<li>Volcano</li>
</ul>
<div class="sourceCode" id="cb587"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb587-1"><a href="examples.html#cb587-1" tabindex="-1"></a><span class="fu">plot_volcano</span>(</span>
<span id="cb587-2"><a href="examples.html#cb587-2" tabindex="-1"></a>    <span class="at">da_res =</span> metaphlan2_ps_wilcox,</span>
<span id="cb587-3"><a href="examples.html#cb587-3" tabindex="-1"></a>    <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb587-4"><a href="examples.html#cb587-4" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;Log2FoldChange (Rank)</span><span class="sc">\n</span><span class="st">AA_vs_BB&quot;</span>,</span>
<span id="cb587-5"><a href="examples.html#cb587-5" tabindex="-1"></a>    <span class="at">x_index_cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb587-6"><a href="examples.html#cb587-6" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb587-7"><a href="examples.html#cb587-7" tabindex="-1"></a>    <span class="at">y_index_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb587-8"><a href="examples.html#cb587-8" tabindex="-1"></a>    <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;grey&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb587-9"><a href="examples.html#cb587-9" tabindex="-1"></a>    <span class="at">topN =</span> <span class="dv">5</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-311"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-311-1.png" alt="Wilcoxon Rank-Sum test (MGS example)" width="768" />
<p class="caption">
Figure 14.18: Wilcoxon Rank-Sum test (MGS example)
</p>
</div>
</div>
</div>
</div>
<div id="systematic-information-10" class="section level2 hasAnchor" number="14.5">
<h2><span class="header-section-number">14.5</span> Systematic Information<a href="examples.html#systematic-information-10" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb588"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb588-1"><a href="examples.html#cb588-1" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 4.1.3 (2022-03-10)
##  os       macOS Monterey 12.2.1
##  system   x86_64, darwin17.0
##  ui       RStudio
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       Asia/Shanghai
##  date     2023-11-30
##  rstudio  2023.09.0+463 Desert Sunflower (desktop)
##  pandoc   3.1.1 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/ (via rmarkdown)
## 
## ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package              * version   date (UTC) lib source
##  abind                  1.4-5     2016-07-21 [2] CRAN (R 4.1.0)
##  ade4                   1.7-22    2023-02-06 [2] CRAN (R 4.1.2)
##  ALDEx2                 1.30.0    2022-11-01 [2] Bioconductor
##  annotate               1.72.0    2021-10-26 [2] Bioconductor
##  AnnotationDbi          1.60.2    2023-03-10 [2] Bioconductor
##  ape                  * 5.7-1     2023-03-13 [2] CRAN (R 4.1.2)
##  askpass                1.1       2019-01-13 [2] CRAN (R 4.1.0)
##  backports              1.4.1     2021-12-13 [2] CRAN (R 4.1.0)
##  base64enc              0.1-3     2015-07-28 [2] CRAN (R 4.1.0)
##  bayesm                 3.1-5     2022-12-02 [2] CRAN (R 4.1.2)
##  Biobase              * 2.54.0    2021-10-26 [2] Bioconductor
##  BiocGenerics         * 0.40.0    2021-10-26 [2] Bioconductor
##  BiocParallel           1.28.3    2021-12-09 [2] Bioconductor
##  biomformat             1.22.0    2021-10-26 [2] Bioconductor
##  Biostrings             2.62.0    2021-10-26 [2] Bioconductor
##  bit                    4.0.5     2022-11-15 [2] CRAN (R 4.1.2)
##  bit64                  4.0.5     2020-08-30 [2] CRAN (R 4.1.0)
##  bitops                 1.0-7     2021-04-24 [2] CRAN (R 4.1.0)
##  blob                   1.2.4     2023-03-17 [2] CRAN (R 4.1.2)
##  bookdown               0.34      2023-05-09 [2] CRAN (R 4.1.2)
##  broom                  1.0.5     2023-06-09 [2] CRAN (R 4.1.3)
##  bslib                  0.6.0     2023-11-21 [1] CRAN (R 4.1.3)
##  cachem                 1.0.8     2023-05-01 [2] CRAN (R 4.1.2)
##  callr                  3.7.3     2022-11-02 [2] CRAN (R 4.1.2)
##  car                    3.1-2     2023-03-30 [2] CRAN (R 4.1.2)
##  carData                3.0-5     2022-01-06 [2] CRAN (R 4.1.2)
##  caTools                1.18.2    2021-03-28 [2] CRAN (R 4.1.0)
##  checkmate              2.2.0     2023-04-27 [2] CRAN (R 4.1.2)
##  class                  7.3-22    2023-05-03 [2] CRAN (R 4.1.2)
##  classInt               0.4-9     2023-02-28 [2] CRAN (R 4.1.2)
##  cli                    3.6.1     2023-03-23 [2] CRAN (R 4.1.2)
##  cluster                2.1.4     2022-08-22 [2] CRAN (R 4.1.2)
##  coda                   0.19-4    2020-09-30 [2] CRAN (R 4.1.0)
##  codetools              0.2-19    2023-02-01 [2] CRAN (R 4.1.2)
##  coin                   1.4-2     2021-10-08 [2] CRAN (R 4.1.0)
##  colorspace             2.1-0     2023-01-23 [2] CRAN (R 4.1.2)
##  compositions           2.0-6     2023-04-13 [2] CRAN (R 4.1.2)
##  conflicted           * 1.2.0     2023-02-01 [2] CRAN (R 4.1.2)
##  corrplot               0.92      2021-11-18 [2] CRAN (R 4.1.0)
##  cowplot                1.1.1     2020-12-30 [2] CRAN (R 4.1.0)
##  crayon                 1.5.2     2022-09-29 [2] CRAN (R 4.1.2)
##  crosstalk              1.2.0     2021-11-04 [2] CRAN (R 4.1.0)
##  data.table             1.14.8    2023-02-17 [2] CRAN (R 4.1.2)
##  DBI                    1.1.3     2022-06-18 [2] CRAN (R 4.1.2)
##  DelayedArray           0.20.0    2021-10-26 [2] Bioconductor
##  DEoptimR               1.0-14    2023-06-09 [2] CRAN (R 4.1.3)
##  DESeq2                 1.34.0    2021-10-26 [2] Bioconductor
##  devtools               2.4.5     2022-10-11 [2] CRAN (R 4.1.2)
##  digest                 0.6.33    2023-07-07 [1] CRAN (R 4.1.3)
##  dplyr                * 1.1.2     2023-04-20 [2] CRAN (R 4.1.2)
##  DT                     0.28      2023-05-18 [2] CRAN (R 4.1.3)
##  e1071                  1.7-13    2023-02-01 [2] CRAN (R 4.1.2)
##  edgeR                  3.36.0    2021-10-26 [2] Bioconductor
##  ellipsis               0.3.2     2021-04-29 [2] CRAN (R 4.1.0)
##  emmeans                1.8.7     2023-06-23 [1] CRAN (R 4.1.3)
##  estimability           1.4.1     2022-08-05 [2] CRAN (R 4.1.2)
##  evaluate               0.21      2023-05-05 [2] CRAN (R 4.1.2)
##  FactoMineR             2.8       2023-03-27 [2] CRAN (R 4.1.2)
##  fansi                  1.0.4     2023-01-22 [2] CRAN (R 4.1.2)
##  farver                 2.1.1     2022-07-06 [2] CRAN (R 4.1.2)
##  fastmap                1.1.1     2023-02-24 [2] CRAN (R 4.1.2)
##  flashClust             1.01-2    2012-08-21 [2] CRAN (R 4.1.0)
##  foreach                1.5.2     2022-02-02 [2] CRAN (R 4.1.2)
##  foreign                0.8-84    2022-12-06 [2] CRAN (R 4.1.2)
##  forestplot             3.1.1     2022-12-06 [2] CRAN (R 4.1.2)
##  Formula                1.2-5     2023-02-24 [2] CRAN (R 4.1.2)
##  fs                     1.6.2     2023-04-25 [2] CRAN (R 4.1.2)
##  genefilter             1.76.0    2021-10-26 [2] Bioconductor
##  geneplotter            1.72.0    2021-10-26 [2] Bioconductor
##  generics               0.1.3     2022-07-05 [2] CRAN (R 4.1.2)
##  GenomeInfoDb         * 1.30.1    2022-01-30 [2] Bioconductor
##  GenomeInfoDbData       1.2.7     2022-03-09 [2] Bioconductor
##  GenomicRanges        * 1.46.1    2021-11-18 [2] Bioconductor
##  ggiraph                0.8.7     2023-03-17 [2] CRAN (R 4.1.2)
##  ggiraphExtra           0.3.0     2020-10-06 [2] CRAN (R 4.1.2)
##  ggplot2              * 3.4.2     2023-04-03 [2] CRAN (R 4.1.2)
##  ggpubr               * 0.6.0     2023-02-10 [2] CRAN (R 4.1.2)
##  ggrepel                0.9.3     2023-02-03 [2] CRAN (R 4.1.2)
##  ggsci                  3.0.0     2023-03-08 [2] CRAN (R 4.1.2)
##  ggsignif               0.6.4     2022-10-13 [2] CRAN (R 4.1.2)
##  ggVennDiagram          1.2.2     2022-09-08 [2] CRAN (R 4.1.2)
##  glmnet                 4.1-7     2023-03-23 [2] CRAN (R 4.1.2)
##  glue                 * 1.6.2     2022-02-24 [2] CRAN (R 4.1.2)
##  Gmisc                * 3.0.2     2023-03-13 [2] CRAN (R 4.1.2)
##  gplots                 3.1.3     2022-04-25 [2] CRAN (R 4.1.2)
##  gridExtra              2.3       2017-09-09 [2] CRAN (R 4.1.0)
##  gtable                 0.3.3     2023-03-21 [2] CRAN (R 4.1.2)
##  gtools                 3.9.4     2022-11-27 [2] CRAN (R 4.1.2)
##  highr                  0.10      2022-12-22 [2] CRAN (R 4.1.2)
##  Hmisc                  5.1-0     2023-05-08 [2] CRAN (R 4.1.2)
##  htmlTable            * 2.4.1     2022-07-07 [2] CRAN (R 4.1.2)
##  htmltools              0.5.7     2023-11-03 [1] CRAN (R 4.1.3)
##  htmlwidgets            1.6.2     2023-03-17 [2] CRAN (R 4.1.2)
##  httpuv                 1.6.11    2023-05-11 [2] CRAN (R 4.1.3)
##  httr                   1.4.6     2023-05-08 [2] CRAN (R 4.1.2)
##  igraph                 1.5.0     2023-06-16 [1] CRAN (R 4.1.3)
##  insight                0.19.3    2023-06-29 [2] CRAN (R 4.1.3)
##  IRanges              * 2.28.0    2021-10-26 [2] Bioconductor
##  iterators              1.0.14    2022-02-05 [2] CRAN (R 4.1.2)
##  jquerylib              0.1.4     2021-04-26 [2] CRAN (R 4.1.0)
##  jsonlite               1.8.7     2023-06-29 [2] CRAN (R 4.1.3)
##  kableExtra             1.3.4     2021-02-20 [2] CRAN (R 4.1.2)
##  KEGGREST               1.34.0    2021-10-26 [2] Bioconductor
##  KernSmooth             2.23-22   2023-07-10 [2] CRAN (R 4.1.3)
##  knitr                  1.43      2023-05-25 [2] CRAN (R 4.1.3)
##  labeling               0.4.2     2020-10-20 [2] CRAN (R 4.1.0)
##  later                  1.3.1     2023-05-02 [2] CRAN (R 4.1.2)
##  lattice              * 0.21-8    2023-04-05 [2] CRAN (R 4.1.2)
##  leaps                  3.1       2020-01-16 [2] CRAN (R 4.1.0)
##  libcoin                1.0-9     2021-09-27 [2] CRAN (R 4.1.0)
##  lifecycle              1.0.3     2022-10-07 [2] CRAN (R 4.1.2)
##  limma                  3.50.3    2022-04-07 [2] Bioconductor
##  locfit                 1.5-9.8   2023-06-11 [2] CRAN (R 4.1.3)
##  LOCOM                  1.1       2022-08-05 [2] Github (yijuanhu/LOCOM@c181e0f)
##  lubridate              1.9.2     2023-02-10 [2] CRAN (R 4.1.2)
##  magrittr             * 2.0.3     2022-03-30 [2] CRAN (R 4.1.2)
##  MASS                   7.3-60    2023-05-04 [2] CRAN (R 4.1.2)
##  Matrix                 1.6-0     2023-07-08 [2] CRAN (R 4.1.3)
##  MatrixGenerics       * 1.6.0     2021-10-26 [2] Bioconductor
##  matrixStats          * 1.0.0     2023-06-02 [2] CRAN (R 4.1.3)
##  mbzinb                 0.2       2022-03-16 [2] local
##  memoise                2.0.1     2021-11-26 [2] CRAN (R 4.1.0)
##  metagenomeSeq          1.36.0    2021-10-26 [2] Bioconductor
##  mgcv                   1.8-42    2023-03-02 [2] CRAN (R 4.1.2)
##  microbiome             1.16.0    2021-10-26 [2] Bioconductor
##  mime                   0.12      2021-09-28 [2] CRAN (R 4.1.0)
##  miniUI                 0.1.1.1   2018-05-18 [2] CRAN (R 4.1.0)
##  modeltools             0.2-23    2020-03-05 [2] CRAN (R 4.1.0)
##  multcomp               1.4-25    2023-06-20 [2] CRAN (R 4.1.3)
##  multcompView           0.1-9     2023-04-09 [2] CRAN (R 4.1.2)
##  multtest               2.50.0    2021-10-26 [2] Bioconductor
##  munsell                0.5.0     2018-06-12 [2] CRAN (R 4.1.0)
##  mvtnorm                1.2-2     2023-06-08 [2] CRAN (R 4.1.3)
##  mycor                  0.1.1     2018-04-10 [2] CRAN (R 4.1.0)
##  NADA                   1.6-1.1   2020-03-22 [2] CRAN (R 4.1.0)
##  nlme                 * 3.1-162   2023-01-31 [2] CRAN (R 4.1.2)
##  nnet                   7.3-19    2023-05-03 [2] CRAN (R 4.1.2)
##  openssl                2.0.6     2023-03-09 [2] CRAN (R 4.1.2)
##  permute              * 0.9-7     2022-01-27 [2] CRAN (R 4.1.2)
##  pheatmap             * 1.0.12    2019-01-04 [2] CRAN (R 4.1.0)
##  phyloseq             * 1.38.0    2021-10-26 [2] Bioconductor
##  picante              * 1.8.2     2020-06-10 [2] CRAN (R 4.1.0)
##  pillar                 1.9.0     2023-03-22 [2] CRAN (R 4.1.2)
##  pkgbuild               1.4.2     2023-06-26 [2] CRAN (R 4.1.3)
##  pkgconfig              2.0.3     2019-09-22 [2] CRAN (R 4.1.0)
##  pkgload                1.3.2.1   2023-07-08 [2] CRAN (R 4.1.3)
##  plyr                   1.8.8     2022-11-11 [2] CRAN (R 4.1.2)
##  png                    0.1-8     2022-11-29 [2] CRAN (R 4.1.2)
##  ppcor                  1.1       2015-12-03 [2] CRAN (R 4.1.0)
##  prettyunits            1.1.1     2020-01-24 [2] CRAN (R 4.1.0)
##  processx               3.8.2     2023-06-30 [2] CRAN (R 4.1.3)
##  profvis                0.3.8     2023-05-02 [2] CRAN (R 4.1.2)
##  promises               1.2.0.1   2021-02-11 [2] CRAN (R 4.1.0)
##  protoclust             1.6.4     2022-04-01 [2] CRAN (R 4.1.2)
##  proxy                  0.4-27    2022-06-09 [2] CRAN (R 4.1.2)
##  ps                     1.7.5     2023-04-18 [2] CRAN (R 4.1.2)
##  pscl                   1.5.5.1   2023-05-10 [2] CRAN (R 4.1.2)
##  purrr                  1.0.1     2023-01-10 [2] CRAN (R 4.1.2)
##  qvalue                 2.26.0    2021-10-26 [2] Bioconductor
##  R6                     2.5.1     2021-08-19 [2] CRAN (R 4.1.0)
##  RAIDA                  1.0       2022-03-14 [2] local
##  RColorBrewer         * 1.1-3     2022-04-03 [2] CRAN (R 4.1.2)
##  Rcpp                 * 1.0.11    2023-07-06 [1] CRAN (R 4.1.3)
##  RcppZiggurat           0.1.6     2020-10-20 [2] CRAN (R 4.1.0)
##  RCurl                  1.98-1.12 2023-03-27 [2] CRAN (R 4.1.2)
##  remotes                2.4.2     2021-11-30 [2] CRAN (R 4.1.0)
##  reshape2               1.4.4     2020-04-09 [2] CRAN (R 4.1.0)
##  reticulate             1.30      2023-06-09 [2] CRAN (R 4.1.3)
##  Rfast                  2.0.8     2023-07-03 [2] CRAN (R 4.1.3)
##  rhdf5                  2.38.1    2022-03-10 [2] Bioconductor
##  rhdf5filters           1.6.0     2021-10-26 [2] Bioconductor
##  Rhdf5lib               1.16.0    2021-10-26 [2] Bioconductor
##  rlang                  1.1.1     2023-04-28 [1] CRAN (R 4.1.2)
##  rmarkdown              2.23      2023-07-01 [2] CRAN (R 4.1.3)
##  robustbase             0.99-0    2023-06-16 [2] CRAN (R 4.1.3)
##  rpart                  4.1.19    2022-10-21 [2] CRAN (R 4.1.2)
##  RSpectra               0.16-1    2022-04-24 [2] CRAN (R 4.1.2)
##  RSQLite                2.3.1     2023-04-03 [2] CRAN (R 4.1.2)
##  rstatix                0.7.2     2023-02-01 [2] CRAN (R 4.1.2)
##  rstudioapi             0.15.0    2023-07-07 [2] CRAN (R 4.1.3)
##  Rtsne                  0.16      2022-04-17 [2] CRAN (R 4.1.2)
##  RVenn                  1.1.0     2019-07-18 [2] CRAN (R 4.1.0)
##  rvest                  1.0.3     2022-08-19 [2] CRAN (R 4.1.2)
##  S4Vectors            * 0.32.4    2022-03-29 [2] Bioconductor
##  sandwich               3.0-2     2022-06-15 [2] CRAN (R 4.1.2)
##  sass                   0.4.6     2023-05-03 [2] CRAN (R 4.1.2)
##  scales                 1.2.1     2022-08-20 [2] CRAN (R 4.1.2)
##  scatterplot3d          0.3-44    2023-05-05 [2] CRAN (R 4.1.2)
##  sessioninfo            1.2.2     2021-12-06 [2] CRAN (R 4.1.0)
##  sf                     1.0-7     2022-03-07 [2] CRAN (R 4.1.2)
##  shape                  1.4.6     2021-05-19 [2] CRAN (R 4.1.0)
##  shiny                  1.7.4.1   2023-07-06 [2] CRAN (R 4.1.3)
##  sjlabelled             1.2.0     2022-04-10 [2] CRAN (R 4.1.2)
##  sjmisc                 2.8.9     2021-12-03 [2] CRAN (R 4.1.0)
##  stringi                1.7.12    2023-01-11 [2] CRAN (R 4.1.2)
##  stringr                1.5.0     2022-12-02 [2] CRAN (R 4.1.2)
##  SummarizedExperiment * 1.24.0    2021-10-26 [2] Bioconductor
##  survival               3.5-5     2023-03-12 [2] CRAN (R 4.1.2)
##  svglite                2.1.1     2023-01-10 [2] CRAN (R 4.1.2)
##  systemfonts            1.0.4     2022-02-11 [2] CRAN (R 4.1.2)
##  tensorA                0.36.2    2020-11-19 [2] CRAN (R 4.1.0)
##  TH.data                1.1-2     2023-04-17 [2] CRAN (R 4.1.2)
##  tibble               * 3.2.1     2023-03-20 [2] CRAN (R 4.1.2)
##  tidyr                  1.3.0     2023-01-24 [2] CRAN (R 4.1.2)
##  tidyselect             1.2.0     2022-10-10 [2] CRAN (R 4.1.2)
##  timechange             0.2.0     2023-01-11 [2] CRAN (R 4.1.2)
##  truncnorm              1.0-9     2023-03-20 [2] CRAN (R 4.1.2)
##  umap                   0.2.10.0  2023-02-01 [2] CRAN (R 4.1.2)
##  units                  0.8-2     2023-04-27 [2] CRAN (R 4.1.2)
##  urlchecker             1.0.1     2021-11-30 [2] CRAN (R 4.1.0)
##  usethis                2.2.2     2023-07-06 [2] CRAN (R 4.1.3)
##  utf8                   1.2.3     2023-01-31 [2] CRAN (R 4.1.2)
##  uuid                   1.1-0     2022-04-19 [2] CRAN (R 4.1.2)
##  vctrs                  0.6.3     2023-06-14 [1] CRAN (R 4.1.3)
##  vegan                * 2.6-4     2022-10-11 [2] CRAN (R 4.1.2)
##  viridis              * 0.6.3     2023-05-03 [2] CRAN (R 4.1.2)
##  viridisLite          * 0.4.2     2023-05-02 [2] CRAN (R 4.1.2)
##  webshot                0.5.5     2023-06-26 [2] CRAN (R 4.1.3)
##  withr                  2.5.0     2022-03-03 [2] CRAN (R 4.1.2)
##  Wrench                 1.12.0    2021-10-26 [2] Bioconductor
##  xfun                   0.40      2023-08-09 [1] CRAN (R 4.1.3)
##  XMAS2                * 2.2.0     2023-11-30 [1] local
##  XML                    3.99-0.14 2023-03-19 [2] CRAN (R 4.1.2)
##  xml2                   1.3.5     2023-07-06 [2] CRAN (R 4.1.3)
##  xtable                 1.8-4     2019-04-21 [2] CRAN (R 4.1.0)
##  XVector                0.34.0    2021-10-26 [2] Bioconductor
##  yaml                   2.3.7     2023-01-23 [2] CRAN (R 4.1.2)
##  zCompositions          1.4.0-1   2022-03-26 [2] CRAN (R 4.1.2)
##  zlibbioc               1.40.0    2021-10-26 [2] Bioconductor
##  zoo                    1.8-12    2023-04-13 [2] CRAN (R 4.1.2)
## 
##  [1] /Users/zouhua/Library/R/x86_64/4.1/library
##  [2] /Library/Frameworks/R.framework/Versions/4.1/Resources/library
## 
## ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-thingholm2019obese" class="csl-entry">
Thingholm, Louise B, Malte C Rühlemann, Manja Koch, Brie Fuqua, Guido Laucke, Ruwen Boehm, Corinna Bang, et al. 2019. <span>“Obese Individuals with and Without Type 2 Diabetes Show Different Gut Microbial Functional Capacity and Composition.”</span> <em>Cell Host &amp; Microbe</em> 26 (2): 252–64.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="principals-of-differential-analysis-methods.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/XMAS_tutorial/blob/main/19-Examples.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["XMAS 2.0 Tutorial.pdf", "XMAS 2.0 Tutorial.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true,
"toc_depth": 3
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
