<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Example | Data Analysis in Metabolomics</title>
  <meta name="description" content="A template for Data Analysis in Metabolomics" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Example | Data Analysis in Metabolomics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A template for Data Analysis in Metabolomics" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Example | Data Analysis in Metabolomics" />
  
  <meta name="twitter:description" content="A template for Data Analysis in Metabolomics" />
  

<meta name="author" content="Hua Zou and Bangzhuo Tong" />


<meta name="date" content="2023-11-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="metorigin-analysis.html"/>
<link rel="next" href="references.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="assets/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Data Analysis in Metabolomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#what-is-metabolomics"><i class="fa fa-check"></i><b>1.1</b> What is metabolomics?</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#workflow-of-metabolomics-data-analysis"><i class="fa fa-check"></i><b>1.2</b> Workflow of metabolomics data analysis</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#statistical-analysis"><i class="fa fa-check"></i><b>1.2.1</b> Statistical Analysis</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#functional-analysis"><i class="fa fa-check"></i><b>1.2.2</b> Functional Analysis</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#software"><i class="fa fa-check"></i><b>1.3</b> Software</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#reference"><i class="fa fa-check"></i><b>1.4</b> Reference</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-processing.html"><a href="data-processing.html"><i class="fa fa-check"></i><b>2</b> Data Processing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-processing.html"><a href="data-processing.html#loading-packages"><i class="fa fa-check"></i><b>2.1</b> Loading packages</a></li>
<li class="chapter" data-level="2.2" data-path="data-processing.html"><a href="data-processing.html#importing-data"><i class="fa fa-check"></i><b>2.2</b> Importing data</a></li>
<li class="chapter" data-level="2.3" data-path="data-processing.html"><a href="data-processing.html#data-checking"><i class="fa fa-check"></i><b>2.3</b> Data Checking</a></li>
<li class="chapter" data-level="2.4" data-path="data-processing.html"><a href="data-processing.html#missing-value-imputation"><i class="fa fa-check"></i><b>2.4</b> Missing value imputation</a></li>
<li class="chapter" data-level="2.5" data-path="data-processing.html"><a href="data-processing.html#data-filtering"><i class="fa fa-check"></i><b>2.5</b> Data Filtering</a></li>
<li class="chapter" data-level="2.6" data-path="data-processing.html"><a href="data-processing.html#data-normalization"><i class="fa fa-check"></i><b>2.6</b> Data Normalization</a>
<ul>
<li class="chapter" data-level="2.6.1" data-path="data-processing.html"><a href="data-processing.html#normalization-by-normalizedata-function"><i class="fa fa-check"></i><b>2.6.1</b> Normalization by <strong>NormalizeData</strong> function</a></li>
<li class="chapter" data-level="2.6.2" data-path="data-processing.html"><a href="data-processing.html#normalization-by-poma-r-package"><i class="fa fa-check"></i><b>2.6.2</b> Normalization by <strong>POMA</strong> R package</a></li>
<li class="chapter" data-level="2.6.3" data-path="data-processing.html"><a href="data-processing.html#comparison-of-unnormalized-and-normalized-dataset"><i class="fa fa-check"></i><b>2.6.3</b> Comparison of unnormalized and normalized dataset</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="data-processing.html"><a href="data-processing.html#removing-outliers"><i class="fa fa-check"></i><b>2.7</b> Removing outliers</a></li>
<li class="chapter" data-level="2.8" data-path="data-processing.html"><a href="data-processing.html#data-distribution"><i class="fa fa-check"></i><b>2.8</b> Data distribution</a></li>
<li class="chapter" data-level="2.9" data-path="data-processing.html"><a href="data-processing.html#saving-datasets-into-rds-files"><i class="fa fa-check"></i><b>2.9</b> Saving datasets into RDS files</a></li>
<li class="chapter" data-level="2.10" data-path="data-processing.html"><a href="data-processing.html#systematic-information"><i class="fa fa-check"></i><b>2.10</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html"><i class="fa fa-check"></i><b>3</b> Statistical Analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-preprocessing"><i class="fa fa-check"></i><b>3.1</b> Data Preprocessing</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#environment-setup"><i class="fa fa-check"></i><b>3.1.1</b> Environment setup</a></li>
<li class="chapter" data-level="3.1.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-data"><i class="fa fa-check"></i><b>3.1.2</b> Loading data</a></li>
<li class="chapter" data-level="3.1.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#object-preparation"><i class="fa fa-check"></i><b>3.1.3</b> Object Preparation</a></li>
<li class="chapter" data-level="3.1.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-checking-1"><i class="fa fa-check"></i><b>3.1.4</b> Data Checking</a></li>
<li class="chapter" data-level="3.1.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#missing-value-imputation-1"><i class="fa fa-check"></i><b>3.1.5</b> Missing value imputation</a></li>
<li class="chapter" data-level="3.1.6" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-filtering-1"><i class="fa fa-check"></i><b>3.1.6</b> Data Filtering</a></li>
<li class="chapter" data-level="3.1.7" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-normalization-1"><i class="fa fa-check"></i><b>3.1.7</b> Data Normalization</a></li>
<li class="chapter" data-level="3.1.8" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#removing-outliers-1"><i class="fa fa-check"></i><b>3.1.8</b> Removing outliers</a></li>
<li class="chapter" data-level="3.1.9" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#saving-datasets-into-rds-files-1"><i class="fa fa-check"></i><b>3.1.9</b> Saving datasets into RDS files</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#cluster-analysis"><i class="fa fa-check"></i><b>3.2</b> Cluster Analysis</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-packages-1"><i class="fa fa-check"></i><b>3.2.1</b> Loading packages</a></li>
<li class="chapter" data-level="3.2.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#importing-data-1"><i class="fa fa-check"></i><b>3.2.2</b> Importing data</a></li>
<li class="chapter" data-level="3.2.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#hierarchical-clustering"><i class="fa fa-check"></i><b>3.2.3</b> Hierarchical Clustering</a></li>
<li class="chapter" data-level="3.2.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#partitional-clustering"><i class="fa fa-check"></i><b>3.2.4</b> Partitional Clustering</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#chemometrics-analysis"><i class="fa fa-check"></i><b>3.3</b> Chemometrics Analysis</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-packages-2"><i class="fa fa-check"></i><b>3.3.1</b> Loading packages</a></li>
<li class="chapter" data-level="3.3.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#importing-data-2"><i class="fa fa-check"></i><b>3.3.2</b> Importing data</a></li>
<li class="chapter" data-level="3.3.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>3.3.3</b> Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="3.3.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#partial-least-squares-discriminant-analysis-pls-da"><i class="fa fa-check"></i><b>3.3.4</b> Partial Least Squares-Discriminant Analysis (PLS-DA)</a></li>
<li class="chapter" data-level="3.3.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#sparse-partial-least-squares-discriminant-analysis-spls-da"><i class="fa fa-check"></i><b>3.3.5</b> Sparse Partial Least Squares-Discriminant Analysis (sPLS-DA)</a></li>
<li class="chapter" data-level="3.3.6" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#orthogonal-partial-least-squares-discriminant-analysis-orthopls-da"><i class="fa fa-check"></i><b>3.3.6</b> Orthogonal Partial Least Squares-Discriminant Analysis (orthoPLS-DA)</a></li>
<li class="chapter" data-level="3.3.7" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#chemometrics-analysis-by-own-scripts"><i class="fa fa-check"></i><b>3.3.7</b> Chemometrics Analysis by own scripts</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#univariate-analysis"><i class="fa fa-check"></i><b>3.4</b> Univariate Analysis</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-packages-3"><i class="fa fa-check"></i><b>3.4.1</b> Loading packages</a></li>
<li class="chapter" data-level="3.4.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#importing-data-3"><i class="fa fa-check"></i><b>3.4.2</b> Importing data</a></li>
<li class="chapter" data-level="3.4.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#fold-change-analysis"><i class="fa fa-check"></i><b>3.4.3</b> Fold Change Analysis</a></li>
<li class="chapter" data-level="3.4.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#t-test"><i class="fa fa-check"></i><b>3.4.4</b> T Test</a></li>
<li class="chapter" data-level="3.4.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#wilcoxon-test"><i class="fa fa-check"></i><b>3.4.5</b> Wilcoxon Test</a></li>
<li class="chapter" data-level="3.4.6" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#limma-test"><i class="fa fa-check"></i><b>3.4.6</b> Limma Test</a></li>
<li class="chapter" data-level="3.4.7" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#volcano-plot"><i class="fa fa-check"></i><b>3.4.7</b> Volcano plot</a></li>
<li class="chapter" data-level="3.4.8" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#vip-variable-influence-on-projection-coefficient"><i class="fa fa-check"></i><b>3.4.8</b> VIP (Variable influence on projection &amp; coefficient)</a></li>
<li class="chapter" data-level="3.4.9" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#t-test-by-local-codes"><i class="fa fa-check"></i><b>3.4.9</b> T-test by local codes</a></li>
<li class="chapter" data-level="3.4.10" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#merging-result"><i class="fa fa-check"></i><b>3.4.10</b> Merging result</a></li>
<li class="chapter" data-level="3.4.11" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#volcano-of-merged-results"><i class="fa fa-check"></i><b>3.4.11</b> Volcano of Merged Results</a></li>
<li class="chapter" data-level="3.4.12" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#correlation-heatmaps"><i class="fa fa-check"></i><b>3.4.12</b> Correlation Heatmaps</a></li>
<li class="chapter" data-level="3.4.13" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#glasso-this-function-will-compute-a-gaussian-graphical-model-using-the-glmnet-package"><i class="fa fa-check"></i><b>3.4.13</b> glasso: this function will compute a Gaussian Graphical Model using the glmnet package</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#multivariate-analysis"><i class="fa fa-check"></i><b>3.5</b> Multivariate analysis</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-packages-4"><i class="fa fa-check"></i><b>3.5.1</b> Loading packages</a></li>
<li class="chapter" data-level="3.5.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#importing-data-4"><i class="fa fa-check"></i><b>3.5.2</b> Importing data</a></li>
<li class="chapter" data-level="3.5.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-curation"><i class="fa fa-check"></i><b>3.5.3</b> Data curation</a></li>
<li class="chapter" data-level="3.5.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#regularized-generalized-linear-models"><i class="fa fa-check"></i><b>3.5.4</b> Regularized Generalized Linear Models</a></li>
<li class="chapter" data-level="3.5.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#classification"><i class="fa fa-check"></i><b>3.5.5</b> Classification</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#network-analysis"><i class="fa fa-check"></i><b>3.6</b> Network Analysis</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#introduction-1"><i class="fa fa-check"></i><b>3.6.1</b> Introduction</a></li>
<li class="chapter" data-level="3.6.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-packages-5"><i class="fa fa-check"></i><b>3.6.2</b> Loading packages</a></li>
<li class="chapter" data-level="3.6.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#importing-data-5"><i class="fa fa-check"></i><b>3.6.3</b> Importing data</a></li>
<li class="chapter" data-level="3.6.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-curation-1"><i class="fa fa-check"></i><b>3.6.4</b> Data curation</a></li>
<li class="chapter" data-level="3.6.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#associations-among-features"><i class="fa fa-check"></i><b>3.6.5</b> Associations Among Features</a></li>
<li class="chapter" data-level="3.6.6" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#network-comparison"><i class="fa fa-check"></i><b>3.6.6</b> Network comparison</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#systematic-information-1"><i class="fa fa-check"></i><b>3.7</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html"><i class="fa fa-check"></i><b>4</b> Functional Analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#enrichment-analysis"><i class="fa fa-check"></i><b>4.1</b> Enrichment Analysis</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#enviroment-setup"><i class="fa fa-check"></i><b>4.1.1</b> Enviroment Setup</a></li>
<li class="chapter" data-level="4.1.2" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#over-representation-analysis-ora"><i class="fa fa-check"></i><b>4.1.2</b> Over representation analysis (ORA)</a></li>
<li class="chapter" data-level="4.1.3" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#clean-environment-1"><i class="fa fa-check"></i><b>4.1.3</b> Clean environment 1</a></li>
<li class="chapter" data-level="4.1.4" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#single-sample-profiling"><i class="fa fa-check"></i><b>4.1.4</b> Single Sample Profiling</a></li>
<li class="chapter" data-level="4.1.5" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#clean-environment-2"><i class="fa fa-check"></i><b>4.1.5</b> Clean environment 2</a></li>
<li class="chapter" data-level="4.1.6" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#quantitative-enrichment-analysis-qea"><i class="fa fa-check"></i><b>4.1.6</b> Quantitative Enrichment Analysis (QEA)</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#pathway-analysis"><i class="fa fa-check"></i><b>4.2</b> Pathway Analysis</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#enviroment-set-up"><i class="fa fa-check"></i><b>4.2.1</b> Enviroment Set up</a></li>
<li class="chapter" data-level="4.2.2" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#over-representation-analysis"><i class="fa fa-check"></i><b>4.2.2</b> Over representation analysis</a></li>
<li class="chapter" data-level="4.2.3" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#clean-environment-3"><i class="fa fa-check"></i><b>4.2.3</b> Clean environment 3</a></li>
<li class="chapter" data-level="4.2.4" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#concentration-table-qea"><i class="fa fa-check"></i><b>4.2.4</b> Concentration Table (QEA)</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#functional-analysis-by-r-package"><i class="fa fa-check"></i><b>4.3</b> Functional Analysis by R package</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#loading-r-packages"><i class="fa fa-check"></i><b>4.3.1</b> Loading R packages</a></li>
<li class="chapter" data-level="4.3.2" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#importing-data-6"><i class="fa fa-check"></i><b>4.3.2</b> Importing Data</a></li>
<li class="chapter" data-level="4.3.3" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#differential-analysis"><i class="fa fa-check"></i><b>4.3.3</b> Differential Analysis</a></li>
<li class="chapter" data-level="4.3.4" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#metabolites-id-for-merged-results"><i class="fa fa-check"></i><b>4.3.4</b> Metabolites ID for Merged Results</a></li>
<li class="chapter" data-level="4.3.5" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#kegg-pathway-compound"><i class="fa fa-check"></i><b>4.3.5</b> KEGG PATHWAY COMPOUND</a></li>
<li class="chapter" data-level="4.3.6" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#functions-related-to-analysis"><i class="fa fa-check"></i><b>4.3.6</b> Functions related to Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#session-info"><i class="fa fa-check"></i><b>4.4</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html"><i class="fa fa-check"></i><b>5</b> MetOrigin Analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#loading-packages-6"><i class="fa fa-check"></i><b>5.1</b> Loading packages</a></li>
<li class="chapter" data-level="5.2" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#importing-data-7"><i class="fa fa-check"></i><b>5.2</b> Importing data</a></li>
<li class="chapter" data-level="5.3" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#differential-analysis-1"><i class="fa fa-check"></i><b>5.3</b> Differential Analysis</a></li>
<li class="chapter" data-level="5.4" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#de-metabolites"><i class="fa fa-check"></i><b>5.4</b> DE metabolites</a></li>
<li class="chapter" data-level="5.5" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#obtain-inputs-for-metorigin"><i class="fa fa-check"></i><b>5.5</b> Obtain inputs for Metorigin</a></li>
<li class="chapter" data-level="5.6" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#save-result"><i class="fa fa-check"></i><b>5.6</b> Save result</a></li>
<li class="chapter" data-level="5.7" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#session-info-1"><i class="fa fa-check"></i><b>5.7</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="example.html"><a href="example.html"><i class="fa fa-check"></i><b>6</b> Example</a>
<ul>
<li class="chapter" data-level="6.1" data-path="example.html"><a href="example.html#loading-packages-7"><i class="fa fa-check"></i><b>6.1</b> Loading packages</a></li>
<li class="chapter" data-level="6.2" data-path="example.html"><a href="example.html#importing-data-8"><i class="fa fa-check"></i><b>6.2</b> Importing data</a></li>
<li class="chapter" data-level="6.3" data-path="example.html"><a href="example.html#data-processing-1"><i class="fa fa-check"></i><b>6.3</b> Data Processing</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="example.html"><a href="example.html#data-checking-2"><i class="fa fa-check"></i><b>6.3.1</b> Data Checking</a></li>
<li class="chapter" data-level="6.3.2" data-path="example.html"><a href="example.html#missing-value-imputation-2"><i class="fa fa-check"></i><b>6.3.2</b> Missing value imputation</a></li>
<li class="chapter" data-level="6.3.3" data-path="example.html"><a href="example.html#data-filtering-2"><i class="fa fa-check"></i><b>6.3.3</b> Data Filtering</a></li>
<li class="chapter" data-level="6.3.4" data-path="example.html"><a href="example.html#data-normalization-2"><i class="fa fa-check"></i><b>6.3.4</b> Data Normalization</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="example.html"><a href="example.html#cluster-analysis-1"><i class="fa fa-check"></i><b>6.4</b> Cluster Analysis</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="example.html"><a href="example.html#hierarchical-clustering-1"><i class="fa fa-check"></i><b>6.4.1</b> Hierarchical Clustering</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="example.html"><a href="example.html#chemometrics-analysis-1"><i class="fa fa-check"></i><b>6.5</b> Chemometrics Analysis</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="example.html"><a href="example.html#partial-least-squares-discriminant-analysis-pls-da-1"><i class="fa fa-check"></i><b>6.5.1</b> Partial Least Squares-Discriminant Analysis (PLS-DA)</a></li>
<li class="chapter" data-level="6.5.2" data-path="example.html"><a href="example.html#sparse-partial-least-squares-discriminant-analysis-spls-da-1"><i class="fa fa-check"></i><b>6.5.2</b> Sparse Partial Least Squares-Discriminant Analysis (sPLS-DA)</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="example.html"><a href="example.html#univariate-analysis-1"><i class="fa fa-check"></i><b>6.6</b> Univariate Analysis</a>
<ul>
<li class="chapter" data-level="6.6.1" data-path="example.html"><a href="example.html#fold-change-analysis-1"><i class="fa fa-check"></i><b>6.6.1</b> Fold Change Analysis</a></li>
<li class="chapter" data-level="6.6.2" data-path="example.html"><a href="example.html#vip-variable-influence-on-projection-coefficient-1"><i class="fa fa-check"></i><b>6.6.2</b> VIP (Variable influence on projection &amp; coefficient)</a></li>
<li class="chapter" data-level="6.6.3" data-path="example.html"><a href="example.html#t-test-by-local-codes-1"><i class="fa fa-check"></i><b>6.6.3</b> T-test by local codes</a></li>
<li class="chapter" data-level="6.6.4" data-path="example.html"><a href="example.html#merging-result-1"><i class="fa fa-check"></i><b>6.6.4</b> Merging result</a></li>
<li class="chapter" data-level="6.6.5" data-path="example.html"><a href="example.html#volcano-of-merged-results-1"><i class="fa fa-check"></i><b>6.6.5</b> Volcano of Merged Results</a></li>
<li class="chapter" data-level="6.6.6" data-path="example.html"><a href="example.html#t-test-1"><i class="fa fa-check"></i><b>6.6.6</b> T Test</a></li>
<li class="chapter" data-level="6.6.7" data-path="example.html"><a href="example.html#volcano-plot-1"><i class="fa fa-check"></i><b>6.6.7</b> Volcano plot</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="example.html"><a href="example.html#feature-selection"><i class="fa fa-check"></i><b>6.7</b> Feature Selection</a>
<ul>
<li class="chapter" data-level="6.7.1" data-path="example.html"><a href="example.html#regularized-generalized-linear-models-lasso-alpha-1"><i class="fa fa-check"></i><b>6.7.1</b> Regularized Generalized Linear Models (Lasso: alpha = 1)</a></li>
<li class="chapter" data-level="6.7.2" data-path="example.html"><a href="example.html#classification-random-forest"><i class="fa fa-check"></i><b>6.7.2</b> Classification (Random Forest)</a></li>
</ul></li>
<li class="chapter" data-level="6.8" data-path="example.html"><a href="example.html#network-analysis-2"><i class="fa fa-check"></i><b>6.8</b> Network Analysis</a>
<ul>
<li class="chapter" data-level="6.8.1" data-path="example.html"><a href="example.html#data-curation-2"><i class="fa fa-check"></i><b>6.8.1</b> Data curation</a></li>
<li class="chapter" data-level="6.8.2" data-path="example.html"><a href="example.html#building-network-model-4"><i class="fa fa-check"></i><b>6.8.2</b> Building network model</a></li>
<li class="chapter" data-level="6.8.3" data-path="example.html"><a href="example.html#visualizing-the-network-4"><i class="fa fa-check"></i><b>6.8.3</b> Visualizing the network</a></li>
</ul></li>
<li class="chapter" data-level="6.9" data-path="example.html"><a href="example.html#network-analysis-by-wgcna"><i class="fa fa-check"></i><b>6.9</b> Network Analysis by WGCNA</a>
<ul>
<li class="chapter" data-level="6.9.1" data-path="example.html"><a href="example.html#data-curation-3"><i class="fa fa-check"></i><b>6.9.1</b> Data curation</a></li>
<li class="chapter" data-level="6.9.2" data-path="example.html"><a href="example.html#tuning-soft-thresholds"><i class="fa fa-check"></i><b>6.9.2</b> Tuning soft thresholds</a></li>
<li class="chapter" data-level="6.9.3" data-path="example.html"><a href="example.html#create-the-network-using-the-blockwisemodules"><i class="fa fa-check"></i><b>6.9.3</b> Create the network using the <code>blockwiseModules</code></a></li>
<li class="chapter" data-level="6.9.4" data-path="example.html"><a href="example.html#plot-modules"><i class="fa fa-check"></i><b>6.9.4</b> plot modules</a></li>
<li class="chapter" data-level="6.9.5" data-path="example.html"><a href="example.html#relationships-among-modules"><i class="fa fa-check"></i><b>6.9.5</b> Relationships among modules</a></li>
<li class="chapter" data-level="6.9.6" data-path="example.html"><a href="example.html#module-eigengene-correlation"><i class="fa fa-check"></i><b>6.9.6</b> Module (Eigengene) correlation</a></li>
<li class="chapter" data-level="6.9.7" data-path="example.html"><a href="example.html#relate-module-cluster-assignments-to-groups"><i class="fa fa-check"></i><b>6.9.7</b> Relate Module (cluster) Assignments to Groups</a></li>
<li class="chapter" data-level="6.9.8" data-path="example.html"><a href="example.html#generate-and-export-networks"><i class="fa fa-check"></i><b>6.9.8</b> Generate and Export Networks</a></li>
<li class="chapter" data-level="6.9.9" data-path="example.html"><a href="example.html#network-visualization"><i class="fa fa-check"></i><b>6.9.9</b> Network visualization</a></li>
</ul></li>
<li class="chapter" data-level="6.10" data-path="example.html"><a href="example.html#systematic-information-2"><i class="fa fa-check"></i><b>6.10</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data Analysis in Metabolomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="example" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> Example<a href="example.html#example" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>We perform all the data analysis on our own metabolomic data in this chapter. There are several datasets:</p>
<ul>
<li><p><strong>GvHD stool metabolites TM wide-target sequencing</strong>: /home/xuxiaomin/project/NanFangHospitalGvHD/00.data/metabolites_TM/MWY-20-522-01_2021-03-25_17-34-06/1.Data_assess/ALL_sample_data.xlsx</p></li>
<li><p><strong>GvHD stool metabolites SCFA</strong>: /home/xuxiaomin/project/NanFangHospitalGvHD/00.data/metabolites_scfa/SCFA.levels.xlsx</p></li>
<li><p><strong>PD-1 mice trial (round 2) serum metabolites TM wide-target sequencing</strong>: /home/xuxiaomin/project/pd1_mice/Round_2/00.data/MetaboliteSerum/MWY-20-049/1.Data_assess/all_group/ALL_sample_data.xlsx</p></li>
</ul>
<p>and the metadata:</p>
<ul>
<li><strong>GvHD metadata</strong>: /home/xuxiaomin/project/NanFangHospitalGvHD/00.data/metadata/metadata_v4.txt</li>
</ul>
<p>We transform their names as following:</p>
<ul>
<li><p>GvHD_stool_metabolites_TM.xlsx</p></li>
<li><p>GvHD_stool_metabolites_SCFA.xlsx</p></li>
<li><p>PD1_mice_serum_metabolites_TM.xlsx</p></li>
<li><p>GvHD_metadata.txt</p></li>
</ul>
<p>Here, we use the <code>GvHD_stool_metabolites_TM.xlsx</code> to practice our template.</p>
<div id="loading-packages-7" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Loading packages<a href="example.html#loading-packages-7" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb231"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb231-1"><a href="example.html#cb231-1" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">warning =</span> F)</span>
<span id="cb231-2"><a href="example.html#cb231-2" tabindex="-1"></a></span>
<span id="cb231-3"><a href="example.html#cb231-3" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb231-4"><a href="example.html#cb231-4" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb231-5"><a href="example.html#cb231-5" tabindex="-1"></a><span class="fu">library</span>(POMA)</span>
<span id="cb231-6"><a href="example.html#cb231-6" tabindex="-1"></a><span class="fu">library</span>(ropls)</span>
<span id="cb231-7"><a href="example.html#cb231-7" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb231-8"><a href="example.html#cb231-8" tabindex="-1"></a><span class="fu">library</span>(ggraph)</span>
<span id="cb231-9"><a href="example.html#cb231-9" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb231-10"><a href="example.html#cb231-10" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb231-11"><a href="example.html#cb231-11" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb231-12"><a href="example.html#cb231-12" tabindex="-1"></a></span>
<span id="cb231-13"><a href="example.html#cb231-13" tabindex="-1"></a><span class="fu">library</span>(cluster)    <span class="co"># clustering algorithms</span></span>
<span id="cb231-14"><a href="example.html#cb231-14" tabindex="-1"></a><span class="fu">library</span>(factoextra) <span class="co"># clustering visualization</span></span>
<span id="cb231-15"><a href="example.html#cb231-15" tabindex="-1"></a><span class="fu">library</span>(dendextend) <span class="co"># for comparing two dendrograms</span></span>
<span id="cb231-16"><a href="example.html#cb231-16" tabindex="-1"></a></span>
<span id="cb231-17"><a href="example.html#cb231-17" tabindex="-1"></a><span class="fu">library</span>(NetCoMi)</span>
<span id="cb231-18"><a href="example.html#cb231-18" tabindex="-1"></a><span class="fu">library</span>(SPRING)</span>
<span id="cb231-19"><a href="example.html#cb231-19" tabindex="-1"></a><span class="fu">library</span>(SpiecEasi)</span>
<span id="cb231-20"><a href="example.html#cb231-20" tabindex="-1"></a></span>
<span id="cb231-21"><a href="example.html#cb231-21" tabindex="-1"></a><span class="fu">library</span>(WGCNA)</span>
<span id="cb231-22"><a href="example.html#cb231-22" tabindex="-1"></a></span>
<span id="cb231-23"><a href="example.html#cb231-23" tabindex="-1"></a></span>
<span id="cb231-24"><a href="example.html#cb231-24" tabindex="-1"></a><span class="co"># rm(list = ls())</span></span>
<span id="cb231-25"><a href="example.html#cb231-25" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb231-26"><a href="example.html#cb231-26" tabindex="-1"></a><span class="fu">options</span>(<span class="at">future.globals.maxSize =</span> <span class="dv">1000</span> <span class="sc">*</span> <span class="dv">1024</span><span class="sc">^</span><span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="importing-data-8" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Importing data<a href="example.html#importing-data-8" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>features table</li>
</ul>
<div class="sourceCode" id="cb232"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb232-1"><a href="example.html#cb232-1" tabindex="-1"></a>profile <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(<span class="st">&quot;./dataset/GvHD_stool_metabolites_TM.xlsx&quot;</span>, <span class="at">sheet =</span> <span class="dv">1</span>)</span>
<span id="cb232-2"><a href="example.html#cb232-2" tabindex="-1"></a><span class="fu">head</span>(profile)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 78
##   Index     Compounds      物质  `Class I` 物质一级分类 `Class II` 物质二级分类 `Q1 (Da)` Molecular Weight (Da…¹ `Ionization model` Formula
##   &lt;chr&gt;     &lt;chr&gt;          &lt;chr&gt; &lt;chr&gt;     &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt;            &lt;dbl&gt;                  &lt;dbl&gt; &lt;chr&gt;              &lt;chr&gt;  
## 1 MEDL00066 2-Hydroxy-2-m… -     -         -            -          -                 117.                   118. [M-H]-             C5H10O3
## 2 MEDL00356 Phe-Asp        -     -         -            -          -                 245.                   348. [M+H-2H2O]+        C13H16…
## 3 MEDL00369 Glu-Phe        -     -         -            -          -                 293.                   294. [M-H]-             C14H18…
## 4 MEDL00375 Glu-Trp        -     -         -            -          -                 334.                   333. [M+H]+             C16H19…
## 5 MEDL00392 1-Oleoyl-sn-g… -     -         -            -          -                 525.                   521. [M+H]+             C26H52…
## 6 MEDL00401 Confertifoline -     -         -            -          -                 233.                   234. [M-H]-             C15H22…
## # ℹ abbreviated name: ¹​`Molecular Weight (Da)`
## # ℹ 67 more variables: `KEGG ID` &lt;chr&gt;, HMDB &lt;chr&gt;, `Pubchem CID` &lt;chr&gt;, CAS &lt;chr&gt;, ChEBI &lt;chr&gt;, Metlin &lt;chr&gt;, mix01 &lt;dbl&gt;, mix02 &lt;dbl&gt;,
## #   mix03 &lt;dbl&gt;, mix04 &lt;dbl&gt;, mix05 &lt;dbl&gt;, mix06 &lt;dbl&gt;, mix07 &lt;dbl&gt;, `TZW-V0` &lt;dbl&gt;, `LBC-V0` &lt;dbl&gt;, `HXZ-V0` &lt;dbl&gt;, `LJY-V0` &lt;dbl&gt;,
## #   `CJY-V0` &lt;dbl&gt;, `WKM-V0` &lt;dbl&gt;, `WJH-V0` &lt;dbl&gt;, `ZTX2-V0` &lt;dbl&gt;, `DGL-V0` &lt;dbl&gt;, `ZHP-V0` &lt;dbl&gt;, `MZH-V0` &lt;dbl&gt;, `WL-V0` &lt;dbl&gt;,
## #   `LWJ2-V0` &lt;dbl&gt;, `SZP-V0` &lt;dbl&gt;, `LPQ-V0` &lt;dbl&gt;, `LHR-V0` &lt;dbl&gt;, `XQ-V0` &lt;dbl&gt;, `LJD-V0` &lt;dbl&gt;, `LXH-V0` &lt;dbl&gt;, `YF-V0` &lt;dbl&gt;,
## #   `ZJW-V0` &lt;dbl&gt;, `LHC-V0` &lt;dbl&gt;, `LQB-V0` &lt;dbl&gt;, `LWL-V0` &lt;dbl&gt;, `XD-V0` &lt;dbl&gt;, `ZCY-V0` &lt;dbl&gt;, `ZTX1-V0` &lt;dbl&gt;, `TZW-V1` &lt;dbl&gt;,
## #   `HXZ-V1` &lt;dbl&gt;, `LJY-V1` &lt;dbl&gt;, `CJY-V1` &lt;dbl&gt;, `WKM-V1` &lt;dbl&gt;, `WJH-V1` &lt;dbl&gt;, `ZTX2-V1` &lt;dbl&gt;, `DGL-V1` &lt;dbl&gt;, `ZHP-V1` &lt;dbl&gt;, …</code></pre>
<ul>
<li>metadata table</li>
</ul>
<div class="sourceCode" id="cb234"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb234-1"><a href="example.html#cb234-1" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;./dataset/GvHD_metadata.txt&quot;</span>)</span>
<span id="cb234-2"><a href="example.html#cb234-2" tabindex="-1"></a><span class="fu">head</span>(metadata)</span></code></pre></div>
<pre><code>##    seq_id           SampleName FMT_status        SampleType Group V1_outcome GVHD_type GVHD_type_phynotype Donor baseline_merge
## 1:   4371                  CJY         V0             stool   kid         CR     cGVHD            diarrhea   DO1               
## 2:   4389                  CJY         V1             stool   kid         CR     cGVHD            diarrhea   DO1               
## 3:   4426 CJY_WKM_LXH_FMTdonor            bacteria_solution donor                                                              
## 4:   4375                  DGL         V0             stool adult         CR     cGVHD               bowel   DO2         medium
## 5:   4393                  DGL         V1             stool adult         CR     cGVHD               bowel   DO2               
## 6:   4435         DGL_FMTdonor            bacteria_solution donor                                                              
##    baseline_aGVHD_sainai baseline_aGVHD_sainai_detail baseline_aGVHD_glucksberg baseline_aGVHD_glucksberg_detail baseline_cGVHD_NIH
## 1:                                                                                                                                 
## 2:                                                                                                                                 
## 3:                                                                                                                                 
## 4:                                                                                                                             中度
## 5:                                                                                                                                 
## 6:                                                                                                                                 
##    baseline_cGVHD_NIH_detail Baseline_infection antibiotics_48h_pre_or_post_FMT GVHDdrug_antiMicrobiomeDrug_2weeksPreV0
## 1:                                                                                                                     
## 2:                                                                                                                     
## 3:                                                                                                                     
## 4:          肠道2级，皮肤1级                                                               他克莫司、麦考酚酯、伊曲康唑
## 5:                                                                                         他克莫司、麦考酚酯、伊曲康唑
## 6:                                                                                                                     
##    GVHDdrug_antiMicrobiomeDrug_V0toV1       FMT_method gender age Diagnosis HSCT_DLI_date
## 1:                                    nasojejunal tube   male   3       AML     2020/5/28
## 2:                                    nasojejunal tube   male   3       AML     2020/5/28
## 3:                                                             NA                        
## 4:       他克莫司、麦考酚酯、伊曲康唑      colonoscope   male  36       AML     2019/12/1
## 5:       他克莫司、麦考酚酯、伊曲康唑      colonoscope   male  36       AML     2019/12/1
## 6:                                                             NA                        
##                                             Pretreatment Pretreatment_type      GVHD_prevention death relapse
## 1: Ara-c+CY+IVBu+ATG+Flu+PTCy+PTCy（Ara+Cy+Bu+Flu+PTCy）  marrow cleansing PTCY+PTFLU+MMF+FK506              
## 2: Ara-c+CY+IVBu+ATG+Flu+PTCy+PTCy（Ara+Cy+Bu+Flu+PTCy）  marrow cleansing PTCY+PTFLU+MMF+FK506              
## 3:                                                                                                           
## 4:                                                  BUCY  marrow cleansing      CSA+MTX+MMF+ATG              
## 5:                                                  BUCY  marrow cleansing      CSA+MTX+MMF+ATG              
## 6:</code></pre>
<ul>
<li><p>Data Preparation: SummarizedExperiment object</p></li>
<li><p>Renaming column</p></li>
<li><p>Replace 9 by NA</p></li>
<li><p>column with <code>mix</code> prefix are regards as <strong>QC</strong> samples</p></li>
</ul>
<div class="sourceCode" id="cb236"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb236-1"><a href="example.html#cb236-1" tabindex="-1"></a>getSEobject <span class="ot">&lt;-</span> <span class="cf">function</span>(x, y) {</span>
<span id="cb236-2"><a href="example.html#cb236-2" tabindex="-1"></a>  </span>
<span id="cb236-3"><a href="example.html#cb236-3" tabindex="-1"></a>  <span class="co"># x = metadata</span></span>
<span id="cb236-4"><a href="example.html#cb236-4" tabindex="-1"></a>  <span class="co"># y = profile</span></span>
<span id="cb236-5"><a href="example.html#cb236-5" tabindex="-1"></a></span>
<span id="cb236-6"><a href="example.html#cb236-6" tabindex="-1"></a>  <span class="co"># mix : qc samples</span></span>
<span id="cb236-7"><a href="example.html#cb236-7" tabindex="-1"></a>  qc_samples <span class="ot">&lt;-</span> <span class="fu">grep</span>(<span class="st">&quot;mix&quot;</span>, <span class="fu">colnames</span>(profile), <span class="at">value =</span> T)</span>
<span id="cb236-8"><a href="example.html#cb236-8" tabindex="-1"></a>  qc_groups <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">SampleID =</span> qc_samples,</span>
<span id="cb236-9"><a href="example.html#cb236-9" tabindex="-1"></a>                          <span class="at">V1_outcome =</span> <span class="fu">rep</span>(<span class="st">&quot;QC&quot;</span>, <span class="fu">length</span>(qc_samples)),</span>
<span id="cb236-10"><a href="example.html#cb236-10" tabindex="-1"></a>                          <span class="at">SampleName =</span> <span class="fu">rep</span>(<span class="st">&quot;SampleName&quot;</span>, <span class="fu">length</span>(qc_samples)),</span>
<span id="cb236-11"><a href="example.html#cb236-11" tabindex="-1"></a>                          <span class="at">FMT_status =</span> <span class="fu">rep</span>(<span class="st">&quot;FMT_status&quot;</span>, <span class="fu">length</span>(qc_samples)),</span>
<span id="cb236-12"><a href="example.html#cb236-12" tabindex="-1"></a>                          <span class="at">gender =</span> <span class="fu">rep</span>(<span class="st">&quot;gender&quot;</span>, <span class="fu">length</span>(qc_samples)),</span>
<span id="cb236-13"><a href="example.html#cb236-13" tabindex="-1"></a>                          <span class="at">age =</span> <span class="fu">rep</span>(<span class="st">&quot;age&quot;</span>, <span class="fu">length</span>(qc_samples)))</span>
<span id="cb236-14"><a href="example.html#cb236-14" tabindex="-1"></a>  <span class="co"># target table</span></span>
<span id="cb236-15"><a href="example.html#cb236-15" tabindex="-1"></a>  target <span class="ot">&lt;-</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb236-16"><a href="example.html#cb236-16" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(FMT_status <span class="sc">!=</span> <span class="st">&quot;&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb236-17"><a href="example.html#cb236-17" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">SampleID =</span> <span class="fu">paste</span>(SampleName, FMT_status, <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb236-18"><a href="example.html#cb236-18" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(SampleID, V1_outcome, SampleName, FMT_status, gender, age) <span class="sc">%&gt;%</span></span>
<span id="cb236-19"><a href="example.html#cb236-19" tabindex="-1"></a>    <span class="fu">rbind</span>(qc_groups)</span>
<span id="cb236-20"><a href="example.html#cb236-20" tabindex="-1"></a>  </span>
<span id="cb236-21"><a href="example.html#cb236-21" tabindex="-1"></a>  <span class="co"># profile column </span></span>
<span id="cb236-22"><a href="example.html#cb236-22" tabindex="-1"></a>  <span class="fu">colnames</span>(y) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>, <span class="st">&quot;_&quot;</span>, <span class="fu">colnames</span>(y))</span>
<span id="cb236-23"><a href="example.html#cb236-23" tabindex="-1"></a></span>
<span id="cb236-24"><a href="example.html#cb236-24" tabindex="-1"></a>  sid <span class="ot">&lt;-</span> <span class="fu">intersect</span>(target<span class="sc">$</span>SampleID, <span class="fu">colnames</span>(y))</span>
<span id="cb236-25"><a href="example.html#cb236-25" tabindex="-1"></a>  </span>
<span id="cb236-26"><a href="example.html#cb236-26" tabindex="-1"></a>  features <span class="ot">&lt;-</span> y <span class="sc">%&gt;%</span> </span>
<span id="cb236-27"><a href="example.html#cb236-27" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(sid)) <span class="sc">%&gt;%</span></span>
<span id="cb236-28"><a href="example.html#cb236-28" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">t</span>()</span>
<span id="cb236-29"><a href="example.html#cb236-29" tabindex="-1"></a>  <span class="fu">colnames</span>(features) <span class="ot">&lt;-</span> y<span class="sc">$</span>Index</span>
<span id="cb236-30"><a href="example.html#cb236-30" tabindex="-1"></a>  </span>
<span id="cb236-31"><a href="example.html#cb236-31" tabindex="-1"></a>  <span class="co"># replace 9 by NA </span></span>
<span id="cb236-32"><a href="example.html#cb236-32" tabindex="-1"></a>  features[features <span class="sc">==</span> <span class="dv">9</span>] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb236-33"><a href="example.html#cb236-33" tabindex="-1"></a>  </span>
<span id="cb236-34"><a href="example.html#cb236-34" tabindex="-1"></a>  target <span class="ot">&lt;-</span> target[<span class="fu">pmatch</span>(sid, target<span class="sc">$</span>SampleID), , F]</span>
<span id="cb236-35"><a href="example.html#cb236-35" tabindex="-1"></a>  </span>
<span id="cb236-36"><a href="example.html#cb236-36" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb236-37"><a href="example.html#cb236-37" tabindex="-1"></a>                                  <span class="at">features =</span> features)</span>
<span id="cb236-38"><a href="example.html#cb236-38" tabindex="-1"></a>  <span class="fu">return</span>(res)  </span>
<span id="cb236-39"><a href="example.html#cb236-39" tabindex="-1"></a>}</span>
<span id="cb236-40"><a href="example.html#cb236-40" tabindex="-1"></a></span>
<span id="cb236-41"><a href="example.html#cb236-41" tabindex="-1"></a>se_raw <span class="ot">&lt;-</span> <span class="fu">getSEobject</span>(metadata, profile)</span>
<span id="cb236-42"><a href="example.html#cb236-42" tabindex="-1"></a></span>
<span id="cb236-43"><a href="example.html#cb236-43" tabindex="-1"></a>se_raw</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 1235 61 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(1235): MEDL00066 MEDL00356 ... MW0169477 MW0169569
## rowData names(0):
## colnames(61): CJY_V0 CJY_V1 ... mix06 mix07
## colData names(5): group SampleName FMT_status gender age</code></pre>
</div>
<div id="data-processing-1" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> Data Processing<a href="example.html#data-processing-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="data-checking-2" class="section level3 hasAnchor" number="6.3.1">
<h3><span class="header-section-number">6.3.1</span> Data Checking<a href="example.html#data-checking-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Features in PomaSummarizedExperiment object must have the following criterion:</p>
<ul>
<li><p>All data values are numeric.</p></li>
<li><p>A total of 0 (0%) missing values were detected.</p></li>
</ul>
<div class="sourceCode" id="cb238"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb238-1"><a href="example.html#cb238-1" tabindex="-1"></a>CheckData <span class="ot">&lt;-</span> <span class="cf">function</span>(object) {</span>
<span id="cb238-2"><a href="example.html#cb238-2" tabindex="-1"></a>  </span>
<span id="cb238-3"><a href="example.html#cb238-3" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object)</span>
<span id="cb238-4"><a href="example.html#cb238-4" tabindex="-1"></a>  </span>
<span id="cb238-5"><a href="example.html#cb238-5" tabindex="-1"></a>  <span class="co"># numeric &amp; missing values</span></span>
<span id="cb238-6"><a href="example.html#cb238-6" tabindex="-1"></a>  int_mat <span class="ot">&lt;-</span> features_tab</span>
<span id="cb238-7"><a href="example.html#cb238-7" tabindex="-1"></a>  rowNms <span class="ot">&lt;-</span> <span class="fu">rownames</span>(int_mat)</span>
<span id="cb238-8"><a href="example.html#cb238-8" tabindex="-1"></a>  colNms <span class="ot">&lt;-</span> <span class="fu">colnames</span>(int_mat)</span>
<span id="cb238-9"><a href="example.html#cb238-9" tabindex="-1"></a>  naNms <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">is.na</span>(int_mat))</span>
<span id="cb238-10"><a href="example.html#cb238-10" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(int_mat)) {</span>
<span id="cb238-11"><a href="example.html#cb238-11" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">class</span>(int_mat[, i]) <span class="sc">==</span> <span class="st">&quot;integer64&quot;</span>) {</span>
<span id="cb238-12"><a href="example.html#cb238-12" tabindex="-1"></a>      int_mat[, i] <span class="ot">&lt;-</span> <span class="fu">as.double</span>(int_mat[, i])</span>
<span id="cb238-13"><a href="example.html#cb238-13" tabindex="-1"></a>    }</span>
<span id="cb238-14"><a href="example.html#cb238-14" tabindex="-1"></a>  }</span>
<span id="cb238-15"><a href="example.html#cb238-15" tabindex="-1"></a>  </span>
<span id="cb238-16"><a href="example.html#cb238-16" tabindex="-1"></a>  num_mat <span class="ot">&lt;-</span> <span class="fu">apply</span>(int_mat, <span class="dv">2</span>, as.numeric)</span>
<span id="cb238-17"><a href="example.html#cb238-17" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">sum</span>(<span class="fu">is.na</span>(num_mat)) <span class="sc">&gt;</span> naNms) {</span>
<span id="cb238-18"><a href="example.html#cb238-18" tabindex="-1"></a>    num_mat <span class="ot">&lt;-</span> <span class="fu">apply</span>(int_mat, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;,&quot;</span>,  <span class="st">&quot;&quot;</span>, x)))</span>
<span id="cb238-19"><a href="example.html#cb238-19" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">sum</span>(<span class="fu">is.na</span>(num_mat)) <span class="sc">&gt;</span> naNms) {</span>
<span id="cb238-20"><a href="example.html#cb238-20" tabindex="-1"></a>      <span class="fu">message</span>(<span class="st">&quot;&lt;font color=</span><span class="sc">\&quot;</span><span class="st">red</span><span class="sc">\&quot;</span><span class="st">&gt;Non-numeric values were found and replaced by NA.&lt;/font&gt;&quot;</span>)</span>
<span id="cb238-21"><a href="example.html#cb238-21" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb238-22"><a href="example.html#cb238-22" tabindex="-1"></a>      <span class="fu">message</span>(<span class="st">&quot;All data values are numeric.&quot;</span>)</span>
<span id="cb238-23"><a href="example.html#cb238-23" tabindex="-1"></a>    }</span>
<span id="cb238-24"><a href="example.html#cb238-24" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb238-25"><a href="example.html#cb238-25" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">&quot;All data values are numeric.&quot;</span>)</span>
<span id="cb238-26"><a href="example.html#cb238-26" tabindex="-1"></a>  }</span>
<span id="cb238-27"><a href="example.html#cb238-27" tabindex="-1"></a>  </span>
<span id="cb238-28"><a href="example.html#cb238-28" tabindex="-1"></a>  int_mat <span class="ot">&lt;-</span> num_mat</span>
<span id="cb238-29"><a href="example.html#cb238-29" tabindex="-1"></a>  <span class="fu">rownames</span>(int_mat) <span class="ot">&lt;-</span> rowNms</span>
<span id="cb238-30"><a href="example.html#cb238-30" tabindex="-1"></a>  <span class="fu">colnames</span>(int_mat) <span class="ot">&lt;-</span> colNms</span>
<span id="cb238-31"><a href="example.html#cb238-31" tabindex="-1"></a>  varCol <span class="ot">&lt;-</span> <span class="fu">apply</span>(int_mat, <span class="dv">2</span>, var, <span class="at">na.rm =</span> T)</span>
<span id="cb238-32"><a href="example.html#cb238-32" tabindex="-1"></a>  constCol <span class="ot">&lt;-</span> (varCol <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> <span class="fu">is.na</span>(varCol))</span>
<span id="cb238-33"><a href="example.html#cb238-33" tabindex="-1"></a>  constNum <span class="ot">&lt;-</span> <span class="fu">sum</span>(constCol, <span class="at">na.rm =</span> T)</span>
<span id="cb238-34"><a href="example.html#cb238-34" tabindex="-1"></a>  <span class="cf">if</span> (constNum <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb238-35"><a href="example.html#cb238-35" tabindex="-1"></a>    <span class="fu">message</span>(<span class="fu">paste</span>(<span class="st">&quot;&lt;font color=</span><span class="sc">\&quot;</span><span class="st">red</span><span class="sc">\&quot;</span><span class="st">&gt;&quot;</span>, constNum, </span>
<span id="cb238-36"><a href="example.html#cb238-36" tabindex="-1"></a>      <span class="st">&quot;features with a constant or single value across samples were found and deleted.&lt;/font&gt;&quot;</span>))</span>
<span id="cb238-37"><a href="example.html#cb238-37" tabindex="-1"></a>    int_mat <span class="ot">&lt;-</span> int_mat[, <span class="sc">!</span>constCol, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb238-38"><a href="example.html#cb238-38" tabindex="-1"></a>  }</span>
<span id="cb238-39"><a href="example.html#cb238-39" tabindex="-1"></a>  </span>
<span id="cb238-40"><a href="example.html#cb238-40" tabindex="-1"></a>  totalCount <span class="ot">&lt;-</span> <span class="fu">nrow</span>(int_mat) <span class="sc">*</span> <span class="fu">ncol</span>(int_mat)</span>
<span id="cb238-41"><a href="example.html#cb238-41" tabindex="-1"></a>  naCount <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">is.na</span>(int_mat))</span>
<span id="cb238-42"><a href="example.html#cb238-42" tabindex="-1"></a>  naPercent <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> naCount<span class="sc">/</span>totalCount, <span class="dv">1</span>)</span>
<span id="cb238-43"><a href="example.html#cb238-43" tabindex="-1"></a></span>
<span id="cb238-44"><a href="example.html#cb238-44" tabindex="-1"></a>  <span class="fu">message</span>(<span class="fu">paste</span>(<span class="st">&quot;A total of &quot;</span>, naCount, <span class="st">&quot; (&quot;</span>, naPercent, </span>
<span id="cb238-45"><a href="example.html#cb238-45" tabindex="-1"></a>    <span class="st">&quot;%) missing values were detected.&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))  </span>
<span id="cb238-46"><a href="example.html#cb238-46" tabindex="-1"></a>  </span>
<span id="cb238-47"><a href="example.html#cb238-47" tabindex="-1"></a>  <span class="co"># save int_mat into se object </span></span>
<span id="cb238-48"><a href="example.html#cb238-48" tabindex="-1"></a>  target <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object) <span class="sc">%&gt;%</span></span>
<span id="cb238-49"><a href="example.html#cb238-49" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb238-50"><a href="example.html#cb238-50" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb238-51"><a href="example.html#cb238-51" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb238-52"><a href="example.html#cb238-52" tabindex="-1"></a>                                  <span class="at">features =</span> <span class="fu">t</span>(int_mat))</span>
<span id="cb238-53"><a href="example.html#cb238-53" tabindex="-1"></a>  </span>
<span id="cb238-54"><a href="example.html#cb238-54" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb238-55"><a href="example.html#cb238-55" tabindex="-1"></a>}</span>
<span id="cb238-56"><a href="example.html#cb238-56" tabindex="-1"></a></span>
<span id="cb238-57"><a href="example.html#cb238-57" tabindex="-1"></a>se_check <span class="ot">&lt;-</span> <span class="fu">CheckData</span>(<span class="at">object =</span> se_raw)</span>
<span id="cb238-58"><a href="example.html#cb238-58" tabindex="-1"></a>se_check</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 1235 61 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(1235): MEDL00066 MEDL00356 ... MW0169477 MW0169569
## rowData names(0):
## colnames(61): CJY_V0 CJY_V1 ... mix06 mix07
## colData names(5): group SampleName FMT_status gender age</code></pre>
</div>
<div id="missing-value-imputation-2" class="section level3 hasAnchor" number="6.3.2">
<h3><span class="header-section-number">6.3.2</span> Missing value imputation<a href="example.html#missing-value-imputation-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>“none”: all missing values will be replaced by zero.</p>
<p>“LOD”: specific Limit Of Detection which provides by user.</p>
<p>“half_min”: half minimal values across samples except zero.</p>
<p>“median”: median values across samples except zero.</p>
<p>“mean”: mean values across samples except zero.</p>
<p>“min”: minimal values across samples except zero.</p>
<p>“knn”: k-nearest neighbors samples.</p>
<p>“rf”: nonparametric missing value imputation using Random Forest.</p>
<p>“QRILC”: missing values imputation based quantile regression. (default: “none”).</p>
<div class="sourceCode" id="cb240"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb240-1"><a href="example.html#cb240-1" tabindex="-1"></a>impute_abundance <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb240-2"><a href="example.html#cb240-2" tabindex="-1"></a>    object,</span>
<span id="cb240-3"><a href="example.html#cb240-3" tabindex="-1"></a>    group,</span>
<span id="cb240-4"><a href="example.html#cb240-4" tabindex="-1"></a>    <span class="at">ZerosAsNA =</span> <span class="cn">FALSE</span>,</span>
<span id="cb240-5"><a href="example.html#cb240-5" tabindex="-1"></a>    <span class="at">RemoveNA =</span> <span class="cn">TRUE</span>,</span>
<span id="cb240-6"><a href="example.html#cb240-6" tabindex="-1"></a>    <span class="at">prevalence =</span> <span class="fl">0.5</span>,</span>
<span id="cb240-7"><a href="example.html#cb240-7" tabindex="-1"></a>    <span class="at">method =</span> <span class="fu">c</span>(<span class="st">&quot;none&quot;</span>, <span class="st">&quot;LOD&quot;</span>, <span class="st">&quot;half_min&quot;</span>, <span class="st">&quot;median&quot;</span>,</span>
<span id="cb240-8"><a href="example.html#cb240-8" tabindex="-1"></a>               <span class="st">&quot;mean&quot;</span>, <span class="st">&quot;min&quot;</span>, <span class="st">&quot;knn&quot;</span>, <span class="st">&quot;rf&quot;</span>, <span class="st">&quot;QRILC&quot;</span>),</span>
<span id="cb240-9"><a href="example.html#cb240-9" tabindex="-1"></a>    <span class="at">LOD =</span> <span class="cn">NULL</span>) {</span>
<span id="cb240-10"><a href="example.html#cb240-10" tabindex="-1"></a></span>
<span id="cb240-11"><a href="example.html#cb240-11" tabindex="-1"></a>  <span class="co"># object = se_check</span></span>
<span id="cb240-12"><a href="example.html#cb240-12" tabindex="-1"></a>  <span class="co"># group = &quot;group&quot;</span></span>
<span id="cb240-13"><a href="example.html#cb240-13" tabindex="-1"></a>  <span class="co"># ZerosAsNA = TRUE</span></span>
<span id="cb240-14"><a href="example.html#cb240-14" tabindex="-1"></a>  <span class="co"># RemoveNA = TRUE</span></span>
<span id="cb240-15"><a href="example.html#cb240-15" tabindex="-1"></a>  <span class="co"># prevalence = 0.5</span></span>
<span id="cb240-16"><a href="example.html#cb240-16" tabindex="-1"></a>  <span class="co"># method = &quot;knn&quot;</span></span>
<span id="cb240-17"><a href="example.html#cb240-17" tabindex="-1"></a></span>
<span id="cb240-18"><a href="example.html#cb240-18" tabindex="-1"></a>  <span class="cf">if</span> (base<span class="sc">::</span><span class="fu">missing</span>(object)) {</span>
<span id="cb240-19"><a href="example.html#cb240-19" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;object argument is empty!&quot;</span>)</span>
<span id="cb240-20"><a href="example.html#cb240-20" tabindex="-1"></a>  }</span>
<span id="cb240-21"><a href="example.html#cb240-21" tabindex="-1"></a></span>
<span id="cb240-22"><a href="example.html#cb240-22" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span>methods<span class="sc">::</span><span class="fu">is</span>(object, <span class="st">&quot;SummarizedExperiment&quot;</span>)) {</span>
<span id="cb240-23"><a href="example.html#cb240-23" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;object is not either a phyloseq or SummarizedExperiment object.&quot;</span>)</span>
<span id="cb240-24"><a href="example.html#cb240-24" tabindex="-1"></a>  }</span>
<span id="cb240-25"><a href="example.html#cb240-25" tabindex="-1"></a></span>
<span id="cb240-26"><a href="example.html#cb240-26" tabindex="-1"></a>  method <span class="ot">&lt;-</span> <span class="fu">match.arg</span>(</span>
<span id="cb240-27"><a href="example.html#cb240-27" tabindex="-1"></a>    method, <span class="fu">c</span>(<span class="st">&quot;none&quot;</span>, <span class="st">&quot;LOD&quot;</span>, <span class="st">&quot;half_min&quot;</span>, <span class="st">&quot;median&quot;</span>,</span>
<span id="cb240-28"><a href="example.html#cb240-28" tabindex="-1"></a>              <span class="st">&quot;mean&quot;</span>, <span class="st">&quot;min&quot;</span>, <span class="st">&quot;knn&quot;</span>, <span class="st">&quot;rf&quot;</span>, <span class="st">&quot;QRILC&quot;</span>)</span>
<span id="cb240-29"><a href="example.html#cb240-29" tabindex="-1"></a>  )</span>
<span id="cb240-30"><a href="example.html#cb240-30" tabindex="-1"></a></span>
<span id="cb240-31"><a href="example.html#cb240-31" tabindex="-1"></a>  <span class="cf">if</span> (base<span class="sc">::</span><span class="fu">missing</span>(method)) {</span>
<span id="cb240-32"><a href="example.html#cb240-32" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">&quot;method argument is empty! KNN will be used&quot;</span>)</span>
<span id="cb240-33"><a href="example.html#cb240-33" tabindex="-1"></a>  }</span>
<span id="cb240-34"><a href="example.html#cb240-34" tabindex="-1"></a></span>
<span id="cb240-35"><a href="example.html#cb240-35" tabindex="-1"></a>  <span class="co"># profile: row-&gt;samples; col-&gt;features</span></span>
<span id="cb240-36"><a href="example.html#cb240-36" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">all</span>(<span class="sc">!</span><span class="fu">is.null</span>(object), <span class="fu">inherits</span>(object, <span class="st">&quot;SummarizedExperiment&quot;</span>))) {</span>
<span id="cb240-37"><a href="example.html#cb240-37" tabindex="-1"></a>    <span class="co"># sample table &amp; profile table</span></span>
<span id="cb240-38"><a href="example.html#cb240-38" tabindex="-1"></a>    sam_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object) <span class="sc">%&gt;%</span></span>
<span id="cb240-39"><a href="example.html#cb240-39" tabindex="-1"></a>      <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb240-40"><a href="example.html#cb240-40" tabindex="-1"></a>      tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;TempRowNames&quot;</span>)</span>
<span id="cb240-41"><a href="example.html#cb240-41" tabindex="-1"></a>    prf_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) <span class="sc">%&gt;%</span></span>
<span id="cb240-42"><a href="example.html#cb240-42" tabindex="-1"></a>      <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb240-43"><a href="example.html#cb240-43" tabindex="-1"></a>      <span class="fu">t</span>()</span>
<span id="cb240-44"><a href="example.html#cb240-44" tabindex="-1"></a>  }</span>
<span id="cb240-45"><a href="example.html#cb240-45" tabindex="-1"></a></span>
<span id="cb240-46"><a href="example.html#cb240-46" tabindex="-1"></a>  group_index <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">colnames</span>(sam_tab) <span class="sc">==</span> group)</span>
<span id="cb240-47"><a href="example.html#cb240-47" tabindex="-1"></a>  samples_groups <span class="ot">&lt;-</span> sam_tab[, group_index]</span>
<span id="cb240-48"><a href="example.html#cb240-48" tabindex="-1"></a>  to_imp_data <span class="ot">&lt;-</span> prf_tab <span class="sc">%&gt;%</span> <span class="fu">as.matrix</span>()</span>
<span id="cb240-49"><a href="example.html#cb240-49" tabindex="-1"></a></span>
<span id="cb240-50"><a href="example.html#cb240-50" tabindex="-1"></a>  <span class="cf">if</span> (ZerosAsNA) {</span>
<span id="cb240-51"><a href="example.html#cb240-51" tabindex="-1"></a>    to_imp_data[to_imp_data <span class="sc">==</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb240-52"><a href="example.html#cb240-52" tabindex="-1"></a>    to_imp_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">cbind</span>(<span class="at">Group =</span> samples_groups, to_imp_data))</span>
<span id="cb240-53"><a href="example.html#cb240-53" tabindex="-1"></a>    <span class="fu">colnames</span>(to_imp_data)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(to_imp_data)] <span class="ot">&lt;-</span> <span class="fu">colnames</span>(prf_tab)</span>
<span id="cb240-54"><a href="example.html#cb240-54" tabindex="-1"></a></span>
<span id="cb240-55"><a href="example.html#cb240-55" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb240-56"><a href="example.html#cb240-56" tabindex="-1"></a>    to_imp_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">cbind</span>(<span class="at">Group =</span> samples_groups, to_imp_data))</span>
<span id="cb240-57"><a href="example.html#cb240-57" tabindex="-1"></a>    <span class="fu">colnames</span>(to_imp_data)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(to_imp_data)] <span class="ot">&lt;-</span> <span class="fu">colnames</span>(prf_tab)</span>
<span id="cb240-58"><a href="example.html#cb240-58" tabindex="-1"></a>  }</span>
<span id="cb240-59"><a href="example.html#cb240-59" tabindex="-1"></a></span>
<span id="cb240-60"><a href="example.html#cb240-60" tabindex="-1"></a>  percent_na <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">is.na</span>(to_imp_data))</span>
<span id="cb240-61"><a href="example.html#cb240-61" tabindex="-1"></a>  <span class="cf">if</span> (percent_na <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb240-62"><a href="example.html#cb240-62" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">&quot;No missing values detected in your data&quot;</span>)</span>
<span id="cb240-63"><a href="example.html#cb240-63" tabindex="-1"></a>    <span class="cf">if</span> (method <span class="sc">!=</span> <span class="st">&quot;none&quot;</span>) {</span>
<span id="cb240-64"><a href="example.html#cb240-64" tabindex="-1"></a>      method <span class="ot">&lt;-</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb240-65"><a href="example.html#cb240-65" tabindex="-1"></a>    }</span>
<span id="cb240-66"><a href="example.html#cb240-66" tabindex="-1"></a>  }</span>
<span id="cb240-67"><a href="example.html#cb240-67" tabindex="-1"></a></span>
<span id="cb240-68"><a href="example.html#cb240-68" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">isTRUE</span>(RemoveNA)) {</span>
<span id="cb240-69"><a href="example.html#cb240-69" tabindex="-1"></a>    count_NA <span class="ot">&lt;-</span> stats<span class="sc">::</span><span class="fu">aggregate</span>(</span>
<span id="cb240-70"><a href="example.html#cb240-70" tabindex="-1"></a>          . <span class="sc">~</span> Group,</span>
<span id="cb240-71"><a href="example.html#cb240-71" tabindex="-1"></a>          <span class="at">data =</span> to_imp_data,</span>
<span id="cb240-72"><a href="example.html#cb240-72" tabindex="-1"></a>          <span class="cf">function</span>(x) {(<span class="fu">sum</span>(<span class="fu">is.na</span>(x)) <span class="sc">/</span> (<span class="fu">sum</span>(<span class="fu">is.na</span>(x)) <span class="sc">+</span> <span class="fu">sum</span>(<span class="sc">!</span><span class="fu">is.na</span>(x))) ) },</span>
<span id="cb240-73"><a href="example.html#cb240-73" tabindex="-1"></a>          <span class="at">na.action =</span> <span class="cn">NULL</span>)</span>
<span id="cb240-74"><a href="example.html#cb240-74" tabindex="-1"></a>    count_NA <span class="ot">&lt;-</span> count_NA <span class="sc">%&gt;%</span></span>
<span id="cb240-75"><a href="example.html#cb240-75" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>Group)</span>
<span id="cb240-76"><a href="example.html#cb240-76" tabindex="-1"></a>    correct_names <span class="ot">&lt;-</span> <span class="fu">names</span>(count_NA)</span>
<span id="cb240-77"><a href="example.html#cb240-77" tabindex="-1"></a>    supress <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">as.data.frame</span>(<span class="fu">lapply</span>(count_NA, <span class="cf">function</span>(x) <span class="fu">any</span>(x <span class="sc">&gt;</span> prevalence))))</span>
<span id="cb240-78"><a href="example.html#cb240-78" tabindex="-1"></a>    <span class="fu">names</span>(supress) <span class="ot">&lt;-</span> correct_names</span>
<span id="cb240-79"><a href="example.html#cb240-79" tabindex="-1"></a>    correct_names <span class="ot">&lt;-</span> <span class="fu">names</span>(supress[supress <span class="sc">==</span> <span class="st">&quot;FALSE&quot;</span>])</span>
<span id="cb240-80"><a href="example.html#cb240-80" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> to_imp_data[, <span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(to_imp_data)][<span class="sc">!</span>supress]</span>
<span id="cb240-81"><a href="example.html#cb240-81" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">sapply</span>(depurdata, <span class="cf">function</span>(x) <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(x)))</span>
<span id="cb240-82"><a href="example.html#cb240-82" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb240-83"><a href="example.html#cb240-83" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> to_imp_data[, <span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(to_imp_data)]</span>
<span id="cb240-84"><a href="example.html#cb240-84" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">sapply</span>(depurdata, <span class="cf">function</span>(x) <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(x)))</span>
<span id="cb240-85"><a href="example.html#cb240-85" tabindex="-1"></a>    correct_names <span class="ot">&lt;-</span> <span class="fu">colnames</span>(prf_tab)</span>
<span id="cb240-86"><a href="example.html#cb240-86" tabindex="-1"></a>  }</span>
<span id="cb240-87"><a href="example.html#cb240-87" tabindex="-1"></a></span>
<span id="cb240-88"><a href="example.html#cb240-88" tabindex="-1"></a>  <span class="co"># Row-&gt;feature;Col-&gt;sample</span></span>
<span id="cb240-89"><a href="example.html#cb240-89" tabindex="-1"></a>  <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;none&quot;</span>) {</span>
<span id="cb240-90"><a href="example.html#cb240-90" tabindex="-1"></a>    depurdata[<span class="fu">is.na</span>(depurdata)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb240-91"><a href="example.html#cb240-91" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;LOD&quot;</span>) {</span>
<span id="cb240-92"><a href="example.html#cb240-92" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.null</span>(LOD)) {</span>
<span id="cb240-93"><a href="example.html#cb240-93" tabindex="-1"></a>      <span class="fu">message</span>(<span class="st">&quot;No LOD provided, regard one-tenth mininal value as LOD&quot;</span>)</span>
<span id="cb240-94"><a href="example.html#cb240-94" tabindex="-1"></a>      depurdata_withoutNA <span class="ot">&lt;-</span> depurdata[<span class="sc">!</span><span class="fu">is.na</span>(depurdata)]</span>
<span id="cb240-95"><a href="example.html#cb240-95" tabindex="-1"></a>      LOD <span class="ot">&lt;-</span> <span class="fu">min</span>(depurdata_withoutNA[depurdata_withoutNA <span class="sc">!=</span> <span class="dv">0</span>]) <span class="sc">/</span> <span class="dv">10</span></span>
<span id="cb240-96"><a href="example.html#cb240-96" tabindex="-1"></a>    }</span>
<span id="cb240-97"><a href="example.html#cb240-97" tabindex="-1"></a>    depurdata[<span class="fu">is.na</span>(depurdata)] <span class="ot">&lt;-</span> LOD</span>
<span id="cb240-98"><a href="example.html#cb240-98" tabindex="-1"></a>    depurdata[depurdata <span class="sc">==</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> LOD</span>
<span id="cb240-99"><a href="example.html#cb240-99" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;half_min&quot;</span>) {</span>
<span id="cb240-100"><a href="example.html#cb240-100" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">apply</span>(depurdata, <span class="dv">2</span>, <span class="cf">function</span>(x) {</span>
<span id="cb240-101"><a href="example.html#cb240-101" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">is.numeric</span>(x)) <span class="fu">ifelse</span>(<span class="fu">is.na</span>(x), <span class="fu">min</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)<span class="sc">/</span><span class="dv">2</span>, x) <span class="cf">else</span> x})</span>
<span id="cb240-102"><a href="example.html#cb240-102" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;median&quot;</span>) {</span>
<span id="cb240-103"><a href="example.html#cb240-103" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">apply</span>(depurdata, <span class="dv">2</span>, <span class="cf">function</span>(x) {</span>
<span id="cb240-104"><a href="example.html#cb240-104" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">is.numeric</span>(x)) <span class="fu">ifelse</span>(<span class="fu">is.na</span>(x), <span class="fu">median</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), x) <span class="cf">else</span> x})</span>
<span id="cb240-105"><a href="example.html#cb240-105" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;mean&quot;</span>) {</span>
<span id="cb240-106"><a href="example.html#cb240-106" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">apply</span>(depurdata, <span class="dv">2</span>, <span class="cf">function</span>(x) {</span>
<span id="cb240-107"><a href="example.html#cb240-107" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">is.numeric</span>(x)) <span class="fu">ifelse</span>(<span class="fu">is.na</span>(x), <span class="fu">mean</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), x) <span class="cf">else</span> x})</span>
<span id="cb240-108"><a href="example.html#cb240-108" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;min&quot;</span>) {</span>
<span id="cb240-109"><a href="example.html#cb240-109" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">apply</span>(depurdata, <span class="dv">2</span>, <span class="cf">function</span>(x) {</span>
<span id="cb240-110"><a href="example.html#cb240-110" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">is.numeric</span>(x)) <span class="fu">ifelse</span>(<span class="fu">is.na</span>(x), <span class="fu">min</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), x) <span class="cf">else</span> x})</span>
<span id="cb240-111"><a href="example.html#cb240-111" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;knn&quot;</span>) {</span>
<span id="cb240-112"><a href="example.html#cb240-112" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">t</span>(depurdata)</span>
<span id="cb240-113"><a href="example.html#cb240-113" tabindex="-1"></a>    datai <span class="ot">&lt;-</span> impute<span class="sc">::</span><span class="fu">impute.knn</span>(depurdata, <span class="at">k =</span> <span class="dv">20</span>)</span>
<span id="cb240-114"><a href="example.html#cb240-114" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">t</span>(datai<span class="sc">$</span>data)</span>
<span id="cb240-115"><a href="example.html#cb240-115" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;rf&quot;</span>) {</span>
<span id="cb240-116"><a href="example.html#cb240-116" tabindex="-1"></a>    fit <span class="ot">&lt;-</span> missForest<span class="sc">::</span><span class="fu">missForest</span>(<span class="fu">t</span>(depurdata))</span>
<span id="cb240-117"><a href="example.html#cb240-117" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> fit<span class="sc">$</span>ximp <span class="sc">%&gt;%</span></span>
<span id="cb240-118"><a href="example.html#cb240-118" tabindex="-1"></a>      <span class="fu">t</span>()</span>
<span id="cb240-119"><a href="example.html#cb240-119" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;QRILC&quot;</span>) {</span>
<span id="cb240-120"><a href="example.html#cb240-120" tabindex="-1"></a>    fit <span class="ot">&lt;-</span> <span class="fu">log</span>(<span class="fu">t</span>(depurdata)) <span class="sc">%&gt;%</span></span>
<span id="cb240-121"><a href="example.html#cb240-121" tabindex="-1"></a>      imputeLCMD<span class="sc">::</span><span class="fu">impute.QRILC</span>()</span>
<span id="cb240-122"><a href="example.html#cb240-122" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">t</span>(fit[[<span class="dv">1</span>]])</span>
<span id="cb240-123"><a href="example.html#cb240-123" tabindex="-1"></a>  }</span>
<span id="cb240-124"><a href="example.html#cb240-124" tabindex="-1"></a></span>
<span id="cb240-125"><a href="example.html#cb240-125" tabindex="-1"></a>  <span class="fu">colnames</span>(depurdata) <span class="ot">&lt;-</span> correct_names</span>
<span id="cb240-126"><a href="example.html#cb240-126" tabindex="-1"></a>  <span class="fu">rownames</span>(depurdata) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(prf_tab)</span>
<span id="cb240-127"><a href="example.html#cb240-127" tabindex="-1"></a></span>
<span id="cb240-128"><a href="example.html#cb240-128" tabindex="-1"></a>  <span class="cf">if</span> (methods<span class="sc">::</span><span class="fu">is</span>(object, <span class="st">&quot;SummarizedExperiment&quot;</span>)) {</span>
<span id="cb240-129"><a href="example.html#cb240-129" tabindex="-1"></a>    target <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object) <span class="sc">%&gt;%</span></span>
<span id="cb240-130"><a href="example.html#cb240-130" tabindex="-1"></a>      <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb240-131"><a href="example.html#cb240-131" tabindex="-1"></a>      <span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb240-132"><a href="example.html#cb240-132" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb240-133"><a href="example.html#cb240-133" tabindex="-1"></a>                                    <span class="at">features =</span> depurdata)</span>
<span id="cb240-134"><a href="example.html#cb240-134" tabindex="-1"></a>    </span>
<span id="cb240-135"><a href="example.html#cb240-135" tabindex="-1"></a>  }</span>
<span id="cb240-136"><a href="example.html#cb240-136" tabindex="-1"></a></span>
<span id="cb240-137"><a href="example.html#cb240-137" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb240-138"><a href="example.html#cb240-138" tabindex="-1"></a>}</span>
<span id="cb240-139"><a href="example.html#cb240-139" tabindex="-1"></a></span>
<span id="cb240-140"><a href="example.html#cb240-140" tabindex="-1"></a>se_impute <span class="ot">&lt;-</span> <span class="fu">impute_abundance</span>(</span>
<span id="cb240-141"><a href="example.html#cb240-141" tabindex="-1"></a>                    se_check, </span>
<span id="cb240-142"><a href="example.html#cb240-142" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;group&quot;</span>,</span>
<span id="cb240-143"><a href="example.html#cb240-143" tabindex="-1"></a>                    <span class="at">ZerosAsNA =</span> <span class="cn">TRUE</span>, </span>
<span id="cb240-144"><a href="example.html#cb240-144" tabindex="-1"></a>                    <span class="at">RemoveNA =</span> <span class="cn">TRUE</span>, </span>
<span id="cb240-145"><a href="example.html#cb240-145" tabindex="-1"></a>                    <span class="at">prevalence =</span> <span class="fl">0.5</span>, </span>
<span id="cb240-146"><a href="example.html#cb240-146" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;knn&quot;</span>)</span>
<span id="cb240-147"><a href="example.html#cb240-147" tabindex="-1"></a>se_impute</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 978 61 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(978): MEDL00066 MEDL00356 ... MW0168866 MW0169569
## rowData names(0):
## colnames(61): CJY_V0 CJY_V1 ... mix06 mix07
## colData names(5): group SampleName FMT_status gender age</code></pre>
<div class="sourceCode" id="cb242"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb242-1"><a href="example.html#cb242-1" tabindex="-1"></a><span class="co"># profile</span></span>
<span id="cb242-2"><a href="example.html#cb242-2" tabindex="-1"></a><span class="fu">head</span>(SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(se_impute))</span></code></pre></div>
<pre><code>##              CJY_V0     CJY_V1 DGL_V0    DGL_V1 HXZ_V0    HXZ_V1    LBC_V0    LBC_V1   LHC_V1    LHC_V0     LHR_V0   LHR_V1 LJD_V0  LJD_V1
## MEDL00066  10447.00 4011900.00 560560 149120.00 442540 181010.00   12620.0 173400.00  97848.0  7682.475  744910.00 95554.00  29999 65406.0
## MEDL00356  50530.92   38175.58  71717 128990.00 162090 106800.00 8524300.0  61881.00  88503.0 21030.000   32277.89 28348.57  30666 34586.0
## MEDL00369  40739.79   85573.00  36339  50110.28 135600  26164.00    7287.5  11808.00  86864.6 34776.883   39771.74 34645.75  52056 17594.0
## MEDL00375  31087.00   88306.00 449150 263050.00 216830  37280.00  302150.0 607850.00  87101.0  7595.700 3162900.00 83160.00 293680 80921.0
## MEDL00392 238370.00  260320.00 250350  21585.00 480800  91366.73  225860.0  53617.11 809940.0  9993.000  964380.00 65519.00  99231  5760.1
## MEDL00401  12589.00   12590.00  15954  16425.00  14961   8432.20   16545.0   9618.60  20929.0 12400.000   17539.00 10249.00  13183 12489.0
##           LJY_V0  LJY_V1   LPQ_V0  LPQ_V1  LQB_V1  LQB_V0  LWJ2_V0  LWJ2_V1    LWL_V1   LWL_V0 LXH_V0   LXH_V1     MZH_V0    MZH_V1 SZP_V0
## MEDL00066 321900 5065100 363640.0  464980 5664000  6312.7 131810.0 262810.0 4501100.0 786330.0 342390  13018.0 1351000.00 2677700.0 574420
## MEDL00356  48010   20193  14003.0 1818200   65151 37127.0  23929.0  17193.0   15555.0 182650.0 277870 121210.8   27022.93   53654.0  90642
## MEDL00369  17115   27850   5519.3  110600   13576  5116.1 175290.0 122980.0   55525.0   8868.3  48983  10114.0   33974.00    9057.1 112340
## MEDL00375  75261   39253 127590.0   96760  166800 69780.0 244030.0 174640.0   77431.0 227830.0  55610  19619.0  291990.00   31441.0 524740
## MEDL00392 223600   37209 679940.0  467250   16911 10461.0 123994.5 125642.8    5700.4  10468.0  52219  25862.0   88710.00 1032412.4  42641
## MEDL00401  12513   10672  17804.0   15920   10570 14307.0  12191.0  12257.0   11169.0  16478.0  13150   9346.6   10180.00    4854.2  15582
##           SZP_V1    TZW_V0   TZW_V1 WJH_V0 WJH_V1 WKM_V0  WKM_V1     WL_V0     WL_V1   XD_V0    XD_V1     XQ_V0    XQ_V1    YF_V1  YF_V0
## MEDL00066 137160 562570.00 10292000 365240  48338 226940 2036500 124550.00 128280.00 1742800 137230.0 257980.00 88932.00 13412.00  12427
## MEDL00356  77216  39512.75    22893  21026  38348  20400   23115  54844.55 232230.00   32944 243850.0 119550.00 57710.00 35455.33  19786
## MEDL00369  56780  80875.26    48794 211510  32971  85272   44528   9265.80  42110.64  422050  15741.0  44267.07 46179.76 34615.38  17800
## MEDL00375 537650 121470.00   431950  90307  81435 205780  140470 135240.00 220610.00  290030 233800.0  53591.00 70224.00 21049.00 202080
## MEDL00392  32157 993620.00    31457  23587  19511 312710  107940  39497.38  39986.00  113640  19644.0  46579.00 44633.00 19111.00 156870
## MEDL00401  13955  15942.00    13843  15067  16507  13660   12523  12381.00  11528.00   10740   8921.5   7834.50 11835.00 13978.00  11395
##              ZCY_V1   ZCY_V0     ZHP_V0     ZHP_V1     ZJW_V1    ZJW_V0  ZTX1_V1  ZTX1_V0   ZTX2_V0    ZTX2_V1   mix01   mix02   mix03
## MEDL00066  121830.0 859300.0 1908600.00 2960800.00  803940.00 498920.00  22789.0  12131.0  74144.00  378000.00 1042600 1159400 1167400
## MEDL00356  260550.0  89027.0   94463.00   64879.00   20510.52  35318.84  84165.0  13540.0  33362.67   58783.00  314180  288500  279550
## MEDL00369   39220.0  52984.0   37201.74   50596.39   13670.00  29996.22   4734.5   8474.4  65541.00   35292.96   33037   37579   29471
## MEDL00375 1038100.0 782630.0  738020.00  609040.00 1744600.00 993080.00 794930.0 111690.0  50044.00 2155000.00  436160  428120  453000
## MEDL00392  101906.1 105220.8   56996.37   84620.86   79583.00  72859.00  80132.8  48640.0 203680.00  236080.00  183070  196370  206210
## MEDL00401   13510.0  15545.0   16938.00   17509.00   11082.00  14200.00  27791.0  12716.0  17282.00   13456.00   11124   11093   11136
##            mix04  mix05   mix06   mix07
## MEDL00066 981970 912960 1092400 1106700
## MEDL00356 265550 276420  299290  291510
## MEDL00369  36386  35111   35431   37024
## MEDL00375 490780 373010  499630  474080
## MEDL00392 195750 191210  169370  134860
## MEDL00401  11020  11076   11432   12582</code></pre>
</div>
<div id="data-filtering-2" class="section level3 hasAnchor" number="6.3.3">
<h3><span class="header-section-number">6.3.3</span> Data Filtering<a href="example.html#data-filtering-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb244"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb244-1"><a href="example.html#cb244-1" tabindex="-1"></a>FilterFeature <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb244-2"><a href="example.html#cb244-2" tabindex="-1"></a>                       object,</span>
<span id="cb244-3"><a href="example.html#cb244-3" tabindex="-1"></a>                       qc_label,</span>
<span id="cb244-4"><a href="example.html#cb244-4" tabindex="-1"></a>                       <span class="at">method =</span> <span class="fu">c</span>(<span class="st">&quot;none&quot;</span>, <span class="st">&quot;iqr&quot;</span>, <span class="st">&quot;rsd&quot;</span>, </span>
<span id="cb244-5"><a href="example.html#cb244-5" tabindex="-1"></a>                                  <span class="st">&quot;nrsd&quot;</span>, <span class="st">&quot;mean&quot;</span>, <span class="st">&quot;sd&quot;</span>,</span>
<span id="cb244-6"><a href="example.html#cb244-6" tabindex="-1"></a>                                  <span class="st">&quot;mad&quot;</span>, <span class="st">&quot;median&quot;</span>),</span>
<span id="cb244-7"><a href="example.html#cb244-7" tabindex="-1"></a>                       <span class="at">rsd_cutoff =</span> <span class="dv">25</span>) {</span>
<span id="cb244-8"><a href="example.html#cb244-8" tabindex="-1"></a>  </span>
<span id="cb244-9"><a href="example.html#cb244-9" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) </span>
<span id="cb244-10"><a href="example.html#cb244-10" tabindex="-1"></a>  metadata_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object) </span>
<span id="cb244-11"><a href="example.html#cb244-11" tabindex="-1"></a>  </span>
<span id="cb244-12"><a href="example.html#cb244-12" tabindex="-1"></a>  <span class="co"># QC samples</span></span>
<span id="cb244-13"><a href="example.html#cb244-13" tabindex="-1"></a>  qc_samples <span class="ot">&lt;-</span> metadata_tab <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb244-14"><a href="example.html#cb244-14" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(group <span class="sc">==</span> qc_label)</span>
<span id="cb244-15"><a href="example.html#cb244-15" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">dim</span>(qc_samples)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb244-16"><a href="example.html#cb244-16" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;No qc samples have been chosen, please check your input&quot;</span>)</span>
<span id="cb244-17"><a href="example.html#cb244-17" tabindex="-1"></a>  }</span>
<span id="cb244-18"><a href="example.html#cb244-18" tabindex="-1"></a>  </span>
<span id="cb244-19"><a href="example.html#cb244-19" tabindex="-1"></a>  <span class="co"># QC samples&#39; feature table</span></span>
<span id="cb244-20"><a href="example.html#cb244-20" tabindex="-1"></a>  qc_feature <span class="ot">&lt;-</span> features_tab[, <span class="fu">colnames</span>(features_tab) <span class="sc">%in%</span> <span class="fu">rownames</span>(qc_samples)] <span class="sc">%&gt;%</span></span>
<span id="cb244-21"><a href="example.html#cb244-21" tabindex="-1"></a>    <span class="fu">t</span>()</span>
<span id="cb244-22"><a href="example.html#cb244-22" tabindex="-1"></a>  </span>
<span id="cb244-23"><a href="example.html#cb244-23" tabindex="-1"></a>  <span class="co"># filter features by QC RSD</span></span>
<span id="cb244-24"><a href="example.html#cb244-24" tabindex="-1"></a>  rsd <span class="ot">&lt;-</span> rsd_cutoff <span class="sc">/</span> <span class="dv">100</span></span>
<span id="cb244-25"><a href="example.html#cb244-25" tabindex="-1"></a>  sds <span class="ot">&lt;-</span> <span class="fu">apply</span>(qc_feature, <span class="dv">2</span>, sd, <span class="at">na.rm =</span> T)</span>
<span id="cb244-26"><a href="example.html#cb244-26" tabindex="-1"></a>  mns <span class="ot">&lt;-</span> <span class="fu">apply</span>(qc_feature, <span class="dv">2</span>, mean, <span class="at">na.rm =</span> T)</span>
<span id="cb244-27"><a href="example.html#cb244-27" tabindex="-1"></a>  rsd_vals <span class="ot">&lt;-</span> <span class="fu">abs</span>(sds<span class="sc">/</span>mns) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb244-28"><a href="example.html#cb244-28" tabindex="-1"></a>  gd_inx <span class="ot">&lt;-</span> rsd_vals <span class="sc">&lt;</span> rsd</span>
<span id="cb244-29"><a href="example.html#cb244-29" tabindex="-1"></a>  int_mat <span class="ot">&lt;-</span> features_tab[gd_inx, ]</span>
<span id="cb244-30"><a href="example.html#cb244-30" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">&quot;Removed &quot;</span>, (<span class="fu">dim</span>(qc_feature)[<span class="dv">2</span>] <span class="sc">-</span> <span class="fu">dim</span>(int_mat)[<span class="dv">1</span>]), </span>
<span id="cb244-31"><a href="example.html#cb244-31" tabindex="-1"></a>  <span class="st">&quot; features based on QC RSD values. QC samples are excluded from downstream functional analysis.&quot;</span>)</span>
<span id="cb244-32"><a href="example.html#cb244-32" tabindex="-1"></a>  </span>
<span id="cb244-33"><a href="example.html#cb244-33" tabindex="-1"></a>  <span class="co"># whether to filter features by percentage according to the number</span></span>
<span id="cb244-34"><a href="example.html#cb244-34" tabindex="-1"></a>  PerformFeatureFilter <span class="ot">&lt;-</span> <span class="cf">function</span>(datMatrix, </span>
<span id="cb244-35"><a href="example.html#cb244-35" tabindex="-1"></a>                                   <span class="at">qc_method =</span> method,</span>
<span id="cb244-36"><a href="example.html#cb244-36" tabindex="-1"></a>                                   <span class="at">remain_num =</span> <span class="cn">NULL</span>) {</span>
<span id="cb244-37"><a href="example.html#cb244-37" tabindex="-1"></a>    </span>
<span id="cb244-38"><a href="example.html#cb244-38" tabindex="-1"></a>    dat <span class="ot">&lt;-</span> datMatrix</span>
<span id="cb244-39"><a href="example.html#cb244-39" tabindex="-1"></a>    feat_num <span class="ot">&lt;-</span> <span class="fu">ncol</span>(dat)</span>
<span id="cb244-40"><a href="example.html#cb244-40" tabindex="-1"></a>    feat_nms <span class="ot">&lt;-</span> <span class="fu">colnames</span>(dat)</span>
<span id="cb244-41"><a href="example.html#cb244-41" tabindex="-1"></a>    nm <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb244-42"><a href="example.html#cb244-42" tabindex="-1"></a>    <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;none&quot;</span> <span class="sc">&amp;&amp;</span> feat_num <span class="sc">&lt;</span> <span class="dv">5000</span>) { <span class="co"># only allow for less than 4000</span></span>
<span id="cb244-43"><a href="example.html#cb244-43" tabindex="-1"></a>      remain <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">TRUE</span>, feat_num)</span>
<span id="cb244-44"><a href="example.html#cb244-44" tabindex="-1"></a>      nm <span class="ot">&lt;-</span> <span class="st">&quot;No filtering was applied&quot;</span></span>
<span id="cb244-45"><a href="example.html#cb244-45" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb244-46"><a href="example.html#cb244-46" tabindex="-1"></a>      <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;rsd&quot;</span>){</span>
<span id="cb244-47"><a href="example.html#cb244-47" tabindex="-1"></a>        sds <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, sd, <span class="at">na.rm =</span> T)</span>
<span id="cb244-48"><a href="example.html#cb244-48" tabindex="-1"></a>        mns <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, mean, <span class="at">na.rm =</span> T)</span>
<span id="cb244-49"><a href="example.html#cb244-49" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">abs</span>(sds<span class="sc">/</span>mns)</span>
<span id="cb244-50"><a href="example.html#cb244-50" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;Relative standard deviation&quot;</span></span>
<span id="cb244-51"><a href="example.html#cb244-51" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;nrsd&quot;</span> ) {</span>
<span id="cb244-52"><a href="example.html#cb244-52" tabindex="-1"></a>        mads <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, mad, <span class="at">na.rm =</span> T)</span>
<span id="cb244-53"><a href="example.html#cb244-53" tabindex="-1"></a>        meds <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, median, <span class="at">na.rm =</span> T)</span>
<span id="cb244-54"><a href="example.html#cb244-54" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">abs</span>(mads<span class="sc">/</span>meds)</span>
<span id="cb244-55"><a href="example.html#cb244-55" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;Non-paramatric relative standard deviation&quot;</span></span>
<span id="cb244-56"><a href="example.html#cb244-56" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;mean&quot;</span>) {</span>
<span id="cb244-57"><a href="example.html#cb244-57" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, mean, <span class="at">na.rm =</span> T)</span>
<span id="cb244-58"><a href="example.html#cb244-58" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;mean&quot;</span></span>
<span id="cb244-59"><a href="example.html#cb244-59" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;sd&quot;</span>) {</span>
<span id="cb244-60"><a href="example.html#cb244-60" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, sd, <span class="at">na.rm =</span> T)</span>
<span id="cb244-61"><a href="example.html#cb244-61" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;standard deviation&quot;</span></span>
<span id="cb244-62"><a href="example.html#cb244-62" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;mad&quot;</span>) {</span>
<span id="cb244-63"><a href="example.html#cb244-63" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, mad, <span class="at">na.rm =</span> T)</span>
<span id="cb244-64"><a href="example.html#cb244-64" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;Median absolute deviation&quot;</span></span>
<span id="cb244-65"><a href="example.html#cb244-65" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;median&quot;</span>) {</span>
<span id="cb244-66"><a href="example.html#cb244-66" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, median, <span class="at">na.rm =</span> T)</span>
<span id="cb244-67"><a href="example.html#cb244-67" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;median&quot;</span></span>
<span id="cb244-68"><a href="example.html#cb244-68" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;iqr&quot;</span>) { <span class="co"># iqr</span></span>
<span id="cb244-69"><a href="example.html#cb244-69" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, IQR, <span class="at">na.rm =</span> T)</span>
<span id="cb244-70"><a href="example.html#cb244-70" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;Interquantile Range&quot;</span></span>
<span id="cb244-71"><a href="example.html#cb244-71" tabindex="-1"></a>      }</span>
<span id="cb244-72"><a href="example.html#cb244-72" tabindex="-1"></a>      </span>
<span id="cb244-73"><a href="example.html#cb244-73" tabindex="-1"></a>      <span class="co"># get the rank of the filtered variables</span></span>
<span id="cb244-74"><a href="example.html#cb244-74" tabindex="-1"></a>      rk <span class="ot">&lt;-</span> <span class="fu">rank</span>(<span class="sc">-</span>filter_val, <span class="at">ties.method =</span> <span class="st">&quot;random&quot;</span>)</span>
<span id="cb244-75"><a href="example.html#cb244-75" tabindex="-1"></a>      </span>
<span id="cb244-76"><a href="example.html#cb244-76" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">is.null</span>(remain_num)) { <span class="co"># apply empirical filtering based on data size</span></span>
<span id="cb244-77"><a href="example.html#cb244-77" tabindex="-1"></a>          <span class="cf">if</span> (feat_num <span class="sc">&lt;</span> <span class="dv">250</span>) { <span class="co"># reduce 5%</span></span>
<span id="cb244-78"><a href="example.html#cb244-78" tabindex="-1"></a>            remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> feat_num <span class="sc">*</span> <span class="fl">0.95</span></span>
<span id="cb244-79"><a href="example.html#cb244-79" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Further feature filtering based on &quot;</span>, nm)</span>
<span id="cb244-80"><a href="example.html#cb244-80" tabindex="-1"></a>          } <span class="cf">else</span> <span class="cf">if</span> (feat_num <span class="sc">&lt;</span> <span class="dv">500</span>) { <span class="co"># reduce 10%</span></span>
<span id="cb244-81"><a href="example.html#cb244-81" tabindex="-1"></a>            remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> feat_num <span class="sc">*</span> <span class="fl">0.9</span></span>
<span id="cb244-82"><a href="example.html#cb244-82" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Further feature filtering based on &quot;</span>, nm)</span>
<span id="cb244-83"><a href="example.html#cb244-83" tabindex="-1"></a>          } <span class="cf">else</span> <span class="cf">if</span> (feat_num <span class="sc">&lt;</span> <span class="dv">1000</span>) { <span class="co"># reduce 25%</span></span>
<span id="cb244-84"><a href="example.html#cb244-84" tabindex="-1"></a>            remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> feat_num <span class="sc">*</span> <span class="fl">0.75</span></span>
<span id="cb244-85"><a href="example.html#cb244-85" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Further feature filtering based on &quot;</span>, nm)</span>
<span id="cb244-86"><a href="example.html#cb244-86" tabindex="-1"></a>          } <span class="cf">else</span> { <span class="co"># reduce 40%, if still over 5000, then only use top 5000</span></span>
<span id="cb244-87"><a href="example.html#cb244-87" tabindex="-1"></a>            remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> feat_num <span class="sc">*</span> <span class="fl">0.6</span></span>
<span id="cb244-88"><a href="example.html#cb244-88" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Further feature filtering based on &quot;</span>, nm)</span>
<span id="cb244-89"><a href="example.html#cb244-89" tabindex="-1"></a>          }</span>
<span id="cb244-90"><a href="example.html#cb244-90" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb244-91"><a href="example.html#cb244-91" tabindex="-1"></a>        remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> remain_num</span>
<span id="cb244-92"><a href="example.html#cb244-92" tabindex="-1"></a>      }</span>
<span id="cb244-93"><a href="example.html#cb244-93" tabindex="-1"></a>    }</span>
<span id="cb244-94"><a href="example.html#cb244-94" tabindex="-1"></a>    </span>
<span id="cb244-95"><a href="example.html#cb244-95" tabindex="-1"></a>    res <span class="ot">&lt;-</span> datMatrix[, remain]</span>
<span id="cb244-96"><a href="example.html#cb244-96" tabindex="-1"></a>    </span>
<span id="cb244-97"><a href="example.html#cb244-97" tabindex="-1"></a>    <span class="fu">return</span>(res)</span>
<span id="cb244-98"><a href="example.html#cb244-98" tabindex="-1"></a>  }  </span>
<span id="cb244-99"><a href="example.html#cb244-99" tabindex="-1"></a>  </span>
<span id="cb244-100"><a href="example.html#cb244-100" tabindex="-1"></a>  feature_res <span class="ot">&lt;-</span> <span class="fu">PerformFeatureFilter</span>(<span class="fu">t</span>(int_mat))</span>
<span id="cb244-101"><a href="example.html#cb244-101" tabindex="-1"></a>  </span>
<span id="cb244-102"><a href="example.html#cb244-102" tabindex="-1"></a>  <span class="co"># remove QC samples </span></span>
<span id="cb244-103"><a href="example.html#cb244-103" tabindex="-1"></a>  feature_final <span class="ot">&lt;-</span> feature_res[<span class="sc">!</span><span class="fu">rownames</span>(feature_res) <span class="sc">%in%</span> <span class="fu">rownames</span>(qc_samples), ]</span>
<span id="cb244-104"><a href="example.html#cb244-104" tabindex="-1"></a>  </span>
<span id="cb244-105"><a href="example.html#cb244-105" tabindex="-1"></a>  <span class="co"># save int_mat into se object </span></span>
<span id="cb244-106"><a href="example.html#cb244-106" tabindex="-1"></a>  target <span class="ot">&lt;-</span> metadata_tab <span class="sc">%&gt;%</span> </span>
<span id="cb244-107"><a href="example.html#cb244-107" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb244-108"><a href="example.html#cb244-108" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb244-109"><a href="example.html#cb244-109" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(SampleID <span class="sc">%in%</span> <span class="fu">rownames</span>(feature_final))</span>
<span id="cb244-110"><a href="example.html#cb244-110" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb244-111"><a href="example.html#cb244-111" tabindex="-1"></a>                                  <span class="at">features =</span> feature_final)</span>
<span id="cb244-112"><a href="example.html#cb244-112" tabindex="-1"></a>  </span>
<span id="cb244-113"><a href="example.html#cb244-113" tabindex="-1"></a>  <span class="fu">return</span>(res) </span>
<span id="cb244-114"><a href="example.html#cb244-114" tabindex="-1"></a>}</span>
<span id="cb244-115"><a href="example.html#cb244-115" tabindex="-1"></a></span>
<span id="cb244-116"><a href="example.html#cb244-116" tabindex="-1"></a>se_filter <span class="ot">&lt;-</span> <span class="fu">FilterFeature</span>(<span class="at">object =</span> se_impute,</span>
<span id="cb244-117"><a href="example.html#cb244-117" tabindex="-1"></a>                           <span class="at">qc_label =</span> <span class="st">&quot;QC&quot;</span>,</span>
<span id="cb244-118"><a href="example.html#cb244-118" tabindex="-1"></a>                           <span class="at">method =</span> <span class="st">&quot;iqr&quot;</span>)</span>
<span id="cb244-119"><a href="example.html#cb244-119" tabindex="-1"></a>se_filter</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 705 54 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(705): MEDL00066 MEDL00375 ... MW0168376 MW0169569
## rowData names(0):
## colnames(54): CJY_V0 CJY_V1 ... ZTX2_V0 ZTX2_V1
## colData names(5): group SampleName FMT_status gender age</code></pre>
</div>
<div id="data-normalization-2" class="section level3 hasAnchor" number="6.3.4">
<h3><span class="header-section-number">6.3.4</span> Data Normalization<a href="example.html#data-normalization-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="normalization-by-normalizedata-function-2" class="section level4 hasAnchor" number="6.3.4.1">
<h4><span class="header-section-number">6.3.4.1</span> Normalization by <strong>NormalizeData</strong> function<a href="example.html#normalization-by-normalizedata-function-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb246"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb246-1"><a href="example.html#cb246-1" tabindex="-1"></a>NormalizeData <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb246-2"><a href="example.html#cb246-2" tabindex="-1"></a>                      object,</span>
<span id="cb246-3"><a href="example.html#cb246-3" tabindex="-1"></a>                      <span class="at">rowNorm =</span> <span class="fu">c</span>(<span class="st">&quot;Quantile&quot;</span>, <span class="st">&quot;GroupPQN&quot;</span>, <span class="st">&quot;SamplePQN&quot;</span>, </span>
<span id="cb246-4"><a href="example.html#cb246-4" tabindex="-1"></a>                                  <span class="st">&quot;CompNorm&quot;</span>, <span class="st">&quot;SumNorm&quot;</span>, <span class="st">&quot;MedianNorm&quot;</span>,</span>
<span id="cb246-5"><a href="example.html#cb246-5" tabindex="-1"></a>                                  <span class="st">&quot;SpecNorm&quot;</span>, <span class="st">&quot;None&quot;</span>),</span>
<span id="cb246-6"><a href="example.html#cb246-6" tabindex="-1"></a>                      <span class="at">transNorm =</span> <span class="fu">c</span>(<span class="st">&quot;LogNorm&quot;</span>, <span class="st">&quot;SrNorm&quot;</span>, <span class="st">&quot;CrNorm&quot;</span>, <span class="st">&quot;None&quot;</span>),</span>
<span id="cb246-7"><a href="example.html#cb246-7" tabindex="-1"></a>                      <span class="at">scaleNorm =</span> <span class="fu">c</span>(<span class="st">&quot;MeanCenter&quot;</span>, <span class="st">&quot;AutoNorm&quot;</span>, <span class="st">&quot;ParetoNorm&quot;</span>, </span>
<span id="cb246-8"><a href="example.html#cb246-8" tabindex="-1"></a>                                    <span class="st">&quot;RangeNorm&quot;</span>, <span class="st">&quot;None&quot;</span>),</span>
<span id="cb246-9"><a href="example.html#cb246-9" tabindex="-1"></a>                      <span class="at">ref =</span> <span class="cn">NULL</span>,</span>
<span id="cb246-10"><a href="example.html#cb246-10" tabindex="-1"></a>                      <span class="at">SpeWeight =</span> <span class="dv">1</span>) {</span>
<span id="cb246-11"><a href="example.html#cb246-11" tabindex="-1"></a>  </span>
<span id="cb246-12"><a href="example.html#cb246-12" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) </span>
<span id="cb246-13"><a href="example.html#cb246-13" tabindex="-1"></a>  metadata_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object)   </span>
<span id="cb246-14"><a href="example.html#cb246-14" tabindex="-1"></a>  </span>
<span id="cb246-15"><a href="example.html#cb246-15" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">t</span>(features_tab)</span>
<span id="cb246-16"><a href="example.html#cb246-16" tabindex="-1"></a>  colNames <span class="ot">&lt;-</span> <span class="fu">colnames</span>(data)</span>
<span id="cb246-17"><a href="example.html#cb246-17" tabindex="-1"></a>  rowNames <span class="ot">&lt;-</span> <span class="fu">rownames</span>(data)</span>
<span id="cb246-18"><a href="example.html#cb246-18" tabindex="-1"></a>  </span>
<span id="cb246-19"><a href="example.html#cb246-19" tabindex="-1"></a>  <span class="do">#############################################</span></span>
<span id="cb246-20"><a href="example.html#cb246-20" tabindex="-1"></a>  <span class="co"># Sample normalization</span></span>
<span id="cb246-21"><a href="example.html#cb246-21" tabindex="-1"></a>  <span class="co"># perform quantile normalization on the raw data (can be log transformed later by user)</span></span>
<span id="cb246-22"><a href="example.html#cb246-22" tabindex="-1"></a>  QuantileNormalize <span class="ot">&lt;-</span> <span class="cf">function</span>(data) {</span>
<span id="cb246-23"><a href="example.html#cb246-23" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">t</span>(preprocessCore<span class="sc">::</span><span class="fu">normalize.quantiles</span>(<span class="fu">t</span>(data), <span class="at">copy=</span><span class="cn">FALSE</span>)));</span>
<span id="cb246-24"><a href="example.html#cb246-24" tabindex="-1"></a>  }</span>
<span id="cb246-25"><a href="example.html#cb246-25" tabindex="-1"></a>  <span class="co"># normalize by a reference sample (probability quotient normalization)</span></span>
<span id="cb246-26"><a href="example.html#cb246-26" tabindex="-1"></a>  <span class="co"># ref should be the name of the reference sample</span></span>
<span id="cb246-27"><a href="example.html#cb246-27" tabindex="-1"></a>  ProbNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x, ref_smpl) {</span>
<span id="cb246-28"><a href="example.html#cb246-28" tabindex="-1"></a>    <span class="fu">return</span>(x<span class="sc">/</span><span class="fu">median</span>(<span class="fu">as.numeric</span>(x<span class="sc">/</span>ref_smpl), <span class="at">na.rm =</span> T))</span>
<span id="cb246-29"><a href="example.html#cb246-29" tabindex="-1"></a>  }</span>
<span id="cb246-30"><a href="example.html#cb246-30" tabindex="-1"></a>  </span>
<span id="cb246-31"><a href="example.html#cb246-31" tabindex="-1"></a>  <span class="co"># normalize by a reference reference (i.e. creatinine)</span></span>
<span id="cb246-32"><a href="example.html#cb246-32" tabindex="-1"></a>  <span class="co"># ref should be the name of the cmpd</span></span>
<span id="cb246-33"><a href="example.html#cb246-33" tabindex="-1"></a>  CompNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x, ref) {</span>
<span id="cb246-34"><a href="example.html#cb246-34" tabindex="-1"></a>    <span class="fu">return</span>(<span class="dv">1000</span><span class="sc">*</span>x<span class="sc">/</span>x[ref])</span>
<span id="cb246-35"><a href="example.html#cb246-35" tabindex="-1"></a>  }</span>
<span id="cb246-36"><a href="example.html#cb246-36" tabindex="-1"></a>  </span>
<span id="cb246-37"><a href="example.html#cb246-37" tabindex="-1"></a>  SumNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb246-38"><a href="example.html#cb246-38" tabindex="-1"></a>    <span class="fu">return</span>(<span class="dv">1000</span><span class="sc">*</span>x<span class="sc">/</span><span class="fu">sum</span>(x, <span class="at">na.rm =</span> T))</span>
<span id="cb246-39"><a href="example.html#cb246-39" tabindex="-1"></a>  }</span>
<span id="cb246-40"><a href="example.html#cb246-40" tabindex="-1"></a>  </span>
<span id="cb246-41"><a href="example.html#cb246-41" tabindex="-1"></a>  <span class="co"># normalize by median</span></span>
<span id="cb246-42"><a href="example.html#cb246-42" tabindex="-1"></a>  MedianNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb246-43"><a href="example.html#cb246-43" tabindex="-1"></a>    <span class="fu">return</span>(x<span class="sc">/</span><span class="fu">median</span>(x, <span class="at">na.rm =</span> T))</span>
<span id="cb246-44"><a href="example.html#cb246-44" tabindex="-1"></a>  }  </span>
<span id="cb246-45"><a href="example.html#cb246-45" tabindex="-1"></a>  </span>
<span id="cb246-46"><a href="example.html#cb246-46" tabindex="-1"></a>  <span class="co"># row-wise normalization</span></span>
<span id="cb246-47"><a href="example.html#cb246-47" tabindex="-1"></a>  <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;Quantile&quot;</span>) {</span>
<span id="cb246-48"><a href="example.html#cb246-48" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">QuantileNormalize</span>(data)</span>
<span id="cb246-49"><a href="example.html#cb246-49" tabindex="-1"></a>    <span class="co"># this can introduce constant variables if a variable is </span></span>
<span id="cb246-50"><a href="example.html#cb246-50" tabindex="-1"></a>    <span class="co"># at the same rank across all samples (replaced by its average across all)</span></span>
<span id="cb246-51"><a href="example.html#cb246-51" tabindex="-1"></a>    varCol <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, var, <span class="at">na.rm =</span> T)</span>
<span id="cb246-52"><a href="example.html#cb246-52" tabindex="-1"></a>    constCol <span class="ot">&lt;-</span> (varCol <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> <span class="fu">is.na</span>(varCol))</span>
<span id="cb246-53"><a href="example.html#cb246-53" tabindex="-1"></a>    constNum <span class="ot">&lt;-</span> <span class="fu">sum</span>(constCol, <span class="at">na.rm =</span> T)</span>
<span id="cb246-54"><a href="example.html#cb246-54" tabindex="-1"></a>    <span class="cf">if</span> (constNum <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb246-55"><a href="example.html#cb246-55" tabindex="-1"></a>      <span class="fu">message</span>(<span class="fu">paste</span>(<span class="st">&quot;After quantile normalization&quot;</span>, constNum, </span>
<span id="cb246-56"><a href="example.html#cb246-56" tabindex="-1"></a>                    <span class="st">&quot;features with a constant value were found and deleted.&quot;</span>))</span>
<span id="cb246-57"><a href="example.html#cb246-57" tabindex="-1"></a>      data <span class="ot">&lt;-</span> data[, <span class="sc">!</span>constCol, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb246-58"><a href="example.html#cb246-58" tabindex="-1"></a>      colNames <span class="ot">&lt;-</span> <span class="fu">colnames</span>(data)</span>
<span id="cb246-59"><a href="example.html#cb246-59" tabindex="-1"></a>      rowNames <span class="ot">&lt;-</span> <span class="fu">rownames</span>(data)</span>
<span id="cb246-60"><a href="example.html#cb246-60" tabindex="-1"></a>    }</span>
<span id="cb246-61"><a href="example.html#cb246-61" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Quantile Normalization&quot;</span></span>
<span id="cb246-62"><a href="example.html#cb246-62" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;GroupPQN&quot;</span>) {</span>
<span id="cb246-63"><a href="example.html#cb246-63" tabindex="-1"></a>    grp_inx <span class="ot">&lt;-</span> metadata_tab<span class="sc">$</span>group <span class="sc">==</span> ref</span>
<span id="cb246-64"><a href="example.html#cb246-64" tabindex="-1"></a>    ref.smpl <span class="ot">&lt;-</span> <span class="fu">apply</span>(data[grp_inx, , <span class="at">drop =</span> <span class="cn">FALSE</span>], <span class="dv">2</span>, mean)</span>
<span id="cb246-65"><a href="example.html#cb246-65" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(data, <span class="dv">1</span>, ProbNorm, ref.smpl))</span>
<span id="cb246-66"><a href="example.html#cb246-66" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Probabilistic Quotient Normalization by a reference group&quot;</span></span>
<span id="cb246-67"><a href="example.html#cb246-67" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;SamplePQN&quot;</span>) {</span>
<span id="cb246-68"><a href="example.html#cb246-68" tabindex="-1"></a>    ref.smpl <span class="ot">&lt;-</span> data[ref, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb246-69"><a href="example.html#cb246-69" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(data, <span class="dv">1</span>, ProbNorm, ref.smpl))</span>
<span id="cb246-70"><a href="example.html#cb246-70" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Probabilistic Quotient Normalization by a reference sample&quot;</span></span>
<span id="cb246-71"><a href="example.html#cb246-71" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;CompNorm&quot;</span>) {</span>
<span id="cb246-72"><a href="example.html#cb246-72" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(<span class="fu">t</span>(data), <span class="dv">1</span>, CompNorm, ref))</span>
<span id="cb246-73"><a href="example.html#cb246-73" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Normalization by a reference feature&quot;</span>;</span>
<span id="cb246-74"><a href="example.html#cb246-74" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;SumNorm&quot;</span>) {</span>
<span id="cb246-75"><a href="example.html#cb246-75" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(data, <span class="dv">1</span>, SumNorm))</span>
<span id="cb246-76"><a href="example.html#cb246-76" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Normalization to constant sum&quot;</span></span>
<span id="cb246-77"><a href="example.html#cb246-77" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;MedianNorm&quot;</span>) {</span>
<span id="cb246-78"><a href="example.html#cb246-78" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(data, <span class="dv">1</span>, MedianNorm))</span>
<span id="cb246-79"><a href="example.html#cb246-79" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Normalization to sample median&quot;</span></span>
<span id="cb246-80"><a href="example.html#cb246-80" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span>(rowNorm <span class="sc">==</span> <span class="st">&quot;SpecNorm&quot;</span>) {</span>
<span id="cb246-81"><a href="example.html#cb246-81" tabindex="-1"></a>    norm.vec <span class="ot">&lt;-</span> <span class="fu">rep</span>(SpeWeight, <span class="fu">nrow</span>(data)) <span class="co"># default all same weight vec to prevent error</span></span>
<span id="cb246-82"><a href="example.html#cb246-82" tabindex="-1"></a>    data <span class="ot">&lt;-</span> data <span class="sc">/</span> norm.vec</span>
<span id="cb246-83"><a href="example.html#cb246-83" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">&quot;No sample specific information were given, all set to 1.0&quot;</span>)</span>
<span id="cb246-84"><a href="example.html#cb246-84" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Normalization by sample-specific factor&quot;</span></span>
<span id="cb246-85"><a href="example.html#cb246-85" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb246-86"><a href="example.html#cb246-86" tabindex="-1"></a>    <span class="co"># nothing to do</span></span>
<span id="cb246-87"><a href="example.html#cb246-87" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;N/A&quot;</span></span>
<span id="cb246-88"><a href="example.html#cb246-88" tabindex="-1"></a>  }</span>
<span id="cb246-89"><a href="example.html#cb246-89" tabindex="-1"></a>  <span class="do">################################################ </span></span>
<span id="cb246-90"><a href="example.html#cb246-90" tabindex="-1"></a>  </span>
<span id="cb246-91"><a href="example.html#cb246-91" tabindex="-1"></a>  <span class="co"># use apply will lose dimension info (i.e. row names and colnames)</span></span>
<span id="cb246-92"><a href="example.html#cb246-92" tabindex="-1"></a>  <span class="fu">rownames</span>(data) <span class="ot">&lt;-</span> rowNames</span>
<span id="cb246-93"><a href="example.html#cb246-93" tabindex="-1"></a>  <span class="fu">colnames</span>(data) <span class="ot">&lt;-</span> colNames</span>
<span id="cb246-94"><a href="example.html#cb246-94" tabindex="-1"></a>  </span>
<span id="cb246-95"><a href="example.html#cb246-95" tabindex="-1"></a>  <span class="co"># if the reference by feature, the feature column should be removed, since it is all 1</span></span>
<span id="cb246-96"><a href="example.html#cb246-96" tabindex="-1"></a>  <span class="cf">if</span>(rowNorm <span class="sc">==</span> <span class="st">&quot;CompNorm&quot;</span> <span class="sc">&amp;&amp;</span> <span class="sc">!</span><span class="fu">is.null</span>(ref)){</span>
<span id="cb246-97"><a href="example.html#cb246-97" tabindex="-1"></a>    inx <span class="ot">&lt;-</span> <span class="fu">match</span>(ref, <span class="fu">colnames</span>(data))</span>
<span id="cb246-98"><a href="example.html#cb246-98" tabindex="-1"></a>    data <span class="ot">&lt;-</span> data[, <span class="sc">-</span>inx, drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb246-99"><a href="example.html#cb246-99" tabindex="-1"></a>    colNames <span class="ot">&lt;-</span> colNames[<span class="sc">-</span>inx]</span>
<span id="cb246-100"><a href="example.html#cb246-100" tabindex="-1"></a>  }</span>
<span id="cb246-101"><a href="example.html#cb246-101" tabindex="-1"></a>  </span>
<span id="cb246-102"><a href="example.html#cb246-102" tabindex="-1"></a>  <span class="do">#############################################</span></span>
<span id="cb246-103"><a href="example.html#cb246-103" tabindex="-1"></a>  <span class="co">#  Data transformation</span></span>
<span id="cb246-104"><a href="example.html#cb246-104" tabindex="-1"></a>  <span class="co"># generalize log, tolerant to 0 and negative values</span></span>
<span id="cb246-105"><a href="example.html#cb246-105" tabindex="-1"></a>  LogNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x, min.val) {</span>
<span id="cb246-106"><a href="example.html#cb246-106" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">log10</span>((x <span class="sc">+</span> <span class="fu">sqrt</span>(x<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> min.val<span class="sc">^</span><span class="dv">2</span>))<span class="sc">/</span><span class="dv">2</span>))</span>
<span id="cb246-107"><a href="example.html#cb246-107" tabindex="-1"></a>  }</span>
<span id="cb246-108"><a href="example.html#cb246-108" tabindex="-1"></a>  </span>
<span id="cb246-109"><a href="example.html#cb246-109" tabindex="-1"></a>  <span class="co"># square root, tolerant to negative values</span></span>
<span id="cb246-110"><a href="example.html#cb246-110" tabindex="-1"></a>  SquareRootNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x, min.val) {</span>
<span id="cb246-111"><a href="example.html#cb246-111" tabindex="-1"></a>    <span class="fu">return</span>(((x <span class="sc">+</span> <span class="fu">sqrt</span>(x<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> min.val<span class="sc">^</span><span class="dv">2</span>))<span class="sc">/</span><span class="dv">2</span>)<span class="sc">^</span>(<span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>))</span>
<span id="cb246-112"><a href="example.html#cb246-112" tabindex="-1"></a>  }</span>
<span id="cb246-113"><a href="example.html#cb246-113" tabindex="-1"></a>  </span>
<span id="cb246-114"><a href="example.html#cb246-114" tabindex="-1"></a>  <span class="cf">if</span> (transNorm <span class="sc">==</span> <span class="st">&quot;LogNorm&quot;</span>) {</span>
<span id="cb246-115"><a href="example.html#cb246-115" tabindex="-1"></a>    min.val <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">abs</span>(data[data <span class="sc">!=</span> <span class="dv">0</span>]))<span class="sc">/</span><span class="dv">10</span></span>
<span id="cb246-116"><a href="example.html#cb246-116" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, LogNorm, min.val)</span>
<span id="cb246-117"><a href="example.html#cb246-117" tabindex="-1"></a>    transnm <span class="ot">&lt;-</span> <span class="st">&quot;Log10 Normalization&quot;</span></span>
<span id="cb246-118"><a href="example.html#cb246-118" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (transNorm <span class="sc">==</span> <span class="st">&quot;SrNorm&quot;</span>) {</span>
<span id="cb246-119"><a href="example.html#cb246-119" tabindex="-1"></a>    min.val <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">abs</span>(data[data <span class="sc">!=</span> <span class="dv">0</span>]))<span class="sc">/</span><span class="dv">10</span></span>
<span id="cb246-120"><a href="example.html#cb246-120" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, SquareRootNorm, min.val)</span>
<span id="cb246-121"><a href="example.html#cb246-121" tabindex="-1"></a>    transnm <span class="ot">&lt;-</span> <span class="st">&quot;Square Root Transformation&quot;</span></span>
<span id="cb246-122"><a href="example.html#cb246-122" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (transNorm <span class="sc">==</span> <span class="st">&quot;CrNorm&quot;</span>) {</span>
<span id="cb246-123"><a href="example.html#cb246-123" tabindex="-1"></a>    norm.data <span class="ot">&lt;-</span> <span class="fu">abs</span>(data)<span class="sc">^</span>(<span class="dv">1</span><span class="sc">/</span><span class="dv">3</span>)</span>
<span id="cb246-124"><a href="example.html#cb246-124" tabindex="-1"></a>    norm.data[data <span class="sc">&lt;</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="sc">-</span>norm.data[data <span class="sc">&lt;</span> <span class="dv">0</span>]</span>
<span id="cb246-125"><a href="example.html#cb246-125" tabindex="-1"></a>    data <span class="ot">&lt;-</span> norm.data</span>
<span id="cb246-126"><a href="example.html#cb246-126" tabindex="-1"></a>    transnm <span class="ot">&lt;-</span> <span class="st">&quot;Cubic Root Transformation&quot;</span></span>
<span id="cb246-127"><a href="example.html#cb246-127" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb246-128"><a href="example.html#cb246-128" tabindex="-1"></a>    transnm <span class="ot">&lt;-</span> <span class="st">&quot;N/A&quot;</span></span>
<span id="cb246-129"><a href="example.html#cb246-129" tabindex="-1"></a>  }</span>
<span id="cb246-130"><a href="example.html#cb246-130" tabindex="-1"></a>  <span class="do">#############################################</span></span>
<span id="cb246-131"><a href="example.html#cb246-131" tabindex="-1"></a>  </span>
<span id="cb246-132"><a href="example.html#cb246-132" tabindex="-1"></a>  <span class="do">#############################################</span></span>
<span id="cb246-133"><a href="example.html#cb246-133" tabindex="-1"></a>  <span class="co"># Data scaling</span></span>
<span id="cb246-134"><a href="example.html#cb246-134" tabindex="-1"></a>  <span class="co"># normalize to zero mean and unit variance</span></span>
<span id="cb246-135"><a href="example.html#cb246-135" tabindex="-1"></a>  AutoNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb246-136"><a href="example.html#cb246-136" tabindex="-1"></a>    <span class="fu">return</span>((x <span class="sc">-</span> <span class="fu">mean</span>(x))<span class="sc">/</span><span class="fu">sd</span>(x, <span class="at">na.rm =</span> T))</span>
<span id="cb246-137"><a href="example.html#cb246-137" tabindex="-1"></a>  }</span>
<span id="cb246-138"><a href="example.html#cb246-138" tabindex="-1"></a>  </span>
<span id="cb246-139"><a href="example.html#cb246-139" tabindex="-1"></a>  <span class="co"># normalize to zero mean but variance/SE</span></span>
<span id="cb246-140"><a href="example.html#cb246-140" tabindex="-1"></a>  ParetoNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb246-141"><a href="example.html#cb246-141" tabindex="-1"></a>    <span class="fu">return</span>((x <span class="sc">-</span> <span class="fu">mean</span>(x))<span class="sc">/</span><span class="fu">sqrt</span>(<span class="fu">sd</span>(x, <span class="at">na.rm =</span> T)))</span>
<span id="cb246-142"><a href="example.html#cb246-142" tabindex="-1"></a>  }</span>
<span id="cb246-143"><a href="example.html#cb246-143" tabindex="-1"></a>  </span>
<span id="cb246-144"><a href="example.html#cb246-144" tabindex="-1"></a>  <span class="co"># normalize to zero mean but variance/SE</span></span>
<span id="cb246-145"><a href="example.html#cb246-145" tabindex="-1"></a>  MeanCenter <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb246-146"><a href="example.html#cb246-146" tabindex="-1"></a>    <span class="fu">return</span>(x <span class="sc">-</span> <span class="fu">mean</span>(x))</span>
<span id="cb246-147"><a href="example.html#cb246-147" tabindex="-1"></a>  }</span>
<span id="cb246-148"><a href="example.html#cb246-148" tabindex="-1"></a>  </span>
<span id="cb246-149"><a href="example.html#cb246-149" tabindex="-1"></a>  <span class="co"># normalize to zero mean but variance/SE</span></span>
<span id="cb246-150"><a href="example.html#cb246-150" tabindex="-1"></a>  RangeNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb246-151"><a href="example.html#cb246-151" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">max</span>(x) <span class="sc">==</span> <span class="fu">min</span>(x)) {</span>
<span id="cb246-152"><a href="example.html#cb246-152" tabindex="-1"></a>      <span class="fu">return</span>(x)</span>
<span id="cb246-153"><a href="example.html#cb246-153" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb246-154"><a href="example.html#cb246-154" tabindex="-1"></a>      <span class="fu">return</span>((x <span class="sc">-</span> <span class="fu">mean</span>(x))<span class="sc">/</span>(<span class="fu">max</span>(x) <span class="sc">-</span> <span class="fu">min</span>(x)))</span>
<span id="cb246-155"><a href="example.html#cb246-155" tabindex="-1"></a>    }</span>
<span id="cb246-156"><a href="example.html#cb246-156" tabindex="-1"></a>  }</span>
<span id="cb246-157"><a href="example.html#cb246-157" tabindex="-1"></a></span>
<span id="cb246-158"><a href="example.html#cb246-158" tabindex="-1"></a>  <span class="cf">if</span> (scaleNorm <span class="sc">==</span> <span class="st">&quot;MeanCenter&quot;</span>) {</span>
<span id="cb246-159"><a href="example.html#cb246-159" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, MeanCenter)</span>
<span id="cb246-160"><a href="example.html#cb246-160" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;Mean Centering&quot;</span></span>
<span id="cb246-161"><a href="example.html#cb246-161" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (scaleNorm <span class="sc">==</span> <span class="st">&quot;AutoNorm&quot;</span>) {</span>
<span id="cb246-162"><a href="example.html#cb246-162" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, AutoNorm)</span>
<span id="cb246-163"><a href="example.html#cb246-163" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;Autoscaling&quot;</span></span>
<span id="cb246-164"><a href="example.html#cb246-164" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (scaleNorm <span class="sc">==</span> <span class="st">&quot;ParetoNorm&quot;</span>) {</span>
<span id="cb246-165"><a href="example.html#cb246-165" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, ParetoNorm)</span>
<span id="cb246-166"><a href="example.html#cb246-166" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;Pareto Scaling&quot;</span></span>
<span id="cb246-167"><a href="example.html#cb246-167" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (scaleNorm <span class="sc">==</span> <span class="st">&quot;RangeNorm&quot;</span>) {</span>
<span id="cb246-168"><a href="example.html#cb246-168" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, RangeNorm)</span>
<span id="cb246-169"><a href="example.html#cb246-169" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;Range Scaling&quot;</span></span>
<span id="cb246-170"><a href="example.html#cb246-170" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb246-171"><a href="example.html#cb246-171" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;N/A&quot;</span></span>
<span id="cb246-172"><a href="example.html#cb246-172" tabindex="-1"></a>  }</span>
<span id="cb246-173"><a href="example.html#cb246-173" tabindex="-1"></a>  <span class="do">############################################# </span></span>
<span id="cb246-174"><a href="example.html#cb246-174" tabindex="-1"></a>  </span>
<span id="cb246-175"><a href="example.html#cb246-175" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">&quot;Row norm: &quot;</span>, rownm, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>,</span>
<span id="cb246-176"><a href="example.html#cb246-176" tabindex="-1"></a>          <span class="st">&quot;Data Transformation norm: &quot;</span>, transnm, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>,</span>
<span id="cb246-177"><a href="example.html#cb246-177" tabindex="-1"></a>          <span class="st">&quot;Data Scaling norm: &quot;</span>, scalenm, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)  </span>
<span id="cb246-178"><a href="example.html#cb246-178" tabindex="-1"></a>  </span>
<span id="cb246-179"><a href="example.html#cb246-179" tabindex="-1"></a>  <span class="co"># note after using &quot;apply&quot; function, all the attribute lost, need to add back</span></span>
<span id="cb246-180"><a href="example.html#cb246-180" tabindex="-1"></a>  <span class="fu">rownames</span>(data) <span class="ot">&lt;-</span> rowNames</span>
<span id="cb246-181"><a href="example.html#cb246-181" tabindex="-1"></a>  <span class="fu">colnames</span>(data) <span class="ot">&lt;-</span> colNames</span>
<span id="cb246-182"><a href="example.html#cb246-182" tabindex="-1"></a>  </span>
<span id="cb246-183"><a href="example.html#cb246-183" tabindex="-1"></a>  target <span class="ot">&lt;-</span> metadata_tab <span class="sc">%&gt;%</span> </span>
<span id="cb246-184"><a href="example.html#cb246-184" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb246-185"><a href="example.html#cb246-185" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb246-186"><a href="example.html#cb246-186" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(SampleID<span class="sc">%in%</span><span class="fu">rownames</span>(data))</span>
<span id="cb246-187"><a href="example.html#cb246-187" tabindex="-1"></a>  se <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb246-188"><a href="example.html#cb246-188" tabindex="-1"></a>                                 <span class="at">features =</span> data)</span>
<span id="cb246-189"><a href="example.html#cb246-189" tabindex="-1"></a>  </span>
<span id="cb246-190"><a href="example.html#cb246-190" tabindex="-1"></a>  <span class="co"># need to do some sanity check, for log there may be Inf values introduced</span></span>
<span id="cb246-191"><a href="example.html#cb246-191" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">CheckData</span>(se)</span>
<span id="cb246-192"><a href="example.html#cb246-192" tabindex="-1"></a>  </span>
<span id="cb246-193"><a href="example.html#cb246-193" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb246-194"><a href="example.html#cb246-194" tabindex="-1"></a>}</span>
<span id="cb246-195"><a href="example.html#cb246-195" tabindex="-1"></a></span>
<span id="cb246-196"><a href="example.html#cb246-196" tabindex="-1"></a>se_normalize <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(</span>
<span id="cb246-197"><a href="example.html#cb246-197" tabindex="-1"></a>                      <span class="at">object =</span> se_impute,</span>
<span id="cb246-198"><a href="example.html#cb246-198" tabindex="-1"></a>                      <span class="at">rowNorm =</span> <span class="st">&quot;None&quot;</span>,</span>
<span id="cb246-199"><a href="example.html#cb246-199" tabindex="-1"></a>                      <span class="at">transNorm =</span> <span class="st">&quot;LogNorm&quot;</span>,</span>
<span id="cb246-200"><a href="example.html#cb246-200" tabindex="-1"></a>                      <span class="at">scaleNorm =</span> <span class="st">&quot;ParetoNorm&quot;</span>)</span>
<span id="cb246-201"><a href="example.html#cb246-201" tabindex="-1"></a></span>
<span id="cb246-202"><a href="example.html#cb246-202" tabindex="-1"></a>se_normalize</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 978 61 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(978): MEDL00066 MEDL00356 ... MW0168866 MW0169569
## rowData names(0):
## colnames(61): CJY_V0 CJY_V1 ... mix06 mix07
## colData names(5): group SampleName FMT_status gender age</code></pre>
</div>
<div id="normalization-by-poma-r-package-2" class="section level4 hasAnchor" number="6.3.4.2">
<h4><span class="header-section-number">6.3.4.2</span> Normalization by <strong>POMA</strong> R package<a href="example.html#normalization-by-poma-r-package-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="example.html#cb248-1" tabindex="-1"></a>se_normalize_v2 <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_impute, <span class="at">method =</span> <span class="st">&quot;log_pareto&quot;</span>)</span>
<span id="cb248-2"><a href="example.html#cb248-2" tabindex="-1"></a>se_normalize_v2</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 978 61 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(978): MEDL00066 MEDL00356 ... MW0168866 MW0169569
## rowData names(0):
## colnames(61): CJY_V0 CJY_V1 ... mix06 mix07
## colData names(5): group SampleName FMT_status gender age</code></pre>
</div>
<div id="comparison-of-unnormalized-and-normalized-dataset-2" class="section level4 hasAnchor" number="6.3.4.3">
<h4><span class="header-section-number">6.3.4.3</span> Comparison of unnormalized and normalized dataset<a href="example.html#comparison-of-unnormalized-and-normalized-dataset-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>boxplot</li>
</ul>
<div class="sourceCode" id="cb250"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb250-1"><a href="example.html#cb250-1" tabindex="-1"></a>pl_unnor <span class="ot">&lt;-</span> <span class="fu">PomaBoxplots</span>(se_impute, <span class="at">group =</span> <span class="st">&quot;samples&quot;</span>, <span class="at">jitter =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb250-2"><a href="example.html#cb250-2" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Not Normalized&quot;</span>) <span class="sc">+</span></span>
<span id="cb250-3"><a href="example.html#cb250-3" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="co"># data before normalization</span></span>
<span id="cb250-4"><a href="example.html#cb250-4" tabindex="-1"></a></span>
<span id="cb250-5"><a href="example.html#cb250-5" tabindex="-1"></a>pl_nor <span class="ot">&lt;-</span> <span class="fu">PomaBoxplots</span>(se_normalize, <span class="at">group =</span> <span class="st">&quot;samples&quot;</span>, <span class="at">jitter =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb250-6"><a href="example.html#cb250-6" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Normalized&quot;</span>) <span class="co"># data after normalization</span></span>
<span id="cb250-7"><a href="example.html#cb250-7" tabindex="-1"></a></span>
<span id="cb250-8"><a href="example.html#cb250-8" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(pl_unnor, pl_nor, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">align =</span> <span class="st">&quot;v&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-176-1.png" width="1344" /></p>
<ul>
<li>density</li>
</ul>
<div class="sourceCode" id="cb251"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb251-1"><a href="example.html#cb251-1" tabindex="-1"></a>pl_unnor <span class="ot">&lt;-</span> <span class="fu">PomaDensity</span>(se_impute, <span class="at">group =</span> <span class="st">&quot;features&quot;</span>) <span class="sc">+</span></span>
<span id="cb251-2"><a href="example.html#cb251-2" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Not Normalized&quot;</span>) <span class="sc">+</span></span>
<span id="cb251-3"><a href="example.html#cb251-3" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="co"># data before normalization</span></span>
<span id="cb251-4"><a href="example.html#cb251-4" tabindex="-1"></a></span>
<span id="cb251-5"><a href="example.html#cb251-5" tabindex="-1"></a>pl_nor <span class="ot">&lt;-</span> <span class="fu">PomaDensity</span>(se_normalize, <span class="at">group =</span> <span class="st">&quot;features&quot;</span>) <span class="sc">+</span></span>
<span id="cb251-6"><a href="example.html#cb251-6" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Normalized&quot;</span>) <span class="co"># data after normalization</span></span>
<span id="cb251-7"><a href="example.html#cb251-7" tabindex="-1"></a></span>
<span id="cb251-8"><a href="example.html#cb251-8" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(pl_unnor, pl_nor, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">align =</span> <span class="st">&quot;v&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-177-1.png" width="960" /></p>
</div>
</div>
</div>
<div id="cluster-analysis-1" class="section level2 hasAnchor" number="6.4">
<h2><span class="header-section-number">6.4</span> Cluster Analysis<a href="example.html#cluster-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="hierarchical-clustering-1" class="section level3 hasAnchor" number="6.4.1">
<h3><span class="header-section-number">6.4.1</span> Hierarchical Clustering<a href="example.html#hierarchical-clustering-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb252"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb252-1"><a href="example.html#cb252-1" tabindex="-1"></a>HieraCluster <span class="ot">&lt;-</span> <span class="cf">function</span>(object,</span>
<span id="cb252-2"><a href="example.html#cb252-2" tabindex="-1"></a>                         <span class="at">method_dis =</span> <span class="fu">c</span>(<span class="st">&quot;euclidean&quot;</span>, <span class="st">&quot;bray&quot;</span>),</span>
<span id="cb252-3"><a href="example.html#cb252-3" tabindex="-1"></a>                         <span class="at">method_cluster =</span> <span class="fu">c</span>(<span class="st">&quot;average&quot;</span>, <span class="st">&quot;single&quot;</span>, <span class="st">&quot;complete&quot;</span>, <span class="st">&quot;ward&quot;</span>, <span class="st">&quot;ward.D2&quot;</span>),</span>
<span id="cb252-4"><a href="example.html#cb252-4" tabindex="-1"></a>                         <span class="at">cluster_type =</span> <span class="fu">c</span>(<span class="st">&quot;Agglomerative&quot;</span>, <span class="st">&quot;Divisive&quot;</span>),</span>
<span id="cb252-5"><a href="example.html#cb252-5" tabindex="-1"></a>                         <span class="at">tree_num =</span> <span class="dv">4</span>) {</span>
<span id="cb252-6"><a href="example.html#cb252-6" tabindex="-1"></a>  </span>
<span id="cb252-7"><a href="example.html#cb252-7" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) </span>
<span id="cb252-8"><a href="example.html#cb252-8" tabindex="-1"></a>  metadata_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object)</span>
<span id="cb252-9"><a href="example.html#cb252-9" tabindex="-1"></a>  </span>
<span id="cb252-10"><a href="example.html#cb252-10" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">t</span>(features_tab)</span>
<span id="cb252-11"><a href="example.html#cb252-11" tabindex="-1"></a>  <span class="cf">if</span> (cluster_type <span class="sc">==</span> <span class="st">&quot;Agglomerative&quot;</span>) {</span>
<span id="cb252-12"><a href="example.html#cb252-12" tabindex="-1"></a>    <span class="co"># Agglomerative Hierarchical Clustering </span></span>
<span id="cb252-13"><a href="example.html#cb252-13" tabindex="-1"></a>    <span class="co"># Dissimilarity matrix</span></span>
<span id="cb252-14"><a href="example.html#cb252-14" tabindex="-1"></a>    d <span class="ot">&lt;-</span> <span class="fu">dist</span>(df, <span class="at">method =</span> method_dis)</span>
<span id="cb252-15"><a href="example.html#cb252-15" tabindex="-1"></a>    </span>
<span id="cb252-16"><a href="example.html#cb252-16" tabindex="-1"></a>    <span class="co"># Hierarchical clustering using Linkage method</span></span>
<span id="cb252-17"><a href="example.html#cb252-17" tabindex="-1"></a>    hc <span class="ot">&lt;-</span> <span class="fu">hclust</span>(d, <span class="at">method =</span> method_cluster)</span>
<span id="cb252-18"><a href="example.html#cb252-18" tabindex="-1"></a>    <span class="co"># hc &lt;- agnes(df, method = method_cluster)</span></span>
<span id="cb252-19"><a href="example.html#cb252-19" tabindex="-1"></a>    </span>
<span id="cb252-20"><a href="example.html#cb252-20" tabindex="-1"></a>    <span class="do">####### identifying the strongest clustering structure ################</span></span>
<span id="cb252-21"><a href="example.html#cb252-21" tabindex="-1"></a>    <span class="co"># # methods to assess</span></span>
<span id="cb252-22"><a href="example.html#cb252-22" tabindex="-1"></a>    <span class="co"># m &lt;- c( &quot;average&quot;, &quot;single&quot;, &quot;complete&quot;, &quot;ward&quot;)</span></span>
<span id="cb252-23"><a href="example.html#cb252-23" tabindex="-1"></a>    <span class="co"># names(m) &lt;- c( &quot;average&quot;, &quot;single&quot;, &quot;complete&quot;, &quot;ward&quot;)</span></span>
<span id="cb252-24"><a href="example.html#cb252-24" tabindex="-1"></a>    <span class="co"># </span></span>
<span id="cb252-25"><a href="example.html#cb252-25" tabindex="-1"></a>    <span class="co"># # function to compute coefficient</span></span>
<span id="cb252-26"><a href="example.html#cb252-26" tabindex="-1"></a>    <span class="co"># ac &lt;- function(x) {</span></span>
<span id="cb252-27"><a href="example.html#cb252-27" tabindex="-1"></a>    <span class="co">#   agnes(df, method = x)$ac</span></span>
<span id="cb252-28"><a href="example.html#cb252-28" tabindex="-1"></a>    <span class="co"># }</span></span>
<span id="cb252-29"><a href="example.html#cb252-29" tabindex="-1"></a>    <span class="co"># </span></span>
<span id="cb252-30"><a href="example.html#cb252-30" tabindex="-1"></a>    <span class="co"># map_dbl(m, ac)     </span></span>
<span id="cb252-31"><a href="example.html#cb252-31" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (cluster_type <span class="sc">==</span> <span class="st">&quot;Divisive&quot;</span>) {</span>
<span id="cb252-32"><a href="example.html#cb252-32" tabindex="-1"></a>    <span class="co"># Divisive Hierarchical Clustering </span></span>
<span id="cb252-33"><a href="example.html#cb252-33" tabindex="-1"></a>    hc <span class="ot">&lt;-</span> <span class="fu">diana</span>(df, <span class="at">metric =</span> method_dis)</span>
<span id="cb252-34"><a href="example.html#cb252-34" tabindex="-1"></a>  }</span>
<span id="cb252-35"><a href="example.html#cb252-35" tabindex="-1"></a>  </span>
<span id="cb252-36"><a href="example.html#cb252-36" tabindex="-1"></a>  hc_res <span class="ot">&lt;-</span> <span class="fu">as.hclust</span>(hc)</span>
<span id="cb252-37"><a href="example.html#cb252-37" tabindex="-1"></a>  sub_grp <span class="ot">&lt;-</span> <span class="fu">cutree</span>(hc_res, <span class="at">k =</span> tree_num)</span>
<span id="cb252-38"><a href="example.html#cb252-38" tabindex="-1"></a></span>
<span id="cb252-39"><a href="example.html#cb252-39" tabindex="-1"></a>  <span class="fu">plot</span>(hc_res, <span class="at">cex =</span> <span class="fl">0.6</span>)</span>
<span id="cb252-40"><a href="example.html#cb252-40" tabindex="-1"></a>  <span class="fu">rect.hclust</span>(hc_res, <span class="at">k =</span> tree_num, <span class="at">border =</span> <span class="dv">2</span><span class="sc">:</span>(tree_num<span class="sc">+</span><span class="dv">1</span>)) </span>
<span id="cb252-41"><a href="example.html#cb252-41" tabindex="-1"></a>  </span>
<span id="cb252-42"><a href="example.html#cb252-42" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">data=</span>df,</span>
<span id="cb252-43"><a href="example.html#cb252-43" tabindex="-1"></a>              <span class="at">cluster=</span>sub_grp,</span>
<span id="cb252-44"><a href="example.html#cb252-44" tabindex="-1"></a>              <span class="at">hc=</span>hc_res)</span>
<span id="cb252-45"><a href="example.html#cb252-45" tabindex="-1"></a>  </span>
<span id="cb252-46"><a href="example.html#cb252-46" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb252-47"><a href="example.html#cb252-47" tabindex="-1"></a>}</span></code></pre></div>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb253"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb253-1"><a href="example.html#cb253-1" tabindex="-1"></a>Agg_hc_res <span class="ot">&lt;-</span> <span class="fu">HieraCluster</span>(</span>
<span id="cb253-2"><a href="example.html#cb253-2" tabindex="-1"></a>      <span class="at">object =</span> se_normalize,</span>
<span id="cb253-3"><a href="example.html#cb253-3" tabindex="-1"></a>      <span class="at">method_dis =</span> <span class="st">&quot;euclidean&quot;</span>,</span>
<span id="cb253-4"><a href="example.html#cb253-4" tabindex="-1"></a>      <span class="at">method_cluster =</span> <span class="st">&quot;ward.D2&quot;</span>,</span>
<span id="cb253-5"><a href="example.html#cb253-5" tabindex="-1"></a>      <span class="at">cluster_type =</span> <span class="st">&quot;Agglomerative&quot;</span>,</span>
<span id="cb253-6"><a href="example.html#cb253-6" tabindex="-1"></a>      <span class="at">tree_num =</span> <span class="dv">3</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-179-1.png" width="768" /></p>
<ul>
<li>Visualization: visualize the result in a scatter plot</li>
</ul>
<div class="sourceCode" id="cb254"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb254-1"><a href="example.html#cb254-1" tabindex="-1"></a><span class="fu">fviz_cluster</span>(<span class="fu">list</span>(<span class="at">data =</span> Agg_hc_res<span class="sc">$</span>data, </span>
<span id="cb254-2"><a href="example.html#cb254-2" tabindex="-1"></a>                  <span class="at">cluster =</span> Agg_hc_res<span class="sc">$</span>cluster))</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-180-1.png" width="768" /></p>
</div>
</div>
<div id="chemometrics-analysis-1" class="section level2 hasAnchor" number="6.5">
<h2><span class="header-section-number">6.5</span> Chemometrics Analysis<a href="example.html#chemometrics-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="partial-least-squares-discriminant-analysis-pls-da-1" class="section level3 hasAnchor" number="6.5.1">
<h3><span class="header-section-number">6.5.1</span> Partial Least Squares-Discriminant Analysis (PLS-DA)<a href="example.html#partial-least-squares-discriminant-analysis-pls-da-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb255"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb255-1"><a href="example.html#cb255-1" tabindex="-1"></a>poma_plsda <span class="ot">&lt;-</span> <span class="fu">PomaMultivariate</span>(se_normalize, <span class="at">method =</span> <span class="st">&quot;plsda&quot;</span>)</span></code></pre></div>
<ul>
<li>scatter plot</li>
</ul>
<div class="sourceCode" id="cb256"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb256-1"><a href="example.html#cb256-1" tabindex="-1"></a>poma_plsda<span class="sc">$</span>scoresplot <span class="sc">+</span></span>
<span id="cb256-2"><a href="example.html#cb256-2" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Scores Plot (plsda)&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-182-1.png" width="768" /></p>
<ul>
<li>errors plot</li>
</ul>
<div class="sourceCode" id="cb257"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb257-1"><a href="example.html#cb257-1" tabindex="-1"></a>poma_plsda<span class="sc">$</span>errors_plsda_plot <span class="sc">+</span></span>
<span id="cb257-2"><a href="example.html#cb257-2" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Error Plot (plsda)&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-183-1.png" width="768" /></p>
</div>
<div id="sparse-partial-least-squares-discriminant-analysis-spls-da-1" class="section level3 hasAnchor" number="6.5.2">
<h3><span class="header-section-number">6.5.2</span> Sparse Partial Least Squares-Discriminant Analysis (sPLS-DA)<a href="example.html#sparse-partial-least-squares-discriminant-analysis-spls-da-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<blockquote>
<p>Even though PLS is highly efficient in a high dimensional context, the interpretability of PLS needed to be improved. sPLS has been recently developed by our team to perform simultaneous variable selection in both data sets X and Y data sets, by including LASSO penalizations in PLS on each pair of loading vectors</p>
</blockquote>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb258"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb258-1"><a href="example.html#cb258-1" tabindex="-1"></a>poma_splsda <span class="ot">&lt;-</span> <span class="fu">PomaMultivariate</span>(se_normalize, <span class="at">method =</span> <span class="st">&quot;splsda&quot;</span>)</span></code></pre></div>
<ul>
<li>scatter plot</li>
</ul>
<div class="sourceCode" id="cb259"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb259-1"><a href="example.html#cb259-1" tabindex="-1"></a>poma_splsda<span class="sc">$</span>scoresplot <span class="sc">+</span></span>
<span id="cb259-2"><a href="example.html#cb259-2" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Scores Plot (splsda)&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-185-1.png" width="768" /></p>
</div>
</div>
<div id="univariate-analysis-1" class="section level2 hasAnchor" number="6.6">
<h2><span class="header-section-number">6.6</span> Univariate Analysis<a href="example.html#univariate-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="fold-change-analysis-1" class="section level3 hasAnchor" number="6.6.1">
<h3><span class="header-section-number">6.6.1</span> Fold Change Analysis<a href="example.html#fold-change-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>RawData (inputed data)</li>
</ul>
<div class="sourceCode" id="cb260"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb260-1"><a href="example.html#cb260-1" tabindex="-1"></a>FoldChange <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb260-2"><a href="example.html#cb260-2" tabindex="-1"></a>    x,</span>
<span id="cb260-3"><a href="example.html#cb260-3" tabindex="-1"></a>    group,</span>
<span id="cb260-4"><a href="example.html#cb260-4" tabindex="-1"></a>    group_names) {</span>
<span id="cb260-5"><a href="example.html#cb260-5" tabindex="-1"></a>  </span>
<span id="cb260-6"><a href="example.html#cb260-6" tabindex="-1"></a>  <span class="co"># dataseat</span></span>
<span id="cb260-7"><a href="example.html#cb260-7" tabindex="-1"></a>  metadata <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(x) <span class="sc">%&gt;%</span></span>
<span id="cb260-8"><a href="example.html#cb260-8" tabindex="-1"></a>      <span class="fu">as.data.frame</span>()</span>
<span id="cb260-9"><a href="example.html#cb260-9" tabindex="-1"></a>  profile <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(x) <span class="sc">%&gt;%</span></span>
<span id="cb260-10"><a href="example.html#cb260-10" tabindex="-1"></a>      <span class="fu">as.data.frame</span>()</span>
<span id="cb260-11"><a href="example.html#cb260-11" tabindex="-1"></a></span>
<span id="cb260-12"><a href="example.html#cb260-12" tabindex="-1"></a>  <span class="fu">colnames</span>(metadata)[<span class="fu">which</span>(<span class="fu">colnames</span>(metadata) <span class="sc">==</span> group)] <span class="ot">&lt;-</span> <span class="st">&quot;CompVar&quot;</span></span>
<span id="cb260-13"><a href="example.html#cb260-13" tabindex="-1"></a>  phenotype <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span></span>
<span id="cb260-14"><a href="example.html#cb260-14" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(CompVar <span class="sc">%in%</span> group_names) <span class="sc">%&gt;%</span></span>
<span id="cb260-15"><a href="example.html#cb260-15" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">CompVar =</span> <span class="fu">as.character</span>(CompVar)) <span class="sc">%&gt;%</span></span>
<span id="cb260-16"><a href="example.html#cb260-16" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">CompVar =</span> <span class="fu">factor</span>(CompVar, <span class="at">levels =</span> group_names))</span>
<span id="cb260-17"><a href="example.html#cb260-17" tabindex="-1"></a>  </span>
<span id="cb260-18"><a href="example.html#cb260-18" tabindex="-1"></a>  sid <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(phenotype), <span class="fu">colnames</span>(profile))</span>
<span id="cb260-19"><a href="example.html#cb260-19" tabindex="-1"></a>  phen <span class="ot">&lt;-</span> phenotype[<span class="fu">pmatch</span>(sid, <span class="fu">rownames</span>(phenotype)), , ]</span>
<span id="cb260-20"><a href="example.html#cb260-20" tabindex="-1"></a>  prof <span class="ot">&lt;-</span> profile <span class="sc">%&gt;%</span></span>
<span id="cb260-21"><a href="example.html#cb260-21" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(sid))</span>
<span id="cb260-22"><a href="example.html#cb260-22" tabindex="-1"></a>  </span>
<span id="cb260-23"><a href="example.html#cb260-23" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">all</span>(<span class="fu">colnames</span>(prof) <span class="sc">==</span> <span class="fu">rownames</span>(phen))) {</span>
<span id="cb260-24"><a href="example.html#cb260-24" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;Wrong Order&quot;</span>)</span>
<span id="cb260-25"><a href="example.html#cb260-25" tabindex="-1"></a>  }</span>
<span id="cb260-26"><a href="example.html#cb260-26" tabindex="-1"></a>  </span>
<span id="cb260-27"><a href="example.html#cb260-27" tabindex="-1"></a>  fc_res <span class="ot">&lt;-</span> <span class="fu">apply</span>(prof, <span class="dv">1</span>, <span class="cf">function</span>(x1, y1) {</span>
<span id="cb260-28"><a href="example.html#cb260-28" tabindex="-1"></a>    dat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">value =</span> <span class="fu">as.numeric</span>(x1), <span class="at">group =</span> y1)</span>
<span id="cb260-29"><a href="example.html#cb260-29" tabindex="-1"></a>    mn <span class="ot">&lt;-</span> <span class="fu">tapply</span>(dat<span class="sc">$</span>value, dat<span class="sc">$</span>group, <span class="cf">function</span>(x){</span>
<span id="cb260-30"><a href="example.html#cb260-30" tabindex="-1"></a>      <span class="fu">mean</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb260-31"><a href="example.html#cb260-31" tabindex="-1"></a>    }) <span class="sc">%&gt;%</span></span>
<span id="cb260-32"><a href="example.html#cb260-32" tabindex="-1"></a>      <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb260-33"><a href="example.html#cb260-33" tabindex="-1"></a>      stats<span class="sc">::</span><span class="fu">setNames</span>(<span class="st">&quot;value&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb260-34"><a href="example.html#cb260-34" tabindex="-1"></a>      tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;Group&quot;</span>)</span>
<span id="cb260-35"><a href="example.html#cb260-35" tabindex="-1"></a></span>
<span id="cb260-36"><a href="example.html#cb260-36" tabindex="-1"></a>    mn1 <span class="ot">&lt;-</span> <span class="fu">with</span>(mn, mn[Group <span class="sc">%in%</span> group_names[<span class="dv">1</span>], <span class="st">&quot;value&quot;</span>])</span>
<span id="cb260-37"><a href="example.html#cb260-37" tabindex="-1"></a>    mn2 <span class="ot">&lt;-</span> <span class="fu">with</span>(mn, mn[Group <span class="sc">%in%</span> group_names[<span class="dv">2</span>], <span class="st">&quot;value&quot;</span>])</span>
<span id="cb260-38"><a href="example.html#cb260-38" tabindex="-1"></a>    mnall <span class="ot">&lt;-</span> <span class="fu">mean</span>(dat<span class="sc">$</span>value, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb260-39"><a href="example.html#cb260-39" tabindex="-1"></a>    </span>
<span id="cb260-40"><a href="example.html#cb260-40" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">all</span>(mn1 <span class="sc">!=</span> <span class="dv">0</span>, mn2 <span class="sc">!=</span> <span class="dv">0</span>)) {</span>
<span id="cb260-41"><a href="example.html#cb260-41" tabindex="-1"></a>      fc <span class="ot">&lt;-</span> mn1 <span class="sc">/</span> mn2</span>
<span id="cb260-42"><a href="example.html#cb260-42" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb260-43"><a href="example.html#cb260-43" tabindex="-1"></a>      fc <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb260-44"><a href="example.html#cb260-44" tabindex="-1"></a>    }</span>
<span id="cb260-45"><a href="example.html#cb260-45" tabindex="-1"></a>    </span>
<span id="cb260-46"><a href="example.html#cb260-46" tabindex="-1"></a>    logfc <span class="ot">&lt;-</span> <span class="fu">log2</span>(fc)</span>
<span id="cb260-47"><a href="example.html#cb260-47" tabindex="-1"></a></span>
<span id="cb260-48"><a href="example.html#cb260-48" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="fu">c</span>(fc, logfc, mnall, mn1, mn2)</span>
<span id="cb260-49"><a href="example.html#cb260-49" tabindex="-1"></a>    <span class="fu">return</span>(res)</span>
<span id="cb260-50"><a href="example.html#cb260-50" tabindex="-1"></a>  }, phen<span class="sc">$</span>CompVar) <span class="sc">%&gt;%</span></span>
<span id="cb260-51"><a href="example.html#cb260-51" tabindex="-1"></a>    base<span class="sc">::</span><span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb260-52"><a href="example.html#cb260-52" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;Feature&quot;</span>)</span>
<span id="cb260-53"><a href="example.html#cb260-53" tabindex="-1"></a>  </span>
<span id="cb260-54"><a href="example.html#cb260-54" tabindex="-1"></a>  <span class="fu">colnames</span>(fc_res) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;FeatureID&quot;</span>, <span class="st">&quot;FoldChange&quot;</span>, </span>
<span id="cb260-55"><a href="example.html#cb260-55" tabindex="-1"></a>                        <span class="st">&quot;Log2FoldChange&quot;</span>,</span>
<span id="cb260-56"><a href="example.html#cb260-56" tabindex="-1"></a>                        <span class="st">&quot;Mean Abundance</span><span class="sc">\n</span><span class="st">(All)&quot;</span>,</span>
<span id="cb260-57"><a href="example.html#cb260-57" tabindex="-1"></a>                     <span class="fu">paste0</span>(<span class="st">&quot;Mean Abundance</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="fu">c</span>(<span class="st">&quot;former&quot;</span>, <span class="st">&quot;latter&quot;</span>)))  </span>
<span id="cb260-58"><a href="example.html#cb260-58" tabindex="-1"></a>  </span>
<span id="cb260-59"><a href="example.html#cb260-59" tabindex="-1"></a>  <span class="co"># Number of Group</span></span>
<span id="cb260-60"><a href="example.html#cb260-60" tabindex="-1"></a>  dat_status <span class="ot">&lt;-</span> <span class="fu">table</span>(phen<span class="sc">$</span>CompVar)</span>
<span id="cb260-61"><a href="example.html#cb260-61" tabindex="-1"></a>  dat_status_number <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(dat_status)</span>
<span id="cb260-62"><a href="example.html#cb260-62" tabindex="-1"></a>  dat_status_name <span class="ot">&lt;-</span> <span class="fu">names</span>(dat_status)</span>
<span id="cb260-63"><a href="example.html#cb260-63" tabindex="-1"></a>  fc_res<span class="sc">$</span>Block <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">paste</span>(dat_status_number[<span class="dv">1</span>], dat_status_name[<span class="dv">1</span>], <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>),</span>
<span id="cb260-64"><a href="example.html#cb260-64" tabindex="-1"></a>                       <span class="st">&quot;vs&quot;</span>,</span>
<span id="cb260-65"><a href="example.html#cb260-65" tabindex="-1"></a>                       <span class="fu">paste</span>(dat_status_number[<span class="dv">2</span>], dat_status_name[<span class="dv">2</span>], <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>))</span>
<span id="cb260-66"><a href="example.html#cb260-66" tabindex="-1"></a>  </span>
<span id="cb260-67"><a href="example.html#cb260-67" tabindex="-1"></a>  res <span class="ot">&lt;-</span> fc_res <span class="sc">%&gt;%</span></span>
<span id="cb260-68"><a href="example.html#cb260-68" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(FeatureID, Block, <span class="fu">everything</span>())    </span>
<span id="cb260-69"><a href="example.html#cb260-69" tabindex="-1"></a></span>
<span id="cb260-70"><a href="example.html#cb260-70" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb260-71"><a href="example.html#cb260-71" tabindex="-1"></a>}</span>
<span id="cb260-72"><a href="example.html#cb260-72" tabindex="-1"></a></span>
<span id="cb260-73"><a href="example.html#cb260-73" tabindex="-1"></a></span>
<span id="cb260-74"><a href="example.html#cb260-74" tabindex="-1"></a>fc_res <span class="ot">&lt;-</span> <span class="fu">FoldChange</span>(</span>
<span id="cb260-75"><a href="example.html#cb260-75" tabindex="-1"></a>           <span class="at">x =</span> se_impute,</span>
<span id="cb260-76"><a href="example.html#cb260-76" tabindex="-1"></a>           <span class="at">group =</span> <span class="st">&quot;group&quot;</span>,</span>
<span id="cb260-77"><a href="example.html#cb260-77" tabindex="-1"></a>           <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;NR&quot;</span>, <span class="st">&quot;PR&quot;</span>))</span>
<span id="cb260-78"><a href="example.html#cb260-78" tabindex="-1"></a></span>
<span id="cb260-79"><a href="example.html#cb260-79" tabindex="-1"></a><span class="fu">head</span>(fc_res)</span></code></pre></div>
<pre><code>##   FeatureID          Block FoldChange Log2FoldChange Mean Abundance\n(All) Mean Abundance\nformer Mean Abundance\nlatter
## 1 MEDL00066 22_NR vs 22_PR  0.3725103    -1.42464778             884972.39              480377.20             1289567.58
## 2 MEDL00356 22_NR vs 22_PR  0.2998209    -1.73782728             296756.21              136901.49              456610.93
## 3 MEDL00369 22_NR vs 22_PR  0.8976511    -0.15577325              57818.66               54700.24               60937.08
## 4 MEDL00375 22_NR vs 22_PR  1.1237471     0.16831736             391121.42              413911.40              368331.45
## 5 MEDL00392 22_NR vs 22_PR  1.6597914     0.73100193             169984.74              212151.38              127818.10
## 6 MEDL00401 22_NR vs 22_PR  0.9665848    -0.04903182              13670.47               13438.19               13902.75</code></pre>
</div>
<div id="vip-variable-influence-on-projection-coefficient-1" class="section level3 hasAnchor" number="6.6.2">
<h3><span class="header-section-number">6.6.2</span> VIP (Variable influence on projection &amp; coefficient)<a href="example.html#vip-variable-influence-on-projection-coefficient-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><p>Variable influence on projection (VIP) for orthogonal projections to latent structures (OPLS)</p></li>
<li><p>Variable influence on projection (VIP) for projections to latent structures (PLS)</p></li>
</ul>
<div class="sourceCode" id="cb262"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb262-1"><a href="example.html#cb262-1" tabindex="-1"></a>VIP_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb262-2"><a href="example.html#cb262-2" tabindex="-1"></a>    x,</span>
<span id="cb262-3"><a href="example.html#cb262-3" tabindex="-1"></a>    group,</span>
<span id="cb262-4"><a href="example.html#cb262-4" tabindex="-1"></a>    group_names,</span>
<span id="cb262-5"><a href="example.html#cb262-5" tabindex="-1"></a>    <span class="at">VIPtype =</span> <span class="fu">c</span>(<span class="st">&quot;OPLS&quot;</span>, <span class="st">&quot;PLS&quot;</span>),</span>
<span id="cb262-6"><a href="example.html#cb262-6" tabindex="-1"></a>    <span class="at">vip_cutoff =</span> <span class="dv">1</span>) {</span>
<span id="cb262-7"><a href="example.html#cb262-7" tabindex="-1"></a>  </span>
<span id="cb262-8"><a href="example.html#cb262-8" tabindex="-1"></a>  metadata <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(x) <span class="sc">%&gt;%</span></span>
<span id="cb262-9"><a href="example.html#cb262-9" tabindex="-1"></a>      <span class="fu">as.data.frame</span>()</span>
<span id="cb262-10"><a href="example.html#cb262-10" tabindex="-1"></a>  profile <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(x) <span class="sc">%&gt;%</span></span>
<span id="cb262-11"><a href="example.html#cb262-11" tabindex="-1"></a>      <span class="fu">as.data.frame</span>()</span>
<span id="cb262-12"><a href="example.html#cb262-12" tabindex="-1"></a></span>
<span id="cb262-13"><a href="example.html#cb262-13" tabindex="-1"></a>  <span class="fu">colnames</span>(metadata)[<span class="fu">which</span>(<span class="fu">colnames</span>(metadata) <span class="sc">==</span> group)] <span class="ot">&lt;-</span> <span class="st">&quot;CompVar&quot;</span></span>
<span id="cb262-14"><a href="example.html#cb262-14" tabindex="-1"></a>  phenotype <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span></span>
<span id="cb262-15"><a href="example.html#cb262-15" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(CompVar <span class="sc">%in%</span> group_names) <span class="sc">%&gt;%</span></span>
<span id="cb262-16"><a href="example.html#cb262-16" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">CompVar =</span> <span class="fu">as.character</span>(CompVar)) <span class="sc">%&gt;%</span></span>
<span id="cb262-17"><a href="example.html#cb262-17" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">CompVar =</span> <span class="fu">factor</span>(CompVar, <span class="at">levels =</span> group_names))</span>
<span id="cb262-18"><a href="example.html#cb262-18" tabindex="-1"></a>  </span>
<span id="cb262-19"><a href="example.html#cb262-19" tabindex="-1"></a>  sid <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(phenotype), <span class="fu">colnames</span>(profile))</span>
<span id="cb262-20"><a href="example.html#cb262-20" tabindex="-1"></a>  phen <span class="ot">&lt;-</span> phenotype[<span class="fu">pmatch</span>(sid, <span class="fu">rownames</span>(phenotype)), , ]</span>
<span id="cb262-21"><a href="example.html#cb262-21" tabindex="-1"></a>  prof <span class="ot">&lt;-</span> profile <span class="sc">%&gt;%</span></span>
<span id="cb262-22"><a href="example.html#cb262-22" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(sid))</span>
<span id="cb262-23"><a href="example.html#cb262-23" tabindex="-1"></a>  </span>
<span id="cb262-24"><a href="example.html#cb262-24" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">all</span>(<span class="fu">colnames</span>(prof) <span class="sc">==</span> <span class="fu">rownames</span>(phen))) {</span>
<span id="cb262-25"><a href="example.html#cb262-25" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;Wrong Order&quot;</span>)</span>
<span id="cb262-26"><a href="example.html#cb262-26" tabindex="-1"></a>  }</span>
<span id="cb262-27"><a href="example.html#cb262-27" tabindex="-1"></a>  </span>
<span id="cb262-28"><a href="example.html#cb262-28" tabindex="-1"></a>  dataMatrix <span class="ot">&lt;-</span> prof <span class="sc">%&gt;%</span> base<span class="sc">::</span><span class="fu">t</span>() <span class="co"># row-&gt;sampleID; col-&gt;features</span></span>
<span id="cb262-29"><a href="example.html#cb262-29" tabindex="-1"></a>  sampleMetadata <span class="ot">&lt;-</span> phen <span class="co"># row-&gt;sampleID; col-&gt;features</span></span>
<span id="cb262-30"><a href="example.html#cb262-30" tabindex="-1"></a>  </span>
<span id="cb262-31"><a href="example.html#cb262-31" tabindex="-1"></a>  comparsionVn <span class="ot">&lt;-</span> sampleMetadata[, <span class="st">&quot;CompVar&quot;</span>]</span>
<span id="cb262-32"><a href="example.html#cb262-32" tabindex="-1"></a>  </span>
<span id="cb262-33"><a href="example.html#cb262-33" tabindex="-1"></a>  <span class="co"># corrlation between group and features</span></span>
<span id="cb262-34"><a href="example.html#cb262-34" tabindex="-1"></a>  pvaVn <span class="ot">&lt;-</span> <span class="fu">apply</span>(dataMatrix, <span class="dv">2</span>,</span>
<span id="cb262-35"><a href="example.html#cb262-35" tabindex="-1"></a>                 <span class="cf">function</span>(feaVn) <span class="fu">cor.test</span>(<span class="fu">as.numeric</span>(comparsionVn), feaVn)[[<span class="st">&quot;p.value&quot;</span>]])</span>
<span id="cb262-36"><a href="example.html#cb262-36" tabindex="-1"></a>  </span>
<span id="cb262-37"><a href="example.html#cb262-37" tabindex="-1"></a>  <span class="fu">library</span>(ropls)</span>
<span id="cb262-38"><a href="example.html#cb262-38" tabindex="-1"></a>  </span>
<span id="cb262-39"><a href="example.html#cb262-39" tabindex="-1"></a>  <span class="cf">if</span> (VIPtype <span class="sc">==</span> <span class="st">&quot;OPLS&quot;</span>) {</span>
<span id="cb262-40"><a href="example.html#cb262-40" tabindex="-1"></a>    vipVn <span class="ot">&lt;-</span> <span class="fu">getVipVn</span>(<span class="fu">opls</span>(dataMatrix, </span>
<span id="cb262-41"><a href="example.html#cb262-41" tabindex="-1"></a>                           comparsionVn,</span>
<span id="cb262-42"><a href="example.html#cb262-42" tabindex="-1"></a>                           <span class="at">predI =</span> <span class="dv">1</span>, </span>
<span id="cb262-43"><a href="example.html#cb262-43" tabindex="-1"></a>                           <span class="at">orthoI =</span> <span class="cn">NA</span>,</span>
<span id="cb262-44"><a href="example.html#cb262-44" tabindex="-1"></a>                           <span class="at">fig.pdfC =</span> <span class="st">&quot;none&quot;</span>))    </span>
<span id="cb262-45"><a href="example.html#cb262-45" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb262-46"><a href="example.html#cb262-46" tabindex="-1"></a>    vipVn <span class="ot">&lt;-</span> <span class="fu">getVipVn</span>(<span class="fu">opls</span>(dataMatrix, </span>
<span id="cb262-47"><a href="example.html#cb262-47" tabindex="-1"></a>                           comparsionVn,</span>
<span id="cb262-48"><a href="example.html#cb262-48" tabindex="-1"></a>                           <span class="at">predI =</span> <span class="dv">1</span>, </span>
<span id="cb262-49"><a href="example.html#cb262-49" tabindex="-1"></a>                           <span class="at">fig.pdfC =</span> <span class="st">&quot;none&quot;</span>))    </span>
<span id="cb262-50"><a href="example.html#cb262-50" tabindex="-1"></a>  }</span>
<span id="cb262-51"><a href="example.html#cb262-51" tabindex="-1"></a>  quantVn <span class="ot">&lt;-</span> <span class="fu">qnorm</span>(<span class="dv">1</span> <span class="sc">-</span> pvaVn <span class="sc">/</span> <span class="dv">2</span>)</span>
<span id="cb262-52"><a href="example.html#cb262-52" tabindex="-1"></a>  rmsQuantN <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="fu">mean</span>(quantVn<span class="sc">^</span><span class="dv">2</span>))</span>
<span id="cb262-53"><a href="example.html#cb262-53" tabindex="-1"></a>  </span>
<span id="cb262-54"><a href="example.html#cb262-54" tabindex="-1"></a>  opar <span class="ot">&lt;-</span> <span class="fu">par</span>(<span class="at">font =</span> <span class="dv">2</span>, <span class="at">font.axis =</span> <span class="dv">2</span>, <span class="at">font.lab =</span> <span class="dv">2</span>,</span>
<span id="cb262-55"><a href="example.html#cb262-55" tabindex="-1"></a>              <span class="at">las =</span> <span class="dv">1</span>,</span>
<span id="cb262-56"><a href="example.html#cb262-56" tabindex="-1"></a>              <span class="at">mar =</span> <span class="fu">c</span>(<span class="fl">5.1</span>, <span class="fl">4.6</span>, <span class="fl">4.1</span>, <span class="fl">2.1</span>),</span>
<span id="cb262-57"><a href="example.html#cb262-57" tabindex="-1"></a>              <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">pch =</span> <span class="dv">16</span>)</span>
<span id="cb262-58"><a href="example.html#cb262-58" tabindex="-1"></a>  <span class="fu">plot</span>(pvaVn, </span>
<span id="cb262-59"><a href="example.html#cb262-59" tabindex="-1"></a>       vipVn,</span>
<span id="cb262-60"><a href="example.html#cb262-60" tabindex="-1"></a>       <span class="at">col =</span> <span class="st">&quot;red&quot;</span>,</span>
<span id="cb262-61"><a href="example.html#cb262-61" tabindex="-1"></a>       <span class="at">pch =</span> <span class="dv">16</span>,</span>
<span id="cb262-62"><a href="example.html#cb262-62" tabindex="-1"></a>       <span class="at">xlab =</span> <span class="st">&quot;p-value&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;VIP&quot;</span>, <span class="at">xaxs =</span> <span class="st">&quot;i&quot;</span>, <span class="at">yaxs =</span> <span class="st">&quot;i&quot;</span>)</span>
<span id="cb262-63"><a href="example.html#cb262-63" tabindex="-1"></a>  <span class="fu">box</span>(<span class="at">lwd =</span> <span class="dv">2</span>)</span>
<span id="cb262-64"><a href="example.html#cb262-64" tabindex="-1"></a>  <span class="fu">curve</span>(<span class="fu">qnorm</span>(<span class="dv">1</span> <span class="sc">-</span> x <span class="sc">/</span> <span class="dv">2</span>) <span class="sc">/</span> rmsQuantN, <span class="dv">0</span>, <span class="dv">1</span>, <span class="at">add =</span> <span class="cn">TRUE</span>, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>, <span class="at">lwd =</span> <span class="dv">3</span>)</span>
<span id="cb262-65"><a href="example.html#cb262-65" tabindex="-1"></a>  <span class="fu">abline</span>(<span class="at">h =</span> <span class="dv">1</span>, <span class="at">col =</span> <span class="st">&quot;blue&quot;</span>)</span>
<span id="cb262-66"><a href="example.html#cb262-66" tabindex="-1"></a>  <span class="fu">abline</span>(<span class="at">v =</span> <span class="fl">0.05</span>, <span class="at">col =</span> <span class="st">&quot;blue&quot;</span>)</span>
<span id="cb262-67"><a href="example.html#cb262-67" tabindex="-1"></a>  </span>
<span id="cb262-68"><a href="example.html#cb262-68" tabindex="-1"></a>  res_temp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb262-69"><a href="example.html#cb262-69" tabindex="-1"></a>      <span class="at">FeatureID =</span> <span class="fu">names</span>(vipVn),</span>
<span id="cb262-70"><a href="example.html#cb262-70" tabindex="-1"></a>      <span class="at">VIP =</span> vipVn,</span>
<span id="cb262-71"><a href="example.html#cb262-71" tabindex="-1"></a>      <span class="at">CorPvalue =</span> pvaVn) <span class="sc">%&gt;%</span></span>
<span id="cb262-72"><a href="example.html#cb262-72" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">desc</span>(VIP))</span>
<span id="cb262-73"><a href="example.html#cb262-73" tabindex="-1"></a></span>
<span id="cb262-74"><a href="example.html#cb262-74" tabindex="-1"></a>  vip_select <span class="ot">&lt;-</span> res_temp <span class="sc">%&gt;%</span></span>
<span id="cb262-75"><a href="example.html#cb262-75" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(VIP <span class="sc">&gt;</span> vip_cutoff) </span>
<span id="cb262-76"><a href="example.html#cb262-76" tabindex="-1"></a>  </span>
<span id="cb262-77"><a href="example.html#cb262-77" tabindex="-1"></a>  pl <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(vip_select, <span class="fu">aes</span>(FeatureID, VIP)) <span class="sc">+</span></span>
<span id="cb262-78"><a href="example.html#cb262-78" tabindex="-1"></a>    <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> FeatureID, <span class="at">xend =</span> FeatureID,</span>
<span id="cb262-79"><a href="example.html#cb262-79" tabindex="-1"></a>                     <span class="at">y =</span> <span class="dv">0</span>, <span class="at">yend =</span> VIP)) <span class="sc">+</span></span>
<span id="cb262-80"><a href="example.html#cb262-80" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">21</span>, <span class="at">size =</span> <span class="dv">5</span>, <span class="at">color =</span> <span class="st">&#39;#008000&#39;</span> ,<span class="at">fill =</span> <span class="st">&#39;#008000&#39;</span>) <span class="sc">+</span></span>
<span id="cb262-81"><a href="example.html#cb262-81" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="dv">1</span>,<span class="fl">2.5</span>), <span class="at">color =</span> <span class="st">&#39;white&#39;</span>) <span class="sc">+</span></span>
<span id="cb262-82"><a href="example.html#cb262-82" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">1</span>, <span class="at">linetype =</span> <span class="st">&#39;dashed&#39;</span>) <span class="sc">+</span></span>
<span id="cb262-83"><a href="example.html#cb262-83" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb262-84"><a href="example.html#cb262-84" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&#39;&#39;</span>, <span class="at">y =</span> <span class="st">&#39;VIP value&#39;</span>) <span class="sc">+</span></span>
<span id="cb262-85"><a href="example.html#cb262-85" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb262-86"><a href="example.html#cb262-86" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&#39;none&#39;</span>,</span>
<span id="cb262-87"><a href="example.html#cb262-87" tabindex="-1"></a>          <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">&#39;black&#39;</span>,<span class="at">size =</span> <span class="dv">12</span>, <span class="at">family =</span> <span class="st">&#39;Arial&#39;</span>, <span class="at">face =</span> <span class="st">&#39;plain&#39;</span>),</span>
<span id="cb262-88"><a href="example.html#cb262-88" tabindex="-1"></a>          <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb262-89"><a href="example.html#cb262-89" tabindex="-1"></a>          <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb262-90"><a href="example.html#cb262-90" tabindex="-1"></a>          <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">&#39;black&#39;</span>,<span class="at">size =</span> <span class="dv">15</span>, <span class="at">family =</span> <span class="st">&#39;Arial&#39;</span>, <span class="at">face =</span> <span class="st">&#39;plain&#39;</span>),</span>
<span id="cb262-91"><a href="example.html#cb262-91" tabindex="-1"></a>          <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>),</span>
<span id="cb262-92"><a href="example.html#cb262-92" tabindex="-1"></a>          <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">&#39;black&#39;</span>,<span class="at">size =</span> <span class="dv">15</span>, <span class="at">family =</span> <span class="st">&#39;Arial&#39;</span>, <span class="at">face =</span> <span class="st">&#39;plain&#39;</span>),</span>
<span id="cb262-93"><a href="example.html#cb262-93" tabindex="-1"></a>          <span class="at">axis.ticks =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb262-94"><a href="example.html#cb262-94" tabindex="-1"></a>          <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>())</span>
<span id="cb262-95"><a href="example.html#cb262-95" tabindex="-1"></a>  </span>
<span id="cb262-96"><a href="example.html#cb262-96" tabindex="-1"></a>  <span class="co"># Number of Group</span></span>
<span id="cb262-97"><a href="example.html#cb262-97" tabindex="-1"></a>  dat_status <span class="ot">&lt;-</span> <span class="fu">table</span>(phen<span class="sc">$</span>CompVar)</span>
<span id="cb262-98"><a href="example.html#cb262-98" tabindex="-1"></a>  dat_status_number <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(dat_status)</span>
<span id="cb262-99"><a href="example.html#cb262-99" tabindex="-1"></a>  dat_status_name <span class="ot">&lt;-</span> <span class="fu">names</span>(dat_status)</span>
<span id="cb262-100"><a href="example.html#cb262-100" tabindex="-1"></a>  res_temp<span class="sc">$</span>Block <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">paste</span>(dat_status_number[<span class="dv">1</span>], dat_status_name[<span class="dv">1</span>], <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>),</span>
<span id="cb262-101"><a href="example.html#cb262-101" tabindex="-1"></a>                       <span class="st">&quot;vs&quot;</span>,</span>
<span id="cb262-102"><a href="example.html#cb262-102" tabindex="-1"></a>                       <span class="fu">paste</span>(dat_status_number[<span class="dv">2</span>], dat_status_name[<span class="dv">2</span>], <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>))</span>
<span id="cb262-103"><a href="example.html#cb262-103" tabindex="-1"></a></span>
<span id="cb262-104"><a href="example.html#cb262-104" tabindex="-1"></a>  res_df <span class="ot">&lt;-</span> res_temp <span class="sc">%&gt;%</span></span>
<span id="cb262-105"><a href="example.html#cb262-105" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(FeatureID, Block, <span class="fu">everything</span>())    </span>
<span id="cb262-106"><a href="example.html#cb262-106" tabindex="-1"></a></span>
<span id="cb262-107"><a href="example.html#cb262-107" tabindex="-1"></a></span>
<span id="cb262-108"><a href="example.html#cb262-108" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">vip =</span> res_df,</span>
<span id="cb262-109"><a href="example.html#cb262-109" tabindex="-1"></a>              <span class="at">plot =</span> pl)</span>
<span id="cb262-110"><a href="example.html#cb262-110" tabindex="-1"></a>  </span>
<span id="cb262-111"><a href="example.html#cb262-111" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb262-112"><a href="example.html#cb262-112" tabindex="-1"></a>}</span>
<span id="cb262-113"><a href="example.html#cb262-113" tabindex="-1"></a></span>
<span id="cb262-114"><a href="example.html#cb262-114" tabindex="-1"></a>vip_res <span class="ot">&lt;-</span> <span class="fu">VIP_fun</span>(</span>
<span id="cb262-115"><a href="example.html#cb262-115" tabindex="-1"></a>  <span class="at">x =</span> se_normalize,</span>
<span id="cb262-116"><a href="example.html#cb262-116" tabindex="-1"></a>  <span class="at">group =</span> <span class="st">&quot;group&quot;</span>,</span>
<span id="cb262-117"><a href="example.html#cb262-117" tabindex="-1"></a>  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;NR&quot;</span>, <span class="st">&quot;PR&quot;</span>),</span>
<span id="cb262-118"><a href="example.html#cb262-118" tabindex="-1"></a>  <span class="at">VIPtype =</span> <span class="st">&quot;PLS&quot;</span>,</span>
<span id="cb262-119"><a href="example.html#cb262-119" tabindex="-1"></a>  <span class="at">vip_cutoff =</span> <span class="dv">1</span>)</span></code></pre></div>
<pre><code>## PLS-DA
## 44 samples x 978 variables and 1 response
## standard scaling of predictors and response(s)
##       R2X(cum) R2Y(cum) Q2(cum) RMSEE pre ort pR2Y pQ2
## Total    0.101    0.483   0.048 0.368   1   0 0.75 0.2</code></pre>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-187-1.png" width="672" /></p>
<div class="sourceCode" id="cb264"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb264-1"><a href="example.html#cb264-1" tabindex="-1"></a><span class="fu">head</span>(vip_res<span class="sc">$</span>vip)</span></code></pre></div>
<pre><code>##           FeatureID          Block      VIP    CorPvalue
## MEDP0151   MEDP0151 22_NR vs 22_PR 2.903799 0.0001855852
## MEDP1276   MEDP1276 22_NR vs 22_PR 2.903799 0.0001855852
## MEDP1177   MEDP1177 22_NR vs 22_PR 2.885603 0.0002072022
## MEDP1040   MEDP1040 22_NR vs 22_PR 2.881680 0.0002121518
## MW0166596 MW0166596 22_NR vs 22_PR 2.712717 0.0005586171
## MW0168376 MW0168376 22_NR vs 22_PR 2.673082 0.0006919501</code></pre>
<div class="sourceCode" id="cb266"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb266-1"><a href="example.html#cb266-1" tabindex="-1"></a>vip_res<span class="sc">$</span>plot</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-187-2.png" width="672" /></p>
</div>
<div id="t-test-by-local-codes-1" class="section level3 hasAnchor" number="6.6.3">
<h3><span class="header-section-number">6.6.3</span> T-test by local codes<a href="example.html#t-test-by-local-codes-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>significant differences between two groups (p value)</li>
</ul>
<div class="sourceCode" id="cb267"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb267-1"><a href="example.html#cb267-1" tabindex="-1"></a>t_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb267-2"><a href="example.html#cb267-2" tabindex="-1"></a>    x,</span>
<span id="cb267-3"><a href="example.html#cb267-3" tabindex="-1"></a>    group,</span>
<span id="cb267-4"><a href="example.html#cb267-4" tabindex="-1"></a>    group_names) {</span>
<span id="cb267-5"><a href="example.html#cb267-5" tabindex="-1"></a>  </span>
<span id="cb267-6"><a href="example.html#cb267-6" tabindex="-1"></a>  <span class="co"># dataseat</span></span>
<span id="cb267-7"><a href="example.html#cb267-7" tabindex="-1"></a>  metadata <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(x) <span class="sc">%&gt;%</span></span>
<span id="cb267-8"><a href="example.html#cb267-8" tabindex="-1"></a>      <span class="fu">as.data.frame</span>()</span>
<span id="cb267-9"><a href="example.html#cb267-9" tabindex="-1"></a>  profile <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(x) <span class="sc">%&gt;%</span></span>
<span id="cb267-10"><a href="example.html#cb267-10" tabindex="-1"></a>      <span class="fu">as.data.frame</span>()</span>
<span id="cb267-11"><a href="example.html#cb267-11" tabindex="-1"></a></span>
<span id="cb267-12"><a href="example.html#cb267-12" tabindex="-1"></a>  <span class="co"># rename variables</span></span>
<span id="cb267-13"><a href="example.html#cb267-13" tabindex="-1"></a>  <span class="fu">colnames</span>(metadata)[<span class="fu">which</span>(<span class="fu">colnames</span>(metadata) <span class="sc">==</span> group)] <span class="ot">&lt;-</span> <span class="st">&quot;CompVar&quot;</span> </span>
<span id="cb267-14"><a href="example.html#cb267-14" tabindex="-1"></a>  </span>
<span id="cb267-15"><a href="example.html#cb267-15" tabindex="-1"></a>  phenotype <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span></span>
<span id="cb267-16"><a href="example.html#cb267-16" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(CompVar <span class="sc">%in%</span> group_names) <span class="sc">%&gt;%</span></span>
<span id="cb267-17"><a href="example.html#cb267-17" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">CompVar =</span> <span class="fu">as.character</span>(CompVar)) <span class="sc">%&gt;%</span></span>
<span id="cb267-18"><a href="example.html#cb267-18" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">CompVar =</span> <span class="fu">factor</span>(CompVar, <span class="at">levels =</span> group_names))</span>
<span id="cb267-19"><a href="example.html#cb267-19" tabindex="-1"></a>  </span>
<span id="cb267-20"><a href="example.html#cb267-20" tabindex="-1"></a>  sid <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(phenotype), <span class="fu">colnames</span>(profile))</span>
<span id="cb267-21"><a href="example.html#cb267-21" tabindex="-1"></a>  phen <span class="ot">&lt;-</span> phenotype[<span class="fu">pmatch</span>(sid, <span class="fu">rownames</span>(phenotype)), , ]</span>
<span id="cb267-22"><a href="example.html#cb267-22" tabindex="-1"></a>  prof <span class="ot">&lt;-</span> profile <span class="sc">%&gt;%</span></span>
<span id="cb267-23"><a href="example.html#cb267-23" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(sid))</span>
<span id="cb267-24"><a href="example.html#cb267-24" tabindex="-1"></a>  </span>
<span id="cb267-25"><a href="example.html#cb267-25" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">all</span>(<span class="fu">colnames</span>(prof) <span class="sc">==</span> <span class="fu">rownames</span>(phen))) {</span>
<span id="cb267-26"><a href="example.html#cb267-26" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;Wrong Order&quot;</span>)</span>
<span id="cb267-27"><a href="example.html#cb267-27" tabindex="-1"></a>  }</span>
<span id="cb267-28"><a href="example.html#cb267-28" tabindex="-1"></a></span>
<span id="cb267-29"><a href="example.html#cb267-29" tabindex="-1"></a>  t_res <span class="ot">&lt;-</span> <span class="fu">apply</span>(prof, <span class="dv">1</span>, <span class="cf">function</span>(x1, y1) {</span>
<span id="cb267-30"><a href="example.html#cb267-30" tabindex="-1"></a>    dat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">value =</span> <span class="fu">as.numeric</span>(x1), <span class="at">group =</span> y1)</span>
<span id="cb267-31"><a href="example.html#cb267-31" tabindex="-1"></a>    </span>
<span id="cb267-32"><a href="example.html#cb267-32" tabindex="-1"></a>    rest <span class="ot">&lt;-</span> <span class="fu">t.test</span>(<span class="at">data =</span> dat, value <span class="sc">~</span> group)</span>
<span id="cb267-33"><a href="example.html#cb267-33" tabindex="-1"></a></span>
<span id="cb267-34"><a href="example.html#cb267-34" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="fu">c</span>(rest<span class="sc">$</span>statistic, rest<span class="sc">$</span>p.value)</span>
<span id="cb267-35"><a href="example.html#cb267-35" tabindex="-1"></a>    <span class="fu">return</span>(res)</span>
<span id="cb267-36"><a href="example.html#cb267-36" tabindex="-1"></a>  }, phen<span class="sc">$</span>CompVar) <span class="sc">%&gt;%</span></span>
<span id="cb267-37"><a href="example.html#cb267-37" tabindex="-1"></a>    base<span class="sc">::</span><span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb267-38"><a href="example.html#cb267-38" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;Feature&quot;</span>)</span>
<span id="cb267-39"><a href="example.html#cb267-39" tabindex="-1"></a>  </span>
<span id="cb267-40"><a href="example.html#cb267-40" tabindex="-1"></a>  <span class="fu">colnames</span>(t_res) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;FeatureID&quot;</span>, <span class="st">&quot;Statistic&quot;</span>, <span class="st">&quot;Pvalue&quot;</span>)</span>
<span id="cb267-41"><a href="example.html#cb267-41" tabindex="-1"></a>  t_res<span class="sc">$</span>AdjustedPvalue <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(<span class="fu">as.numeric</span>(t_res<span class="sc">$</span>Pvalue), <span class="at">method =</span> <span class="st">&quot;BH&quot;</span>)</span>
<span id="cb267-42"><a href="example.html#cb267-42" tabindex="-1"></a>  </span>
<span id="cb267-43"><a href="example.html#cb267-43" tabindex="-1"></a>  <span class="co"># Number of Group</span></span>
<span id="cb267-44"><a href="example.html#cb267-44" tabindex="-1"></a>  dat_status <span class="ot">&lt;-</span> <span class="fu">table</span>(phen<span class="sc">$</span>CompVar)</span>
<span id="cb267-45"><a href="example.html#cb267-45" tabindex="-1"></a>  dat_status_number <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(dat_status)</span>
<span id="cb267-46"><a href="example.html#cb267-46" tabindex="-1"></a>  dat_status_name <span class="ot">&lt;-</span> <span class="fu">names</span>(dat_status)</span>
<span id="cb267-47"><a href="example.html#cb267-47" tabindex="-1"></a>  t_res<span class="sc">$</span>Block <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">paste</span>(dat_status_number[<span class="dv">1</span>], dat_status_name[<span class="dv">1</span>], <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>),</span>
<span id="cb267-48"><a href="example.html#cb267-48" tabindex="-1"></a>                       <span class="st">&quot;vs&quot;</span>,</span>
<span id="cb267-49"><a href="example.html#cb267-49" tabindex="-1"></a>                       <span class="fu">paste</span>(dat_status_number[<span class="dv">2</span>], dat_status_name[<span class="dv">2</span>], <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>))</span>
<span id="cb267-50"><a href="example.html#cb267-50" tabindex="-1"></a>  </span>
<span id="cb267-51"><a href="example.html#cb267-51" tabindex="-1"></a>  res <span class="ot">&lt;-</span> t_res <span class="sc">%&gt;%</span></span>
<span id="cb267-52"><a href="example.html#cb267-52" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(FeatureID, Block, <span class="fu">everything</span>())    </span>
<span id="cb267-53"><a href="example.html#cb267-53" tabindex="-1"></a></span>
<span id="cb267-54"><a href="example.html#cb267-54" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb267-55"><a href="example.html#cb267-55" tabindex="-1"></a>}</span>
<span id="cb267-56"><a href="example.html#cb267-56" tabindex="-1"></a></span>
<span id="cb267-57"><a href="example.html#cb267-57" tabindex="-1"></a>ttest_res <span class="ot">&lt;-</span> <span class="fu">t_fun</span>(</span>
<span id="cb267-58"><a href="example.html#cb267-58" tabindex="-1"></a>  <span class="at">x =</span> se_normalize,</span>
<span id="cb267-59"><a href="example.html#cb267-59" tabindex="-1"></a>  <span class="at">group =</span> <span class="st">&quot;group&quot;</span>,</span>
<span id="cb267-60"><a href="example.html#cb267-60" tabindex="-1"></a>  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;NR&quot;</span>, <span class="st">&quot;PR&quot;</span>))</span>
<span id="cb267-61"><a href="example.html#cb267-61" tabindex="-1"></a></span>
<span id="cb267-62"><a href="example.html#cb267-62" tabindex="-1"></a><span class="fu">head</span>(ttest_res)</span></code></pre></div>
<pre><code>##   FeatureID          Block  Statistic    Pvalue AdjustedPvalue
## 1 MEDL00066 22_NR vs 22_PR -0.7561238 0.4541166      0.8276388
## 2 MEDL00356 22_NR vs 22_PR -0.6643555 0.5102652      0.8479872
## 3 MEDL00369 22_NR vs 22_PR  0.6039380 0.5493194      0.8636438
## 4 MEDL00375 22_NR vs 22_PR -0.6759149 0.5030887      0.8449287
## 5 MEDL00392 22_NR vs 22_PR  1.1826181 0.2436510      0.7285392
## 6 MEDL00401 22_NR vs 22_PR -0.2537679 0.8010020      0.9393045</code></pre>
</div>
<div id="merging-result-1" class="section level3 hasAnchor" number="6.6.4">
<h3><span class="header-section-number">6.6.4</span> Merging result<a href="example.html#merging-result-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><p>Foldchange by Raw Data</p></li>
<li><p>VIP by Normalized Data</p></li>
<li><p>test Pvalue by Normalized Data</p></li>
</ul>
<div class="sourceCode" id="cb269"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb269-1"><a href="example.html#cb269-1" tabindex="-1"></a>mergedResults <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb269-2"><a href="example.html#cb269-2" tabindex="-1"></a>    fc_result,</span>
<span id="cb269-3"><a href="example.html#cb269-3" tabindex="-1"></a>    vip_result,</span>
<span id="cb269-4"><a href="example.html#cb269-4" tabindex="-1"></a>    test_result,</span>
<span id="cb269-5"><a href="example.html#cb269-5" tabindex="-1"></a>    group_names,</span>
<span id="cb269-6"><a href="example.html#cb269-6" tabindex="-1"></a>    group_labels) {</span>
<span id="cb269-7"><a href="example.html#cb269-7" tabindex="-1"></a>  </span>
<span id="cb269-8"><a href="example.html#cb269-8" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(vip_result)) {</span>
<span id="cb269-9"><a href="example.html#cb269-9" tabindex="-1"></a>    mdat <span class="ot">&lt;-</span> fc_result <span class="sc">%&gt;%</span></span>
<span id="cb269-10"><a href="example.html#cb269-10" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Block2 =</span> <span class="fu">paste</span>(group_labels, <span class="at">collapse =</span> <span class="st">&quot; vs &quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb269-11"><a href="example.html#cb269-11" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">FeatureID =</span> <span class="fu">make.names</span>(FeatureID)) <span class="sc">%&gt;%</span></span>
<span id="cb269-12"><a href="example.html#cb269-12" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">&quot;Mean Abundance</span><span class="sc">\n</span><span class="st">(All)&quot;</span>, </span>
<span id="cb269-13"><a href="example.html#cb269-13" tabindex="-1"></a>                              <span class="st">&quot;Mean Abundance</span><span class="sc">\n</span><span class="st">former&quot;</span>, </span>
<span id="cb269-14"><a href="example.html#cb269-14" tabindex="-1"></a>                              <span class="st">&quot;Mean Abundance</span><span class="sc">\n</span><span class="st">latter&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb269-15"><a href="example.html#cb269-15" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">inner_join</span>(test_result <span class="sc">%&gt;%</span></span>
<span id="cb269-16"><a href="example.html#cb269-16" tabindex="-1"></a>                          dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>Block) <span class="sc">%&gt;%</span></span>
<span id="cb269-17"><a href="example.html#cb269-17" tabindex="-1"></a>                          dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">FeatureID =</span> <span class="fu">make.names</span>(FeatureID)),</span>
<span id="cb269-18"><a href="example.html#cb269-18" tabindex="-1"></a>                        <span class="at">by =</span> <span class="st">&quot;FeatureID&quot;</span>) </span>
<span id="cb269-19"><a href="example.html#cb269-19" tabindex="-1"></a>    </span>
<span id="cb269-20"><a href="example.html#cb269-20" tabindex="-1"></a>    res <span class="ot">&lt;-</span> mdat <span class="sc">%&gt;%</span></span>
<span id="cb269-21"><a href="example.html#cb269-21" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(FeatureID, Block2, Block,</span>
<span id="cb269-22"><a href="example.html#cb269-22" tabindex="-1"></a>                    FoldChange, Log2FoldChange,</span>
<span id="cb269-23"><a href="example.html#cb269-23" tabindex="-1"></a>                    Statistic, Pvalue, AdjustedPvalue,</span>
<span id="cb269-24"><a href="example.html#cb269-24" tabindex="-1"></a>                    <span class="fu">everything</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb269-25"><a href="example.html#cb269-25" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">arrange</span>(AdjustedPvalue, Log2FoldChange)    </span>
<span id="cb269-26"><a href="example.html#cb269-26" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb269-27"><a href="example.html#cb269-27" tabindex="-1"></a>    mdat <span class="ot">&lt;-</span> fc_result <span class="sc">%&gt;%</span></span>
<span id="cb269-28"><a href="example.html#cb269-28" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Block2 =</span> <span class="fu">paste</span>(group_labels, <span class="at">collapse =</span> <span class="st">&quot; vs &quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb269-29"><a href="example.html#cb269-29" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">FeatureID =</span> <span class="fu">make.names</span>(FeatureID)) <span class="sc">%&gt;%</span></span>
<span id="cb269-30"><a href="example.html#cb269-30" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">&quot;Mean Abundance</span><span class="sc">\n</span><span class="st">(All)&quot;</span>, </span>
<span id="cb269-31"><a href="example.html#cb269-31" tabindex="-1"></a>                              <span class="st">&quot;Mean Abundance</span><span class="sc">\n</span><span class="st">former&quot;</span>, </span>
<span id="cb269-32"><a href="example.html#cb269-32" tabindex="-1"></a>                              <span class="st">&quot;Mean Abundance</span><span class="sc">\n</span><span class="st">latter&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb269-33"><a href="example.html#cb269-33" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">inner_join</span>(vip_result <span class="sc">%&gt;%</span></span>
<span id="cb269-34"><a href="example.html#cb269-34" tabindex="-1"></a>                          dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>Block) <span class="sc">%&gt;%</span></span>
<span id="cb269-35"><a href="example.html#cb269-35" tabindex="-1"></a>                          dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">FeatureID =</span> <span class="fu">make.names</span>(FeatureID)),</span>
<span id="cb269-36"><a href="example.html#cb269-36" tabindex="-1"></a>                        <span class="at">by =</span> <span class="st">&quot;FeatureID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb269-37"><a href="example.html#cb269-37" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">inner_join</span>(test_result <span class="sc">%&gt;%</span></span>
<span id="cb269-38"><a href="example.html#cb269-38" tabindex="-1"></a>                          dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>Block) <span class="sc">%&gt;%</span></span>
<span id="cb269-39"><a href="example.html#cb269-39" tabindex="-1"></a>                          dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">FeatureID =</span> <span class="fu">make.names</span>(FeatureID)),</span>
<span id="cb269-40"><a href="example.html#cb269-40" tabindex="-1"></a>                        <span class="at">by =</span> <span class="st">&quot;FeatureID&quot;</span>) </span>
<span id="cb269-41"><a href="example.html#cb269-41" tabindex="-1"></a>    </span>
<span id="cb269-42"><a href="example.html#cb269-42" tabindex="-1"></a>    res <span class="ot">&lt;-</span> mdat <span class="sc">%&gt;%</span></span>
<span id="cb269-43"><a href="example.html#cb269-43" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(FeatureID, Block2, Block,</span>
<span id="cb269-44"><a href="example.html#cb269-44" tabindex="-1"></a>                    FoldChange, Log2FoldChange,</span>
<span id="cb269-45"><a href="example.html#cb269-45" tabindex="-1"></a>                    VIP, CorPvalue,</span>
<span id="cb269-46"><a href="example.html#cb269-46" tabindex="-1"></a>                    Statistic, Pvalue, AdjustedPvalue,</span>
<span id="cb269-47"><a href="example.html#cb269-47" tabindex="-1"></a>                    <span class="fu">everything</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb269-48"><a href="example.html#cb269-48" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">arrange</span>(AdjustedPvalue, Log2FoldChange)    </span>
<span id="cb269-49"><a href="example.html#cb269-49" tabindex="-1"></a>  }</span>
<span id="cb269-50"><a href="example.html#cb269-50" tabindex="-1"></a>  </span>
<span id="cb269-51"><a href="example.html#cb269-51" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb269-52"><a href="example.html#cb269-52" tabindex="-1"></a>}</span>
<span id="cb269-53"><a href="example.html#cb269-53" tabindex="-1"></a></span>
<span id="cb269-54"><a href="example.html#cb269-54" tabindex="-1"></a>m_results <span class="ot">&lt;-</span> <span class="fu">mergedResults</span>(</span>
<span id="cb269-55"><a href="example.html#cb269-55" tabindex="-1"></a>  <span class="at">fc_result =</span> fc_res,</span>
<span id="cb269-56"><a href="example.html#cb269-56" tabindex="-1"></a>  <span class="at">vip_result =</span> vip_res<span class="sc">$</span>vip,</span>
<span id="cb269-57"><a href="example.html#cb269-57" tabindex="-1"></a>  <span class="at">test_result =</span> ttest_res,</span>
<span id="cb269-58"><a href="example.html#cb269-58" tabindex="-1"></a>  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;NR&quot;</span>, <span class="st">&quot;PR&quot;</span>),</span>
<span id="cb269-59"><a href="example.html#cb269-59" tabindex="-1"></a>  <span class="at">group_labels =</span> <span class="fu">c</span>(<span class="st">&quot;NR&quot;</span>, <span class="st">&quot;PR&quot;</span>))</span>
<span id="cb269-60"><a href="example.html#cb269-60" tabindex="-1"></a></span>
<span id="cb269-61"><a href="example.html#cb269-61" tabindex="-1"></a><span class="fu">head</span>(m_results)</span></code></pre></div>
<pre><code>##   FeatureID   Block2          Block FoldChange Log2FoldChange      VIP    CorPvalue Statistic       Pvalue AdjustedPvalue
## 1  MEDP0151 NR vs PR 22_NR vs 22_PR 0.06937088      -3.849526 2.903799 0.0001855852 -4.099188 0.0002012845     0.05342607
## 2  MEDP1276 NR vs PR 22_NR vs 22_PR 0.06937088      -3.849526 2.903799 0.0001855852 -4.099188 0.0002012845     0.05342607
## 3  MEDP1177 NR vs PR 22_NR vs 22_PR 0.17373447      -2.525044 2.885603 0.0002072022 -4.063360 0.0002078791     0.05342607
## 4  MEDP1040 NR vs PR 22_NR vs 22_PR 0.25634493      -1.963842 2.881680 0.0002121518 -4.055669 0.0002185115     0.05342607
## 5 MW0166596 NR vs PR 22_NR vs 22_PR 0.34037949      -1.554784 2.712717 0.0005586171 -3.735517 0.0005936658     0.11612104
## 6 MW0168376 NR vs PR 22_NR vs 22_PR 0.28801993      -1.795759 2.673082 0.0006919501 -3.663326 0.0007354806     0.11988334</code></pre>
</div>
<div id="volcano-of-merged-results-1" class="section level3 hasAnchor" number="6.6.5">
<h3><span class="header-section-number">6.6.5</span> Volcano of Merged Results<a href="example.html#volcano-of-merged-results-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb271"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb271-1"><a href="example.html#cb271-1" tabindex="-1"></a>get_volcano <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb271-2"><a href="example.html#cb271-2" tabindex="-1"></a>    inputdata,</span>
<span id="cb271-3"><a href="example.html#cb271-3" tabindex="-1"></a>    group_names,</span>
<span id="cb271-4"><a href="example.html#cb271-4" tabindex="-1"></a>    group_labels,</span>
<span id="cb271-5"><a href="example.html#cb271-5" tabindex="-1"></a>    group_colors,</span>
<span id="cb271-6"><a href="example.html#cb271-6" tabindex="-1"></a>    x_index,</span>
<span id="cb271-7"><a href="example.html#cb271-7" tabindex="-1"></a>    x_cutoff,</span>
<span id="cb271-8"><a href="example.html#cb271-8" tabindex="-1"></a>    y_index,</span>
<span id="cb271-9"><a href="example.html#cb271-9" tabindex="-1"></a>    y_cutoff,</span>
<span id="cb271-10"><a href="example.html#cb271-10" tabindex="-1"></a>    <span class="at">plot =</span> <span class="cn">TRUE</span>) {</span>
<span id="cb271-11"><a href="example.html#cb271-11" tabindex="-1"></a>  </span>
<span id="cb271-12"><a href="example.html#cb271-12" tabindex="-1"></a>  selected_group2 <span class="ot">&lt;-</span> <span class="fu">paste</span>(group_labels, <span class="at">collapse =</span> <span class="st">&quot; vs &quot;</span>) </span>
<span id="cb271-13"><a href="example.html#cb271-13" tabindex="-1"></a>  dat <span class="ot">&lt;-</span> inputdata <span class="sc">%&gt;%</span></span>
<span id="cb271-14"><a href="example.html#cb271-14" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(Block2 <span class="sc">%in%</span> selected_group2) </span>
<span id="cb271-15"><a href="example.html#cb271-15" tabindex="-1"></a>  plotdata <span class="ot">&lt;-</span> dat <span class="sc">%&gt;%</span></span>
<span id="cb271-16"><a href="example.html#cb271-16" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">FeatureID =</span> <span class="fu">paste</span>(FeatureID, <span class="at">sep =</span> <span class="st">&quot;:&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb271-17"><a href="example.html#cb271-17" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">&quot;FeatureID&quot;</span>, <span class="st">&quot;Block2&quot;</span>, x_index, y_index)))</span>
<span id="cb271-18"><a href="example.html#cb271-18" tabindex="-1"></a>  </span>
<span id="cb271-19"><a href="example.html#cb271-19" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">any</span>(<span class="fu">colnames</span>(plotdata) <span class="sc">%in%</span> <span class="st">&quot;TaxaID&quot;</span>)) {</span>
<span id="cb271-20"><a href="example.html#cb271-20" tabindex="-1"></a>    <span class="fu">colnames</span>(plotdata)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;TaxaID&quot;</span></span>
<span id="cb271-21"><a href="example.html#cb271-21" tabindex="-1"></a>  }</span>
<span id="cb271-22"><a href="example.html#cb271-22" tabindex="-1"></a>  <span class="cf">if</span> (y_index <span class="sc">==</span> <span class="st">&quot;CorPvalue&quot;</span>) {</span>
<span id="cb271-23"><a href="example.html#cb271-23" tabindex="-1"></a>    <span class="fu">colnames</span>(plotdata)[<span class="fu">which</span>(<span class="fu">colnames</span>(plotdata) <span class="sc">==</span> y_index)] <span class="ot">&lt;-</span> <span class="st">&quot;Pvalue&quot;</span></span>
<span id="cb271-24"><a href="example.html#cb271-24" tabindex="-1"></a>    y_index <span class="ot">&lt;-</span> <span class="st">&quot;Pvalue&quot;</span></span>
<span id="cb271-25"><a href="example.html#cb271-25" tabindex="-1"></a>  }</span>
<span id="cb271-26"><a href="example.html#cb271-26" tabindex="-1"></a> </span>
<span id="cb271-27"><a href="example.html#cb271-27" tabindex="-1"></a>  </span>
<span id="cb271-28"><a href="example.html#cb271-28" tabindex="-1"></a>  pl <span class="ot">&lt;-</span> <span class="fu">plot_volcano</span>(</span>
<span id="cb271-29"><a href="example.html#cb271-29" tabindex="-1"></a>      <span class="at">da_res =</span> plotdata,</span>
<span id="cb271-30"><a href="example.html#cb271-30" tabindex="-1"></a>      <span class="at">group_names =</span> group_labels,</span>
<span id="cb271-31"><a href="example.html#cb271-31" tabindex="-1"></a>      <span class="at">x_index =</span> x_index,</span>
<span id="cb271-32"><a href="example.html#cb271-32" tabindex="-1"></a>      <span class="at">x_index_cutoff =</span> x_cutoff,</span>
<span id="cb271-33"><a href="example.html#cb271-33" tabindex="-1"></a>      <span class="at">y_index =</span> y_index,</span>
<span id="cb271-34"><a href="example.html#cb271-34" tabindex="-1"></a>      <span class="at">y_index_cutoff =</span> y_cutoff,</span>
<span id="cb271-35"><a href="example.html#cb271-35" tabindex="-1"></a>      <span class="at">group_color =</span> <span class="fu">c</span>(group_colors[<span class="dv">1</span>], <span class="st">&quot;grey&quot;</span>, group_colors[<span class="dv">2</span>]))</span>
<span id="cb271-36"><a href="example.html#cb271-36" tabindex="-1"></a>  </span>
<span id="cb271-37"><a href="example.html#cb271-37" tabindex="-1"></a>  <span class="cf">if</span> (plot) {</span>
<span id="cb271-38"><a href="example.html#cb271-38" tabindex="-1"></a>    res <span class="ot">&lt;-</span> pl</span>
<span id="cb271-39"><a href="example.html#cb271-39" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb271-40"><a href="example.html#cb271-40" tabindex="-1"></a>    <span class="fu">colnames</span>(plotdata)[<span class="fu">which</span>(<span class="fu">colnames</span>(plotdata) <span class="sc">==</span> x_index)] <span class="ot">&lt;-</span> <span class="st">&quot;Xindex&quot;</span></span>
<span id="cb271-41"><a href="example.html#cb271-41" tabindex="-1"></a>    <span class="fu">colnames</span>(plotdata)[<span class="fu">which</span>(<span class="fu">colnames</span>(plotdata) <span class="sc">==</span> y_index)] <span class="ot">&lt;-</span> <span class="st">&quot;Yindex&quot;</span></span>
<span id="cb271-42"><a href="example.html#cb271-42" tabindex="-1"></a>    </span>
<span id="cb271-43"><a href="example.html#cb271-43" tabindex="-1"></a>    datsignif <span class="ot">&lt;-</span> plotdata <span class="sc">%&gt;%</span></span>
<span id="cb271-44"><a href="example.html#cb271-44" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">abs</span>(Xindex) <span class="sc">&gt;</span> x_cutoff) <span class="sc">%&gt;%</span></span>
<span id="cb271-45"><a href="example.html#cb271-45" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">filter</span>(Yindex <span class="sc">&lt;</span> y_cutoff)</span>
<span id="cb271-46"><a href="example.html#cb271-46" tabindex="-1"></a>    </span>
<span id="cb271-47"><a href="example.html#cb271-47" tabindex="-1"></a>    <span class="fu">colnames</span>(datsignif)[<span class="fu">which</span>(<span class="fu">colnames</span>(datsignif) <span class="sc">==</span> <span class="st">&quot;Xindex&quot;</span>)] <span class="ot">&lt;-</span> x_index</span>
<span id="cb271-48"><a href="example.html#cb271-48" tabindex="-1"></a>    <span class="fu">colnames</span>(datsignif)[<span class="fu">which</span>(<span class="fu">colnames</span>(datsignif) <span class="sc">==</span> <span class="st">&quot;Yindex&quot;</span>)] <span class="ot">&lt;-</span> y_index</span>
<span id="cb271-49"><a href="example.html#cb271-49" tabindex="-1"></a>        </span>
<span id="cb271-50"><a href="example.html#cb271-50" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">figure =</span> pl,</span>
<span id="cb271-51"><a href="example.html#cb271-51" tabindex="-1"></a>                <span class="at">data =</span> datsignif)</span>
<span id="cb271-52"><a href="example.html#cb271-52" tabindex="-1"></a>    </span>
<span id="cb271-53"><a href="example.html#cb271-53" tabindex="-1"></a>  }</span>
<span id="cb271-54"><a href="example.html#cb271-54" tabindex="-1"></a>  </span>
<span id="cb271-55"><a href="example.html#cb271-55" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb271-56"><a href="example.html#cb271-56" tabindex="-1"></a>}</span>
<span id="cb271-57"><a href="example.html#cb271-57" tabindex="-1"></a></span>
<span id="cb271-58"><a href="example.html#cb271-58" tabindex="-1"></a>lgfc_FDR_vol <span class="ot">&lt;-</span> <span class="fu">get_volcano</span>(</span>
<span id="cb271-59"><a href="example.html#cb271-59" tabindex="-1"></a>  <span class="at">inputdata =</span> m_results,</span>
<span id="cb271-60"><a href="example.html#cb271-60" tabindex="-1"></a>  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;NR&quot;</span>, <span class="st">&quot;PR&quot;</span>),</span>
<span id="cb271-61"><a href="example.html#cb271-61" tabindex="-1"></a>  <span class="at">group_labels =</span> <span class="fu">c</span>(<span class="st">&quot;NR&quot;</span>, <span class="st">&quot;PR&quot;</span>),</span>
<span id="cb271-62"><a href="example.html#cb271-62" tabindex="-1"></a>  <span class="at">group_colors =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb271-63"><a href="example.html#cb271-63" tabindex="-1"></a>  <span class="at">x_index =</span> <span class="st">&quot;Log2FoldChange&quot;</span>,</span>
<span id="cb271-64"><a href="example.html#cb271-64" tabindex="-1"></a>  <span class="at">x_cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb271-65"><a href="example.html#cb271-65" tabindex="-1"></a>  <span class="at">y_index =</span> <span class="st">&quot;AdjustedPvalue&quot;</span>,</span>
<span id="cb271-66"><a href="example.html#cb271-66" tabindex="-1"></a>  <span class="at">y_cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb271-67"><a href="example.html#cb271-67" tabindex="-1"></a>  <span class="at">plot =</span> <span class="cn">FALSE</span>)</span>
<span id="cb271-68"><a href="example.html#cb271-68" tabindex="-1"></a></span>
<span id="cb271-69"><a href="example.html#cb271-69" tabindex="-1"></a>lgfc_FDR_vol<span class="sc">$</span>figure</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-190-1.png" width="672" /></p>
</div>
<div id="t-test-1" class="section level3 hasAnchor" number="6.6.6">
<h3><span class="header-section-number">6.6.6</span> T Test<a href="example.html#t-test-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb272"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb272-1"><a href="example.html#cb272-1" tabindex="-1"></a>group_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;NR&quot;</span>, <span class="st">&quot;PR&quot;</span>)</span>
<span id="cb272-2"><a href="example.html#cb272-2" tabindex="-1"></a></span>
<span id="cb272-3"><a href="example.html#cb272-3" tabindex="-1"></a>se_normalize_subset <span class="ot">&lt;-</span> se_normalize[, se_normalize<span class="sc">$</span>group <span class="sc">%in%</span> group_names]</span>
<span id="cb272-4"><a href="example.html#cb272-4" tabindex="-1"></a>se_normalize_subset<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">as.character</span>(se_normalize_subset<span class="sc">$</span>group))</span>
<span id="cb272-5"><a href="example.html#cb272-5" tabindex="-1"></a>ttest_res <span class="ot">&lt;-</span> <span class="fu">PomaUnivariate</span>(se_normalize_subset, <span class="at">method =</span> <span class="st">&quot;ttest&quot;</span>)</span>
<span id="cb272-6"><a href="example.html#cb272-6" tabindex="-1"></a><span class="fu">head</span>(ttest_res)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 9
##   feature       FC diff_means pvalue pvalueAdj mean_NR mean_PR sd_NR sd_PR
##   &lt;chr&gt;      &lt;dbl&gt;      &lt;dbl&gt;  &lt;dbl&gt;     &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;
## 1 MEDL00066  0.215      0.201  0.454     0.828 -0.255  -0.0548 0.749 0.994
## 2 MEDL00356  0.309      0.151  0.510     0.848 -0.218  -0.0675 0.665 0.830
## 3 MEDL00369 -0.936     -0.131  0.549     0.864  0.0676 -0.0633 0.626 0.800
## 4 MEDL00375 -0.436      0.151  0.503     0.845 -0.105   0.0459 0.838 0.629
## 5 MEDL00392 -5.65      -0.279  0.244     0.729  0.0419 -0.237  0.811 0.751
## 6 MEDL00401  1.81       0.024  0.801     0.939  0.0295  0.0533 0.266 0.350</code></pre>
</div>
<div id="volcano-plot-1" class="section level3 hasAnchor" number="6.6.7">
<h3><span class="header-section-number">6.6.7</span> Volcano plot<a href="example.html#volcano-plot-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb274"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb274-1"><a href="example.html#cb274-1" tabindex="-1"></a>se_impute_subset <span class="ot">&lt;-</span> se_impute[, se_impute<span class="sc">$</span>group <span class="sc">%in%</span> group_names]</span>
<span id="cb274-2"><a href="example.html#cb274-2" tabindex="-1"></a>se_impute_subset<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">as.character</span>(se_impute_subset<span class="sc">$</span>group))</span>
<span id="cb274-3"><a href="example.html#cb274-3" tabindex="-1"></a></span>
<span id="cb274-4"><a href="example.html#cb274-4" tabindex="-1"></a><span class="fu">PomaVolcano</span>(se_impute_subset, </span>
<span id="cb274-5"><a href="example.html#cb274-5" tabindex="-1"></a>            <span class="at">pval =</span> <span class="st">&quot;raw&quot;</span>,</span>
<span id="cb274-6"><a href="example.html#cb274-6" tabindex="-1"></a>            <span class="at">pval_cutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb274-7"><a href="example.html#cb274-7" tabindex="-1"></a>            <span class="at">log2FC =</span> <span class="fl">0.5</span>,</span>
<span id="cb274-8"><a href="example.html#cb274-8" tabindex="-1"></a>            <span class="at">xlim =</span> <span class="dv">3</span>,</span>
<span id="cb274-9"><a href="example.html#cb274-9" tabindex="-1"></a>            <span class="at">labels =</span> <span class="cn">TRUE</span>,</span>
<span id="cb274-10"><a href="example.html#cb274-10" tabindex="-1"></a>            <span class="at">plot_title =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-192-1.png" width="576" /></p>
</div>
</div>
<div id="feature-selection" class="section level2 hasAnchor" number="6.7">
<h2><span class="header-section-number">6.7</span> Feature Selection<a href="example.html#feature-selection" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="regularized-generalized-linear-models-lasso-alpha-1" class="section level3 hasAnchor" number="6.7.1">
<h3><span class="header-section-number">6.7.1</span> Regularized Generalized Linear Models (Lasso: alpha = 1)<a href="example.html#regularized-generalized-linear-models-lasso-alpha-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb275"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb275-1"><a href="example.html#cb275-1" tabindex="-1"></a>lasso_res <span class="ot">&lt;-</span> <span class="fu">PomaLasso</span>(se_normalize_subset, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">labels =</span> <span class="cn">TRUE</span>)</span>
<span id="cb275-2"><a href="example.html#cb275-2" tabindex="-1"></a></span>
<span id="cb275-3"><a href="example.html#cb275-3" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(lasso_res<span class="sc">$</span>cvLassoPlot,</span>
<span id="cb275-4"><a href="example.html#cb275-4" tabindex="-1"></a>                   lasso_res<span class="sc">$</span>coefficientPlot,</span>
<span id="cb275-5"><a href="example.html#cb275-5" tabindex="-1"></a>                   <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">align =</span> <span class="st">&quot;h&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-193-1.png" width="1152" /></p>
<div class="sourceCode" id="cb276"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb276-1"><a href="example.html#cb276-1" tabindex="-1"></a>lasso_res<span class="sc">$</span>coefficients</span></code></pre></div>
<pre><code>## # A tibble: 22 × 2
##    feature     coefficient
##    &lt;chr&gt;             &lt;dbl&gt;
##  1 (Intercept)     0.00959
##  2 MEDN0366       -0.427  
##  3 MEDN0411       -0.0107 
##  4 MEDN0490       -0.316  
##  5 MEDN1135        0.485  
##  6 MEDN1285       -0.00470
##  7 MEDP0006        0.265  
##  8 MEDP1040        1.71   
##  9 MEDP1170        1.23   
## 10 MEDP1177        1.47   
## # ℹ 12 more rows</code></pre>
</div>
<div id="classification-random-forest" class="section level3 hasAnchor" number="6.7.2">
<h3><span class="header-section-number">6.7.2</span> Classification (Random Forest)<a href="example.html#classification-random-forest" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb278"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb278-1"><a href="example.html#cb278-1" tabindex="-1"></a>poma_rf <span class="ot">&lt;-</span> <span class="fu">PomaRandForest</span>(se_normalize_subset, <span class="at">ntest =</span> <span class="dv">10</span>, <span class="at">nvar =</span> <span class="dv">10</span>)</span>
<span id="cb278-2"><a href="example.html#cb278-2" tabindex="-1"></a>poma_rf<span class="sc">$</span>error_tree</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-194-1.png" width="576" /></p>
<ul>
<li>table</li>
</ul>
<div class="sourceCode" id="cb279"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb279-1"><a href="example.html#cb279-1" tabindex="-1"></a>poma_rf<span class="sc">$</span>confusionMatrix<span class="sc">$</span>table</span></code></pre></div>
<pre><code>##           Reference
## Prediction 1 2
##          1 1 0
##          2 1 2</code></pre>
<ul>
<li>Important features</li>
</ul>
<div class="sourceCode" id="cb281"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb281-1"><a href="example.html#cb281-1" tabindex="-1"></a>poma_rf<span class="sc">$</span>MeanDecreaseGini_plot</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-196-1.png" width="576" /></p>
</div>
</div>
<div id="network-analysis-2" class="section level2 hasAnchor" number="6.8">
<h2><span class="header-section-number">6.8</span> Network Analysis<a href="example.html#network-analysis-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="data-curation-2" class="section level3 hasAnchor" number="6.8.1">
<h3><span class="header-section-number">6.8.1</span> Data curation<a href="example.html#data-curation-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb282"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb282-1"><a href="example.html#cb282-1" tabindex="-1"></a>features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(se_filter) <span class="sc">%&gt;%</span> </span>
<span id="cb282-2"><a href="example.html#cb282-2" tabindex="-1"></a>  <span class="fu">t</span>()</span>
<span id="cb282-3"><a href="example.html#cb282-3" tabindex="-1"></a>features_tab[<span class="fu">is.na</span>(features_tab)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb282-4"><a href="example.html#cb282-4" tabindex="-1"></a></span>
<span id="cb282-5"><a href="example.html#cb282-5" tabindex="-1"></a><span class="fu">print</span>(features_tab[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>])</span></code></pre></div>
<pre><code>##        MEDL00066 MEDL00375 MEDL00392 MEDL00568 MEDL00587 MEDL01764 MEDL01799 MEDL01837 MEDL01844 MEDL01922
## CJY_V0     10447     31087 238370.00  10248.00    618200  113031.3  335795.7    801660    127250  309560.3
## CJY_V1   4011900     88306 260320.00  96595.00    563130  233660.0   51267.0    830010   4172100  107560.0
## DGL_V0    560560    449150 250350.00  78849.00   1079300  109130.2  304040.0    739880  19462000   39149.0
## DGL_V1    149120    263050  21585.00  44336.38   2047500  277960.0   24342.0    866470  14373000  964600.0
## HXZ_V0    442540    216830 480800.00 134607.77   1399700 2022600.0 1481800.0    842220   7830000  269570.0
## HXZ_V1    181010     37280  91366.73  69526.92   1171700   12301.0  166435.5    735560   5171400  196770.0</code></pre>
</div>
<div id="building-network-model-4" class="section level3 hasAnchor" number="6.8.2">
<h3><span class="header-section-number">6.8.2</span> Building network model<a href="example.html#building-network-model-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb284"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb284-1"><a href="example.html#cb284-1" tabindex="-1"></a>net_single <span class="ot">&lt;-</span> <span class="fu">netConstruct</span>(features_tab, </span>
<span id="cb284-2"><a href="example.html#cb284-2" tabindex="-1"></a>                           <span class="at">measure =</span> <span class="st">&quot;sparcc&quot;</span>,</span>
<span id="cb284-3"><a href="example.html#cb284-3" tabindex="-1"></a>                           <span class="at">measurePar =</span> <span class="fu">list</span>(<span class="at">iter =</span> <span class="dv">20</span>,</span>
<span id="cb284-4"><a href="example.html#cb284-4" tabindex="-1"></a>                                             <span class="at">inner_iter =</span> <span class="dv">10</span>,</span>
<span id="cb284-5"><a href="example.html#cb284-5" tabindex="-1"></a>                                             <span class="at">th =</span> <span class="fl">0.1</span>),</span>
<span id="cb284-6"><a href="example.html#cb284-6" tabindex="-1"></a>                           <span class="at">filtTax =</span> <span class="st">&quot;highestVar&quot;</span>,</span>
<span id="cb284-7"><a href="example.html#cb284-7" tabindex="-1"></a>                           <span class="at">filtTaxPar =</span> <span class="fu">list</span>(<span class="at">highestVar =</span> <span class="dv">50</span>),</span>
<span id="cb284-8"><a href="example.html#cb284-8" tabindex="-1"></a>                           <span class="at">filtSamp =</span> <span class="st">&quot;totalReads&quot;</span>,</span>
<span id="cb284-9"><a href="example.html#cb284-9" tabindex="-1"></a>                           <span class="at">filtSampPar =</span> <span class="fu">list</span>(<span class="at">totalReads =</span> <span class="dv">100</span>),</span>
<span id="cb284-10"><a href="example.html#cb284-10" tabindex="-1"></a>                           <span class="at">verbose =</span> <span class="dv">3</span>,</span>
<span id="cb284-11"><a href="example.html#cb284-11" tabindex="-1"></a>                           <span class="at">seed =</span> <span class="dv">123</span>)</span></code></pre></div>
</div>
<div id="visualizing-the-network-4" class="section level3 hasAnchor" number="6.8.3">
<h3><span class="header-section-number">6.8.3</span> Visualizing the network<a href="example.html#visualizing-the-network-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb285"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb285-1"><a href="example.html#cb285-1" tabindex="-1"></a>props_single <span class="ot">&lt;-</span> <span class="fu">netAnalyze</span>(net_single, <span class="at">clustMethod =</span> <span class="st">&quot;cluster_fast_greedy&quot;</span>)</span>
<span id="cb285-2"><a href="example.html#cb285-2" tabindex="-1"></a></span>
<span id="cb285-3"><a href="example.html#cb285-3" tabindex="-1"></a><span class="fu">plot</span>(props_single, </span>
<span id="cb285-4"><a href="example.html#cb285-4" tabindex="-1"></a>     <span class="at">nodeColor =</span> <span class="st">&quot;cluster&quot;</span>, </span>
<span id="cb285-5"><a href="example.html#cb285-5" tabindex="-1"></a>     <span class="at">nodeSize =</span> <span class="st">&quot;eigenvector&quot;</span>,</span>
<span id="cb285-6"><a href="example.html#cb285-6" tabindex="-1"></a>     <span class="at">repulsion =</span> <span class="fl">0.8</span>,</span>
<span id="cb285-7"><a href="example.html#cb285-7" tabindex="-1"></a>     <span class="at">rmSingles =</span> <span class="cn">TRUE</span>,</span>
<span id="cb285-8"><a href="example.html#cb285-8" tabindex="-1"></a>     <span class="at">labelScale =</span> <span class="cn">FALSE</span>,</span>
<span id="cb285-9"><a href="example.html#cb285-9" tabindex="-1"></a>     <span class="at">cexLabels =</span> <span class="fl">1.6</span>,</span>
<span id="cb285-10"><a href="example.html#cb285-10" tabindex="-1"></a>     <span class="at">nodeSizeSpread =</span> <span class="dv">3</span>,</span>
<span id="cb285-11"><a href="example.html#cb285-11" tabindex="-1"></a>     <span class="at">cexNodes =</span> <span class="dv">2</span>,</span>
<span id="cb285-12"><a href="example.html#cb285-12" tabindex="-1"></a>     <span class="at">title1 =</span> <span class="st">&quot;Network on metabolomics with Pearson correlations&quot;</span>, </span>
<span id="cb285-13"><a href="example.html#cb285-13" tabindex="-1"></a>     <span class="at">showTitle =</span> <span class="cn">TRUE</span>,</span>
<span id="cb285-14"><a href="example.html#cb285-14" tabindex="-1"></a>     <span class="at">cexTitle =</span> <span class="fl">1.5</span>)</span>
<span id="cb285-15"><a href="example.html#cb285-15" tabindex="-1"></a></span>
<span id="cb285-16"><a href="example.html#cb285-16" tabindex="-1"></a><span class="fu">legend</span>(<span class="fl">0.7</span>, <span class="fl">1.1</span>, <span class="at">cex =</span> <span class="fl">1.2</span>, <span class="at">title =</span> <span class="st">&quot;estimated correlation:&quot;</span>,</span>
<span id="cb285-17"><a href="example.html#cb285-17" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;+&quot;</span>,<span class="st">&quot;-&quot;</span>), <span class="at">lty =</span> <span class="dv">1</span>, <span class="at">lwd =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;#009900&quot;</span>,<span class="st">&quot;red&quot;</span>),</span>
<span id="cb285-18"><a href="example.html#cb285-18" tabindex="-1"></a>       <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">horiz =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-199-1.png" width="1152" /></p>
</div>
</div>
<div id="network-analysis-by-wgcna" class="section level2 hasAnchor" number="6.9">
<h2><span class="header-section-number">6.9</span> Network Analysis by WGCNA<a href="example.html#network-analysis-by-wgcna" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Performing Network Analysis step by step through <strong>WGCNA</strong> R package.</p>
<div id="data-curation-3" class="section level3 hasAnchor" number="6.9.1">
<h3><span class="header-section-number">6.9.1</span> Data curation<a href="example.html#data-curation-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><p>Data Matrix</p>
<ul>
<li><p>Row -&gt; metabolites</p></li>
<li><p>Column -&gt; samples</p></li>
</ul></li>
</ul>
<div class="sourceCode" id="cb286"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb286-1"><a href="example.html#cb286-1" tabindex="-1"></a>features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(se_impute)</span>
<span id="cb286-2"><a href="example.html#cb286-2" tabindex="-1"></a><span class="fu">print</span>(features_tab[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>])</span></code></pre></div>
<pre><code>##              CJY_V0     CJY_V1 DGL_V0    DGL_V1 HXZ_V0    HXZ_V1    LBC_V0    LBC_V1   LHC_V1    LHC_V0
## MEDL00066  10447.00 4011900.00 560560 149120.00 442540 181010.00   12620.0 173400.00  97848.0  7682.475
## MEDL00356  50530.92   38175.58  71717 128990.00 162090 106800.00 8524300.0  61881.00  88503.0 21030.000
## MEDL00369  40739.79   85573.00  36339  50110.28 135600  26164.00    7287.5  11808.00  86864.6 34776.883
## MEDL00375  31087.00   88306.00 449150 263050.00 216830  37280.00  302150.0 607850.00  87101.0  7595.700
## MEDL00392 238370.00  260320.00 250350  21585.00 480800  91366.73  225860.0  53617.11 809940.0  9993.000
## MEDL00401  12589.00   12590.00  15954  16425.00  14961   8432.20   16545.0   9618.60  20929.0 12400.000</code></pre>
<ul>
<li>Data normalization</li>
</ul>
<div class="sourceCode" id="cb288"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb288-1"><a href="example.html#cb288-1" tabindex="-1"></a><span class="co"># TSS</span></span>
<span id="cb288-2"><a href="example.html#cb288-2" tabindex="-1"></a>features_tab_norm <span class="ot">&lt;-</span> XMAS2<span class="sc">::</span><span class="fu">norm_tss</span>(phyloseq<span class="sc">::</span><span class="fu">otu_table</span>(features_tab, <span class="at">taxa_are_rows =</span> T)) <span class="sc">%&gt;%</span> </span>
<span id="cb288-3"><a href="example.html#cb288-3" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">t</span>()</span>
<span id="cb288-4"><a href="example.html#cb288-4" tabindex="-1"></a><span class="fu">print</span>(features_tab_norm[<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>])</span></code></pre></div>
<pre><code>##           MEDL00066    MEDL00356    MEDL00369    MEDL00375    MEDL00392    MEDL00401    MEDL00404    MEDL00416    MEDL00568    MEDL00587
## CJY_V0 7.044680e-06 3.407429e-05 2.747189e-05 2.096276e-05 1.607390e-04 8.489085e-06 2.581116e-04 1.401990e-05 6.910489e-06 0.0004168681
## CJY_V1 1.930100e-03 1.836604e-05 4.116864e-05 4.248347e-05 1.252383e-04 6.056971e-06 9.460709e-06 7.009055e-05 4.647125e-05 0.0002709183
## DGL_V0 1.373244e-04 1.756903e-05 8.902225e-06 1.100315e-04 6.133003e-05 3.908366e-06 6.912275e-06 6.819184e-06 1.931620e-05 0.0002644038
## DGL_V1 3.576173e-05 3.093418e-05 1.201737e-05 6.308424e-05 5.176481e-06 3.939018e-06 1.145229e-05 1.549515e-05 1.063268e-05 0.0004910283
## HXZ_V0 1.046211e-04 3.831977e-05 3.205725e-05 5.126087e-05 1.136661e-04 3.536936e-06 2.507133e-05 6.972689e-06 3.182268e-05 0.0003309037
## HXZ_V1 5.993074e-05 3.536049e-05 8.662658e-06 1.234306e-05 3.025068e-05 2.791823e-06 2.311676e-06 3.888661e-05 2.301972e-05 0.0003879390</code></pre>
<div class="sourceCode" id="cb290"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb290-1"><a href="example.html#cb290-1" tabindex="-1"></a><span class="co"># # CSS</span></span>
<span id="cb290-2"><a href="example.html#cb290-2" tabindex="-1"></a><span class="co"># features_tab_norm &lt;- XMAS2::norm_css(phyloseq::otu_table(features_tab, taxa_are_rows = T)) %&gt;% </span></span>
<span id="cb290-3"><a href="example.html#cb290-3" tabindex="-1"></a><span class="co">#   data.frame() %&gt;% t()</span></span>
<span id="cb290-4"><a href="example.html#cb290-4" tabindex="-1"></a><span class="co"># print(features_tab_norm[1:6, 1:10])</span></span></code></pre></div>
</div>
<div id="tuning-soft-thresholds" class="section level3 hasAnchor" number="6.9.2">
<h3><span class="header-section-number">6.9.2</span> Tuning soft thresholds<a href="example.html#tuning-soft-thresholds" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Picking a threshhold value (if correlation is below threshold, remove the edge). WGCNA will try a range of soft thresholds and create a diagnostic plot.</p>
<ul>
<li>Choose a set of soft-thresholding powers</li>
</ul>
<div class="sourceCode" id="cb291"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb291-1"><a href="example.html#cb291-1" tabindex="-1"></a>powers <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>), <span class="fu">seq</span>(<span class="dv">12</span>, <span class="dv">20</span>, <span class="dv">2</span>))</span></code></pre></div>
<ul>
<li><p>Call the network topology analysis function</p>
<ul>
<li><p>Row -&gt; samples</p></li>
<li><p>Column -&gt; metabolites</p></li>
</ul></li>
</ul>
<div class="sourceCode" id="cb292"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb292-1"><a href="example.html#cb292-1" tabindex="-1"></a>sft <span class="ot">&lt;-</span> <span class="fu">pickSoftThreshold</span>(</span>
<span id="cb292-2"><a href="example.html#cb292-2" tabindex="-1"></a>  features_tab_norm,</span>
<span id="cb292-3"><a href="example.html#cb292-3" tabindex="-1"></a>  <span class="at">powerVector =</span> powers,</span>
<span id="cb292-4"><a href="example.html#cb292-4" tabindex="-1"></a>  <span class="at">networkType =</span> <span class="st">&quot;unsigned&quot;</span>,</span>
<span id="cb292-5"><a href="example.html#cb292-5" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="dv">2</span>)</span></code></pre></div>
<pre><code>## pickSoftThreshold: will use block size 978.
##  pickSoftThreshold: calculating connectivity for given powers...
##    ..working on genes 1 through 978 of 978
##    Power SFT.R.sq  slope truncated.R.sq mean.k. median.k. max.k.
## 1      1    0.236 -0.883          0.803  138.00   131.000  228.0
## 2      2    0.724 -1.560          0.655   40.30    32.500  116.0
## 3      3    0.820 -1.520          0.787   19.00    12.300   83.3
## 4      4    0.843 -1.420          0.834   11.70     6.020   66.8
## 5      5    0.872 -1.350          0.877    8.29     3.570   56.5
## 6      6    0.906 -1.290          0.921    6.39     2.420   49.3
## 7      7    0.919 -1.250          0.938    5.19     1.760   44.0
## 8      8    0.876 -1.240          0.887    4.36     1.320   39.8
## 9      9    0.880 -1.230          0.880    3.76     1.090   36.4
## 10    10    0.923 -1.210          0.942    3.30     0.934   33.6
## 11    12    0.944 -1.200          0.959    2.64     0.674   29.3
## 12    14    0.970 -1.200          0.982    2.20     0.507   26.0
## 13    16    0.917 -1.230          0.908    1.88     0.354   23.3
## 14    18    0.949 -1.240          0.945    1.63     0.261   21.1
## 15    20    0.948 -1.250          0.943    1.45     0.180   19.3</code></pre>
<ul>
<li>the optimal power value</li>
</ul>
<div class="sourceCode" id="cb294"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb294-1"><a href="example.html#cb294-1" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb294-2"><a href="example.html#cb294-2" tabindex="-1"></a>cex1 <span class="ot">=</span> <span class="fl">1.2</span></span>
<span id="cb294-3"><a href="example.html#cb294-3" tabindex="-1"></a></span>
<span id="cb294-4"><a href="example.html#cb294-4" tabindex="-1"></a><span class="fu">plot</span>(sft<span class="sc">$</span>fitIndices[, <span class="dv">1</span>],</span>
<span id="cb294-5"><a href="example.html#cb294-5" tabindex="-1"></a>     <span class="sc">-</span><span class="fu">sign</span>(sft<span class="sc">$</span>fitIndices[, <span class="dv">3</span>]) <span class="sc">*</span> sft<span class="sc">$</span>fitIndices[, <span class="dv">2</span>],</span>
<span id="cb294-6"><a href="example.html#cb294-6" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">&quot;Soft Threshold (power)&quot;</span>,</span>
<span id="cb294-7"><a href="example.html#cb294-7" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">&quot;Scale Free Topology Model Fit, signed R^2&quot;</span>,</span>
<span id="cb294-8"><a href="example.html#cb294-8" tabindex="-1"></a>     <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">&quot;Scale independence&quot;</span>)</span>
<span id="cb294-9"><a href="example.html#cb294-9" tabindex="-1"></a>)</span>
<span id="cb294-10"><a href="example.html#cb294-10" tabindex="-1"></a><span class="fu">text</span>(sft<span class="sc">$</span>fitIndices[, <span class="dv">1</span>],</span>
<span id="cb294-11"><a href="example.html#cb294-11" tabindex="-1"></a>     <span class="sc">-</span><span class="fu">sign</span>(sft<span class="sc">$</span>fitIndices[, <span class="dv">3</span>]) <span class="sc">*</span> sft<span class="sc">$</span>fitIndices[, <span class="dv">2</span>],</span>
<span id="cb294-12"><a href="example.html#cb294-12" tabindex="-1"></a>     <span class="at">labels =</span> powers, <span class="at">cex =</span> cex1, <span class="at">col =</span> <span class="st">&quot;red&quot;</span></span>
<span id="cb294-13"><a href="example.html#cb294-13" tabindex="-1"></a>)</span>
<span id="cb294-14"><a href="example.html#cb294-14" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="fl">0.90</span>, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>)</span>
<span id="cb294-15"><a href="example.html#cb294-15" tabindex="-1"></a><span class="fu">plot</span>(sft<span class="sc">$</span>fitIndices[, <span class="dv">1</span>],</span>
<span id="cb294-16"><a href="example.html#cb294-16" tabindex="-1"></a>     sft<span class="sc">$</span>fitIndices[, <span class="dv">5</span>],</span>
<span id="cb294-17"><a href="example.html#cb294-17" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">&quot;Soft Threshold (power)&quot;</span>,</span>
<span id="cb294-18"><a href="example.html#cb294-18" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">&quot;Mean Connectivity&quot;</span>,</span>
<span id="cb294-19"><a href="example.html#cb294-19" tabindex="-1"></a>     <span class="at">type =</span> <span class="st">&quot;n&quot;</span>,</span>
<span id="cb294-20"><a href="example.html#cb294-20" tabindex="-1"></a>     <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">&quot;Mean connectivity&quot;</span>)</span>
<span id="cb294-21"><a href="example.html#cb294-21" tabindex="-1"></a>)</span>
<span id="cb294-22"><a href="example.html#cb294-22" tabindex="-1"></a><span class="fu">text</span>(sft<span class="sc">$</span>fitIndices[, <span class="dv">1</span>],</span>
<span id="cb294-23"><a href="example.html#cb294-23" tabindex="-1"></a>     sft<span class="sc">$</span>fitIndices[, <span class="dv">5</span>],</span>
<span id="cb294-24"><a href="example.html#cb294-24" tabindex="-1"></a>     <span class="at">labels =</span> powers,</span>
<span id="cb294-25"><a href="example.html#cb294-25" tabindex="-1"></a>     <span class="at">cex =</span> cex1, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-204-1.png" width="768" /></p>
<p><strong>Notice</strong>: We’ pick <strong>5</strong> but feel free to experiment with other powers to see how it affects your results.</p>
</div>
<div id="create-the-network-using-the-blockwisemodules" class="section level3 hasAnchor" number="6.9.3">
<h3><span class="header-section-number">6.9.3</span> Create the network using the <code>blockwiseModules</code><a href="example.html#create-the-network-using-the-blockwisemodules" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>building network</li>
</ul>
<div class="sourceCode" id="cb295"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb295-1"><a href="example.html#cb295-1" tabindex="-1"></a>picked_power <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb295-2"><a href="example.html#cb295-2" tabindex="-1"></a><span class="co"># temp_cor &lt;- cor       </span></span>
<span id="cb295-3"><a href="example.html#cb295-3" tabindex="-1"></a><span class="co"># cor &lt;- WGCNA::cor </span></span>
<span id="cb295-4"><a href="example.html#cb295-4" tabindex="-1"></a>netwk <span class="ot">&lt;-</span> <span class="fu">blockwiseModules</span>(features_tab_norm,</span>
<span id="cb295-5"><a href="example.html#cb295-5" tabindex="-1"></a>                          </span>
<span id="cb295-6"><a href="example.html#cb295-6" tabindex="-1"></a>                          <span class="co"># == Adjacency Function ==</span></span>
<span id="cb295-7"><a href="example.html#cb295-7" tabindex="-1"></a>                          <span class="at">power =</span> picked_power,                <span class="co"># &lt;= power here</span></span>
<span id="cb295-8"><a href="example.html#cb295-8" tabindex="-1"></a>                          <span class="at">networkType =</span> <span class="st">&quot;signed&quot;</span>,</span>
<span id="cb295-9"><a href="example.html#cb295-9" tabindex="-1"></a>                          </span>
<span id="cb295-10"><a href="example.html#cb295-10" tabindex="-1"></a>                          <span class="co"># == Network construction arguments: correlation options</span></span>
<span id="cb295-11"><a href="example.html#cb295-11" tabindex="-1"></a>                          <span class="at">corType =</span> <span class="st">&quot;bicor&quot;</span>,</span>
<span id="cb295-12"><a href="example.html#cb295-12" tabindex="-1"></a>                          <span class="at">maxPOutliers =</span> <span class="fl">0.05</span>,</span>
<span id="cb295-13"><a href="example.html#cb295-13" tabindex="-1"></a></span>
<span id="cb295-14"><a href="example.html#cb295-14" tabindex="-1"></a>                          <span class="co"># == Tree and Block Options ==</span></span>
<span id="cb295-15"><a href="example.html#cb295-15" tabindex="-1"></a>                          <span class="at">deepSplit =</span> <span class="dv">2</span>,</span>
<span id="cb295-16"><a href="example.html#cb295-16" tabindex="-1"></a>                          <span class="at">pamRespectsDendro =</span> F,</span>
<span id="cb295-17"><a href="example.html#cb295-17" tabindex="-1"></a>                          <span class="at">minModuleSize =</span> <span class="dv">20</span>,</span>
<span id="cb295-18"><a href="example.html#cb295-18" tabindex="-1"></a>                          <span class="at">maxBlockSize =</span> <span class="dv">4000</span>,</span>
<span id="cb295-19"><a href="example.html#cb295-19" tabindex="-1"></a></span>
<span id="cb295-20"><a href="example.html#cb295-20" tabindex="-1"></a>                          <span class="co"># == Module Adjustments ==</span></span>
<span id="cb295-21"><a href="example.html#cb295-21" tabindex="-1"></a>                          <span class="at">reassignThreshold =</span> <span class="dv">0</span>,</span>
<span id="cb295-22"><a href="example.html#cb295-22" tabindex="-1"></a>                          <span class="at">mergeCutHeight =</span> <span class="fl">0.25</span>,</span>
<span id="cb295-23"><a href="example.html#cb295-23" tabindex="-1"></a></span>
<span id="cb295-24"><a href="example.html#cb295-24" tabindex="-1"></a>                          <span class="co"># == TOM == Archive the run results in TOM file (saves time)</span></span>
<span id="cb295-25"><a href="example.html#cb295-25" tabindex="-1"></a>                          <span class="at">saveTOMs =</span> T,</span>
<span id="cb295-26"><a href="example.html#cb295-26" tabindex="-1"></a>                          <span class="at">saveTOMFileBase =</span> <span class="fu">paste0</span>(<span class="st">&quot;./dataset/&quot;</span>, <span class="st">&quot;GvHD&quot;</span>),</span>
<span id="cb295-27"><a href="example.html#cb295-27" tabindex="-1"></a></span>
<span id="cb295-28"><a href="example.html#cb295-28" tabindex="-1"></a>                          <span class="co"># == Output Options</span></span>
<span id="cb295-29"><a href="example.html#cb295-29" tabindex="-1"></a>                          <span class="at">numericLabels =</span> T,</span>
<span id="cb295-30"><a href="example.html#cb295-30" tabindex="-1"></a>                          <span class="at">verbose =</span> <span class="dv">3</span>,</span>
<span id="cb295-31"><a href="example.html#cb295-31" tabindex="-1"></a>                          <span class="at">randomSeed =</span> <span class="dv">123</span>)</span></code></pre></div>
<pre><code>##  Calculating module eigengenes block-wise from all genes
##    Flagging genes and samples with too many missing values...
##     ..step 1
##  ..Working on block 1 .
##     TOM calculation: adjacency..
##     ..will not use multithreading.
##      Fraction of slow calculations: 0.000000
##     ..connectivity..
##     ..matrix multiplication (system BLAS)..
##     ..normalization..
##     ..done.
##    ..saving TOM for block 1 into file ./dataset/GvHD-block.1.RData
##  ....clustering..
##  ....detecting modules..
##  ....calculating module eigengenes..
##  ....checking kME in modules..
##      ..removing 19 genes from module 1 because their KME is too low.
##      ..removing 113 genes from module 2 because their KME is too low.
##      ..removing 32 genes from module 3 because their KME is too low.
##      ..removing 45 genes from module 4 because their KME is too low.
##      ..removing 4 genes from module 5 because their KME is too low.
##      ..removing 17 genes from module 6 because their KME is too low.
##      ..removing 1 genes from module 7 because their KME is too low.
##  ..merging modules that are too close..
##      mergeCloseModules: Merging modules whose distance is less than 0.25
##        Calculating new MEs...</code></pre>
<ul>
<li>Modules’ number</li>
</ul>
<div class="sourceCode" id="cb297"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb297-1"><a href="example.html#cb297-1" tabindex="-1"></a><span class="fu">table</span>(netwk<span class="sc">$</span>colors)</span></code></pre></div>
<pre><code>## 
##   0   1   2   3   4   5   6   7 
## 231 224 139  96  86  73  71  58</code></pre>
<ul>
<li>hubs</li>
</ul>
<div class="sourceCode" id="cb299"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb299-1"><a href="example.html#cb299-1" tabindex="-1"></a><span class="fu">rownames</span>(netwk<span class="sc">$</span>MEs) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(features_tab_norm)</span>
<span id="cb299-2"><a href="example.html#cb299-2" tabindex="-1"></a><span class="fu">names</span>(netwk<span class="sc">$</span>colors) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(features_tab_norm)</span>
<span id="cb299-3"><a href="example.html#cb299-3" tabindex="-1"></a><span class="fu">names</span>(netwk<span class="sc">$</span>unmergedColors) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(features_tab_norm)</span>
<span id="cb299-4"><a href="example.html#cb299-4" tabindex="-1"></a></span>
<span id="cb299-5"><a href="example.html#cb299-5" tabindex="-1"></a>hubs <span class="ot">&lt;-</span> <span class="fu">chooseTopHubInEachModule</span>(features_tab_norm, netwk<span class="sc">$</span>colors)</span>
<span id="cb299-6"><a href="example.html#cb299-6" tabindex="-1"></a></span>
<span id="cb299-7"><a href="example.html#cb299-7" tabindex="-1"></a>hubs</span></code></pre></div>
<pre><code>##           0           1           2           3           4           5           6           7 
## &quot;MW0146102&quot; &quot;MW0155806&quot; &quot;MEDL02751&quot;  &quot;MEDP1201&quot; &quot;MW0145567&quot;  &quot;MEDP1033&quot; &quot;MW0156874&quot;  &quot;MEDP0434&quot;</code></pre>
<ul>
<li>The number of features per module</li>
</ul>
<div class="sourceCode" id="cb301"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb301-1"><a href="example.html#cb301-1" tabindex="-1"></a><span class="fu">table</span>(netwk<span class="sc">$</span>colors) <span class="sc">%&gt;%</span> </span>
<span id="cb301-2"><a href="example.html#cb301-2" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb301-3"><a href="example.html#cb301-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Module =</span> Var1, <span class="at">Number =</span> Freq) <span class="sc">%&gt;%</span> </span>
<span id="cb301-4"><a href="example.html#cb301-4" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Module_color =</span> <span class="fu">labels2colors</span>(<span class="fu">as.numeric</span>(<span class="fu">as.character</span>(Module)))) <span class="sc">%&gt;%</span> </span>
<span id="cb301-5"><a href="example.html#cb301-5" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Module, <span class="at">y =</span> Number, <span class="at">fill =</span> Module)) <span class="sc">+</span></span>
<span id="cb301-6"><a href="example.html#cb301-6" tabindex="-1"></a>      <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> Number), <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.18</span>, <span class="at">size =</span> <span class="fl">3.5</span>) <span class="sc">+</span></span>
<span id="cb301-7"><a href="example.html#cb301-7" tabindex="-1"></a>      <span class="fu">geom_col</span>(<span class="at">color =</span> <span class="st">&quot;#000000&quot;</span>) <span class="sc">+</span></span>
<span id="cb301-8"><a href="example.html#cb301-8" tabindex="-1"></a>      <span class="fu">ggtitle</span>(<span class="st">&quot;Number of features per module&quot;</span>) <span class="sc">+</span></span>
<span id="cb301-9"><a href="example.html#cb301-9" tabindex="-1"></a>      <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb301-10"><a href="example.html#cb301-10" tabindex="-1"></a>      <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb301-11"><a href="example.html#cb301-11" tabindex="-1"></a>      <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb301-12"><a href="example.html#cb301-12" tabindex="-1"></a>      <span class="fu">theme</span>(<span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="st">&quot;pt&quot;</span>),</span>
<span id="cb301-13"><a href="example.html#cb301-13" tabindex="-1"></a>            <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>),</span>
<span id="cb301-14"><a href="example.html#cb301-14" tabindex="-1"></a>            <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-208-1.png" width="768" /></p>
</div>
<div id="plot-modules" class="section level3 hasAnchor" number="6.9.4">
<h3><span class="header-section-number">6.9.4</span> plot modules<a href="example.html#plot-modules" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb302"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb302-1"><a href="example.html#cb302-1" tabindex="-1"></a>mergedColors <span class="ot">&lt;-</span> <span class="fu">labels2colors</span>(netwk<span class="sc">$</span>colors)</span>
<span id="cb302-2"><a href="example.html#cb302-2" tabindex="-1"></a></span>
<span id="cb302-3"><a href="example.html#cb302-3" tabindex="-1"></a><span class="fu">plotDendroAndColors</span>(</span>
<span id="cb302-4"><a href="example.html#cb302-4" tabindex="-1"></a>  netwk<span class="sc">$</span>dendrograms[[<span class="dv">1</span>]],</span>
<span id="cb302-5"><a href="example.html#cb302-5" tabindex="-1"></a>  mergedColors[netwk<span class="sc">$</span>blockGenes[[<span class="dv">1</span>]]],</span>
<span id="cb302-6"><a href="example.html#cb302-6" tabindex="-1"></a>  <span class="st">&quot;Module colors&quot;</span>,</span>
<span id="cb302-7"><a href="example.html#cb302-7" tabindex="-1"></a>  <span class="at">dendroLabels =</span> <span class="cn">FALSE</span>,</span>
<span id="cb302-8"><a href="example.html#cb302-8" tabindex="-1"></a>  <span class="at">hang =</span> <span class="fl">0.03</span>,</span>
<span id="cb302-9"><a href="example.html#cb302-9" tabindex="-1"></a>  <span class="at">addGuide =</span> <span class="cn">TRUE</span>,</span>
<span id="cb302-10"><a href="example.html#cb302-10" tabindex="-1"></a>  <span class="at">guideHang =</span> <span class="fl">0.05</span> )</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-209-1.png" width="768" /></p>
</div>
<div id="relationships-among-modules" class="section level3 hasAnchor" number="6.9.5">
<h3><span class="header-section-number">6.9.5</span> Relationships among modules<a href="example.html#relationships-among-modules" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb303"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb303-1"><a href="example.html#cb303-1" tabindex="-1"></a><span class="fu">plotEigengeneNetworks</span>(netwk<span class="sc">$</span>MEs, </span>
<span id="cb303-2"><a href="example.html#cb303-2" tabindex="-1"></a>                      <span class="st">&quot;Eigengene adjacency heatmap&quot;</span>,</span>
<span id="cb303-3"><a href="example.html#cb303-3" tabindex="-1"></a>                      <span class="at">marDendro =</span> <span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">2</span>, <span class="dv">4</span>),</span>
<span id="cb303-4"><a href="example.html#cb303-4" tabindex="-1"></a>                      <span class="at">marHeatmap =</span> <span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">2</span>, <span class="dv">2</span>), </span>
<span id="cb303-5"><a href="example.html#cb303-5" tabindex="-1"></a>                      <span class="at">plotDendrograms =</span> T,</span>
<span id="cb303-6"><a href="example.html#cb303-6" tabindex="-1"></a>                      <span class="at">xLabelsAngle =</span> <span class="dv">90</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-210-1.png" width="768" /></p>
</div>
<div id="module-eigengene-correlation" class="section level3 hasAnchor" number="6.9.6">
<h3><span class="header-section-number">6.9.6</span> Module (Eigengene) correlation<a href="example.html#module-eigengene-correlation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb304"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb304-1"><a href="example.html#cb304-1" tabindex="-1"></a>MEs <span class="ot">&lt;-</span> netwk<span class="sc">$</span>MEs</span>
<span id="cb304-2"><a href="example.html#cb304-2" tabindex="-1"></a>MEs_R <span class="ot">&lt;-</span> <span class="fu">bicor</span>(MEs, MEs, <span class="at">maxPOutliers =</span> <span class="fl">0.05</span>)</span>
<span id="cb304-3"><a href="example.html#cb304-3" tabindex="-1"></a></span>
<span id="cb304-4"><a href="example.html#cb304-4" tabindex="-1"></a>idx.r <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">rownames</span>(MEs_R) <span class="sc">==</span> <span class="st">&quot;ME0&quot;</span>)</span>
<span id="cb304-5"><a href="example.html#cb304-5" tabindex="-1"></a>idx.c <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">colnames</span>(MEs_R) <span class="sc">==</span> <span class="st">&quot;ME0&quot;</span>)</span>
<span id="cb304-6"><a href="example.html#cb304-6" tabindex="-1"></a></span>
<span id="cb304-7"><a href="example.html#cb304-7" tabindex="-1"></a>MEs_R_noME0 <span class="ot">&lt;-</span> MEs_R[<span class="sc">-</span>idx.r, <span class="sc">-</span>idx.c]</span>
<span id="cb304-8"><a href="example.html#cb304-8" tabindex="-1"></a></span>
<span id="cb304-9"><a href="example.html#cb304-9" tabindex="-1"></a>MEs_R_density <span class="ot">&lt;-</span> MEs_R[<span class="fu">upper.tri</span>(MEs_R_noME0)] <span class="sc">%&gt;%</span> </span>
<span id="cb304-10"><a href="example.html#cb304-10" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb304-11"><a href="example.html#cb304-11" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;correlation&quot;</span> <span class="ot">=</span> <span class="st">&quot;.&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb304-12"><a href="example.html#cb304-12" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>correlation)) <span class="sc">+</span> </span>
<span id="cb304-13"><a href="example.html#cb304-13" tabindex="-1"></a>  <span class="fu">geom_density</span>() <span class="sc">+</span> </span>
<span id="cb304-14"><a href="example.html#cb304-14" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(<span class="st">&quot;ME correlation density</span><span class="sc">\n</span><span class="st"> without &quot;</span>, <span class="st">&quot;ME0&quot;</span>))</span>
<span id="cb304-15"><a href="example.html#cb304-15" tabindex="-1"></a></span>
<span id="cb304-16"><a href="example.html#cb304-16" tabindex="-1"></a>MEs_R_Corr <span class="ot">&lt;-</span> pheatmap<span class="sc">::</span><span class="fu">pheatmap</span>(MEs_R, <span class="at">color =</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">&quot;Blue&quot;</span>, <span class="st">&quot;White&quot;</span>, <span class="st">&quot;Red&quot;</span>))(<span class="dv">100</span>),</span>
<span id="cb304-17"><a href="example.html#cb304-17" tabindex="-1"></a>                   <span class="at">silent =</span> T, </span>
<span id="cb304-18"><a href="example.html#cb304-18" tabindex="-1"></a>                   <span class="at">breaks =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>,<span class="at">length.out =</span> <span class="dv">101</span>),</span>
<span id="cb304-19"><a href="example.html#cb304-19" tabindex="-1"></a>                   <span class="at">treeheight_row =</span> <span class="dv">5</span>, </span>
<span id="cb304-20"><a href="example.html#cb304-20" tabindex="-1"></a>                   <span class="at">treeheight_col =</span> <span class="dv">5</span>,</span>
<span id="cb304-21"><a href="example.html#cb304-21" tabindex="-1"></a>                   <span class="at">main =</span> <span class="fu">paste0</span>(<span class="st">&quot;ME correlation heatmap&quot;</span>),</span>
<span id="cb304-22"><a href="example.html#cb304-22" tabindex="-1"></a>                   <span class="at">labels_row =</span> <span class="fu">rownames</span>(MEs_R),</span>
<span id="cb304-23"><a href="example.html#cb304-23" tabindex="-1"></a>                   <span class="at">labels_col =</span>  <span class="fu">colnames</span>(MEs_R)) </span>
<span id="cb304-24"><a href="example.html#cb304-24" tabindex="-1"></a></span>
<span id="cb304-25"><a href="example.html#cb304-25" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(MEs_R_density, MEs_R_Corr<span class="sc">$</span>gtable, </span>
<span id="cb304-26"><a href="example.html#cb304-26" tabindex="-1"></a>                   <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>), </span>
<span id="cb304-27"><a href="example.html#cb304-27" tabindex="-1"></a>                   <span class="at">label_size =</span> <span class="dv">15</span>, </span>
<span id="cb304-28"><a href="example.html#cb304-28" tabindex="-1"></a>                   <span class="at">rel_widths =</span> <span class="fu">c</span>(<span class="fl">0.6</span>, <span class="dv">1</span>),</span>
<span id="cb304-29"><a href="example.html#cb304-29" tabindex="-1"></a>                   <span class="at">align =</span> <span class="st">&quot;h&quot;</span>) </span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-211-1.png" width="576" /></p>
</div>
<div id="relate-module-cluster-assignments-to-groups" class="section level3 hasAnchor" number="6.9.7">
<h3><span class="header-section-number">6.9.7</span> Relate Module (cluster) Assignments to Groups<a href="example.html#relate-module-cluster-assignments-to-groups" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb305"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb305-1"><a href="example.html#cb305-1" tabindex="-1"></a>module_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb305-2"><a href="example.html#cb305-2" tabindex="-1"></a>  <span class="at">featureID =</span> <span class="fu">names</span>(netwk<span class="sc">$</span>colors),</span>
<span id="cb305-3"><a href="example.html#cb305-3" tabindex="-1"></a>  <span class="at">colors =</span> <span class="fu">labels2colors</span>(netwk<span class="sc">$</span>colors)</span>
<span id="cb305-4"><a href="example.html#cb305-4" tabindex="-1"></a>)</span>
<span id="cb305-5"><a href="example.html#cb305-5" tabindex="-1"></a></span>
<span id="cb305-6"><a href="example.html#cb305-6" tabindex="-1"></a><span class="co"># Get Module Eigengenes per cluster</span></span>
<span id="cb305-7"><a href="example.html#cb305-7" tabindex="-1"></a>MEs0 <span class="ot">&lt;-</span> <span class="fu">moduleEigengenes</span>(features_tab_norm, mergedColors)<span class="sc">$</span>eigengenes</span>
<span id="cb305-8"><a href="example.html#cb305-8" tabindex="-1"></a></span>
<span id="cb305-9"><a href="example.html#cb305-9" tabindex="-1"></a><span class="co"># Reorder modules so similar modules are next to each other</span></span>
<span id="cb305-10"><a href="example.html#cb305-10" tabindex="-1"></a>MEs0 <span class="ot">&lt;-</span> <span class="fu">orderMEs</span>(MEs0)</span>
<span id="cb305-11"><a href="example.html#cb305-11" tabindex="-1"></a>module_order <span class="ot">&lt;-</span> <span class="fu">names</span>(MEs0) <span class="sc">%&gt;%</span> <span class="fu">gsub</span>(<span class="st">&quot;ME&quot;</span>,<span class="st">&quot;&quot;</span>, .)</span>
<span id="cb305-12"><a href="example.html#cb305-12" tabindex="-1"></a></span>
<span id="cb305-13"><a href="example.html#cb305-13" tabindex="-1"></a><span class="co"># Add group names</span></span>
<span id="cb305-14"><a href="example.html#cb305-14" tabindex="-1"></a>MEs0<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">paste0</span>(se_impute<span class="sc">$</span>group, <span class="fu">rownames</span>(<span class="fu">colData</span>(se_impute)))  <span class="co"># row.names(MEs0) == rownames(colData(se_impute))</span></span>
<span id="cb305-15"><a href="example.html#cb305-15" tabindex="-1"></a></span>
<span id="cb305-16"><a href="example.html#cb305-16" tabindex="-1"></a><span class="co"># tidy &amp; plot data</span></span>
<span id="cb305-17"><a href="example.html#cb305-17" tabindex="-1"></a>mME <span class="ot">&lt;-</span> MEs0 <span class="sc">%&gt;%</span></span>
<span id="cb305-18"><a href="example.html#cb305-18" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(<span class="sc">-</span>group) <span class="sc">%&gt;%</span></span>
<span id="cb305-19"><a href="example.html#cb305-19" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb305-20"><a href="example.html#cb305-20" tabindex="-1"></a>    <span class="at">name =</span> <span class="fu">gsub</span>(<span class="st">&quot;ME&quot;</span>, <span class="st">&quot;&quot;</span>, name),</span>
<span id="cb305-21"><a href="example.html#cb305-21" tabindex="-1"></a>    <span class="at">name =</span> <span class="fu">factor</span>(name, <span class="at">levels =</span> module_order)</span>
<span id="cb305-22"><a href="example.html#cb305-22" tabindex="-1"></a>  )</span>
<span id="cb305-23"><a href="example.html#cb305-23" tabindex="-1"></a></span>
<span id="cb305-24"><a href="example.html#cb305-24" tabindex="-1"></a>mME <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x=</span>group, <span class="at">y=</span>name, <span class="at">fill=</span>value)) <span class="sc">+</span></span>
<span id="cb305-25"><a href="example.html#cb305-25" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb305-26"><a href="example.html#cb305-26" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Module-samples Relationships&quot;</span>, <span class="at">y =</span> <span class="st">&quot;Modules&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;corr&quot;</span>) <span class="sc">+</span>  </span>
<span id="cb305-27"><a href="example.html#cb305-27" tabindex="-1"></a>  <span class="fu">scale_fill_gradient2</span>(</span>
<span id="cb305-28"><a href="example.html#cb305-28" tabindex="-1"></a>    <span class="at">low =</span> <span class="st">&quot;blue&quot;</span>,</span>
<span id="cb305-29"><a href="example.html#cb305-29" tabindex="-1"></a>    <span class="at">high =</span> <span class="st">&quot;red&quot;</span>,</span>
<span id="cb305-30"><a href="example.html#cb305-30" tabindex="-1"></a>    <span class="at">mid =</span> <span class="st">&quot;white&quot;</span>,</span>
<span id="cb305-31"><a href="example.html#cb305-31" tabindex="-1"></a>    <span class="at">midpoint =</span> <span class="dv">0</span>,</span>
<span id="cb305-32"><a href="example.html#cb305-32" tabindex="-1"></a>    <span class="at">limit =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb305-33"><a href="example.html#cb305-33" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb305-34"><a href="example.html#cb305-34" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>))</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-212-1.png" width="768" /></p>
<p><strong>Result</strong>:</p>
<ol style="list-style-type: decimal">
<li>the <em>black</em> modules seems negatively associated (red shading) with the PR groups.</li>
</ol>
</div>
<div id="generate-and-export-networks" class="section level3 hasAnchor" number="6.9.8">
<h3><span class="header-section-number">6.9.8</span> Generate and Export Networks<a href="example.html#generate-and-export-networks" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb306"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb306-1"><a href="example.html#cb306-1" tabindex="-1"></a><span class="co"># modules_of_interest &lt;- c(&quot;green&quot;, &quot;brown&quot;, &quot;black&quot;)</span></span>
<span id="cb306-2"><a href="example.html#cb306-2" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb306-3"><a href="example.html#cb306-3" tabindex="-1"></a><span class="co"># genes_of_interest &lt;- module_df %&gt;%</span></span>
<span id="cb306-4"><a href="example.html#cb306-4" tabindex="-1"></a><span class="co">#   subset(colors %in% modules_of_interest)</span></span>
<span id="cb306-5"><a href="example.html#cb306-5" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb306-6"><a href="example.html#cb306-6" tabindex="-1"></a><span class="co"># expr_of_interest &lt;- features_tab_norm[, genes_of_interest$featureID]</span></span>
<span id="cb306-7"><a href="example.html#cb306-7" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb306-8"><a href="example.html#cb306-8" tabindex="-1"></a><span class="co"># # Only recalculate TOM for modules of interest </span></span>
<span id="cb306-9"><a href="example.html#cb306-9" tabindex="-1"></a><span class="co"># TOM &lt;- TOMsimilarityFromExpr(expr_of_interest,</span></span>
<span id="cb306-10"><a href="example.html#cb306-10" tabindex="-1"></a><span class="co">#                              power = picked_power)</span></span>
<span id="cb306-11"><a href="example.html#cb306-11" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb306-12"><a href="example.html#cb306-12" tabindex="-1"></a><span class="co"># # Add feature id to row and columns</span></span>
<span id="cb306-13"><a href="example.html#cb306-13" tabindex="-1"></a><span class="co"># rownames(TOM) &lt;- colnames(expr_of_interest)</span></span>
<span id="cb306-14"><a href="example.html#cb306-14" tabindex="-1"></a><span class="co"># colnames(TOM) &lt;- colnames(expr_of_interest)</span></span>
<span id="cb306-15"><a href="example.html#cb306-15" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb306-16"><a href="example.html#cb306-16" tabindex="-1"></a><span class="co"># edge_list &lt;- data.frame(TOM) %&gt;%</span></span>
<span id="cb306-17"><a href="example.html#cb306-17" tabindex="-1"></a><span class="co">#   tibble::rownames_to_column(&quot;featureID&quot;) %&gt;%</span></span>
<span id="cb306-18"><a href="example.html#cb306-18" tabindex="-1"></a><span class="co">#   tidyr::pivot_longer(-featureID) %&gt;%</span></span>
<span id="cb306-19"><a href="example.html#cb306-19" tabindex="-1"></a><span class="co">#   dplyr::rename(featureID2 = name, correlation = value) %&gt;%</span></span>
<span id="cb306-20"><a href="example.html#cb306-20" tabindex="-1"></a><span class="co">#   unique() %&gt;%</span></span>
<span id="cb306-21"><a href="example.html#cb306-21" tabindex="-1"></a><span class="co">#   subset(!(featureID == featureID2)) %&gt;%</span></span>
<span id="cb306-22"><a href="example.html#cb306-22" tabindex="-1"></a><span class="co">#   dplyr::mutate(</span></span>
<span id="cb306-23"><a href="example.html#cb306-23" tabindex="-1"></a><span class="co">#     module1 = module_df[featureID, ]$colors,</span></span>
<span id="cb306-24"><a href="example.html#cb306-24" tabindex="-1"></a><span class="co">#     module2 = module_df[featureID2, ]$colors)</span></span>
<span id="cb306-25"><a href="example.html#cb306-25" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb306-26"><a href="example.html#cb306-26" tabindex="-1"></a><span class="co"># head(edge_list)</span></span></code></pre></div>
</div>
<div id="network-visualization" class="section level3 hasAnchor" number="6.9.9">
<h3><span class="header-section-number">6.9.9</span> Network visualization<a href="example.html#network-visualization" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb307"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb307-1"><a href="example.html#cb307-1" tabindex="-1"></a><span class="fu">library</span>(igraph)</span>
<span id="cb307-2"><a href="example.html#cb307-2" tabindex="-1"></a></span>
<span id="cb307-3"><a href="example.html#cb307-3" tabindex="-1"></a>strength_adjust <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb307-4"><a href="example.html#cb307-4" tabindex="-1"></a>TOM <span class="ot">&lt;-</span> <span class="fu">TOMsimilarityFromExpr</span>(features_tab_norm,</span>
<span id="cb307-5"><a href="example.html#cb307-5" tabindex="-1"></a>                             <span class="at">power =</span> picked_power)</span></code></pre></div>
<pre><code>## TOM calculation: adjacency..
## ..will not use multithreading.
##  Fraction of slow calculations: 0.000000
## ..connectivity..
## ..matrix multiplication (system BLAS)..
## ..normalization..
## ..done.</code></pre>
<div class="sourceCode" id="cb309"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb309-1"><a href="example.html#cb309-1" tabindex="-1"></a><span class="co"># Add feature id to row and columns</span></span>
<span id="cb309-2"><a href="example.html#cb309-2" tabindex="-1"></a><span class="fu">rownames</span>(TOM) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(features_tab_norm)</span>
<span id="cb309-3"><a href="example.html#cb309-3" tabindex="-1"></a><span class="fu">colnames</span>(TOM) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(features_tab_norm)</span>
<span id="cb309-4"><a href="example.html#cb309-4" tabindex="-1"></a></span>
<span id="cb309-5"><a href="example.html#cb309-5" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">graph.adjacency</span>(TOM, <span class="at">mode=</span><span class="st">&quot;undirected&quot;</span>, <span class="at">weighted=</span> <span class="cn">TRUE</span>)</span>
<span id="cb309-6"><a href="example.html#cb309-6" tabindex="-1"></a><span class="fu">delete.edges</span>(g, <span class="fu">which</span>(<span class="fu">E</span>(g)<span class="sc">$</span>weight <span class="sc">&lt;</span><span class="dv">1</span>))</span></code></pre></div>
<pre><code>## IGRAPH d50076b UNW- 978 978 -- 
## + attr: name (v/c), weight (e/n)
## + edges from d50076b (vertex names):
##  [1] MEDL00066--MEDL00066 MEDL00356--MEDL00356 MEDL00369--MEDL00369 MEDL00375--MEDL00375 MEDL00392--MEDL00392 MEDL00401--MEDL00401
##  [7] MEDL00404--MEDL00404 MEDL00416--MEDL00416 MEDL00568--MEDL00568 MEDL00587--MEDL00587 MEDL01764--MEDL01764 MEDL01799--MEDL01799
## [13] MEDL01837--MEDL01837 MEDL01844--MEDL01844 MEDL01857--MEDL01857 MEDL01867--MEDL01867 MEDL01879--MEDL01879 MEDL01922--MEDL01922
## [19] MEDL01994--MEDL01994 MEDL01999--MEDL01999 MEDL02002--MEDL02002 MEDL02018--MEDL02018 MEDL02044--MEDL02044 MEDL02106--MEDL02106
## [25] MEDL02174--MEDL02174 MEDL02177--MEDL02177 MEDL02187--MEDL02187 MEDL02519--MEDL02519 MEDL02561--MEDL02561 MEDL02562--MEDL02562
## [31] MEDL02582--MEDL02582 MEDL02630--MEDL02630 MEDL02681--MEDL02681 MEDL02751--MEDL02751 MEDN0004 --MEDN0004  MEDN0005 --MEDN0005 
## [37] MEDN0006 --MEDN0006  MEDN0007 --MEDN0007  MEDN0010 --MEDN0010  MEDN0015 --MEDN0015  MEDN0018 --MEDN0018  MEDN0025 --MEDN0025 
## [43] MEDN0027 --MEDN0027  MEDN0028 --MEDN0028  MEDN0032 --MEDN0032  MEDN0036 --MEDN0036  MEDN0041 --MEDN0041  MEDN0042 --MEDN0042 
## + ... omitted several edges</code></pre>
<div class="sourceCode" id="cb311"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb311-1"><a href="example.html#cb311-1" tabindex="-1"></a><span class="fu">E</span>(g)<span class="sc">$</span>width <span class="ot">&lt;-</span> <span class="fu">E</span>(g)<span class="sc">$</span>weight<span class="sc">*</span>strength_adjust <span class="sc">+</span> <span class="fu">min</span>(<span class="fu">E</span>(g)<span class="sc">$</span>weight)</span>
<span id="cb311-2"><a href="example.html#cb311-2" tabindex="-1"></a><span class="fu">E</span>(g)<span class="sc">$</span>color <span class="ot">&lt;-</span> <span class="st">&quot;red&quot;</span></span>
<span id="cb311-3"><a href="example.html#cb311-3" tabindex="-1"></a></span>
<span id="cb311-4"><a href="example.html#cb311-4" tabindex="-1"></a><span class="fu">plot</span>(g)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-214-1.png" width="768" /></p>
</div>
</div>
<div id="systematic-information-2" class="section level2 hasAnchor" number="6.10">
<h2><span class="header-section-number">6.10</span> Systematic Information<a href="example.html#systematic-information-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb312"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb312-1"><a href="example.html#cb312-1" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 4.1.3 (2022-03-10)
##  os       macOS Monterey 12.2.1
##  system   x86_64, darwin17.0
##  ui       RStudio
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       Asia/Shanghai
##  date     2023-11-23
##  rstudio  2023.09.0+463 Desert Sunflower (desktop)
##  pandoc   3.1.1 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/ (via rmarkdown)
## 
## ─ Packages ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package              * version    date (UTC) lib source
##  abind                  1.4-5      2016-07-21 [2] CRAN (R 4.1.0)
##  ade4                   1.7-22     2023-02-06 [2] CRAN (R 4.1.2)
##  affy                   1.72.0     2021-10-26 [2] Bioconductor
##  affyio                 1.64.0     2021-10-26 [2] Bioconductor
##  annotate               1.72.0     2021-10-26 [2] Bioconductor
##  AnnotationDbi        * 1.60.2     2023-03-10 [2] Bioconductor
##  ape                    5.7-1      2023-03-13 [2] CRAN (R 4.1.2)
##  aplot                  0.1.10     2023-03-08 [2] CRAN (R 4.1.2)
##  attempt                0.3.1      2020-05-03 [2] CRAN (R 4.1.0)
##  backports              1.4.1      2021-12-13 [2] CRAN (R 4.1.0)
##  base64enc              0.1-3      2015-07-28 [2] CRAN (R 4.1.0)
##  Biobase              * 2.54.0     2021-10-26 [2] Bioconductor
##  BiocGenerics         * 0.40.0     2021-10-26 [2] Bioconductor
##  BiocManager            1.30.21    2023-06-10 [2] CRAN (R 4.1.3)
##  BiocParallel           1.28.3     2021-12-09 [2] Bioconductor
##  biomformat             1.22.0     2021-10-26 [2] Bioconductor
##  Biostrings             2.62.0     2021-10-26 [2] Bioconductor
##  bit                    4.0.5      2022-11-15 [2] CRAN (R 4.1.2)
##  bit64                  4.0.5      2020-08-30 [2] CRAN (R 4.1.0)
##  bitops                 1.0-7      2021-04-24 [2] CRAN (R 4.1.0)
##  blob                   1.2.4      2023-03-17 [2] CRAN (R 4.1.2)
##  bookdown               0.34       2023-05-09 [2] CRAN (R 4.1.2)
##  broom                  1.0.5      2023-06-09 [2] CRAN (R 4.1.3)
##  bslib                  0.5.0      2023-06-09 [2] CRAN (R 4.1.3)
##  cachem                 1.0.8      2023-05-01 [2] CRAN (R 4.1.2)
##  Cairo                  1.6-0      2022-07-05 [2] CRAN (R 4.1.2)
##  callr                  3.7.3      2022-11-02 [2] CRAN (R 4.1.2)
##  car                    3.1-2      2023-03-30 [2] CRAN (R 4.1.2)
##  carData                3.0-5      2022-01-06 [2] CRAN (R 4.1.2)
##  caret                  6.0-94     2023-03-21 [2] CRAN (R 4.1.2)
##  caTools                1.18.2     2021-03-28 [2] CRAN (R 4.1.0)
##  cellranger             1.1.0      2016-07-27 [2] CRAN (R 4.1.0)
##  checkmate              2.2.0      2023-04-27 [2] CRAN (R 4.1.2)
##  circlize               0.4.15     2022-05-10 [2] CRAN (R 4.1.2)
##  class                  7.3-22     2023-05-03 [2] CRAN (R 4.1.2)
##  cli                    3.6.1      2023-03-23 [2] CRAN (R 4.1.2)
##  clue                   0.3-64     2023-01-31 [2] CRAN (R 4.1.2)
##  cluster              * 2.1.4      2022-08-22 [2] CRAN (R 4.1.2)
##  clusterProfiler      * 4.2.2      2022-01-13 [2] Bioconductor
##  codetools              0.2-19     2023-02-01 [2] CRAN (R 4.1.2)
##  coin                   1.4-2      2021-10-08 [2] CRAN (R 4.1.0)
##  colorspace             2.1-0      2023-01-23 [2] CRAN (R 4.1.2)
##  ComplexHeatmap         2.10.0     2021-10-26 [2] Bioconductor
##  config                 0.3.1      2020-12-17 [2] CRAN (R 4.1.0)
##  corpcor                1.6.10     2021-09-16 [2] CRAN (R 4.1.0)
##  cowplot                1.1.1      2020-12-30 [2] CRAN (R 4.1.0)
##  crayon                 1.5.2      2022-09-29 [2] CRAN (R 4.1.2)
##  crmn                   0.0.21     2020-02-10 [2] CRAN (R 4.1.0)
##  curl                   5.0.1      2023-06-07 [2] CRAN (R 4.1.3)
##  data.table             1.14.8     2023-02-17 [2] CRAN (R 4.1.2)
##  DBI                    1.1.3      2022-06-18 [2] CRAN (R 4.1.2)
##  DelayedArray           0.20.0     2021-10-26 [2] Bioconductor
##  dendextend           * 1.17.1     2023-03-25 [2] CRAN (R 4.1.2)
##  DESeq2                 1.34.0     2021-10-26 [2] Bioconductor
##  devtools               2.4.5      2022-10-11 [2] CRAN (R 4.1.2)
##  digest                 0.6.33     2023-07-07 [1] CRAN (R 4.1.3)
##  DO.db                  2.9        2022-04-11 [2] Bioconductor
##  doParallel             1.0.17     2022-02-07 [2] CRAN (R 4.1.2)
##  doRNG                  1.8.6      2023-01-16 [2] CRAN (R 4.1.2)
##  DOSE                   3.20.1     2021-11-18 [2] Bioconductor
##  doSNOW                 1.0.20     2022-02-04 [2] CRAN (R 4.1.2)
##  downloader             0.4        2015-07-09 [2] CRAN (R 4.1.0)
##  dplyr                * 1.1.2      2023-04-20 [2] CRAN (R 4.1.2)
##  DT                     0.28       2023-05-18 [2] CRAN (R 4.1.3)
##  dynamicTreeCut       * 1.63-1     2016-03-11 [2] CRAN (R 4.1.0)
##  e1071                  1.7-13     2023-02-01 [2] CRAN (R 4.1.2)
##  edgeR                  3.36.0     2021-10-26 [2] Bioconductor
##  ellipse                0.4.5      2023-04-05 [2] CRAN (R 4.1.2)
##  ellipsis               0.3.2      2021-04-29 [2] CRAN (R 4.1.0)
##  enrichplot             1.14.2     2022-02-24 [2] Bioconductor
##  evaluate               0.21       2023-05-05 [2] CRAN (R 4.1.2)
##  factoextra           * 1.0.7      2020-04-01 [2] CRAN (R 4.1.0)
##  fansi                  1.0.4      2023-01-22 [2] CRAN (R 4.1.2)
##  farver                 2.1.1      2022-07-06 [2] CRAN (R 4.1.2)
##  fastcluster          * 1.2.3      2021-05-24 [2] CRAN (R 4.1.0)
##  fastmap                1.1.1      2023-02-24 [2] CRAN (R 4.1.2)
##  fastmatch              1.1-3      2021-07-23 [2] CRAN (R 4.1.0)
##  fdrtool                1.2.17     2021-11-13 [2] CRAN (R 4.1.0)
##  fgsea                  1.20.0     2021-10-26 [2] Bioconductor
##  filematrix             1.3        2018-02-27 [2] CRAN (R 4.1.0)
##  foreach                1.5.2      2022-02-02 [2] CRAN (R 4.1.2)
##  foreign                0.8-84     2022-12-06 [2] CRAN (R 4.1.2)
##  forestplot             3.1.1      2022-12-06 [2] CRAN (R 4.1.2)
##  Formula                1.2-5      2023-02-24 [2] CRAN (R 4.1.2)
##  fs                     1.6.2      2023-04-25 [2] CRAN (R 4.1.2)
##  furrr                  0.3.1      2022-08-15 [2] CRAN (R 4.1.2)
##  future                 1.33.0     2023-07-01 [2] CRAN (R 4.1.3)
##  future.apply           1.11.0     2023-05-21 [2] CRAN (R 4.1.3)
##  genefilter             1.76.0     2021-10-26 [2] Bioconductor
##  geneplotter            1.72.0     2021-10-26 [2] Bioconductor
##  generics               0.1.3      2022-07-05 [2] CRAN (R 4.1.2)
##  GenomeInfoDb         * 1.30.1     2022-01-30 [2] Bioconductor
##  GenomeInfoDbData       1.2.7      2022-03-09 [2] Bioconductor
##  GenomicRanges        * 1.46.1     2021-11-18 [2] Bioconductor
##  GetoptLong             1.0.5      2020-12-15 [2] CRAN (R 4.1.0)
##  ggforce                0.4.1      2022-10-04 [2] CRAN (R 4.1.2)
##  ggfun                  0.1.1      2023-06-24 [2] CRAN (R 4.1.3)
##  ggplot2              * 3.4.2      2023-04-03 [2] CRAN (R 4.1.2)
##  ggplotify              0.1.1      2023-06-27 [2] CRAN (R 4.1.3)
##  ggpubr                 0.6.0      2023-02-10 [2] CRAN (R 4.1.2)
##  ggraph               * 2.1.0.9000 2023-07-11 [1] Github (thomasp85/ggraph@febab71)
##  ggrepel                0.9.3      2023-02-03 [2] CRAN (R 4.1.2)
##  ggsignif               0.6.4      2022-10-13 [2] CRAN (R 4.1.2)
##  ggtree                 3.2.1      2021-11-16 [2] Bioconductor
##  glasso                 1.11       2019-10-01 [2] CRAN (R 4.1.0)
##  glmnet               * 4.1-7      2023-03-23 [2] CRAN (R 4.1.2)
##  GlobalOptions          0.1.2      2020-06-10 [2] CRAN (R 4.1.0)
##  globals                0.16.2     2022-11-21 [2] CRAN (R 4.1.2)
##  globaltest             5.48.0     2021-10-26 [2] Bioconductor
##  glue                 * 1.6.2      2022-02-24 [2] CRAN (R 4.1.2)
##  Gmisc                * 3.0.2      2023-03-13 [2] CRAN (R 4.1.2)
##  gmm                    1.8        2023-06-06 [2] CRAN (R 4.1.3)
##  gmp                    0.7-1      2023-02-07 [2] CRAN (R 4.1.2)
##  GO.db                  3.14.0     2022-04-11 [2] Bioconductor
##  golem                  0.4.1      2023-06-05 [2] CRAN (R 4.1.3)
##  GOSemSim               2.20.0     2021-10-26 [2] Bioconductor
##  gower                  1.0.1      2022-12-22 [2] CRAN (R 4.1.2)
##  gplots                 3.1.3      2022-04-25 [2] CRAN (R 4.1.2)
##  graphlayouts           1.0.0      2023-05-01 [2] CRAN (R 4.1.2)
##  gridExtra              2.3        2017-09-09 [2] CRAN (R 4.1.0)
##  gridGraphics           0.5-1      2020-12-13 [2] CRAN (R 4.1.0)
##  gtable                 0.3.3      2023-03-21 [2] CRAN (R 4.1.2)
##  gtools                 3.9.4      2022-11-27 [2] CRAN (R 4.1.2)
##  hardhat                1.3.0      2023-03-30 [2] CRAN (R 4.1.2)
##  highr                  0.10       2022-12-22 [2] CRAN (R 4.1.2)
##  Hmisc                  5.1-0      2023-05-08 [2] CRAN (R 4.1.2)
##  hms                    1.1.3      2023-03-21 [2] CRAN (R 4.1.2)
##  htmlTable            * 2.4.1      2022-07-07 [2] CRAN (R 4.1.2)
##  htmltools              0.5.5      2023-03-23 [2] CRAN (R 4.1.2)
##  htmlwidgets            1.6.2      2023-03-17 [2] CRAN (R 4.1.2)
##  httpuv                 1.6.11     2023-05-11 [2] CRAN (R 4.1.3)
##  httr                 * 1.4.6      2023-05-08 [2] CRAN (R 4.1.2)
##  huge                   1.3.5      2021-06-30 [2] CRAN (R 4.1.0)
##  igraph               * 1.5.0      2023-06-16 [1] CRAN (R 4.1.3)
##  impute                 1.68.0     2021-10-26 [2] Bioconductor
##  imputeLCMD             2.1        2022-06-10 [2] CRAN (R 4.1.2)
##  ipred                  0.9-14     2023-03-09 [2] CRAN (R 4.1.2)
##  IRanges              * 2.28.0     2021-10-26 [2] Bioconductor
##  irlba                  2.3.5.1    2022-10-03 [2] CRAN (R 4.1.2)
##  iterators              1.0.14     2022-02-05 [2] CRAN (R 4.1.2)
##  itertools              0.1-3      2014-03-12 [2] CRAN (R 4.1.0)
##  jpeg                   0.1-10     2022-11-29 [2] CRAN (R 4.1.2)
##  jquerylib              0.1.4      2021-04-26 [2] CRAN (R 4.1.0)
##  jsonlite               1.8.7      2023-06-29 [2] CRAN (R 4.1.3)
##  KEGGREST               1.34.0     2021-10-26 [2] Bioconductor
##  KernSmooth             2.23-22    2023-07-10 [2] CRAN (R 4.1.3)
##  knitr                  1.43       2023-05-25 [2] CRAN (R 4.1.3)
##  labeling               0.4.2      2020-10-20 [2] CRAN (R 4.1.0)
##  later                  1.3.1      2023-05-02 [2] CRAN (R 4.1.2)
##  lattice                0.21-8     2023-04-05 [2] CRAN (R 4.1.2)
##  lava                   1.7.2.1    2023-02-27 [2] CRAN (R 4.1.2)
##  lavaan                 0.6-15     2023-03-14 [2] CRAN (R 4.1.2)
##  lazyeval               0.2.2      2019-03-15 [2] CRAN (R 4.1.0)
##  libcoin                1.0-9      2021-09-27 [2] CRAN (R 4.1.0)
##  lifecycle              1.0.3      2022-10-07 [2] CRAN (R 4.1.2)
##  limma                  3.50.3     2022-04-07 [2] Bioconductor
##  listenv                0.9.0      2022-12-16 [2] CRAN (R 4.1.2)
##  locfit                 1.5-9.8    2023-06-11 [2] CRAN (R 4.1.3)
##  lubridate              1.9.2      2023-02-10 [2] CRAN (R 4.1.2)
##  magrittr             * 2.0.3      2022-03-30 [2] CRAN (R 4.1.2)
##  MALDIquant             1.22.1     2023-03-20 [2] CRAN (R 4.1.2)
##  MASS                   7.3-60     2023-05-04 [2] CRAN (R 4.1.2)
##  massdatabase         * 1.0.7      2023-05-30 [2] gitlab (jaspershen/massdatabase@df83e93)
##  massdataset          * 1.0.24     2023-05-30 [2] gitlab (jaspershen/massdataset@b397116)
##  masstools            * 1.0.10     2023-05-30 [2] gitlab (jaspershen/masstools@b3c73bc)
##  Matrix               * 1.6-0      2023-07-08 [2] CRAN (R 4.1.3)
##  MatrixGenerics       * 1.6.0      2021-10-26 [2] Bioconductor
##  matrixStats          * 1.0.0      2023-06-02 [2] CRAN (R 4.1.3)
##  memoise                2.0.1      2021-11-26 [2] CRAN (R 4.1.0)
##  MetaboAnalystR       * 3.2.0      2022-06-28 [2] Github (xia-lab/MetaboAnalystR@892a31b)
##  metagenomeSeq          1.36.0     2021-10-26 [2] Bioconductor
##  metid                * 1.2.26     2023-05-30 [2] gitlab (jaspershen/metid@6bde121)
##  metpath              * 1.0.5      2023-05-30 [2] gitlab (jaspershen/metpath@adcad4f)
##  mgcv                   1.8-42     2023-03-02 [2] CRAN (R 4.1.2)
##  MicrobiomeProfiler   * 1.0.0      2021-10-26 [2] Bioconductor
##  mime                   0.12       2021-09-28 [2] CRAN (R 4.1.0)
##  miniUI                 0.1.1.1    2018-05-18 [2] CRAN (R 4.1.0)
##  missForest             1.5        2022-04-14 [2] CRAN (R 4.1.2)
##  mixedCCA               1.6.2      2022-09-09 [2] CRAN (R 4.1.2)
##  mixOmics               6.18.1     2021-11-18 [2] Bioconductor (R 4.1.2)
##  mnormt                 2.1.1      2022-09-26 [2] CRAN (R 4.1.2)
##  ModelMetrics           1.2.2.2    2020-03-17 [2] CRAN (R 4.1.0)
##  modeltools             0.2-23     2020-03-05 [2] CRAN (R 4.1.0)
##  MsCoreUtils            1.6.2      2022-02-24 [2] Bioconductor
##  MSnbase              * 2.20.4     2022-01-16 [2] Bioconductor
##  multcomp               1.4-25     2023-06-20 [2] CRAN (R 4.1.3)
##  multtest               2.50.0     2021-10-26 [2] Bioconductor
##  munsell                0.5.0      2018-06-12 [2] CRAN (R 4.1.0)
##  mvtnorm                1.2-2      2023-06-08 [2] CRAN (R 4.1.3)
##  mzID                   1.32.0     2021-10-26 [2] Bioconductor
##  mzR                  * 2.28.0     2021-10-27 [2] Bioconductor
##  ncdf4                  1.21       2023-01-07 [2] CRAN (R 4.1.2)
##  NetCoMi              * 1.0.3      2022-07-14 [2] Github (stefpeschel/NetCoMi@d4d80d3)
##  nlme                   3.1-162    2023-01-31 [2] CRAN (R 4.1.2)
##  nnet                   7.3-19     2023-05-03 [2] CRAN (R 4.1.2)
##  norm                   1.0-11.1   2023-06-18 [2] CRAN (R 4.1.3)
##  openxlsx               4.2.5.2    2023-02-06 [2] CRAN (R 4.1.2)
##  org.Mm.eg.db         * 3.14.0     2022-11-23 [2] Bioconductor
##  parallelly             1.36.0     2023-05-26 [2] CRAN (R 4.1.3)
##  patchwork              1.1.2      2022-08-19 [2] CRAN (R 4.1.2)
##  pbapply                1.7-2      2023-06-27 [2] CRAN (R 4.1.3)
##  pbivnorm               0.6.0      2015-01-23 [2] CRAN (R 4.1.0)
##  pcaMethods             1.86.0     2021-10-26 [2] Bioconductor
##  pcaPP                  2.0-3      2022-10-24 [2] CRAN (R 4.1.2)
##  permute                0.9-7      2022-01-27 [2] CRAN (R 4.1.2)
##  pheatmap               1.0.12     2019-01-04 [2] CRAN (R 4.1.0)
##  phyloseq               1.38.0     2021-10-26 [2] Bioconductor
##  pillar                 1.9.0      2023-03-22 [2] CRAN (R 4.1.2)
##  pkgbuild               1.4.2      2023-06-26 [2] CRAN (R 4.1.3)
##  pkgconfig              2.0.3      2019-09-22 [2] CRAN (R 4.1.0)
##  pkgload                1.3.2.1    2023-07-08 [2] CRAN (R 4.1.3)
##  plotly               * 4.10.2     2023-06-03 [2] CRAN (R 4.1.3)
##  plyr                   1.8.8      2022-11-11 [2] CRAN (R 4.1.2)
##  png                    0.1-8      2022-11-29 [2] CRAN (R 4.1.2)
##  polyclip               1.10-4     2022-10-20 [2] CRAN (R 4.1.2)
##  POMA                 * 1.7.2      2022-07-26 [2] Github (pcastellanoescuder/POMA@bc8a972)
##  preprocessCore         1.56.0     2021-10-26 [2] Bioconductor
##  prettyunits            1.1.1      2020-01-24 [2] CRAN (R 4.1.0)
##  pROC                   1.18.4     2023-07-06 [2] CRAN (R 4.1.3)
##  processx               3.8.2      2023-06-30 [2] CRAN (R 4.1.3)
##  prodlim                2023.03.31 2023-04-02 [2] CRAN (R 4.1.2)
##  profvis                0.3.8      2023-05-02 [2] CRAN (R 4.1.2)
##  progress               1.2.2      2019-05-16 [2] CRAN (R 4.1.0)
##  promises               1.2.0.1    2021-02-11 [2] CRAN (R 4.1.0)
##  ProtGenerics         * 1.26.0     2021-10-26 [2] Bioconductor
##  proxy                  0.4-27     2022-06-09 [2] CRAN (R 4.1.2)
##  ps                     1.7.5      2023-04-18 [2] CRAN (R 4.1.2)
##  psych                  2.3.6      2023-06-21 [2] CRAN (R 4.1.3)
##  pulsar                 0.3.10     2023-01-26 [2] CRAN (R 4.1.2)
##  purrr                  1.0.1      2023-01-10 [2] CRAN (R 4.1.2)
##  qgraph                 1.9.5      2023-05-16 [2] CRAN (R 4.1.3)
##  qs                     0.25.5     2023-02-22 [2] CRAN (R 4.1.2)
##  quadprog               1.5-8      2019-11-20 [2] CRAN (R 4.1.0)
##  qvalue                 2.26.0     2021-10-26 [2] Bioconductor
##  R6                     2.5.1      2021-08-19 [2] CRAN (R 4.1.0)
##  ragg                   1.2.5      2023-01-12 [2] CRAN (R 4.1.2)
##  randomForest           4.7-1.1    2022-05-23 [2] CRAN (R 4.1.2)
##  RankProd               3.20.0     2021-10-26 [2] Bioconductor
##  RApiSerialize          0.1.2      2022-08-25 [2] CRAN (R 4.1.2)
##  rARPACK                0.11-0     2016-03-10 [2] CRAN (R 4.1.0)
##  rbibutils              2.2.13     2023-01-13 [2] CRAN (R 4.1.2)
##  RColorBrewer           1.1-3      2022-04-03 [2] CRAN (R 4.1.2)
##  Rcpp                 * 1.0.11     2023-07-06 [1] CRAN (R 4.1.3)
##  RcppParallel           5.1.7      2023-02-27 [2] CRAN (R 4.1.2)
##  RCurl                  1.98-1.12  2023-03-27 [2] CRAN (R 4.1.2)
##  Rdisop                 1.54.0     2021-10-26 [2] Bioconductor
##  Rdpack                 2.4        2022-07-20 [2] CRAN (R 4.1.2)
##  readr                  2.1.4      2023-02-10 [2] CRAN (R 4.1.2)
##  readxl               * 1.4.3      2023-07-06 [2] CRAN (R 4.1.3)
##  recipes                1.0.6      2023-04-25 [2] CRAN (R 4.1.2)
##  remotes                2.4.2      2021-11-30 [2] CRAN (R 4.1.0)
##  reshape2               1.4.4      2020-04-09 [2] CRAN (R 4.1.0)
##  rhdf5                  2.38.1     2022-03-10 [2] Bioconductor
##  rhdf5filters           1.6.0      2021-10-26 [2] Bioconductor
##  Rhdf5lib               1.16.0     2021-10-26 [2] Bioconductor
##  rJava                  1.0-6      2021-12-10 [2] CRAN (R 4.1.0)
##  rjson                  0.2.21     2022-01-09 [2] CRAN (R 4.1.2)
##  rlang                  1.1.1      2023-04-28 [1] CRAN (R 4.1.2)
##  rmarkdown              2.23       2023-07-01 [2] CRAN (R 4.1.3)
##  Rmpfr                  0.9-2      2023-04-22 [2] CRAN (R 4.1.2)
##  rngtools               1.5.2      2021-09-20 [2] CRAN (R 4.1.0)
##  rootSolve              1.8.2.3    2021-09-29 [2] CRAN (R 4.1.0)
##  ropls                * 1.26.4     2022-01-11 [2] Bioconductor
##  rpart                  4.1.19     2022-10-21 [2] CRAN (R 4.1.2)
##  Rserve               * 1.8-11     2022-11-28 [2] CRAN (R 4.1.2)
##  RSpectra               0.16-1     2022-04-24 [2] CRAN (R 4.1.2)
##  RSQLite                2.3.1      2023-04-03 [2] CRAN (R 4.1.2)
##  rstatix                0.7.2      2023-02-01 [2] CRAN (R 4.1.2)
##  rstudioapi             0.15.0     2023-07-07 [2] CRAN (R 4.1.3)
##  rvest                  1.0.3      2022-08-19 [2] CRAN (R 4.1.2)
##  S4Vectors            * 0.32.4     2022-03-29 [2] Bioconductor
##  sandwich               3.0-2      2022-06-15 [2] CRAN (R 4.1.2)
##  sass                   0.4.6      2023-05-03 [2] CRAN (R 4.1.2)
##  scales                 1.2.1      2022-08-20 [2] CRAN (R 4.1.2)
##  scatterpie             0.2.1      2023-06-07 [2] CRAN (R 4.1.3)
##  scrime                 1.3.5      2018-12-01 [2] CRAN (R 4.1.0)
##  sessioninfo            1.2.2      2021-12-06 [2] CRAN (R 4.1.0)
##  shadowtext             0.1.2      2022-04-22 [2] CRAN (R 4.1.2)
##  shape                  1.4.6      2021-05-19 [2] CRAN (R 4.1.0)
##  shiny                  1.7.4.1    2023-07-06 [2] CRAN (R 4.1.3)
##  shinycustomloader      0.9.0      2018-03-27 [2] CRAN (R 4.1.0)
##  shinyWidgets           0.7.6      2023-01-08 [2] CRAN (R 4.1.2)
##  siggenes               1.68.0     2021-10-26 [2] Bioconductor
##  snow                   0.4-4      2021-10-27 [2] CRAN (R 4.1.0)
##  SpiecEasi            * 1.1.2      2022-07-14 [2] Github (zdk123/SpiecEasi@c463727)
##  SPRING               * 1.0.4      2022-08-03 [2] Github (GraceYoon/SPRING@3d641a4)
##  stringdist             0.9.10     2022-11-07 [2] CRAN (R 4.1.2)
##  stringfish             0.15.8     2023-05-30 [2] CRAN (R 4.1.3)
##  stringi                1.7.12     2023-01-11 [2] CRAN (R 4.1.2)
##  stringr                1.5.0      2022-12-02 [2] CRAN (R 4.1.2)
##  SummarizedExperiment * 1.24.0     2021-10-26 [2] Bioconductor
##  survival               3.5-5      2023-03-12 [2] CRAN (R 4.1.2)
##  systemfonts            1.0.4      2022-02-11 [2] CRAN (R 4.1.2)
##  textshaping            0.3.6      2021-10-13 [2] CRAN (R 4.1.0)
##  TH.data                1.1-2      2023-04-17 [2] CRAN (R 4.1.2)
##  tibble               * 3.2.1      2023-03-20 [2] CRAN (R 4.1.2)
##  tidygraph              1.2.3      2023-02-01 [2] CRAN (R 4.1.2)
##  tidyr                  1.3.0      2023-01-24 [2] CRAN (R 4.1.2)
##  tidyselect             1.2.0      2022-10-10 [2] CRAN (R 4.1.2)
##  tidytree               0.4.2      2022-12-18 [2] CRAN (R 4.1.2)
##  timechange             0.2.0      2023-01-11 [2] CRAN (R 4.1.2)
##  timeDate               4022.108   2023-01-07 [2] CRAN (R 4.1.2)
##  tmvtnorm               1.5        2022-03-22 [2] CRAN (R 4.1.2)
##  treeio                 1.18.1     2021-11-14 [2] Bioconductor
##  tweenr                 2.0.2      2022-09-06 [2] CRAN (R 4.1.2)
##  tzdb                   0.4.0      2023-05-12 [2] CRAN (R 4.1.3)
##  urlchecker             1.0.1      2021-11-30 [2] CRAN (R 4.1.0)
##  usethis                2.2.2      2023-07-06 [2] CRAN (R 4.1.3)
##  utf8                   1.2.3      2023-01-31 [2] CRAN (R 4.1.2)
##  vctrs                  0.6.3      2023-06-14 [1] CRAN (R 4.1.3)
##  vegan                  2.6-4      2022-10-11 [2] CRAN (R 4.1.2)
##  VGAM                   1.1-8      2023-03-09 [2] CRAN (R 4.1.2)
##  viridis                0.6.3      2023-05-03 [2] CRAN (R 4.1.2)
##  viridisLite            0.4.2      2023-05-02 [2] CRAN (R 4.1.2)
##  vsn                    3.62.0     2021-10-26 [2] Bioconductor
##  WGCNA                * 1.72-1     2023-01-18 [2] CRAN (R 4.1.2)
##  withr                  2.5.0      2022-03-03 [2] CRAN (R 4.1.2)
##  Wrench                 1.12.0     2021-10-26 [2] Bioconductor
##  writexl                1.4.2      2023-01-06 [2] CRAN (R 4.1.2)
##  xfun                   0.40       2023-08-09 [1] CRAN (R 4.1.3)
##  xlsx                   0.6.5      2020-11-10 [2] CRAN (R 4.1.0)
##  xlsxjars               0.6.1      2014-08-22 [2] CRAN (R 4.1.0)
##  XMAS2                * 2.2.0      2023-10-27 [1] local
##  XML                    3.99-0.14  2023-03-19 [2] CRAN (R 4.1.2)
##  xml2                   1.3.5      2023-07-06 [2] CRAN (R 4.1.3)
##  xtable                 1.8-4      2019-04-21 [2] CRAN (R 4.1.0)
##  XVector                0.34.0     2021-10-26 [2] Bioconductor
##  yaml                   2.3.7      2023-01-23 [2] CRAN (R 4.1.2)
##  yulab.utils            0.0.6      2022-12-20 [2] CRAN (R 4.1.2)
##  zip                    2.3.0      2023-04-17 [2] CRAN (R 4.1.2)
##  zlibbioc               1.40.0     2021-10-26 [2] Bioconductor
##  zoo                    1.8-12     2023-04-13 [2] CRAN (R 4.1.2)
## 
##  [1] /Users/zouhua/Library/R/x86_64/4.1/library
##  [2] /Library/Frameworks/R.framework/Versions/4.1/Resources/library
## 
## ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="metorigin-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/Metabolomics_Aanlysis/blob/main/09-Example.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true,
"toc_depth": 3
}
});
});
</script>

</body>

</html>
