<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 MetOrigin Analysis | Data Analysis in Metabolomics</title>
  <meta name="description" content="A template for Data Analysis in Metabolomics" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 MetOrigin Analysis | Data Analysis in Metabolomics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A template for Data Analysis in Metabolomics" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 MetOrigin Analysis | Data Analysis in Metabolomics" />
  
  <meta name="twitter:description" content="A template for Data Analysis in Metabolomics" />
  

<meta name="author" content="Hua Zou and Bangzhuo Tong" />


<meta name="date" content="2023-11-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="functional-analysis-1.html"/>
<link rel="next" href="example.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="assets/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Data Analysis in Metabolomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#what-is-metabolomics"><i class="fa fa-check"></i><b>1.1</b> What is metabolomics?</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#workflow-of-metabolomics-data-analysis"><i class="fa fa-check"></i><b>1.2</b> Workflow of metabolomics data analysis</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#statistical-analysis"><i class="fa fa-check"></i><b>1.2.1</b> Statistical Analysis</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#functional-analysis"><i class="fa fa-check"></i><b>1.2.2</b> Functional Analysis</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#software"><i class="fa fa-check"></i><b>1.3</b> Software</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#reference"><i class="fa fa-check"></i><b>1.4</b> Reference</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-processing.html"><a href="data-processing.html"><i class="fa fa-check"></i><b>2</b> Data Processing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-processing.html"><a href="data-processing.html#loading-packages"><i class="fa fa-check"></i><b>2.1</b> Loading packages</a></li>
<li class="chapter" data-level="2.2" data-path="data-processing.html"><a href="data-processing.html#importing-data"><i class="fa fa-check"></i><b>2.2</b> Importing data</a></li>
<li class="chapter" data-level="2.3" data-path="data-processing.html"><a href="data-processing.html#data-checking"><i class="fa fa-check"></i><b>2.3</b> Data Checking</a></li>
<li class="chapter" data-level="2.4" data-path="data-processing.html"><a href="data-processing.html#missing-value-imputation"><i class="fa fa-check"></i><b>2.4</b> Missing value imputation</a></li>
<li class="chapter" data-level="2.5" data-path="data-processing.html"><a href="data-processing.html#data-filtering"><i class="fa fa-check"></i><b>2.5</b> Data Filtering</a></li>
<li class="chapter" data-level="2.6" data-path="data-processing.html"><a href="data-processing.html#data-normalization"><i class="fa fa-check"></i><b>2.6</b> Data Normalization</a>
<ul>
<li class="chapter" data-level="2.6.1" data-path="data-processing.html"><a href="data-processing.html#normalization-by-normalizedata-function"><i class="fa fa-check"></i><b>2.6.1</b> Normalization by <strong>NormalizeData</strong> function</a></li>
<li class="chapter" data-level="2.6.2" data-path="data-processing.html"><a href="data-processing.html#normalization-by-poma-r-package"><i class="fa fa-check"></i><b>2.6.2</b> Normalization by <strong>POMA</strong> R package</a></li>
<li class="chapter" data-level="2.6.3" data-path="data-processing.html"><a href="data-processing.html#comparison-of-unnormalized-and-normalized-dataset"><i class="fa fa-check"></i><b>2.6.3</b> Comparison of unnormalized and normalized dataset</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="data-processing.html"><a href="data-processing.html#removing-outliers"><i class="fa fa-check"></i><b>2.7</b> Removing outliers</a></li>
<li class="chapter" data-level="2.8" data-path="data-processing.html"><a href="data-processing.html#data-distribution"><i class="fa fa-check"></i><b>2.8</b> Data distribution</a></li>
<li class="chapter" data-level="2.9" data-path="data-processing.html"><a href="data-processing.html#saving-datasets-into-rds-files"><i class="fa fa-check"></i><b>2.9</b> Saving datasets into RDS files</a></li>
<li class="chapter" data-level="2.10" data-path="data-processing.html"><a href="data-processing.html#systematic-information"><i class="fa fa-check"></i><b>2.10</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html"><i class="fa fa-check"></i><b>3</b> Statistical Analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-preprocessing"><i class="fa fa-check"></i><b>3.1</b> Data Preprocessing</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#environment-setup"><i class="fa fa-check"></i><b>3.1.1</b> Environment setup</a></li>
<li class="chapter" data-level="3.1.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-data"><i class="fa fa-check"></i><b>3.1.2</b> Loading data</a></li>
<li class="chapter" data-level="3.1.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-checking-1"><i class="fa fa-check"></i><b>3.1.3</b> Data Checking</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#missing-value-imputation-1"><i class="fa fa-check"></i><b>3.2</b> Missing value imputation</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-filtering-1"><i class="fa fa-check"></i><b>3.2.1</b> Data Filtering</a></li>
<li class="chapter" data-level="3.2.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-normalization-1"><i class="fa fa-check"></i><b>3.2.2</b> Data Normalization</a></li>
<li class="chapter" data-level="3.2.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#removing-outliers-1"><i class="fa fa-check"></i><b>3.2.3</b> Removing outliers</a></li>
<li class="chapter" data-level="3.2.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#saving-datasets-into-rds-files-1"><i class="fa fa-check"></i><b>3.2.4</b> Saving datasets into RDS files</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#cluster-analysis"><i class="fa fa-check"></i><b>3.3</b> Cluster Analysis</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-packages-1"><i class="fa fa-check"></i><b>3.3.1</b> Loading packages</a></li>
<li class="chapter" data-level="3.3.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#importing-data-1"><i class="fa fa-check"></i><b>3.3.2</b> Importing data</a></li>
<li class="chapter" data-level="3.3.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#hierarchical-clustering"><i class="fa fa-check"></i><b>3.3.3</b> Hierarchical Clustering</a></li>
<li class="chapter" data-level="3.3.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#partitional-clustering"><i class="fa fa-check"></i><b>3.3.4</b> Partitional Clustering</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#chemometrics-analysis"><i class="fa fa-check"></i><b>3.4</b> Chemometrics Analysis</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-packages-2"><i class="fa fa-check"></i><b>3.4.1</b> Loading packages</a></li>
<li class="chapter" data-level="3.4.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#importing-data-2"><i class="fa fa-check"></i><b>3.4.2</b> Importing data</a></li>
<li class="chapter" data-level="3.4.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>3.4.3</b> Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="3.4.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#partial-least-squares-discriminant-analysis-pls-da"><i class="fa fa-check"></i><b>3.4.4</b> Partial Least Squares-Discriminant Analysis (PLS-DA)</a></li>
<li class="chapter" data-level="3.4.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#sparse-partial-least-squares-discriminant-analysis-spls-da"><i class="fa fa-check"></i><b>3.4.5</b> Sparse Partial Least Squares-Discriminant Analysis (sPLS-DA)</a></li>
<li class="chapter" data-level="3.4.6" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#orthogonal-partial-least-squares-discriminant-analysis-orthopls-da"><i class="fa fa-check"></i><b>3.4.6</b> Orthogonal Partial Least Squares-Discriminant Analysis (orthoPLS-DA)</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#univariate-analysis"><i class="fa fa-check"></i><b>3.5</b> Univariate Analysis</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-packages-3"><i class="fa fa-check"></i><b>3.5.1</b> Loading packages</a></li>
<li class="chapter" data-level="3.5.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#importing-data-3"><i class="fa fa-check"></i><b>3.5.2</b> Importing data</a></li>
<li class="chapter" data-level="3.5.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#fold-change-analysis"><i class="fa fa-check"></i><b>3.5.3</b> Fold Change Analysis</a></li>
<li class="chapter" data-level="3.5.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#t-test"><i class="fa fa-check"></i><b>3.5.4</b> T Test</a></li>
<li class="chapter" data-level="3.5.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#wilcoxon-test"><i class="fa fa-check"></i><b>3.5.5</b> Wilcoxon Test</a></li>
<li class="chapter" data-level="3.5.6" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#limma-test"><i class="fa fa-check"></i><b>3.5.6</b> Limma Test</a></li>
<li class="chapter" data-level="3.5.7" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#volcano-plot"><i class="fa fa-check"></i><b>3.5.7</b> Volcano plot</a></li>
<li class="chapter" data-level="3.5.8" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#vip-variable-influence-on-projection-coefficient"><i class="fa fa-check"></i><b>3.5.8</b> VIP (Variable influence on projection &amp; coefficient)</a></li>
<li class="chapter" data-level="3.5.9" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#t-test-by-local-codes"><i class="fa fa-check"></i><b>3.5.9</b> T-test by local codes</a></li>
<li class="chapter" data-level="3.5.10" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#merging-result"><i class="fa fa-check"></i><b>3.5.10</b> Merging result</a></li>
<li class="chapter" data-level="3.5.11" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#volcano-of-merged-results"><i class="fa fa-check"></i><b>3.5.11</b> Volcano of Merged Results</a></li>
<li class="chapter" data-level="3.5.12" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#correlation-heatmaps"><i class="fa fa-check"></i><b>3.5.12</b> Correlation Heatmaps</a></li>
<li class="chapter" data-level="3.5.13" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#glasso-this-function-will-compute-a-gaussian-graphical-model-using-the-glmnet-package"><i class="fa fa-check"></i><b>3.5.13</b> glasso: this function will compute a Gaussian Graphical Model using the glmnet package</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#multivariate-analysis"><i class="fa fa-check"></i><b>3.6</b> Multivariate analysis</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-packages-4"><i class="fa fa-check"></i><b>3.6.1</b> Loading packages</a></li>
<li class="chapter" data-level="3.6.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#importing-data-4"><i class="fa fa-check"></i><b>3.6.2</b> Importing data</a></li>
<li class="chapter" data-level="3.6.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-curation"><i class="fa fa-check"></i><b>3.6.3</b> Data curation</a></li>
<li class="chapter" data-level="3.6.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#regularized-generalized-linear-models"><i class="fa fa-check"></i><b>3.6.4</b> Regularized Generalized Linear Models</a></li>
<li class="chapter" data-level="3.6.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#classification"><i class="fa fa-check"></i><b>3.6.5</b> Classification</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#network-analysis"><i class="fa fa-check"></i><b>3.7</b> Network Analysis</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#introduction-1"><i class="fa fa-check"></i><b>3.7.1</b> Introduction</a></li>
<li class="chapter" data-level="3.7.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-packages-5"><i class="fa fa-check"></i><b>3.7.2</b> Loading packages</a></li>
<li class="chapter" data-level="3.7.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#importing-data-5"><i class="fa fa-check"></i><b>3.7.3</b> Importing data</a></li>
<li class="chapter" data-level="3.7.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-curation-1"><i class="fa fa-check"></i><b>3.7.4</b> Data curation</a></li>
<li class="chapter" data-level="3.7.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#associations-among-features"><i class="fa fa-check"></i><b>3.7.5</b> Associations Among Features</a></li>
<li class="chapter" data-level="3.7.6" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#network-comparison"><i class="fa fa-check"></i><b>3.7.6</b> Network comparison</a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#systematic-information-1"><i class="fa fa-check"></i><b>3.8</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html"><i class="fa fa-check"></i><b>4</b> Functional Analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#enrichment-analysis"><i class="fa fa-check"></i><b>4.1</b> Enrichment Analysis</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#enviroment-setup"><i class="fa fa-check"></i><b>4.1.1</b> Enviroment Setup</a></li>
<li class="chapter" data-level="4.1.2" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#over-representation-analysis-ora"><i class="fa fa-check"></i><b>4.1.2</b> Over representation analysis (ORA)</a></li>
<li class="chapter" data-level="4.1.3" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#clean-environment-1"><i class="fa fa-check"></i><b>4.1.3</b> Clean environment 1</a></li>
<li class="chapter" data-level="4.1.4" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#single-sample-profiling"><i class="fa fa-check"></i><b>4.1.4</b> Single Sample Profiling</a></li>
<li class="chapter" data-level="4.1.5" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#clean-environment-2"><i class="fa fa-check"></i><b>4.1.5</b> Clean environment 2</a></li>
<li class="chapter" data-level="4.1.6" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#quantitative-enrichment-analysis-qea"><i class="fa fa-check"></i><b>4.1.6</b> Quantitative Enrichment Analysis (QEA)</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#pathway-analysis"><i class="fa fa-check"></i><b>4.2</b> Pathway Analysis</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#enviroment-set-up"><i class="fa fa-check"></i><b>4.2.1</b> Enviroment Set up</a></li>
<li class="chapter" data-level="4.2.2" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#over-representation-analysis"><i class="fa fa-check"></i><b>4.2.2</b> Over representation analysis</a></li>
<li class="chapter" data-level="4.2.3" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#clean-environment-3"><i class="fa fa-check"></i><b>4.2.3</b> Clean environment 3</a></li>
<li class="chapter" data-level="4.2.4" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#concentration-table-qea"><i class="fa fa-check"></i><b>4.2.4</b> Concentration Table (QEA)</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#functional-analysis-by-r-package"><i class="fa fa-check"></i><b>4.3</b> Functional Analysis by R package</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#loading-r-packages"><i class="fa fa-check"></i><b>4.3.1</b> Loading R packages</a></li>
<li class="chapter" data-level="4.3.2" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#importing-data-6"><i class="fa fa-check"></i><b>4.3.2</b> Importing Data</a></li>
<li class="chapter" data-level="4.3.3" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#differential-analysis"><i class="fa fa-check"></i><b>4.3.3</b> Differential Analysis</a></li>
<li class="chapter" data-level="4.3.4" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#metabolites-id-for-merged-results"><i class="fa fa-check"></i><b>4.3.4</b> Metabolites ID for Merged Results</a></li>
<li class="chapter" data-level="4.3.5" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#kegg-pathway-compound"><i class="fa fa-check"></i><b>4.3.5</b> KEGG PATHWAY COMPOUND</a></li>
<li class="chapter" data-level="4.3.6" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#functions-related-to-analysis"><i class="fa fa-check"></i><b>4.3.6</b> Functions related to Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#session-info"><i class="fa fa-check"></i><b>4.4</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html"><i class="fa fa-check"></i><b>5</b> MetOrigin Analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#loading-packages-6"><i class="fa fa-check"></i><b>5.1</b> Loading packages</a></li>
<li class="chapter" data-level="5.2" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#importing-data-7"><i class="fa fa-check"></i><b>5.2</b> Importing data</a></li>
<li class="chapter" data-level="5.3" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#differential-analysis-1"><i class="fa fa-check"></i><b>5.3</b> Differential Analysis</a></li>
<li class="chapter" data-level="5.4" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#de-metabolites"><i class="fa fa-check"></i><b>5.4</b> DE metabolites</a></li>
<li class="chapter" data-level="5.5" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#obtain-inputs-for-metorigin"><i class="fa fa-check"></i><b>5.5</b> Obtain inputs for Metorigin</a></li>
<li class="chapter" data-level="5.6" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#save-result"><i class="fa fa-check"></i><b>5.6</b> Save result</a></li>
<li class="chapter" data-level="5.7" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#session-info-1"><i class="fa fa-check"></i><b>5.7</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="example.html"><a href="example.html"><i class="fa fa-check"></i><b>6</b> Example</a>
<ul>
<li class="chapter" data-level="6.1" data-path="example.html"><a href="example.html#loading-packages-7"><i class="fa fa-check"></i><b>6.1</b> Loading packages</a></li>
<li class="chapter" data-level="6.2" data-path="example.html"><a href="example.html#importing-data-8"><i class="fa fa-check"></i><b>6.2</b> Importing data</a></li>
<li class="chapter" data-level="6.3" data-path="example.html"><a href="example.html#data-processing-1"><i class="fa fa-check"></i><b>6.3</b> Data Processing</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="example.html"><a href="example.html#data-checking-2"><i class="fa fa-check"></i><b>6.3.1</b> Data Checking</a></li>
<li class="chapter" data-level="6.3.2" data-path="example.html"><a href="example.html#missing-value-imputation-2"><i class="fa fa-check"></i><b>6.3.2</b> Missing value imputation</a></li>
<li class="chapter" data-level="6.3.3" data-path="example.html"><a href="example.html#data-filtering-2"><i class="fa fa-check"></i><b>6.3.3</b> Data Filtering</a></li>
<li class="chapter" data-level="6.3.4" data-path="example.html"><a href="example.html#data-normalization-2"><i class="fa fa-check"></i><b>6.3.4</b> Data Normalization</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="example.html"><a href="example.html#cluster-analysis-1"><i class="fa fa-check"></i><b>6.4</b> Cluster Analysis</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="example.html"><a href="example.html#hierarchical-clustering-1"><i class="fa fa-check"></i><b>6.4.1</b> Hierarchical Clustering</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="example.html"><a href="example.html#chemometrics-analysis-1"><i class="fa fa-check"></i><b>6.5</b> Chemometrics Analysis</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="example.html"><a href="example.html#partial-least-squares-discriminant-analysis-pls-da-1"><i class="fa fa-check"></i><b>6.5.1</b> Partial Least Squares-Discriminant Analysis (PLS-DA)</a></li>
<li class="chapter" data-level="6.5.2" data-path="example.html"><a href="example.html#sparse-partial-least-squares-discriminant-analysis-spls-da-1"><i class="fa fa-check"></i><b>6.5.2</b> Sparse Partial Least Squares-Discriminant Analysis (sPLS-DA)</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="example.html"><a href="example.html#univariate-analysis-1"><i class="fa fa-check"></i><b>6.6</b> Univariate Analysis</a>
<ul>
<li class="chapter" data-level="6.6.1" data-path="example.html"><a href="example.html#fold-change-analysis-1"><i class="fa fa-check"></i><b>6.6.1</b> Fold Change Analysis</a></li>
<li class="chapter" data-level="6.6.2" data-path="example.html"><a href="example.html#vip-variable-influence-on-projection-coefficient-1"><i class="fa fa-check"></i><b>6.6.2</b> VIP (Variable influence on projection &amp; coefficient)</a></li>
<li class="chapter" data-level="6.6.3" data-path="example.html"><a href="example.html#t-test-by-local-codes-1"><i class="fa fa-check"></i><b>6.6.3</b> T-test by local codes</a></li>
<li class="chapter" data-level="6.6.4" data-path="example.html"><a href="example.html#merging-result-1"><i class="fa fa-check"></i><b>6.6.4</b> Merging result</a></li>
<li class="chapter" data-level="6.6.5" data-path="example.html"><a href="example.html#volcano-of-merged-results-1"><i class="fa fa-check"></i><b>6.6.5</b> Volcano of Merged Results</a></li>
<li class="chapter" data-level="6.6.6" data-path="example.html"><a href="example.html#t-test-1"><i class="fa fa-check"></i><b>6.6.6</b> T Test</a></li>
<li class="chapter" data-level="6.6.7" data-path="example.html"><a href="example.html#volcano-plot-1"><i class="fa fa-check"></i><b>6.6.7</b> Volcano plot</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="example.html"><a href="example.html#feature-selection"><i class="fa fa-check"></i><b>6.7</b> Feature Selection</a>
<ul>
<li class="chapter" data-level="6.7.1" data-path="example.html"><a href="example.html#regularized-generalized-linear-models-lasso-alpha-1"><i class="fa fa-check"></i><b>6.7.1</b> Regularized Generalized Linear Models (Lasso: alpha = 1)</a></li>
<li class="chapter" data-level="6.7.2" data-path="example.html"><a href="example.html#classification-random-forest"><i class="fa fa-check"></i><b>6.7.2</b> Classification (Random Forest)</a></li>
</ul></li>
<li class="chapter" data-level="6.8" data-path="example.html"><a href="example.html#network-analysis-2"><i class="fa fa-check"></i><b>6.8</b> Network Analysis</a>
<ul>
<li class="chapter" data-level="6.8.1" data-path="example.html"><a href="example.html#data-curation-2"><i class="fa fa-check"></i><b>6.8.1</b> Data curation</a></li>
<li class="chapter" data-level="6.8.2" data-path="example.html"><a href="example.html#building-network-model-4"><i class="fa fa-check"></i><b>6.8.2</b> Building network model</a></li>
<li class="chapter" data-level="6.8.3" data-path="example.html"><a href="example.html#visualizing-the-network-4"><i class="fa fa-check"></i><b>6.8.3</b> Visualizing the network</a></li>
</ul></li>
<li class="chapter" data-level="6.9" data-path="example.html"><a href="example.html#network-analysis-by-wgcna"><i class="fa fa-check"></i><b>6.9</b> Network Analysis by WGCNA</a>
<ul>
<li class="chapter" data-level="6.9.1" data-path="example.html"><a href="example.html#data-curation-3"><i class="fa fa-check"></i><b>6.9.1</b> Data curation</a></li>
<li class="chapter" data-level="6.9.2" data-path="example.html"><a href="example.html#tuning-soft-thresholds"><i class="fa fa-check"></i><b>6.9.2</b> Tuning soft thresholds</a></li>
<li class="chapter" data-level="6.9.3" data-path="example.html"><a href="example.html#create-the-network-using-the-blockwisemodules"><i class="fa fa-check"></i><b>6.9.3</b> Create the network using the <code>blockwiseModules</code></a></li>
<li class="chapter" data-level="6.9.4" data-path="example.html"><a href="example.html#plot-modules"><i class="fa fa-check"></i><b>6.9.4</b> plot modules</a></li>
<li class="chapter" data-level="6.9.5" data-path="example.html"><a href="example.html#relationships-among-modules"><i class="fa fa-check"></i><b>6.9.5</b> Relationships among modules</a></li>
<li class="chapter" data-level="6.9.6" data-path="example.html"><a href="example.html#module-eigengene-correlation"><i class="fa fa-check"></i><b>6.9.6</b> Module (Eigengene) correlation</a></li>
<li class="chapter" data-level="6.9.7" data-path="example.html"><a href="example.html#relate-module-cluster-assignments-to-groups"><i class="fa fa-check"></i><b>6.9.7</b> Relate Module (cluster) Assignments to Groups</a></li>
<li class="chapter" data-level="6.9.8" data-path="example.html"><a href="example.html#generate-and-export-networks"><i class="fa fa-check"></i><b>6.9.8</b> Generate and Export Networks</a></li>
<li class="chapter" data-level="6.9.9" data-path="example.html"><a href="example.html#network-visualization"><i class="fa fa-check"></i><b>6.9.9</b> Network visualization</a></li>
</ul></li>
<li class="chapter" data-level="6.10" data-path="example.html"><a href="example.html#systematic-information-2"><i class="fa fa-check"></i><b>6.10</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data Analysis in Metabolomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="metorigin-analysis" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> MetOrigin Analysis<a href="metorigin-analysis.html#metorigin-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<blockquote>
<p>Microbiome and its metabolites are closely associated with human health and diseases. However, it is challenging to understand the complex interplay between microbiome and metabolites. MetOrigin is a bioinformatics tool, aiming to identify which bacteria and how they participate in certain metabolic reactions, helping us to understand where metabolites come from: host, bacteria, or both?</p>
</blockquote>
<p>The tutorial was from <a href="https://metorigin.met-bioinformatics.cn/home/" class="uri">https://metorigin.met-bioinformatics.cn/home/</a>. Here, we gave example for how to prepare inputs to MetOrigin website.</p>
<div id="loading-packages-6" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Loading packages<a href="metorigin-analysis.html#loading-packages-6" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb232"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb232-1"><a href="metorigin-analysis.html#cb232-1" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">warning =</span> F)</span>
<span id="cb232-2"><a href="metorigin-analysis.html#cb232-2" tabindex="-1"></a></span>
<span id="cb232-3"><a href="metorigin-analysis.html#cb232-3" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb232-4"><a href="metorigin-analysis.html#cb232-4" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb232-5"><a href="metorigin-analysis.html#cb232-5" tabindex="-1"></a><span class="fu">library</span>(Biobase)</span>
<span id="cb232-6"><a href="metorigin-analysis.html#cb232-6" tabindex="-1"></a><span class="fu">library</span>(POMA)</span>
<span id="cb232-7"><a href="metorigin-analysis.html#cb232-7" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb232-8"><a href="metorigin-analysis.html#cb232-8" tabindex="-1"></a><span class="fu">library</span>(ropls)</span>
<span id="cb232-9"><a href="metorigin-analysis.html#cb232-9" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb232-10"><a href="metorigin-analysis.html#cb232-10" tabindex="-1"></a></span>
<span id="cb232-11"><a href="metorigin-analysis.html#cb232-11" tabindex="-1"></a><span class="co"># rm(list = ls())</span></span>
<span id="cb232-12"><a href="metorigin-analysis.html#cb232-12" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb232-13"><a href="metorigin-analysis.html#cb232-13" tabindex="-1"></a><span class="fu">options</span>(<span class="at">future.globals.maxSize =</span> <span class="dv">1000</span> <span class="sc">*</span> <span class="dv">1024</span><span class="sc">^</span><span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="importing-data-7" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Importing data<a href="metorigin-analysis.html#importing-data-7" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The dataset is from the Zeybel-2022 published paper <span class="citation">(<a href="#ref-zeybel2022multiomics">Zeybel et al. 2022</a>)</span>.</p>
<div class="sourceCode" id="cb233"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb233-1"><a href="metorigin-analysis.html#cb233-1" tabindex="-1"></a>ExprSet <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;./dataset/POMA/ExprSet_raw.RDS&quot;</span>)</span>
<span id="cb233-2"><a href="metorigin-analysis.html#cb233-2" tabindex="-1"></a>se_impute <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;./dataset/POMA/se_impute.RDS&quot;</span>)</span>
<span id="cb233-3"><a href="metorigin-analysis.html#cb233-3" tabindex="-1"></a>se_norm <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;./dataset/POMA/se_normalize.RDS&quot;</span>)</span></code></pre></div>
</div>
<div id="differential-analysis-1" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Differential Analysis<a href="metorigin-analysis.html#differential-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li><p>Fold change (group1 vs group2)</p>
<ul>
<li>RawData</li>
</ul></li>
<li><p>VIP (Variable influence on projection &amp; coefficient)</p>
<ul>
<li><p>Variable influence on projection (VIP) for orthogonal projections to latent structures (OPLS)</p></li>
<li><p>Variable influence on projection (VIP) for projections to latent structures (PLS)</p></li>
</ul></li>
<li><p>T-test</p>
<ul>
<li>significant differences between two groups (p value)</li>
</ul></li>
<li><p>Merging result</p>
<ul>
<li><p>Foldchange by Raw Data</p></li>
<li><p>VIP by Normalized Data</p></li>
<li><p>test Pvalue by Normalized Data</p></li>
</ul></li>
</ul>
<div class="sourceCode" id="cb234"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb234-1"><a href="metorigin-analysis.html#cb234-1" tabindex="-1"></a>FoldChange <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb234-2"><a href="metorigin-analysis.html#cb234-2" tabindex="-1"></a>    x,</span>
<span id="cb234-3"><a href="metorigin-analysis.html#cb234-3" tabindex="-1"></a>    group,</span>
<span id="cb234-4"><a href="metorigin-analysis.html#cb234-4" tabindex="-1"></a>    group_names) {</span>
<span id="cb234-5"><a href="metorigin-analysis.html#cb234-5" tabindex="-1"></a>  </span>
<span id="cb234-6"><a href="metorigin-analysis.html#cb234-6" tabindex="-1"></a>  <span class="co"># dataseat</span></span>
<span id="cb234-7"><a href="metorigin-analysis.html#cb234-7" tabindex="-1"></a>  metadata <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(x) <span class="sc">%&gt;%</span></span>
<span id="cb234-8"><a href="metorigin-analysis.html#cb234-8" tabindex="-1"></a>      <span class="fu">as.data.frame</span>()</span>
<span id="cb234-9"><a href="metorigin-analysis.html#cb234-9" tabindex="-1"></a>  profile <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(x) <span class="sc">%&gt;%</span></span>
<span id="cb234-10"><a href="metorigin-analysis.html#cb234-10" tabindex="-1"></a>      <span class="fu">as.data.frame</span>()</span>
<span id="cb234-11"><a href="metorigin-analysis.html#cb234-11" tabindex="-1"></a></span>
<span id="cb234-12"><a href="metorigin-analysis.html#cb234-12" tabindex="-1"></a>  <span class="fu">colnames</span>(metadata)[<span class="fu">which</span>(<span class="fu">colnames</span>(metadata) <span class="sc">==</span> group)] <span class="ot">&lt;-</span> <span class="st">&quot;CompVar&quot;</span></span>
<span id="cb234-13"><a href="metorigin-analysis.html#cb234-13" tabindex="-1"></a>  phenotype <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span></span>
<span id="cb234-14"><a href="metorigin-analysis.html#cb234-14" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(CompVar <span class="sc">%in%</span> group_names) <span class="sc">%&gt;%</span></span>
<span id="cb234-15"><a href="metorigin-analysis.html#cb234-15" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">CompVar =</span> <span class="fu">as.character</span>(CompVar)) <span class="sc">%&gt;%</span></span>
<span id="cb234-16"><a href="metorigin-analysis.html#cb234-16" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">CompVar =</span> <span class="fu">factor</span>(CompVar, <span class="at">levels =</span> group_names))</span>
<span id="cb234-17"><a href="metorigin-analysis.html#cb234-17" tabindex="-1"></a>  </span>
<span id="cb234-18"><a href="metorigin-analysis.html#cb234-18" tabindex="-1"></a>  sid <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(phenotype), <span class="fu">colnames</span>(profile))</span>
<span id="cb234-19"><a href="metorigin-analysis.html#cb234-19" tabindex="-1"></a>  phen <span class="ot">&lt;-</span> phenotype[<span class="fu">pmatch</span>(sid, <span class="fu">rownames</span>(phenotype)), , ]</span>
<span id="cb234-20"><a href="metorigin-analysis.html#cb234-20" tabindex="-1"></a>  prof <span class="ot">&lt;-</span> profile <span class="sc">%&gt;%</span></span>
<span id="cb234-21"><a href="metorigin-analysis.html#cb234-21" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(sid))</span>
<span id="cb234-22"><a href="metorigin-analysis.html#cb234-22" tabindex="-1"></a>  </span>
<span id="cb234-23"><a href="metorigin-analysis.html#cb234-23" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">all</span>(<span class="fu">colnames</span>(prof) <span class="sc">==</span> <span class="fu">rownames</span>(phen))) {</span>
<span id="cb234-24"><a href="metorigin-analysis.html#cb234-24" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;Wrong Order&quot;</span>)</span>
<span id="cb234-25"><a href="metorigin-analysis.html#cb234-25" tabindex="-1"></a>  }</span>
<span id="cb234-26"><a href="metorigin-analysis.html#cb234-26" tabindex="-1"></a>  </span>
<span id="cb234-27"><a href="metorigin-analysis.html#cb234-27" tabindex="-1"></a>  fc_res <span class="ot">&lt;-</span> <span class="fu">apply</span>(prof, <span class="dv">1</span>, <span class="cf">function</span>(x1, y1) {</span>
<span id="cb234-28"><a href="metorigin-analysis.html#cb234-28" tabindex="-1"></a>    dat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">value =</span> <span class="fu">as.numeric</span>(x1), <span class="at">group =</span> y1)</span>
<span id="cb234-29"><a href="metorigin-analysis.html#cb234-29" tabindex="-1"></a>    mn <span class="ot">&lt;-</span> <span class="fu">tapply</span>(dat<span class="sc">$</span>value, dat<span class="sc">$</span>group, <span class="cf">function</span>(x){</span>
<span id="cb234-30"><a href="metorigin-analysis.html#cb234-30" tabindex="-1"></a>      <span class="fu">mean</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb234-31"><a href="metorigin-analysis.html#cb234-31" tabindex="-1"></a>    }) <span class="sc">%&gt;%</span></span>
<span id="cb234-32"><a href="metorigin-analysis.html#cb234-32" tabindex="-1"></a>      <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb234-33"><a href="metorigin-analysis.html#cb234-33" tabindex="-1"></a>      stats<span class="sc">::</span><span class="fu">setNames</span>(<span class="st">&quot;value&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb234-34"><a href="metorigin-analysis.html#cb234-34" tabindex="-1"></a>      tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;Group&quot;</span>)</span>
<span id="cb234-35"><a href="metorigin-analysis.html#cb234-35" tabindex="-1"></a></span>
<span id="cb234-36"><a href="metorigin-analysis.html#cb234-36" tabindex="-1"></a>    mn1 <span class="ot">&lt;-</span> <span class="fu">with</span>(mn, mn[Group <span class="sc">%in%</span> group_names[<span class="dv">1</span>], <span class="st">&quot;value&quot;</span>])</span>
<span id="cb234-37"><a href="metorigin-analysis.html#cb234-37" tabindex="-1"></a>    mn2 <span class="ot">&lt;-</span> <span class="fu">with</span>(mn, mn[Group <span class="sc">%in%</span> group_names[<span class="dv">2</span>], <span class="st">&quot;value&quot;</span>])</span>
<span id="cb234-38"><a href="metorigin-analysis.html#cb234-38" tabindex="-1"></a>    mnall <span class="ot">&lt;-</span> <span class="fu">mean</span>(dat<span class="sc">$</span>value, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb234-39"><a href="metorigin-analysis.html#cb234-39" tabindex="-1"></a>    </span>
<span id="cb234-40"><a href="metorigin-analysis.html#cb234-40" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">all</span>(mn1 <span class="sc">!=</span> <span class="dv">0</span>, mn2 <span class="sc">!=</span> <span class="dv">0</span>)) {</span>
<span id="cb234-41"><a href="metorigin-analysis.html#cb234-41" tabindex="-1"></a>      fc <span class="ot">&lt;-</span> mn1 <span class="sc">/</span> mn2</span>
<span id="cb234-42"><a href="metorigin-analysis.html#cb234-42" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb234-43"><a href="metorigin-analysis.html#cb234-43" tabindex="-1"></a>      fc <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb234-44"><a href="metorigin-analysis.html#cb234-44" tabindex="-1"></a>    }</span>
<span id="cb234-45"><a href="metorigin-analysis.html#cb234-45" tabindex="-1"></a>    </span>
<span id="cb234-46"><a href="metorigin-analysis.html#cb234-46" tabindex="-1"></a>    logfc <span class="ot">&lt;-</span> <span class="fu">log2</span>(fc)</span>
<span id="cb234-47"><a href="metorigin-analysis.html#cb234-47" tabindex="-1"></a></span>
<span id="cb234-48"><a href="metorigin-analysis.html#cb234-48" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="fu">c</span>(fc, logfc, mnall, mn1, mn2)</span>
<span id="cb234-49"><a href="metorigin-analysis.html#cb234-49" tabindex="-1"></a>    <span class="fu">return</span>(res)</span>
<span id="cb234-50"><a href="metorigin-analysis.html#cb234-50" tabindex="-1"></a>  }, phen<span class="sc">$</span>CompVar) <span class="sc">%&gt;%</span></span>
<span id="cb234-51"><a href="metorigin-analysis.html#cb234-51" tabindex="-1"></a>    base<span class="sc">::</span><span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb234-52"><a href="metorigin-analysis.html#cb234-52" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;Feature&quot;</span>)</span>
<span id="cb234-53"><a href="metorigin-analysis.html#cb234-53" tabindex="-1"></a>  </span>
<span id="cb234-54"><a href="metorigin-analysis.html#cb234-54" tabindex="-1"></a>  <span class="fu">colnames</span>(fc_res) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;FeatureID&quot;</span>, <span class="st">&quot;FoldChange&quot;</span>, </span>
<span id="cb234-55"><a href="metorigin-analysis.html#cb234-55" tabindex="-1"></a>                        <span class="st">&quot;Log2FoldChange&quot;</span>,</span>
<span id="cb234-56"><a href="metorigin-analysis.html#cb234-56" tabindex="-1"></a>                        <span class="st">&quot;Mean Abundance</span><span class="sc">\n</span><span class="st">(All)&quot;</span>,</span>
<span id="cb234-57"><a href="metorigin-analysis.html#cb234-57" tabindex="-1"></a>                     <span class="fu">paste0</span>(<span class="st">&quot;Mean Abundance</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="fu">c</span>(<span class="st">&quot;former&quot;</span>, <span class="st">&quot;latter&quot;</span>)))  </span>
<span id="cb234-58"><a href="metorigin-analysis.html#cb234-58" tabindex="-1"></a>  </span>
<span id="cb234-59"><a href="metorigin-analysis.html#cb234-59" tabindex="-1"></a>  <span class="co"># Number of Group</span></span>
<span id="cb234-60"><a href="metorigin-analysis.html#cb234-60" tabindex="-1"></a>  dat_status <span class="ot">&lt;-</span> <span class="fu">table</span>(phen<span class="sc">$</span>CompVar)</span>
<span id="cb234-61"><a href="metorigin-analysis.html#cb234-61" tabindex="-1"></a>  dat_status_number <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(dat_status)</span>
<span id="cb234-62"><a href="metorigin-analysis.html#cb234-62" tabindex="-1"></a>  dat_status_name <span class="ot">&lt;-</span> <span class="fu">names</span>(dat_status)</span>
<span id="cb234-63"><a href="metorigin-analysis.html#cb234-63" tabindex="-1"></a>  fc_res<span class="sc">$</span>Block <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">paste</span>(dat_status_number[<span class="dv">1</span>], dat_status_name[<span class="dv">1</span>], <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>),</span>
<span id="cb234-64"><a href="metorigin-analysis.html#cb234-64" tabindex="-1"></a>                       <span class="st">&quot;vs&quot;</span>,</span>
<span id="cb234-65"><a href="metorigin-analysis.html#cb234-65" tabindex="-1"></a>                       <span class="fu">paste</span>(dat_status_number[<span class="dv">2</span>], dat_status_name[<span class="dv">2</span>], <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>))</span>
<span id="cb234-66"><a href="metorigin-analysis.html#cb234-66" tabindex="-1"></a>  </span>
<span id="cb234-67"><a href="metorigin-analysis.html#cb234-67" tabindex="-1"></a>  res <span class="ot">&lt;-</span> fc_res <span class="sc">%&gt;%</span></span>
<span id="cb234-68"><a href="metorigin-analysis.html#cb234-68" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(FeatureID, Block, <span class="fu">everything</span>())    </span>
<span id="cb234-69"><a href="metorigin-analysis.html#cb234-69" tabindex="-1"></a></span>
<span id="cb234-70"><a href="metorigin-analysis.html#cb234-70" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb234-71"><a href="metorigin-analysis.html#cb234-71" tabindex="-1"></a>}</span>
<span id="cb234-72"><a href="metorigin-analysis.html#cb234-72" tabindex="-1"></a></span>
<span id="cb234-73"><a href="metorigin-analysis.html#cb234-73" tabindex="-1"></a>VIP_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb234-74"><a href="metorigin-analysis.html#cb234-74" tabindex="-1"></a>    x,</span>
<span id="cb234-75"><a href="metorigin-analysis.html#cb234-75" tabindex="-1"></a>    group,</span>
<span id="cb234-76"><a href="metorigin-analysis.html#cb234-76" tabindex="-1"></a>    group_names,</span>
<span id="cb234-77"><a href="metorigin-analysis.html#cb234-77" tabindex="-1"></a>    <span class="at">VIPtype =</span> <span class="fu">c</span>(<span class="st">&quot;OPLS&quot;</span>, <span class="st">&quot;PLS&quot;</span>),</span>
<span id="cb234-78"><a href="metorigin-analysis.html#cb234-78" tabindex="-1"></a>    <span class="at">vip_cutoff =</span> <span class="dv">1</span>) {</span>
<span id="cb234-79"><a href="metorigin-analysis.html#cb234-79" tabindex="-1"></a>  </span>
<span id="cb234-80"><a href="metorigin-analysis.html#cb234-80" tabindex="-1"></a>  metadata <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(x) <span class="sc">%&gt;%</span></span>
<span id="cb234-81"><a href="metorigin-analysis.html#cb234-81" tabindex="-1"></a>      <span class="fu">as.data.frame</span>()</span>
<span id="cb234-82"><a href="metorigin-analysis.html#cb234-82" tabindex="-1"></a>  profile <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(x) <span class="sc">%&gt;%</span></span>
<span id="cb234-83"><a href="metorigin-analysis.html#cb234-83" tabindex="-1"></a>      <span class="fu">as.data.frame</span>()</span>
<span id="cb234-84"><a href="metorigin-analysis.html#cb234-84" tabindex="-1"></a></span>
<span id="cb234-85"><a href="metorigin-analysis.html#cb234-85" tabindex="-1"></a>  <span class="fu">colnames</span>(metadata)[<span class="fu">which</span>(<span class="fu">colnames</span>(metadata) <span class="sc">==</span> group)] <span class="ot">&lt;-</span> <span class="st">&quot;CompVar&quot;</span></span>
<span id="cb234-86"><a href="metorigin-analysis.html#cb234-86" tabindex="-1"></a>  phenotype <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span></span>
<span id="cb234-87"><a href="metorigin-analysis.html#cb234-87" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(CompVar <span class="sc">%in%</span> group_names) <span class="sc">%&gt;%</span></span>
<span id="cb234-88"><a href="metorigin-analysis.html#cb234-88" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">CompVar =</span> <span class="fu">as.character</span>(CompVar)) <span class="sc">%&gt;%</span></span>
<span id="cb234-89"><a href="metorigin-analysis.html#cb234-89" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">CompVar =</span> <span class="fu">factor</span>(CompVar, <span class="at">levels =</span> group_names))</span>
<span id="cb234-90"><a href="metorigin-analysis.html#cb234-90" tabindex="-1"></a>  </span>
<span id="cb234-91"><a href="metorigin-analysis.html#cb234-91" tabindex="-1"></a>  sid <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(phenotype), <span class="fu">colnames</span>(profile))</span>
<span id="cb234-92"><a href="metorigin-analysis.html#cb234-92" tabindex="-1"></a>  phen <span class="ot">&lt;-</span> phenotype[<span class="fu">pmatch</span>(sid, <span class="fu">rownames</span>(phenotype)), , ]</span>
<span id="cb234-93"><a href="metorigin-analysis.html#cb234-93" tabindex="-1"></a>  prof <span class="ot">&lt;-</span> profile <span class="sc">%&gt;%</span></span>
<span id="cb234-94"><a href="metorigin-analysis.html#cb234-94" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(sid))</span>
<span id="cb234-95"><a href="metorigin-analysis.html#cb234-95" tabindex="-1"></a>  </span>
<span id="cb234-96"><a href="metorigin-analysis.html#cb234-96" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">all</span>(<span class="fu">colnames</span>(prof) <span class="sc">==</span> <span class="fu">rownames</span>(phen))) {</span>
<span id="cb234-97"><a href="metorigin-analysis.html#cb234-97" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;Wrong Order&quot;</span>)</span>
<span id="cb234-98"><a href="metorigin-analysis.html#cb234-98" tabindex="-1"></a>  }</span>
<span id="cb234-99"><a href="metorigin-analysis.html#cb234-99" tabindex="-1"></a>  </span>
<span id="cb234-100"><a href="metorigin-analysis.html#cb234-100" tabindex="-1"></a>  dataMatrix <span class="ot">&lt;-</span> prof <span class="sc">%&gt;%</span> base<span class="sc">::</span><span class="fu">t</span>() <span class="co"># row-&gt;sampleID; col-&gt;features</span></span>
<span id="cb234-101"><a href="metorigin-analysis.html#cb234-101" tabindex="-1"></a>  sampleMetadata <span class="ot">&lt;-</span> phen <span class="co"># row-&gt;sampleID; col-&gt;features</span></span>
<span id="cb234-102"><a href="metorigin-analysis.html#cb234-102" tabindex="-1"></a>  </span>
<span id="cb234-103"><a href="metorigin-analysis.html#cb234-103" tabindex="-1"></a>  comparsionVn <span class="ot">&lt;-</span> sampleMetadata[, <span class="st">&quot;CompVar&quot;</span>]</span>
<span id="cb234-104"><a href="metorigin-analysis.html#cb234-104" tabindex="-1"></a>  </span>
<span id="cb234-105"><a href="metorigin-analysis.html#cb234-105" tabindex="-1"></a>  <span class="co"># corrlation between group and features</span></span>
<span id="cb234-106"><a href="metorigin-analysis.html#cb234-106" tabindex="-1"></a>  pvaVn <span class="ot">&lt;-</span> <span class="fu">apply</span>(dataMatrix, <span class="dv">2</span>,</span>
<span id="cb234-107"><a href="metorigin-analysis.html#cb234-107" tabindex="-1"></a>                 <span class="cf">function</span>(feaVn) <span class="fu">cor.test</span>(<span class="fu">as.numeric</span>(comparsionVn), feaVn)[[<span class="st">&quot;p.value&quot;</span>]])</span>
<span id="cb234-108"><a href="metorigin-analysis.html#cb234-108" tabindex="-1"></a>  </span>
<span id="cb234-109"><a href="metorigin-analysis.html#cb234-109" tabindex="-1"></a>  <span class="cf">if</span> (VIPtype <span class="sc">==</span> <span class="st">&quot;OPLS&quot;</span>) {</span>
<span id="cb234-110"><a href="metorigin-analysis.html#cb234-110" tabindex="-1"></a>    vipVn <span class="ot">&lt;-</span> <span class="fu">getVipVn</span>(<span class="fu">opls</span>(dataMatrix, </span>
<span id="cb234-111"><a href="metorigin-analysis.html#cb234-111" tabindex="-1"></a>                           comparsionVn,</span>
<span id="cb234-112"><a href="metorigin-analysis.html#cb234-112" tabindex="-1"></a>                           <span class="at">predI =</span> <span class="dv">1</span>, </span>
<span id="cb234-113"><a href="metorigin-analysis.html#cb234-113" tabindex="-1"></a>                           <span class="at">orthoI =</span> <span class="cn">NA</span>,</span>
<span id="cb234-114"><a href="metorigin-analysis.html#cb234-114" tabindex="-1"></a>                           <span class="at">fig.pdfC =</span> <span class="st">&quot;none&quot;</span>))    </span>
<span id="cb234-115"><a href="metorigin-analysis.html#cb234-115" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb234-116"><a href="metorigin-analysis.html#cb234-116" tabindex="-1"></a>    vipVn <span class="ot">&lt;-</span> <span class="fu">getVipVn</span>(<span class="fu">opls</span>(dataMatrix, </span>
<span id="cb234-117"><a href="metorigin-analysis.html#cb234-117" tabindex="-1"></a>                           comparsionVn,</span>
<span id="cb234-118"><a href="metorigin-analysis.html#cb234-118" tabindex="-1"></a>                           <span class="at">predI =</span> <span class="dv">1</span>, </span>
<span id="cb234-119"><a href="metorigin-analysis.html#cb234-119" tabindex="-1"></a>                           <span class="at">fig.pdfC =</span> <span class="st">&quot;none&quot;</span>))    </span>
<span id="cb234-120"><a href="metorigin-analysis.html#cb234-120" tabindex="-1"></a>  }</span>
<span id="cb234-121"><a href="metorigin-analysis.html#cb234-121" tabindex="-1"></a>  quantVn <span class="ot">&lt;-</span> <span class="fu">qnorm</span>(<span class="dv">1</span> <span class="sc">-</span> pvaVn <span class="sc">/</span> <span class="dv">2</span>)</span>
<span id="cb234-122"><a href="metorigin-analysis.html#cb234-122" tabindex="-1"></a>  rmsQuantN <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="fu">mean</span>(quantVn<span class="sc">^</span><span class="dv">2</span>))</span>
<span id="cb234-123"><a href="metorigin-analysis.html#cb234-123" tabindex="-1"></a>  </span>
<span id="cb234-124"><a href="metorigin-analysis.html#cb234-124" tabindex="-1"></a>  opar <span class="ot">&lt;-</span> <span class="fu">par</span>(<span class="at">font =</span> <span class="dv">2</span>, <span class="at">font.axis =</span> <span class="dv">2</span>, <span class="at">font.lab =</span> <span class="dv">2</span>,</span>
<span id="cb234-125"><a href="metorigin-analysis.html#cb234-125" tabindex="-1"></a>              <span class="at">las =</span> <span class="dv">1</span>,</span>
<span id="cb234-126"><a href="metorigin-analysis.html#cb234-126" tabindex="-1"></a>              <span class="at">mar =</span> <span class="fu">c</span>(<span class="fl">5.1</span>, <span class="fl">4.6</span>, <span class="fl">4.1</span>, <span class="fl">2.1</span>),</span>
<span id="cb234-127"><a href="metorigin-analysis.html#cb234-127" tabindex="-1"></a>              <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">pch =</span> <span class="dv">16</span>)</span>
<span id="cb234-128"><a href="metorigin-analysis.html#cb234-128" tabindex="-1"></a>  <span class="fu">plot</span>(pvaVn, </span>
<span id="cb234-129"><a href="metorigin-analysis.html#cb234-129" tabindex="-1"></a>       vipVn,</span>
<span id="cb234-130"><a href="metorigin-analysis.html#cb234-130" tabindex="-1"></a>       <span class="at">col =</span> <span class="st">&quot;red&quot;</span>,</span>
<span id="cb234-131"><a href="metorigin-analysis.html#cb234-131" tabindex="-1"></a>       <span class="at">pch =</span> <span class="dv">16</span>,</span>
<span id="cb234-132"><a href="metorigin-analysis.html#cb234-132" tabindex="-1"></a>       <span class="at">xlab =</span> <span class="st">&quot;p-value&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;VIP&quot;</span>, <span class="at">xaxs =</span> <span class="st">&quot;i&quot;</span>, <span class="at">yaxs =</span> <span class="st">&quot;i&quot;</span>)</span>
<span id="cb234-133"><a href="metorigin-analysis.html#cb234-133" tabindex="-1"></a>  <span class="fu">box</span>(<span class="at">lwd =</span> <span class="dv">2</span>)</span>
<span id="cb234-134"><a href="metorigin-analysis.html#cb234-134" tabindex="-1"></a>  <span class="fu">curve</span>(<span class="fu">qnorm</span>(<span class="dv">1</span> <span class="sc">-</span> x <span class="sc">/</span> <span class="dv">2</span>) <span class="sc">/</span> rmsQuantN, <span class="dv">0</span>, <span class="dv">1</span>, <span class="at">add =</span> <span class="cn">TRUE</span>, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>, <span class="at">lwd =</span> <span class="dv">3</span>)</span>
<span id="cb234-135"><a href="metorigin-analysis.html#cb234-135" tabindex="-1"></a>  <span class="fu">abline</span>(<span class="at">h =</span> <span class="dv">1</span>, <span class="at">col =</span> <span class="st">&quot;blue&quot;</span>)</span>
<span id="cb234-136"><a href="metorigin-analysis.html#cb234-136" tabindex="-1"></a>  <span class="fu">abline</span>(<span class="at">v =</span> <span class="fl">0.05</span>, <span class="at">col =</span> <span class="st">&quot;blue&quot;</span>)</span>
<span id="cb234-137"><a href="metorigin-analysis.html#cb234-137" tabindex="-1"></a>  </span>
<span id="cb234-138"><a href="metorigin-analysis.html#cb234-138" tabindex="-1"></a>  res_temp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb234-139"><a href="metorigin-analysis.html#cb234-139" tabindex="-1"></a>      <span class="at">FeatureID =</span> <span class="fu">names</span>(vipVn),</span>
<span id="cb234-140"><a href="metorigin-analysis.html#cb234-140" tabindex="-1"></a>      <span class="at">VIP =</span> vipVn,</span>
<span id="cb234-141"><a href="metorigin-analysis.html#cb234-141" tabindex="-1"></a>      <span class="at">CorPvalue =</span> pvaVn) <span class="sc">%&gt;%</span></span>
<span id="cb234-142"><a href="metorigin-analysis.html#cb234-142" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">desc</span>(VIP))</span>
<span id="cb234-143"><a href="metorigin-analysis.html#cb234-143" tabindex="-1"></a></span>
<span id="cb234-144"><a href="metorigin-analysis.html#cb234-144" tabindex="-1"></a>  vip_select <span class="ot">&lt;-</span> res_temp <span class="sc">%&gt;%</span></span>
<span id="cb234-145"><a href="metorigin-analysis.html#cb234-145" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(VIP <span class="sc">&gt;</span> vip_cutoff) </span>
<span id="cb234-146"><a href="metorigin-analysis.html#cb234-146" tabindex="-1"></a>  </span>
<span id="cb234-147"><a href="metorigin-analysis.html#cb234-147" tabindex="-1"></a>  pl <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(vip_select, <span class="fu">aes</span>(FeatureID, VIP)) <span class="sc">+</span></span>
<span id="cb234-148"><a href="metorigin-analysis.html#cb234-148" tabindex="-1"></a>    <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> FeatureID, <span class="at">xend =</span> FeatureID,</span>
<span id="cb234-149"><a href="metorigin-analysis.html#cb234-149" tabindex="-1"></a>                     <span class="at">y =</span> <span class="dv">0</span>, <span class="at">yend =</span> VIP)) <span class="sc">+</span></span>
<span id="cb234-150"><a href="metorigin-analysis.html#cb234-150" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">21</span>, <span class="at">size =</span> <span class="dv">5</span>, <span class="at">color =</span> <span class="st">&#39;#008000&#39;</span> ,<span class="at">fill =</span> <span class="st">&#39;#008000&#39;</span>) <span class="sc">+</span></span>
<span id="cb234-151"><a href="metorigin-analysis.html#cb234-151" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="dv">1</span>,<span class="fl">2.5</span>), <span class="at">color =</span> <span class="st">&#39;white&#39;</span>) <span class="sc">+</span></span>
<span id="cb234-152"><a href="metorigin-analysis.html#cb234-152" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">1</span>, <span class="at">linetype =</span> <span class="st">&#39;dashed&#39;</span>) <span class="sc">+</span></span>
<span id="cb234-153"><a href="metorigin-analysis.html#cb234-153" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb234-154"><a href="metorigin-analysis.html#cb234-154" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&#39;&#39;</span>, <span class="at">y =</span> <span class="st">&#39;VIP value&#39;</span>) <span class="sc">+</span></span>
<span id="cb234-155"><a href="metorigin-analysis.html#cb234-155" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb234-156"><a href="metorigin-analysis.html#cb234-156" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&#39;none&#39;</span>,</span>
<span id="cb234-157"><a href="metorigin-analysis.html#cb234-157" tabindex="-1"></a>          <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">&#39;black&#39;</span>,<span class="at">size =</span> <span class="dv">12</span>, <span class="at">family =</span> <span class="st">&#39;Arial&#39;</span>, <span class="at">face =</span> <span class="st">&#39;plain&#39;</span>),</span>
<span id="cb234-158"><a href="metorigin-analysis.html#cb234-158" tabindex="-1"></a>          <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb234-159"><a href="metorigin-analysis.html#cb234-159" tabindex="-1"></a>          <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb234-160"><a href="metorigin-analysis.html#cb234-160" tabindex="-1"></a>          <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">&#39;black&#39;</span>,<span class="at">size =</span> <span class="dv">15</span>, <span class="at">family =</span> <span class="st">&#39;Arial&#39;</span>, <span class="at">face =</span> <span class="st">&#39;plain&#39;</span>),</span>
<span id="cb234-161"><a href="metorigin-analysis.html#cb234-161" tabindex="-1"></a>          <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>),</span>
<span id="cb234-162"><a href="metorigin-analysis.html#cb234-162" tabindex="-1"></a>          <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">&#39;black&#39;</span>,<span class="at">size =</span> <span class="dv">15</span>, <span class="at">family =</span> <span class="st">&#39;Arial&#39;</span>, <span class="at">face =</span> <span class="st">&#39;plain&#39;</span>),</span>
<span id="cb234-163"><a href="metorigin-analysis.html#cb234-163" tabindex="-1"></a>          <span class="at">axis.ticks =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb234-164"><a href="metorigin-analysis.html#cb234-164" tabindex="-1"></a>          <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>())</span>
<span id="cb234-165"><a href="metorigin-analysis.html#cb234-165" tabindex="-1"></a>  </span>
<span id="cb234-166"><a href="metorigin-analysis.html#cb234-166" tabindex="-1"></a>  <span class="co"># Number of Group</span></span>
<span id="cb234-167"><a href="metorigin-analysis.html#cb234-167" tabindex="-1"></a>  dat_status <span class="ot">&lt;-</span> <span class="fu">table</span>(phen<span class="sc">$</span>CompVar)</span>
<span id="cb234-168"><a href="metorigin-analysis.html#cb234-168" tabindex="-1"></a>  dat_status_number <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(dat_status)</span>
<span id="cb234-169"><a href="metorigin-analysis.html#cb234-169" tabindex="-1"></a>  dat_status_name <span class="ot">&lt;-</span> <span class="fu">names</span>(dat_status)</span>
<span id="cb234-170"><a href="metorigin-analysis.html#cb234-170" tabindex="-1"></a>  res_temp<span class="sc">$</span>Block <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">paste</span>(dat_status_number[<span class="dv">1</span>], dat_status_name[<span class="dv">1</span>], <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>),</span>
<span id="cb234-171"><a href="metorigin-analysis.html#cb234-171" tabindex="-1"></a>                       <span class="st">&quot;vs&quot;</span>,</span>
<span id="cb234-172"><a href="metorigin-analysis.html#cb234-172" tabindex="-1"></a>                       <span class="fu">paste</span>(dat_status_number[<span class="dv">2</span>], dat_status_name[<span class="dv">2</span>], <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>))</span>
<span id="cb234-173"><a href="metorigin-analysis.html#cb234-173" tabindex="-1"></a></span>
<span id="cb234-174"><a href="metorigin-analysis.html#cb234-174" tabindex="-1"></a>  res_df <span class="ot">&lt;-</span> res_temp <span class="sc">%&gt;%</span></span>
<span id="cb234-175"><a href="metorigin-analysis.html#cb234-175" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(FeatureID, Block, <span class="fu">everything</span>())    </span>
<span id="cb234-176"><a href="metorigin-analysis.html#cb234-176" tabindex="-1"></a></span>
<span id="cb234-177"><a href="metorigin-analysis.html#cb234-177" tabindex="-1"></a></span>
<span id="cb234-178"><a href="metorigin-analysis.html#cb234-178" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">vip =</span> res_df,</span>
<span id="cb234-179"><a href="metorigin-analysis.html#cb234-179" tabindex="-1"></a>              <span class="at">plot =</span> pl)</span>
<span id="cb234-180"><a href="metorigin-analysis.html#cb234-180" tabindex="-1"></a>  </span>
<span id="cb234-181"><a href="metorigin-analysis.html#cb234-181" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb234-182"><a href="metorigin-analysis.html#cb234-182" tabindex="-1"></a>}</span>
<span id="cb234-183"><a href="metorigin-analysis.html#cb234-183" tabindex="-1"></a></span>
<span id="cb234-184"><a href="metorigin-analysis.html#cb234-184" tabindex="-1"></a>t_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb234-185"><a href="metorigin-analysis.html#cb234-185" tabindex="-1"></a>    x,</span>
<span id="cb234-186"><a href="metorigin-analysis.html#cb234-186" tabindex="-1"></a>    group,</span>
<span id="cb234-187"><a href="metorigin-analysis.html#cb234-187" tabindex="-1"></a>    group_names) {</span>
<span id="cb234-188"><a href="metorigin-analysis.html#cb234-188" tabindex="-1"></a>  </span>
<span id="cb234-189"><a href="metorigin-analysis.html#cb234-189" tabindex="-1"></a>  <span class="co"># dataseat</span></span>
<span id="cb234-190"><a href="metorigin-analysis.html#cb234-190" tabindex="-1"></a>  metadata <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(x) <span class="sc">%&gt;%</span></span>
<span id="cb234-191"><a href="metorigin-analysis.html#cb234-191" tabindex="-1"></a>      <span class="fu">as.data.frame</span>()</span>
<span id="cb234-192"><a href="metorigin-analysis.html#cb234-192" tabindex="-1"></a>  profile <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(x) <span class="sc">%&gt;%</span></span>
<span id="cb234-193"><a href="metorigin-analysis.html#cb234-193" tabindex="-1"></a>      <span class="fu">as.data.frame</span>()</span>
<span id="cb234-194"><a href="metorigin-analysis.html#cb234-194" tabindex="-1"></a></span>
<span id="cb234-195"><a href="metorigin-analysis.html#cb234-195" tabindex="-1"></a>  <span class="co"># rename variables</span></span>
<span id="cb234-196"><a href="metorigin-analysis.html#cb234-196" tabindex="-1"></a>  <span class="fu">colnames</span>(metadata)[<span class="fu">which</span>(<span class="fu">colnames</span>(metadata) <span class="sc">==</span> group)] <span class="ot">&lt;-</span> <span class="st">&quot;CompVar&quot;</span> </span>
<span id="cb234-197"><a href="metorigin-analysis.html#cb234-197" tabindex="-1"></a>  </span>
<span id="cb234-198"><a href="metorigin-analysis.html#cb234-198" tabindex="-1"></a>  phenotype <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span></span>
<span id="cb234-199"><a href="metorigin-analysis.html#cb234-199" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(CompVar <span class="sc">%in%</span> group_names) <span class="sc">%&gt;%</span></span>
<span id="cb234-200"><a href="metorigin-analysis.html#cb234-200" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">CompVar =</span> <span class="fu">as.character</span>(CompVar)) <span class="sc">%&gt;%</span></span>
<span id="cb234-201"><a href="metorigin-analysis.html#cb234-201" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">CompVar =</span> <span class="fu">factor</span>(CompVar, <span class="at">levels =</span> group_names))</span>
<span id="cb234-202"><a href="metorigin-analysis.html#cb234-202" tabindex="-1"></a>  </span>
<span id="cb234-203"><a href="metorigin-analysis.html#cb234-203" tabindex="-1"></a>  sid <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(phenotype), <span class="fu">colnames</span>(profile))</span>
<span id="cb234-204"><a href="metorigin-analysis.html#cb234-204" tabindex="-1"></a>  phen <span class="ot">&lt;-</span> phenotype[<span class="fu">pmatch</span>(sid, <span class="fu">rownames</span>(phenotype)), , ]</span>
<span id="cb234-205"><a href="metorigin-analysis.html#cb234-205" tabindex="-1"></a>  prof <span class="ot">&lt;-</span> profile <span class="sc">%&gt;%</span></span>
<span id="cb234-206"><a href="metorigin-analysis.html#cb234-206" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(sid))</span>
<span id="cb234-207"><a href="metorigin-analysis.html#cb234-207" tabindex="-1"></a>  </span>
<span id="cb234-208"><a href="metorigin-analysis.html#cb234-208" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">all</span>(<span class="fu">colnames</span>(prof) <span class="sc">==</span> <span class="fu">rownames</span>(phen))) {</span>
<span id="cb234-209"><a href="metorigin-analysis.html#cb234-209" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;Wrong Order&quot;</span>)</span>
<span id="cb234-210"><a href="metorigin-analysis.html#cb234-210" tabindex="-1"></a>  }</span>
<span id="cb234-211"><a href="metorigin-analysis.html#cb234-211" tabindex="-1"></a></span>
<span id="cb234-212"><a href="metorigin-analysis.html#cb234-212" tabindex="-1"></a>  t_res <span class="ot">&lt;-</span> <span class="fu">apply</span>(prof, <span class="dv">1</span>, <span class="cf">function</span>(x1, y1) {</span>
<span id="cb234-213"><a href="metorigin-analysis.html#cb234-213" tabindex="-1"></a>    dat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">value =</span> <span class="fu">as.numeric</span>(x1), <span class="at">group =</span> y1)</span>
<span id="cb234-214"><a href="metorigin-analysis.html#cb234-214" tabindex="-1"></a>    </span>
<span id="cb234-215"><a href="metorigin-analysis.html#cb234-215" tabindex="-1"></a>    rest <span class="ot">&lt;-</span> <span class="fu">t.test</span>(<span class="at">data =</span> dat, value <span class="sc">~</span> group)</span>
<span id="cb234-216"><a href="metorigin-analysis.html#cb234-216" tabindex="-1"></a></span>
<span id="cb234-217"><a href="metorigin-analysis.html#cb234-217" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="fu">c</span>(rest<span class="sc">$</span>statistic, rest<span class="sc">$</span>p.value)</span>
<span id="cb234-218"><a href="metorigin-analysis.html#cb234-218" tabindex="-1"></a>    <span class="fu">return</span>(res)</span>
<span id="cb234-219"><a href="metorigin-analysis.html#cb234-219" tabindex="-1"></a>  }, phen<span class="sc">$</span>CompVar) <span class="sc">%&gt;%</span></span>
<span id="cb234-220"><a href="metorigin-analysis.html#cb234-220" tabindex="-1"></a>    base<span class="sc">::</span><span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb234-221"><a href="metorigin-analysis.html#cb234-221" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;Feature&quot;</span>)</span>
<span id="cb234-222"><a href="metorigin-analysis.html#cb234-222" tabindex="-1"></a>  </span>
<span id="cb234-223"><a href="metorigin-analysis.html#cb234-223" tabindex="-1"></a>  <span class="fu">colnames</span>(t_res) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;FeatureID&quot;</span>, <span class="st">&quot;Statistic&quot;</span>, <span class="st">&quot;Pvalue&quot;</span>)</span>
<span id="cb234-224"><a href="metorigin-analysis.html#cb234-224" tabindex="-1"></a>  t_res<span class="sc">$</span>AdjustedPvalue <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(<span class="fu">as.numeric</span>(t_res<span class="sc">$</span>Pvalue), <span class="at">method =</span> <span class="st">&quot;BH&quot;</span>)</span>
<span id="cb234-225"><a href="metorigin-analysis.html#cb234-225" tabindex="-1"></a>  </span>
<span id="cb234-226"><a href="metorigin-analysis.html#cb234-226" tabindex="-1"></a>  <span class="co"># Number of Group</span></span>
<span id="cb234-227"><a href="metorigin-analysis.html#cb234-227" tabindex="-1"></a>  dat_status <span class="ot">&lt;-</span> <span class="fu">table</span>(phen<span class="sc">$</span>CompVar)</span>
<span id="cb234-228"><a href="metorigin-analysis.html#cb234-228" tabindex="-1"></a>  dat_status_number <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(dat_status)</span>
<span id="cb234-229"><a href="metorigin-analysis.html#cb234-229" tabindex="-1"></a>  dat_status_name <span class="ot">&lt;-</span> <span class="fu">names</span>(dat_status)</span>
<span id="cb234-230"><a href="metorigin-analysis.html#cb234-230" tabindex="-1"></a>  t_res<span class="sc">$</span>Block <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">paste</span>(dat_status_number[<span class="dv">1</span>], dat_status_name[<span class="dv">1</span>], <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>),</span>
<span id="cb234-231"><a href="metorigin-analysis.html#cb234-231" tabindex="-1"></a>                       <span class="st">&quot;vs&quot;</span>,</span>
<span id="cb234-232"><a href="metorigin-analysis.html#cb234-232" tabindex="-1"></a>                       <span class="fu">paste</span>(dat_status_number[<span class="dv">2</span>], dat_status_name[<span class="dv">2</span>], <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>))</span>
<span id="cb234-233"><a href="metorigin-analysis.html#cb234-233" tabindex="-1"></a>  </span>
<span id="cb234-234"><a href="metorigin-analysis.html#cb234-234" tabindex="-1"></a>  res <span class="ot">&lt;-</span> t_res <span class="sc">%&gt;%</span></span>
<span id="cb234-235"><a href="metorigin-analysis.html#cb234-235" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(FeatureID, Block, <span class="fu">everything</span>())    </span>
<span id="cb234-236"><a href="metorigin-analysis.html#cb234-236" tabindex="-1"></a></span>
<span id="cb234-237"><a href="metorigin-analysis.html#cb234-237" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb234-238"><a href="metorigin-analysis.html#cb234-238" tabindex="-1"></a>}</span>
<span id="cb234-239"><a href="metorigin-analysis.html#cb234-239" tabindex="-1"></a></span>
<span id="cb234-240"><a href="metorigin-analysis.html#cb234-240" tabindex="-1"></a>mergedResults <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb234-241"><a href="metorigin-analysis.html#cb234-241" tabindex="-1"></a>    fc_result,</span>
<span id="cb234-242"><a href="metorigin-analysis.html#cb234-242" tabindex="-1"></a>    vip_result,</span>
<span id="cb234-243"><a href="metorigin-analysis.html#cb234-243" tabindex="-1"></a>    test_result,</span>
<span id="cb234-244"><a href="metorigin-analysis.html#cb234-244" tabindex="-1"></a>    group_names,</span>
<span id="cb234-245"><a href="metorigin-analysis.html#cb234-245" tabindex="-1"></a>    group_labels) {</span>
<span id="cb234-246"><a href="metorigin-analysis.html#cb234-246" tabindex="-1"></a>  </span>
<span id="cb234-247"><a href="metorigin-analysis.html#cb234-247" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(vip_result)) {</span>
<span id="cb234-248"><a href="metorigin-analysis.html#cb234-248" tabindex="-1"></a>    mdat <span class="ot">&lt;-</span> fc_result <span class="sc">%&gt;%</span></span>
<span id="cb234-249"><a href="metorigin-analysis.html#cb234-249" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Block2 =</span> <span class="fu">paste</span>(group_labels, <span class="at">collapse =</span> <span class="st">&quot; vs &quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb234-250"><a href="metorigin-analysis.html#cb234-250" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">FeatureID =</span> <span class="fu">make.names</span>(FeatureID)) <span class="sc">%&gt;%</span></span>
<span id="cb234-251"><a href="metorigin-analysis.html#cb234-251" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">&quot;Mean Abundance</span><span class="sc">\n</span><span class="st">(All)&quot;</span>, </span>
<span id="cb234-252"><a href="metorigin-analysis.html#cb234-252" tabindex="-1"></a>                              <span class="st">&quot;Mean Abundance</span><span class="sc">\n</span><span class="st">former&quot;</span>, </span>
<span id="cb234-253"><a href="metorigin-analysis.html#cb234-253" tabindex="-1"></a>                              <span class="st">&quot;Mean Abundance</span><span class="sc">\n</span><span class="st">latter&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb234-254"><a href="metorigin-analysis.html#cb234-254" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">inner_join</span>(test_result <span class="sc">%&gt;%</span></span>
<span id="cb234-255"><a href="metorigin-analysis.html#cb234-255" tabindex="-1"></a>                          dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>Block) <span class="sc">%&gt;%</span></span>
<span id="cb234-256"><a href="metorigin-analysis.html#cb234-256" tabindex="-1"></a>                          dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">FeatureID =</span> <span class="fu">make.names</span>(FeatureID)),</span>
<span id="cb234-257"><a href="metorigin-analysis.html#cb234-257" tabindex="-1"></a>                        <span class="at">by =</span> <span class="st">&quot;FeatureID&quot;</span>) </span>
<span id="cb234-258"><a href="metorigin-analysis.html#cb234-258" tabindex="-1"></a>    </span>
<span id="cb234-259"><a href="metorigin-analysis.html#cb234-259" tabindex="-1"></a>    res <span class="ot">&lt;-</span> mdat <span class="sc">%&gt;%</span></span>
<span id="cb234-260"><a href="metorigin-analysis.html#cb234-260" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(FeatureID, Block2, Block,</span>
<span id="cb234-261"><a href="metorigin-analysis.html#cb234-261" tabindex="-1"></a>                    FoldChange, Log2FoldChange,</span>
<span id="cb234-262"><a href="metorigin-analysis.html#cb234-262" tabindex="-1"></a>                    Statistic, Pvalue, AdjustedPvalue,</span>
<span id="cb234-263"><a href="metorigin-analysis.html#cb234-263" tabindex="-1"></a>                    <span class="fu">everything</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb234-264"><a href="metorigin-analysis.html#cb234-264" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">arrange</span>(AdjustedPvalue, Log2FoldChange)    </span>
<span id="cb234-265"><a href="metorigin-analysis.html#cb234-265" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb234-266"><a href="metorigin-analysis.html#cb234-266" tabindex="-1"></a>    mdat <span class="ot">&lt;-</span> fc_result <span class="sc">%&gt;%</span></span>
<span id="cb234-267"><a href="metorigin-analysis.html#cb234-267" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Block2 =</span> <span class="fu">paste</span>(group_labels, <span class="at">collapse =</span> <span class="st">&quot; vs &quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb234-268"><a href="metorigin-analysis.html#cb234-268" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">FeatureID =</span> <span class="fu">make.names</span>(FeatureID)) <span class="sc">%&gt;%</span></span>
<span id="cb234-269"><a href="metorigin-analysis.html#cb234-269" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">&quot;Mean Abundance</span><span class="sc">\n</span><span class="st">(All)&quot;</span>, </span>
<span id="cb234-270"><a href="metorigin-analysis.html#cb234-270" tabindex="-1"></a>                              <span class="st">&quot;Mean Abundance</span><span class="sc">\n</span><span class="st">former&quot;</span>, </span>
<span id="cb234-271"><a href="metorigin-analysis.html#cb234-271" tabindex="-1"></a>                              <span class="st">&quot;Mean Abundance</span><span class="sc">\n</span><span class="st">latter&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb234-272"><a href="metorigin-analysis.html#cb234-272" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">inner_join</span>(vip_result <span class="sc">%&gt;%</span></span>
<span id="cb234-273"><a href="metorigin-analysis.html#cb234-273" tabindex="-1"></a>                          dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>Block) <span class="sc">%&gt;%</span></span>
<span id="cb234-274"><a href="metorigin-analysis.html#cb234-274" tabindex="-1"></a>                          dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">FeatureID =</span> <span class="fu">make.names</span>(FeatureID)),</span>
<span id="cb234-275"><a href="metorigin-analysis.html#cb234-275" tabindex="-1"></a>                        <span class="at">by =</span> <span class="st">&quot;FeatureID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb234-276"><a href="metorigin-analysis.html#cb234-276" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">inner_join</span>(test_result <span class="sc">%&gt;%</span></span>
<span id="cb234-277"><a href="metorigin-analysis.html#cb234-277" tabindex="-1"></a>                          dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>Block) <span class="sc">%&gt;%</span></span>
<span id="cb234-278"><a href="metorigin-analysis.html#cb234-278" tabindex="-1"></a>                          dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">FeatureID =</span> <span class="fu">make.names</span>(FeatureID)),</span>
<span id="cb234-279"><a href="metorigin-analysis.html#cb234-279" tabindex="-1"></a>                        <span class="at">by =</span> <span class="st">&quot;FeatureID&quot;</span>) </span>
<span id="cb234-280"><a href="metorigin-analysis.html#cb234-280" tabindex="-1"></a>    </span>
<span id="cb234-281"><a href="metorigin-analysis.html#cb234-281" tabindex="-1"></a>    res <span class="ot">&lt;-</span> mdat <span class="sc">%&gt;%</span></span>
<span id="cb234-282"><a href="metorigin-analysis.html#cb234-282" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(FeatureID, Block2, Block,</span>
<span id="cb234-283"><a href="metorigin-analysis.html#cb234-283" tabindex="-1"></a>                    FoldChange, Log2FoldChange,</span>
<span id="cb234-284"><a href="metorigin-analysis.html#cb234-284" tabindex="-1"></a>                    VIP, CorPvalue,</span>
<span id="cb234-285"><a href="metorigin-analysis.html#cb234-285" tabindex="-1"></a>                    Statistic, Pvalue, AdjustedPvalue,</span>
<span id="cb234-286"><a href="metorigin-analysis.html#cb234-286" tabindex="-1"></a>                    <span class="fu">everything</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb234-287"><a href="metorigin-analysis.html#cb234-287" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">arrange</span>(AdjustedPvalue, Log2FoldChange)    </span>
<span id="cb234-288"><a href="metorigin-analysis.html#cb234-288" tabindex="-1"></a>  }</span>
<span id="cb234-289"><a href="metorigin-analysis.html#cb234-289" tabindex="-1"></a>  </span>
<span id="cb234-290"><a href="metorigin-analysis.html#cb234-290" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb234-291"><a href="metorigin-analysis.html#cb234-291" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="de-metabolites" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">5.4</span> DE metabolites<a href="metorigin-analysis.html#de-metabolites" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb235"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb235-1"><a href="metorigin-analysis.html#cb235-1" tabindex="-1"></a>fc_res <span class="ot">&lt;-</span> <span class="fu">FoldChange</span>(</span>
<span id="cb235-2"><a href="metorigin-analysis.html#cb235-2" tabindex="-1"></a>           <span class="at">x =</span> se_impute,</span>
<span id="cb235-3"><a href="metorigin-analysis.html#cb235-3" tabindex="-1"></a>           <span class="at">group =</span> <span class="st">&quot;group&quot;</span>,</span>
<span id="cb235-4"><a href="metorigin-analysis.html#cb235-4" tabindex="-1"></a>           <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Moderate&quot;</span>))</span>
<span id="cb235-5"><a href="metorigin-analysis.html#cb235-5" tabindex="-1"></a></span>
<span id="cb235-6"><a href="metorigin-analysis.html#cb235-6" tabindex="-1"></a>vip_res <span class="ot">&lt;-</span> <span class="fu">VIP_fun</span>(</span>
<span id="cb235-7"><a href="metorigin-analysis.html#cb235-7" tabindex="-1"></a>  <span class="at">x =</span> se_norm,</span>
<span id="cb235-8"><a href="metorigin-analysis.html#cb235-8" tabindex="-1"></a>  <span class="at">group =</span> <span class="st">&quot;group&quot;</span>,</span>
<span id="cb235-9"><a href="metorigin-analysis.html#cb235-9" tabindex="-1"></a>  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Severe&quot;</span>),</span>
<span id="cb235-10"><a href="metorigin-analysis.html#cb235-10" tabindex="-1"></a>  <span class="at">VIPtype =</span> <span class="st">&quot;PLS&quot;</span>,</span>
<span id="cb235-11"><a href="metorigin-analysis.html#cb235-11" tabindex="-1"></a>  <span class="at">vip_cutoff =</span> <span class="dv">1</span>)</span></code></pre></div>
<pre><code>## PLS-DA
## 26 samples x 167 variables and 1 response
## standard scaling of predictors and response(s)
##       R2X(cum) R2Y(cum) Q2(cum) RMSEE pre ort pR2Y  pQ2
## Total     0.12    0.692    0.33 0.288   1   0  0.2 0.05</code></pre>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-141-1.png" width="672" /></p>
<div class="sourceCode" id="cb237"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb237-1"><a href="metorigin-analysis.html#cb237-1" tabindex="-1"></a>ttest_res <span class="ot">&lt;-</span> <span class="fu">t_fun</span>(</span>
<span id="cb237-2"><a href="metorigin-analysis.html#cb237-2" tabindex="-1"></a>  <span class="at">x =</span> se_norm,</span>
<span id="cb237-3"><a href="metorigin-analysis.html#cb237-3" tabindex="-1"></a>  <span class="at">group =</span> <span class="st">&quot;group&quot;</span>,</span>
<span id="cb237-4"><a href="metorigin-analysis.html#cb237-4" tabindex="-1"></a>  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Severe&quot;</span>))</span>
<span id="cb237-5"><a href="metorigin-analysis.html#cb237-5" tabindex="-1"></a></span>
<span id="cb237-6"><a href="metorigin-analysis.html#cb237-6" tabindex="-1"></a>m_results <span class="ot">&lt;-</span> <span class="fu">mergedResults</span>(</span>
<span id="cb237-7"><a href="metorigin-analysis.html#cb237-7" tabindex="-1"></a>  <span class="at">fc_result =</span> fc_res,</span>
<span id="cb237-8"><a href="metorigin-analysis.html#cb237-8" tabindex="-1"></a>  <span class="at">vip_result =</span> vip_res<span class="sc">$</span>vip,</span>
<span id="cb237-9"><a href="metorigin-analysis.html#cb237-9" tabindex="-1"></a>  <span class="at">test_result =</span> ttest_res,</span>
<span id="cb237-10"><a href="metorigin-analysis.html#cb237-10" tabindex="-1"></a>  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Severe&quot;</span>),</span>
<span id="cb237-11"><a href="metorigin-analysis.html#cb237-11" tabindex="-1"></a>  <span class="at">group_labels =</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Severe&quot;</span>))</span>
<span id="cb237-12"><a href="metorigin-analysis.html#cb237-12" tabindex="-1"></a></span>
<span id="cb237-13"><a href="metorigin-analysis.html#cb237-13" tabindex="-1"></a><span class="fu">head</span>(m_results)</span></code></pre></div>
<pre><code>##   FeatureID         Block2                  Block FoldChange Log2FoldChange      VIP   CorPvalue Statistic      Pvalue AdjustedPvalue
## 1   M_42449 Mild vs Severe 14_Mild vs 19_Moderate  0.6571801     -0.6056393 2.362694 0.002247897 -3.498896 0.001881585      0.2132448
## 2   M_52446 Mild vs Severe 14_Mild vs 19_Moderate  0.7040861     -0.5061763 2.059461 0.009476225 -2.892892 0.008125613      0.2132448
## 3    M_1566 Mild vs Severe 14_Mild vs 19_Moderate  0.7161609     -0.4816443 2.143655 0.006556886 -3.077688 0.005431005      0.2132448
## 4   M_19263 Mild vs Severe 14_Mild vs 19_Moderate  0.7165011     -0.4809592 2.128317 0.007023483 -3.073546 0.005774977      0.2132448
## 5   M_42448 Mild vs Severe 14_Mild vs 19_Moderate  0.7577644     -0.4001788 2.006246 0.011828403 -2.793941 0.010215320      0.2132448
## 6   M_43761 Mild vs Severe 14_Mild vs 19_Moderate  0.8685495     -0.2033201 2.043515 0.010136007 -2.822338 0.009428795      0.2132448</code></pre>
</div>
<div id="obtain-inputs-for-metorigin" class="section level2 hasAnchor" number="5.5">
<h2><span class="header-section-number">5.5</span> Obtain inputs for Metorigin<a href="metorigin-analysis.html#obtain-inputs-for-metorigin" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<blockquote>
<p>The metabolite table must contain at least one column of “HMDBID”, “KEGGID” or “Name”, and a column of 0/1 values indicating statistical significance (1-significant, 0-nonsignificant). If the “Diff” column is missing, all metabolites will be considered as differential metabolites.</p>
</blockquote>
<div class="sourceCode" id="cb239"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb239-1"><a href="metorigin-analysis.html#cb239-1" tabindex="-1"></a>get_metabolites <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb239-2"><a href="metorigin-analysis.html#cb239-2" tabindex="-1"></a>  <span class="at">x =</span> ExprSet,</span>
<span id="cb239-3"><a href="metorigin-analysis.html#cb239-3" tabindex="-1"></a>  dat,</span>
<span id="cb239-4"><a href="metorigin-analysis.html#cb239-4" tabindex="-1"></a>  group_names,</span>
<span id="cb239-5"><a href="metorigin-analysis.html#cb239-5" tabindex="-1"></a>  <span class="at">index_names =</span> <span class="fu">c</span>(<span class="st">&quot;FoldChange&quot;</span>, <span class="st">&quot;Log2FoldChange&quot;</span>, <span class="st">&quot;VIP&quot;</span>, <span class="st">&quot;CorPvalue&quot;</span>, <span class="st">&quot;Pvalue&quot;</span>, <span class="st">&quot;AdjustedPvalue&quot;</span>),</span>
<span id="cb239-6"><a href="metorigin-analysis.html#cb239-6" tabindex="-1"></a>  <span class="at">index_cutoff =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="fl">0.05</span>, <span class="fl">0.05</span>, <span class="fl">0.2</span>)) {</span>
<span id="cb239-7"><a href="metorigin-analysis.html#cb239-7" tabindex="-1"></a>  </span>
<span id="cb239-8"><a href="metorigin-analysis.html#cb239-8" tabindex="-1"></a>  <span class="co"># x = ExprSet</span></span>
<span id="cb239-9"><a href="metorigin-analysis.html#cb239-9" tabindex="-1"></a>  <span class="co"># dat = m_results</span></span>
<span id="cb239-10"><a href="metorigin-analysis.html#cb239-10" tabindex="-1"></a>  <span class="co"># group_names = &quot;Mild vs Severe&quot;</span></span>
<span id="cb239-11"><a href="metorigin-analysis.html#cb239-11" tabindex="-1"></a>  <span class="co"># index_names = c(&quot;Log2FoldChange&quot;, &quot;AdjustedPvalue&quot;)</span></span>
<span id="cb239-12"><a href="metorigin-analysis.html#cb239-12" tabindex="-1"></a>  <span class="co"># index_cutoff = c(0, 0.2)</span></span>
<span id="cb239-13"><a href="metorigin-analysis.html#cb239-13" tabindex="-1"></a>  </span>
<span id="cb239-14"><a href="metorigin-analysis.html#cb239-14" tabindex="-1"></a>  feature <span class="ot">&lt;-</span> Biobase<span class="sc">::</span><span class="fu">fData</span>(x)</span>
<span id="cb239-15"><a href="metorigin-analysis.html#cb239-15" tabindex="-1"></a>  <span class="fu">colnames</span>(feature)[<span class="fu">which</span>(<span class="fu">colnames</span>(feature) <span class="sc">==</span> <span class="st">&quot;SampleID HMDBID&quot;</span>)] <span class="ot">&lt;-</span> <span class="st">&quot;HMDB&quot;</span></span>
<span id="cb239-16"><a href="metorigin-analysis.html#cb239-16" tabindex="-1"></a>  <span class="fu">colnames</span>(feature)[<span class="fu">which</span>(<span class="fu">colnames</span>(feature) <span class="sc">==</span> <span class="st">&quot;KEGG&quot;</span>)] <span class="ot">&lt;-</span> <span class="st">&quot;cpd_ID&quot;</span></span>
<span id="cb239-17"><a href="metorigin-analysis.html#cb239-17" tabindex="-1"></a>  <span class="fu">colnames</span>(feature)[<span class="fu">which</span>(<span class="fu">colnames</span>(feature) <span class="sc">==</span> <span class="st">&quot;BIOCHEMICAL&quot;</span>)] <span class="ot">&lt;-</span> <span class="st">&quot;Compounds&quot;</span></span>
<span id="cb239-18"><a href="metorigin-analysis.html#cb239-18" tabindex="-1"></a>  feature<span class="sc">$</span>HMDB <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;,</span><span class="sc">\\</span><span class="st">S+&quot;</span>, <span class="st">&quot;&quot;</span>, feature<span class="sc">$</span>HMDB)</span>
<span id="cb239-19"><a href="metorigin-analysis.html#cb239-19" tabindex="-1"></a>  </span>
<span id="cb239-20"><a href="metorigin-analysis.html#cb239-20" tabindex="-1"></a>  temp_dat <span class="ot">&lt;-</span> dat <span class="sc">%&gt;%</span></span>
<span id="cb239-21"><a href="metorigin-analysis.html#cb239-21" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(Block2 <span class="sc">%in%</span> group_names) <span class="sc">%&gt;%</span></span>
<span id="cb239-22"><a href="metorigin-analysis.html#cb239-22" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">inner_join</span>(feature <span class="sc">%&gt;%</span></span>
<span id="cb239-23"><a href="metorigin-analysis.html#cb239-23" tabindex="-1"></a>                        tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;FeatureID&quot;</span>),</span>
<span id="cb239-24"><a href="metorigin-analysis.html#cb239-24" tabindex="-1"></a>                      <span class="at">by =</span> <span class="st">&quot;FeatureID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb239-25"><a href="metorigin-analysis.html#cb239-25" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(HMDB <span class="sc">!=</span> <span class="st">&quot;-&quot;</span>)</span>
<span id="cb239-26"><a href="metorigin-analysis.html#cb239-26" tabindex="-1"></a>  </span>
<span id="cb239-27"><a href="metorigin-analysis.html#cb239-27" tabindex="-1"></a>  <span class="fu">colnames</span>(temp_dat)[<span class="fu">which</span>(<span class="fu">colnames</span>(temp_dat) <span class="sc">==</span> index_names[<span class="dv">1</span>])] <span class="ot">&lt;-</span> <span class="st">&quot;DA_index1&quot;</span></span>
<span id="cb239-28"><a href="metorigin-analysis.html#cb239-28" tabindex="-1"></a>  <span class="fu">colnames</span>(temp_dat)[<span class="fu">which</span>(<span class="fu">colnames</span>(temp_dat) <span class="sc">==</span> index_names[<span class="dv">2</span>])] <span class="ot">&lt;-</span> <span class="st">&quot;DA_index2&quot;</span></span>
<span id="cb239-29"><a href="metorigin-analysis.html#cb239-29" tabindex="-1"></a>  </span>
<span id="cb239-30"><a href="metorigin-analysis.html#cb239-30" tabindex="-1"></a>  temp_dat_diff <span class="ot">&lt;-</span> temp_dat <span class="sc">%&gt;%</span></span>
<span id="cb239-31"><a href="metorigin-analysis.html#cb239-31" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">abs</span>(DA_index1) <span class="sc">&gt;</span> index_cutoff[<span class="dv">1</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb239-32"><a href="metorigin-analysis.html#cb239-32" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(DA_index2 <span class="sc">&lt;</span> index_cutoff[<span class="dv">2</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb239-33"><a href="metorigin-analysis.html#cb239-33" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Diff =</span> <span class="dv">1</span>)</span>
<span id="cb239-34"><a href="metorigin-analysis.html#cb239-34" tabindex="-1"></a>  </span>
<span id="cb239-35"><a href="metorigin-analysis.html#cb239-35" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">nrow</span>(temp_dat_diff) <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb239-36"><a href="metorigin-analysis.html#cb239-36" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;Beyond these thresholds, no significant metabolites were selected&quot;</span>)</span>
<span id="cb239-37"><a href="metorigin-analysis.html#cb239-37" tabindex="-1"></a>  }</span>
<span id="cb239-38"><a href="metorigin-analysis.html#cb239-38" tabindex="-1"></a>  </span>
<span id="cb239-39"><a href="metorigin-analysis.html#cb239-39" tabindex="-1"></a>  temp_dat_nodiff <span class="ot">&lt;-</span> temp_dat <span class="sc">%&gt;%</span></span>
<span id="cb239-40"><a href="metorigin-analysis.html#cb239-40" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>HMDB <span class="sc">%in%</span> temp_dat_diff<span class="sc">$</span>HMDB) <span class="sc">%&gt;%</span></span>
<span id="cb239-41"><a href="metorigin-analysis.html#cb239-41" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Diff =</span> <span class="dv">0</span>)</span>
<span id="cb239-42"><a href="metorigin-analysis.html#cb239-42" tabindex="-1"></a>  </span>
<span id="cb239-43"><a href="metorigin-analysis.html#cb239-43" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">rbind</span>(temp_dat_diff, temp_dat_nodiff) <span class="sc">%&gt;%</span></span>
<span id="cb239-44"><a href="metorigin-analysis.html#cb239-44" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(HMDB, cpd_ID, Compounds, DA_index1, DA_index2, Diff) <span class="sc">%&gt;%</span></span>
<span id="cb239-45"><a href="metorigin-analysis.html#cb239-45" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">HMDBID =</span> HMDB,</span>
<span id="cb239-46"><a href="metorigin-analysis.html#cb239-46" tabindex="-1"></a>                  <span class="at">KEGGID =</span> cpd_ID,</span>
<span id="cb239-47"><a href="metorigin-analysis.html#cb239-47" tabindex="-1"></a>                  <span class="at">Name =</span> Compounds) <span class="sc">%&gt;%</span></span>
<span id="cb239-48"><a href="metorigin-analysis.html#cb239-48" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(HMDBID, KEGGID, Name, Diff)</span>
<span id="cb239-49"><a href="metorigin-analysis.html#cb239-49" tabindex="-1"></a></span>
<span id="cb239-50"><a href="metorigin-analysis.html#cb239-50" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb239-51"><a href="metorigin-analysis.html#cb239-51" tabindex="-1"></a>}</span>
<span id="cb239-52"><a href="metorigin-analysis.html#cb239-52" tabindex="-1"></a></span>
<span id="cb239-53"><a href="metorigin-analysis.html#cb239-53" tabindex="-1"></a>pre_result <span class="ot">&lt;-</span> <span class="fu">get_metabolites</span>(</span>
<span id="cb239-54"><a href="metorigin-analysis.html#cb239-54" tabindex="-1"></a>  <span class="at">x =</span> ExprSet,</span>
<span id="cb239-55"><a href="metorigin-analysis.html#cb239-55" tabindex="-1"></a>  <span class="at">dat =</span> m_results,</span>
<span id="cb239-56"><a href="metorigin-analysis.html#cb239-56" tabindex="-1"></a>  <span class="at">group_names =</span> <span class="st">&quot;Mild vs Severe&quot;</span>,</span>
<span id="cb239-57"><a href="metorigin-analysis.html#cb239-57" tabindex="-1"></a>  <span class="at">index_names =</span> <span class="fu">c</span>(<span class="st">&quot;Log2FoldChange&quot;</span>, <span class="st">&quot;AdjustedPvalue&quot;</span>),</span>
<span id="cb239-58"><a href="metorigin-analysis.html#cb239-58" tabindex="-1"></a>  <span class="at">index_cutoff =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>))</span>
<span id="cb239-59"><a href="metorigin-analysis.html#cb239-59" tabindex="-1"></a>  </span>
<span id="cb239-60"><a href="metorigin-analysis.html#cb239-60" tabindex="-1"></a><span class="fu">head</span>(pre_result)</span></code></pre></div>
<pre><code>##        HMDBID KEGGID                                    Name Diff
## 1 HMDB0005322   &lt;NA&gt; 1-palmitoyl-2-linoleoyl-GPE (16:0/18:2)    1
## 2 HMDB0008994   &lt;NA&gt; 1-stearoyl-2-linoleoyl-GPE (18:0/18:2)*    1
## 3 HMDB0002166 C05145                      3-aminoisobutyrate    1
## 4 HMDB0005320   &lt;NA&gt;    1-palmitoyl-2-oleoyl-GPE (16:0/18:1)    1
## 5 HMDB0008993   &lt;NA&gt;     1-stearoyl-2-oleoyl-GPE (18:0/18:1)    1
## 6 HMDB0094649   &lt;NA&gt;                       2-aminoheptanoate    1</code></pre>
</div>
<div id="save-result" class="section level2 hasAnchor" number="5.6">
<h2><span class="header-section-number">5.6</span> Save result<a href="metorigin-analysis.html#save-result" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>the csv file could be used as inputs in the MetOrigin website.</li>
</ul>
<div class="sourceCode" id="cb241"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb241-1"><a href="metorigin-analysis.html#cb241-1" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="st">&quot;./dataset/MetOrigin&quot;</span>)) {</span>
<span id="cb241-2"><a href="metorigin-analysis.html#cb241-2" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="st">&quot;./dataset/MetOrigin&quot;</span>, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb241-3"><a href="metorigin-analysis.html#cb241-3" tabindex="-1"></a>}</span>
<span id="cb241-4"><a href="metorigin-analysis.html#cb241-4" tabindex="-1"></a></span>
<span id="cb241-5"><a href="metorigin-analysis.html#cb241-5" tabindex="-1"></a><span class="fu">write.csv</span>(pre_result, <span class="st">&quot;./dataset/MetOrigin/MetOrigin_inputs.csv&quot;</span>, <span class="at">row.names =</span> F)</span></code></pre></div>
</div>
<div id="session-info-1" class="section level2 hasAnchor" number="5.7">
<h2><span class="header-section-number">5.7</span> Session info<a href="metorigin-analysis.html#session-info-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb242"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb242-1"><a href="metorigin-analysis.html#cb242-1" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 4.1.3 (2022-03-10)
##  os       macOS Monterey 12.2.1
##  system   x86_64, darwin17.0
##  ui       RStudio
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       Asia/Shanghai
##  date     2023-11-23
##  rstudio  2023.09.0+463 Desert Sunflower (desktop)
##  pandoc   3.1.1 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/ (via rmarkdown)
## 
## ─ Packages ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package              * version    date (UTC) lib source
##  abind                  1.4-5      2016-07-21 [2] CRAN (R 4.1.0)
##  ade4                   1.7-22     2023-02-06 [2] CRAN (R 4.1.2)
##  affy                   1.72.0     2021-10-26 [2] Bioconductor
##  affyio                 1.64.0     2021-10-26 [2] Bioconductor
##  annotate               1.72.0     2021-10-26 [2] Bioconductor
##  AnnotationDbi        * 1.60.2     2023-03-10 [2] Bioconductor
##  ape                    5.7-1      2023-03-13 [2] CRAN (R 4.1.2)
##  aplot                  0.1.10     2023-03-08 [2] CRAN (R 4.1.2)
##  attempt                0.3.1      2020-05-03 [2] CRAN (R 4.1.0)
##  backports              1.4.1      2021-12-13 [2] CRAN (R 4.1.0)
##  base64enc              0.1-3      2015-07-28 [2] CRAN (R 4.1.0)
##  Biobase              * 2.54.0     2021-10-26 [2] Bioconductor
##  BiocGenerics         * 0.40.0     2021-10-26 [2] Bioconductor
##  BiocManager            1.30.21    2023-06-10 [2] CRAN (R 4.1.3)
##  BiocParallel           1.28.3     2021-12-09 [2] Bioconductor
##  biomformat             1.22.0     2021-10-26 [2] Bioconductor
##  Biostrings             2.62.0     2021-10-26 [2] Bioconductor
##  bit                    4.0.5      2022-11-15 [2] CRAN (R 4.1.2)
##  bit64                  4.0.5      2020-08-30 [2] CRAN (R 4.1.0)
##  bitops                 1.0-7      2021-04-24 [2] CRAN (R 4.1.0)
##  blob                   1.2.4      2023-03-17 [2] CRAN (R 4.1.2)
##  bookdown               0.34       2023-05-09 [2] CRAN (R 4.1.2)
##  broom                  1.0.5      2023-06-09 [2] CRAN (R 4.1.3)
##  bslib                  0.5.0      2023-06-09 [2] CRAN (R 4.1.3)
##  cachem                 1.0.8      2023-05-01 [2] CRAN (R 4.1.2)
##  Cairo                  1.6-0      2022-07-05 [2] CRAN (R 4.1.2)
##  callr                  3.7.3      2022-11-02 [2] CRAN (R 4.1.2)
##  car                    3.1-2      2023-03-30 [2] CRAN (R 4.1.2)
##  carData                3.0-5      2022-01-06 [2] CRAN (R 4.1.2)
##  caret                  6.0-94     2023-03-21 [2] CRAN (R 4.1.2)
##  caTools                1.18.2     2021-03-28 [2] CRAN (R 4.1.0)
##  cellranger             1.1.0      2016-07-27 [2] CRAN (R 4.1.0)
##  checkmate              2.2.0      2023-04-27 [2] CRAN (R 4.1.2)
##  circlize               0.4.15     2022-05-10 [2] CRAN (R 4.1.2)
##  class                  7.3-22     2023-05-03 [2] CRAN (R 4.1.2)
##  cli                    3.6.1      2023-03-23 [2] CRAN (R 4.1.2)
##  clue                   0.3-64     2023-01-31 [2] CRAN (R 4.1.2)
##  cluster              * 2.1.4      2022-08-22 [2] CRAN (R 4.1.2)
##  clusterProfiler      * 4.2.2      2022-01-13 [2] Bioconductor
##  codetools              0.2-19     2023-02-01 [2] CRAN (R 4.1.2)
##  coin                   1.4-2      2021-10-08 [2] CRAN (R 4.1.0)
##  colorspace             2.1-0      2023-01-23 [2] CRAN (R 4.1.2)
##  ComplexHeatmap         2.10.0     2021-10-26 [2] Bioconductor
##  config                 0.3.1      2020-12-17 [2] CRAN (R 4.1.0)
##  corpcor                1.6.10     2021-09-16 [2] CRAN (R 4.1.0)
##  cowplot                1.1.1      2020-12-30 [2] CRAN (R 4.1.0)
##  crayon                 1.5.2      2022-09-29 [2] CRAN (R 4.1.2)
##  crmn                   0.0.21     2020-02-10 [2] CRAN (R 4.1.0)
##  curl                   5.0.1      2023-06-07 [2] CRAN (R 4.1.3)
##  data.table             1.14.8     2023-02-17 [2] CRAN (R 4.1.2)
##  DBI                    1.1.3      2022-06-18 [2] CRAN (R 4.1.2)
##  DelayedArray           0.20.0     2021-10-26 [2] Bioconductor
##  dendextend           * 1.17.1     2023-03-25 [2] CRAN (R 4.1.2)
##  DESeq2                 1.34.0     2021-10-26 [2] Bioconductor
##  devtools               2.4.5      2022-10-11 [2] CRAN (R 4.1.2)
##  digest                 0.6.33     2023-07-07 [1] CRAN (R 4.1.3)
##  DO.db                  2.9        2022-04-11 [2] Bioconductor
##  doParallel             1.0.17     2022-02-07 [2] CRAN (R 4.1.2)
##  doRNG                  1.8.6      2023-01-16 [2] CRAN (R 4.1.2)
##  DOSE                   3.20.1     2021-11-18 [2] Bioconductor
##  doSNOW                 1.0.20     2022-02-04 [2] CRAN (R 4.1.2)
##  downloader             0.4        2015-07-09 [2] CRAN (R 4.1.0)
##  dplyr                * 1.1.2      2023-04-20 [2] CRAN (R 4.1.2)
##  DT                     0.28       2023-05-18 [2] CRAN (R 4.1.3)
##  dynamicTreeCut         1.63-1     2016-03-11 [2] CRAN (R 4.1.0)
##  e1071                  1.7-13     2023-02-01 [2] CRAN (R 4.1.2)
##  edgeR                  3.36.0     2021-10-26 [2] Bioconductor
##  ellipse                0.4.5      2023-04-05 [2] CRAN (R 4.1.2)
##  ellipsis               0.3.2      2021-04-29 [2] CRAN (R 4.1.0)
##  enrichplot             1.14.2     2022-02-24 [2] Bioconductor
##  evaluate               0.21       2023-05-05 [2] CRAN (R 4.1.2)
##  factoextra           * 1.0.7      2020-04-01 [2] CRAN (R 4.1.0)
##  fansi                  1.0.4      2023-01-22 [2] CRAN (R 4.1.2)
##  farver                 2.1.1      2022-07-06 [2] CRAN (R 4.1.2)
##  fastcluster            1.2.3      2021-05-24 [2] CRAN (R 4.1.0)
##  fastmap                1.1.1      2023-02-24 [2] CRAN (R 4.1.2)
##  fastmatch              1.1-3      2021-07-23 [2] CRAN (R 4.1.0)
##  fdrtool                1.2.17     2021-11-13 [2] CRAN (R 4.1.0)
##  fgsea                  1.20.0     2021-10-26 [2] Bioconductor
##  filematrix             1.3        2018-02-27 [2] CRAN (R 4.1.0)
##  foreach                1.5.2      2022-02-02 [2] CRAN (R 4.1.2)
##  foreign                0.8-84     2022-12-06 [2] CRAN (R 4.1.2)
##  forestplot             3.1.1      2022-12-06 [2] CRAN (R 4.1.2)
##  Formula                1.2-5      2023-02-24 [2] CRAN (R 4.1.2)
##  fs                     1.6.2      2023-04-25 [2] CRAN (R 4.1.2)
##  furrr                  0.3.1      2022-08-15 [2] CRAN (R 4.1.2)
##  future                 1.33.0     2023-07-01 [2] CRAN (R 4.1.3)
##  future.apply           1.11.0     2023-05-21 [2] CRAN (R 4.1.3)
##  genefilter             1.76.0     2021-10-26 [2] Bioconductor
##  geneplotter            1.72.0     2021-10-26 [2] Bioconductor
##  generics               0.1.3      2022-07-05 [2] CRAN (R 4.1.2)
##  GenomeInfoDb         * 1.30.1     2022-01-30 [2] Bioconductor
##  GenomeInfoDbData       1.2.7      2022-03-09 [2] Bioconductor
##  GenomicRanges        * 1.46.1     2021-11-18 [2] Bioconductor
##  GetoptLong             1.0.5      2020-12-15 [2] CRAN (R 4.1.0)
##  ggforce                0.4.1      2022-10-04 [2] CRAN (R 4.1.2)
##  ggfun                  0.1.1      2023-06-24 [2] CRAN (R 4.1.3)
##  ggplot2              * 3.4.2      2023-04-03 [2] CRAN (R 4.1.2)
##  ggplotify              0.1.1      2023-06-27 [2] CRAN (R 4.1.3)
##  ggpubr                 0.6.0      2023-02-10 [2] CRAN (R 4.1.2)
##  ggraph               * 2.1.0.9000 2023-07-11 [1] Github (thomasp85/ggraph@febab71)
##  ggrepel                0.9.3      2023-02-03 [2] CRAN (R 4.1.2)
##  ggsignif               0.6.4      2022-10-13 [2] CRAN (R 4.1.2)
##  ggtree                 3.2.1      2021-11-16 [2] Bioconductor
##  glasso                 1.11       2019-10-01 [2] CRAN (R 4.1.0)
##  glmnet               * 4.1-7      2023-03-23 [2] CRAN (R 4.1.2)
##  GlobalOptions          0.1.2      2020-06-10 [2] CRAN (R 4.1.0)
##  globals                0.16.2     2022-11-21 [2] CRAN (R 4.1.2)
##  globaltest             5.48.0     2021-10-26 [2] Bioconductor
##  glue                 * 1.6.2      2022-02-24 [2] CRAN (R 4.1.2)
##  Gmisc                * 3.0.2      2023-03-13 [2] CRAN (R 4.1.2)
##  gmm                    1.8        2023-06-06 [2] CRAN (R 4.1.3)
##  gmp                    0.7-1      2023-02-07 [2] CRAN (R 4.1.2)
##  GO.db                  3.14.0     2022-04-11 [2] Bioconductor
##  golem                  0.4.1      2023-06-05 [2] CRAN (R 4.1.3)
##  GOSemSim               2.20.0     2021-10-26 [2] Bioconductor
##  gower                  1.0.1      2022-12-22 [2] CRAN (R 4.1.2)
##  gplots                 3.1.3      2022-04-25 [2] CRAN (R 4.1.2)
##  graphlayouts           1.0.0      2023-05-01 [2] CRAN (R 4.1.2)
##  gridExtra              2.3        2017-09-09 [2] CRAN (R 4.1.0)
##  gridGraphics           0.5-1      2020-12-13 [2] CRAN (R 4.1.0)
##  gtable                 0.3.3      2023-03-21 [2] CRAN (R 4.1.2)
##  gtools                 3.9.4      2022-11-27 [2] CRAN (R 4.1.2)
##  hardhat                1.3.0      2023-03-30 [2] CRAN (R 4.1.2)
##  highr                  0.10       2022-12-22 [2] CRAN (R 4.1.2)
##  Hmisc                  5.1-0      2023-05-08 [2] CRAN (R 4.1.2)
##  hms                    1.1.3      2023-03-21 [2] CRAN (R 4.1.2)
##  htmlTable            * 2.4.1      2022-07-07 [2] CRAN (R 4.1.2)
##  htmltools              0.5.5      2023-03-23 [2] CRAN (R 4.1.2)
##  htmlwidgets            1.6.2      2023-03-17 [2] CRAN (R 4.1.2)
##  httpuv                 1.6.11     2023-05-11 [2] CRAN (R 4.1.3)
##  httr                 * 1.4.6      2023-05-08 [2] CRAN (R 4.1.2)
##  huge                   1.3.5      2021-06-30 [2] CRAN (R 4.1.0)
##  igraph                 1.5.0      2023-06-16 [1] CRAN (R 4.1.3)
##  impute                 1.68.0     2021-10-26 [2] Bioconductor
##  imputeLCMD             2.1        2022-06-10 [2] CRAN (R 4.1.2)
##  ipred                  0.9-14     2023-03-09 [2] CRAN (R 4.1.2)
##  IRanges              * 2.28.0     2021-10-26 [2] Bioconductor
##  irlba                  2.3.5.1    2022-10-03 [2] CRAN (R 4.1.2)
##  iterators              1.0.14     2022-02-05 [2] CRAN (R 4.1.2)
##  itertools              0.1-3      2014-03-12 [2] CRAN (R 4.1.0)
##  jpeg                   0.1-10     2022-11-29 [2] CRAN (R 4.1.2)
##  jquerylib              0.1.4      2021-04-26 [2] CRAN (R 4.1.0)
##  jsonlite               1.8.7      2023-06-29 [2] CRAN (R 4.1.3)
##  KEGGREST               1.34.0     2021-10-26 [2] Bioconductor
##  KernSmooth             2.23-22    2023-07-10 [2] CRAN (R 4.1.3)
##  knitr                  1.43       2023-05-25 [2] CRAN (R 4.1.3)
##  labeling               0.4.2      2020-10-20 [2] CRAN (R 4.1.0)
##  later                  1.3.1      2023-05-02 [2] CRAN (R 4.1.2)
##  lattice                0.21-8     2023-04-05 [2] CRAN (R 4.1.2)
##  lava                   1.7.2.1    2023-02-27 [2] CRAN (R 4.1.2)
##  lavaan                 0.6-15     2023-03-14 [2] CRAN (R 4.1.2)
##  lazyeval               0.2.2      2019-03-15 [2] CRAN (R 4.1.0)
##  libcoin                1.0-9      2021-09-27 [2] CRAN (R 4.1.0)
##  lifecycle              1.0.3      2022-10-07 [2] CRAN (R 4.1.2)
##  limma                  3.50.3     2022-04-07 [2] Bioconductor
##  listenv                0.9.0      2022-12-16 [2] CRAN (R 4.1.2)
##  locfit                 1.5-9.8    2023-06-11 [2] CRAN (R 4.1.3)
##  lubridate              1.9.2      2023-02-10 [2] CRAN (R 4.1.2)
##  magrittr             * 2.0.3      2022-03-30 [2] CRAN (R 4.1.2)
##  MALDIquant             1.22.1     2023-03-20 [2] CRAN (R 4.1.2)
##  MASS                   7.3-60     2023-05-04 [2] CRAN (R 4.1.2)
##  massdatabase         * 1.0.7      2023-05-30 [2] gitlab (jaspershen/massdatabase@df83e93)
##  massdataset          * 1.0.24     2023-05-30 [2] gitlab (jaspershen/massdataset@b397116)
##  masstools            * 1.0.10     2023-05-30 [2] gitlab (jaspershen/masstools@b3c73bc)
##  Matrix               * 1.6-0      2023-07-08 [2] CRAN (R 4.1.3)
##  MatrixGenerics       * 1.6.0      2021-10-26 [2] Bioconductor
##  matrixStats          * 1.0.0      2023-06-02 [2] CRAN (R 4.1.3)
##  memoise                2.0.1      2021-11-26 [2] CRAN (R 4.1.0)
##  MetaboAnalystR       * 3.2.0      2022-06-28 [2] Github (xia-lab/MetaboAnalystR@892a31b)
##  metagenomeSeq          1.36.0     2021-10-26 [2] Bioconductor
##  metid                * 1.2.26     2023-05-30 [2] gitlab (jaspershen/metid@6bde121)
##  metpath              * 1.0.5      2023-05-30 [2] gitlab (jaspershen/metpath@adcad4f)
##  mgcv                   1.8-42     2023-03-02 [2] CRAN (R 4.1.2)
##  MicrobiomeProfiler   * 1.0.0      2021-10-26 [2] Bioconductor
##  mime                   0.12       2021-09-28 [2] CRAN (R 4.1.0)
##  miniUI                 0.1.1.1    2018-05-18 [2] CRAN (R 4.1.0)
##  missForest             1.5        2022-04-14 [2] CRAN (R 4.1.2)
##  mixedCCA               1.6.2      2022-09-09 [2] CRAN (R 4.1.2)
##  mixOmics               6.18.1     2021-11-18 [2] Bioconductor (R 4.1.2)
##  mnormt                 2.1.1      2022-09-26 [2] CRAN (R 4.1.2)
##  ModelMetrics           1.2.2.2    2020-03-17 [2] CRAN (R 4.1.0)
##  modeltools             0.2-23     2020-03-05 [2] CRAN (R 4.1.0)
##  MsCoreUtils            1.6.2      2022-02-24 [2] Bioconductor
##  MSnbase              * 2.20.4     2022-01-16 [2] Bioconductor
##  multcomp               1.4-25     2023-06-20 [2] CRAN (R 4.1.3)
##  multtest               2.50.0     2021-10-26 [2] Bioconductor
##  munsell                0.5.0      2018-06-12 [2] CRAN (R 4.1.0)
##  mvtnorm                1.2-2      2023-06-08 [2] CRAN (R 4.1.3)
##  mzID                   1.32.0     2021-10-26 [2] Bioconductor
##  mzR                  * 2.28.0     2021-10-27 [2] Bioconductor
##  ncdf4                  1.21       2023-01-07 [2] CRAN (R 4.1.2)
##  NetCoMi              * 1.0.3      2022-07-14 [2] Github (stefpeschel/NetCoMi@d4d80d3)
##  nlme                   3.1-162    2023-01-31 [2] CRAN (R 4.1.2)
##  nnet                   7.3-19     2023-05-03 [2] CRAN (R 4.1.2)
##  norm                   1.0-11.1   2023-06-18 [2] CRAN (R 4.1.3)
##  openxlsx               4.2.5.2    2023-02-06 [2] CRAN (R 4.1.2)
##  org.Mm.eg.db         * 3.14.0     2022-11-23 [2] Bioconductor
##  parallelly             1.36.0     2023-05-26 [2] CRAN (R 4.1.3)
##  patchwork              1.1.2      2022-08-19 [2] CRAN (R 4.1.2)
##  pbapply                1.7-2      2023-06-27 [2] CRAN (R 4.1.3)
##  pbivnorm               0.6.0      2015-01-23 [2] CRAN (R 4.1.0)
##  pcaMethods             1.86.0     2021-10-26 [2] Bioconductor
##  pcaPP                  2.0-3      2022-10-24 [2] CRAN (R 4.1.2)
##  permute                0.9-7      2022-01-27 [2] CRAN (R 4.1.2)
##  pheatmap               1.0.12     2019-01-04 [2] CRAN (R 4.1.0)
##  phyloseq               1.38.0     2021-10-26 [2] Bioconductor
##  pillar                 1.9.0      2023-03-22 [2] CRAN (R 4.1.2)
##  pkgbuild               1.4.2      2023-06-26 [2] CRAN (R 4.1.3)
##  pkgconfig              2.0.3      2019-09-22 [2] CRAN (R 4.1.0)
##  pkgload                1.3.2.1    2023-07-08 [2] CRAN (R 4.1.3)
##  plotly               * 4.10.2     2023-06-03 [2] CRAN (R 4.1.3)
##  plyr                   1.8.8      2022-11-11 [2] CRAN (R 4.1.2)
##  png                    0.1-8      2022-11-29 [2] CRAN (R 4.1.2)
##  polyclip               1.10-4     2022-10-20 [2] CRAN (R 4.1.2)
##  POMA                 * 1.7.2      2022-07-26 [2] Github (pcastellanoescuder/POMA@bc8a972)
##  preprocessCore         1.56.0     2021-10-26 [2] Bioconductor
##  prettyunits            1.1.1      2020-01-24 [2] CRAN (R 4.1.0)
##  pROC                   1.18.4     2023-07-06 [2] CRAN (R 4.1.3)
##  processx               3.8.2      2023-06-30 [2] CRAN (R 4.1.3)
##  prodlim                2023.03.31 2023-04-02 [2] CRAN (R 4.1.2)
##  profvis                0.3.8      2023-05-02 [2] CRAN (R 4.1.2)
##  progress               1.2.2      2019-05-16 [2] CRAN (R 4.1.0)
##  promises               1.2.0.1    2021-02-11 [2] CRAN (R 4.1.0)
##  ProtGenerics         * 1.26.0     2021-10-26 [2] Bioconductor
##  proxy                  0.4-27     2022-06-09 [2] CRAN (R 4.1.2)
##  ps                     1.7.5      2023-04-18 [2] CRAN (R 4.1.2)
##  psych                  2.3.6      2023-06-21 [2] CRAN (R 4.1.3)
##  pulsar                 0.3.10     2023-01-26 [2] CRAN (R 4.1.2)
##  purrr                  1.0.1      2023-01-10 [2] CRAN (R 4.1.2)
##  qgraph                 1.9.5      2023-05-16 [2] CRAN (R 4.1.3)
##  qs                     0.25.5     2023-02-22 [2] CRAN (R 4.1.2)
##  quadprog               1.5-8      2019-11-20 [2] CRAN (R 4.1.0)
##  qvalue                 2.26.0     2021-10-26 [2] Bioconductor
##  R6                     2.5.1      2021-08-19 [2] CRAN (R 4.1.0)
##  ragg                   1.2.5      2023-01-12 [2] CRAN (R 4.1.2)
##  randomForest           4.7-1.1    2022-05-23 [2] CRAN (R 4.1.2)
##  RankProd               3.20.0     2021-10-26 [2] Bioconductor
##  RApiSerialize          0.1.2      2022-08-25 [2] CRAN (R 4.1.2)
##  rARPACK                0.11-0     2016-03-10 [2] CRAN (R 4.1.0)
##  rbibutils              2.2.13     2023-01-13 [2] CRAN (R 4.1.2)
##  RColorBrewer           1.1-3      2022-04-03 [2] CRAN (R 4.1.2)
##  Rcpp                 * 1.0.11     2023-07-06 [1] CRAN (R 4.1.3)
##  RcppParallel           5.1.7      2023-02-27 [2] CRAN (R 4.1.2)
##  RCurl                  1.98-1.12  2023-03-27 [2] CRAN (R 4.1.2)
##  Rdisop                 1.54.0     2021-10-26 [2] Bioconductor
##  Rdpack                 2.4        2022-07-20 [2] CRAN (R 4.1.2)
##  readr                  2.1.4      2023-02-10 [2] CRAN (R 4.1.2)
##  readxl               * 1.4.3      2023-07-06 [2] CRAN (R 4.1.3)
##  recipes                1.0.6      2023-04-25 [2] CRAN (R 4.1.2)
##  remotes                2.4.2      2021-11-30 [2] CRAN (R 4.1.0)
##  reshape2               1.4.4      2020-04-09 [2] CRAN (R 4.1.0)
##  rhdf5                  2.38.1     2022-03-10 [2] Bioconductor
##  rhdf5filters           1.6.0      2021-10-26 [2] Bioconductor
##  Rhdf5lib               1.16.0     2021-10-26 [2] Bioconductor
##  rjson                  0.2.21     2022-01-09 [2] CRAN (R 4.1.2)
##  rlang                  1.1.1      2023-04-28 [1] CRAN (R 4.1.2)
##  rmarkdown              2.23       2023-07-01 [2] CRAN (R 4.1.3)
##  Rmpfr                  0.9-2      2023-04-22 [2] CRAN (R 4.1.2)
##  rngtools               1.5.2      2021-09-20 [2] CRAN (R 4.1.0)
##  rootSolve              1.8.2.3    2021-09-29 [2] CRAN (R 4.1.0)
##  ropls                * 1.26.4     2022-01-11 [2] Bioconductor
##  rpart                  4.1.19     2022-10-21 [2] CRAN (R 4.1.2)
##  Rserve               * 1.8-11     2022-11-28 [2] CRAN (R 4.1.2)
##  RSpectra               0.16-1     2022-04-24 [2] CRAN (R 4.1.2)
##  RSQLite                2.3.1      2023-04-03 [2] CRAN (R 4.1.2)
##  rstatix                0.7.2      2023-02-01 [2] CRAN (R 4.1.2)
##  rstudioapi             0.15.0     2023-07-07 [2] CRAN (R 4.1.3)
##  rvest                  1.0.3      2022-08-19 [2] CRAN (R 4.1.2)
##  S4Vectors            * 0.32.4     2022-03-29 [2] Bioconductor
##  sandwich               3.0-2      2022-06-15 [2] CRAN (R 4.1.2)
##  sass                   0.4.6      2023-05-03 [2] CRAN (R 4.1.2)
##  scales                 1.2.1      2022-08-20 [2] CRAN (R 4.1.2)
##  scatterpie             0.2.1      2023-06-07 [2] CRAN (R 4.1.3)
##  scrime                 1.3.5      2018-12-01 [2] CRAN (R 4.1.0)
##  sessioninfo            1.2.2      2021-12-06 [2] CRAN (R 4.1.0)
##  shadowtext             0.1.2      2022-04-22 [2] CRAN (R 4.1.2)
##  shape                  1.4.6      2021-05-19 [2] CRAN (R 4.1.0)
##  shiny                  1.7.4.1    2023-07-06 [2] CRAN (R 4.1.3)
##  shinycustomloader      0.9.0      2018-03-27 [2] CRAN (R 4.1.0)
##  shinyWidgets           0.7.6      2023-01-08 [2] CRAN (R 4.1.2)
##  siggenes               1.68.0     2021-10-26 [2] Bioconductor
##  snow                   0.4-4      2021-10-27 [2] CRAN (R 4.1.0)
##  SpiecEasi            * 1.1.2      2022-07-14 [2] Github (zdk123/SpiecEasi@c463727)
##  SPRING               * 1.0.4      2022-08-03 [2] Github (GraceYoon/SPRING@3d641a4)
##  stringdist             0.9.10     2022-11-07 [2] CRAN (R 4.1.2)
##  stringfish             0.15.8     2023-05-30 [2] CRAN (R 4.1.3)
##  stringi                1.7.12     2023-01-11 [2] CRAN (R 4.1.2)
##  stringr                1.5.0      2022-12-02 [2] CRAN (R 4.1.2)
##  SummarizedExperiment * 1.24.0     2021-10-26 [2] Bioconductor
##  survival               3.5-5      2023-03-12 [2] CRAN (R 4.1.2)
##  systemfonts            1.0.4      2022-02-11 [2] CRAN (R 4.1.2)
##  textshaping            0.3.6      2021-10-13 [2] CRAN (R 4.1.0)
##  TH.data                1.1-2      2023-04-17 [2] CRAN (R 4.1.2)
##  tibble               * 3.2.1      2023-03-20 [2] CRAN (R 4.1.2)
##  tidygraph              1.2.3      2023-02-01 [2] CRAN (R 4.1.2)
##  tidyr                  1.3.0      2023-01-24 [2] CRAN (R 4.1.2)
##  tidyselect             1.2.0      2022-10-10 [2] CRAN (R 4.1.2)
##  tidytree               0.4.2      2022-12-18 [2] CRAN (R 4.1.2)
##  timechange             0.2.0      2023-01-11 [2] CRAN (R 4.1.2)
##  timeDate               4022.108   2023-01-07 [2] CRAN (R 4.1.2)
##  tmvtnorm               1.5        2022-03-22 [2] CRAN (R 4.1.2)
##  treeio                 1.18.1     2021-11-14 [2] Bioconductor
##  tweenr                 2.0.2      2022-09-06 [2] CRAN (R 4.1.2)
##  tzdb                   0.4.0      2023-05-12 [2] CRAN (R 4.1.3)
##  urlchecker             1.0.1      2021-11-30 [2] CRAN (R 4.1.0)
##  usethis                2.2.2      2023-07-06 [2] CRAN (R 4.1.3)
##  utf8                   1.2.3      2023-01-31 [2] CRAN (R 4.1.2)
##  vctrs                  0.6.3      2023-06-14 [1] CRAN (R 4.1.3)
##  vegan                  2.6-4      2022-10-11 [2] CRAN (R 4.1.2)
##  VGAM                   1.1-8      2023-03-09 [2] CRAN (R 4.1.2)
##  viridis                0.6.3      2023-05-03 [2] CRAN (R 4.1.2)
##  viridisLite            0.4.2      2023-05-02 [2] CRAN (R 4.1.2)
##  vsn                    3.62.0     2021-10-26 [2] Bioconductor
##  WGCNA                  1.72-1     2023-01-18 [2] CRAN (R 4.1.2)
##  withr                  2.5.0      2022-03-03 [2] CRAN (R 4.1.2)
##  Wrench                 1.12.0     2021-10-26 [2] Bioconductor
##  xfun                   0.40       2023-08-09 [1] CRAN (R 4.1.3)
##  XMAS2                * 2.2.0      2023-10-27 [1] local
##  XML                    3.99-0.14  2023-03-19 [2] CRAN (R 4.1.2)
##  xml2                   1.3.5      2023-07-06 [2] CRAN (R 4.1.3)
##  xtable                 1.8-4      2019-04-21 [2] CRAN (R 4.1.0)
##  XVector                0.34.0     2021-10-26 [2] Bioconductor
##  yaml                   2.3.7      2023-01-23 [2] CRAN (R 4.1.2)
##  yulab.utils            0.0.6      2022-12-20 [2] CRAN (R 4.1.2)
##  zip                    2.3.0      2023-04-17 [2] CRAN (R 4.1.2)
##  zlibbioc               1.40.0     2021-10-26 [2] Bioconductor
##  zoo                    1.8-12     2023-04-13 [2] CRAN (R 4.1.2)
## 
##  [1] /Users/zouhua/Library/R/x86_64/4.1/library
##  [2] /Library/Frameworks/R.framework/Versions/4.1/Resources/library
## 
## ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-zeybel2022multiomics" class="csl-entry">
Zeybel, Mujdat, Muhammad Arif, Xiangyu Li, Ozlem Altay, Hong Yang, Mengnan Shi, Murat Akyildiz, et al. 2022. <span>“Multiomics Analysis Reveals the Impact of Microbiota on Host Metabolism in Hepatic Steatosis.”</span> <em>Advanced Science</em> 9 (11): 2104373.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="functional-analysis-1.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="example.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/Metabolomics_Aanlysis/blob/main/03-MetOrigin-Analysis.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true,
"toc_depth": 3
}
});
});
</script>

</body>

</html>
