<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 Network Analysis | Data Analysis in Metabolomics</title>
  <meta name="description" content="A template for Data Analysis in Metabolomics" />
  <meta name="generator" content="bookdown 0.27 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 Network Analysis | Data Analysis in Metabolomics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A template for Data Analysis in Metabolomics" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 Network Analysis | Data Analysis in Metabolomics" />
  
  <meta name="twitter:description" content="A template for Data Analysis in Metabolomics" />
  

<meta name="author" content="Hua Zou and Bangzhuo Tong" />


<meta name="date" content="2022-08-16" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="multivariate-analysis.html"/>
<link rel="next" href="enrichment-analysis.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="assets/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Data Analysis in Metabolomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#what-is-metabolomics"><i class="fa fa-check"></i><b>1.1</b> What is metabolomics?</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#what-are-the-procedures-in-this-template"><i class="fa fa-check"></i><b>1.2</b> What are the procedures in this template?</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#statistical-analysis"><i class="fa fa-check"></i><b>1.2.1</b> Statistical Analysis</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#functional-analysis"><i class="fa fa-check"></i><b>1.2.2</b> Functional Analysis</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#software"><i class="fa fa-check"></i><b>1.3</b> Software</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#reference"><i class="fa fa-check"></i><b>1.4</b> Reference</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-processing.html"><a href="data-processing.html"><i class="fa fa-check"></i><b>2</b> Data Processing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-processing.html"><a href="data-processing.html#loading-packages"><i class="fa fa-check"></i><b>2.1</b> Loading packages</a></li>
<li class="chapter" data-level="2.2" data-path="data-processing.html"><a href="data-processing.html#importing-data"><i class="fa fa-check"></i><b>2.2</b> Importing data</a></li>
<li class="chapter" data-level="2.3" data-path="data-processing.html"><a href="data-processing.html#data-checking"><i class="fa fa-check"></i><b>2.3</b> Data Checking</a></li>
<li class="chapter" data-level="2.4" data-path="data-processing.html"><a href="data-processing.html#data-filtering"><i class="fa fa-check"></i><b>2.4</b> Data Filtering</a></li>
<li class="chapter" data-level="2.5" data-path="data-processing.html"><a href="data-processing.html#missing-value-imputation"><i class="fa fa-check"></i><b>2.5</b> Missing Value Imputation</a></li>
<li class="chapter" data-level="2.6" data-path="data-processing.html"><a href="data-processing.html#data-normalization"><i class="fa fa-check"></i><b>2.6</b> Data Normalization</a>
<ul>
<li class="chapter" data-level="2.6.1" data-path="data-processing.html"><a href="data-processing.html#normalization-by-normalizedata-function"><i class="fa fa-check"></i><b>2.6.1</b> Normalization by <strong>NormalizeData</strong> function</a></li>
<li class="chapter" data-level="2.6.2" data-path="data-processing.html"><a href="data-processing.html#normalization-by-poma-r-package"><i class="fa fa-check"></i><b>2.6.2</b> Normalization by <strong>POMA</strong> R package</a></li>
<li class="chapter" data-level="2.6.3" data-path="data-processing.html"><a href="data-processing.html#comparison-of-unnormalized-and-normalized-dataset"><i class="fa fa-check"></i><b>2.6.3</b> Comparison of unnormalized and normalized dataset</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="data-processing.html"><a href="data-processing.html#removing-outliers"><i class="fa fa-check"></i><b>2.7</b> Removing outliers</a></li>
<li class="chapter" data-level="2.8" data-path="data-processing.html"><a href="data-processing.html#saving-datasets-into-rds-files"><i class="fa fa-check"></i><b>2.8</b> Saving datasets into RDS files</a></li>
<li class="chapter" data-level="2.9" data-path="data-processing.html"><a href="data-processing.html#systematic-information"><i class="fa fa-check"></i><b>2.9</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cluster-analysis.html"><a href="cluster-analysis.html"><i class="fa fa-check"></i><b>3</b> Cluster Analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="cluster-analysis.html"><a href="cluster-analysis.html#loading-packages-1"><i class="fa fa-check"></i><b>3.1</b> Loading packages</a></li>
<li class="chapter" data-level="3.2" data-path="cluster-analysis.html"><a href="cluster-analysis.html#importing-data-1"><i class="fa fa-check"></i><b>3.2</b> Importing data</a></li>
<li class="chapter" data-level="3.3" data-path="cluster-analysis.html"><a href="cluster-analysis.html#hierarchical-clustering"><i class="fa fa-check"></i><b>3.3</b> Hierarchical Clustering</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="cluster-analysis.html"><a href="cluster-analysis.html#agglomerative-hierarchical-clustering"><i class="fa fa-check"></i><b>3.3.1</b> Agglomerative Hierarchical Clustering</a></li>
<li class="chapter" data-level="3.3.2" data-path="cluster-analysis.html"><a href="cluster-analysis.html#divisive-hierarchical-clustering"><i class="fa fa-check"></i><b>3.3.2</b> Divisive Hierarchical Clustering</a></li>
<li class="chapter" data-level="3.3.3" data-path="cluster-analysis.html"><a href="cluster-analysis.html#comparison"><i class="fa fa-check"></i><b>3.3.3</b> Comparison</a></li>
<li class="chapter" data-level="3.3.4" data-path="cluster-analysis.html"><a href="cluster-analysis.html#determining-optimal-clusters"><i class="fa fa-check"></i><b>3.3.4</b> Determining Optimal Clusters</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="cluster-analysis.html"><a href="cluster-analysis.html#partitional-clustering"><i class="fa fa-check"></i><b>3.4</b> Partitional Clustering</a></li>
<li class="chapter" data-level="3.5" data-path="cluster-analysis.html"><a href="cluster-analysis.html#systematic-information-1"><i class="fa fa-check"></i><b>3.5</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="chemometrics-analysis.html"><a href="chemometrics-analysis.html"><i class="fa fa-check"></i><b>4</b> Chemometrics Analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="chemometrics-analysis.html"><a href="chemometrics-analysis.html#loading-packages-2"><i class="fa fa-check"></i><b>4.1</b> Loading packages</a></li>
<li class="chapter" data-level="4.2" data-path="chemometrics-analysis.html"><a href="chemometrics-analysis.html#importing-data-2"><i class="fa fa-check"></i><b>4.2</b> Importing data</a></li>
<li class="chapter" data-level="4.3" data-path="chemometrics-analysis.html"><a href="chemometrics-analysis.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>4.3</b> Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="4.4" data-path="chemometrics-analysis.html"><a href="chemometrics-analysis.html#partial-least-squares-discriminant-analysis-pls-da"><i class="fa fa-check"></i><b>4.4</b> Partial Least Squares-Discriminant Analysis (PLS-DA)</a></li>
<li class="chapter" data-level="4.5" data-path="chemometrics-analysis.html"><a href="chemometrics-analysis.html#sparse-partial-least-squares-discriminant-analysis-spls-da"><i class="fa fa-check"></i><b>4.5</b> Sparse Partial Least Squares-Discriminant Analysis (sPLS-DA)</a></li>
<li class="chapter" data-level="4.6" data-path="chemometrics-analysis.html"><a href="chemometrics-analysis.html#orthogonal-partial-least-squares-discriminant-analysis-orthopls-da"><i class="fa fa-check"></i><b>4.6</b> Orthogonal Partial Least Squares-Discriminant Analysis (orthoPLS-DA)</a></li>
<li class="chapter" data-level="4.7" data-path="chemometrics-analysis.html"><a href="chemometrics-analysis.html#systematic-information-2"><i class="fa fa-check"></i><b>4.7</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="univariate-analysis.html"><a href="univariate-analysis.html"><i class="fa fa-check"></i><b>5</b> Univariate Analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="univariate-analysis.html"><a href="univariate-analysis.html#loading-packages-3"><i class="fa fa-check"></i><b>5.1</b> Loading packages</a></li>
<li class="chapter" data-level="5.2" data-path="univariate-analysis.html"><a href="univariate-analysis.html#importing-data-3"><i class="fa fa-check"></i><b>5.2</b> Importing data</a></li>
<li class="chapter" data-level="5.3" data-path="univariate-analysis.html"><a href="univariate-analysis.html#fold-change-analysis"><i class="fa fa-check"></i><b>5.3</b> Fold Change Analysis</a></li>
<li class="chapter" data-level="5.4" data-path="univariate-analysis.html"><a href="univariate-analysis.html#t-test"><i class="fa fa-check"></i><b>5.4</b> T Test</a></li>
<li class="chapter" data-level="5.5" data-path="univariate-analysis.html"><a href="univariate-analysis.html#wilcoxon-test"><i class="fa fa-check"></i><b>5.5</b> Wilcoxon Test</a></li>
<li class="chapter" data-level="5.6" data-path="univariate-analysis.html"><a href="univariate-analysis.html#limma-test"><i class="fa fa-check"></i><b>5.6</b> Limma Test</a></li>
<li class="chapter" data-level="5.7" data-path="univariate-analysis.html"><a href="univariate-analysis.html#volcano-plot"><i class="fa fa-check"></i><b>5.7</b> Volcano plot</a></li>
<li class="chapter" data-level="5.8" data-path="univariate-analysis.html"><a href="univariate-analysis.html#correlation-heatmaps"><i class="fa fa-check"></i><b>5.8</b> Correlation Heatmaps</a></li>
<li class="chapter" data-level="5.9" data-path="univariate-analysis.html"><a href="univariate-analysis.html#glasso-this-function-will-compute-a-gaussian-graphical-model-using-the-glmnet-package"><i class="fa fa-check"></i><b>5.9</b> glasso: this function will compute a Gaussian Graphical Model using the glmnet package</a></li>
<li class="chapter" data-level="5.10" data-path="univariate-analysis.html"><a href="univariate-analysis.html#systematic-information-3"><i class="fa fa-check"></i><b>5.10</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html"><i class="fa fa-check"></i><b>6</b> Multivariate analysis</a>
<ul>
<li class="chapter" data-level="6.1" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#loading-packages-4"><i class="fa fa-check"></i><b>6.1</b> Loading packages</a></li>
<li class="chapter" data-level="6.2" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#importing-data-4"><i class="fa fa-check"></i><b>6.2</b> Importing data</a></li>
<li class="chapter" data-level="6.3" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#data-curation"><i class="fa fa-check"></i><b>6.3</b> Data curation</a></li>
<li class="chapter" data-level="6.4" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#regularized-generalized-linear-models"><i class="fa fa-check"></i><b>6.4</b> Regularized Generalized Linear Models</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#lasso-alpha-1"><i class="fa fa-check"></i><b>6.4.1</b> Lasso: alpha = 1</a></li>
<li class="chapter" data-level="6.4.2" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#ridge-alpha-0"><i class="fa fa-check"></i><b>6.4.2</b> Ridge: alpha = 0</a></li>
<li class="chapter" data-level="6.4.3" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#elasticnet-0-alpha-1"><i class="fa fa-check"></i><b>6.4.3</b> Elasticnet: 0 &lt; alpha &lt; 1</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#classification"><i class="fa fa-check"></i><b>6.5</b> Classification</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#random-forest"><i class="fa fa-check"></i><b>6.5.1</b> Random Forest</a></li>
<li class="chapter" data-level="6.5.2" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#support-vector-machine-svm"><i class="fa fa-check"></i><b>6.5.2</b> Support Vector Machine (SVM)</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#systematic-information-4"><i class="fa fa-check"></i><b>6.6</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="network-analysis.html"><a href="network-analysis.html"><i class="fa fa-check"></i><b>7</b> Network Analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="network-analysis.html"><a href="network-analysis.html#introduction-1"><i class="fa fa-check"></i><b>7.1</b> Introduction</a></li>
<li class="chapter" data-level="7.2" data-path="network-analysis.html"><a href="network-analysis.html#loading-packages-5"><i class="fa fa-check"></i><b>7.2</b> Loading packages</a></li>
<li class="chapter" data-level="7.3" data-path="network-analysis.html"><a href="network-analysis.html#importing-data-5"><i class="fa fa-check"></i><b>7.3</b> Importing data</a></li>
<li class="chapter" data-level="7.4" data-path="network-analysis.html"><a href="network-analysis.html#data-curation-1"><i class="fa fa-check"></i><b>7.4</b> Data curation</a></li>
<li class="chapter" data-level="7.5" data-path="network-analysis.html"><a href="network-analysis.html#associations-among-features"><i class="fa fa-check"></i><b>7.5</b> Associations Among Features</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="network-analysis.html"><a href="network-analysis.html#single-network-with-pearson-correlation-as-association-measure"><i class="fa fa-check"></i><b>7.5.1</b> Single network with Pearson correlation as association measure</a></li>
<li class="chapter" data-level="7.5.2" data-path="network-analysis.html"><a href="network-analysis.html#single-network-with-spearman-correlation-as-association-measure"><i class="fa fa-check"></i><b>7.5.2</b> Single network with spearman correlation as association measure</a></li>
<li class="chapter" data-level="7.5.3" data-path="network-analysis.html"><a href="network-analysis.html#single-network-with-wgcna-bicor-as-association-measure"><i class="fa fa-check"></i><b>7.5.3</b> Single network with WGCNA (bicor) as association measure</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="network-analysis.html"><a href="network-analysis.html#network-comparison"><i class="fa fa-check"></i><b>7.6</b> Network comparison</a>
<ul>
<li class="chapter" data-level="7.6.1" data-path="network-analysis.html"><a href="network-analysis.html#data-preparing"><i class="fa fa-check"></i><b>7.6.1</b> Data preparing</a></li>
<li class="chapter" data-level="7.6.2" data-path="network-analysis.html"><a href="network-analysis.html#building-network-model-3"><i class="fa fa-check"></i><b>7.6.2</b> Building network model</a></li>
<li class="chapter" data-level="7.6.3" data-path="network-analysis.html"><a href="network-analysis.html#network-analysis-1"><i class="fa fa-check"></i><b>7.6.3</b> Network analysis</a></li>
<li class="chapter" data-level="7.6.4" data-path="network-analysis.html"><a href="network-analysis.html#visualizing-the-network-3"><i class="fa fa-check"></i><b>7.6.4</b> Visualizing the network</a></li>
<li class="chapter" data-level="7.6.5" data-path="network-analysis.html"><a href="network-analysis.html#quantitative-network-comparison"><i class="fa fa-check"></i><b>7.6.5</b> Quantitative network comparison</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="network-analysis.html"><a href="network-analysis.html#systematic-information-5"><i class="fa fa-check"></i><b>7.7</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html"><i class="fa fa-check"></i><b>8</b> Enrichment Analysis</a>
<ul>
<li class="chapter" data-level="8.1" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#enviroment-set-up"><i class="fa fa-check"></i><b>8.1</b> Enviroment Set up</a></li>
<li class="chapter" data-level="8.2" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#over-representation-analysis-ora"><i class="fa fa-check"></i><b>8.2</b> Over representation analysis (ORA)</a></li>
<li class="chapter" data-level="8.3" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#clean-environment-1"><i class="fa fa-check"></i><b>8.3</b> Clean environment 1</a></li>
<li class="chapter" data-level="8.4" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#single-sample-profiling"><i class="fa fa-check"></i><b>8.4</b> Single Sample Profiling</a></li>
<li class="chapter" data-level="8.5" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#clean-environment-2"><i class="fa fa-check"></i><b>8.5</b> Clean environment 2</a></li>
<li class="chapter" data-level="8.6" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#quantitative-enrichment-analysis-qea"><i class="fa fa-check"></i><b>8.6</b> Quantitative Enrichment Analysis (QEA)</a></li>
<li class="chapter" data-level="8.7" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#systematic-information-6"><i class="fa fa-check"></i><b>8.7</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="pathway-analysis.html"><a href="pathway-analysis.html"><i class="fa fa-check"></i><b>9</b> Pathway Analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="pathway-analysis.html"><a href="pathway-analysis.html#enviroment-set-up-1"><i class="fa fa-check"></i><b>9.1</b> Enviroment Set up</a></li>
<li class="chapter" data-level="9.2" data-path="pathway-analysis.html"><a href="pathway-analysis.html#over-representation-analysis"><i class="fa fa-check"></i><b>9.2</b> Over representation analysis</a></li>
<li class="chapter" data-level="9.3" data-path="pathway-analysis.html"><a href="pathway-analysis.html#clean-environment-1-1"><i class="fa fa-check"></i><b>9.3</b> Clean environment 1</a></li>
<li class="chapter" data-level="9.4" data-path="pathway-analysis.html"><a href="pathway-analysis.html#concentration-table-qea"><i class="fa fa-check"></i><b>9.4</b> Concentration Table (QEA)</a></li>
<li class="chapter" data-level="9.5" data-path="pathway-analysis.html"><a href="pathway-analysis.html#systematic-information-7"><i class="fa fa-check"></i><b>9.5</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="example.html"><a href="example.html"><i class="fa fa-check"></i><b>10</b> Example</a>
<ul>
<li class="chapter" data-level="10.1" data-path="example.html"><a href="example.html#loading-packages-6"><i class="fa fa-check"></i><b>10.1</b> Loading packages</a></li>
<li class="chapter" data-level="10.2" data-path="example.html"><a href="example.html#importing-data-6"><i class="fa fa-check"></i><b>10.2</b> Importing data</a></li>
<li class="chapter" data-level="10.3" data-path="example.html"><a href="example.html#data-processing-1"><i class="fa fa-check"></i><b>10.3</b> Data Processing</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="example.html"><a href="example.html#data-checking-1"><i class="fa fa-check"></i><b>10.3.1</b> Data Checking</a></li>
<li class="chapter" data-level="10.3.2" data-path="example.html"><a href="example.html#data-filtering-1"><i class="fa fa-check"></i><b>10.3.2</b> Data Filtering</a></li>
<li class="chapter" data-level="10.3.3" data-path="example.html"><a href="example.html#missing-value-imputation-1"><i class="fa fa-check"></i><b>10.3.3</b> Missing Value Imputation</a></li>
<li class="chapter" data-level="10.3.4" data-path="example.html"><a href="example.html#data-normalization-1"><i class="fa fa-check"></i><b>10.3.4</b> Data Normalization</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="example.html"><a href="example.html#cluster-analysis-1"><i class="fa fa-check"></i><b>10.4</b> Cluster Analysis</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="example.html"><a href="example.html#hierarchical-clustering-1"><i class="fa fa-check"></i><b>10.4.1</b> Hierarchical Clustering</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="example.html"><a href="example.html#chemometrics-analysis-1"><i class="fa fa-check"></i><b>10.5</b> Chemometrics Analysis</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="example.html"><a href="example.html#partial-least-squares-discriminant-analysis-pls-da-1"><i class="fa fa-check"></i><b>10.5.1</b> Partial Least Squares-Discriminant Analysis (PLS-DA)</a></li>
<li class="chapter" data-level="10.5.2" data-path="example.html"><a href="example.html#sparse-partial-least-squares-discriminant-analysis-spls-da-1"><i class="fa fa-check"></i><b>10.5.2</b> Sparse Partial Least Squares-Discriminant Analysis (sPLS-DA)</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="example.html"><a href="example.html#univariate-analysis-1"><i class="fa fa-check"></i><b>10.6</b> Univariate Analysis</a>
<ul>
<li class="chapter" data-level="10.6.1" data-path="example.html"><a href="example.html#fold-change-analysis-1"><i class="fa fa-check"></i><b>10.6.1</b> Fold Change Analysis</a></li>
<li class="chapter" data-level="10.6.2" data-path="example.html"><a href="example.html#t-test-1"><i class="fa fa-check"></i><b>10.6.2</b> T Test</a></li>
<li class="chapter" data-level="10.6.3" data-path="example.html"><a href="example.html#volcano-plot-1"><i class="fa fa-check"></i><b>10.6.3</b> Volcano plot</a></li>
</ul></li>
<li class="chapter" data-level="10.7" data-path="example.html"><a href="example.html#feature-selection"><i class="fa fa-check"></i><b>10.7</b> Feature Selection</a>
<ul>
<li class="chapter" data-level="10.7.1" data-path="example.html"><a href="example.html#regularized-generalized-linear-models-lasso-alpha-1"><i class="fa fa-check"></i><b>10.7.1</b> Regularized Generalized Linear Models (Lasso: alpha = 1)</a></li>
<li class="chapter" data-level="10.7.2" data-path="example.html"><a href="example.html#classification-random-forest"><i class="fa fa-check"></i><b>10.7.2</b> Classification (Random Forest)</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="example.html"><a href="example.html#network-analysis-2"><i class="fa fa-check"></i><b>10.8</b> Network Analysis</a>
<ul>
<li class="chapter" data-level="10.8.1" data-path="example.html"><a href="example.html#data-curation-2"><i class="fa fa-check"></i><b>10.8.1</b> Data curation</a></li>
<li class="chapter" data-level="10.8.2" data-path="example.html"><a href="example.html#building-network-model-4"><i class="fa fa-check"></i><b>10.8.2</b> Building network model</a></li>
<li class="chapter" data-level="10.8.3" data-path="example.html"><a href="example.html#visualizing-the-network-4"><i class="fa fa-check"></i><b>10.8.3</b> Visualizing the network</a></li>
</ul></li>
<li class="chapter" data-level="10.9" data-path="example.html"><a href="example.html#network-analysis-by-wgcna"><i class="fa fa-check"></i><b>10.9</b> Network Analysis by WGCNA</a>
<ul>
<li class="chapter" data-level="10.9.1" data-path="example.html"><a href="example.html#data-curation-3"><i class="fa fa-check"></i><b>10.9.1</b> Data curation</a></li>
<li class="chapter" data-level="10.9.2" data-path="example.html"><a href="example.html#tuning-soft-thresholds"><i class="fa fa-check"></i><b>10.9.2</b> Tuning soft thresholds</a></li>
<li class="chapter" data-level="10.9.3" data-path="example.html"><a href="example.html#create-the-network-using-the-blockwisemodules"><i class="fa fa-check"></i><b>10.9.3</b> Create the network using the <code>blockwiseModules</code></a></li>
<li class="chapter" data-level="10.9.4" data-path="example.html"><a href="example.html#plot-modules"><i class="fa fa-check"></i><b>10.9.4</b> plot modules</a></li>
<li class="chapter" data-level="10.9.5" data-path="example.html"><a href="example.html#relationships-among-modules"><i class="fa fa-check"></i><b>10.9.5</b> Relationships among modules</a></li>
<li class="chapter" data-level="10.9.6" data-path="example.html"><a href="example.html#module-eigengene-correlation"><i class="fa fa-check"></i><b>10.9.6</b> Module (Eigengene) correlation</a></li>
<li class="chapter" data-level="10.9.7" data-path="example.html"><a href="example.html#relate-module-cluster-assignments-to-groups"><i class="fa fa-check"></i><b>10.9.7</b> Relate Module (cluster) Assignments to Groups</a></li>
<li class="chapter" data-level="10.9.8" data-path="example.html"><a href="example.html#generate-and-export-networks"><i class="fa fa-check"></i><b>10.9.8</b> Generate and Export Networks</a></li>
<li class="chapter" data-level="10.9.9" data-path="example.html"><a href="example.html#network-visualization"><i class="fa fa-check"></i><b>10.9.9</b> Network visualization</a></li>
</ul></li>
<li class="chapter" data-level="10.10" data-path="example.html"><a href="example.html#systematic-information-8"><i class="fa fa-check"></i><b>10.10</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data Analysis in Metabolomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="network-analysis" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">Chapter 7</span> Network Analysis<a href="network-analysis.html#network-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="introduction-1" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Introduction<a href="network-analysis.html#introduction-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<blockquote>
<p>Estimating microbial association networks from high-throughput sequencing data is a common exploratory data analysis approach aiming at understanding the complex interplay of microbial communities in their natural habitat. Statistical network estimation workflows comprise several analysis steps, including methods for zero handling, data normalization and computing microbial associations. Since microbial interactions are likely to change between conditions, e.g. between healthy individuals and patients, identifying network differences between groups is often an integral secondary analysis step.</p>
</blockquote>
<p>NetCoMi (Network Construction and Comparison for Microbiome Data) <span class="citation">(<a href="#ref-peschel2021netcomi" role="doc-biblioref">Peschel et al. 2021</a>)</span> provides functionality for constructing, analyzing, and comparing networks suitable for the application on microbial compositional data.</p>
<p>The following information is from <a href="https://github.com/stefpeschel/NetCoMi">NetCoMi github</a>.</p>
<p><strong>Association measures:</strong></p>
<ul>
<li>Pearson coefficient
(<a href="https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/cor"><code>cor()</code></a>
from <code>stats</code> package)</li>
<li>Spearman coefficient
(<a href="https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/cor"><code>cor()</code></a>
from <code>stats</code> package)</li>
<li>Biweight Midcorrelation
<a href="https://rdrr.io/cran/WGCNA/man/bicor.html"><code>bicor()</code></a> from <code>WGCNA</code>
package</li>
</ul>
<p><strong>Methods for zero replacement:</strong></p>
<ul>
<li>Adding a predefined pseudo count</li>
<li>Multiplicative replacement
(<a href="https://rdrr.io/cran/zCompositions/man/multRepl.html"><code>multRepl</code></a>
from <code>zCompositions</code> package)</li>
<li>Modified EM alr-algorithm
(<a href="https://rdrr.io/cran/zCompositions/man/lrEM.html"><code>lrEM</code></a> from
<code>zCompositions</code> package)</li>
<li>Bayesian-multiplicative replacement
(<a href="https://rdrr.io/cran/zCompositions/man/cmultRepl.html"><code>cmultRepl</code></a>
from <code>zCompositions</code> package)</li>
</ul>
<p><strong>Normalization methods:</strong></p>
<ul>
<li>Total Sum Scaling (TSS) (own implementation)</li>
<li>Cumulative Sum Scaling (CSS) (<a href=""><code>cumNormMat</code></a> from <code>metagenomeSeq</code>
package)</li>
<li>Common Sum Scaling (COM) (own implementation)</li>
<li>Rarefying (<a href=""><code>rrarefy</code></a> from <code>vegan</code> package)</li>
<li>Variance Stabilizing Transformation (VST)
(<a href=""><code>varianceStabilizingTransformation</code></a> from <code>DESeq2</code> package)</li>
<li>Centered log-ratio (clr) transformation
(<a href="https://rdrr.io/github/zdk123/SpiecEasi/man/clr.html"><code>clr()</code></a>
from <code>SpiecEasi</code> package))</li>
</ul>
<p>TSS, CSS, COM, VST, and the clr transformation are described in [Badri et al., 2020].</p>
</div>
<div id="loading-packages-5" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> Loading packages<a href="network-analysis.html#loading-packages-5" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="network-analysis.html#cb101-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">warning =</span> F)</span>
<span id="cb101-2"><a href="network-analysis.html#cb101-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-3"><a href="network-analysis.html#cb101-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb101-4"><a href="network-analysis.html#cb101-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb101-5"><a href="network-analysis.html#cb101-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(POMA)</span>
<span id="cb101-6"><a href="network-analysis.html#cb101-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb101-7"><a href="network-analysis.html#cb101-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggraph)</span>
<span id="cb101-8"><a href="network-analysis.html#cb101-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb101-9"><a href="network-analysis.html#cb101-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb101-10"><a href="network-analysis.html#cb101-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-11"><a href="network-analysis.html#cb101-11" aria-hidden="true" tabindex="-1"></a><span class="co"># installing package</span></span>
<span id="cb101-12"><a href="network-analysis.html#cb101-12" aria-hidden="true" tabindex="-1"></a><span class="co"># devtools::install_github(&quot;stefpeschel/NetCoMi&quot;, </span></span>
<span id="cb101-13"><a href="network-analysis.html#cb101-13" aria-hidden="true" tabindex="-1"></a><span class="co">#                          dependencies = c(&quot;Depends&quot;, &quot;Imports&quot;, &quot;LinkingTo&quot;),</span></span>
<span id="cb101-14"><a href="network-analysis.html#cb101-14" aria-hidden="true" tabindex="-1"></a><span class="co">#                          repos = c(&quot;https://cloud.r-project.org/&quot;,</span></span>
<span id="cb101-15"><a href="network-analysis.html#cb101-15" aria-hidden="true" tabindex="-1"></a><span class="co">#                                    BiocManager::repositories()))</span></span>
<span id="cb101-16"><a href="network-analysis.html#cb101-16" aria-hidden="true" tabindex="-1"></a><span class="co"># devtools::install_github(&quot;GraceYoon/SPRING&quot;, force = TRUE)</span></span>
<span id="cb101-17"><a href="network-analysis.html#cb101-17" aria-hidden="true" tabindex="-1"></a><span class="co"># devtools::install_github(&quot;zdk123/SpiecEasi&quot;, force = TRUE)</span></span>
<span id="cb101-18"><a href="network-analysis.html#cb101-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(NetCoMi)</span>
<span id="cb101-19"><a href="network-analysis.html#cb101-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SPRING)</span>
<span id="cb101-20"><a href="network-analysis.html#cb101-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SpiecEasi)</span>
<span id="cb101-21"><a href="network-analysis.html#cb101-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-22"><a href="network-analysis.html#cb101-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-23"><a href="network-analysis.html#cb101-23" aria-hidden="true" tabindex="-1"></a><span class="co"># rm(list = ls())</span></span>
<span id="cb101-24"><a href="network-analysis.html#cb101-24" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb101-25"><a href="network-analysis.html#cb101-25" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">future.globals.maxSize =</span> <span class="dv">1000</span> <span class="sc">*</span> <span class="dv">1024</span><span class="sc">^</span><span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="importing-data-5" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> Importing data<a href="network-analysis.html#importing-data-5" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The input data sets are from the previous chapter.</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="network-analysis.html#cb102-1" aria-hidden="true" tabindex="-1"></a>se_filter <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;./dataset/POMA/se_filter.RDS&quot;</span>)</span></code></pre></div>
</div>
<div id="data-curation-1" class="section level2 hasAnchor" number="7.4">
<h2><span class="header-section-number">7.4</span> Data curation<a href="network-analysis.html#data-curation-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="network-analysis.html#cb103-1" aria-hidden="true" tabindex="-1"></a>features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(se_filter) <span class="sc">%&gt;%</span> </span>
<span id="cb103-2"><a href="network-analysis.html#cb103-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>()</span>
<span id="cb103-3"><a href="network-analysis.html#cb103-3" aria-hidden="true" tabindex="-1"></a>features_tab[<span class="fu">is.na</span>(features_tab)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb103-4"><a href="network-analysis.html#cb103-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-5"><a href="network-analysis.html#cb103-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(features_tab)</span></code></pre></div>
<pre><code>##           M_52603  M_19130   M_39270 M_35186 M_34214  M_49617  M_52710  M_53189   M_34419  M_36600  M_54885  M_32350 M_55041
## P101001  94392176 25632184  959249.2 8141286 8284910 381540.1 47739292 392911.0 306346432  9887348 298886.1 627613.7 3655240
## P101004 115155104 25106562  785895.0 4303854 7923489 588488.2 43678784 300383.9 263333424  9703074 356274.3 339586.8 1637351
## P101007  79582632 31371314 1903592.4 7129884 9211485 464548.8 53975068 539361.5 191261648  6431966 393623.0 662555.6 7284474
## P101009 118408760 27787270  863701.9 5394354 8559147 497342.0 47619564 487186.2 232171184 11914472 599783.6 492430.9 3065707
## P101010  92508664 26685844 1177860.8 6270727 7623603 457409.1 61906028 236719.5 183104336  9366287 280123.8 373484.2 2493687
## P101011  94076424 27780988  638271.3 4292087 6074633 238781.8 40534280 279118.7 164623888  6003700 349482.3 608671.7 2653816
##           M_48258 M_35628  M_33230   M_52462  M_52464   M_52454   M_52610   M_42446   M_52461  M_52616   M_33955 M_35631   M_42450
## P101001 181136336 6548460 19098412 486910976  6794782 274026048 206279104 712253440 486655872 12365879 448634784 4035864 358896736
## P101004 115752200 3455001 10641834 380887616  4090405 282066016 253920496 807180352 439253152 10655923 413690720 3446139 282910752
## P101007 102036920 3640571 14977371 515795264 13183206 309709728 316142400 715500800 448578400 14010174 368849632 3764448 404918560
## P101009 118663320 5087503 13575508 390977568  7596680 325723168 232534688 742576768 473622240 14814230 365712768 3444159 298822048
## P101010 154183360 5244878 19464998 478479936  5395172 382690720 262177072 642967424 647036736 12873290 376101984 3276579 373251712
## P101011  76137216 2238570  8797510 420938976  4854076 261022832 264648720 728199680 378921056 13152187 348108832 2364665 335196864
##          M_52447  M_52449   M_52452   M_52438   M_33961 M_42398  M_47403  M_57547   M_19266   M_63739  M_57663 M_42489  M_37253
## P101001 30252208 43173148 510461472 181406848 191147264 6137137 209255.5 169297.7 176944.80  761146.2 719237.9 2865827 411578.0
## P101004 19424238 30030992 573170816 122556968 166379712 5143481 220424.3 426227.5      0.00  549097.4 364556.1 1546570 425030.0
## P101007 57376512 50766284 588643712 181572112 189915296 7412535 369675.0 561078.5 385203.22  792883.2 500680.8 1403190 742298.0
## P101009 26429340 43420376 631965888 208484000 186599152 5421984 172995.7 457043.3      0.00  727360.0 523549.1 1653830 466825.0
## P101010 21732290 43622724 590247424 306379392 217632736 5767624 168503.4 602024.4      0.00  913964.4 455201.1 1007203 232876.0
## P101011 21960100 39447380 560182336 136970272 170865408 3588338 375137.6 304329.9  90888.59 1609550.0 449288.0 1818438 651777.4
##          M_45095   M_53229   M_55072  M_47118  M_35253   M_46115 M_62952  M_63681 M_63361
## P101001 751527.3  768160.4  488544.9 281162.7 27446600 1578011.4 5104806 434480.3 2540280
## P101004 378033.7  953194.9  653107.0 362380.0 28088842  440408.7 3084011 373229.5 2165454
## P101007 589506.3 1619436.8  522721.5 482046.2 21613430  794592.2 4728273 400386.6 2231037
## P101009 503910.2  877530.2  873616.8 619943.9 28872470  671497.6 4140767 301840.8       0
## P101010 489013.9  993923.4  864624.8 677774.7 25375946 1133763.9 2434359 548330.4 1125607
## P101011 785751.6 1092817.9 1360312.8 208040.6 17388928 1635119.5 5722723 344799.8       0</code></pre>
</div>
<div id="associations-among-features" class="section level2 hasAnchor" number="7.5">
<h2><span class="header-section-number">7.5</span> Associations Among Features<a href="network-analysis.html#associations-among-features" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="single-network-with-pearson-correlation-as-association-measure" class="section level3 hasAnchor" number="7.5.1">
<h3><span class="header-section-number">7.5.1</span> Single network with Pearson correlation as association measure<a href="network-analysis.html#single-network-with-pearson-correlation-as-association-measure" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Since Pearson correlations may lead to compositional effects when applied to sequencing data, we use the clr transformation as normalization method. Zero treatment is necessary in this case.</p>
<ul>
<li><p>InputData: numeric matrix. Can be a count matrix (rows are samples, columns are Features).</p></li>
<li><p>Method to compute the associations between features (argument measure).</p></li>
<li><p>Normalization method:</p>
<ul>
<li><p>normMethod: clr</p></li>
<li><p>zeroMethod: multRepl</p></li>
<li><p>sparsMethod: threshold</p></li>
</ul></li>
</ul>
<p>A threshold of 0.3 is used as sparsification method, so that only OTUs with an absolute correlation greater than or equal to 0.3 are connected.</p>
<div id="building-network-model" class="section level4 hasAnchor" number="7.5.1.1">
<h4><span class="header-section-number">7.5.1.1</span> Building network model<a href="network-analysis.html#building-network-model" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="network-analysis.html#cb105-1" aria-hidden="true" tabindex="-1"></a>net_single <span class="ot">&lt;-</span> <span class="fu">netConstruct</span>(features_tab,  </span>
<span id="cb105-2"><a href="network-analysis.html#cb105-2" aria-hidden="true" tabindex="-1"></a>                           <span class="at">measure =</span> <span class="st">&quot;pearson&quot;</span>,</span>
<span id="cb105-3"><a href="network-analysis.html#cb105-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">normMethod =</span> <span class="st">&quot;clr&quot;</span>, </span>
<span id="cb105-4"><a href="network-analysis.html#cb105-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">zeroMethod =</span> <span class="st">&quot;multRepl&quot;</span>,</span>
<span id="cb105-5"><a href="network-analysis.html#cb105-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">sparsMethod =</span> <span class="st">&quot;threshold&quot;</span>, </span>
<span id="cb105-6"><a href="network-analysis.html#cb105-6" aria-hidden="true" tabindex="-1"></a>                           <span class="at">thresh =</span> <span class="fl">0.3</span>,</span>
<span id="cb105-7"><a href="network-analysis.html#cb105-7" aria-hidden="true" tabindex="-1"></a>                           <span class="at">verbose =</span> <span class="dv">3</span>,</span>
<span id="cb105-8"><a href="network-analysis.html#cb105-8" aria-hidden="true" tabindex="-1"></a>                           <span class="at">seed =</span> <span class="dv">123</span>)</span></code></pre></div>
</div>
<div id="visualizing-the-network" class="section level4 hasAnchor" number="7.5.1.2">
<h4><span class="header-section-number">7.5.1.2</span> Visualizing the network<a href="network-analysis.html#visualizing-the-network" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>primary plot</li>
</ul>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="network-analysis.html#cb106-1" aria-hidden="true" tabindex="-1"></a>props_single <span class="ot">&lt;-</span> <span class="fu">netAnalyze</span>(net_single, <span class="at">clustMethod =</span> <span class="st">&quot;cluster_fast_greedy&quot;</span>)</span>
<span id="cb106-2"><a href="network-analysis.html#cb106-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-3"><a href="network-analysis.html#cb106-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(props_single, </span>
<span id="cb106-4"><a href="network-analysis.html#cb106-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeColor =</span> <span class="st">&quot;cluster&quot;</span>, </span>
<span id="cb106-5"><a href="network-analysis.html#cb106-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeSize =</span> <span class="st">&quot;eigenvector&quot;</span>,</span>
<span id="cb106-6"><a href="network-analysis.html#cb106-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">title1 =</span> <span class="st">&quot;Network on metabolomics with Pearson correlations&quot;</span>, </span>
<span id="cb106-7"><a href="network-analysis.html#cb106-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">showTitle =</span> <span class="cn">TRUE</span>,</span>
<span id="cb106-8"><a href="network-analysis.html#cb106-8" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexTitle =</span> <span class="fl">1.5</span>)</span>
<span id="cb106-9"><a href="network-analysis.html#cb106-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-10"><a href="network-analysis.html#cb106-10" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="fl">0.7</span>, <span class="fl">1.1</span>, <span class="at">cex =</span> <span class="dv">1</span>, <span class="at">title =</span> <span class="st">&quot;estimated correlation:&quot;</span>, </span>
<span id="cb106-11"><a href="network-analysis.html#cb106-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;+&quot;</span>,<span class="st">&quot;-&quot;</span>), <span class="at">lty =</span> <span class="dv">1</span>, <span class="at">lwd =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;#009900&quot;</span>,<span class="st">&quot;red&quot;</span>), </span>
<span id="cb106-12"><a href="network-analysis.html#cb106-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">horiz =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-67-1.png" width="768" /></p>
<ul>
<li><p>improve the visualization by changing the following arguments:</p>
<ul>
<li><p><code>repulsion = 0.8</code>: Place the nodes further apart</p></li>
<li><p><code>rmSingles = TRUE</code>: Single nodes are removed</p></li>
<li><p><code>labelScale = FALSE</code> and <code>cexLabels = 1.6</code>: All labels have equal size and are enlarged to improve readability of small node’s labels</p></li>
<li><p><code>nodeSizeSpread = 3</code> (default is 4): Node sizes are more similar if the value is decreased. This argument (in combination with <code>cexNodes</code>) is useful to enlarge small nodes while keeping the size of big nodes.</p></li>
</ul></li>
</ul>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="network-analysis.html#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(props_single, </span>
<span id="cb107-2"><a href="network-analysis.html#cb107-2" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeColor =</span> <span class="st">&quot;cluster&quot;</span>, </span>
<span id="cb107-3"><a href="network-analysis.html#cb107-3" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeSize =</span> <span class="st">&quot;eigenvector&quot;</span>,</span>
<span id="cb107-4"><a href="network-analysis.html#cb107-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">repulsion =</span> <span class="fl">0.8</span>,</span>
<span id="cb107-5"><a href="network-analysis.html#cb107-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">rmSingles =</span> <span class="cn">TRUE</span>,</span>
<span id="cb107-6"><a href="network-analysis.html#cb107-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">labelScale =</span> <span class="cn">FALSE</span>,</span>
<span id="cb107-7"><a href="network-analysis.html#cb107-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexLabels =</span> <span class="fl">1.6</span>,</span>
<span id="cb107-8"><a href="network-analysis.html#cb107-8" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeSizeSpread =</span> <span class="dv">3</span>,</span>
<span id="cb107-9"><a href="network-analysis.html#cb107-9" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexNodes =</span> <span class="dv">2</span>,</span>
<span id="cb107-10"><a href="network-analysis.html#cb107-10" aria-hidden="true" tabindex="-1"></a>     <span class="at">title1 =</span> <span class="st">&quot;Network on metabolomics with Pearson correlations&quot;</span>, </span>
<span id="cb107-11"><a href="network-analysis.html#cb107-11" aria-hidden="true" tabindex="-1"></a>     <span class="at">showTitle =</span> <span class="cn">TRUE</span>,</span>
<span id="cb107-12"><a href="network-analysis.html#cb107-12" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexTitle =</span> <span class="fl">1.5</span>)</span>
<span id="cb107-13"><a href="network-analysis.html#cb107-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-14"><a href="network-analysis.html#cb107-14" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="fl">0.7</span>, <span class="fl">1.1</span>, <span class="at">cex =</span> <span class="fl">1.2</span>, <span class="at">title =</span> <span class="st">&quot;estimated correlation:&quot;</span>,</span>
<span id="cb107-15"><a href="network-analysis.html#cb107-15" aria-hidden="true" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;+&quot;</span>,<span class="st">&quot;-&quot;</span>), <span class="at">lty =</span> <span class="dv">1</span>, <span class="at">lwd =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;#009900&quot;</span>,<span class="st">&quot;red&quot;</span>),</span>
<span id="cb107-16"><a href="network-analysis.html#cb107-16" aria-hidden="true" tabindex="-1"></a>       <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">horiz =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-68-1.png" width="1152" /></p>
</div>
</div>
<div id="single-network-with-spearman-correlation-as-association-measure" class="section level3 hasAnchor" number="7.5.2">
<h3><span class="header-section-number">7.5.2</span> Single network with spearman correlation as association measure<a href="network-analysis.html#single-network-with-spearman-correlation-as-association-measure" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="building-network-model-1" class="section level4 hasAnchor" number="7.5.2.1">
<h4><span class="header-section-number">7.5.2.1</span> Building network model<a href="network-analysis.html#building-network-model-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="network-analysis.html#cb108-1" aria-hidden="true" tabindex="-1"></a>net_single2 <span class="ot">&lt;-</span> <span class="fu">netConstruct</span>(features_tab,  </span>
<span id="cb108-2"><a href="network-analysis.html#cb108-2" aria-hidden="true" tabindex="-1"></a>                            <span class="at">measure =</span> <span class="st">&quot;spearman&quot;</span>,</span>
<span id="cb108-3"><a href="network-analysis.html#cb108-3" aria-hidden="true" tabindex="-1"></a>                            <span class="at">normMethod =</span> <span class="st">&quot;clr&quot;</span>, </span>
<span id="cb108-4"><a href="network-analysis.html#cb108-4" aria-hidden="true" tabindex="-1"></a>                            <span class="at">zeroMethod =</span> <span class="st">&quot;multRepl&quot;</span>,</span>
<span id="cb108-5"><a href="network-analysis.html#cb108-5" aria-hidden="true" tabindex="-1"></a>                            <span class="at">sparsMethod =</span> <span class="st">&quot;threshold&quot;</span>, </span>
<span id="cb108-6"><a href="network-analysis.html#cb108-6" aria-hidden="true" tabindex="-1"></a>                            <span class="at">thresh =</span> <span class="fl">0.3</span>,</span>
<span id="cb108-7"><a href="network-analysis.html#cb108-7" aria-hidden="true" tabindex="-1"></a>                            <span class="at">verbose =</span> <span class="dv">3</span>,</span>
<span id="cb108-8"><a href="network-analysis.html#cb108-8" aria-hidden="true" tabindex="-1"></a>                            <span class="at">seed =</span> <span class="dv">123</span>)</span></code></pre></div>
</div>
<div id="visualizing-the-network-1" class="section level4 hasAnchor" number="7.5.2.2">
<h4><span class="header-section-number">7.5.2.2</span> Visualizing the network<a href="network-analysis.html#visualizing-the-network-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="network-analysis.html#cb109-1" aria-hidden="true" tabindex="-1"></a>props_single2 <span class="ot">&lt;-</span> <span class="fu">netAnalyze</span>(net_single2, <span class="at">clustMethod =</span> <span class="st">&quot;cluster_fast_greedy&quot;</span>)</span>
<span id="cb109-2"><a href="network-analysis.html#cb109-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-3"><a href="network-analysis.html#cb109-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(props_single2, </span>
<span id="cb109-4"><a href="network-analysis.html#cb109-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeColor =</span> <span class="st">&quot;cluster&quot;</span>, </span>
<span id="cb109-5"><a href="network-analysis.html#cb109-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeSize =</span> <span class="st">&quot;eigenvector&quot;</span>,</span>
<span id="cb109-6"><a href="network-analysis.html#cb109-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">repulsion =</span> <span class="fl">0.8</span>,</span>
<span id="cb109-7"><a href="network-analysis.html#cb109-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">rmSingles =</span> <span class="cn">TRUE</span>,</span>
<span id="cb109-8"><a href="network-analysis.html#cb109-8" aria-hidden="true" tabindex="-1"></a>     <span class="at">labelScale =</span> <span class="cn">FALSE</span>,</span>
<span id="cb109-9"><a href="network-analysis.html#cb109-9" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexLabels =</span> <span class="fl">1.6</span>,</span>
<span id="cb109-10"><a href="network-analysis.html#cb109-10" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeSizeSpread =</span> <span class="dv">3</span>,</span>
<span id="cb109-11"><a href="network-analysis.html#cb109-11" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexNodes =</span> <span class="dv">2</span>,</span>
<span id="cb109-12"><a href="network-analysis.html#cb109-12" aria-hidden="true" tabindex="-1"></a>     <span class="at">title1 =</span> <span class="st">&quot;Network on metabolomics with Spearman correlations&quot;</span>, </span>
<span id="cb109-13"><a href="network-analysis.html#cb109-13" aria-hidden="true" tabindex="-1"></a>     <span class="at">showTitle =</span> <span class="cn">TRUE</span>,</span>
<span id="cb109-14"><a href="network-analysis.html#cb109-14" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexTitle =</span> <span class="fl">1.5</span>)</span>
<span id="cb109-15"><a href="network-analysis.html#cb109-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-16"><a href="network-analysis.html#cb109-16" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="fl">0.7</span>, <span class="fl">1.1</span>, <span class="at">cex =</span> <span class="fl">1.2</span>, <span class="at">title =</span> <span class="st">&quot;estimated correlation:&quot;</span>,</span>
<span id="cb109-17"><a href="network-analysis.html#cb109-17" aria-hidden="true" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;+&quot;</span>,<span class="st">&quot;-&quot;</span>), <span class="at">lty =</span> <span class="dv">1</span>, <span class="at">lwd =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;#009900&quot;</span>,<span class="st">&quot;red&quot;</span>),</span>
<span id="cb109-18"><a href="network-analysis.html#cb109-18" aria-hidden="true" tabindex="-1"></a>       <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">horiz =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-70-1.png" width="1152" /></p>
</div>
</div>
<div id="single-network-with-wgcna-bicor-as-association-measure" class="section level3 hasAnchor" number="7.5.3">
<h3><span class="header-section-number">7.5.3</span> Single network with WGCNA (bicor) as association measure<a href="network-analysis.html#single-network-with-wgcna-bicor-as-association-measure" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Biweight Midcorrelation <code>bicor()</code> from WGCNA package.</p>
<div id="building-network-model-2" class="section level4 hasAnchor" number="7.5.3.1">
<h4><span class="header-section-number">7.5.3.1</span> Building network model<a href="network-analysis.html#building-network-model-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="network-analysis.html#cb110-1" aria-hidden="true" tabindex="-1"></a>net_single3 <span class="ot">&lt;-</span> <span class="fu">netConstruct</span>(features_tab, </span>
<span id="cb110-2"><a href="network-analysis.html#cb110-2" aria-hidden="true" tabindex="-1"></a>                           <span class="at">measure =</span> <span class="st">&quot;bicor&quot;</span>,</span>
<span id="cb110-3"><a href="network-analysis.html#cb110-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">measurePar =</span> <span class="fu">list</span>(<span class="at">use =</span> <span class="st">&quot;all.obs&quot;</span>,</span>
<span id="cb110-4"><a href="network-analysis.html#cb110-4" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">maxPOutliers =</span> <span class="dv">1</span>,</span>
<span id="cb110-5"><a href="network-analysis.html#cb110-5" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">nThreads =</span> <span class="dv">2</span>),</span>
<span id="cb110-6"><a href="network-analysis.html#cb110-6" aria-hidden="true" tabindex="-1"></a>                           <span class="at">filtTax =</span> <span class="st">&quot;highestVar&quot;</span>,</span>
<span id="cb110-7"><a href="network-analysis.html#cb110-7" aria-hidden="true" tabindex="-1"></a>                           <span class="at">filtTaxPar =</span> <span class="fu">list</span>(<span class="at">highestVar =</span> <span class="dv">50</span>),</span>
<span id="cb110-8"><a href="network-analysis.html#cb110-8" aria-hidden="true" tabindex="-1"></a>                           <span class="at">filtSamp =</span> <span class="st">&quot;totalReads&quot;</span>,</span>
<span id="cb110-9"><a href="network-analysis.html#cb110-9" aria-hidden="true" tabindex="-1"></a>                           <span class="at">filtSampPar =</span> <span class="fu">list</span>(<span class="at">totalReads =</span> <span class="dv">100</span>),</span>
<span id="cb110-10"><a href="network-analysis.html#cb110-10" aria-hidden="true" tabindex="-1"></a>                           <span class="at">dissFunc =</span> <span class="st">&quot;TOMdiss&quot;</span>,</span>
<span id="cb110-11"><a href="network-analysis.html#cb110-11" aria-hidden="true" tabindex="-1"></a>                           <span class="at">verbose =</span> <span class="dv">3</span>)</span></code></pre></div>
<pre><code>## ..will use 2 parallel threads.
##  Fraction of slow calculations: 0.000000</code></pre>
</div>
<div id="visualizing-the-network-2" class="section level4 hasAnchor" number="7.5.3.2">
<h4><span class="header-section-number">7.5.3.2</span> Visualizing the network<a href="network-analysis.html#visualizing-the-network-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="network-analysis.html#cb112-1" aria-hidden="true" tabindex="-1"></a>props_single3 <span class="ot">&lt;-</span> <span class="fu">netAnalyze</span>(net_single3, <span class="at">clustMethod =</span> <span class="st">&quot;cluster_fast_greedy&quot;</span>)</span>
<span id="cb112-2"><a href="network-analysis.html#cb112-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-3"><a href="network-analysis.html#cb112-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(props_single3, </span>
<span id="cb112-4"><a href="network-analysis.html#cb112-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeColor =</span> <span class="st">&quot;cluster&quot;</span>, </span>
<span id="cb112-5"><a href="network-analysis.html#cb112-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeSize =</span> <span class="st">&quot;eigenvector&quot;</span>,</span>
<span id="cb112-6"><a href="network-analysis.html#cb112-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">repulsion =</span> <span class="fl">0.8</span>,</span>
<span id="cb112-7"><a href="network-analysis.html#cb112-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">rmSingles =</span> <span class="cn">TRUE</span>,</span>
<span id="cb112-8"><a href="network-analysis.html#cb112-8" aria-hidden="true" tabindex="-1"></a>     <span class="at">labelScale =</span> <span class="cn">FALSE</span>,</span>
<span id="cb112-9"><a href="network-analysis.html#cb112-9" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexLabels =</span> <span class="fl">1.6</span>,</span>
<span id="cb112-10"><a href="network-analysis.html#cb112-10" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeSizeSpread =</span> <span class="dv">3</span>,</span>
<span id="cb112-11"><a href="network-analysis.html#cb112-11" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexNodes =</span> <span class="dv">2</span>,</span>
<span id="cb112-12"><a href="network-analysis.html#cb112-12" aria-hidden="true" tabindex="-1"></a>     <span class="at">title1 =</span> <span class="st">&quot;Network on metabolomics with WGCNA correlations&quot;</span>, </span>
<span id="cb112-13"><a href="network-analysis.html#cb112-13" aria-hidden="true" tabindex="-1"></a>     <span class="at">showTitle =</span> <span class="cn">TRUE</span>,</span>
<span id="cb112-14"><a href="network-analysis.html#cb112-14" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexTitle =</span> <span class="fl">1.5</span>)</span>
<span id="cb112-15"><a href="network-analysis.html#cb112-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-16"><a href="network-analysis.html#cb112-16" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="fl">0.7</span>, <span class="fl">1.1</span>, <span class="at">cex =</span> <span class="fl">1.2</span>, <span class="at">title =</span> <span class="st">&quot;estimated correlation:&quot;</span>,</span>
<span id="cb112-17"><a href="network-analysis.html#cb112-17" aria-hidden="true" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;+&quot;</span>,<span class="st">&quot;-&quot;</span>), <span class="at">lty =</span> <span class="dv">1</span>, <span class="at">lwd =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;#009900&quot;</span>,<span class="st">&quot;red&quot;</span>),</span>
<span id="cb112-18"><a href="network-analysis.html#cb112-18" aria-hidden="true" tabindex="-1"></a>       <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">horiz =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-72-1.png" width="1152" /></p>
</div>
</div>
</div>
<div id="network-comparison" class="section level2 hasAnchor" number="7.6">
<h2><span class="header-section-number">7.6</span> Network comparison<a href="network-analysis.html#network-comparison" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Comparing two networks by <strong>NetCoMi</strong>.</p>
<div id="data-preparing" class="section level3 hasAnchor" number="7.6.1">
<h3><span class="header-section-number">7.6.1</span> Data preparing<a href="network-analysis.html#data-preparing" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="network-analysis.html#cb113-1" aria-hidden="true" tabindex="-1"></a>group_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Severe&quot;</span>)</span>
<span id="cb113-2"><a href="network-analysis.html#cb113-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-3"><a href="network-analysis.html#cb113-3" aria-hidden="true" tabindex="-1"></a>se_filter_subset <span class="ot">&lt;-</span> se_filter[, se_filter<span class="sc">$</span>group <span class="sc">%in%</span> group_names]</span>
<span id="cb113-4"><a href="network-analysis.html#cb113-4" aria-hidden="true" tabindex="-1"></a>se_filter_subset<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">as.character</span>(se_filter_subset<span class="sc">$</span>group))</span>
<span id="cb113-5"><a href="network-analysis.html#cb113-5" aria-hidden="true" tabindex="-1"></a>features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(se_filter_subset) <span class="sc">%&gt;%</span> </span>
<span id="cb113-6"><a href="network-analysis.html#cb113-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>()</span>
<span id="cb113-7"><a href="network-analysis.html#cb113-7" aria-hidden="true" tabindex="-1"></a>features_tab[<span class="fu">is.na</span>(features_tab)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb113-8"><a href="network-analysis.html#cb113-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-9"><a href="network-analysis.html#cb113-9" aria-hidden="true" tabindex="-1"></a>group_vector <span class="ot">&lt;-</span> se_filter_subset<span class="sc">$</span>group</span></code></pre></div>
</div>
<div id="building-network-model-3" class="section level3 hasAnchor" number="7.6.2">
<h3><span class="header-section-number">7.6.2</span> Building network model<a href="network-analysis.html#building-network-model-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="network-analysis.html#cb114-1" aria-hidden="true" tabindex="-1"></a>net_group <span class="ot">&lt;-</span> <span class="fu">netConstruct</span>(features_tab, </span>
<span id="cb114-2"><a href="network-analysis.html#cb114-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">group =</span> group_vector, </span>
<span id="cb114-3"><a href="network-analysis.html#cb114-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">measure =</span> <span class="st">&quot;pearson&quot;</span>,</span>
<span id="cb114-4"><a href="network-analysis.html#cb114-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">normMethod =</span> <span class="st">&quot;clr&quot;</span>, </span>
<span id="cb114-5"><a href="network-analysis.html#cb114-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">zeroMethod =</span> <span class="st">&quot;multRepl&quot;</span>,</span>
<span id="cb114-6"><a href="network-analysis.html#cb114-6" aria-hidden="true" tabindex="-1"></a>                          <span class="at">sparsMethod =</span> <span class="st">&quot;threshold&quot;</span>, </span>
<span id="cb114-7"><a href="network-analysis.html#cb114-7" aria-hidden="true" tabindex="-1"></a>                          <span class="at">thresh =</span> <span class="fl">0.3</span>,</span>
<span id="cb114-8"><a href="network-analysis.html#cb114-8" aria-hidden="true" tabindex="-1"></a>                          <span class="at">verbose =</span> <span class="dv">3</span>,</span>
<span id="cb114-9"><a href="network-analysis.html#cb114-9" aria-hidden="true" tabindex="-1"></a>                          <span class="at">seed =</span> <span class="dv">123</span>)</span></code></pre></div>
</div>
<div id="network-analysis-1" class="section level3 hasAnchor" number="7.6.3">
<h3><span class="header-section-number">7.6.3</span> Network analysis<a href="network-analysis.html#network-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="network-analysis.html#cb115-1" aria-hidden="true" tabindex="-1"></a>props_group <span class="ot">&lt;-</span> <span class="fu">netAnalyze</span>(net_group, </span>
<span id="cb115-2"><a href="network-analysis.html#cb115-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">centrLCC =</span> <span class="cn">FALSE</span>,</span>
<span id="cb115-3"><a href="network-analysis.html#cb115-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">avDissIgnoreInf =</span> <span class="cn">TRUE</span>,</span>
<span id="cb115-4"><a href="network-analysis.html#cb115-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">sPathNorm =</span> <span class="cn">FALSE</span>,</span>
<span id="cb115-5"><a href="network-analysis.html#cb115-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">clustMethod =</span> <span class="st">&quot;cluster_fast_greedy&quot;</span>,</span>
<span id="cb115-6"><a href="network-analysis.html#cb115-6" aria-hidden="true" tabindex="-1"></a>                          <span class="at">hubPar =</span> <span class="fu">c</span>(<span class="st">&quot;degree&quot;</span>, <span class="st">&quot;between&quot;</span>, <span class="st">&quot;closeness&quot;</span>),</span>
<span id="cb115-7"><a href="network-analysis.html#cb115-7" aria-hidden="true" tabindex="-1"></a>                          <span class="at">hubQuant =</span> <span class="fl">0.9</span>,</span>
<span id="cb115-8"><a href="network-analysis.html#cb115-8" aria-hidden="true" tabindex="-1"></a>                          <span class="at">lnormFit =</span> <span class="cn">TRUE</span>,</span>
<span id="cb115-9"><a href="network-analysis.html#cb115-9" aria-hidden="true" tabindex="-1"></a>                          <span class="at">normDeg =</span> <span class="cn">FALSE</span>,</span>
<span id="cb115-10"><a href="network-analysis.html#cb115-10" aria-hidden="true" tabindex="-1"></a>                          <span class="at">normBetw =</span> <span class="cn">FALSE</span>,</span>
<span id="cb115-11"><a href="network-analysis.html#cb115-11" aria-hidden="true" tabindex="-1"></a>                          <span class="at">normClose =</span> <span class="cn">FALSE</span>,</span>
<span id="cb115-12"><a href="network-analysis.html#cb115-12" aria-hidden="true" tabindex="-1"></a>                          <span class="at">normEigen =</span> <span class="cn">FALSE</span>)</span>
<span id="cb115-13"><a href="network-analysis.html#cb115-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb115-14"><a href="network-analysis.html#cb115-14" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(props_group)</span></code></pre></div>
<pre><code>## 
## Component sizes
## ```````````````
## Group 1:        
## size: 48
##    #:  1
## 
## Group 2:        
## size: 48
##    #:  1
## ______________________________
## Global network properties
## `````````````````````````
##                          group &#39;1&#39; group &#39;2&#39;
## Number of components       1.00000   1.00000
## Clustering coefficient     0.50151   0.48747
## Modularity                 0.15580   0.08807
## Positive edge percentage  51.11607  49.24078
## Edge density               0.39716   0.40869
## Natural connectivity       0.09563   0.08721
## Vertex connectivity        9.00000   7.00000
## Edge connectivity          9.00000   7.00000
## Average dissimilarity*     0.67288   0.68152
## Average path length**      0.92286   0.91581
## 
##  *Dissimilarity = 1 - edge weight
## **Path length: Sum of dissimilarities along the path
## 
## ______________________________
## Clusters
## - In the whole network
## - Algorithm: cluster_fast_greedy
## ```````````````````````````````` 
## group &#39;1&#39;:                
## name: 1  2  3  4
##    #: 6 15 14 13
## 
## group &#39;2&#39;:                
## name:  1  2  3 4
##    #: 18 10 12 8
## 
## ______________________________
## Hubs
## - In alphabetical/numerical order
## - Based on log-normal quantiles of centralities
## ```````````````````````````````````````````````
## No hubs detected.
## ______________________________
## Centrality measures
## - In decreasing order
## - Computed for the complete network
## ````````````````````````````````````
## Degree (unnormalized):
##          group &#39;1&#39; group &#39;2&#39;
##  M_48258        29        19
##  M_42398        26        25
##  M_47118        26        27
##  M_35186        25        22
##  M_55072        24        21
##             ______    ______
##  M_47118        26        27
##  M_19130        14        26
##  M_42446        18        26
##  M_33230        22        25
##  M_52452        23        25
## 
## Betweenness centrality (unnormalized):
##          group &#39;1&#39; group &#39;2&#39;
##  M_35186        51         5
##  M_42450        31        27
##  M_52452        31        18
##  M_42398        31        20
##  M_35253        31         5
##             ______    ______
##  M_19130         5        34
##  M_33230         3        32
##  M_63361         5        31
##  M_53189         7        27
##  M_42450        31        27
## 
## Closeness centrality (unnormalized):
##          group &#39;1&#39; group &#39;2&#39;
##  M_35253  71.79398  51.58148
##  M_42398   70.9484  63.74822
##  M_33961  68.32261  55.35701
##  M_48258  67.49245   61.7771
##  M_33955  66.93392  57.32211
##             ______    ______
##  M_52447  63.54214  66.40753
##  M_52464  62.24541  65.07936
##  M_33230  62.53671  63.84338
##  M_42398   70.9484  63.74822
##  M_47118   64.7125  63.10391
## 
## Eigenvector centrality (unnormalized):
##          group &#39;1&#39; group &#39;2&#39;
##  M_42398   0.26682   0.21086
##  M_35253   0.26167   0.08052
##  M_48258   0.25696    0.1782
##  M_35186   0.23664   0.20434
##  M_33961   0.23245   0.11227
##             ______    ______
##  M_33230   0.18321   0.21563
##  M_47118   0.20899   0.21435
##  M_42398   0.26682   0.21086
##  M_35186   0.23664   0.20434
##  M_52447   0.13304   0.18604</code></pre>
</div>
<div id="visualizing-the-network-3" class="section level3 hasAnchor" number="7.6.4">
<h3><span class="header-section-number">7.6.4</span> Visualizing the network<a href="network-analysis.html#visualizing-the-network-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="network-analysis.html#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(props_group, </span>
<span id="cb117-2"><a href="network-analysis.html#cb117-2" aria-hidden="true" tabindex="-1"></a>     <span class="at">sameLayout =</span> <span class="cn">TRUE</span>, </span>
<span id="cb117-3"><a href="network-analysis.html#cb117-3" aria-hidden="true" tabindex="-1"></a>     <span class="at">layoutGroup =</span> <span class="dv">1</span>,</span>
<span id="cb117-4"><a href="network-analysis.html#cb117-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">rmSingles =</span> <span class="st">&quot;inboth&quot;</span>, </span>
<span id="cb117-5"><a href="network-analysis.html#cb117-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeSize =</span> <span class="st">&quot;mclr&quot;</span>, </span>
<span id="cb117-6"><a href="network-analysis.html#cb117-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">labelScale =</span> <span class="cn">FALSE</span>,</span>
<span id="cb117-7"><a href="network-analysis.html#cb117-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexNodes =</span> <span class="dv">1</span>, </span>
<span id="cb117-8"><a href="network-analysis.html#cb117-8" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexLabels =</span> <span class="fl">1.5</span>,</span>
<span id="cb117-9"><a href="network-analysis.html#cb117-9" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexHubLabels =</span> <span class="dv">2</span>,</span>
<span id="cb117-10"><a href="network-analysis.html#cb117-10" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexTitle =</span> <span class="dv">2</span>,</span>
<span id="cb117-11"><a href="network-analysis.html#cb117-11" aria-hidden="true" tabindex="-1"></a>     <span class="at">groupNames =</span> group_names,</span>
<span id="cb117-12"><a href="network-analysis.html#cb117-12" aria-hidden="true" tabindex="-1"></a>     <span class="at">hubBorderCol  =</span> <span class="st">&quot;gray40&quot;</span>)</span>
<span id="cb117-13"><a href="network-analysis.html#cb117-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-14"><a href="network-analysis.html#cb117-14" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;bottom&quot;</span>, <span class="at">title =</span> <span class="st">&quot;estimated association:&quot;</span>, <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;+&quot;</span>, <span class="st">&quot;-&quot;</span>), </span>
<span id="cb117-15"><a href="network-analysis.html#cb117-15" aria-hidden="true" tabindex="-1"></a>       <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;#009900&quot;</span>,<span class="st">&quot;red&quot;</span>), <span class="at">inset =</span> <span class="fl">0.04</span>, <span class="at">cex =</span> <span class="dv">3</span>, <span class="at">lty =</span> <span class="dv">1</span>, <span class="at">lwd =</span> <span class="dv">4</span>, </span>
<span id="cb117-16"><a href="network-analysis.html#cb117-16" aria-hidden="true" tabindex="-1"></a>       <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">horiz =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-76-1.png" width="1920" /></p>
</div>
<div id="quantitative-network-comparison" class="section level3 hasAnchor" number="7.6.5">
<h3><span class="header-section-number">7.6.5</span> Quantitative network comparison<a href="network-analysis.html#quantitative-network-comparison" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="network-analysis.html#cb118-1" aria-hidden="true" tabindex="-1"></a>comp_group <span class="ot">&lt;-</span> <span class="fu">netCompare</span>(props_group, <span class="at">permTest =</span> <span class="cn">FALSE</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb118-2"><a href="network-analysis.html#cb118-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-3"><a href="network-analysis.html#cb118-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(comp_group, </span>
<span id="cb118-4"><a href="network-analysis.html#cb118-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">groupNames =</span> group_names,</span>
<span id="cb118-5"><a href="network-analysis.html#cb118-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">showCentr =</span> <span class="fu">c</span>(<span class="st">&quot;degree&quot;</span>, <span class="st">&quot;between&quot;</span>, <span class="st">&quot;closeness&quot;</span>), </span>
<span id="cb118-6"><a href="network-analysis.html#cb118-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">numbNodes =</span> <span class="dv">5</span>)</span></code></pre></div>
<pre><code>## 
## Comparison of Network Properties
## ----------------------------------
## CALL: 
## netCompare(x = props_group, permTest = FALSE, verbose = FALSE)
## 
## ______________________________
## Global network properties
## `````````````````````````
##                            Mild   Severe    difference
## Number of components      1.000    1.000         0.000
## Clustering coefficient    0.502    0.487         0.014
## Moduarity                 0.156    0.088         0.068
## Positive edge percentage 51.116   49.241         1.875
## Edge density              0.397    0.409         0.012
## Natural connectivity      0.096    0.087         0.008
## Vertex connectivity       9.000    7.000         2.000
## Edge connectivity         9.000    7.000         2.000
## Average dissimilarity*    0.673    0.682         0.009
## Average path length**     0.923    0.916         0.007
## -----
##  *: Dissimilarity = 1 - edge weight
## **Path length: Sum of dissimilarities along the path
## 
## ______________________________
## Jaccard index (similarity betw. sets of most central nodes)
## ``````````````````````````````````````````````````````````
##                     Jacc   P(&lt;=Jacc)     P(&gt;=Jacc)   
## degree             0.278    0.412243      0.768928   
## betweenness centr. 0.120    0.014890 *    0.996495   
## closeness centr.   0.300    0.479343      0.702786   
## eigenvec. centr.   0.444    0.892398      0.223260   
## hub taxa           0.000    1.000000      1.000000   
## -----
## Jaccard index ranges from 0 (compl. different) to 1 (sets equal)
## 
## ______________________________
## Adjusted Rand index (similarity betw. clusterings)
## ``````````````````````````````````````````````````
##    ARI       p-value
##  0.121             0
## -----
## ARI in [-1,1] with ARI=1: perfect agreement betw. clusterings,
##                    ARI=0: expected for two random clusterings
## p-value: two-tailed test with null hypothesis ARI=0
## 
## ______________________________
## Centrality measures
## - In decreasing order
## - Computed for the complete network
## ````````````````````````````````````
## Degree (unnormalized):
##         Mild Severe abs.diff.
## M_19130   14     26        12
## M_19266   21     10        11
## M_48258   29     19        10
## M_49617   10     19         9
## M_35253   24     15         9
## 
## Betweenness centrality (unnormalized):
##         Mild Severe abs.diff.
## M_35186   51      5        46
## M_19130    5     34        29
## M_33230    3     32        29
## M_63361    5     31        26
## M_53229   27      1        26
## 
## Closeness centrality (unnormalized):
##           Mild Severe abs.diff.
## M_35253 71.794 51.581    20.213
## M_33961 68.323 55.357    12.966
## M_49617 46.281 57.240    10.958
## M_53189 49.398 59.611    10.213
## M_33955 66.934 57.322     9.612
## 
## _________________________________________________________
## Significance codes: ***: 0.001, **: 0.01, *: 0.05, .: 0.1</code></pre>
</div>
</div>
<div id="systematic-information-5" class="section level2 hasAnchor" number="7.7">
<h2><span class="header-section-number">7.7</span> Systematic Information<a href="network-analysis.html#systematic-information-5" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="network-analysis.html#cb120-1" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 4.1.2 (2021-11-01)
##  os       macOS Monterey 12.2.1
##  system   x86_64, darwin17.0
##  ui       RStudio
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       Asia/Shanghai
##  date     2022-08-16
##  rstudio  2022.07.1+554 Spotted Wakerobin (desktop)
##  pandoc   2.18 @ /Applications/RStudio.app/Contents/MacOS/quarto/bin/tools/ (via rmarkdown)
## 
## ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package              * version    date (UTC) lib source
##  abind                  1.4-5      2016-07-21 [1] CRAN (R 4.1.0)
##  ade4                   1.7-18     2021-09-16 [1] CRAN (R 4.1.0)
##  ALDEx2                 1.26.0     2021-10-26 [1] Bioconductor
##  annotate               1.72.0     2021-10-26 [1] Bioconductor
##  AnnotationDbi          1.56.2     2021-11-09 [1] Bioconductor
##  ape                  * 5.6-2      2022-03-02 [1] CRAN (R 4.1.2)
##  askpass                1.1        2019-01-13 [1] CRAN (R 4.1.0)
##  assertthat             0.2.1      2019-03-21 [1] CRAN (R 4.1.0)
##  backports              1.4.1      2021-12-13 [1] CRAN (R 4.1.0)
##  base64enc              0.1-3      2015-07-28 [1] CRAN (R 4.1.0)
##  bayesm                 3.1-4      2019-10-15 [1] CRAN (R 4.1.0)
##  Biobase              * 2.54.0     2021-10-26 [1] Bioconductor
##  BiocGenerics         * 0.40.0     2021-10-26 [1] Bioconductor
##  BiocParallel           1.28.3     2021-12-09 [1] Bioconductor
##  biomformat             1.22.0     2021-10-26 [1] Bioconductor
##  Biostrings             2.62.0     2021-10-26 [1] Bioconductor
##  bit                    4.0.4      2020-08-04 [1] CRAN (R 4.1.0)
##  bit64                  4.0.5      2020-08-30 [1] CRAN (R 4.1.0)
##  bitops                 1.0-7      2021-04-24 [1] CRAN (R 4.1.0)
##  blob                   1.2.2      2021-07-23 [1] CRAN (R 4.1.0)
##  bookdown               0.27       2022-06-14 [1] CRAN (R 4.1.2)
##  brio                   1.1.3      2021-11-30 [1] CRAN (R 4.1.0)
##  broom                  0.7.12     2022-01-28 [1] CRAN (R 4.1.2)
##  bslib                  0.3.1      2021-10-06 [1] CRAN (R 4.1.0)
##  cachem                 1.0.6      2021-08-19 [1] CRAN (R 4.1.0)
##  callr                  3.7.0      2021-04-20 [1] CRAN (R 4.1.0)
##  car                    3.0-12     2021-11-06 [1] CRAN (R 4.1.0)
##  carData                3.0-5      2022-01-06 [1] CRAN (R 4.1.2)
##  caret                  6.0-92     2022-04-19 [1] CRAN (R 4.1.2)
##  caTools                1.18.2     2021-03-28 [1] CRAN (R 4.1.0)
##  cellranger             1.1.0      2016-07-27 [1] CRAN (R 4.1.0)
##  checkmate              2.0.0      2020-02-06 [1] CRAN (R 4.1.0)
##  circlize               0.4.14     2022-02-11 [1] CRAN (R 4.1.2)
##  class                  7.3-20     2022-01-13 [1] CRAN (R 4.1.2)
##  classInt               0.4-3      2020-04-07 [1] CRAN (R 4.1.0)
##  cli                    3.3.0      2022-04-25 [1] CRAN (R 4.1.2)
##  clue                   0.3-60     2021-10-11 [1] CRAN (R 4.1.0)
##  cluster              * 2.1.2      2021-04-17 [1] CRAN (R 4.1.2)
##  codetools              0.2-18     2020-11-04 [1] CRAN (R 4.1.2)
##  coin                   1.4-2      2021-10-08 [1] CRAN (R 4.1.0)
##  colorspace             2.0-3      2022-02-21 [1] CRAN (R 4.1.2)
##  ComplexHeatmap         2.10.0     2021-10-26 [1] Bioconductor
##  compositions           2.0-4      2022-01-05 [1] CRAN (R 4.1.2)
##  conflicted           * 1.1.0      2021-11-26 [1] CRAN (R 4.1.0)
##  corpcor                1.6.10     2021-09-16 [1] CRAN (R 4.1.0)
##  corrplot               0.92       2021-11-18 [1] CRAN (R 4.1.0)
##  cowplot                1.1.1      2020-12-30 [1] CRAN (R 4.1.0)
##  crayon                 1.5.0      2022-02-14 [1] CRAN (R 4.1.2)
##  crosstalk              1.2.0      2021-11-04 [1] CRAN (R 4.1.0)
##  data.table             1.14.2     2021-09-27 [1] CRAN (R 4.1.0)
##  DBI                    1.1.2      2021-12-20 [1] CRAN (R 4.1.0)
##  DelayedArray           0.20.0     2021-10-26 [1] Bioconductor
##  dendextend           * 1.15.2     2021-10-28 [1] CRAN (R 4.1.0)
##  DEoptimR               1.0-10     2022-01-03 [1] CRAN (R 4.1.2)
##  desc                   1.4.1      2022-03-06 [1] CRAN (R 4.1.2)
##  DESeq2                 1.34.0     2021-10-26 [1] Bioconductor
##  devtools             * 2.4.3      2021-11-30 [1] CRAN (R 4.1.0)
##  digest                 0.6.29     2021-12-01 [1] CRAN (R 4.1.0)
##  doParallel             1.0.17     2022-02-07 [1] CRAN (R 4.1.2)
##  doSNOW                 1.0.20     2022-02-04 [1] CRAN (R 4.1.2)
##  dplyr                * 1.0.8      2022-02-08 [1] CRAN (R 4.1.2)
##  DT                     0.21       2022-02-26 [1] CRAN (R 4.1.2)
##  dynamicTreeCut         1.63-1     2016-03-11 [1] CRAN (R 4.1.0)
##  e1071                  1.7-9      2021-09-16 [1] CRAN (R 4.1.0)
##  edgeR                  3.36.0     2021-10-26 [1] Bioconductor
##  ellipse                0.4.2      2020-05-27 [1] CRAN (R 4.1.0)
##  ellipsis               0.3.2      2021-04-29 [1] CRAN (R 4.1.0)
##  evaluate               0.15       2022-02-18 [1] CRAN (R 4.1.2)
##  factoextra           * 1.0.7      2020-04-01 [1] CRAN (R 4.1.0)
##  FactoMineR             2.4        2020-12-11 [1] CRAN (R 4.1.0)
##  fansi                  1.0.2      2022-01-14 [1] CRAN (R 4.1.2)
##  farver                 2.1.0      2021-02-28 [1] CRAN (R 4.1.0)
##  fastcluster            1.2.3      2021-05-24 [1] CRAN (R 4.1.0)
##  fastmap                1.1.0      2021-01-25 [1] CRAN (R 4.1.0)
##  fdrtool                1.2.17     2021-11-13 [1] CRAN (R 4.1.0)
##  filematrix             1.3        2018-02-27 [1] CRAN (R 4.1.0)
##  flashClust             1.01-2     2012-08-21 [1] CRAN (R 4.1.0)
##  foreach                1.5.2      2022-02-02 [1] CRAN (R 4.1.2)
##  foreign                0.8-82     2022-01-13 [1] CRAN (R 4.1.2)
##  forestplot             2.0.1      2021-09-03 [1] CRAN (R 4.1.0)
##  formatR                1.11       2021-06-01 [1] CRAN (R 4.1.0)
##  Formula                1.2-4      2020-10-16 [1] CRAN (R 4.1.0)
##  fs                     1.5.2      2021-12-08 [1] CRAN (R 4.1.0)
##  futile.logger        * 1.4.3      2016-07-10 [1] CRAN (R 4.1.0)
##  futile.options         1.0.1      2018-04-20 [1] CRAN (R 4.1.0)
##  future                 1.24.0     2022-02-19 [1] CRAN (R 4.1.2)
##  future.apply           1.8.1      2021-08-10 [1] CRAN (R 4.1.0)
##  genefilter             1.76.0     2021-10-26 [1] Bioconductor
##  geneplotter            1.72.0     2021-10-26 [1] Bioconductor
##  generics               0.1.2      2022-01-31 [1] CRAN (R 4.1.2)
##  GenomeInfoDb         * 1.30.1     2022-01-30 [1] Bioconductor
##  GenomeInfoDbData       1.2.7      2022-03-09 [1] Bioconductor
##  GenomicRanges        * 1.46.1     2021-11-18 [1] Bioconductor
##  GetoptLong             1.0.5      2020-12-15 [1] CRAN (R 4.1.0)
##  ggforce                0.3.3      2021-03-05 [1] CRAN (R 4.1.0)
##  ggiraph                0.8.2      2022-02-22 [1] CRAN (R 4.1.2)
##  ggiraphExtra           0.3.0      2020-10-06 [1] CRAN (R 4.1.2)
##  ggplot2              * 3.3.5      2021-06-25 [1] CRAN (R 4.1.0)
##  ggpubr               * 0.4.0      2020-06-27 [1] CRAN (R 4.1.0)
##  ggraph               * 2.0.5      2021-02-23 [1] CRAN (R 4.1.0)
##  ggrepel                0.9.1      2021-01-15 [1] CRAN (R 4.1.0)
##  ggsci                  2.9        2018-05-14 [1] CRAN (R 4.1.0)
##  ggsignif               0.6.3      2021-09-09 [1] CRAN (R 4.1.0)
##  ggVennDiagram          1.2.1      2022-04-13 [1] Github (gaospecial/ggVennDiagram@db6742d)
##  glasso                 1.11       2019-10-01 [1] CRAN (R 4.1.0)
##  glmnet               * 4.1-3      2021-11-02 [1] CRAN (R 4.1.0)
##  GlobalOptions          0.1.2      2020-06-10 [1] CRAN (R 4.1.0)
##  globals                0.14.0     2020-11-22 [1] CRAN (R 4.1.0)
##  glue                 * 1.6.2      2022-02-24 [1] CRAN (R 4.1.2)
##  Gmisc                * 3.0.0      2022-01-03 [1] CRAN (R 4.1.2)
##  gmp                    0.6-5      2022-03-17 [1] CRAN (R 4.1.2)
##  GO.db                  3.14.0     2022-04-11 [1] Bioconductor
##  gower                  1.0.0      2022-02-03 [1] CRAN (R 4.1.2)
##  gplots                 3.1.1      2020-11-28 [1] CRAN (R 4.1.0)
##  graphlayouts           0.8.0      2022-01-03 [1] CRAN (R 4.1.2)
##  gridExtra              2.3        2017-09-09 [1] CRAN (R 4.1.0)
##  gtable                 0.3.0      2019-03-25 [1] CRAN (R 4.1.0)
##  gtools                 3.9.2      2021-06-06 [1] CRAN (R 4.1.0)
##  hardhat                0.2.0      2022-01-24 [1] CRAN (R 4.1.2)
##  highr                  0.9        2021-04-16 [1] CRAN (R 4.1.0)
##  Hmisc                  4.6-0      2021-10-07 [1] CRAN (R 4.1.0)
##  hms                    1.1.1      2021-09-26 [1] CRAN (R 4.1.0)
##  htmlTable            * 2.4.0      2022-01-04 [1] CRAN (R 4.1.2)
##  htmltools              0.5.2      2021-08-25 [1] CRAN (R 4.1.0)
##  htmlwidgets            1.5.4      2021-09-08 [1] CRAN (R 4.1.0)
##  httr                   1.4.2      2020-07-20 [1] CRAN (R 4.1.0)
##  huge                   1.3.5      2021-06-30 [1] CRAN (R 4.1.0)
##  igraph                 1.2.11     2022-01-04 [1] CRAN (R 4.1.2)
##  impute                 1.68.0     2021-10-26 [1] Bioconductor
##  insight                0.17.0     2022-03-29 [1] CRAN (R 4.1.2)
##  ipred                  0.9-12     2021-09-15 [1] CRAN (R 4.1.0)
##  IRanges              * 2.28.0     2021-10-26 [1] Bioconductor
##  irlba                  2.3.5      2021-12-06 [1] CRAN (R 4.1.0)
##  iterators              1.0.14     2022-02-05 [1] CRAN (R 4.1.2)
##  jpeg                   0.1-9      2021-07-24 [1] CRAN (R 4.1.0)
##  jquerylib              0.1.4      2021-04-26 [1] CRAN (R 4.1.0)
##  jsonlite               1.8.0      2022-02-22 [1] CRAN (R 4.1.2)
##  kableExtra             1.3.4      2021-02-20 [1] CRAN (R 4.1.2)
##  KEGGREST               1.34.0     2021-10-26 [1] Bioconductor
##  KernSmooth             2.23-20    2021-05-03 [1] CRAN (R 4.1.2)
##  knitr                  1.39       2022-04-26 [1] CRAN (R 4.1.2)
##  labeling               0.4.2      2020-10-20 [1] CRAN (R 4.1.0)
##  lambda.r               1.2.4      2019-09-18 [1] CRAN (R 4.1.0)
##  lattice              * 0.20-45    2021-09-22 [1] CRAN (R 4.1.2)
##  latticeExtra           0.6-29     2019-12-19 [1] CRAN (R 4.1.0)
##  lava                   1.6.10     2021-09-02 [1] CRAN (R 4.1.0)
##  lavaan                 0.6-11     2022-03-31 [1] CRAN (R 4.1.2)
##  lazyeval               0.2.2      2019-03-15 [1] CRAN (R 4.1.0)
##  leaps                  3.1        2020-01-16 [1] CRAN (R 4.1.0)
##  libcoin                1.0-9      2021-09-27 [1] CRAN (R 4.1.0)
##  lifecycle              1.0.1      2021-09-24 [1] CRAN (R 4.1.0)
##  limma                  3.50.1     2022-02-17 [1] Bioconductor
##  listenv                0.8.0      2019-12-05 [1] CRAN (R 4.1.0)
##  locfit                 1.5-9.5    2022-03-03 [1] CRAN (R 4.1.2)
##  LOCOM                  1.1        2022-08-05 [1] Github (yijuanhu/LOCOM@c181e0f)
##  lubridate              1.8.0      2021-10-07 [1] CRAN (R 4.1.0)
##  magrittr             * 2.0.2      2022-01-26 [1] CRAN (R 4.1.2)
##  MASS                   7.3-55     2022-01-13 [1] CRAN (R 4.1.2)
##  Matrix               * 1.4-0      2021-12-08 [1] CRAN (R 4.1.0)
##  MatrixGenerics       * 1.6.0      2021-10-26 [1] Bioconductor
##  matrixStats          * 0.61.0     2021-09-17 [1] CRAN (R 4.1.0)
##  mbzinb                 0.2        2022-03-16 [1] local
##  memoise                2.0.1      2021-11-26 [1] CRAN (R 4.1.0)
##  metagenomeSeq          1.36.0     2021-10-26 [1] Bioconductor
##  mgcv                   1.8-39     2022-02-24 [1] CRAN (R 4.1.2)
##  microbiome             1.16.0     2021-10-26 [1] Bioconductor
##  mixedCCA               1.5.2      2022-07-14 [1] Github (irinagain/mixedCCA@c6d41a3)
##  mixOmics               6.18.1     2021-11-18 [1] Bioconductor (R 4.1.2)
##  mnormt                 2.0.2      2020-09-01 [1] CRAN (R 4.1.0)
##  ModelMetrics           1.2.2.2    2020-03-17 [1] CRAN (R 4.1.0)
##  modeltools             0.2-23     2020-03-05 [1] CRAN (R 4.1.0)
##  multcomp               1.4-18     2022-01-04 [1] CRAN (R 4.1.2)
##  multtest               2.50.0     2021-10-26 [1] Bioconductor
##  munsell                0.5.0      2018-06-12 [1] CRAN (R 4.1.0)
##  mvtnorm                1.1-3      2021-10-08 [1] CRAN (R 4.1.0)
##  mycor                  0.1.1      2018-04-10 [1] CRAN (R 4.1.0)
##  NADA                   1.6-1.1    2020-03-22 [1] CRAN (R 4.1.0)
##  NetCoMi              * 1.0.3      2022-07-14 [1] Github (stefpeschel/NetCoMi@d4d80d3)
##  nlme                 * 3.1-155    2022-01-13 [1] CRAN (R 4.1.2)
##  nnet                   7.3-17     2022-01-13 [1] CRAN (R 4.1.2)
##  openssl                2.0.0      2022-03-02 [1] CRAN (R 4.1.2)
##  parallelly             1.31.0     2022-04-07 [1] CRAN (R 4.1.2)
##  pbapply                1.5-0      2021-09-16 [1] CRAN (R 4.1.0)
##  pbivnorm               0.6.0      2015-01-23 [1] CRAN (R 4.1.0)
##  pcaPP                  1.9-74     2021-04-23 [1] CRAN (R 4.1.0)
##  permute              * 0.9-7      2022-01-27 [1] CRAN (R 4.1.2)
##  pheatmap             * 1.0.12     2019-01-04 [1] CRAN (R 4.1.0)
##  phyloseq             * 1.38.0     2021-10-26 [1] Bioconductor
##  picante              * 1.8.2      2020-06-10 [1] CRAN (R 4.1.0)
##  pillar                 1.7.0      2022-02-01 [1] CRAN (R 4.1.2)
##  pkgbuild               1.3.1      2021-12-20 [1] CRAN (R 4.1.0)
##  pkgconfig              2.0.3      2019-09-22 [1] CRAN (R 4.1.0)
##  pkgload                1.2.4      2021-11-30 [1] CRAN (R 4.1.0)
##  plotly               * 4.10.0     2021-10-09 [1] CRAN (R 4.1.0)
##  plyr                   1.8.6      2020-03-03 [1] CRAN (R 4.1.0)
##  png                    0.1-7      2013-12-03 [1] CRAN (R 4.1.0)
##  polyclip               1.10-0     2019-03-14 [1] CRAN (R 4.1.0)
##  POMA                 * 1.7.2      2022-07-26 [1] Github (pcastellanoescuder/POMA@bc8a972)
##  ppcor                  1.1        2015-12-03 [1] CRAN (R 4.1.0)
##  preprocessCore         1.56.0     2021-10-26 [1] Bioconductor
##  prettyunits            1.1.1      2020-01-24 [1] CRAN (R 4.1.0)
##  pROC                   1.18.0     2021-09-03 [1] CRAN (R 4.1.0)
##  processx               3.5.2      2021-04-30 [1] CRAN (R 4.1.0)
##  prodlim                2019.11.13 2019-11-17 [1] CRAN (R 4.1.0)
##  protoclust             1.6.3      2019-01-31 [1] CRAN (R 4.1.0)
##  proxy                  0.4-26     2021-06-07 [1] CRAN (R 4.1.0)
##  ps                     1.6.0      2021-02-28 [1] CRAN (R 4.1.0)
##  pscl                   1.5.5      2020-03-07 [1] CRAN (R 4.1.0)
##  psych                  2.2.5      2022-05-10 [1] CRAN (R 4.1.2)
##  pulsar                 0.3.7      2020-08-07 [1] CRAN (R 4.1.0)
##  purrr                * 0.3.4      2020-04-17 [1] CRAN (R 4.1.0)
##  qgraph                 1.9.2      2022-03-04 [1] CRAN (R 4.1.2)
##  qvalue                 2.26.0     2021-10-26 [1] Bioconductor
##  R6                     2.5.1      2021-08-19 [1] CRAN (R 4.1.0)
##  RAIDA                  1.0        2022-03-14 [1] local
##  randomForest           4.7-1      2022-02-03 [1] CRAN (R 4.1.2)
##  RankProd               3.20.0     2021-10-26 [1] Bioconductor
##  rARPACK                0.11-0     2016-03-10 [1] CRAN (R 4.1.0)
##  rbibutils              2.2.7      2021-12-07 [1] CRAN (R 4.1.0)
##  RColorBrewer         * 1.1-2      2014-12-07 [1] CRAN (R 4.1.0)
##  Rcpp                 * 1.0.8.2    2022-03-11 [1] CRAN (R 4.1.2)
##  RcppZiggurat           0.1.6      2020-10-20 [1] CRAN (R 4.1.0)
##  RCurl                  1.98-1.6   2022-02-08 [1] CRAN (R 4.1.2)
##  Rdpack                 2.2        2022-03-19 [1] CRAN (R 4.1.2)
##  readr                * 2.1.2      2022-01-30 [1] CRAN (R 4.1.2)
##  readxl               * 1.4.0      2022-03-28 [1] CRAN (R 4.1.2)
##  recipes                0.2.0      2022-02-18 [1] CRAN (R 4.1.2)
##  remotes                2.4.2      2021-11-30 [1] CRAN (R 4.1.0)
##  reshape2               1.4.4      2020-04-09 [1] CRAN (R 4.1.0)
##  reticulate             1.24       2022-01-26 [1] CRAN (R 4.1.2)
##  Rfast                  2.0.6      2022-02-16 [1] CRAN (R 4.1.2)
##  rhdf5                  2.38.1     2022-03-10 [1] Bioconductor
##  rhdf5filters           1.6.0      2021-10-26 [1] Bioconductor
##  Rhdf5lib               1.16.0     2021-10-26 [1] Bioconductor
##  rjson                  0.2.21     2022-01-09 [1] CRAN (R 4.1.2)
##  rlang                  1.0.2      2022-03-04 [1] CRAN (R 4.1.2)
##  rmarkdown              2.14       2022-04-25 [1] CRAN (R 4.1.2)
##  Rmpfr                  0.8-9      2022-06-03 [1] CRAN (R 4.1.2)
##  robustbase             0.93-9     2021-09-27 [1] CRAN (R 4.1.0)
##  rootSolve              1.8.2.3    2021-09-29 [1] CRAN (R 4.1.0)
##  rpart                  4.1.16     2022-01-24 [1] CRAN (R 4.1.2)
##  rprojroot              2.0.2      2020-11-15 [1] CRAN (R 4.1.0)
##  RSpectra               0.16-0     2019-12-01 [1] CRAN (R 4.1.0)
##  RSQLite                2.2.10     2022-02-17 [1] CRAN (R 4.1.2)
##  rstatix                0.7.0      2021-02-13 [1] CRAN (R 4.1.0)
##  rstudioapi             0.13       2020-11-12 [1] CRAN (R 4.1.0)
##  Rtsne                  0.15       2018-11-10 [1] CRAN (R 4.1.0)
##  RVenn                  1.1.0      2019-07-18 [1] CRAN (R 4.1.0)
##  rvest                  1.0.2      2021-10-16 [1] CRAN (R 4.1.0)
##  S4Vectors            * 0.32.3     2021-11-21 [1] Bioconductor
##  sandwich               3.0-1      2021-05-18 [1] CRAN (R 4.1.0)
##  sass                   0.4.0      2021-05-12 [1] CRAN (R 4.1.0)
##  scales                 1.1.1      2020-05-11 [1] CRAN (R 4.1.0)
##  scatterplot3d          0.3-41     2018-03-14 [1] CRAN (R 4.1.0)
##  sessioninfo            1.2.2      2021-12-06 [1] CRAN (R 4.1.0)
##  sf                     1.0-7      2022-03-07 [1] CRAN (R 4.1.2)
##  shape                  1.4.6      2021-05-19 [1] CRAN (R 4.1.0)
##  sjlabelled             1.2.0      2022-04-10 [1] CRAN (R 4.1.2)
##  sjmisc                 2.8.9      2021-12-03 [1] CRAN (R 4.1.0)
##  snow                   0.4-4      2021-10-27 [1] CRAN (R 4.1.0)
##  SpiecEasi            * 1.1.2      2022-07-14 [1] Github (zdk123/SpiecEasi@c463727)
##  SPRING               * 1.0.4      2022-08-03 [1] Github (GraceYoon/SPRING@3d641a4)
##  stringi                1.7.6      2021-11-29 [1] CRAN (R 4.1.0)
##  stringr                1.4.0      2019-02-10 [1] CRAN (R 4.1.0)
##  SummarizedExperiment * 1.24.0     2021-10-26 [1] Bioconductor
##  survival               3.3-1      2022-03-03 [1] CRAN (R 4.1.2)
##  svglite                2.1.0      2022-02-03 [1] CRAN (R 4.1.2)
##  systemfonts            1.0.4      2022-02-11 [1] CRAN (R 4.1.2)
##  tensorA                0.36.2     2020-11-19 [1] CRAN (R 4.1.0)
##  testthat               3.1.2      2022-01-20 [1] CRAN (R 4.1.2)
##  TH.data                1.1-0      2021-09-27 [1] CRAN (R 4.1.0)
##  tibble               * 3.1.6      2021-11-07 [1] CRAN (R 4.1.0)
##  tidygraph              1.2.1      2022-04-05 [1] CRAN (R 4.1.2)
##  tidyr                * 1.2.0      2022-02-01 [1] CRAN (R 4.1.2)
##  tidyselect             1.1.2      2022-02-21 [1] CRAN (R 4.1.2)
##  timeDate               3043.102   2018-02-21 [1] CRAN (R 4.1.0)
##  tmvnsim                1.0-2      2016-12-15 [1] CRAN (R 4.1.0)
##  truncnorm              1.0-8      2018-02-27 [1] CRAN (R 4.1.0)
##  tweenr                 1.0.2      2021-03-23 [1] CRAN (R 4.1.0)
##  tzdb                   0.2.0      2021-10-27 [1] CRAN (R 4.1.0)
##  umap                   0.2.8.0    2022-03-23 [1] CRAN (R 4.1.2)
##  units                  0.8-0      2022-02-05 [1] CRAN (R 4.1.2)
##  usethis              * 2.1.5      2021-12-09 [1] CRAN (R 4.1.0)
##  utf8                   1.2.2      2021-07-24 [1] CRAN (R 4.1.0)
##  uuid                   1.0-3      2021-11-01 [1] CRAN (R 4.1.0)
##  vctrs                  0.3.8      2021-04-29 [1] CRAN (R 4.1.0)
##  vegan                * 2.5-7      2020-11-28 [1] CRAN (R 4.1.0)
##  VennDiagram          * 1.7.3      2022-04-12 [1] CRAN (R 4.1.2)
##  VGAM                   1.1-6      2022-02-14 [1] CRAN (R 4.1.2)
##  viridis              * 0.6.2      2021-10-13 [1] CRAN (R 4.1.0)
##  viridisLite          * 0.4.0      2021-04-13 [1] CRAN (R 4.1.0)
##  vroom                  1.5.7      2021-11-30 [1] CRAN (R 4.1.0)
##  webshot                0.5.3      2022-04-14 [1] CRAN (R 4.1.2)
##  WGCNA                  1.71       2022-04-22 [1] CRAN (R 4.1.2)
##  withr                  2.5.0      2022-03-03 [1] CRAN (R 4.1.2)
##  Wrench                 1.12.0     2021-10-26 [1] Bioconductor
##  xfun                   0.30       2022-03-02 [1] CRAN (R 4.1.2)
##  XMAS2                * 2.1.7.5    2022-08-16 [1] local
##  XML                    3.99-0.9   2022-02-24 [1] CRAN (R 4.1.2)
##  xml2                   1.3.3      2021-11-30 [1] CRAN (R 4.1.0)
##  xtable                 1.8-4      2019-04-21 [1] CRAN (R 4.1.0)
##  XVector                0.34.0     2021-10-26 [1] Bioconductor
##  yaml                   2.3.5      2022-02-21 [1] CRAN (R 4.1.2)
##  zCompositions          1.4.0      2022-01-13 [1] CRAN (R 4.1.2)
##  zlibbioc               1.40.0     2021-10-26 [1] Bioconductor
##  zoo                    1.8-9      2021-03-09 [1] CRAN (R 4.1.0)
## 
##  [1] /Library/Frameworks/R.framework/Versions/4.1/Resources/library
## 
## ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-peschel2021netcomi" class="csl-entry">
Peschel, Stefanie, Christian L Müller, Erika von Mutius, Anne-Laure Boulesteix, and Martin Depner. 2021. <span>“NetCoMi: Network Construction and Comparison for Microbiome Data in r.”</span> <em>Briefings in Bioinformatics</em> 22 (4): bbaa290.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="multivariate-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="enrichment-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/Metabolomics_Aanlysis/blob/main/06-Network-Analysis.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection",
"scroll_highlight": true,
"toc_depth": 3
}
});
});
</script>

</body>

</html>
