<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Pathway Analysis | Data Analysis in Metabolomics</title>
  <meta name="description" content="A template for Data Analysis in Metabolomics" />
  <meta name="generator" content="bookdown 0.27 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Pathway Analysis | Data Analysis in Metabolomics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A template for Data Analysis in Metabolomics" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Pathway Analysis | Data Analysis in Metabolomics" />
  
  <meta name="twitter:description" content="A template for Data Analysis in Metabolomics" />
  

<meta name="author" content="Hua Zou and Bangzhuo Tong" />


<meta name="date" content="2022-08-11" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="enrichment-analysis.html"/>
<link rel="next" href="example.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="assets/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Data Analysis in Metabolomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#what-is-metabolomics"><i class="fa fa-check"></i><b>1.1</b> What is metabolomics?</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#what-are-the-procedures-in-this-template"><i class="fa fa-check"></i><b>1.2</b> What are the procedures in this template?</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#statistical-analysis"><i class="fa fa-check"></i><b>1.2.1</b> Statistical Analysis</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#functional-analysis"><i class="fa fa-check"></i><b>1.2.2</b> Functional Analysis</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#software"><i class="fa fa-check"></i><b>1.3</b> Software</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#reference"><i class="fa fa-check"></i><b>1.4</b> Reference</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-processing.html"><a href="data-processing.html"><i class="fa fa-check"></i><b>2</b> Data Processing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-processing.html"><a href="data-processing.html#loading-packages"><i class="fa fa-check"></i><b>2.1</b> Loading packages</a></li>
<li class="chapter" data-level="2.2" data-path="data-processing.html"><a href="data-processing.html#importing-data"><i class="fa fa-check"></i><b>2.2</b> Importing data</a></li>
<li class="chapter" data-level="2.3" data-path="data-processing.html"><a href="data-processing.html#data-checking"><i class="fa fa-check"></i><b>2.3</b> Data Checking</a></li>
<li class="chapter" data-level="2.4" data-path="data-processing.html"><a href="data-processing.html#data-filtering"><i class="fa fa-check"></i><b>2.4</b> Data Filtering</a></li>
<li class="chapter" data-level="2.5" data-path="data-processing.html"><a href="data-processing.html#missing-value-imputation"><i class="fa fa-check"></i><b>2.5</b> Missing Value Imputation</a></li>
<li class="chapter" data-level="2.6" data-path="data-processing.html"><a href="data-processing.html#data-normalization"><i class="fa fa-check"></i><b>2.6</b> Data Normalization</a>
<ul>
<li class="chapter" data-level="2.6.1" data-path="data-processing.html"><a href="data-processing.html#normalization-by-normalizedata-function"><i class="fa fa-check"></i><b>2.6.1</b> Normalization by <strong>NormalizeData</strong> function</a></li>
<li class="chapter" data-level="2.6.2" data-path="data-processing.html"><a href="data-processing.html#normalization-by-poma-r-package"><i class="fa fa-check"></i><b>2.6.2</b> Normalization by <strong>POMA</strong> R package</a></li>
<li class="chapter" data-level="2.6.3" data-path="data-processing.html"><a href="data-processing.html#comparison-of-unnormalized-and-normalized-dataset"><i class="fa fa-check"></i><b>2.6.3</b> Comparison of unnormalized and normalized dataset</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="data-processing.html"><a href="data-processing.html#removing-outliers"><i class="fa fa-check"></i><b>2.7</b> Removing outliers</a></li>
<li class="chapter" data-level="2.8" data-path="data-processing.html"><a href="data-processing.html#saving-datasets-into-rds-files"><i class="fa fa-check"></i><b>2.8</b> Saving datasets into RDS files</a></li>
<li class="chapter" data-level="2.9" data-path="data-processing.html"><a href="data-processing.html#systematic-information"><i class="fa fa-check"></i><b>2.9</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="cluster-analysis.html"><a href="cluster-analysis.html"><i class="fa fa-check"></i><b>3</b> Cluster Analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="cluster-analysis.html"><a href="cluster-analysis.html#loading-packages-1"><i class="fa fa-check"></i><b>3.1</b> Loading packages</a></li>
<li class="chapter" data-level="3.2" data-path="cluster-analysis.html"><a href="cluster-analysis.html#importing-data-1"><i class="fa fa-check"></i><b>3.2</b> Importing data</a></li>
<li class="chapter" data-level="3.3" data-path="cluster-analysis.html"><a href="cluster-analysis.html#hierarchical-clustering"><i class="fa fa-check"></i><b>3.3</b> Hierarchical Clustering</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="cluster-analysis.html"><a href="cluster-analysis.html#agglomerative-hierarchical-clustering"><i class="fa fa-check"></i><b>3.3.1</b> Agglomerative Hierarchical Clustering</a></li>
<li class="chapter" data-level="3.3.2" data-path="cluster-analysis.html"><a href="cluster-analysis.html#divisive-hierarchical-clustering"><i class="fa fa-check"></i><b>3.3.2</b> Divisive Hierarchical Clustering</a></li>
<li class="chapter" data-level="3.3.3" data-path="cluster-analysis.html"><a href="cluster-analysis.html#comparison"><i class="fa fa-check"></i><b>3.3.3</b> Comparison</a></li>
<li class="chapter" data-level="3.3.4" data-path="cluster-analysis.html"><a href="cluster-analysis.html#determining-optimal-clusters"><i class="fa fa-check"></i><b>3.3.4</b> Determining Optimal Clusters</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="cluster-analysis.html"><a href="cluster-analysis.html#partitional-clustering"><i class="fa fa-check"></i><b>3.4</b> Partitional Clustering</a></li>
<li class="chapter" data-level="3.5" data-path="cluster-analysis.html"><a href="cluster-analysis.html#systematic-information-1"><i class="fa fa-check"></i><b>3.5</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="chemometrics-analysis.html"><a href="chemometrics-analysis.html"><i class="fa fa-check"></i><b>4</b> Chemometrics Analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="chemometrics-analysis.html"><a href="chemometrics-analysis.html#loading-packages-2"><i class="fa fa-check"></i><b>4.1</b> Loading packages</a></li>
<li class="chapter" data-level="4.2" data-path="chemometrics-analysis.html"><a href="chemometrics-analysis.html#importing-data-2"><i class="fa fa-check"></i><b>4.2</b> Importing data</a></li>
<li class="chapter" data-level="4.3" data-path="chemometrics-analysis.html"><a href="chemometrics-analysis.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>4.3</b> Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="4.4" data-path="chemometrics-analysis.html"><a href="chemometrics-analysis.html#partial-least-squares-discriminant-analysis-pls-da"><i class="fa fa-check"></i><b>4.4</b> Partial Least Squares-Discriminant Analysis (PLS-DA)</a></li>
<li class="chapter" data-level="4.5" data-path="chemometrics-analysis.html"><a href="chemometrics-analysis.html#sparse-partial-least-squares-discriminant-analysis-spls-da"><i class="fa fa-check"></i><b>4.5</b> Sparse Partial Least Squares-Discriminant Analysis (sPLS-DA)</a></li>
<li class="chapter" data-level="4.6" data-path="chemometrics-analysis.html"><a href="chemometrics-analysis.html#orthogonal-partial-least-squares-discriminant-analysis-orthopls-da"><i class="fa fa-check"></i><b>4.6</b> Orthogonal Partial Least Squares-Discriminant Analysis (orthoPLS-DA)</a></li>
<li class="chapter" data-level="4.7" data-path="chemometrics-analysis.html"><a href="chemometrics-analysis.html#systematic-information-2"><i class="fa fa-check"></i><b>4.7</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="univariate-analysis.html"><a href="univariate-analysis.html"><i class="fa fa-check"></i><b>5</b> Univariate Analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="univariate-analysis.html"><a href="univariate-analysis.html#loading-packages-3"><i class="fa fa-check"></i><b>5.1</b> Loading packages</a></li>
<li class="chapter" data-level="5.2" data-path="univariate-analysis.html"><a href="univariate-analysis.html#importing-data-3"><i class="fa fa-check"></i><b>5.2</b> Importing data</a></li>
<li class="chapter" data-level="5.3" data-path="univariate-analysis.html"><a href="univariate-analysis.html#fold-change-analysis"><i class="fa fa-check"></i><b>5.3</b> Fold Change Analysis</a></li>
<li class="chapter" data-level="5.4" data-path="univariate-analysis.html"><a href="univariate-analysis.html#t-test"><i class="fa fa-check"></i><b>5.4</b> T Test</a></li>
<li class="chapter" data-level="5.5" data-path="univariate-analysis.html"><a href="univariate-analysis.html#wilcoxon-test"><i class="fa fa-check"></i><b>5.5</b> Wilcoxon Test</a></li>
<li class="chapter" data-level="5.6" data-path="univariate-analysis.html"><a href="univariate-analysis.html#limma-test"><i class="fa fa-check"></i><b>5.6</b> Limma Test</a></li>
<li class="chapter" data-level="5.7" data-path="univariate-analysis.html"><a href="univariate-analysis.html#volcano-plot"><i class="fa fa-check"></i><b>5.7</b> Volcano plot</a></li>
<li class="chapter" data-level="5.8" data-path="univariate-analysis.html"><a href="univariate-analysis.html#correlation-heatmaps"><i class="fa fa-check"></i><b>5.8</b> Correlation Heatmaps</a></li>
<li class="chapter" data-level="5.9" data-path="univariate-analysis.html"><a href="univariate-analysis.html#glasso-this-function-will-compute-a-gaussian-graphical-model-using-the-glmnet-package"><i class="fa fa-check"></i><b>5.9</b> glasso: this function will compute a Gaussian Graphical Model using the glmnet package</a></li>
<li class="chapter" data-level="5.10" data-path="univariate-analysis.html"><a href="univariate-analysis.html#systematic-information-3"><i class="fa fa-check"></i><b>5.10</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html"><i class="fa fa-check"></i><b>6</b> Multivariate analysis</a>
<ul>
<li class="chapter" data-level="6.1" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#loading-packages-4"><i class="fa fa-check"></i><b>6.1</b> Loading packages</a></li>
<li class="chapter" data-level="6.2" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#importing-data-4"><i class="fa fa-check"></i><b>6.2</b> Importing data</a></li>
<li class="chapter" data-level="6.3" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#data-curation"><i class="fa fa-check"></i><b>6.3</b> Data curation</a></li>
<li class="chapter" data-level="6.4" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#regularized-generalized-linear-models"><i class="fa fa-check"></i><b>6.4</b> Regularized Generalized Linear Models</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#lasso-alpha-1"><i class="fa fa-check"></i><b>6.4.1</b> Lasso: alpha = 1</a></li>
<li class="chapter" data-level="6.4.2" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#ridge-alpha-0"><i class="fa fa-check"></i><b>6.4.2</b> Ridge: alpha = 0</a></li>
<li class="chapter" data-level="6.4.3" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#elasticnet-0-alpha-1"><i class="fa fa-check"></i><b>6.4.3</b> Elasticnet: 0 &lt; alpha &lt; 1</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#classification"><i class="fa fa-check"></i><b>6.5</b> Classification</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#random-forest"><i class="fa fa-check"></i><b>6.5.1</b> Random Forest</a></li>
<li class="chapter" data-level="6.5.2" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#support-vector-machine-svm"><i class="fa fa-check"></i><b>6.5.2</b> Support Vector Machine (SVM)</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#systematic-information-4"><i class="fa fa-check"></i><b>6.6</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="network-analysis.html"><a href="network-analysis.html"><i class="fa fa-check"></i><b>7</b> Network Analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="network-analysis.html"><a href="network-analysis.html#introduction-1"><i class="fa fa-check"></i><b>7.1</b> Introduction</a></li>
<li class="chapter" data-level="7.2" data-path="network-analysis.html"><a href="network-analysis.html#loading-packages-5"><i class="fa fa-check"></i><b>7.2</b> Loading packages</a></li>
<li class="chapter" data-level="7.3" data-path="network-analysis.html"><a href="network-analysis.html#importing-data-5"><i class="fa fa-check"></i><b>7.3</b> Importing data</a></li>
<li class="chapter" data-level="7.4" data-path="network-analysis.html"><a href="network-analysis.html#data-curation-1"><i class="fa fa-check"></i><b>7.4</b> Data curation</a></li>
<li class="chapter" data-level="7.5" data-path="network-analysis.html"><a href="network-analysis.html#associations-among-features"><i class="fa fa-check"></i><b>7.5</b> Associations Among Features</a>
<ul>
<li class="chapter" data-level="7.5.1" data-path="network-analysis.html"><a href="network-analysis.html#single-network-with-pearson-correlation-as-association-measure"><i class="fa fa-check"></i><b>7.5.1</b> Single network with Pearson correlation as association measure</a></li>
<li class="chapter" data-level="7.5.2" data-path="network-analysis.html"><a href="network-analysis.html#single-network-with-spearman-correlation-as-association-measure"><i class="fa fa-check"></i><b>7.5.2</b> Single network with spearman correlation as association measure</a></li>
<li class="chapter" data-level="7.5.3" data-path="network-analysis.html"><a href="network-analysis.html#single-network-with-wgcna-bicor-as-association-measure"><i class="fa fa-check"></i><b>7.5.3</b> Single network with WGCNA (bicor) as association measure</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="network-analysis.html"><a href="network-analysis.html#network-comparison"><i class="fa fa-check"></i><b>7.6</b> Network comparison</a>
<ul>
<li class="chapter" data-level="7.6.1" data-path="network-analysis.html"><a href="network-analysis.html#data-preparing"><i class="fa fa-check"></i><b>7.6.1</b> Data preparing</a></li>
<li class="chapter" data-level="7.6.2" data-path="network-analysis.html"><a href="network-analysis.html#building-network-model-3"><i class="fa fa-check"></i><b>7.6.2</b> Building network model</a></li>
<li class="chapter" data-level="7.6.3" data-path="network-analysis.html"><a href="network-analysis.html#network-analysis-1"><i class="fa fa-check"></i><b>7.6.3</b> Network analysis</a></li>
<li class="chapter" data-level="7.6.4" data-path="network-analysis.html"><a href="network-analysis.html#visualizing-the-network-3"><i class="fa fa-check"></i><b>7.6.4</b> Visualizing the network</a></li>
<li class="chapter" data-level="7.6.5" data-path="network-analysis.html"><a href="network-analysis.html#quantitative-network-comparison"><i class="fa fa-check"></i><b>7.6.5</b> Quantitative network comparison</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="network-analysis.html"><a href="network-analysis.html#systematic-information-5"><i class="fa fa-check"></i><b>7.7</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html"><i class="fa fa-check"></i><b>8</b> Enrichment Analysis</a>
<ul>
<li class="chapter" data-level="8.1" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#enviroment-set-up"><i class="fa fa-check"></i><b>8.1</b> Enviroment Set up</a></li>
<li class="chapter" data-level="8.2" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#over-representation-analysis-ora"><i class="fa fa-check"></i><b>8.2</b> Over representation analysis (ORA)</a></li>
<li class="chapter" data-level="8.3" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#clean-environment-1"><i class="fa fa-check"></i><b>8.3</b> Clean environment 1</a></li>
<li class="chapter" data-level="8.4" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#single-sample-profiling"><i class="fa fa-check"></i><b>8.4</b> Single Sample Profiling</a></li>
<li class="chapter" data-level="8.5" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#clean-environment-2"><i class="fa fa-check"></i><b>8.5</b> Clean environment 2</a></li>
<li class="chapter" data-level="8.6" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#quantitative-enrichment-analysis-qea"><i class="fa fa-check"></i><b>8.6</b> Quantitative Enrichment Analysis (QEA)</a></li>
<li class="chapter" data-level="8.7" data-path="enrichment-analysis.html"><a href="enrichment-analysis.html#systematic-information-6"><i class="fa fa-check"></i><b>8.7</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="pathway-analysis.html"><a href="pathway-analysis.html"><i class="fa fa-check"></i><b>9</b> Pathway Analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="pathway-analysis.html"><a href="pathway-analysis.html#enviroment-set-up-1"><i class="fa fa-check"></i><b>9.1</b> Enviroment Set up</a></li>
<li class="chapter" data-level="9.2" data-path="pathway-analysis.html"><a href="pathway-analysis.html#over-representation-analysis"><i class="fa fa-check"></i><b>9.2</b> Over representation analysis</a></li>
<li class="chapter" data-level="9.3" data-path="pathway-analysis.html"><a href="pathway-analysis.html#clean-environment-1-1"><i class="fa fa-check"></i><b>9.3</b> Clean environment 1</a></li>
<li class="chapter" data-level="9.4" data-path="pathway-analysis.html"><a href="pathway-analysis.html#concentration-table-qea"><i class="fa fa-check"></i><b>9.4</b> Concentration Table (QEA)</a></li>
<li class="chapter" data-level="9.5" data-path="pathway-analysis.html"><a href="pathway-analysis.html#systematic-information-7"><i class="fa fa-check"></i><b>9.5</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="example.html"><a href="example.html"><i class="fa fa-check"></i><b>10</b> Example</a>
<ul>
<li class="chapter" data-level="10.1" data-path="example.html"><a href="example.html#loading-packages-6"><i class="fa fa-check"></i><b>10.1</b> Loading packages</a></li>
<li class="chapter" data-level="10.2" data-path="example.html"><a href="example.html#importing-data-6"><i class="fa fa-check"></i><b>10.2</b> Importing data</a></li>
<li class="chapter" data-level="10.3" data-path="example.html"><a href="example.html#data-processing-1"><i class="fa fa-check"></i><b>10.3</b> Data Processing</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="example.html"><a href="example.html#data-checking-1"><i class="fa fa-check"></i><b>10.3.1</b> Data Checking</a></li>
<li class="chapter" data-level="10.3.2" data-path="example.html"><a href="example.html#data-filtering-1"><i class="fa fa-check"></i><b>10.3.2</b> Data Filtering</a></li>
<li class="chapter" data-level="10.3.3" data-path="example.html"><a href="example.html#missing-value-imputation-1"><i class="fa fa-check"></i><b>10.3.3</b> Missing Value Imputation</a></li>
<li class="chapter" data-level="10.3.4" data-path="example.html"><a href="example.html#data-normalization-1"><i class="fa fa-check"></i><b>10.3.4</b> Data Normalization</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="example.html"><a href="example.html#cluster-analysis-1"><i class="fa fa-check"></i><b>10.4</b> Cluster Analysis</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="example.html"><a href="example.html#hierarchical-clustering-1"><i class="fa fa-check"></i><b>10.4.1</b> Hierarchical Clustering</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="example.html"><a href="example.html#chemometrics-analysis-1"><i class="fa fa-check"></i><b>10.5</b> Chemometrics Analysis</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="example.html"><a href="example.html#partial-least-squares-discriminant-analysis-pls-da-1"><i class="fa fa-check"></i><b>10.5.1</b> Partial Least Squares-Discriminant Analysis (PLS-DA)</a></li>
<li class="chapter" data-level="10.5.2" data-path="example.html"><a href="example.html#sparse-partial-least-squares-discriminant-analysis-spls-da-1"><i class="fa fa-check"></i><b>10.5.2</b> Sparse Partial Least Squares-Discriminant Analysis (sPLS-DA)</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="example.html"><a href="example.html#univariate-analysis-1"><i class="fa fa-check"></i><b>10.6</b> Univariate Analysis</a>
<ul>
<li class="chapter" data-level="10.6.1" data-path="example.html"><a href="example.html#fold-change-analysis-1"><i class="fa fa-check"></i><b>10.6.1</b> Fold Change Analysis</a></li>
<li class="chapter" data-level="10.6.2" data-path="example.html"><a href="example.html#t-test-1"><i class="fa fa-check"></i><b>10.6.2</b> T Test</a></li>
<li class="chapter" data-level="10.6.3" data-path="example.html"><a href="example.html#volcano-plot-1"><i class="fa fa-check"></i><b>10.6.3</b> Volcano plot</a></li>
</ul></li>
<li class="chapter" data-level="10.7" data-path="example.html"><a href="example.html#feature-selection"><i class="fa fa-check"></i><b>10.7</b> Feature Selection</a>
<ul>
<li class="chapter" data-level="10.7.1" data-path="example.html"><a href="example.html#regularized-generalized-linear-models-lasso-alpha-1"><i class="fa fa-check"></i><b>10.7.1</b> Regularized Generalized Linear Models (Lasso: alpha = 1)</a></li>
<li class="chapter" data-level="10.7.2" data-path="example.html"><a href="example.html#classification-random-forest"><i class="fa fa-check"></i><b>10.7.2</b> Classification (Random Forest)</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="example.html"><a href="example.html#network-analysis-2"><i class="fa fa-check"></i><b>10.8</b> Network Analysis</a>
<ul>
<li class="chapter" data-level="10.8.1" data-path="example.html"><a href="example.html#data-curation-2"><i class="fa fa-check"></i><b>10.8.1</b> Data curation</a></li>
<li class="chapter" data-level="10.8.2" data-path="example.html"><a href="example.html#building-network-model-4"><i class="fa fa-check"></i><b>10.8.2</b> Building network model</a></li>
<li class="chapter" data-level="10.8.3" data-path="example.html"><a href="example.html#visualizing-the-network-4"><i class="fa fa-check"></i><b>10.8.3</b> Visualizing the network</a></li>
</ul></li>
<li class="chapter" data-level="10.9" data-path="example.html"><a href="example.html#network-analysis-by-wgcna"><i class="fa fa-check"></i><b>10.9</b> Network Analysis by WGCNA</a>
<ul>
<li class="chapter" data-level="10.9.1" data-path="example.html"><a href="example.html#data-curation-3"><i class="fa fa-check"></i><b>10.9.1</b> Data curation</a></li>
<li class="chapter" data-level="10.9.2" data-path="example.html"><a href="example.html#tuning-soft-thresholds"><i class="fa fa-check"></i><b>10.9.2</b> Tuning soft thresholds</a></li>
<li class="chapter" data-level="10.9.3" data-path="example.html"><a href="example.html#create-the-network-using-the-blockwisemodules"><i class="fa fa-check"></i><b>10.9.3</b> Create the network using the <code>blockwiseModules</code></a></li>
<li class="chapter" data-level="10.9.4" data-path="example.html"><a href="example.html#plot-modules"><i class="fa fa-check"></i><b>10.9.4</b> plot modules</a></li>
<li class="chapter" data-level="10.9.5" data-path="example.html"><a href="example.html#relationships-among-modules"><i class="fa fa-check"></i><b>10.9.5</b> Relationships among modules</a></li>
<li class="chapter" data-level="10.9.6" data-path="example.html"><a href="example.html#module-eigengene-correlation"><i class="fa fa-check"></i><b>10.9.6</b> Module (Eigengene) correlation</a></li>
<li class="chapter" data-level="10.9.7" data-path="example.html"><a href="example.html#relate-module-cluster-assignments-to-groups"><i class="fa fa-check"></i><b>10.9.7</b> Relate Module (cluster) Assignments to Groups</a></li>
<li class="chapter" data-level="10.9.8" data-path="example.html"><a href="example.html#generate-and-export-networks"><i class="fa fa-check"></i><b>10.9.8</b> Generate and Export Networks</a></li>
<li class="chapter" data-level="10.9.9" data-path="example.html"><a href="example.html#network-visualization"><i class="fa fa-check"></i><b>10.9.9</b> Network visualization</a></li>
</ul></li>
<li class="chapter" data-level="10.10" data-path="example.html"><a href="example.html#systematic-information-8"><i class="fa fa-check"></i><b>10.10</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data Analysis in Metabolomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="pathway-analysis" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">Chapter 9</span> Pathway Analysis<a href="pathway-analysis.html#pathway-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>In this tutorial, we aim to help you to walk through the pathway analysis in Metaboanalyst5.</p>
<p>This module supports pathway analysis (integrating enrichment analysis and pathway topology analysis) and visualization for 26 model organisms, including Human, Mouse, Rat, Cow, Chicken, Zebrafish, Arabidopsis thaliana, Rice, Drosophila, Malaria, S. cerevisae, E.coli, and others species.</p>
<p>Here, we apply TM metabolites sequencing data of stool samples in GvHD project as demo data. Moreover, to reduce data noise, we selected differentially abundant metabolites in aGVHD patients from the differential analysis result in Xu XiaoMin’s analysis.</p>
<div id="enviroment-set-up-1" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> Enviroment Set up<a href="pathway-analysis.html#enviroment-set-up-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
</div>
<div id="over-representation-analysis" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> Over representation analysis<a href="pathway-analysis.html#over-representation-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Similar to ORA in Enrichment analysis.</p>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="pathway-analysis.html#cb153-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Read in differentially abundant TM metanolites data from GVHD project</span></span>
<span id="cb153-2"><a href="pathway-analysis.html#cb153-2" aria-hidden="true" tabindex="-1"></a>DA_metabolites <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(<span class="st">&#39;./dataset/InputFiles/DA_metabolites_agvhd_adult_result.xlsx&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb153-3"><a href="pathway-analysis.html#cb153-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb153-4"><a href="pathway-analysis.html#cb153-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb153-5"><a href="pathway-analysis.html#cb153-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Create vector consisting of compounds for enrichment analysis</span></span>
<span id="cb153-6"><a href="pathway-analysis.html#cb153-6" aria-hidden="true" tabindex="-1"></a>tmp.vec <span class="ot">&lt;-</span> DA_metabolites<span class="sc">$</span>Compounds</span>
<span id="cb153-7"><a href="pathway-analysis.html#cb153-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb153-8"><a href="pathway-analysis.html#cb153-8" aria-hidden="true" tabindex="-1"></a><span class="do">## Create mSetObj for storing objects created during your analysis</span></span>
<span id="cb153-9"><a href="pathway-analysis.html#cb153-9" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">InitDataObjects</span>(<span class="st">&quot;conc&quot;</span>, <span class="st">&quot;pathora&quot;</span>, <span class="cn">FALSE</span>)</span>
<span id="cb153-10"><a href="pathway-analysis.html#cb153-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb153-11"><a href="pathway-analysis.html#cb153-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Set up mSetObj with the list of compounds</span></span>
<span id="cb153-12"><a href="pathway-analysis.html#cb153-12" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">Setup.MapData</span>(mSet, tmp.vec)</span>
<span id="cb153-13"><a href="pathway-analysis.html#cb153-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb153-14"><a href="pathway-analysis.html#cb153-14" aria-hidden="true" tabindex="-1"></a><span class="do">## Cross reference list of compounds against libraries (hmdb, pubchem, chebi, kegg, metlin). This step is to make sure that all metabolites names provided match the metabolites&#39; names in KEGG database.</span></span>
<span id="cb153-15"><a href="pathway-analysis.html#cb153-15" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">CrossReferencing</span>(mSet, <span class="st">&quot;name&quot;</span>)</span></code></pre></div>
<p>[1] “Loaded files from MetaboAnalyst web-server.”
[1] “Loaded files from MetaboAnalyst web-server.”
[1] “0”<br />
[2] “Over half of the compound IDs could not be matched to our database. Please make sure that correct compound IDs or common compound names are used.”</p>
<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb154-1"><a href="pathway-analysis.html#cb154-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Remove metabolites failed to match the metabolites&#39; names in KEGG database.</span></span>
<span id="cb154-2"><a href="pathway-analysis.html#cb154-2" aria-hidden="true" tabindex="-1"></a>tmp.vec <span class="ot">&lt;-</span> tmp.vec[mSet<span class="sc">$</span>name.map<span class="sc">$</span>match.state <span class="sc">==</span> <span class="dv">1</span>]</span>
<span id="cb154-3"><a href="pathway-analysis.html#cb154-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb154-4"><a href="pathway-analysis.html#cb154-4" aria-hidden="true" tabindex="-1"></a><span class="do">## Re-initialize mSet object</span></span>
<span id="cb154-5"><a href="pathway-analysis.html#cb154-5" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">InitDataObjects</span>(<span class="st">&quot;conc&quot;</span>, <span class="st">&quot;pathora&quot;</span>, <span class="cn">FALSE</span>)</span>
<span id="cb154-6"><a href="pathway-analysis.html#cb154-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb154-7"><a href="pathway-analysis.html#cb154-7" aria-hidden="true" tabindex="-1"></a><span class="do">## Set up mSetObj with the matched list of compounds</span></span>
<span id="cb154-8"><a href="pathway-analysis.html#cb154-8" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">Setup.MapData</span>(mSet, tmp.vec)</span>
<span id="cb154-9"><a href="pathway-analysis.html#cb154-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb154-10"><a href="pathway-analysis.html#cb154-10" aria-hidden="true" tabindex="-1"></a><span class="do">## Cross reference list of compounds against libraries (hmdb, pubchem, chebi, kegg, metlin). Check again if all metabolites&#39; names have their matches in KEGG database</span></span>
<span id="cb154-11"><a href="pathway-analysis.html#cb154-11" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">CrossReferencing</span>(mSet, <span class="st">&quot;name&quot;</span>)</span></code></pre></div>
<p>[1] “Loaded files from MetaboAnalyst web-server.”
[1] “Loaded files from MetaboAnalyst web-server.”
[1] “1”<br />
[2] “Name matching OK, please inspect (and manual correct) the results then proceed.”</p>
<div class="sourceCode" id="cb155"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb155-1"><a href="pathway-analysis.html#cb155-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Creates a mapping result table shows HMDB, KEGG, PubChem, etc. IDs. Saved as &quot;name_map.csv&quot; or can be found in mSet$dataSet$map.table. Compounds with no hits will contain NAs across the columns.</span></span>
<span id="cb155-2"><a href="pathway-analysis.html#cb155-2" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">CreateMappingResultTable</span>(mSet)</span></code></pre></div>
<p>[1] “Loaded files from MetaboAnalyst web-server.”</p>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="pathway-analysis.html#cb156-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Select the pathway library, ranging from mammals to prokaryotes</span></span>
<span id="cb156-2"><a href="pathway-analysis.html#cb156-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Note the third parameter, where users need to input the KEGG pathway version.</span></span>
<span id="cb156-3"><a href="pathway-analysis.html#cb156-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Use &quot;current&quot; for the latest KEGG pathway library or &quot;v2018&quot; for the KEGG pathway library version prior to November 2019.</span></span>
<span id="cb156-4"><a href="pathway-analysis.html#cb156-4" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">SetKEGG.PathLib</span>(mSet, <span class="st">&quot;hsa&quot;</span>, <span class="st">&quot;current&quot;</span>)</span>
<span id="cb156-5"><a href="pathway-analysis.html#cb156-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-6"><a href="pathway-analysis.html#cb156-6" aria-hidden="true" tabindex="-1"></a><span class="do">## Set the metabolite filter. Default set to false</span></span>
<span id="cb156-7"><a href="pathway-analysis.html#cb156-7" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">SetMetabolomeFilter</span>(mSet, F)</span>
<span id="cb156-8"><a href="pathway-analysis.html#cb156-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb156-9"><a href="pathway-analysis.html#cb156-9" aria-hidden="true" tabindex="-1"></a><span class="do">## Calculate the over representation analysis score, here we selected to use the hypergeometric test (alternative is Fisher&#39;s exact test)</span></span>
<span id="cb156-10"><a href="pathway-analysis.html#cb156-10" aria-hidden="true" tabindex="-1"></a><span class="do">## A results table &quot;pathway_results.csv&quot; will be created and found within your working directory</span></span>
<span id="cb156-11"><a href="pathway-analysis.html#cb156-11" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">CalculateOraScore</span>(mSet, <span class="st">&quot;rbc&quot;</span>, <span class="st">&quot;hyperg&quot;</span>)</span></code></pre></div>
<p>[1] “Loaded files from MetaboAnalyst web-server.”
[1] “<a href="http://api.xialab.ca/pathwayora" class="uri">http://api.xialab.ca/pathwayora</a>”</p>
<div class="sourceCode" id="cb157"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb157-1"><a href="pathway-analysis.html#cb157-1" aria-hidden="true" tabindex="-1"></a>pathway_res <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&#39;./pathway_results.csv&#39;</span>)</span>
<span id="cb157-2"><a href="pathway-analysis.html#cb157-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb157-3"><a href="pathway-analysis.html#cb157-3" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(pathway_res))</span></code></pre></div>
<table>
<colgroup>
<col width="12%" />
<col width="8%" />
<col width="12%" />
<col width="7%" />
<col width="11%" />
<col width="12%" />
<col width="17%" />
<col width="5%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">X</th>
<th align="right">Total</th>
<th align="right">Expected</th>
<th align="right">Hits</th>
<th align="right">Raw.p</th>
<th align="right">X.log.p.</th>
<th align="right">Holm.adjust</th>
<th align="right">FDR</th>
<th align="right">Impact</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">hsa00130</td>
<td align="right">9</td>
<td align="right">0.046452</td>
<td align="right">1</td>
<td align="right">0.04562</td>
<td align="right">3.0874</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.00000</td>
</tr>
<tr class="even">
<td align="left">hsa00030</td>
<td align="right">22</td>
<td align="right">0.113550</td>
<td align="right">1</td>
<td align="right">0.10830</td>
<td align="right">2.2229</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.04712</td>
</tr>
<tr class="odd">
<td align="left">hsa00564</td>
<td align="right">36</td>
<td align="right">0.185810</td>
<td align="right">1</td>
<td align="right">0.17174</td>
<td align="right">1.7618</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.00000</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="pathway-analysis.html#cb158-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot of the Pathway Analysis Overview</span></span>
<span id="cb158-2"><a href="pathway-analysis.html#cb158-2" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">PlotPathSummary</span>(mSet,<span class="at">show.grid=</span><span class="cn">FALSE</span>, <span class="st">&quot;./dataset/OutputFiles/path_view_0_&quot;</span>, <span class="st">&quot;png&quot;</span>, <span class="at">dpi =</span><span class="dv">250</span>, <span class="at">width=</span><span class="cn">NA</span>)</span>
<span id="cb158-3"><a href="pathway-analysis.html#cb158-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb158-4"><a href="pathway-analysis.html#cb158-4" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">&#39;./dataset/OutputFiles/path_view_0_dpi250.png&#39;</span>)</span></code></pre></div>
<p><img src="dataset/OutputFiles/path_view_0_dpi250.png" width="875" /></p>
<div class="sourceCode" id="cb159"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb159-1"><a href="pathway-analysis.html#cb159-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot a specific metabolic pathway, in this case &quot;Glycine, serine and threonine metabolism&quot;</span></span>
<span id="cb159-2"><a href="pathway-analysis.html#cb159-2" aria-hidden="true" tabindex="-1"></a><span class="co"># mSet &lt;- PlotKEGGPath(mSetObj = mSet, pathName = &quot;Glycine, serine and threonine metabolism&quot;, 528, 480, &quot;png&quot;, dpi=72)</span></span></code></pre></div>
</div>
<div id="clean-environment-1-1" class="section level2 hasAnchor" number="9.3">
<h2><span class="header-section-number">9.3</span> Clean environment 1<a href="pathway-analysis.html#clean-environment-1-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb160"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb160-1"><a href="pathway-analysis.html#cb160-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span></code></pre></div>
</div>
<div id="concentration-table-qea" class="section level2 hasAnchor" number="9.4">
<h2><span class="header-section-number">9.4</span> Concentration Table (QEA)<a href="pathway-analysis.html#concentration-table-qea" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Similar to QEA in Enrichment analysis.</p>
<p>KO enrichment is calculated via the global test algorithm when abundance of metabolites are provided.</p>
<p>Global test evaluates whether a set of genes (i.e. KEGG pathways) is significantly associated with a variable of interest. Compared to ORA, which uses only the total number of KO hits in a pathway, global test considers the gene abundance values and is considered to be more sensitive than ORA. <strong>It assumes that if a gene set can be used to predict an outcome of interest, the gene expression patterns per outcome must be different</strong>.</p>
<p>The global test algorithm is implemented in MicrobiomeAnalyst using the globaltest R package. P-values for both methods are corrected for multiple-testing using the Benjamini and Hochberg’s False-Discovery Rate (FDR).</p>
<p>In this module, we use differentially abundant metabolites in aGVHD patients from the differential analysis result in Xu XiaoMin’s analysis and their abundance as input.</p>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="pathway-analysis.html#cb161-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Read in differentially abundant TM metanolites data from GVHD project</span></span>
<span id="cb161-2"><a href="pathway-analysis.html#cb161-2" aria-hidden="true" tabindex="-1"></a>DA_metabolites <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(<span class="st">&#39;./dataset/InputFiles/DA_metabolites_agvhd_adult_result.xlsx&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb161-3"><a href="pathway-analysis.html#cb161-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb161-4"><a href="pathway-analysis.html#cb161-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb161-5"><a href="pathway-analysis.html#cb161-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Initialize data object</span></span>
<span id="cb161-6"><a href="pathway-analysis.html#cb161-6" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">InitDataObjects</span>(<span class="st">&quot;conc&quot;</span>, <span class="st">&quot;pathqea&quot;</span>, <span class="cn">FALSE</span>)</span></code></pre></div>
<p>Starting Rserve:
/Library/Frameworks/R.framework/Resources/bin/R CMD /Library/Frameworks/R.framework/Versions/4.1/Resources/library/Rserve/libs//Rserve –no-save</p>
<p>[1] “MetaboAnalyst R objects initialized …”</p>
<div class="sourceCode" id="cb162"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb162-1"><a href="pathway-analysis.html#cb162-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Read in data table</span></span>
<span id="cb162-2"><a href="pathway-analysis.html#cb162-2" aria-hidden="true" tabindex="-1"></a>input_df <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(<span class="st">&#39;./dataset/InputFiles/GvHD_stool_metabolites_TM.xlsx&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb162-3"><a href="pathway-analysis.html#cb162-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb162-4"><a href="pathway-analysis.html#cb162-4" aria-hidden="true" tabindex="-1"></a>  .[,<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">25</span><span class="sc">:</span><span class="fu">ncol</span>(.))] <span class="sc">%&gt;%</span> </span>
<span id="cb162-5"><a href="pathway-analysis.html#cb162-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">&#39;Compounds&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb162-6"><a href="pathway-analysis.html#cb162-6" aria-hidden="true" tabindex="-1"></a>  .[DA_metabolites<span class="sc">$</span>Compounds,] <span class="sc">%&gt;%</span> </span>
<span id="cb162-7"><a href="pathway-analysis.html#cb162-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb162-8"><a href="pathway-analysis.html#cb162-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb162-9"><a href="pathway-analysis.html#cb162-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb162-10"><a href="pathway-analysis.html#cb162-10" aria-hidden="true" tabindex="-1"></a>Groupinfo <span class="ot">&lt;-</span> <span class="fu">rownames</span>(input_df) <span class="sc">%&gt;%</span> </span>
<span id="cb162-11"><a href="pathway-analysis.html#cb162-11" aria-hidden="true" tabindex="-1"></a>  stringr<span class="sc">::</span><span class="fu">str_split</span>(<span class="st">&#39;-&#39;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb162-12"><a href="pathway-analysis.html#cb162-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb162-13"><a href="pathway-analysis.html#cb162-13" aria-hidden="true" tabindex="-1"></a>  .[<span class="dv">2</span>,] <span class="sc">%&gt;%</span> </span>
<span id="cb162-14"><a href="pathway-analysis.html#cb162-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unlist</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb162-15"><a href="pathway-analysis.html#cb162-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.vector</span>()</span>
<span id="cb162-16"><a href="pathway-analysis.html#cb162-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb162-17"><a href="pathway-analysis.html#cb162-17" aria-hidden="true" tabindex="-1"></a>input_df <span class="sc">%&lt;&gt;%</span> dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Group =</span> Groupinfo) <span class="sc">%&gt;%</span> </span>
<span id="cb162-18"><a href="pathway-analysis.html#cb162-18" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(<span class="st">&#39;Group&#39;</span>, <span class="fu">colnames</span>(.)[<span class="fu">colnames</span>(.) <span class="sc">!=</span> <span class="st">&#39;Group&#39;</span>])) <span class="sc">%&gt;%</span> </span>
<span id="cb162-19"><a href="pathway-analysis.html#cb162-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&#39;Sample&#39;</span>)</span>
<span id="cb162-20"><a href="pathway-analysis.html#cb162-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb162-21"><a href="pathway-analysis.html#cb162-21" aria-hidden="true" tabindex="-1"></a><span class="do">## Write reformed metabolites abundance table into csv file</span></span>
<span id="cb162-22"><a href="pathway-analysis.html#cb162-22" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(input_df, <span class="st">&#39;./dataset/OutputFiles/GvHD_stool_metabolites_TM.csv&#39;</span>,<span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb162-23"><a href="pathway-analysis.html#cb162-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb162-24"><a href="pathway-analysis.html#cb162-24" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">Read.TextData</span>(mSet, <span class="st">&#39;./dataset/OutputFiles/GvHD_stool_metabolites_TM.csv&#39;</span>, <span class="st">&quot;rowu&quot;</span>, <span class="st">&quot;disc&quot;</span>)</span>
<span id="cb162-25"><a href="pathway-analysis.html#cb162-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb162-26"><a href="pathway-analysis.html#cb162-26" aria-hidden="true" tabindex="-1"></a><span class="do">## Show first few rows in data table</span></span>
<span id="cb162-27"><a href="pathway-analysis.html#cb162-27" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(input_df))</span></code></pre></div>
<table style="width:100%;">
<colgroup>
<col width="0%" />
<col width="0%" />
<col width="1%" />
<col width="2%" />
<col width="2%" />
<col width="1%" />
<col width="4%" />
<col width="2%" />
<col width="5%" />
<col width="1%" />
<col width="3%" />
<col width="2%" />
<col width="1%" />
<col width="1%" />
<col width="2%" />
<col width="4%" />
<col width="1%" />
<col width="3%" />
<col width="1%" />
<col width="1%" />
<col width="1%" />
<col width="1%" />
<col width="2%" />
<col width="2%" />
<col width="1%" />
<col width="9%" />
<col width="14%" />
<col width="1%" />
<col width="1%" />
<col width="1%" />
<col width="1%" />
<col width="1%" />
<col width="2%" />
<col width="1%" />
<col width="1%" />
<col width="5%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Sample</th>
<th align="left">Group</th>
<th align="right">Erucic acid</th>
<th align="right">5-Methylcytosine</th>
<th align="right">D-Gluconic Acid</th>
<th align="right">LPE(14:0/0:0)</th>
<th align="right">N-Acetylglucosamine 1-Phosphate</th>
<th align="right">7-ketodeoxycholic acid</th>
<th align="right">Bis(1-inositol) -3,1’-phosphate 1-phosphate</th>
<th align="right">Asp-phe</th>
<th align="right">Phenylacetyl-L-Glutamine</th>
<th align="right">N-Acetylglycine</th>
<th align="right">Carnitine C2:0</th>
<th align="right">Acetylcholine</th>
<th align="right">Lithocholic acid</th>
<th align="right">3-Carboxypropyltrimethylammonium</th>
<th align="right">Biotinamide</th>
<th align="right">5’-deoxy-5’-fluoroadenosine</th>
<th align="right">Carnitine C5:1</th>
<th align="right">Val-Val</th>
<th align="right">Gly Leu Tyr</th>
<th align="right">Ile-Thr</th>
<th align="right">Isoleucyl-Serine</th>
<th align="right">Isoleucyl-Tyrosine</th>
<th align="right">Phe Val Ala</th>
<th align="right">Phosphonic acid, P-[(3R)-3-amino-4-[(3-hexylphenyl)amino]-4-oxobutyl]-</th>
<th align="right">2-[2,4-dihydroxy-3-(3-methylbut-2-en-1-yl)phenyl]-1-[2,4,6-trihydroxy-3-(3-methylbut-2-en-1-yl)phenyl]propan-1-one</th>
<th align="right">Asp Ile Ile</th>
<th align="right">Gly Leu Val</th>
<th align="right">Ile Glu Leu</th>
<th align="right">Ile Ile Asn</th>
<th align="right">Lys Pro Ile</th>
<th align="right">Phylloquinone oxide</th>
<th align="right">Thr Val Met</th>
<th align="right">Val Leu Thr</th>
<th align="right">(2R,3S,4R,5R)-2,3,4,5,6-pentahydroxyhexanoate</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">TZW-V0</td>
<td align="left">V0</td>
<td align="right">1629500</td>
<td align="right">630750</td>
<td align="right">11598000</td>
<td align="right">1900000</td>
<td align="right">1048900</td>
<td align="right">142580.0</td>
<td align="right">4516900</td>
<td align="right">1896000</td>
<td align="right">483960</td>
<td align="right">75306000</td>
<td align="right">35820000</td>
<td align="right">169220</td>
<td align="right">30462</td>
<td align="right">169220</td>
<td align="right">72050</td>
<td align="right">52622</td>
<td align="right">48756</td>
<td align="right">6690500</td>
<td align="right">382840</td>
<td align="right">151620</td>
<td align="right">863010</td>
<td align="right">3696700</td>
<td align="right">371890</td>
<td align="right">419900</td>
<td align="right">376300</td>
<td align="right">1992200</td>
<td align="right">304420</td>
<td align="right">321110</td>
<td align="right">243390</td>
<td align="right">17617</td>
<td align="right">3885</td>
<td align="right">1950000</td>
<td align="right">582200</td>
<td align="right">208490000</td>
</tr>
<tr class="even">
<td align="left">LBC-V0</td>
<td align="left">V0</td>
<td align="right">17057</td>
<td align="right">220290</td>
<td align="right">8239500</td>
<td align="right">68958</td>
<td align="right">9880900</td>
<td align="right">6446.1</td>
<td align="right">313960</td>
<td align="right">19488000</td>
<td align="right">569570</td>
<td align="right">11310000</td>
<td align="right">81713000</td>
<td align="right">90812</td>
<td align="right">9</td>
<td align="right">90812</td>
<td align="right">533990</td>
<td align="right">9</td>
<td align="right">338900</td>
<td align="right">1044600</td>
<td align="right">9</td>
<td align="right">99102</td>
<td align="right">468170</td>
<td align="right">7789300</td>
<td align="right">46193</td>
<td align="right">88988</td>
<td align="right">14154</td>
<td align="right">463070</td>
<td align="right">10903</td>
<td align="right">32430</td>
<td align="right">29187</td>
<td align="right">12902000</td>
<td align="right">9</td>
<td align="right">333220</td>
<td align="right">210680</td>
<td align="right">166450000</td>
</tr>
<tr class="odd">
<td align="left">HXZ-V0</td>
<td align="left">V0</td>
<td align="right">3719000</td>
<td align="right">316360</td>
<td align="right">91189</td>
<td align="right">14390000</td>
<td align="right">37784000</td>
<td align="right">21965000.0</td>
<td align="right">459320</td>
<td align="right">5671200</td>
<td align="right">8502200</td>
<td align="right">6758300</td>
<td align="right">71495000</td>
<td align="right">518950</td>
<td align="right">438800</td>
<td align="right">518950</td>
<td align="right">1222000</td>
<td align="right">36295</td>
<td align="right">953350</td>
<td align="right">8538900</td>
<td align="right">88531</td>
<td align="right">454250</td>
<td align="right">952750</td>
<td align="right">56697</td>
<td align="right">45953</td>
<td align="right">115110</td>
<td align="right">2798900</td>
<td align="right">348560</td>
<td align="right">79613</td>
<td align="right">134400</td>
<td align="right">438750</td>
<td align="right">7569200</td>
<td align="right">24432</td>
<td align="right">747830</td>
<td align="right">86317</td>
<td align="right">2163600</td>
</tr>
<tr class="even">
<td align="left">LJY-V0</td>
<td align="left">V0</td>
<td align="right">1641600</td>
<td align="right">285270</td>
<td align="right">72194</td>
<td align="right">7366800</td>
<td align="right">9557700</td>
<td align="right">3289400.0</td>
<td align="right">71782</td>
<td align="right">756830</td>
<td align="right">30602</td>
<td align="right">531930</td>
<td align="right">387590</td>
<td align="right">1345400</td>
<td align="right">6488400</td>
<td align="right">1345400</td>
<td align="right">544550</td>
<td align="right">27188</td>
<td align="right">295820</td>
<td align="right">3163500</td>
<td align="right">136380</td>
<td align="right">180510</td>
<td align="right">819430</td>
<td align="right">21168</td>
<td align="right">171330</td>
<td align="right">159850</td>
<td align="right">1478200</td>
<td align="right">804870</td>
<td align="right">128420</td>
<td align="right">910850</td>
<td align="right">379890</td>
<td align="right">78433</td>
<td align="right">459240</td>
<td align="right">576110</td>
<td align="right">123100</td>
<td align="right">1502600</td>
</tr>
<tr class="odd">
<td align="left">CJY-V0</td>
<td align="left">V0</td>
<td align="right">9</td>
<td align="right">45013</td>
<td align="right">25589</td>
<td align="right">882500</td>
<td align="right">561320</td>
<td align="right">1603.1</td>
<td align="right">9</td>
<td align="right">7463500</td>
<td align="right">556470</td>
<td align="right">5814900</td>
<td align="right">13723000</td>
<td align="right">26319</td>
<td align="right">9</td>
<td align="right">26319</td>
<td align="right">182350</td>
<td align="right">18175</td>
<td align="right">135980</td>
<td align="right">1555700</td>
<td align="right">38015</td>
<td align="right">130400</td>
<td align="right">441760</td>
<td align="right">3275600</td>
<td align="right">235320</td>
<td align="right">154660</td>
<td align="right">169880</td>
<td align="right">568150</td>
<td align="right">35916</td>
<td align="right">48067</td>
<td align="right">215340</td>
<td align="right">328930</td>
<td align="right">9</td>
<td align="right">763870</td>
<td align="right">97575</td>
<td align="right">900440</td>
</tr>
<tr class="even">
<td align="left">WKM-V0</td>
<td align="left">V0</td>
<td align="right">3079600</td>
<td align="right">228130</td>
<td align="right">396980</td>
<td align="right">8410000</td>
<td align="right">12296000</td>
<td align="right">7266400.0</td>
<td align="right">42405</td>
<td align="right">13785000</td>
<td align="right">4553400</td>
<td align="right">9620400</td>
<td align="right">1311800</td>
<td align="right">4392400</td>
<td align="right">9</td>
<td align="right">4392400</td>
<td align="right">74831</td>
<td align="right">79472</td>
<td align="right">48558</td>
<td align="right">5309200</td>
<td align="right">88741</td>
<td align="right">221440</td>
<td align="right">593130</td>
<td align="right">44678</td>
<td align="right">250550</td>
<td align="right">140720</td>
<td align="right">1660200</td>
<td align="right">522490</td>
<td align="right">234090</td>
<td align="right">263740</td>
<td align="right">415690</td>
<td align="right">9312900</td>
<td align="right">586940</td>
<td align="right">619550</td>
<td align="right">136990</td>
<td align="right">8894600</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb163"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb163-1"><a href="pathway-analysis.html#cb163-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Check metabolites Names of input table. Found &gt;15 compounds without matches.</span></span>
<span id="cb163-2"><a href="pathway-analysis.html#cb163-2" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">CrossReferencing</span>(mSet, <span class="st">&quot;name&quot;</span>)</span></code></pre></div>
<p>[1] “Loaded files from MetaboAnalyst web-server.”
[1] “Loaded files from MetaboAnalyst web-server.”
[1] “0”<br />
[2] “Over half of the compound IDs could not be matched to our database. Please make sure that correct compound IDs or common compound names are used.”</p>
<div class="sourceCode" id="cb164"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb164-1"><a href="pathway-analysis.html#cb164-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Keep matched metabolites</span></span>
<span id="cb164-2"><a href="pathway-analysis.html#cb164-2" aria-hidden="true" tabindex="-1"></a>metabolites_keep <span class="ot">&lt;-</span> mSet<span class="sc">$</span>name.map<span class="sc">$</span>query.vec[mSet<span class="sc">$</span>name.map<span class="sc">$</span>match.state <span class="sc">==</span> <span class="dv">1</span>]</span>
<span id="cb164-3"><a href="pathway-analysis.html#cb164-3" aria-hidden="true" tabindex="-1"></a>input_df <span class="ot">&lt;-</span> input_df[<span class="fu">c</span>(<span class="st">&#39;Sample&#39;</span>,<span class="st">&#39;Group&#39;</span>, metabolites_keep)]</span>
<span id="cb164-4"><a href="pathway-analysis.html#cb164-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb164-5"><a href="pathway-analysis.html#cb164-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Overwrite previously generated abundance csv file with matched metabolites</span></span>
<span id="cb164-6"><a href="pathway-analysis.html#cb164-6" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(input_df, <span class="st">&#39;./dataset/OutputFiles/GvHD_stool_metabolites_TM.csv&#39;</span>,<span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb164-7"><a href="pathway-analysis.html#cb164-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb164-8"><a href="pathway-analysis.html#cb164-8" aria-hidden="true" tabindex="-1"></a><span class="do">## Initialize data object again</span></span>
<span id="cb164-9"><a href="pathway-analysis.html#cb164-9" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">InitDataObjects</span>(<span class="st">&quot;conc&quot;</span>, <span class="st">&quot;pathqea&quot;</span>, <span class="cn">FALSE</span>)</span>
<span id="cb164-10"><a href="pathway-analysis.html#cb164-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb164-11"><a href="pathway-analysis.html#cb164-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Read in updated metabolites abundance table</span></span>
<span id="cb164-12"><a href="pathway-analysis.html#cb164-12" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">Read.TextData</span>(mSet, <span class="st">&#39;./dataset/OutputFiles/GvHD_stool_metabolites_TM.csv&#39;</span>, <span class="st">&quot;rowu&quot;</span>, <span class="st">&quot;disc&quot;</span>)</span>
<span id="cb164-13"><a href="pathway-analysis.html#cb164-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb164-14"><a href="pathway-analysis.html#cb164-14" aria-hidden="true" tabindex="-1"></a><span class="do">## Check Sanity and replace 0</span></span>
<span id="cb164-15"><a href="pathway-analysis.html#cb164-15" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">SanityCheckData</span>(mSet)</span></code></pre></div>
<p>[1] “Successfully passed sanity check!”<br />
[2] “Samples are not paired.”<br />
[3] “2 groups were detected in samples.”<br />
[4] “Only English letters, numbers, underscore, hyphen and forward slash (/) are allowed.”<br />
[5] “<font color=\"orange\">Other special characters or punctuations (if any) will be stripped off.</font>”<br />
[6] “All data values are numeric.”<br />
[7] “A total of 0 (0%) missing values were detected.”<br />
[8] “<u>By default, missing values will be replaced by 1/5 of min positive values of their corresponding variables</u>”
[9] “Click the <b>Proceed</b> button if you accept the default practice;”<br />
[10] “Or click the <b>Missing Values</b> button to use other methods.”</p>
<div class="sourceCode" id="cb165"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb165-1"><a href="pathway-analysis.html#cb165-1" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">ReplaceMin</span>(mSet)</span>
<span id="cb165-2"><a href="pathway-analysis.html#cb165-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb165-3"><a href="pathway-analysis.html#cb165-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Check metabolites Names of input table again</span></span>
<span id="cb165-4"><a href="pathway-analysis.html#cb165-4" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">CrossReferencing</span>(mSet, <span class="st">&quot;name&quot;</span>)</span></code></pre></div>
<p>[1] “Loaded files from MetaboAnalyst web-server.”
[1] “Loaded files from MetaboAnalyst web-server.”
[1] “1”<br />
[2] “Name matching OK, please inspect (and manual correct) the results then proceed.”</p>
<div class="sourceCode" id="cb166"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb166-1"><a href="pathway-analysis.html#cb166-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Creates a mapping result table shows HMDB, KEGG, PubChem, etc. IDs. Saved as &quot;name_map.csv&quot; or can be found in mSet$dataSet$map.table. Compounds with no hits will contain NAs across the columns.</span></span>
<span id="cb166-2"><a href="pathway-analysis.html#cb166-2" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">CreateMappingResultTable</span>(mSet)</span></code></pre></div>
<p>[1] “Loaded files from MetaboAnalyst web-server.”</p>
<div class="sourceCode" id="cb167"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb167-1"><a href="pathway-analysis.html#cb167-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Normalize</span></span>
<span id="cb167-2"><a href="pathway-analysis.html#cb167-2" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">PreparePrenormData</span>(mSet)</span>
<span id="cb167-3"><a href="pathway-analysis.html#cb167-3" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">Normalization</span>(mSet, <span class="at">rowNorm =</span> <span class="st">&quot;SumNorm&quot;</span>, <span class="at">transNorm =</span> <span class="st">&quot;LogNorm&quot;</span>, <span class="at">scaleNorm =</span> <span class="st">&quot;ParetoNorm&quot;</span>)</span>
<span id="cb167-4"><a href="pathway-analysis.html#cb167-4" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">PlotNormSummary</span>(mSet, </span>
<span id="cb167-5"><a href="pathway-analysis.html#cb167-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">imgName =</span> <span class="st">&quot;./dataset/OutputFiles/kegg_pathnorm_0_&quot;</span>, </span>
<span id="cb167-6"><a href="pathway-analysis.html#cb167-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">format =</span> <span class="st">&quot;png&quot;</span>, <span class="at">dpi =</span> <span class="dv">250</span>, <span class="at">width=</span><span class="cn">NA</span>)</span>
<span id="cb167-7"><a href="pathway-analysis.html#cb167-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb167-8"><a href="pathway-analysis.html#cb167-8" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">&#39;./dataset/OutputFiles/kegg_pathnorm_0_dpi250.png&#39;</span>)</span></code></pre></div>
<p><img src="dataset/OutputFiles/kegg_pathnorm_0_dpi250.png" width="1312" /></p>
<div class="sourceCode" id="cb168"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb168-1"><a href="pathway-analysis.html#cb168-1" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">PlotSampleNormSummary</span>(mSet, <span class="st">&quot;./dataset/OutputFiles/kegg_pathsnorm_0_&quot;</span>, <span class="st">&quot;png&quot;</span>, <span class="dv">250</span>, <span class="at">width=</span><span class="cn">NA</span>)</span>
<span id="cb168-2"><a href="pathway-analysis.html#cb168-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb168-3"><a href="pathway-analysis.html#cb168-3" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">&#39;./dataset/OutputFiles/kegg_pathsnorm_0_dpi250.png&#39;</span>)</span></code></pre></div>
<p><img src="dataset/OutputFiles/kegg_pathsnorm_0_dpi250.png" width="1312" /></p>
<div class="sourceCode" id="cb169"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb169-1"><a href="pathway-analysis.html#cb169-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Enrich to KEGG</span></span>
<span id="cb169-2"><a href="pathway-analysis.html#cb169-2" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">SetKEGG.PathLib</span>(mSet, <span class="st">&quot;hsa&quot;</span>, <span class="st">&quot;current&quot;</span>)</span>
<span id="cb169-3"><a href="pathway-analysis.html#cb169-3" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">SetMetabolomeFilter</span>(mSet, F)</span>
<span id="cb169-4"><a href="pathway-analysis.html#cb169-4" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">CalculateQeaScore</span>(mSet, <span class="st">&quot;rbc&quot;</span>, <span class="st">&quot;gt&quot;</span>)</span></code></pre></div>
<p>[1] “Loaded files from MetaboAnalyst web-server.”
[1] “<a href="http://api.xialab.ca/pathwayqea" class="uri">http://api.xialab.ca/pathwayqea</a>”
[1] “Pathway QEA via api.metaboanalyst.ca successful!”</p>
<div class="sourceCode" id="cb170"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb170-1"><a href="pathway-analysis.html#cb170-1" aria-hidden="true" tabindex="-1"></a>pathway_res <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&#39;./pathway_results.csv&#39;</span>, <span class="at">check.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb170-2"><a href="pathway-analysis.html#cb170-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb170-3"><a href="pathway-analysis.html#cb170-3" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(pathway_res))</span></code></pre></div>
<table>
<colgroup>
<col width="44%" />
<col width="9%" />
<col width="4%" />
<col width="8%" />
<col width="6%" />
<col width="10%" />
<col width="8%" />
<col width="6%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">Total Cmpd</th>
<th align="right">Hits</th>
<th align="right">Raw p</th>
<th align="right">-log(p)</th>
<th align="right">Holm adjust</th>
<th align="right">FDR</th>
<th align="right">Impact</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Pentose phosphate pathway</td>
<td align="right">22</td>
<td align="right">1</td>
<td align="right">0.0024452</td>
<td align="right">6.0136</td>
<td align="right">0.0073355</td>
<td align="right">0.0073355</td>
<td align="right">0.04712</td>
</tr>
<tr class="even">
<td align="left">Glycerophospholipid metabolism</td>
<td align="right">36</td>
<td align="right">1</td>
<td align="right">0.0206320</td>
<td align="right">3.8809</td>
<td align="right">0.0412630</td>
<td align="right">0.0267100</td>
<td align="right">0.00000</td>
</tr>
<tr class="odd">
<td align="left">Ubiquinone and other terpenoid-quinone biosynthesis</td>
<td align="right">9</td>
<td align="right">1</td>
<td align="right">0.0267100</td>
<td align="right">3.6227</td>
<td align="right">0.0412630</td>
<td align="right">0.0267100</td>
<td align="right">0.00000</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb171"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb171-1"><a href="pathway-analysis.html#cb171-1" aria-hidden="true" tabindex="-1"></a>mSet <span class="ot">&lt;-</span> <span class="fu">PlotPathSummary</span>(mSet, F, <span class="st">&quot;./dataset/OutputFiles/kegg_path_view_0_&quot;</span>, <span class="st">&quot;png&quot;</span>, <span class="dv">250</span>, <span class="at">width=</span><span class="cn">NA</span>, <span class="cn">NA</span>, <span class="cn">NA</span> )</span>
<span id="cb171-2"><a href="pathway-analysis.html#cb171-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb171-3"><a href="pathway-analysis.html#cb171-3" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">&#39;./dataset/OutputFiles/kegg_path_view_0_dpi250.png&#39;</span>)</span></code></pre></div>
<p><img src="dataset/OutputFiles/kegg_path_view_0_dpi250.png" width="875" /></p>
<div class="sourceCode" id="cb172"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb172-1"><a href="pathway-analysis.html#cb172-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Plot specified pathway map</span></span>
<span id="cb172-2"><a href="pathway-analysis.html#cb172-2" aria-hidden="true" tabindex="-1"></a><span class="co"># mSet &lt;- PlotKEGGPath(mSet, &quot;Phenylalanine, tyrosine and tryptophan biosynthesis&quot;,576, 480, &quot;png&quot;, NULL)</span></span>
<span id="cb172-3"><a href="pathway-analysis.html#cb172-3" aria-hidden="true" tabindex="-1"></a><span class="co"># mSet &lt;- RerenderMetPAGraph(mSet, &quot;zoom1658398230135.png&quot;,576.0, 480.0, 100.0)</span></span>
<span id="cb172-4"><a href="pathway-analysis.html#cb172-4" aria-hidden="true" tabindex="-1"></a><span class="co"># mSet &lt;- PlotKEGGPath(mSet, &quot;Glycine, serine and threonine metabolism&quot;,576, 480, &quot;png&quot;, NULL)</span></span></code></pre></div>
</div>
<div id="systematic-information-7" class="section level2 hasAnchor" number="9.5">
<h2><span class="header-section-number">9.5</span> Systematic Information<a href="pathway-analysis.html#systematic-information-7" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb173"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb173-1"><a href="pathway-analysis.html#cb173-1" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ─────────────────────────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 4.1.2 (2021-11-01)
##  os       macOS Monterey 12.2.1
##  system   x86_64, darwin17.0
##  ui       RStudio
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       Asia/Shanghai
##  date     2022-08-11
##  rstudio  2022.07.1+554 Spotted Wakerobin (desktop)
##  pandoc   2.18 @ /Applications/RStudio.app/Contents/MacOS/quarto/bin/tools/ (via rmarkdown)
## 
## ─ Packages ─────────────────────────────────────────────────────────────────────────────────────
##  package              * version    date (UTC) lib source
##  abind                  1.4-5      2016-07-21 [1] CRAN (R 4.1.0)
##  ade4                   1.7-18     2021-09-16 [1] CRAN (R 4.1.0)
##  annotate               1.72.0     2021-10-26 [1] Bioconductor
##  AnnotationDbi          1.56.2     2021-11-09 [1] Bioconductor
##  ape                    5.6-2      2022-03-02 [1] CRAN (R 4.1.2)
##  assertthat             0.2.1      2019-03-21 [1] CRAN (R 4.1.0)
##  backports              1.4.1      2021-12-13 [1] CRAN (R 4.1.0)
##  base64enc              0.1-3      2015-07-28 [1] CRAN (R 4.1.0)
##  Biobase              * 2.54.0     2021-10-26 [1] Bioconductor
##  BiocGenerics         * 0.40.0     2021-10-26 [1] Bioconductor
##  BiocParallel           1.28.3     2021-12-09 [1] Bioconductor
##  biomformat             1.22.0     2021-10-26 [1] Bioconductor
##  Biostrings             2.62.0     2021-10-26 [1] Bioconductor
##  bit                    4.0.4      2020-08-04 [1] CRAN (R 4.1.0)
##  bit64                  4.0.5      2020-08-30 [1] CRAN (R 4.1.0)
##  bitops                 1.0-7      2021-04-24 [1] CRAN (R 4.1.0)
##  blob                   1.2.2      2021-07-23 [1] CRAN (R 4.1.0)
##  bookdown               0.27       2022-06-14 [1] CRAN (R 4.1.2)
##  brio                   1.1.3      2021-11-30 [1] CRAN (R 4.1.0)
##  broom                  0.7.12     2022-01-28 [1] CRAN (R 4.1.2)
##  bslib                  0.3.1      2021-10-06 [1] CRAN (R 4.1.0)
##  cachem                 1.0.6      2021-08-19 [1] CRAN (R 4.1.0)
##  Cairo                  1.5-14     2021-12-21 [1] CRAN (R 4.1.2)
##  callr                  3.7.0      2021-04-20 [1] CRAN (R 4.1.0)
##  car                    3.0-12     2021-11-06 [1] CRAN (R 4.1.0)
##  carData                3.0-5      2022-01-06 [1] CRAN (R 4.1.2)
##  caret                  6.0-92     2022-04-19 [1] CRAN (R 4.1.2)
##  caTools                1.18.2     2021-03-28 [1] CRAN (R 4.1.0)
##  cellranger             1.1.0      2016-07-27 [1] CRAN (R 4.1.0)
##  checkmate              2.0.0      2020-02-06 [1] CRAN (R 4.1.0)
##  circlize               0.4.14     2022-02-11 [1] CRAN (R 4.1.2)
##  class                  7.3-20     2022-01-13 [1] CRAN (R 4.1.2)
##  cli                    3.3.0      2022-04-25 [1] CRAN (R 4.1.2)
##  clue                   0.3-60     2021-10-11 [1] CRAN (R 4.1.0)
##  cluster              * 2.1.2      2021-04-17 [1] CRAN (R 4.1.2)
##  codetools              0.2-18     2020-11-04 [1] CRAN (R 4.1.2)
##  coin                   1.4-2      2021-10-08 [1] CRAN (R 4.1.0)
##  colorspace             2.0-3      2022-02-21 [1] CRAN (R 4.1.2)
##  ComplexHeatmap         2.10.0     2021-10-26 [1] Bioconductor
##  corpcor                1.6.10     2021-09-16 [1] CRAN (R 4.1.0)
##  cowplot                1.1.1      2020-12-30 [1] CRAN (R 4.1.0)
##  crayon                 1.5.0      2022-02-14 [1] CRAN (R 4.1.2)
##  crmn                   0.0.21     2020-02-10 [1] CRAN (R 4.1.0)
##  curl                   4.3.2      2021-06-23 [1] CRAN (R 4.1.0)
##  data.table             1.14.2     2021-09-27 [1] CRAN (R 4.1.0)
##  DBI                    1.1.2      2021-12-20 [1] CRAN (R 4.1.0)
##  DelayedArray           0.20.0     2021-10-26 [1] Bioconductor
##  dendextend           * 1.15.2     2021-10-28 [1] CRAN (R 4.1.0)
##  desc                   1.4.1      2022-03-06 [1] CRAN (R 4.1.2)
##  DESeq2                 1.34.0     2021-10-26 [1] Bioconductor
##  devtools               2.4.3      2021-11-30 [1] CRAN (R 4.1.0)
##  digest                 0.6.29     2021-12-01 [1] CRAN (R 4.1.0)
##  doParallel             1.0.17     2022-02-07 [1] CRAN (R 4.1.2)
##  doSNOW                 1.0.20     2022-02-04 [1] CRAN (R 4.1.2)
##  dplyr                * 1.0.8      2022-02-08 [1] CRAN (R 4.1.2)
##  dynamicTreeCut       * 1.63-1     2016-03-11 [1] CRAN (R 4.1.0)
##  e1071                  1.7-9      2021-09-16 [1] CRAN (R 4.1.0)
##  edgeR                  3.36.0     2021-10-26 [1] Bioconductor
##  ellipse                0.4.2      2020-05-27 [1] CRAN (R 4.1.0)
##  ellipsis               0.3.2      2021-04-29 [1] CRAN (R 4.1.0)
##  evaluate               0.15       2022-02-18 [1] CRAN (R 4.1.2)
##  factoextra           * 1.0.7      2020-04-01 [1] CRAN (R 4.1.0)
##  fansi                  1.0.2      2022-01-14 [1] CRAN (R 4.1.2)
##  farver                 2.1.0      2021-02-28 [1] CRAN (R 4.1.0)
##  fastcluster          * 1.2.3      2021-05-24 [1] CRAN (R 4.1.0)
##  fastmap                1.1.0      2021-01-25 [1] CRAN (R 4.1.0)
##  fastmatch              1.1-3      2021-07-23 [1] CRAN (R 4.1.0)
##  fdrtool                1.2.17     2021-11-13 [1] CRAN (R 4.1.0)
##  fgsea                  1.20.0     2021-10-26 [1] Bioconductor
##  filematrix             1.3        2018-02-27 [1] CRAN (R 4.1.0)
##  foreach                1.5.2      2022-02-02 [1] CRAN (R 4.1.2)
##  foreign                0.8-82     2022-01-13 [1] CRAN (R 4.1.2)
##  forestplot             2.0.1      2021-09-03 [1] CRAN (R 4.1.0)
##  Formula                1.2-4      2020-10-16 [1] CRAN (R 4.1.0)
##  fs                     1.5.2      2021-12-08 [1] CRAN (R 4.1.0)
##  future                 1.24.0     2022-02-19 [1] CRAN (R 4.1.2)
##  future.apply           1.8.1      2021-08-10 [1] CRAN (R 4.1.0)
##  genefilter             1.76.0     2021-10-26 [1] Bioconductor
##  geneplotter            1.72.0     2021-10-26 [1] Bioconductor
##  generics               0.1.2      2022-01-31 [1] CRAN (R 4.1.2)
##  GenomeInfoDb         * 1.30.1     2022-01-30 [1] Bioconductor
##  GenomeInfoDbData       1.2.7      2022-03-09 [1] Bioconductor
##  GenomicRanges        * 1.46.1     2021-11-18 [1] Bioconductor
##  GetoptLong             1.0.5      2020-12-15 [1] CRAN (R 4.1.0)
##  ggforce                0.3.3      2021-03-05 [1] CRAN (R 4.1.0)
##  ggplot2              * 3.3.5      2021-06-25 [1] CRAN (R 4.1.0)
##  ggpubr                 0.4.0      2020-06-27 [1] CRAN (R 4.1.0)
##  ggraph               * 2.0.5      2021-02-23 [1] CRAN (R 4.1.0)
##  ggrepel                0.9.1      2021-01-15 [1] CRAN (R 4.1.0)
##  ggsignif               0.6.3      2021-09-09 [1] CRAN (R 4.1.0)
##  glasso                 1.11       2019-10-01 [1] CRAN (R 4.1.0)
##  glmnet               * 4.1-3      2021-11-02 [1] CRAN (R 4.1.0)
##  GlobalOptions          0.1.2      2020-06-10 [1] CRAN (R 4.1.0)
##  globals                0.14.0     2020-11-22 [1] CRAN (R 4.1.0)
##  globaltest             5.48.0     2021-10-26 [1] Bioconductor
##  glue                 * 1.6.2      2022-02-24 [1] CRAN (R 4.1.2)
##  Gmisc                * 3.0.0      2022-01-03 [1] CRAN (R 4.1.2)
##  gmp                    0.6-5      2022-03-17 [1] CRAN (R 4.1.2)
##  GO.db                  3.14.0     2022-04-11 [1] Bioconductor
##  gower                  1.0.0      2022-02-03 [1] CRAN (R 4.1.2)
##  gplots                 3.1.1      2020-11-28 [1] CRAN (R 4.1.0)
##  graphlayouts           0.8.0      2022-01-03 [1] CRAN (R 4.1.2)
##  gridExtra              2.3        2017-09-09 [1] CRAN (R 4.1.0)
##  gtable                 0.3.0      2019-03-25 [1] CRAN (R 4.1.0)
##  gtools                 3.9.2      2021-06-06 [1] CRAN (R 4.1.0)
##  hardhat                0.2.0      2022-01-24 [1] CRAN (R 4.1.2)
##  highr                  0.9        2021-04-16 [1] CRAN (R 4.1.0)
##  Hmisc                  4.6-0      2021-10-07 [1] CRAN (R 4.1.0)
##  htmlTable            * 2.4.0      2022-01-04 [1] CRAN (R 4.1.2)
##  htmltools              0.5.2      2021-08-25 [1] CRAN (R 4.1.0)
##  htmlwidgets            1.5.4      2021-09-08 [1] CRAN (R 4.1.0)
##  httr                 * 1.4.2      2020-07-20 [1] CRAN (R 4.1.0)
##  huge                   1.3.5      2021-06-30 [1] CRAN (R 4.1.0)
##  igraph               * 1.2.11     2022-01-04 [1] CRAN (R 4.1.2)
##  impute                 1.68.0     2021-10-26 [1] Bioconductor
##  ipred                  0.9-12     2021-09-15 [1] CRAN (R 4.1.0)
##  IRanges              * 2.28.0     2021-10-26 [1] Bioconductor
##  irlba                  2.3.5      2021-12-06 [1] CRAN (R 4.1.0)
##  iterators              1.0.14     2022-02-05 [1] CRAN (R 4.1.2)
##  jpeg                   0.1-9      2021-07-24 [1] CRAN (R 4.1.0)
##  jquerylib              0.1.4      2021-04-26 [1] CRAN (R 4.1.0)
##  jsonlite               1.8.0      2022-02-22 [1] CRAN (R 4.1.2)
##  KEGGREST               1.34.0     2021-10-26 [1] Bioconductor
##  KernSmooth             2.23-20    2021-05-03 [1] CRAN (R 4.1.2)
##  knitr                  1.39       2022-04-26 [1] CRAN (R 4.1.2)
##  labeling               0.4.2      2020-10-20 [1] CRAN (R 4.1.0)
##  lattice                0.20-45    2021-09-22 [1] CRAN (R 4.1.2)
##  latticeExtra           0.6-29     2019-12-19 [1] CRAN (R 4.1.0)
##  lava                   1.6.10     2021-09-02 [1] CRAN (R 4.1.0)
##  lavaan                 0.6-11     2022-03-31 [1] CRAN (R 4.1.2)
##  lazyeval               0.2.2      2019-03-15 [1] CRAN (R 4.1.0)
##  libcoin                1.0-9      2021-09-27 [1] CRAN (R 4.1.0)
##  lifecycle              1.0.1      2021-09-24 [1] CRAN (R 4.1.0)
##  limma                  3.50.1     2022-02-17 [1] Bioconductor
##  listenv                0.8.0      2019-12-05 [1] CRAN (R 4.1.0)
##  locfit                 1.5-9.5    2022-03-03 [1] CRAN (R 4.1.2)
##  lubridate              1.8.0      2021-10-07 [1] CRAN (R 4.1.0)
##  magrittr             * 2.0.2      2022-01-26 [1] CRAN (R 4.1.2)
##  MASS                   7.3-55     2022-01-13 [1] CRAN (R 4.1.2)
##  Matrix               * 1.4-0      2021-12-08 [1] CRAN (R 4.1.0)
##  MatrixGenerics       * 1.6.0      2021-10-26 [1] Bioconductor
##  matrixStats          * 0.61.0     2021-09-17 [1] CRAN (R 4.1.0)
##  memoise                2.0.1      2021-11-26 [1] CRAN (R 4.1.0)
##  MetaboAnalystR       * 3.2.0      2022-06-28 [1] Github (xia-lab/MetaboAnalystR@892a31b)
##  metagenomeSeq          1.36.0     2021-10-26 [1] Bioconductor
##  mgcv                   1.8-39     2022-02-24 [1] CRAN (R 4.1.2)
##  mixedCCA               1.5.2      2022-07-14 [1] Github (irinagain/mixedCCA@c6d41a3)
##  mixOmics               6.18.1     2021-11-18 [1] Bioconductor (R 4.1.2)
##  mnormt                 2.0.2      2020-09-01 [1] CRAN (R 4.1.0)
##  ModelMetrics           1.2.2.2    2020-03-17 [1] CRAN (R 4.1.0)
##  modeltools             0.2-23     2020-03-05 [1] CRAN (R 4.1.0)
##  multcomp               1.4-18     2022-01-04 [1] CRAN (R 4.1.2)
##  multtest               2.50.0     2021-10-26 [1] Bioconductor
##  munsell                0.5.0      2018-06-12 [1] CRAN (R 4.1.0)
##  mvtnorm                1.1-3      2021-10-08 [1] CRAN (R 4.1.0)
##  NADA                   1.6-1.1    2020-03-22 [1] CRAN (R 4.1.0)
##  NetCoMi              * 1.0.3      2022-07-14 [1] Github (stefpeschel/NetCoMi@d4d80d3)
##  nlme                   3.1-155    2022-01-13 [1] CRAN (R 4.1.2)
##  nnet                   7.3-17     2022-01-13 [1] CRAN (R 4.1.2)
##  parallelly             1.31.0     2022-04-07 [1] CRAN (R 4.1.2)
##  pbapply                1.5-0      2021-09-16 [1] CRAN (R 4.1.0)
##  pbivnorm               0.6.0      2015-01-23 [1] CRAN (R 4.1.0)
##  pcaMethods             1.86.0     2021-10-26 [1] Bioconductor
##  pcaPP                  1.9-74     2021-04-23 [1] CRAN (R 4.1.0)
##  permute                0.9-7      2022-01-27 [1] CRAN (R 4.1.2)
##  pheatmap               1.0.12     2019-01-04 [1] CRAN (R 4.1.0)
##  phyloseq               1.38.0     2021-10-26 [1] Bioconductor
##  pillar                 1.7.0      2022-02-01 [1] CRAN (R 4.1.2)
##  pkgbuild               1.3.1      2021-12-20 [1] CRAN (R 4.1.0)
##  pkgconfig              2.0.3      2019-09-22 [1] CRAN (R 4.1.0)
##  pkgload                1.2.4      2021-11-30 [1] CRAN (R 4.1.0)
##  plotly               * 4.10.0     2021-10-09 [1] CRAN (R 4.1.0)
##  plyr                   1.8.6      2020-03-03 [1] CRAN (R 4.1.0)
##  png                    0.1-7      2013-12-03 [1] CRAN (R 4.1.0)
##  polyclip               1.10-0     2019-03-14 [1] CRAN (R 4.1.0)
##  POMA                 * 1.7.2      2022-07-26 [1] Github (pcastellanoescuder/POMA@bc8a972)
##  preprocessCore         1.56.0     2021-10-26 [1] Bioconductor
##  prettyunits            1.1.1      2020-01-24 [1] CRAN (R 4.1.0)
##  pROC                   1.18.0     2021-09-03 [1] CRAN (R 4.1.0)
##  processx               3.5.2      2021-04-30 [1] CRAN (R 4.1.0)
##  prodlim                2019.11.13 2019-11-17 [1] CRAN (R 4.1.0)
##  proxy                  0.4-26     2021-06-07 [1] CRAN (R 4.1.0)
##  ps                     1.6.0      2021-02-28 [1] CRAN (R 4.1.0)
##  psych                  2.2.5      2022-05-10 [1] CRAN (R 4.1.2)
##  pulsar                 0.3.7      2020-08-07 [1] CRAN (R 4.1.0)
##  purrr                  0.3.4      2020-04-17 [1] CRAN (R 4.1.0)
##  qgraph                 1.9.2      2022-03-04 [1] CRAN (R 4.1.2)
##  qs                     0.25.3     2022-02-22 [1] CRAN (R 4.1.2)
##  R6                     2.5.1      2021-08-19 [1] CRAN (R 4.1.0)
##  ragg                   1.2.2      2022-02-21 [1] CRAN (R 4.1.2)
##  randomForest           4.7-1      2022-02-03 [1] CRAN (R 4.1.2)
##  RankProd               3.20.0     2021-10-26 [1] Bioconductor
##  RApiSerialize          0.1.0      2014-04-19 [1] CRAN (R 4.1.0)
##  rARPACK                0.11-0     2016-03-10 [1] CRAN (R 4.1.0)
##  rbibutils              2.2.7      2021-12-07 [1] CRAN (R 4.1.0)
##  RColorBrewer           1.1-2      2014-12-07 [1] CRAN (R 4.1.0)
##  Rcpp                 * 1.0.8.2    2022-03-11 [1] CRAN (R 4.1.2)
##  RcppParallel           5.1.5      2022-01-05 [1] CRAN (R 4.1.2)
##  RCurl                  1.98-1.6   2022-02-08 [1] CRAN (R 4.1.2)
##  Rdpack                 2.2        2022-03-19 [1] CRAN (R 4.1.2)
##  readxl               * 1.4.0      2022-03-28 [1] CRAN (R 4.1.2)
##  recipes                0.2.0      2022-02-18 [1] CRAN (R 4.1.2)
##  remotes                2.4.2      2021-11-30 [1] CRAN (R 4.1.0)
##  reshape2               1.4.4      2020-04-09 [1] CRAN (R 4.1.0)
##  rhdf5                  2.38.1     2022-03-10 [1] Bioconductor
##  rhdf5filters           1.6.0      2021-10-26 [1] Bioconductor
##  Rhdf5lib               1.16.0     2021-10-26 [1] Bioconductor
##  rjson                  0.2.21     2022-01-09 [1] CRAN (R 4.1.2)
##  rlang                  1.0.2      2022-03-04 [1] CRAN (R 4.1.2)
##  rmarkdown              2.14       2022-04-25 [1] CRAN (R 4.1.2)
##  Rmpfr                  0.8-9      2022-06-03 [1] CRAN (R 4.1.2)
##  rootSolve              1.8.2.3    2021-09-29 [1] CRAN (R 4.1.0)
##  rpart                  4.1.16     2022-01-24 [1] CRAN (R 4.1.2)
##  rprojroot              2.0.2      2020-11-15 [1] CRAN (R 4.1.0)
##  Rserve               * 1.8-10     2021-11-25 [1] CRAN (R 4.1.0)
##  RSpectra               0.16-0     2019-12-01 [1] CRAN (R 4.1.0)
##  RSQLite                2.2.10     2022-02-17 [1] CRAN (R 4.1.2)
##  rstatix                0.7.0      2021-02-13 [1] CRAN (R 4.1.0)
##  rstudioapi             0.13       2020-11-12 [1] CRAN (R 4.1.0)
##  S4Vectors            * 0.32.3     2021-11-21 [1] Bioconductor
##  sandwich               3.0-1      2021-05-18 [1] CRAN (R 4.1.0)
##  sass                   0.4.0      2021-05-12 [1] CRAN (R 4.1.0)
##  scales                 1.1.1      2020-05-11 [1] CRAN (R 4.1.0)
##  scrime                 1.3.5      2018-12-01 [1] CRAN (R 4.1.0)
##  sessioninfo            1.2.2      2021-12-06 [1] CRAN (R 4.1.0)
##  shape                  1.4.6      2021-05-19 [1] CRAN (R 4.1.0)
##  siggenes               1.68.0     2021-10-26 [1] Bioconductor
##  snow                   0.4-4      2021-10-27 [1] CRAN (R 4.1.0)
##  SpiecEasi            * 1.1.2      2022-07-14 [1] Github (zdk123/SpiecEasi@c463727)
##  SPRING               * 1.0.4      2022-08-03 [1] Github (GraceYoon/SPRING@3d641a4)
##  stringfish             0.15.5     2021-12-01 [1] CRAN (R 4.1.0)
##  stringi                1.7.6      2021-11-29 [1] CRAN (R 4.1.0)
##  stringr                1.4.0      2019-02-10 [1] CRAN (R 4.1.0)
##  SummarizedExperiment * 1.24.0     2021-10-26 [1] Bioconductor
##  survival               3.3-1      2022-03-03 [1] CRAN (R 4.1.2)
##  systemfonts            1.0.4      2022-02-11 [1] CRAN (R 4.1.2)
##  testthat               3.1.2      2022-01-20 [1] CRAN (R 4.1.2)
##  textshaping            0.3.6      2021-10-13 [1] CRAN (R 4.1.0)
##  TH.data                1.1-0      2021-09-27 [1] CRAN (R 4.1.0)
##  tibble               * 3.1.6      2021-11-07 [1] CRAN (R 4.1.0)
##  tidygraph              1.2.1      2022-04-05 [1] CRAN (R 4.1.2)
##  tidyr                  1.2.0      2022-02-01 [1] CRAN (R 4.1.2)
##  tidyselect             1.1.2      2022-02-21 [1] CRAN (R 4.1.2)
##  timeDate               3043.102   2018-02-21 [1] CRAN (R 4.1.0)
##  tmvnsim                1.0-2      2016-12-15 [1] CRAN (R 4.1.0)
##  truncnorm              1.0-8      2018-02-27 [1] CRAN (R 4.1.0)
##  tweenr                 1.0.2      2021-03-23 [1] CRAN (R 4.1.0)
##  usethis                2.1.5      2021-12-09 [1] CRAN (R 4.1.0)
##  utf8                   1.2.2      2021-07-24 [1] CRAN (R 4.1.0)
##  vctrs                  0.3.8      2021-04-29 [1] CRAN (R 4.1.0)
##  vegan                  2.5-7      2020-11-28 [1] CRAN (R 4.1.0)
##  VGAM                   1.1-6      2022-02-14 [1] CRAN (R 4.1.2)
##  viridis                0.6.2      2021-10-13 [1] CRAN (R 4.1.0)
##  viridisLite            0.4.0      2021-04-13 [1] CRAN (R 4.1.0)
##  WGCNA                * 1.71       2022-04-22 [1] CRAN (R 4.1.2)
##  withr                  2.5.0      2022-03-03 [1] CRAN (R 4.1.2)
##  Wrench                 1.12.0     2021-10-26 [1] Bioconductor
##  xfun                   0.30       2022-03-02 [1] CRAN (R 4.1.2)
##  XMAS2                  2.1.7.4    2022-08-09 [1] local
##  XML                    3.99-0.9   2022-02-24 [1] CRAN (R 4.1.2)
##  xtable                 1.8-4      2019-04-21 [1] CRAN (R 4.1.0)
##  XVector                0.34.0     2021-10-26 [1] Bioconductor
##  yaml                   2.3.5      2022-02-21 [1] CRAN (R 4.1.2)
##  zCompositions          1.4.0      2022-01-13 [1] CRAN (R 4.1.2)
##  zlibbioc               1.40.0     2021-10-26 [1] Bioconductor
##  zoo                    1.8-9      2021-03-09 [1] CRAN (R 4.1.0)
## 
##  [1] /Library/Frameworks/R.framework/Versions/4.1/Resources/library
## 
## ────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="enrichment-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="example.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/Metabolomics_Aanlysis/blob/main/08-Pathway_Analysis.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection",
"scroll_highlight": true,
"toc_depth": 3
}
});
});
</script>

</body>

</html>
