<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Statistical Analysis | Functional Analysis</title>
  <meta name="description" content="A template for Data Analysis in Metabolomics" />
  <meta name="generator" content="bookdown 0.27 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Statistical Analysis | Functional Analysis" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A template for Data Analysis in Metabolomics" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Statistical Analysis | Functional Analysis" />
  
  <meta name="twitter:description" content="A template for Data Analysis in Metabolomics" />
  

<meta name="author" content="BangzhuoTong" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="functional-analysis-1.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="assets/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Data Analysis in Metabolomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#what-is-metabolomics"><i class="fa fa-check"></i><b>1.1</b> What is metabolomics?</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#workflow-of-metabolomics-data-analysis"><i class="fa fa-check"></i><b>1.2</b> Workflow of metabolomics data analysis</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#statistical-analysis"><i class="fa fa-check"></i><b>1.2.1</b> Statistical Analysis</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#functional-analysis"><i class="fa fa-check"></i><b>1.2.2</b> Functional Analysis</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#software"><i class="fa fa-check"></i><b>1.3</b> Software</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#reference"><i class="fa fa-check"></i><b>1.4</b> Reference</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html"><i class="fa fa-check"></i><b>2</b> Statistical Analysis</a>
<ul>
<li class="chapter" data-level="2.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-preprocessing"><i class="fa fa-check"></i><b>2.1</b> Data Preprocessing</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#environment-setup"><i class="fa fa-check"></i><b>2.1.1</b> Environment setup</a></li>
<li class="chapter" data-level="2.1.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-data"><i class="fa fa-check"></i><b>2.1.2</b> Loading data</a></li>
<li class="chapter" data-level="2.1.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-checking"><i class="fa fa-check"></i><b>2.1.3</b> Data Checking</a></li>
<li class="chapter" data-level="2.1.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-filtering"><i class="fa fa-check"></i><b>2.1.4</b> Data Filtering</a></li>
<li class="chapter" data-level="2.1.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#missing-value-imputation"><i class="fa fa-check"></i><b>2.1.5</b> Missing Value Imputation</a></li>
<li class="chapter" data-level="2.1.6" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-normalization"><i class="fa fa-check"></i><b>2.1.6</b> Data Normalization</a></li>
<li class="chapter" data-level="2.1.7" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#removing-outliers"><i class="fa fa-check"></i><b>2.1.7</b> Removing outliers</a></li>
<li class="chapter" data-level="2.1.8" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#saving-datasets-into-rds-files"><i class="fa fa-check"></i><b>2.1.8</b> Saving datasets into RDS files</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#cluster-analysis"><i class="fa fa-check"></i><b>2.2</b> Cluster Analysis</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-packages"><i class="fa fa-check"></i><b>2.2.1</b> Loading packages</a></li>
<li class="chapter" data-level="2.2.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#importing-data"><i class="fa fa-check"></i><b>2.2.2</b> Importing data</a></li>
<li class="chapter" data-level="2.2.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#hierarchical-clustering"><i class="fa fa-check"></i><b>2.2.3</b> Hierarchical Clustering</a></li>
<li class="chapter" data-level="2.2.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#partitional-clustering"><i class="fa fa-check"></i><b>2.2.4</b> Partitional Clustering</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#chemometrics-analysis"><i class="fa fa-check"></i><b>2.3</b> Chemometrics Analysis</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-packages-1"><i class="fa fa-check"></i><b>2.3.1</b> Loading packages</a></li>
<li class="chapter" data-level="2.3.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#importing-data-1"><i class="fa fa-check"></i><b>2.3.2</b> Importing data</a></li>
<li class="chapter" data-level="2.3.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>2.3.3</b> Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="2.3.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#partial-least-squares-discriminant-analysis-pls-da"><i class="fa fa-check"></i><b>2.3.4</b> Partial Least Squares-Discriminant Analysis (PLS-DA)</a></li>
<li class="chapter" data-level="2.3.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#sparse-partial-least-squares-discriminant-analysis-spls-da"><i class="fa fa-check"></i><b>2.3.5</b> Sparse Partial Least Squares-Discriminant Analysis (sPLS-DA)</a></li>
<li class="chapter" data-level="2.3.6" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#orthogonal-partial-least-squares-discriminant-analysis-orthopls-da"><i class="fa fa-check"></i><b>2.3.6</b> Orthogonal Partial Least Squares-Discriminant Analysis (orthoPLS-DA)</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#univariate-analysis"><i class="fa fa-check"></i><b>2.4</b> Univariate Analysis</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-packages-2"><i class="fa fa-check"></i><b>2.4.1</b> Loading packages</a></li>
<li class="chapter" data-level="2.4.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#importing-data-2"><i class="fa fa-check"></i><b>2.4.2</b> Importing data</a></li>
<li class="chapter" data-level="2.4.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#fold-change-analysis"><i class="fa fa-check"></i><b>2.4.3</b> Fold Change Analysis</a></li>
<li class="chapter" data-level="2.4.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#t-test"><i class="fa fa-check"></i><b>2.4.4</b> T Test</a></li>
<li class="chapter" data-level="2.4.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#wilcoxon-test"><i class="fa fa-check"></i><b>2.4.5</b> Wilcoxon Test</a></li>
<li class="chapter" data-level="2.4.6" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#limma-test"><i class="fa fa-check"></i><b>2.4.6</b> Limma Test</a></li>
<li class="chapter" data-level="2.4.7" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#volcano-plot"><i class="fa fa-check"></i><b>2.4.7</b> Volcano plot</a></li>
<li class="chapter" data-level="2.4.8" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#correlation-heatmaps"><i class="fa fa-check"></i><b>2.4.8</b> Correlation Heatmaps</a></li>
<li class="chapter" data-level="2.4.9" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#glasso-this-function-will-compute-a-gaussian-graphical-model-using-the-glmnet-package"><i class="fa fa-check"></i><b>2.4.9</b> glasso: this function will compute a Gaussian Graphical Model using the glmnet package</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#multivariate-analysis"><i class="fa fa-check"></i><b>2.5</b> Multivariate analysis</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-packages-3"><i class="fa fa-check"></i><b>2.5.1</b> Loading packages</a></li>
<li class="chapter" data-level="2.5.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#importing-data-3"><i class="fa fa-check"></i><b>2.5.2</b> Importing data</a></li>
<li class="chapter" data-level="2.5.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-curation"><i class="fa fa-check"></i><b>2.5.3</b> Data curation</a></li>
<li class="chapter" data-level="2.5.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#regularized-generalized-linear-models"><i class="fa fa-check"></i><b>2.5.4</b> Regularized Generalized Linear Models</a></li>
<li class="chapter" data-level="2.5.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#classification"><i class="fa fa-check"></i><b>2.5.5</b> Classification</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#network-analysis"><i class="fa fa-check"></i><b>2.6</b> Network Analysis</a>
<ul>
<li class="chapter" data-level="2.6.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#introduction-1"><i class="fa fa-check"></i><b>2.6.1</b> Introduction</a></li>
<li class="chapter" data-level="2.6.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-packages-4"><i class="fa fa-check"></i><b>2.6.2</b> Loading packages</a></li>
<li class="chapter" data-level="2.6.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#importing-data-4"><i class="fa fa-check"></i><b>2.6.3</b> Importing data</a></li>
<li class="chapter" data-level="2.6.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-curation-1"><i class="fa fa-check"></i><b>2.6.4</b> Data curation</a></li>
<li class="chapter" data-level="2.6.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#associations-among-features"><i class="fa fa-check"></i><b>2.6.5</b> Associations Among Features</a></li>
<li class="chapter" data-level="2.6.6" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#network-comparison"><i class="fa fa-check"></i><b>2.6.6</b> Network comparison</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#systematic-information"><i class="fa fa-check"></i><b>2.7</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html"><i class="fa fa-check"></i><b>3</b> Functional Analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#enrichment-analysis"><i class="fa fa-check"></i><b>3.1</b> Enrichment Analysis</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#enviroment-setup"><i class="fa fa-check"></i><b>3.1.1</b> Enviroment Setup</a></li>
<li class="chapter" data-level="3.1.2" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#over-representation-analysis-ora"><i class="fa fa-check"></i><b>3.1.2</b> Over representation analysis (ORA)</a></li>
<li class="chapter" data-level="3.1.3" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#clean-environment-1"><i class="fa fa-check"></i><b>3.1.3</b> Clean environment 1</a></li>
<li class="chapter" data-level="3.1.4" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#single-sample-profiling"><i class="fa fa-check"></i><b>3.1.4</b> Single Sample Profiling</a></li>
<li class="chapter" data-level="3.1.5" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#clean-environment-2"><i class="fa fa-check"></i><b>3.1.5</b> Clean environment 2</a></li>
<li class="chapter" data-level="3.1.6" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#quantitative-enrichment-analysis-qea"><i class="fa fa-check"></i><b>3.1.6</b> Quantitative Enrichment Analysis (QEA)</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#pathway-analysis"><i class="fa fa-check"></i><b>3.2</b> Pathway Analysis</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#enviroment-set-up"><i class="fa fa-check"></i><b>3.2.1</b> Enviroment Set up</a></li>
<li class="chapter" data-level="3.2.2" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#over-representation-analysis"><i class="fa fa-check"></i><b>3.2.2</b> Over representation analysis</a></li>
<li class="chapter" data-level="3.2.3" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#clean-environment-3"><i class="fa fa-check"></i><b>3.2.3</b> Clean environment 3</a></li>
<li class="chapter" data-level="3.2.4" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#concentration-table-qea"><i class="fa fa-check"></i><b>3.2.4</b> Concentration Table (QEA)</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#session-info"><i class="fa fa-check"></i><b>3.3</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="example.html"><a href="example.html"><i class="fa fa-check"></i><b>4</b> Example</a>
<ul>
<li class="chapter" data-level="4.1" data-path="example.html"><a href="example.html#loading-packages-5"><i class="fa fa-check"></i><b>4.1</b> Loading packages</a></li>
<li class="chapter" data-level="4.2" data-path="example.html"><a href="example.html#importing-data-5"><i class="fa fa-check"></i><b>4.2</b> Importing data</a></li>
<li class="chapter" data-level="4.3" data-path="example.html"><a href="example.html#data-processing"><i class="fa fa-check"></i><b>4.3</b> Data Processing</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="example.html"><a href="example.html#data-checking-1"><i class="fa fa-check"></i><b>4.3.1</b> Data Checking</a></li>
<li class="chapter" data-level="4.3.2" data-path="example.html"><a href="example.html#data-filtering-1"><i class="fa fa-check"></i><b>4.3.2</b> Data Filtering</a></li>
<li class="chapter" data-level="4.3.3" data-path="example.html"><a href="example.html#missing-value-imputation-1"><i class="fa fa-check"></i><b>4.3.3</b> Missing Value Imputation</a></li>
<li class="chapter" data-level="4.3.4" data-path="example.html"><a href="example.html#data-normalization-1"><i class="fa fa-check"></i><b>4.3.4</b> Data Normalization</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="example.html"><a href="example.html#cluster-analysis-1"><i class="fa fa-check"></i><b>4.4</b> Cluster Analysis</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="example.html"><a href="example.html#hierarchical-clustering-1"><i class="fa fa-check"></i><b>4.4.1</b> Hierarchical Clustering</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="example.html"><a href="example.html#chemometrics-analysis-1"><i class="fa fa-check"></i><b>4.5</b> Chemometrics Analysis</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="example.html"><a href="example.html#partial-least-squares-discriminant-analysis-pls-da-1"><i class="fa fa-check"></i><b>4.5.1</b> Partial Least Squares-Discriminant Analysis (PLS-DA)</a></li>
<li class="chapter" data-level="4.5.2" data-path="example.html"><a href="example.html#sparse-partial-least-squares-discriminant-analysis-spls-da-1"><i class="fa fa-check"></i><b>4.5.2</b> Sparse Partial Least Squares-Discriminant Analysis (sPLS-DA)</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="example.html"><a href="example.html#univariate-analysis-1"><i class="fa fa-check"></i><b>4.6</b> Univariate Analysis</a>
<ul>
<li class="chapter" data-level="4.6.1" data-path="example.html"><a href="example.html#fold-change-analysis-1"><i class="fa fa-check"></i><b>4.6.1</b> Fold Change Analysis</a></li>
<li class="chapter" data-level="4.6.2" data-path="example.html"><a href="example.html#t-test-1"><i class="fa fa-check"></i><b>4.6.2</b> T Test</a></li>
<li class="chapter" data-level="4.6.3" data-path="example.html"><a href="example.html#volcano-plot-1"><i class="fa fa-check"></i><b>4.6.3</b> Volcano plot</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="example.html"><a href="example.html#feature-selection"><i class="fa fa-check"></i><b>4.7</b> Feature Selection</a>
<ul>
<li class="chapter" data-level="4.7.1" data-path="example.html"><a href="example.html#regularized-generalized-linear-models-lasso-alpha-1"><i class="fa fa-check"></i><b>4.7.1</b> Regularized Generalized Linear Models (Lasso: alpha = 1)</a></li>
<li class="chapter" data-level="4.7.2" data-path="example.html"><a href="example.html#classification-random-forest"><i class="fa fa-check"></i><b>4.7.2</b> Classification (Random Forest)</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="example.html"><a href="example.html#network-analysis-2"><i class="fa fa-check"></i><b>4.8</b> Network Analysis</a>
<ul>
<li class="chapter" data-level="4.8.1" data-path="example.html"><a href="example.html#data-curation-2"><i class="fa fa-check"></i><b>4.8.1</b> Data curation</a></li>
<li class="chapter" data-level="4.8.2" data-path="example.html"><a href="example.html#building-network-model-4"><i class="fa fa-check"></i><b>4.8.2</b> Building network model</a></li>
<li class="chapter" data-level="4.8.3" data-path="example.html"><a href="example.html#visualizing-the-network-4"><i class="fa fa-check"></i><b>4.8.3</b> Visualizing the network</a></li>
</ul></li>
<li class="chapter" data-level="4.9" data-path="example.html"><a href="example.html#network-analysis-by-wgcna"><i class="fa fa-check"></i><b>4.9</b> Network Analysis by WGCNA</a>
<ul>
<li class="chapter" data-level="4.9.1" data-path="example.html"><a href="example.html#data-curation-3"><i class="fa fa-check"></i><b>4.9.1</b> Data curation</a></li>
<li class="chapter" data-level="4.9.2" data-path="example.html"><a href="example.html#tuning-soft-thresholds"><i class="fa fa-check"></i><b>4.9.2</b> Tuning soft thresholds</a></li>
<li class="chapter" data-level="4.9.3" data-path="example.html"><a href="example.html#create-the-network-using-the-blockwisemodules"><i class="fa fa-check"></i><b>4.9.3</b> Create the network using the <code>blockwiseModules</code></a></li>
<li class="chapter" data-level="4.9.4" data-path="example.html"><a href="example.html#plot-modules"><i class="fa fa-check"></i><b>4.9.4</b> plot modules</a></li>
<li class="chapter" data-level="4.9.5" data-path="example.html"><a href="example.html#relationships-among-modules"><i class="fa fa-check"></i><b>4.9.5</b> Relationships among modules</a></li>
<li class="chapter" data-level="4.9.6" data-path="example.html"><a href="example.html#module-eigengene-correlation"><i class="fa fa-check"></i><b>4.9.6</b> Module (Eigengene) correlation</a></li>
<li class="chapter" data-level="4.9.7" data-path="example.html"><a href="example.html#relate-module-cluster-assignments-to-groups"><i class="fa fa-check"></i><b>4.9.7</b> Relate Module (cluster) Assignments to Groups</a></li>
<li class="chapter" data-level="4.9.8" data-path="example.html"><a href="example.html#generate-and-export-networks"><i class="fa fa-check"></i><b>4.9.8</b> Generate and Export Networks</a></li>
<li class="chapter" data-level="4.9.9" data-path="example.html"><a href="example.html#network-visualization"><i class="fa fa-check"></i><b>4.9.9</b> Network visualization</a></li>
</ul></li>
<li class="chapter" data-level="4.10" data-path="example.html"><a href="example.html#systematic-information-1"><i class="fa fa-check"></i><b>4.10</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Functional Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="statistical-analysis-1" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Statistical Analysis<a href="statistical-analysis-1.html#statistical-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Although the horrible experience of data analysis by using <strong>MetaboAnalystR</strong> R package <span class="citation">(<a href="#ref-pang2020metaboanalystr" role="doc-biblioref">Pang et al. 2020</a>)</span>, its thought of data processing are very useful. Therefore, this template is based on the workflow from <strong>MetaboAnalystR</strong>.</p>
<p><strong>In this chapter, very detailed explaination of the available methods in each step in Statistical Analysis would be introduced. Users can nevertheless go through the whole analysis from with example data in Chapter 4.</strong></p>
<hr />
<div id="data-preprocessing" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Data Preprocessing<a href="statistical-analysis-1.html#data-preprocessing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We integrated R packages and our own scripts to build the data analysis template for metabolomics data. Particularly, we thanks very much for <strong>POMA</strong> R package <span class="citation">(<a href="#ref-castellano2021pomashiny" role="doc-biblioref">Castellano-Escuder et al. 2021</a>)</span>. <strong>POMA</strong> is a flexible data cleaning and statistical analysis processes in one comprehensible and user-friendly R package.</p>
<p><strong>Note: Please remember to preprocess your data before <a href="statistical-analysis-1.html#cluster-analysis">Cluster Analysis</a> and other steps below.</strong></p>
<div id="environment-setup" class="section level3 hasAnchor" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> Environment setup<a href="statistical-analysis-1.html#environment-setup" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="statistical-analysis-1.html#cb1-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">warning =</span> F)</span>
<span id="cb1-2"><a href="statistical-analysis-1.html#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="statistical-analysis-1.html#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-4"><a href="statistical-analysis-1.html#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb1-5"><a href="statistical-analysis-1.html#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(POMA)</span>
<span id="cb1-6"><a href="statistical-analysis-1.html#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-7"><a href="statistical-analysis-1.html#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggraph)</span>
<span id="cb1-8"><a href="statistical-analysis-1.html#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb1-9"><a href="statistical-analysis-1.html#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb1-10"><a href="statistical-analysis-1.html#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb1-11"><a href="statistical-analysis-1.html#cb1-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-12"><a href="statistical-analysis-1.html#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="co"># rm(list = ls())</span></span>
<span id="cb1-13"><a href="statistical-analysis-1.html#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb1-14"><a href="statistical-analysis-1.html#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">future.globals.maxSize =</span> <span class="dv">1000</span> <span class="sc">*</span> <span class="dv">1024</span><span class="sc">^</span><span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="loading-data" class="section level3 hasAnchor" number="2.1.2">
<h3><span class="header-section-number">2.1.2</span> Loading data<a href="statistical-analysis-1.html#loading-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The dataset is from the Zeybel-2022 published paper <span class="citation">(<a href="#ref-zeybel2022multiomics" role="doc-biblioref">Zeybel et al. 2022</a>)</span>.</p>
<ul>
<li>features table</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="statistical-analysis-1.html#cb2-1" aria-hidden="true" tabindex="-1"></a>profile <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(<span class="st">&quot;./dataset/OmicsDataSet-Zeybel-2022.xlsx&quot;</span>, <span class="at">sheet =</span> <span class="dv">6</span>)</span>
<span id="cb2-2"><a href="statistical-analysis-1.html#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(profile)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 67
##   BIOCHEMICAL        `SUPER PATHWAY` `SUB PATHWAY` `COMP ID` PLATFORM `CHEMICAL ID`    RI  MASS PUBCHEM CAS   KEGG  `SampleID HMDB…` P101001 P101012 P101030 P101031 P101050 P101059 P101071 P101072 P101084
##   &lt;chr&gt;              &lt;chr&gt;           &lt;chr&gt;             &lt;dbl&gt; &lt;chr&gt;            &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;              &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;
## 1 (14 or 15)-methyl… Lipid           Fatty Acid, …     38768 LC/MS N…     100002945  5695  269. 8181;1… &lt;NA&gt;  C169… HMDB0061859       5.11e7  5.12e7  3.84e7  5.91e7  2.20e7  2.26e7  3.24e7  3.42e7  2.91e7
## 2 (16 or 17)-methyl… Lipid           Fatty Acid, …     38296 LC/MS N…     100002356  5993  297. 3083779 2724… &lt;NA&gt;  HMDB0037397       5.11e6  6.00e6  2.86e6  4.72e6  2.90e6  2.20e6  3.00e6  4.68e6  3.65e6
## 3 (2 or 3)-decenoat… Lipid           Medium Chain…     63436 LC/MS N…     100021502  4990  169. &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;  &lt;NA&gt;              7.57e5  5.98e5  3.67e5  5.12e5  3.17e5  7.35e5  5.97e5  3.17e5  4.97e5
## 4 (2,4 or 2,5)-dime… Xenobiotics     Food Compone…     62533 LC/MS N…     100020519  3474  201. &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;  &lt;NA&gt;             NA      NA       5.64e5  7.38e4  1.14e5 NA      NA       6.80e5 NA     
## 5 (N(1) + N(8))-ace… Amino Acid      Polyamine Me…     57814 LC/MS P…     100016038  3080  188. 123689… &lt;NA&gt;  C006… HMDB0001276,HMD…  2.82e5  2.49e5  2.31e5  3.15e5  2.54e5  1.82e5  4.12e5  3.36e5  2.28e5
## 6 (R)-3-hydroxybuty… Lipid           Fatty Acid M…     43264 LC/MS P…     100003926  2400  248. 534816… &lt;NA&gt;  &lt;NA&gt;  HMDB0013127      NA      NA      NA      NA       1.27e5 NA      NA       1.23e5  2.01e5
## # … with 46 more variables: P101003 &lt;dbl&gt;, P101004 &lt;dbl&gt;, P101013 &lt;dbl&gt;, P101016 &lt;dbl&gt;, P101017 &lt;dbl&gt;, P101038 &lt;dbl&gt;, P101051 &lt;dbl&gt;, P101061 &lt;dbl&gt;, P101062 &lt;dbl&gt;, P101074 &lt;dbl&gt;, P101075 &lt;dbl&gt;,
## #   P101076 &lt;dbl&gt;, P101085 &lt;dbl&gt;, P101088 &lt;dbl&gt;, P101007 &lt;dbl&gt;, P101018 &lt;dbl&gt;, P101019 &lt;dbl&gt;, P101041 &lt;dbl&gt;, P101052 &lt;dbl&gt;, P101064 &lt;dbl&gt;, P101065 &lt;dbl&gt;, P101077 &lt;dbl&gt;, P101090 &lt;dbl&gt;, P101094 &lt;dbl&gt;,
## #   P101009 &lt;dbl&gt;, P101010 &lt;dbl&gt;, P101021 &lt;dbl&gt;, P101022 &lt;dbl&gt;, P101042 &lt;dbl&gt;, P101054 &lt;dbl&gt;, P101056 &lt;dbl&gt;, P101067 &lt;dbl&gt;, P101068 &lt;dbl&gt;, P101079 &lt;dbl&gt;, P101095 &lt;dbl&gt;, P101096 &lt;dbl&gt;, P101011 &lt;dbl&gt;,
## #   P101024 &lt;dbl&gt;, P101025 &lt;dbl&gt;, P101027 &lt;dbl&gt;, P101047 &lt;dbl&gt;, P101057 &lt;dbl&gt;, P101069 &lt;dbl&gt;, P101080 &lt;dbl&gt;, P101081 &lt;dbl&gt;, P101082 &lt;dbl&gt;</code></pre>
<ul>
<li>metadata table</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="statistical-analysis-1.html#cb4-1" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(<span class="st">&quot;./dataset/OmicsDataSet-Zeybel-2022.xlsx&quot;</span>, <span class="at">sheet =</span> <span class="dv">2</span>)</span>
<span id="cb4-2"><a href="statistical-analysis-1.html#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(metadata)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 11
##   PatientID Stage  Metabolomics Proteomics GutMetagenomics OralMetagenomics LiverFatClass Gender AlcoholConsumption Smoker   Age
##   &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt;           &lt;chr&gt;            &lt;chr&gt;         &lt;chr&gt;  &lt;chr&gt;              &lt;chr&gt;  &lt;dbl&gt;
## 1 P101001   Before Send         Send       Send            Send             Severe        Male   No                 No        52
## 2 P101003   Before Send         Send       Send            Send             None          Female No                 No        31
## 3 P101004   Before Send         Send       Send            Send             Moderate      Male   Yes                No        43
## 4 P101007   Before Send         Send       Send            Send             Severe        Female No                 No        61
## 5 P101009   Before Send         Send       Send            Send             Moderate      Male   No                 Yes       51
## 6 P101010   Before Send         Send       Send            Send             Mild          Male   Yes                No        27</code></pre>
<ul>
<li>Data Preparation: SummarizedExperiment object</li>
</ul>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="statistical-analysis-1.html#cb6-1" aria-hidden="true" tabindex="-1"></a>getSEobject <span class="ot">&lt;-</span> <span class="cf">function</span>(x, y) {</span>
<span id="cb6-2"><a href="statistical-analysis-1.html#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="statistical-analysis-1.html#cb6-3" aria-hidden="true" tabindex="-1"></a>  target <span class="ot">&lt;-</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb6-4"><a href="statistical-analysis-1.html#cb6-4" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(Metabolomics <span class="sc">==</span> <span class="st">&quot;Send&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-5"><a href="statistical-analysis-1.html#cb6-5" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(PatientID, LiverFatClass, Gender, Smoker, Age, AlcoholConsumption)</span>
<span id="cb6-6"><a href="statistical-analysis-1.html#cb6-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-7"><a href="statistical-analysis-1.html#cb6-7" aria-hidden="true" tabindex="-1"></a>  sid <span class="ot">&lt;-</span> <span class="fu">intersect</span>(target<span class="sc">$</span>PatientID, <span class="fu">colnames</span>(profile))</span>
<span id="cb6-8"><a href="statistical-analysis-1.html#cb6-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-9"><a href="statistical-analysis-1.html#cb6-9" aria-hidden="true" tabindex="-1"></a>  features <span class="ot">&lt;-</span> y <span class="sc">%&gt;%</span> </span>
<span id="cb6-10"><a href="statistical-analysis-1.html#cb6-10" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(sid)) <span class="sc">%&gt;%</span></span>
<span id="cb6-11"><a href="statistical-analysis-1.html#cb6-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">t</span>()</span>
<span id="cb6-12"><a href="statistical-analysis-1.html#cb6-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(features) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;M_&quot;</span>, profile<span class="sc">$</span><span class="st">`</span><span class="at">COMP ID</span><span class="st">`</span>)</span>
<span id="cb6-13"><a href="statistical-analysis-1.html#cb6-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-14"><a href="statistical-analysis-1.html#cb6-14" aria-hidden="true" tabindex="-1"></a>  target <span class="ot">&lt;-</span> target[<span class="fu">pmatch</span>(sid, target<span class="sc">$</span>PatientID), , F]</span>
<span id="cb6-15"><a href="statistical-analysis-1.html#cb6-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-16"><a href="statistical-analysis-1.html#cb6-16" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb6-17"><a href="statistical-analysis-1.html#cb6-17" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">features =</span> features)</span>
<span id="cb6-18"><a href="statistical-analysis-1.html#cb6-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(res)  </span>
<span id="cb6-19"><a href="statistical-analysis-1.html#cb6-19" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-20"><a href="statistical-analysis-1.html#cb6-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-21"><a href="statistical-analysis-1.html#cb6-21" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">getSEobject</span>(metadata, profile)</span>
<span id="cb6-22"><a href="statistical-analysis-1.html#cb6-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-23"><a href="statistical-analysis-1.html#cb6-23" aria-hidden="true" tabindex="-1"></a>se</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 1032 55 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(1032): M_38768 M_38296 ... M_57517 M_15581
## rowData names(0):
## colnames(55): P101001 P101003 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
<ul>
<li>Extract data for test dataset</li>
</ul>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="statistical-analysis-1.html#cb8-1" aria-hidden="true" tabindex="-1"></a>get_testData <span class="ot">&lt;-</span> <span class="cf">function</span>(object, <span class="at">num =</span> <span class="dv">200</span>) {</span>
<span id="cb8-2"><a href="statistical-analysis-1.html#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="statistical-analysis-1.html#cb8-3" aria-hidden="true" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) <span class="sc">%&gt;%</span></span>
<span id="cb8-4"><a href="statistical-analysis-1.html#cb8-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">t</span>()</span>
<span id="cb8-5"><a href="statistical-analysis-1.html#cb8-5" aria-hidden="true" tabindex="-1"></a>  metadata_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object) <span class="sc">%&gt;%</span></span>
<span id="cb8-6"><a href="statistical-analysis-1.html#cb8-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb8-7"><a href="statistical-analysis-1.html#cb8-7" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;ID&quot;</span>)</span>
<span id="cb8-8"><a href="statistical-analysis-1.html#cb8-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb8-9"><a href="statistical-analysis-1.html#cb8-9" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> metadata_tab, </span>
<span id="cb8-10"><a href="statistical-analysis-1.html#cb8-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">features =</span> features_tab[, <span class="dv">1</span><span class="sc">:</span>num])</span>
<span id="cb8-11"><a href="statistical-analysis-1.html#cb8-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb8-12"><a href="statistical-analysis-1.html#cb8-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-13"><a href="statistical-analysis-1.html#cb8-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-14"><a href="statistical-analysis-1.html#cb8-14" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">get_testData</span>(<span class="at">object =</span> se)</span>
<span id="cb8-15"><a href="statistical-analysis-1.html#cb8-15" aria-hidden="true" tabindex="-1"></a>se</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 200 55 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(200): M_38768 M_38296 ... M_31787 M_63361
## rowData names(0):
## colnames(55): P101001 P101003 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="data-checking" class="section level3 hasAnchor" number="2.1.3">
<h3><span class="header-section-number">2.1.3</span> Data Checking<a href="statistical-analysis-1.html#data-checking" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Features in PomaSummarizedExperiment object must have the following criterion:</p>
<ul>
<li><p>All data values are numeric.</p></li>
<li><p>A total of 0 (0%) missing values were detected.</p></li>
</ul>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="statistical-analysis-1.html#cb10-1" aria-hidden="true" tabindex="-1"></a>CheckData <span class="ot">&lt;-</span> <span class="cf">function</span>(object) {</span>
<span id="cb10-2"><a href="statistical-analysis-1.html#cb10-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb10-3"><a href="statistical-analysis-1.html#cb10-3" aria-hidden="true" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object)</span>
<span id="cb10-4"><a href="statistical-analysis-1.html#cb10-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb10-5"><a href="statistical-analysis-1.html#cb10-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># numeric &amp; missing values</span></span>
<span id="cb10-6"><a href="statistical-analysis-1.html#cb10-6" aria-hidden="true" tabindex="-1"></a>  int_mat <span class="ot">&lt;-</span> features_tab</span>
<span id="cb10-7"><a href="statistical-analysis-1.html#cb10-7" aria-hidden="true" tabindex="-1"></a>  rowNms <span class="ot">&lt;-</span> <span class="fu">rownames</span>(int_mat)</span>
<span id="cb10-8"><a href="statistical-analysis-1.html#cb10-8" aria-hidden="true" tabindex="-1"></a>  colNms <span class="ot">&lt;-</span> <span class="fu">colnames</span>(int_mat)</span>
<span id="cb10-9"><a href="statistical-analysis-1.html#cb10-9" aria-hidden="true" tabindex="-1"></a>  naNms <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">is.na</span>(int_mat))</span>
<span id="cb10-10"><a href="statistical-analysis-1.html#cb10-10" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(int_mat)) {</span>
<span id="cb10-11"><a href="statistical-analysis-1.html#cb10-11" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">class</span>(int_mat[, i]) <span class="sc">==</span> <span class="st">&quot;integer64&quot;</span>) {</span>
<span id="cb10-12"><a href="statistical-analysis-1.html#cb10-12" aria-hidden="true" tabindex="-1"></a>      int_mat[, i] <span class="ot">&lt;-</span> <span class="fu">as.double</span>(int_mat[, i])</span>
<span id="cb10-13"><a href="statistical-analysis-1.html#cb10-13" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb10-14"><a href="statistical-analysis-1.html#cb10-14" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb10-15"><a href="statistical-analysis-1.html#cb10-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb10-16"><a href="statistical-analysis-1.html#cb10-16" aria-hidden="true" tabindex="-1"></a>  num_mat <span class="ot">&lt;-</span> <span class="fu">apply</span>(int_mat, <span class="dv">2</span>, as.numeric)</span>
<span id="cb10-17"><a href="statistical-analysis-1.html#cb10-17" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">sum</span>(<span class="fu">is.na</span>(num_mat)) <span class="sc">&gt;</span> naNms) {</span>
<span id="cb10-18"><a href="statistical-analysis-1.html#cb10-18" aria-hidden="true" tabindex="-1"></a>    num_mat <span class="ot">&lt;-</span> <span class="fu">apply</span>(int_mat, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;,&quot;</span>,  <span class="st">&quot;&quot;</span>, x)))</span>
<span id="cb10-19"><a href="statistical-analysis-1.html#cb10-19" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">sum</span>(<span class="fu">is.na</span>(num_mat)) <span class="sc">&gt;</span> naNms) {</span>
<span id="cb10-20"><a href="statistical-analysis-1.html#cb10-20" aria-hidden="true" tabindex="-1"></a>      <span class="fu">message</span>(<span class="st">&quot;&lt;font color=</span><span class="sc">\&quot;</span><span class="st">red</span><span class="sc">\&quot;</span><span class="st">&gt;Non-numeric values were found and replaced by NA.&lt;/font&gt;&quot;</span>)</span>
<span id="cb10-21"><a href="statistical-analysis-1.html#cb10-21" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb10-22"><a href="statistical-analysis-1.html#cb10-22" aria-hidden="true" tabindex="-1"></a>      <span class="fu">message</span>(<span class="st">&quot;All data values are numeric.&quot;</span>)</span>
<span id="cb10-23"><a href="statistical-analysis-1.html#cb10-23" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb10-24"><a href="statistical-analysis-1.html#cb10-24" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb10-25"><a href="statistical-analysis-1.html#cb10-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">&quot;All data values are numeric.&quot;</span>)</span>
<span id="cb10-26"><a href="statistical-analysis-1.html#cb10-26" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb10-27"><a href="statistical-analysis-1.html#cb10-27" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb10-28"><a href="statistical-analysis-1.html#cb10-28" aria-hidden="true" tabindex="-1"></a>  int_mat <span class="ot">&lt;-</span> num_mat</span>
<span id="cb10-29"><a href="statistical-analysis-1.html#cb10-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(int_mat) <span class="ot">&lt;-</span> rowNms</span>
<span id="cb10-30"><a href="statistical-analysis-1.html#cb10-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(int_mat) <span class="ot">&lt;-</span> colNms</span>
<span id="cb10-31"><a href="statistical-analysis-1.html#cb10-31" aria-hidden="true" tabindex="-1"></a>  varCol <span class="ot">&lt;-</span> <span class="fu">apply</span>(int_mat, <span class="dv">2</span>, var, <span class="at">na.rm =</span> T)</span>
<span id="cb10-32"><a href="statistical-analysis-1.html#cb10-32" aria-hidden="true" tabindex="-1"></a>  constCol <span class="ot">&lt;-</span> (varCol <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> <span class="fu">is.na</span>(varCol))</span>
<span id="cb10-33"><a href="statistical-analysis-1.html#cb10-33" aria-hidden="true" tabindex="-1"></a>  constNum <span class="ot">&lt;-</span> <span class="fu">sum</span>(constCol, <span class="at">na.rm =</span> T)</span>
<span id="cb10-34"><a href="statistical-analysis-1.html#cb10-34" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (constNum <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb10-35"><a href="statistical-analysis-1.html#cb10-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="fu">paste</span>(<span class="st">&quot;&lt;font color=</span><span class="sc">\&quot;</span><span class="st">red</span><span class="sc">\&quot;</span><span class="st">&gt;&quot;</span>, constNum, </span>
<span id="cb10-36"><a href="statistical-analysis-1.html#cb10-36" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot;features with a constant or single value across samples were found and deleted.&lt;/font&gt;&quot;</span>))</span>
<span id="cb10-37"><a href="statistical-analysis-1.html#cb10-37" aria-hidden="true" tabindex="-1"></a>    int_mat <span class="ot">&lt;-</span> int_mat[, <span class="sc">!</span>constCol, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb10-38"><a href="statistical-analysis-1.html#cb10-38" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb10-39"><a href="statistical-analysis-1.html#cb10-39" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb10-40"><a href="statistical-analysis-1.html#cb10-40" aria-hidden="true" tabindex="-1"></a>  totalCount <span class="ot">&lt;-</span> <span class="fu">nrow</span>(int_mat) <span class="sc">*</span> <span class="fu">ncol</span>(int_mat)</span>
<span id="cb10-41"><a href="statistical-analysis-1.html#cb10-41" aria-hidden="true" tabindex="-1"></a>  naCount <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">is.na</span>(int_mat))</span>
<span id="cb10-42"><a href="statistical-analysis-1.html#cb10-42" aria-hidden="true" tabindex="-1"></a>  naPercent <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> naCount<span class="sc">/</span>totalCount, <span class="dv">1</span>)</span>
<span id="cb10-43"><a href="statistical-analysis-1.html#cb10-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-44"><a href="statistical-analysis-1.html#cb10-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="fu">paste</span>(<span class="st">&quot;A total of &quot;</span>, naCount, <span class="st">&quot; (&quot;</span>, naPercent, </span>
<span id="cb10-45"><a href="statistical-analysis-1.html#cb10-45" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;%) missing values were detected.&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))  </span>
<span id="cb10-46"><a href="statistical-analysis-1.html#cb10-46" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb10-47"><a href="statistical-analysis-1.html#cb10-47" aria-hidden="true" tabindex="-1"></a>  <span class="co"># save int_mat into se object </span></span>
<span id="cb10-48"><a href="statistical-analysis-1.html#cb10-48" aria-hidden="true" tabindex="-1"></a>  target <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object) <span class="sc">%&gt;%</span></span>
<span id="cb10-49"><a href="statistical-analysis-1.html#cb10-49" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb10-50"><a href="statistical-analysis-1.html#cb10-50" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb10-51"><a href="statistical-analysis-1.html#cb10-51" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb10-52"><a href="statistical-analysis-1.html#cb10-52" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">features =</span> <span class="fu">t</span>(int_mat))</span>
<span id="cb10-53"><a href="statistical-analysis-1.html#cb10-53" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb10-54"><a href="statistical-analysis-1.html#cb10-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb10-55"><a href="statistical-analysis-1.html#cb10-55" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-56"><a href="statistical-analysis-1.html#cb10-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-57"><a href="statistical-analysis-1.html#cb10-57" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="fu">CheckData</span>(<span class="at">object =</span> se)</span></code></pre></div>
<pre><code>## All data values are numeric.</code></pre>
<pre><code>## A total of 1146 (10.4%) missing values were detected.</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="statistical-analysis-1.html#cb13-1" aria-hidden="true" tabindex="-1"></a>se</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 200 55 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(200): M_38768 M_38296 ... M_31787 M_63361
## rowData names(0):
## colnames(55): P101001 P101003 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="data-filtering" class="section level3 hasAnchor" number="2.1.4">
<h3><span class="header-section-number">2.1.4</span> Data Filtering<a href="statistical-analysis-1.html#data-filtering" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The purpose of the data filtering is to identify and remove variables that are unlikely to be of use when modeling the data. No phenotype information are used in the filtering process, so the result can be used with any downstream analysis. This step is strongly recommended for untargeted metabolomics datasets (i.e. spectral binning data, peak lists) with large number of variables, many of them are from baseline noises. Filtering can usually improve the results. For details, please refer to the paper by Hackstadt, et al.</p>
<p>Non-informative variables can be characterized in three groups: 1) variables of very small values (close to baseline or detection limit) - these variables can be detected using mean or median; 2) variables that are near-constant values throughout the experiment conditions (housekeeping or homeostasis) - these variables can be detected using standard deviation (SD); or the robust estimate such as interquantile range (IQR); and 3) variables that show low repeatability - this can be measured using QC samples using the relative standard deviation(RSD = SD/mean). Features with high percent RSD should be removed from the subsequent analysis (the suggested threshold is 20% for LC-MS and 30% for GC-MS). For data filtering based on the first two categories, the following empirical rules are applied during data filtering:</p>
<ul>
<li>Less than 250 variables: 5% will be filtered;</li>
<li>Between 250 - 500 variables: 10% will be filtered;</li>
<li>Between 500 - 1000 variables: 25% will be filtered;</li>
<li>Over 1000 variables: 40% will be filtered;</li>
</ul>
<p>Filtering features if their RSDs are &gt; 25% in QC samples</p>
<ul>
<li><p>Interquantile range (IQR)</p></li>
<li><p>Standard deviation (SD)</p></li>
<li><p>Median absolute deviation (MAD)</p></li>
<li><p>Relative standard deviation (RSD = SD/mean)</p></li>
<li><p>Non-parametric relative standard deviation (MAD/median)</p></li>
<li><p>Mean intensity value</p></li>
<li><p>Median intensity value</p></li>
</ul>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="statistical-analysis-1.html#cb15-1" aria-hidden="true" tabindex="-1"></a>FilterFeature <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb15-2"><a href="statistical-analysis-1.html#cb15-2" aria-hidden="true" tabindex="-1"></a>                       object,</span>
<span id="cb15-3"><a href="statistical-analysis-1.html#cb15-3" aria-hidden="true" tabindex="-1"></a>                       qc_label,</span>
<span id="cb15-4"><a href="statistical-analysis-1.html#cb15-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">method =</span> <span class="fu">c</span>(<span class="st">&quot;none&quot;</span>, <span class="st">&quot;iqr&quot;</span>, <span class="st">&quot;rsd&quot;</span>, </span>
<span id="cb15-5"><a href="statistical-analysis-1.html#cb15-5" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;nrsd&quot;</span>, <span class="st">&quot;mean&quot;</span>, <span class="st">&quot;sd&quot;</span>,</span>
<span id="cb15-6"><a href="statistical-analysis-1.html#cb15-6" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;mad&quot;</span>, <span class="st">&quot;median&quot;</span>),</span>
<span id="cb15-7"><a href="statistical-analysis-1.html#cb15-7" aria-hidden="true" tabindex="-1"></a>                       <span class="at">rsd_cutoff =</span> <span class="dv">25</span>) {</span>
<span id="cb15-8"><a href="statistical-analysis-1.html#cb15-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-9"><a href="statistical-analysis-1.html#cb15-9" aria-hidden="true" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) </span>
<span id="cb15-10"><a href="statistical-analysis-1.html#cb15-10" aria-hidden="true" tabindex="-1"></a>  metadata_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object) </span>
<span id="cb15-11"><a href="statistical-analysis-1.html#cb15-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-12"><a href="statistical-analysis-1.html#cb15-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># QC samples</span></span>
<span id="cb15-13"><a href="statistical-analysis-1.html#cb15-13" aria-hidden="true" tabindex="-1"></a>  qc_samples <span class="ot">&lt;-</span> metadata_tab <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb15-14"><a href="statistical-analysis-1.html#cb15-14" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(group <span class="sc">==</span> qc_label)</span>
<span id="cb15-15"><a href="statistical-analysis-1.html#cb15-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">dim</span>(qc_samples)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb15-16"><a href="statistical-analysis-1.html#cb15-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;No qc samples have been chosen, please check your input&quot;</span>)</span>
<span id="cb15-17"><a href="statistical-analysis-1.html#cb15-17" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb15-18"><a href="statistical-analysis-1.html#cb15-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-19"><a href="statistical-analysis-1.html#cb15-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># QC samples&#39; feature table</span></span>
<span id="cb15-20"><a href="statistical-analysis-1.html#cb15-20" aria-hidden="true" tabindex="-1"></a>  qc_feature <span class="ot">&lt;-</span> features_tab[, <span class="fu">colnames</span>(features_tab)<span class="sc">%in%</span><span class="fu">rownames</span>(qc_samples)] <span class="sc">%&gt;%</span></span>
<span id="cb15-21"><a href="statistical-analysis-1.html#cb15-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">t</span>()</span>
<span id="cb15-22"><a href="statistical-analysis-1.html#cb15-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-23"><a href="statistical-analysis-1.html#cb15-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># filter features by QC RSD</span></span>
<span id="cb15-24"><a href="statistical-analysis-1.html#cb15-24" aria-hidden="true" tabindex="-1"></a>  rsd <span class="ot">&lt;-</span> rsd_cutoff <span class="sc">/</span> <span class="dv">100</span></span>
<span id="cb15-25"><a href="statistical-analysis-1.html#cb15-25" aria-hidden="true" tabindex="-1"></a>  sds <span class="ot">&lt;-</span> <span class="fu">apply</span>(qc_feature, <span class="dv">2</span>, sd, <span class="at">na.rm =</span> T)</span>
<span id="cb15-26"><a href="statistical-analysis-1.html#cb15-26" aria-hidden="true" tabindex="-1"></a>  mns <span class="ot">&lt;-</span> <span class="fu">apply</span>(qc_feature, <span class="dv">2</span>, mean, <span class="at">na.rm =</span> T)</span>
<span id="cb15-27"><a href="statistical-analysis-1.html#cb15-27" aria-hidden="true" tabindex="-1"></a>  rsd_vals <span class="ot">&lt;-</span> <span class="fu">abs</span>(sds<span class="sc">/</span>mns) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb15-28"><a href="statistical-analysis-1.html#cb15-28" aria-hidden="true" tabindex="-1"></a>  gd_inx <span class="ot">&lt;-</span> rsd_vals <span class="sc">&lt;</span> rsd</span>
<span id="cb15-29"><a href="statistical-analysis-1.html#cb15-29" aria-hidden="true" tabindex="-1"></a>  int_mat <span class="ot">&lt;-</span> features_tab[gd_inx, ]</span>
<span id="cb15-30"><a href="statistical-analysis-1.html#cb15-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">&quot;Removed &quot;</span>, (<span class="fu">dim</span>(qc_feature)[<span class="dv">2</span>] <span class="sc">-</span> <span class="fu">dim</span>(int_mat)[<span class="dv">1</span>]), </span>
<span id="cb15-31"><a href="statistical-analysis-1.html#cb15-31" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot; features based on QC RSD values. QC samples are excluded from downstream functional analysis.&quot;</span>)</span>
<span id="cb15-32"><a href="statistical-analysis-1.html#cb15-32" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-33"><a href="statistical-analysis-1.html#cb15-33" aria-hidden="true" tabindex="-1"></a>  <span class="co"># whether to filter features by percentage according to the number</span></span>
<span id="cb15-34"><a href="statistical-analysis-1.html#cb15-34" aria-hidden="true" tabindex="-1"></a>  PerformFeatureFilter <span class="ot">&lt;-</span> <span class="cf">function</span>(datMatrix, </span>
<span id="cb15-35"><a href="statistical-analysis-1.html#cb15-35" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">qc_method =</span> method,</span>
<span id="cb15-36"><a href="statistical-analysis-1.html#cb15-36" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">remain_num =</span> <span class="cn">NULL</span>) {</span>
<span id="cb15-37"><a href="statistical-analysis-1.html#cb15-37" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb15-38"><a href="statistical-analysis-1.html#cb15-38" aria-hidden="true" tabindex="-1"></a>    dat <span class="ot">&lt;-</span> datMatrix</span>
<span id="cb15-39"><a href="statistical-analysis-1.html#cb15-39" aria-hidden="true" tabindex="-1"></a>    feat_num <span class="ot">&lt;-</span> <span class="fu">ncol</span>(dat)</span>
<span id="cb15-40"><a href="statistical-analysis-1.html#cb15-40" aria-hidden="true" tabindex="-1"></a>    feat_nms <span class="ot">&lt;-</span> <span class="fu">colnames</span>(dat)</span>
<span id="cb15-41"><a href="statistical-analysis-1.html#cb15-41" aria-hidden="true" tabindex="-1"></a>    nm <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb15-42"><a href="statistical-analysis-1.html#cb15-42" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;none&quot;</span> <span class="sc">&amp;&amp;</span> feat_num <span class="sc">&lt;</span> <span class="dv">5000</span>) { <span class="co"># only allow for less than 4000</span></span>
<span id="cb15-43"><a href="statistical-analysis-1.html#cb15-43" aria-hidden="true" tabindex="-1"></a>      remain <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">TRUE</span>, feat_num)</span>
<span id="cb15-44"><a href="statistical-analysis-1.html#cb15-44" aria-hidden="true" tabindex="-1"></a>      nm <span class="ot">&lt;-</span> <span class="st">&quot;No filtering was applied&quot;</span></span>
<span id="cb15-45"><a href="statistical-analysis-1.html#cb15-45" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb15-46"><a href="statistical-analysis-1.html#cb15-46" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;rsd&quot;</span>){</span>
<span id="cb15-47"><a href="statistical-analysis-1.html#cb15-47" aria-hidden="true" tabindex="-1"></a>        sds <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, sd, <span class="at">na.rm =</span> T)</span>
<span id="cb15-48"><a href="statistical-analysis-1.html#cb15-48" aria-hidden="true" tabindex="-1"></a>        mns <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, mean, <span class="at">na.rm =</span> T)</span>
<span id="cb15-49"><a href="statistical-analysis-1.html#cb15-49" aria-hidden="true" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">abs</span>(sds<span class="sc">/</span>mns)</span>
<span id="cb15-50"><a href="statistical-analysis-1.html#cb15-50" aria-hidden="true" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;Relative standard deviation&quot;</span></span>
<span id="cb15-51"><a href="statistical-analysis-1.html#cb15-51" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;nrsd&quot;</span> ) {</span>
<span id="cb15-52"><a href="statistical-analysis-1.html#cb15-52" aria-hidden="true" tabindex="-1"></a>        mads <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, mad, <span class="at">na.rm =</span> T)</span>
<span id="cb15-53"><a href="statistical-analysis-1.html#cb15-53" aria-hidden="true" tabindex="-1"></a>        meds <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, median, <span class="at">na.rm =</span> T)</span>
<span id="cb15-54"><a href="statistical-analysis-1.html#cb15-54" aria-hidden="true" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">abs</span>(mads<span class="sc">/</span>meds)</span>
<span id="cb15-55"><a href="statistical-analysis-1.html#cb15-55" aria-hidden="true" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;Non-paramatric relative standard deviation&quot;</span></span>
<span id="cb15-56"><a href="statistical-analysis-1.html#cb15-56" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;mean&quot;</span>) {</span>
<span id="cb15-57"><a href="statistical-analysis-1.html#cb15-57" aria-hidden="true" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, mean, <span class="at">na.rm =</span> T)</span>
<span id="cb15-58"><a href="statistical-analysis-1.html#cb15-58" aria-hidden="true" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;mean&quot;</span></span>
<span id="cb15-59"><a href="statistical-analysis-1.html#cb15-59" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;sd&quot;</span>) {</span>
<span id="cb15-60"><a href="statistical-analysis-1.html#cb15-60" aria-hidden="true" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, sd, <span class="at">na.rm =</span> T)</span>
<span id="cb15-61"><a href="statistical-analysis-1.html#cb15-61" aria-hidden="true" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;standard deviation&quot;</span></span>
<span id="cb15-62"><a href="statistical-analysis-1.html#cb15-62" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;mad&quot;</span>) {</span>
<span id="cb15-63"><a href="statistical-analysis-1.html#cb15-63" aria-hidden="true" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, mad, <span class="at">na.rm =</span> T)</span>
<span id="cb15-64"><a href="statistical-analysis-1.html#cb15-64" aria-hidden="true" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;Median absolute deviation&quot;</span></span>
<span id="cb15-65"><a href="statistical-analysis-1.html#cb15-65" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;median&quot;</span>) {</span>
<span id="cb15-66"><a href="statistical-analysis-1.html#cb15-66" aria-hidden="true" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, median, <span class="at">na.rm =</span> T)</span>
<span id="cb15-67"><a href="statistical-analysis-1.html#cb15-67" aria-hidden="true" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;median&quot;</span></span>
<span id="cb15-68"><a href="statistical-analysis-1.html#cb15-68" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;iqr&quot;</span>) { <span class="co"># iqr</span></span>
<span id="cb15-69"><a href="statistical-analysis-1.html#cb15-69" aria-hidden="true" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, IQR, <span class="at">na.rm =</span> T)</span>
<span id="cb15-70"><a href="statistical-analysis-1.html#cb15-70" aria-hidden="true" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;Interquantile Range&quot;</span></span>
<span id="cb15-71"><a href="statistical-analysis-1.html#cb15-71" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb15-72"><a href="statistical-analysis-1.html#cb15-72" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb15-73"><a href="statistical-analysis-1.html#cb15-73" aria-hidden="true" tabindex="-1"></a>      <span class="co"># get the rank of the filtered variables</span></span>
<span id="cb15-74"><a href="statistical-analysis-1.html#cb15-74" aria-hidden="true" tabindex="-1"></a>      rk <span class="ot">&lt;-</span> <span class="fu">rank</span>(<span class="sc">-</span>filter_val, <span class="at">ties.method =</span> <span class="st">&quot;random&quot;</span>)</span>
<span id="cb15-75"><a href="statistical-analysis-1.html#cb15-75" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb15-76"><a href="statistical-analysis-1.html#cb15-76" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">is.null</span>(remain_num)) { <span class="co"># apply empirical filtering based on data size</span></span>
<span id="cb15-77"><a href="statistical-analysis-1.html#cb15-77" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span> (feat_num <span class="sc">&lt;</span> <span class="dv">250</span>) { <span class="co"># reduce 5%</span></span>
<span id="cb15-78"><a href="statistical-analysis-1.html#cb15-78" aria-hidden="true" tabindex="-1"></a>            remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> feat_num <span class="sc">*</span> <span class="fl">0.95</span></span>
<span id="cb15-79"><a href="statistical-analysis-1.html#cb15-79" aria-hidden="true" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Further feature filtering based on &quot;</span>, nm)</span>
<span id="cb15-80"><a href="statistical-analysis-1.html#cb15-80" aria-hidden="true" tabindex="-1"></a>          } <span class="cf">else</span> <span class="cf">if</span> (feat_num <span class="sc">&lt;</span> <span class="dv">500</span>) { <span class="co"># reduce 10%</span></span>
<span id="cb15-81"><a href="statistical-analysis-1.html#cb15-81" aria-hidden="true" tabindex="-1"></a>            remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> feat_num <span class="sc">*</span> <span class="fl">0.9</span></span>
<span id="cb15-82"><a href="statistical-analysis-1.html#cb15-82" aria-hidden="true" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Further feature filtering based on &quot;</span>, nm)</span>
<span id="cb15-83"><a href="statistical-analysis-1.html#cb15-83" aria-hidden="true" tabindex="-1"></a>          } <span class="cf">else</span> <span class="cf">if</span> (feat_num <span class="sc">&lt;</span> <span class="dv">1000</span>) { <span class="co"># reduce 25%</span></span>
<span id="cb15-84"><a href="statistical-analysis-1.html#cb15-84" aria-hidden="true" tabindex="-1"></a>            remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> feat_num <span class="sc">*</span> <span class="fl">0.75</span></span>
<span id="cb15-85"><a href="statistical-analysis-1.html#cb15-85" aria-hidden="true" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Further feature filtering based on &quot;</span>, nm)</span>
<span id="cb15-86"><a href="statistical-analysis-1.html#cb15-86" aria-hidden="true" tabindex="-1"></a>          } <span class="cf">else</span> { <span class="co"># reduce 40%, if still over 5000, then only use top 5000</span></span>
<span id="cb15-87"><a href="statistical-analysis-1.html#cb15-87" aria-hidden="true" tabindex="-1"></a>            remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> feat_num <span class="sc">*</span> <span class="fl">0.6</span></span>
<span id="cb15-88"><a href="statistical-analysis-1.html#cb15-88" aria-hidden="true" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Further feature filtering based on &quot;</span>, nm)</span>
<span id="cb15-89"><a href="statistical-analysis-1.html#cb15-89" aria-hidden="true" tabindex="-1"></a>          }</span>
<span id="cb15-90"><a href="statistical-analysis-1.html#cb15-90" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb15-91"><a href="statistical-analysis-1.html#cb15-91" aria-hidden="true" tabindex="-1"></a>        remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> remain_num</span>
<span id="cb15-92"><a href="statistical-analysis-1.html#cb15-92" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb15-93"><a href="statistical-analysis-1.html#cb15-93" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb15-94"><a href="statistical-analysis-1.html#cb15-94" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb15-95"><a href="statistical-analysis-1.html#cb15-95" aria-hidden="true" tabindex="-1"></a>    res <span class="ot">&lt;-</span> datMatrix[, remain]</span>
<span id="cb15-96"><a href="statistical-analysis-1.html#cb15-96" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb15-97"><a href="statistical-analysis-1.html#cb15-97" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(res)</span>
<span id="cb15-98"><a href="statistical-analysis-1.html#cb15-98" aria-hidden="true" tabindex="-1"></a>  }  </span>
<span id="cb15-99"><a href="statistical-analysis-1.html#cb15-99" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-100"><a href="statistical-analysis-1.html#cb15-100" aria-hidden="true" tabindex="-1"></a>  feature_res <span class="ot">&lt;-</span> <span class="fu">PerformFeatureFilter</span>(<span class="fu">t</span>(int_mat))</span>
<span id="cb15-101"><a href="statistical-analysis-1.html#cb15-101" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-102"><a href="statistical-analysis-1.html#cb15-102" aria-hidden="true" tabindex="-1"></a>  <span class="co"># remove QC samples </span></span>
<span id="cb15-103"><a href="statistical-analysis-1.html#cb15-103" aria-hidden="true" tabindex="-1"></a>  feature_final <span class="ot">&lt;-</span> feature_res[<span class="sc">!</span><span class="fu">rownames</span>(feature_res) <span class="sc">%in%</span> <span class="fu">rownames</span>(qc_samples), ]</span>
<span id="cb15-104"><a href="statistical-analysis-1.html#cb15-104" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-105"><a href="statistical-analysis-1.html#cb15-105" aria-hidden="true" tabindex="-1"></a>  <span class="co"># save int_mat into se object </span></span>
<span id="cb15-106"><a href="statistical-analysis-1.html#cb15-106" aria-hidden="true" tabindex="-1"></a>  target <span class="ot">&lt;-</span> metadata_tab <span class="sc">%&gt;%</span> </span>
<span id="cb15-107"><a href="statistical-analysis-1.html#cb15-107" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb15-108"><a href="statistical-analysis-1.html#cb15-108" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-109"><a href="statistical-analysis-1.html#cb15-109" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(SampleID <span class="sc">%in%</span> <span class="fu">rownames</span>(feature_final))</span>
<span id="cb15-110"><a href="statistical-analysis-1.html#cb15-110" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb15-111"><a href="statistical-analysis-1.html#cb15-111" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">features =</span> feature_final)</span>
<span id="cb15-112"><a href="statistical-analysis-1.html#cb15-112" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-113"><a href="statistical-analysis-1.html#cb15-113" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(res) </span>
<span id="cb15-114"><a href="statistical-analysis-1.html#cb15-114" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-115"><a href="statistical-analysis-1.html#cb15-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-116"><a href="statistical-analysis-1.html#cb15-116" aria-hidden="true" tabindex="-1"></a>se_filter <span class="ot">&lt;-</span> <span class="fu">FilterFeature</span>(<span class="at">object =</span> se,</span>
<span id="cb15-117"><a href="statistical-analysis-1.html#cb15-117" aria-hidden="true" tabindex="-1"></a>                           <span class="at">qc_label =</span> <span class="st">&quot;None&quot;</span>,</span>
<span id="cb15-118"><a href="statistical-analysis-1.html#cb15-118" aria-hidden="true" tabindex="-1"></a>                           <span class="at">method =</span> <span class="st">&quot;iqr&quot;</span>)</span></code></pre></div>
<pre><code>## Removed 149 features based on QC RSD values. QC samples are excluded from downstream functional analysis.</code></pre>
<pre><code>## Further feature filtering based on Interquantile Range</code></pre>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="statistical-analysis-1.html#cb18-1" aria-hidden="true" tabindex="-1"></a>se_filter</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 48 45 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(48): M_52603 M_19130 ... M_63681 M_63361
## rowData names(0):
## colnames(45): P101001 P101004 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="missing-value-imputation" class="section level3 hasAnchor" number="2.1.5">
<h3><span class="header-section-number">2.1.5</span> Missing Value Imputation<a href="statistical-analysis-1.html#missing-value-imputation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="statistical-analysis-1.html#cb20-1" aria-hidden="true" tabindex="-1"></a>se_impute <span class="ot">&lt;-</span> <span class="fu">PomaImpute</span>(</span>
<span id="cb20-2"><a href="statistical-analysis-1.html#cb20-2" aria-hidden="true" tabindex="-1"></a>                    se_filter, </span>
<span id="cb20-3"><a href="statistical-analysis-1.html#cb20-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ZerosAsNA =</span> <span class="cn">TRUE</span>, </span>
<span id="cb20-4"><a href="statistical-analysis-1.html#cb20-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">RemoveNA =</span> <span class="cn">TRUE</span>, </span>
<span id="cb20-5"><a href="statistical-analysis-1.html#cb20-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">cutoff =</span> <span class="dv">20</span>, </span>
<span id="cb20-6"><a href="statistical-analysis-1.html#cb20-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;knn&quot;</span>)</span>
<span id="cb20-7"><a href="statistical-analysis-1.html#cb20-7" aria-hidden="true" tabindex="-1"></a>se_impute</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 48 45 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(48): M_52603 M_19130 ... M_63681 M_63361
## rowData names(0):
## colnames(45): P101001 P101004 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="data-normalization" class="section level3 hasAnchor" number="2.1.6">
<h3><span class="header-section-number">2.1.6</span> Data Normalization<a href="statistical-analysis-1.html#data-normalization" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The normalization procedures are grouped into three categories. You can use one or combine them to achieve better results.</p>
<ul>
<li><p>Sample normalization is for general-purpose adjustment for systematic differences among samples;</p>
<ul>
<li><p>Sample-specific normalization (i.e. weight, volume)</p></li>
<li><p>Normalization by sum</p></li>
<li><p>Normalization by median</p></li>
<li><p>Normalization by a reference sample (PQN)</p></li>
<li><p>Normalization by a pooled sample from group (group PQN)</p></li>
<li><p>Normalization by reference feature</p></li>
<li><p>Quantile normalization (suggested only for &gt; 1000 features)</p></li>
</ul></li>
<li><p>Data transformation applies a mathematical transformation on individual values themselves. A simple mathematical approach is used to deal with negative values in log and square root.</p>
<ul>
<li><p>Log transformation (base 10)</p></li>
<li><p>Square root transformation (square root of data values)</p></li>
<li><p>Cube root transformation (cube root of data values)</p></li>
</ul></li>
<li><p>Data scaling adjusts each variable/feature by a scaling factor computed based on the dispersion of the variable.</p>
<ul>
<li><p>Mean centering (mean-centered only)</p></li>
<li><p>Auto scaling (mean-centered and divided by the standard deviation of each variable)</p></li>
<li><p>Pareto scaling (mean-centered and divided by the square root of the standard deviation of each variable)</p></li>
<li><p>Range scaling (mean-centered and divided by the range of each variable)</p></li>
</ul></li>
</ul>
<div id="normalization-by-normalizedata-function" class="section level4 hasAnchor" number="2.1.6.1">
<h4><span class="header-section-number">2.1.6.1</span> Normalization by <strong>NormalizeData</strong> function<a href="statistical-analysis-1.html#normalization-by-normalizedata-function" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="statistical-analysis-1.html#cb22-1" aria-hidden="true" tabindex="-1"></a>NormalizeData <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb22-2"><a href="statistical-analysis-1.html#cb22-2" aria-hidden="true" tabindex="-1"></a>                      object,</span>
<span id="cb22-3"><a href="statistical-analysis-1.html#cb22-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">rowNorm =</span> <span class="fu">c</span>(<span class="st">&quot;Quantile&quot;</span>, <span class="st">&quot;GroupPQN&quot;</span>, <span class="st">&quot;SamplePQN&quot;</span>, </span>
<span id="cb22-4"><a href="statistical-analysis-1.html#cb22-4" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;CompNorm&quot;</span>, <span class="st">&quot;SumNorm&quot;</span>, <span class="st">&quot;MedianNorm&quot;</span>,</span>
<span id="cb22-5"><a href="statistical-analysis-1.html#cb22-5" aria-hidden="true" tabindex="-1"></a>                                  <span class="st">&quot;SpecNorm&quot;</span>, <span class="st">&quot;None&quot;</span>),</span>
<span id="cb22-6"><a href="statistical-analysis-1.html#cb22-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">transNorm =</span> <span class="fu">c</span>(<span class="st">&quot;LogNorm&quot;</span>, <span class="st">&quot;SrNorm&quot;</span>, <span class="st">&quot;CrNorm&quot;</span>, <span class="st">&quot;None&quot;</span>),</span>
<span id="cb22-7"><a href="statistical-analysis-1.html#cb22-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">scaleNorm =</span> <span class="fu">c</span>(<span class="st">&quot;MeanCenter&quot;</span>, <span class="st">&quot;AutoNorm&quot;</span>, <span class="st">&quot;ParetoNorm&quot;</span>, </span>
<span id="cb22-8"><a href="statistical-analysis-1.html#cb22-8" aria-hidden="true" tabindex="-1"></a>                                    <span class="st">&quot;RangeNorm&quot;</span>, <span class="st">&quot;None&quot;</span>),</span>
<span id="cb22-9"><a href="statistical-analysis-1.html#cb22-9" aria-hidden="true" tabindex="-1"></a>                      <span class="at">ref =</span> <span class="cn">NULL</span>,</span>
<span id="cb22-10"><a href="statistical-analysis-1.html#cb22-10" aria-hidden="true" tabindex="-1"></a>                      <span class="at">SpeWeight =</span> <span class="dv">1</span>) {</span>
<span id="cb22-11"><a href="statistical-analysis-1.html#cb22-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-12"><a href="statistical-analysis-1.html#cb22-12" aria-hidden="true" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) </span>
<span id="cb22-13"><a href="statistical-analysis-1.html#cb22-13" aria-hidden="true" tabindex="-1"></a>  metadata_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object)   </span>
<span id="cb22-14"><a href="statistical-analysis-1.html#cb22-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-15"><a href="statistical-analysis-1.html#cb22-15" aria-hidden="true" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">t</span>(features_tab)</span>
<span id="cb22-16"><a href="statistical-analysis-1.html#cb22-16" aria-hidden="true" tabindex="-1"></a>  colNames <span class="ot">&lt;-</span> <span class="fu">colnames</span>(data)</span>
<span id="cb22-17"><a href="statistical-analysis-1.html#cb22-17" aria-hidden="true" tabindex="-1"></a>  rowNames <span class="ot">&lt;-</span> <span class="fu">rownames</span>(data)</span>
<span id="cb22-18"><a href="statistical-analysis-1.html#cb22-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-19"><a href="statistical-analysis-1.html#cb22-19" aria-hidden="true" tabindex="-1"></a>  <span class="do">#############################################</span></span>
<span id="cb22-20"><a href="statistical-analysis-1.html#cb22-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Sample normalization</span></span>
<span id="cb22-21"><a href="statistical-analysis-1.html#cb22-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># perform quantile normalization on the raw data (can be log transformed later by user)</span></span>
<span id="cb22-22"><a href="statistical-analysis-1.html#cb22-22" aria-hidden="true" tabindex="-1"></a>  QuantileNormalize <span class="ot">&lt;-</span> <span class="cf">function</span>(data) {</span>
<span id="cb22-23"><a href="statistical-analysis-1.html#cb22-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">t</span>(preprocessCore<span class="sc">::</span><span class="fu">normalize.quantiles</span>(<span class="fu">t</span>(data), <span class="at">copy=</span><span class="cn">FALSE</span>)));</span>
<span id="cb22-24"><a href="statistical-analysis-1.html#cb22-24" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb22-25"><a href="statistical-analysis-1.html#cb22-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># normalize by a reference sample (probability quotient normalization)</span></span>
<span id="cb22-26"><a href="statistical-analysis-1.html#cb22-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ref should be the name of the reference sample</span></span>
<span id="cb22-27"><a href="statistical-analysis-1.html#cb22-27" aria-hidden="true" tabindex="-1"></a>  ProbNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x, ref_smpl) {</span>
<span id="cb22-28"><a href="statistical-analysis-1.html#cb22-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(x<span class="sc">/</span><span class="fu">median</span>(<span class="fu">as.numeric</span>(x<span class="sc">/</span>ref_smpl), <span class="at">na.rm =</span> T))</span>
<span id="cb22-29"><a href="statistical-analysis-1.html#cb22-29" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb22-30"><a href="statistical-analysis-1.html#cb22-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-31"><a href="statistical-analysis-1.html#cb22-31" aria-hidden="true" tabindex="-1"></a>  <span class="co"># normalize by a reference reference (i.e. creatinine)</span></span>
<span id="cb22-32"><a href="statistical-analysis-1.html#cb22-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ref should be the name of the cmpd</span></span>
<span id="cb22-33"><a href="statistical-analysis-1.html#cb22-33" aria-hidden="true" tabindex="-1"></a>  CompNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x, ref) {</span>
<span id="cb22-34"><a href="statistical-analysis-1.html#cb22-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="dv">1000</span><span class="sc">*</span>x<span class="sc">/</span>x[ref])</span>
<span id="cb22-35"><a href="statistical-analysis-1.html#cb22-35" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb22-36"><a href="statistical-analysis-1.html#cb22-36" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-37"><a href="statistical-analysis-1.html#cb22-37" aria-hidden="true" tabindex="-1"></a>  SumNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb22-38"><a href="statistical-analysis-1.html#cb22-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="dv">1000</span><span class="sc">*</span>x<span class="sc">/</span><span class="fu">sum</span>(x, <span class="at">na.rm =</span> T))</span>
<span id="cb22-39"><a href="statistical-analysis-1.html#cb22-39" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb22-40"><a href="statistical-analysis-1.html#cb22-40" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-41"><a href="statistical-analysis-1.html#cb22-41" aria-hidden="true" tabindex="-1"></a>  <span class="co"># normalize by median</span></span>
<span id="cb22-42"><a href="statistical-analysis-1.html#cb22-42" aria-hidden="true" tabindex="-1"></a>  MedianNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb22-43"><a href="statistical-analysis-1.html#cb22-43" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(x<span class="sc">/</span><span class="fu">median</span>(x, <span class="at">na.rm =</span> T))</span>
<span id="cb22-44"><a href="statistical-analysis-1.html#cb22-44" aria-hidden="true" tabindex="-1"></a>  }  </span>
<span id="cb22-45"><a href="statistical-analysis-1.html#cb22-45" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-46"><a href="statistical-analysis-1.html#cb22-46" aria-hidden="true" tabindex="-1"></a>  <span class="co"># row-wise normalization</span></span>
<span id="cb22-47"><a href="statistical-analysis-1.html#cb22-47" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;Quantile&quot;</span>) {</span>
<span id="cb22-48"><a href="statistical-analysis-1.html#cb22-48" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">QuantileNormalize</span>(data)</span>
<span id="cb22-49"><a href="statistical-analysis-1.html#cb22-49" aria-hidden="true" tabindex="-1"></a>    <span class="co"># this can introduce constant variables if a variable is </span></span>
<span id="cb22-50"><a href="statistical-analysis-1.html#cb22-50" aria-hidden="true" tabindex="-1"></a>    <span class="co"># at the same rank across all samples (replaced by its average across all)</span></span>
<span id="cb22-51"><a href="statistical-analysis-1.html#cb22-51" aria-hidden="true" tabindex="-1"></a>    varCol <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, var, <span class="at">na.rm =</span> T)</span>
<span id="cb22-52"><a href="statistical-analysis-1.html#cb22-52" aria-hidden="true" tabindex="-1"></a>    constCol <span class="ot">&lt;-</span> (varCol <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> <span class="fu">is.na</span>(varCol))</span>
<span id="cb22-53"><a href="statistical-analysis-1.html#cb22-53" aria-hidden="true" tabindex="-1"></a>    constNum <span class="ot">&lt;-</span> <span class="fu">sum</span>(constCol, <span class="at">na.rm =</span> T)</span>
<span id="cb22-54"><a href="statistical-analysis-1.html#cb22-54" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (constNum <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb22-55"><a href="statistical-analysis-1.html#cb22-55" aria-hidden="true" tabindex="-1"></a>      <span class="fu">message</span>(<span class="fu">paste</span>(<span class="st">&quot;After quantile normalization&quot;</span>, constNum, </span>
<span id="cb22-56"><a href="statistical-analysis-1.html#cb22-56" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;features with a constant value were found and deleted.&quot;</span>))</span>
<span id="cb22-57"><a href="statistical-analysis-1.html#cb22-57" aria-hidden="true" tabindex="-1"></a>      data <span class="ot">&lt;-</span> data[, <span class="sc">!</span>constCol, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb22-58"><a href="statistical-analysis-1.html#cb22-58" aria-hidden="true" tabindex="-1"></a>      colNames <span class="ot">&lt;-</span> <span class="fu">colnames</span>(data)</span>
<span id="cb22-59"><a href="statistical-analysis-1.html#cb22-59" aria-hidden="true" tabindex="-1"></a>      rowNames <span class="ot">&lt;-</span> <span class="fu">rownames</span>(data)</span>
<span id="cb22-60"><a href="statistical-analysis-1.html#cb22-60" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb22-61"><a href="statistical-analysis-1.html#cb22-61" aria-hidden="true" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Quantile Normalization&quot;</span></span>
<span id="cb22-62"><a href="statistical-analysis-1.html#cb22-62" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;GroupPQN&quot;</span>) {</span>
<span id="cb22-63"><a href="statistical-analysis-1.html#cb22-63" aria-hidden="true" tabindex="-1"></a>    grp_inx <span class="ot">&lt;-</span> metadata_tab<span class="sc">$</span>group <span class="sc">==</span> ref</span>
<span id="cb22-64"><a href="statistical-analysis-1.html#cb22-64" aria-hidden="true" tabindex="-1"></a>    ref.smpl <span class="ot">&lt;-</span> <span class="fu">apply</span>(data[grp_inx, , <span class="at">drop =</span> <span class="cn">FALSE</span>], <span class="dv">2</span>, mean)</span>
<span id="cb22-65"><a href="statistical-analysis-1.html#cb22-65" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(data, <span class="dv">1</span>, ProbNorm, ref.smpl))</span>
<span id="cb22-66"><a href="statistical-analysis-1.html#cb22-66" aria-hidden="true" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Probabilistic Quotient Normalization by a reference group&quot;</span></span>
<span id="cb22-67"><a href="statistical-analysis-1.html#cb22-67" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;SamplePQN&quot;</span>) {</span>
<span id="cb22-68"><a href="statistical-analysis-1.html#cb22-68" aria-hidden="true" tabindex="-1"></a>    ref.smpl <span class="ot">&lt;-</span> data[ref, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb22-69"><a href="statistical-analysis-1.html#cb22-69" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(data, <span class="dv">1</span>, ProbNorm, ref.smpl))</span>
<span id="cb22-70"><a href="statistical-analysis-1.html#cb22-70" aria-hidden="true" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Probabilistic Quotient Normalization by a reference sample&quot;</span></span>
<span id="cb22-71"><a href="statistical-analysis-1.html#cb22-71" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;CompNorm&quot;</span>) {</span>
<span id="cb22-72"><a href="statistical-analysis-1.html#cb22-72" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(<span class="fu">t</span>(data), <span class="dv">1</span>, CompNorm, ref))</span>
<span id="cb22-73"><a href="statistical-analysis-1.html#cb22-73" aria-hidden="true" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Normalization by a reference feature&quot;</span>;</span>
<span id="cb22-74"><a href="statistical-analysis-1.html#cb22-74" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;SumNorm&quot;</span>) {</span>
<span id="cb22-75"><a href="statistical-analysis-1.html#cb22-75" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(data, <span class="dv">1</span>, SumNorm))</span>
<span id="cb22-76"><a href="statistical-analysis-1.html#cb22-76" aria-hidden="true" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Normalization to constant sum&quot;</span></span>
<span id="cb22-77"><a href="statistical-analysis-1.html#cb22-77" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;MedianNorm&quot;</span>) {</span>
<span id="cb22-78"><a href="statistical-analysis-1.html#cb22-78" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(data, <span class="dv">1</span>, MedianNorm))</span>
<span id="cb22-79"><a href="statistical-analysis-1.html#cb22-79" aria-hidden="true" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Normalization to sample median&quot;</span></span>
<span id="cb22-80"><a href="statistical-analysis-1.html#cb22-80" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span>(rowNorm <span class="sc">==</span> <span class="st">&quot;SpecNorm&quot;</span>) {</span>
<span id="cb22-81"><a href="statistical-analysis-1.html#cb22-81" aria-hidden="true" tabindex="-1"></a>    norm.vec <span class="ot">&lt;-</span> <span class="fu">rep</span>(SpeWeight, <span class="fu">nrow</span>(data)) <span class="co"># default all same weight vec to prevent error</span></span>
<span id="cb22-82"><a href="statistical-analysis-1.html#cb22-82" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> data <span class="sc">/</span> norm.vec</span>
<span id="cb22-83"><a href="statistical-analysis-1.html#cb22-83" aria-hidden="true" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">&quot;No sample specific information were given, all set to 1.0&quot;</span>)</span>
<span id="cb22-84"><a href="statistical-analysis-1.html#cb22-84" aria-hidden="true" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Normalization by sample-specific factor&quot;</span></span>
<span id="cb22-85"><a href="statistical-analysis-1.html#cb22-85" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb22-86"><a href="statistical-analysis-1.html#cb22-86" aria-hidden="true" tabindex="-1"></a>    <span class="co"># nothing to do</span></span>
<span id="cb22-87"><a href="statistical-analysis-1.html#cb22-87" aria-hidden="true" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;N/A&quot;</span></span>
<span id="cb22-88"><a href="statistical-analysis-1.html#cb22-88" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb22-89"><a href="statistical-analysis-1.html#cb22-89" aria-hidden="true" tabindex="-1"></a>  <span class="do">################################################ </span></span>
<span id="cb22-90"><a href="statistical-analysis-1.html#cb22-90" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-91"><a href="statistical-analysis-1.html#cb22-91" aria-hidden="true" tabindex="-1"></a>  <span class="co"># use apply will lose dimension info (i.e. row names and colnames)</span></span>
<span id="cb22-92"><a href="statistical-analysis-1.html#cb22-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(data) <span class="ot">&lt;-</span> rowNames</span>
<span id="cb22-93"><a href="statistical-analysis-1.html#cb22-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(data) <span class="ot">&lt;-</span> colNames</span>
<span id="cb22-94"><a href="statistical-analysis-1.html#cb22-94" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-95"><a href="statistical-analysis-1.html#cb22-95" aria-hidden="true" tabindex="-1"></a>  <span class="co"># if the reference by feature, the feature column should be removed, since it is all 1</span></span>
<span id="cb22-96"><a href="statistical-analysis-1.html#cb22-96" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(rowNorm <span class="sc">==</span> <span class="st">&quot;CompNorm&quot;</span> <span class="sc">&amp;&amp;</span> <span class="sc">!</span><span class="fu">is.null</span>(ref)){</span>
<span id="cb22-97"><a href="statistical-analysis-1.html#cb22-97" aria-hidden="true" tabindex="-1"></a>    inx <span class="ot">&lt;-</span> <span class="fu">match</span>(ref, <span class="fu">colnames</span>(data))</span>
<span id="cb22-98"><a href="statistical-analysis-1.html#cb22-98" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> data[, <span class="sc">-</span>inx, drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb22-99"><a href="statistical-analysis-1.html#cb22-99" aria-hidden="true" tabindex="-1"></a>    colNames <span class="ot">&lt;-</span> colNames[<span class="sc">-</span>inx]</span>
<span id="cb22-100"><a href="statistical-analysis-1.html#cb22-100" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb22-101"><a href="statistical-analysis-1.html#cb22-101" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-102"><a href="statistical-analysis-1.html#cb22-102" aria-hidden="true" tabindex="-1"></a>  <span class="do">#############################################</span></span>
<span id="cb22-103"><a href="statistical-analysis-1.html#cb22-103" aria-hidden="true" tabindex="-1"></a>  <span class="co">#  Data transformation</span></span>
<span id="cb22-104"><a href="statistical-analysis-1.html#cb22-104" aria-hidden="true" tabindex="-1"></a>  <span class="co"># generalize log, tolerant to 0 and negative values</span></span>
<span id="cb22-105"><a href="statistical-analysis-1.html#cb22-105" aria-hidden="true" tabindex="-1"></a>  LogNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x, min.val) {</span>
<span id="cb22-106"><a href="statistical-analysis-1.html#cb22-106" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">log10</span>((x <span class="sc">+</span> <span class="fu">sqrt</span>(x<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> min.val<span class="sc">^</span><span class="dv">2</span>))<span class="sc">/</span><span class="dv">2</span>))</span>
<span id="cb22-107"><a href="statistical-analysis-1.html#cb22-107" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb22-108"><a href="statistical-analysis-1.html#cb22-108" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-109"><a href="statistical-analysis-1.html#cb22-109" aria-hidden="true" tabindex="-1"></a>  <span class="co"># square root, tolerant to negative values</span></span>
<span id="cb22-110"><a href="statistical-analysis-1.html#cb22-110" aria-hidden="true" tabindex="-1"></a>  SquareRootNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x, min.val) {</span>
<span id="cb22-111"><a href="statistical-analysis-1.html#cb22-111" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(((x <span class="sc">+</span> <span class="fu">sqrt</span>(x<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> min.val<span class="sc">^</span><span class="dv">2</span>))<span class="sc">/</span><span class="dv">2</span>)<span class="sc">^</span>(<span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>))</span>
<span id="cb22-112"><a href="statistical-analysis-1.html#cb22-112" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb22-113"><a href="statistical-analysis-1.html#cb22-113" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-114"><a href="statistical-analysis-1.html#cb22-114" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (transNorm <span class="sc">==</span> <span class="st">&quot;LogNorm&quot;</span>) {</span>
<span id="cb22-115"><a href="statistical-analysis-1.html#cb22-115" aria-hidden="true" tabindex="-1"></a>    min.val <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">abs</span>(data[data <span class="sc">!=</span> <span class="dv">0</span>]))<span class="sc">/</span><span class="dv">10</span></span>
<span id="cb22-116"><a href="statistical-analysis-1.html#cb22-116" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, LogNorm, min.val)</span>
<span id="cb22-117"><a href="statistical-analysis-1.html#cb22-117" aria-hidden="true" tabindex="-1"></a>    transnm <span class="ot">&lt;-</span> <span class="st">&quot;Log10 Normalization&quot;</span></span>
<span id="cb22-118"><a href="statistical-analysis-1.html#cb22-118" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (transNorm <span class="sc">==</span> <span class="st">&quot;SrNorm&quot;</span>) {</span>
<span id="cb22-119"><a href="statistical-analysis-1.html#cb22-119" aria-hidden="true" tabindex="-1"></a>    min.val <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">abs</span>(data[data <span class="sc">!=</span> <span class="dv">0</span>]))<span class="sc">/</span><span class="dv">10</span></span>
<span id="cb22-120"><a href="statistical-analysis-1.html#cb22-120" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, SquareRootNorm, min.val)</span>
<span id="cb22-121"><a href="statistical-analysis-1.html#cb22-121" aria-hidden="true" tabindex="-1"></a>    transnm <span class="ot">&lt;-</span> <span class="st">&quot;Square Root Transformation&quot;</span></span>
<span id="cb22-122"><a href="statistical-analysis-1.html#cb22-122" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (transNorm <span class="sc">==</span> <span class="st">&quot;CrNorm&quot;</span>) {</span>
<span id="cb22-123"><a href="statistical-analysis-1.html#cb22-123" aria-hidden="true" tabindex="-1"></a>    norm.data <span class="ot">&lt;-</span> <span class="fu">abs</span>(data)<span class="sc">^</span>(<span class="dv">1</span><span class="sc">/</span><span class="dv">3</span>)</span>
<span id="cb22-124"><a href="statistical-analysis-1.html#cb22-124" aria-hidden="true" tabindex="-1"></a>    norm.data[data <span class="sc">&lt;</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="sc">-</span>norm.data[data <span class="sc">&lt;</span> <span class="dv">0</span>]</span>
<span id="cb22-125"><a href="statistical-analysis-1.html#cb22-125" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> norm.data</span>
<span id="cb22-126"><a href="statistical-analysis-1.html#cb22-126" aria-hidden="true" tabindex="-1"></a>    transnm <span class="ot">&lt;-</span> <span class="st">&quot;Cubic Root Transformation&quot;</span></span>
<span id="cb22-127"><a href="statistical-analysis-1.html#cb22-127" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb22-128"><a href="statistical-analysis-1.html#cb22-128" aria-hidden="true" tabindex="-1"></a>    transnm <span class="ot">&lt;-</span> <span class="st">&quot;N/A&quot;</span></span>
<span id="cb22-129"><a href="statistical-analysis-1.html#cb22-129" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb22-130"><a href="statistical-analysis-1.html#cb22-130" aria-hidden="true" tabindex="-1"></a>  <span class="do">#############################################</span></span>
<span id="cb22-131"><a href="statistical-analysis-1.html#cb22-131" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-132"><a href="statistical-analysis-1.html#cb22-132" aria-hidden="true" tabindex="-1"></a>  <span class="do">#############################################</span></span>
<span id="cb22-133"><a href="statistical-analysis-1.html#cb22-133" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Data scaling</span></span>
<span id="cb22-134"><a href="statistical-analysis-1.html#cb22-134" aria-hidden="true" tabindex="-1"></a>  <span class="co"># normalize to zero mean and unit variance</span></span>
<span id="cb22-135"><a href="statistical-analysis-1.html#cb22-135" aria-hidden="true" tabindex="-1"></a>  AutoNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb22-136"><a href="statistical-analysis-1.html#cb22-136" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>((x <span class="sc">-</span> <span class="fu">mean</span>(x))<span class="sc">/</span><span class="fu">sd</span>(x, <span class="at">na.rm =</span> T))</span>
<span id="cb22-137"><a href="statistical-analysis-1.html#cb22-137" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb22-138"><a href="statistical-analysis-1.html#cb22-138" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-139"><a href="statistical-analysis-1.html#cb22-139" aria-hidden="true" tabindex="-1"></a>  <span class="co"># normalize to zero mean but variance/SE</span></span>
<span id="cb22-140"><a href="statistical-analysis-1.html#cb22-140" aria-hidden="true" tabindex="-1"></a>  ParetoNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb22-141"><a href="statistical-analysis-1.html#cb22-141" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>((x <span class="sc">-</span> <span class="fu">mean</span>(x))<span class="sc">/</span><span class="fu">sqrt</span>(<span class="fu">sd</span>(x, <span class="at">na.rm =</span> T)))</span>
<span id="cb22-142"><a href="statistical-analysis-1.html#cb22-142" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb22-143"><a href="statistical-analysis-1.html#cb22-143" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-144"><a href="statistical-analysis-1.html#cb22-144" aria-hidden="true" tabindex="-1"></a>  <span class="co"># normalize to zero mean but variance/SE</span></span>
<span id="cb22-145"><a href="statistical-analysis-1.html#cb22-145" aria-hidden="true" tabindex="-1"></a>  MeanCenter <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb22-146"><a href="statistical-analysis-1.html#cb22-146" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(x <span class="sc">-</span> <span class="fu">mean</span>(x))</span>
<span id="cb22-147"><a href="statistical-analysis-1.html#cb22-147" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb22-148"><a href="statistical-analysis-1.html#cb22-148" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-149"><a href="statistical-analysis-1.html#cb22-149" aria-hidden="true" tabindex="-1"></a>  <span class="co"># normalize to zero mean but variance/SE</span></span>
<span id="cb22-150"><a href="statistical-analysis-1.html#cb22-150" aria-hidden="true" tabindex="-1"></a>  RangeNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb22-151"><a href="statistical-analysis-1.html#cb22-151" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">max</span>(x) <span class="sc">==</span> <span class="fu">min</span>(x)) {</span>
<span id="cb22-152"><a href="statistical-analysis-1.html#cb22-152" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>(x)</span>
<span id="cb22-153"><a href="statistical-analysis-1.html#cb22-153" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb22-154"><a href="statistical-analysis-1.html#cb22-154" aria-hidden="true" tabindex="-1"></a>      <span class="fu">return</span>((x <span class="sc">-</span> <span class="fu">mean</span>(x))<span class="sc">/</span>(<span class="fu">max</span>(x) <span class="sc">-</span> <span class="fu">min</span>(x)))</span>
<span id="cb22-155"><a href="statistical-analysis-1.html#cb22-155" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb22-156"><a href="statistical-analysis-1.html#cb22-156" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb22-157"><a href="statistical-analysis-1.html#cb22-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-158"><a href="statistical-analysis-1.html#cb22-158" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (scaleNorm <span class="sc">==</span> <span class="st">&quot;MeanCenter&quot;</span>) {</span>
<span id="cb22-159"><a href="statistical-analysis-1.html#cb22-159" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, MeanCenter)</span>
<span id="cb22-160"><a href="statistical-analysis-1.html#cb22-160" aria-hidden="true" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;Mean Centering&quot;</span></span>
<span id="cb22-161"><a href="statistical-analysis-1.html#cb22-161" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (scaleNorm <span class="sc">==</span> <span class="st">&quot;AutoNorm&quot;</span>) {</span>
<span id="cb22-162"><a href="statistical-analysis-1.html#cb22-162" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, AutoNorm)</span>
<span id="cb22-163"><a href="statistical-analysis-1.html#cb22-163" aria-hidden="true" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;Autoscaling&quot;</span></span>
<span id="cb22-164"><a href="statistical-analysis-1.html#cb22-164" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (scaleNorm <span class="sc">==</span> <span class="st">&quot;ParetoNorm&quot;</span>) {</span>
<span id="cb22-165"><a href="statistical-analysis-1.html#cb22-165" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, ParetoNorm)</span>
<span id="cb22-166"><a href="statistical-analysis-1.html#cb22-166" aria-hidden="true" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;Pareto Scaling&quot;</span></span>
<span id="cb22-167"><a href="statistical-analysis-1.html#cb22-167" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (scaleNorm <span class="sc">==</span> <span class="st">&quot;RangeNorm&quot;</span>) {</span>
<span id="cb22-168"><a href="statistical-analysis-1.html#cb22-168" aria-hidden="true" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, RangeNorm)</span>
<span id="cb22-169"><a href="statistical-analysis-1.html#cb22-169" aria-hidden="true" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;Range Scaling&quot;</span></span>
<span id="cb22-170"><a href="statistical-analysis-1.html#cb22-170" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb22-171"><a href="statistical-analysis-1.html#cb22-171" aria-hidden="true" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;N/A&quot;</span></span>
<span id="cb22-172"><a href="statistical-analysis-1.html#cb22-172" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb22-173"><a href="statistical-analysis-1.html#cb22-173" aria-hidden="true" tabindex="-1"></a>  <span class="do">############################################# </span></span>
<span id="cb22-174"><a href="statistical-analysis-1.html#cb22-174" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-175"><a href="statistical-analysis-1.html#cb22-175" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">&quot;Row norm: &quot;</span>, rownm, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>,</span>
<span id="cb22-176"><a href="statistical-analysis-1.html#cb22-176" aria-hidden="true" tabindex="-1"></a>          <span class="st">&quot;Data Transformation norm: &quot;</span>, transnm, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>,</span>
<span id="cb22-177"><a href="statistical-analysis-1.html#cb22-177" aria-hidden="true" tabindex="-1"></a>          <span class="st">&quot;Data Scaling norm: &quot;</span>, scalenm, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)  </span>
<span id="cb22-178"><a href="statistical-analysis-1.html#cb22-178" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-179"><a href="statistical-analysis-1.html#cb22-179" aria-hidden="true" tabindex="-1"></a>  <span class="co"># note after using &quot;apply&quot; function, all the attribute lost, need to add back</span></span>
<span id="cb22-180"><a href="statistical-analysis-1.html#cb22-180" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(data) <span class="ot">&lt;-</span> rowNames</span>
<span id="cb22-181"><a href="statistical-analysis-1.html#cb22-181" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(data) <span class="ot">&lt;-</span> colNames</span>
<span id="cb22-182"><a href="statistical-analysis-1.html#cb22-182" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-183"><a href="statistical-analysis-1.html#cb22-183" aria-hidden="true" tabindex="-1"></a>  target <span class="ot">&lt;-</span> metadata_tab <span class="sc">%&gt;%</span> </span>
<span id="cb22-184"><a href="statistical-analysis-1.html#cb22-184" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb22-185"><a href="statistical-analysis-1.html#cb22-185" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-186"><a href="statistical-analysis-1.html#cb22-186" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(SampleID<span class="sc">%in%</span><span class="fu">rownames</span>(data))</span>
<span id="cb22-187"><a href="statistical-analysis-1.html#cb22-187" aria-hidden="true" tabindex="-1"></a>  se <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb22-188"><a href="statistical-analysis-1.html#cb22-188" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">features =</span> data)</span>
<span id="cb22-189"><a href="statistical-analysis-1.html#cb22-189" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-190"><a href="statistical-analysis-1.html#cb22-190" aria-hidden="true" tabindex="-1"></a>  <span class="co"># need to do some sanity check, for log there may be Inf values introduced</span></span>
<span id="cb22-191"><a href="statistical-analysis-1.html#cb22-191" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">CheckData</span>(se)</span>
<span id="cb22-192"><a href="statistical-analysis-1.html#cb22-192" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb22-193"><a href="statistical-analysis-1.html#cb22-193" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb22-194"><a href="statistical-analysis-1.html#cb22-194" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb22-195"><a href="statistical-analysis-1.html#cb22-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-196"><a href="statistical-analysis-1.html#cb22-196" aria-hidden="true" tabindex="-1"></a>se_normalize <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(</span>
<span id="cb22-197"><a href="statistical-analysis-1.html#cb22-197" aria-hidden="true" tabindex="-1"></a>                      <span class="at">object =</span> se_impute,</span>
<span id="cb22-198"><a href="statistical-analysis-1.html#cb22-198" aria-hidden="true" tabindex="-1"></a>                      <span class="at">rowNorm =</span> <span class="st">&quot;None&quot;</span>,</span>
<span id="cb22-199"><a href="statistical-analysis-1.html#cb22-199" aria-hidden="true" tabindex="-1"></a>                      <span class="at">transNorm =</span> <span class="st">&quot;LogNorm&quot;</span>,</span>
<span id="cb22-200"><a href="statistical-analysis-1.html#cb22-200" aria-hidden="true" tabindex="-1"></a>                      <span class="at">scaleNorm =</span> <span class="st">&quot;ParetoNorm&quot;</span>)</span></code></pre></div>
<pre><code>## Row norm: N/A
## Data Transformation norm: Log10 Normalization
## Data Scaling norm: Pareto Scaling</code></pre>
<pre><code>## All data values are numeric.</code></pre>
<pre><code>## A total of 0 (0%) missing values were detected.</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="statistical-analysis-1.html#cb26-1" aria-hidden="true" tabindex="-1"></a>se_normalize</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 48 45 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(48): M_52603 M_19130 ... M_63681 M_63361
## rowData names(0):
## colnames(45): P101001 P101004 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="normalization-by-poma-r-package" class="section level4 hasAnchor" number="2.1.6.2">
<h4><span class="header-section-number">2.1.6.2</span> Normalization by <strong>POMA</strong> R package<a href="statistical-analysis-1.html#normalization-by-poma-r-package" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="statistical-analysis-1.html#cb28-1" aria-hidden="true" tabindex="-1"></a>none <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_impute, <span class="at">method =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb28-2"><a href="statistical-analysis-1.html#cb28-2" aria-hidden="true" tabindex="-1"></a>auto_scaling <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_impute, <span class="at">method =</span> <span class="st">&quot;auto_scaling&quot;</span>)</span>
<span id="cb28-3"><a href="statistical-analysis-1.html#cb28-3" aria-hidden="true" tabindex="-1"></a>evel_scaling <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_impute, <span class="at">method =</span> <span class="st">&quot;level_scaling&quot;</span>)</span>
<span id="cb28-4"><a href="statistical-analysis-1.html#cb28-4" aria-hidden="true" tabindex="-1"></a>log_scaling <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_impute, <span class="at">method =</span> <span class="st">&quot;log_scaling&quot;</span>)</span>
<span id="cb28-5"><a href="statistical-analysis-1.html#cb28-5" aria-hidden="true" tabindex="-1"></a>log_transformation <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_impute, <span class="at">method =</span> <span class="st">&quot;log_transformation&quot;</span>)</span>
<span id="cb28-6"><a href="statistical-analysis-1.html#cb28-6" aria-hidden="true" tabindex="-1"></a>vast_scaling <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_impute, <span class="at">method =</span> <span class="st">&quot;vast_scaling&quot;</span>)</span>
<span id="cb28-7"><a href="statistical-analysis-1.html#cb28-7" aria-hidden="true" tabindex="-1"></a>se_normalize_v2 <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_impute, <span class="at">method =</span> <span class="st">&quot;log_pareto&quot;</span>)</span>
<span id="cb28-8"><a href="statistical-analysis-1.html#cb28-8" aria-hidden="true" tabindex="-1"></a>se_normalize_v2</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 48 45 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(48): M_52603 M_19130 ... M_63681 M_63361
## rowData names(0):
## colnames(45): P101001 P101004 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="comparison-of-unnormalized-and-normalized-dataset" class="section level4 hasAnchor" number="2.1.6.3">
<h4><span class="header-section-number">2.1.6.3</span> Comparison of unnormalized and normalized dataset<a href="statistical-analysis-1.html#comparison-of-unnormalized-and-normalized-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>boxplot</li>
</ul>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="statistical-analysis-1.html#cb30-1" aria-hidden="true" tabindex="-1"></a>pl_unnor <span class="ot">&lt;-</span> <span class="fu">PomaBoxplots</span>(se_impute, <span class="at">group =</span> <span class="st">&quot;samples&quot;</span>, <span class="at">jitter =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb30-2"><a href="statistical-analysis-1.html#cb30-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Not Normalized&quot;</span>) <span class="sc">+</span></span>
<span id="cb30-3"><a href="statistical-analysis-1.html#cb30-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="co"># data before normalization</span></span>
<span id="cb30-4"><a href="statistical-analysis-1.html#cb30-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-5"><a href="statistical-analysis-1.html#cb30-5" aria-hidden="true" tabindex="-1"></a>pl_nor <span class="ot">&lt;-</span> <span class="fu">PomaBoxplots</span>(se_normalize, <span class="at">group =</span> <span class="st">&quot;samples&quot;</span>, <span class="at">jitter =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb30-6"><a href="statistical-analysis-1.html#cb30-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Normalized&quot;</span>) <span class="co"># data after normalization</span></span>
<span id="cb30-7"><a href="statistical-analysis-1.html#cb30-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-8"><a href="statistical-analysis-1.html#cb30-8" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(pl_unnor, pl_nor, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">align =</span> <span class="st">&quot;v&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-13-1.png" width="1344" /></p>
<ul>
<li>density</li>
</ul>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="statistical-analysis-1.html#cb31-1" aria-hidden="true" tabindex="-1"></a>pl_unnor <span class="ot">&lt;-</span> <span class="fu">PomaDensity</span>(se_impute, <span class="at">group =</span> <span class="st">&quot;features&quot;</span>) <span class="sc">+</span></span>
<span id="cb31-2"><a href="statistical-analysis-1.html#cb31-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Not Normalized&quot;</span>) <span class="sc">+</span></span>
<span id="cb31-3"><a href="statistical-analysis-1.html#cb31-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="co"># data before normalization</span></span>
<span id="cb31-4"><a href="statistical-analysis-1.html#cb31-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-5"><a href="statistical-analysis-1.html#cb31-5" aria-hidden="true" tabindex="-1"></a>pl_nor <span class="ot">&lt;-</span> <span class="fu">PomaDensity</span>(se_normalize, <span class="at">group =</span> <span class="st">&quot;features&quot;</span>) <span class="sc">+</span></span>
<span id="cb31-6"><a href="statistical-analysis-1.html#cb31-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Normalized&quot;</span>) <span class="co"># data after normalization</span></span>
<span id="cb31-7"><a href="statistical-analysis-1.html#cb31-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-8"><a href="statistical-analysis-1.html#cb31-8" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(pl_unnor, pl_nor, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">align =</span> <span class="st">&quot;v&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-14-1.png" width="960" /></p>
</div>
</div>
<div id="removing-outliers" class="section level3 hasAnchor" number="2.1.7">
<h3><span class="header-section-number">2.1.7</span> Removing outliers<a href="statistical-analysis-1.html#removing-outliers" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="statistical-analysis-1.html#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">PomaOutliers</span>(se_normalize, <span class="at">do =</span> <span class="st">&quot;analyze&quot;</span>)<span class="sc">$</span>polygon_plot <span class="co"># to explore</span></span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-15-1.png" width="960" /></p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="statistical-analysis-1.html#cb33-1" aria-hidden="true" tabindex="-1"></a>se_processed <span class="ot">&lt;-</span> <span class="fu">PomaOutliers</span>(se_normalize, <span class="at">do =</span> <span class="st">&quot;clean&quot;</span>) <span class="co"># to remove outliers</span></span>
<span id="cb33-2"><a href="statistical-analysis-1.html#cb33-2" aria-hidden="true" tabindex="-1"></a>se_processed</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 48 43 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(48): M_52603 M_19130 ... M_63681 M_63361
## rowData names(0):
## colnames(43): P101001 P101004 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="saving-datasets-into-rds-files" class="section level3 hasAnchor" number="2.1.8">
<h3><span class="header-section-number">2.1.8</span> Saving datasets into RDS files<a href="statistical-analysis-1.html#saving-datasets-into-rds-files" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="statistical-analysis-1.html#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="st">&quot;./dataset/POMA/&quot;</span>)) {</span>
<span id="cb35-2"><a href="statistical-analysis-1.html#cb35-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="st">&quot;./dataset/POMA/&quot;</span>)</span>
<span id="cb35-3"><a href="statistical-analysis-1.html#cb35-3" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb35-4"><a href="statistical-analysis-1.html#cb35-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-5"><a href="statistical-analysis-1.html#cb35-5" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(se_filter, <span class="st">&quot;./dataset/POMA/se_filter.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span>
<span id="cb35-6"><a href="statistical-analysis-1.html#cb35-6" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(se_impute, <span class="st">&quot;./dataset/POMA/se_impute.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span>
<span id="cb35-7"><a href="statistical-analysis-1.html#cb35-7" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(se_normalize, <span class="st">&quot;./dataset/POMA/se_normalize.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span>
<span id="cb35-8"><a href="statistical-analysis-1.html#cb35-8" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(se_processed, <span class="st">&quot;./dataset/POMA/se_processed.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<hr />
</div>
</div>
<div id="cluster-analysis" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Cluster Analysis<a href="statistical-analysis-1.html#cluster-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<blockquote>
<p>Hierarchical clustering is an alternative approach to k-means clustering for identifying groups in the dataset. It does not require us to pre-specify the number of clusters to be generated as is required by the k-means approach. Furthermore, hierarchical clustering has an added advantage over K-means clustering in that it results in an attractive tree-based representation of the observations, called a dendrogram.</p>
</blockquote>
<p><strong>Note: Please remember to <a href="statistical-analysis-1.html#data-preprocessing">preprocess</a> your data before clustering</strong></p>
<div id="loading-packages" class="section level3 hasAnchor" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> Loading packages<a href="statistical-analysis-1.html#loading-packages" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="statistical-analysis-1.html#cb36-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">warning =</span> F)</span>
<span id="cb36-2"><a href="statistical-analysis-1.html#cb36-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-3"><a href="statistical-analysis-1.html#cb36-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb36-4"><a href="statistical-analysis-1.html#cb36-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb36-5"><a href="statistical-analysis-1.html#cb36-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(POMA)</span>
<span id="cb36-6"><a href="statistical-analysis-1.html#cb36-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb36-7"><a href="statistical-analysis-1.html#cb36-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb36-8"><a href="statistical-analysis-1.html#cb36-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cluster)    <span class="co"># clustering algorithms</span></span>
<span id="cb36-9"><a href="statistical-analysis-1.html#cb36-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(factoextra) <span class="co"># clustering visualization</span></span>
<span id="cb36-10"><a href="statistical-analysis-1.html#cb36-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dendextend) <span class="co"># for comparing two dendrograms</span></span>
<span id="cb36-11"><a href="statistical-analysis-1.html#cb36-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-12"><a href="statistical-analysis-1.html#cb36-12" aria-hidden="true" tabindex="-1"></a><span class="co"># rm(list = ls())</span></span>
<span id="cb36-13"><a href="statistical-analysis-1.html#cb36-13" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb36-14"><a href="statistical-analysis-1.html#cb36-14" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">future.globals.maxSize =</span> <span class="dv">1000</span> <span class="sc">*</span> <span class="dv">1024</span><span class="sc">^</span><span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="importing-data" class="section level3 hasAnchor" number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> Importing data<a href="statistical-analysis-1.html#importing-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The input data sets are from the previous chapter.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="statistical-analysis-1.html#cb37-1" aria-hidden="true" tabindex="-1"></a>se_normalize <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;./dataset/POMA/se_normalize.RDS&quot;</span>)</span></code></pre></div>
</div>
<div id="hierarchical-clustering" class="section level3 hasAnchor" number="2.2.3">
<h3><span class="header-section-number">2.2.3</span> Hierarchical Clustering<a href="statistical-analysis-1.html#hierarchical-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Hierarchical clustering can be divided into two main types: <a href="statistical-analysis-1.html#agglomerative-hierarchical-clustering"><strong>agglomerative</strong></a> and <a href="statistical-analysis-1.html#divisive-hierarchical-clustering"><strong>divisive</strong></a>.</p>
<p>Calculate dissimilarity</p>
<blockquote>
<p>However, a bigger question is: How do we measure the dissimilarity between two clusters of observations? A number of different cluster agglomeration methods (i.e, linkage methods) have been developed to answer to this question. The most common types methods are:</p>
<ol style="list-style-type: decimal">
<li><strong>Maximum or complete linkage clustering</strong>: It computes all pairwise dissimilarities between the elements in cluster 1 and the elements in cluster 2, and considers the largest value (i.e., maximum value) of these dissimilarities as the distance between the two clusters. It tends to produce more compact clusters.</li>
</ol>
</blockquote>
<ol start="2" style="list-style-type: decimal">
<li><p><strong>Minimum or single linkage clustering</strong>: It computes all pairwise dissimilarities between the elements in cluster 1 and the elements in cluster 2, and considers the smallest of these dissimilarities as a linkage criterion. It tends to produce long, “loose” clusters.</p></li>
<li><p><strong>Mean or average linkage clustering</strong>: It computes all pairwise dissimilarities between the elements in cluster 1 and the elements in cluster 2, and considers the average of these dissimilarities as the distance between the two clusters.</p></li>
<li><p><strong>Centroid linkage clustering</strong>: It computes the dissimilarity between the centroid for cluster 1 (a mean vector of length p variables) and the centroid for cluster 2.</p></li>
<li><p><strong>Ward’s minimum variance method</strong>: It minimizes the total within-cluster variance. At each step the pair of clusters with minimum between-cluster distance are merged.</p></li>
</ol>
<p>Data Processing:</p>
<ol style="list-style-type: decimal">
<li><p>Rows are observations (individuals) and columns are variables.</p></li>
<li><p>Any missing value in the data must be removed or estimated.</p></li>
<li><p>The data must be standardized (i.e., scaled) to make variables comparable. Recall that, standardization consists of transforming the variables such that they have mean zero and standard deviation one.</p></li>
</ol>
<p>Functions to computing hierarchical clustering:</p>
<ol style="list-style-type: decimal">
<li><p><em>hclust</em> [in stats package] and <em>agnes</em> [in cluster package] for agglomerative hierarchical clustering.</p></li>
<li><p><em>diana</em> [in cluster package] for divisive hierarchical clustering.</p></li>
</ol>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="statistical-analysis-1.html#cb38-1" aria-hidden="true" tabindex="-1"></a>HieraCluster <span class="ot">&lt;-</span> <span class="cf">function</span>(object,</span>
<span id="cb38-2"><a href="statistical-analysis-1.html#cb38-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">method_dis =</span> <span class="fu">c</span>(<span class="st">&quot;euclidean&quot;</span>, <span class="st">&quot;bray&quot;</span>),</span>
<span id="cb38-3"><a href="statistical-analysis-1.html#cb38-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">method_cluster =</span> <span class="fu">c</span>(<span class="st">&quot;average&quot;</span>, <span class="st">&quot;single&quot;</span>, <span class="st">&quot;complete&quot;</span>, <span class="st">&quot;ward&quot;</span>, <span class="st">&quot;ward.D2&quot;</span>),</span>
<span id="cb38-4"><a href="statistical-analysis-1.html#cb38-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">cluster_type =</span> <span class="fu">c</span>(<span class="st">&quot;Agglomerative&quot;</span>, <span class="st">&quot;Divisive&quot;</span>),</span>
<span id="cb38-5"><a href="statistical-analysis-1.html#cb38-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">tree_num =</span> <span class="dv">4</span>) {</span>
<span id="cb38-6"><a href="statistical-analysis-1.html#cb38-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb38-7"><a href="statistical-analysis-1.html#cb38-7" aria-hidden="true" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) </span>
<span id="cb38-8"><a href="statistical-analysis-1.html#cb38-8" aria-hidden="true" tabindex="-1"></a>  metadata_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object)</span>
<span id="cb38-9"><a href="statistical-analysis-1.html#cb38-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb38-10"><a href="statistical-analysis-1.html#cb38-10" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">t</span>(features_tab)</span>
<span id="cb38-11"><a href="statistical-analysis-1.html#cb38-11" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (cluster_type <span class="sc">==</span> <span class="st">&quot;Agglomerative&quot;</span>) {</span>
<span id="cb38-12"><a href="statistical-analysis-1.html#cb38-12" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Agglomerative Hierarchical Clustering </span></span>
<span id="cb38-13"><a href="statistical-analysis-1.html#cb38-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Dissimilarity matrix</span></span>
<span id="cb38-14"><a href="statistical-analysis-1.html#cb38-14" aria-hidden="true" tabindex="-1"></a>    d <span class="ot">&lt;-</span> <span class="fu">dist</span>(df, <span class="at">method =</span> method_dis)</span>
<span id="cb38-15"><a href="statistical-analysis-1.html#cb38-15" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-16"><a href="statistical-analysis-1.html#cb38-16" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Hierarchical clustering using Linkage method</span></span>
<span id="cb38-17"><a href="statistical-analysis-1.html#cb38-17" aria-hidden="true" tabindex="-1"></a>    hc <span class="ot">&lt;-</span> <span class="fu">hclust</span>(d, <span class="at">method =</span> method_cluster)</span>
<span id="cb38-18"><a href="statistical-analysis-1.html#cb38-18" aria-hidden="true" tabindex="-1"></a>    <span class="co"># hc &lt;- agnes(df, method = method_cluster)</span></span>
<span id="cb38-19"><a href="statistical-analysis-1.html#cb38-19" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-20"><a href="statistical-analysis-1.html#cb38-20" aria-hidden="true" tabindex="-1"></a>    <span class="do">####### identifying the strongest clustering structure ################</span></span>
<span id="cb38-21"><a href="statistical-analysis-1.html#cb38-21" aria-hidden="true" tabindex="-1"></a>    <span class="co"># # methods to assess</span></span>
<span id="cb38-22"><a href="statistical-analysis-1.html#cb38-22" aria-hidden="true" tabindex="-1"></a>    <span class="co"># m &lt;- c( &quot;average&quot;, &quot;single&quot;, &quot;complete&quot;, &quot;ward&quot;)</span></span>
<span id="cb38-23"><a href="statistical-analysis-1.html#cb38-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># names(m) &lt;- c( &quot;average&quot;, &quot;single&quot;, &quot;complete&quot;, &quot;ward&quot;)</span></span>
<span id="cb38-24"><a href="statistical-analysis-1.html#cb38-24" aria-hidden="true" tabindex="-1"></a>    <span class="co"># </span></span>
<span id="cb38-25"><a href="statistical-analysis-1.html#cb38-25" aria-hidden="true" tabindex="-1"></a>    <span class="co"># # function to compute coefficient</span></span>
<span id="cb38-26"><a href="statistical-analysis-1.html#cb38-26" aria-hidden="true" tabindex="-1"></a>    <span class="co"># ac &lt;- function(x) {</span></span>
<span id="cb38-27"><a href="statistical-analysis-1.html#cb38-27" aria-hidden="true" tabindex="-1"></a>    <span class="co">#   agnes(df, method = x)$ac</span></span>
<span id="cb38-28"><a href="statistical-analysis-1.html#cb38-28" aria-hidden="true" tabindex="-1"></a>    <span class="co"># }</span></span>
<span id="cb38-29"><a href="statistical-analysis-1.html#cb38-29" aria-hidden="true" tabindex="-1"></a>    <span class="co"># </span></span>
<span id="cb38-30"><a href="statistical-analysis-1.html#cb38-30" aria-hidden="true" tabindex="-1"></a>    <span class="co"># map_dbl(m, ac)     </span></span>
<span id="cb38-31"><a href="statistical-analysis-1.html#cb38-31" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (cluster_type <span class="sc">==</span> <span class="st">&quot;Divisive&quot;</span>) {</span>
<span id="cb38-32"><a href="statistical-analysis-1.html#cb38-32" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Divisive Hierarchical Clustering </span></span>
<span id="cb38-33"><a href="statistical-analysis-1.html#cb38-33" aria-hidden="true" tabindex="-1"></a>    hc <span class="ot">&lt;-</span> <span class="fu">diana</span>(df, <span class="at">metric =</span> method_dis)</span>
<span id="cb38-34"><a href="statistical-analysis-1.html#cb38-34" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb38-35"><a href="statistical-analysis-1.html#cb38-35" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb38-36"><a href="statistical-analysis-1.html#cb38-36" aria-hidden="true" tabindex="-1"></a>  hc_res <span class="ot">&lt;-</span> <span class="fu">as.hclust</span>(hc)</span>
<span id="cb38-37"><a href="statistical-analysis-1.html#cb38-37" aria-hidden="true" tabindex="-1"></a>  sub_grp <span class="ot">&lt;-</span> <span class="fu">cutree</span>(hc_res, <span class="at">k =</span> tree_num)</span>
<span id="cb38-38"><a href="statistical-analysis-1.html#cb38-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-39"><a href="statistical-analysis-1.html#cb38-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(hc_res, <span class="at">cex =</span> <span class="fl">0.6</span>)</span>
<span id="cb38-40"><a href="statistical-analysis-1.html#cb38-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rect.hclust</span>(hc_res, <span class="at">k =</span> tree_num, <span class="at">border =</span> <span class="dv">2</span><span class="sc">:</span>(tree_num<span class="sc">+</span><span class="dv">1</span>)) </span>
<span id="cb38-41"><a href="statistical-analysis-1.html#cb38-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb38-42"><a href="statistical-analysis-1.html#cb38-42" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">data=</span>df,</span>
<span id="cb38-43"><a href="statistical-analysis-1.html#cb38-43" aria-hidden="true" tabindex="-1"></a>              <span class="at">cluster=</span>sub_grp,</span>
<span id="cb38-44"><a href="statistical-analysis-1.html#cb38-44" aria-hidden="true" tabindex="-1"></a>              <span class="at">hc=</span>hc_res)</span>
<span id="cb38-45"><a href="statistical-analysis-1.html#cb38-45" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb38-46"><a href="statistical-analysis-1.html#cb38-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb38-47"><a href="statistical-analysis-1.html#cb38-47" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<div id="agglomerative-hierarchical-clustering" class="section level4 hasAnchor" number="2.2.3.1">
<h4><span class="header-section-number">2.2.3.1</span> Agglomerative Hierarchical Clustering<a href="statistical-analysis-1.html#agglomerative-hierarchical-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<blockquote>
<p><strong>Agglomerative clustering</strong>: It’s also known as AGNES (Agglomerative Nesting). It works in a bottom-up manner. That is, each object is initially considered as a single-element cluster (leaf). At each step of the algorithm, the two clusters that are the most similar are combined into a new bigger cluster (nodes). This procedure is iterated until all points are member of just one single big cluster (root). The result is a tree which can be plotted as a dendrogram.</p>
</blockquote>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="statistical-analysis-1.html#cb39-1" aria-hidden="true" tabindex="-1"></a>Agg_hc_res <span class="ot">&lt;-</span> <span class="fu">HieraCluster</span>(</span>
<span id="cb39-2"><a href="statistical-analysis-1.html#cb39-2" aria-hidden="true" tabindex="-1"></a>      <span class="at">object =</span> se_normalize,</span>
<span id="cb39-3"><a href="statistical-analysis-1.html#cb39-3" aria-hidden="true" tabindex="-1"></a>      <span class="at">method_dis =</span> <span class="st">&quot;euclidean&quot;</span>,</span>
<span id="cb39-4"><a href="statistical-analysis-1.html#cb39-4" aria-hidden="true" tabindex="-1"></a>      <span class="at">method_cluster =</span> <span class="st">&quot;ward.D2&quot;</span>,</span>
<span id="cb39-5"><a href="statistical-analysis-1.html#cb39-5" aria-hidden="true" tabindex="-1"></a>      <span class="at">cluster_type =</span> <span class="st">&quot;Agglomerative&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-20-1.png" width="768" /></p>
<ul>
<li>Visualization: visualize the result in a scatter plot</li>
</ul>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="statistical-analysis-1.html#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_cluster</span>(<span class="fu">list</span>(<span class="at">data =</span> Agg_hc_res<span class="sc">$</span>data, </span>
<span id="cb40-2"><a href="statistical-analysis-1.html#cb40-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> Agg_hc_res<span class="sc">$</span>cluster))</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-21-1.png" width="768" /></p>
</div>
<div id="divisive-hierarchical-clustering" class="section level4 hasAnchor" number="2.2.3.2">
<h4><span class="header-section-number">2.2.3.2</span> Divisive Hierarchical Clustering<a href="statistical-analysis-1.html#divisive-hierarchical-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<blockquote>
<p><strong>Divisive hierarchical clustering</strong>: It’s also known as DIANA (Divise Analysis) and it works in a top-down manner. The algorithm is an inverse order of AGNES. It begins with the root, in which all objects are included in a single cluster. At each step of iteration, the most heterogeneous cluster is divided into two. The process is iterated until all objects are in their own cluster.</p>
</blockquote>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="statistical-analysis-1.html#cb41-1" aria-hidden="true" tabindex="-1"></a>Div_hc_res <span class="ot">&lt;-</span> <span class="fu">HieraCluster</span>(</span>
<span id="cb41-2"><a href="statistical-analysis-1.html#cb41-2" aria-hidden="true" tabindex="-1"></a>      <span class="at">object =</span> se_normalize,</span>
<span id="cb41-3"><a href="statistical-analysis-1.html#cb41-3" aria-hidden="true" tabindex="-1"></a>      <span class="at">method_dis =</span> <span class="st">&quot;euclidean&quot;</span>,</span>
<span id="cb41-4"><a href="statistical-analysis-1.html#cb41-4" aria-hidden="true" tabindex="-1"></a>      <span class="at">method_cluster =</span> <span class="st">&quot;ward&quot;</span>,</span>
<span id="cb41-5"><a href="statistical-analysis-1.html#cb41-5" aria-hidden="true" tabindex="-1"></a>      <span class="at">cluster_type =</span> <span class="st">&quot;Divisive&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-22-1.png" width="768" /></p>
<ul>
<li>Visualization: visualize the result in a scatter plot</li>
</ul>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="statistical-analysis-1.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_cluster</span>(<span class="fu">list</span>(<span class="at">data =</span> Div_hc_res<span class="sc">$</span>data, </span>
<span id="cb42-2"><a href="statistical-analysis-1.html#cb42-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">cluster =</span> Div_hc_res<span class="sc">$</span>cluster))</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-23-1.png" width="768" /></p>
</div>
<div id="comparison" class="section level4 hasAnchor" number="2.2.3.3">
<h4><span class="header-section-number">2.2.3.3</span> Comparison<a href="statistical-analysis-1.html#comparison" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>dendrograms: In the dendrogram displayed above, each leaf corresponds to one observation.</li>
</ul>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="statistical-analysis-1.html#cb43-1" aria-hidden="true" tabindex="-1"></a>Agg_hc_dend <span class="ot">&lt;-</span> <span class="fu">as.dendrogram</span>(Agg_hc_res<span class="sc">$</span>hc)</span>
<span id="cb43-2"><a href="statistical-analysis-1.html#cb43-2" aria-hidden="true" tabindex="-1"></a>Div_hc_dend <span class="ot">&lt;-</span> <span class="fu">as.dendrogram</span>(Div_hc_res<span class="sc">$</span>hc)</span>
<span id="cb43-3"><a href="statistical-analysis-1.html#cb43-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-4"><a href="statistical-analysis-1.html#cb43-4" aria-hidden="true" tabindex="-1"></a><span class="fu">tanglegram</span>(Agg_hc_dend, Div_hc_dend)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-24-1.png" width="768" /></p>
<ul>
<li>tanglegrams</li>
</ul>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="statistical-analysis-1.html#cb44-1" aria-hidden="true" tabindex="-1"></a>dend_list <span class="ot">&lt;-</span> <span class="fu">dendlist</span>(Agg_hc_dend, Div_hc_dend)</span>
<span id="cb44-2"><a href="statistical-analysis-1.html#cb44-2" aria-hidden="true" tabindex="-1"></a><span class="fu">tanglegram</span>(Agg_hc_dend, Div_hc_dend,</span>
<span id="cb44-3"><a href="statistical-analysis-1.html#cb44-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">highlight_distinct_edges =</span> <span class="cn">FALSE</span>, <span class="co"># Turn-off dashed lines</span></span>
<span id="cb44-4"><a href="statistical-analysis-1.html#cb44-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">common_subtrees_color_lines =</span> <span class="cn">FALSE</span>, <span class="co"># Turn-off line colors</span></span>
<span id="cb44-5"><a href="statistical-analysis-1.html#cb44-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">common_subtrees_color_branches =</span> <span class="cn">TRUE</span>, <span class="co"># Color common branches </span></span>
<span id="cb44-6"><a href="statistical-analysis-1.html#cb44-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">&quot;entanglement =&quot;</span>, <span class="fu">round</span>(<span class="fu">entanglement</span>(dend_list), <span class="dv">2</span>)))</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-25-1.png" width="768" /></p>
</div>
<div id="determining-optimal-clusters" class="section level4 hasAnchor" number="2.2.3.4">
<h4><span class="header-section-number">2.2.3.4</span> Determining Optimal Clusters<a href="statistical-analysis-1.html#determining-optimal-clusters" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Elbow plot</li>
</ul>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="statistical-analysis-1.html#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_nbclust</span>(Agg_hc_res<span class="sc">$</span>data, <span class="at">FUN =</span> hcut, <span class="at">method =</span> <span class="st">&quot;wss&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-26-1.png" width="576" /></p>
<ul>
<li>Average Silhouette Method</li>
</ul>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="statistical-analysis-1.html#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_nbclust</span>(Agg_hc_res<span class="sc">$</span>data, <span class="at">FUN =</span> hcut, <span class="at">method =</span> <span class="st">&quot;silhouette&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-27-1.png" width="576" /></p>
<ul>
<li>Gap Statistic Method</li>
</ul>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="statistical-analysis-1.html#cb47-1" aria-hidden="true" tabindex="-1"></a>gap_stat <span class="ot">&lt;-</span> <span class="fu">clusGap</span>(Agg_hc_res<span class="sc">$</span>data, <span class="at">FUN =</span> hcut, <span class="at">nstart =</span> <span class="dv">25</span>, <span class="at">K.max =</span> <span class="dv">10</span>, <span class="at">B =</span> <span class="dv">50</span>)</span></code></pre></div>
<pre><code>## Clustering k = 1,2,..., K.max (= 10): .. done
## Bootstrapping, b = 1,2,..., B (= 50)  [one &quot;.&quot; per sample]:
## .................................................. 50</code></pre>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="statistical-analysis-1.html#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_gap_stat</span>(gap_stat)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-28-1.png" width="576" /></p>
</div>
</div>
<div id="partitional-clustering" class="section level3 hasAnchor" number="2.2.4">
<h3><span class="header-section-number">2.2.4</span> Partitional Clustering<a href="statistical-analysis-1.html#partitional-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<blockquote>
<p>K-means clustering is the most commonly used unsupervised machine learning algorithm for partitioning a given data set into a set of k groups (i.e. k clusters), where k represents the number of groups pre-specified by the analyst.</p>
</blockquote>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="statistical-analysis-1.html#cb50-1" aria-hidden="true" tabindex="-1"></a>PartCluster <span class="ot">&lt;-</span> <span class="cf">function</span>(object,</span>
<span id="cb50-2"><a href="statistical-analysis-1.html#cb50-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">cluster_num =</span> <span class="dv">4</span>) {</span>
<span id="cb50-3"><a href="statistical-analysis-1.html#cb50-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-4"><a href="statistical-analysis-1.html#cb50-4" aria-hidden="true" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) </span>
<span id="cb50-5"><a href="statistical-analysis-1.html#cb50-5" aria-hidden="true" tabindex="-1"></a>  metadata_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object)</span>
<span id="cb50-6"><a href="statistical-analysis-1.html#cb50-6" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">t</span>(features_tab)</span>
<span id="cb50-7"><a href="statistical-analysis-1.html#cb50-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-8"><a href="statistical-analysis-1.html#cb50-8" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">kmeans</span>(df, <span class="at">centers =</span> cluster_num)  </span>
<span id="cb50-9"><a href="statistical-analysis-1.html#cb50-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># show clusters</span></span>
<span id="cb50-10"><a href="statistical-analysis-1.html#cb50-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">fviz_cluster</span>(<span class="fu">list</span>(<span class="at">data =</span> df, </span>
<span id="cb50-11"><a href="statistical-analysis-1.html#cb50-11" aria-hidden="true" tabindex="-1"></a>                    <span class="at">cluster =</span> res<span class="sc">$</span>cluster))) </span>
<span id="cb50-12"><a href="statistical-analysis-1.html#cb50-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb50-13"><a href="statistical-analysis-1.html#cb50-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb50-14"><a href="statistical-analysis-1.html#cb50-14" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb50-15"><a href="statistical-analysis-1.html#cb50-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-16"><a href="statistical-analysis-1.html#cb50-16" aria-hidden="true" tabindex="-1"></a>Kcluster_res <span class="ot">&lt;-</span> <span class="fu">PartCluster</span>(</span>
<span id="cb50-17"><a href="statistical-analysis-1.html#cb50-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">object =</span> se_normalize,</span>
<span id="cb50-18"><a href="statistical-analysis-1.html#cb50-18" aria-hidden="true" tabindex="-1"></a>        <span class="at">cluster_num =</span> <span class="dv">4</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-29-1.png" width="768" /></p>
<hr />
</div>
</div>
<div id="chemometrics-analysis" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Chemometrics Analysis<a href="statistical-analysis-1.html#chemometrics-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The functions for Chemometrics Analysis in <strong>POMA</strong> <span class="citation">(<a href="#ref-castellano2021pomashiny" role="doc-biblioref">Castellano-Escuder et al. 2021</a>)</span> implemented from <strong>mixOmics</strong> <span class="citation">(<a href="#ref-rohart2017mixomics" role="doc-biblioref">Rohart et al. 2017</a>)</span>.</p>
<p><strong>Note: Please also remember to <a href="statistical-analysis-1.html#data-preprocessing">preprocess</a> your data before running this sub-chapter.</strong></p>
<div id="loading-packages-1" class="section level3 hasAnchor" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> Loading packages<a href="statistical-analysis-1.html#loading-packages-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="statistical-analysis-1.html#cb51-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">warning =</span> F, <span class="at">message =</span> F)</span>
<span id="cb51-2"><a href="statistical-analysis-1.html#cb51-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-3"><a href="statistical-analysis-1.html#cb51-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb51-4"><a href="statistical-analysis-1.html#cb51-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb51-5"><a href="statistical-analysis-1.html#cb51-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(POMA)</span>
<span id="cb51-6"><a href="statistical-analysis-1.html#cb51-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb51-7"><a href="statistical-analysis-1.html#cb51-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggraph)</span>
<span id="cb51-8"><a href="statistical-analysis-1.html#cb51-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb51-9"><a href="statistical-analysis-1.html#cb51-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb51-10"><a href="statistical-analysis-1.html#cb51-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-11"><a href="statistical-analysis-1.html#cb51-11" aria-hidden="true" tabindex="-1"></a><span class="co"># rm(list = ls())</span></span>
<span id="cb51-12"><a href="statistical-analysis-1.html#cb51-12" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb51-13"><a href="statistical-analysis-1.html#cb51-13" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">future.globals.maxSize =</span> <span class="dv">1000</span> <span class="sc">*</span> <span class="dv">1024</span><span class="sc">^</span><span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="importing-data-1" class="section level3 hasAnchor" number="2.3.2">
<h3><span class="header-section-number">2.3.2</span> Importing data<a href="statistical-analysis-1.html#importing-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The input data sets are from the previous chapter.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="statistical-analysis-1.html#cb52-1" aria-hidden="true" tabindex="-1"></a>se_processed <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;./dataset/POMA/se_processed.RDS&quot;</span>)</span></code></pre></div>
</div>
<div id="principal-component-analysis-pca" class="section level3 hasAnchor" number="2.3.3">
<h3><span class="header-section-number">2.3.3</span> Principal Component Analysis (PCA)<a href="statistical-analysis-1.html#principal-component-analysis-pca" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<blockquote>
<p>The aim of PCA (Jolliffe 2005) is to reduce the dimensionality of the data whilst retaining as much information as possible. ‘Information’ is referred here as variance. The idea is to create uncorrelated artificial variables called principal components (PCs) that combine in a linear manner the original (possibly correlated) variables.</p>
</blockquote>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="statistical-analysis-1.html#cb53-1" aria-hidden="true" tabindex="-1"></a>poma_pca <span class="ot">&lt;-</span> <span class="fu">PomaMultivariate</span>(se_processed, <span class="at">method =</span> <span class="st">&quot;pca&quot;</span>)</span>
<span id="cb53-2"><a href="statistical-analysis-1.html#cb53-2" aria-hidden="true" tabindex="-1"></a>poma_pca<span class="sc">$</span>scoresplot <span class="sc">+</span></span>
<span id="cb53-3"><a href="statistical-analysis-1.html#cb53-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Scores Plot (pca)&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-32-1.png" width="768" /></p>
</div>
<div id="partial-least-squares-discriminant-analysis-pls-da" class="section level3 hasAnchor" number="2.3.4">
<h3><span class="header-section-number">2.3.4</span> Partial Least Squares-Discriminant Analysis (PLS-DA)<a href="statistical-analysis-1.html#partial-least-squares-discriminant-analysis-pls-da" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<blockquote>
<p>Partial Least Squares (PLS) regression is a multivariate methodology which relates two data matrices X (e.g. transcriptomics) and Y (e.g. lipids). PLS goes beyond traditional multiple regression by modelling the structure of both matrices. Unlike traditional multiple regression models, it is not limited to uncorrelated variables. One of the many advantages of PLS is that it can handle many noisy, collinear (correlated) and missing variables and can also simultaneously model several response variables in Y.</p>
</blockquote>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="statistical-analysis-1.html#cb54-1" aria-hidden="true" tabindex="-1"></a>poma_plsda <span class="ot">&lt;-</span> <span class="fu">PomaMultivariate</span>(se_processed, <span class="at">method =</span> <span class="st">&quot;plsda&quot;</span>)</span></code></pre></div>
<ul>
<li>scatter plot</li>
</ul>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="statistical-analysis-1.html#cb55-1" aria-hidden="true" tabindex="-1"></a>poma_plsda<span class="sc">$</span>scoresplot <span class="sc">+</span></span>
<span id="cb55-2"><a href="statistical-analysis-1.html#cb55-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Scores Plot (plsda)&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-34-1.png" width="768" /></p>
<ul>
<li>errors plot</li>
</ul>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="statistical-analysis-1.html#cb56-1" aria-hidden="true" tabindex="-1"></a>poma_plsda<span class="sc">$</span>errors_plsda_plot <span class="sc">+</span></span>
<span id="cb56-2"><a href="statistical-analysis-1.html#cb56-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Error Plot (plsda)&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-35-1.png" width="768" /></p>
</div>
<div id="sparse-partial-least-squares-discriminant-analysis-spls-da" class="section level3 hasAnchor" number="2.3.5">
<h3><span class="header-section-number">2.3.5</span> Sparse Partial Least Squares-Discriminant Analysis (sPLS-DA)<a href="statistical-analysis-1.html#sparse-partial-least-squares-discriminant-analysis-spls-da" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<blockquote>
<p>Even though PLS is highly efficient in a high dimensional context, the interpretability of PLS needed to be improved. sPLS has been recently developed by our team to perform simultaneous variable selection in both data sets X and Y data sets, by including LASSO penalizations in PLS on each pair of loading vectors</p>
</blockquote>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="statistical-analysis-1.html#cb57-1" aria-hidden="true" tabindex="-1"></a>poma_splsda <span class="ot">&lt;-</span> <span class="fu">PomaMultivariate</span>(se_processed, <span class="at">method =</span> <span class="st">&quot;splsda&quot;</span>)</span></code></pre></div>
<ul>
<li>scatter plot</li>
</ul>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="statistical-analysis-1.html#cb58-1" aria-hidden="true" tabindex="-1"></a>poma_splsda<span class="sc">$</span>scoresplot <span class="sc">+</span></span>
<span id="cb58-2"><a href="statistical-analysis-1.html#cb58-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Scores Plot (splsda)&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-37-1.png" width="768" /></p>
</div>
<div id="orthogonal-partial-least-squares-discriminant-analysis-orthopls-da" class="section level3 hasAnchor" number="2.3.6">
<h3><span class="header-section-number">2.3.6</span> Orthogonal Partial Least Squares-Discriminant Analysis (orthoPLS-DA)<a href="statistical-analysis-1.html#orthogonal-partial-least-squares-discriminant-analysis-orthopls-da" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Still in development…</p>
<hr />
</div>
</div>
<div id="univariate-analysis" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Univariate Analysis<a href="statistical-analysis-1.html#univariate-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Univariate analysis explores each variable in a data set, separately and it uses traditional statistical methods on single variable to calculate the statistics, such as fold change, p-value, etc.
<strong>Note: Please also remember to <a href="statistical-analysis-1.html#data-preprocessing">preprocess</a> your data before running this sub-chapter.</strong></p>
<div id="loading-packages-2" class="section level3 hasAnchor" number="2.4.1">
<h3><span class="header-section-number">2.4.1</span> Loading packages<a href="statistical-analysis-1.html#loading-packages-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="statistical-analysis-1.html#cb59-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">warning =</span> F)</span>
<span id="cb59-2"><a href="statistical-analysis-1.html#cb59-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-3"><a href="statistical-analysis-1.html#cb59-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb59-4"><a href="statistical-analysis-1.html#cb59-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb59-5"><a href="statistical-analysis-1.html#cb59-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(POMA)</span>
<span id="cb59-6"><a href="statistical-analysis-1.html#cb59-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb59-7"><a href="statistical-analysis-1.html#cb59-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggraph)</span>
<span id="cb59-8"><a href="statistical-analysis-1.html#cb59-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb59-9"><a href="statistical-analysis-1.html#cb59-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb59-10"><a href="statistical-analysis-1.html#cb59-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-11"><a href="statistical-analysis-1.html#cb59-11" aria-hidden="true" tabindex="-1"></a><span class="co"># rm(list = ls())</span></span>
<span id="cb59-12"><a href="statistical-analysis-1.html#cb59-12" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb59-13"><a href="statistical-analysis-1.html#cb59-13" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">future.globals.maxSize =</span> <span class="dv">1000</span> <span class="sc">*</span> <span class="dv">1024</span><span class="sc">^</span><span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="importing-data-2" class="section level3 hasAnchor" number="2.4.2">
<h3><span class="header-section-number">2.4.2</span> Importing data<a href="statistical-analysis-1.html#importing-data-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The input data sets are from the previous chapter.</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="statistical-analysis-1.html#cb60-1" aria-hidden="true" tabindex="-1"></a>se_impute <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;./dataset/POMA/se_impute.RDS&quot;</span>)</span>
<span id="cb60-2"><a href="statistical-analysis-1.html#cb60-2" aria-hidden="true" tabindex="-1"></a>se_normalize <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;./dataset/POMA/se_normalize.RDS&quot;</span>)</span>
<span id="cb60-3"><a href="statistical-analysis-1.html#cb60-3" aria-hidden="true" tabindex="-1"></a>se_processed <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;./dataset/POMA/se_processed.RDS&quot;</span>)</span></code></pre></div>
</div>
<div id="fold-change-analysis" class="section level3 hasAnchor" number="2.4.3">
<h3><span class="header-section-number">2.4.3</span> Fold Change Analysis<a href="statistical-analysis-1.html#fold-change-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="statistical-analysis-1.html#cb61-1" aria-hidden="true" tabindex="-1"></a>FoldChange <span class="ot">&lt;-</span> <span class="cf">function</span>(object,</span>
<span id="cb61-2"><a href="statistical-analysis-1.html#cb61-2" aria-hidden="true" tabindex="-1"></a>                       group_names,</span>
<span id="cb61-3"><a href="statistical-analysis-1.html#cb61-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">fc_cutoff =</span> <span class="dv">2</span>,</span>
<span id="cb61-4"><a href="statistical-analysis-1.html#cb61-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">cmp_type =</span> <span class="dv">0</span>, </span>
<span id="cb61-5"><a href="statistical-analysis-1.html#cb61-5" aria-hidden="true" tabindex="-1"></a>                       <span class="at">paired =</span> <span class="cn">FALSE</span>) {</span>
<span id="cb61-6"><a href="statistical-analysis-1.html#cb61-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb61-7"><a href="statistical-analysis-1.html#cb61-7" aria-hidden="true" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) </span>
<span id="cb61-8"><a href="statistical-analysis-1.html#cb61-8" aria-hidden="true" tabindex="-1"></a>  metadata_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object)   </span>
<span id="cb61-9"><a href="statistical-analysis-1.html#cb61-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb61-10"><a href="statistical-analysis-1.html#cb61-10" aria-hidden="true" tabindex="-1"></a>  metadata <span class="ot">&lt;-</span> metadata_tab <span class="sc">%&gt;%</span></span>
<span id="cb61-11"><a href="statistical-analysis-1.html#cb61-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb61-12"><a href="statistical-analysis-1.html#cb61-12" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(group <span class="sc">%in%</span> group_names)</span>
<span id="cb61-13"><a href="statistical-analysis-1.html#cb61-13" aria-hidden="true" tabindex="-1"></a>  profile <span class="ot">&lt;-</span> features_tab[, <span class="fu">pmatch</span>(<span class="fu">rownames</span>(metadata), <span class="fu">colnames</span>(features_tab))]</span>
<span id="cb61-14"><a href="statistical-analysis-1.html#cb61-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb61-15"><a href="statistical-analysis-1.html#cb61-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># make sure threshold is above 1</span></span>
<span id="cb61-16"><a href="statistical-analysis-1.html#cb61-16" aria-hidden="true" tabindex="-1"></a>  fc_cutoff <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(fc_cutoff <span class="sc">&gt;</span> <span class="dv">1</span>, fc_cutoff, <span class="dv">1</span><span class="sc">/</span>fc_cutoff)</span>
<span id="cb61-17"><a href="statistical-analysis-1.html#cb61-17" aria-hidden="true" tabindex="-1"></a>  max_cutoff <span class="ot">&lt;-</span> fc_cutoff</span>
<span id="cb61-18"><a href="statistical-analysis-1.html#cb61-18" aria-hidden="true" tabindex="-1"></a>  min_cutoff <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">/</span> fc_cutoff</span>
<span id="cb61-19"><a href="statistical-analysis-1.html#cb61-19" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb61-20"><a href="statistical-analysis-1.html#cb61-20" aria-hidden="true" tabindex="-1"></a>  GetFC <span class="ot">&lt;-</span> <span class="cf">function</span>(x,</span>
<span id="cb61-21"><a href="statistical-analysis-1.html#cb61-21" aria-hidden="true" tabindex="-1"></a>                    y,</span>
<span id="cb61-22"><a href="statistical-analysis-1.html#cb61-22" aria-hidden="true" tabindex="-1"></a>                    cmpType, </span>
<span id="cb61-23"><a href="statistical-analysis-1.html#cb61-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">paired =</span> <span class="cn">FALSE</span>) {</span>
<span id="cb61-24"><a href="statistical-analysis-1.html#cb61-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-25"><a href="statistical-analysis-1.html#cb61-25" aria-hidden="true" tabindex="-1"></a>    <span class="co"># x = profile</span></span>
<span id="cb61-26"><a href="statistical-analysis-1.html#cb61-26" aria-hidden="true" tabindex="-1"></a>    <span class="co"># y = metadata</span></span>
<span id="cb61-27"><a href="statistical-analysis-1.html#cb61-27" aria-hidden="true" tabindex="-1"></a>    <span class="co"># cmpType = cmp_type</span></span>
<span id="cb61-28"><a href="statistical-analysis-1.html#cb61-28" aria-hidden="true" tabindex="-1"></a>    <span class="co"># paired = paired    </span></span>
<span id="cb61-29"><a href="statistical-analysis-1.html#cb61-29" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb61-30"><a href="statistical-analysis-1.html#cb61-30" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> <span class="fu">t</span>(x)</span>
<span id="cb61-31"><a href="statistical-analysis-1.html#cb61-31" aria-hidden="true" tabindex="-1"></a>    colNames <span class="ot">&lt;-</span> <span class="fu">colnames</span>(x)</span>
<span id="cb61-32"><a href="statistical-analysis-1.html#cb61-32" aria-hidden="true" tabindex="-1"></a>    rowNames <span class="ot">&lt;-</span> <span class="fu">rownames</span>(x)</span>
<span id="cb61-33"><a href="statistical-analysis-1.html#cb61-33" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb61-34"><a href="statistical-analysis-1.html#cb61-34" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (paired) { </span>
<span id="cb61-35"><a href="statistical-analysis-1.html#cb61-35" aria-hidden="true" tabindex="-1"></a>      <span class="co"># compute the average of paired FC (unit is pair)</span></span>
<span id="cb61-36"><a href="statistical-analysis-1.html#cb61-36" aria-hidden="true" tabindex="-1"></a>      G1 <span class="ot">&lt;-</span> x[<span class="fu">which</span>(y<span class="sc">$</span>group <span class="sc">==</span> group_names[<span class="dv">1</span>]), ]</span>
<span id="cb61-37"><a href="statistical-analysis-1.html#cb61-37" aria-hidden="true" tabindex="-1"></a>      G2 <span class="ot">&lt;-</span> x[<span class="fu">which</span>(y<span class="sc">$</span>group <span class="sc">==</span> group_names[<span class="dv">2</span>]), ]</span>
<span id="cb61-38"><a href="statistical-analysis-1.html#cb61-38" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb61-39"><a href="statistical-analysis-1.html#cb61-39" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (cmpType <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb61-40"><a href="statistical-analysis-1.html#cb61-40" aria-hidden="true" tabindex="-1"></a>        fc.mat <span class="ot">&lt;-</span> G1 <span class="sc">-</span> G2</span>
<span id="cb61-41"><a href="statistical-analysis-1.html#cb61-41" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb61-42"><a href="statistical-analysis-1.html#cb61-42" aria-hidden="true" tabindex="-1"></a>        fc.mat <span class="ot">&lt;-</span> G2 <span class="sc">-</span> G1</span>
<span id="cb61-43"><a href="statistical-analysis-1.html#cb61-43" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb61-44"><a href="statistical-analysis-1.html#cb61-44" aria-hidden="true" tabindex="-1"></a>      fc.log <span class="ot">&lt;-</span> <span class="fu">apply</span>(fc.mat, <span class="dv">2</span>, mean)</span>
<span id="cb61-45"><a href="statistical-analysis-1.html#cb61-45" aria-hidden="true" tabindex="-1"></a>      fc.all <span class="ot">&lt;-</span> <span class="fu">signif</span>(<span class="dv">2</span><span class="sc">^</span>fc.log, <span class="dv">5</span>)</span>
<span id="cb61-46"><a href="statistical-analysis-1.html#cb61-46" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb61-47"><a href="statistical-analysis-1.html#cb61-47" aria-hidden="true" tabindex="-1"></a>      m1 <span class="ot">&lt;-</span> <span class="fu">colMeans</span>(x[<span class="fu">which</span>(y<span class="sc">$</span>group <span class="sc">==</span> group_names[<span class="dv">1</span>]), ])</span>
<span id="cb61-48"><a href="statistical-analysis-1.html#cb61-48" aria-hidden="true" tabindex="-1"></a>      m2 <span class="ot">&lt;-</span> <span class="fu">colMeans</span>(x[<span class="fu">which</span>(y<span class="sc">$</span>group <span class="sc">==</span> group_names[<span class="dv">2</span>]), ])</span>
<span id="cb61-49"><a href="statistical-analysis-1.html#cb61-49" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb61-50"><a href="statistical-analysis-1.html#cb61-50" aria-hidden="true" tabindex="-1"></a>      <span class="co"># create a named matrix of sig vars for display</span></span>
<span id="cb61-51"><a href="statistical-analysis-1.html#cb61-51" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (cmpType <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb61-52"><a href="statistical-analysis-1.html#cb61-52" aria-hidden="true" tabindex="-1"></a>          ratio <span class="ot">&lt;-</span> m1<span class="sc">/</span>m2</span>
<span id="cb61-53"><a href="statistical-analysis-1.html#cb61-53" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb61-54"><a href="statistical-analysis-1.html#cb61-54" aria-hidden="true" tabindex="-1"></a>          ratio <span class="ot">&lt;-</span> m2<span class="sc">/</span>m1</span>
<span id="cb61-55"><a href="statistical-analysis-1.html#cb61-55" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb61-56"><a href="statistical-analysis-1.html#cb61-56" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb61-57"><a href="statistical-analysis-1.html#cb61-57" aria-hidden="true" tabindex="-1"></a>      fc.all <span class="ot">&lt;-</span> <span class="fu">signif</span>(ratio, <span class="dv">5</span>)</span>
<span id="cb61-58"><a href="statistical-analysis-1.html#cb61-58" aria-hidden="true" tabindex="-1"></a>      ratio[ratio <span class="sc">&lt;</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb61-59"><a href="statistical-analysis-1.html#cb61-59" aria-hidden="true" tabindex="-1"></a>      fc.log <span class="ot">&lt;-</span> <span class="fu">signif</span>(<span class="fu">log2</span>(ratio), <span class="dv">5</span>)</span>
<span id="cb61-60"><a href="statistical-analysis-1.html#cb61-60" aria-hidden="true" tabindex="-1"></a>      fc.log[<span class="fu">is.infinite</span>(fc.log) <span class="sc">&amp;</span> fc.log <span class="sc">&lt;</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="sc">-</span><span class="dv">99</span></span>
<span id="cb61-61"><a href="statistical-analysis-1.html#cb61-61" aria-hidden="true" tabindex="-1"></a>      fc.log[<span class="fu">is.infinite</span>(fc.log) <span class="sc">&amp;</span> fc.log <span class="sc">&gt;</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="dv">99</span></span>
<span id="cb61-62"><a href="statistical-analysis-1.html#cb61-62" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb61-63"><a href="statistical-analysis-1.html#cb61-63" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb61-64"><a href="statistical-analysis-1.html#cb61-64" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(fc.all) <span class="ot">&lt;-</span> <span class="fu">names</span>(fc.log) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(x)</span>
<span id="cb61-65"><a href="statistical-analysis-1.html#cb61-65" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb61-66"><a href="statistical-analysis-1.html#cb61-66" aria-hidden="true" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">fc.all =</span> fc.all, <span class="at">fc.log =</span> fc.log)</span>
<span id="cb61-67"><a href="statistical-analysis-1.html#cb61-67" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(res)</span>
<span id="cb61-68"><a href="statistical-analysis-1.html#cb61-68" aria-hidden="true" tabindex="-1"></a>  }  </span>
<span id="cb61-69"><a href="statistical-analysis-1.html#cb61-69" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb61-70"><a href="statistical-analysis-1.html#cb61-70" aria-hidden="true" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">GetFC</span>(profile, metadata, cmp_type, paired)</span>
<span id="cb61-71"><a href="statistical-analysis-1.html#cb61-71" aria-hidden="true" tabindex="-1"></a>  fc.all <span class="ot">&lt;-</span> res<span class="sc">$</span>fc.all</span>
<span id="cb61-72"><a href="statistical-analysis-1.html#cb61-72" aria-hidden="true" tabindex="-1"></a>  fc.log <span class="ot">&lt;-</span> res<span class="sc">$</span>fc.log</span>
<span id="cb61-73"><a href="statistical-analysis-1.html#cb61-73" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb61-74"><a href="statistical-analysis-1.html#cb61-74" aria-hidden="true" tabindex="-1"></a>  inx.up <span class="ot">&lt;-</span> fc.all <span class="sc">&gt;</span> max_cutoff</span>
<span id="cb61-75"><a href="statistical-analysis-1.html#cb61-75" aria-hidden="true" tabindex="-1"></a>  inx.down <span class="ot">&lt;-</span> fc.all <span class="sc">&lt;</span> min_cutoff</span>
<span id="cb61-76"><a href="statistical-analysis-1.html#cb61-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(inx.up) <span class="ot">&lt;-</span> <span class="fu">names</span>(inx.down) <span class="ot">&lt;-</span> <span class="fu">names</span>(fc.all)</span>
<span id="cb61-77"><a href="statistical-analysis-1.html#cb61-77" aria-hidden="true" tabindex="-1"></a>  imp.inx <span class="ot">&lt;-</span> inx.up <span class="sc">|</span> inx.down</span>
<span id="cb61-78"><a href="statistical-analysis-1.html#cb61-78" aria-hidden="true" tabindex="-1"></a>  sig.mat <span class="ot">&lt;-</span> <span class="fu">cbind</span>(fc.all[imp.inx, <span class="at">drop =</span> F], fc.log[imp.inx, <span class="at">drop =</span> F])</span>
<span id="cb61-79"><a href="statistical-analysis-1.html#cb61-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(sig.mat) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Fold Change&quot;</span>, <span class="st">&quot;log2(FC)&quot;</span>)</span>
<span id="cb61-80"><a href="statistical-analysis-1.html#cb61-80" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb61-81"><a href="statistical-analysis-1.html#cb61-81" aria-hidden="true" tabindex="-1"></a>  <span class="co"># order by absolute log value (since symmetrical in pos and neg)</span></span>
<span id="cb61-82"><a href="statistical-analysis-1.html#cb61-82" aria-hidden="true" tabindex="-1"></a>  inx.ord <span class="ot">&lt;-</span> <span class="fu">order</span>(<span class="fu">abs</span>(sig.mat[, <span class="dv">2</span>]), <span class="at">decreasing =</span> T)</span>
<span id="cb61-83"><a href="statistical-analysis-1.html#cb61-83" aria-hidden="true" tabindex="-1"></a>  sig.mat <span class="ot">&lt;-</span> sig.mat[inx.ord, , drop <span class="ot">=</span> F]</span>
<span id="cb61-84"><a href="statistical-analysis-1.html#cb61-84" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb61-85"><a href="statistical-analysis-1.html#cb61-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(sig.mat)</span>
<span id="cb61-86"><a href="statistical-analysis-1.html#cb61-86" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb61-87"><a href="statistical-analysis-1.html#cb61-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-88"><a href="statistical-analysis-1.html#cb61-88" aria-hidden="true" tabindex="-1"></a>FC_res <span class="ot">&lt;-</span> <span class="fu">FoldChange</span>(</span>
<span id="cb61-89"><a href="statistical-analysis-1.html#cb61-89" aria-hidden="true" tabindex="-1"></a>           <span class="at">object =</span> se_normalize, </span>
<span id="cb61-90"><a href="statistical-analysis-1.html#cb61-90" aria-hidden="true" tabindex="-1"></a>           <span class="at">fc_cutoff =</span> <span class="dv">2</span>,</span>
<span id="cb61-91"><a href="statistical-analysis-1.html#cb61-91" aria-hidden="true" tabindex="-1"></a>           <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Moderate&quot;</span>),</span>
<span id="cb61-92"><a href="statistical-analysis-1.html#cb61-92" aria-hidden="true" tabindex="-1"></a>           <span class="at">cmp_type =</span> <span class="dv">0</span>, </span>
<span id="cb61-93"><a href="statistical-analysis-1.html#cb61-93" aria-hidden="true" tabindex="-1"></a>           <span class="at">paired =</span> <span class="cn">FALSE</span>)</span>
<span id="cb61-94"><a href="statistical-analysis-1.html#cb61-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-95"><a href="statistical-analysis-1.html#cb61-95" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(FC_res)</span></code></pre></div>
<pre><code>##         Fold Change log2(FC)
## M_52603    -0.43650      -99
## M_39270    -0.28201      -99
## M_35186    -1.17830      -99
## M_34214    -1.03940      -99
## M_49617    -4.72130      -99
## M_53189    -0.97524      -99</code></pre>
</div>
<div id="t-test" class="section level3 hasAnchor" number="2.4.4">
<h3><span class="header-section-number">2.4.4</span> T Test<a href="statistical-analysis-1.html#t-test" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="statistical-analysis-1.html#cb63-1" aria-hidden="true" tabindex="-1"></a>group_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Severe&quot;</span>)</span>
<span id="cb63-2"><a href="statistical-analysis-1.html#cb63-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-3"><a href="statistical-analysis-1.html#cb63-3" aria-hidden="true" tabindex="-1"></a>se_processed_subset <span class="ot">&lt;-</span> se_processed[, se_processed<span class="sc">$</span>group <span class="sc">%in%</span> group_names]</span>
<span id="cb63-4"><a href="statistical-analysis-1.html#cb63-4" aria-hidden="true" tabindex="-1"></a>se_processed_subset<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">as.character</span>(se_processed_subset<span class="sc">$</span>group))</span>
<span id="cb63-5"><a href="statistical-analysis-1.html#cb63-5" aria-hidden="true" tabindex="-1"></a>ttest_res <span class="ot">&lt;-</span> <span class="fu">PomaUnivariate</span>(se_processed_subset, <span class="at">method =</span> <span class="st">&quot;ttest&quot;</span>)</span>
<span id="cb63-6"><a href="statistical-analysis-1.html#cb63-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ttest_res)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 9
##   feature     FC diff_means pvalue pvalueAdj mean_Mild mean_Severe sd_Mild sd_Severe
##   &lt;chr&gt;    &lt;dbl&gt;      &lt;dbl&gt;  &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;       &lt;dbl&gt;   &lt;dbl&gt;     &lt;dbl&gt;
## 1 M_52603  3.93       0.069  0.604     0.745   0.0236      0.0926    0.238     0.360
## 2 M_19130  2.94       0.015  0.854     0.891   0.00797     0.0234    0.269     0.130
## 3 M_39270 -0.297      0.023  0.880     0.898  -0.0177      0.00526   0.460     0.272
## 4 M_35186 -0.656      0.086  0.510     0.745  -0.0518      0.0340    0.328     0.294
## 5 M_34214 -0.232      0.144  0.329     0.636  -0.117       0.0271    0.466     0.229
## 6 M_49617  0.064     -0.128  0.218     0.636   0.137       0.00881   0.249     0.240</code></pre>
</div>
<div id="wilcoxon-test" class="section level3 hasAnchor" number="2.4.5">
<h3><span class="header-section-number">2.4.5</span> Wilcoxon Test<a href="statistical-analysis-1.html#wilcoxon-test" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="statistical-analysis-1.html#cb65-1" aria-hidden="true" tabindex="-1"></a>wilcox_res <span class="ot">&lt;-</span> <span class="fu">PomaUnivariate</span>(se_processed_subset, <span class="at">method =</span> <span class="st">&quot;mann&quot;</span>)</span>
<span id="cb65-2"><a href="statistical-analysis-1.html#cb65-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(wilcox_res)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 9
##   feature     FC diff_means pvalue pvalueAdj mean_Mild mean_Severe sd_Mild sd_Severe
##   &lt;chr&gt;    &lt;dbl&gt;      &lt;dbl&gt;  &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;       &lt;dbl&gt;   &lt;dbl&gt;     &lt;dbl&gt;
## 1 M_52603  3.93       0.069  0.709     0.830   0.0236      0.0926    0.238     0.360
## 2 M_19130  2.94       0.015  0.931     0.951   0.00797     0.0234    0.269     0.130
## 3 M_39270 -0.297      0.023  0.752     0.840  -0.0177      0.00526   0.460     0.272
## 4 M_35186 -0.656      0.086  0.546     0.748  -0.0518      0.0340    0.328     0.294
## 5 M_34214 -0.232      0.144  0.472     0.730  -0.117       0.0271    0.466     0.229
## 6 M_49617  0.064     -0.128  0.138     0.600   0.137       0.00881   0.249     0.240</code></pre>
</div>
<div id="limma-test" class="section level3 hasAnchor" number="2.4.6">
<h3><span class="header-section-number">2.4.6</span> Limma Test<a href="statistical-analysis-1.html#limma-test" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="statistical-analysis-1.html#cb67-1" aria-hidden="true" tabindex="-1"></a>Limma_res <span class="ot">&lt;-</span> <span class="fu">PomaLimma</span>(se_processed_subset, <span class="at">contrast =</span> <span class="fu">paste</span>(group_names, <span class="at">collapse =</span> <span class="st">&quot;-&quot;</span>), <span class="at">adjust =</span> <span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb67-2"><a href="statistical-analysis-1.html#cb67-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Limma_res)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 7
##   feature  logFC  AveExpr     t P.Value adj.P.Val     B
##   &lt;chr&gt;    &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt;
## 1 M_52464 -0.419 -0.0672  -2.94 0.00584     0.278 -2.26
## 2 M_52447 -0.369 -0.0657  -2.67 0.0116      0.278 -2.77
## 3 M_55041 -0.320 -0.0764  -2.23 0.0326      0.505 -3.52
## 4 M_35631 -0.269 -0.0382  -2.05 0.0477      0.505 -3.79
## 5 M_57663 -0.261  0.0104  -2.01 0.0526      0.505 -3.86
## 6 M_54885 -0.265  0.00557 -1.77 0.0853      0.655 -4.20</code></pre>
</div>
<div id="volcano-plot" class="section level3 hasAnchor" number="2.4.7">
<h3><span class="header-section-number">2.4.7</span> Volcano plot<a href="statistical-analysis-1.html#volcano-plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="statistical-analysis-1.html#cb69-1" aria-hidden="true" tabindex="-1"></a>se_impute_subset <span class="ot">&lt;-</span> se_impute[, se_impute<span class="sc">$</span>group <span class="sc">%in%</span> group_names]</span>
<span id="cb69-2"><a href="statistical-analysis-1.html#cb69-2" aria-hidden="true" tabindex="-1"></a>se_impute_subset<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">as.character</span>(se_impute_subset<span class="sc">$</span>group))</span>
<span id="cb69-3"><a href="statistical-analysis-1.html#cb69-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-4"><a href="statistical-analysis-1.html#cb69-4" aria-hidden="true" tabindex="-1"></a><span class="fu">PomaVolcano</span>(se_impute_subset, </span>
<span id="cb69-5"><a href="statistical-analysis-1.html#cb69-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">pval =</span> <span class="st">&quot;raw&quot;</span>,</span>
<span id="cb69-6"><a href="statistical-analysis-1.html#cb69-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">pval_cutoff =</span> <span class="fl">0.1</span>,</span>
<span id="cb69-7"><a href="statistical-analysis-1.html#cb69-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">log2FC =</span> <span class="fl">0.2</span>,</span>
<span id="cb69-8"><a href="statistical-analysis-1.html#cb69-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">xlim =</span> <span class="dv">1</span>,</span>
<span id="cb69-9"><a href="statistical-analysis-1.html#cb69-9" aria-hidden="true" tabindex="-1"></a>            <span class="at">labels =</span> <span class="cn">TRUE</span>,</span>
<span id="cb69-10"><a href="statistical-analysis-1.html#cb69-10" aria-hidden="true" tabindex="-1"></a>            <span class="at">plot_title =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-44-1.png" width="576" /></p>
</div>
<div id="correlation-heatmaps" class="section level3 hasAnchor" number="2.4.8">
<h3><span class="header-section-number">2.4.8</span> Correlation Heatmaps<a href="statistical-analysis-1.html#correlation-heatmaps" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="statistical-analysis-1.html#cb70-1" aria-hidden="true" tabindex="-1"></a>poma_cor <span class="ot">&lt;-</span> <span class="fu">PomaCorr</span>(se_processed_subset, <span class="at">label_size =</span> <span class="dv">8</span>, <span class="at">coeff =</span> <span class="fl">0.6</span>)</span>
<span id="cb70-2"><a href="statistical-analysis-1.html#cb70-2" aria-hidden="true" tabindex="-1"></a>poma_cor<span class="sc">$</span>correlations</span></code></pre></div>
<pre><code>## # A tibble: 1,128 × 5
##    feature1 feature2     R   pvalue          FDR
##    &lt;chr&gt;    &lt;chr&gt;    &lt;dbl&gt;    &lt;dbl&gt;        &lt;dbl&gt;
##  1 M_52464  M_52447  0.933 3.15e-11 0.0000000356
##  2 M_55041  M_52464  0.911 6.07e-10 0.000000252 
##  3 M_33955  M_33961  0.910 6.71e-10 0.000000252 
##  4 M_55041  M_52447  0.905 1.20e- 9 0.000000337 
##  5 M_33961  M_42398  0.901 2.04e- 9 0.000000460 
##  6 M_33955  M_42398  0.898 2.72e- 9 0.000000511 
##  7 M_52461  M_52438  0.887 7.73e- 9 0.00000116  
##  8 M_33955  M_35253  0.886 8.19e- 9 0.00000116  
##  9 M_33230  M_47118  0.884 9.95e- 9 0.00000125  
## 10 M_33961  M_35253  0.881 1.28e- 8 0.00000144  
## # … with 1,118 more rows</code></pre>
<ul>
<li>correlation plot</li>
</ul>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="statistical-analysis-1.html#cb72-1" aria-hidden="true" tabindex="-1"></a>poma_cor<span class="sc">$</span>corrplot</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-46-1.png" width="960" /></p>
<ul>
<li>Network</li>
</ul>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="statistical-analysis-1.html#cb73-1" aria-hidden="true" tabindex="-1"></a>poma_cor<span class="sc">$</span>graph</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-47-1.png" width="960" /></p>
</div>
<div id="glasso-this-function-will-compute-a-gaussian-graphical-model-using-the-glmnet-package" class="section level3 hasAnchor" number="2.4.9">
<h3><span class="header-section-number">2.4.9</span> glasso: this function will compute a Gaussian Graphical Model using the glmnet package<a href="statistical-analysis-1.html#glasso-this-function-will-compute-a-gaussian-graphical-model-using-the-glmnet-package" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="statistical-analysis-1.html#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">PomaCorr</span>(se_processed_subset, <span class="at">corr_type =</span> <span class="st">&quot;glasso&quot;</span>, <span class="at">coeff =</span> <span class="fl">0.6</span>)<span class="sc">$</span>graph</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-48-1.png" width="960" /></p>
<hr />
</div>
</div>
<div id="multivariate-analysis" class="section level2 hasAnchor" number="2.5">
<h2><span class="header-section-number">2.5</span> Multivariate analysis<a href="statistical-analysis-1.html#multivariate-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Comparing to <strong>univariate analysis</strong>, <strong>multivariate analysis</strong> is defined as a process of involving multiple dependent variables resulting in one outcome for feature selection. Here, we use Regularized Generalized Linear Models and Random forest model to identify the biomarkers associated with Outcomes.</p>
<ul>
<li><p>Lasso, Ridge and Elasticnet Regularized Generalized Linear Models for Binary Outcomes</p></li>
<li><p>Random forest model to select the important features</p></li>
</ul>
<p><strong>Note: Please also remember to <a href="statistical-analysis-1.html#data-preprocessing">preprocess</a> your data before running this sub-chapter.</strong></p>
<div id="loading-packages-3" class="section level3 hasAnchor" number="2.5.1">
<h3><span class="header-section-number">2.5.1</span> Loading packages<a href="statistical-analysis-1.html#loading-packages-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="statistical-analysis-1.html#cb75-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">warning =</span> F)</span>
<span id="cb75-2"><a href="statistical-analysis-1.html#cb75-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-3"><a href="statistical-analysis-1.html#cb75-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb75-4"><a href="statistical-analysis-1.html#cb75-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb75-5"><a href="statistical-analysis-1.html#cb75-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(POMA)</span>
<span id="cb75-6"><a href="statistical-analysis-1.html#cb75-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb75-7"><a href="statistical-analysis-1.html#cb75-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggraph)</span>
<span id="cb75-8"><a href="statistical-analysis-1.html#cb75-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb75-9"><a href="statistical-analysis-1.html#cb75-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb75-10"><a href="statistical-analysis-1.html#cb75-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmnet)</span>
<span id="cb75-11"><a href="statistical-analysis-1.html#cb75-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-12"><a href="statistical-analysis-1.html#cb75-12" aria-hidden="true" tabindex="-1"></a><span class="co"># rm(list = ls())</span></span>
<span id="cb75-13"><a href="statistical-analysis-1.html#cb75-13" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb75-14"><a href="statistical-analysis-1.html#cb75-14" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">future.globals.maxSize =</span> <span class="dv">1000</span> <span class="sc">*</span> <span class="dv">1024</span><span class="sc">^</span><span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="importing-data-3" class="section level3 hasAnchor" number="2.5.2">
<h3><span class="header-section-number">2.5.2</span> Importing data<a href="statistical-analysis-1.html#importing-data-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The input data sets are from the previous chapter.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="statistical-analysis-1.html#cb76-1" aria-hidden="true" tabindex="-1"></a>se_processed <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;./dataset/POMA/se_processed.RDS&quot;</span>)</span></code></pre></div>
</div>
<div id="data-curation" class="section level3 hasAnchor" number="2.5.3">
<h3><span class="header-section-number">2.5.3</span> Data curation<a href="statistical-analysis-1.html#data-curation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="statistical-analysis-1.html#cb77-1" aria-hidden="true" tabindex="-1"></a>group_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Severe&quot;</span>)</span>
<span id="cb77-2"><a href="statistical-analysis-1.html#cb77-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb77-3"><a href="statistical-analysis-1.html#cb77-3" aria-hidden="true" tabindex="-1"></a>se_processed_subset <span class="ot">&lt;-</span> se_processed[, se_processed<span class="sc">$</span>group <span class="sc">%in%</span> group_names]</span>
<span id="cb77-4"><a href="statistical-analysis-1.html#cb77-4" aria-hidden="true" tabindex="-1"></a>se_processed_subset<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">as.character</span>(se_processed_subset<span class="sc">$</span>group))</span></code></pre></div>
</div>
<div id="regularized-generalized-linear-models" class="section level3 hasAnchor" number="2.5.4">
<h3><span class="header-section-number">2.5.4</span> Regularized Generalized Linear Models<a href="statistical-analysis-1.html#regularized-generalized-linear-models" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="lasso-alpha-1" class="section level4 hasAnchor" number="2.5.4.1">
<h4><span class="header-section-number">2.5.4.1</span> Lasso: alpha = 1<a href="statistical-analysis-1.html#lasso-alpha-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="statistical-analysis-1.html#cb78-1" aria-hidden="true" tabindex="-1"></a>lasso_res <span class="ot">&lt;-</span> <span class="fu">PomaLasso</span>(se_processed_subset, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">labels =</span> <span class="cn">TRUE</span>)</span>
<span id="cb78-2"><a href="statistical-analysis-1.html#cb78-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-3"><a href="statistical-analysis-1.html#cb78-3" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(lasso_res<span class="sc">$</span>cvLassoPlot,</span>
<span id="cb78-4"><a href="statistical-analysis-1.html#cb78-4" aria-hidden="true" tabindex="-1"></a>                   lasso_res<span class="sc">$</span>coefficientPlot,</span>
<span id="cb78-5"><a href="statistical-analysis-1.html#cb78-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">align =</span> <span class="st">&quot;hv&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-52-1.png" width="1152" /></p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="statistical-analysis-1.html#cb79-1" aria-hidden="true" tabindex="-1"></a>lasso_res<span class="sc">$</span>coefficients</span></code></pre></div>
<pre><code>## # A tibble: 2 × 2
##   feature     coefficient
##   &lt;chr&gt;             &lt;dbl&gt;
## 1 (Intercept)      -0.292
## 2 M_52464           0.762</code></pre>
</div>
<div id="ridge-alpha-0" class="section level4 hasAnchor" number="2.5.4.2">
<h4><span class="header-section-number">2.5.4.2</span> Ridge: alpha = 0<a href="statistical-analysis-1.html#ridge-alpha-0" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="statistical-analysis-1.html#cb81-1" aria-hidden="true" tabindex="-1"></a>ridge_res <span class="ot">&lt;-</span> <span class="fu">PomaLasso</span>(se_processed_subset, <span class="at">alpha =</span> <span class="dv">0</span>, <span class="at">labels =</span> <span class="cn">TRUE</span>)</span>
<span id="cb81-2"><a href="statistical-analysis-1.html#cb81-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-3"><a href="statistical-analysis-1.html#cb81-3" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(ridge_res<span class="sc">$</span>cvLassoPlot,</span>
<span id="cb81-4"><a href="statistical-analysis-1.html#cb81-4" aria-hidden="true" tabindex="-1"></a>                   ridge_res<span class="sc">$</span>coefficientPlot,</span>
<span id="cb81-5"><a href="statistical-analysis-1.html#cb81-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">align =</span> <span class="st">&quot;hv&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-53-1.png" width="1152" /></p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="statistical-analysis-1.html#cb82-1" aria-hidden="true" tabindex="-1"></a>ridge_res<span class="sc">$</span>coefficients</span></code></pre></div>
<pre><code>## # A tibble: 49 × 2
##    feature     coefficient
##    &lt;chr&gt;             &lt;dbl&gt;
##  1 (Intercept)    -0.269  
##  2 M_52603         0.0416 
##  3 M_19130        -0.0311 
##  4 M_39270        -0.0294 
##  5 M_35186         0.0101 
##  6 M_34214         0.0329 
##  7 M_49617        -0.214  
##  8 M_52710        -0.186  
##  9 M_53189         0.00193
## 10 M_34419         0.0266 
## # … with 39 more rows</code></pre>
</div>
<div id="elasticnet-0-alpha-1" class="section level4 hasAnchor" number="2.5.4.3">
<h4><span class="header-section-number">2.5.4.3</span> Elasticnet: 0 &lt; alpha &lt; 1<a href="statistical-analysis-1.html#elasticnet-0-alpha-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="statistical-analysis-1.html#cb84-1" aria-hidden="true" tabindex="-1"></a>elastic_res <span class="ot">&lt;-</span> <span class="fu">PomaLasso</span>(se_processed_subset, <span class="at">alpha =</span> <span class="fl">0.4</span>, <span class="at">labels =</span> <span class="cn">TRUE</span>)</span>
<span id="cb84-2"><a href="statistical-analysis-1.html#cb84-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-3"><a href="statistical-analysis-1.html#cb84-3" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(elastic_res<span class="sc">$</span>cvLassoPlot,</span>
<span id="cb84-4"><a href="statistical-analysis-1.html#cb84-4" aria-hidden="true" tabindex="-1"></a>                   elastic_res<span class="sc">$</span>coefficientPlot,</span>
<span id="cb84-5"><a href="statistical-analysis-1.html#cb84-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">align =</span> <span class="st">&quot;hv&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-54-1.png" width="1152" /></p>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="statistical-analysis-1.html#cb85-1" aria-hidden="true" tabindex="-1"></a>elastic_res<span class="sc">$</span>coefficients</span></code></pre></div>
<pre><code>## # A tibble: 3 × 2
##   feature     coefficient
##   &lt;chr&gt;             &lt;dbl&gt;
## 1 (Intercept)      -0.305
## 2 M_52464           0.319
## 3 M_52447           0.199</code></pre>
</div>
</div>
<div id="classification" class="section level3 hasAnchor" number="2.5.5">
<h3><span class="header-section-number">2.5.5</span> Classification<a href="statistical-analysis-1.html#classification" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="random-forest" class="section level4 hasAnchor" number="2.5.5.1">
<h4><span class="header-section-number">2.5.5.1</span> Random Forest<a href="statistical-analysis-1.html#random-forest" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="statistical-analysis-1.html#cb87-1" aria-hidden="true" tabindex="-1"></a>poma_rf <span class="ot">&lt;-</span> <span class="fu">PomaRandForest</span>(se_processed_subset, <span class="at">ntest =</span> <span class="dv">10</span>, <span class="at">nvar =</span> <span class="dv">10</span>)</span>
<span id="cb87-2"><a href="statistical-analysis-1.html#cb87-2" aria-hidden="true" tabindex="-1"></a>poma_rf<span class="sc">$</span>error_tree</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-55-1.png" width="576" /></p>
<ul>
<li>table</li>
</ul>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="statistical-analysis-1.html#cb88-1" aria-hidden="true" tabindex="-1"></a>poma_rf<span class="sc">$</span>confusionMatrix<span class="sc">$</span>table</span></code></pre></div>
<pre><code>##           Reference
## Prediction 1 2
##          1 1 0
##          2 0 1</code></pre>
<ul>
<li>Important features</li>
</ul>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="statistical-analysis-1.html#cb90-1" aria-hidden="true" tabindex="-1"></a>poma_rf<span class="sc">$</span>MeanDecreaseGini_plot</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-57-1.png" width="576" /></p>
</div>
<div id="support-vector-machine-svm" class="section level4 hasAnchor" number="2.5.5.2">
<h4><span class="header-section-number">2.5.5.2</span> Support Vector Machine (SVM)<a href="statistical-analysis-1.html#support-vector-machine-svm" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Still in development…</p>
<hr />
</div>
</div>
</div>
<div id="network-analysis" class="section level2 hasAnchor" number="2.6">
<h2><span class="header-section-number">2.6</span> Network Analysis<a href="statistical-analysis-1.html#network-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="introduction-1" class="section level3 hasAnchor" number="2.6.1">
<h3><span class="header-section-number">2.6.1</span> Introduction<a href="statistical-analysis-1.html#introduction-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<blockquote>
<p>Estimating microbial association networks from high-throughput sequencing data is a common exploratory data analysis approach aiming at understanding the complex interplay of microbial communities in their natural habitat. Statistical network estimation workflows comprise several analysis steps, including methods for zero handling, data normalization and computing microbial associations. Since microbial interactions are likely to change between conditions, e.g. between healthy individuals and patients, identifying network differences between groups is often an integral secondary analysis step.</p>
</blockquote>
<p>NetCoMi (Network Construction and Comparison for Microbiome Data) <span class="citation">(<a href="#ref-peschel2021netcomi" role="doc-biblioref">Peschel et al. 2021</a>)</span> provides functionality for constructing, analyzing, and comparing networks suitable for the application on microbial compositional data.</p>
<p>The following information is from <a href="https://github.com/stefpeschel/NetCoMi">NetCoMi github</a>.</p>
<p><strong>Association measures:</strong></p>
<ul>
<li><p>Pearson coefficient
(<a href="https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/cor"><code>cor()</code></a>
from <code>stats</code> package)</p></li>
<li><p>Spearman coefficient
(<a href="https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/cor"><code>cor()</code></a>
from <code>stats</code> package)</p></li>
<li><p>Biweight Midcorrelation
<a href="https://rdrr.io/cran/WGCNA/man/bicor.html"><code>bicor()</code></a> from <code>WGCNA</code>
package</p></li>
</ul>
<p><strong>Methods for zero replacement:</strong></p>
<ul>
<li><p>Adding a predefined pseudo count</p></li>
<li><p>Multiplicative replacement
(<a href="https://rdrr.io/cran/zCompositions/man/multRepl.html"><code>multRepl</code></a>
from <code>zCompositions</code> package)</p></li>
<li><p>Modified EM alr-algorithm
(<a href="https://rdrr.io/cran/zCompositions/man/lrEM.html"><code>lrEM</code></a> from
<code>zCompositions</code> package)</p></li>
<li><p>Bayesian-multiplicative replacement
(<a href="https://rdrr.io/cran/zCompositions/man/cmultRepl.html"><code>cmultRepl</code></a>
from <code>zCompositions</code> package)</p></li>
</ul>
<p><strong>Normalization methods:</strong></p>
<ul>
<li><p>Total Sum Scaling (TSS) (own implementation)</p></li>
<li><p>Cumulative Sum Scaling (CSS) (<a href=""><code>cumNormMat</code></a> from <code>metagenomeSeq</code>
package)</p></li>
<li><p>Common Sum Scaling (COM) (own implementation)</p></li>
<li><p>Rarefying (<a href=""><code>rrarefy</code></a> from <code>vegan</code> package)</p></li>
<li><p>Variance Stabilizing Transformation (VST)<br />
(<a href=""><code>varianceStabilizingTransformation</code></a> from <code>DESeq2</code> package)</p></li>
<li><p>Centered log-ratio (clr) transformation
(<a href="https://rdrr.io/github/zdk123/SpiecEasi/man/clr.html"><code>clr()</code></a>
from <code>SpiecEasi</code> package))</p></li>
</ul>
<p>TSS, CSS, COM, VST, and the clr transformation are described in [Badri et al., 2020].</p>
</div>
<div id="loading-packages-4" class="section level3 hasAnchor" number="2.6.2">
<h3><span class="header-section-number">2.6.2</span> Loading packages<a href="statistical-analysis-1.html#loading-packages-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="statistical-analysis-1.html#cb91-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">warning =</span> F)</span>
<span id="cb91-2"><a href="statistical-analysis-1.html#cb91-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-3"><a href="statistical-analysis-1.html#cb91-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb91-4"><a href="statistical-analysis-1.html#cb91-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb91-5"><a href="statistical-analysis-1.html#cb91-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(POMA)</span>
<span id="cb91-6"><a href="statistical-analysis-1.html#cb91-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb91-7"><a href="statistical-analysis-1.html#cb91-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggraph)</span>
<span id="cb91-8"><a href="statistical-analysis-1.html#cb91-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb91-9"><a href="statistical-analysis-1.html#cb91-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb91-10"><a href="statistical-analysis-1.html#cb91-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(NetCoMi)</span>
<span id="cb91-11"><a href="statistical-analysis-1.html#cb91-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SPRING)</span>
<span id="cb91-12"><a href="statistical-analysis-1.html#cb91-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SpiecEasi)</span>
<span id="cb91-13"><a href="statistical-analysis-1.html#cb91-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-14"><a href="statistical-analysis-1.html#cb91-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-15"><a href="statistical-analysis-1.html#cb91-15" aria-hidden="true" tabindex="-1"></a><span class="co"># rm(list = ls())</span></span>
<span id="cb91-16"><a href="statistical-analysis-1.html#cb91-16" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb91-17"><a href="statistical-analysis-1.html#cb91-17" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">future.globals.maxSize =</span> <span class="dv">1000</span> <span class="sc">*</span> <span class="dv">1024</span><span class="sc">^</span><span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="importing-data-4" class="section level3 hasAnchor" number="2.6.3">
<h3><span class="header-section-number">2.6.3</span> Importing data<a href="statistical-analysis-1.html#importing-data-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The input data sets are from the previous chapter.</p>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="statistical-analysis-1.html#cb92-1" aria-hidden="true" tabindex="-1"></a>se_filter <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;./dataset/POMA/se_filter.RDS&quot;</span>)</span></code></pre></div>
</div>
<div id="data-curation-1" class="section level3 hasAnchor" number="2.6.4">
<h3><span class="header-section-number">2.6.4</span> Data curation<a href="statistical-analysis-1.html#data-curation-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="statistical-analysis-1.html#cb93-1" aria-hidden="true" tabindex="-1"></a>features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(se_filter) <span class="sc">%&gt;%</span> </span>
<span id="cb93-2"><a href="statistical-analysis-1.html#cb93-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>()</span>
<span id="cb93-3"><a href="statistical-analysis-1.html#cb93-3" aria-hidden="true" tabindex="-1"></a>features_tab[<span class="fu">is.na</span>(features_tab)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb93-4"><a href="statistical-analysis-1.html#cb93-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-5"><a href="statistical-analysis-1.html#cb93-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(features_tab)</span></code></pre></div>
<pre><code>##           M_52603  M_19130   M_39270 M_35186 M_34214  M_49617  M_52710  M_53189   M_34419  M_36600  M_54885  M_32350 M_55041   M_48258 M_35628  M_33230   M_52462  M_52464   M_52454   M_52610   M_42446
## P101001  94392176 25632184  959249.2 8141286 8284910 381540.1 47739292 392911.0 306346432  9887348 298886.1 627613.7 3655240 181136336 6548460 19098412 486910976  6794782 274026048 206279104 712253440
## P101004 115155104 25106562  785895.0 4303854 7923489 588488.2 43678784 300383.9 263333424  9703074 356274.3 339586.8 1637351 115752200 3455001 10641834 380887616  4090405 282066016 253920496 807180352
## P101007  79582632 31371314 1903592.4 7129884 9211485 464548.8 53975068 539361.5 191261648  6431966 393623.0 662555.6 7284474 102036920 3640571 14977371 515795264 13183206 309709728 316142400 715500800
## P101009 118408760 27787270  863701.9 5394354 8559147 497342.0 47619564 487186.2 232171184 11914472 599783.6 492430.9 3065707 118663320 5087503 13575508 390977568  7596680 325723168 232534688 742576768
## P101010  92508664 26685844 1177860.8 6270727 7623603 457409.1 61906028 236719.5 183104336  9366287 280123.8 373484.2 2493687 154183360 5244878 19464998 478479936  5395172 382690720 262177072 642967424
## P101011  94076424 27780988  638271.3 4292087 6074633 238781.8 40534280 279118.7 164623888  6003700 349482.3 608671.7 2653816  76137216 2238570  8797510 420938976  4854076 261022832 264648720 728199680
##           M_52461  M_52616   M_33955 M_35631   M_42450  M_52447  M_52449   M_52452   M_52438   M_33961 M_42398  M_47403  M_57547   M_19266   M_63739  M_57663 M_42489  M_37253  M_45095   M_53229   M_55072
## P101001 486655872 12365879 448634784 4035864 358896736 30252208 43173148 510461472 181406848 191147264 6137137 209255.5 169297.7 176944.80  761146.2 719237.9 2865827 411578.0 751527.3  768160.4  488544.9
## P101004 439253152 10655923 413690720 3446139 282910752 19424238 30030992 573170816 122556968 166379712 5143481 220424.3 426227.5      0.00  549097.4 364556.1 1546570 425030.0 378033.7  953194.9  653107.0
## P101007 448578400 14010174 368849632 3764448 404918560 57376512 50766284 588643712 181572112 189915296 7412535 369675.0 561078.5 385203.22  792883.2 500680.8 1403190 742298.0 589506.3 1619436.8  522721.5
## P101009 473622240 14814230 365712768 3444159 298822048 26429340 43420376 631965888 208484000 186599152 5421984 172995.7 457043.3      0.00  727360.0 523549.1 1653830 466825.0 503910.2  877530.2  873616.8
## P101010 647036736 12873290 376101984 3276579 373251712 21732290 43622724 590247424 306379392 217632736 5767624 168503.4 602024.4      0.00  913964.4 455201.1 1007203 232876.0 489013.9  993923.4  864624.8
## P101011 378921056 13152187 348108832 2364665 335196864 21960100 39447380 560182336 136970272 170865408 3588338 375137.6 304329.9  90888.59 1609550.0 449288.0 1818438 651777.4 785751.6 1092817.9 1360312.8
##          M_47118  M_35253   M_46115 M_62952  M_63681 M_63361
## P101001 281162.7 27446600 1578011.4 5104806 434480.3 2540280
## P101004 362380.0 28088842  440408.7 3084011 373229.5 2165454
## P101007 482046.2 21613430  794592.2 4728273 400386.6 2231037
## P101009 619943.9 28872470  671497.6 4140767 301840.8       0
## P101010 677774.7 25375946 1133763.9 2434359 548330.4 1125607
## P101011 208040.6 17388928 1635119.5 5722723 344799.8       0</code></pre>
</div>
<div id="associations-among-features" class="section level3 hasAnchor" number="2.6.5">
<h3><span class="header-section-number">2.6.5</span> Associations Among Features<a href="statistical-analysis-1.html#associations-among-features" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="single-network-with-pearson-correlation-as-association-measure" class="section level4 hasAnchor" number="2.6.5.1">
<h4><span class="header-section-number">2.6.5.1</span> Single network with Pearson correlation as association measure<a href="statistical-analysis-1.html#single-network-with-pearson-correlation-as-association-measure" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Since Pearson correlations may lead to compositional effects when applied to sequencing data, we use the clr transformation as normalization method. Zero treatment is necessary in this case.</p>
<ul>
<li><p>InputData: numeric matrix. Can be a count matrix (rows are samples, columns are Features).</p></li>
<li><p>Method to compute the associations between features (argument measure).</p></li>
<li><p>Normalization method:</p>
<ul>
<li><p>normMethod: clr</p></li>
<li><p>zeroMethod: multRepl</p></li>
<li><p>sparsMethod: threshold</p></li>
</ul></li>
</ul>
<p>A threshold of 0.3 is used as sparsification method, so that only OTUs with an absolute correlation greater than or equal to 0.3 are connected.</p>
<div id="building-network-model" class="section level5 hasAnchor" number="2.6.5.1.1">
<h5><span class="header-section-number">2.6.5.1.1</span> Building network model<a href="statistical-analysis-1.html#building-network-model" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="statistical-analysis-1.html#cb95-1" aria-hidden="true" tabindex="-1"></a>net_single <span class="ot">&lt;-</span> <span class="fu">netConstruct</span>(features_tab,  </span>
<span id="cb95-2"><a href="statistical-analysis-1.html#cb95-2" aria-hidden="true" tabindex="-1"></a>                           <span class="at">measure =</span> <span class="st">&quot;pearson&quot;</span>,</span>
<span id="cb95-3"><a href="statistical-analysis-1.html#cb95-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">normMethod =</span> <span class="st">&quot;clr&quot;</span>, </span>
<span id="cb95-4"><a href="statistical-analysis-1.html#cb95-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">zeroMethod =</span> <span class="st">&quot;multRepl&quot;</span>,</span>
<span id="cb95-5"><a href="statistical-analysis-1.html#cb95-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">sparsMethod =</span> <span class="st">&quot;threshold&quot;</span>, </span>
<span id="cb95-6"><a href="statistical-analysis-1.html#cb95-6" aria-hidden="true" tabindex="-1"></a>                           <span class="at">thresh =</span> <span class="fl">0.3</span>,</span>
<span id="cb95-7"><a href="statistical-analysis-1.html#cb95-7" aria-hidden="true" tabindex="-1"></a>                           <span class="at">verbose =</span> <span class="dv">3</span>,</span>
<span id="cb95-8"><a href="statistical-analysis-1.html#cb95-8" aria-hidden="true" tabindex="-1"></a>                           <span class="at">seed =</span> <span class="dv">123</span>)</span></code></pre></div>
</div>
<div id="visualizing-the-network" class="section level5 hasAnchor" number="2.6.5.1.2">
<h5><span class="header-section-number">2.6.5.1.2</span> Visualizing the network<a href="statistical-analysis-1.html#visualizing-the-network" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li>primary plot</li>
</ul>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="statistical-analysis-1.html#cb96-1" aria-hidden="true" tabindex="-1"></a>props_single <span class="ot">&lt;-</span> <span class="fu">netAnalyze</span>(net_single, <span class="at">clustMethod =</span> <span class="st">&quot;cluster_fast_greedy&quot;</span>)</span>
<span id="cb96-2"><a href="statistical-analysis-1.html#cb96-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-3"><a href="statistical-analysis-1.html#cb96-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(props_single, </span>
<span id="cb96-4"><a href="statistical-analysis-1.html#cb96-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeColor =</span> <span class="st">&quot;cluster&quot;</span>, </span>
<span id="cb96-5"><a href="statistical-analysis-1.html#cb96-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeSize =</span> <span class="st">&quot;eigenvector&quot;</span>,</span>
<span id="cb96-6"><a href="statistical-analysis-1.html#cb96-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">title1 =</span> <span class="st">&quot;Network on metabolomics with Pearson correlations&quot;</span>, </span>
<span id="cb96-7"><a href="statistical-analysis-1.html#cb96-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">showTitle =</span> <span class="cn">TRUE</span>,</span>
<span id="cb96-8"><a href="statistical-analysis-1.html#cb96-8" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexTitle =</span> <span class="fl">1.5</span>)</span>
<span id="cb96-9"><a href="statistical-analysis-1.html#cb96-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-10"><a href="statistical-analysis-1.html#cb96-10" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="fl">0.7</span>, <span class="fl">1.1</span>, <span class="at">cex =</span> <span class="dv">1</span>, <span class="at">title =</span> <span class="st">&quot;estimated correlation:&quot;</span>, </span>
<span id="cb96-11"><a href="statistical-analysis-1.html#cb96-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;+&quot;</span>,<span class="st">&quot;-&quot;</span>), <span class="at">lty =</span> <span class="dv">1</span>, <span class="at">lwd =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;#009900&quot;</span>,<span class="st">&quot;red&quot;</span>), </span>
<span id="cb96-12"><a href="statistical-analysis-1.html#cb96-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">horiz =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-62-1.png" width="768" /></p>
<ul>
<li><p>improve the visualization by changing the following arguments:</p>
<ul>
<li><p><code>repulsion = 0.8</code>: Place the nodes further apart</p></li>
<li><p><code>rmSingles = TRUE</code>: Single nodes are removed</p></li>
<li><p><code>labelScale = FALSE</code> and <code>cexLabels = 1.6</code>: All labels have equal size and are enlarged to improve readability of small node’s labels</p></li>
<li><p><code>nodeSizeSpread = 3</code> (default is 4): Node sizes are more similar if the value is decreased. This argument (in combination with <code>cexNodes</code>) is useful to enlarge small nodes while keeping the size of big nodes.</p></li>
</ul></li>
</ul>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="statistical-analysis-1.html#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(props_single, </span>
<span id="cb97-2"><a href="statistical-analysis-1.html#cb97-2" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeColor =</span> <span class="st">&quot;cluster&quot;</span>, </span>
<span id="cb97-3"><a href="statistical-analysis-1.html#cb97-3" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeSize =</span> <span class="st">&quot;eigenvector&quot;</span>,</span>
<span id="cb97-4"><a href="statistical-analysis-1.html#cb97-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">repulsion =</span> <span class="fl">0.8</span>,</span>
<span id="cb97-5"><a href="statistical-analysis-1.html#cb97-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">rmSingles =</span> <span class="cn">TRUE</span>,</span>
<span id="cb97-6"><a href="statistical-analysis-1.html#cb97-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">labelScale =</span> <span class="cn">FALSE</span>,</span>
<span id="cb97-7"><a href="statistical-analysis-1.html#cb97-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexLabels =</span> <span class="fl">1.6</span>,</span>
<span id="cb97-8"><a href="statistical-analysis-1.html#cb97-8" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeSizeSpread =</span> <span class="dv">3</span>,</span>
<span id="cb97-9"><a href="statistical-analysis-1.html#cb97-9" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexNodes =</span> <span class="dv">2</span>,</span>
<span id="cb97-10"><a href="statistical-analysis-1.html#cb97-10" aria-hidden="true" tabindex="-1"></a>     <span class="at">title1 =</span> <span class="st">&quot;Network on metabolomics with Pearson correlations&quot;</span>, </span>
<span id="cb97-11"><a href="statistical-analysis-1.html#cb97-11" aria-hidden="true" tabindex="-1"></a>     <span class="at">showTitle =</span> <span class="cn">TRUE</span>,</span>
<span id="cb97-12"><a href="statistical-analysis-1.html#cb97-12" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexTitle =</span> <span class="fl">1.5</span>)</span>
<span id="cb97-13"><a href="statistical-analysis-1.html#cb97-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-14"><a href="statistical-analysis-1.html#cb97-14" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="fl">0.7</span>, <span class="fl">1.1</span>, <span class="at">cex =</span> <span class="fl">1.2</span>, <span class="at">title =</span> <span class="st">&quot;estimated correlation:&quot;</span>,</span>
<span id="cb97-15"><a href="statistical-analysis-1.html#cb97-15" aria-hidden="true" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;+&quot;</span>,<span class="st">&quot;-&quot;</span>), <span class="at">lty =</span> <span class="dv">1</span>, <span class="at">lwd =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;#009900&quot;</span>,<span class="st">&quot;red&quot;</span>),</span>
<span id="cb97-16"><a href="statistical-analysis-1.html#cb97-16" aria-hidden="true" tabindex="-1"></a>       <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">horiz =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-63-1.png" width="1152" /></p>
</div>
</div>
<div id="single-network-with-spearman-correlation-as-association-measure" class="section level4 hasAnchor" number="2.6.5.2">
<h4><span class="header-section-number">2.6.5.2</span> Single network with spearman correlation as association measure<a href="statistical-analysis-1.html#single-network-with-spearman-correlation-as-association-measure" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div id="building-network-model-1" class="section level5 hasAnchor" number="2.6.5.2.1">
<h5><span class="header-section-number">2.6.5.2.1</span> Building network model<a href="statistical-analysis-1.html#building-network-model-1" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="statistical-analysis-1.html#cb98-1" aria-hidden="true" tabindex="-1"></a>net_single2 <span class="ot">&lt;-</span> <span class="fu">netConstruct</span>(features_tab,  </span>
<span id="cb98-2"><a href="statistical-analysis-1.html#cb98-2" aria-hidden="true" tabindex="-1"></a>                            <span class="at">measure =</span> <span class="st">&quot;spearman&quot;</span>,</span>
<span id="cb98-3"><a href="statistical-analysis-1.html#cb98-3" aria-hidden="true" tabindex="-1"></a>                            <span class="at">normMethod =</span> <span class="st">&quot;clr&quot;</span>, </span>
<span id="cb98-4"><a href="statistical-analysis-1.html#cb98-4" aria-hidden="true" tabindex="-1"></a>                            <span class="at">zeroMethod =</span> <span class="st">&quot;multRepl&quot;</span>,</span>
<span id="cb98-5"><a href="statistical-analysis-1.html#cb98-5" aria-hidden="true" tabindex="-1"></a>                            <span class="at">sparsMethod =</span> <span class="st">&quot;threshold&quot;</span>, </span>
<span id="cb98-6"><a href="statistical-analysis-1.html#cb98-6" aria-hidden="true" tabindex="-1"></a>                            <span class="at">thresh =</span> <span class="fl">0.3</span>,</span>
<span id="cb98-7"><a href="statistical-analysis-1.html#cb98-7" aria-hidden="true" tabindex="-1"></a>                            <span class="at">verbose =</span> <span class="dv">3</span>,</span>
<span id="cb98-8"><a href="statistical-analysis-1.html#cb98-8" aria-hidden="true" tabindex="-1"></a>                            <span class="at">seed =</span> <span class="dv">123</span>)</span></code></pre></div>
</div>
<div id="visualizing-the-network-1" class="section level5 hasAnchor" number="2.6.5.2.2">
<h5><span class="header-section-number">2.6.5.2.2</span> Visualizing the network<a href="statistical-analysis-1.html#visualizing-the-network-1" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="statistical-analysis-1.html#cb99-1" aria-hidden="true" tabindex="-1"></a>props_single2 <span class="ot">&lt;-</span> <span class="fu">netAnalyze</span>(net_single2, <span class="at">clustMethod =</span> <span class="st">&quot;cluster_fast_greedy&quot;</span>)</span>
<span id="cb99-2"><a href="statistical-analysis-1.html#cb99-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-3"><a href="statistical-analysis-1.html#cb99-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(props_single2, </span>
<span id="cb99-4"><a href="statistical-analysis-1.html#cb99-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeColor =</span> <span class="st">&quot;cluster&quot;</span>, </span>
<span id="cb99-5"><a href="statistical-analysis-1.html#cb99-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeSize =</span> <span class="st">&quot;eigenvector&quot;</span>,</span>
<span id="cb99-6"><a href="statistical-analysis-1.html#cb99-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">repulsion =</span> <span class="fl">0.8</span>,</span>
<span id="cb99-7"><a href="statistical-analysis-1.html#cb99-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">rmSingles =</span> <span class="cn">TRUE</span>,</span>
<span id="cb99-8"><a href="statistical-analysis-1.html#cb99-8" aria-hidden="true" tabindex="-1"></a>     <span class="at">labelScale =</span> <span class="cn">FALSE</span>,</span>
<span id="cb99-9"><a href="statistical-analysis-1.html#cb99-9" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexLabels =</span> <span class="fl">1.6</span>,</span>
<span id="cb99-10"><a href="statistical-analysis-1.html#cb99-10" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeSizeSpread =</span> <span class="dv">3</span>,</span>
<span id="cb99-11"><a href="statistical-analysis-1.html#cb99-11" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexNodes =</span> <span class="dv">2</span>,</span>
<span id="cb99-12"><a href="statistical-analysis-1.html#cb99-12" aria-hidden="true" tabindex="-1"></a>     <span class="at">title1 =</span> <span class="st">&quot;Network on metabolomics with Spearman correlations&quot;</span>, </span>
<span id="cb99-13"><a href="statistical-analysis-1.html#cb99-13" aria-hidden="true" tabindex="-1"></a>     <span class="at">showTitle =</span> <span class="cn">TRUE</span>,</span>
<span id="cb99-14"><a href="statistical-analysis-1.html#cb99-14" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexTitle =</span> <span class="fl">1.5</span>)</span>
<span id="cb99-15"><a href="statistical-analysis-1.html#cb99-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb99-16"><a href="statistical-analysis-1.html#cb99-16" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="fl">0.7</span>, <span class="fl">1.1</span>, <span class="at">cex =</span> <span class="fl">1.2</span>, <span class="at">title =</span> <span class="st">&quot;estimated correlation:&quot;</span>,</span>
<span id="cb99-17"><a href="statistical-analysis-1.html#cb99-17" aria-hidden="true" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;+&quot;</span>,<span class="st">&quot;-&quot;</span>), <span class="at">lty =</span> <span class="dv">1</span>, <span class="at">lwd =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;#009900&quot;</span>,<span class="st">&quot;red&quot;</span>),</span>
<span id="cb99-18"><a href="statistical-analysis-1.html#cb99-18" aria-hidden="true" tabindex="-1"></a>       <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">horiz =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-65-1.png" width="1152" /></p>
</div>
</div>
<div id="single-network-with-wgcna-bicor-as-association-measure" class="section level4 hasAnchor" number="2.6.5.3">
<h4><span class="header-section-number">2.6.5.3</span> Single network with WGCNA (bicor) as association measure<a href="statistical-analysis-1.html#single-network-with-wgcna-bicor-as-association-measure" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Biweight Midcorrelation <code>bicor()</code> from WGCNA package.</p>
<div id="building-network-model-2" class="section level5 hasAnchor" number="2.6.5.3.1">
<h5><span class="header-section-number">2.6.5.3.1</span> Building network model<a href="statistical-analysis-1.html#building-network-model-2" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="statistical-analysis-1.html#cb100-1" aria-hidden="true" tabindex="-1"></a>net_single3 <span class="ot">&lt;-</span> <span class="fu">netConstruct</span>(features_tab, </span>
<span id="cb100-2"><a href="statistical-analysis-1.html#cb100-2" aria-hidden="true" tabindex="-1"></a>                           <span class="at">measure =</span> <span class="st">&quot;bicor&quot;</span>,</span>
<span id="cb100-3"><a href="statistical-analysis-1.html#cb100-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">measurePar =</span> <span class="fu">list</span>(<span class="at">use =</span> <span class="st">&quot;all.obs&quot;</span>,</span>
<span id="cb100-4"><a href="statistical-analysis-1.html#cb100-4" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">maxPOutliers =</span> <span class="dv">1</span>,</span>
<span id="cb100-5"><a href="statistical-analysis-1.html#cb100-5" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">nThreads =</span> <span class="dv">2</span>),</span>
<span id="cb100-6"><a href="statistical-analysis-1.html#cb100-6" aria-hidden="true" tabindex="-1"></a>                           <span class="at">filtTax =</span> <span class="st">&quot;highestVar&quot;</span>,</span>
<span id="cb100-7"><a href="statistical-analysis-1.html#cb100-7" aria-hidden="true" tabindex="-1"></a>                           <span class="at">filtTaxPar =</span> <span class="fu">list</span>(<span class="at">highestVar =</span> <span class="dv">50</span>),</span>
<span id="cb100-8"><a href="statistical-analysis-1.html#cb100-8" aria-hidden="true" tabindex="-1"></a>                           <span class="at">filtSamp =</span> <span class="st">&quot;totalReads&quot;</span>,</span>
<span id="cb100-9"><a href="statistical-analysis-1.html#cb100-9" aria-hidden="true" tabindex="-1"></a>                           <span class="at">filtSampPar =</span> <span class="fu">list</span>(<span class="at">totalReads =</span> <span class="dv">100</span>),</span>
<span id="cb100-10"><a href="statistical-analysis-1.html#cb100-10" aria-hidden="true" tabindex="-1"></a>                           <span class="at">dissFunc =</span> <span class="st">&quot;TOMdiss&quot;</span>,</span>
<span id="cb100-11"><a href="statistical-analysis-1.html#cb100-11" aria-hidden="true" tabindex="-1"></a>                           <span class="at">verbose =</span> <span class="dv">3</span>)</span></code></pre></div>
<pre><code>## ..will use 2 parallel threads.
##  Fraction of slow calculations: 0.000000</code></pre>
</div>
<div id="visualizing-the-network-2" class="section level5 hasAnchor" number="2.6.5.3.2">
<h5><span class="header-section-number">2.6.5.3.2</span> Visualizing the network<a href="statistical-analysis-1.html#visualizing-the-network-2" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="statistical-analysis-1.html#cb102-1" aria-hidden="true" tabindex="-1"></a>props_single3 <span class="ot">&lt;-</span> <span class="fu">netAnalyze</span>(net_single3, <span class="at">clustMethod =</span> <span class="st">&quot;cluster_fast_greedy&quot;</span>)</span>
<span id="cb102-2"><a href="statistical-analysis-1.html#cb102-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-3"><a href="statistical-analysis-1.html#cb102-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(props_single3, </span>
<span id="cb102-4"><a href="statistical-analysis-1.html#cb102-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeColor =</span> <span class="st">&quot;cluster&quot;</span>, </span>
<span id="cb102-5"><a href="statistical-analysis-1.html#cb102-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeSize =</span> <span class="st">&quot;eigenvector&quot;</span>,</span>
<span id="cb102-6"><a href="statistical-analysis-1.html#cb102-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">repulsion =</span> <span class="fl">0.8</span>,</span>
<span id="cb102-7"><a href="statistical-analysis-1.html#cb102-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">rmSingles =</span> <span class="cn">TRUE</span>,</span>
<span id="cb102-8"><a href="statistical-analysis-1.html#cb102-8" aria-hidden="true" tabindex="-1"></a>     <span class="at">labelScale =</span> <span class="cn">FALSE</span>,</span>
<span id="cb102-9"><a href="statistical-analysis-1.html#cb102-9" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexLabels =</span> <span class="fl">1.6</span>,</span>
<span id="cb102-10"><a href="statistical-analysis-1.html#cb102-10" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeSizeSpread =</span> <span class="dv">3</span>,</span>
<span id="cb102-11"><a href="statistical-analysis-1.html#cb102-11" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexNodes =</span> <span class="dv">2</span>,</span>
<span id="cb102-12"><a href="statistical-analysis-1.html#cb102-12" aria-hidden="true" tabindex="-1"></a>     <span class="at">title1 =</span> <span class="st">&quot;Network on metabolomics with WGCNA correlations&quot;</span>, </span>
<span id="cb102-13"><a href="statistical-analysis-1.html#cb102-13" aria-hidden="true" tabindex="-1"></a>     <span class="at">showTitle =</span> <span class="cn">TRUE</span>,</span>
<span id="cb102-14"><a href="statistical-analysis-1.html#cb102-14" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexTitle =</span> <span class="fl">1.5</span>)</span>
<span id="cb102-15"><a href="statistical-analysis-1.html#cb102-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-16"><a href="statistical-analysis-1.html#cb102-16" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="fl">0.7</span>, <span class="fl">1.1</span>, <span class="at">cex =</span> <span class="fl">1.2</span>, <span class="at">title =</span> <span class="st">&quot;estimated correlation:&quot;</span>,</span>
<span id="cb102-17"><a href="statistical-analysis-1.html#cb102-17" aria-hidden="true" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;+&quot;</span>,<span class="st">&quot;-&quot;</span>), <span class="at">lty =</span> <span class="dv">1</span>, <span class="at">lwd =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;#009900&quot;</span>,<span class="st">&quot;red&quot;</span>),</span>
<span id="cb102-18"><a href="statistical-analysis-1.html#cb102-18" aria-hidden="true" tabindex="-1"></a>       <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">horiz =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-67-1.png" width="1152" /></p>
</div>
</div>
</div>
<div id="network-comparison" class="section level3 hasAnchor" number="2.6.6">
<h3><span class="header-section-number">2.6.6</span> Network comparison<a href="statistical-analysis-1.html#network-comparison" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Comparing two networks by <strong>NetCoMi</strong>.</p>
<div id="data-preparing" class="section level4 hasAnchor" number="2.6.6.1">
<h4><span class="header-section-number">2.6.6.1</span> Data preparing<a href="statistical-analysis-1.html#data-preparing" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="statistical-analysis-1.html#cb103-1" aria-hidden="true" tabindex="-1"></a>group_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Severe&quot;</span>)</span>
<span id="cb103-2"><a href="statistical-analysis-1.html#cb103-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-3"><a href="statistical-analysis-1.html#cb103-3" aria-hidden="true" tabindex="-1"></a>se_filter_subset <span class="ot">&lt;-</span> se_filter[, se_filter<span class="sc">$</span>group <span class="sc">%in%</span> group_names]</span>
<span id="cb103-4"><a href="statistical-analysis-1.html#cb103-4" aria-hidden="true" tabindex="-1"></a>se_filter_subset<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">as.character</span>(se_filter_subset<span class="sc">$</span>group))</span>
<span id="cb103-5"><a href="statistical-analysis-1.html#cb103-5" aria-hidden="true" tabindex="-1"></a>features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(se_filter_subset) <span class="sc">%&gt;%</span> </span>
<span id="cb103-6"><a href="statistical-analysis-1.html#cb103-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>()</span>
<span id="cb103-7"><a href="statistical-analysis-1.html#cb103-7" aria-hidden="true" tabindex="-1"></a>features_tab[<span class="fu">is.na</span>(features_tab)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb103-8"><a href="statistical-analysis-1.html#cb103-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-9"><a href="statistical-analysis-1.html#cb103-9" aria-hidden="true" tabindex="-1"></a>group_vector <span class="ot">&lt;-</span> se_filter_subset<span class="sc">$</span>group</span></code></pre></div>
</div>
<div id="building-network-model-3" class="section level4 hasAnchor" number="2.6.6.2">
<h4><span class="header-section-number">2.6.6.2</span> Building network model<a href="statistical-analysis-1.html#building-network-model-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="statistical-analysis-1.html#cb104-1" aria-hidden="true" tabindex="-1"></a>net_group <span class="ot">&lt;-</span> <span class="fu">netConstruct</span>(features_tab, </span>
<span id="cb104-2"><a href="statistical-analysis-1.html#cb104-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">group =</span> group_vector, </span>
<span id="cb104-3"><a href="statistical-analysis-1.html#cb104-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">measure =</span> <span class="st">&quot;pearson&quot;</span>,</span>
<span id="cb104-4"><a href="statistical-analysis-1.html#cb104-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">normMethod =</span> <span class="st">&quot;clr&quot;</span>, </span>
<span id="cb104-5"><a href="statistical-analysis-1.html#cb104-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">zeroMethod =</span> <span class="st">&quot;multRepl&quot;</span>,</span>
<span id="cb104-6"><a href="statistical-analysis-1.html#cb104-6" aria-hidden="true" tabindex="-1"></a>                          <span class="at">sparsMethod =</span> <span class="st">&quot;threshold&quot;</span>, </span>
<span id="cb104-7"><a href="statistical-analysis-1.html#cb104-7" aria-hidden="true" tabindex="-1"></a>                          <span class="at">thresh =</span> <span class="fl">0.3</span>,</span>
<span id="cb104-8"><a href="statistical-analysis-1.html#cb104-8" aria-hidden="true" tabindex="-1"></a>                          <span class="at">verbose =</span> <span class="dv">3</span>,</span>
<span id="cb104-9"><a href="statistical-analysis-1.html#cb104-9" aria-hidden="true" tabindex="-1"></a>                          <span class="at">seed =</span> <span class="dv">123</span>)</span></code></pre></div>
</div>
<div id="network-analysis-1" class="section level4 hasAnchor" number="2.6.6.3">
<h4><span class="header-section-number">2.6.6.3</span> Network analysis<a href="statistical-analysis-1.html#network-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="statistical-analysis-1.html#cb105-1" aria-hidden="true" tabindex="-1"></a>props_group <span class="ot">&lt;-</span> <span class="fu">netAnalyze</span>(net_group, </span>
<span id="cb105-2"><a href="statistical-analysis-1.html#cb105-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">centrLCC =</span> <span class="cn">FALSE</span>,</span>
<span id="cb105-3"><a href="statistical-analysis-1.html#cb105-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">avDissIgnoreInf =</span> <span class="cn">TRUE</span>,</span>
<span id="cb105-4"><a href="statistical-analysis-1.html#cb105-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">sPathNorm =</span> <span class="cn">FALSE</span>,</span>
<span id="cb105-5"><a href="statistical-analysis-1.html#cb105-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">clustMethod =</span> <span class="st">&quot;cluster_fast_greedy&quot;</span>,</span>
<span id="cb105-6"><a href="statistical-analysis-1.html#cb105-6" aria-hidden="true" tabindex="-1"></a>                          <span class="at">hubPar =</span> <span class="fu">c</span>(<span class="st">&quot;degree&quot;</span>, <span class="st">&quot;between&quot;</span>, <span class="st">&quot;closeness&quot;</span>),</span>
<span id="cb105-7"><a href="statistical-analysis-1.html#cb105-7" aria-hidden="true" tabindex="-1"></a>                          <span class="at">hubQuant =</span> <span class="fl">0.9</span>,</span>
<span id="cb105-8"><a href="statistical-analysis-1.html#cb105-8" aria-hidden="true" tabindex="-1"></a>                          <span class="at">lnormFit =</span> <span class="cn">TRUE</span>,</span>
<span id="cb105-9"><a href="statistical-analysis-1.html#cb105-9" aria-hidden="true" tabindex="-1"></a>                          <span class="at">normDeg =</span> <span class="cn">FALSE</span>,</span>
<span id="cb105-10"><a href="statistical-analysis-1.html#cb105-10" aria-hidden="true" tabindex="-1"></a>                          <span class="at">normBetw =</span> <span class="cn">FALSE</span>,</span>
<span id="cb105-11"><a href="statistical-analysis-1.html#cb105-11" aria-hidden="true" tabindex="-1"></a>                          <span class="at">normClose =</span> <span class="cn">FALSE</span>,</span>
<span id="cb105-12"><a href="statistical-analysis-1.html#cb105-12" aria-hidden="true" tabindex="-1"></a>                          <span class="at">normEigen =</span> <span class="cn">FALSE</span>)</span>
<span id="cb105-13"><a href="statistical-analysis-1.html#cb105-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-14"><a href="statistical-analysis-1.html#cb105-14" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(props_group)</span></code></pre></div>
<pre><code>## 
## Component sizes
## ```````````````
## Group 1:        
## size: 48
##    #:  1
## 
## Group 2:        
## size: 48
##    #:  1
## ______________________________
## Global network properties
## `````````````````````````
##                          group &#39;1&#39; group &#39;2&#39;
## Number of components       1.00000   1.00000
## Clustering coefficient     0.50151   0.48747
## Modularity                 0.15580   0.08807
## Positive edge percentage  51.11607  49.24078
## Edge density               0.39716   0.40869
## Natural connectivity       0.09563   0.08721
## Vertex connectivity        9.00000   7.00000
## Edge connectivity          9.00000   7.00000
## Average dissimilarity*     0.67288   0.68152
## Average path length**      0.92286   0.91581
## 
##  *Dissimilarity = 1 - edge weight
## **Path length: Sum of dissimilarities along the path
## 
## ______________________________
## Clusters
## - In the whole network
## - Algorithm: cluster_fast_greedy
## ```````````````````````````````` 
## group &#39;1&#39;:                
## name: 1  2  3  4
##    #: 6 15 14 13
## 
## group &#39;2&#39;:                
## name:  1  2  3 4
##    #: 18 10 12 8
## 
## ______________________________
## Hubs
## - In alphabetical/numerical order
## - Based on log-normal quantiles of centralities
## ```````````````````````````````````````````````
## No hubs detected.
## ______________________________
## Centrality measures
## - In decreasing order
## - Computed for the complete network
## ````````````````````````````````````
## Degree (unnormalized):
##          group &#39;1&#39; group &#39;2&#39;
##  M_48258        29        19
##  M_42398        26        25
##  M_47118        26        27
##  M_35186        25        22
##  M_55072        24        21
##             ______    ______
##  M_47118        26        27
##  M_19130        14        26
##  M_42446        18        26
##  M_33230        22        25
##  M_52452        23        25
## 
## Betweenness centrality (unnormalized):
##          group &#39;1&#39; group &#39;2&#39;
##  M_35186        51         5
##  M_42450        31        27
##  M_52452        31        18
##  M_42398        31        20
##  M_35253        31         5
##             ______    ______
##  M_19130         5        34
##  M_33230         3        32
##  M_63361         5        31
##  M_53189         7        27
##  M_42450        31        27
## 
## Closeness centrality (unnormalized):
##          group &#39;1&#39; group &#39;2&#39;
##  M_35253  71.79398  51.58148
##  M_42398   70.9484  63.74822
##  M_33961  68.32261  55.35701
##  M_48258  67.49245   61.7771
##  M_33955  66.93392  57.32211
##             ______    ______
##  M_52447  63.54214  66.40753
##  M_52464  62.24541  65.07936
##  M_33230  62.53671  63.84338
##  M_42398   70.9484  63.74822
##  M_47118   64.7125  63.10391
## 
## Eigenvector centrality (unnormalized):
##          group &#39;1&#39; group &#39;2&#39;
##  M_42398   0.26682   0.21086
##  M_35253   0.26167   0.08052
##  M_48258   0.25696    0.1782
##  M_35186   0.23664   0.20434
##  M_33961   0.23245   0.11227
##             ______    ______
##  M_33230   0.18321   0.21563
##  M_47118   0.20899   0.21435
##  M_42398   0.26682   0.21086
##  M_35186   0.23664   0.20434
##  M_52447   0.13304   0.18604</code></pre>
</div>
<div id="visualizing-the-network-3" class="section level4 hasAnchor" number="2.6.6.4">
<h4><span class="header-section-number">2.6.6.4</span> Visualizing the network<a href="statistical-analysis-1.html#visualizing-the-network-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="statistical-analysis-1.html#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(props_group, </span>
<span id="cb107-2"><a href="statistical-analysis-1.html#cb107-2" aria-hidden="true" tabindex="-1"></a>     <span class="at">sameLayout =</span> <span class="cn">TRUE</span>, </span>
<span id="cb107-3"><a href="statistical-analysis-1.html#cb107-3" aria-hidden="true" tabindex="-1"></a>     <span class="at">layoutGroup =</span> <span class="dv">1</span>,</span>
<span id="cb107-4"><a href="statistical-analysis-1.html#cb107-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">rmSingles =</span> <span class="st">&quot;inboth&quot;</span>, </span>
<span id="cb107-5"><a href="statistical-analysis-1.html#cb107-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">nodeSize =</span> <span class="st">&quot;mclr&quot;</span>, </span>
<span id="cb107-6"><a href="statistical-analysis-1.html#cb107-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">labelScale =</span> <span class="cn">FALSE</span>,</span>
<span id="cb107-7"><a href="statistical-analysis-1.html#cb107-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexNodes =</span> <span class="dv">1</span>, </span>
<span id="cb107-8"><a href="statistical-analysis-1.html#cb107-8" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexLabels =</span> <span class="fl">1.5</span>,</span>
<span id="cb107-9"><a href="statistical-analysis-1.html#cb107-9" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexHubLabels =</span> <span class="dv">2</span>,</span>
<span id="cb107-10"><a href="statistical-analysis-1.html#cb107-10" aria-hidden="true" tabindex="-1"></a>     <span class="at">cexTitle =</span> <span class="dv">2</span>,</span>
<span id="cb107-11"><a href="statistical-analysis-1.html#cb107-11" aria-hidden="true" tabindex="-1"></a>     <span class="at">groupNames =</span> group_names,</span>
<span id="cb107-12"><a href="statistical-analysis-1.html#cb107-12" aria-hidden="true" tabindex="-1"></a>     <span class="at">hubBorderCol  =</span> <span class="st">&quot;gray40&quot;</span>)</span>
<span id="cb107-13"><a href="statistical-analysis-1.html#cb107-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-14"><a href="statistical-analysis-1.html#cb107-14" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;bottom&quot;</span>, <span class="at">title =</span> <span class="st">&quot;estimated association:&quot;</span>, <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;+&quot;</span>, <span class="st">&quot;-&quot;</span>), </span>
<span id="cb107-15"><a href="statistical-analysis-1.html#cb107-15" aria-hidden="true" tabindex="-1"></a>       <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;#009900&quot;</span>,<span class="st">&quot;red&quot;</span>), <span class="at">inset =</span> <span class="fl">0.04</span>, <span class="at">cex =</span> <span class="dv">3</span>, <span class="at">lty =</span> <span class="dv">1</span>, <span class="at">lwd =</span> <span class="dv">4</span>, </span>
<span id="cb107-16"><a href="statistical-analysis-1.html#cb107-16" aria-hidden="true" tabindex="-1"></a>       <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">horiz =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-71-1.png" width="1920" /></p>
</div>
<div id="quantitative-network-comparison" class="section level4 hasAnchor" number="2.6.6.5">
<h4><span class="header-section-number">2.6.6.5</span> Quantitative network comparison<a href="statistical-analysis-1.html#quantitative-network-comparison" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="statistical-analysis-1.html#cb108-1" aria-hidden="true" tabindex="-1"></a>comp_group <span class="ot">&lt;-</span> <span class="fu">netCompare</span>(props_group, <span class="at">permTest =</span> <span class="cn">FALSE</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb108-2"><a href="statistical-analysis-1.html#cb108-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-3"><a href="statistical-analysis-1.html#cb108-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(comp_group, </span>
<span id="cb108-4"><a href="statistical-analysis-1.html#cb108-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">groupNames =</span> group_names,</span>
<span id="cb108-5"><a href="statistical-analysis-1.html#cb108-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">showCentr =</span> <span class="fu">c</span>(<span class="st">&quot;degree&quot;</span>, <span class="st">&quot;between&quot;</span>, <span class="st">&quot;closeness&quot;</span>), </span>
<span id="cb108-6"><a href="statistical-analysis-1.html#cb108-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">numbNodes =</span> <span class="dv">5</span>)</span></code></pre></div>
<pre><code>## 
## Comparison of Network Properties
## ----------------------------------
## CALL: 
## netCompare(x = props_group, permTest = FALSE, verbose = FALSE)
## 
## ______________________________
## Global network properties
## `````````````````````````
##                            Mild   Severe    difference
## Number of components      1.000    1.000         0.000
## Clustering coefficient    0.502    0.487         0.014
## Moduarity                 0.156    0.088         0.068
## Positive edge percentage 51.116   49.241         1.875
## Edge density              0.397    0.409         0.012
## Natural connectivity      0.096    0.087         0.008
## Vertex connectivity       9.000    7.000         2.000
## Edge connectivity         9.000    7.000         2.000
## Average dissimilarity*    0.673    0.682         0.009
## Average path length**     0.923    0.916         0.007
## -----
##  *: Dissimilarity = 1 - edge weight
## **Path length: Sum of dissimilarities along the path
## 
## ______________________________
## Jaccard index (similarity betw. sets of most central nodes)
## ``````````````````````````````````````````````````````````
##                     Jacc   P(&lt;=Jacc)     P(&gt;=Jacc)   
## degree             0.278    0.412243      0.768928   
## betweenness centr. 0.120    0.014890 *    0.996495   
## closeness centr.   0.300    0.479343      0.702786   
## eigenvec. centr.   0.444    0.892398      0.223260   
## hub taxa           0.000    1.000000      1.000000   
## -----
## Jaccard index ranges from 0 (compl. different) to 1 (sets equal)
## 
## ______________________________
## Adjusted Rand index (similarity betw. clusterings)
## ``````````````````````````````````````````````````
##    ARI       p-value
##  0.121             0
## -----
## ARI in [-1,1] with ARI=1: perfect agreement betw. clusterings,
##                    ARI=0: expected for two random clusterings
## p-value: two-tailed test with null hypothesis ARI=0
## 
## ______________________________
## Centrality measures
## - In decreasing order
## - Computed for the complete network
## ````````````````````````````````````
## Degree (unnormalized):
##         Mild Severe abs.diff.
## M_19130   14     26        12
## M_19266   21     10        11
## M_48258   29     19        10
## M_49617   10     19         9
## M_35253   24     15         9
## 
## Betweenness centrality (unnormalized):
##         Mild Severe abs.diff.
## M_35186   51      5        46
## M_19130    5     34        29
## M_33230    3     32        29
## M_63361    5     31        26
## M_53229   27      1        26
## 
## Closeness centrality (unnormalized):
##           Mild Severe abs.diff.
## M_35253 71.794 51.581    20.213
## M_33961 68.323 55.357    12.966
## M_49617 46.281 57.240    10.958
## M_53189 49.398 59.611    10.213
## M_33955 66.934 57.322     9.612
## 
## _________________________________________________________
## Significance codes: ***: 0.001, **: 0.01, *: 0.05, .: 0.1</code></pre>
<hr />
</div>
</div>
</div>
<div id="systematic-information" class="section level2 hasAnchor" number="2.7">
<h2><span class="header-section-number">2.7</span> Systematic Information<a href="statistical-analysis-1.html#systematic-information" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="statistical-analysis-1.html#cb110-1" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 4.2.0 (2022-04-22)
##  os       macOS Monterey 12.5
##  system   x86_64, darwin17.0
##  ui       RStudio
##  language (EN)
##  collate  zh_CN.UTF-8
##  ctype    zh_CN.UTF-8
##  tz       Asia/Shanghai
##  date     2022-09-19
##  rstudio  2021.09.1+372 Ghost Orchid (desktop)
##  pandoc   2.18 @ /usr/local/bin/ (via rmarkdown)
## 
## ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package              * version    date (UTC) lib source
##  abind                  1.4-5      2016-07-21 [1] CRAN (R 4.2.0)
##  ade4                   1.7-19     2022-04-19 [1] CRAN (R 4.2.0)
##  AnnotationDbi          1.58.0     2022-04-26 [1] Bioconductor
##  ape                    5.6-2      2022-03-02 [1] CRAN (R 4.2.0)
##  assertthat             0.2.1      2019-03-21 [1] CRAN (R 4.2.0)
##  backports              1.4.1      2021-12-13 [1] CRAN (R 4.2.0)
##  base64enc              0.1-3      2015-07-28 [1] CRAN (R 4.2.0)
##  Biobase              * 2.56.0     2022-04-26 [1] Bioconductor
##  BiocGenerics         * 0.42.0     2022-04-26 [1] Bioconductor
##  BiocParallel           1.30.3     2022-06-05 [1] Bioconductor
##  biomformat             1.24.0     2022-04-26 [1] Bioconductor
##  Biostrings             2.64.0     2022-04-26 [1] Bioconductor
##  bit                    4.0.4      2020-08-04 [1] CRAN (R 4.2.0)
##  bit64                  4.0.5      2020-08-30 [1] CRAN (R 4.2.0)
##  bitops                 1.0-7      2021-04-24 [1] CRAN (R 4.2.0)
##  blob                   1.2.3      2022-04-10 [1] CRAN (R 4.2.0)
##  bookdown               0.27       2022-06-14 [1] CRAN (R 4.2.0)
##  brio                   1.1.3      2021-11-30 [1] CRAN (R 4.2.0)
##  broom                  0.8.0      2022-04-13 [1] CRAN (R 4.2.0)
##  bslib                  0.3.1      2021-10-06 [1] CRAN (R 4.2.0)
##  cachem                 1.0.6      2021-08-19 [1] CRAN (R 4.2.0)
##  callr                  3.7.0      2021-04-20 [1] CRAN (R 4.2.0)
##  car                    3.0-13     2022-05-02 [1] CRAN (R 4.2.0)
##  carData                3.0-5      2022-01-06 [1] CRAN (R 4.2.0)
##  caret                  6.0-92     2022-04-19 [1] CRAN (R 4.2.0)
##  cellranger             1.1.0      2016-07-27 [1] CRAN (R 4.2.0)
##  checkmate              2.1.0      2022-04-21 [1] CRAN (R 4.2.0)
##  circlize               0.4.15     2022-05-10 [1] CRAN (R 4.2.0)
##  class                  7.3-20     2022-01-16 [1] CRAN (R 4.2.0)
##  cli                    3.3.0      2022-04-25 [1] CRAN (R 4.2.0)
##  clue                   0.3-61     2022-05-30 [1] CRAN (R 4.2.0)
##  cluster              * 2.1.3      2022-03-28 [1] CRAN (R 4.2.0)
##  codetools              0.2-18     2020-11-04 [1] CRAN (R 4.2.0)
##  colorspace             2.0-3      2022-02-21 [1] CRAN (R 4.2.0)
##  ComplexHeatmap         2.12.0     2022-04-26 [1] Bioconductor
##  corpcor                1.6.10     2021-09-16 [1] CRAN (R 4.2.0)
##  cowplot                1.1.1      2020-12-30 [1] CRAN (R 4.2.0)
##  crayon                 1.5.1      2022-03-26 [1] CRAN (R 4.2.0)
##  data.table             1.14.2     2021-09-27 [1] CRAN (R 4.2.0)
##  DBI                    1.1.2      2021-12-20 [1] CRAN (R 4.2.0)
##  DelayedArray           0.22.0     2022-04-26 [1] Bioconductor
##  deldir                 1.0-6      2021-10-23 [1] CRAN (R 4.2.0)
##  dendextend           * 1.15.2     2021-10-28 [1] CRAN (R 4.2.0)
##  desc                   1.4.1      2022-03-06 [1] CRAN (R 4.2.0)
##  devtools               2.4.3      2021-11-30 [1] CRAN (R 4.2.0)
##  digest                 0.6.29     2021-12-01 [1] CRAN (R 4.2.0)
##  doParallel             1.0.17     2022-02-07 [1] CRAN (R 4.2.0)
##  doSNOW                 1.0.20     2022-02-04 [1] CRAN (R 4.2.0)
##  dplyr                * 1.0.9      2022-04-28 [1] CRAN (R 4.2.0)
##  dynamicTreeCut         1.63-1     2016-03-11 [1] CRAN (R 4.2.0)
##  e1071                  1.7-11     2022-06-07 [1] CRAN (R 4.2.0)
##  ellipse                0.4.3      2022-05-31 [1] CRAN (R 4.2.0)
##  ellipsis               0.3.2      2021-04-29 [1] CRAN (R 4.2.0)
##  evaluate               0.15       2022-02-18 [1] CRAN (R 4.2.0)
##  factoextra           * 1.0.7      2020-04-01 [1] CRAN (R 4.2.0)
##  fansi                  1.0.3      2022-03-24 [1] CRAN (R 4.2.0)
##  farver                 2.1.0      2021-02-28 [1] CRAN (R 4.2.0)
##  fastcluster            1.2.3      2021-05-24 [1] CRAN (R 4.2.0)
##  fastmap                1.1.0      2021-01-25 [1] CRAN (R 4.2.0)
##  fdrtool                1.2.17     2021-11-13 [1] CRAN (R 4.2.0)
##  filematrix             1.3        2018-02-27 [1] CRAN (R 4.2.0)
##  foreach                1.5.2      2022-02-02 [1] CRAN (R 4.2.0)
##  foreign                0.8-82     2022-01-16 [1] CRAN (R 4.2.0)
##  forestplot             2.0.1      2021-09-03 [1] CRAN (R 4.2.0)
##  Formula                1.2-4      2020-10-16 [1] CRAN (R 4.2.0)
##  fs                     1.5.2      2021-12-08 [1] CRAN (R 4.2.0)
##  future                 1.26.1     2022-05-27 [1] CRAN (R 4.2.0)
##  future.apply           1.9.0      2022-04-25 [1] CRAN (R 4.2.0)
##  generics               0.1.2      2022-01-31 [1] CRAN (R 4.2.0)
##  GenomeInfoDb         * 1.32.1     2022-04-28 [1] Bioconductor
##  GenomeInfoDbData       1.2.8      2022-05-16 [1] Bioconductor
##  GenomicRanges        * 1.48.0     2022-04-26 [1] Bioconductor
##  GetoptLong             1.0.5      2020-12-15 [1] CRAN (R 4.2.0)
##  ggforce                0.3.4      2022-08-18 [1] CRAN (R 4.2.0)
##  ggplot2              * 3.3.6      2022-05-03 [1] CRAN (R 4.2.0)
##  ggpubr                 0.4.0      2020-06-27 [1] CRAN (R 4.2.0)
##  ggraph               * 2.0.6      2022-08-08 [1] CRAN (R 4.2.0)
##  ggrepel                0.9.1      2021-01-15 [1] CRAN (R 4.2.0)
##  ggsignif               0.6.3      2021-09-09 [1] CRAN (R 4.2.0)
##  glasso                 1.11       2019-10-01 [1] CRAN (R 4.2.0)
##  glmnet               * 4.1-4      2022-04-15 [1] CRAN (R 4.2.0)
##  GlobalOptions          0.1.2      2020-06-10 [1] CRAN (R 4.2.0)
##  globals                0.15.1     2022-06-24 [1] CRAN (R 4.2.0)
##  glue                 * 1.6.2      2022-02-24 [1] CRAN (R 4.2.0)
##  Gmisc                * 3.0.0      2022-01-03 [1] CRAN (R 4.2.0)
##  gmp                    0.6-5      2022-03-17 [1] CRAN (R 4.2.0)
##  GO.db                  3.15.0     2022-09-14 [1] Bioconductor
##  gower                  1.0.0      2022-02-03 [1] CRAN (R 4.2.0)
##  graphlayouts           0.8.1      2022-08-11 [1] CRAN (R 4.2.0)
##  gridExtra              2.3        2017-09-09 [1] CRAN (R 4.2.0)
##  gtable                 0.3.0      2019-03-25 [1] CRAN (R 4.2.0)
##  gtools                 3.9.2.2    2022-06-13 [1] CRAN (R 4.2.0)
##  hardhat                1.1.0      2022-06-10 [1] CRAN (R 4.2.0)
##  highr                  0.9        2021-04-16 [1] CRAN (R 4.2.0)
##  Hmisc                  4.7-1      2022-08-15 [1] CRAN (R 4.2.0)
##  htmlTable            * 2.4.1      2022-07-07 [1] CRAN (R 4.2.0)
##  htmltools              0.5.2      2021-08-25 [1] CRAN (R 4.2.0)
##  htmlwidgets            1.5.4      2021-09-08 [1] CRAN (R 4.2.0)
##  httr                   1.4.3      2022-05-04 [1] CRAN (R 4.2.0)
##  huge                   1.3.5      2021-06-30 [1] CRAN (R 4.2.0)
##  igraph                 1.3.1      2022-04-20 [1] CRAN (R 4.2.0)
##  impute                 1.70.0     2022-04-26 [1] Bioconductor
##  interp                 1.1-3      2022-07-13 [1] CRAN (R 4.2.0)
##  ipred                  0.9-13     2022-06-02 [1] CRAN (R 4.2.0)
##  IRanges              * 2.30.0     2022-04-26 [1] Bioconductor
##  irlba                  2.3.5      2021-12-06 [1] CRAN (R 4.2.0)
##  iterators              1.0.14     2022-02-05 [1] CRAN (R 4.2.0)
##  jpeg                   0.1-9      2021-07-24 [1] CRAN (R 4.2.0)
##  jquerylib              0.1.4      2021-04-26 [1] CRAN (R 4.2.0)
##  jsonlite               1.8.0      2022-02-22 [1] CRAN (R 4.2.0)
##  KEGGREST               1.36.2     2022-06-09 [1] Bioconductor
##  knitr                  1.39       2022-04-26 [1] CRAN (R 4.2.0)
##  labeling               0.4.2      2020-10-20 [1] CRAN (R 4.2.0)
##  lattice                0.20-45    2021-09-22 [1] CRAN (R 4.2.0)
##  latticeExtra           0.6-30     2022-07-04 [1] CRAN (R 4.2.0)
##  lava                   1.6.10     2021-09-02 [1] CRAN (R 4.2.0)
##  lavaan                 0.6-12     2022-07-04 [1] CRAN (R 4.2.0)
##  lazyeval               0.2.2      2019-03-15 [1] CRAN (R 4.2.0)
##  lifecycle              1.0.1      2021-09-24 [1] CRAN (R 4.2.0)
##  limma                  3.52.2     2022-06-21 [1] Bioconductor
##  listenv                0.8.0      2019-12-05 [1] CRAN (R 4.2.0)
##  lubridate              1.8.0      2021-10-07 [1] CRAN (R 4.2.0)
##  magrittr             * 2.0.3      2022-03-30 [1] CRAN (R 4.2.0)
##  MASS                   7.3-56     2022-03-23 [1] CRAN (R 4.2.0)
##  Matrix               * 1.4-1      2022-03-23 [1] CRAN (R 4.2.0)
##  MatrixGenerics       * 1.8.1      2022-06-30 [1] Bioconductor
##  matrixStats          * 0.62.0     2022-04-19 [1] CRAN (R 4.2.0)
##  memoise                2.0.1      2021-11-26 [1] CRAN (R 4.2.0)
##  mgcv                   1.8-40     2022-03-29 [1] CRAN (R 4.2.0)
##  mixedCCA               1.6.2      2022-09-19 [1] Github (irinagain/mixedCCA@4c2b63f)
##  mixOmics               6.20.0     2022-04-26 [1] Bioconductor (R 4.2.0)
##  mnormt                 2.1.0      2022-06-07 [1] CRAN (R 4.2.0)
##  ModelMetrics           1.2.2.2    2020-03-17 [1] CRAN (R 4.2.0)
##  multtest               2.52.0     2022-04-26 [1] Bioconductor
##  munsell                0.5.0      2018-06-12 [1] CRAN (R 4.2.0)
##  mvtnorm                1.1-3      2021-10-08 [1] CRAN (R 4.2.0)
##  NADA                   1.6-1.1    2020-03-22 [1] CRAN (R 4.2.0)
##  NetCoMi              * 1.0.3      2022-09-19 [1] Github (stefpeschel/NetCoMi@1260971)
##  nlme                   3.1-157    2022-03-25 [1] CRAN (R 4.2.0)
##  nnet                   7.3-17     2022-01-16 [1] CRAN (R 4.2.0)
##  parallelly             1.32.0     2022-06-07 [1] CRAN (R 4.2.0)
##  pbapply                1.5-0      2021-09-16 [1] CRAN (R 4.2.0)
##  pbivnorm               0.6.0      2015-01-23 [1] CRAN (R 4.2.0)
##  pcaPP                  2.0-2      2022-07-08 [1] CRAN (R 4.2.0)
##  permute                0.9-7      2022-01-27 [1] CRAN (R 4.2.0)
##  phyloseq               1.40.0     2022-04-26 [1] Bioconductor
##  pillar                 1.7.0      2022-02-01 [1] CRAN (R 4.2.0)
##  pkgbuild               1.3.1      2021-12-20 [1] CRAN (R 4.2.0)
##  pkgconfig              2.0.3      2019-09-22 [1] CRAN (R 4.2.0)
##  pkgload                1.2.4      2021-11-30 [1] CRAN (R 4.2.0)
##  plotly               * 4.10.0     2021-10-09 [1] CRAN (R 4.2.0)
##  plyr                   1.8.7      2022-03-24 [1] CRAN (R 4.2.0)
##  png                    0.1-7      2013-12-03 [1] CRAN (R 4.2.0)
##  polyclip               1.10-0     2019-03-14 [1] CRAN (R 4.2.0)
##  POMA                 * 1.6.0      2022-04-26 [1] Bioconductor
##  preprocessCore         1.58.0     2022-04-26 [1] Bioconductor
##  prettyunits            1.1.1      2020-01-24 [1] CRAN (R 4.2.0)
##  pROC                   1.18.0     2021-09-03 [1] CRAN (R 4.2.0)
##  processx               3.5.3      2022-03-25 [1] CRAN (R 4.2.0)
##  prodlim                2019.11.13 2019-11-17 [1] CRAN (R 4.2.0)
##  proxy                  0.4-27     2022-06-09 [1] CRAN (R 4.2.0)
##  ps                     1.7.0      2022-04-23 [1] CRAN (R 4.2.0)
##  psych                  2.2.5      2022-05-10 [1] CRAN (R 4.2.0)
##  pulsar                 0.3.8      2022-08-27 [1] CRAN (R 4.2.0)
##  purrr                  0.3.4      2020-04-17 [1] CRAN (R 4.2.0)
##  qgraph                 1.9.2      2022-03-04 [1] CRAN (R 4.2.0)
##  R6                     2.5.1      2021-08-19 [1] CRAN (R 4.2.0)
##  randomForest           4.7-1.1    2022-05-23 [1] CRAN (R 4.2.0)
##  RankProd               3.22.0     2022-04-26 [1] Bioconductor
##  rARPACK                0.11-0     2016-03-10 [1] CRAN (R 4.2.0)
##  rbibutils              2.2.9      2022-08-15 [1] CRAN (R 4.2.0)
##  RColorBrewer           1.1-3      2022-04-03 [1] CRAN (R 4.2.0)
##  Rcpp                 * 1.0.8.3    2022-03-17 [1] CRAN (R 4.2.0)
##  RCurl                  1.98-1.6   2022-02-08 [1] CRAN (R 4.2.0)
##  Rdpack                 2.4        2022-07-20 [1] CRAN (R 4.2.0)
##  readxl               * 1.4.0      2022-03-28 [1] CRAN (R 4.2.0)
##  recipes                0.2.0      2022-02-18 [1] CRAN (R 4.2.0)
##  remotes                2.4.2      2021-11-30 [1] CRAN (R 4.2.0)
##  reshape2               1.4.4      2020-04-09 [1] CRAN (R 4.2.0)
##  rhdf5                  2.40.0     2022-04-26 [1] Bioconductor
##  rhdf5filters           1.8.0      2022-04-26 [1] Bioconductor
##  Rhdf5lib               1.18.0     2022-04-26 [1] Bioconductor
##  rjson                  0.2.21     2022-01-09 [1] CRAN (R 4.2.0)
##  rlang                  1.0.2      2022-03-04 [1] CRAN (R 4.2.0)
##  rmarkdown              2.14       2022-04-25 [1] CRAN (R 4.2.0)
##  Rmpfr                  0.8-9      2022-06-03 [1] CRAN (R 4.2.0)
##  rootSolve              1.8.2.3    2021-09-29 [1] CRAN (R 4.2.0)
##  rpart                  4.1.16     2022-01-24 [1] CRAN (R 4.2.0)
##  rprojroot              2.0.3      2022-04-02 [1] CRAN (R 4.2.0)
##  RSpectra               0.16-1     2022-04-24 [1] CRAN (R 4.2.0)
##  RSQLite                2.2.14     2022-05-07 [1] CRAN (R 4.2.0)
##  rstatix                0.7.0      2021-02-13 [1] CRAN (R 4.2.0)
##  rstudioapi             0.13       2020-11-12 [1] CRAN (R 4.2.0)
##  S4Vectors            * 0.34.0     2022-04-26 [1] Bioconductor
##  sass                   0.4.1      2022-03-23 [1] CRAN (R 4.2.0)
##  scales                 1.2.0      2022-04-13 [1] CRAN (R 4.2.0)
##  sessioninfo            1.2.2      2021-12-06 [1] CRAN (R 4.2.0)
##  shape                  1.4.6      2021-05-19 [1] CRAN (R 4.2.0)
##  snow                   0.4-4      2021-10-27 [1] CRAN (R 4.2.0)
##  SpiecEasi            * 1.1.2      2022-09-19 [1] Github (zdk123/SpiecEasi@c463727)
##  SPRING               * 1.0.4      2022-09-19 [1] Github (GraceYoon/SPRING@3d641a4)
##  stringi                1.7.6      2021-11-29 [1] CRAN (R 4.2.0)
##  stringr                1.4.0      2019-02-10 [1] CRAN (R 4.2.0)
##  SummarizedExperiment * 1.26.1     2022-05-01 [1] Bioconductor
##  survival               3.3-1      2022-03-03 [1] CRAN (R 4.2.0)
##  testthat               3.1.4      2022-04-26 [1] CRAN (R 4.2.0)
##  tibble               * 3.1.7      2022-05-03 [1] CRAN (R 4.2.0)
##  tidygraph              1.2.2      2022-08-22 [1] CRAN (R 4.2.0)
##  tidyr                  1.2.0      2022-02-01 [1] CRAN (R 4.2.0)
##  tidyselect             1.1.2      2022-02-21 [1] CRAN (R 4.2.0)
##  timeDate               3043.102   2018-02-21 [1] CRAN (R 4.2.0)
##  truncnorm              1.0-8      2018-02-27 [1] CRAN (R 4.2.0)
##  tweenr                 2.0.2      2022-09-06 [1] CRAN (R 4.2.0)
##  usethis                2.1.6      2022-05-25 [1] CRAN (R 4.2.0)
##  utf8                   1.2.2      2021-07-24 [1] CRAN (R 4.2.0)
##  vctrs                  0.4.1      2022-04-13 [1] CRAN (R 4.2.0)
##  vegan                  2.6-2      2022-04-17 [1] CRAN (R 4.2.0)
##  VGAM                   1.1-6      2022-02-14 [1] CRAN (R 4.2.0)
##  viridis                0.6.2      2021-10-13 [1] CRAN (R 4.2.0)
##  viridisLite            0.4.0      2021-04-13 [1] CRAN (R 4.2.0)
##  WGCNA                  1.71       2022-04-22 [1] CRAN (R 4.2.0)
##  withr                  2.5.0      2022-03-03 [1] CRAN (R 4.2.0)
##  xfun                   0.31       2022-05-10 [1] CRAN (R 4.2.0)
##  XML                    3.99-0.10  2022-06-09 [1] CRAN (R 4.2.0)
##  XVector                0.36.0     2022-04-26 [1] Bioconductor
##  yaml                   2.3.5      2022-02-21 [1] CRAN (R 4.2.0)
##  zCompositions          1.4.0-1    2022-03-26 [1] CRAN (R 4.2.0)
##  zlibbioc               1.42.0     2022-04-26 [1] Bioconductor
## 
##  [1] /Library/Frameworks/R.framework/Versions/4.2/Resources/library
## 
## ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-castellano2021pomashiny" class="csl-entry">
Castellano-Escuder, Pol, Raúl González-Domı́nguez, Francesc Carmona-Pontaque, Cristina Andrés-Lacueva, and Alex Sánchez-Pla. 2021. <span>“POMAShiny: A User-Friendly Web-Based Workflow for Metabolomics and Proteomics Data Analysis.”</span> <em>PLoS Computational Biology</em> 17 (7): e1009148.
</div>
<div id="ref-pang2020metaboanalystr" class="csl-entry">
Pang, Zhiqiang, Jasmine Chong, Shuzhao Li, and Jianguo Xia. 2020. <span>“MetaboAnalystR 3.0: Toward an Optimized Workflow for Global Metabolomics.”</span> <em>Metabolites</em> 10 (5): 186.
</div>
<div id="ref-peschel2021netcomi" class="csl-entry">
Peschel, Stefanie, Christian L Müller, Erika von Mutius, Anne-Laure Boulesteix, and Martin Depner. 2021. <span>“NetCoMi: Network Construction and Comparison for Microbiome Data in r.”</span> <em>Briefings in Bioinformatics</em> 22 (4): bbaa290.
</div>
<div id="ref-rohart2017mixomics" class="csl-entry">
Rohart, Florian, Benoı̂t Gautier, Amrit Singh, and Kim-Anh Lê Cao. 2017. <span>“mixOmics: An r Package for ‘Omics Feature Selection and Multiple Data Integration.”</span> <em>PLoS Computational Biology</em> 13 (11): e1005752.
</div>
<div id="ref-zeybel2022multiomics" class="csl-entry">
Zeybel, Mujdat, Muhammad Arif, Xiangyu Li, Ozlem Altay, Hong Yang, Mengnan Shi, Murat Akyildiz, et al. 2022. <span>“Multiomics Analysis Reveals the Impact of Microbiota on Host Metabolism in Hepatic Steatosis.”</span> <em>Advanced Science</em> 9 (11): 2104373.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="functional-analysis-1.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/Metabolomics_Aanlysis/blob/main/01-Statistical-Analysis.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection",
"scroll_highlight": true,
"toc_depth": 3
}
});
});
</script>

</body>

</html>
