<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Statistical Analysis | Data Analysis in Metabolomics</title>
  <meta name="description" content="A template for Data Analysis in Metabolomics" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Statistical Analysis | Data Analysis in Metabolomics" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A template for Data Analysis in Metabolomics" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Statistical Analysis | Data Analysis in Metabolomics" />
  
  <meta name="twitter:description" content="A template for Data Analysis in Metabolomics" />
  

<meta name="author" content="Hua Zou and Bangzhuo Tong" />


<meta name="date" content="2023-12-07" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="data-processing.html"/>
<link rel="next" href="functional-analysis-1.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="assets/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Data Analysis in Metabolomics</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#what-is-metabolomics"><i class="fa fa-check"></i><b>1.1</b> What is metabolomics?</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#workflow-of-metabolomics-data-analysis"><i class="fa fa-check"></i><b>1.2</b> Workflow of metabolomics data analysis</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#statistical-analysis"><i class="fa fa-check"></i><b>1.2.1</b> Statistical Analysis</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#functional-analysis"><i class="fa fa-check"></i><b>1.2.2</b> Functional Analysis</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#software"><i class="fa fa-check"></i><b>1.3</b> Software</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#reference"><i class="fa fa-check"></i><b>1.4</b> Reference</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-processing.html"><a href="data-processing.html"><i class="fa fa-check"></i><b>2</b> Data Processing</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-processing.html"><a href="data-processing.html#loading-packages"><i class="fa fa-check"></i><b>2.1</b> Loading packages</a></li>
<li class="chapter" data-level="2.2" data-path="data-processing.html"><a href="data-processing.html#importing-data"><i class="fa fa-check"></i><b>2.2</b> Importing data</a></li>
<li class="chapter" data-level="2.3" data-path="data-processing.html"><a href="data-processing.html#data-checking"><i class="fa fa-check"></i><b>2.3</b> Data Checking</a></li>
<li class="chapter" data-level="2.4" data-path="data-processing.html"><a href="data-processing.html#data-trimming"><i class="fa fa-check"></i><b>2.4</b> Data Trimming</a></li>
<li class="chapter" data-level="2.5" data-path="data-processing.html"><a href="data-processing.html#missing-value-imputation"><i class="fa fa-check"></i><b>2.5</b> Missing value imputation</a></li>
<li class="chapter" data-level="2.6" data-path="data-processing.html"><a href="data-processing.html#data-filtering"><i class="fa fa-check"></i><b>2.6</b> Data Filtering</a></li>
<li class="chapter" data-level="2.7" data-path="data-processing.html"><a href="data-processing.html#data-normalization"><i class="fa fa-check"></i><b>2.7</b> Data Normalization</a>
<ul>
<li class="chapter" data-level="2.7.1" data-path="data-processing.html"><a href="data-processing.html#normalization-by-normalizedata-function"><i class="fa fa-check"></i><b>2.7.1</b> Normalization by <strong>NormalizeData</strong> function</a></li>
<li class="chapter" data-level="2.7.2" data-path="data-processing.html"><a href="data-processing.html#normalization-by-poma-r-package"><i class="fa fa-check"></i><b>2.7.2</b> Normalization by <strong>POMA</strong> R package</a></li>
<li class="chapter" data-level="2.7.3" data-path="data-processing.html"><a href="data-processing.html#comparison-of-unnormalized-and-normalized-dataset"><i class="fa fa-check"></i><b>2.7.3</b> Comparison of unnormalized and normalized dataset</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="data-processing.html"><a href="data-processing.html#removing-outliers"><i class="fa fa-check"></i><b>2.8</b> Removing outliers</a></li>
<li class="chapter" data-level="2.9" data-path="data-processing.html"><a href="data-processing.html#data-distribution"><i class="fa fa-check"></i><b>2.9</b> Data distribution</a></li>
<li class="chapter" data-level="2.10" data-path="data-processing.html"><a href="data-processing.html#saving-datasets-into-rds-files"><i class="fa fa-check"></i><b>2.10</b> Saving datasets into RDS files</a></li>
<li class="chapter" data-level="2.11" data-path="data-processing.html"><a href="data-processing.html#systematic-information"><i class="fa fa-check"></i><b>2.11</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html"><i class="fa fa-check"></i><b>3</b> Statistical Analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-preprocessing"><i class="fa fa-check"></i><b>3.1</b> Data Preprocessing</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#environment-setup"><i class="fa fa-check"></i><b>3.1.1</b> Environment setup</a></li>
<li class="chapter" data-level="3.1.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-data"><i class="fa fa-check"></i><b>3.1.2</b> Loading data</a></li>
<li class="chapter" data-level="3.1.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#object-preparation"><i class="fa fa-check"></i><b>3.1.3</b> Object Preparation</a></li>
<li class="chapter" data-level="3.1.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-checking-1"><i class="fa fa-check"></i><b>3.1.4</b> Data Checking</a></li>
<li class="chapter" data-level="3.1.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#missing-value-imputation-1"><i class="fa fa-check"></i><b>3.1.5</b> Missing value imputation</a></li>
<li class="chapter" data-level="3.1.6" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-filtering-1"><i class="fa fa-check"></i><b>3.1.6</b> Data Filtering</a></li>
<li class="chapter" data-level="3.1.7" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-normalization-1"><i class="fa fa-check"></i><b>3.1.7</b> Data Normalization</a></li>
<li class="chapter" data-level="3.1.8" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#removing-outliers-1"><i class="fa fa-check"></i><b>3.1.8</b> Removing outliers</a></li>
<li class="chapter" data-level="3.1.9" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#saving-datasets-into-rds-files-1"><i class="fa fa-check"></i><b>3.1.9</b> Saving datasets into RDS files</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#cluster-analysis"><i class="fa fa-check"></i><b>3.2</b> Cluster Analysis</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-packages-1"><i class="fa fa-check"></i><b>3.2.1</b> Loading packages</a></li>
<li class="chapter" data-level="3.2.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#importing-data-1"><i class="fa fa-check"></i><b>3.2.2</b> Importing data</a></li>
<li class="chapter" data-level="3.2.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#hierarchical-clustering"><i class="fa fa-check"></i><b>3.2.3</b> Hierarchical Clustering</a></li>
<li class="chapter" data-level="3.2.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#partitional-clustering"><i class="fa fa-check"></i><b>3.2.4</b> Partitional Clustering</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#chemometrics-analysis"><i class="fa fa-check"></i><b>3.3</b> Chemometrics Analysis</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-packages-2"><i class="fa fa-check"></i><b>3.3.1</b> Loading packages</a></li>
<li class="chapter" data-level="3.3.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#importing-data-2"><i class="fa fa-check"></i><b>3.3.2</b> Importing data</a></li>
<li class="chapter" data-level="3.3.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>3.3.3</b> Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="3.3.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#partial-least-squares-discriminant-analysis-pls-da"><i class="fa fa-check"></i><b>3.3.4</b> Partial Least Squares-Discriminant Analysis (PLS-DA)</a></li>
<li class="chapter" data-level="3.3.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#sparse-partial-least-squares-discriminant-analysis-spls-da"><i class="fa fa-check"></i><b>3.3.5</b> Sparse Partial Least Squares-Discriminant Analysis (sPLS-DA)</a></li>
<li class="chapter" data-level="3.3.6" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#orthogonal-partial-least-squares-discriminant-analysis-orthopls-da"><i class="fa fa-check"></i><b>3.3.6</b> Orthogonal Partial Least Squares-Discriminant Analysis (orthoPLS-DA)</a></li>
<li class="chapter" data-level="3.3.7" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#chemometrics-analysis-by-own-scripts"><i class="fa fa-check"></i><b>3.3.7</b> Chemometrics Analysis by own scripts</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#univariate-analysis"><i class="fa fa-check"></i><b>3.4</b> Univariate Analysis</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-packages-3"><i class="fa fa-check"></i><b>3.4.1</b> Loading packages</a></li>
<li class="chapter" data-level="3.4.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#importing-data-3"><i class="fa fa-check"></i><b>3.4.2</b> Importing data</a></li>
<li class="chapter" data-level="3.4.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#fold-change-analysis"><i class="fa fa-check"></i><b>3.4.3</b> Fold Change Analysis</a></li>
<li class="chapter" data-level="3.4.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#t-test"><i class="fa fa-check"></i><b>3.4.4</b> T Test</a></li>
<li class="chapter" data-level="3.4.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#wilcoxon-test"><i class="fa fa-check"></i><b>3.4.5</b> Wilcoxon Test</a></li>
<li class="chapter" data-level="3.4.6" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#limma-test"><i class="fa fa-check"></i><b>3.4.6</b> Limma Test</a></li>
<li class="chapter" data-level="3.4.7" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#volcano-plot"><i class="fa fa-check"></i><b>3.4.7</b> Volcano plot</a></li>
<li class="chapter" data-level="3.4.8" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#vip-variable-influence-on-projection-coefficient"><i class="fa fa-check"></i><b>3.4.8</b> VIP (Variable influence on projection &amp; coefficient)</a></li>
<li class="chapter" data-level="3.4.9" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#t-test-by-local-codes"><i class="fa fa-check"></i><b>3.4.9</b> T-test by local codes</a></li>
<li class="chapter" data-level="3.4.10" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#merging-result"><i class="fa fa-check"></i><b>3.4.10</b> Merging result</a></li>
<li class="chapter" data-level="3.4.11" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#volcano-of-merged-results"><i class="fa fa-check"></i><b>3.4.11</b> Volcano of Merged Results</a></li>
<li class="chapter" data-level="3.4.12" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#adjusting-for-confounding-variables"><i class="fa fa-check"></i><b>3.4.12</b> Adjusting for confounding variables</a></li>
<li class="chapter" data-level="3.4.13" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#correlation-heatmaps"><i class="fa fa-check"></i><b>3.4.13</b> Correlation Heatmaps</a></li>
<li class="chapter" data-level="3.4.14" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#glasso-this-function-will-compute-a-gaussian-graphical-model-using-the-glmnet-package"><i class="fa fa-check"></i><b>3.4.14</b> glasso: this function will compute a Gaussian Graphical Model using the glmnet package</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#multivariate-analysis"><i class="fa fa-check"></i><b>3.5</b> Multivariate analysis</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-packages-4"><i class="fa fa-check"></i><b>3.5.1</b> Loading packages</a></li>
<li class="chapter" data-level="3.5.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#importing-data-4"><i class="fa fa-check"></i><b>3.5.2</b> Importing data</a></li>
<li class="chapter" data-level="3.5.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-curation"><i class="fa fa-check"></i><b>3.5.3</b> Data curation</a></li>
<li class="chapter" data-level="3.5.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#regularized-generalized-linear-models"><i class="fa fa-check"></i><b>3.5.4</b> Regularized Generalized Linear Models</a></li>
<li class="chapter" data-level="3.5.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#classification"><i class="fa fa-check"></i><b>3.5.5</b> Classification</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#network-analysis"><i class="fa fa-check"></i><b>3.6</b> Network Analysis</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#introduction-1"><i class="fa fa-check"></i><b>3.6.1</b> Introduction</a></li>
<li class="chapter" data-level="3.6.2" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#loading-packages-5"><i class="fa fa-check"></i><b>3.6.2</b> Loading packages</a></li>
<li class="chapter" data-level="3.6.3" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#importing-data-5"><i class="fa fa-check"></i><b>3.6.3</b> Importing data</a></li>
<li class="chapter" data-level="3.6.4" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#data-curation-1"><i class="fa fa-check"></i><b>3.6.4</b> Data curation</a></li>
<li class="chapter" data-level="3.6.5" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#associations-among-features"><i class="fa fa-check"></i><b>3.6.5</b> Associations Among Features</a></li>
<li class="chapter" data-level="3.6.6" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#network-comparison"><i class="fa fa-check"></i><b>3.6.6</b> Network comparison</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="statistical-analysis-1.html"><a href="statistical-analysis-1.html#systematic-information-1"><i class="fa fa-check"></i><b>3.7</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html"><i class="fa fa-check"></i><b>4</b> Functional Analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#enrichment-analysis"><i class="fa fa-check"></i><b>4.1</b> Enrichment Analysis</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#enviroment-setup"><i class="fa fa-check"></i><b>4.1.1</b> Enviroment Setup</a></li>
<li class="chapter" data-level="4.1.2" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#over-representation-analysis-ora"><i class="fa fa-check"></i><b>4.1.2</b> Over representation analysis (ORA)</a></li>
<li class="chapter" data-level="4.1.3" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#clean-environment-1"><i class="fa fa-check"></i><b>4.1.3</b> Clean environment 1</a></li>
<li class="chapter" data-level="4.1.4" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#single-sample-profiling"><i class="fa fa-check"></i><b>4.1.4</b> Single Sample Profiling</a></li>
<li class="chapter" data-level="4.1.5" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#clean-environment-2"><i class="fa fa-check"></i><b>4.1.5</b> Clean environment 2</a></li>
<li class="chapter" data-level="4.1.6" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#quantitative-enrichment-analysis-qea"><i class="fa fa-check"></i><b>4.1.6</b> Quantitative Enrichment Analysis (QEA)</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#pathway-analysis"><i class="fa fa-check"></i><b>4.2</b> Pathway Analysis</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#enviroment-set-up"><i class="fa fa-check"></i><b>4.2.1</b> Enviroment Set up</a></li>
<li class="chapter" data-level="4.2.2" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#over-representation-analysis"><i class="fa fa-check"></i><b>4.2.2</b> Over representation analysis</a></li>
<li class="chapter" data-level="4.2.3" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#clean-environment-3"><i class="fa fa-check"></i><b>4.2.3</b> Clean environment 3</a></li>
<li class="chapter" data-level="4.2.4" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#concentration-table-qea"><i class="fa fa-check"></i><b>4.2.4</b> Concentration Table (QEA)</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#functional-analysis-by-r-package"><i class="fa fa-check"></i><b>4.3</b> Functional Analysis by R package</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#loading-r-packages"><i class="fa fa-check"></i><b>4.3.1</b> Loading R packages</a></li>
<li class="chapter" data-level="4.3.2" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#importing-data-6"><i class="fa fa-check"></i><b>4.3.2</b> Importing Data</a></li>
<li class="chapter" data-level="4.3.3" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#differential-analysis"><i class="fa fa-check"></i><b>4.3.3</b> Differential Analysis</a></li>
<li class="chapter" data-level="4.3.4" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#metabolites-id-for-merged-results"><i class="fa fa-check"></i><b>4.3.4</b> Metabolites ID for Merged Results</a></li>
<li class="chapter" data-level="4.3.5" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#kegg-pathway-compound"><i class="fa fa-check"></i><b>4.3.5</b> KEGG PATHWAY COMPOUND</a></li>
<li class="chapter" data-level="4.3.6" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#functions-related-to-analysis"><i class="fa fa-check"></i><b>4.3.6</b> Functions related to Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#session-info"><i class="fa fa-check"></i><b>4.4</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html"><i class="fa fa-check"></i><b>5</b> MetOrigin Analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#loading-packages-6"><i class="fa fa-check"></i><b>5.1</b> Loading packages</a></li>
<li class="chapter" data-level="5.2" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#importing-data-7"><i class="fa fa-check"></i><b>5.2</b> Importing data</a></li>
<li class="chapter" data-level="5.3" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#differential-analysis-1"><i class="fa fa-check"></i><b>5.3</b> Differential Analysis</a></li>
<li class="chapter" data-level="5.4" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#de-metabolites"><i class="fa fa-check"></i><b>5.4</b> DE metabolites</a></li>
<li class="chapter" data-level="5.5" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#obtain-inputs-for-metorigin"><i class="fa fa-check"></i><b>5.5</b> Obtain inputs for Metorigin</a></li>
<li class="chapter" data-level="5.6" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#save-result"><i class="fa fa-check"></i><b>5.6</b> Save result</a></li>
<li class="chapter" data-level="5.7" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#metorigin-user-tutorial"><i class="fa fa-check"></i><b>5.7</b> MetOrigin User Tutorial</a></li>
<li class="chapter" data-level="5.8" data-path="metorigin-analysis.html"><a href="metorigin-analysis.html#session-info-1"><i class="fa fa-check"></i><b>5.8</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="example.html"><a href="example.html"><i class="fa fa-check"></i><b>6</b> Example</a>
<ul>
<li class="chapter" data-level="6.1" data-path="example.html"><a href="example.html#loading-packages-7"><i class="fa fa-check"></i><b>6.1</b> Loading packages</a></li>
<li class="chapter" data-level="6.2" data-path="example.html"><a href="example.html#importing-data-8"><i class="fa fa-check"></i><b>6.2</b> Importing data</a></li>
<li class="chapter" data-level="6.3" data-path="example.html"><a href="example.html#data-processing-1"><i class="fa fa-check"></i><b>6.3</b> Data Processing</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="example.html"><a href="example.html#data-checking-2"><i class="fa fa-check"></i><b>6.3.1</b> Data Checking</a></li>
<li class="chapter" data-level="6.3.2" data-path="example.html"><a href="example.html#missing-value-imputation-2"><i class="fa fa-check"></i><b>6.3.2</b> Missing value imputation</a></li>
<li class="chapter" data-level="6.3.3" data-path="example.html"><a href="example.html#data-filtering-2"><i class="fa fa-check"></i><b>6.3.3</b> Data Filtering</a></li>
<li class="chapter" data-level="6.3.4" data-path="example.html"><a href="example.html#data-normalization-2"><i class="fa fa-check"></i><b>6.3.4</b> Data Normalization</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="example.html"><a href="example.html#cluster-analysis-1"><i class="fa fa-check"></i><b>6.4</b> Cluster Analysis</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="example.html"><a href="example.html#hierarchical-clustering-1"><i class="fa fa-check"></i><b>6.4.1</b> Hierarchical Clustering</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="example.html"><a href="example.html#chemometrics-analysis-1"><i class="fa fa-check"></i><b>6.5</b> Chemometrics Analysis</a>
<ul>
<li class="chapter" data-level="6.5.1" data-path="example.html"><a href="example.html#partial-least-squares-discriminant-analysis-pls-da-1"><i class="fa fa-check"></i><b>6.5.1</b> Partial Least Squares-Discriminant Analysis (PLS-DA)</a></li>
<li class="chapter" data-level="6.5.2" data-path="example.html"><a href="example.html#sparse-partial-least-squares-discriminant-analysis-spls-da-1"><i class="fa fa-check"></i><b>6.5.2</b> Sparse Partial Least Squares-Discriminant Analysis (sPLS-DA)</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="example.html"><a href="example.html#univariate-analysis-1"><i class="fa fa-check"></i><b>6.6</b> Univariate Analysis</a>
<ul>
<li class="chapter" data-level="6.6.1" data-path="example.html"><a href="example.html#fold-change-analysis-1"><i class="fa fa-check"></i><b>6.6.1</b> Fold Change Analysis</a></li>
<li class="chapter" data-level="6.6.2" data-path="example.html"><a href="example.html#vip-variable-influence-on-projection-coefficient-1"><i class="fa fa-check"></i><b>6.6.2</b> VIP (Variable influence on projection &amp; coefficient)</a></li>
<li class="chapter" data-level="6.6.3" data-path="example.html"><a href="example.html#t-test-by-local-codes-1"><i class="fa fa-check"></i><b>6.6.3</b> T-test by local codes</a></li>
<li class="chapter" data-level="6.6.4" data-path="example.html"><a href="example.html#merging-result-1"><i class="fa fa-check"></i><b>6.6.4</b> Merging result</a></li>
<li class="chapter" data-level="6.6.5" data-path="example.html"><a href="example.html#volcano-of-merged-results-1"><i class="fa fa-check"></i><b>6.6.5</b> Volcano of Merged Results</a></li>
<li class="chapter" data-level="6.6.6" data-path="example.html"><a href="example.html#t-test-1"><i class="fa fa-check"></i><b>6.6.6</b> T Test</a></li>
<li class="chapter" data-level="6.6.7" data-path="example.html"><a href="example.html#volcano-plot-1"><i class="fa fa-check"></i><b>6.6.7</b> Volcano plot</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="example.html"><a href="example.html#feature-selection"><i class="fa fa-check"></i><b>6.7</b> Feature Selection</a>
<ul>
<li class="chapter" data-level="6.7.1" data-path="example.html"><a href="example.html#regularized-generalized-linear-models-lasso-alpha-1"><i class="fa fa-check"></i><b>6.7.1</b> Regularized Generalized Linear Models (Lasso: alpha = 1)</a></li>
<li class="chapter" data-level="6.7.2" data-path="example.html"><a href="example.html#classification-random-forest"><i class="fa fa-check"></i><b>6.7.2</b> Classification (Random Forest)</a></li>
</ul></li>
<li class="chapter" data-level="6.8" data-path="example.html"><a href="example.html#network-analysis-2"><i class="fa fa-check"></i><b>6.8</b> Network Analysis</a>
<ul>
<li class="chapter" data-level="6.8.1" data-path="example.html"><a href="example.html#data-curation-2"><i class="fa fa-check"></i><b>6.8.1</b> Data curation</a></li>
<li class="chapter" data-level="6.8.2" data-path="example.html"><a href="example.html#building-network-model-4"><i class="fa fa-check"></i><b>6.8.2</b> Building network model</a></li>
<li class="chapter" data-level="6.8.3" data-path="example.html"><a href="example.html#visualizing-the-network-4"><i class="fa fa-check"></i><b>6.8.3</b> Visualizing the network</a></li>
</ul></li>
<li class="chapter" data-level="6.9" data-path="example.html"><a href="example.html#network-analysis-by-wgcna"><i class="fa fa-check"></i><b>6.9</b> Network Analysis by WGCNA</a>
<ul>
<li class="chapter" data-level="6.9.1" data-path="example.html"><a href="example.html#data-curation-3"><i class="fa fa-check"></i><b>6.9.1</b> Data curation</a></li>
<li class="chapter" data-level="6.9.2" data-path="example.html"><a href="example.html#tuning-soft-thresholds"><i class="fa fa-check"></i><b>6.9.2</b> Tuning soft thresholds</a></li>
<li class="chapter" data-level="6.9.3" data-path="example.html"><a href="example.html#create-the-network-using-the-blockwisemodules"><i class="fa fa-check"></i><b>6.9.3</b> Create the network using the <code>blockwiseModules</code></a></li>
<li class="chapter" data-level="6.9.4" data-path="example.html"><a href="example.html#plot-modules"><i class="fa fa-check"></i><b>6.9.4</b> plot modules</a></li>
<li class="chapter" data-level="6.9.5" data-path="example.html"><a href="example.html#relationships-among-modules"><i class="fa fa-check"></i><b>6.9.5</b> Relationships among modules</a></li>
<li class="chapter" data-level="6.9.6" data-path="example.html"><a href="example.html#module-eigengene-correlation"><i class="fa fa-check"></i><b>6.9.6</b> Module (Eigengene) correlation</a></li>
<li class="chapter" data-level="6.9.7" data-path="example.html"><a href="example.html#relate-module-cluster-assignments-to-groups"><i class="fa fa-check"></i><b>6.9.7</b> Relate Module (cluster) Assignments to Groups</a></li>
<li class="chapter" data-level="6.9.8" data-path="example.html"><a href="example.html#generate-and-export-networks"><i class="fa fa-check"></i><b>6.9.8</b> Generate and Export Networks</a></li>
<li class="chapter" data-level="6.9.9" data-path="example.html"><a href="example.html#network-visualization"><i class="fa fa-check"></i><b>6.9.9</b> Network visualization</a></li>
</ul></li>
<li class="chapter" data-level="6.10" data-path="example.html"><a href="example.html#systematic-information-2"><i class="fa fa-check"></i><b>6.10</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data Analysis in Metabolomics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="statistical-analysis-1" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> Statistical Analysis<a href="statistical-analysis-1.html#statistical-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Although the horrible experience of data analysis by using <strong>MetaboAnalystR</strong> R package <span class="citation">(<a href="#ref-pang2020metaboanalystr">Pang et al. 2020</a>)</span>, its thought of data processing are very useful. Therefore, this template is based on the workflow from <strong>MetaboAnalystR</strong>.</p>
<p><strong>In this chapter, very detailed explaination of the available methods in each step in Statistical Analysis would be introduced. Users can nevertheless go through the whole analysis from with example data in Chapter 4.</strong></p>
<hr />
<div id="data-preprocessing" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Data Preprocessing<a href="statistical-analysis-1.html#data-preprocessing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We integrated R packages and our own scripts to build the data analysis template for metabolomics data. Particularly, we thanks very much for <strong>POMA</strong> R package <span class="citation">(<a href="#ref-castellano2021pomashiny">Castellano-Escuder et al. 2021</a>)</span>. <strong>POMA</strong> is a flexible data cleaning and statistical analysis processes in one comprehensible and user-friendly R package.</p>
<p><strong>Note: Please remember to preprocess your data before <a href="statistical-analysis-1.html#cluster-analysis">Cluster Analysis</a> and other steps below.</strong></p>
<div id="environment-setup" class="section level3 hasAnchor" number="3.1.1">
<h3><span class="header-section-number">3.1.1</span> Environment setup<a href="statistical-analysis-1.html#environment-setup" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="statistical-analysis-1.html#cb45-1" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">warning =</span> F)</span>
<span id="cb45-2"><a href="statistical-analysis-1.html#cb45-2" tabindex="-1"></a></span>
<span id="cb45-3"><a href="statistical-analysis-1.html#cb45-3" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb45-4"><a href="statistical-analysis-1.html#cb45-4" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb45-5"><a href="statistical-analysis-1.html#cb45-5" tabindex="-1"></a><span class="fu">library</span>(Biobase)</span>
<span id="cb45-6"><a href="statistical-analysis-1.html#cb45-6" tabindex="-1"></a><span class="fu">library</span>(POMA)</span>
<span id="cb45-7"><a href="statistical-analysis-1.html#cb45-7" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb45-8"><a href="statistical-analysis-1.html#cb45-8" tabindex="-1"></a><span class="fu">library</span>(ggraph)</span>
<span id="cb45-9"><a href="statistical-analysis-1.html#cb45-9" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb45-10"><a href="statistical-analysis-1.html#cb45-10" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb45-11"><a href="statistical-analysis-1.html#cb45-11" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb45-12"><a href="statistical-analysis-1.html#cb45-12" tabindex="-1"></a><span class="fu">library</span>(ropls)</span>
<span id="cb45-13"><a href="statistical-analysis-1.html#cb45-13" tabindex="-1"></a><span class="fu">library</span>(XMAS2)</span>
<span id="cb45-14"><a href="statistical-analysis-1.html#cb45-14" tabindex="-1"></a></span>
<span id="cb45-15"><a href="statistical-analysis-1.html#cb45-15" tabindex="-1"></a><span class="co"># rm(list = ls())</span></span>
<span id="cb45-16"><a href="statistical-analysis-1.html#cb45-16" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb45-17"><a href="statistical-analysis-1.html#cb45-17" tabindex="-1"></a><span class="fu">options</span>(<span class="at">future.globals.maxSize =</span> <span class="dv">1000</span> <span class="sc">*</span> <span class="dv">1024</span><span class="sc">^</span><span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="loading-data" class="section level3 hasAnchor" number="3.1.2">
<h3><span class="header-section-number">3.1.2</span> Loading data<a href="statistical-analysis-1.html#loading-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The dataset is from the Zeybel-2022 published paper <span class="citation">(<a href="#ref-zeybel2022multiomics">Zeybel et al. 2022</a>)</span>.</p>
<ul>
<li>features table</li>
</ul>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="statistical-analysis-1.html#cb46-1" tabindex="-1"></a>profile <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(<span class="st">&quot;./dataset/OmicsDataSet-Zeybel-2022.xlsx&quot;</span>, <span class="at">sheet =</span> <span class="dv">6</span>)</span>
<span id="cb46-2"><a href="statistical-analysis-1.html#cb46-2" tabindex="-1"></a><span class="fu">head</span>(profile)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 67
##   BIOCHEMICAL `SUPER PATHWAY` `SUB PATHWAY` `COMP ID` PLATFORM `CHEMICAL ID`    RI  MASS PUBCHEM CAS   KEGG  `SampleID HMDBID` P101001 P101012 P101030
##   &lt;chr&gt;       &lt;chr&gt;           &lt;chr&gt;             &lt;dbl&gt; &lt;chr&gt;            &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;               &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;
## 1 (14 or 15)… Lipid           Fatty Acid, …     38768 LC/MS N…     100002945  5695  269. 8181;1… &lt;NA&gt;  C169… HMDB0061859        5.11e7  5.12e7  3.84e7
## 2 (16 or 17)… Lipid           Fatty Acid, …     38296 LC/MS N…     100002356  5993  297. 3083779 2724… &lt;NA&gt;  HMDB0037397        5.11e6  6.00e6  2.86e6
## 3 (2 or 3)-d… Lipid           Medium Chain…     63436 LC/MS N…     100021502  4990  169. &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;  &lt;NA&gt;               7.57e5  5.98e5  3.67e5
## 4 (2,4 or 2,… Xenobiotics     Food Compone…     62533 LC/MS N…     100020519  3474  201. &lt;NA&gt;    &lt;NA&gt;  &lt;NA&gt;  &lt;NA&gt;              NA      NA       5.64e5
## 5 (N(1) + N(… Amino Acid      Polyamine Me…     57814 LC/MS P…     100016038  3080  188. 123689… &lt;NA&gt;  C006… HMDB0001276,HMDB…  2.82e5  2.49e5  2.31e5
## 6 (R)-3-hydr… Lipid           Fatty Acid M…     43264 LC/MS P…     100003926  2400  248. 534816… &lt;NA&gt;  &lt;NA&gt;  HMDB0013127       NA      NA      NA     
## # ℹ 52 more variables: P101031 &lt;dbl&gt;, P101050 &lt;dbl&gt;, P101059 &lt;dbl&gt;, P101071 &lt;dbl&gt;, P101072 &lt;dbl&gt;, P101084 &lt;dbl&gt;, P101003 &lt;dbl&gt;, P101004 &lt;dbl&gt;,
## #   P101013 &lt;dbl&gt;, P101016 &lt;dbl&gt;, P101017 &lt;dbl&gt;, P101038 &lt;dbl&gt;, P101051 &lt;dbl&gt;, P101061 &lt;dbl&gt;, P101062 &lt;dbl&gt;, P101074 &lt;dbl&gt;, P101075 &lt;dbl&gt;,
## #   P101076 &lt;dbl&gt;, P101085 &lt;dbl&gt;, P101088 &lt;dbl&gt;, P101007 &lt;dbl&gt;, P101018 &lt;dbl&gt;, P101019 &lt;dbl&gt;, P101041 &lt;dbl&gt;, P101052 &lt;dbl&gt;, P101064 &lt;dbl&gt;,
## #   P101065 &lt;dbl&gt;, P101077 &lt;dbl&gt;, P101090 &lt;dbl&gt;, P101094 &lt;dbl&gt;, P101009 &lt;dbl&gt;, P101010 &lt;dbl&gt;, P101021 &lt;dbl&gt;, P101022 &lt;dbl&gt;, P101042 &lt;dbl&gt;,
## #   P101054 &lt;dbl&gt;, P101056 &lt;dbl&gt;, P101067 &lt;dbl&gt;, P101068 &lt;dbl&gt;, P101079 &lt;dbl&gt;, P101095 &lt;dbl&gt;, P101096 &lt;dbl&gt;, P101011 &lt;dbl&gt;, P101024 &lt;dbl&gt;,
## #   P101025 &lt;dbl&gt;, P101027 &lt;dbl&gt;, P101047 &lt;dbl&gt;, P101057 &lt;dbl&gt;, P101069 &lt;dbl&gt;, P101080 &lt;dbl&gt;, P101081 &lt;dbl&gt;, P101082 &lt;dbl&gt;</code></pre>
<ul>
<li>metadata table</li>
</ul>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="statistical-analysis-1.html#cb48-1" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> readxl<span class="sc">::</span><span class="fu">read_xlsx</span>(<span class="st">&quot;./dataset/OmicsDataSet-Zeybel-2022.xlsx&quot;</span>, <span class="at">sheet =</span> <span class="dv">2</span>)</span>
<span id="cb48-2"><a href="statistical-analysis-1.html#cb48-2" tabindex="-1"></a><span class="fu">head</span>(metadata)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 11
##   PatientID Stage  Metabolomics Proteomics GutMetagenomics OralMetagenomics LiverFatClass Gender AlcoholConsumption Smoker   Age
##   &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt;           &lt;chr&gt;            &lt;chr&gt;         &lt;chr&gt;  &lt;chr&gt;              &lt;chr&gt;  &lt;dbl&gt;
## 1 P101001   Before Send         Send       Send            Send             Severe        Male   No                 No        52
## 2 P101003   Before Send         Send       Send            Send             None          Female No                 No        31
## 3 P101004   Before Send         Send       Send            Send             Moderate      Male   Yes                No        43
## 4 P101007   Before Send         Send       Send            Send             Severe        Female No                 No        61
## 5 P101009   Before Send         Send       Send            Send             Moderate      Male   No                 Yes       51
## 6 P101010   Before Send         Send       Send            Send             Mild          Male   Yes                No        27</code></pre>
</div>
<div id="object-preparation" class="section level3 hasAnchor" number="3.1.3">
<h3><span class="header-section-number">3.1.3</span> Object Preparation<a href="statistical-analysis-1.html#object-preparation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Data Preparation: ExpressionSet object</li>
</ul>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="statistical-analysis-1.html#cb50-1" tabindex="-1"></a>get_ExpressionSet <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb50-2"><a href="statistical-analysis-1.html#cb50-2" tabindex="-1"></a>    x,</span>
<span id="cb50-3"><a href="statistical-analysis-1.html#cb50-3" tabindex="-1"></a>    y) {</span>
<span id="cb50-4"><a href="statistical-analysis-1.html#cb50-4" tabindex="-1"></a>  </span>
<span id="cb50-5"><a href="statistical-analysis-1.html#cb50-5" tabindex="-1"></a>  <span class="co"># x = metadata</span></span>
<span id="cb50-6"><a href="statistical-analysis-1.html#cb50-6" tabindex="-1"></a>  <span class="co"># y = profile</span></span>
<span id="cb50-7"><a href="statistical-analysis-1.html#cb50-7" tabindex="-1"></a> </span>
<span id="cb50-8"><a href="statistical-analysis-1.html#cb50-8" tabindex="-1"></a>  phen <span class="ot">&lt;-</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb50-9"><a href="statistical-analysis-1.html#cb50-9" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(Metabolomics <span class="sc">==</span> <span class="st">&quot;Send&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-10"><a href="statistical-analysis-1.html#cb50-10" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(PatientID, LiverFatClass, Gender, Smoker, Age, AlcoholConsumption)</span>
<span id="cb50-11"><a href="statistical-analysis-1.html#cb50-11" tabindex="-1"></a></span>
<span id="cb50-12"><a href="statistical-analysis-1.html#cb50-12" tabindex="-1"></a>  sid <span class="ot">&lt;-</span> <span class="fu">intersect</span>(phen<span class="sc">$</span>PatientID, <span class="fu">colnames</span>(y))</span>
<span id="cb50-13"><a href="statistical-analysis-1.html#cb50-13" tabindex="-1"></a>  </span>
<span id="cb50-14"><a href="statistical-analysis-1.html#cb50-14" tabindex="-1"></a>  prof <span class="ot">&lt;-</span> y <span class="sc">%&gt;%</span> </span>
<span id="cb50-15"><a href="statistical-analysis-1.html#cb50-15" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(sid)) <span class="sc">%&gt;%</span></span>
<span id="cb50-16"><a href="statistical-analysis-1.html#cb50-16" tabindex="-1"></a>    <span class="fu">data.frame</span>()</span>
<span id="cb50-17"><a href="statistical-analysis-1.html#cb50-17" tabindex="-1"></a>  <span class="fu">rownames</span>(prof) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;M_&quot;</span>, y<span class="sc">$</span><span class="st">`</span><span class="at">COMP ID</span><span class="st">`</span>)</span>
<span id="cb50-18"><a href="statistical-analysis-1.html#cb50-18" tabindex="-1"></a>  </span>
<span id="cb50-19"><a href="statistical-analysis-1.html#cb50-19" tabindex="-1"></a>  phen <span class="ot">&lt;-</span> phen[<span class="fu">pmatch</span>(sid, phen<span class="sc">$</span>PatientID), , F] <span class="sc">%&gt;%</span></span>
<span id="cb50-20"><a href="statistical-analysis-1.html#cb50-20" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">&quot;PatientID&quot;</span>)</span>
<span id="cb50-21"><a href="statistical-analysis-1.html#cb50-21" tabindex="-1"></a>  </span>
<span id="cb50-22"><a href="statistical-analysis-1.html#cb50-22" tabindex="-1"></a>  feat <span class="ot">&lt;-</span> y <span class="sc">%&gt;%</span> </span>
<span id="cb50-23"><a href="statistical-analysis-1.html#cb50-23" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">12</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-24"><a href="statistical-analysis-1.html#cb50-24" tabindex="-1"></a>    <span class="fu">as.data.frame</span>()</span>
<span id="cb50-25"><a href="statistical-analysis-1.html#cb50-25" tabindex="-1"></a>  <span class="fu">rownames</span>(feat) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;M_&quot;</span>, y<span class="sc">$</span><span class="st">`</span><span class="at">COMP ID</span><span class="st">`</span>)</span>
<span id="cb50-26"><a href="statistical-analysis-1.html#cb50-26" tabindex="-1"></a>  </span>
<span id="cb50-27"><a href="statistical-analysis-1.html#cb50-27" tabindex="-1"></a>  <span class="co"># expressionSet</span></span>
<span id="cb50-28"><a href="statistical-analysis-1.html#cb50-28" tabindex="-1"></a>  phen_ADF <span class="ot">&lt;-</span>  <span class="fu">new</span>(<span class="st">&quot;AnnotatedDataFrame&quot;</span>, <span class="at">data=</span>phen)</span>
<span id="cb50-29"><a href="statistical-analysis-1.html#cb50-29" tabindex="-1"></a>  feature_ADF <span class="ot">&lt;-</span>  <span class="fu">new</span>(<span class="st">&quot;AnnotatedDataFrame&quot;</span>, <span class="at">data=</span>feat)</span>
<span id="cb50-30"><a href="statistical-analysis-1.html#cb50-30" tabindex="-1"></a>  experimentData <span class="ot">&lt;-</span> <span class="fu">new</span>(</span>
<span id="cb50-31"><a href="statistical-analysis-1.html#cb50-31" tabindex="-1"></a>    <span class="st">&quot;MIAME&quot;</span>,</span>
<span id="cb50-32"><a href="statistical-analysis-1.html#cb50-32" tabindex="-1"></a>    <span class="at">name=</span><span class="st">&quot;Hua&quot;</span>, </span>
<span id="cb50-33"><a href="statistical-analysis-1.html#cb50-33" tabindex="-1"></a>    <span class="at">lab=</span><span class="st">&quot;Xbiome Company&quot;</span>,</span>
<span id="cb50-34"><a href="statistical-analysis-1.html#cb50-34" tabindex="-1"></a>    <span class="at">contact=</span><span class="st">&quot;Hua@xbiome.com&quot;</span>,</span>
<span id="cb50-35"><a href="statistical-analysis-1.html#cb50-35" tabindex="-1"></a>    <span class="at">title=</span><span class="st">&quot;Metabolomics&quot;</span>,</span>
<span id="cb50-36"><a href="statistical-analysis-1.html#cb50-36" tabindex="-1"></a>    <span class="at">abstract=</span><span class="st">&quot;The Mass Spectrometry ExpressionSet without imputation value&quot;</span>,</span>
<span id="cb50-37"><a href="statistical-analysis-1.html#cb50-37" tabindex="-1"></a>    <span class="at">url=</span><span class="st">&quot;www.xbiome.cn&quot;</span>,</span>
<span id="cb50-38"><a href="statistical-analysis-1.html#cb50-38" tabindex="-1"></a>    <span class="at">other=</span><span class="fu">list</span>(<span class="at">notes=</span><span class="st">&quot;Metabolomics&quot;</span>))</span>
<span id="cb50-39"><a href="statistical-analysis-1.html#cb50-39" tabindex="-1"></a>  </span>
<span id="cb50-40"><a href="statistical-analysis-1.html#cb50-40" tabindex="-1"></a>  expressionSet <span class="ot">&lt;-</span> <span class="fu">new</span>(</span>
<span id="cb50-41"><a href="statistical-analysis-1.html#cb50-41" tabindex="-1"></a>    <span class="st">&quot;ExpressionSet&quot;</span>,</span>
<span id="cb50-42"><a href="statistical-analysis-1.html#cb50-42" tabindex="-1"></a>    <span class="at">exprs=</span>prof,</span>
<span id="cb50-43"><a href="statistical-analysis-1.html#cb50-43" tabindex="-1"></a>    <span class="at">phenoData=</span>phen_ADF,</span>
<span id="cb50-44"><a href="statistical-analysis-1.html#cb50-44" tabindex="-1"></a>    <span class="at">featureData=</span>feature_ADF,</span>
<span id="cb50-45"><a href="statistical-analysis-1.html#cb50-45" tabindex="-1"></a>    <span class="at">experimentData=</span>experimentData)</span>
<span id="cb50-46"><a href="statistical-analysis-1.html#cb50-46" tabindex="-1"></a>  </span>
<span id="cb50-47"><a href="statistical-analysis-1.html#cb50-47" tabindex="-1"></a>  <span class="fu">return</span>(expressionSet)</span>
<span id="cb50-48"><a href="statistical-analysis-1.html#cb50-48" tabindex="-1"></a>}</span>
<span id="cb50-49"><a href="statistical-analysis-1.html#cb50-49" tabindex="-1"></a></span>
<span id="cb50-50"><a href="statistical-analysis-1.html#cb50-50" tabindex="-1"></a></span>
<span id="cb50-51"><a href="statistical-analysis-1.html#cb50-51" tabindex="-1"></a>ExprSet <span class="ot">&lt;-</span> <span class="fu">get_ExpressionSet</span>(<span class="at">x =</span> metadata, <span class="at">y =</span> profile)</span>
<span id="cb50-52"><a href="statistical-analysis-1.html#cb50-52" tabindex="-1"></a></span>
<span id="cb50-53"><a href="statistical-analysis-1.html#cb50-53" tabindex="-1"></a>ExprSet</span></code></pre></div>
<pre><code>## ExpressionSet (storageMode: lockedEnvironment)
## assayData: 1032 features, 55 samples 
##   element names: exprs 
## protocolData: none
## phenoData
##   sampleNames: P101001 P101003 ... P101096 (55 total)
##   varLabels: LiverFatClass Gender ... AlcoholConsumption (5 total)
##   varMetadata: labelDescription
## featureData
##   featureNames: M_38768 M_38296 ... M_15581 (1032 total)
##   fvarLabels: BIOCHEMICAL SUPER PATHWAY ... SampleID HMDBID (12 total)
##   fvarMetadata: labelDescription
## experimentData: use &#39;experimentData(object)&#39;
## Annotation:</code></pre>
<ul>
<li>Data Preparation: SummarizedExperiment object</li>
</ul>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="statistical-analysis-1.html#cb52-1" tabindex="-1"></a>getSEobject <span class="ot">&lt;-</span> <span class="cf">function</span>(x, y) {</span>
<span id="cb52-2"><a href="statistical-analysis-1.html#cb52-2" tabindex="-1"></a></span>
<span id="cb52-3"><a href="statistical-analysis-1.html#cb52-3" tabindex="-1"></a>  target <span class="ot">&lt;-</span> x <span class="sc">%&gt;%</span> </span>
<span id="cb52-4"><a href="statistical-analysis-1.html#cb52-4" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(Metabolomics <span class="sc">==</span> <span class="st">&quot;Send&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb52-5"><a href="statistical-analysis-1.html#cb52-5" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(PatientID, LiverFatClass, Gender, Smoker, Age, AlcoholConsumption)</span>
<span id="cb52-6"><a href="statistical-analysis-1.html#cb52-6" tabindex="-1"></a></span>
<span id="cb52-7"><a href="statistical-analysis-1.html#cb52-7" tabindex="-1"></a>  sid <span class="ot">&lt;-</span> <span class="fu">intersect</span>(target<span class="sc">$</span>PatientID, <span class="fu">colnames</span>(profile))</span>
<span id="cb52-8"><a href="statistical-analysis-1.html#cb52-8" tabindex="-1"></a>  </span>
<span id="cb52-9"><a href="statistical-analysis-1.html#cb52-9" tabindex="-1"></a>  features <span class="ot">&lt;-</span> y <span class="sc">%&gt;%</span> </span>
<span id="cb52-10"><a href="statistical-analysis-1.html#cb52-10" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(sid)) <span class="sc">%&gt;%</span></span>
<span id="cb52-11"><a href="statistical-analysis-1.html#cb52-11" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">t</span>()</span>
<span id="cb52-12"><a href="statistical-analysis-1.html#cb52-12" tabindex="-1"></a>  <span class="fu">colnames</span>(features) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;M_&quot;</span>, profile<span class="sc">$</span><span class="st">`</span><span class="at">COMP ID</span><span class="st">`</span>)</span>
<span id="cb52-13"><a href="statistical-analysis-1.html#cb52-13" tabindex="-1"></a>  </span>
<span id="cb52-14"><a href="statistical-analysis-1.html#cb52-14" tabindex="-1"></a>  target <span class="ot">&lt;-</span> target[<span class="fu">pmatch</span>(sid, target<span class="sc">$</span>PatientID), , F]</span>
<span id="cb52-15"><a href="statistical-analysis-1.html#cb52-15" tabindex="-1"></a>  </span>
<span id="cb52-16"><a href="statistical-analysis-1.html#cb52-16" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb52-17"><a href="statistical-analysis-1.html#cb52-17" tabindex="-1"></a>                                  <span class="at">features =</span> features)</span>
<span id="cb52-18"><a href="statistical-analysis-1.html#cb52-18" tabindex="-1"></a>  <span class="fu">return</span>(res)  </span>
<span id="cb52-19"><a href="statistical-analysis-1.html#cb52-19" tabindex="-1"></a>}</span>
<span id="cb52-20"><a href="statistical-analysis-1.html#cb52-20" tabindex="-1"></a></span>
<span id="cb52-21"><a href="statistical-analysis-1.html#cb52-21" tabindex="-1"></a>se_raw <span class="ot">&lt;-</span> <span class="fu">getSEobject</span>(metadata, profile)</span>
<span id="cb52-22"><a href="statistical-analysis-1.html#cb52-22" tabindex="-1"></a></span>
<span id="cb52-23"><a href="statistical-analysis-1.html#cb52-23" tabindex="-1"></a>se_raw</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 1032 55 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(1032): M_38768 M_38296 ... M_57517 M_15581
## rowData names(0):
## colnames(55): P101001 P101003 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
<ul>
<li>Extract data for test dataset</li>
</ul>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="statistical-analysis-1.html#cb54-1" tabindex="-1"></a>get_testData <span class="ot">&lt;-</span> <span class="cf">function</span>(object, <span class="at">num =</span> <span class="dv">200</span>) {</span>
<span id="cb54-2"><a href="statistical-analysis-1.html#cb54-2" tabindex="-1"></a></span>
<span id="cb54-3"><a href="statistical-analysis-1.html#cb54-3" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) <span class="sc">%&gt;%</span></span>
<span id="cb54-4"><a href="statistical-analysis-1.html#cb54-4" tabindex="-1"></a>    <span class="fu">t</span>()</span>
<span id="cb54-5"><a href="statistical-analysis-1.html#cb54-5" tabindex="-1"></a>  metadata_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object) <span class="sc">%&gt;%</span></span>
<span id="cb54-6"><a href="statistical-analysis-1.html#cb54-6" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb54-7"><a href="statistical-analysis-1.html#cb54-7" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;ID&quot;</span>)</span>
<span id="cb54-8"><a href="statistical-analysis-1.html#cb54-8" tabindex="-1"></a>  </span>
<span id="cb54-9"><a href="statistical-analysis-1.html#cb54-9" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> metadata_tab, </span>
<span id="cb54-10"><a href="statistical-analysis-1.html#cb54-10" tabindex="-1"></a>                                  <span class="at">features =</span> features_tab[, <span class="dv">1</span><span class="sc">:</span>num])</span>
<span id="cb54-11"><a href="statistical-analysis-1.html#cb54-11" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb54-12"><a href="statistical-analysis-1.html#cb54-12" tabindex="-1"></a>}</span>
<span id="cb54-13"><a href="statistical-analysis-1.html#cb54-13" tabindex="-1"></a></span>
<span id="cb54-14"><a href="statistical-analysis-1.html#cb54-14" tabindex="-1"></a>se_raw <span class="ot">&lt;-</span> <span class="fu">get_testData</span>(<span class="at">object =</span> se_raw)</span>
<span id="cb54-15"><a href="statistical-analysis-1.html#cb54-15" tabindex="-1"></a>se_raw</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 200 55 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(200): M_38768 M_38296 ... M_31787 M_63361
## rowData names(0):
## colnames(55): P101001 P101003 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="data-checking-1" class="section level3 hasAnchor" number="3.1.4">
<h3><span class="header-section-number">3.1.4</span> Data Checking<a href="statistical-analysis-1.html#data-checking-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Features in PomaSummarizedExperiment object must have the following criterion:</p>
<ul>
<li><p>All data values are numeric.</p></li>
<li><p>A total of 0 (0%) missing values were detected.</p></li>
</ul>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="statistical-analysis-1.html#cb56-1" tabindex="-1"></a>CheckData <span class="ot">&lt;-</span> <span class="cf">function</span>(object) {</span>
<span id="cb56-2"><a href="statistical-analysis-1.html#cb56-2" tabindex="-1"></a>  </span>
<span id="cb56-3"><a href="statistical-analysis-1.html#cb56-3" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object)</span>
<span id="cb56-4"><a href="statistical-analysis-1.html#cb56-4" tabindex="-1"></a>  </span>
<span id="cb56-5"><a href="statistical-analysis-1.html#cb56-5" tabindex="-1"></a>  <span class="co"># numeric &amp; missing values</span></span>
<span id="cb56-6"><a href="statistical-analysis-1.html#cb56-6" tabindex="-1"></a>  int_mat <span class="ot">&lt;-</span> features_tab</span>
<span id="cb56-7"><a href="statistical-analysis-1.html#cb56-7" tabindex="-1"></a>  rowNms <span class="ot">&lt;-</span> <span class="fu">rownames</span>(int_mat)</span>
<span id="cb56-8"><a href="statistical-analysis-1.html#cb56-8" tabindex="-1"></a>  colNms <span class="ot">&lt;-</span> <span class="fu">colnames</span>(int_mat)</span>
<span id="cb56-9"><a href="statistical-analysis-1.html#cb56-9" tabindex="-1"></a>  naNms <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">is.na</span>(int_mat))</span>
<span id="cb56-10"><a href="statistical-analysis-1.html#cb56-10" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(int_mat)) {</span>
<span id="cb56-11"><a href="statistical-analysis-1.html#cb56-11" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">class</span>(int_mat[, i]) <span class="sc">==</span> <span class="st">&quot;integer64&quot;</span>) {</span>
<span id="cb56-12"><a href="statistical-analysis-1.html#cb56-12" tabindex="-1"></a>      int_mat[, i] <span class="ot">&lt;-</span> <span class="fu">as.double</span>(int_mat[, i])</span>
<span id="cb56-13"><a href="statistical-analysis-1.html#cb56-13" tabindex="-1"></a>    }</span>
<span id="cb56-14"><a href="statistical-analysis-1.html#cb56-14" tabindex="-1"></a>  }</span>
<span id="cb56-15"><a href="statistical-analysis-1.html#cb56-15" tabindex="-1"></a>  </span>
<span id="cb56-16"><a href="statistical-analysis-1.html#cb56-16" tabindex="-1"></a>  num_mat <span class="ot">&lt;-</span> <span class="fu">apply</span>(int_mat, <span class="dv">2</span>, as.numeric)</span>
<span id="cb56-17"><a href="statistical-analysis-1.html#cb56-17" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">sum</span>(<span class="fu">is.na</span>(num_mat)) <span class="sc">&gt;</span> naNms) {</span>
<span id="cb56-18"><a href="statistical-analysis-1.html#cb56-18" tabindex="-1"></a>    num_mat <span class="ot">&lt;-</span> <span class="fu">apply</span>(int_mat, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">as.numeric</span>(<span class="fu">gsub</span>(<span class="st">&quot;,&quot;</span>,  <span class="st">&quot;&quot;</span>, x)))</span>
<span id="cb56-19"><a href="statistical-analysis-1.html#cb56-19" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">sum</span>(<span class="fu">is.na</span>(num_mat)) <span class="sc">&gt;</span> naNms) {</span>
<span id="cb56-20"><a href="statistical-analysis-1.html#cb56-20" tabindex="-1"></a>      <span class="fu">message</span>(<span class="st">&quot;&lt;font color=</span><span class="sc">\&quot;</span><span class="st">red</span><span class="sc">\&quot;</span><span class="st">&gt;Non-numeric values were found and replaced by NA.&lt;/font&gt;&quot;</span>)</span>
<span id="cb56-21"><a href="statistical-analysis-1.html#cb56-21" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb56-22"><a href="statistical-analysis-1.html#cb56-22" tabindex="-1"></a>      <span class="fu">message</span>(<span class="st">&quot;All data values are numeric.&quot;</span>)</span>
<span id="cb56-23"><a href="statistical-analysis-1.html#cb56-23" tabindex="-1"></a>    }</span>
<span id="cb56-24"><a href="statistical-analysis-1.html#cb56-24" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb56-25"><a href="statistical-analysis-1.html#cb56-25" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">&quot;All data values are numeric.&quot;</span>)</span>
<span id="cb56-26"><a href="statistical-analysis-1.html#cb56-26" tabindex="-1"></a>  }</span>
<span id="cb56-27"><a href="statistical-analysis-1.html#cb56-27" tabindex="-1"></a>  </span>
<span id="cb56-28"><a href="statistical-analysis-1.html#cb56-28" tabindex="-1"></a>  int_mat <span class="ot">&lt;-</span> num_mat</span>
<span id="cb56-29"><a href="statistical-analysis-1.html#cb56-29" tabindex="-1"></a>  <span class="fu">rownames</span>(int_mat) <span class="ot">&lt;-</span> rowNms</span>
<span id="cb56-30"><a href="statistical-analysis-1.html#cb56-30" tabindex="-1"></a>  <span class="fu">colnames</span>(int_mat) <span class="ot">&lt;-</span> colNms</span>
<span id="cb56-31"><a href="statistical-analysis-1.html#cb56-31" tabindex="-1"></a>  varCol <span class="ot">&lt;-</span> <span class="fu">apply</span>(int_mat, <span class="dv">2</span>, var, <span class="at">na.rm =</span> T)</span>
<span id="cb56-32"><a href="statistical-analysis-1.html#cb56-32" tabindex="-1"></a>  constCol <span class="ot">&lt;-</span> (varCol <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> <span class="fu">is.na</span>(varCol))</span>
<span id="cb56-33"><a href="statistical-analysis-1.html#cb56-33" tabindex="-1"></a>  constNum <span class="ot">&lt;-</span> <span class="fu">sum</span>(constCol, <span class="at">na.rm =</span> T)</span>
<span id="cb56-34"><a href="statistical-analysis-1.html#cb56-34" tabindex="-1"></a>  <span class="cf">if</span> (constNum <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb56-35"><a href="statistical-analysis-1.html#cb56-35" tabindex="-1"></a>    <span class="fu">message</span>(<span class="fu">paste</span>(<span class="st">&quot;&lt;font color=</span><span class="sc">\&quot;</span><span class="st">red</span><span class="sc">\&quot;</span><span class="st">&gt;&quot;</span>, constNum, </span>
<span id="cb56-36"><a href="statistical-analysis-1.html#cb56-36" tabindex="-1"></a>      <span class="st">&quot;features with a constant or single value across samples were found and deleted.&lt;/font&gt;&quot;</span>))</span>
<span id="cb56-37"><a href="statistical-analysis-1.html#cb56-37" tabindex="-1"></a>    int_mat <span class="ot">&lt;-</span> int_mat[, <span class="sc">!</span>constCol, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb56-38"><a href="statistical-analysis-1.html#cb56-38" tabindex="-1"></a>  }</span>
<span id="cb56-39"><a href="statistical-analysis-1.html#cb56-39" tabindex="-1"></a>  </span>
<span id="cb56-40"><a href="statistical-analysis-1.html#cb56-40" tabindex="-1"></a>  totalCount <span class="ot">&lt;-</span> <span class="fu">nrow</span>(int_mat) <span class="sc">*</span> <span class="fu">ncol</span>(int_mat)</span>
<span id="cb56-41"><a href="statistical-analysis-1.html#cb56-41" tabindex="-1"></a>  naCount <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">is.na</span>(int_mat))</span>
<span id="cb56-42"><a href="statistical-analysis-1.html#cb56-42" tabindex="-1"></a>  naPercent <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> naCount<span class="sc">/</span>totalCount, <span class="dv">1</span>)</span>
<span id="cb56-43"><a href="statistical-analysis-1.html#cb56-43" tabindex="-1"></a></span>
<span id="cb56-44"><a href="statistical-analysis-1.html#cb56-44" tabindex="-1"></a>  <span class="fu">message</span>(<span class="fu">paste</span>(<span class="st">&quot;A total of &quot;</span>, naCount, <span class="st">&quot; (&quot;</span>, naPercent, </span>
<span id="cb56-45"><a href="statistical-analysis-1.html#cb56-45" tabindex="-1"></a>    <span class="st">&quot;%) missing values were detected.&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))  </span>
<span id="cb56-46"><a href="statistical-analysis-1.html#cb56-46" tabindex="-1"></a>  </span>
<span id="cb56-47"><a href="statistical-analysis-1.html#cb56-47" tabindex="-1"></a>  <span class="co"># save int_mat into se object </span></span>
<span id="cb56-48"><a href="statistical-analysis-1.html#cb56-48" tabindex="-1"></a>  target <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object) <span class="sc">%&gt;%</span></span>
<span id="cb56-49"><a href="statistical-analysis-1.html#cb56-49" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb56-50"><a href="statistical-analysis-1.html#cb56-50" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb56-51"><a href="statistical-analysis-1.html#cb56-51" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb56-52"><a href="statistical-analysis-1.html#cb56-52" tabindex="-1"></a>                                  <span class="at">features =</span> <span class="fu">t</span>(int_mat))</span>
<span id="cb56-53"><a href="statistical-analysis-1.html#cb56-53" tabindex="-1"></a>  </span>
<span id="cb56-54"><a href="statistical-analysis-1.html#cb56-54" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb56-55"><a href="statistical-analysis-1.html#cb56-55" tabindex="-1"></a>}</span>
<span id="cb56-56"><a href="statistical-analysis-1.html#cb56-56" tabindex="-1"></a></span>
<span id="cb56-57"><a href="statistical-analysis-1.html#cb56-57" tabindex="-1"></a>se_check <span class="ot">&lt;-</span> <span class="fu">CheckData</span>(<span class="at">object =</span> se_raw)</span></code></pre></div>
<pre><code>## All data values are numeric.</code></pre>
<pre><code>## A total of 1146 (10.4%) missing values were detected.</code></pre>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="statistical-analysis-1.html#cb59-1" tabindex="-1"></a>se_check</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 200 55 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(200): M_38768 M_38296 ... M_31787 M_63361
## rowData names(0):
## colnames(55): P101001 P101003 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="missing-value-imputation-1" class="section level3 hasAnchor" number="3.1.5">
<h3><span class="header-section-number">3.1.5</span> Missing value imputation<a href="statistical-analysis-1.html#missing-value-imputation-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>“none”: all missing values will be replaced by zero.</p>
<p>“LOD”: specific Limit Of Detection which provides by user.</p>
<p>“half_min”: half minimal values across samples except zero.</p>
<p>“median”: median values across samples except zero.</p>
<p>“mean”: mean values across samples except zero.</p>
<p>“min”: minimal values across samples except zero.</p>
<p>“knn”: k-nearest neighbors samples.</p>
<p>“rf”: nonparametric missing value imputation using Random Forest.</p>
<p>“QRILC”: missing values imputation based quantile regression. (default: “none”).</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="statistical-analysis-1.html#cb61-1" tabindex="-1"></a>impute_abundance <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb61-2"><a href="statistical-analysis-1.html#cb61-2" tabindex="-1"></a>    object,</span>
<span id="cb61-3"><a href="statistical-analysis-1.html#cb61-3" tabindex="-1"></a>    group,</span>
<span id="cb61-4"><a href="statistical-analysis-1.html#cb61-4" tabindex="-1"></a>    <span class="at">ZerosAsNA =</span> <span class="cn">FALSE</span>,</span>
<span id="cb61-5"><a href="statistical-analysis-1.html#cb61-5" tabindex="-1"></a>    <span class="at">RemoveNA =</span> <span class="cn">TRUE</span>,</span>
<span id="cb61-6"><a href="statistical-analysis-1.html#cb61-6" tabindex="-1"></a>    <span class="at">prevalence =</span> <span class="fl">0.5</span>,</span>
<span id="cb61-7"><a href="statistical-analysis-1.html#cb61-7" tabindex="-1"></a>    <span class="at">method =</span> <span class="fu">c</span>(<span class="st">&quot;none&quot;</span>, <span class="st">&quot;LOD&quot;</span>, <span class="st">&quot;half_min&quot;</span>, <span class="st">&quot;median&quot;</span>,</span>
<span id="cb61-8"><a href="statistical-analysis-1.html#cb61-8" tabindex="-1"></a>               <span class="st">&quot;mean&quot;</span>, <span class="st">&quot;min&quot;</span>, <span class="st">&quot;knn&quot;</span>, <span class="st">&quot;rf&quot;</span>, <span class="st">&quot;QRILC&quot;</span>),</span>
<span id="cb61-9"><a href="statistical-analysis-1.html#cb61-9" tabindex="-1"></a>    <span class="at">LOD =</span> <span class="cn">NULL</span>) {</span>
<span id="cb61-10"><a href="statistical-analysis-1.html#cb61-10" tabindex="-1"></a></span>
<span id="cb61-11"><a href="statistical-analysis-1.html#cb61-11" tabindex="-1"></a>  <span class="co"># object = se_check</span></span>
<span id="cb61-12"><a href="statistical-analysis-1.html#cb61-12" tabindex="-1"></a>  <span class="co"># group = &quot;group&quot;</span></span>
<span id="cb61-13"><a href="statistical-analysis-1.html#cb61-13" tabindex="-1"></a>  <span class="co"># ZerosAsNA = TRUE</span></span>
<span id="cb61-14"><a href="statistical-analysis-1.html#cb61-14" tabindex="-1"></a>  <span class="co"># RemoveNA = TRUE</span></span>
<span id="cb61-15"><a href="statistical-analysis-1.html#cb61-15" tabindex="-1"></a>  <span class="co"># prevalence = 0.5</span></span>
<span id="cb61-16"><a href="statistical-analysis-1.html#cb61-16" tabindex="-1"></a>  <span class="co"># method = &quot;knn&quot;</span></span>
<span id="cb61-17"><a href="statistical-analysis-1.html#cb61-17" tabindex="-1"></a></span>
<span id="cb61-18"><a href="statistical-analysis-1.html#cb61-18" tabindex="-1"></a>  <span class="cf">if</span> (base<span class="sc">::</span><span class="fu">missing</span>(object)) {</span>
<span id="cb61-19"><a href="statistical-analysis-1.html#cb61-19" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;object argument is empty!&quot;</span>)</span>
<span id="cb61-20"><a href="statistical-analysis-1.html#cb61-20" tabindex="-1"></a>  }</span>
<span id="cb61-21"><a href="statistical-analysis-1.html#cb61-21" tabindex="-1"></a></span>
<span id="cb61-22"><a href="statistical-analysis-1.html#cb61-22" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span>methods<span class="sc">::</span><span class="fu">is</span>(object, <span class="st">&quot;SummarizedExperiment&quot;</span>)) {</span>
<span id="cb61-23"><a href="statistical-analysis-1.html#cb61-23" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;object is not either a phyloseq or SummarizedExperiment object.&quot;</span>)</span>
<span id="cb61-24"><a href="statistical-analysis-1.html#cb61-24" tabindex="-1"></a>  }</span>
<span id="cb61-25"><a href="statistical-analysis-1.html#cb61-25" tabindex="-1"></a></span>
<span id="cb61-26"><a href="statistical-analysis-1.html#cb61-26" tabindex="-1"></a>  method <span class="ot">&lt;-</span> <span class="fu">match.arg</span>(</span>
<span id="cb61-27"><a href="statistical-analysis-1.html#cb61-27" tabindex="-1"></a>    method, <span class="fu">c</span>(<span class="st">&quot;none&quot;</span>, <span class="st">&quot;LOD&quot;</span>, <span class="st">&quot;half_min&quot;</span>, <span class="st">&quot;median&quot;</span>,</span>
<span id="cb61-28"><a href="statistical-analysis-1.html#cb61-28" tabindex="-1"></a>              <span class="st">&quot;mean&quot;</span>, <span class="st">&quot;min&quot;</span>, <span class="st">&quot;knn&quot;</span>, <span class="st">&quot;rf&quot;</span>, <span class="st">&quot;QRILC&quot;</span>)</span>
<span id="cb61-29"><a href="statistical-analysis-1.html#cb61-29" tabindex="-1"></a>  )</span>
<span id="cb61-30"><a href="statistical-analysis-1.html#cb61-30" tabindex="-1"></a></span>
<span id="cb61-31"><a href="statistical-analysis-1.html#cb61-31" tabindex="-1"></a>  <span class="cf">if</span> (base<span class="sc">::</span><span class="fu">missing</span>(method)) {</span>
<span id="cb61-32"><a href="statistical-analysis-1.html#cb61-32" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">&quot;method argument is empty! KNN will be used&quot;</span>)</span>
<span id="cb61-33"><a href="statistical-analysis-1.html#cb61-33" tabindex="-1"></a>  }</span>
<span id="cb61-34"><a href="statistical-analysis-1.html#cb61-34" tabindex="-1"></a></span>
<span id="cb61-35"><a href="statistical-analysis-1.html#cb61-35" tabindex="-1"></a>  <span class="co"># profile: row-&gt;samples; col-&gt;features</span></span>
<span id="cb61-36"><a href="statistical-analysis-1.html#cb61-36" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">all</span>(<span class="sc">!</span><span class="fu">is.null</span>(object), <span class="fu">inherits</span>(object, <span class="st">&quot;SummarizedExperiment&quot;</span>))) {</span>
<span id="cb61-37"><a href="statistical-analysis-1.html#cb61-37" tabindex="-1"></a>    <span class="co"># sample table &amp; profile table</span></span>
<span id="cb61-38"><a href="statistical-analysis-1.html#cb61-38" tabindex="-1"></a>    sam_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object) <span class="sc">%&gt;%</span></span>
<span id="cb61-39"><a href="statistical-analysis-1.html#cb61-39" tabindex="-1"></a>      <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb61-40"><a href="statistical-analysis-1.html#cb61-40" tabindex="-1"></a>      tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;TempRowNames&quot;</span>)</span>
<span id="cb61-41"><a href="statistical-analysis-1.html#cb61-41" tabindex="-1"></a>    prf_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) <span class="sc">%&gt;%</span></span>
<span id="cb61-42"><a href="statistical-analysis-1.html#cb61-42" tabindex="-1"></a>      <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb61-43"><a href="statistical-analysis-1.html#cb61-43" tabindex="-1"></a>      <span class="fu">t</span>()</span>
<span id="cb61-44"><a href="statistical-analysis-1.html#cb61-44" tabindex="-1"></a>  }</span>
<span id="cb61-45"><a href="statistical-analysis-1.html#cb61-45" tabindex="-1"></a></span>
<span id="cb61-46"><a href="statistical-analysis-1.html#cb61-46" tabindex="-1"></a>  group_index <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">colnames</span>(sam_tab) <span class="sc">==</span> group)</span>
<span id="cb61-47"><a href="statistical-analysis-1.html#cb61-47" tabindex="-1"></a>  samples_groups <span class="ot">&lt;-</span> sam_tab[, group_index]</span>
<span id="cb61-48"><a href="statistical-analysis-1.html#cb61-48" tabindex="-1"></a>  to_imp_data <span class="ot">&lt;-</span> prf_tab <span class="sc">%&gt;%</span> <span class="fu">as.matrix</span>()</span>
<span id="cb61-49"><a href="statistical-analysis-1.html#cb61-49" tabindex="-1"></a></span>
<span id="cb61-50"><a href="statistical-analysis-1.html#cb61-50" tabindex="-1"></a>  <span class="cf">if</span> (ZerosAsNA) {</span>
<span id="cb61-51"><a href="statistical-analysis-1.html#cb61-51" tabindex="-1"></a>    to_imp_data[to_imp_data <span class="sc">==</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb61-52"><a href="statistical-analysis-1.html#cb61-52" tabindex="-1"></a>    to_imp_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">cbind</span>(<span class="at">Group =</span> samples_groups, to_imp_data))</span>
<span id="cb61-53"><a href="statistical-analysis-1.html#cb61-53" tabindex="-1"></a>    <span class="fu">colnames</span>(to_imp_data)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(to_imp_data)] <span class="ot">&lt;-</span> <span class="fu">colnames</span>(prf_tab)</span>
<span id="cb61-54"><a href="statistical-analysis-1.html#cb61-54" tabindex="-1"></a></span>
<span id="cb61-55"><a href="statistical-analysis-1.html#cb61-55" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb61-56"><a href="statistical-analysis-1.html#cb61-56" tabindex="-1"></a>    to_imp_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">cbind</span>(<span class="at">Group =</span> samples_groups, to_imp_data))</span>
<span id="cb61-57"><a href="statistical-analysis-1.html#cb61-57" tabindex="-1"></a>    <span class="fu">colnames</span>(to_imp_data)[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(to_imp_data)] <span class="ot">&lt;-</span> <span class="fu">colnames</span>(prf_tab)</span>
<span id="cb61-58"><a href="statistical-analysis-1.html#cb61-58" tabindex="-1"></a>  }</span>
<span id="cb61-59"><a href="statistical-analysis-1.html#cb61-59" tabindex="-1"></a></span>
<span id="cb61-60"><a href="statistical-analysis-1.html#cb61-60" tabindex="-1"></a>  percent_na <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">is.na</span>(to_imp_data))</span>
<span id="cb61-61"><a href="statistical-analysis-1.html#cb61-61" tabindex="-1"></a>  <span class="cf">if</span> (percent_na <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb61-62"><a href="statistical-analysis-1.html#cb61-62" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">&quot;No missing values detected in your data&quot;</span>)</span>
<span id="cb61-63"><a href="statistical-analysis-1.html#cb61-63" tabindex="-1"></a>    <span class="cf">if</span> (method <span class="sc">!=</span> <span class="st">&quot;none&quot;</span>) {</span>
<span id="cb61-64"><a href="statistical-analysis-1.html#cb61-64" tabindex="-1"></a>      method <span class="ot">&lt;-</span> <span class="st">&quot;none&quot;</span></span>
<span id="cb61-65"><a href="statistical-analysis-1.html#cb61-65" tabindex="-1"></a>    }</span>
<span id="cb61-66"><a href="statistical-analysis-1.html#cb61-66" tabindex="-1"></a>  }</span>
<span id="cb61-67"><a href="statistical-analysis-1.html#cb61-67" tabindex="-1"></a></span>
<span id="cb61-68"><a href="statistical-analysis-1.html#cb61-68" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">isTRUE</span>(RemoveNA)) {</span>
<span id="cb61-69"><a href="statistical-analysis-1.html#cb61-69" tabindex="-1"></a>    count_NA <span class="ot">&lt;-</span> stats<span class="sc">::</span><span class="fu">aggregate</span>(</span>
<span id="cb61-70"><a href="statistical-analysis-1.html#cb61-70" tabindex="-1"></a>          . <span class="sc">~</span> Group,</span>
<span id="cb61-71"><a href="statistical-analysis-1.html#cb61-71" tabindex="-1"></a>          <span class="at">data =</span> to_imp_data,</span>
<span id="cb61-72"><a href="statistical-analysis-1.html#cb61-72" tabindex="-1"></a>          <span class="cf">function</span>(x) {(<span class="fu">sum</span>(<span class="fu">is.na</span>(x)) <span class="sc">/</span> (<span class="fu">sum</span>(<span class="fu">is.na</span>(x)) <span class="sc">+</span> <span class="fu">sum</span>(<span class="sc">!</span><span class="fu">is.na</span>(x))) ) },</span>
<span id="cb61-73"><a href="statistical-analysis-1.html#cb61-73" tabindex="-1"></a>          <span class="at">na.action =</span> <span class="cn">NULL</span>)</span>
<span id="cb61-74"><a href="statistical-analysis-1.html#cb61-74" tabindex="-1"></a>    count_NA <span class="ot">&lt;-</span> count_NA <span class="sc">%&gt;%</span></span>
<span id="cb61-75"><a href="statistical-analysis-1.html#cb61-75" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>Group)</span>
<span id="cb61-76"><a href="statistical-analysis-1.html#cb61-76" tabindex="-1"></a>    correct_names <span class="ot">&lt;-</span> <span class="fu">names</span>(count_NA)</span>
<span id="cb61-77"><a href="statistical-analysis-1.html#cb61-77" tabindex="-1"></a>    supress <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">as.data.frame</span>(<span class="fu">lapply</span>(count_NA, <span class="cf">function</span>(x) <span class="fu">any</span>(x <span class="sc">&gt;</span> prevalence))))</span>
<span id="cb61-78"><a href="statistical-analysis-1.html#cb61-78" tabindex="-1"></a>    <span class="fu">names</span>(supress) <span class="ot">&lt;-</span> correct_names</span>
<span id="cb61-79"><a href="statistical-analysis-1.html#cb61-79" tabindex="-1"></a>    correct_names <span class="ot">&lt;-</span> <span class="fu">names</span>(supress[supress <span class="sc">==</span> <span class="st">&quot;FALSE&quot;</span>])</span>
<span id="cb61-80"><a href="statistical-analysis-1.html#cb61-80" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> to_imp_data[, <span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(to_imp_data)][<span class="sc">!</span>supress]</span>
<span id="cb61-81"><a href="statistical-analysis-1.html#cb61-81" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">sapply</span>(depurdata, <span class="cf">function</span>(x) <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(x)))</span>
<span id="cb61-82"><a href="statistical-analysis-1.html#cb61-82" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb61-83"><a href="statistical-analysis-1.html#cb61-83" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> to_imp_data[, <span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(to_imp_data)]</span>
<span id="cb61-84"><a href="statistical-analysis-1.html#cb61-84" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">sapply</span>(depurdata, <span class="cf">function</span>(x) <span class="fu">as.numeric</span>(<span class="fu">as.character</span>(x)))</span>
<span id="cb61-85"><a href="statistical-analysis-1.html#cb61-85" tabindex="-1"></a>    correct_names <span class="ot">&lt;-</span> <span class="fu">colnames</span>(prf_tab)</span>
<span id="cb61-86"><a href="statistical-analysis-1.html#cb61-86" tabindex="-1"></a>  }</span>
<span id="cb61-87"><a href="statistical-analysis-1.html#cb61-87" tabindex="-1"></a></span>
<span id="cb61-88"><a href="statistical-analysis-1.html#cb61-88" tabindex="-1"></a>  <span class="co"># Row-&gt;feature;Col-&gt;sample</span></span>
<span id="cb61-89"><a href="statistical-analysis-1.html#cb61-89" tabindex="-1"></a>  <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;none&quot;</span>) {</span>
<span id="cb61-90"><a href="statistical-analysis-1.html#cb61-90" tabindex="-1"></a>    depurdata[<span class="fu">is.na</span>(depurdata)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb61-91"><a href="statistical-analysis-1.html#cb61-91" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;LOD&quot;</span>) {</span>
<span id="cb61-92"><a href="statistical-analysis-1.html#cb61-92" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">is.null</span>(LOD)) {</span>
<span id="cb61-93"><a href="statistical-analysis-1.html#cb61-93" tabindex="-1"></a>      <span class="fu">message</span>(<span class="st">&quot;No LOD provided, regard one-tenth mininal value as LOD&quot;</span>)</span>
<span id="cb61-94"><a href="statistical-analysis-1.html#cb61-94" tabindex="-1"></a>      depurdata_withoutNA <span class="ot">&lt;-</span> depurdata[<span class="sc">!</span><span class="fu">is.na</span>(depurdata)]</span>
<span id="cb61-95"><a href="statistical-analysis-1.html#cb61-95" tabindex="-1"></a>      LOD <span class="ot">&lt;-</span> <span class="fu">min</span>(depurdata_withoutNA[depurdata_withoutNA <span class="sc">!=</span> <span class="dv">0</span>]) <span class="sc">/</span> <span class="dv">10</span></span>
<span id="cb61-96"><a href="statistical-analysis-1.html#cb61-96" tabindex="-1"></a>    }</span>
<span id="cb61-97"><a href="statistical-analysis-1.html#cb61-97" tabindex="-1"></a>    depurdata[<span class="fu">is.na</span>(depurdata)] <span class="ot">&lt;-</span> LOD</span>
<span id="cb61-98"><a href="statistical-analysis-1.html#cb61-98" tabindex="-1"></a>    depurdata[depurdata <span class="sc">==</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> LOD</span>
<span id="cb61-99"><a href="statistical-analysis-1.html#cb61-99" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;half_min&quot;</span>) {</span>
<span id="cb61-100"><a href="statistical-analysis-1.html#cb61-100" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">apply</span>(depurdata, <span class="dv">2</span>, <span class="cf">function</span>(x) {</span>
<span id="cb61-101"><a href="statistical-analysis-1.html#cb61-101" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">is.numeric</span>(x)) <span class="fu">ifelse</span>(<span class="fu">is.na</span>(x), <span class="fu">min</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)<span class="sc">/</span><span class="dv">2</span>, x) <span class="cf">else</span> x})</span>
<span id="cb61-102"><a href="statistical-analysis-1.html#cb61-102" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;median&quot;</span>) {</span>
<span id="cb61-103"><a href="statistical-analysis-1.html#cb61-103" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">apply</span>(depurdata, <span class="dv">2</span>, <span class="cf">function</span>(x) {</span>
<span id="cb61-104"><a href="statistical-analysis-1.html#cb61-104" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">is.numeric</span>(x)) <span class="fu">ifelse</span>(<span class="fu">is.na</span>(x), <span class="fu">median</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), x) <span class="cf">else</span> x})</span>
<span id="cb61-105"><a href="statistical-analysis-1.html#cb61-105" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;mean&quot;</span>) {</span>
<span id="cb61-106"><a href="statistical-analysis-1.html#cb61-106" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">apply</span>(depurdata, <span class="dv">2</span>, <span class="cf">function</span>(x) {</span>
<span id="cb61-107"><a href="statistical-analysis-1.html#cb61-107" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">is.numeric</span>(x)) <span class="fu">ifelse</span>(<span class="fu">is.na</span>(x), <span class="fu">mean</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), x) <span class="cf">else</span> x})</span>
<span id="cb61-108"><a href="statistical-analysis-1.html#cb61-108" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;min&quot;</span>) {</span>
<span id="cb61-109"><a href="statistical-analysis-1.html#cb61-109" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">apply</span>(depurdata, <span class="dv">2</span>, <span class="cf">function</span>(x) {</span>
<span id="cb61-110"><a href="statistical-analysis-1.html#cb61-110" tabindex="-1"></a>      <span class="cf">if</span>(<span class="fu">is.numeric</span>(x)) <span class="fu">ifelse</span>(<span class="fu">is.na</span>(x), <span class="fu">min</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), x) <span class="cf">else</span> x})</span>
<span id="cb61-111"><a href="statistical-analysis-1.html#cb61-111" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;knn&quot;</span>) {</span>
<span id="cb61-112"><a href="statistical-analysis-1.html#cb61-112" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">t</span>(depurdata)</span>
<span id="cb61-113"><a href="statistical-analysis-1.html#cb61-113" tabindex="-1"></a>    datai <span class="ot">&lt;-</span> impute<span class="sc">::</span><span class="fu">impute.knn</span>(depurdata, <span class="at">k =</span> <span class="dv">20</span>)</span>
<span id="cb61-114"><a href="statistical-analysis-1.html#cb61-114" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">t</span>(datai<span class="sc">$</span>data)</span>
<span id="cb61-115"><a href="statistical-analysis-1.html#cb61-115" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;rf&quot;</span>) {</span>
<span id="cb61-116"><a href="statistical-analysis-1.html#cb61-116" tabindex="-1"></a>    fit <span class="ot">&lt;-</span> missForest<span class="sc">::</span><span class="fu">missForest</span>(<span class="fu">t</span>(depurdata))</span>
<span id="cb61-117"><a href="statistical-analysis-1.html#cb61-117" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> fit<span class="sc">$</span>ximp <span class="sc">%&gt;%</span></span>
<span id="cb61-118"><a href="statistical-analysis-1.html#cb61-118" tabindex="-1"></a>      <span class="fu">t</span>()</span>
<span id="cb61-119"><a href="statistical-analysis-1.html#cb61-119" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (method <span class="sc">==</span> <span class="st">&quot;QRILC&quot;</span>) {</span>
<span id="cb61-120"><a href="statistical-analysis-1.html#cb61-120" tabindex="-1"></a>    fit <span class="ot">&lt;-</span> <span class="fu">log</span>(<span class="fu">t</span>(depurdata)) <span class="sc">%&gt;%</span></span>
<span id="cb61-121"><a href="statistical-analysis-1.html#cb61-121" tabindex="-1"></a>      imputeLCMD<span class="sc">::</span><span class="fu">impute.QRILC</span>()</span>
<span id="cb61-122"><a href="statistical-analysis-1.html#cb61-122" tabindex="-1"></a>    depurdata <span class="ot">&lt;-</span> <span class="fu">t</span>(fit[[<span class="dv">1</span>]])</span>
<span id="cb61-123"><a href="statistical-analysis-1.html#cb61-123" tabindex="-1"></a>  }</span>
<span id="cb61-124"><a href="statistical-analysis-1.html#cb61-124" tabindex="-1"></a></span>
<span id="cb61-125"><a href="statistical-analysis-1.html#cb61-125" tabindex="-1"></a>  <span class="fu">colnames</span>(depurdata) <span class="ot">&lt;-</span> correct_names</span>
<span id="cb61-126"><a href="statistical-analysis-1.html#cb61-126" tabindex="-1"></a>  <span class="fu">rownames</span>(depurdata) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(prf_tab)</span>
<span id="cb61-127"><a href="statistical-analysis-1.html#cb61-127" tabindex="-1"></a></span>
<span id="cb61-128"><a href="statistical-analysis-1.html#cb61-128" tabindex="-1"></a>  <span class="cf">if</span> (methods<span class="sc">::</span><span class="fu">is</span>(object, <span class="st">&quot;SummarizedExperiment&quot;</span>)) {</span>
<span id="cb61-129"><a href="statistical-analysis-1.html#cb61-129" tabindex="-1"></a>    target <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object) <span class="sc">%&gt;%</span></span>
<span id="cb61-130"><a href="statistical-analysis-1.html#cb61-130" tabindex="-1"></a>      <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb61-131"><a href="statistical-analysis-1.html#cb61-131" tabindex="-1"></a>      <span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>)</span>
<span id="cb61-132"><a href="statistical-analysis-1.html#cb61-132" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb61-133"><a href="statistical-analysis-1.html#cb61-133" tabindex="-1"></a>                                    <span class="at">features =</span> depurdata)</span>
<span id="cb61-134"><a href="statistical-analysis-1.html#cb61-134" tabindex="-1"></a>    </span>
<span id="cb61-135"><a href="statistical-analysis-1.html#cb61-135" tabindex="-1"></a>  }</span>
<span id="cb61-136"><a href="statistical-analysis-1.html#cb61-136" tabindex="-1"></a></span>
<span id="cb61-137"><a href="statistical-analysis-1.html#cb61-137" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb61-138"><a href="statistical-analysis-1.html#cb61-138" tabindex="-1"></a>}</span>
<span id="cb61-139"><a href="statistical-analysis-1.html#cb61-139" tabindex="-1"></a></span>
<span id="cb61-140"><a href="statistical-analysis-1.html#cb61-140" tabindex="-1"></a>se_impute <span class="ot">&lt;-</span> <span class="fu">impute_abundance</span>(</span>
<span id="cb61-141"><a href="statistical-analysis-1.html#cb61-141" tabindex="-1"></a>                    se_check, </span>
<span id="cb61-142"><a href="statistical-analysis-1.html#cb61-142" tabindex="-1"></a>                    <span class="at">group =</span> <span class="st">&quot;group&quot;</span>,</span>
<span id="cb61-143"><a href="statistical-analysis-1.html#cb61-143" tabindex="-1"></a>                    <span class="at">ZerosAsNA =</span> <span class="cn">TRUE</span>, </span>
<span id="cb61-144"><a href="statistical-analysis-1.html#cb61-144" tabindex="-1"></a>                    <span class="at">RemoveNA =</span> <span class="cn">TRUE</span>, </span>
<span id="cb61-145"><a href="statistical-analysis-1.html#cb61-145" tabindex="-1"></a>                    <span class="at">prevalence =</span> <span class="fl">0.5</span>, </span>
<span id="cb61-146"><a href="statistical-analysis-1.html#cb61-146" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;knn&quot;</span>)</span>
<span id="cb61-147"><a href="statistical-analysis-1.html#cb61-147" tabindex="-1"></a>se_impute</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 180 55 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(180): M_38768 M_38296 ... M_31787 M_63361
## rowData names(0):
## colnames(55): P101001 P101003 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="statistical-analysis-1.html#cb63-1" tabindex="-1"></a><span class="co"># profile</span></span>
<span id="cb63-2"><a href="statistical-analysis-1.html#cb63-2" tabindex="-1"></a><span class="fu">head</span>(SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(se_impute))</span></code></pre></div>
<pre><code>##            P101001    P101003     P101004    P101007     P101009    P101010    P101011     P101012    P101013    P101016     P101017     P101018
## M_38768 51127588.0 42040432.0 34940596.00 58518636.0 51118832.00 83783688.0 29017984.0 51222064.00 77550128.0 30949554.0 26923596.00 56720032.00
## M_38296  5105020.5  4006120.2  3885477.00  4285129.5  6665653.50  9057441.0  2802655.2  5996555.00 11367511.0  3874736.8  2817151.00  8029728.00
## M_63436   756686.2   983889.2   851026.50   726593.9   232959.52   650261.1   541954.8   598491.00   438885.6  1625844.8   566466.94   427850.62
## M_57814   281502.0   175893.0   297304.38   319016.7   242172.20   200135.8   242804.9   248842.02   377212.6   183718.3   232514.53   279062.56
## M_52984   125465.8   154494.6   128320.37   176100.9    77345.62   122282.8   131924.3    80226.58   102010.7   129011.9    97377.79    98456.04
## M_48762   559069.1   596473.9    53293.76   627140.5  7016015.00  1914246.9  2762589.2   140576.45   723530.9   295995.2  3321584.25  3489639.25
##            P101019     P101021    P101022    P101024    P101025    P101027    P101030    P101031    P101038     P101041     P101042     P101047
## M_38768 27956064.0 48723600.00 16282054.0 77028824.0 32022342.0 22589448.0 38449788.0 59134052.0 32038030.0 20833830.00 33809080.00 18637508.00
## M_38296  3766663.8  5174967.00  1746182.9  5519105.5  2557365.0  1882902.6  2860324.2  4721201.0  4011627.5  2938779.00  3017260.50  1935144.12
## M_63436   519559.0  1301591.25  1474247.4   970475.8   628680.1   635516.6   367246.8   512037.9   852000.1   634488.56  1680135.75   326005.62
## M_57814   195083.9   290545.53   514479.0   420295.6   181825.7   196115.8   231290.3   315392.4   248820.8   214474.08   295862.56   397441.94
## M_52984   361975.5   119374.53   169170.4   101209.9   105117.3   104063.2   153193.4   181312.4   235948.1    68496.68   112239.87    94186.81
## M_48762   417727.6    16078.62 24787876.0  2287562.2  4724330.0  2960643.8  1503947.2   338791.1  1028211.6 16129281.00    87286.19  2211343.75
##            P101050    P101051     P101052    P101054    P101056    P101057    P101059     P101061    P101062    P101064     P101065     P101067
## M_38768 21978476.0 24265162.0 52203780.00 12836384.0 18546636.0 32301820.0 22645984.0 23683254.00 29027646.0 32629048.0 22950806.00 33555116.00
## M_38296  2897211.0  2476279.5  5928454.00  1685760.6  1650011.0  3419157.8  2196044.2  3217499.25  4060367.8  3031529.5  2467147.25  3567913.25
## M_63436   316650.2   737202.9   459385.94   346176.6   585470.2   417958.5   734586.3   337035.72   982299.4  1255148.0   637699.06   284516.12
## M_57814   253910.2   295948.6   246857.86   330762.7   209301.7   265086.1   181746.3   256601.55   187952.7   165441.1   203872.41   238433.59
## M_52984   112969.0   118040.4   115115.18   142644.3   103307.2   113114.2    89261.8    56381.96   186628.3   179940.6    82205.24    56987.89
## M_48762    98800.9  1933066.2    54885.02  4003695.2  1786579.6  2497416.7 10217042.0  7272486.00  4477679.0   395835.7   460618.81  8156659.00
##            P101068    P101069    P101071    P101072    P101074     P101075    P101076    P101077    P101079    P101080    P101081    P101082
## M_38768 44283972.0 52685972.0 32415040.0 34170948.0 22550616.0 22058076.00 24455466.0 25225170.0 15718590.0 29120336.0 65904836.0 22908578.0
## M_38296  6525382.0  3984333.5  3001414.5  4679519.0  2529255.5  2583265.50  3515218.2  3272875.0  2449462.5  2695001.5  6474709.5  2110243.8
## M_63436   664800.1   684813.6   596846.1   316855.0   646136.8   198381.73   255897.7   547243.4   508791.6  1256550.2   339909.3   596292.2
## M_57814   220632.9   209510.0   412294.9   335871.3   229869.7   225894.31   201601.9   285970.2   191453.5   264148.1   212220.1   335886.6
## M_52984   144673.7   151451.9   131184.4   127776.5   102868.8    64535.59   105523.9   130333.8   127343.7   153657.1   125355.2   107572.9
## M_48762  3966085.0  2887422.5   190455.1 46928844.0  3240584.5    91241.58  3088418.0  6992567.5  1325582.1   600080.8  3410091.7  1303324.6
##            P101084    P101085     P101088     P101090     P101094    P101095     P101096
## M_38768 29140440.0 20427124.0 29199012.00 24042020.00 36910084.00 35662068.0 66402192.00
## M_38296  3648091.2  3253531.8  4154170.75  2396959.75  4759584.50  3452283.2  6374383.00
## M_63436   497300.8   309859.3   601515.12   794206.00   414972.84  3606340.5  1077637.50
## M_57814   228471.4   345303.3   333549.22   321148.53   313197.78   500135.6   226660.25
## M_52984   151915.6   106491.1    89181.83   147634.67    91856.74   340070.0   137341.48
## M_48762   684199.1  2319273.2   854781.06    92700.81  1132143.00 31216882.0    34001.17</code></pre>
</div>
<div id="data-filtering-1" class="section level3 hasAnchor" number="3.1.6">
<h3><span class="header-section-number">3.1.6</span> Data Filtering<a href="statistical-analysis-1.html#data-filtering-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The purpose of the data filtering is to identify and remove variables that are unlikely to be of use when modeling the data. No phenotype information are used in the filtering process, so the result can be used with any downstream analysis. This step is strongly recommended for untargeted metabolomics datasets (i.e. spectral binning data, peak lists) with large number of variables, many of them are from baseline noises. Filtering can usually improve the results. For details, please refer to the paper by Hackstadt, et al.</p>
<p>Non-informative variables can be characterized in three groups: 1) variables of very small values (close to baseline or detection limit) - these variables can be detected using mean or median; 2) variables that are near-constant values throughout the experiment conditions (housekeeping or homeostasis) - these variables can be detected using standard deviation (SD); or the robust estimate such as interquantile range (IQR); and 3) variables that show low repeatability - this can be measured using QC samples using the relative standard deviation(RSD = SD/mean). Features with high percent RSD should be removed from the subsequent analysis (the suggested threshold is 20% for LC-MS and 30% for GC-MS). For data filtering based on the first two categories, the following empirical rules are applied during data filtering:</p>
<ul>
<li>Less than 250 variables: 5% will be filtered;</li>
<li>Between 250 - 500 variables: 10% will be filtered;</li>
<li>Between 500 - 1000 variables: 25% will be filtered;</li>
<li>Over 1000 variables: 40% will be filtered;</li>
</ul>
<p>Filtering features if their RSDs are &gt; 25% in QC samples</p>
<ul>
<li><p>Interquantile range (IQR)</p></li>
<li><p>Standard deviation (SD)</p></li>
<li><p>Median absolute deviation (MAD)</p></li>
<li><p>Relative standard deviation (RSD = SD/mean)</p></li>
<li><p>Non-parametric relative standard deviation (MAD/median)</p></li>
<li><p>Mean intensity value</p></li>
<li><p>Median intensity value</p></li>
</ul>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="statistical-analysis-1.html#cb65-1" tabindex="-1"></a>FilterFeature <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb65-2"><a href="statistical-analysis-1.html#cb65-2" tabindex="-1"></a>    object,</span>
<span id="cb65-3"><a href="statistical-analysis-1.html#cb65-3" tabindex="-1"></a>    qc_label,</span>
<span id="cb65-4"><a href="statistical-analysis-1.html#cb65-4" tabindex="-1"></a>    <span class="at">method =</span> <span class="fu">c</span>(<span class="st">&quot;none&quot;</span>, <span class="st">&quot;iqr&quot;</span>, <span class="st">&quot;rsd&quot;</span>, </span>
<span id="cb65-5"><a href="statistical-analysis-1.html#cb65-5" tabindex="-1"></a>               <span class="st">&quot;nrsd&quot;</span>, <span class="st">&quot;mean&quot;</span>, <span class="st">&quot;sd&quot;</span>,</span>
<span id="cb65-6"><a href="statistical-analysis-1.html#cb65-6" tabindex="-1"></a>               <span class="st">&quot;mad&quot;</span>, <span class="st">&quot;median&quot;</span>),</span>
<span id="cb65-7"><a href="statistical-analysis-1.html#cb65-7" tabindex="-1"></a>    <span class="at">rsd_cutoff =</span> <span class="dv">25</span>) {</span>
<span id="cb65-8"><a href="statistical-analysis-1.html#cb65-8" tabindex="-1"></a>  </span>
<span id="cb65-9"><a href="statistical-analysis-1.html#cb65-9" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) </span>
<span id="cb65-10"><a href="statistical-analysis-1.html#cb65-10" tabindex="-1"></a>  metadata_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object) </span>
<span id="cb65-11"><a href="statistical-analysis-1.html#cb65-11" tabindex="-1"></a>  </span>
<span id="cb65-12"><a href="statistical-analysis-1.html#cb65-12" tabindex="-1"></a>  <span class="co"># QC samples</span></span>
<span id="cb65-13"><a href="statistical-analysis-1.html#cb65-13" tabindex="-1"></a>  qc_samples <span class="ot">&lt;-</span> metadata_tab <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb65-14"><a href="statistical-analysis-1.html#cb65-14" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(group <span class="sc">==</span> qc_label)</span>
<span id="cb65-15"><a href="statistical-analysis-1.html#cb65-15" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">dim</span>(qc_samples)[<span class="dv">1</span>] <span class="sc">==</span> <span class="dv">0</span>) {</span>
<span id="cb65-16"><a href="statistical-analysis-1.html#cb65-16" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;No qc samples have been chosen, please check your input&quot;</span>)</span>
<span id="cb65-17"><a href="statistical-analysis-1.html#cb65-17" tabindex="-1"></a>  }</span>
<span id="cb65-18"><a href="statistical-analysis-1.html#cb65-18" tabindex="-1"></a>  </span>
<span id="cb65-19"><a href="statistical-analysis-1.html#cb65-19" tabindex="-1"></a>  <span class="co"># QC samples&#39; feature table</span></span>
<span id="cb65-20"><a href="statistical-analysis-1.html#cb65-20" tabindex="-1"></a>  qc_feature <span class="ot">&lt;-</span> features_tab[, <span class="fu">colnames</span>(features_tab)<span class="sc">%in%</span><span class="fu">rownames</span>(qc_samples)] <span class="sc">%&gt;%</span></span>
<span id="cb65-21"><a href="statistical-analysis-1.html#cb65-21" tabindex="-1"></a>    <span class="fu">t</span>()</span>
<span id="cb65-22"><a href="statistical-analysis-1.html#cb65-22" tabindex="-1"></a>  </span>
<span id="cb65-23"><a href="statistical-analysis-1.html#cb65-23" tabindex="-1"></a>  <span class="co"># filter features by QC RSD</span></span>
<span id="cb65-24"><a href="statistical-analysis-1.html#cb65-24" tabindex="-1"></a>  rsd <span class="ot">&lt;-</span> rsd_cutoff <span class="sc">/</span> <span class="dv">100</span></span>
<span id="cb65-25"><a href="statistical-analysis-1.html#cb65-25" tabindex="-1"></a>  sds <span class="ot">&lt;-</span> <span class="fu">apply</span>(qc_feature, <span class="dv">2</span>, sd, <span class="at">na.rm =</span> T)</span>
<span id="cb65-26"><a href="statistical-analysis-1.html#cb65-26" tabindex="-1"></a>  mns <span class="ot">&lt;-</span> <span class="fu">apply</span>(qc_feature, <span class="dv">2</span>, mean, <span class="at">na.rm =</span> T)</span>
<span id="cb65-27"><a href="statistical-analysis-1.html#cb65-27" tabindex="-1"></a>  rsd_vals <span class="ot">&lt;-</span> <span class="fu">abs</span>(sds<span class="sc">/</span>mns) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb65-28"><a href="statistical-analysis-1.html#cb65-28" tabindex="-1"></a>  gd_inx <span class="ot">&lt;-</span> rsd_vals <span class="sc">&lt;</span> rsd</span>
<span id="cb65-29"><a href="statistical-analysis-1.html#cb65-29" tabindex="-1"></a>  int_mat <span class="ot">&lt;-</span> features_tab[gd_inx, ]</span>
<span id="cb65-30"><a href="statistical-analysis-1.html#cb65-30" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">&quot;Removed &quot;</span>, (<span class="fu">dim</span>(qc_feature)[<span class="dv">2</span>] <span class="sc">-</span> <span class="fu">dim</span>(int_mat)[<span class="dv">1</span>]), </span>
<span id="cb65-31"><a href="statistical-analysis-1.html#cb65-31" tabindex="-1"></a>  <span class="st">&quot; features based on QC RSD values. QC samples are excluded from downstream functional analysis.&quot;</span>)</span>
<span id="cb65-32"><a href="statistical-analysis-1.html#cb65-32" tabindex="-1"></a>  </span>
<span id="cb65-33"><a href="statistical-analysis-1.html#cb65-33" tabindex="-1"></a>  <span class="co"># whether to filter features by percentage according to the number</span></span>
<span id="cb65-34"><a href="statistical-analysis-1.html#cb65-34" tabindex="-1"></a>  PerformFeatureFilter <span class="ot">&lt;-</span> <span class="cf">function</span>(datMatrix, </span>
<span id="cb65-35"><a href="statistical-analysis-1.html#cb65-35" tabindex="-1"></a>                                   <span class="at">qc_method =</span> method,</span>
<span id="cb65-36"><a href="statistical-analysis-1.html#cb65-36" tabindex="-1"></a>                                   <span class="at">remain_num =</span> <span class="cn">NULL</span>) {</span>
<span id="cb65-37"><a href="statistical-analysis-1.html#cb65-37" tabindex="-1"></a>    </span>
<span id="cb65-38"><a href="statistical-analysis-1.html#cb65-38" tabindex="-1"></a>    dat <span class="ot">&lt;-</span> datMatrix</span>
<span id="cb65-39"><a href="statistical-analysis-1.html#cb65-39" tabindex="-1"></a>    feat_num <span class="ot">&lt;-</span> <span class="fu">ncol</span>(dat)</span>
<span id="cb65-40"><a href="statistical-analysis-1.html#cb65-40" tabindex="-1"></a>    feat_nms <span class="ot">&lt;-</span> <span class="fu">colnames</span>(dat)</span>
<span id="cb65-41"><a href="statistical-analysis-1.html#cb65-41" tabindex="-1"></a>    nm <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb65-42"><a href="statistical-analysis-1.html#cb65-42" tabindex="-1"></a>    <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;none&quot;</span> <span class="sc">&amp;&amp;</span> feat_num <span class="sc">&lt;</span> <span class="dv">5000</span>) { <span class="co"># only allow for less than 4000</span></span>
<span id="cb65-43"><a href="statistical-analysis-1.html#cb65-43" tabindex="-1"></a>      remain <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">TRUE</span>, feat_num)</span>
<span id="cb65-44"><a href="statistical-analysis-1.html#cb65-44" tabindex="-1"></a>      nm <span class="ot">&lt;-</span> <span class="st">&quot;No filtering was applied&quot;</span></span>
<span id="cb65-45"><a href="statistical-analysis-1.html#cb65-45" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb65-46"><a href="statistical-analysis-1.html#cb65-46" tabindex="-1"></a>      <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;rsd&quot;</span>){</span>
<span id="cb65-47"><a href="statistical-analysis-1.html#cb65-47" tabindex="-1"></a>        sds <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, sd, <span class="at">na.rm =</span> T)</span>
<span id="cb65-48"><a href="statistical-analysis-1.html#cb65-48" tabindex="-1"></a>        mns <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, mean, <span class="at">na.rm =</span> T)</span>
<span id="cb65-49"><a href="statistical-analysis-1.html#cb65-49" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">abs</span>(sds<span class="sc">/</span>mns)</span>
<span id="cb65-50"><a href="statistical-analysis-1.html#cb65-50" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;Relative standard deviation&quot;</span></span>
<span id="cb65-51"><a href="statistical-analysis-1.html#cb65-51" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;nrsd&quot;</span> ) {</span>
<span id="cb65-52"><a href="statistical-analysis-1.html#cb65-52" tabindex="-1"></a>        mads <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, mad, <span class="at">na.rm =</span> T)</span>
<span id="cb65-53"><a href="statistical-analysis-1.html#cb65-53" tabindex="-1"></a>        meds <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, median, <span class="at">na.rm =</span> T)</span>
<span id="cb65-54"><a href="statistical-analysis-1.html#cb65-54" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">abs</span>(mads<span class="sc">/</span>meds)</span>
<span id="cb65-55"><a href="statistical-analysis-1.html#cb65-55" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;Non-paramatric relative standard deviation&quot;</span></span>
<span id="cb65-56"><a href="statistical-analysis-1.html#cb65-56" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;mean&quot;</span>) {</span>
<span id="cb65-57"><a href="statistical-analysis-1.html#cb65-57" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, mean, <span class="at">na.rm =</span> T)</span>
<span id="cb65-58"><a href="statistical-analysis-1.html#cb65-58" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;mean&quot;</span></span>
<span id="cb65-59"><a href="statistical-analysis-1.html#cb65-59" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;sd&quot;</span>) {</span>
<span id="cb65-60"><a href="statistical-analysis-1.html#cb65-60" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, sd, <span class="at">na.rm =</span> T)</span>
<span id="cb65-61"><a href="statistical-analysis-1.html#cb65-61" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;standard deviation&quot;</span></span>
<span id="cb65-62"><a href="statistical-analysis-1.html#cb65-62" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;mad&quot;</span>) {</span>
<span id="cb65-63"><a href="statistical-analysis-1.html#cb65-63" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, mad, <span class="at">na.rm =</span> T)</span>
<span id="cb65-64"><a href="statistical-analysis-1.html#cb65-64" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;Median absolute deviation&quot;</span></span>
<span id="cb65-65"><a href="statistical-analysis-1.html#cb65-65" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;median&quot;</span>) {</span>
<span id="cb65-66"><a href="statistical-analysis-1.html#cb65-66" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, median, <span class="at">na.rm =</span> T)</span>
<span id="cb65-67"><a href="statistical-analysis-1.html#cb65-67" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;median&quot;</span></span>
<span id="cb65-68"><a href="statistical-analysis-1.html#cb65-68" tabindex="-1"></a>      } <span class="cf">else</span> <span class="cf">if</span> (qc_method <span class="sc">==</span> <span class="st">&quot;iqr&quot;</span>) { <span class="co"># iqr</span></span>
<span id="cb65-69"><a href="statistical-analysis-1.html#cb65-69" tabindex="-1"></a>        filter_val <span class="ot">&lt;-</span> <span class="fu">apply</span>(dat, <span class="dv">2</span>, IQR, <span class="at">na.rm =</span> T)</span>
<span id="cb65-70"><a href="statistical-analysis-1.html#cb65-70" tabindex="-1"></a>        nm <span class="ot">&lt;-</span> <span class="st">&quot;Interquantile Range&quot;</span></span>
<span id="cb65-71"><a href="statistical-analysis-1.html#cb65-71" tabindex="-1"></a>      }</span>
<span id="cb65-72"><a href="statistical-analysis-1.html#cb65-72" tabindex="-1"></a>      </span>
<span id="cb65-73"><a href="statistical-analysis-1.html#cb65-73" tabindex="-1"></a>      <span class="co"># get the rank of the filtered variables</span></span>
<span id="cb65-74"><a href="statistical-analysis-1.html#cb65-74" tabindex="-1"></a>      rk <span class="ot">&lt;-</span> <span class="fu">rank</span>(<span class="sc">-</span>filter_val, <span class="at">ties.method =</span> <span class="st">&quot;random&quot;</span>)</span>
<span id="cb65-75"><a href="statistical-analysis-1.html#cb65-75" tabindex="-1"></a>      </span>
<span id="cb65-76"><a href="statistical-analysis-1.html#cb65-76" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">is.null</span>(remain_num)) { <span class="co"># apply empirical filtering based on data size</span></span>
<span id="cb65-77"><a href="statistical-analysis-1.html#cb65-77" tabindex="-1"></a>          <span class="cf">if</span> (feat_num <span class="sc">&lt;</span> <span class="dv">250</span>) { <span class="co"># reduce 5%</span></span>
<span id="cb65-78"><a href="statistical-analysis-1.html#cb65-78" tabindex="-1"></a>            remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> feat_num <span class="sc">*</span> <span class="fl">0.95</span></span>
<span id="cb65-79"><a href="statistical-analysis-1.html#cb65-79" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Further feature filtering based on &quot;</span>, nm)</span>
<span id="cb65-80"><a href="statistical-analysis-1.html#cb65-80" tabindex="-1"></a>          } <span class="cf">else</span> <span class="cf">if</span> (feat_num <span class="sc">&lt;</span> <span class="dv">500</span>) { <span class="co"># reduce 10%</span></span>
<span id="cb65-81"><a href="statistical-analysis-1.html#cb65-81" tabindex="-1"></a>            remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> feat_num <span class="sc">*</span> <span class="fl">0.9</span></span>
<span id="cb65-82"><a href="statistical-analysis-1.html#cb65-82" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Further feature filtering based on &quot;</span>, nm)</span>
<span id="cb65-83"><a href="statistical-analysis-1.html#cb65-83" tabindex="-1"></a>          } <span class="cf">else</span> <span class="cf">if</span> (feat_num <span class="sc">&lt;</span> <span class="dv">1000</span>) { <span class="co"># reduce 25%</span></span>
<span id="cb65-84"><a href="statistical-analysis-1.html#cb65-84" tabindex="-1"></a>            remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> feat_num <span class="sc">*</span> <span class="fl">0.75</span></span>
<span id="cb65-85"><a href="statistical-analysis-1.html#cb65-85" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Further feature filtering based on &quot;</span>, nm)</span>
<span id="cb65-86"><a href="statistical-analysis-1.html#cb65-86" tabindex="-1"></a>          } <span class="cf">else</span> { <span class="co"># reduce 40%, if still over 5000, then only use top 5000</span></span>
<span id="cb65-87"><a href="statistical-analysis-1.html#cb65-87" tabindex="-1"></a>            remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> feat_num <span class="sc">*</span> <span class="fl">0.6</span></span>
<span id="cb65-88"><a href="statistical-analysis-1.html#cb65-88" tabindex="-1"></a>            <span class="fu">message</span>(<span class="st">&quot;Further feature filtering based on &quot;</span>, nm)</span>
<span id="cb65-89"><a href="statistical-analysis-1.html#cb65-89" tabindex="-1"></a>          }</span>
<span id="cb65-90"><a href="statistical-analysis-1.html#cb65-90" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb65-91"><a href="statistical-analysis-1.html#cb65-91" tabindex="-1"></a>        remain <span class="ot">&lt;-</span> rk <span class="sc">&lt;</span> remain_num</span>
<span id="cb65-92"><a href="statistical-analysis-1.html#cb65-92" tabindex="-1"></a>      }</span>
<span id="cb65-93"><a href="statistical-analysis-1.html#cb65-93" tabindex="-1"></a>    }</span>
<span id="cb65-94"><a href="statistical-analysis-1.html#cb65-94" tabindex="-1"></a>    </span>
<span id="cb65-95"><a href="statistical-analysis-1.html#cb65-95" tabindex="-1"></a>    res <span class="ot">&lt;-</span> datMatrix[, remain]</span>
<span id="cb65-96"><a href="statistical-analysis-1.html#cb65-96" tabindex="-1"></a>    </span>
<span id="cb65-97"><a href="statistical-analysis-1.html#cb65-97" tabindex="-1"></a>    <span class="fu">return</span>(res)</span>
<span id="cb65-98"><a href="statistical-analysis-1.html#cb65-98" tabindex="-1"></a>  }  </span>
<span id="cb65-99"><a href="statistical-analysis-1.html#cb65-99" tabindex="-1"></a>  </span>
<span id="cb65-100"><a href="statistical-analysis-1.html#cb65-100" tabindex="-1"></a>  feature_res <span class="ot">&lt;-</span> <span class="fu">PerformFeatureFilter</span>(<span class="fu">t</span>(int_mat))</span>
<span id="cb65-101"><a href="statistical-analysis-1.html#cb65-101" tabindex="-1"></a>  </span>
<span id="cb65-102"><a href="statistical-analysis-1.html#cb65-102" tabindex="-1"></a>  <span class="co"># remove QC samples </span></span>
<span id="cb65-103"><a href="statistical-analysis-1.html#cb65-103" tabindex="-1"></a>  feature_final <span class="ot">&lt;-</span> feature_res[<span class="sc">!</span><span class="fu">rownames</span>(feature_res) <span class="sc">%in%</span> <span class="fu">rownames</span>(qc_samples), ]</span>
<span id="cb65-104"><a href="statistical-analysis-1.html#cb65-104" tabindex="-1"></a>  </span>
<span id="cb65-105"><a href="statistical-analysis-1.html#cb65-105" tabindex="-1"></a>  <span class="co"># save int_mat into se object </span></span>
<span id="cb65-106"><a href="statistical-analysis-1.html#cb65-106" tabindex="-1"></a>  target <span class="ot">&lt;-</span> metadata_tab <span class="sc">%&gt;%</span> </span>
<span id="cb65-107"><a href="statistical-analysis-1.html#cb65-107" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb65-108"><a href="statistical-analysis-1.html#cb65-108" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb65-109"><a href="statistical-analysis-1.html#cb65-109" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(SampleID <span class="sc">%in%</span> <span class="fu">rownames</span>(feature_final))</span>
<span id="cb65-110"><a href="statistical-analysis-1.html#cb65-110" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb65-111"><a href="statistical-analysis-1.html#cb65-111" tabindex="-1"></a>                                  <span class="at">features =</span> feature_final)</span>
<span id="cb65-112"><a href="statistical-analysis-1.html#cb65-112" tabindex="-1"></a>  </span>
<span id="cb65-113"><a href="statistical-analysis-1.html#cb65-113" tabindex="-1"></a>  <span class="fu">return</span>(res) </span>
<span id="cb65-114"><a href="statistical-analysis-1.html#cb65-114" tabindex="-1"></a>}</span>
<span id="cb65-115"><a href="statistical-analysis-1.html#cb65-115" tabindex="-1"></a></span>
<span id="cb65-116"><a href="statistical-analysis-1.html#cb65-116" tabindex="-1"></a>se_filter <span class="ot">&lt;-</span> <span class="fu">FilterFeature</span>(</span>
<span id="cb65-117"><a href="statistical-analysis-1.html#cb65-117" tabindex="-1"></a>  <span class="at">object =</span> se_impute,</span>
<span id="cb65-118"><a href="statistical-analysis-1.html#cb65-118" tabindex="-1"></a>  <span class="at">qc_label =</span> <span class="st">&quot;None&quot;</span>,</span>
<span id="cb65-119"><a href="statistical-analysis-1.html#cb65-119" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">&quot;iqr&quot;</span>,</span>
<span id="cb65-120"><a href="statistical-analysis-1.html#cb65-120" tabindex="-1"></a>  <span class="at">rsd_cutoff =</span> <span class="dv">100</span>) <span class="co"># default values: rsd_cutoff = 25</span></span></code></pre></div>
<pre><code>## Removed 4 features based on QC RSD values. QC samples are excluded from downstream functional analysis.</code></pre>
<pre><code>## Further feature filtering based on Interquantile Range</code></pre>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="statistical-analysis-1.html#cb68-1" tabindex="-1"></a>se_filter</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 167 45 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(167): M_38768 M_38296 ... M_31787 M_63361
## rowData names(0):
## colnames(45): P101001 P101004 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="data-normalization-1" class="section level3 hasAnchor" number="3.1.7">
<h3><span class="header-section-number">3.1.7</span> Data Normalization<a href="statistical-analysis-1.html#data-normalization-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The normalization procedures are grouped into three categories. You can use one or combine them to achieve better results.</p>
<ul>
<li><p>Sample normalization is for general-purpose adjustment for systematic differences among samples;</p>
<ul>
<li><p>Sample-specific normalization (i.e. weight, volume)</p></li>
<li><p>Normalization by sum</p></li>
<li><p>Normalization by median</p></li>
<li><p>Normalization by a reference sample (PQN)</p></li>
<li><p>Normalization by a pooled sample from group (group PQN)</p></li>
<li><p>Normalization by reference feature</p></li>
<li><p>Quantile normalization (suggested only for &gt; 1000 features)</p></li>
</ul></li>
<li><p>Data transformation applies a mathematical transformation on individual values themselves. A simple mathematical approach is used to deal with negative values in log and square root.</p>
<ul>
<li><p>Log transformation (base 10)</p></li>
<li><p>Square root transformation (square root of data values)</p></li>
<li><p>Cube root transformation (cube root of data values)</p></li>
</ul></li>
<li><p>Data scaling adjusts each variable/feature by a scaling factor computed based on the dispersion of the variable.</p>
<ul>
<li><p>Mean centering (mean-centered only)</p></li>
<li><p>Auto scaling (mean-centered and divided by the standard deviation of each variable)</p></li>
<li><p>Pareto scaling (mean-centered and divided by the square root of the standard deviation of each variable)</p></li>
<li><p>Range scaling (mean-centered and divided by the range of each variable)</p></li>
</ul></li>
</ul>
<div id="normalization-by-normalizedata-function-1" class="section level4 hasAnchor" number="3.1.7.1">
<h4><span class="header-section-number">3.1.7.1</span> Normalization by <strong>NormalizeData</strong> function<a href="statistical-analysis-1.html#normalization-by-normalizedata-function-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="statistical-analysis-1.html#cb70-1" tabindex="-1"></a>NormalizeData <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb70-2"><a href="statistical-analysis-1.html#cb70-2" tabindex="-1"></a>                      object,</span>
<span id="cb70-3"><a href="statistical-analysis-1.html#cb70-3" tabindex="-1"></a>                      <span class="at">rowNorm =</span> <span class="fu">c</span>(<span class="st">&quot;Quantile&quot;</span>, <span class="st">&quot;GroupPQN&quot;</span>, <span class="st">&quot;SamplePQN&quot;</span>, </span>
<span id="cb70-4"><a href="statistical-analysis-1.html#cb70-4" tabindex="-1"></a>                                  <span class="st">&quot;CompNorm&quot;</span>, <span class="st">&quot;SumNorm&quot;</span>, <span class="st">&quot;MedianNorm&quot;</span>,</span>
<span id="cb70-5"><a href="statistical-analysis-1.html#cb70-5" tabindex="-1"></a>                                  <span class="st">&quot;SpecNorm&quot;</span>, <span class="st">&quot;None&quot;</span>),</span>
<span id="cb70-6"><a href="statistical-analysis-1.html#cb70-6" tabindex="-1"></a>                      <span class="at">transNorm =</span> <span class="fu">c</span>(<span class="st">&quot;LogNorm&quot;</span>, <span class="st">&quot;SrNorm&quot;</span>, <span class="st">&quot;CrNorm&quot;</span>, <span class="st">&quot;None&quot;</span>),</span>
<span id="cb70-7"><a href="statistical-analysis-1.html#cb70-7" tabindex="-1"></a>                      <span class="at">scaleNorm =</span> <span class="fu">c</span>(<span class="st">&quot;MeanCenter&quot;</span>, <span class="st">&quot;AutoNorm&quot;</span>, <span class="st">&quot;ParetoNorm&quot;</span>, </span>
<span id="cb70-8"><a href="statistical-analysis-1.html#cb70-8" tabindex="-1"></a>                                    <span class="st">&quot;RangeNorm&quot;</span>, <span class="st">&quot;None&quot;</span>),</span>
<span id="cb70-9"><a href="statistical-analysis-1.html#cb70-9" tabindex="-1"></a>                      <span class="at">ref =</span> <span class="cn">NULL</span>,</span>
<span id="cb70-10"><a href="statistical-analysis-1.html#cb70-10" tabindex="-1"></a>                      <span class="at">SpeWeight =</span> <span class="dv">1</span>) {</span>
<span id="cb70-11"><a href="statistical-analysis-1.html#cb70-11" tabindex="-1"></a>  </span>
<span id="cb70-12"><a href="statistical-analysis-1.html#cb70-12" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) </span>
<span id="cb70-13"><a href="statistical-analysis-1.html#cb70-13" tabindex="-1"></a>  metadata_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object)   </span>
<span id="cb70-14"><a href="statistical-analysis-1.html#cb70-14" tabindex="-1"></a>  </span>
<span id="cb70-15"><a href="statistical-analysis-1.html#cb70-15" tabindex="-1"></a>  data <span class="ot">&lt;-</span> <span class="fu">t</span>(features_tab)</span>
<span id="cb70-16"><a href="statistical-analysis-1.html#cb70-16" tabindex="-1"></a>  colNames <span class="ot">&lt;-</span> <span class="fu">colnames</span>(data)</span>
<span id="cb70-17"><a href="statistical-analysis-1.html#cb70-17" tabindex="-1"></a>  rowNames <span class="ot">&lt;-</span> <span class="fu">rownames</span>(data)</span>
<span id="cb70-18"><a href="statistical-analysis-1.html#cb70-18" tabindex="-1"></a>  </span>
<span id="cb70-19"><a href="statistical-analysis-1.html#cb70-19" tabindex="-1"></a>  <span class="do">#############################################</span></span>
<span id="cb70-20"><a href="statistical-analysis-1.html#cb70-20" tabindex="-1"></a>  <span class="co"># Sample normalization</span></span>
<span id="cb70-21"><a href="statistical-analysis-1.html#cb70-21" tabindex="-1"></a>  <span class="co"># perform quantile normalization on the raw data (can be log transformed later by user)</span></span>
<span id="cb70-22"><a href="statistical-analysis-1.html#cb70-22" tabindex="-1"></a>  QuantileNormalize <span class="ot">&lt;-</span> <span class="cf">function</span>(data) {</span>
<span id="cb70-23"><a href="statistical-analysis-1.html#cb70-23" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">t</span>(preprocessCore<span class="sc">::</span><span class="fu">normalize.quantiles</span>(<span class="fu">t</span>(data), <span class="at">copy=</span><span class="cn">FALSE</span>)));</span>
<span id="cb70-24"><a href="statistical-analysis-1.html#cb70-24" tabindex="-1"></a>  }</span>
<span id="cb70-25"><a href="statistical-analysis-1.html#cb70-25" tabindex="-1"></a>  <span class="co"># normalize by a reference sample (probability quotient normalization)</span></span>
<span id="cb70-26"><a href="statistical-analysis-1.html#cb70-26" tabindex="-1"></a>  <span class="co"># ref should be the name of the reference sample</span></span>
<span id="cb70-27"><a href="statistical-analysis-1.html#cb70-27" tabindex="-1"></a>  ProbNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x, ref_smpl) {</span>
<span id="cb70-28"><a href="statistical-analysis-1.html#cb70-28" tabindex="-1"></a>    <span class="fu">return</span>(x<span class="sc">/</span><span class="fu">median</span>(<span class="fu">as.numeric</span>(x<span class="sc">/</span>ref_smpl), <span class="at">na.rm =</span> T))</span>
<span id="cb70-29"><a href="statistical-analysis-1.html#cb70-29" tabindex="-1"></a>  }</span>
<span id="cb70-30"><a href="statistical-analysis-1.html#cb70-30" tabindex="-1"></a>  </span>
<span id="cb70-31"><a href="statistical-analysis-1.html#cb70-31" tabindex="-1"></a>  <span class="co"># normalize by a reference reference (i.e. creatinine)</span></span>
<span id="cb70-32"><a href="statistical-analysis-1.html#cb70-32" tabindex="-1"></a>  <span class="co"># ref should be the name of the cmpd</span></span>
<span id="cb70-33"><a href="statistical-analysis-1.html#cb70-33" tabindex="-1"></a>  CompNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x, ref) {</span>
<span id="cb70-34"><a href="statistical-analysis-1.html#cb70-34" tabindex="-1"></a>    <span class="fu">return</span>(<span class="dv">1000</span><span class="sc">*</span>x<span class="sc">/</span>x[ref])</span>
<span id="cb70-35"><a href="statistical-analysis-1.html#cb70-35" tabindex="-1"></a>  }</span>
<span id="cb70-36"><a href="statistical-analysis-1.html#cb70-36" tabindex="-1"></a>  </span>
<span id="cb70-37"><a href="statistical-analysis-1.html#cb70-37" tabindex="-1"></a>  SumNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb70-38"><a href="statistical-analysis-1.html#cb70-38" tabindex="-1"></a>    <span class="fu">return</span>(<span class="dv">1000</span><span class="sc">*</span>x<span class="sc">/</span><span class="fu">sum</span>(x, <span class="at">na.rm =</span> T))</span>
<span id="cb70-39"><a href="statistical-analysis-1.html#cb70-39" tabindex="-1"></a>  }</span>
<span id="cb70-40"><a href="statistical-analysis-1.html#cb70-40" tabindex="-1"></a>  </span>
<span id="cb70-41"><a href="statistical-analysis-1.html#cb70-41" tabindex="-1"></a>  <span class="co"># normalize by median</span></span>
<span id="cb70-42"><a href="statistical-analysis-1.html#cb70-42" tabindex="-1"></a>  MedianNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb70-43"><a href="statistical-analysis-1.html#cb70-43" tabindex="-1"></a>    <span class="fu">return</span>(x<span class="sc">/</span><span class="fu">median</span>(x, <span class="at">na.rm =</span> T))</span>
<span id="cb70-44"><a href="statistical-analysis-1.html#cb70-44" tabindex="-1"></a>  }  </span>
<span id="cb70-45"><a href="statistical-analysis-1.html#cb70-45" tabindex="-1"></a>  </span>
<span id="cb70-46"><a href="statistical-analysis-1.html#cb70-46" tabindex="-1"></a>  <span class="co"># row-wise normalization</span></span>
<span id="cb70-47"><a href="statistical-analysis-1.html#cb70-47" tabindex="-1"></a>  <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;Quantile&quot;</span>) {</span>
<span id="cb70-48"><a href="statistical-analysis-1.html#cb70-48" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">QuantileNormalize</span>(data)</span>
<span id="cb70-49"><a href="statistical-analysis-1.html#cb70-49" tabindex="-1"></a>    <span class="co"># this can introduce constant variables if a variable is </span></span>
<span id="cb70-50"><a href="statistical-analysis-1.html#cb70-50" tabindex="-1"></a>    <span class="co"># at the same rank across all samples (replaced by its average across all)</span></span>
<span id="cb70-51"><a href="statistical-analysis-1.html#cb70-51" tabindex="-1"></a>    varCol <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, var, <span class="at">na.rm =</span> T)</span>
<span id="cb70-52"><a href="statistical-analysis-1.html#cb70-52" tabindex="-1"></a>    constCol <span class="ot">&lt;-</span> (varCol <span class="sc">==</span> <span class="dv">0</span> <span class="sc">|</span> <span class="fu">is.na</span>(varCol))</span>
<span id="cb70-53"><a href="statistical-analysis-1.html#cb70-53" tabindex="-1"></a>    constNum <span class="ot">&lt;-</span> <span class="fu">sum</span>(constCol, <span class="at">na.rm =</span> T)</span>
<span id="cb70-54"><a href="statistical-analysis-1.html#cb70-54" tabindex="-1"></a>    <span class="cf">if</span> (constNum <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb70-55"><a href="statistical-analysis-1.html#cb70-55" tabindex="-1"></a>      <span class="fu">message</span>(<span class="fu">paste</span>(<span class="st">&quot;After quantile normalization&quot;</span>, constNum, </span>
<span id="cb70-56"><a href="statistical-analysis-1.html#cb70-56" tabindex="-1"></a>                    <span class="st">&quot;features with a constant value were found and deleted.&quot;</span>))</span>
<span id="cb70-57"><a href="statistical-analysis-1.html#cb70-57" tabindex="-1"></a>      data <span class="ot">&lt;-</span> data[, <span class="sc">!</span>constCol, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb70-58"><a href="statistical-analysis-1.html#cb70-58" tabindex="-1"></a>      colNames <span class="ot">&lt;-</span> <span class="fu">colnames</span>(data)</span>
<span id="cb70-59"><a href="statistical-analysis-1.html#cb70-59" tabindex="-1"></a>      rowNames <span class="ot">&lt;-</span> <span class="fu">rownames</span>(data)</span>
<span id="cb70-60"><a href="statistical-analysis-1.html#cb70-60" tabindex="-1"></a>    }</span>
<span id="cb70-61"><a href="statistical-analysis-1.html#cb70-61" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Quantile Normalization&quot;</span></span>
<span id="cb70-62"><a href="statistical-analysis-1.html#cb70-62" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;GroupPQN&quot;</span>) {</span>
<span id="cb70-63"><a href="statistical-analysis-1.html#cb70-63" tabindex="-1"></a>    grp_inx <span class="ot">&lt;-</span> metadata_tab<span class="sc">$</span>group <span class="sc">==</span> ref</span>
<span id="cb70-64"><a href="statistical-analysis-1.html#cb70-64" tabindex="-1"></a>    ref.smpl <span class="ot">&lt;-</span> <span class="fu">apply</span>(data[grp_inx, , <span class="at">drop =</span> <span class="cn">FALSE</span>], <span class="dv">2</span>, mean)</span>
<span id="cb70-65"><a href="statistical-analysis-1.html#cb70-65" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(data, <span class="dv">1</span>, ProbNorm, ref.smpl))</span>
<span id="cb70-66"><a href="statistical-analysis-1.html#cb70-66" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Probabilistic Quotient Normalization by a reference group&quot;</span></span>
<span id="cb70-67"><a href="statistical-analysis-1.html#cb70-67" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;SamplePQN&quot;</span>) {</span>
<span id="cb70-68"><a href="statistical-analysis-1.html#cb70-68" tabindex="-1"></a>    ref.smpl <span class="ot">&lt;-</span> data[ref, , drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb70-69"><a href="statistical-analysis-1.html#cb70-69" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(data, <span class="dv">1</span>, ProbNorm, ref.smpl))</span>
<span id="cb70-70"><a href="statistical-analysis-1.html#cb70-70" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Probabilistic Quotient Normalization by a reference sample&quot;</span></span>
<span id="cb70-71"><a href="statistical-analysis-1.html#cb70-71" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;CompNorm&quot;</span>) {</span>
<span id="cb70-72"><a href="statistical-analysis-1.html#cb70-72" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(<span class="fu">t</span>(data), <span class="dv">1</span>, CompNorm, ref))</span>
<span id="cb70-73"><a href="statistical-analysis-1.html#cb70-73" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Normalization by a reference feature&quot;</span>;</span>
<span id="cb70-74"><a href="statistical-analysis-1.html#cb70-74" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;SumNorm&quot;</span>) {</span>
<span id="cb70-75"><a href="statistical-analysis-1.html#cb70-75" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(data, <span class="dv">1</span>, SumNorm))</span>
<span id="cb70-76"><a href="statistical-analysis-1.html#cb70-76" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Normalization to constant sum&quot;</span></span>
<span id="cb70-77"><a href="statistical-analysis-1.html#cb70-77" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (rowNorm <span class="sc">==</span> <span class="st">&quot;MedianNorm&quot;</span>) {</span>
<span id="cb70-78"><a href="statistical-analysis-1.html#cb70-78" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">apply</span>(data, <span class="dv">1</span>, MedianNorm))</span>
<span id="cb70-79"><a href="statistical-analysis-1.html#cb70-79" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Normalization to sample median&quot;</span></span>
<span id="cb70-80"><a href="statistical-analysis-1.html#cb70-80" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span>(rowNorm <span class="sc">==</span> <span class="st">&quot;SpecNorm&quot;</span>) {</span>
<span id="cb70-81"><a href="statistical-analysis-1.html#cb70-81" tabindex="-1"></a>    norm.vec <span class="ot">&lt;-</span> <span class="fu">rep</span>(SpeWeight, <span class="fu">nrow</span>(data)) <span class="co"># default all same weight vec to prevent error</span></span>
<span id="cb70-82"><a href="statistical-analysis-1.html#cb70-82" tabindex="-1"></a>    data <span class="ot">&lt;-</span> data <span class="sc">/</span> norm.vec</span>
<span id="cb70-83"><a href="statistical-analysis-1.html#cb70-83" tabindex="-1"></a>    <span class="fu">message</span>(<span class="st">&quot;No sample specific information were given, all set to 1.0&quot;</span>)</span>
<span id="cb70-84"><a href="statistical-analysis-1.html#cb70-84" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;Normalization by sample-specific factor&quot;</span></span>
<span id="cb70-85"><a href="statistical-analysis-1.html#cb70-85" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb70-86"><a href="statistical-analysis-1.html#cb70-86" tabindex="-1"></a>    <span class="co"># nothing to do</span></span>
<span id="cb70-87"><a href="statistical-analysis-1.html#cb70-87" tabindex="-1"></a>    rownm <span class="ot">&lt;-</span> <span class="st">&quot;N/A&quot;</span></span>
<span id="cb70-88"><a href="statistical-analysis-1.html#cb70-88" tabindex="-1"></a>  }</span>
<span id="cb70-89"><a href="statistical-analysis-1.html#cb70-89" tabindex="-1"></a>  <span class="do">################################################ </span></span>
<span id="cb70-90"><a href="statistical-analysis-1.html#cb70-90" tabindex="-1"></a>  </span>
<span id="cb70-91"><a href="statistical-analysis-1.html#cb70-91" tabindex="-1"></a>  <span class="co"># use apply will lose dimension info (i.e. row names and colnames)</span></span>
<span id="cb70-92"><a href="statistical-analysis-1.html#cb70-92" tabindex="-1"></a>  <span class="fu">rownames</span>(data) <span class="ot">&lt;-</span> rowNames</span>
<span id="cb70-93"><a href="statistical-analysis-1.html#cb70-93" tabindex="-1"></a>  <span class="fu">colnames</span>(data) <span class="ot">&lt;-</span> colNames</span>
<span id="cb70-94"><a href="statistical-analysis-1.html#cb70-94" tabindex="-1"></a>  </span>
<span id="cb70-95"><a href="statistical-analysis-1.html#cb70-95" tabindex="-1"></a>  <span class="co"># if the reference by feature, the feature column should be removed, since it is all 1</span></span>
<span id="cb70-96"><a href="statistical-analysis-1.html#cb70-96" tabindex="-1"></a>  <span class="cf">if</span>(rowNorm <span class="sc">==</span> <span class="st">&quot;CompNorm&quot;</span> <span class="sc">&amp;&amp;</span> <span class="sc">!</span><span class="fu">is.null</span>(ref)){</span>
<span id="cb70-97"><a href="statistical-analysis-1.html#cb70-97" tabindex="-1"></a>    inx <span class="ot">&lt;-</span> <span class="fu">match</span>(ref, <span class="fu">colnames</span>(data))</span>
<span id="cb70-98"><a href="statistical-analysis-1.html#cb70-98" tabindex="-1"></a>    data <span class="ot">&lt;-</span> data[, <span class="sc">-</span>inx, drop<span class="ot">=</span><span class="cn">FALSE</span>]</span>
<span id="cb70-99"><a href="statistical-analysis-1.html#cb70-99" tabindex="-1"></a>    colNames <span class="ot">&lt;-</span> colNames[<span class="sc">-</span>inx]</span>
<span id="cb70-100"><a href="statistical-analysis-1.html#cb70-100" tabindex="-1"></a>  }</span>
<span id="cb70-101"><a href="statistical-analysis-1.html#cb70-101" tabindex="-1"></a>  </span>
<span id="cb70-102"><a href="statistical-analysis-1.html#cb70-102" tabindex="-1"></a>  <span class="do">#############################################</span></span>
<span id="cb70-103"><a href="statistical-analysis-1.html#cb70-103" tabindex="-1"></a>  <span class="co">#  Data transformation</span></span>
<span id="cb70-104"><a href="statistical-analysis-1.html#cb70-104" tabindex="-1"></a>  <span class="co"># generalize log, tolerant to 0 and negative values</span></span>
<span id="cb70-105"><a href="statistical-analysis-1.html#cb70-105" tabindex="-1"></a>  LogNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x, min.val) {</span>
<span id="cb70-106"><a href="statistical-analysis-1.html#cb70-106" tabindex="-1"></a>    <span class="fu">return</span>(<span class="fu">log10</span>((x <span class="sc">+</span> <span class="fu">sqrt</span>(x<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> min.val<span class="sc">^</span><span class="dv">2</span>))<span class="sc">/</span><span class="dv">2</span>))</span>
<span id="cb70-107"><a href="statistical-analysis-1.html#cb70-107" tabindex="-1"></a>  }</span>
<span id="cb70-108"><a href="statistical-analysis-1.html#cb70-108" tabindex="-1"></a>  </span>
<span id="cb70-109"><a href="statistical-analysis-1.html#cb70-109" tabindex="-1"></a>  <span class="co"># square root, tolerant to negative values</span></span>
<span id="cb70-110"><a href="statistical-analysis-1.html#cb70-110" tabindex="-1"></a>  SquareRootNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x, min.val) {</span>
<span id="cb70-111"><a href="statistical-analysis-1.html#cb70-111" tabindex="-1"></a>    <span class="fu">return</span>(((x <span class="sc">+</span> <span class="fu">sqrt</span>(x<span class="sc">^</span><span class="dv">2</span> <span class="sc">+</span> min.val<span class="sc">^</span><span class="dv">2</span>))<span class="sc">/</span><span class="dv">2</span>)<span class="sc">^</span>(<span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>))</span>
<span id="cb70-112"><a href="statistical-analysis-1.html#cb70-112" tabindex="-1"></a>  }</span>
<span id="cb70-113"><a href="statistical-analysis-1.html#cb70-113" tabindex="-1"></a>  </span>
<span id="cb70-114"><a href="statistical-analysis-1.html#cb70-114" tabindex="-1"></a>  <span class="cf">if</span> (transNorm <span class="sc">==</span> <span class="st">&quot;LogNorm&quot;</span>) {</span>
<span id="cb70-115"><a href="statistical-analysis-1.html#cb70-115" tabindex="-1"></a>    min.val <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">abs</span>(data[data <span class="sc">!=</span> <span class="dv">0</span>]))<span class="sc">/</span><span class="dv">10</span></span>
<span id="cb70-116"><a href="statistical-analysis-1.html#cb70-116" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, LogNorm, min.val)</span>
<span id="cb70-117"><a href="statistical-analysis-1.html#cb70-117" tabindex="-1"></a>    transnm <span class="ot">&lt;-</span> <span class="st">&quot;Log10 Normalization&quot;</span></span>
<span id="cb70-118"><a href="statistical-analysis-1.html#cb70-118" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (transNorm <span class="sc">==</span> <span class="st">&quot;SrNorm&quot;</span>) {</span>
<span id="cb70-119"><a href="statistical-analysis-1.html#cb70-119" tabindex="-1"></a>    min.val <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">abs</span>(data[data <span class="sc">!=</span> <span class="dv">0</span>]))<span class="sc">/</span><span class="dv">10</span></span>
<span id="cb70-120"><a href="statistical-analysis-1.html#cb70-120" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, SquareRootNorm, min.val)</span>
<span id="cb70-121"><a href="statistical-analysis-1.html#cb70-121" tabindex="-1"></a>    transnm <span class="ot">&lt;-</span> <span class="st">&quot;Square Root Transformation&quot;</span></span>
<span id="cb70-122"><a href="statistical-analysis-1.html#cb70-122" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (transNorm <span class="sc">==</span> <span class="st">&quot;CrNorm&quot;</span>) {</span>
<span id="cb70-123"><a href="statistical-analysis-1.html#cb70-123" tabindex="-1"></a>    norm.data <span class="ot">&lt;-</span> <span class="fu">abs</span>(data)<span class="sc">^</span>(<span class="dv">1</span><span class="sc">/</span><span class="dv">3</span>)</span>
<span id="cb70-124"><a href="statistical-analysis-1.html#cb70-124" tabindex="-1"></a>    norm.data[data <span class="sc">&lt;</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="sc">-</span>norm.data[data <span class="sc">&lt;</span> <span class="dv">0</span>]</span>
<span id="cb70-125"><a href="statistical-analysis-1.html#cb70-125" tabindex="-1"></a>    data <span class="ot">&lt;-</span> norm.data</span>
<span id="cb70-126"><a href="statistical-analysis-1.html#cb70-126" tabindex="-1"></a>    transnm <span class="ot">&lt;-</span> <span class="st">&quot;Cubic Root Transformation&quot;</span></span>
<span id="cb70-127"><a href="statistical-analysis-1.html#cb70-127" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb70-128"><a href="statistical-analysis-1.html#cb70-128" tabindex="-1"></a>    transnm <span class="ot">&lt;-</span> <span class="st">&quot;N/A&quot;</span></span>
<span id="cb70-129"><a href="statistical-analysis-1.html#cb70-129" tabindex="-1"></a>  }</span>
<span id="cb70-130"><a href="statistical-analysis-1.html#cb70-130" tabindex="-1"></a>  <span class="do">#############################################</span></span>
<span id="cb70-131"><a href="statistical-analysis-1.html#cb70-131" tabindex="-1"></a>  </span>
<span id="cb70-132"><a href="statistical-analysis-1.html#cb70-132" tabindex="-1"></a>  <span class="do">#############################################</span></span>
<span id="cb70-133"><a href="statistical-analysis-1.html#cb70-133" tabindex="-1"></a>  <span class="co"># Data scaling</span></span>
<span id="cb70-134"><a href="statistical-analysis-1.html#cb70-134" tabindex="-1"></a>  <span class="co"># normalize to zero mean and unit variance</span></span>
<span id="cb70-135"><a href="statistical-analysis-1.html#cb70-135" tabindex="-1"></a>  AutoNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb70-136"><a href="statistical-analysis-1.html#cb70-136" tabindex="-1"></a>    <span class="fu">return</span>((x <span class="sc">-</span> <span class="fu">mean</span>(x))<span class="sc">/</span><span class="fu">sd</span>(x, <span class="at">na.rm =</span> T))</span>
<span id="cb70-137"><a href="statistical-analysis-1.html#cb70-137" tabindex="-1"></a>  }</span>
<span id="cb70-138"><a href="statistical-analysis-1.html#cb70-138" tabindex="-1"></a>  </span>
<span id="cb70-139"><a href="statistical-analysis-1.html#cb70-139" tabindex="-1"></a>  <span class="co"># normalize to zero mean but variance/SE</span></span>
<span id="cb70-140"><a href="statistical-analysis-1.html#cb70-140" tabindex="-1"></a>  ParetoNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb70-141"><a href="statistical-analysis-1.html#cb70-141" tabindex="-1"></a>    <span class="fu">return</span>((x <span class="sc">-</span> <span class="fu">mean</span>(x))<span class="sc">/</span><span class="fu">sqrt</span>(<span class="fu">sd</span>(x, <span class="at">na.rm =</span> T)))</span>
<span id="cb70-142"><a href="statistical-analysis-1.html#cb70-142" tabindex="-1"></a>  }</span>
<span id="cb70-143"><a href="statistical-analysis-1.html#cb70-143" tabindex="-1"></a>  </span>
<span id="cb70-144"><a href="statistical-analysis-1.html#cb70-144" tabindex="-1"></a>  <span class="co"># normalize to zero mean but variance/SE</span></span>
<span id="cb70-145"><a href="statistical-analysis-1.html#cb70-145" tabindex="-1"></a>  MeanCenter <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb70-146"><a href="statistical-analysis-1.html#cb70-146" tabindex="-1"></a>    <span class="fu">return</span>(x <span class="sc">-</span> <span class="fu">mean</span>(x))</span>
<span id="cb70-147"><a href="statistical-analysis-1.html#cb70-147" tabindex="-1"></a>  }</span>
<span id="cb70-148"><a href="statistical-analysis-1.html#cb70-148" tabindex="-1"></a>  </span>
<span id="cb70-149"><a href="statistical-analysis-1.html#cb70-149" tabindex="-1"></a>  <span class="co"># normalize to zero mean but variance/SE</span></span>
<span id="cb70-150"><a href="statistical-analysis-1.html#cb70-150" tabindex="-1"></a>  RangeNorm <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb70-151"><a href="statistical-analysis-1.html#cb70-151" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">max</span>(x) <span class="sc">==</span> <span class="fu">min</span>(x)) {</span>
<span id="cb70-152"><a href="statistical-analysis-1.html#cb70-152" tabindex="-1"></a>      <span class="fu">return</span>(x)</span>
<span id="cb70-153"><a href="statistical-analysis-1.html#cb70-153" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb70-154"><a href="statistical-analysis-1.html#cb70-154" tabindex="-1"></a>      <span class="fu">return</span>((x <span class="sc">-</span> <span class="fu">mean</span>(x))<span class="sc">/</span>(<span class="fu">max</span>(x) <span class="sc">-</span> <span class="fu">min</span>(x)))</span>
<span id="cb70-155"><a href="statistical-analysis-1.html#cb70-155" tabindex="-1"></a>    }</span>
<span id="cb70-156"><a href="statistical-analysis-1.html#cb70-156" tabindex="-1"></a>  }</span>
<span id="cb70-157"><a href="statistical-analysis-1.html#cb70-157" tabindex="-1"></a></span>
<span id="cb70-158"><a href="statistical-analysis-1.html#cb70-158" tabindex="-1"></a>  <span class="cf">if</span> (scaleNorm <span class="sc">==</span> <span class="st">&quot;MeanCenter&quot;</span>) {</span>
<span id="cb70-159"><a href="statistical-analysis-1.html#cb70-159" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, MeanCenter)</span>
<span id="cb70-160"><a href="statistical-analysis-1.html#cb70-160" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;Mean Centering&quot;</span></span>
<span id="cb70-161"><a href="statistical-analysis-1.html#cb70-161" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (scaleNorm <span class="sc">==</span> <span class="st">&quot;AutoNorm&quot;</span>) {</span>
<span id="cb70-162"><a href="statistical-analysis-1.html#cb70-162" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, AutoNorm)</span>
<span id="cb70-163"><a href="statistical-analysis-1.html#cb70-163" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;Autoscaling&quot;</span></span>
<span id="cb70-164"><a href="statistical-analysis-1.html#cb70-164" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (scaleNorm <span class="sc">==</span> <span class="st">&quot;ParetoNorm&quot;</span>) {</span>
<span id="cb70-165"><a href="statistical-analysis-1.html#cb70-165" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, ParetoNorm)</span>
<span id="cb70-166"><a href="statistical-analysis-1.html#cb70-166" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;Pareto Scaling&quot;</span></span>
<span id="cb70-167"><a href="statistical-analysis-1.html#cb70-167" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (scaleNorm <span class="sc">==</span> <span class="st">&quot;RangeNorm&quot;</span>) {</span>
<span id="cb70-168"><a href="statistical-analysis-1.html#cb70-168" tabindex="-1"></a>    data <span class="ot">&lt;-</span> <span class="fu">apply</span>(data, <span class="dv">2</span>, RangeNorm)</span>
<span id="cb70-169"><a href="statistical-analysis-1.html#cb70-169" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;Range Scaling&quot;</span></span>
<span id="cb70-170"><a href="statistical-analysis-1.html#cb70-170" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb70-171"><a href="statistical-analysis-1.html#cb70-171" tabindex="-1"></a>    scalenm <span class="ot">&lt;-</span> <span class="st">&quot;N/A&quot;</span></span>
<span id="cb70-172"><a href="statistical-analysis-1.html#cb70-172" tabindex="-1"></a>  }</span>
<span id="cb70-173"><a href="statistical-analysis-1.html#cb70-173" tabindex="-1"></a>  <span class="do">############################################# </span></span>
<span id="cb70-174"><a href="statistical-analysis-1.html#cb70-174" tabindex="-1"></a>  </span>
<span id="cb70-175"><a href="statistical-analysis-1.html#cb70-175" tabindex="-1"></a>  <span class="fu">message</span>(<span class="st">&quot;Row norm: &quot;</span>, rownm, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>,</span>
<span id="cb70-176"><a href="statistical-analysis-1.html#cb70-176" tabindex="-1"></a>          <span class="st">&quot;Data Transformation norm: &quot;</span>, transnm, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>,</span>
<span id="cb70-177"><a href="statistical-analysis-1.html#cb70-177" tabindex="-1"></a>          <span class="st">&quot;Data Scaling norm: &quot;</span>, scalenm, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)  </span>
<span id="cb70-178"><a href="statistical-analysis-1.html#cb70-178" tabindex="-1"></a>  </span>
<span id="cb70-179"><a href="statistical-analysis-1.html#cb70-179" tabindex="-1"></a>  <span class="co"># note after using &quot;apply&quot; function, all the attribute lost, need to add back</span></span>
<span id="cb70-180"><a href="statistical-analysis-1.html#cb70-180" tabindex="-1"></a>  <span class="fu">rownames</span>(data) <span class="ot">&lt;-</span> rowNames</span>
<span id="cb70-181"><a href="statistical-analysis-1.html#cb70-181" tabindex="-1"></a>  <span class="fu">colnames</span>(data) <span class="ot">&lt;-</span> colNames</span>
<span id="cb70-182"><a href="statistical-analysis-1.html#cb70-182" tabindex="-1"></a>  </span>
<span id="cb70-183"><a href="statistical-analysis-1.html#cb70-183" tabindex="-1"></a>  target <span class="ot">&lt;-</span> metadata_tab <span class="sc">%&gt;%</span> </span>
<span id="cb70-184"><a href="statistical-analysis-1.html#cb70-184" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb70-185"><a href="statistical-analysis-1.html#cb70-185" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;SampleID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb70-186"><a href="statistical-analysis-1.html#cb70-186" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(SampleID<span class="sc">%in%</span><span class="fu">rownames</span>(data))</span>
<span id="cb70-187"><a href="statistical-analysis-1.html#cb70-187" tabindex="-1"></a>  se <span class="ot">&lt;-</span> <span class="fu">PomaSummarizedExperiment</span>(<span class="at">target =</span> target, </span>
<span id="cb70-188"><a href="statistical-analysis-1.html#cb70-188" tabindex="-1"></a>                                 <span class="at">features =</span> data)</span>
<span id="cb70-189"><a href="statistical-analysis-1.html#cb70-189" tabindex="-1"></a>  </span>
<span id="cb70-190"><a href="statistical-analysis-1.html#cb70-190" tabindex="-1"></a>  <span class="co"># need to do some sanity check, for log there may be Inf values introduced</span></span>
<span id="cb70-191"><a href="statistical-analysis-1.html#cb70-191" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">CheckData</span>(se)</span>
<span id="cb70-192"><a href="statistical-analysis-1.html#cb70-192" tabindex="-1"></a>  </span>
<span id="cb70-193"><a href="statistical-analysis-1.html#cb70-193" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb70-194"><a href="statistical-analysis-1.html#cb70-194" tabindex="-1"></a>}</span>
<span id="cb70-195"><a href="statistical-analysis-1.html#cb70-195" tabindex="-1"></a></span>
<span id="cb70-196"><a href="statistical-analysis-1.html#cb70-196" tabindex="-1"></a>se_normalize <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(</span>
<span id="cb70-197"><a href="statistical-analysis-1.html#cb70-197" tabindex="-1"></a>                      <span class="at">object =</span> se_filter,</span>
<span id="cb70-198"><a href="statistical-analysis-1.html#cb70-198" tabindex="-1"></a>                      <span class="at">rowNorm =</span> <span class="st">&quot;None&quot;</span>,</span>
<span id="cb70-199"><a href="statistical-analysis-1.html#cb70-199" tabindex="-1"></a>                      <span class="at">transNorm =</span> <span class="st">&quot;LogNorm&quot;</span>,</span>
<span id="cb70-200"><a href="statistical-analysis-1.html#cb70-200" tabindex="-1"></a>                      <span class="at">scaleNorm =</span> <span class="st">&quot;ParetoNorm&quot;</span>)</span></code></pre></div>
<pre><code>## Row norm: N/A
## Data Transformation norm: Log10 Normalization
## Data Scaling norm: Pareto Scaling</code></pre>
<pre><code>## All data values are numeric.</code></pre>
<pre><code>## A total of 0 (0%) missing values were detected.</code></pre>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="statistical-analysis-1.html#cb74-1" tabindex="-1"></a>se_normalize</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 167 45 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(167): M_38768 M_38296 ... M_31787 M_63361
## rowData names(0):
## colnames(45): P101001 P101004 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="normalization-by-poma-r-package-1" class="section level4 hasAnchor" number="3.1.7.2">
<h4><span class="header-section-number">3.1.7.2</span> Normalization by <strong>POMA</strong> R package<a href="statistical-analysis-1.html#normalization-by-poma-r-package-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="statistical-analysis-1.html#cb76-1" tabindex="-1"></a>none <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_filter, <span class="at">method =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb76-2"><a href="statistical-analysis-1.html#cb76-2" tabindex="-1"></a>auto_scaling <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_filter, <span class="at">method =</span> <span class="st">&quot;auto_scaling&quot;</span>)</span>
<span id="cb76-3"><a href="statistical-analysis-1.html#cb76-3" tabindex="-1"></a>evel_scaling <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_filter, <span class="at">method =</span> <span class="st">&quot;level_scaling&quot;</span>)</span>
<span id="cb76-4"><a href="statistical-analysis-1.html#cb76-4" tabindex="-1"></a>log_scaling <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_filter, <span class="at">method =</span> <span class="st">&quot;log_scaling&quot;</span>)</span>
<span id="cb76-5"><a href="statistical-analysis-1.html#cb76-5" tabindex="-1"></a>log_transformation <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_filter, <span class="at">method =</span> <span class="st">&quot;log_transformation&quot;</span>)</span>
<span id="cb76-6"><a href="statistical-analysis-1.html#cb76-6" tabindex="-1"></a>vast_scaling <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_filter, <span class="at">method =</span> <span class="st">&quot;vast_scaling&quot;</span>)</span>
<span id="cb76-7"><a href="statistical-analysis-1.html#cb76-7" tabindex="-1"></a>se_normalize_v2 <span class="ot">&lt;-</span> <span class="fu">PomaNorm</span>(se_filter, <span class="at">method =</span> <span class="st">&quot;log_pareto&quot;</span>)</span>
<span id="cb76-8"><a href="statistical-analysis-1.html#cb76-8" tabindex="-1"></a>se_normalize_v2</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 167 45 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(167): M_38768 M_38296 ... M_31787 M_63361
## rowData names(0):
## colnames(45): P101001 P101004 ... P101095 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="comparison-of-unnormalized-and-normalized-dataset-1" class="section level4 hasAnchor" number="3.1.7.3">
<h4><span class="header-section-number">3.1.7.3</span> Comparison of unnormalized and normalized dataset<a href="statistical-analysis-1.html#comparison-of-unnormalized-and-normalized-dataset-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>boxplot</li>
</ul>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="statistical-analysis-1.html#cb78-1" tabindex="-1"></a>pl_unnor <span class="ot">&lt;-</span> <span class="fu">PomaBoxplots</span>(se_filter, <span class="at">group =</span> <span class="st">&quot;samples&quot;</span>, <span class="at">jitter =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb78-2"><a href="statistical-analysis-1.html#cb78-2" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Not Normalized&quot;</span>) <span class="sc">+</span></span>
<span id="cb78-3"><a href="statistical-analysis-1.html#cb78-3" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="co"># data before normalization</span></span>
<span id="cb78-4"><a href="statistical-analysis-1.html#cb78-4" tabindex="-1"></a></span>
<span id="cb78-5"><a href="statistical-analysis-1.html#cb78-5" tabindex="-1"></a>pl_nor <span class="ot">&lt;-</span> <span class="fu">PomaBoxplots</span>(se_normalize, <span class="at">group =</span> <span class="st">&quot;samples&quot;</span>, <span class="at">jitter =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb78-6"><a href="statistical-analysis-1.html#cb78-6" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Normalized&quot;</span>) <span class="co"># data after normalization</span></span>
<span id="cb78-7"><a href="statistical-analysis-1.html#cb78-7" tabindex="-1"></a></span>
<span id="cb78-8"><a href="statistical-analysis-1.html#cb78-8" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(pl_unnor, pl_nor, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">align =</span> <span class="st">&quot;v&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-105-1.png" width="1344" /></p>
<ul>
<li>density</li>
</ul>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="statistical-analysis-1.html#cb79-1" tabindex="-1"></a>pl_unnor <span class="ot">&lt;-</span> <span class="fu">PomaDensity</span>(se_filter, <span class="at">group =</span> <span class="st">&quot;features&quot;</span>) <span class="sc">+</span></span>
<span id="cb79-2"><a href="statistical-analysis-1.html#cb79-2" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Not Normalized&quot;</span>) <span class="sc">+</span></span>
<span id="cb79-3"><a href="statistical-analysis-1.html#cb79-3" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>) <span class="co"># data before normalization</span></span>
<span id="cb79-4"><a href="statistical-analysis-1.html#cb79-4" tabindex="-1"></a></span>
<span id="cb79-5"><a href="statistical-analysis-1.html#cb79-5" tabindex="-1"></a>pl_nor <span class="ot">&lt;-</span> <span class="fu">PomaDensity</span>(se_normalize, <span class="at">group =</span> <span class="st">&quot;features&quot;</span>) <span class="sc">+</span></span>
<span id="cb79-6"><a href="statistical-analysis-1.html#cb79-6" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Normalized&quot;</span>) <span class="co"># data after normalization</span></span>
<span id="cb79-7"><a href="statistical-analysis-1.html#cb79-7" tabindex="-1"></a></span>
<span id="cb79-8"><a href="statistical-analysis-1.html#cb79-8" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(pl_unnor, pl_nor, <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">align =</span> <span class="st">&quot;v&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-106-1.png" width="960" /></p>
</div>
</div>
<div id="removing-outliers-1" class="section level3 hasAnchor" number="3.1.8">
<h3><span class="header-section-number">3.1.8</span> Removing outliers<a href="statistical-analysis-1.html#removing-outliers-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="statistical-analysis-1.html#cb80-1" tabindex="-1"></a><span class="fu">PomaOutliers</span>(se_normalize, <span class="at">do =</span> <span class="st">&quot;analyze&quot;</span>)<span class="sc">$</span>polygon_plot <span class="co"># to explore</span></span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-107-1.png" width="960" /></p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="statistical-analysis-1.html#cb81-1" tabindex="-1"></a>se_processed <span class="ot">&lt;-</span> <span class="fu">PomaOutliers</span>(se_normalize, <span class="at">do =</span> <span class="st">&quot;clean&quot;</span>) <span class="co"># to remove outliers</span></span>
<span id="cb81-2"><a href="statistical-analysis-1.html#cb81-2" tabindex="-1"></a>se_processed</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 167 42 
## metadata(0):
## assays(1): &#39;&#39;
## rownames(167): M_38768 M_38296 ... M_31787 M_63361
## rowData names(0):
## colnames(42): P101001 P101004 ... P101094 P101096
## colData names(5): group Gender Smoker Age AlcoholConsumption</code></pre>
</div>
<div id="saving-datasets-into-rds-files-1" class="section level3 hasAnchor" number="3.1.9">
<h3><span class="header-section-number">3.1.9</span> Saving datasets into RDS files<a href="statistical-analysis-1.html#saving-datasets-into-rds-files-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="statistical-analysis-1.html#cb83-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="st">&quot;./dataset/POMA/&quot;</span>)) {</span>
<span id="cb83-2"><a href="statistical-analysis-1.html#cb83-2" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="st">&quot;./dataset/POMA/&quot;</span>)</span>
<span id="cb83-3"><a href="statistical-analysis-1.html#cb83-3" tabindex="-1"></a>}</span>
<span id="cb83-4"><a href="statistical-analysis-1.html#cb83-4" tabindex="-1"></a></span>
<span id="cb83-5"><a href="statistical-analysis-1.html#cb83-5" tabindex="-1"></a><span class="fu">saveRDS</span>(ExprSet, <span class="st">&quot;./dataset/POMA/ExprSet_raw.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span>
<span id="cb83-6"><a href="statistical-analysis-1.html#cb83-6" tabindex="-1"></a><span class="fu">saveRDS</span>(se_raw, <span class="st">&quot;./dataset/POMA/se_raw.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span>
<span id="cb83-7"><a href="statistical-analysis-1.html#cb83-7" tabindex="-1"></a><span class="fu">saveRDS</span>(se_check, <span class="st">&quot;./dataset/POMA/se_check.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span>
<span id="cb83-8"><a href="statistical-analysis-1.html#cb83-8" tabindex="-1"></a><span class="fu">saveRDS</span>(se_filter, <span class="st">&quot;./dataset/POMA/se_filter.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span>
<span id="cb83-9"><a href="statistical-analysis-1.html#cb83-9" tabindex="-1"></a><span class="fu">saveRDS</span>(se_impute, <span class="st">&quot;./dataset/POMA/se_impute.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span>
<span id="cb83-10"><a href="statistical-analysis-1.html#cb83-10" tabindex="-1"></a><span class="fu">saveRDS</span>(se_normalize, <span class="st">&quot;./dataset/POMA/se_normalize.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span>
<span id="cb83-11"><a href="statistical-analysis-1.html#cb83-11" tabindex="-1"></a><span class="fu">saveRDS</span>(se_processed, <span class="st">&quot;./dataset/POMA/se_processed.RDS&quot;</span>, <span class="at">compress =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<hr />
</div>
</div>
<div id="cluster-analysis" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Cluster Analysis<a href="statistical-analysis-1.html#cluster-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<blockquote>
<p>Hierarchical clustering is an alternative approach to k-means clustering for identifying groups in the dataset. It does not require us to pre-specify the number of clusters to be generated as is required by the k-means approach. Furthermore, hierarchical clustering has an added advantage over K-means clustering in that it results in an attractive tree-based representation of the observations, called a dendrogram.</p>
</blockquote>
<p><strong>Note: Please remember to <a href="statistical-analysis-1.html#data-preprocessing">preprocess</a> your data before clustering</strong></p>
<div id="loading-packages-1" class="section level3 hasAnchor" number="3.2.1">
<h3><span class="header-section-number">3.2.1</span> Loading packages<a href="statistical-analysis-1.html#loading-packages-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="statistical-analysis-1.html#cb84-1" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">warning =</span> F)</span>
<span id="cb84-2"><a href="statistical-analysis-1.html#cb84-2" tabindex="-1"></a></span>
<span id="cb84-3"><a href="statistical-analysis-1.html#cb84-3" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb84-4"><a href="statistical-analysis-1.html#cb84-4" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb84-5"><a href="statistical-analysis-1.html#cb84-5" tabindex="-1"></a><span class="fu">library</span>(POMA)</span>
<span id="cb84-6"><a href="statistical-analysis-1.html#cb84-6" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb84-7"><a href="statistical-analysis-1.html#cb84-7" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb84-8"><a href="statistical-analysis-1.html#cb84-8" tabindex="-1"></a><span class="fu">library</span>(cluster)    <span class="co"># clustering algorithms</span></span>
<span id="cb84-9"><a href="statistical-analysis-1.html#cb84-9" tabindex="-1"></a><span class="fu">library</span>(factoextra) <span class="co"># clustering visualization</span></span>
<span id="cb84-10"><a href="statistical-analysis-1.html#cb84-10" tabindex="-1"></a><span class="fu">library</span>(dendextend) <span class="co"># for comparing two dendrograms</span></span>
<span id="cb84-11"><a href="statistical-analysis-1.html#cb84-11" tabindex="-1"></a></span>
<span id="cb84-12"><a href="statistical-analysis-1.html#cb84-12" tabindex="-1"></a><span class="co"># rm(list = ls())</span></span>
<span id="cb84-13"><a href="statistical-analysis-1.html#cb84-13" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb84-14"><a href="statistical-analysis-1.html#cb84-14" tabindex="-1"></a><span class="fu">options</span>(<span class="at">future.globals.maxSize =</span> <span class="dv">1000</span> <span class="sc">*</span> <span class="dv">1024</span><span class="sc">^</span><span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="importing-data-1" class="section level3 hasAnchor" number="3.2.2">
<h3><span class="header-section-number">3.2.2</span> Importing data<a href="statistical-analysis-1.html#importing-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The input data sets are from the previous chapter.</p>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="statistical-analysis-1.html#cb85-1" tabindex="-1"></a>se_normalize <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;./dataset/POMA/se_normalize.RDS&quot;</span>)</span></code></pre></div>
</div>
<div id="hierarchical-clustering" class="section level3 hasAnchor" number="3.2.3">
<h3><span class="header-section-number">3.2.3</span> Hierarchical Clustering<a href="statistical-analysis-1.html#hierarchical-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Hierarchical clustering can be divided into two main types: <a href="statistical-analysis-1.html#agglomerative-hierarchical-clustering"><strong>agglomerative</strong></a> and <a href="statistical-analysis-1.html#divisive-hierarchical-clustering"><strong>divisive</strong></a>.</p>
<p>Calculate dissimilarity</p>
<blockquote>
<p>However, a bigger question is: How do we measure the dissimilarity between two clusters of observations? A number of different cluster agglomeration methods (i.e, linkage methods) have been developed to answer to this question. The most common types methods are:</p>
<ol style="list-style-type: decimal">
<li><strong>Maximum or complete linkage clustering</strong>: It computes all pairwise dissimilarities between the elements in cluster 1 and the elements in cluster 2, and considers the largest value (i.e., maximum value) of these dissimilarities as the distance between the two clusters. It tends to produce more compact clusters.</li>
</ol>
</blockquote>
<ol start="2" style="list-style-type: decimal">
<li><p><strong>Minimum or single linkage clustering</strong>: It computes all pairwise dissimilarities between the elements in cluster 1 and the elements in cluster 2, and considers the smallest of these dissimilarities as a linkage criterion. It tends to produce long, “loose” clusters.</p></li>
<li><p><strong>Mean or average linkage clustering</strong>: It computes all pairwise dissimilarities between the elements in cluster 1 and the elements in cluster 2, and considers the average of these dissimilarities as the distance between the two clusters.</p></li>
<li><p><strong>Centroid linkage clustering</strong>: It computes the dissimilarity between the centroid for cluster 1 (a mean vector of length p variables) and the centroid for cluster 2.</p></li>
<li><p><strong>Ward’s minimum variance method</strong>: It minimizes the total within-cluster variance. At each step the pair of clusters with minimum between-cluster distance are merged.</p></li>
</ol>
<p>Data Processing:</p>
<ol style="list-style-type: decimal">
<li><p>Rows are observations (individuals) and columns are variables.</p></li>
<li><p>Any missing value in the data must be removed or estimated.</p></li>
<li><p>The data must be standardized (i.e., scaled) to make variables comparable. Recall that, standardization consists of transforming the variables such that they have mean zero and standard deviation one.</p></li>
</ol>
<p>Functions to computing hierarchical clustering:</p>
<ol style="list-style-type: decimal">
<li><p><em>hclust</em> [in stats package] and <em>agnes</em> [in cluster package] for agglomerative hierarchical clustering.</p></li>
<li><p><em>diana</em> [in cluster package] for divisive hierarchical clustering.</p></li>
</ol>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="statistical-analysis-1.html#cb86-1" tabindex="-1"></a>HieraCluster <span class="ot">&lt;-</span> <span class="cf">function</span>(object,</span>
<span id="cb86-2"><a href="statistical-analysis-1.html#cb86-2" tabindex="-1"></a>                         <span class="at">method_dis =</span> <span class="fu">c</span>(<span class="st">&quot;euclidean&quot;</span>, <span class="st">&quot;bray&quot;</span>),</span>
<span id="cb86-3"><a href="statistical-analysis-1.html#cb86-3" tabindex="-1"></a>                         <span class="at">method_cluster =</span> <span class="fu">c</span>(<span class="st">&quot;average&quot;</span>, <span class="st">&quot;single&quot;</span>, <span class="st">&quot;complete&quot;</span>, <span class="st">&quot;ward&quot;</span>, <span class="st">&quot;ward.D2&quot;</span>),</span>
<span id="cb86-4"><a href="statistical-analysis-1.html#cb86-4" tabindex="-1"></a>                         <span class="at">cluster_type =</span> <span class="fu">c</span>(<span class="st">&quot;Agglomerative&quot;</span>, <span class="st">&quot;Divisive&quot;</span>),</span>
<span id="cb86-5"><a href="statistical-analysis-1.html#cb86-5" tabindex="-1"></a>                         <span class="at">tree_num =</span> <span class="dv">4</span>) {</span>
<span id="cb86-6"><a href="statistical-analysis-1.html#cb86-6" tabindex="-1"></a>  </span>
<span id="cb86-7"><a href="statistical-analysis-1.html#cb86-7" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) </span>
<span id="cb86-8"><a href="statistical-analysis-1.html#cb86-8" tabindex="-1"></a>  metadata_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object)</span>
<span id="cb86-9"><a href="statistical-analysis-1.html#cb86-9" tabindex="-1"></a>  </span>
<span id="cb86-10"><a href="statistical-analysis-1.html#cb86-10" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">t</span>(features_tab)</span>
<span id="cb86-11"><a href="statistical-analysis-1.html#cb86-11" tabindex="-1"></a>  <span class="cf">if</span> (cluster_type <span class="sc">==</span> <span class="st">&quot;Agglomerative&quot;</span>) {</span>
<span id="cb86-12"><a href="statistical-analysis-1.html#cb86-12" tabindex="-1"></a>    <span class="co"># Agglomerative Hierarchical Clustering </span></span>
<span id="cb86-13"><a href="statistical-analysis-1.html#cb86-13" tabindex="-1"></a>    <span class="co"># Dissimilarity matrix</span></span>
<span id="cb86-14"><a href="statistical-analysis-1.html#cb86-14" tabindex="-1"></a>    d <span class="ot">&lt;-</span> <span class="fu">dist</span>(df, <span class="at">method =</span> method_dis)</span>
<span id="cb86-15"><a href="statistical-analysis-1.html#cb86-15" tabindex="-1"></a>    </span>
<span id="cb86-16"><a href="statistical-analysis-1.html#cb86-16" tabindex="-1"></a>    <span class="co"># Hierarchical clustering using Linkage method</span></span>
<span id="cb86-17"><a href="statistical-analysis-1.html#cb86-17" tabindex="-1"></a>    hc <span class="ot">&lt;-</span> <span class="fu">hclust</span>(d, <span class="at">method =</span> method_cluster)</span>
<span id="cb86-18"><a href="statistical-analysis-1.html#cb86-18" tabindex="-1"></a>    <span class="co"># hc &lt;- agnes(df, method = method_cluster)</span></span>
<span id="cb86-19"><a href="statistical-analysis-1.html#cb86-19" tabindex="-1"></a>    </span>
<span id="cb86-20"><a href="statistical-analysis-1.html#cb86-20" tabindex="-1"></a>    <span class="do">####### identifying the strongest clustering structure ################</span></span>
<span id="cb86-21"><a href="statistical-analysis-1.html#cb86-21" tabindex="-1"></a>    <span class="co"># # methods to assess</span></span>
<span id="cb86-22"><a href="statistical-analysis-1.html#cb86-22" tabindex="-1"></a>    <span class="co"># m &lt;- c( &quot;average&quot;, &quot;single&quot;, &quot;complete&quot;, &quot;ward&quot;)</span></span>
<span id="cb86-23"><a href="statistical-analysis-1.html#cb86-23" tabindex="-1"></a>    <span class="co"># names(m) &lt;- c( &quot;average&quot;, &quot;single&quot;, &quot;complete&quot;, &quot;ward&quot;)</span></span>
<span id="cb86-24"><a href="statistical-analysis-1.html#cb86-24" tabindex="-1"></a>    <span class="co"># </span></span>
<span id="cb86-25"><a href="statistical-analysis-1.html#cb86-25" tabindex="-1"></a>    <span class="co"># # function to compute coefficient</span></span>
<span id="cb86-26"><a href="statistical-analysis-1.html#cb86-26" tabindex="-1"></a>    <span class="co"># ac &lt;- function(x) {</span></span>
<span id="cb86-27"><a href="statistical-analysis-1.html#cb86-27" tabindex="-1"></a>    <span class="co">#   agnes(df, method = x)$ac</span></span>
<span id="cb86-28"><a href="statistical-analysis-1.html#cb86-28" tabindex="-1"></a>    <span class="co"># }</span></span>
<span id="cb86-29"><a href="statistical-analysis-1.html#cb86-29" tabindex="-1"></a>    <span class="co"># </span></span>
<span id="cb86-30"><a href="statistical-analysis-1.html#cb86-30" tabindex="-1"></a>    <span class="co"># map_dbl(m, ac)     </span></span>
<span id="cb86-31"><a href="statistical-analysis-1.html#cb86-31" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (cluster_type <span class="sc">==</span> <span class="st">&quot;Divisive&quot;</span>) {</span>
<span id="cb86-32"><a href="statistical-analysis-1.html#cb86-32" tabindex="-1"></a>    <span class="co"># Divisive Hierarchical Clustering </span></span>
<span id="cb86-33"><a href="statistical-analysis-1.html#cb86-33" tabindex="-1"></a>    hc <span class="ot">&lt;-</span> <span class="fu">diana</span>(df, <span class="at">metric =</span> method_dis)</span>
<span id="cb86-34"><a href="statistical-analysis-1.html#cb86-34" tabindex="-1"></a>  }</span>
<span id="cb86-35"><a href="statistical-analysis-1.html#cb86-35" tabindex="-1"></a>  </span>
<span id="cb86-36"><a href="statistical-analysis-1.html#cb86-36" tabindex="-1"></a>  hc_res <span class="ot">&lt;-</span> <span class="fu">as.hclust</span>(hc)</span>
<span id="cb86-37"><a href="statistical-analysis-1.html#cb86-37" tabindex="-1"></a>  sub_grp <span class="ot">&lt;-</span> <span class="fu">cutree</span>(hc_res, <span class="at">k =</span> tree_num)</span>
<span id="cb86-38"><a href="statistical-analysis-1.html#cb86-38" tabindex="-1"></a></span>
<span id="cb86-39"><a href="statistical-analysis-1.html#cb86-39" tabindex="-1"></a>  <span class="fu">plot</span>(hc_res, <span class="at">cex =</span> <span class="fl">0.6</span>)</span>
<span id="cb86-40"><a href="statistical-analysis-1.html#cb86-40" tabindex="-1"></a>  <span class="fu">rect.hclust</span>(hc_res, <span class="at">k =</span> tree_num, <span class="at">border =</span> <span class="dv">2</span><span class="sc">:</span>(tree_num<span class="sc">+</span><span class="dv">1</span>)) </span>
<span id="cb86-41"><a href="statistical-analysis-1.html#cb86-41" tabindex="-1"></a>  </span>
<span id="cb86-42"><a href="statistical-analysis-1.html#cb86-42" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">data=</span>df,</span>
<span id="cb86-43"><a href="statistical-analysis-1.html#cb86-43" tabindex="-1"></a>              <span class="at">cluster=</span>sub_grp,</span>
<span id="cb86-44"><a href="statistical-analysis-1.html#cb86-44" tabindex="-1"></a>              <span class="at">hc=</span>hc_res)</span>
<span id="cb86-45"><a href="statistical-analysis-1.html#cb86-45" tabindex="-1"></a>  </span>
<span id="cb86-46"><a href="statistical-analysis-1.html#cb86-46" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb86-47"><a href="statistical-analysis-1.html#cb86-47" tabindex="-1"></a>}</span></code></pre></div>
<div id="agglomerative-hierarchical-clustering" class="section level4 hasAnchor" number="3.2.3.1">
<h4><span class="header-section-number">3.2.3.1</span> Agglomerative Hierarchical Clustering<a href="statistical-analysis-1.html#agglomerative-hierarchical-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<blockquote>
<p><strong>Agglomerative clustering</strong>: It’s also known as AGNES (Agglomerative Nesting). It works in a bottom-up manner. That is, each object is initially considered as a single-element cluster (leaf). At each step of the algorithm, the two clusters that are the most similar are combined into a new bigger cluster (nodes). This procedure is iterated until all points are member of just one single big cluster (root). The result is a tree which can be plotted as a dendrogram.</p>
</blockquote>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="statistical-analysis-1.html#cb87-1" tabindex="-1"></a>Agg_hc_res <span class="ot">&lt;-</span> <span class="fu">HieraCluster</span>(</span>
<span id="cb87-2"><a href="statistical-analysis-1.html#cb87-2" tabindex="-1"></a>      <span class="at">object =</span> se_normalize,</span>
<span id="cb87-3"><a href="statistical-analysis-1.html#cb87-3" tabindex="-1"></a>      <span class="at">method_dis =</span> <span class="st">&quot;euclidean&quot;</span>,</span>
<span id="cb87-4"><a href="statistical-analysis-1.html#cb87-4" tabindex="-1"></a>      <span class="at">method_cluster =</span> <span class="st">&quot;ward.D2&quot;</span>,</span>
<span id="cb87-5"><a href="statistical-analysis-1.html#cb87-5" tabindex="-1"></a>      <span class="at">cluster_type =</span> <span class="st">&quot;Agglomerative&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-112-1.png" width="768" /></p>
<ul>
<li>Visualization: visualize the result in a scatter plot</li>
</ul>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="statistical-analysis-1.html#cb88-1" tabindex="-1"></a><span class="fu">fviz_cluster</span>(<span class="fu">list</span>(<span class="at">data =</span> Agg_hc_res<span class="sc">$</span>data, </span>
<span id="cb88-2"><a href="statistical-analysis-1.html#cb88-2" tabindex="-1"></a>                  <span class="at">cluster =</span> Agg_hc_res<span class="sc">$</span>cluster))</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-113-1.png" width="768" /></p>
</div>
<div id="divisive-hierarchical-clustering" class="section level4 hasAnchor" number="3.2.3.2">
<h4><span class="header-section-number">3.2.3.2</span> Divisive Hierarchical Clustering<a href="statistical-analysis-1.html#divisive-hierarchical-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<blockquote>
<p><strong>Divisive hierarchical clustering</strong>: It’s also known as DIANA (Divise Analysis) and it works in a top-down manner. The algorithm is an inverse order of AGNES. It begins with the root, in which all objects are included in a single cluster. At each step of iteration, the most heterogeneous cluster is divided into two. The process is iterated until all objects are in their own cluster.</p>
</blockquote>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="statistical-analysis-1.html#cb89-1" tabindex="-1"></a>Div_hc_res <span class="ot">&lt;-</span> <span class="fu">HieraCluster</span>(</span>
<span id="cb89-2"><a href="statistical-analysis-1.html#cb89-2" tabindex="-1"></a>      <span class="at">object =</span> se_normalize,</span>
<span id="cb89-3"><a href="statistical-analysis-1.html#cb89-3" tabindex="-1"></a>      <span class="at">method_dis =</span> <span class="st">&quot;euclidean&quot;</span>,</span>
<span id="cb89-4"><a href="statistical-analysis-1.html#cb89-4" tabindex="-1"></a>      <span class="at">method_cluster =</span> <span class="st">&quot;ward&quot;</span>,</span>
<span id="cb89-5"><a href="statistical-analysis-1.html#cb89-5" tabindex="-1"></a>      <span class="at">cluster_type =</span> <span class="st">&quot;Divisive&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-114-1.png" width="768" /></p>
<ul>
<li>Visualization: visualize the result in a scatter plot</li>
</ul>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="statistical-analysis-1.html#cb90-1" tabindex="-1"></a><span class="fu">fviz_cluster</span>(<span class="fu">list</span>(<span class="at">data =</span> Div_hc_res<span class="sc">$</span>data, </span>
<span id="cb90-2"><a href="statistical-analysis-1.html#cb90-2" tabindex="-1"></a>                  <span class="at">cluster =</span> Div_hc_res<span class="sc">$</span>cluster))</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-115-1.png" width="768" /></p>
</div>
<div id="comparison" class="section level4 hasAnchor" number="3.2.3.3">
<h4><span class="header-section-number">3.2.3.3</span> Comparison<a href="statistical-analysis-1.html#comparison" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>dendrograms: In the dendrogram displayed above, each leaf corresponds to one observation.</li>
</ul>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="statistical-analysis-1.html#cb91-1" tabindex="-1"></a>Agg_hc_dend <span class="ot">&lt;-</span> <span class="fu">as.dendrogram</span>(Agg_hc_res<span class="sc">$</span>hc)</span>
<span id="cb91-2"><a href="statistical-analysis-1.html#cb91-2" tabindex="-1"></a>Div_hc_dend <span class="ot">&lt;-</span> <span class="fu">as.dendrogram</span>(Div_hc_res<span class="sc">$</span>hc)</span>
<span id="cb91-3"><a href="statistical-analysis-1.html#cb91-3" tabindex="-1"></a></span>
<span id="cb91-4"><a href="statistical-analysis-1.html#cb91-4" tabindex="-1"></a><span class="fu">tanglegram</span>(Agg_hc_dend, Div_hc_dend)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-116-1.png" width="768" /></p>
<ul>
<li>tanglegrams</li>
</ul>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="statistical-analysis-1.html#cb92-1" tabindex="-1"></a>dend_list <span class="ot">&lt;-</span> <span class="fu">dendlist</span>(Agg_hc_dend, Div_hc_dend)</span>
<span id="cb92-2"><a href="statistical-analysis-1.html#cb92-2" tabindex="-1"></a><span class="fu">tanglegram</span>(Agg_hc_dend, Div_hc_dend,</span>
<span id="cb92-3"><a href="statistical-analysis-1.html#cb92-3" tabindex="-1"></a>  <span class="at">highlight_distinct_edges =</span> <span class="cn">FALSE</span>, <span class="co"># Turn-off dashed lines</span></span>
<span id="cb92-4"><a href="statistical-analysis-1.html#cb92-4" tabindex="-1"></a>  <span class="at">common_subtrees_color_lines =</span> <span class="cn">FALSE</span>, <span class="co"># Turn-off line colors</span></span>
<span id="cb92-5"><a href="statistical-analysis-1.html#cb92-5" tabindex="-1"></a>  <span class="at">common_subtrees_color_branches =</span> <span class="cn">TRUE</span>, <span class="co"># Color common branches </span></span>
<span id="cb92-6"><a href="statistical-analysis-1.html#cb92-6" tabindex="-1"></a>  <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">&quot;entanglement =&quot;</span>, <span class="fu">round</span>(<span class="fu">entanglement</span>(dend_list), <span class="dv">2</span>)))</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-117-1.png" width="768" /></p>
</div>
<div id="determining-optimal-clusters" class="section level4 hasAnchor" number="3.2.3.4">
<h4><span class="header-section-number">3.2.3.4</span> Determining Optimal Clusters<a href="statistical-analysis-1.html#determining-optimal-clusters" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Elbow plot</li>
</ul>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="statistical-analysis-1.html#cb93-1" tabindex="-1"></a><span class="fu">fviz_nbclust</span>(Agg_hc_res<span class="sc">$</span>data, <span class="at">FUN =</span> hcut, <span class="at">method =</span> <span class="st">&quot;wss&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-118-1.png" width="576" /></p>
<ul>
<li>Average Silhouette Method</li>
</ul>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="statistical-analysis-1.html#cb94-1" tabindex="-1"></a><span class="fu">fviz_nbclust</span>(Agg_hc_res<span class="sc">$</span>data, <span class="at">FUN =</span> hcut, <span class="at">method =</span> <span class="st">&quot;silhouette&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-119-1.png" width="576" /></p>
<ul>
<li>Gap Statistic Method</li>
</ul>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="statistical-analysis-1.html#cb95-1" tabindex="-1"></a>gap_stat <span class="ot">&lt;-</span> <span class="fu">clusGap</span>(Agg_hc_res<span class="sc">$</span>data, <span class="at">FUN =</span> hcut, <span class="at">nstart =</span> <span class="dv">25</span>, <span class="at">K.max =</span> <span class="dv">10</span>, <span class="at">B =</span> <span class="dv">50</span>)</span></code></pre></div>
<pre><code>## Clustering k = 1,2,..., K.max (= 10): .. done
## Bootstrapping, b = 1,2,..., B (= 50)  [one &quot;.&quot; per sample]:
## .................................................. 50</code></pre>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="statistical-analysis-1.html#cb97-1" tabindex="-1"></a><span class="fu">fviz_gap_stat</span>(gap_stat)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-120-1.png" width="576" /></p>
</div>
</div>
<div id="partitional-clustering" class="section level3 hasAnchor" number="3.2.4">
<h3><span class="header-section-number">3.2.4</span> Partitional Clustering<a href="statistical-analysis-1.html#partitional-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<blockquote>
<p>K-means clustering is the most commonly used unsupervised machine learning algorithm for partitioning a given data set into a set of k groups (i.e. k clusters), where k represents the number of groups pre-specified by the analyst.</p>
</blockquote>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="statistical-analysis-1.html#cb98-1" tabindex="-1"></a>PartCluster <span class="ot">&lt;-</span> <span class="cf">function</span>(object,</span>
<span id="cb98-2"><a href="statistical-analysis-1.html#cb98-2" tabindex="-1"></a>                        <span class="at">cluster_num =</span> <span class="dv">4</span>) {</span>
<span id="cb98-3"><a href="statistical-analysis-1.html#cb98-3" tabindex="-1"></a>  </span>
<span id="cb98-4"><a href="statistical-analysis-1.html#cb98-4" tabindex="-1"></a>  features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(object) </span>
<span id="cb98-5"><a href="statistical-analysis-1.html#cb98-5" tabindex="-1"></a>  metadata_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(object)</span>
<span id="cb98-6"><a href="statistical-analysis-1.html#cb98-6" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">t</span>(features_tab)</span>
<span id="cb98-7"><a href="statistical-analysis-1.html#cb98-7" tabindex="-1"></a>  </span>
<span id="cb98-8"><a href="statistical-analysis-1.html#cb98-8" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">kmeans</span>(df, <span class="at">centers =</span> cluster_num)  </span>
<span id="cb98-9"><a href="statistical-analysis-1.html#cb98-9" tabindex="-1"></a>  <span class="co"># show clusters</span></span>
<span id="cb98-10"><a href="statistical-analysis-1.html#cb98-10" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">fviz_cluster</span>(<span class="fu">list</span>(<span class="at">data =</span> df, </span>
<span id="cb98-11"><a href="statistical-analysis-1.html#cb98-11" tabindex="-1"></a>                    <span class="at">cluster =</span> res<span class="sc">$</span>cluster))) </span>
<span id="cb98-12"><a href="statistical-analysis-1.html#cb98-12" tabindex="-1"></a>  </span>
<span id="cb98-13"><a href="statistical-analysis-1.html#cb98-13" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb98-14"><a href="statistical-analysis-1.html#cb98-14" tabindex="-1"></a>}</span>
<span id="cb98-15"><a href="statistical-analysis-1.html#cb98-15" tabindex="-1"></a></span>
<span id="cb98-16"><a href="statistical-analysis-1.html#cb98-16" tabindex="-1"></a>Kcluster_res <span class="ot">&lt;-</span> <span class="fu">PartCluster</span>(</span>
<span id="cb98-17"><a href="statistical-analysis-1.html#cb98-17" tabindex="-1"></a>        <span class="at">object =</span> se_normalize,</span>
<span id="cb98-18"><a href="statistical-analysis-1.html#cb98-18" tabindex="-1"></a>        <span class="at">cluster_num =</span> <span class="dv">4</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-121-1.png" width="768" /></p>
<hr />
</div>
</div>
<div id="chemometrics-analysis" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Chemometrics Analysis<a href="statistical-analysis-1.html#chemometrics-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The functions for Chemometrics Analysis in <strong>POMA</strong> <span class="citation">(<a href="#ref-castellano2021pomashiny">Castellano-Escuder et al. 2021</a>)</span> implemented from <strong>mixOmics</strong> <span class="citation">(<a href="#ref-rohart2017mixomics">Rohart et al. 2017</a>)</span>.</p>
<p><strong>Note: Please also remember to <a href="statistical-analysis-1.html#data-preprocessing">preprocess</a> your data before running this sub-chapter.</strong></p>
<div id="loading-packages-2" class="section level3 hasAnchor" number="3.3.1">
<h3><span class="header-section-number">3.3.1</span> Loading packages<a href="statistical-analysis-1.html#loading-packages-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="statistical-analysis-1.html#cb99-1" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">warning =</span> F, <span class="at">message =</span> F)</span>
<span id="cb99-2"><a href="statistical-analysis-1.html#cb99-2" tabindex="-1"></a></span>
<span id="cb99-3"><a href="statistical-analysis-1.html#cb99-3" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb99-4"><a href="statistical-analysis-1.html#cb99-4" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb99-5"><a href="statistical-analysis-1.html#cb99-5" tabindex="-1"></a><span class="fu">library</span>(POMA)</span>
<span id="cb99-6"><a href="statistical-analysis-1.html#cb99-6" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb99-7"><a href="statistical-analysis-1.html#cb99-7" tabindex="-1"></a><span class="fu">library</span>(ggraph)</span>
<span id="cb99-8"><a href="statistical-analysis-1.html#cb99-8" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb99-9"><a href="statistical-analysis-1.html#cb99-9" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb99-10"><a href="statistical-analysis-1.html#cb99-10" tabindex="-1"></a></span>
<span id="cb99-11"><a href="statistical-analysis-1.html#cb99-11" tabindex="-1"></a><span class="co"># rm(list = ls())</span></span>
<span id="cb99-12"><a href="statistical-analysis-1.html#cb99-12" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb99-13"><a href="statistical-analysis-1.html#cb99-13" tabindex="-1"></a><span class="fu">options</span>(<span class="at">future.globals.maxSize =</span> <span class="dv">1000</span> <span class="sc">*</span> <span class="dv">1024</span><span class="sc">^</span><span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="importing-data-2" class="section level3 hasAnchor" number="3.3.2">
<h3><span class="header-section-number">3.3.2</span> Importing data<a href="statistical-analysis-1.html#importing-data-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The input data sets are from the previous chapter.</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="statistical-analysis-1.html#cb100-1" tabindex="-1"></a>se_processed <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;./dataset/POMA/se_processed.RDS&quot;</span>)</span></code></pre></div>
</div>
<div id="principal-component-analysis-pca" class="section level3 hasAnchor" number="3.3.3">
<h3><span class="header-section-number">3.3.3</span> Principal Component Analysis (PCA)<a href="statistical-analysis-1.html#principal-component-analysis-pca" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<blockquote>
<p>The aim of PCA (Jolliffe 2005) is to reduce the dimensionality of the data whilst retaining as much information as possible. ‘Information’ is referred here as variance. The idea is to create uncorrelated artificial variables called principal components (PCs) that combine in a linear manner the original (possibly correlated) variables.</p>
</blockquote>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="statistical-analysis-1.html#cb101-1" tabindex="-1"></a>poma_pca <span class="ot">&lt;-</span> <span class="fu">PomaMultivariate</span>(se_processed, <span class="at">method =</span> <span class="st">&quot;pca&quot;</span>)</span>
<span id="cb101-2"><a href="statistical-analysis-1.html#cb101-2" tabindex="-1"></a>poma_pca<span class="sc">$</span>scoresplot <span class="sc">+</span></span>
<span id="cb101-3"><a href="statistical-analysis-1.html#cb101-3" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Scores Plot (pca)&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-124-1.png" width="768" /></p>
</div>
<div id="partial-least-squares-discriminant-analysis-pls-da" class="section level3 hasAnchor" number="3.3.4">
<h3><span class="header-section-number">3.3.4</span> Partial Least Squares-Discriminant Analysis (PLS-DA)<a href="statistical-analysis-1.html#partial-least-squares-discriminant-analysis-pls-da" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<blockquote>
<p>Partial Least Squares (PLS) regression is a multivariate methodology which relates two data matrices X (e.g. transcriptomics) and Y (e.g. lipids). PLS goes beyond traditional multiple regression by modelling the structure of both matrices. Unlike traditional multiple regression models, it is not limited to uncorrelated variables. One of the many advantages of PLS is that it can handle many noisy, collinear (correlated) and missing variables and can also simultaneously model several response variables in Y.</p>
</blockquote>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="statistical-analysis-1.html#cb102-1" tabindex="-1"></a>poma_plsda <span class="ot">&lt;-</span> <span class="fu">PomaMultivariate</span>(se_processed, <span class="at">method =</span> <span class="st">&quot;plsda&quot;</span>)</span></code></pre></div>
<ul>
<li>scatter plot</li>
</ul>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="statistical-analysis-1.html#cb103-1" tabindex="-1"></a>poma_plsda<span class="sc">$</span>scoresplot <span class="sc">+</span></span>
<span id="cb103-2"><a href="statistical-analysis-1.html#cb103-2" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Scores Plot (plsda)&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-126-1.png" width="768" /></p>
<ul>
<li>errors plot</li>
</ul>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="statistical-analysis-1.html#cb104-1" tabindex="-1"></a>poma_plsda<span class="sc">$</span>errors_plsda_plot <span class="sc">+</span></span>
<span id="cb104-2"><a href="statistical-analysis-1.html#cb104-2" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Error Plot (plsda)&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-127-1.png" width="768" /></p>
</div>
<div id="sparse-partial-least-squares-discriminant-analysis-spls-da" class="section level3 hasAnchor" number="3.3.5">
<h3><span class="header-section-number">3.3.5</span> Sparse Partial Least Squares-Discriminant Analysis (sPLS-DA)<a href="statistical-analysis-1.html#sparse-partial-least-squares-discriminant-analysis-spls-da" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<blockquote>
<p>Even though PLS is highly efficient in a high dimensional context, the interpretability of PLS needed to be improved. sPLS has been recently developed by our team to perform simultaneous variable selection in both data sets X and Y data sets, by including LASSO penalizations in PLS on each pair of loading vectors</p>
</blockquote>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="statistical-analysis-1.html#cb105-1" tabindex="-1"></a>poma_splsda <span class="ot">&lt;-</span> <span class="fu">PomaMultivariate</span>(se_processed, <span class="at">method =</span> <span class="st">&quot;splsda&quot;</span>)</span></code></pre></div>
<ul>
<li>scatter plot</li>
</ul>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="statistical-analysis-1.html#cb106-1" tabindex="-1"></a>poma_splsda<span class="sc">$</span>scoresplot <span class="sc">+</span></span>
<span id="cb106-2"><a href="statistical-analysis-1.html#cb106-2" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Scores Plot (splsda)&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-129-1.png" width="768" /></p>
</div>
<div id="orthogonal-partial-least-squares-discriminant-analysis-orthopls-da" class="section level3 hasAnchor" number="3.3.6">
<h3><span class="header-section-number">3.3.6</span> Orthogonal Partial Least Squares-Discriminant Analysis (orthoPLS-DA)<a href="statistical-analysis-1.html#orthogonal-partial-least-squares-discriminant-analysis-orthopls-da" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Still in development…</p>
</div>
<div id="chemometrics-analysis-by-own-scripts" class="section level3 hasAnchor" number="3.3.7">
<h3><span class="header-section-number">3.3.7</span> Chemometrics Analysis by own scripts<a href="statistical-analysis-1.html#chemometrics-analysis-by-own-scripts" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="statistical-analysis-1.html#cb107-1" tabindex="-1"></a>DR_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb107-2"><a href="statistical-analysis-1.html#cb107-2" tabindex="-1"></a>    x,</span>
<span id="cb107-3"><a href="statistical-analysis-1.html#cb107-3" tabindex="-1"></a>    group,</span>
<span id="cb107-4"><a href="statistical-analysis-1.html#cb107-4" tabindex="-1"></a>    group_names,</span>
<span id="cb107-5"><a href="statistical-analysis-1.html#cb107-5" tabindex="-1"></a>    <span class="at">DRtype =</span> <span class="fu">c</span>(<span class="st">&quot;PCA&quot;</span>, <span class="st">&quot;PLS&quot;</span>, <span class="st">&quot;OPLS&quot;</span>)) {</span>
<span id="cb107-6"><a href="statistical-analysis-1.html#cb107-6" tabindex="-1"></a>  </span>
<span id="cb107-7"><a href="statistical-analysis-1.html#cb107-7" tabindex="-1"></a>  <span class="co"># x = se_processed</span></span>
<span id="cb107-8"><a href="statistical-analysis-1.html#cb107-8" tabindex="-1"></a>  <span class="co"># group = &quot;group&quot;</span></span>
<span id="cb107-9"><a href="statistical-analysis-1.html#cb107-9" tabindex="-1"></a>  <span class="co"># group_names = c(&quot;Mild&quot;, &quot;Severe&quot;)</span></span>
<span id="cb107-10"><a href="statistical-analysis-1.html#cb107-10" tabindex="-1"></a>  <span class="co"># DRtype = &quot;PCA&quot;</span></span>
<span id="cb107-11"><a href="statistical-analysis-1.html#cb107-11" tabindex="-1"></a>  </span>
<span id="cb107-12"><a href="statistical-analysis-1.html#cb107-12" tabindex="-1"></a>  <span class="co"># dataseat</span></span>
<span id="cb107-13"><a href="statistical-analysis-1.html#cb107-13" tabindex="-1"></a>  metadata <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(x) <span class="sc">%&gt;%</span></span>
<span id="cb107-14"><a href="statistical-analysis-1.html#cb107-14" tabindex="-1"></a>      <span class="fu">as.data.frame</span>()</span>
<span id="cb107-15"><a href="statistical-analysis-1.html#cb107-15" tabindex="-1"></a>  profile <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(x) <span class="sc">%&gt;%</span></span>
<span id="cb107-16"><a href="statistical-analysis-1.html#cb107-16" tabindex="-1"></a>      <span class="fu">as.data.frame</span>()</span>
<span id="cb107-17"><a href="statistical-analysis-1.html#cb107-17" tabindex="-1"></a>  </span>
<span id="cb107-18"><a href="statistical-analysis-1.html#cb107-18" tabindex="-1"></a>  <span class="fu">colnames</span>(metadata)[<span class="fu">which</span>(<span class="fu">colnames</span>(metadata) <span class="sc">==</span> group)] <span class="ot">&lt;-</span> <span class="st">&quot;CompVar&quot;</span></span>
<span id="cb107-19"><a href="statistical-analysis-1.html#cb107-19" tabindex="-1"></a>  phenotype <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span></span>
<span id="cb107-20"><a href="statistical-analysis-1.html#cb107-20" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(CompVar <span class="sc">%in%</span> group_names) </span>
<span id="cb107-21"><a href="statistical-analysis-1.html#cb107-21" tabindex="-1"></a>    </span>
<span id="cb107-22"><a href="statistical-analysis-1.html#cb107-22" tabindex="-1"></a>  match_order_index <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">pmatch</span>(<span class="fu">unique</span>(phenotype<span class="sc">$</span>CompVar), group_names), <span class="at">decreasing =</span> F)</span>
<span id="cb107-23"><a href="statistical-analysis-1.html#cb107-23" tabindex="-1"></a>  group_names_new <span class="ot">&lt;-</span> group_names[match_order_index]</span>
<span id="cb107-24"><a href="statistical-analysis-1.html#cb107-24" tabindex="-1"></a>  </span>
<span id="cb107-25"><a href="statistical-analysis-1.html#cb107-25" tabindex="-1"></a>  phenotype<span class="sc">$</span>CompVar <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">as.character</span>(phenotype<span class="sc">$</span>CompVar), </span>
<span id="cb107-26"><a href="statistical-analysis-1.html#cb107-26" tabindex="-1"></a>                              <span class="at">levels =</span> group_names_new)</span>
<span id="cb107-27"><a href="statistical-analysis-1.html#cb107-27" tabindex="-1"></a></span>
<span id="cb107-28"><a href="statistical-analysis-1.html#cb107-28" tabindex="-1"></a>  sid <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(phenotype), <span class="fu">colnames</span>(profile))</span>
<span id="cb107-29"><a href="statistical-analysis-1.html#cb107-29" tabindex="-1"></a>  phen <span class="ot">&lt;-</span> phenotype[<span class="fu">pmatch</span>(sid, <span class="fu">rownames</span>(phenotype)), , ]</span>
<span id="cb107-30"><a href="statistical-analysis-1.html#cb107-30" tabindex="-1"></a>  prof <span class="ot">&lt;-</span> profile <span class="sc">%&gt;%</span></span>
<span id="cb107-31"><a href="statistical-analysis-1.html#cb107-31" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(sid))</span>
<span id="cb107-32"><a href="statistical-analysis-1.html#cb107-32" tabindex="-1"></a>  </span>
<span id="cb107-33"><a href="statistical-analysis-1.html#cb107-33" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">all</span>(<span class="fu">colnames</span>(prof) <span class="sc">==</span> <span class="fu">rownames</span>(phen))) {</span>
<span id="cb107-34"><a href="statistical-analysis-1.html#cb107-34" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;Wrong Order&quot;</span>)</span>
<span id="cb107-35"><a href="statistical-analysis-1.html#cb107-35" tabindex="-1"></a>  }</span>
<span id="cb107-36"><a href="statistical-analysis-1.html#cb107-36" tabindex="-1"></a></span>
<span id="cb107-37"><a href="statistical-analysis-1.html#cb107-37" tabindex="-1"></a>  prof_cln <span class="ot">&lt;-</span> prof</span>
<span id="cb107-38"><a href="statistical-analysis-1.html#cb107-38" tabindex="-1"></a>  dataMatrix <span class="ot">&lt;-</span> prof_cln <span class="sc">%&gt;%</span> <span class="fu">t</span>() <span class="co"># row-&gt;sampleID; col-&gt;features</span></span>
<span id="cb107-39"><a href="statistical-analysis-1.html#cb107-39" tabindex="-1"></a>  sampleMetadata <span class="ot">&lt;-</span> phen <span class="sc">%&gt;%</span> <span class="co"># row-&gt;sampleID; col-&gt;features</span></span>
<span id="cb107-40"><a href="statistical-analysis-1.html#cb107-40" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">CompVar =</span> <span class="fu">factor</span>(<span class="fu">as.character</span>(CompVar), </span>
<span id="cb107-41"><a href="statistical-analysis-1.html#cb107-41" tabindex="-1"></a>                                  <span class="at">levels =</span> group_names_new)) <span class="sc">%&gt;%</span></span>
<span id="cb107-42"><a href="statistical-analysis-1.html#cb107-42" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Color =</span> <span class="fu">factor</span>(<span class="fu">as.character</span>(CompVar), </span>
<span id="cb107-43"><a href="statistical-analysis-1.html#cb107-43" tabindex="-1"></a>                                  <span class="at">levels =</span> group_names_new),</span>
<span id="cb107-44"><a href="statistical-analysis-1.html#cb107-44" tabindex="-1"></a>                  <span class="at">Color =</span> <span class="fu">as.character</span>(Color))    </span>
<span id="cb107-45"><a href="statistical-analysis-1.html#cb107-45" tabindex="-1"></a>  </span>
<span id="cb107-46"><a href="statistical-analysis-1.html#cb107-46" tabindex="-1"></a>  <span class="cf">if</span> (DRtype <span class="sc">==</span> <span class="st">&quot;PCA&quot;</span>) {</span>
<span id="cb107-47"><a href="statistical-analysis-1.html#cb107-47" tabindex="-1"></a>    fit <span class="ot">&lt;-</span> ropls<span class="sc">::</span><span class="fu">opls</span>(dataMatrix)</span>
<span id="cb107-48"><a href="statistical-analysis-1.html#cb107-48" tabindex="-1"></a>    <span class="fu">plot</span>(fit,</span>
<span id="cb107-49"><a href="statistical-analysis-1.html#cb107-49" tabindex="-1"></a>         <span class="at">typeVc =</span> <span class="st">&quot;x-score&quot;</span>,</span>
<span id="cb107-50"><a href="statistical-analysis-1.html#cb107-50" tabindex="-1"></a>         <span class="at">parAsColFcVn =</span> sampleMetadata<span class="sc">$</span>CompVar,</span>
<span id="cb107-51"><a href="statistical-analysis-1.html#cb107-51" tabindex="-1"></a>         )</span>
<span id="cb107-52"><a href="statistical-analysis-1.html#cb107-52" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (DRtype <span class="sc">==</span> <span class="st">&quot;PLS&quot;</span>) {</span>
<span id="cb107-53"><a href="statistical-analysis-1.html#cb107-53" tabindex="-1"></a>    fit <span class="ot">&lt;-</span> ropls<span class="sc">::</span><span class="fu">opls</span>(dataMatrix, sampleMetadata<span class="sc">$</span>CompVar)</span>
<span id="cb107-54"><a href="statistical-analysis-1.html#cb107-54" tabindex="-1"></a>    <span class="fu">plot</span>(fit,</span>
<span id="cb107-55"><a href="statistical-analysis-1.html#cb107-55" tabindex="-1"></a>         <span class="at">typeVc =</span> <span class="st">&quot;x-score&quot;</span>,</span>
<span id="cb107-56"><a href="statistical-analysis-1.html#cb107-56" tabindex="-1"></a>         <span class="at">parAsColFcVn =</span> sampleMetadata<span class="sc">$</span>CompVar,</span>
<span id="cb107-57"><a href="statistical-analysis-1.html#cb107-57" tabindex="-1"></a>         )</span>
<span id="cb107-58"><a href="statistical-analysis-1.html#cb107-58" tabindex="-1"></a>    <span class="co"># return(fit)    </span></span>
<span id="cb107-59"><a href="statistical-analysis-1.html#cb107-59" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (DRtype <span class="sc">==</span> <span class="st">&quot;OPLS&quot;</span>) {</span>
<span id="cb107-60"><a href="statistical-analysis-1.html#cb107-60" tabindex="-1"></a>    <span class="co"># only for binary classification</span></span>
<span id="cb107-61"><a href="statistical-analysis-1.html#cb107-61" tabindex="-1"></a>    fit <span class="ot">&lt;-</span> ropls<span class="sc">::</span><span class="fu">opls</span>(dataMatrix, sampleMetadata<span class="sc">$</span>CompVar,</span>
<span id="cb107-62"><a href="statistical-analysis-1.html#cb107-62" tabindex="-1"></a>                <span class="at">predI =</span> <span class="dv">1</span>, <span class="at">orthoI =</span> <span class="cn">NA</span>)</span>
<span id="cb107-63"><a href="statistical-analysis-1.html#cb107-63" tabindex="-1"></a>    <span class="fu">plot</span>(fit,</span>
<span id="cb107-64"><a href="statistical-analysis-1.html#cb107-64" tabindex="-1"></a>         <span class="at">typeVc =</span> <span class="st">&quot;x-score&quot;</span>,</span>
<span id="cb107-65"><a href="statistical-analysis-1.html#cb107-65" tabindex="-1"></a>         <span class="at">parAsColFcVn =</span> sampleMetadata<span class="sc">$</span>CompVar,</span>
<span id="cb107-66"><a href="statistical-analysis-1.html#cb107-66" tabindex="-1"></a>         )</span>
<span id="cb107-67"><a href="statistical-analysis-1.html#cb107-67" tabindex="-1"></a>    <span class="co"># return(fit)</span></span>
<span id="cb107-68"><a href="statistical-analysis-1.html#cb107-68" tabindex="-1"></a>  }</span>
<span id="cb107-69"><a href="statistical-analysis-1.html#cb107-69" tabindex="-1"></a>  </span>
<span id="cb107-70"><a href="statistical-analysis-1.html#cb107-70" tabindex="-1"></a>  <span class="fu">return</span>(fit)  </span>
<span id="cb107-71"><a href="statistical-analysis-1.html#cb107-71" tabindex="-1"></a>}</span></code></pre></div>
<ul>
<li>Principal Component Analysis (PCA)</li>
</ul>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="statistical-analysis-1.html#cb108-1" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">DR_fun</span>(</span>
<span id="cb108-2"><a href="statistical-analysis-1.html#cb108-2" tabindex="-1"></a>  <span class="at">x =</span> se_processed,</span>
<span id="cb108-3"><a href="statistical-analysis-1.html#cb108-3" tabindex="-1"></a>  <span class="at">group =</span> <span class="st">&quot;group&quot;</span>,</span>
<span id="cb108-4"><a href="statistical-analysis-1.html#cb108-4" tabindex="-1"></a>  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Severe&quot;</span>),</span>
<span id="cb108-5"><a href="statistical-analysis-1.html#cb108-5" tabindex="-1"></a>  <span class="at">DRtype =</span> <span class="st">&quot;PCA&quot;</span>)</span></code></pre></div>
<pre><code>## PCA
## 26 samples x 167 variables
## standard scaling of predictors
##       R2X(cum) pre ort
## Total    0.528   5   0</code></pre>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-131-1.png" width="672" /><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-131-2.png" width="672" /></p>
<ul>
<li>Partial Least Squares-Discriminant Analysis (PLS-DA)</li>
</ul>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="statistical-analysis-1.html#cb110-1" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">DR_fun</span>(</span>
<span id="cb110-2"><a href="statistical-analysis-1.html#cb110-2" tabindex="-1"></a>  <span class="at">x =</span> se_processed,</span>
<span id="cb110-3"><a href="statistical-analysis-1.html#cb110-3" tabindex="-1"></a>  <span class="at">group =</span> <span class="st">&quot;group&quot;</span>,</span>
<span id="cb110-4"><a href="statistical-analysis-1.html#cb110-4" tabindex="-1"></a>  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Severe&quot;</span>),</span>
<span id="cb110-5"><a href="statistical-analysis-1.html#cb110-5" tabindex="-1"></a>  <span class="at">DRtype =</span> <span class="st">&quot;PLS&quot;</span>)</span></code></pre></div>
<pre><code>## PLS-DA
## 26 samples x 167 variables and 1 response
## standard scaling of predictors and response(s)
##       R2X(cum) R2Y(cum) Q2(cum) RMSEE pre ort pR2Y  pQ2
## Total    0.296    0.857   0.537   0.2   2   0 0.25 0.05</code></pre>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-132-1.png" width="672" /><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-132-2.png" width="672" /></p>
<ul>
<li>Orthogonal Partial Least Squares-Discriminant Analysis (orthoPLS-DA)</li>
</ul>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="statistical-analysis-1.html#cb112-1" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">DR_fun</span>(</span>
<span id="cb112-2"><a href="statistical-analysis-1.html#cb112-2" tabindex="-1"></a>  <span class="at">x =</span> se_processed,</span>
<span id="cb112-3"><a href="statistical-analysis-1.html#cb112-3" tabindex="-1"></a>  <span class="at">group =</span> <span class="st">&quot;group&quot;</span>,</span>
<span id="cb112-4"><a href="statistical-analysis-1.html#cb112-4" tabindex="-1"></a>  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Severe&quot;</span>),</span>
<span id="cb112-5"><a href="statistical-analysis-1.html#cb112-5" tabindex="-1"></a>  <span class="at">DRtype =</span> <span class="st">&quot;OPLS&quot;</span>)</span></code></pre></div>
<pre><code>## OPLS-DA
## 26 samples x 167 variables and 1 response
## standard scaling of predictors and response(s)
##       R2X(cum) R2Y(cum) Q2(cum)  RMSEE pre ort pR2Y  pQ2
## Total    0.402     0.99   0.521 0.0566   1   3 0.05 0.05</code></pre>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-133-1.png" width="672" /><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-133-2.png" width="672" /></p>
<hr />
</div>
</div>
<div id="univariate-analysis" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> Univariate Analysis<a href="statistical-analysis-1.html#univariate-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Univariate analysis explores each variable in a data set, separately and it uses traditional statistical methods on single variable to calculate the statistics, such as fold change, p-value, etc.
<strong>Note: Please also remember to <a href="statistical-analysis-1.html#data-preprocessing">preprocess</a> your data before running this sub-chapter.</strong></p>
<div id="loading-packages-3" class="section level3 hasAnchor" number="3.4.1">
<h3><span class="header-section-number">3.4.1</span> Loading packages<a href="statistical-analysis-1.html#loading-packages-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="statistical-analysis-1.html#cb114-1" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">warning =</span> F)</span>
<span id="cb114-2"><a href="statistical-analysis-1.html#cb114-2" tabindex="-1"></a></span>
<span id="cb114-3"><a href="statistical-analysis-1.html#cb114-3" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb114-4"><a href="statistical-analysis-1.html#cb114-4" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb114-5"><a href="statistical-analysis-1.html#cb114-5" tabindex="-1"></a><span class="fu">library</span>(POMA)</span>
<span id="cb114-6"><a href="statistical-analysis-1.html#cb114-6" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb114-7"><a href="statistical-analysis-1.html#cb114-7" tabindex="-1"></a><span class="fu">library</span>(ggraph)</span>
<span id="cb114-8"><a href="statistical-analysis-1.html#cb114-8" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb114-9"><a href="statistical-analysis-1.html#cb114-9" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb114-10"><a href="statistical-analysis-1.html#cb114-10" tabindex="-1"></a></span>
<span id="cb114-11"><a href="statistical-analysis-1.html#cb114-11" tabindex="-1"></a><span class="co"># rm(list = ls())</span></span>
<span id="cb114-12"><a href="statistical-analysis-1.html#cb114-12" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb114-13"><a href="statistical-analysis-1.html#cb114-13" tabindex="-1"></a><span class="fu">options</span>(<span class="at">future.globals.maxSize =</span> <span class="dv">1000</span> <span class="sc">*</span> <span class="dv">1024</span><span class="sc">^</span><span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="importing-data-3" class="section level3 hasAnchor" number="3.4.2">
<h3><span class="header-section-number">3.4.2</span> Importing data<a href="statistical-analysis-1.html#importing-data-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The input data sets are from the previous chapter.</p>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="statistical-analysis-1.html#cb115-1" tabindex="-1"></a>se_impute <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;./dataset/POMA/se_impute.RDS&quot;</span>)</span>
<span id="cb115-2"><a href="statistical-analysis-1.html#cb115-2" tabindex="-1"></a>se_normalize <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;./dataset/POMA/se_normalize.RDS&quot;</span>)</span>
<span id="cb115-3"><a href="statistical-analysis-1.html#cb115-3" tabindex="-1"></a>se_processed <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;./dataset/POMA/se_processed.RDS&quot;</span>)</span></code></pre></div>
</div>
<div id="fold-change-analysis" class="section level3 hasAnchor" number="3.4.3">
<h3><span class="header-section-number">3.4.3</span> Fold Change Analysis<a href="statistical-analysis-1.html#fold-change-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>RawData (inputed data)</li>
</ul>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="statistical-analysis-1.html#cb116-1" tabindex="-1"></a>FoldChange <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb116-2"><a href="statistical-analysis-1.html#cb116-2" tabindex="-1"></a>    x,</span>
<span id="cb116-3"><a href="statistical-analysis-1.html#cb116-3" tabindex="-1"></a>    group,</span>
<span id="cb116-4"><a href="statistical-analysis-1.html#cb116-4" tabindex="-1"></a>    group_names) {</span>
<span id="cb116-5"><a href="statistical-analysis-1.html#cb116-5" tabindex="-1"></a>  </span>
<span id="cb116-6"><a href="statistical-analysis-1.html#cb116-6" tabindex="-1"></a>  <span class="co"># x = se_impute</span></span>
<span id="cb116-7"><a href="statistical-analysis-1.html#cb116-7" tabindex="-1"></a>  <span class="co"># group = &quot;group&quot;</span></span>
<span id="cb116-8"><a href="statistical-analysis-1.html#cb116-8" tabindex="-1"></a>  <span class="co"># group_names = c(&quot;Mild&quot;, &quot;Severe&quot;)</span></span>
<span id="cb116-9"><a href="statistical-analysis-1.html#cb116-9" tabindex="-1"></a>  </span>
<span id="cb116-10"><a href="statistical-analysis-1.html#cb116-10" tabindex="-1"></a>  <span class="co"># dataseat</span></span>
<span id="cb116-11"><a href="statistical-analysis-1.html#cb116-11" tabindex="-1"></a>  metadata <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(x) <span class="sc">%&gt;%</span></span>
<span id="cb116-12"><a href="statistical-analysis-1.html#cb116-12" tabindex="-1"></a>      <span class="fu">as.data.frame</span>()</span>
<span id="cb116-13"><a href="statistical-analysis-1.html#cb116-13" tabindex="-1"></a>  profile <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(x) <span class="sc">%&gt;%</span></span>
<span id="cb116-14"><a href="statistical-analysis-1.html#cb116-14" tabindex="-1"></a>      <span class="fu">as.data.frame</span>()</span>
<span id="cb116-15"><a href="statistical-analysis-1.html#cb116-15" tabindex="-1"></a></span>
<span id="cb116-16"><a href="statistical-analysis-1.html#cb116-16" tabindex="-1"></a>  <span class="fu">colnames</span>(metadata)[<span class="fu">which</span>(<span class="fu">colnames</span>(metadata) <span class="sc">==</span> group)] <span class="ot">&lt;-</span> <span class="st">&quot;CompVar&quot;</span></span>
<span id="cb116-17"><a href="statistical-analysis-1.html#cb116-17" tabindex="-1"></a>  phenotype <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span></span>
<span id="cb116-18"><a href="statistical-analysis-1.html#cb116-18" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(CompVar <span class="sc">%in%</span> group_names) <span class="sc">%&gt;%</span></span>
<span id="cb116-19"><a href="statistical-analysis-1.html#cb116-19" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">CompVar =</span> <span class="fu">as.character</span>(CompVar)) <span class="sc">%&gt;%</span></span>
<span id="cb116-20"><a href="statistical-analysis-1.html#cb116-20" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">CompVar =</span> <span class="fu">factor</span>(CompVar, <span class="at">levels =</span> group_names))</span>
<span id="cb116-21"><a href="statistical-analysis-1.html#cb116-21" tabindex="-1"></a>  </span>
<span id="cb116-22"><a href="statistical-analysis-1.html#cb116-22" tabindex="-1"></a>  sid <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(phenotype), <span class="fu">colnames</span>(profile))</span>
<span id="cb116-23"><a href="statistical-analysis-1.html#cb116-23" tabindex="-1"></a>  phen <span class="ot">&lt;-</span> phenotype[<span class="fu">pmatch</span>(sid, <span class="fu">rownames</span>(phenotype)), , ]</span>
<span id="cb116-24"><a href="statistical-analysis-1.html#cb116-24" tabindex="-1"></a>  prof <span class="ot">&lt;-</span> profile <span class="sc">%&gt;%</span></span>
<span id="cb116-25"><a href="statistical-analysis-1.html#cb116-25" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(sid))</span>
<span id="cb116-26"><a href="statistical-analysis-1.html#cb116-26" tabindex="-1"></a>  </span>
<span id="cb116-27"><a href="statistical-analysis-1.html#cb116-27" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">all</span>(<span class="fu">colnames</span>(prof) <span class="sc">==</span> <span class="fu">rownames</span>(phen))) {</span>
<span id="cb116-28"><a href="statistical-analysis-1.html#cb116-28" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;Wrong Order&quot;</span>)</span>
<span id="cb116-29"><a href="statistical-analysis-1.html#cb116-29" tabindex="-1"></a>  }</span>
<span id="cb116-30"><a href="statistical-analysis-1.html#cb116-30" tabindex="-1"></a>  </span>
<span id="cb116-31"><a href="statistical-analysis-1.html#cb116-31" tabindex="-1"></a>  fc_res <span class="ot">&lt;-</span> <span class="fu">apply</span>(prof, <span class="dv">1</span>, <span class="cf">function</span>(x1, y1) {</span>
<span id="cb116-32"><a href="statistical-analysis-1.html#cb116-32" tabindex="-1"></a>    dat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">value =</span> <span class="fu">as.numeric</span>(x1), <span class="at">group =</span> y1)</span>
<span id="cb116-33"><a href="statistical-analysis-1.html#cb116-33" tabindex="-1"></a>    mn <span class="ot">&lt;-</span> <span class="fu">tapply</span>(dat<span class="sc">$</span>value, dat<span class="sc">$</span>group, <span class="cf">function</span>(x){</span>
<span id="cb116-34"><a href="statistical-analysis-1.html#cb116-34" tabindex="-1"></a>      <span class="fu">mean</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb116-35"><a href="statistical-analysis-1.html#cb116-35" tabindex="-1"></a>    }) <span class="sc">%&gt;%</span></span>
<span id="cb116-36"><a href="statistical-analysis-1.html#cb116-36" tabindex="-1"></a>      <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb116-37"><a href="statistical-analysis-1.html#cb116-37" tabindex="-1"></a>      stats<span class="sc">::</span><span class="fu">setNames</span>(<span class="st">&quot;value&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb116-38"><a href="statistical-analysis-1.html#cb116-38" tabindex="-1"></a>      tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;Group&quot;</span>)</span>
<span id="cb116-39"><a href="statistical-analysis-1.html#cb116-39" tabindex="-1"></a></span>
<span id="cb116-40"><a href="statistical-analysis-1.html#cb116-40" tabindex="-1"></a>    mn1 <span class="ot">&lt;-</span> <span class="fu">with</span>(mn, mn[Group <span class="sc">%in%</span> group_names[<span class="dv">1</span>], <span class="st">&quot;value&quot;</span>])</span>
<span id="cb116-41"><a href="statistical-analysis-1.html#cb116-41" tabindex="-1"></a>    mn2 <span class="ot">&lt;-</span> <span class="fu">with</span>(mn, mn[Group <span class="sc">%in%</span> group_names[<span class="dv">2</span>], <span class="st">&quot;value&quot;</span>])</span>
<span id="cb116-42"><a href="statistical-analysis-1.html#cb116-42" tabindex="-1"></a>    mnall <span class="ot">&lt;-</span> <span class="fu">mean</span>(dat<span class="sc">$</span>value, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb116-43"><a href="statistical-analysis-1.html#cb116-43" tabindex="-1"></a>    </span>
<span id="cb116-44"><a href="statistical-analysis-1.html#cb116-44" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">all</span>(mn1 <span class="sc">!=</span> <span class="dv">0</span>, mn2 <span class="sc">!=</span> <span class="dv">0</span>)) {</span>
<span id="cb116-45"><a href="statistical-analysis-1.html#cb116-45" tabindex="-1"></a>      fc <span class="ot">&lt;-</span> mn1 <span class="sc">/</span> mn2</span>
<span id="cb116-46"><a href="statistical-analysis-1.html#cb116-46" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb116-47"><a href="statistical-analysis-1.html#cb116-47" tabindex="-1"></a>      fc <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb116-48"><a href="statistical-analysis-1.html#cb116-48" tabindex="-1"></a>    }</span>
<span id="cb116-49"><a href="statistical-analysis-1.html#cb116-49" tabindex="-1"></a>    </span>
<span id="cb116-50"><a href="statistical-analysis-1.html#cb116-50" tabindex="-1"></a>    logfc <span class="ot">&lt;-</span> <span class="fu">log2</span>(fc)</span>
<span id="cb116-51"><a href="statistical-analysis-1.html#cb116-51" tabindex="-1"></a></span>
<span id="cb116-52"><a href="statistical-analysis-1.html#cb116-52" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="fu">c</span>(fc, logfc, mnall, mn1, mn2)</span>
<span id="cb116-53"><a href="statistical-analysis-1.html#cb116-53" tabindex="-1"></a>    <span class="fu">return</span>(res)</span>
<span id="cb116-54"><a href="statistical-analysis-1.html#cb116-54" tabindex="-1"></a>  }, phen<span class="sc">$</span>CompVar) <span class="sc">%&gt;%</span></span>
<span id="cb116-55"><a href="statistical-analysis-1.html#cb116-55" tabindex="-1"></a>    base<span class="sc">::</span><span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb116-56"><a href="statistical-analysis-1.html#cb116-56" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;Feature&quot;</span>)</span>
<span id="cb116-57"><a href="statistical-analysis-1.html#cb116-57" tabindex="-1"></a>  </span>
<span id="cb116-58"><a href="statistical-analysis-1.html#cb116-58" tabindex="-1"></a>  <span class="fu">colnames</span>(fc_res) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;FeatureID&quot;</span>, <span class="st">&quot;FoldChange&quot;</span>, </span>
<span id="cb116-59"><a href="statistical-analysis-1.html#cb116-59" tabindex="-1"></a>                        <span class="st">&quot;Log2FoldChange&quot;</span>,</span>
<span id="cb116-60"><a href="statistical-analysis-1.html#cb116-60" tabindex="-1"></a>                        <span class="st">&quot;Mean Abundance</span><span class="sc">\n</span><span class="st">(All)&quot;</span>,</span>
<span id="cb116-61"><a href="statistical-analysis-1.html#cb116-61" tabindex="-1"></a>                     <span class="fu">paste0</span>(<span class="st">&quot;Mean Abundance</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="fu">c</span>(<span class="st">&quot;former&quot;</span>, <span class="st">&quot;latter&quot;</span>)))  </span>
<span id="cb116-62"><a href="statistical-analysis-1.html#cb116-62" tabindex="-1"></a>  </span>
<span id="cb116-63"><a href="statistical-analysis-1.html#cb116-63" tabindex="-1"></a>  <span class="co"># Number of Group</span></span>
<span id="cb116-64"><a href="statistical-analysis-1.html#cb116-64" tabindex="-1"></a>  dat_status <span class="ot">&lt;-</span> <span class="fu">table</span>(phen<span class="sc">$</span>CompVar)</span>
<span id="cb116-65"><a href="statistical-analysis-1.html#cb116-65" tabindex="-1"></a>  dat_status_number <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(dat_status)</span>
<span id="cb116-66"><a href="statistical-analysis-1.html#cb116-66" tabindex="-1"></a>  dat_status_name <span class="ot">&lt;-</span> <span class="fu">names</span>(dat_status)</span>
<span id="cb116-67"><a href="statistical-analysis-1.html#cb116-67" tabindex="-1"></a>  fc_res<span class="sc">$</span>Block <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">paste</span>(dat_status_number[<span class="dv">1</span>], dat_status_name[<span class="dv">1</span>], <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>),</span>
<span id="cb116-68"><a href="statistical-analysis-1.html#cb116-68" tabindex="-1"></a>                       <span class="st">&quot;vs&quot;</span>,</span>
<span id="cb116-69"><a href="statistical-analysis-1.html#cb116-69" tabindex="-1"></a>                       <span class="fu">paste</span>(dat_status_number[<span class="dv">2</span>], dat_status_name[<span class="dv">2</span>], <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>))</span>
<span id="cb116-70"><a href="statistical-analysis-1.html#cb116-70" tabindex="-1"></a>  </span>
<span id="cb116-71"><a href="statistical-analysis-1.html#cb116-71" tabindex="-1"></a>  res <span class="ot">&lt;-</span> fc_res <span class="sc">%&gt;%</span></span>
<span id="cb116-72"><a href="statistical-analysis-1.html#cb116-72" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(FeatureID, Block, <span class="fu">everything</span>())    </span>
<span id="cb116-73"><a href="statistical-analysis-1.html#cb116-73" tabindex="-1"></a></span>
<span id="cb116-74"><a href="statistical-analysis-1.html#cb116-74" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb116-75"><a href="statistical-analysis-1.html#cb116-75" tabindex="-1"></a>}</span>
<span id="cb116-76"><a href="statistical-analysis-1.html#cb116-76" tabindex="-1"></a></span>
<span id="cb116-77"><a href="statistical-analysis-1.html#cb116-77" tabindex="-1"></a></span>
<span id="cb116-78"><a href="statistical-analysis-1.html#cb116-78" tabindex="-1"></a>fc_res <span class="ot">&lt;-</span> <span class="fu">FoldChange</span>(</span>
<span id="cb116-79"><a href="statistical-analysis-1.html#cb116-79" tabindex="-1"></a>           <span class="at">x =</span> se_impute,</span>
<span id="cb116-80"><a href="statistical-analysis-1.html#cb116-80" tabindex="-1"></a>           <span class="at">group =</span> <span class="st">&quot;group&quot;</span>,</span>
<span id="cb116-81"><a href="statistical-analysis-1.html#cb116-81" tabindex="-1"></a>           <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Moderate&quot;</span>))</span>
<span id="cb116-82"><a href="statistical-analysis-1.html#cb116-82" tabindex="-1"></a></span>
<span id="cb116-83"><a href="statistical-analysis-1.html#cb116-83" tabindex="-1"></a><span class="fu">head</span>(fc_res)</span></code></pre></div>
<pre><code>##   FeatureID                  Block FoldChange Log2FoldChange Mean Abundance\n(All) Mean Abundance\nformer Mean Abundance\nlatter
## 1   M_38768 14_Mild vs 19_Moderate  0.8977727    -0.15557781            37464768.7             35159693.7             39163245.1
## 2   M_38296 14_Mild vs 19_Moderate  0.7827914    -0.35330013             4140693.4              3570299.1              4560983.9
## 3   M_63436 14_Mild vs 19_Moderate  0.7982418    -0.32510221              734958.7               641591.4               803755.6
## 4   M_57814 14_Mild vs 19_Moderate  0.9414267    -0.08707934              267247.3               258005.0               274057.4
## 5   M_52984 14_Mild vs 19_Moderate  0.8476930    -0.23838621              136374.4               123589.4               145795.0
## 6   M_48762 14_Mild vs 19_Moderate  0.3214408    -1.63737503             4371547.5              1973236.4              6138724.2</code></pre>
</div>
<div id="t-test" class="section level3 hasAnchor" number="3.4.4">
<h3><span class="header-section-number">3.4.4</span> T Test<a href="statistical-analysis-1.html#t-test" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="statistical-analysis-1.html#cb118-1" tabindex="-1"></a>group_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Severe&quot;</span>)</span>
<span id="cb118-2"><a href="statistical-analysis-1.html#cb118-2" tabindex="-1"></a></span>
<span id="cb118-3"><a href="statistical-analysis-1.html#cb118-3" tabindex="-1"></a>se_processed_subset <span class="ot">&lt;-</span> se_processed[, se_processed<span class="sc">$</span>group <span class="sc">%in%</span> group_names]</span>
<span id="cb118-4"><a href="statistical-analysis-1.html#cb118-4" tabindex="-1"></a>se_processed_subset<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">as.character</span>(se_processed_subset<span class="sc">$</span>group))</span>
<span id="cb118-5"><a href="statistical-analysis-1.html#cb118-5" tabindex="-1"></a>ttest_res <span class="ot">&lt;-</span> <span class="fu">PomaUnivariate</span>(se_processed_subset, <span class="at">method =</span> <span class="st">&quot;ttest&quot;</span>)</span>
<span id="cb118-6"><a href="statistical-analysis-1.html#cb118-6" tabindex="-1"></a><span class="fu">head</span>(ttest_res)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 9
##   feature     FC diff_means pvalue pvalueAdj mean_Mild mean_Severe sd_Mild sd_Severe
##   &lt;chr&gt;    &lt;dbl&gt;      &lt;dbl&gt;  &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;       &lt;dbl&gt;   &lt;dbl&gt;     &lt;dbl&gt;
## 1 M_38768  0.116      0.087  0.621     0.893   -0.0990    -0.0115    0.528     0.354
## 2 M_38296 -0.028      0.168  0.309     0.730   -0.163      0.00462   0.504     0.307
## 3 M_63436 -0.395      0.082  0.594     0.878   -0.0584     0.0231    0.402     0.368
## 4 M_57814  2.24      -0.024  0.827     0.952   -0.0192    -0.0431    0.327     0.223
## 5 M_52603  2.46       0.034  0.771     0.952    0.0236     0.0580    0.238     0.339
## 6 M_53174  0.178      0.103  0.546     0.878   -0.126     -0.0224    0.375     0.470</code></pre>
</div>
<div id="wilcoxon-test" class="section level3 hasAnchor" number="3.4.5">
<h3><span class="header-section-number">3.4.5</span> Wilcoxon Test<a href="statistical-analysis-1.html#wilcoxon-test" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="statistical-analysis-1.html#cb120-1" tabindex="-1"></a>wilcox_res <span class="ot">&lt;-</span> <span class="fu">PomaUnivariate</span>(se_processed_subset, <span class="at">method =</span> <span class="st">&quot;mann&quot;</span>)</span>
<span id="cb120-2"><a href="statistical-analysis-1.html#cb120-2" tabindex="-1"></a><span class="fu">head</span>(wilcox_res)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 9
##   feature     FC diff_means pvalue pvalueAdj mean_Mild mean_Severe sd_Mild sd_Severe
##   &lt;chr&gt;    &lt;dbl&gt;      &lt;dbl&gt;  &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;       &lt;dbl&gt;   &lt;dbl&gt;     &lt;dbl&gt;
## 1 M_38768  0.116      0.087  0.560     0.867   -0.0990    -0.0115    0.528     0.354
## 2 M_38296 -0.028      0.168  0.145     0.576   -0.163      0.00462   0.504     0.307
## 3 M_63436 -0.395      0.082  0.560     0.867   -0.0584     0.0231    0.402     0.368
## 4 M_57814  2.24      -0.024  0.940     0.981   -0.0192    -0.0431    0.327     0.223
## 5 M_52603  2.46       0.034  0.940     0.981    0.0236     0.0580    0.238     0.339
## 6 M_53174  0.178      0.103  0.705     0.912   -0.126     -0.0224    0.375     0.470</code></pre>
</div>
<div id="limma-test" class="section level3 hasAnchor" number="3.4.6">
<h3><span class="header-section-number">3.4.6</span> Limma Test<a href="statistical-analysis-1.html#limma-test" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="statistical-analysis-1.html#cb122-1" tabindex="-1"></a>Limma_res <span class="ot">&lt;-</span> <span class="fu">PomaLimma</span>(se_processed_subset, <span class="at">contrast =</span> <span class="fu">paste</span>(group_names, <span class="at">collapse =</span> <span class="st">&quot;-&quot;</span>), <span class="at">adjust =</span> <span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb122-2"><a href="statistical-analysis-1.html#cb122-2" tabindex="-1"></a><span class="fu">head</span>(Limma_res)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 7
##   feature  logFC AveExpr     t P.Value adj.P.Val     B
##   &lt;chr&gt;    &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt;
## 1 M_42449 -0.455 -0.0649 -3.24 0.00253     0.217 -2.34
## 2 M_37482  0.535  0.0923  2.99 0.00499     0.217 -2.69
## 3 M_62566 -0.389  0.0428 -2.95 0.00550     0.217 -2.74
## 4 M_19263 -0.420 -0.0546 -2.87 0.00679     0.217 -2.85
## 5 M_1566  -0.397 -0.0456 -2.83 0.00761     0.217 -2.91
## 6 M_43761 -0.440  0.0168 -2.82 0.00780     0.217 -2.92</code></pre>
</div>
<div id="volcano-plot" class="section level3 hasAnchor" number="3.4.7">
<h3><span class="header-section-number">3.4.7</span> Volcano plot<a href="statistical-analysis-1.html#volcano-plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="statistical-analysis-1.html#cb124-1" tabindex="-1"></a>se_impute_subset <span class="ot">&lt;-</span> se_impute[, se_impute<span class="sc">$</span>group <span class="sc">%in%</span> group_names]</span>
<span id="cb124-2"><a href="statistical-analysis-1.html#cb124-2" tabindex="-1"></a>se_impute_subset<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">as.character</span>(se_impute_subset<span class="sc">$</span>group))</span>
<span id="cb124-3"><a href="statistical-analysis-1.html#cb124-3" tabindex="-1"></a></span>
<span id="cb124-4"><a href="statistical-analysis-1.html#cb124-4" tabindex="-1"></a><span class="fu">PomaVolcano</span>(se_impute_subset, </span>
<span id="cb124-5"><a href="statistical-analysis-1.html#cb124-5" tabindex="-1"></a>            <span class="at">pval =</span> <span class="st">&quot;raw&quot;</span>,</span>
<span id="cb124-6"><a href="statistical-analysis-1.html#cb124-6" tabindex="-1"></a>            <span class="at">pval_cutoff =</span> <span class="fl">0.1</span>,</span>
<span id="cb124-7"><a href="statistical-analysis-1.html#cb124-7" tabindex="-1"></a>            <span class="at">log2FC =</span> <span class="fl">0.2</span>,</span>
<span id="cb124-8"><a href="statistical-analysis-1.html#cb124-8" tabindex="-1"></a>            <span class="at">xlim =</span> <span class="dv">1</span>,</span>
<span id="cb124-9"><a href="statistical-analysis-1.html#cb124-9" tabindex="-1"></a>            <span class="at">labels =</span> <span class="cn">TRUE</span>,</span>
<span id="cb124-10"><a href="statistical-analysis-1.html#cb124-10" tabindex="-1"></a>            <span class="at">plot_title =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-140-1.png" width="576" /></p>
</div>
<div id="vip-variable-influence-on-projection-coefficient" class="section level3 hasAnchor" number="3.4.8">
<h3><span class="header-section-number">3.4.8</span> VIP (Variable influence on projection &amp; coefficient)<a href="statistical-analysis-1.html#vip-variable-influence-on-projection-coefficient" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><p>Variable influence on projection (VIP) for orthogonal projections to latent structures (OPLS)</p></li>
<li><p>Variable influence on projection (VIP) for projections to latent structures (PLS)</p></li>
</ul>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="statistical-analysis-1.html#cb125-1" tabindex="-1"></a>VIP_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb125-2"><a href="statistical-analysis-1.html#cb125-2" tabindex="-1"></a>    x,</span>
<span id="cb125-3"><a href="statistical-analysis-1.html#cb125-3" tabindex="-1"></a>    group,</span>
<span id="cb125-4"><a href="statistical-analysis-1.html#cb125-4" tabindex="-1"></a>    group_names,</span>
<span id="cb125-5"><a href="statistical-analysis-1.html#cb125-5" tabindex="-1"></a>    <span class="at">VIPtype =</span> <span class="fu">c</span>(<span class="st">&quot;OPLS&quot;</span>, <span class="st">&quot;PLS&quot;</span>),</span>
<span id="cb125-6"><a href="statistical-analysis-1.html#cb125-6" tabindex="-1"></a>    <span class="at">vip_cutoff =</span> <span class="dv">1</span>) {</span>
<span id="cb125-7"><a href="statistical-analysis-1.html#cb125-7" tabindex="-1"></a>  </span>
<span id="cb125-8"><a href="statistical-analysis-1.html#cb125-8" tabindex="-1"></a>  <span class="co"># x = se_normalize</span></span>
<span id="cb125-9"><a href="statistical-analysis-1.html#cb125-9" tabindex="-1"></a>  <span class="co"># group = &quot;group&quot;</span></span>
<span id="cb125-10"><a href="statistical-analysis-1.html#cb125-10" tabindex="-1"></a>  <span class="co"># group_names = c(&quot;Mild&quot;, &quot;Severe&quot;)</span></span>
<span id="cb125-11"><a href="statistical-analysis-1.html#cb125-11" tabindex="-1"></a>  <span class="co"># VIPtype = &quot;PLS&quot;</span></span>
<span id="cb125-12"><a href="statistical-analysis-1.html#cb125-12" tabindex="-1"></a>  <span class="co"># vip_cutoff = 1</span></span>
<span id="cb125-13"><a href="statistical-analysis-1.html#cb125-13" tabindex="-1"></a>  </span>
<span id="cb125-14"><a href="statistical-analysis-1.html#cb125-14" tabindex="-1"></a>  metadata <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(x) <span class="sc">%&gt;%</span></span>
<span id="cb125-15"><a href="statistical-analysis-1.html#cb125-15" tabindex="-1"></a>      <span class="fu">as.data.frame</span>()</span>
<span id="cb125-16"><a href="statistical-analysis-1.html#cb125-16" tabindex="-1"></a>  profile <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(x) <span class="sc">%&gt;%</span></span>
<span id="cb125-17"><a href="statistical-analysis-1.html#cb125-17" tabindex="-1"></a>      <span class="fu">as.data.frame</span>()</span>
<span id="cb125-18"><a href="statistical-analysis-1.html#cb125-18" tabindex="-1"></a></span>
<span id="cb125-19"><a href="statistical-analysis-1.html#cb125-19" tabindex="-1"></a>  <span class="fu">colnames</span>(metadata)[<span class="fu">which</span>(<span class="fu">colnames</span>(metadata) <span class="sc">==</span> group)] <span class="ot">&lt;-</span> <span class="st">&quot;CompVar&quot;</span></span>
<span id="cb125-20"><a href="statistical-analysis-1.html#cb125-20" tabindex="-1"></a>  phenotype <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span></span>
<span id="cb125-21"><a href="statistical-analysis-1.html#cb125-21" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(CompVar <span class="sc">%in%</span> group_names) <span class="sc">%&gt;%</span></span>
<span id="cb125-22"><a href="statistical-analysis-1.html#cb125-22" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">CompVar =</span> <span class="fu">as.character</span>(CompVar)) <span class="sc">%&gt;%</span></span>
<span id="cb125-23"><a href="statistical-analysis-1.html#cb125-23" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">CompVar =</span> <span class="fu">factor</span>(CompVar, <span class="at">levels =</span> group_names))</span>
<span id="cb125-24"><a href="statistical-analysis-1.html#cb125-24" tabindex="-1"></a>  </span>
<span id="cb125-25"><a href="statistical-analysis-1.html#cb125-25" tabindex="-1"></a>  sid <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(phenotype), <span class="fu">colnames</span>(profile))</span>
<span id="cb125-26"><a href="statistical-analysis-1.html#cb125-26" tabindex="-1"></a>  phen <span class="ot">&lt;-</span> phenotype[<span class="fu">pmatch</span>(sid, <span class="fu">rownames</span>(phenotype)), , ]</span>
<span id="cb125-27"><a href="statistical-analysis-1.html#cb125-27" tabindex="-1"></a>  prof <span class="ot">&lt;-</span> profile <span class="sc">%&gt;%</span></span>
<span id="cb125-28"><a href="statistical-analysis-1.html#cb125-28" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(sid))</span>
<span id="cb125-29"><a href="statistical-analysis-1.html#cb125-29" tabindex="-1"></a>  </span>
<span id="cb125-30"><a href="statistical-analysis-1.html#cb125-30" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">all</span>(<span class="fu">colnames</span>(prof) <span class="sc">==</span> <span class="fu">rownames</span>(phen))) {</span>
<span id="cb125-31"><a href="statistical-analysis-1.html#cb125-31" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;Wrong Order&quot;</span>)</span>
<span id="cb125-32"><a href="statistical-analysis-1.html#cb125-32" tabindex="-1"></a>  }</span>
<span id="cb125-33"><a href="statistical-analysis-1.html#cb125-33" tabindex="-1"></a>  </span>
<span id="cb125-34"><a href="statistical-analysis-1.html#cb125-34" tabindex="-1"></a>  dataMatrix <span class="ot">&lt;-</span> prof <span class="sc">%&gt;%</span> base<span class="sc">::</span><span class="fu">t</span>() <span class="co"># row-&gt;sampleID; col-&gt;features</span></span>
<span id="cb125-35"><a href="statistical-analysis-1.html#cb125-35" tabindex="-1"></a>  sampleMetadata <span class="ot">&lt;-</span> phen <span class="co"># row-&gt;sampleID; col-&gt;features</span></span>
<span id="cb125-36"><a href="statistical-analysis-1.html#cb125-36" tabindex="-1"></a>  </span>
<span id="cb125-37"><a href="statistical-analysis-1.html#cb125-37" tabindex="-1"></a>  comparsionVn <span class="ot">&lt;-</span> sampleMetadata[, <span class="st">&quot;CompVar&quot;</span>]</span>
<span id="cb125-38"><a href="statistical-analysis-1.html#cb125-38" tabindex="-1"></a>  </span>
<span id="cb125-39"><a href="statistical-analysis-1.html#cb125-39" tabindex="-1"></a>  <span class="co"># corrlation between group and features</span></span>
<span id="cb125-40"><a href="statistical-analysis-1.html#cb125-40" tabindex="-1"></a>  pvaVn <span class="ot">&lt;-</span> <span class="fu">apply</span>(dataMatrix, <span class="dv">2</span>,</span>
<span id="cb125-41"><a href="statistical-analysis-1.html#cb125-41" tabindex="-1"></a>                 <span class="cf">function</span>(feaVn) <span class="fu">cor.test</span>(<span class="fu">as.numeric</span>(comparsionVn), feaVn)[[<span class="st">&quot;p.value&quot;</span>]])</span>
<span id="cb125-42"><a href="statistical-analysis-1.html#cb125-42" tabindex="-1"></a>  </span>
<span id="cb125-43"><a href="statistical-analysis-1.html#cb125-43" tabindex="-1"></a>  <span class="fu">library</span>(ropls)</span>
<span id="cb125-44"><a href="statistical-analysis-1.html#cb125-44" tabindex="-1"></a>  </span>
<span id="cb125-45"><a href="statistical-analysis-1.html#cb125-45" tabindex="-1"></a>  <span class="cf">if</span> (VIPtype <span class="sc">==</span> <span class="st">&quot;OPLS&quot;</span>) {</span>
<span id="cb125-46"><a href="statistical-analysis-1.html#cb125-46" tabindex="-1"></a>    vipVn <span class="ot">&lt;-</span> <span class="fu">getVipVn</span>(<span class="fu">opls</span>(dataMatrix, </span>
<span id="cb125-47"><a href="statistical-analysis-1.html#cb125-47" tabindex="-1"></a>                           comparsionVn,</span>
<span id="cb125-48"><a href="statistical-analysis-1.html#cb125-48" tabindex="-1"></a>                           <span class="at">predI =</span> <span class="dv">1</span>, </span>
<span id="cb125-49"><a href="statistical-analysis-1.html#cb125-49" tabindex="-1"></a>                           <span class="at">orthoI =</span> <span class="cn">NA</span>,</span>
<span id="cb125-50"><a href="statistical-analysis-1.html#cb125-50" tabindex="-1"></a>                           <span class="at">fig.pdfC =</span> <span class="st">&quot;none&quot;</span>))    </span>
<span id="cb125-51"><a href="statistical-analysis-1.html#cb125-51" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb125-52"><a href="statistical-analysis-1.html#cb125-52" tabindex="-1"></a>    vipVn <span class="ot">&lt;-</span> <span class="fu">getVipVn</span>(<span class="fu">opls</span>(dataMatrix, </span>
<span id="cb125-53"><a href="statistical-analysis-1.html#cb125-53" tabindex="-1"></a>                           comparsionVn,</span>
<span id="cb125-54"><a href="statistical-analysis-1.html#cb125-54" tabindex="-1"></a>                           <span class="at">predI =</span> <span class="dv">1</span>, </span>
<span id="cb125-55"><a href="statistical-analysis-1.html#cb125-55" tabindex="-1"></a>                           <span class="at">fig.pdfC =</span> <span class="st">&quot;none&quot;</span>))    </span>
<span id="cb125-56"><a href="statistical-analysis-1.html#cb125-56" tabindex="-1"></a>  }</span>
<span id="cb125-57"><a href="statistical-analysis-1.html#cb125-57" tabindex="-1"></a>  quantVn <span class="ot">&lt;-</span> <span class="fu">qnorm</span>(<span class="dv">1</span> <span class="sc">-</span> pvaVn <span class="sc">/</span> <span class="dv">2</span>)</span>
<span id="cb125-58"><a href="statistical-analysis-1.html#cb125-58" tabindex="-1"></a>  rmsQuantN <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="fu">mean</span>(quantVn<span class="sc">^</span><span class="dv">2</span>))</span>
<span id="cb125-59"><a href="statistical-analysis-1.html#cb125-59" tabindex="-1"></a>  </span>
<span id="cb125-60"><a href="statistical-analysis-1.html#cb125-60" tabindex="-1"></a>  opar <span class="ot">&lt;-</span> <span class="fu">par</span>(<span class="at">font =</span> <span class="dv">2</span>, <span class="at">font.axis =</span> <span class="dv">2</span>, <span class="at">font.lab =</span> <span class="dv">2</span>,</span>
<span id="cb125-61"><a href="statistical-analysis-1.html#cb125-61" tabindex="-1"></a>              <span class="at">las =</span> <span class="dv">1</span>,</span>
<span id="cb125-62"><a href="statistical-analysis-1.html#cb125-62" tabindex="-1"></a>              <span class="at">mar =</span> <span class="fu">c</span>(<span class="fl">5.1</span>, <span class="fl">4.6</span>, <span class="fl">4.1</span>, <span class="fl">2.1</span>),</span>
<span id="cb125-63"><a href="statistical-analysis-1.html#cb125-63" tabindex="-1"></a>              <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">pch =</span> <span class="dv">16</span>)</span>
<span id="cb125-64"><a href="statistical-analysis-1.html#cb125-64" tabindex="-1"></a>  <span class="fu">plot</span>(pvaVn, </span>
<span id="cb125-65"><a href="statistical-analysis-1.html#cb125-65" tabindex="-1"></a>       vipVn,</span>
<span id="cb125-66"><a href="statistical-analysis-1.html#cb125-66" tabindex="-1"></a>       <span class="at">col =</span> <span class="st">&quot;red&quot;</span>,</span>
<span id="cb125-67"><a href="statistical-analysis-1.html#cb125-67" tabindex="-1"></a>       <span class="at">pch =</span> <span class="dv">16</span>,</span>
<span id="cb125-68"><a href="statistical-analysis-1.html#cb125-68" tabindex="-1"></a>       <span class="at">xlab =</span> <span class="st">&quot;p-value&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;VIP&quot;</span>, <span class="at">xaxs =</span> <span class="st">&quot;i&quot;</span>, <span class="at">yaxs =</span> <span class="st">&quot;i&quot;</span>)</span>
<span id="cb125-69"><a href="statistical-analysis-1.html#cb125-69" tabindex="-1"></a>  <span class="fu">box</span>(<span class="at">lwd =</span> <span class="dv">2</span>)</span>
<span id="cb125-70"><a href="statistical-analysis-1.html#cb125-70" tabindex="-1"></a>  <span class="fu">curve</span>(<span class="fu">qnorm</span>(<span class="dv">1</span> <span class="sc">-</span> x <span class="sc">/</span> <span class="dv">2</span>) <span class="sc">/</span> rmsQuantN, <span class="dv">0</span>, <span class="dv">1</span>, <span class="at">add =</span> <span class="cn">TRUE</span>, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>, <span class="at">lwd =</span> <span class="dv">3</span>)</span>
<span id="cb125-71"><a href="statistical-analysis-1.html#cb125-71" tabindex="-1"></a>  <span class="fu">abline</span>(<span class="at">h =</span> <span class="dv">1</span>, <span class="at">col =</span> <span class="st">&quot;blue&quot;</span>)</span>
<span id="cb125-72"><a href="statistical-analysis-1.html#cb125-72" tabindex="-1"></a>  <span class="fu">abline</span>(<span class="at">v =</span> <span class="fl">0.05</span>, <span class="at">col =</span> <span class="st">&quot;blue&quot;</span>)</span>
<span id="cb125-73"><a href="statistical-analysis-1.html#cb125-73" tabindex="-1"></a>  </span>
<span id="cb125-74"><a href="statistical-analysis-1.html#cb125-74" tabindex="-1"></a>  res_temp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb125-75"><a href="statistical-analysis-1.html#cb125-75" tabindex="-1"></a>      <span class="at">FeatureID =</span> <span class="fu">names</span>(vipVn),</span>
<span id="cb125-76"><a href="statistical-analysis-1.html#cb125-76" tabindex="-1"></a>      <span class="at">VIP =</span> vipVn,</span>
<span id="cb125-77"><a href="statistical-analysis-1.html#cb125-77" tabindex="-1"></a>      <span class="at">CorPvalue =</span> pvaVn) <span class="sc">%&gt;%</span></span>
<span id="cb125-78"><a href="statistical-analysis-1.html#cb125-78" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">arrange</span>(<span class="fu">desc</span>(VIP))</span>
<span id="cb125-79"><a href="statistical-analysis-1.html#cb125-79" tabindex="-1"></a></span>
<span id="cb125-80"><a href="statistical-analysis-1.html#cb125-80" tabindex="-1"></a>  vip_select <span class="ot">&lt;-</span> res_temp <span class="sc">%&gt;%</span></span>
<span id="cb125-81"><a href="statistical-analysis-1.html#cb125-81" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(VIP <span class="sc">&gt;</span> vip_cutoff) </span>
<span id="cb125-82"><a href="statistical-analysis-1.html#cb125-82" tabindex="-1"></a>  </span>
<span id="cb125-83"><a href="statistical-analysis-1.html#cb125-83" tabindex="-1"></a>  pl <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(vip_select, <span class="fu">aes</span>(FeatureID, VIP)) <span class="sc">+</span></span>
<span id="cb125-84"><a href="statistical-analysis-1.html#cb125-84" tabindex="-1"></a>    <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> FeatureID, <span class="at">xend =</span> FeatureID,</span>
<span id="cb125-85"><a href="statistical-analysis-1.html#cb125-85" tabindex="-1"></a>                     <span class="at">y =</span> <span class="dv">0</span>, <span class="at">yend =</span> VIP)) <span class="sc">+</span></span>
<span id="cb125-86"><a href="statistical-analysis-1.html#cb125-86" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">21</span>, <span class="at">size =</span> <span class="dv">5</span>, <span class="at">color =</span> <span class="st">&#39;#008000&#39;</span> ,<span class="at">fill =</span> <span class="st">&#39;#008000&#39;</span>) <span class="sc">+</span></span>
<span id="cb125-87"><a href="statistical-analysis-1.html#cb125-87" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="dv">1</span>,<span class="fl">2.5</span>), <span class="at">color =</span> <span class="st">&#39;white&#39;</span>) <span class="sc">+</span></span>
<span id="cb125-88"><a href="statistical-analysis-1.html#cb125-88" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">1</span>, <span class="at">linetype =</span> <span class="st">&#39;dashed&#39;</span>) <span class="sc">+</span></span>
<span id="cb125-89"><a href="statistical-analysis-1.html#cb125-89" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb125-90"><a href="statistical-analysis-1.html#cb125-90" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&#39;&#39;</span>, <span class="at">y =</span> <span class="st">&#39;VIP value&#39;</span>) <span class="sc">+</span></span>
<span id="cb125-91"><a href="statistical-analysis-1.html#cb125-91" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb125-92"><a href="statistical-analysis-1.html#cb125-92" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&#39;none&#39;</span>,</span>
<span id="cb125-93"><a href="statistical-analysis-1.html#cb125-93" tabindex="-1"></a>          <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">&#39;black&#39;</span>,<span class="at">size =</span> <span class="dv">12</span>, <span class="at">family =</span> <span class="st">&#39;Arial&#39;</span>, <span class="at">face =</span> <span class="st">&#39;plain&#39;</span>),</span>
<span id="cb125-94"><a href="statistical-analysis-1.html#cb125-94" tabindex="-1"></a>          <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb125-95"><a href="statistical-analysis-1.html#cb125-95" tabindex="-1"></a>          <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb125-96"><a href="statistical-analysis-1.html#cb125-96" tabindex="-1"></a>          <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">&#39;black&#39;</span>,<span class="at">size =</span> <span class="dv">15</span>, <span class="at">family =</span> <span class="st">&#39;Arial&#39;</span>, <span class="at">face =</span> <span class="st">&#39;plain&#39;</span>),</span>
<span id="cb125-97"><a href="statistical-analysis-1.html#cb125-97" tabindex="-1"></a>          <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>),</span>
<span id="cb125-98"><a href="statistical-analysis-1.html#cb125-98" tabindex="-1"></a>          <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">&#39;black&#39;</span>,<span class="at">size =</span> <span class="dv">15</span>, <span class="at">family =</span> <span class="st">&#39;Arial&#39;</span>, <span class="at">face =</span> <span class="st">&#39;plain&#39;</span>),</span>
<span id="cb125-99"><a href="statistical-analysis-1.html#cb125-99" tabindex="-1"></a>          <span class="at">axis.ticks =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">&#39;black&#39;</span>),</span>
<span id="cb125-100"><a href="statistical-analysis-1.html#cb125-100" tabindex="-1"></a>          <span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>())</span>
<span id="cb125-101"><a href="statistical-analysis-1.html#cb125-101" tabindex="-1"></a>  </span>
<span id="cb125-102"><a href="statistical-analysis-1.html#cb125-102" tabindex="-1"></a>  <span class="co"># Number of Group</span></span>
<span id="cb125-103"><a href="statistical-analysis-1.html#cb125-103" tabindex="-1"></a>  dat_status <span class="ot">&lt;-</span> <span class="fu">table</span>(phen<span class="sc">$</span>CompVar)</span>
<span id="cb125-104"><a href="statistical-analysis-1.html#cb125-104" tabindex="-1"></a>  dat_status_number <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(dat_status)</span>
<span id="cb125-105"><a href="statistical-analysis-1.html#cb125-105" tabindex="-1"></a>  dat_status_name <span class="ot">&lt;-</span> <span class="fu">names</span>(dat_status)</span>
<span id="cb125-106"><a href="statistical-analysis-1.html#cb125-106" tabindex="-1"></a>  res_temp<span class="sc">$</span>Block <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">paste</span>(dat_status_number[<span class="dv">1</span>], dat_status_name[<span class="dv">1</span>], <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>),</span>
<span id="cb125-107"><a href="statistical-analysis-1.html#cb125-107" tabindex="-1"></a>                       <span class="st">&quot;vs&quot;</span>,</span>
<span id="cb125-108"><a href="statistical-analysis-1.html#cb125-108" tabindex="-1"></a>                       <span class="fu">paste</span>(dat_status_number[<span class="dv">2</span>], dat_status_name[<span class="dv">2</span>], <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>))</span>
<span id="cb125-109"><a href="statistical-analysis-1.html#cb125-109" tabindex="-1"></a></span>
<span id="cb125-110"><a href="statistical-analysis-1.html#cb125-110" tabindex="-1"></a>  res_df <span class="ot">&lt;-</span> res_temp <span class="sc">%&gt;%</span></span>
<span id="cb125-111"><a href="statistical-analysis-1.html#cb125-111" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(FeatureID, Block, <span class="fu">everything</span>())    </span>
<span id="cb125-112"><a href="statistical-analysis-1.html#cb125-112" tabindex="-1"></a></span>
<span id="cb125-113"><a href="statistical-analysis-1.html#cb125-113" tabindex="-1"></a></span>
<span id="cb125-114"><a href="statistical-analysis-1.html#cb125-114" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">vip =</span> res_df,</span>
<span id="cb125-115"><a href="statistical-analysis-1.html#cb125-115" tabindex="-1"></a>              <span class="at">plot =</span> pl)</span>
<span id="cb125-116"><a href="statistical-analysis-1.html#cb125-116" tabindex="-1"></a>  </span>
<span id="cb125-117"><a href="statistical-analysis-1.html#cb125-117" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb125-118"><a href="statistical-analysis-1.html#cb125-118" tabindex="-1"></a>}</span>
<span id="cb125-119"><a href="statistical-analysis-1.html#cb125-119" tabindex="-1"></a></span>
<span id="cb125-120"><a href="statistical-analysis-1.html#cb125-120" tabindex="-1"></a>vip_res <span class="ot">&lt;-</span> <span class="fu">VIP_fun</span>(</span>
<span id="cb125-121"><a href="statistical-analysis-1.html#cb125-121" tabindex="-1"></a>  <span class="at">x =</span> se_normalize,</span>
<span id="cb125-122"><a href="statistical-analysis-1.html#cb125-122" tabindex="-1"></a>  <span class="at">group =</span> <span class="st">&quot;group&quot;</span>,</span>
<span id="cb125-123"><a href="statistical-analysis-1.html#cb125-123" tabindex="-1"></a>  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Severe&quot;</span>),</span>
<span id="cb125-124"><a href="statistical-analysis-1.html#cb125-124" tabindex="-1"></a>  <span class="at">VIPtype =</span> <span class="st">&quot;PLS&quot;</span>,</span>
<span id="cb125-125"><a href="statistical-analysis-1.html#cb125-125" tabindex="-1"></a>  <span class="at">vip_cutoff =</span> <span class="dv">1</span>)</span></code></pre></div>
<pre><code>## PLS-DA
## 26 samples x 167 variables and 1 response
## standard scaling of predictors and response(s)
##       R2X(cum) R2Y(cum) Q2(cum) RMSEE pre ort pR2Y  pQ2
## Total     0.12    0.692    0.33 0.288   1   0  0.1 0.05</code></pre>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-141-1.png" width="672" /></p>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="statistical-analysis-1.html#cb127-1" tabindex="-1"></a><span class="fu">head</span>(vip_res<span class="sc">$</span>vip)</span></code></pre></div>
<pre><code>##         FeatureID                Block      VIP   CorPvalue
## M_42449   M_42449 14_Mild vs 12_Severe 2.362694 0.002247897
## M_62566   M_62566 14_Mild vs 12_Severe 2.288128 0.003300438
## M_1566     M_1566 14_Mild vs 12_Severe 2.143655 0.006556886
## M_19263   M_19263 14_Mild vs 12_Severe 2.128317 0.007023483
## M_52446   M_52446 14_Mild vs 12_Severe 2.059461 0.009476225
## M_37482   M_37482 14_Mild vs 12_Severe 2.052135 0.009774742</code></pre>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="statistical-analysis-1.html#cb129-1" tabindex="-1"></a>vip_res<span class="sc">$</span>plot</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-141-2.png" width="672" /></p>
</div>
<div id="t-test-by-local-codes" class="section level3 hasAnchor" number="3.4.9">
<h3><span class="header-section-number">3.4.9</span> T-test by local codes<a href="statistical-analysis-1.html#t-test-by-local-codes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>significant differences between two groups (p value)</li>
</ul>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="statistical-analysis-1.html#cb130-1" tabindex="-1"></a>t_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb130-2"><a href="statistical-analysis-1.html#cb130-2" tabindex="-1"></a>    x,</span>
<span id="cb130-3"><a href="statistical-analysis-1.html#cb130-3" tabindex="-1"></a>    group,</span>
<span id="cb130-4"><a href="statistical-analysis-1.html#cb130-4" tabindex="-1"></a>    group_names) {</span>
<span id="cb130-5"><a href="statistical-analysis-1.html#cb130-5" tabindex="-1"></a>  </span>
<span id="cb130-6"><a href="statistical-analysis-1.html#cb130-6" tabindex="-1"></a>  <span class="co"># x = se_normalize</span></span>
<span id="cb130-7"><a href="statistical-analysis-1.html#cb130-7" tabindex="-1"></a>  <span class="co"># group = &quot;group&quot;</span></span>
<span id="cb130-8"><a href="statistical-analysis-1.html#cb130-8" tabindex="-1"></a>  <span class="co"># group_names = c(&quot;Mild&quot;, &quot;Severe&quot;)</span></span>
<span id="cb130-9"><a href="statistical-analysis-1.html#cb130-9" tabindex="-1"></a>  </span>
<span id="cb130-10"><a href="statistical-analysis-1.html#cb130-10" tabindex="-1"></a>  <span class="co"># dataseat</span></span>
<span id="cb130-11"><a href="statistical-analysis-1.html#cb130-11" tabindex="-1"></a>  metadata <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">colData</span>(x) <span class="sc">%&gt;%</span></span>
<span id="cb130-12"><a href="statistical-analysis-1.html#cb130-12" tabindex="-1"></a>      <span class="fu">as.data.frame</span>()</span>
<span id="cb130-13"><a href="statistical-analysis-1.html#cb130-13" tabindex="-1"></a>  profile <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(x) <span class="sc">%&gt;%</span></span>
<span id="cb130-14"><a href="statistical-analysis-1.html#cb130-14" tabindex="-1"></a>      <span class="fu">as.data.frame</span>()</span>
<span id="cb130-15"><a href="statistical-analysis-1.html#cb130-15" tabindex="-1"></a></span>
<span id="cb130-16"><a href="statistical-analysis-1.html#cb130-16" tabindex="-1"></a>  <span class="co"># rename variables</span></span>
<span id="cb130-17"><a href="statistical-analysis-1.html#cb130-17" tabindex="-1"></a>  <span class="fu">colnames</span>(metadata)[<span class="fu">which</span>(<span class="fu">colnames</span>(metadata) <span class="sc">==</span> group)] <span class="ot">&lt;-</span> <span class="st">&quot;CompVar&quot;</span> </span>
<span id="cb130-18"><a href="statistical-analysis-1.html#cb130-18" tabindex="-1"></a>  </span>
<span id="cb130-19"><a href="statistical-analysis-1.html#cb130-19" tabindex="-1"></a>  phenotype <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span></span>
<span id="cb130-20"><a href="statistical-analysis-1.html#cb130-20" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(CompVar <span class="sc">%in%</span> group_names) <span class="sc">%&gt;%</span></span>
<span id="cb130-21"><a href="statistical-analysis-1.html#cb130-21" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">CompVar =</span> <span class="fu">as.character</span>(CompVar)) <span class="sc">%&gt;%</span></span>
<span id="cb130-22"><a href="statistical-analysis-1.html#cb130-22" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">CompVar =</span> <span class="fu">factor</span>(CompVar, <span class="at">levels =</span> group_names))</span>
<span id="cb130-23"><a href="statistical-analysis-1.html#cb130-23" tabindex="-1"></a>  </span>
<span id="cb130-24"><a href="statistical-analysis-1.html#cb130-24" tabindex="-1"></a>  sid <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(phenotype), <span class="fu">colnames</span>(profile))</span>
<span id="cb130-25"><a href="statistical-analysis-1.html#cb130-25" tabindex="-1"></a>  phen <span class="ot">&lt;-</span> phenotype[<span class="fu">pmatch</span>(sid, <span class="fu">rownames</span>(phenotype)), , ]</span>
<span id="cb130-26"><a href="statistical-analysis-1.html#cb130-26" tabindex="-1"></a>  prof <span class="ot">&lt;-</span> profile <span class="sc">%&gt;%</span></span>
<span id="cb130-27"><a href="statistical-analysis-1.html#cb130-27" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(sid))</span>
<span id="cb130-28"><a href="statistical-analysis-1.html#cb130-28" tabindex="-1"></a>  </span>
<span id="cb130-29"><a href="statistical-analysis-1.html#cb130-29" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">all</span>(<span class="fu">colnames</span>(prof) <span class="sc">==</span> <span class="fu">rownames</span>(phen))) {</span>
<span id="cb130-30"><a href="statistical-analysis-1.html#cb130-30" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;Wrong Order&quot;</span>)</span>
<span id="cb130-31"><a href="statistical-analysis-1.html#cb130-31" tabindex="-1"></a>  }</span>
<span id="cb130-32"><a href="statistical-analysis-1.html#cb130-32" tabindex="-1"></a></span>
<span id="cb130-33"><a href="statistical-analysis-1.html#cb130-33" tabindex="-1"></a>  t_res <span class="ot">&lt;-</span> <span class="fu">apply</span>(prof, <span class="dv">1</span>, <span class="cf">function</span>(x1, y1) {</span>
<span id="cb130-34"><a href="statistical-analysis-1.html#cb130-34" tabindex="-1"></a>    dat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">value =</span> <span class="fu">as.numeric</span>(x1), <span class="at">group =</span> y1)</span>
<span id="cb130-35"><a href="statistical-analysis-1.html#cb130-35" tabindex="-1"></a>    </span>
<span id="cb130-36"><a href="statistical-analysis-1.html#cb130-36" tabindex="-1"></a>    rest <span class="ot">&lt;-</span> <span class="fu">t.test</span>(<span class="at">data =</span> dat, value <span class="sc">~</span> group)</span>
<span id="cb130-37"><a href="statistical-analysis-1.html#cb130-37" tabindex="-1"></a></span>
<span id="cb130-38"><a href="statistical-analysis-1.html#cb130-38" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="fu">c</span>(rest<span class="sc">$</span>statistic, rest<span class="sc">$</span>p.value)</span>
<span id="cb130-39"><a href="statistical-analysis-1.html#cb130-39" tabindex="-1"></a>    <span class="fu">return</span>(res)</span>
<span id="cb130-40"><a href="statistical-analysis-1.html#cb130-40" tabindex="-1"></a>  }, phen<span class="sc">$</span>CompVar) <span class="sc">%&gt;%</span></span>
<span id="cb130-41"><a href="statistical-analysis-1.html#cb130-41" tabindex="-1"></a>    base<span class="sc">::</span><span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb130-42"><a href="statistical-analysis-1.html#cb130-42" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">&quot;Feature&quot;</span>)</span>
<span id="cb130-43"><a href="statistical-analysis-1.html#cb130-43" tabindex="-1"></a>  </span>
<span id="cb130-44"><a href="statistical-analysis-1.html#cb130-44" tabindex="-1"></a>  <span class="fu">colnames</span>(t_res) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;FeatureID&quot;</span>, <span class="st">&quot;Statistic&quot;</span>, <span class="st">&quot;Pvalue&quot;</span>)</span>
<span id="cb130-45"><a href="statistical-analysis-1.html#cb130-45" tabindex="-1"></a>  t_res<span class="sc">$</span>AdjustedPvalue <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(<span class="fu">as.numeric</span>(t_res<span class="sc">$</span>Pvalue), <span class="at">method =</span> <span class="st">&quot;BH&quot;</span>)</span>
<span id="cb130-46"><a href="statistical-analysis-1.html#cb130-46" tabindex="-1"></a>  </span>
<span id="cb130-47"><a href="statistical-analysis-1.html#cb130-47" tabindex="-1"></a>  <span class="co"># Number of Group</span></span>
<span id="cb130-48"><a href="statistical-analysis-1.html#cb130-48" tabindex="-1"></a>  dat_status <span class="ot">&lt;-</span> <span class="fu">table</span>(phen<span class="sc">$</span>CompVar)</span>
<span id="cb130-49"><a href="statistical-analysis-1.html#cb130-49" tabindex="-1"></a>  dat_status_number <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(dat_status)</span>
<span id="cb130-50"><a href="statistical-analysis-1.html#cb130-50" tabindex="-1"></a>  dat_status_name <span class="ot">&lt;-</span> <span class="fu">names</span>(dat_status)</span>
<span id="cb130-51"><a href="statistical-analysis-1.html#cb130-51" tabindex="-1"></a>  t_res<span class="sc">$</span>Block <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">paste</span>(dat_status_number[<span class="dv">1</span>], dat_status_name[<span class="dv">1</span>], <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>),</span>
<span id="cb130-52"><a href="statistical-analysis-1.html#cb130-52" tabindex="-1"></a>                       <span class="st">&quot;vs&quot;</span>,</span>
<span id="cb130-53"><a href="statistical-analysis-1.html#cb130-53" tabindex="-1"></a>                       <span class="fu">paste</span>(dat_status_number[<span class="dv">2</span>], dat_status_name[<span class="dv">2</span>], <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>))</span>
<span id="cb130-54"><a href="statistical-analysis-1.html#cb130-54" tabindex="-1"></a>  </span>
<span id="cb130-55"><a href="statistical-analysis-1.html#cb130-55" tabindex="-1"></a>  res <span class="ot">&lt;-</span> t_res <span class="sc">%&gt;%</span></span>
<span id="cb130-56"><a href="statistical-analysis-1.html#cb130-56" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(FeatureID, Block, <span class="fu">everything</span>())    </span>
<span id="cb130-57"><a href="statistical-analysis-1.html#cb130-57" tabindex="-1"></a></span>
<span id="cb130-58"><a href="statistical-analysis-1.html#cb130-58" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb130-59"><a href="statistical-analysis-1.html#cb130-59" tabindex="-1"></a>}</span>
<span id="cb130-60"><a href="statistical-analysis-1.html#cb130-60" tabindex="-1"></a></span>
<span id="cb130-61"><a href="statistical-analysis-1.html#cb130-61" tabindex="-1"></a>ttest_res <span class="ot">&lt;-</span> <span class="fu">t_fun</span>(</span>
<span id="cb130-62"><a href="statistical-analysis-1.html#cb130-62" tabindex="-1"></a>  <span class="at">x =</span> se_normalize,</span>
<span id="cb130-63"><a href="statistical-analysis-1.html#cb130-63" tabindex="-1"></a>  <span class="at">group =</span> <span class="st">&quot;group&quot;</span>,</span>
<span id="cb130-64"><a href="statistical-analysis-1.html#cb130-64" tabindex="-1"></a>  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Severe&quot;</span>))</span>
<span id="cb130-65"><a href="statistical-analysis-1.html#cb130-65" tabindex="-1"></a></span>
<span id="cb130-66"><a href="statistical-analysis-1.html#cb130-66" tabindex="-1"></a><span class="fu">head</span>(ttest_res)</span></code></pre></div>
<pre><code>##   FeatureID                Block  Statistic    Pvalue AdjustedPvalue
## 1   M_38768 14_Mild vs 12_Severe -0.5018649 0.6205743      0.8928355
## 2   M_38296 14_Mild vs 12_Severe -1.0405745 0.3094567      0.7300466
## 3   M_63436 14_Mild vs 12_Severe -0.5398558 0.5942962      0.8782962
## 4   M_57814 14_Mild vs 12_Severe  0.2204845 0.8274429      0.9515249
## 5   M_52603 14_Mild vs 12_Severe -0.2952738 0.7709351      0.9515249
## 6   M_53174 14_Mild vs 12_Severe -0.6134155 0.5461908      0.8782962</code></pre>
</div>
<div id="merging-result" class="section level3 hasAnchor" number="3.4.10">
<h3><span class="header-section-number">3.4.10</span> Merging result<a href="statistical-analysis-1.html#merging-result" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><p>Foldchange by Raw Data</p></li>
<li><p>VIP by Normalized Data</p></li>
<li><p>test Pvalue by Normalized Data</p></li>
</ul>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="statistical-analysis-1.html#cb132-1" tabindex="-1"></a>mergedResults <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb132-2"><a href="statistical-analysis-1.html#cb132-2" tabindex="-1"></a>    fc_result,</span>
<span id="cb132-3"><a href="statistical-analysis-1.html#cb132-3" tabindex="-1"></a>    vip_result,</span>
<span id="cb132-4"><a href="statistical-analysis-1.html#cb132-4" tabindex="-1"></a>    test_result,</span>
<span id="cb132-5"><a href="statistical-analysis-1.html#cb132-5" tabindex="-1"></a>    group_names,</span>
<span id="cb132-6"><a href="statistical-analysis-1.html#cb132-6" tabindex="-1"></a>    group_labels) {</span>
<span id="cb132-7"><a href="statistical-analysis-1.html#cb132-7" tabindex="-1"></a>  </span>
<span id="cb132-8"><a href="statistical-analysis-1.html#cb132-8" tabindex="-1"></a>  <span class="co"># fc_result = fc_res</span></span>
<span id="cb132-9"><a href="statistical-analysis-1.html#cb132-9" tabindex="-1"></a>  <span class="co"># vip_result = vip_res$vip</span></span>
<span id="cb132-10"><a href="statistical-analysis-1.html#cb132-10" tabindex="-1"></a>  <span class="co"># test_result = ttest_res</span></span>
<span id="cb132-11"><a href="statistical-analysis-1.html#cb132-11" tabindex="-1"></a>  <span class="co"># group_names = c(&quot;Mild&quot;, &quot;Severe&quot;)</span></span>
<span id="cb132-12"><a href="statistical-analysis-1.html#cb132-12" tabindex="-1"></a>  <span class="co"># group_labels = c(&quot;Mild&quot;, &quot;Severe&quot;)</span></span>
<span id="cb132-13"><a href="statistical-analysis-1.html#cb132-13" tabindex="-1"></a>  </span>
<span id="cb132-14"><a href="statistical-analysis-1.html#cb132-14" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">is.null</span>(vip_result)) {</span>
<span id="cb132-15"><a href="statistical-analysis-1.html#cb132-15" tabindex="-1"></a>    mdat <span class="ot">&lt;-</span> fc_result <span class="sc">%&gt;%</span></span>
<span id="cb132-16"><a href="statistical-analysis-1.html#cb132-16" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Block2 =</span> <span class="fu">paste</span>(group_labels, <span class="at">collapse =</span> <span class="st">&quot; vs &quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb132-17"><a href="statistical-analysis-1.html#cb132-17" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">FeatureID =</span> <span class="fu">make.names</span>(FeatureID)) <span class="sc">%&gt;%</span></span>
<span id="cb132-18"><a href="statistical-analysis-1.html#cb132-18" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">&quot;Mean Abundance</span><span class="sc">\n</span><span class="st">(All)&quot;</span>, </span>
<span id="cb132-19"><a href="statistical-analysis-1.html#cb132-19" tabindex="-1"></a>                              <span class="st">&quot;Mean Abundance</span><span class="sc">\n</span><span class="st">former&quot;</span>, </span>
<span id="cb132-20"><a href="statistical-analysis-1.html#cb132-20" tabindex="-1"></a>                              <span class="st">&quot;Mean Abundance</span><span class="sc">\n</span><span class="st">latter&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb132-21"><a href="statistical-analysis-1.html#cb132-21" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">inner_join</span>(test_result <span class="sc">%&gt;%</span></span>
<span id="cb132-22"><a href="statistical-analysis-1.html#cb132-22" tabindex="-1"></a>                          dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>Block) <span class="sc">%&gt;%</span></span>
<span id="cb132-23"><a href="statistical-analysis-1.html#cb132-23" tabindex="-1"></a>                          dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">FeatureID =</span> <span class="fu">make.names</span>(FeatureID)),</span>
<span id="cb132-24"><a href="statistical-analysis-1.html#cb132-24" tabindex="-1"></a>                        <span class="at">by =</span> <span class="st">&quot;FeatureID&quot;</span>) </span>
<span id="cb132-25"><a href="statistical-analysis-1.html#cb132-25" tabindex="-1"></a>    </span>
<span id="cb132-26"><a href="statistical-analysis-1.html#cb132-26" tabindex="-1"></a>    res <span class="ot">&lt;-</span> mdat <span class="sc">%&gt;%</span></span>
<span id="cb132-27"><a href="statistical-analysis-1.html#cb132-27" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(FeatureID, Block2, Block,</span>
<span id="cb132-28"><a href="statistical-analysis-1.html#cb132-28" tabindex="-1"></a>                    FoldChange, Log2FoldChange,</span>
<span id="cb132-29"><a href="statistical-analysis-1.html#cb132-29" tabindex="-1"></a>                    Statistic, Pvalue, AdjustedPvalue,</span>
<span id="cb132-30"><a href="statistical-analysis-1.html#cb132-30" tabindex="-1"></a>                    <span class="fu">everything</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb132-31"><a href="statistical-analysis-1.html#cb132-31" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">arrange</span>(AdjustedPvalue, Log2FoldChange)    </span>
<span id="cb132-32"><a href="statistical-analysis-1.html#cb132-32" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb132-33"><a href="statistical-analysis-1.html#cb132-33" tabindex="-1"></a>    mdat <span class="ot">&lt;-</span> fc_result <span class="sc">%&gt;%</span></span>
<span id="cb132-34"><a href="statistical-analysis-1.html#cb132-34" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">Block2 =</span> <span class="fu">paste</span>(group_labels, <span class="at">collapse =</span> <span class="st">&quot; vs &quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb132-35"><a href="statistical-analysis-1.html#cb132-35" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">FeatureID =</span> <span class="fu">make.names</span>(FeatureID)) <span class="sc">%&gt;%</span></span>
<span id="cb132-36"><a href="statistical-analysis-1.html#cb132-36" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">&quot;Mean Abundance</span><span class="sc">\n</span><span class="st">(All)&quot;</span>, </span>
<span id="cb132-37"><a href="statistical-analysis-1.html#cb132-37" tabindex="-1"></a>                              <span class="st">&quot;Mean Abundance</span><span class="sc">\n</span><span class="st">former&quot;</span>, </span>
<span id="cb132-38"><a href="statistical-analysis-1.html#cb132-38" tabindex="-1"></a>                              <span class="st">&quot;Mean Abundance</span><span class="sc">\n</span><span class="st">latter&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb132-39"><a href="statistical-analysis-1.html#cb132-39" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">inner_join</span>(vip_result <span class="sc">%&gt;%</span></span>
<span id="cb132-40"><a href="statistical-analysis-1.html#cb132-40" tabindex="-1"></a>                          dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>Block) <span class="sc">%&gt;%</span></span>
<span id="cb132-41"><a href="statistical-analysis-1.html#cb132-41" tabindex="-1"></a>                          dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">FeatureID =</span> <span class="fu">make.names</span>(FeatureID)),</span>
<span id="cb132-42"><a href="statistical-analysis-1.html#cb132-42" tabindex="-1"></a>                        <span class="at">by =</span> <span class="st">&quot;FeatureID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb132-43"><a href="statistical-analysis-1.html#cb132-43" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">inner_join</span>(test_result <span class="sc">%&gt;%</span></span>
<span id="cb132-44"><a href="statistical-analysis-1.html#cb132-44" tabindex="-1"></a>                          dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>Block) <span class="sc">%&gt;%</span></span>
<span id="cb132-45"><a href="statistical-analysis-1.html#cb132-45" tabindex="-1"></a>                          dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">FeatureID =</span> <span class="fu">make.names</span>(FeatureID)),</span>
<span id="cb132-46"><a href="statistical-analysis-1.html#cb132-46" tabindex="-1"></a>                        <span class="at">by =</span> <span class="st">&quot;FeatureID&quot;</span>) </span>
<span id="cb132-47"><a href="statistical-analysis-1.html#cb132-47" tabindex="-1"></a>    </span>
<span id="cb132-48"><a href="statistical-analysis-1.html#cb132-48" tabindex="-1"></a>    res <span class="ot">&lt;-</span> mdat <span class="sc">%&gt;%</span></span>
<span id="cb132-49"><a href="statistical-analysis-1.html#cb132-49" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">select</span>(FeatureID, Block2, Block,</span>
<span id="cb132-50"><a href="statistical-analysis-1.html#cb132-50" tabindex="-1"></a>                    FoldChange, Log2FoldChange,</span>
<span id="cb132-51"><a href="statistical-analysis-1.html#cb132-51" tabindex="-1"></a>                    VIP, CorPvalue,</span>
<span id="cb132-52"><a href="statistical-analysis-1.html#cb132-52" tabindex="-1"></a>                    Statistic, Pvalue, AdjustedPvalue,</span>
<span id="cb132-53"><a href="statistical-analysis-1.html#cb132-53" tabindex="-1"></a>                    <span class="fu">everything</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb132-54"><a href="statistical-analysis-1.html#cb132-54" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">arrange</span>(AdjustedPvalue, Log2FoldChange)    </span>
<span id="cb132-55"><a href="statistical-analysis-1.html#cb132-55" tabindex="-1"></a>  }</span>
<span id="cb132-56"><a href="statistical-analysis-1.html#cb132-56" tabindex="-1"></a>  </span>
<span id="cb132-57"><a href="statistical-analysis-1.html#cb132-57" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb132-58"><a href="statistical-analysis-1.html#cb132-58" tabindex="-1"></a>}</span>
<span id="cb132-59"><a href="statistical-analysis-1.html#cb132-59" tabindex="-1"></a></span>
<span id="cb132-60"><a href="statistical-analysis-1.html#cb132-60" tabindex="-1"></a>m_results <span class="ot">&lt;-</span> <span class="fu">mergedResults</span>(</span>
<span id="cb132-61"><a href="statistical-analysis-1.html#cb132-61" tabindex="-1"></a>  <span class="at">fc_result =</span> fc_res,</span>
<span id="cb132-62"><a href="statistical-analysis-1.html#cb132-62" tabindex="-1"></a>  <span class="at">vip_result =</span> vip_res<span class="sc">$</span>vip,</span>
<span id="cb132-63"><a href="statistical-analysis-1.html#cb132-63" tabindex="-1"></a>  <span class="at">test_result =</span> ttest_res,</span>
<span id="cb132-64"><a href="statistical-analysis-1.html#cb132-64" tabindex="-1"></a>  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Severe&quot;</span>),</span>
<span id="cb132-65"><a href="statistical-analysis-1.html#cb132-65" tabindex="-1"></a>  <span class="at">group_labels =</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Severe&quot;</span>))</span>
<span id="cb132-66"><a href="statistical-analysis-1.html#cb132-66" tabindex="-1"></a></span>
<span id="cb132-67"><a href="statistical-analysis-1.html#cb132-67" tabindex="-1"></a><span class="fu">head</span>(m_results)</span></code></pre></div>
<pre><code>##   FeatureID         Block2                  Block FoldChange Log2FoldChange      VIP   CorPvalue Statistic      Pvalue AdjustedPvalue
## 1   M_42449 Mild vs Severe 14_Mild vs 19_Moderate  0.6571801     -0.6056393 2.362694 0.002247897 -3.498896 0.001881585      0.2132448
## 2   M_52446 Mild vs Severe 14_Mild vs 19_Moderate  0.7040861     -0.5061763 2.059461 0.009476225 -2.892892 0.008125613      0.2132448
## 3    M_1566 Mild vs Severe 14_Mild vs 19_Moderate  0.7161609     -0.4816443 2.143655 0.006556886 -3.077688 0.005431005      0.2132448
## 4   M_19263 Mild vs Severe 14_Mild vs 19_Moderate  0.7165011     -0.4809592 2.128317 0.007023483 -3.073546 0.005774977      0.2132448
## 5   M_42448 Mild vs Severe 14_Mild vs 19_Moderate  0.7577644     -0.4001788 2.006246 0.011828403 -2.793941 0.010215320      0.2132448
## 6   M_43761 Mild vs Severe 14_Mild vs 19_Moderate  0.8685495     -0.2033201 2.043515 0.010136007 -2.822338 0.009428795      0.2132448</code></pre>
</div>
<div id="volcano-of-merged-results" class="section level3 hasAnchor" number="3.4.11">
<h3><span class="header-section-number">3.4.11</span> Volcano of Merged Results<a href="statistical-analysis-1.html#volcano-of-merged-results" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="statistical-analysis-1.html#cb134-1" tabindex="-1"></a>get_volcano <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb134-2"><a href="statistical-analysis-1.html#cb134-2" tabindex="-1"></a>    inputdata,</span>
<span id="cb134-3"><a href="statistical-analysis-1.html#cb134-3" tabindex="-1"></a>    group_names,</span>
<span id="cb134-4"><a href="statistical-analysis-1.html#cb134-4" tabindex="-1"></a>    group_labels,</span>
<span id="cb134-5"><a href="statistical-analysis-1.html#cb134-5" tabindex="-1"></a>    group_colors,</span>
<span id="cb134-6"><a href="statistical-analysis-1.html#cb134-6" tabindex="-1"></a>    x_index,</span>
<span id="cb134-7"><a href="statistical-analysis-1.html#cb134-7" tabindex="-1"></a>    x_cutoff,</span>
<span id="cb134-8"><a href="statistical-analysis-1.html#cb134-8" tabindex="-1"></a>    y_index,</span>
<span id="cb134-9"><a href="statistical-analysis-1.html#cb134-9" tabindex="-1"></a>    y_cutoff,</span>
<span id="cb134-10"><a href="statistical-analysis-1.html#cb134-10" tabindex="-1"></a>    <span class="at">plot =</span> <span class="cn">TRUE</span>) {</span>
<span id="cb134-11"><a href="statistical-analysis-1.html#cb134-11" tabindex="-1"></a>  </span>
<span id="cb134-12"><a href="statistical-analysis-1.html#cb134-12" tabindex="-1"></a>  <span class="co"># inputdata = m_results</span></span>
<span id="cb134-13"><a href="statistical-analysis-1.html#cb134-13" tabindex="-1"></a>  <span class="co"># group_names = c(&quot;Mild&quot;, &quot;Severe&quot;)</span></span>
<span id="cb134-14"><a href="statistical-analysis-1.html#cb134-14" tabindex="-1"></a>  <span class="co"># group_labels = c(&quot;Mild&quot;, &quot;Severe&quot;)</span></span>
<span id="cb134-15"><a href="statistical-analysis-1.html#cb134-15" tabindex="-1"></a>  <span class="co"># group_colors = c(&quot;red&quot;, &quot;blue&quot;)</span></span>
<span id="cb134-16"><a href="statistical-analysis-1.html#cb134-16" tabindex="-1"></a>  <span class="co"># x_index = &quot;Log2FoldChange&quot;</span></span>
<span id="cb134-17"><a href="statistical-analysis-1.html#cb134-17" tabindex="-1"></a>  <span class="co"># x_cutoff = 0.5</span></span>
<span id="cb134-18"><a href="statistical-analysis-1.html#cb134-18" tabindex="-1"></a>  <span class="co"># y_index = &quot;AdjustedPvalue&quot;</span></span>
<span id="cb134-19"><a href="statistical-analysis-1.html#cb134-19" tabindex="-1"></a>  <span class="co"># y_cutoff = 0.5</span></span>
<span id="cb134-20"><a href="statistical-analysis-1.html#cb134-20" tabindex="-1"></a>  <span class="co"># plot = FALSE</span></span>
<span id="cb134-21"><a href="statistical-analysis-1.html#cb134-21" tabindex="-1"></a>  </span>
<span id="cb134-22"><a href="statistical-analysis-1.html#cb134-22" tabindex="-1"></a>  <span class="co"># selected_group &lt;- paste(group_names, collapse = &quot; vs &quot;) </span></span>
<span id="cb134-23"><a href="statistical-analysis-1.html#cb134-23" tabindex="-1"></a>  selected_group2 <span class="ot">&lt;-</span> <span class="fu">paste</span>(group_labels, <span class="at">collapse =</span> <span class="st">&quot; vs &quot;</span>) </span>
<span id="cb134-24"><a href="statistical-analysis-1.html#cb134-24" tabindex="-1"></a>  dat <span class="ot">&lt;-</span> inputdata <span class="sc">%&gt;%</span></span>
<span id="cb134-25"><a href="statistical-analysis-1.html#cb134-25" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(Block2 <span class="sc">%in%</span> selected_group2) </span>
<span id="cb134-26"><a href="statistical-analysis-1.html#cb134-26" tabindex="-1"></a>  plotdata <span class="ot">&lt;-</span> dat <span class="sc">%&gt;%</span></span>
<span id="cb134-27"><a href="statistical-analysis-1.html#cb134-27" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">FeatureID =</span> <span class="fu">paste</span>(FeatureID, <span class="at">sep =</span> <span class="st">&quot;:&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb134-28"><a href="statistical-analysis-1.html#cb134-28" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(<span class="fu">c</span>(<span class="st">&quot;FeatureID&quot;</span>, <span class="st">&quot;Block2&quot;</span>, x_index, y_index)))</span>
<span id="cb134-29"><a href="statistical-analysis-1.html#cb134-29" tabindex="-1"></a>  </span>
<span id="cb134-30"><a href="statistical-analysis-1.html#cb134-30" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">any</span>(<span class="fu">colnames</span>(plotdata) <span class="sc">%in%</span> <span class="st">&quot;TaxaID&quot;</span>)) {</span>
<span id="cb134-31"><a href="statistical-analysis-1.html#cb134-31" tabindex="-1"></a>    <span class="fu">colnames</span>(plotdata)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;TaxaID&quot;</span></span>
<span id="cb134-32"><a href="statistical-analysis-1.html#cb134-32" tabindex="-1"></a>  }</span>
<span id="cb134-33"><a href="statistical-analysis-1.html#cb134-33" tabindex="-1"></a>  <span class="cf">if</span> (y_index <span class="sc">==</span> <span class="st">&quot;CorPvalue&quot;</span>) {</span>
<span id="cb134-34"><a href="statistical-analysis-1.html#cb134-34" tabindex="-1"></a>    <span class="fu">colnames</span>(plotdata)[<span class="fu">which</span>(<span class="fu">colnames</span>(plotdata) <span class="sc">==</span> y_index)] <span class="ot">&lt;-</span> <span class="st">&quot;Pvalue&quot;</span></span>
<span id="cb134-35"><a href="statistical-analysis-1.html#cb134-35" tabindex="-1"></a>    y_index <span class="ot">&lt;-</span> <span class="st">&quot;Pvalue&quot;</span></span>
<span id="cb134-36"><a href="statistical-analysis-1.html#cb134-36" tabindex="-1"></a>  }</span>
<span id="cb134-37"><a href="statistical-analysis-1.html#cb134-37" tabindex="-1"></a> </span>
<span id="cb134-38"><a href="statistical-analysis-1.html#cb134-38" tabindex="-1"></a>  </span>
<span id="cb134-39"><a href="statistical-analysis-1.html#cb134-39" tabindex="-1"></a>  pl <span class="ot">&lt;-</span> <span class="fu">plot_volcano</span>(</span>
<span id="cb134-40"><a href="statistical-analysis-1.html#cb134-40" tabindex="-1"></a>      <span class="at">da_res =</span> plotdata,</span>
<span id="cb134-41"><a href="statistical-analysis-1.html#cb134-41" tabindex="-1"></a>      <span class="at">group_names =</span> group_labels,</span>
<span id="cb134-42"><a href="statistical-analysis-1.html#cb134-42" tabindex="-1"></a>      <span class="at">x_index =</span> x_index,</span>
<span id="cb134-43"><a href="statistical-analysis-1.html#cb134-43" tabindex="-1"></a>      <span class="at">x_index_cutoff =</span> x_cutoff,</span>
<span id="cb134-44"><a href="statistical-analysis-1.html#cb134-44" tabindex="-1"></a>      <span class="at">y_index =</span> y_index,</span>
<span id="cb134-45"><a href="statistical-analysis-1.html#cb134-45" tabindex="-1"></a>      <span class="at">y_index_cutoff =</span> y_cutoff,</span>
<span id="cb134-46"><a href="statistical-analysis-1.html#cb134-46" tabindex="-1"></a>      <span class="at">group_color =</span> <span class="fu">c</span>(group_colors[<span class="dv">1</span>], <span class="st">&quot;grey&quot;</span>, group_colors[<span class="dv">2</span>]),</span>
<span id="cb134-47"><a href="statistical-analysis-1.html#cb134-47" tabindex="-1"></a>      <span class="at">add_enrich_arrow =</span> <span class="cn">TRUE</span>)</span>
<span id="cb134-48"><a href="statistical-analysis-1.html#cb134-48" tabindex="-1"></a>  </span>
<span id="cb134-49"><a href="statistical-analysis-1.html#cb134-49" tabindex="-1"></a>  <span class="cf">if</span> (plot) {</span>
<span id="cb134-50"><a href="statistical-analysis-1.html#cb134-50" tabindex="-1"></a>    res <span class="ot">&lt;-</span> pl</span>
<span id="cb134-51"><a href="statistical-analysis-1.html#cb134-51" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb134-52"><a href="statistical-analysis-1.html#cb134-52" tabindex="-1"></a>    <span class="fu">colnames</span>(plotdata)[<span class="fu">which</span>(<span class="fu">colnames</span>(plotdata) <span class="sc">==</span> x_index)] <span class="ot">&lt;-</span> <span class="st">&quot;Xindex&quot;</span></span>
<span id="cb134-53"><a href="statistical-analysis-1.html#cb134-53" tabindex="-1"></a>    <span class="fu">colnames</span>(plotdata)[<span class="fu">which</span>(<span class="fu">colnames</span>(plotdata) <span class="sc">==</span> y_index)] <span class="ot">&lt;-</span> <span class="st">&quot;Yindex&quot;</span></span>
<span id="cb134-54"><a href="statistical-analysis-1.html#cb134-54" tabindex="-1"></a>    </span>
<span id="cb134-55"><a href="statistical-analysis-1.html#cb134-55" tabindex="-1"></a>    datsignif <span class="ot">&lt;-</span> plotdata <span class="sc">%&gt;%</span></span>
<span id="cb134-56"><a href="statistical-analysis-1.html#cb134-56" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">abs</span>(Xindex) <span class="sc">&gt;</span> x_cutoff) <span class="sc">%&gt;%</span></span>
<span id="cb134-57"><a href="statistical-analysis-1.html#cb134-57" tabindex="-1"></a>      dplyr<span class="sc">::</span><span class="fu">filter</span>(Yindex <span class="sc">&lt;</span> y_cutoff)</span>
<span id="cb134-58"><a href="statistical-analysis-1.html#cb134-58" tabindex="-1"></a>    </span>
<span id="cb134-59"><a href="statistical-analysis-1.html#cb134-59" tabindex="-1"></a>    <span class="fu">colnames</span>(datsignif)[<span class="fu">which</span>(<span class="fu">colnames</span>(datsignif) <span class="sc">==</span> <span class="st">&quot;Xindex&quot;</span>)] <span class="ot">&lt;-</span> x_index</span>
<span id="cb134-60"><a href="statistical-analysis-1.html#cb134-60" tabindex="-1"></a>    <span class="fu">colnames</span>(datsignif)[<span class="fu">which</span>(<span class="fu">colnames</span>(datsignif) <span class="sc">==</span> <span class="st">&quot;Yindex&quot;</span>)] <span class="ot">&lt;-</span> y_index</span>
<span id="cb134-61"><a href="statistical-analysis-1.html#cb134-61" tabindex="-1"></a>        </span>
<span id="cb134-62"><a href="statistical-analysis-1.html#cb134-62" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">figure =</span> pl,</span>
<span id="cb134-63"><a href="statistical-analysis-1.html#cb134-63" tabindex="-1"></a>                <span class="at">data =</span> datsignif)</span>
<span id="cb134-64"><a href="statistical-analysis-1.html#cb134-64" tabindex="-1"></a>    </span>
<span id="cb134-65"><a href="statistical-analysis-1.html#cb134-65" tabindex="-1"></a>  }</span>
<span id="cb134-66"><a href="statistical-analysis-1.html#cb134-66" tabindex="-1"></a>  </span>
<span id="cb134-67"><a href="statistical-analysis-1.html#cb134-67" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb134-68"><a href="statistical-analysis-1.html#cb134-68" tabindex="-1"></a>}</span>
<span id="cb134-69"><a href="statistical-analysis-1.html#cb134-69" tabindex="-1"></a></span>
<span id="cb134-70"><a href="statistical-analysis-1.html#cb134-70" tabindex="-1"></a>lgfc_FDR_vol <span class="ot">&lt;-</span> <span class="fu">get_volcano</span>(</span>
<span id="cb134-71"><a href="statistical-analysis-1.html#cb134-71" tabindex="-1"></a>  <span class="at">inputdata =</span> m_results,</span>
<span id="cb134-72"><a href="statistical-analysis-1.html#cb134-72" tabindex="-1"></a>  <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Severe&quot;</span>),</span>
<span id="cb134-73"><a href="statistical-analysis-1.html#cb134-73" tabindex="-1"></a>  <span class="at">group_labels =</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Severe&quot;</span>),</span>
<span id="cb134-74"><a href="statistical-analysis-1.html#cb134-74" tabindex="-1"></a>  <span class="at">group_colors =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb134-75"><a href="statistical-analysis-1.html#cb134-75" tabindex="-1"></a>  <span class="at">x_index =</span> <span class="st">&quot;Log2FoldChange&quot;</span>,</span>
<span id="cb134-76"><a href="statistical-analysis-1.html#cb134-76" tabindex="-1"></a>  <span class="at">x_cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb134-77"><a href="statistical-analysis-1.html#cb134-77" tabindex="-1"></a>  <span class="at">y_index =</span> <span class="st">&quot;AdjustedPvalue&quot;</span>,</span>
<span id="cb134-78"><a href="statistical-analysis-1.html#cb134-78" tabindex="-1"></a>  <span class="at">y_cutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb134-79"><a href="statistical-analysis-1.html#cb134-79" tabindex="-1"></a>  <span class="at">plot =</span> <span class="cn">FALSE</span>)</span>
<span id="cb134-80"><a href="statistical-analysis-1.html#cb134-80" tabindex="-1"></a></span>
<span id="cb134-81"><a href="statistical-analysis-1.html#cb134-81" tabindex="-1"></a>lgfc_FDR_vol<span class="sc">$</span>figure</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-144-1.png" width="672" /></p>
</div>
<div id="adjusting-for-confounding-variables" class="section level3 hasAnchor" number="3.4.12">
<h3><span class="header-section-number">3.4.12</span> Adjusting for confounding variables<a href="statistical-analysis-1.html#adjusting-for-confounding-variables" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Controlling confounding effects by the following statistical approaches or R packages</p>
<ul>
<li><p><a href="https://huttenhower.sph.harvard.edu/maaslin/"><strong>MaAsLin2</strong></a>: Microbiome Multivariable Association with Linear Models.</p></li>
<li><p><a href="https://www.bioconductor.org/packages/release/bioc/vignettes/ALDEx2/inst/doc/ALDEx2_vignette.html"><strong>ALDEx2</strong></a>: differential abundance analysis across two or more conditions.</p></li>
<li><p><a href="https://stats.oarc.ucla.edu/other/mult-pkg/introduction-to-linear-mixed-models/">Linear Mixed Model</a></p></li>
</ul>
</div>
<div id="correlation-heatmaps" class="section level3 hasAnchor" number="3.4.13">
<h3><span class="header-section-number">3.4.13</span> Correlation Heatmaps<a href="statistical-analysis-1.html#correlation-heatmaps" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="statistical-analysis-1.html#cb135-1" tabindex="-1"></a>poma_cor <span class="ot">&lt;-</span> <span class="fu">PomaCorr</span>(se_processed_subset, <span class="at">label_size =</span> <span class="dv">8</span>, <span class="at">coeff =</span> <span class="fl">0.6</span>)</span>
<span id="cb135-2"><a href="statistical-analysis-1.html#cb135-2" tabindex="-1"></a>poma_cor<span class="sc">$</span>correlations</span></code></pre></div>
<pre><code>## # A tibble: 13,861 × 5
##    feature1 feature2     R   pvalue      FDR
##    &lt;chr&gt;    &lt;chr&gt;    &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;
##  1 M_52465  M_52466  0.955 3.19e-14 4.42e-10
##  2 M_44621  M_39270  0.946 3.36e-13 2.33e- 9
##  3 M_52610  M_52611  0.941 9.33e-13 4.31e- 9
##  4 M_33971  M_33972  0.939 1.39e-12 4.83e- 9
##  5 M_52464  M_52447  0.936 2.33e-12 6.10e- 9
##  6 M_52687  M_42449  0.935 2.64e-12 6.10e- 9
##  7 M_22036  M_63120  0.926 1.22e-11 2.41e- 8
##  8 M_38768  M_33972  0.925 1.42e-11 2.46e- 8
##  9 M_42449  M_52446  0.921 2.49e-11 3.83e- 8
## 10 M_44621  M_39271  0.920 2.97e-11 4.12e- 8
## # ℹ 13,851 more rows</code></pre>
<ul>
<li>correlation plot</li>
</ul>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="statistical-analysis-1.html#cb137-1" tabindex="-1"></a>poma_cor<span class="sc">$</span>corrplot</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-146-1.png" width="960" /></p>
<ul>
<li>Network</li>
</ul>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="statistical-analysis-1.html#cb138-1" tabindex="-1"></a>poma_cor<span class="sc">$</span>graph</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-147-1.png" width="960" /></p>
</div>
<div id="glasso-this-function-will-compute-a-gaussian-graphical-model-using-the-glmnet-package" class="section level3 hasAnchor" number="3.4.14">
<h3><span class="header-section-number">3.4.14</span> glasso: this function will compute a Gaussian Graphical Model using the glmnet package<a href="statistical-analysis-1.html#glasso-this-function-will-compute-a-gaussian-graphical-model-using-the-glmnet-package" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="statistical-analysis-1.html#cb139-1" tabindex="-1"></a><span class="fu">PomaCorr</span>(se_processed_subset, <span class="at">corr_type =</span> <span class="st">&quot;glasso&quot;</span>, <span class="at">coeff =</span> <span class="fl">0.6</span>)<span class="sc">$</span>graph</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-148-1.png" width="960" /></p>
<hr />
</div>
</div>
<div id="multivariate-analysis" class="section level2 hasAnchor" number="3.5">
<h2><span class="header-section-number">3.5</span> Multivariate analysis<a href="statistical-analysis-1.html#multivariate-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Comparing to <strong>univariate analysis</strong>, <strong>multivariate analysis</strong> is defined as a process of involving multiple dependent variables resulting in one outcome for feature selection. Here, we use Regularized Generalized Linear Models and Random forest model to identify the biomarkers associated with Outcomes.</p>
<ul>
<li><p>Lasso, Ridge and Elasticnet Regularized Generalized Linear Models for Binary Outcomes</p></li>
<li><p>Random forest model to select the important features</p></li>
</ul>
<p><strong>Note: Please also remember to <a href="statistical-analysis-1.html#data-preprocessing">preprocess</a> your data before running this sub-chapter.</strong></p>
<div id="loading-packages-4" class="section level3 hasAnchor" number="3.5.1">
<h3><span class="header-section-number">3.5.1</span> Loading packages<a href="statistical-analysis-1.html#loading-packages-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="statistical-analysis-1.html#cb140-1" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">warning =</span> F)</span>
<span id="cb140-2"><a href="statistical-analysis-1.html#cb140-2" tabindex="-1"></a></span>
<span id="cb140-3"><a href="statistical-analysis-1.html#cb140-3" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb140-4"><a href="statistical-analysis-1.html#cb140-4" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb140-5"><a href="statistical-analysis-1.html#cb140-5" tabindex="-1"></a><span class="fu">library</span>(POMA)</span>
<span id="cb140-6"><a href="statistical-analysis-1.html#cb140-6" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb140-7"><a href="statistical-analysis-1.html#cb140-7" tabindex="-1"></a><span class="fu">library</span>(ggraph)</span>
<span id="cb140-8"><a href="statistical-analysis-1.html#cb140-8" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb140-9"><a href="statistical-analysis-1.html#cb140-9" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb140-10"><a href="statistical-analysis-1.html#cb140-10" tabindex="-1"></a><span class="fu">library</span>(glmnet)</span>
<span id="cb140-11"><a href="statistical-analysis-1.html#cb140-11" tabindex="-1"></a></span>
<span id="cb140-12"><a href="statistical-analysis-1.html#cb140-12" tabindex="-1"></a><span class="co"># rm(list = ls())</span></span>
<span id="cb140-13"><a href="statistical-analysis-1.html#cb140-13" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb140-14"><a href="statistical-analysis-1.html#cb140-14" tabindex="-1"></a><span class="fu">options</span>(<span class="at">future.globals.maxSize =</span> <span class="dv">1000</span> <span class="sc">*</span> <span class="dv">1024</span><span class="sc">^</span><span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="importing-data-4" class="section level3 hasAnchor" number="3.5.2">
<h3><span class="header-section-number">3.5.2</span> Importing data<a href="statistical-analysis-1.html#importing-data-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The input data sets are from the previous chapter.</p>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="statistical-analysis-1.html#cb141-1" tabindex="-1"></a>se_processed <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;./dataset/POMA/se_processed.RDS&quot;</span>)</span></code></pre></div>
</div>
<div id="data-curation" class="section level3 hasAnchor" number="3.5.3">
<h3><span class="header-section-number">3.5.3</span> Data curation<a href="statistical-analysis-1.html#data-curation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="statistical-analysis-1.html#cb142-1" tabindex="-1"></a>group_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Severe&quot;</span>)</span>
<span id="cb142-2"><a href="statistical-analysis-1.html#cb142-2" tabindex="-1"></a></span>
<span id="cb142-3"><a href="statistical-analysis-1.html#cb142-3" tabindex="-1"></a>se_processed_subset <span class="ot">&lt;-</span> se_processed[, se_processed<span class="sc">$</span>group <span class="sc">%in%</span> group_names]</span>
<span id="cb142-4"><a href="statistical-analysis-1.html#cb142-4" tabindex="-1"></a>se_processed_subset<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">as.character</span>(se_processed_subset<span class="sc">$</span>group))</span></code></pre></div>
</div>
<div id="regularized-generalized-linear-models" class="section level3 hasAnchor" number="3.5.4">
<h3><span class="header-section-number">3.5.4</span> Regularized Generalized Linear Models<a href="statistical-analysis-1.html#regularized-generalized-linear-models" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="lasso-alpha-1" class="section level4 hasAnchor" number="3.5.4.1">
<h4><span class="header-section-number">3.5.4.1</span> Lasso: alpha = 1<a href="statistical-analysis-1.html#lasso-alpha-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb143-1"><a href="statistical-analysis-1.html#cb143-1" tabindex="-1"></a>lasso_res <span class="ot">&lt;-</span> <span class="fu">PomaLasso</span>(se_processed_subset, <span class="at">alpha =</span> <span class="dv">1</span>, <span class="at">labels =</span> <span class="cn">TRUE</span>)</span>
<span id="cb143-2"><a href="statistical-analysis-1.html#cb143-2" tabindex="-1"></a></span>
<span id="cb143-3"><a href="statistical-analysis-1.html#cb143-3" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(lasso_res<span class="sc">$</span>cvLassoPlot,</span>
<span id="cb143-4"><a href="statistical-analysis-1.html#cb143-4" tabindex="-1"></a>                   lasso_res<span class="sc">$</span>coefficientPlot,</span>
<span id="cb143-5"><a href="statistical-analysis-1.html#cb143-5" tabindex="-1"></a>                   <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">align =</span> <span class="st">&quot;hv&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-152-1.png" width="1152" /></p>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="statistical-analysis-1.html#cb144-1" tabindex="-1"></a>lasso_res<span class="sc">$</span>coefficients</span></code></pre></div>
<pre><code>## # A tibble: 11 × 2
##    feature     coefficient
##    &lt;chr&gt;             &lt;dbl&gt;
##  1 (Intercept)      0.102 
##  2 M_52682         -1.55  
##  3 M_62566          0.336 
##  4 M_49617         -1.26  
##  5 M_52710         -0.0742
##  6 M_42449          1.92  
##  7 M_45970          0.324 
##  8 M_42374         -0.117 
##  9 M_43761          0.319 
## 10 M_41220          0.249 
## 11 M_1566           1.44</code></pre>
</div>
<div id="ridge-alpha-0" class="section level4 hasAnchor" number="3.5.4.2">
<h4><span class="header-section-number">3.5.4.2</span> Ridge: alpha = 0<a href="statistical-analysis-1.html#ridge-alpha-0" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="statistical-analysis-1.html#cb146-1" tabindex="-1"></a>ridge_res <span class="ot">&lt;-</span> <span class="fu">PomaLasso</span>(se_processed_subset, <span class="at">alpha =</span> <span class="dv">0</span>, <span class="at">labels =</span> <span class="cn">TRUE</span>)</span>
<span id="cb146-2"><a href="statistical-analysis-1.html#cb146-2" tabindex="-1"></a></span>
<span id="cb146-3"><a href="statistical-analysis-1.html#cb146-3" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(ridge_res<span class="sc">$</span>cvLassoPlot,</span>
<span id="cb146-4"><a href="statistical-analysis-1.html#cb146-4" tabindex="-1"></a>                   ridge_res<span class="sc">$</span>coefficientPlot,</span>
<span id="cb146-5"><a href="statistical-analysis-1.html#cb146-5" tabindex="-1"></a>                   <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">align =</span> <span class="st">&quot;hv&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-153-1.png" width="1152" /></p>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="statistical-analysis-1.html#cb147-1" tabindex="-1"></a>ridge_res<span class="sc">$</span>coefficients</span></code></pre></div>
<pre><code>## # A tibble: 168 × 2
##    feature     coefficient
##    &lt;chr&gt;             &lt;dbl&gt;
##  1 (Intercept)   -0.118   
##  2 M_38768        0.0163  
##  3 M_38296        0.0426  
##  4 M_63436        0.0351  
##  5 M_57814        0.00998 
##  6 M_52603       -0.0159  
##  7 M_53174       -0.000721
##  8 M_19130       -0.0635  
##  9 M_34404       -0.0326  
## 10 M_32391       -0.0262  
## # ℹ 158 more rows</code></pre>
</div>
<div id="elasticnet-0-alpha-1" class="section level4 hasAnchor" number="3.5.4.3">
<h4><span class="header-section-number">3.5.4.3</span> Elasticnet: 0 &lt; alpha &lt; 1<a href="statistical-analysis-1.html#elasticnet-0-alpha-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="statistical-analysis-1.html#cb149-1" tabindex="-1"></a>elastic_res <span class="ot">&lt;-</span> <span class="fu">PomaLasso</span>(se_processed_subset, <span class="at">alpha =</span> <span class="fl">0.4</span>, <span class="at">labels =</span> <span class="cn">TRUE</span>)</span>
<span id="cb149-2"><a href="statistical-analysis-1.html#cb149-2" tabindex="-1"></a></span>
<span id="cb149-3"><a href="statistical-analysis-1.html#cb149-3" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(elastic_res<span class="sc">$</span>cvLassoPlot,</span>
<span id="cb149-4"><a href="statistical-analysis-1.html#cb149-4" tabindex="-1"></a>                   elastic_res<span class="sc">$</span>coefficientPlot,</span>
<span id="cb149-5"><a href="statistical-analysis-1.html#cb149-5" tabindex="-1"></a>                   <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">align =</span> <span class="st">&quot;hv&quot;</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-154-1.png" width="1152" /></p>
<div class="sourceCode" id="cb150"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb150-1"><a href="statistical-analysis-1.html#cb150-1" tabindex="-1"></a>elastic_res<span class="sc">$</span>coefficients</span></code></pre></div>
<pre><code>## # A tibble: 7 × 2
##   feature     coefficient
##   &lt;chr&gt;             &lt;dbl&gt;
## 1 (Intercept)     -0.141 
## 2 M_62566          0.224 
## 3 M_42449          0.237 
## 4 M_19263          0.0537
## 5 M_37482         -0.0116
## 6 M_43761          0.0507
## 7 M_1566           0.125</code></pre>
</div>
</div>
<div id="classification" class="section level3 hasAnchor" number="3.5.5">
<h3><span class="header-section-number">3.5.5</span> Classification<a href="statistical-analysis-1.html#classification" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="random-forest" class="section level4 hasAnchor" number="3.5.5.1">
<h4><span class="header-section-number">3.5.5.1</span> Random Forest<a href="statistical-analysis-1.html#random-forest" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li>Calculation</li>
</ul>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="statistical-analysis-1.html#cb152-1" tabindex="-1"></a>poma_rf <span class="ot">&lt;-</span> <span class="fu">PomaRandForest</span>(se_processed_subset, <span class="at">ntest =</span> <span class="dv">10</span>, <span class="at">nvar =</span> <span class="dv">10</span>)</span>
<span id="cb152-2"><a href="statistical-analysis-1.html#cb152-2" tabindex="-1"></a>poma_rf<span class="sc">$</span>error_tree</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-155-1.png" width="576" /></p>
<ul>
<li>table</li>
</ul>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="statistical-analysis-1.html#cb153-1" tabindex="-1"></a>poma_rf<span class="sc">$</span>confusionMatrix<span class="sc">$</span>table</span></code></pre></div>
<pre><code>##           Reference
## Prediction 1 2
##          1 0 0
##          2 1 1</code></pre>
<ul>
<li>Important features</li>
</ul>
<div class="sourceCode" id="cb155"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb155-1"><a href="statistical-analysis-1.html#cb155-1" tabindex="-1"></a>poma_rf<span class="sc">$</span>MeanDecreaseGini_plot</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-157-1.png" width="576" /></p>
</div>
<div id="support-vector-machine-svm" class="section level4 hasAnchor" number="3.5.5.2">
<h4><span class="header-section-number">3.5.5.2</span> Support Vector Machine (SVM)<a href="statistical-analysis-1.html#support-vector-machine-svm" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Still in development…</p>
<hr />
</div>
</div>
</div>
<div id="network-analysis" class="section level2 hasAnchor" number="3.6">
<h2><span class="header-section-number">3.6</span> Network Analysis<a href="statistical-analysis-1.html#network-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="introduction-1" class="section level3 hasAnchor" number="3.6.1">
<h3><span class="header-section-number">3.6.1</span> Introduction<a href="statistical-analysis-1.html#introduction-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<blockquote>
<p>Estimating microbial association networks from high-throughput sequencing data is a common exploratory data analysis approach aiming at understanding the complex interplay of microbial communities in their natural habitat. Statistical network estimation workflows comprise several analysis steps, including methods for zero handling, data normalization and computing microbial associations. Since microbial interactions are likely to change between conditions, e.g. between healthy individuals and patients, identifying network differences between groups is often an integral secondary analysis step.</p>
</blockquote>
<p>NetCoMi (Network Construction and Comparison for Microbiome Data) <span class="citation">(<a href="#ref-peschel2021netcomi">Peschel et al. 2021</a>)</span> provides functionality for constructing, analyzing, and comparing networks suitable for the application on microbial compositional data.</p>
<p>The following information is from <a href="https://github.com/stefpeschel/NetCoMi">NetCoMi github</a>.</p>
<p><strong>Association measures:</strong></p>
<ul>
<li><p>Pearson coefficient
(<a href="https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/cor"><code>cor()</code></a>
from <code>stats</code> package)</p></li>
<li><p>Spearman coefficient
(<a href="https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/cor"><code>cor()</code></a>
from <code>stats</code> package)</p></li>
<li><p>Biweight Midcorrelation
<a href="https://rdrr.io/cran/WGCNA/man/bicor.html"><code>bicor()</code></a> from <code>WGCNA</code>
package</p></li>
</ul>
<p><strong>Methods for zero replacement:</strong></p>
<ul>
<li><p>Adding a predefined pseudo count</p></li>
<li><p>Multiplicative replacement
(<a href="https://rdrr.io/cran/zCompositions/man/multRepl.html"><code>multRepl</code></a>
from <code>zCompositions</code> package)</p></li>
<li><p>Modified EM alr-algorithm
(<a href="https://rdrr.io/cran/zCompositions/man/lrEM.html"><code>lrEM</code></a> from
<code>zCompositions</code> package)</p></li>
<li><p>Bayesian-multiplicative replacement
(<a href="https://rdrr.io/cran/zCompositions/man/cmultRepl.html"><code>cmultRepl</code></a>
from <code>zCompositions</code> package)</p></li>
</ul>
<p><strong>Normalization methods:</strong></p>
<ul>
<li><p>Total Sum Scaling (TSS) (own implementation)</p></li>
<li><p>Cumulative Sum Scaling (CSS) (<a href=""><code>cumNormMat</code></a> from <code>metagenomeSeq</code>
package)</p></li>
<li><p>Common Sum Scaling (COM) (own implementation)</p></li>
<li><p>Rarefying (<a href=""><code>rrarefy</code></a> from <code>vegan</code> package)</p></li>
<li><p>Variance Stabilizing Transformation (VST)<br />
(<a href=""><code>varianceStabilizingTransformation</code></a> from <code>DESeq2</code> package)</p></li>
<li><p>Centered log-ratio (clr) transformation
(<a href="https://rdrr.io/github/zdk123/SpiecEasi/man/clr.html"><code>clr()</code></a>
from <code>SpiecEasi</code> package))</p></li>
</ul>
<p>TSS, CSS, COM, VST, and the clr transformation are described in [Badri et al., 2020].</p>
</div>
<div id="loading-packages-5" class="section level3 hasAnchor" number="3.6.2">
<h3><span class="header-section-number">3.6.2</span> Loading packages<a href="statistical-analysis-1.html#loading-packages-5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="statistical-analysis-1.html#cb156-1" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">warning =</span> F)</span>
<span id="cb156-2"><a href="statistical-analysis-1.html#cb156-2" tabindex="-1"></a></span>
<span id="cb156-3"><a href="statistical-analysis-1.html#cb156-3" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb156-4"><a href="statistical-analysis-1.html#cb156-4" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb156-5"><a href="statistical-analysis-1.html#cb156-5" tabindex="-1"></a><span class="fu">library</span>(POMA)</span>
<span id="cb156-6"><a href="statistical-analysis-1.html#cb156-6" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb156-7"><a href="statistical-analysis-1.html#cb156-7" tabindex="-1"></a><span class="fu">library</span>(ggraph)</span>
<span id="cb156-8"><a href="statistical-analysis-1.html#cb156-8" tabindex="-1"></a><span class="fu">library</span>(plotly)</span>
<span id="cb156-9"><a href="statistical-analysis-1.html#cb156-9" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb156-10"><a href="statistical-analysis-1.html#cb156-10" tabindex="-1"></a><span class="fu">library</span>(NetCoMi)</span>
<span id="cb156-11"><a href="statistical-analysis-1.html#cb156-11" tabindex="-1"></a><span class="fu">library</span>(SPRING)</span>
<span id="cb156-12"><a href="statistical-analysis-1.html#cb156-12" tabindex="-1"></a><span class="fu">library</span>(SpiecEasi)</span>
<span id="cb156-13"><a href="statistical-analysis-1.html#cb156-13" tabindex="-1"></a></span>
<span id="cb156-14"><a href="statistical-analysis-1.html#cb156-14" tabindex="-1"></a></span>
<span id="cb156-15"><a href="statistical-analysis-1.html#cb156-15" tabindex="-1"></a><span class="co"># rm(list = ls())</span></span>
<span id="cb156-16"><a href="statistical-analysis-1.html#cb156-16" tabindex="-1"></a><span class="fu">options</span>(<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb156-17"><a href="statistical-analysis-1.html#cb156-17" tabindex="-1"></a><span class="fu">options</span>(<span class="at">future.globals.maxSize =</span> <span class="dv">1000</span> <span class="sc">*</span> <span class="dv">1024</span><span class="sc">^</span><span class="dv">2</span>)</span></code></pre></div>
</div>
<div id="importing-data-5" class="section level3 hasAnchor" number="3.6.3">
<h3><span class="header-section-number">3.6.3</span> Importing data<a href="statistical-analysis-1.html#importing-data-5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The input data sets are from the previous chapter.</p>
<div class="sourceCode" id="cb157"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb157-1"><a href="statistical-analysis-1.html#cb157-1" tabindex="-1"></a>se_filter <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;./dataset/POMA/se_filter.RDS&quot;</span>)</span></code></pre></div>
</div>
<div id="data-curation-1" class="section level3 hasAnchor" number="3.6.4">
<h3><span class="header-section-number">3.6.4</span> Data curation<a href="statistical-analysis-1.html#data-curation-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="statistical-analysis-1.html#cb158-1" tabindex="-1"></a>features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(se_filter) <span class="sc">%&gt;%</span> </span>
<span id="cb158-2"><a href="statistical-analysis-1.html#cb158-2" tabindex="-1"></a>  <span class="fu">t</span>()</span>
<span id="cb158-3"><a href="statistical-analysis-1.html#cb158-3" tabindex="-1"></a>features_tab[<span class="fu">is.na</span>(features_tab)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb158-4"><a href="statistical-analysis-1.html#cb158-4" tabindex="-1"></a></span>
<span id="cb158-5"><a href="statistical-analysis-1.html#cb158-5" tabindex="-1"></a><span class="fu">head</span>(features_tab)</span></code></pre></div>
<pre><code>##          M_38768 M_38296  M_63436  M_57814   M_52603  M_53174  M_19130   M_34404  M_32391   M_20675   M_34400  M_44621  M_52689  M_52673  M_52682
## P101001 51127588 5105020 756686.2 281502.0  94392176 144048.2 25632184 123026.38 321794.2 151752512 2205732.2 402949.6 23821584 11552554 28373504
## P101004 34940596 3885477 851026.5 297304.4 115155104 217577.4 25106562  22810.19 210540.2  57703932  627949.3 288060.6 13722696  8489023 24738246
## P101007 58518636 4285130 726593.9 319016.7  79582632 211262.5 31371314 375686.47 477073.6 198430704 2856552.5 746095.9 26451790 15090030 27171710
## P101009 51118832 6665654 232959.5 242172.2 118408760 431295.3 27787270 118662.19 672746.8 105392656 1831869.2 347963.7 21134366 11239614 33480956
## P101010 83783688 9057441 650261.1 200135.8  92508664 135226.5 26685844 130040.21 263299.9 103049880 1225102.6 550802.9 32726406 19644278 36381128
##         M_52677  M_52478 M_52477  M_52713  M_52716 M_52474   M_39270  M_52475  M_52748 M_52614   M_39271  M_33228 M_35186 M_34214  M_34397  M_62559
## P101001 2494246  9753736 2463866 594237.8  7622086 6879792  959249.2 40933908 14099088 3790466 1572987.8 96375552 8141286 8284910 712147.4 162056.6
## P101004 2398333  8732374 1651473 528063.9  7439745 3780738  785895.0 23139750 12378493 1889665 1101383.0 49699956 4303854 7923489 194718.0 178442.0
## P101007 2891707  9955330 3774870 780014.5  8345656 5111358 1903592.4 51188676 17399748 4683262 2026999.0 70649888 7129884 9211485 505635.2 238782.9
## P101009 2989538 10679558 2826825 978802.8 10357009 4263671  863701.9 36418292 19482396 4177870 1169707.9 41290160 5394354 8559147 159987.3 152987.9
## P101010 4624334 10877791 4131897 702736.2  7783024 5172609 1177860.8 68994568 29016166 6862688 1722600.4 57308316 6270727 7623603 446015.2 263668.8
##          M_62566  M_62562  M_48341  M_35153  M_49617 M_45951  M_52710  M_53189 M_53176   M_34419  M_36600  M_54885 M_36594 M_27447  M_32350   M_62946
## P101001 265921.8 285496.1 292845.1 128661.8 381540.1 3736284 47739292 392911.0 3305130 306346432  9887348 298886.1 3238505 9004255 627613.7  395496.2
## P101004 211541.5 265036.3 331437.5 276938.1 588488.2 1839252 43678784 300383.9 3552664 263333424  9703074 356274.3 4642443 2944057 339586.8 1079166.8
## P101007 456498.2 463207.5 514893.1 185715.0 464548.8 1545863 53975068 539361.5 2035337 191261648  6431966 393623.0 3509099 3230152 662555.6  484653.9
## P101009 262397.8 322379.6 297725.7 320728.0 497342.0 3145905 47619564 487186.2 3802829 232171184 11914472 599783.6 5404004 2030452 492430.9  403058.4
## P101010 356271.0 471981.4 594546.3 113578.3 457409.1 5152494 61906028 236719.5 3049962 183104336  9366287 280123.8 3737352 2431808 373484.2 1065601.8
##            M_64596  M_30460   M_27665  M_34395 M_34389  M_53195  M_19258  M_35625 M_55041  M_52697 M_52687   M_48258 M_35628 M_36602  M_21184
## P101001   46295.48 270857.8 1335229.0 199466.6 2138968 13858347  9863770 574852.4 3655240 12883511 2398777 181136336 6548460 1749937  7613153
## P101004 1017875.06 204924.9 1279771.8 226695.4 2153556  6059094  5239331 236322.3 1637351 13617571 2017056 115752200 3455001 1409849 15023415
## P101007  463587.81 320279.5  676678.7 208125.5 1348283 18901246  8920123 296148.1 7284474 19369632 2768957 102036920 3640571 1495550 23832470
## P101009  370624.53 289090.7 1098703.2 583087.2 4572006  9496899  7639714 269586.8 3065707 14302725 3650640 118663320 5087503 3099956 12286962
## P101010 1228528.25 216454.2  520595.4 259017.0 3558800 19940270 12844156 585791.2 2493687 19721992 1901558 154183360 5244878 3420659 14516694
##           M_53180  M_33230  M_52431   M_52462  M_52464 M_52467   M_52454   M_52610 M_52465   M_42446 M_42449 M_52450   M_52461 M_19263 M_52669
## P101001  748319.2 19098412 824253.8 486910976  6794782 2280194 274026048 206279104 2204234 712253440 4006186 2641266 486655872 3874469 2759973
## P101004  917020.4 10641834 328446.0 380887616  4090405 1003667 282066016 253920496 3084690 807180352 4623029 2342063 439253152 2842332 1193472
## P101007  799016.0 14977371 469095.1 515795264 13183206 2950405 309709728 316142400 8800303 715500800 5532970 3922614 448578400 5864154 2724709
## P101009 1118435.0 13575508 518623.8 390977568  7596680 3766843 325723168 232534688 4486037 742576768 7445768 6170018 473622240 4717213 4360214
## P101010 1685150.8 19464998 795436.6 478479936  5395172 6117689 382690720 262177072 2910989 642967424 3822049 6121756 647036736 4579096 5908542
##          M_52470  M_52616   M_57388   M_33955 M_35631  M_45970 M_35305  M_21127  M_61868   M_42450  M_52447  M_52449   M_52235   M_52611 M_52466
## P101001 35859212 12365879  79676.23 448634784 4035864 409595.2  863723  4804332 504846.8 358896736 30252208 43173148 1118858.8  61304088 1859952
## P101004 23531182 10655923  37269.65 413690720 3446139 230301.7 1016275  6962388 361061.4 282910752 19424238 30030992 3035421.1  78508544 2158371
## P101007 44479796 14010174 103174.72 368849632 3764448 323317.6 1214598 10822974 530824.9 404918560 57376512 50766284 7742330.5 133758544 6886556
## P101009 41899896 14814230  65198.78 365712768 3444159 420762.3 1920743  9249804 507524.6 298822048 26429340 43420376  872709.2  80545136 2995264
## P101010 64131204 12873290  41349.66 376101984 3276579 293613.2 2983894 17823292 309155.8 373251712 21732290 43622724  336274.6 104073768 2487264
##           M_52452  M_52446  M_52468   M_52438 M_42448 M_52726 M_19265   M_33961 M_42398 M_19324  M_33971  M_33972 M_32497   M_63251   M_64418
## P101001 510461472 17331080  7814518 181406848 5754168 2246755 1156850 191147264 6137137 3277043 34683340 21778388 7502807  702985.6 138656.33
## P101004 573170816 17896104  8339710 122556968 3069182  863899 1564114 166379712 5143481 7207395 23345216 10599812 6293102  369620.4 143054.61
## P101007 588643712 27738444 11004061 181572112 6107326 1798405 6813304 189915296 7412535 7613864 68319096 22969486 7799882  514191.1 183824.64
## P101009 631965888 28258140 15869048 208484000 6196910 3176317 1385724 186599152 5421984 6803468 37778060 13476085 3700375  811957.7 162973.59
## P101010 590247424 18904148 11231769 306379392 6164263 4295183 2180185 217632736 5767624 7363716 62689384 24592120 5778318  392289.5 121719.19
##          M_37536   M_37752   M_38168  M_47403   M_37482  M_57547 M_62805  M_38276   M_64328  M_63042   M_48580   M_6146  M_42374   M_43761   M_43343
## P101001 213331.9  914067.1 5222754.0 209255.5 142595.88 169297.7 3250618 228644.9 110957136 12292479  91572.95 177446.7 42943284 1946942.6 242105.86
## P101004 234348.0 1492370.2  156811.9 220424.3 152280.70 426227.5 2357354 281935.7 110705416  7593832 760372.12 352992.7 42325744  839404.1 133500.70
## P101007 580629.7 2570282.5 5744456.0 369675.0  38006.62 561078.5 4624754 131296.8 137532672 31897926 356734.56 480009.0 44412548 2357434.8 165855.44
## P101009 119383.3  714076.2 1239502.1 172995.7  84628.77 457043.3 2861962 149611.1 209583808  6453900  93497.58 354210.8 24181052 1745628.5 135844.00
## P101010 253655.1 1706573.6  914122.9 168503.4 101636.99 602024.4 2523801 246180.4 166666528  5296456  57886.30 650030.8 42762020  361423.8  45789.42
##           M_43266   M_19266   M_36746   M_63739  M_61700  M_57663   M_52281 M_42489  M_37253   M_18281   M_52916 M_61698  M_22036  M_35675    M_1432
## P101001 1163203.9 176944.80  479072.6  761146.2 472302.3 719237.9  78466792 2865827 411578.0 179893.86  733452.4 2130324 774797.4 38006460 1354798.0
## P101004 5113189.5 189413.64 1372193.2  549097.4 265781.7 364556.1  81241496 1546570 425030.0 108887.77  683209.3 1221117 427803.9 49110836  927115.9
## P101007 3111430.8 385203.22  642121.2  792883.2 403507.7 500680.8  95847768 1403190 742298.0 769990.44  767281.1 2035627 455143.1 38735820 2731939.8
## P101009  890192.9 211997.21  778356.4  727360.0 495207.8 523549.1  35564448 1653830 466825.0 267979.69 1044029.8 2243073 422308.4 38389060  933468.1
## P101010 1139527.8 178331.05 2020790.5  913964.4 318258.2 455201.1 108681912 1007203 232876.0  40788.35  716821.7 1380906 199201.0 41784488  652360.4
##          M_17945  M_15667 M_48141   M_63120   M_32506    M_64049  M_45095   M_53229    M_62520  M_63990   M_21232   M_55072  M_47118  M_35253 M_43400
## P101001 17661872 410808.2 5025640 1647630.5  969193.1 1200940.00 751527.3  768160.4   63648.33 10428390  904260.1  488544.9 281162.7 27446600 3614323
## P101004 20479238 364068.7 2705928  715426.6  784027.9  259731.38 378033.7  953194.9  203543.30  6402490 2594437.2  653107.0 362380.0 28088842 1483176
## P101007 16393367 236559.3 2406770  918737.2 1068945.8 1829885.38 589506.3 1619436.8  115621.80 10938208 5687539.5  522721.5 482046.2 21613430 5746994
## P101009 13751929 354344.8 1891979  858821.5  400273.7  109314.27 503910.2  877530.2 1488895.25  7829000 1906269.8  873616.8 619943.9 28872470 5682455
## P101010 18558476 557050.1 1919326  441160.8  552550.0   62541.41 489013.9  993923.4  235431.48  9571510 2910665.2  864624.8 677774.7 25375946 6362144
##         M_41220   M_46115   M_37174  M_63109 M_62952 M_63589    M_35635  M_45415  M_32197  M_62853  M_1566  M_63681  M_61871  M_31787   M_63361
## P101001 3649905 1578011.4  70308.07 29603070 5104806 2036826   77650.82 148777.2  9709386 483999.6 2625320 434480.3 12926878 10650574 2540280.0
## P101004 1132894  440408.7 156658.04 22345812 3084011 1527389  492895.75  30709.0  7547276 299234.0 2387450 373229.5 25459396 19791136 2165454.0
## P101007 1522508  794592.2  31187.21 26736000 4728273 1855034 1017754.44  90723.0 12608340 440482.2 5896738 400386.6 15742145 52497536 2231037.2
## P101009 1882729  671497.6  21561.75 31187900 4140767 1444713  118820.93 168136.0 10253790 491346.8 2785016 301840.8 23710054 48267136  992205.4
## P101010 1265978 1133763.9 115934.38 17916018 2434359 3420016 1812900.50 163988.0 14821266 359500.1 1552196 548330.4 22917482 34226984 1125607.0
##  [ reached getOption(&quot;max.print&quot;) -- omitted 1 row ]</code></pre>
</div>
<div id="associations-among-features" class="section level3 hasAnchor" number="3.6.5">
<h3><span class="header-section-number">3.6.5</span> Associations Among Features<a href="statistical-analysis-1.html#associations-among-features" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="single-network-with-pearson-correlation-as-association-measure" class="section level4 hasAnchor" number="3.6.5.1">
<h4><span class="header-section-number">3.6.5.1</span> Single network with Pearson correlation as association measure<a href="statistical-analysis-1.html#single-network-with-pearson-correlation-as-association-measure" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Since Pearson correlations may lead to compositional effects when applied to sequencing data, we use the clr transformation as normalization method. Zero treatment is necessary in this case.</p>
<ul>
<li><p>InputData: numeric matrix. Can be a count matrix (rows are samples, columns are Features).</p></li>
<li><p>Method to compute the associations between features (argument measure).</p></li>
<li><p>Normalization method:</p>
<ul>
<li><p>normMethod: clr</p></li>
<li><p>zeroMethod: multRepl</p></li>
<li><p>sparsMethod: threshold</p></li>
</ul></li>
</ul>
<p>A threshold of 0.3 is used as sparsification method, so that only OTUs with an absolute correlation greater than or equal to 0.3 are connected.</p>
<div id="building-network-model" class="section level5 hasAnchor" number="3.6.5.1.1">
<h5><span class="header-section-number">3.6.5.1.1</span> Building network model<a href="statistical-analysis-1.html#building-network-model" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb160"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb160-1"><a href="statistical-analysis-1.html#cb160-1" tabindex="-1"></a>net_single <span class="ot">&lt;-</span> <span class="fu">netConstruct</span>(features_tab,  </span>
<span id="cb160-2"><a href="statistical-analysis-1.html#cb160-2" tabindex="-1"></a>                           <span class="at">measure =</span> <span class="st">&quot;pearson&quot;</span>,</span>
<span id="cb160-3"><a href="statistical-analysis-1.html#cb160-3" tabindex="-1"></a>                           <span class="at">normMethod =</span> <span class="st">&quot;clr&quot;</span>, </span>
<span id="cb160-4"><a href="statistical-analysis-1.html#cb160-4" tabindex="-1"></a>                           <span class="at">zeroMethod =</span> <span class="st">&quot;multRepl&quot;</span>,</span>
<span id="cb160-5"><a href="statistical-analysis-1.html#cb160-5" tabindex="-1"></a>                           <span class="at">sparsMethod =</span> <span class="st">&quot;threshold&quot;</span>, </span>
<span id="cb160-6"><a href="statistical-analysis-1.html#cb160-6" tabindex="-1"></a>                           <span class="at">thresh =</span> <span class="fl">0.3</span>,</span>
<span id="cb160-7"><a href="statistical-analysis-1.html#cb160-7" tabindex="-1"></a>                           <span class="at">verbose =</span> <span class="dv">3</span>,</span>
<span id="cb160-8"><a href="statistical-analysis-1.html#cb160-8" tabindex="-1"></a>                           <span class="at">seed =</span> <span class="dv">123</span>)</span></code></pre></div>
</div>
<div id="visualizing-the-network" class="section level5 hasAnchor" number="3.6.5.1.2">
<h5><span class="header-section-number">3.6.5.1.2</span> Visualizing the network<a href="statistical-analysis-1.html#visualizing-the-network" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li>primary plot</li>
</ul>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="statistical-analysis-1.html#cb161-1" tabindex="-1"></a>props_single <span class="ot">&lt;-</span> <span class="fu">netAnalyze</span>(net_single, <span class="at">clustMethod =</span> <span class="st">&quot;cluster_fast_greedy&quot;</span>)</span>
<span id="cb161-2"><a href="statistical-analysis-1.html#cb161-2" tabindex="-1"></a></span>
<span id="cb161-3"><a href="statistical-analysis-1.html#cb161-3" tabindex="-1"></a><span class="fu">plot</span>(props_single, </span>
<span id="cb161-4"><a href="statistical-analysis-1.html#cb161-4" tabindex="-1"></a>     <span class="at">nodeColor =</span> <span class="st">&quot;cluster&quot;</span>, </span>
<span id="cb161-5"><a href="statistical-analysis-1.html#cb161-5" tabindex="-1"></a>     <span class="at">nodeSize =</span> <span class="st">&quot;eigenvector&quot;</span>,</span>
<span id="cb161-6"><a href="statistical-analysis-1.html#cb161-6" tabindex="-1"></a>     <span class="at">title1 =</span> <span class="st">&quot;Network on metabolomics with Pearson correlations&quot;</span>, </span>
<span id="cb161-7"><a href="statistical-analysis-1.html#cb161-7" tabindex="-1"></a>     <span class="at">showTitle =</span> <span class="cn">TRUE</span>,</span>
<span id="cb161-8"><a href="statistical-analysis-1.html#cb161-8" tabindex="-1"></a>     <span class="at">cexTitle =</span> <span class="fl">1.5</span>)</span>
<span id="cb161-9"><a href="statistical-analysis-1.html#cb161-9" tabindex="-1"></a></span>
<span id="cb161-10"><a href="statistical-analysis-1.html#cb161-10" tabindex="-1"></a><span class="fu">legend</span>(<span class="fl">0.7</span>, <span class="fl">1.1</span>, <span class="at">cex =</span> <span class="dv">1</span>, <span class="at">title =</span> <span class="st">&quot;estimated correlation:&quot;</span>, </span>
<span id="cb161-11"><a href="statistical-analysis-1.html#cb161-11" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;+&quot;</span>,<span class="st">&quot;-&quot;</span>), <span class="at">lty =</span> <span class="dv">1</span>, <span class="at">lwd =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;#009900&quot;</span>,<span class="st">&quot;red&quot;</span>), </span>
<span id="cb161-12"><a href="statistical-analysis-1.html#cb161-12" tabindex="-1"></a>       <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">horiz =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-162-1.png" width="768" /></p>
<ul>
<li><p>improve the visualization by changing the following arguments:</p>
<ul>
<li><p><code>repulsion = 0.8</code>: Place the nodes further apart</p></li>
<li><p><code>rmSingles = TRUE</code>: Single nodes are removed</p></li>
<li><p><code>labelScale = FALSE</code> and <code>cexLabels = 1.6</code>: All labels have equal size and are enlarged to improve readability of small node’s labels</p></li>
<li><p><code>nodeSizeSpread = 3</code> (default is 4): Node sizes are more similar if the value is decreased. This argument (in combination with <code>cexNodes</code>) is useful to enlarge small nodes while keeping the size of big nodes.</p></li>
</ul></li>
</ul>
<div class="sourceCode" id="cb162"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb162-1"><a href="statistical-analysis-1.html#cb162-1" tabindex="-1"></a><span class="fu">plot</span>(props_single, </span>
<span id="cb162-2"><a href="statistical-analysis-1.html#cb162-2" tabindex="-1"></a>     <span class="at">nodeColor =</span> <span class="st">&quot;cluster&quot;</span>, </span>
<span id="cb162-3"><a href="statistical-analysis-1.html#cb162-3" tabindex="-1"></a>     <span class="at">nodeSize =</span> <span class="st">&quot;eigenvector&quot;</span>,</span>
<span id="cb162-4"><a href="statistical-analysis-1.html#cb162-4" tabindex="-1"></a>     <span class="at">repulsion =</span> <span class="fl">0.8</span>,</span>
<span id="cb162-5"><a href="statistical-analysis-1.html#cb162-5" tabindex="-1"></a>     <span class="at">rmSingles =</span> <span class="cn">TRUE</span>,</span>
<span id="cb162-6"><a href="statistical-analysis-1.html#cb162-6" tabindex="-1"></a>     <span class="at">labelScale =</span> <span class="cn">FALSE</span>,</span>
<span id="cb162-7"><a href="statistical-analysis-1.html#cb162-7" tabindex="-1"></a>     <span class="at">cexLabels =</span> <span class="fl">1.6</span>,</span>
<span id="cb162-8"><a href="statistical-analysis-1.html#cb162-8" tabindex="-1"></a>     <span class="at">nodeSizeSpread =</span> <span class="dv">3</span>,</span>
<span id="cb162-9"><a href="statistical-analysis-1.html#cb162-9" tabindex="-1"></a>     <span class="at">cexNodes =</span> <span class="dv">2</span>,</span>
<span id="cb162-10"><a href="statistical-analysis-1.html#cb162-10" tabindex="-1"></a>     <span class="at">title1 =</span> <span class="st">&quot;Network on metabolomics with Pearson correlations&quot;</span>, </span>
<span id="cb162-11"><a href="statistical-analysis-1.html#cb162-11" tabindex="-1"></a>     <span class="at">showTitle =</span> <span class="cn">TRUE</span>,</span>
<span id="cb162-12"><a href="statistical-analysis-1.html#cb162-12" tabindex="-1"></a>     <span class="at">cexTitle =</span> <span class="fl">1.5</span>)</span>
<span id="cb162-13"><a href="statistical-analysis-1.html#cb162-13" tabindex="-1"></a></span>
<span id="cb162-14"><a href="statistical-analysis-1.html#cb162-14" tabindex="-1"></a><span class="fu">legend</span>(<span class="fl">0.7</span>, <span class="fl">1.1</span>, <span class="at">cex =</span> <span class="fl">1.2</span>, <span class="at">title =</span> <span class="st">&quot;estimated correlation:&quot;</span>,</span>
<span id="cb162-15"><a href="statistical-analysis-1.html#cb162-15" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;+&quot;</span>,<span class="st">&quot;-&quot;</span>), <span class="at">lty =</span> <span class="dv">1</span>, <span class="at">lwd =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;#009900&quot;</span>,<span class="st">&quot;red&quot;</span>),</span>
<span id="cb162-16"><a href="statistical-analysis-1.html#cb162-16" tabindex="-1"></a>       <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">horiz =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-163-1.png" width="1152" /></p>
</div>
</div>
<div id="single-network-with-spearman-correlation-as-association-measure" class="section level4 hasAnchor" number="3.6.5.2">
<h4><span class="header-section-number">3.6.5.2</span> Single network with spearman correlation as association measure<a href="statistical-analysis-1.html#single-network-with-spearman-correlation-as-association-measure" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div id="building-network-model-1" class="section level5 hasAnchor" number="3.6.5.2.1">
<h5><span class="header-section-number">3.6.5.2.1</span> Building network model<a href="statistical-analysis-1.html#building-network-model-1" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb163"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb163-1"><a href="statistical-analysis-1.html#cb163-1" tabindex="-1"></a>net_single2 <span class="ot">&lt;-</span> <span class="fu">netConstruct</span>(features_tab,  </span>
<span id="cb163-2"><a href="statistical-analysis-1.html#cb163-2" tabindex="-1"></a>                            <span class="at">measure =</span> <span class="st">&quot;spearman&quot;</span>,</span>
<span id="cb163-3"><a href="statistical-analysis-1.html#cb163-3" tabindex="-1"></a>                            <span class="at">normMethod =</span> <span class="st">&quot;clr&quot;</span>, </span>
<span id="cb163-4"><a href="statistical-analysis-1.html#cb163-4" tabindex="-1"></a>                            <span class="at">zeroMethod =</span> <span class="st">&quot;multRepl&quot;</span>,</span>
<span id="cb163-5"><a href="statistical-analysis-1.html#cb163-5" tabindex="-1"></a>                            <span class="at">sparsMethod =</span> <span class="st">&quot;threshold&quot;</span>, </span>
<span id="cb163-6"><a href="statistical-analysis-1.html#cb163-6" tabindex="-1"></a>                            <span class="at">thresh =</span> <span class="fl">0.3</span>,</span>
<span id="cb163-7"><a href="statistical-analysis-1.html#cb163-7" tabindex="-1"></a>                            <span class="at">verbose =</span> <span class="dv">3</span>,</span>
<span id="cb163-8"><a href="statistical-analysis-1.html#cb163-8" tabindex="-1"></a>                            <span class="at">seed =</span> <span class="dv">123</span>)</span></code></pre></div>
</div>
<div id="visualizing-the-network-1" class="section level5 hasAnchor" number="3.6.5.2.2">
<h5><span class="header-section-number">3.6.5.2.2</span> Visualizing the network<a href="statistical-analysis-1.html#visualizing-the-network-1" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb164"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb164-1"><a href="statistical-analysis-1.html#cb164-1" tabindex="-1"></a>props_single2 <span class="ot">&lt;-</span> <span class="fu">netAnalyze</span>(net_single2, <span class="at">clustMethod =</span> <span class="st">&quot;cluster_fast_greedy&quot;</span>)</span>
<span id="cb164-2"><a href="statistical-analysis-1.html#cb164-2" tabindex="-1"></a></span>
<span id="cb164-3"><a href="statistical-analysis-1.html#cb164-3" tabindex="-1"></a><span class="fu">plot</span>(props_single2, </span>
<span id="cb164-4"><a href="statistical-analysis-1.html#cb164-4" tabindex="-1"></a>     <span class="at">nodeColor =</span> <span class="st">&quot;cluster&quot;</span>, </span>
<span id="cb164-5"><a href="statistical-analysis-1.html#cb164-5" tabindex="-1"></a>     <span class="at">nodeSize =</span> <span class="st">&quot;eigenvector&quot;</span>,</span>
<span id="cb164-6"><a href="statistical-analysis-1.html#cb164-6" tabindex="-1"></a>     <span class="at">repulsion =</span> <span class="fl">0.8</span>,</span>
<span id="cb164-7"><a href="statistical-analysis-1.html#cb164-7" tabindex="-1"></a>     <span class="at">rmSingles =</span> <span class="cn">TRUE</span>,</span>
<span id="cb164-8"><a href="statistical-analysis-1.html#cb164-8" tabindex="-1"></a>     <span class="at">labelScale =</span> <span class="cn">FALSE</span>,</span>
<span id="cb164-9"><a href="statistical-analysis-1.html#cb164-9" tabindex="-1"></a>     <span class="at">cexLabels =</span> <span class="fl">1.6</span>,</span>
<span id="cb164-10"><a href="statistical-analysis-1.html#cb164-10" tabindex="-1"></a>     <span class="at">nodeSizeSpread =</span> <span class="dv">3</span>,</span>
<span id="cb164-11"><a href="statistical-analysis-1.html#cb164-11" tabindex="-1"></a>     <span class="at">cexNodes =</span> <span class="dv">2</span>,</span>
<span id="cb164-12"><a href="statistical-analysis-1.html#cb164-12" tabindex="-1"></a>     <span class="at">title1 =</span> <span class="st">&quot;Network on metabolomics with Spearman correlations&quot;</span>, </span>
<span id="cb164-13"><a href="statistical-analysis-1.html#cb164-13" tabindex="-1"></a>     <span class="at">showTitle =</span> <span class="cn">TRUE</span>,</span>
<span id="cb164-14"><a href="statistical-analysis-1.html#cb164-14" tabindex="-1"></a>     <span class="at">cexTitle =</span> <span class="fl">1.5</span>)</span>
<span id="cb164-15"><a href="statistical-analysis-1.html#cb164-15" tabindex="-1"></a></span>
<span id="cb164-16"><a href="statistical-analysis-1.html#cb164-16" tabindex="-1"></a><span class="fu">legend</span>(<span class="fl">0.7</span>, <span class="fl">1.1</span>, <span class="at">cex =</span> <span class="fl">1.2</span>, <span class="at">title =</span> <span class="st">&quot;estimated correlation:&quot;</span>,</span>
<span id="cb164-17"><a href="statistical-analysis-1.html#cb164-17" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;+&quot;</span>,<span class="st">&quot;-&quot;</span>), <span class="at">lty =</span> <span class="dv">1</span>, <span class="at">lwd =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;#009900&quot;</span>,<span class="st">&quot;red&quot;</span>),</span>
<span id="cb164-18"><a href="statistical-analysis-1.html#cb164-18" tabindex="-1"></a>       <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">horiz =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-165-1.png" width="1152" /></p>
</div>
</div>
<div id="single-network-with-wgcna-bicor-as-association-measure" class="section level4 hasAnchor" number="3.6.5.3">
<h4><span class="header-section-number">3.6.5.3</span> Single network with WGCNA (bicor) as association measure<a href="statistical-analysis-1.html#single-network-with-wgcna-bicor-as-association-measure" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Biweight Midcorrelation <code>bicor()</code> from WGCNA package.</p>
<div id="building-network-model-2" class="section level5 hasAnchor" number="3.6.5.3.1">
<h5><span class="header-section-number">3.6.5.3.1</span> Building network model<a href="statistical-analysis-1.html#building-network-model-2" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb165"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb165-1"><a href="statistical-analysis-1.html#cb165-1" tabindex="-1"></a>net_single3 <span class="ot">&lt;-</span> <span class="fu">netConstruct</span>(features_tab, </span>
<span id="cb165-2"><a href="statistical-analysis-1.html#cb165-2" tabindex="-1"></a>                           <span class="at">measure =</span> <span class="st">&quot;bicor&quot;</span>,</span>
<span id="cb165-3"><a href="statistical-analysis-1.html#cb165-3" tabindex="-1"></a>                           <span class="at">measurePar =</span> <span class="fu">list</span>(<span class="at">use =</span> <span class="st">&quot;all.obs&quot;</span>,</span>
<span id="cb165-4"><a href="statistical-analysis-1.html#cb165-4" tabindex="-1"></a>                                             <span class="at">maxPOutliers =</span> <span class="dv">1</span>,</span>
<span id="cb165-5"><a href="statistical-analysis-1.html#cb165-5" tabindex="-1"></a>                                             <span class="at">nThreads =</span> <span class="dv">2</span>),</span>
<span id="cb165-6"><a href="statistical-analysis-1.html#cb165-6" tabindex="-1"></a>                           <span class="at">filtTax =</span> <span class="st">&quot;highestVar&quot;</span>,</span>
<span id="cb165-7"><a href="statistical-analysis-1.html#cb165-7" tabindex="-1"></a>                           <span class="at">filtTaxPar =</span> <span class="fu">list</span>(<span class="at">highestVar =</span> <span class="dv">50</span>),</span>
<span id="cb165-8"><a href="statistical-analysis-1.html#cb165-8" tabindex="-1"></a>                           <span class="at">filtSamp =</span> <span class="st">&quot;totalReads&quot;</span>,</span>
<span id="cb165-9"><a href="statistical-analysis-1.html#cb165-9" tabindex="-1"></a>                           <span class="at">filtSampPar =</span> <span class="fu">list</span>(<span class="at">totalReads =</span> <span class="dv">100</span>),</span>
<span id="cb165-10"><a href="statistical-analysis-1.html#cb165-10" tabindex="-1"></a>                           <span class="at">dissFunc =</span> <span class="st">&quot;TOMdiss&quot;</span>,</span>
<span id="cb165-11"><a href="statistical-analysis-1.html#cb165-11" tabindex="-1"></a>                           <span class="at">verbose =</span> <span class="dv">3</span>)</span></code></pre></div>
<pre><code>## ..will use 2 parallel threads.
##  Fraction of slow calculations: 0.000000</code></pre>
</div>
<div id="visualizing-the-network-2" class="section level5 hasAnchor" number="3.6.5.3.2">
<h5><span class="header-section-number">3.6.5.3.2</span> Visualizing the network<a href="statistical-analysis-1.html#visualizing-the-network-2" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb167"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb167-1"><a href="statistical-analysis-1.html#cb167-1" tabindex="-1"></a>props_single3 <span class="ot">&lt;-</span> <span class="fu">netAnalyze</span>(net_single3, <span class="at">clustMethod =</span> <span class="st">&quot;cluster_fast_greedy&quot;</span>)</span>
<span id="cb167-2"><a href="statistical-analysis-1.html#cb167-2" tabindex="-1"></a></span>
<span id="cb167-3"><a href="statistical-analysis-1.html#cb167-3" tabindex="-1"></a><span class="fu">plot</span>(props_single3, </span>
<span id="cb167-4"><a href="statistical-analysis-1.html#cb167-4" tabindex="-1"></a>     <span class="at">nodeColor =</span> <span class="st">&quot;cluster&quot;</span>, </span>
<span id="cb167-5"><a href="statistical-analysis-1.html#cb167-5" tabindex="-1"></a>     <span class="at">nodeSize =</span> <span class="st">&quot;eigenvector&quot;</span>,</span>
<span id="cb167-6"><a href="statistical-analysis-1.html#cb167-6" tabindex="-1"></a>     <span class="at">repulsion =</span> <span class="fl">0.8</span>,</span>
<span id="cb167-7"><a href="statistical-analysis-1.html#cb167-7" tabindex="-1"></a>     <span class="at">rmSingles =</span> <span class="cn">TRUE</span>,</span>
<span id="cb167-8"><a href="statistical-analysis-1.html#cb167-8" tabindex="-1"></a>     <span class="at">labelScale =</span> <span class="cn">FALSE</span>,</span>
<span id="cb167-9"><a href="statistical-analysis-1.html#cb167-9" tabindex="-1"></a>     <span class="at">cexLabels =</span> <span class="fl">1.6</span>,</span>
<span id="cb167-10"><a href="statistical-analysis-1.html#cb167-10" tabindex="-1"></a>     <span class="at">nodeSizeSpread =</span> <span class="dv">3</span>,</span>
<span id="cb167-11"><a href="statistical-analysis-1.html#cb167-11" tabindex="-1"></a>     <span class="at">cexNodes =</span> <span class="dv">2</span>,</span>
<span id="cb167-12"><a href="statistical-analysis-1.html#cb167-12" tabindex="-1"></a>     <span class="at">title1 =</span> <span class="st">&quot;Network on metabolomics with WGCNA correlations&quot;</span>, </span>
<span id="cb167-13"><a href="statistical-analysis-1.html#cb167-13" tabindex="-1"></a>     <span class="at">showTitle =</span> <span class="cn">TRUE</span>,</span>
<span id="cb167-14"><a href="statistical-analysis-1.html#cb167-14" tabindex="-1"></a>     <span class="at">cexTitle =</span> <span class="fl">1.5</span>)</span>
<span id="cb167-15"><a href="statistical-analysis-1.html#cb167-15" tabindex="-1"></a></span>
<span id="cb167-16"><a href="statistical-analysis-1.html#cb167-16" tabindex="-1"></a><span class="fu">legend</span>(<span class="fl">0.7</span>, <span class="fl">1.1</span>, <span class="at">cex =</span> <span class="fl">1.2</span>, <span class="at">title =</span> <span class="st">&quot;estimated correlation:&quot;</span>,</span>
<span id="cb167-17"><a href="statistical-analysis-1.html#cb167-17" tabindex="-1"></a>       <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;+&quot;</span>,<span class="st">&quot;-&quot;</span>), <span class="at">lty =</span> <span class="dv">1</span>, <span class="at">lwd =</span> <span class="dv">3</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;#009900&quot;</span>,<span class="st">&quot;red&quot;</span>),</span>
<span id="cb167-18"><a href="statistical-analysis-1.html#cb167-18" tabindex="-1"></a>       <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">horiz =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-167-1.png" width="1152" /></p>
</div>
</div>
</div>
<div id="network-comparison" class="section level3 hasAnchor" number="3.6.6">
<h3><span class="header-section-number">3.6.6</span> Network comparison<a href="statistical-analysis-1.html#network-comparison" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Comparing two networks by <strong>NetCoMi</strong>.</p>
<div id="data-preparing" class="section level4 hasAnchor" number="3.6.6.1">
<h4><span class="header-section-number">3.6.6.1</span> Data preparing<a href="statistical-analysis-1.html#data-preparing" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb168"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb168-1"><a href="statistical-analysis-1.html#cb168-1" tabindex="-1"></a>group_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Mild&quot;</span>, <span class="st">&quot;Severe&quot;</span>)</span>
<span id="cb168-2"><a href="statistical-analysis-1.html#cb168-2" tabindex="-1"></a></span>
<span id="cb168-3"><a href="statistical-analysis-1.html#cb168-3" tabindex="-1"></a>se_filter_subset <span class="ot">&lt;-</span> se_filter[, se_filter<span class="sc">$</span>group <span class="sc">%in%</span> group_names]</span>
<span id="cb168-4"><a href="statistical-analysis-1.html#cb168-4" tabindex="-1"></a>se_filter_subset<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">as.character</span>(se_filter_subset<span class="sc">$</span>group))</span>
<span id="cb168-5"><a href="statistical-analysis-1.html#cb168-5" tabindex="-1"></a>features_tab <span class="ot">&lt;-</span> SummarizedExperiment<span class="sc">::</span><span class="fu">assay</span>(se_filter_subset) <span class="sc">%&gt;%</span> </span>
<span id="cb168-6"><a href="statistical-analysis-1.html#cb168-6" tabindex="-1"></a>  <span class="fu">t</span>()</span>
<span id="cb168-7"><a href="statistical-analysis-1.html#cb168-7" tabindex="-1"></a>features_tab[<span class="fu">is.na</span>(features_tab)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb168-8"><a href="statistical-analysis-1.html#cb168-8" tabindex="-1"></a></span>
<span id="cb168-9"><a href="statistical-analysis-1.html#cb168-9" tabindex="-1"></a>group_vector <span class="ot">&lt;-</span> se_filter_subset<span class="sc">$</span>group</span></code></pre></div>
</div>
<div id="building-network-model-3" class="section level4 hasAnchor" number="3.6.6.2">
<h4><span class="header-section-number">3.6.6.2</span> Building network model<a href="statistical-analysis-1.html#building-network-model-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb169"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb169-1"><a href="statistical-analysis-1.html#cb169-1" tabindex="-1"></a>net_group <span class="ot">&lt;-</span> <span class="fu">netConstruct</span>(features_tab, </span>
<span id="cb169-2"><a href="statistical-analysis-1.html#cb169-2" tabindex="-1"></a>                          <span class="at">group =</span> group_vector, </span>
<span id="cb169-3"><a href="statistical-analysis-1.html#cb169-3" tabindex="-1"></a>                          <span class="at">measure =</span> <span class="st">&quot;pearson&quot;</span>,</span>
<span id="cb169-4"><a href="statistical-analysis-1.html#cb169-4" tabindex="-1"></a>                          <span class="at">normMethod =</span> <span class="st">&quot;clr&quot;</span>, </span>
<span id="cb169-5"><a href="statistical-analysis-1.html#cb169-5" tabindex="-1"></a>                          <span class="at">zeroMethod =</span> <span class="st">&quot;multRepl&quot;</span>,</span>
<span id="cb169-6"><a href="statistical-analysis-1.html#cb169-6" tabindex="-1"></a>                          <span class="at">sparsMethod =</span> <span class="st">&quot;threshold&quot;</span>, </span>
<span id="cb169-7"><a href="statistical-analysis-1.html#cb169-7" tabindex="-1"></a>                          <span class="at">thresh =</span> <span class="fl">0.3</span>,</span>
<span id="cb169-8"><a href="statistical-analysis-1.html#cb169-8" tabindex="-1"></a>                          <span class="at">verbose =</span> <span class="dv">3</span>,</span>
<span id="cb169-9"><a href="statistical-analysis-1.html#cb169-9" tabindex="-1"></a>                          <span class="at">seed =</span> <span class="dv">123</span>)</span></code></pre></div>
</div>
<div id="network-analysis-1" class="section level4 hasAnchor" number="3.6.6.3">
<h4><span class="header-section-number">3.6.6.3</span> Network analysis<a href="statistical-analysis-1.html#network-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb170"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb170-1"><a href="statistical-analysis-1.html#cb170-1" tabindex="-1"></a>props_group <span class="ot">&lt;-</span> <span class="fu">netAnalyze</span>(net_group, </span>
<span id="cb170-2"><a href="statistical-analysis-1.html#cb170-2" tabindex="-1"></a>                          <span class="at">centrLCC =</span> <span class="cn">FALSE</span>,</span>
<span id="cb170-3"><a href="statistical-analysis-1.html#cb170-3" tabindex="-1"></a>                          <span class="at">avDissIgnoreInf =</span> <span class="cn">TRUE</span>,</span>
<span id="cb170-4"><a href="statistical-analysis-1.html#cb170-4" tabindex="-1"></a>                          <span class="at">sPathNorm =</span> <span class="cn">FALSE</span>,</span>
<span id="cb170-5"><a href="statistical-analysis-1.html#cb170-5" tabindex="-1"></a>                          <span class="at">clustMethod =</span> <span class="st">&quot;cluster_fast_greedy&quot;</span>,</span>
<span id="cb170-6"><a href="statistical-analysis-1.html#cb170-6" tabindex="-1"></a>                          <span class="at">hubPar =</span> <span class="fu">c</span>(<span class="st">&quot;degree&quot;</span>, <span class="st">&quot;between&quot;</span>, <span class="st">&quot;closeness&quot;</span>),</span>
<span id="cb170-7"><a href="statistical-analysis-1.html#cb170-7" tabindex="-1"></a>                          <span class="at">hubQuant =</span> <span class="fl">0.9</span>,</span>
<span id="cb170-8"><a href="statistical-analysis-1.html#cb170-8" tabindex="-1"></a>                          <span class="at">lnormFit =</span> <span class="cn">TRUE</span>,</span>
<span id="cb170-9"><a href="statistical-analysis-1.html#cb170-9" tabindex="-1"></a>                          <span class="at">normDeg =</span> <span class="cn">FALSE</span>,</span>
<span id="cb170-10"><a href="statistical-analysis-1.html#cb170-10" tabindex="-1"></a>                          <span class="at">normBetw =</span> <span class="cn">FALSE</span>,</span>
<span id="cb170-11"><a href="statistical-analysis-1.html#cb170-11" tabindex="-1"></a>                          <span class="at">normClose =</span> <span class="cn">FALSE</span>,</span>
<span id="cb170-12"><a href="statistical-analysis-1.html#cb170-12" tabindex="-1"></a>                          <span class="at">normEigen =</span> <span class="cn">FALSE</span>)</span>
<span id="cb170-13"><a href="statistical-analysis-1.html#cb170-13" tabindex="-1"></a></span>
<span id="cb170-14"><a href="statistical-analysis-1.html#cb170-14" tabindex="-1"></a><span class="fu">summary</span>(props_group)</span></code></pre></div>
<pre><code>## 
## Component sizes
## ```````````````
## Group 1:         
## size: 167
##    #:   1
## 
## Group 2:         
## size: 167
##    #:   1
## ______________________________
## Global network properties
## `````````````````````````
##                          group &#39;1&#39; group &#39;2&#39;
## Number of components       1.00000   1.00000
## Clustering coefficient     0.46890   0.45147
## Modularity                 0.06069   0.06436
## Positive edge percentage  50.68785  52.11131
## Edge density               0.37234   0.37075
## Natural connectivity       0.10764   0.10844
## Vertex connectivity       33.00000  35.00000
## Edge connectivity         33.00000  35.00000
## Average dissimilarity*     0.67686   0.67218
## Average path length**      0.87528   0.86466
## 
##  *Dissimilarity = 1 - edge weight
## **Path length: Sum of dissimilarities along the path
## 
## ______________________________
## Clusters
## - In the whole network
## - Algorithm: cluster_fast_greedy
## ```````````````````````````````` 
## group &#39;1&#39;:              
## name:  1  2  3
##    #: 81 33 53
## 
## group &#39;2&#39;:              
## name:  1  2  3
##    #: 55 52 60
## 
## ______________________________
## Hubs
## - In alphabetical/numerical order
## - Based on log-normal quantiles of centralities
## ```````````````````````````````````````````````
##  group &#39;1&#39; group &#39;2&#39;
##    M_48258   M_33230
##    M_52447   M_35631
##    M_52467   M_42398
##              M_52452
##              M_52462
## 
## ______________________________
## Centrality measures
## - In decreasing order
## - Computed for the complete network
## ````````````````````````````````````
## Degree (unnormalized):
##          group &#39;1&#39; group &#39;2&#39;
##  M_53180        90        67
##  M_48258        89        63
##  M_52467        84        45
##  M_52447        83        56
##  M_46115        82        57
##             ______    ______
##  M_35631        57        83
##  M_33230        80        82
##  M_52474        77        81
##  M_52462        64        79
##  M_52452        61        79
## 
## Betweenness centrality (unnormalized):
##          group &#39;1&#39; group &#39;2&#39;
##  M_53180       127        70
##  M_52447       122        53
##  M_62805       117        45
##  M_52467       113        23
##  M_42449       107        76
##             ______    ______
##  M_42398        65       126
##  M_36600        63       104
##  M_52281        39        97
##  M_52462        70        96
##  M_52452        75        96
## 
## Closeness centrality (unnormalized):
##          group &#39;1&#39; group &#39;2&#39;
##  M_52464 234.21854 218.76093
##  M_52447 233.58239 219.26645
##  M_52669 233.55223 188.04641
##  M_33961  233.5356   237.105
##  M_19263 232.73107 211.31774
##             ______    ______
##  M_33230 228.08708 240.52716
##  M_52462 220.20663 240.09054
##  M_33955 231.77567 239.49465
##  M_35631 206.06295 239.35578
##  M_52474 228.59113 239.13909
## 
## Eigenvector centrality (unnormalized):
##          group &#39;1&#39; group &#39;2&#39;
##  M_52467   0.12638   0.05312
##  M_33230   0.12329   0.13395
##  M_48258   0.12211   0.11358
##  M_52447   0.12038   0.07532
##  M_35186   0.12025   0.13431
##             ______    ______
##  M_35631   0.07576   0.13915
##  M_52474   0.10472   0.13506
##  M_33955   0.12001   0.13501
##  M_35186   0.12025   0.13431
##  M_33230   0.12329   0.13395</code></pre>
</div>
<div id="visualizing-the-network-3" class="section level4 hasAnchor" number="3.6.6.4">
<h4><span class="header-section-number">3.6.6.4</span> Visualizing the network<a href="statistical-analysis-1.html#visualizing-the-network-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb172"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb172-1"><a href="statistical-analysis-1.html#cb172-1" tabindex="-1"></a><span class="fu">plot</span>(props_group, </span>
<span id="cb172-2"><a href="statistical-analysis-1.html#cb172-2" tabindex="-1"></a>     <span class="at">sameLayout =</span> <span class="cn">TRUE</span>, </span>
<span id="cb172-3"><a href="statistical-analysis-1.html#cb172-3" tabindex="-1"></a>     <span class="at">layoutGroup =</span> <span class="dv">1</span>,</span>
<span id="cb172-4"><a href="statistical-analysis-1.html#cb172-4" tabindex="-1"></a>     <span class="at">rmSingles =</span> <span class="st">&quot;inboth&quot;</span>, </span>
<span id="cb172-5"><a href="statistical-analysis-1.html#cb172-5" tabindex="-1"></a>     <span class="at">nodeSize =</span> <span class="st">&quot;mclr&quot;</span>, </span>
<span id="cb172-6"><a href="statistical-analysis-1.html#cb172-6" tabindex="-1"></a>     <span class="at">labelScale =</span> <span class="cn">FALSE</span>,</span>
<span id="cb172-7"><a href="statistical-analysis-1.html#cb172-7" tabindex="-1"></a>     <span class="at">cexNodes =</span> <span class="dv">1</span>, </span>
<span id="cb172-8"><a href="statistical-analysis-1.html#cb172-8" tabindex="-1"></a>     <span class="at">cexLabels =</span> <span class="fl">1.5</span>,</span>
<span id="cb172-9"><a href="statistical-analysis-1.html#cb172-9" tabindex="-1"></a>     <span class="at">cexHubLabels =</span> <span class="dv">2</span>,</span>
<span id="cb172-10"><a href="statistical-analysis-1.html#cb172-10" tabindex="-1"></a>     <span class="at">cexTitle =</span> <span class="dv">2</span>,</span>
<span id="cb172-11"><a href="statistical-analysis-1.html#cb172-11" tabindex="-1"></a>     <span class="at">groupNames =</span> group_names,</span>
<span id="cb172-12"><a href="statistical-analysis-1.html#cb172-12" tabindex="-1"></a>     <span class="at">hubBorderCol  =</span> <span class="st">&quot;gray40&quot;</span>)</span>
<span id="cb172-13"><a href="statistical-analysis-1.html#cb172-13" tabindex="-1"></a></span>
<span id="cb172-14"><a href="statistical-analysis-1.html#cb172-14" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;bottom&quot;</span>, <span class="at">title =</span> <span class="st">&quot;estimated association:&quot;</span>, <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;+&quot;</span>, <span class="st">&quot;-&quot;</span>), </span>
<span id="cb172-15"><a href="statistical-analysis-1.html#cb172-15" tabindex="-1"></a>       <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;#009900&quot;</span>,<span class="st">&quot;red&quot;</span>), <span class="at">inset =</span> <span class="fl">0.04</span>, <span class="at">cex =</span> <span class="dv">3</span>, <span class="at">lty =</span> <span class="dv">1</span>, <span class="at">lwd =</span> <span class="dv">4</span>, </span>
<span id="cb172-16"><a href="statistical-analysis-1.html#cb172-16" tabindex="-1"></a>       <span class="at">bty =</span> <span class="st">&quot;n&quot;</span>, <span class="at">horiz =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="Data-Analysis-in-Metabolomics_files/figure-html/unnamed-chunk-171-1.png" width="1920" /></p>
</div>
<div id="quantitative-network-comparison" class="section level4 hasAnchor" number="3.6.6.5">
<h4><span class="header-section-number">3.6.6.5</span> Quantitative network comparison<a href="statistical-analysis-1.html#quantitative-network-comparison" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb173"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb173-1"><a href="statistical-analysis-1.html#cb173-1" tabindex="-1"></a>comp_group <span class="ot">&lt;-</span> <span class="fu">netCompare</span>(props_group, <span class="at">permTest =</span> <span class="cn">FALSE</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb173-2"><a href="statistical-analysis-1.html#cb173-2" tabindex="-1"></a></span>
<span id="cb173-3"><a href="statistical-analysis-1.html#cb173-3" tabindex="-1"></a><span class="fu">summary</span>(comp_group, </span>
<span id="cb173-4"><a href="statistical-analysis-1.html#cb173-4" tabindex="-1"></a>        <span class="at">groupNames =</span> group_names,</span>
<span id="cb173-5"><a href="statistical-analysis-1.html#cb173-5" tabindex="-1"></a>        <span class="at">showCentr =</span> <span class="fu">c</span>(<span class="st">&quot;degree&quot;</span>, <span class="st">&quot;between&quot;</span>, <span class="st">&quot;closeness&quot;</span>), </span>
<span id="cb173-6"><a href="statistical-analysis-1.html#cb173-6" tabindex="-1"></a>        <span class="at">numbNodes =</span> <span class="dv">5</span>)</span></code></pre></div>
<pre><code>## 
## Comparison of Network Properties
## ----------------------------------
## CALL: 
## netCompare(x = props_group, permTest = FALSE, verbose = FALSE)
## 
## ______________________________
## Global network properties
## `````````````````````````
##                            Mild   Severe    difference
## Number of components      1.000    1.000         0.000
## Clustering coefficient    0.469    0.451         0.017
## Moduarity                 0.061    0.064         0.004
## Positive edge percentage 50.688   52.111         1.423
## Edge density              0.372    0.371         0.002
## Natural connectivity      0.108    0.108         0.001
## Vertex connectivity      33.000   35.000         2.000
## Edge connectivity        33.000   35.000         2.000
## Average dissimilarity*    0.677    0.672         0.005
## Average path length**     0.875    0.865         0.011
## -----
##  *: Dissimilarity = 1 - edge weight
## **Path length: Sum of dissimilarities along the path
## 
## ______________________________
## Jaccard index (similarity betw. sets of most central nodes)
## ``````````````````````````````````````````````````````````
##                     Jacc   P(&lt;=Jacc)     P(&gt;=Jacc)   
## degree             0.171    0.001255 **   0.999517   
## betweenness centr. 0.237    0.045289 *    0.974501   
## closeness centr.   0.348    0.656742      0.442460   
## eigenvec. centr.   0.382    0.838125      0.231086   
## hub taxa           0.000    0.039018 *    1.000000   
## -----
## Jaccard index ranges from 0 (compl. different) to 1 (sets equal)
## 
## ______________________________
## Adjusted Rand index (similarity betw. clusterings)
## ``````````````````````````````````````````````````
##    ARI       p-value
##  0.048         0.001
## -----
## ARI in [-1,1] with ARI=1: perfect agreement betw. clusterings,
##                    ARI=0: expected for two random clusterings
## p-value: two-tailed test with null hypothesis ARI=0
## 
## ______________________________
## Centrality measures
## - In decreasing order
## - Computed for the complete network
## ````````````````````````````````````
## Degree (unnormalized):
##         Mild Severe abs.diff.
## M_52610   35     75        40
## M_52467   84     45        39
## M_52669   77     38        39
## M_62805   81     45        36
## M_52478   36     68        32
## 
## Betweenness centrality (unnormalized):
##         Mild Severe abs.diff.
## M_52467  113     23        90
## M_57663  101     24        77
## M_62805  117     45        72
## M_52669   87     16        71
## M_52447  122     53        69
## 
## Closeness centrality (unnormalized):
##            Mild  Severe abs.diff.
## M_52669 233.552 188.046    45.506
## M_52467 232.354 194.418    37.936
## M_35631 206.063 239.356    33.293
## M_52610 187.991 219.596    31.605
## M_35153 191.201 222.693    31.491
## 
## _________________________________________________________
## Significance codes: ***: 0.001, **: 0.01, *: 0.05, .: 0.1</code></pre>
<hr />
</div>
</div>
</div>
<div id="systematic-information-1" class="section level2 hasAnchor" number="3.7">
<h2><span class="header-section-number">3.7</span> Systematic Information<a href="statistical-analysis-1.html#systematic-information-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb175"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb175-1"><a href="statistical-analysis-1.html#cb175-1" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 4.1.3 (2022-03-10)
##  os       macOS Monterey 12.2.1
##  system   x86_64, darwin17.0
##  ui       RStudio
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       Asia/Shanghai
##  date     2023-12-07
##  rstudio  2023.09.0+463 Desert Sunflower (desktop)
##  pandoc   3.1.1 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/ (via rmarkdown)
## 
## ─ Packages ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package              * version    date (UTC) lib source
##  abind                  1.4-5      2016-07-21 [2] CRAN (R 4.1.0)
##  ade4                   1.7-22     2023-02-06 [2] CRAN (R 4.1.2)
##  annotate               1.72.0     2021-10-26 [2] Bioconductor
##  AnnotationDbi          1.60.2     2023-03-10 [2] Bioconductor
##  ape                    5.7-1      2023-03-13 [2] CRAN (R 4.1.2)
##  backports              1.4.1      2021-12-13 [2] CRAN (R 4.1.0)
##  base64enc              0.1-3      2015-07-28 [2] CRAN (R 4.1.0)
##  Biobase              * 2.54.0     2021-10-26 [2] Bioconductor
##  BiocGenerics         * 0.40.0     2021-10-26 [2] Bioconductor
##  BiocParallel           1.28.3     2021-12-09 [2] Bioconductor
##  biomformat             1.22.0     2021-10-26 [2] Bioconductor
##  Biostrings             2.62.0     2021-10-26 [2] Bioconductor
##  bit                    4.0.5      2022-11-15 [2] CRAN (R 4.1.2)
##  bit64                  4.0.5      2020-08-30 [2] CRAN (R 4.1.0)
##  bitops                 1.0-7      2021-04-24 [2] CRAN (R 4.1.0)
##  blob                   1.2.4      2023-03-17 [2] CRAN (R 4.1.2)
##  bookdown               0.34       2023-05-09 [2] CRAN (R 4.1.2)
##  broom                  1.0.5      2023-06-09 [2] CRAN (R 4.1.3)
##  bslib                  0.6.0      2023-11-21 [1] CRAN (R 4.1.3)
##  cachem                 1.0.8      2023-05-01 [2] CRAN (R 4.1.2)
##  callr                  3.7.3      2022-11-02 [2] CRAN (R 4.1.2)
##  car                    3.1-2      2023-03-30 [2] CRAN (R 4.1.2)
##  carData                3.0-5      2022-01-06 [2] CRAN (R 4.1.2)
##  caret                  6.0-94     2023-03-21 [2] CRAN (R 4.1.2)
##  caTools                1.18.2     2021-03-28 [2] CRAN (R 4.1.0)
##  cellranger             1.1.0      2016-07-27 [2] CRAN (R 4.1.0)
##  checkmate              2.2.0      2023-04-27 [2] CRAN (R 4.1.2)
##  circlize               0.4.15     2022-05-10 [2] CRAN (R 4.1.2)
##  class                  7.3-22     2023-05-03 [2] CRAN (R 4.1.2)
##  cli                    3.6.1      2023-03-23 [2] CRAN (R 4.1.2)
##  clue                   0.3-64     2023-01-31 [2] CRAN (R 4.1.2)
##  cluster              * 2.1.4      2022-08-22 [2] CRAN (R 4.1.2)
##  codetools              0.2-19     2023-02-01 [2] CRAN (R 4.1.2)
##  coin                   1.4-2      2021-10-08 [2] CRAN (R 4.1.0)
##  colorspace             2.1-0      2023-01-23 [2] CRAN (R 4.1.2)
##  ComplexHeatmap         2.10.0     2021-10-26 [2] Bioconductor
##  corpcor                1.6.10     2021-09-16 [2] CRAN (R 4.1.0)
##  cowplot                1.1.1      2020-12-30 [2] CRAN (R 4.1.0)
##  crayon                 1.5.2      2022-09-29 [2] CRAN (R 4.1.2)
##  data.table             1.14.8     2023-02-17 [2] CRAN (R 4.1.2)
##  DBI                    1.1.3      2022-06-18 [2] CRAN (R 4.1.2)
##  DelayedArray           0.20.0     2021-10-26 [2] Bioconductor
##  dendextend           * 1.17.1     2023-03-25 [2] CRAN (R 4.1.2)
##  DESeq2                 1.34.0     2021-10-26 [2] Bioconductor
##  devtools               2.4.5      2022-10-11 [2] CRAN (R 4.1.2)
##  digest                 0.6.33     2023-07-07 [1] CRAN (R 4.1.3)
##  doParallel             1.0.17     2022-02-07 [2] CRAN (R 4.1.2)
##  doRNG                  1.8.6      2023-01-16 [2] CRAN (R 4.1.2)
##  doSNOW                 1.0.20     2022-02-04 [2] CRAN (R 4.1.2)
##  dplyr                * 1.1.2      2023-04-20 [2] CRAN (R 4.1.2)
##  dynamicTreeCut         1.63-1     2016-03-11 [2] CRAN (R 4.1.0)
##  e1071                  1.7-13     2023-02-01 [2] CRAN (R 4.1.2)
##  edgeR                  3.36.0     2021-10-26 [2] Bioconductor
##  ellipse                0.4.5      2023-04-05 [2] CRAN (R 4.1.2)
##  ellipsis               0.3.2      2021-04-29 [2] CRAN (R 4.1.0)
##  evaluate               0.21       2023-05-05 [2] CRAN (R 4.1.2)
##  factoextra           * 1.0.7      2020-04-01 [2] CRAN (R 4.1.0)
##  fansi                  1.0.4      2023-01-22 [2] CRAN (R 4.1.2)
##  farver                 2.1.1      2022-07-06 [2] CRAN (R 4.1.2)
##  fastcluster            1.2.3      2021-05-24 [2] CRAN (R 4.1.0)
##  fastmap                1.1.1      2023-02-24 [2] CRAN (R 4.1.2)
##  fdrtool                1.2.17     2021-11-13 [2] CRAN (R 4.1.0)
##  filematrix             1.3        2018-02-27 [2] CRAN (R 4.1.0)
##  foreach                1.5.2      2022-02-02 [2] CRAN (R 4.1.2)
##  foreign                0.8-84     2022-12-06 [2] CRAN (R 4.1.2)
##  forestplot             3.1.1      2022-12-06 [2] CRAN (R 4.1.2)
##  Formula                1.2-5      2023-02-24 [2] CRAN (R 4.1.2)
##  fs                     1.6.2      2023-04-25 [2] CRAN (R 4.1.2)
##  future                 1.33.0     2023-07-01 [2] CRAN (R 4.1.3)
##  future.apply           1.11.0     2023-05-21 [2] CRAN (R 4.1.3)
##  genefilter             1.76.0     2021-10-26 [2] Bioconductor
##  geneplotter            1.72.0     2021-10-26 [2] Bioconductor
##  generics               0.1.3      2022-07-05 [2] CRAN (R 4.1.2)
##  GenomeInfoDb         * 1.30.1     2022-01-30 [2] Bioconductor
##  GenomeInfoDbData       1.2.7      2022-03-09 [2] Bioconductor
##  GenomicRanges        * 1.46.1     2021-11-18 [2] Bioconductor
##  GetoptLong             1.0.5      2020-12-15 [2] CRAN (R 4.1.0)
##  ggforce                0.4.1      2022-10-04 [2] CRAN (R 4.1.2)
##  ggplot2              * 3.4.2      2023-04-03 [2] CRAN (R 4.1.2)
##  ggpubr                 0.6.0      2023-02-10 [2] CRAN (R 4.1.2)
##  ggraph               * 2.1.0.9000 2023-07-11 [1] Github (thomasp85/ggraph@febab71)
##  ggrepel                0.9.3      2023-02-03 [2] CRAN (R 4.1.2)
##  ggsignif               0.6.4      2022-10-13 [2] CRAN (R 4.1.2)
##  glasso                 1.11       2019-10-01 [2] CRAN (R 4.1.0)
##  glmnet               * 4.1-7      2023-03-23 [2] CRAN (R 4.1.2)
##  GlobalOptions          0.1.2      2020-06-10 [2] CRAN (R 4.1.0)
##  globals                0.16.2     2022-11-21 [2] CRAN (R 4.1.2)
##  glue                 * 1.6.2      2022-02-24 [2] CRAN (R 4.1.2)
##  Gmisc                * 3.0.2      2023-03-13 [2] CRAN (R 4.1.2)
##  gmm                    1.8        2023-06-06 [2] CRAN (R 4.1.3)
##  gmp                    0.7-1      2023-02-07 [2] CRAN (R 4.1.2)
##  GO.db                  3.14.0     2022-04-11 [2] Bioconductor
##  gower                  1.0.1      2022-12-22 [2] CRAN (R 4.1.2)
##  gplots                 3.1.3      2022-04-25 [2] CRAN (R 4.1.2)
##  graphlayouts           1.0.0      2023-05-01 [2] CRAN (R 4.1.2)
##  gridExtra              2.3        2017-09-09 [2] CRAN (R 4.1.0)
##  gtable                 0.3.3      2023-03-21 [2] CRAN (R 4.1.2)
##  gtools                 3.9.4      2022-11-27 [2] CRAN (R 4.1.2)
##  hardhat                1.3.0      2023-03-30 [2] CRAN (R 4.1.2)
##  highr                  0.10       2022-12-22 [2] CRAN (R 4.1.2)
##  Hmisc                  5.1-0      2023-05-08 [2] CRAN (R 4.1.2)
##  htmlTable            * 2.4.1      2022-07-07 [2] CRAN (R 4.1.2)
##  htmltools              0.5.7      2023-11-03 [1] CRAN (R 4.1.3)
##  htmlwidgets            1.6.2      2023-03-17 [2] CRAN (R 4.1.2)
##  httpuv                 1.6.11     2023-05-11 [2] CRAN (R 4.1.3)
##  httr                   1.4.6      2023-05-08 [2] CRAN (R 4.1.2)
##  huge                   1.3.5      2021-06-30 [2] CRAN (R 4.1.0)
##  igraph                 1.5.0      2023-06-16 [1] CRAN (R 4.1.3)
##  impute                 1.68.0     2021-10-26 [2] Bioconductor
##  imputeLCMD             2.1        2022-06-10 [2] CRAN (R 4.1.2)
##  ipred                  0.9-14     2023-03-09 [2] CRAN (R 4.1.2)
##  IRanges              * 2.28.0     2021-10-26 [2] Bioconductor
##  irlba                  2.3.5.1    2022-10-03 [2] CRAN (R 4.1.2)
##  iterators              1.0.14     2022-02-05 [2] CRAN (R 4.1.2)
##  itertools              0.1-3      2014-03-12 [2] CRAN (R 4.1.0)
##  jpeg                   0.1-10     2022-11-29 [2] CRAN (R 4.1.2)
##  jquerylib              0.1.4      2021-04-26 [2] CRAN (R 4.1.0)
##  jsonlite               1.8.7      2023-06-29 [2] CRAN (R 4.1.3)
##  KEGGREST               1.34.0     2021-10-26 [2] Bioconductor
##  KernSmooth             2.23-22    2023-07-10 [2] CRAN (R 4.1.3)
##  knitr                  1.43       2023-05-25 [2] CRAN (R 4.1.3)
##  labeling               0.4.2      2020-10-20 [2] CRAN (R 4.1.0)
##  later                  1.3.1      2023-05-02 [2] CRAN (R 4.1.2)
##  lattice                0.21-8     2023-04-05 [2] CRAN (R 4.1.2)
##  lava                   1.7.2.1    2023-02-27 [2] CRAN (R 4.1.2)
##  lavaan                 0.6-15     2023-03-14 [2] CRAN (R 4.1.2)
##  lazyeval               0.2.2      2019-03-15 [2] CRAN (R 4.1.0)
##  libcoin                1.0-9      2021-09-27 [2] CRAN (R 4.1.0)
##  lifecycle              1.0.3      2022-10-07 [2] CRAN (R 4.1.2)
##  limma                  3.50.3     2022-04-07 [2] Bioconductor
##  listenv                0.9.0      2022-12-16 [2] CRAN (R 4.1.2)
##  locfit                 1.5-9.8    2023-06-11 [2] CRAN (R 4.1.3)
##  lubridate              1.9.2      2023-02-10 [2] CRAN (R 4.1.2)
##  magrittr             * 2.0.3      2022-03-30 [2] CRAN (R 4.1.2)
##  MASS                   7.3-60     2023-05-04 [2] CRAN (R 4.1.2)
##  Matrix               * 1.6-0      2023-07-08 [2] CRAN (R 4.1.3)
##  MatrixGenerics       * 1.6.0      2021-10-26 [2] Bioconductor
##  matrixStats          * 1.0.0      2023-06-02 [2] CRAN (R 4.1.3)
##  memoise                2.0.1      2021-11-26 [2] CRAN (R 4.1.0)
##  metagenomeSeq          1.36.0     2021-10-26 [2] Bioconductor
##  mgcv                   1.8-42     2023-03-02 [2] CRAN (R 4.1.2)
##  mime                   0.12       2021-09-28 [2] CRAN (R 4.1.0)
##  miniUI                 0.1.1.1    2018-05-18 [2] CRAN (R 4.1.0)
##  missForest             1.5        2022-04-14 [2] CRAN (R 4.1.2)
##  mixedCCA               1.6.2      2022-09-09 [2] CRAN (R 4.1.2)
##  mixOmics               6.18.1     2021-11-18 [2] Bioconductor (R 4.1.2)
##  mnormt                 2.1.1      2022-09-26 [2] CRAN (R 4.1.2)
##  ModelMetrics           1.2.2.2    2020-03-17 [2] CRAN (R 4.1.0)
##  modeltools             0.2-23     2020-03-05 [2] CRAN (R 4.1.0)
##  multcomp               1.4-25     2023-06-20 [2] CRAN (R 4.1.3)
##  multtest               2.50.0     2021-10-26 [2] Bioconductor
##  munsell                0.5.0      2018-06-12 [2] CRAN (R 4.1.0)
##  mvtnorm                1.2-2      2023-06-08 [2] CRAN (R 4.1.3)
##  NetCoMi              * 1.0.3      2022-07-14 [2] Github (stefpeschel/NetCoMi@d4d80d3)
##  nlme                   3.1-162    2023-01-31 [2] CRAN (R 4.1.2)
##  nnet                   7.3-19     2023-05-03 [2] CRAN (R 4.1.2)
##  norm                   1.0-11.1   2023-06-18 [2] CRAN (R 4.1.3)
##  parallelly             1.36.0     2023-05-26 [2] CRAN (R 4.1.3)
##  pbapply                1.7-2      2023-06-27 [2] CRAN (R 4.1.3)
##  pbivnorm               0.6.0      2015-01-23 [2] CRAN (R 4.1.0)
##  pcaMethods             1.86.0     2021-10-26 [2] Bioconductor
##  pcaPP                  2.0-3      2022-10-24 [2] CRAN (R 4.1.2)
##  permute                0.9-7      2022-01-27 [2] CRAN (R 4.1.2)
##  phyloseq               1.38.0     2021-10-26 [2] Bioconductor
##  pillar                 1.9.0      2023-03-22 [2] CRAN (R 4.1.2)
##  pkgbuild               1.4.2      2023-06-26 [2] CRAN (R 4.1.3)
##  pkgconfig              2.0.3      2019-09-22 [2] CRAN (R 4.1.0)
##  pkgload                1.3.2.1    2023-07-08 [2] CRAN (R 4.1.3)
##  plotly               * 4.10.2     2023-06-03 [2] CRAN (R 4.1.3)
##  plyr                   1.8.8      2022-11-11 [2] CRAN (R 4.1.2)
##  png                    0.1-8      2022-11-29 [2] CRAN (R 4.1.2)
##  polyclip               1.10-4     2022-10-20 [2] CRAN (R 4.1.2)
##  POMA                 * 1.7.2      2022-07-26 [2] Github (pcastellanoescuder/POMA@bc8a972)
##  preprocessCore         1.56.0     2021-10-26 [2] Bioconductor
##  prettyunits            1.1.1      2020-01-24 [2] CRAN (R 4.1.0)
##  pROC                   1.18.4     2023-07-06 [2] CRAN (R 4.1.3)
##  processx               3.8.2      2023-06-30 [2] CRAN (R 4.1.3)
##  prodlim                2023.03.31 2023-04-02 [2] CRAN (R 4.1.2)
##  profvis                0.3.8      2023-05-02 [2] CRAN (R 4.1.2)
##  promises               1.2.0.1    2021-02-11 [2] CRAN (R 4.1.0)
##  proxy                  0.4-27     2022-06-09 [2] CRAN (R 4.1.2)
##  ps                     1.7.5      2023-04-18 [2] CRAN (R 4.1.2)
##  psych                  2.3.6      2023-06-21 [2] CRAN (R 4.1.3)
##  pulsar                 0.3.10     2023-01-26 [2] CRAN (R 4.1.2)
##  purrr                  1.0.1      2023-01-10 [2] CRAN (R 4.1.2)
##  qgraph                 1.9.5      2023-05-16 [2] CRAN (R 4.1.3)
##  quadprog               1.5-8      2019-11-20 [2] CRAN (R 4.1.0)
##  R6                     2.5.1      2021-08-19 [2] CRAN (R 4.1.0)
##  randomForest           4.7-1.1    2022-05-23 [2] CRAN (R 4.1.2)
##  RankProd               3.20.0     2021-10-26 [2] Bioconductor
##  rARPACK                0.11-0     2016-03-10 [2] CRAN (R 4.1.0)
##  rbibutils              2.2.13     2023-01-13 [2] CRAN (R 4.1.2)
##  RColorBrewer           1.1-3      2022-04-03 [2] CRAN (R 4.1.2)
##  Rcpp                 * 1.0.11     2023-07-06 [1] CRAN (R 4.1.3)
##  RCurl                  1.98-1.12  2023-03-27 [2] CRAN (R 4.1.2)
##  Rdpack                 2.4        2022-07-20 [2] CRAN (R 4.1.2)
##  readxl               * 1.4.3      2023-07-06 [2] CRAN (R 4.1.3)
##  recipes                1.0.6      2023-04-25 [2] CRAN (R 4.1.2)
##  remotes                2.4.2      2021-11-30 [2] CRAN (R 4.1.0)
##  reshape2               1.4.4      2020-04-09 [2] CRAN (R 4.1.0)
##  rhdf5                  2.38.1     2022-03-10 [2] Bioconductor
##  rhdf5filters           1.6.0      2021-10-26 [2] Bioconductor
##  Rhdf5lib               1.16.0     2021-10-26 [2] Bioconductor
##  rjson                  0.2.21     2022-01-09 [2] CRAN (R 4.1.2)
##  rlang                  1.1.1      2023-04-28 [1] CRAN (R 4.1.2)
##  rmarkdown              2.23       2023-07-01 [2] CRAN (R 4.1.3)
##  Rmpfr                  0.9-2      2023-04-22 [2] CRAN (R 4.1.2)
##  rngtools               1.5.2      2021-09-20 [2] CRAN (R 4.1.0)
##  rootSolve              1.8.2.3    2021-09-29 [2] CRAN (R 4.1.0)
##  ropls                * 1.26.4     2022-01-11 [2] Bioconductor
##  rpart                  4.1.19     2022-10-21 [2] CRAN (R 4.1.2)
##  RSpectra               0.16-1     2022-04-24 [2] CRAN (R 4.1.2)
##  RSQLite                2.3.1      2023-04-03 [2] CRAN (R 4.1.2)
##  rstatix                0.7.2      2023-02-01 [2] CRAN (R 4.1.2)
##  rstudioapi             0.15.0     2023-07-07 [2] CRAN (R 4.1.3)
##  S4Vectors            * 0.32.4     2022-03-29 [2] Bioconductor
##  sandwich               3.0-2      2022-06-15 [2] CRAN (R 4.1.2)
##  sass                   0.4.6      2023-05-03 [2] CRAN (R 4.1.2)
##  scales                 1.2.1      2022-08-20 [2] CRAN (R 4.1.2)
##  sessioninfo            1.2.2      2021-12-06 [2] CRAN (R 4.1.0)
##  shape                  1.4.6      2021-05-19 [2] CRAN (R 4.1.0)
##  shiny                  1.7.4.1    2023-07-06 [2] CRAN (R 4.1.3)
##  snow                   0.4-4      2021-10-27 [2] CRAN (R 4.1.0)
##  SpiecEasi            * 1.1.2      2022-07-14 [2] Github (zdk123/SpiecEasi@c463727)
##  SPRING               * 1.0.4      2022-08-03 [2] Github (GraceYoon/SPRING@3d641a4)
##  stringi                1.7.12     2023-01-11 [2] CRAN (R 4.1.2)
##  stringr                1.5.0      2022-12-02 [2] CRAN (R 4.1.2)
##  SummarizedExperiment * 1.24.0     2021-10-26 [2] Bioconductor
##  survival               3.5-5      2023-03-12 [2] CRAN (R 4.1.2)
##  TH.data                1.1-2      2023-04-17 [2] CRAN (R 4.1.2)
##  tibble               * 3.2.1      2023-03-20 [2] CRAN (R 4.1.2)
##  tidygraph              1.2.3      2023-02-01 [2] CRAN (R 4.1.2)
##  tidyr                  1.3.0      2023-01-24 [2] CRAN (R 4.1.2)
##  tidyselect             1.2.0      2022-10-10 [2] CRAN (R 4.1.2)
##  timechange             0.2.0      2023-01-11 [2] CRAN (R 4.1.2)
##  timeDate               4022.108   2023-01-07 [2] CRAN (R 4.1.2)
##  tmvtnorm               1.5        2022-03-22 [2] CRAN (R 4.1.2)
##  tweenr                 2.0.2      2022-09-06 [2] CRAN (R 4.1.2)
##  urlchecker             1.0.1      2021-11-30 [2] CRAN (R 4.1.0)
##  usethis                2.2.2      2023-07-06 [2] CRAN (R 4.1.3)
##  utf8                   1.2.3      2023-01-31 [2] CRAN (R 4.1.2)
##  vctrs                  0.6.3      2023-06-14 [1] CRAN (R 4.1.3)
##  vegan                  2.6-4      2022-10-11 [2] CRAN (R 4.1.2)
##  VGAM                   1.1-8      2023-03-09 [2] CRAN (R 4.1.2)
##  viridis                0.6.3      2023-05-03 [2] CRAN (R 4.1.2)
##  viridisLite            0.4.2      2023-05-02 [2] CRAN (R 4.1.2)
##  WGCNA                  1.72-1     2023-01-18 [2] CRAN (R 4.1.2)
##  withr                  2.5.0      2022-03-03 [2] CRAN (R 4.1.2)
##  Wrench                 1.12.0     2021-10-26 [2] Bioconductor
##  xfun                   0.40       2023-08-09 [1] CRAN (R 4.1.3)
##  XMAS2                * 2.2.0      2023-12-06 [1] local
##  XML                    3.99-0.14  2023-03-19 [2] CRAN (R 4.1.2)
##  xtable                 1.8-4      2019-04-21 [2] CRAN (R 4.1.0)
##  XVector                0.34.0     2021-10-26 [2] Bioconductor
##  yaml                   2.3.7      2023-01-23 [2] CRAN (R 4.1.2)
##  zlibbioc               1.40.0     2021-10-26 [2] Bioconductor
##  zoo                    1.8-12     2023-04-13 [2] CRAN (R 4.1.2)
## 
##  [1] /Users/zouhua/Library/R/x86_64/4.1/library
##  [2] /Library/Frameworks/R.framework/Versions/4.1/Resources/library
## 
## ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>

</div>
</div>
<h3>References<a href="references.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-castellano2021pomashiny" class="csl-entry">
Castellano-Escuder, Pol, Raúl González-Domı́nguez, Francesc Carmona-Pontaque, Cristina Andrés-Lacueva, and Alex Sánchez-Pla. 2021. <span>“POMAShiny: A User-Friendly Web-Based Workflow for Metabolomics and Proteomics Data Analysis.”</span> <em>PLoS Computational Biology</em> 17 (7): e1009148.
</div>
<div id="ref-pang2020metaboanalystr" class="csl-entry">
Pang, Zhiqiang, Jasmine Chong, Shuzhao Li, and Jianguo Xia. 2020. <span>“MetaboAnalystR 3.0: Toward an Optimized Workflow for Global Metabolomics.”</span> <em>Metabolites</em> 10 (5): 186.
</div>
<div id="ref-peschel2021netcomi" class="csl-entry">
Peschel, Stefanie, Christian L Müller, Erika von Mutius, Anne-Laure Boulesteix, and Martin Depner. 2021. <span>“NetCoMi: Network Construction and Comparison for Microbiome Data in r.”</span> <em>Briefings in Bioinformatics</em> 22 (4): bbaa290.
</div>
<div id="ref-rohart2017mixomics" class="csl-entry">
Rohart, Florian, Benoı̂t Gautier, Amrit Singh, and Kim-Anh Lê Cao. 2017. <span>“mixOmics: An r Package for ‘Omics Feature Selection and Multiple Data Integration.”</span> <em>PLoS Computational Biology</em> 13 (11): e1005752.
</div>
<div id="ref-zeybel2022multiomics" class="csl-entry">
Zeybel, Mujdat, Muhammad Arif, Xiangyu Li, Ozlem Altay, Hong Yang, Mengnan Shi, Murat Akyildiz, et al. 2022. <span>“Multiomics Analysis Reveals the Impact of Microbiota on Host Metabolism in Hepatic Steatosis.”</span> <em>Advanced Science</em> 9 (11): 2104373.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="data-processing.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="functional-analysis-1.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/Metabolomics_Aanlysis/blob/main/01-Statistical-Analysis.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true,
"toc_depth": 3
}
});
});
</script>

</body>

</html>
