<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 11 Visualization | XMAS 2.0 Tutorial</title>
  <meta name="description" content="A tutorial of XMAS 2.0 package for in-house training" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 11 Visualization | XMAS 2.0 Tutorial" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 11 Visualization | XMAS 2.0 Tutorial" />
  
  <meta name="twitter:description" content="A tutorial of XMAS 2.0 package for in-house training" />
  

<meta name="author" content="Hua Zou" />


<meta name="date" content="2023-11-30" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="differential-analysis.html"/>
<link rel="next" href="association-analysis.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.28/datatables.js"></script>
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="assets/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">XMAS 2.0 Tutorial</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#overview-of-the-design-of-xmas"><i class="fa fa-check"></i><b>1.1</b> Overview of the design of <strong>XMAS</strong></a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#input-data"><i class="fa fa-check"></i><b>1.2</b> Input data</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#methods"><i class="fa fa-check"></i><b>1.3</b> Methods</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#outline-of-this-tutorial"><i class="fa fa-check"></i><b>1.4</b> Outline of this Tutorial</a></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#key-publications"><i class="fa fa-check"></i><b>1.5</b> Key publications</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#citation"><i class="fa fa-check"></i><b>1.6</b> Citation</a></li>
<li class="chapter" data-level="1.7" data-path="index.html"><a href="index.html#change-log"><i class="fa fa-check"></i><b>1.7</b> Change log</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html"><i class="fa fa-check"></i><b>2</b> How to install XMAS</a>
<ul>
<li class="chapter" data-level="2.1" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#description"><i class="fa fa-check"></i><b>2.1</b> Description</a></li>
<li class="chapter" data-level="2.2" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#installation-of-dependencies"><i class="fa fa-check"></i><b>2.2</b> Installation of dependencies</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#packages-in-cran-bioconductor"><i class="fa fa-check"></i><b>2.2.1</b> Packages in CRAN &amp; Bioconductor</a></li>
<li class="chapter" data-level="2.2.2" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#packages-in-github"><i class="fa fa-check"></i><b>2.2.2</b> Packages in github</a></li>
<li class="chapter" data-level="2.2.3" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#packages-from-website"><i class="fa fa-check"></i><b>2.2.3</b> Packages from website</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#installation"><i class="fa fa-check"></i><b>2.3</b> Installation</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#installation-in-linux-or-mac-os-through-command-line"><i class="fa fa-check"></i><b>2.3.1</b> Installation in linux or Mac os through command line</a></li>
<li class="chapter" data-level="2.3.2" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#installation-by-r-from-gitlab"><i class="fa fa-check"></i><b>2.3.2</b> Installation by R from gitlab</a></li>
<li class="chapter" data-level="2.3.3" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#installation-by-r-from-local-xmas.tar.gz-file"><i class="fa fa-check"></i><b>2.3.3</b> Installation by R from local XMAS*.tar.gz file</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#vignette"><i class="fa fa-check"></i><b>2.4</b> Vignette</a></li>
<li class="chapter" data-level="2.5" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#authors"><i class="fa fa-check"></i><b>2.5</b> Authors</a></li>
<li class="chapter" data-level="2.6" data-path="how-to-install-xmas.html"><a href="how-to-install-xmas.html#future-plan"><i class="fa fa-check"></i><b>2.6</b> Future plan</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html"><i class="fa fa-check"></i><b>3</b> Convert inputdata into phyloseq object</a>
<ul>
<li class="chapter" data-level="3.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#loading-packages"><i class="fa fa-check"></i><b>3.1</b> Loading Packages</a></li>
<li class="chapter" data-level="3.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#importing-data"><i class="fa fa-check"></i><b>3.2</b> Importing Data</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#result-from-dada2"><i class="fa fa-check"></i><b>3.2.1</b> Result from dada2</a></li>
<li class="chapter" data-level="3.2.2" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#result-from-metaphlan"><i class="fa fa-check"></i><b>3.2.2</b> Result from Metaphlan</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="convert-inputdata-into-phyloseq-object.html"><a href="convert-inputdata-into-phyloseq-object.html#systematic-information"><i class="fa fa-check"></i><b>3.3</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="quality-evaluation.html"><a href="quality-evaluation.html"><i class="fa fa-check"></i><b>4</b> Quality Evaluation</a>
<ul>
<li class="chapter" data-level="4.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#loading-packages-1"><i class="fa fa-check"></i><b>4.1</b> Loading Packages</a></li>
<li class="chapter" data-level="4.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#importing-data-1"><i class="fa fa-check"></i><b>4.2</b> Importing Data</a></li>
<li class="chapter" data-level="4.3" data-path="quality-evaluation.html"><a href="quality-evaluation.html#reads-track"><i class="fa fa-check"></i><b>4.3</b> Reads track</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#plotting-dad2"><i class="fa fa-check"></i><b>4.3.1</b> Plotting DAD2</a></li>
<li class="chapter" data-level="4.3.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#plotting-metaphlan"><i class="fa fa-check"></i><b>4.3.2</b> Plotting Metaphlan</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="quality-evaluation.html"><a href="quality-evaluation.html#spike-in-brs-sample-assessment"><i class="fa fa-check"></i><b>4.4</b> Spike-in (BRS) sample assessment</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#s-data"><i class="fa fa-check"></i><b>4.4.1</b> 16s data</a></li>
<li class="chapter" data-level="4.4.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#metagenomic-data"><i class="fa fa-check"></i><b>4.4.2</b> Metagenomic data</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="quality-evaluation.html"><a href="quality-evaluation.html#removing-brs"><i class="fa fa-check"></i><b>4.5</b> Removing BRS</a></li>
<li class="chapter" data-level="4.6" data-path="quality-evaluation.html"><a href="quality-evaluation.html#rarefaction-curves"><i class="fa fa-check"></i><b>4.6</b> Rarefaction Curves</a></li>
<li class="chapter" data-level="4.7" data-path="quality-evaluation.html"><a href="quality-evaluation.html#summarize-phyloseq-class-object"><i class="fa fa-check"></i><b>4.7</b> Summarize phyloseq-class object</a></li>
<li class="chapter" data-level="4.8" data-path="quality-evaluation.html"><a href="quality-evaluation.html#variability"><i class="fa fa-check"></i><b>4.8</b> Variability</a>
<ul>
<li class="chapter" data-level="4.8.1" data-path="quality-evaluation.html"><a href="quality-evaluation.html#coefficient-of-variation"><i class="fa fa-check"></i><b>4.8.1</b> Coefficient of variation</a></li>
<li class="chapter" data-level="4.8.2" data-path="quality-evaluation.html"><a href="quality-evaluation.html#distribution-of-number-of-sequences"><i class="fa fa-check"></i><b>4.8.2</b> Distribution of number of sequences</a></li>
</ul></li>
<li class="chapter" data-level="4.9" data-path="quality-evaluation.html"><a href="quality-evaluation.html#systematic-information-1"><i class="fa fa-check"></i><b>4.9</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="pre-processing.html"><a href="pre-processing.html"><i class="fa fa-check"></i><b>5</b> Pre-processing</a>
<ul>
<li class="chapter" data-level="5.1" data-path="pre-processing.html"><a href="pre-processing.html#loading-packages-2"><i class="fa fa-check"></i><b>5.1</b> Loading Packages</a></li>
<li class="chapter" data-level="5.2" data-path="pre-processing.html"><a href="pre-processing.html#importing-data-2"><i class="fa fa-check"></i><b>5.2</b> Importing Data</a></li>
<li class="chapter" data-level="5.3" data-path="pre-processing.html"><a href="pre-processing.html#transformation"><i class="fa fa-check"></i><b>5.3</b> Transformation</a></li>
<li class="chapter" data-level="5.4" data-path="pre-processing.html"><a href="pre-processing.html#normalization"><i class="fa fa-check"></i><b>5.4</b> Normalization</a></li>
<li class="chapter" data-level="5.5" data-path="pre-processing.html"><a href="pre-processing.html#filtering"><i class="fa fa-check"></i><b>5.5</b> Filtering</a></li>
<li class="chapter" data-level="5.6" data-path="pre-processing.html"><a href="pre-processing.html#trimming"><i class="fa fa-check"></i><b>5.6</b> Trimming</a></li>
<li class="chapter" data-level="5.7" data-path="pre-processing.html"><a href="pre-processing.html#imputation"><i class="fa fa-check"></i><b>5.7</b> Imputation</a></li>
<li class="chapter" data-level="5.8" data-path="pre-processing.html"><a href="pre-processing.html#extracting-specific-taxa-phyloseq-class-object"><i class="fa fa-check"></i><b>5.8</b> Extracting specific taxa phyloseq-class object</a></li>
<li class="chapter" data-level="5.9" data-path="pre-processing.html"><a href="pre-processing.html#aggregating-low-relative-abundance-or-unclassified-taxa-into-others"><i class="fa fa-check"></i><b>5.9</b> Aggregating low relative abundance or unclassified taxa into others</a></li>
<li class="chapter" data-level="5.10" data-path="pre-processing.html"><a href="pre-processing.html#transform-the-abundance-of-taxa-whose-relative-abundance-under-limit-of-detection-lod-into-zeros-only-in-metaphlan23"><i class="fa fa-check"></i><b>5.10</b> Transform the abundance of taxa whose relative abundance under Limit Of Detection (LOD) into Zeros (only in metaphlan2/3)</a></li>
<li class="chapter" data-level="5.11" data-path="pre-processing.html"><a href="pre-processing.html#systematic-information-2"><i class="fa fa-check"></i><b>5.11</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="aplha-diversity.html"><a href="aplha-diversity.html"><i class="fa fa-check"></i><b>6</b> Aplha diversity</a>
<ul>
<li class="chapter" data-level="6.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#loading-packages-3"><i class="fa fa-check"></i><b>6.1</b> Loading Packages</a></li>
<li class="chapter" data-level="6.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#importing-data-3"><i class="fa fa-check"></i><b>6.2</b> Importing Data</a></li>
<li class="chapter" data-level="6.3" data-path="aplha-diversity.html"><a href="aplha-diversity.html#equal-sample-sums"><i class="fa fa-check"></i><b>6.3</b> Equal sample sums</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#dada2_ps-dataset"><i class="fa fa-check"></i><b>6.3.1</b> dada2_ps dataset</a></li>
<li class="chapter" data-level="6.3.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#amplicon_ps-dataset"><i class="fa fa-check"></i><b>6.3.2</b> amplicon_ps dataset</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="aplha-diversity.html"><a href="aplha-diversity.html#non-phylogenetic-diversities"><i class="fa fa-check"></i><b>6.4</b> Non-phylogenetic diversities</a>
<ul>
<li class="chapter" data-level="6.4.1" data-path="aplha-diversity.html"><a href="aplha-diversity.html#xmas2-r-package"><i class="fa fa-check"></i><b>6.4.1</b> XMAS2 R package</a></li>
<li class="chapter" data-level="6.4.2" data-path="aplha-diversity.html"><a href="aplha-diversity.html#microbiome-r-package"><i class="fa fa-check"></i><b>6.4.2</b> microbiome R package</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="aplha-diversity.html"><a href="aplha-diversity.html#phylogenetic-diversity"><i class="fa fa-check"></i><b>6.5</b> Phylogenetic diversity</a></li>
<li class="chapter" data-level="6.6" data-path="aplha-diversity.html"><a href="aplha-diversity.html#rank-abundance"><i class="fa fa-check"></i><b>6.6</b> Rank Abundance</a></li>
<li class="chapter" data-level="6.7" data-path="aplha-diversity.html"><a href="aplha-diversity.html#evenness"><i class="fa fa-check"></i><b>6.7</b> Evenness</a></li>
<li class="chapter" data-level="6.8" data-path="aplha-diversity.html"><a href="aplha-diversity.html#systematic-information-3"><i class="fa fa-check"></i><b>6.8</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="beta-diversity.html"><a href="beta-diversity.html"><i class="fa fa-check"></i><b>7</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="beta-diversity.html"><a href="beta-diversity.html#loading-packages-4"><i class="fa fa-check"></i><b>7.1</b> Loading Packages</a></li>
<li class="chapter" data-level="7.2" data-path="beta-diversity.html"><a href="beta-diversity.html#importing-data-4"><i class="fa fa-check"></i><b>7.2</b> Importing Data</a></li>
<li class="chapter" data-level="7.3" data-path="beta-diversity.html"><a href="beta-diversity.html#phylogenetic-beta-diversity-metrics"><i class="fa fa-check"></i><b>7.3</b> Phylogenetic beta-diversity metrics</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="beta-diversity.html"><a href="beta-diversity.html#unweighted-unifrac"><i class="fa fa-check"></i><b>7.3.1</b> Unweighted Unifrac</a></li>
<li class="chapter" data-level="7.3.2" data-path="beta-diversity.html"><a href="beta-diversity.html#weighted-unifrac"><i class="fa fa-check"></i><b>7.3.2</b> Weighted Unifrac</a></li>
<li class="chapter" data-level="7.3.3" data-path="beta-diversity.html"><a href="beta-diversity.html#generalized-unifrac-distance-metrics"><i class="fa fa-check"></i><b>7.3.3</b> Generalized UniFrac Distance Metrics</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="beta-diversity.html"><a href="beta-diversity.html#distance-dissimilarity-coefficients-bray-curtis-index"><i class="fa fa-check"></i><b>7.4</b> Distance (Dissimilarity) Coefficients: Bray-Curtis Index</a></li>
<li class="chapter" data-level="7.5" data-path="beta-diversity.html"><a href="beta-diversity.html#checking-the-homogeneity-condition"><i class="fa fa-check"></i><b>7.5</b> Checking the homogeneity condition</a></li>
<li class="chapter" data-level="7.6" data-path="beta-diversity.html"><a href="beta-diversity.html#permutational-multivariate-analysis-of-variance-permanova"><i class="fa fa-check"></i><b>7.6</b> Permutational Multivariate Analysis of Variance (PERMANOVA)</a></li>
<li class="chapter" data-level="7.7" data-path="beta-diversity.html"><a href="beta-diversity.html#mantel-test"><i class="fa fa-check"></i><b>7.7</b> Mantel Test</a></li>
<li class="chapter" data-level="7.8" data-path="beta-diversity.html"><a href="beta-diversity.html#analysis-of-similarity-anosim"><i class="fa fa-check"></i><b>7.8</b> Analysis of Similarity (ANOSIM)</a></li>
<li class="chapter" data-level="7.9" data-path="beta-diversity.html"><a href="beta-diversity.html#multi-response-permutation-procedures-mrpp"><i class="fa fa-check"></i><b>7.9</b> Multi-response permutation procedures (MRPP)</a></li>
<li class="chapter" data-level="7.10" data-path="beta-diversity.html"><a href="beta-diversity.html#ordination"><i class="fa fa-check"></i><b>7.10</b> Ordination</a>
<ul>
<li class="chapter" data-level="7.10.1" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>7.10.1</b> Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="7.10.2" data-path="beta-diversity.html"><a href="beta-diversity.html#principal-coordinate-analysis-pcoa"><i class="fa fa-check"></i><b>7.10.2</b> Principal Coordinate Analysis (PCoA)</a></li>
<li class="chapter" data-level="7.10.3" data-path="beta-diversity.html"><a href="beta-diversity.html#t-distributed-stochastic-neighbor-embedding-t-sne"><i class="fa fa-check"></i><b>7.10.3</b> t-distributed stochastic neighbor embedding (t-SNE)</a></li>
<li class="chapter" data-level="7.10.4" data-path="beta-diversity.html"><a href="beta-diversity.html#uniform-manifold-approximation-and-projection-for-dimension-reduction-umap"><i class="fa fa-check"></i><b>7.10.4</b> Uniform Manifold Approximation and Projection for Dimension Reduction (UMAP)</a></li>
</ul></li>
<li class="chapter" data-level="7.11" data-path="beta-diversity.html"><a href="beta-diversity.html#systematic-information-4"><i class="fa fa-check"></i><b>7.11</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="microbial-composition.html"><a href="microbial-composition.html"><i class="fa fa-check"></i><b>8</b> Microbial Composition</a>
<ul>
<li class="chapter" data-level="8.1" data-path="microbial-composition.html"><a href="microbial-composition.html#loading-packages-5"><i class="fa fa-check"></i><b>8.1</b> Loading Packages</a></li>
<li class="chapter" data-level="8.2" data-path="microbial-composition.html"><a href="microbial-composition.html#importing-data-5"><i class="fa fa-check"></i><b>8.2</b> Importing Data</a></li>
<li class="chapter" data-level="8.3" data-path="microbial-composition.html"><a href="microbial-composition.html#plot_stacked_bar-from-xviz"><i class="fa fa-check"></i><b>8.3</b> plot_stacked_bar from XVIZ</a></li>
<li class="chapter" data-level="8.4" data-path="microbial-composition.html"><a href="microbial-composition.html#plot_stackbarplot-from-xams2"><i class="fa fa-check"></i><b>8.4</b> plot_StackBarPlot from XAMS2</a></li>
<li class="chapter" data-level="8.5" data-path="microbial-composition.html"><a href="microbial-composition.html#heatmaps"><i class="fa fa-check"></i><b>8.5</b> Heatmaps</a></li>
<li class="chapter" data-level="8.6" data-path="microbial-composition.html"><a href="microbial-composition.html#systematic-information-5"><i class="fa fa-check"></i><b>8.6</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="core-microbiota.html"><a href="core-microbiota.html"><i class="fa fa-check"></i><b>9</b> Core microbiota</a>
<ul>
<li class="chapter" data-level="9.1" data-path="core-microbiota.html"><a href="core-microbiota.html#loading-packages-6"><i class="fa fa-check"></i><b>9.1</b> Loading Packages</a></li>
<li class="chapter" data-level="9.2" data-path="core-microbiota.html"><a href="core-microbiota.html#importing-data-6"><i class="fa fa-check"></i><b>9.2</b> Importing Data</a></li>
<li class="chapter" data-level="9.3" data-path="core-microbiota.html"><a href="core-microbiota.html#obtaining-core-microbiota"><i class="fa fa-check"></i><b>9.3</b> Obtaining core microbiota</a></li>
<li class="chapter" data-level="9.4" data-path="core-microbiota.html"><a href="core-microbiota.html#showing-core-abundance-and-diversity"><i class="fa fa-check"></i><b>9.4</b> Showing core abundance and diversity</a></li>
<li class="chapter" data-level="9.5" data-path="core-microbiota.html"><a href="core-microbiota.html#visualizing-core-taxa"><i class="fa fa-check"></i><b>9.5</b> Visualizing core taxa</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="core-microbiota.html"><a href="core-microbiota.html#heatmap"><i class="fa fa-check"></i><b>9.5.1</b> heatmap</a></li>
<li class="chapter" data-level="9.5.2" data-path="core-microbiota.html"><a href="core-microbiota.html#linechart"><i class="fa fa-check"></i><b>9.5.2</b> linechart</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="core-microbiota.html"><a href="core-microbiota.html#systematic-information-6"><i class="fa fa-check"></i><b>9.6</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="differential-analysis.html"><a href="differential-analysis.html"><i class="fa fa-check"></i><b>10</b> Differential Analysis</a>
<ul>
<li class="chapter" data-level="10.1" data-path="differential-analysis.html"><a href="differential-analysis.html#loading-packages-7"><i class="fa fa-check"></i><b>10.1</b> Loading Packages</a></li>
<li class="chapter" data-level="10.2" data-path="differential-analysis.html"><a href="differential-analysis.html#importing-data-7"><i class="fa fa-check"></i><b>10.2</b> Importing Data</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="differential-analysis.html"><a href="differential-analysis.html#s-data-1"><i class="fa fa-check"></i><b>10.2.1</b> 16s data</a></li>
<li class="chapter" data-level="10.2.2" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomic-data-1"><i class="fa fa-check"></i><b>10.2.2</b> Metagenomic data</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="differential-analysis.html"><a href="differential-analysis.html#amplicon-sequencing-microbial-data-16s"><i class="fa fa-check"></i><b>10.3</b> Amplicon sequencing microbial data (16s)</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="differential-analysis.html"><a href="differential-analysis.html#aldex2"><i class="fa fa-check"></i><b>10.3.1</b> ALDEx2</a></li>
<li class="chapter" data-level="10.3.2" data-path="differential-analysis.html"><a href="differential-analysis.html#limma_voom"><i class="fa fa-check"></i><b>10.3.2</b> limma_voom</a></li>
<li class="chapter" data-level="10.3.3" data-path="differential-analysis.html"><a href="differential-analysis.html#mbzinb"><i class="fa fa-check"></i><b>10.3.3</b> mbzinb</a></li>
<li class="chapter" data-level="10.3.4" data-path="differential-analysis.html"><a href="differential-analysis.html#omnibus"><i class="fa fa-check"></i><b>10.3.4</b> omnibus</a></li>
<li class="chapter" data-level="10.3.5" data-path="differential-analysis.html"><a href="differential-analysis.html#raida"><i class="fa fa-check"></i><b>10.3.5</b> RAIDA</a></li>
<li class="chapter" data-level="10.3.6" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests"><i class="fa fa-check"></i><b>10.3.6</b> Wilcoxon Rank Sum and Signed Rank Tests</a></li>
<li class="chapter" data-level="10.3.7" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse"><i class="fa fa-check"></i><b>10.3.7</b> Liner discriminant analysis (LDA) effect size (LEfSe)</a></li>
<li class="chapter" data-level="10.3.8" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test"><i class="fa fa-check"></i><b>10.3.8</b> t-test</a></li>
<li class="chapter" data-level="10.3.9" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomeseq"><i class="fa fa-check"></i><b>10.3.9</b> MetagenomeSeq</a></li>
<li class="chapter" data-level="10.3.10" data-path="differential-analysis.html"><a href="differential-analysis.html#deseq2"><i class="fa fa-check"></i><b>10.3.10</b> DESeq2</a></li>
<li class="chapter" data-level="10.3.11" data-path="differential-analysis.html"><a href="differential-analysis.html#edger"><i class="fa fa-check"></i><b>10.3.11</b> EdgeR</a></li>
<li class="chapter" data-level="10.3.12" data-path="differential-analysis.html"><a href="differential-analysis.html#ancom"><i class="fa fa-check"></i><b>10.3.12</b> ANCOM</a></li>
<li class="chapter" data-level="10.3.13" data-path="differential-analysis.html"><a href="differential-analysis.html#corncob"><i class="fa fa-check"></i><b>10.3.13</b> Corncob</a></li>
<li class="chapter" data-level="10.3.14" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models"><i class="fa fa-check"></i><b>10.3.14</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)</a></li>
<li class="chapter" data-level="10.3.15" data-path="differential-analysis.html"><a href="differential-analysis.html#locom"><i class="fa fa-check"></i><b>10.3.15</b> LOCOM</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="differential-analysis.html"><a href="differential-analysis.html#metagenomic-sequencing-microbial-data-metaphlan23"><i class="fa fa-check"></i><b>10.4</b> Metagenomic sequencing microbial data (metaphlan2/3)</a>
<ul>
<li class="chapter" data-level="10.4.1" data-path="differential-analysis.html"><a href="differential-analysis.html#wilcoxon-rank-sum-and-signed-rank-tests-1"><i class="fa fa-check"></i><b>10.4.1</b> Wilcoxon Rank Sum and Signed Rank Tests</a></li>
<li class="chapter" data-level="10.4.2" data-path="differential-analysis.html"><a href="differential-analysis.html#liner-discriminant-analysis-lda-effect-size-lefse-1"><i class="fa fa-check"></i><b>10.4.2</b> Liner discriminant analysis (LDA) effect size (LEfSe)</a></li>
<li class="chapter" data-level="10.4.3" data-path="differential-analysis.html"><a href="differential-analysis.html#t-test-1"><i class="fa fa-check"></i><b>10.4.3</b> t-test</a></li>
<li class="chapter" data-level="10.4.4" data-path="differential-analysis.html"><a href="differential-analysis.html#maaslin2-microbiome-multivariable-association-with-linear-models-1"><i class="fa fa-check"></i><b>10.4.4</b> Maaslin2 (Microbiome Multivariable Association with Linear Models)</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="differential-analysis.html"><a href="differential-analysis.html#visualization-4"><i class="fa fa-check"></i><b>10.5</b> Visualization</a>
<ul>
<li class="chapter" data-level="10.5.1" data-path="differential-analysis.html"><a href="differential-analysis.html#volcano-plot"><i class="fa fa-check"></i><b>10.5.1</b> Volcano plot</a></li>
<li class="chapter" data-level="10.5.2" data-path="differential-analysis.html"><a href="differential-analysis.html#barplot-in-lefse"><i class="fa fa-check"></i><b>10.5.2</b> barplot in lefse</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="differential-analysis.html"><a href="differential-analysis.html#dominant-taxa"><i class="fa fa-check"></i><b>10.6</b> Dominant taxa</a></li>
<li class="chapter" data-level="10.7" data-path="differential-analysis.html"><a href="differential-analysis.html#multiple-differential-analysis-by-one-function"><i class="fa fa-check"></i><b>10.7</b> Multiple differential analysis by one function</a></li>
<li class="chapter" data-level="10.8" data-path="differential-analysis.html"><a href="differential-analysis.html#systematic-information-7"><i class="fa fa-check"></i><b>10.8</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="visualization-5.html"><a href="visualization-5.html"><i class="fa fa-check"></i><b>11</b> Visualization</a>
<ul>
<li class="chapter" data-level="11.1" data-path="visualization-5.html"><a href="visualization-5.html#loading-packages-8"><i class="fa fa-check"></i><b>11.1</b> Loading Packages</a></li>
<li class="chapter" data-level="11.2" data-path="visualization-5.html"><a href="visualization-5.html#importing-data-8"><i class="fa fa-check"></i><b>11.2</b> Importing Data</a></li>
<li class="chapter" data-level="11.3" data-path="visualization-5.html"><a href="visualization-5.html#plot_boxplot"><i class="fa fa-check"></i><b>11.3</b> plot_boxplot</a></li>
<li class="chapter" data-level="11.4" data-path="visualization-5.html"><a href="visualization-5.html#plot_barplot"><i class="fa fa-check"></i><b>11.4</b> plot_barplot</a></li>
<li class="chapter" data-level="11.5" data-path="visualization-5.html"><a href="visualization-5.html#plot_dotplot"><i class="fa fa-check"></i><b>11.5</b> plot_dotplot</a></li>
<li class="chapter" data-level="11.6" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_boxplot"><i class="fa fa-check"></i><b>11.6</b> plot_correlation_boxplot</a></li>
<li class="chapter" data-level="11.7" data-path="visualization-5.html"><a href="visualization-5.html#plot_correlation_density"><i class="fa fa-check"></i><b>11.7</b> plot_correlation_density</a></li>
<li class="chapter" data-level="11.8" data-path="visualization-5.html"><a href="visualization-5.html#plot_ordination"><i class="fa fa-check"></i><b>11.8</b> plot_Ordination</a></li>
<li class="chapter" data-level="11.9" data-path="visualization-5.html"><a href="visualization-5.html#plot_ggbiplot"><i class="fa fa-check"></i><b>11.9</b> plot_ggbiplot</a></li>
<li class="chapter" data-level="11.10" data-path="visualization-5.html"><a href="visualization-5.html#plot_corrplot"><i class="fa fa-check"></i><b>11.10</b> plot_corrplot</a></li>
<li class="chapter" data-level="11.11" data-path="visualization-5.html"><a href="visualization-5.html#plot_2da_venn"><i class="fa fa-check"></i><b>11.11</b> plot_2DA_venn</a></li>
<li class="chapter" data-level="11.12" data-path="visualization-5.html"><a href="visualization-5.html#plot-the-da-results-from-the-significant-taxa-by-double-barplot"><i class="fa fa-check"></i><b>11.12</b> plot the DA results from the significant taxa by double barplot</a></li>
<li class="chapter" data-level="11.13" data-path="visualization-5.html"><a href="visualization-5.html#plot_stacked_bar_xivz"><i class="fa fa-check"></i><b>11.13</b> plot_stacked_bar_XIVZ</a></li>
<li class="chapter" data-level="11.14" data-path="visualization-5.html"><a href="visualization-5.html#plot_stackbarplot"><i class="fa fa-check"></i><b>11.14</b> plot_StackBarPlot</a></li>
<li class="chapter" data-level="11.15" data-path="visualization-5.html"><a href="visualization-5.html#color-palettes"><i class="fa fa-check"></i><b>11.15</b> Color Palettes</a>
<ul>
<li class="chapter" data-level="11.15.1" data-path="visualization-5.html"><a href="visualization-5.html#wes-anderson-palettes"><i class="fa fa-check"></i><b>11.15.1</b> Wes Anderson Palettes</a></li>
<li class="chapter" data-level="11.15.2" data-path="visualization-5.html"><a href="visualization-5.html#usemycol"><i class="fa fa-check"></i><b>11.15.2</b> useMyCol</a></li>
</ul></li>
<li class="chapter" data-level="11.16" data-path="visualization-5.html"><a href="visualization-5.html#ordination-plots-with-ggplot2"><i class="fa fa-check"></i><b>11.16</b> Ordination plots with ggplot2</a>
<ul>
<li class="chapter" data-level="11.16.1" data-path="visualization-5.html"><a href="visualization-5.html#principal-components-analysis-with-the-iris-data-set"><i class="fa fa-check"></i><b>11.16.1</b> principal components analysis with the iris data set</a></li>
<li class="chapter" data-level="11.16.2" data-path="visualization-5.html"><a href="visualization-5.html#multiple-correspondence-analysis-with-the-tea-dataset"><i class="fa fa-check"></i><b>11.16.2</b> multiple correspondence analysis with the tea dataset</a></li>
<li class="chapter" data-level="11.16.3" data-path="visualization-5.html"><a href="visualization-5.html#linear-discriminant-analysis"><i class="fa fa-check"></i><b>11.16.3</b> linear discriminant analysis</a></li>
</ul></li>
<li class="chapter" data-level="11.17" data-path="visualization-5.html"><a href="visualization-5.html#systematic-information-8"><i class="fa fa-check"></i><b>11.17</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="association-analysis.html"><a href="association-analysis.html"><i class="fa fa-check"></i><b>12</b> Association Analysis</a>
<ul>
<li class="chapter" data-level="12.1" data-path="association-analysis.html"><a href="association-analysis.html#loading-packages-9"><i class="fa fa-check"></i><b>12.1</b> Loading Packages</a></li>
<li class="chapter" data-level="12.2" data-path="association-analysis.html"><a href="association-analysis.html#importing-data-9"><i class="fa fa-check"></i><b>12.2</b> Importing Data</a></li>
<li class="chapter" data-level="12.3" data-path="association-analysis.html"><a href="association-analysis.html#spearman-correlation-analysis"><i class="fa fa-check"></i><b>12.3</b> Spearman Correlation Analysis</a></li>
<li class="chapter" data-level="12.4" data-path="association-analysis.html"><a href="association-analysis.html#partial-correlation-analysis"><i class="fa fa-check"></i><b>12.4</b> Partial Correlation Analysis</a></li>
<li class="chapter" data-level="12.5" data-path="association-analysis.html"><a href="association-analysis.html#systematic-information-9"><i class="fa fa-check"></i><b>12.5</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html"><i class="fa fa-check"></i><b>13</b> Principals of Differential Analysis methods</a>
<ul>
<li class="chapter" data-level="13.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#aldex2-1"><i class="fa fa-check"></i><b>13.1</b> ALDEx2</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-1"><i class="fa fa-check"></i><b>13.1.1</b> Vignette</a></li>
<li class="chapter" data-level="13.1.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type"><i class="fa fa-check"></i><b>13.1.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.1.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments"><i class="fa fa-check"></i><b>13.1.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.1.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures"><i class="fa fa-check"></i><b>13.1.4</b> Procedures</a></li>
<li class="chapter" data-level="13.1.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output"><i class="fa fa-check"></i><b>13.1.5</b> Key output</a></li>
<li class="chapter" data-level="13.1.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax"><i class="fa fa-check"></i><b>13.1.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.1.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax"><i class="fa fa-check"></i><b>13.1.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="13.1.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes"><i class="fa fa-check"></i><b>13.1.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#ancom-ii"><i class="fa fa-check"></i><b>13.2</b> ANCOM-II</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-1"><i class="fa fa-check"></i><b>13.2.1</b> Input Data Type</a></li>
<li class="chapter" data-level="13.2.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-1"><i class="fa fa-check"></i><b>13.2.2</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.2.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-1"><i class="fa fa-check"></i><b>13.2.3</b> Procedures</a></li>
<li class="chapter" data-level="13.2.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-1"><i class="fa fa-check"></i><b>13.2.4</b> Key output</a></li>
<li class="chapter" data-level="13.2.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-1"><i class="fa fa-check"></i><b>13.2.5</b> Original Syntax</a></li>
<li class="chapter" data-level="13.2.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-1"><i class="fa fa-check"></i><b>13.2.6</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="13.2.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-1"><i class="fa fa-check"></i><b>13.2.7</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#corncob-1"><i class="fa fa-check"></i><b>13.3</b> corncob</a>
<ul>
<li class="chapter" data-level="13.3.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-2"><i class="fa fa-check"></i><b>13.3.1</b> Vignette</a></li>
<li class="chapter" data-level="13.3.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-2"><i class="fa fa-check"></i><b>13.3.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.3.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-2"><i class="fa fa-check"></i><b>13.3.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.3.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-2"><i class="fa fa-check"></i><b>13.3.4</b> Procedures</a></li>
<li class="chapter" data-level="13.3.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-2"><i class="fa fa-check"></i><b>13.3.5</b> Key output</a></li>
<li class="chapter" data-level="13.3.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-2"><i class="fa fa-check"></i><b>13.3.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.3.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-2"><i class="fa fa-check"></i><b>13.3.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="13.3.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-2"><i class="fa fa-check"></i><b>13.3.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#deseq2-1"><i class="fa fa-check"></i><b>13.4</b> DESeq2</a>
<ul>
<li class="chapter" data-level="13.4.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-3"><i class="fa fa-check"></i><b>13.4.1</b> Vignette</a></li>
<li class="chapter" data-level="13.4.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-3"><i class="fa fa-check"></i><b>13.4.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.4.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-3"><i class="fa fa-check"></i><b>13.4.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.4.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-3"><i class="fa fa-check"></i><b>13.4.4</b> Procedures</a></li>
<li class="chapter" data-level="13.4.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-3"><i class="fa fa-check"></i><b>13.4.5</b> Key output</a></li>
<li class="chapter" data-level="13.4.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-3"><i class="fa fa-check"></i><b>13.4.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.4.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-3"><i class="fa fa-check"></i><b>13.4.7</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="13.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#edger-1"><i class="fa fa-check"></i><b>13.5</b> edgeR</a>
<ul>
<li class="chapter" data-level="13.5.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-4"><i class="fa fa-check"></i><b>13.5.1</b> Vignette</a></li>
<li class="chapter" data-level="13.5.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-4"><i class="fa fa-check"></i><b>13.5.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.5.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-4"><i class="fa fa-check"></i><b>13.5.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.5.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-4"><i class="fa fa-check"></i><b>13.5.4</b> Procedures</a></li>
<li class="chapter" data-level="13.5.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-4"><i class="fa fa-check"></i><b>13.5.5</b> Key output</a></li>
<li class="chapter" data-level="13.5.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-4"><i class="fa fa-check"></i><b>13.5.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.5.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-4"><i class="fa fa-check"></i><b>13.5.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="13.5.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-3"><i class="fa fa-check"></i><b>13.5.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="13.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#lefse"><i class="fa fa-check"></i><b>13.6</b> LEfSe</a>
<ul>
<li class="chapter" data-level="13.6.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-5"><i class="fa fa-check"></i><b>13.6.1</b> Vignette</a></li>
<li class="chapter" data-level="13.6.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-5"><i class="fa fa-check"></i><b>13.6.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.6.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-5"><i class="fa fa-check"></i><b>13.6.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.6.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-5"><i class="fa fa-check"></i><b>13.6.4</b> Procedures</a></li>
<li class="chapter" data-level="13.6.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-5"><i class="fa fa-check"></i><b>13.6.5</b> Key output</a></li>
<li class="chapter" data-level="13.6.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-5"><i class="fa fa-check"></i><b>13.6.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.6.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-5"><i class="fa fa-check"></i><b>13.6.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="13.6.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-4"><i class="fa fa-check"></i><b>13.6.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="13.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#limma-voom"><i class="fa fa-check"></i><b>13.7</b> limma voom</a>
<ul>
<li class="chapter" data-level="13.7.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-6"><i class="fa fa-check"></i><b>13.7.1</b> Vignette</a></li>
<li class="chapter" data-level="13.7.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-6"><i class="fa fa-check"></i><b>13.7.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.7.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-6"><i class="fa fa-check"></i><b>13.7.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.7.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-6"><i class="fa fa-check"></i><b>13.7.4</b> Procedures</a></li>
<li class="chapter" data-level="13.7.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-6"><i class="fa fa-check"></i><b>13.7.5</b> Key output</a></li>
<li class="chapter" data-level="13.7.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-6"><i class="fa fa-check"></i><b>13.7.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.7.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-6"><i class="fa fa-check"></i><b>13.7.7</b> XMAS2 Syntax</a></li>
<li class="chapter" data-level="13.7.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#notes-5"><i class="fa fa-check"></i><b>13.7.8</b> Notes</a></li>
</ul></li>
<li class="chapter" data-level="13.8" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#maaslin2"><i class="fa fa-check"></i><b>13.8</b> MaAsLin2</a>
<ul>
<li class="chapter" data-level="13.8.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-7"><i class="fa fa-check"></i><b>13.8.1</b> Vignette</a></li>
<li class="chapter" data-level="13.8.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-7"><i class="fa fa-check"></i><b>13.8.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.8.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-7"><i class="fa fa-check"></i><b>13.8.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.8.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-7"><i class="fa fa-check"></i><b>13.8.4</b> Procedures</a></li>
<li class="chapter" data-level="13.8.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-7"><i class="fa fa-check"></i><b>13.8.5</b> Key output</a></li>
<li class="chapter" data-level="13.8.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-7"><i class="fa fa-check"></i><b>13.8.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.8.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-7"><i class="fa fa-check"></i><b>13.8.7</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="13.9" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#metagenomeseq-1"><i class="fa fa-check"></i><b>13.9</b> metagenomeSeq</a>
<ul>
<li class="chapter" data-level="13.9.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-8"><i class="fa fa-check"></i><b>13.9.1</b> Vignette</a></li>
<li class="chapter" data-level="13.9.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-8"><i class="fa fa-check"></i><b>13.9.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.9.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-8"><i class="fa fa-check"></i><b>13.9.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.9.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-8"><i class="fa fa-check"></i><b>13.9.4</b> Procedures</a></li>
<li class="chapter" data-level="13.9.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-8"><i class="fa fa-check"></i><b>13.9.5</b> Key output</a></li>
<li class="chapter" data-level="13.9.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-8"><i class="fa fa-check"></i><b>13.9.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.9.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-8"><i class="fa fa-check"></i><b>13.9.7</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="13.10" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#t-test-2"><i class="fa fa-check"></i><b>13.10</b> t-test</a>
<ul>
<li class="chapter" data-level="13.10.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-9"><i class="fa fa-check"></i><b>13.10.1</b> Input Data Type</a></li>
<li class="chapter" data-level="13.10.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-9"><i class="fa fa-check"></i><b>13.10.2</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.10.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-9"><i class="fa fa-check"></i><b>13.10.3</b> Procedures</a></li>
<li class="chapter" data-level="13.10.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-9"><i class="fa fa-check"></i><b>13.10.4</b> Key output</a></li>
<li class="chapter" data-level="13.10.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-9"><i class="fa fa-check"></i><b>13.10.5</b> Original Syntax</a></li>
<li class="chapter" data-level="13.10.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-9"><i class="fa fa-check"></i><b>13.10.6</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="13.11" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#wilcoxon-test"><i class="fa fa-check"></i><b>13.11</b> Wilcoxon test</a>
<ul>
<li class="chapter" data-level="13.11.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-10"><i class="fa fa-check"></i><b>13.11.1</b> Input Data Type</a></li>
<li class="chapter" data-level="13.11.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-10"><i class="fa fa-check"></i><b>13.11.2</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.11.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-10"><i class="fa fa-check"></i><b>13.11.3</b> Procedures</a></li>
<li class="chapter" data-level="13.11.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-10"><i class="fa fa-check"></i><b>13.11.4</b> Key output</a></li>
<li class="chapter" data-level="13.11.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-10"><i class="fa fa-check"></i><b>13.11.5</b> Original Syntax</a></li>
<li class="chapter" data-level="13.11.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-10"><i class="fa fa-check"></i><b>13.11.6</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="13.12" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#raida-1"><i class="fa fa-check"></i><b>13.12</b> RAIDA</a>
<ul>
<li class="chapter" data-level="13.12.1" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#vignette-9"><i class="fa fa-check"></i><b>13.12.1</b> Vignette</a></li>
<li class="chapter" data-level="13.12.2" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-data-type-11"><i class="fa fa-check"></i><b>13.12.2</b> Input Data Type</a></li>
<li class="chapter" data-level="13.12.3" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#input-key-arguments-11"><i class="fa fa-check"></i><b>13.12.3</b> Input Key Arguments</a></li>
<li class="chapter" data-level="13.12.4" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#procedures-11"><i class="fa fa-check"></i><b>13.12.4</b> Procedures</a></li>
<li class="chapter" data-level="13.12.5" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#key-output-11"><i class="fa fa-check"></i><b>13.12.5</b> Key output</a></li>
<li class="chapter" data-level="13.12.6" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#original-syntax-11"><i class="fa fa-check"></i><b>13.12.6</b> Original Syntax</a></li>
<li class="chapter" data-level="13.12.7" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#xmas2-syntax-11"><i class="fa fa-check"></i><b>13.12.7</b> XMAS2 Syntax</a></li>
</ul></li>
<li class="chapter" data-level="13.13" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#summary"><i class="fa fa-check"></i><b>13.13</b> Summary</a></li>
<li class="chapter" data-level="13.14" data-path="principals-of-differential-analysis-methods.html"><a href="principals-of-differential-analysis-methods.html#advantages-and-disadvantages"><i class="fa fa-check"></i><b>13.14</b> Advantages and disadvantages</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="examples.html"><a href="examples.html"><i class="fa fa-check"></i><b>14</b> Examples</a>
<ul>
<li class="chapter" data-level="14.1" data-path="examples.html"><a href="examples.html#loading-packages-10"><i class="fa fa-check"></i><b>14.1</b> Loading Packages</a></li>
<li class="chapter" data-level="14.2" data-path="examples.html"><a href="examples.html#workflow-description"><i class="fa fa-check"></i><b>14.2</b> Workflow description</a></li>
<li class="chapter" data-level="14.3" data-path="examples.html"><a href="examples.html#amplicon-sequencing-16s"><i class="fa fa-check"></i><b>14.3</b> Amplicon sequencing (16s)</a>
<ul>
<li class="chapter" data-level="14.3.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object"><i class="fa fa-check"></i><b>14.3.1</b> Converting inputs into phyloseq-class object</a></li>
<li class="chapter" data-level="14.3.2" data-path="examples.html"><a href="examples.html#quality-control"><i class="fa fa-check"></i><b>14.3.2</b> Quality Control</a></li>
<li class="chapter" data-level="14.3.3" data-path="examples.html"><a href="examples.html#data-processing"><i class="fa fa-check"></i><b>14.3.3</b> Data processing</a></li>
<li class="chapter" data-level="14.3.4" data-path="examples.html"><a href="examples.html#diversity-analysis"><i class="fa fa-check"></i><b>14.3.4</b> Diversity analysis</a></li>
<li class="chapter" data-level="14.3.5" data-path="examples.html"><a href="examples.html#permanova-ordination"><i class="fa fa-check"></i><b>14.3.5</b> PERMANOVA + Ordination</a></li>
<li class="chapter" data-level="14.3.6" data-path="examples.html"><a href="examples.html#microbial-composition-1"><i class="fa fa-check"></i><b>14.3.6</b> Microbial composition</a></li>
<li class="chapter" data-level="14.3.7" data-path="examples.html"><a href="examples.html#differential-analysis-1"><i class="fa fa-check"></i><b>14.3.7</b> Differential Analysis</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="examples.html"><a href="examples.html#metagenomics-mgs"><i class="fa fa-check"></i><b>14.4</b> Metagenomics (MGS)</a>
<ul>
<li class="chapter" data-level="14.4.1" data-path="examples.html"><a href="examples.html#converting-inputs-into-phyloseq-class-object-1"><i class="fa fa-check"></i><b>14.4.1</b> Converting inputs into phyloseq-class object</a></li>
<li class="chapter" data-level="14.4.2" data-path="examples.html"><a href="examples.html#quality-control-1"><i class="fa fa-check"></i><b>14.4.2</b> Quality Control</a></li>
<li class="chapter" data-level="14.4.3" data-path="examples.html"><a href="examples.html#data-processing-1"><i class="fa fa-check"></i><b>14.4.3</b> Data processing</a></li>
<li class="chapter" data-level="14.4.4" data-path="examples.html"><a href="examples.html#diversity-analysis-1"><i class="fa fa-check"></i><b>14.4.4</b> Diversity analysis</a></li>
<li class="chapter" data-level="14.4.5" data-path="examples.html"><a href="examples.html#permanova-ordination-1"><i class="fa fa-check"></i><b>14.4.5</b> PERMANOVA + Ordination</a></li>
<li class="chapter" data-level="14.4.6" data-path="examples.html"><a href="examples.html#microbial-composition-2"><i class="fa fa-check"></i><b>14.4.6</b> Microbial composition</a></li>
<li class="chapter" data-level="14.4.7" data-path="examples.html"><a href="examples.html#differential-analysis-2"><i class="fa fa-check"></i><b>14.4.7</b> Differential Analysis</a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="examples.html"><a href="examples.html#systematic-information-10"><i class="fa fa-check"></i><b>14.5</b> Systematic Information</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">XMAS 2.0 Tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="visualization-5" class="section level1 hasAnchor" number="11">
<h1><span class="header-section-number">Chapter 11</span> Visualization<a href="visualization-5.html#visualization-5" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><strong>XMAS 2.0</strong> provides mulitple functions for visualization. For instance, using <code>plot_volcano</code> to display the results of differential analysis.</p>
<p>Outline of this Chapter:</p>
<ul>
<li><p><a href="convert-inputdata-into-phyloseq-object.html#loading-packages">Loading Packages</a></p></li>
<li><p><a href="convert-inputdata-into-phyloseq-object.html#importing-data">Importing Data</a></p></li>
<li><p><a href="visualization-5.html#plot_boxplot">plot_boxplot</a></p></li>
<li><p><a href="visualization-5.html#plot_barplot">plot_barplot</a></p></li>
<li><p><a href="visualization-5.html#plot_dotplot">plot_dotplot</a></p></li>
<li><p><a href="visualization-5.html#plot_correlation_boxplot">plot_correlation_boxplot</a></p></li>
<li><p><a href="visualization-5.html#plot_correlation_density">plot_correlation_density</a></p></li>
<li><p><a href="visualization-5.html#plot_ordination">plot_Ordination</a></p></li>
<li><p><a href="visualization-5.html#plot_ggbiplot">plot_ggbiplot</a></p></li>
<li><p><a href="visualization-5.html#plot_corrplot">plot_corrplot</a></p></li>
<li><p><a href="visualization-5.html#plot_2da_venn">plot_2DA_venn</a></p></li>
<li><p><a href="visualization-5.html#plot-the-da-results-from-the-significant-taxa-by-double-barplot">plot the DA results from the significant taxa by double barplot</a></p></li>
<li><p><a href="visualization-5.html#plot_stacked_bar_xivz">plot_stacked_bar_XIVZ</a></p></li>
<li><p><a href="visualization-5.html#plot_stackbarplot">plot_StackBarPlot</a></p></li>
<li><p><a href="visualization-5.html#color-palettes">Color Palettes</a></p></li>
<li><p><a href="visualization-5.html#ordination-plots-with-ggplot2">Ordination plots with ggplot2</a></p></li>
</ul>
<div id="loading-packages-8" class="section level2 hasAnchor" number="11.1">
<h2><span class="header-section-number">11.1</span> Loading Packages<a href="visualization-5.html#loading-packages-8" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb382"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb382-1"><a href="visualization-5.html#cb382-1" tabindex="-1"></a><span class="fu">library</span>(XMAS2)</span>
<span id="cb382-2"><a href="visualization-5.html#cb382-2" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb382-3"><a href="visualization-5.html#cb382-3" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb382-4"><a href="visualization-5.html#cb382-4" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb382-5"><a href="visualization-5.html#cb382-5" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb382-6"><a href="visualization-5.html#cb382-6" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span></code></pre></div>
</div>
<div id="importing-data-8" class="section level2 hasAnchor" number="11.2">
<h2><span class="header-section-number">11.2</span> Importing Data<a href="visualization-5.html#importing-data-8" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb383"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb383-1"><a href="visualization-5.html#cb383-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;amplicon_ps&quot;</span>)</span>
<span id="cb383-2"><a href="visualization-5.html#cb383-2" tabindex="-1"></a>amplicon_ps_rarefy <span class="ot">&lt;-</span> <span class="fu">norm_rarefy</span>(<span class="at">object =</span> amplicon_ps, </span>
<span id="cb383-3"><a href="visualization-5.html#cb383-3" tabindex="-1"></a>                                  <span class="at">size =</span> <span class="dv">1114</span>)</span>
<span id="cb383-4"><a href="visualization-5.html#cb383-4" tabindex="-1"></a>amplicon_ps_rarefy</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 2308 taxa and 34 samples ]
## sample_data() Sample Data:       [ 34 samples by 8 sample variables ]
## tax_table()   Taxonomy Table:    [ 2308 taxa by 7 taxonomic ranks ]
## phy_tree()    Phylogenetic Tree: [ 2308 tips and 2306 internal nodes ]</code></pre>
</div>
<div id="plot_boxplot" class="section level2 hasAnchor" number="11.3">
<h2><span class="header-section-number">11.3</span> plot_boxplot<a href="visualization-5.html#plot_boxplot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>calculate alpha diversity</li>
</ul>
<div class="sourceCode" id="cb385"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb385-1"><a href="visualization-5.html#cb385-1" tabindex="-1"></a>dat_alpha <span class="ot">&lt;-</span> <span class="fu">run_alpha_diversity</span>(<span class="at">ps =</span> amplicon_ps_rarefy, </span>
<span id="cb385-2"><a href="visualization-5.html#cb385-2" tabindex="-1"></a>                                 <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Chao1&quot;</span>, <span class="st">&quot;Observed&quot;</span>))</span>
<span id="cb385-3"><a href="visualization-5.html#cb385-3" tabindex="-1"></a><span class="fu">head</span>(dat_alpha)</span></code></pre></div>
<pre><code>##   TempRowNames SampleType Year Month Day Subject ReportedAntibioticUsage DaysSinceExperimentStart         Description Observed  Chao1
## 1       L1S140        gut 2008    10  28       2                     Yes                        0   2_Fece_10_28_2008       27  74.50
## 2       L1S208        gut 2009     1  20       2                      No                       84    2_Fece_1_20_2009       40 148.75
## 3         L1S8        gut 2008    10  28       1                     Yes                        0   1_Fece_10_28_2008       19  54.00
## 4       L1S281        gut 2009     4  14       2                      No                      168    2_Fece_4_14_2009       60 256.00
## 5       L3S242 right palm 2008    10  28       1                     Yes                        0 1_R_Palm_10_28_2008       16  42.00
## 6       L2S309  left palm 2009     1  20       2                      No                       84  2_L_Palm_1_20_2009       12  57.00
##   se.chao1  Shannon
## 1 30.70970 3.126005
## 2 62.86107 3.303186
## 3 25.57190 2.688337
## 4 93.42352 3.664947
## 5 19.97805 2.692311
## 6 30.06061 2.082828</code></pre>
<p><code>plot_boxplot</code> has many parameters, and help you enjoy it.</p>
<ul>
<li>single measure</li>
</ul>
<div class="sourceCode" id="cb387"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb387-1"><a href="visualization-5.html#cb387-1" tabindex="-1"></a><span class="fu">plot_boxplot</span>(<span class="at">data =</span> dat_alpha,</span>
<span id="cb387-2"><a href="visualization-5.html#cb387-2" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="st">&quot;Shannon&quot;</span>,</span>
<span id="cb387-3"><a href="visualization-5.html#cb387-3" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb387-4"><a href="visualization-5.html#cb387-4" tabindex="-1"></a>             <span class="at">do_test =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-198"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-198-1.png" alt="boxplot(single measure)" width="576" />
<p class="caption">
Figure 11.1: boxplot(single measure)
</p>
</div>
<ul>
<li>single measure with significant results</li>
</ul>
<div class="sourceCode" id="cb388"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb388-1"><a href="visualization-5.html#cb388-1" tabindex="-1"></a><span class="fu">plot_boxplot</span>(<span class="at">data =</span> dat_alpha,</span>
<span id="cb388-2"><a href="visualization-5.html#cb388-2" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="st">&quot;Shannon&quot;</span>,</span>
<span id="cb388-3"><a href="visualization-5.html#cb388-3" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb388-4"><a href="visualization-5.html#cb388-4" tabindex="-1"></a>             <span class="at">do_test =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-199"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-199-1.png" alt="boxplot(single measure with significant results)" width="576" />
<p class="caption">
Figure 11.2: boxplot(single measure with significant results)
</p>
</div>
<ul>
<li>single measure with significant results of pairwises</li>
</ul>
<div class="sourceCode" id="cb389"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb389-1"><a href="visualization-5.html#cb389-1" tabindex="-1"></a><span class="fu">plot_boxplot</span>(<span class="at">data =</span> dat_alpha,</span>
<span id="cb389-2"><a href="visualization-5.html#cb389-2" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="st">&quot;Shannon&quot;</span>,</span>
<span id="cb389-3"><a href="visualization-5.html#cb389-3" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb389-4"><a href="visualization-5.html#cb389-4" tabindex="-1"></a>             <span class="at">do_test =</span> <span class="cn">TRUE</span>,</span>
<span id="cb389-5"><a href="visualization-5.html#cb389-5" tabindex="-1"></a>             <span class="at">cmp_list =</span> <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">&quot;gut&quot;</span>, <span class="st">&quot;right palm&quot;</span>), </span>
<span id="cb389-6"><a href="visualization-5.html#cb389-6" tabindex="-1"></a>                             <span class="fu">c</span>(<span class="st">&quot;gut&quot;</span>, <span class="st">&quot;tongue&quot;</span>)))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-200"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-200-1.png" alt="boxplot(single measure with significant results of pairwises)" width="576" />
<p class="caption">
Figure 11.3: boxplot(single measure with significant results of pairwises)
</p>
</div>
<ul>
<li>single measure with significant results of pairwises and outlier</li>
</ul>
<div class="sourceCode" id="cb390"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb390-1"><a href="visualization-5.html#cb390-1" tabindex="-1"></a><span class="fu">plot_boxplot</span>(<span class="at">data =</span> dat_alpha,</span>
<span id="cb390-2"><a href="visualization-5.html#cb390-2" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="st">&quot;Shannon&quot;</span>,</span>
<span id="cb390-3"><a href="visualization-5.html#cb390-3" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb390-4"><a href="visualization-5.html#cb390-4" tabindex="-1"></a>             <span class="at">do_test =</span> <span class="cn">TRUE</span>,</span>
<span id="cb390-5"><a href="visualization-5.html#cb390-5" tabindex="-1"></a>             <span class="at">cmp_list =</span> <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">&quot;gut&quot;</span>, <span class="st">&quot;right palm&quot;</span>), </span>
<span id="cb390-6"><a href="visualization-5.html#cb390-6" tabindex="-1"></a>                             <span class="fu">c</span>(<span class="st">&quot;gut&quot;</span>, <span class="st">&quot;tongue&quot;</span>)),</span>
<span id="cb390-7"><a href="visualization-5.html#cb390-7" tabindex="-1"></a>             <span class="at">outlier =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-201"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-201-1.png" alt="boxplot(single measure with significant results of pairwises and outlier)" width="576" />
<p class="caption">
Figure 11.4: boxplot(single measure with significant results of pairwises and outlier)
</p>
</div>
<ul>
<li>single measure with significant results of ref_group</li>
</ul>
<div class="sourceCode" id="cb391"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb391-1"><a href="visualization-5.html#cb391-1" tabindex="-1"></a><span class="fu">plot_boxplot</span>(<span class="at">data =</span> dat_alpha,</span>
<span id="cb391-2"><a href="visualization-5.html#cb391-2" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="st">&quot;Shannon&quot;</span>,</span>
<span id="cb391-3"><a href="visualization-5.html#cb391-3" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb391-4"><a href="visualization-5.html#cb391-4" tabindex="-1"></a>             <span class="at">do_test =</span> <span class="cn">TRUE</span>,</span>
<span id="cb391-5"><a href="visualization-5.html#cb391-5" tabindex="-1"></a>             <span class="at">ref_group =</span> <span class="st">&quot;gut&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-202"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-202-1.png" alt="boxplot(single measure with significant results of ref_group)" width="576" />
<p class="caption">
Figure 11.5: boxplot(single measure with significant results of ref_group)
</p>
</div>
<ul>
<li>multiple measures</li>
</ul>
<div class="sourceCode" id="cb392"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb392-1"><a href="visualization-5.html#cb392-1" tabindex="-1"></a><span class="fu">plot_boxplot</span>(<span class="at">data =</span> dat_alpha,</span>
<span id="cb392-2"><a href="visualization-5.html#cb392-2" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Chao1&quot;</span>, <span class="st">&quot;Observed&quot;</span>),</span>
<span id="cb392-3"><a href="visualization-5.html#cb392-3" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb392-4"><a href="visualization-5.html#cb392-4" tabindex="-1"></a>             <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;gut&quot;</span>, <span class="st">&quot;right palm&quot;</span>, <span class="st">&quot;tongue&quot;</span>),</span>
<span id="cb392-5"><a href="visualization-5.html#cb392-5" tabindex="-1"></a>             <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;green&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb392-6"><a href="visualization-5.html#cb392-6" tabindex="-1"></a>             <span class="at">ref_group =</span> <span class="st">&quot;gut&quot;</span>,</span>
<span id="cb392-7"><a href="visualization-5.html#cb392-7" tabindex="-1"></a>             <span class="at">method =</span> <span class="st">&quot;wilcox.test&quot;</span>,</span>
<span id="cb392-8"><a href="visualization-5.html#cb392-8" tabindex="-1"></a>             <span class="at">outlier =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-203"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-203-1.png" alt="boxplot(multiple measure with group number)" width="864" />
<p class="caption">
Figure 11.6: boxplot(multiple measure with group number)
</p>
</div>
<ul>
<li>show group_number in the x-axis break</li>
</ul>
<div class="sourceCode" id="cb393"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb393-1"><a href="visualization-5.html#cb393-1" tabindex="-1"></a><span class="fu">plot_boxplot</span>(<span class="at">data =</span> dat_alpha,</span>
<span id="cb393-2"><a href="visualization-5.html#cb393-2" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Chao1&quot;</span>, <span class="st">&quot;Observed&quot;</span>),</span>
<span id="cb393-3"><a href="visualization-5.html#cb393-3" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb393-4"><a href="visualization-5.html#cb393-4" tabindex="-1"></a>             <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;gut&quot;</span>, <span class="st">&quot;right palm&quot;</span>, <span class="st">&quot;tongue&quot;</span>),</span>
<span id="cb393-5"><a href="visualization-5.html#cb393-5" tabindex="-1"></a>             <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;green&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb393-6"><a href="visualization-5.html#cb393-6" tabindex="-1"></a>             <span class="at">do_test =</span> <span class="cn">TRUE</span>,</span>
<span id="cb393-7"><a href="visualization-5.html#cb393-7" tabindex="-1"></a>             <span class="at">method =</span> <span class="st">&quot;wilcox.test&quot;</span>,</span>
<span id="cb393-8"><a href="visualization-5.html#cb393-8" tabindex="-1"></a>             <span class="at">group_number =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-204"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-204-1.png" alt="boxplot(group_number)" width="768" />
<p class="caption">
Figure 11.7: boxplot(group_number)
</p>
</div>
</div>
<div id="plot_barplot" class="section level2 hasAnchor" number="11.4">
<h2><span class="header-section-number">11.4</span> plot_barplot<a href="visualization-5.html#plot_barplot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>plot_barplot</code> has many parameters, and help you enjoy it.</p>
<ul>
<li>single measure</li>
</ul>
<div class="sourceCode" id="cb394"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb394-1"><a href="visualization-5.html#cb394-1" tabindex="-1"></a><span class="fu">plot_barplot</span>(<span class="at">data =</span> dat_alpha,</span>
<span id="cb394-2"><a href="visualization-5.html#cb394-2" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="st">&quot;Shannon&quot;</span>,</span>
<span id="cb394-3"><a href="visualization-5.html#cb394-3" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb394-4"><a href="visualization-5.html#cb394-4" tabindex="-1"></a>             <span class="at">do_test =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-205"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-205-1.png" alt="barplot(single measure)" width="576" />
<p class="caption">
Figure 11.8: barplot(single measure)
</p>
</div>
<ul>
<li>single measure with significant results</li>
</ul>
<div class="sourceCode" id="cb395"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb395-1"><a href="visualization-5.html#cb395-1" tabindex="-1"></a><span class="fu">plot_barplot</span>(<span class="at">data =</span> dat_alpha,</span>
<span id="cb395-2"><a href="visualization-5.html#cb395-2" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="st">&quot;Shannon&quot;</span>,</span>
<span id="cb395-3"><a href="visualization-5.html#cb395-3" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb395-4"><a href="visualization-5.html#cb395-4" tabindex="-1"></a>             <span class="at">do_test =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-206"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-206-1.png" alt="barplot(single measure with significant results)" width="576" />
<p class="caption">
Figure 11.9: barplot(single measure with significant results)
</p>
</div>
<ul>
<li>single measure with significant results of pairwises</li>
</ul>
<div class="sourceCode" id="cb396"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb396-1"><a href="visualization-5.html#cb396-1" tabindex="-1"></a><span class="fu">plot_barplot</span>(<span class="at">data =</span> dat_alpha,</span>
<span id="cb396-2"><a href="visualization-5.html#cb396-2" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="st">&quot;Shannon&quot;</span>,</span>
<span id="cb396-3"><a href="visualization-5.html#cb396-3" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb396-4"><a href="visualization-5.html#cb396-4" tabindex="-1"></a>             <span class="at">do_test =</span> <span class="cn">TRUE</span>,</span>
<span id="cb396-5"><a href="visualization-5.html#cb396-5" tabindex="-1"></a>             <span class="at">cmp_list =</span> <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">&quot;gut&quot;</span>, <span class="st">&quot;right palm&quot;</span>), <span class="fu">c</span>(<span class="st">&quot;gut&quot;</span>, <span class="st">&quot;tongue&quot;</span>)))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-207"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-207-1.png" alt="barplot(single measure with significant results of pairwises)" width="576" />
<p class="caption">
Figure 11.10: barplot(single measure with significant results of pairwises)
</p>
</div>
<ul>
<li>single measure with significant results of ref_group</li>
</ul>
<div class="sourceCode" id="cb397"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb397-1"><a href="visualization-5.html#cb397-1" tabindex="-1"></a><span class="fu">plot_barplot</span>(<span class="at">data =</span> dat_alpha,</span>
<span id="cb397-2"><a href="visualization-5.html#cb397-2" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="st">&quot;Shannon&quot;</span>,</span>
<span id="cb397-3"><a href="visualization-5.html#cb397-3" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb397-4"><a href="visualization-5.html#cb397-4" tabindex="-1"></a>             <span class="at">do_test =</span> <span class="cn">TRUE</span>,</span>
<span id="cb397-5"><a href="visualization-5.html#cb397-5" tabindex="-1"></a>             <span class="at">ref_group =</span> <span class="st">&quot;gut&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-208"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-208-1.png" alt="barplot(single measure with significant results of ref_group)" width="576" />
<p class="caption">
Figure 11.11: barplot(single measure with significant results of ref_group)
</p>
</div>
<ul>
<li>multiple index</li>
</ul>
<div class="sourceCode" id="cb398"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb398-1"><a href="visualization-5.html#cb398-1" tabindex="-1"></a><span class="fu">plot_barplot</span>(<span class="at">data =</span> dat_alpha,</span>
<span id="cb398-2"><a href="visualization-5.html#cb398-2" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Chao1&quot;</span>, <span class="st">&quot;Observed&quot;</span>),</span>
<span id="cb398-3"><a href="visualization-5.html#cb398-3" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb398-4"><a href="visualization-5.html#cb398-4" tabindex="-1"></a>             <span class="at">do_test =</span> <span class="cn">TRUE</span>,</span>
<span id="cb398-5"><a href="visualization-5.html#cb398-5" tabindex="-1"></a>             <span class="at">method =</span> <span class="st">&quot;wilcox.test&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-209"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-209-1.png" alt="barplot(multiple index)" width="768" />
<p class="caption">
Figure 11.12: barplot(multiple index)
</p>
</div>
<ul>
<li>show group_number in the x-axis break</li>
</ul>
<div class="sourceCode" id="cb399"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb399-1"><a href="visualization-5.html#cb399-1" tabindex="-1"></a><span class="fu">plot_barplot</span>(<span class="at">data =</span> dat_alpha,</span>
<span id="cb399-2"><a href="visualization-5.html#cb399-2" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Chao1&quot;</span>, <span class="st">&quot;Observed&quot;</span>),</span>
<span id="cb399-3"><a href="visualization-5.html#cb399-3" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb399-4"><a href="visualization-5.html#cb399-4" tabindex="-1"></a>             <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;gut&quot;</span>, <span class="st">&quot;right palm&quot;</span>, <span class="st">&quot;tongue&quot;</span>),</span>
<span id="cb399-5"><a href="visualization-5.html#cb399-5" tabindex="-1"></a>             <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;green&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb399-6"><a href="visualization-5.html#cb399-6" tabindex="-1"></a>             <span class="at">do_test =</span> <span class="cn">TRUE</span>,</span>
<span id="cb399-7"><a href="visualization-5.html#cb399-7" tabindex="-1"></a>             <span class="at">method =</span> <span class="st">&quot;wilcox.test&quot;</span>,</span>
<span id="cb399-8"><a href="visualization-5.html#cb399-8" tabindex="-1"></a>             <span class="at">group_number =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-210"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-210-1.png" alt="barplot(group_number)" width="768" />
<p class="caption">
Figure 11.13: barplot(group_number)
</p>
</div>
</div>
<div id="plot_dotplot" class="section level2 hasAnchor" number="11.5">
<h2><span class="header-section-number">11.5</span> plot_dotplot<a href="visualization-5.html#plot_dotplot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>plot_dotplot</code> has many parameters, and help you enjoy it.</p>
<ul>
<li>single measure</li>
</ul>
<div class="sourceCode" id="cb400"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb400-1"><a href="visualization-5.html#cb400-1" tabindex="-1"></a><span class="fu">plot_dotplot</span>(<span class="at">data =</span> dat_alpha,</span>
<span id="cb400-2"><a href="visualization-5.html#cb400-2" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="st">&quot;Shannon&quot;</span>,</span>
<span id="cb400-3"><a href="visualization-5.html#cb400-3" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb400-4"><a href="visualization-5.html#cb400-4" tabindex="-1"></a>             <span class="at">do_test =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## Bin width defaults to 1/30 of the range of the data. Pick better value with `binwidth`.</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-211"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-211-1.png" alt="dotplot(single measure)" width="576" />
<p class="caption">
Figure 11.14: dotplot(single measure)
</p>
</div>
<ul>
<li>single measure with significant results</li>
</ul>
<div class="sourceCode" id="cb402"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb402-1"><a href="visualization-5.html#cb402-1" tabindex="-1"></a><span class="fu">plot_dotplot</span>(<span class="at">data =</span> dat_alpha,</span>
<span id="cb402-2"><a href="visualization-5.html#cb402-2" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="st">&quot;Shannon&quot;</span>,</span>
<span id="cb402-3"><a href="visualization-5.html#cb402-3" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb402-4"><a href="visualization-5.html#cb402-4" tabindex="-1"></a>             <span class="at">do_test =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Bin width defaults to 1/30 of the range of the data. Pick better value with `binwidth`.</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-212"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-212-1.png" alt="dotplot(single measure with significant results)" width="576" />
<p class="caption">
Figure 11.15: dotplot(single measure with significant results)
</p>
</div>
<ul>
<li>single measure with significant results of pairwises</li>
</ul>
<div class="sourceCode" id="cb404"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb404-1"><a href="visualization-5.html#cb404-1" tabindex="-1"></a><span class="fu">plot_dotplot</span>(<span class="at">data =</span> dat_alpha,</span>
<span id="cb404-2"><a href="visualization-5.html#cb404-2" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="st">&quot;Shannon&quot;</span>,</span>
<span id="cb404-3"><a href="visualization-5.html#cb404-3" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb404-4"><a href="visualization-5.html#cb404-4" tabindex="-1"></a>             <span class="at">do_test =</span> <span class="cn">TRUE</span>,</span>
<span id="cb404-5"><a href="visualization-5.html#cb404-5" tabindex="-1"></a>             <span class="at">cmp_list =</span> <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">&quot;gut&quot;</span>, <span class="st">&quot;right palm&quot;</span>), </span>
<span id="cb404-6"><a href="visualization-5.html#cb404-6" tabindex="-1"></a>                             <span class="fu">c</span>(<span class="st">&quot;gut&quot;</span>, <span class="st">&quot;tongue&quot;</span>)))</span></code></pre></div>
<pre><code>## Bin width defaults to 1/30 of the range of the data. Pick better value with `binwidth`.</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-213"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-213-1.png" alt="dotplot(single measure with significant results of pairwises)" width="576" />
<p class="caption">
Figure 11.16: dotplot(single measure with significant results of pairwises)
</p>
</div>
<ul>
<li>single measure with significant results of ref_group</li>
</ul>
<div class="sourceCode" id="cb406"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb406-1"><a href="visualization-5.html#cb406-1" tabindex="-1"></a><span class="fu">plot_dotplot</span>(</span>
<span id="cb406-2"><a href="visualization-5.html#cb406-2" tabindex="-1"></a>    <span class="at">data =</span> dat_alpha,</span>
<span id="cb406-3"><a href="visualization-5.html#cb406-3" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Shannon&quot;</span>,</span>
<span id="cb406-4"><a href="visualization-5.html#cb406-4" tabindex="-1"></a>    <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb406-5"><a href="visualization-5.html#cb406-5" tabindex="-1"></a>    <span class="at">do_test =</span> <span class="cn">TRUE</span>,</span>
<span id="cb406-6"><a href="visualization-5.html#cb406-6" tabindex="-1"></a>    <span class="at">ref_group =</span> <span class="st">&quot;gut&quot;</span>)</span></code></pre></div>
<pre><code>## Bin width defaults to 1/30 of the range of the data. Pick better value with `binwidth`.</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-214"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-214-1.png" alt="dotplot(single measure with significant results of ref_group)" width="576" />
<p class="caption">
Figure 11.17: dotplot(single measure with significant results of ref_group)
</p>
</div>
<ul>
<li>dot size and median size</li>
</ul>
<div class="sourceCode" id="cb408"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb408-1"><a href="visualization-5.html#cb408-1" tabindex="-1"></a><span class="fu">plot_dotplot</span>(</span>
<span id="cb408-2"><a href="visualization-5.html#cb408-2" tabindex="-1"></a>    <span class="at">data =</span> dat_alpha,</span>
<span id="cb408-3"><a href="visualization-5.html#cb408-3" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Shannon&quot;</span>,</span>
<span id="cb408-4"><a href="visualization-5.html#cb408-4" tabindex="-1"></a>    <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb408-5"><a href="visualization-5.html#cb408-5" tabindex="-1"></a>    <span class="at">do_test =</span> <span class="cn">TRUE</span>,</span>
<span id="cb408-6"><a href="visualization-5.html#cb408-6" tabindex="-1"></a>    <span class="at">ref_group =</span> <span class="st">&quot;gut&quot;</span>,</span>
<span id="cb408-7"><a href="visualization-5.html#cb408-7" tabindex="-1"></a>    <span class="at">dotsize =</span> <span class="fl">0.5</span>,</span>
<span id="cb408-8"><a href="visualization-5.html#cb408-8" tabindex="-1"></a>    <span class="at">mediansize =</span> <span class="dv">2</span>)</span></code></pre></div>
<pre><code>## Bin width defaults to 1/30 of the range of the data. Pick better value with `binwidth`.</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-215"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-215-1.png" alt="dotplot(dot size and median size)" width="576" />
<p class="caption">
Figure 11.18: dotplot(dot size and median size)
</p>
</div>
<ul>
<li>multiple index</li>
</ul>
<div class="sourceCode" id="cb410"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb410-1"><a href="visualization-5.html#cb410-1" tabindex="-1"></a><span class="fu">plot_dotplot</span>(</span>
<span id="cb410-2"><a href="visualization-5.html#cb410-2" tabindex="-1"></a>    <span class="at">data =</span> dat_alpha,</span>
<span id="cb410-3"><a href="visualization-5.html#cb410-3" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Chao1&quot;</span>, <span class="st">&quot;Observed&quot;</span>),</span>
<span id="cb410-4"><a href="visualization-5.html#cb410-4" tabindex="-1"></a>    <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb410-5"><a href="visualization-5.html#cb410-5" tabindex="-1"></a>    <span class="at">do_test =</span> <span class="cn">TRUE</span>,</span>
<span id="cb410-6"><a href="visualization-5.html#cb410-6" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">&quot;wilcox.test&quot;</span>)</span></code></pre></div>
<pre><code>## Bin width defaults to 1/30 of the range of the data. Pick better value with `binwidth`.</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-216"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-216-1.png" alt="dotplot(multiple index)" width="768" />
<p class="caption">
Figure 11.19: dotplot(multiple index)
</p>
</div>
<ul>
<li>multiple index with errorbar</li>
</ul>
<div class="sourceCode" id="cb412"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb412-1"><a href="visualization-5.html#cb412-1" tabindex="-1"></a><span class="fu">plot_dotplot</span>(</span>
<span id="cb412-2"><a href="visualization-5.html#cb412-2" tabindex="-1"></a>    <span class="at">data =</span> dat_alpha,</span>
<span id="cb412-3"><a href="visualization-5.html#cb412-3" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Chao1&quot;</span>, <span class="st">&quot;Observed&quot;</span>),</span>
<span id="cb412-4"><a href="visualization-5.html#cb412-4" tabindex="-1"></a>    <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb412-5"><a href="visualization-5.html#cb412-5" tabindex="-1"></a>    <span class="at">do_test =</span> <span class="cn">TRUE</span>,</span>
<span id="cb412-6"><a href="visualization-5.html#cb412-6" tabindex="-1"></a>    <span class="at">show_type =</span> <span class="st">&quot;errorbar&quot;</span>,</span>
<span id="cb412-7"><a href="visualization-5.html#cb412-7" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">&quot;wilcox.test&quot;</span>)</span></code></pre></div>
<pre><code>## Bin width defaults to 1/30 of the range of the data. Pick better value with `binwidth`.</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-217"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-217-1.png" alt="dotplot(multiple index errorbar)" width="768" />
<p class="caption">
Figure 11.20: dotplot(multiple index errorbar)
</p>
</div>
<ul>
<li>show group_number in the x-axis break</li>
</ul>
<div class="sourceCode" id="cb414"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb414-1"><a href="visualization-5.html#cb414-1" tabindex="-1"></a><span class="fu">plot_dotplot</span>(<span class="at">data =</span> dat_alpha,</span>
<span id="cb414-2"><a href="visualization-5.html#cb414-2" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Chao1&quot;</span>, <span class="st">&quot;Observed&quot;</span>),</span>
<span id="cb414-3"><a href="visualization-5.html#cb414-3" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb414-4"><a href="visualization-5.html#cb414-4" tabindex="-1"></a>             <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;gut&quot;</span>, <span class="st">&quot;right palm&quot;</span>, <span class="st">&quot;tongue&quot;</span>),</span>
<span id="cb414-5"><a href="visualization-5.html#cb414-5" tabindex="-1"></a>             <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;green&quot;</span>, <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb414-6"><a href="visualization-5.html#cb414-6" tabindex="-1"></a>             <span class="at">show_type =</span> <span class="st">&quot;errorbar&quot;</span>,</span>
<span id="cb414-7"><a href="visualization-5.html#cb414-7" tabindex="-1"></a>             <span class="at">do_test =</span> <span class="cn">TRUE</span>,</span>
<span id="cb414-8"><a href="visualization-5.html#cb414-8" tabindex="-1"></a>             <span class="at">method =</span> <span class="st">&quot;wilcox.test&quot;</span>,</span>
<span id="cb414-9"><a href="visualization-5.html#cb414-9" tabindex="-1"></a>             <span class="at">group_number =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Bin width defaults to 1/30 of the range of the data. Pick better value with `binwidth`.</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-218"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-218-1.png" alt="dotplot(group_number)" width="768" />
<p class="caption">
Figure 11.21: dotplot(group_number)
</p>
</div>
</div>
<div id="plot_correlation_boxplot" class="section level2 hasAnchor" number="11.6">
<h2><span class="header-section-number">11.6</span> plot_correlation_boxplot<a href="visualization-5.html#plot_correlation_boxplot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Help you enjoy <code>plot_correlation_boxplot</code>.</p>
<div class="sourceCode" id="cb416"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb416-1"><a href="visualization-5.html#cb416-1" tabindex="-1"></a><span class="fu">plot_correlation_boxplot</span>(</span>
<span id="cb416-2"><a href="visualization-5.html#cb416-2" tabindex="-1"></a>    <span class="at">data =</span> dat_alpha,</span>
<span id="cb416-3"><a href="visualization-5.html#cb416-3" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;Chao1&quot;</span>,</span>
<span id="cb416-4"><a href="visualization-5.html#cb416-4" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Shannon&quot;</span>,</span>
<span id="cb416-5"><a href="visualization-5.html#cb416-5" tabindex="-1"></a>    <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-219"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-219-1.png" alt="correlation with boxplot" width="768" />
<p class="caption">
Figure 11.22: correlation with boxplot
</p>
</div>
</div>
<div id="plot_correlation_density" class="section level2 hasAnchor" number="11.7">
<h2><span class="header-section-number">11.7</span> plot_correlation_density<a href="visualization-5.html#plot_correlation_density" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Help you enjoy <code>plot_correlation_density</code>.</p>
<div class="sourceCode" id="cb417"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb417-1"><a href="visualization-5.html#cb417-1" tabindex="-1"></a><span class="fu">plot_correlation_density</span>(</span>
<span id="cb417-2"><a href="visualization-5.html#cb417-2" tabindex="-1"></a>    <span class="at">data =</span> dat_alpha,</span>
<span id="cb417-3"><a href="visualization-5.html#cb417-3" tabindex="-1"></a>    <span class="at">x_index =</span> <span class="st">&quot;Chao1&quot;</span>,</span>
<span id="cb417-4"><a href="visualization-5.html#cb417-4" tabindex="-1"></a>    <span class="at">y_index =</span> <span class="st">&quot;Shannon&quot;</span>,</span>
<span id="cb417-5"><a href="visualization-5.html#cb417-5" tabindex="-1"></a>    <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-220"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-220-1.png" alt="correlation with density" width="768" />
<p class="caption">
Figure 11.23: correlation with density
</p>
</div>
</div>
<div id="plot_ordination" class="section level2 hasAnchor" number="11.8">
<h2><span class="header-section-number">11.8</span> plot_Ordination<a href="visualization-5.html#plot_ordination" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>plot_Ordination</code> provides too many parameters for users to display the ordination results by using ggplot2 format. Here is the ordinary pattern.</p>
<div class="sourceCode" id="cb418"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb418-1"><a href="visualization-5.html#cb418-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;dada2_ps&quot;</span>)</span>
<span id="cb418-2"><a href="visualization-5.html#cb418-2" tabindex="-1"></a></span>
<span id="cb418-3"><a href="visualization-5.html#cb418-3" tabindex="-1"></a><span class="co"># step1: Removing samples of specific group in phyloseq-class object</span></span>
<span id="cb418-4"><a href="visualization-5.html#cb418-4" tabindex="-1"></a>dada2_ps_remove_BRS <span class="ot">&lt;-</span> <span class="fu">get_GroupPhyloseq</span>(</span>
<span id="cb418-5"><a href="visualization-5.html#cb418-5" tabindex="-1"></a>                     <span class="at">ps =</span> dada2_ps,</span>
<span id="cb418-6"><a href="visualization-5.html#cb418-6" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb418-7"><a href="visualization-5.html#cb418-7" tabindex="-1"></a>                     <span class="at">group_names =</span> <span class="st">&quot;QC&quot;</span>)</span>
<span id="cb418-8"><a href="visualization-5.html#cb418-8" tabindex="-1"></a></span>
<span id="cb418-9"><a href="visualization-5.html#cb418-9" tabindex="-1"></a><span class="co"># step2: Rarefying counts in phyloseq-class object</span></span>
<span id="cb418-10"><a href="visualization-5.html#cb418-10" tabindex="-1"></a>dada2_ps_rarefy <span class="ot">&lt;-</span> <span class="fu">norm_rarefy</span>(<span class="at">object =</span> dada2_ps_remove_BRS,</span>
<span id="cb418-11"><a href="visualization-5.html#cb418-11" tabindex="-1"></a>                               <span class="at">size =</span> <span class="dv">51181</span>)</span>
<span id="cb418-12"><a href="visualization-5.html#cb418-12" tabindex="-1"></a></span>
<span id="cb418-13"><a href="visualization-5.html#cb418-13" tabindex="-1"></a><span class="co"># step3: Extracting specific taxa phyloseq-class object </span></span>
<span id="cb418-14"><a href="visualization-5.html#cb418-14" tabindex="-1"></a>dada2_ps_rare_genus <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(<span class="at">ps =</span> dada2_ps_rarefy, </span>
<span id="cb418-15"><a href="visualization-5.html#cb418-15" tabindex="-1"></a>                                      <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>, </span>
<span id="cb418-16"><a href="visualization-5.html#cb418-16" tabindex="-1"></a>                                      <span class="at">absolute =</span> <span class="cn">TRUE</span>)</span>
<span id="cb418-17"><a href="visualization-5.html#cb418-17" tabindex="-1"></a></span>
<span id="cb418-18"><a href="visualization-5.html#cb418-18" tabindex="-1"></a><span class="co"># step4: Aggregating low relative abundance or unclassified taxa into others</span></span>
<span id="cb418-19"><a href="visualization-5.html#cb418-19" tabindex="-1"></a><span class="co"># dada2_ps_genus_LRA &lt;- summarize_LowAbundance_taxa(ps = dada2_ps_rare_genus, </span></span>
<span id="cb418-20"><a href="visualization-5.html#cb418-20" tabindex="-1"></a><span class="co">#                                                   cutoff = 10, </span></span>
<span id="cb418-21"><a href="visualization-5.html#cb418-21" tabindex="-1"></a><span class="co">#                                                   unclass = TRUE)</span></span>
<span id="cb418-22"><a href="visualization-5.html#cb418-22" tabindex="-1"></a></span>
<span id="cb418-23"><a href="visualization-5.html#cb418-23" tabindex="-1"></a><span class="co"># step4: Filtering the low relative abundance or unclassified taxa by the threshold</span></span>
<span id="cb418-24"><a href="visualization-5.html#cb418-24" tabindex="-1"></a>dada2_ps_genus_filter <span class="ot">&lt;-</span> <span class="fu">run_filter</span>(<span class="at">ps =</span> dada2_ps_rare_genus, </span>
<span id="cb418-25"><a href="visualization-5.html#cb418-25" tabindex="-1"></a>                                    <span class="at">cutoff =</span> <span class="dv">10</span>, </span>
<span id="cb418-26"><a href="visualization-5.html#cb418-26" tabindex="-1"></a>                                    <span class="at">unclass =</span> <span class="cn">TRUE</span>)</span>
<span id="cb418-27"><a href="visualization-5.html#cb418-27" tabindex="-1"></a></span>
<span id="cb418-28"><a href="visualization-5.html#cb418-28" tabindex="-1"></a><span class="co"># step5: Trimming the taxa with low occurrence less than threshold</span></span>
<span id="cb418-29"><a href="visualization-5.html#cb418-29" tabindex="-1"></a>dada2_ps_genus_filter_trim <span class="ot">&lt;-</span> <span class="fu">run_trim</span>(<span class="at">object =</span> dada2_ps_genus_filter, </span>
<span id="cb418-30"><a href="visualization-5.html#cb418-30" tabindex="-1"></a>                                       <span class="at">cutoff =</span> <span class="fl">0.2</span>, </span>
<span id="cb418-31"><a href="visualization-5.html#cb418-31" tabindex="-1"></a>                                       <span class="at">trim =</span> <span class="st">&quot;feature&quot;</span>)</span>
<span id="cb418-32"><a href="visualization-5.html#cb418-32" tabindex="-1"></a>dada2_ps_genus_filter_trim</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 83 taxa and 23 samples ]
## sample_data() Sample Data:       [ 23 samples by 1 sample variables ]
## tax_table()   Taxonomy Table:    [ 83 taxa by 6 taxonomic ranks ]</code></pre>
<div class="sourceCode" id="cb420"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb420-1"><a href="visualization-5.html#cb420-1" tabindex="-1"></a>ordination_PCA <span class="ot">&lt;-</span> <span class="fu">run_ordination</span>(</span>
<span id="cb420-2"><a href="visualization-5.html#cb420-2" tabindex="-1"></a>                       <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb420-3"><a href="visualization-5.html#cb420-3" tabindex="-1"></a>                       <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb420-4"><a href="visualization-5.html#cb420-4" tabindex="-1"></a>                       <span class="at">method =</span> <span class="st">&quot;PCA&quot;</span>)</span>
<span id="cb420-5"><a href="visualization-5.html#cb420-5" tabindex="-1"></a></span>
<span id="cb420-6"><a href="visualization-5.html#cb420-6" tabindex="-1"></a><span class="fu">names</span>(ordination_PCA)</span></code></pre></div>
<pre><code>## [1] &quot;fit&quot;           &quot;dat&quot;           &quot;explains&quot;      &quot;eigvalue&quot;      &quot;PERMANOVA&quot;     &quot;BETADISPER&quot;    &quot;axis_taxa_cor&quot;</code></pre>
<ul>
<li>Ordinary pattern</li>
</ul>
<div class="sourceCode" id="cb422"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb422-1"><a href="visualization-5.html#cb422-1" tabindex="-1"></a><span class="fu">plot_Ordination</span>(<span class="at">ResultList =</span> ordination_PCA, </span>
<span id="cb422-2"><a href="visualization-5.html#cb422-2" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-222"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-222-1.png" alt="plot_Ordination (Ordinary pattern)" width="768" />
<p class="caption">
Figure 11.24: plot_Ordination (Ordinary pattern)
</p>
</div>
<ul>
<li>plot with SampleID and setting group colors</li>
</ul>
<div class="sourceCode" id="cb423"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb423-1"><a href="visualization-5.html#cb423-1" tabindex="-1"></a><span class="fu">plot_Ordination</span>(<span class="at">ResultList =</span> ordination_PCA, </span>
<span id="cb423-2"><a href="visualization-5.html#cb423-2" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>, </span>
<span id="cb423-3"><a href="visualization-5.html#cb423-3" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb423-4"><a href="visualization-5.html#cb423-4" tabindex="-1"></a>                <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>),</span>
<span id="cb423-5"><a href="visualization-5.html#cb423-5" tabindex="-1"></a>                <span class="at">circle_type =</span> <span class="st">&quot;ellipse&quot;</span>,</span>
<span id="cb423-6"><a href="visualization-5.html#cb423-6" tabindex="-1"></a>                <span class="at">sample =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-223"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-223-1.png" alt="plot_Ordination (Ordinary pattern with SampleID)" width="768" />
<p class="caption">
Figure 11.25: plot_Ordination (Ordinary pattern with SampleID)
</p>
</div>
<ul>
<li>ellipse with 95% confidence interval</li>
</ul>
<div class="sourceCode" id="cb424"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb424-1"><a href="visualization-5.html#cb424-1" tabindex="-1"></a><span class="fu">plot_Ordination</span>(<span class="at">ResultList =</span> ordination_PCA, </span>
<span id="cb424-2"><a href="visualization-5.html#cb424-2" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>, </span>
<span id="cb424-3"><a href="visualization-5.html#cb424-3" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb424-4"><a href="visualization-5.html#cb424-4" tabindex="-1"></a>                <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>),</span>
<span id="cb424-5"><a href="visualization-5.html#cb424-5" tabindex="-1"></a>                <span class="at">circle_type =</span> <span class="st">&quot;ellipse_CI&quot;</span>,</span>
<span id="cb424-6"><a href="visualization-5.html#cb424-6" tabindex="-1"></a>                <span class="at">sample =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-224"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-224-1.png" alt="plot_Ordination (ellipse with 95% confidence interval)" width="768" />
<p class="caption">
Figure 11.26: plot_Ordination (ellipse with 95% confidence interval)
</p>
</div>
<ul>
<li>ellipse with groups</li>
</ul>
<div class="sourceCode" id="cb425"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb425-1"><a href="visualization-5.html#cb425-1" tabindex="-1"></a><span class="fu">plot_Ordination</span>(<span class="at">ResultList =</span> ordination_PCA, </span>
<span id="cb425-2"><a href="visualization-5.html#cb425-2" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>, </span>
<span id="cb425-3"><a href="visualization-5.html#cb425-3" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb425-4"><a href="visualization-5.html#cb425-4" tabindex="-1"></a>                <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>),</span>
<span id="cb425-5"><a href="visualization-5.html#cb425-5" tabindex="-1"></a>                <span class="at">circle_type =</span> <span class="st">&quot;ellipse_groups&quot;</span>,</span>
<span id="cb425-6"><a href="visualization-5.html#cb425-6" tabindex="-1"></a>                <span class="at">sample =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-225"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-225-1.png" alt="plot_Ordination (ellipse with groups)" width="768" />
<p class="caption">
Figure 11.27: plot_Ordination (ellipse with groups)
</p>
</div>
<ul>
<li>ellipse with border line</li>
</ul>
<div class="sourceCode" id="cb426"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb426-1"><a href="visualization-5.html#cb426-1" tabindex="-1"></a><span class="fu">plot_Ordination</span>(<span class="at">ResultList =</span> ordination_PCA, </span>
<span id="cb426-2"><a href="visualization-5.html#cb426-2" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>, </span>
<span id="cb426-3"><a href="visualization-5.html#cb426-3" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb426-4"><a href="visualization-5.html#cb426-4" tabindex="-1"></a>                <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>),</span>
<span id="cb426-5"><a href="visualization-5.html#cb426-5" tabindex="-1"></a>                <span class="at">circle_type =</span> <span class="st">&quot;ellipse_line&quot;</span>,</span>
<span id="cb426-6"><a href="visualization-5.html#cb426-6" tabindex="-1"></a>                <span class="at">sample =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-226"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-226-1.png" alt="plot_Ordination (ellipse with border line)" width="768" />
<p class="caption">
Figure 11.28: plot_Ordination (ellipse with border line)
</p>
</div>
<ul>
<li>plot with SampleID and sideboxplot and setting group colors</li>
</ul>
<div class="sourceCode" id="cb427"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb427-1"><a href="visualization-5.html#cb427-1" tabindex="-1"></a><span class="fu">plot_Ordination</span>(<span class="at">ResultList =</span> ordination_PCA, </span>
<span id="cb427-2"><a href="visualization-5.html#cb427-2" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>, </span>
<span id="cb427-3"><a href="visualization-5.html#cb427-3" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb427-4"><a href="visualization-5.html#cb427-4" tabindex="-1"></a>                <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>),</span>
<span id="cb427-5"><a href="visualization-5.html#cb427-5" tabindex="-1"></a>                <span class="at">circle_type =</span> <span class="st">&quot;ellipse&quot;</span>,</span>
<span id="cb427-6"><a href="visualization-5.html#cb427-6" tabindex="-1"></a>                <span class="at">sidelinechart =</span> <span class="cn">FALSE</span>,</span>
<span id="cb427-7"><a href="visualization-5.html#cb427-7" tabindex="-1"></a>                <span class="at">sideboxplot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb427-8"><a href="visualization-5.html#cb427-8" tabindex="-1"></a>                <span class="at">sample =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-227"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-227-1.png" alt="plot_Ordination (Ordinary pattern with SampleID sideboxplot)" width="576" />
<p class="caption">
Figure 11.29: plot_Ordination (Ordinary pattern with SampleID sideboxplot)
</p>
</div>
<ul>
<li>plot with SampleID and sideboxplot and setting group colors 2</li>
</ul>
<div class="sourceCode" id="cb428"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb428-1"><a href="visualization-5.html#cb428-1" tabindex="-1"></a><span class="fu">plot_Ordination</span>(<span class="at">ResultList =</span> ordination_PCA, </span>
<span id="cb428-2"><a href="visualization-5.html#cb428-2" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>, </span>
<span id="cb428-3"><a href="visualization-5.html#cb428-3" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb428-4"><a href="visualization-5.html#cb428-4" tabindex="-1"></a>                <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>),</span>
<span id="cb428-5"><a href="visualization-5.html#cb428-5" tabindex="-1"></a>                <span class="at">circle_type =</span> <span class="st">&quot;ellipse_CI&quot;</span>,</span>
<span id="cb428-6"><a href="visualization-5.html#cb428-6" tabindex="-1"></a>                <span class="at">sidelinechart =</span> <span class="cn">FALSE</span>,</span>
<span id="cb428-7"><a href="visualization-5.html#cb428-7" tabindex="-1"></a>                <span class="at">sideboxplot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb428-8"><a href="visualization-5.html#cb428-8" tabindex="-1"></a>                <span class="at">sample =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-228"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-228-1.png" alt="plot_Ordination (ellipse_CI with SampleID sideboxplot)" width="576" />
<p class="caption">
Figure 11.30: plot_Ordination (ellipse_CI with SampleID sideboxplot)
</p>
</div>
<ul>
<li>plot with SampleID and sideboxplot and setting group colors 3</li>
</ul>
<div class="sourceCode" id="cb429"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb429-1"><a href="visualization-5.html#cb429-1" tabindex="-1"></a><span class="fu">plot_Ordination</span>(<span class="at">ResultList =</span> ordination_PCA, </span>
<span id="cb429-2"><a href="visualization-5.html#cb429-2" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>, </span>
<span id="cb429-3"><a href="visualization-5.html#cb429-3" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb429-4"><a href="visualization-5.html#cb429-4" tabindex="-1"></a>                <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>),</span>
<span id="cb429-5"><a href="visualization-5.html#cb429-5" tabindex="-1"></a>                <span class="at">circle_type =</span> <span class="st">&quot;ellipse_groups&quot;</span>,</span>
<span id="cb429-6"><a href="visualization-5.html#cb429-6" tabindex="-1"></a>                <span class="at">sidelinechart =</span> <span class="cn">FALSE</span>,</span>
<span id="cb429-7"><a href="visualization-5.html#cb429-7" tabindex="-1"></a>                <span class="at">sideboxplot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb429-8"><a href="visualization-5.html#cb429-8" tabindex="-1"></a>                <span class="at">sample =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-229"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-229-1.png" alt="plot_Ordination (ellipse_groups with SampleID sideboxplot)" width="576" />
<p class="caption">
Figure 11.31: plot_Ordination (ellipse_groups with SampleID sideboxplot)
</p>
</div>
<ul>
<li>plot with SampleID and sideboxplot and setting group colors 4</li>
</ul>
<div class="sourceCode" id="cb430"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb430-1"><a href="visualization-5.html#cb430-1" tabindex="-1"></a><span class="fu">plot_Ordination</span>(<span class="at">ResultList =</span> ordination_PCA, </span>
<span id="cb430-2"><a href="visualization-5.html#cb430-2" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>, </span>
<span id="cb430-3"><a href="visualization-5.html#cb430-3" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb430-4"><a href="visualization-5.html#cb430-4" tabindex="-1"></a>                <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>),</span>
<span id="cb430-5"><a href="visualization-5.html#cb430-5" tabindex="-1"></a>                <span class="at">circle_type =</span> <span class="st">&quot;ellipse_line&quot;</span>,</span>
<span id="cb430-6"><a href="visualization-5.html#cb430-6" tabindex="-1"></a>                <span class="at">sidelinechart =</span> <span class="cn">FALSE</span>,</span>
<span id="cb430-7"><a href="visualization-5.html#cb430-7" tabindex="-1"></a>                <span class="at">sideboxplot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb430-8"><a href="visualization-5.html#cb430-8" tabindex="-1"></a>                <span class="at">sample =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-230"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-230-1.png" alt="plot_Ordination (ellipse_line with SampleID sideboxplot)" width="576" />
<p class="caption">
Figure 11.32: plot_Ordination (ellipse_line with SampleID sideboxplot)
</p>
</div>
<ul>
<li>plot with SampleID and sideboxplot and setting group colors and shape</li>
</ul>
<div class="sourceCode" id="cb431"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb431-1"><a href="visualization-5.html#cb431-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;amplicon_ps&quot;</span>)</span>
<span id="cb431-2"><a href="visualization-5.html#cb431-2" tabindex="-1"></a>amplicon_ps_genus <span class="ot">&lt;-</span> <span class="fu">summarize_taxa</span>(<span class="at">ps =</span> amplicon_ps, </span>
<span id="cb431-3"><a href="visualization-5.html#cb431-3" tabindex="-1"></a>                                    <span class="at">taxa_level =</span> <span class="st">&quot;Genus&quot;</span>)</span>
<span id="cb431-4"><a href="visualization-5.html#cb431-4" tabindex="-1"></a>amplicon_res_ordination <span class="ot">&lt;-</span> <span class="fu">run_ordination</span>(</span>
<span id="cb431-5"><a href="visualization-5.html#cb431-5" tabindex="-1"></a>                              <span class="at">ps =</span> amplicon_ps_genus,</span>
<span id="cb431-6"><a href="visualization-5.html#cb431-6" tabindex="-1"></a>                              <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb431-7"><a href="visualization-5.html#cb431-7" tabindex="-1"></a>                              <span class="at">method =</span> <span class="st">&quot;PCoA&quot;</span>)</span></code></pre></div>
<pre><code>## [1] &quot;Pvalue of beta dispersion less than 0.05&quot;</code></pre>
<div class="sourceCode" id="cb433"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb433-1"><a href="visualization-5.html#cb433-1" tabindex="-1"></a><span class="fu">plot_Ordination</span>(<span class="at">ResultList =</span> amplicon_res_ordination, </span>
<span id="cb433-2"><a href="visualization-5.html#cb433-2" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb433-3"><a href="visualization-5.html#cb433-3" tabindex="-1"></a>                <span class="at">shape_column =</span> <span class="st">&quot;ReportedAntibioticUsage&quot;</span>,</span>
<span id="cb433-4"><a href="visualization-5.html#cb433-4" tabindex="-1"></a>                <span class="at">shape_values =</span> <span class="fu">c</span>(<span class="dv">16</span>, <span class="dv">17</span>),</span>
<span id="cb433-5"><a href="visualization-5.html#cb433-5" tabindex="-1"></a>                <span class="at">circle_type =</span> <span class="st">&quot;ellipse_line&quot;</span>,</span>
<span id="cb433-6"><a href="visualization-5.html#cb433-6" tabindex="-1"></a>                <span class="at">sidelinechart =</span> <span class="cn">FALSE</span>,</span>
<span id="cb433-7"><a href="visualization-5.html#cb433-7" tabindex="-1"></a>                <span class="at">sideboxplot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb433-8"><a href="visualization-5.html#cb433-8" tabindex="-1"></a>                <span class="at">sample =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-231"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-231-1.png" alt="plot_Ordination (ellipse_line with SampleID sideboxplot)" width="576" />
<p class="caption">
Figure 11.33: plot_Ordination (ellipse_line with SampleID sideboxplot)
</p>
</div>
</div>
<div id="plot_ggbiplot" class="section level2 hasAnchor" number="11.9">
<h2><span class="header-section-number">11.9</span> plot_ggbiplot<a href="visualization-5.html#plot_ggbiplot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>biplot with topN dominant taxa</li>
</ul>
<div class="sourceCode" id="cb434"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb434-1"><a href="visualization-5.html#cb434-1" tabindex="-1"></a><span class="fu">plot_ggbiplot</span>(<span class="at">ResultList =</span> ordination_PCA,</span>
<span id="cb434-2"><a href="visualization-5.html#cb434-2" tabindex="-1"></a>              <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb434-3"><a href="visualization-5.html#cb434-3" tabindex="-1"></a>              <span class="at">group_color =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;red&quot;</span>),</span>
<span id="cb434-4"><a href="visualization-5.html#cb434-4" tabindex="-1"></a>              <span class="at">topN =</span> <span class="dv">5</span>,</span>
<span id="cb434-5"><a href="visualization-5.html#cb434-5" tabindex="-1"></a>              <span class="at">ellipse =</span> <span class="cn">TRUE</span>,</span>
<span id="cb434-6"><a href="visualization-5.html#cb434-6" tabindex="-1"></a>              <span class="at">labels =</span> <span class="st">&quot;SampleID&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-232"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-232-1.png" alt="plot_ggbiplot (biplot)" width="576" />
<p class="caption">
Figure 11.34: plot_ggbiplot (biplot)
</p>
</div>
</div>
<div id="plot_corrplot" class="section level2 hasAnchor" number="11.10">
<h2><span class="header-section-number">11.10</span> plot_corrplot<a href="visualization-5.html#plot_corrplot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb435"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb435-1"><a href="visualization-5.html#cb435-1" tabindex="-1"></a>dada2_beta <span class="ot">&lt;-</span> <span class="fu">run_beta_diversity</span>(<span class="at">ps =</span> dada2_ps_rarefy, </span>
<span id="cb435-2"><a href="visualization-5.html#cb435-2" tabindex="-1"></a>                                 <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>)</span>
<span id="cb435-3"><a href="visualization-5.html#cb435-3" tabindex="-1"></a><span class="fu">plot_distance_corrplot</span>(<span class="at">datMatrix =</span> dada2_beta<span class="sc">$</span>BetaDistance)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-233"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-233-1.png" alt="plot_corrplot (distance)" width="576" />
<p class="caption">
Figure 11.35: plot_corrplot (distance)
</p>
</div>
</div>
<div id="plot_2da_venn" class="section level2 hasAnchor" number="11.11">
<h2><span class="header-section-number">11.11</span> plot_2DA_venn<a href="visualization-5.html#plot_2da_venn" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb436"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb436-1"><a href="visualization-5.html#cb436-1" tabindex="-1"></a>da_wilcox <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb436-2"><a href="visualization-5.html#cb436-2" tabindex="-1"></a>                <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb436-3"><a href="visualization-5.html#cb436-3" tabindex="-1"></a>                <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb436-4"><a href="visualization-5.html#cb436-4" tabindex="-1"></a>                <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb436-5"><a href="visualization-5.html#cb436-5" tabindex="-1"></a></span>
<span id="cb436-6"><a href="visualization-5.html#cb436-6" tabindex="-1"></a>da_ttest <span class="ot">&lt;-</span> <span class="fu">run_ttest</span>(</span>
<span id="cb436-7"><a href="visualization-5.html#cb436-7" tabindex="-1"></a>               <span class="at">ps =</span> dada2_ps_genus_filter_trim,</span>
<span id="cb436-8"><a href="visualization-5.html#cb436-8" tabindex="-1"></a>               <span class="at">group =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb436-9"><a href="visualization-5.html#cb436-9" tabindex="-1"></a>               <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>))</span>
<span id="cb436-10"><a href="visualization-5.html#cb436-10" tabindex="-1"></a></span>
<span id="cb436-11"><a href="visualization-5.html#cb436-11" tabindex="-1"></a>DA_venn_res <span class="ot">&lt;-</span> <span class="fu">plot_2DA_venn</span>(</span>
<span id="cb436-12"><a href="visualization-5.html#cb436-12" tabindex="-1"></a>                   <span class="at">daTest1 =</span> da_wilcox,</span>
<span id="cb436-13"><a href="visualization-5.html#cb436-13" tabindex="-1"></a>                   <span class="at">daTest2 =</span> da_ttest,</span>
<span id="cb436-14"><a href="visualization-5.html#cb436-14" tabindex="-1"></a>                   <span class="at">datType1 =</span> <span class="st">&quot;AA vs BB(wilcox)&quot;</span>,</span>
<span id="cb436-15"><a href="visualization-5.html#cb436-15" tabindex="-1"></a>                   <span class="at">datType2 =</span> <span class="st">&quot;AA vs BB(t-test)&quot;</span>,</span>
<span id="cb436-16"><a href="visualization-5.html#cb436-16" tabindex="-1"></a>                   <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;AA&quot;</span>, <span class="st">&quot;BB&quot;</span>),</span>
<span id="cb436-17"><a href="visualization-5.html#cb436-17" tabindex="-1"></a>                   <span class="at">Pvalue_name =</span> <span class="st">&quot;Pvalue&quot;</span>,</span>
<span id="cb436-18"><a href="visualization-5.html#cb436-18" tabindex="-1"></a>                   <span class="at">logFc_name1 =</span> <span class="st">&quot;Log2FoldChange (Rank)</span><span class="sc">\n</span><span class="st">AA_vs_BB&quot;</span>,</span>
<span id="cb436-19"><a href="visualization-5.html#cb436-19" tabindex="-1"></a>                   <span class="at">logFc_name2 =</span> <span class="st">&quot;Log2FoldChange (Mean)</span><span class="sc">\n</span><span class="st">AA_vs_BB&quot;</span>,</span>
<span id="cb436-20"><a href="visualization-5.html#cb436-20" tabindex="-1"></a>                   <span class="at">Pvalue_cutoff =</span> <span class="fl">0.8</span>,</span>
<span id="cb436-21"><a href="visualization-5.html#cb436-21" tabindex="-1"></a>                   <span class="at">logFC_cutoff =</span> <span class="fl">0.2</span>)</span>
<span id="cb436-22"><a href="visualization-5.html#cb436-22" tabindex="-1"></a>DA_venn_res<span class="sc">$</span>pl</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-234"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-234-1.png" alt="plot_2DA_venn (wilcox vs t_test)" width="576" />
<p class="caption">
Figure 11.36: plot_2DA_venn (wilcox vs t_test)
</p>
</div>
</div>
<div id="plot-the-da-results-from-the-significant-taxa-by-double-barplot" class="section level2 hasAnchor" number="11.12">
<h2><span class="header-section-number">11.12</span> plot the DA results from the significant taxa by double barplot<a href="visualization-5.html#plot-the-da-results-from-the-significant-taxa-by-double-barplot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb437"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb437-1"><a href="visualization-5.html#cb437-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;amplicon_ps&quot;</span>)</span>
<span id="cb437-2"><a href="visualization-5.html#cb437-2" tabindex="-1"></a>DA_res <span class="ot">&lt;-</span> <span class="fu">run_wilcox</span>(</span>
<span id="cb437-3"><a href="visualization-5.html#cb437-3" tabindex="-1"></a>              <span class="at">ps =</span> amplicon_ps, </span>
<span id="cb437-4"><a href="visualization-5.html#cb437-4" tabindex="-1"></a>              <span class="at">taxa_level =</span> <span class="st">&quot;Family&quot;</span>, </span>
<span id="cb437-5"><a href="visualization-5.html#cb437-5" tabindex="-1"></a>              <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>, </span>
<span id="cb437-6"><a href="visualization-5.html#cb437-6" tabindex="-1"></a>              <span class="at">group_names =</span> <span class="fu">c</span>(<span class="st">&quot;tongue&quot;</span>, <span class="st">&quot;gut&quot;</span>))</span>
<span id="cb437-7"><a href="visualization-5.html#cb437-7" tabindex="-1"></a></span>
<span id="cb437-8"><a href="visualization-5.html#cb437-8" tabindex="-1"></a><span class="fu">plot_double_barplot</span>(<span class="at">data =</span> DA_res,</span>
<span id="cb437-9"><a href="visualization-5.html#cb437-9" tabindex="-1"></a>              <span class="at">x_index =</span> <span class="st">&quot;Log2FoldChange (Rank)</span><span class="sc">\n</span><span class="st">tongue_vs_gut&quot;</span>,</span>
<span id="cb437-10"><a href="visualization-5.html#cb437-10" tabindex="-1"></a>              <span class="at">x_index_cutoff =</span> <span class="dv">1</span>,</span>
<span id="cb437-11"><a href="visualization-5.html#cb437-11" tabindex="-1"></a>              <span class="at">y_index =</span> <span class="st">&quot;AdjustedPvalue&quot;</span>,</span>
<span id="cb437-12"><a href="visualization-5.html#cb437-12" tabindex="-1"></a>              <span class="at">y_index_cutoff =</span> <span class="fl">0.05</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-235"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-235-1.png" alt="double barplot for DA results" width="672" />
<p class="caption">
Figure 11.37: double barplot for DA results
</p>
</div>
</div>
<div id="plot_stacked_bar_xivz" class="section level2 hasAnchor" number="11.13">
<h2><span class="header-section-number">11.13</span> plot_stacked_bar_XIVZ<a href="visualization-5.html#plot_stacked_bar_xivz" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Minimum usage: plot in relative abundance</li>
</ul>
<div class="sourceCode" id="cb438"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb438-1"><a href="visualization-5.html#cb438-1" tabindex="-1"></a><span class="fu">plot_stacked_bar_XIVZ</span>(<span class="at">phyloseq =</span> dada2_ps_rarefy, </span>
<span id="cb438-2"><a href="visualization-5.html#cb438-2" tabindex="-1"></a>                      <span class="at">level =</span> <span class="st">&quot;Family&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-236"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-236-1.png" alt="plot_stacked_bar_XIVZ (test1)" width="1152" />
<p class="caption">
Figure 11.38: plot_stacked_bar_XIVZ (test1)
</p>
</div>
<ul>
<li>Set feature parameter to show feature information</li>
</ul>
<div class="sourceCode" id="cb439"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb439-1"><a href="visualization-5.html#cb439-1" tabindex="-1"></a><span class="fu">plot_stacked_bar_XIVZ</span>(<span class="at">phyloseq =</span> dada2_ps_rarefy, </span>
<span id="cb439-2"><a href="visualization-5.html#cb439-2" tabindex="-1"></a>                 <span class="at">level =</span> <span class="st">&quot;Family&quot;</span>,</span>
<span id="cb439-3"><a href="visualization-5.html#cb439-3" tabindex="-1"></a>                 <span class="at">feature =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-237"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-237-1.png" alt="plot_stacked_bar_XIVZ (test2)" width="1152" />
<p class="caption">
Figure 11.39: plot_stacked_bar_XIVZ (test2)
</p>
</div>
<ul>
<li>Pass ordered sample names to order parameter to plot in specific order</li>
</ul>
<div class="sourceCode" id="cb440"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb440-1"><a href="visualization-5.html#cb440-1" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> phyloseq<span class="sc">::</span><span class="fu">sample_data</span>(dada2_ps_rarefy) <span class="sc">%&gt;%</span> </span>
<span id="cb440-2"><a href="visualization-5.html#cb440-2" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb440-3"><a href="visualization-5.html#cb440-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(Group)</span>
<span id="cb440-4"><a href="visualization-5.html#cb440-4" tabindex="-1"></a><span class="fu">plot_stacked_bar_XIVZ</span>(<span class="at">phyloseq =</span> dada2_ps_rarefy, </span>
<span id="cb440-5"><a href="visualization-5.html#cb440-5" tabindex="-1"></a>                 <span class="at">level =</span> <span class="st">&quot;Family&quot;</span>,</span>
<span id="cb440-6"><a href="visualization-5.html#cb440-6" tabindex="-1"></a>                 <span class="at">feature =</span> <span class="st">&quot;Group&quot;</span>,</span>
<span id="cb440-7"><a href="visualization-5.html#cb440-7" tabindex="-1"></a>                 <span class="at">order =</span> <span class="fu">rownames</span>(metadata))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-238"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-238-1.png" alt="plot_stacked_bar_XIVZ (test3)" width="1152" />
<p class="caption">
Figure 11.40: plot_stacked_bar_XIVZ (test3)
</p>
</div>
<ul>
<li>Use facet_wrap(vars(), scale=“free”) funciton to facet stacked barplot</li>
</ul>
<div class="sourceCode" id="cb441"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb441-1"><a href="visualization-5.html#cb441-1" tabindex="-1"></a><span class="fu">plot_stacked_bar_XIVZ</span>(<span class="at">phyloseq =</span> dada2_ps_rarefy, </span>
<span id="cb441-2"><a href="visualization-5.html#cb441-2" tabindex="-1"></a>                 <span class="at">level =</span> <span class="st">&quot;Family&quot;</span>, </span>
<span id="cb441-3"><a href="visualization-5.html#cb441-3" tabindex="-1"></a>                 <span class="at">relative_abundance =</span> <span class="cn">TRUE</span>, </span>
<span id="cb441-4"><a href="visualization-5.html#cb441-4" tabindex="-1"></a>                 <span class="at">order =</span> <span class="fu">rownames</span>(metadata)) <span class="sc">+</span> </span>
<span id="cb441-5"><a href="visualization-5.html#cb441-5" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="fu">vars</span>(Group), <span class="at">scale=</span><span class="st">&quot;free&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-239"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-239-1.png" alt="plot_stacked_bar_XIVZ (test4)" width="1152" />
<p class="caption">
Figure 11.41: plot_stacked_bar_XIVZ (test4)
</p>
</div>
</div>
<div id="plot_stackbarplot" class="section level2 hasAnchor" number="11.14">
<h2><span class="header-section-number">11.14</span> plot_StackBarPlot<a href="visualization-5.html#plot_stackbarplot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>plot_StackBarPlot</code> provides too many parameters for users to display the Stacked barplot of microbial composition by using ggplot2 format. Here is the ordinary pattern. More details to see <code>help(plot_StackBarPlot)</code>.</p>
<ul>
<li>Ordinary pattern</li>
</ul>
<div class="sourceCode" id="cb442"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb442-1"><a href="visualization-5.html#cb442-1" tabindex="-1"></a><span class="fu">plot_StackBarPlot</span>(<span class="at">ps =</span> amplicon_ps_rarefy, </span>
<span id="cb442-2"><a href="visualization-5.html#cb442-2" tabindex="-1"></a>                  <span class="at">taxa_level=</span><span class="st">&quot;Phylum&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-240"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-240-1.png" alt="plot_StackBarPlot(Ordinary pattern)" width="576" />
<p class="caption">
Figure 11.42: plot_StackBarPlot(Ordinary pattern)
</p>
</div>
<ul>
<li>Metadata with <code>SampleType</code> phenotype</li>
</ul>
<div class="sourceCode" id="cb443"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb443-1"><a href="visualization-5.html#cb443-1" tabindex="-1"></a><span class="fu">plot_StackBarPlot</span>(</span>
<span id="cb443-2"><a href="visualization-5.html#cb443-2" tabindex="-1"></a>        <span class="at">ps =</span> amplicon_ps_rarefy,</span>
<span id="cb443-3"><a href="visualization-5.html#cb443-3" tabindex="-1"></a>        <span class="at">taxa_level =</span> <span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb443-4"><a href="visualization-5.html#cb443-4" tabindex="-1"></a>        <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>)</span></code></pre></div>
<pre><code>## [1] &quot;This palatte have 20 colors!&quot;</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-241"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-241-1.png" alt="plot_StackBarPlot (Metadata with group)" width="624" />
<p class="caption">
Figure 11.43: plot_StackBarPlot (Metadata with group)
</p>
</div>
<ul>
<li>Metadata with <code>SampleType</code> phenotype in cluster mode</li>
</ul>
<div class="sourceCode" id="cb445"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb445-1"><a href="visualization-5.html#cb445-1" tabindex="-1"></a><span class="fu">plot_StackBarPlot</span>(</span>
<span id="cb445-2"><a href="visualization-5.html#cb445-2" tabindex="-1"></a>        <span class="at">ps =</span> amplicon_ps_rarefy,</span>
<span id="cb445-3"><a href="visualization-5.html#cb445-3" tabindex="-1"></a>        <span class="at">taxa_level =</span> <span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb445-4"><a href="visualization-5.html#cb445-4" tabindex="-1"></a>        <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb445-5"><a href="visualization-5.html#cb445-5" tabindex="-1"></a>        <span class="at">cluster =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## [1] &quot;This palatte have 20 colors!&quot;</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-242"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-242-1.png" alt="plot_StackBarPlot (Metadata with group in cluster mode)" width="624" />
<p class="caption">
Figure 11.44: plot_StackBarPlot (Metadata with group in cluster mode)
</p>
</div>
<ul>
<li>Metadata with <code>SampleType</code> phenotype in facet</li>
</ul>
<div class="sourceCode" id="cb447"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb447-1"><a href="visualization-5.html#cb447-1" tabindex="-1"></a><span class="fu">plot_StackBarPlot</span>(</span>
<span id="cb447-2"><a href="visualization-5.html#cb447-2" tabindex="-1"></a>        <span class="at">ps =</span> amplicon_ps_rarefy,</span>
<span id="cb447-3"><a href="visualization-5.html#cb447-3" tabindex="-1"></a>        <span class="at">taxa_level =</span> <span class="st">&quot;Phylum&quot;</span>,</span>
<span id="cb447-4"><a href="visualization-5.html#cb447-4" tabindex="-1"></a>        <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb447-5"><a href="visualization-5.html#cb447-5" tabindex="-1"></a>        <span class="at">facet =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-243"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-243-1.png" alt="plot_StackBarPlot (Metadata with group in facet)" width="672" />
<p class="caption">
Figure 11.45: plot_StackBarPlot (Metadata with group in facet)
</p>
</div>
<ul>
<li>Metadata with two groups to display samples</li>
</ul>
<div class="sourceCode" id="cb448"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb448-1"><a href="visualization-5.html#cb448-1" tabindex="-1"></a><span class="fu">plot_StackBarPlot</span>(</span>
<span id="cb448-2"><a href="visualization-5.html#cb448-2" tabindex="-1"></a>  <span class="at">ps =</span> amplicon_ps_rarefy,</span>
<span id="cb448-3"><a href="visualization-5.html#cb448-3" tabindex="-1"></a>  <span class="at">taxa_level =</span> <span class="st">&quot;Order&quot;</span>,</span>
<span id="cb448-4"><a href="visualization-5.html#cb448-4" tabindex="-1"></a>  <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb448-5"><a href="visualization-5.html#cb448-5" tabindex="-1"></a>  <span class="at">subgroup =</span> <span class="st">&quot;Year&quot;</span>)</span></code></pre></div>
<pre><code>## [1] &quot;This palatte have 19 colors!&quot;
## [1] &quot;This palatte have 20 colors!&quot;
## [1] &quot;This palatte have 20 colors!&quot;
## [1] &quot;This palatte have 20 colors!&quot;</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-244"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-244-1.png" alt="plot_StackBarPlot (Metadata with two groups ot display samples)" width="864" />
<p class="caption">
Figure 11.46: plot_StackBarPlot (Metadata with two groups ot display samples)
</p>
</div>
<ul>
<li>Metadata with three groups to display samples</li>
</ul>
<div class="sourceCode" id="cb450"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb450-1"><a href="visualization-5.html#cb450-1" tabindex="-1"></a><span class="fu">plot_StackBarPlot</span>(</span>
<span id="cb450-2"><a href="visualization-5.html#cb450-2" tabindex="-1"></a>  <span class="at">ps =</span> amplicon_ps_rarefy,</span>
<span id="cb450-3"><a href="visualization-5.html#cb450-3" tabindex="-1"></a>  <span class="at">taxa_level =</span> <span class="st">&quot;Order&quot;</span>,</span>
<span id="cb450-4"><a href="visualization-5.html#cb450-4" tabindex="-1"></a>  <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb450-5"><a href="visualization-5.html#cb450-5" tabindex="-1"></a>  <span class="at">subgroup =</span> <span class="fu">c</span>(<span class="st">&quot;Year&quot;</span>, <span class="st">&quot;Month&quot;</span>))</span></code></pre></div>
<pre><code>## [1] &quot;This palatte have 19 colors!&quot;
## [1] &quot;This palatte have 20 colors!&quot;
## [1] &quot;This palatte have 20 colors!&quot;
## [1] &quot;This palatte have 20 colors!&quot;
## [1] &quot;This palatte have 20 colors!&quot;
## [1] &quot;This palatte have 20 colors!&quot;</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-245"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-245-1.png" alt="plot_StackBarPlot (Metadata with three groups ot display samples)" width="864" />
<p class="caption">
Figure 11.47: plot_StackBarPlot (Metadata with three groups ot display samples)
</p>
</div>
<ul>
<li>Order SampleID by <em>orderSample</em> parameter</li>
</ul>
<div class="sourceCode" id="cb452"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb452-1"><a href="visualization-5.html#cb452-1" tabindex="-1"></a><span class="fu">plot_StackBarPlot</span>(</span>
<span id="cb452-2"><a href="visualization-5.html#cb452-2" tabindex="-1"></a>  <span class="at">ps =</span> amplicon_ps_rarefy,</span>
<span id="cb452-3"><a href="visualization-5.html#cb452-3" tabindex="-1"></a>  <span class="at">taxa_level =</span> <span class="st">&quot;Order&quot;</span>,</span>
<span id="cb452-4"><a href="visualization-5.html#cb452-4" tabindex="-1"></a>  <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb452-5"><a href="visualization-5.html#cb452-5" tabindex="-1"></a>  <span class="at">orderSample =</span> phyloseq<span class="sc">::</span><span class="fu">sample_names</span>(amplicon_ps)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>])</span></code></pre></div>
<pre><code>## [1] &quot;This palatte have 20 colors!&quot;</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-246"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-246-1.png" alt="Stacked barplot with Ordered Samples" width="768" />
<p class="caption">
Figure 11.48: Stacked barplot with Ordered Samples
</p>
</div>
<ul>
<li>Hide sample names by <em>sample_label</em> parameter</li>
</ul>
<div class="sourceCode" id="cb454"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb454-1"><a href="visualization-5.html#cb454-1" tabindex="-1"></a><span class="fu">plot_StackBarPlot</span>(</span>
<span id="cb454-2"><a href="visualization-5.html#cb454-2" tabindex="-1"></a>  <span class="at">ps =</span> amplicon_ps_rarefy,</span>
<span id="cb454-3"><a href="visualization-5.html#cb454-3" tabindex="-1"></a>  <span class="at">taxa_level =</span> <span class="st">&quot;Order&quot;</span>,</span>
<span id="cb454-4"><a href="visualization-5.html#cb454-4" tabindex="-1"></a>  <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb454-5"><a href="visualization-5.html#cb454-5" tabindex="-1"></a>  <span class="at">orderSample =</span> phyloseq<span class="sc">::</span><span class="fu">sample_names</span>(amplicon_ps)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],</span>
<span id="cb454-6"><a href="visualization-5.html#cb454-6" tabindex="-1"></a>  <span class="at">sample_label =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## [1] &quot;This palatte have 20 colors!&quot;</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-247"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-247-1.png" alt="Stacked barplot with hiding Samples' names" width="768" />
<p class="caption">
Figure 11.49: Stacked barplot with hiding Samples’ names
</p>
</div>
</div>
<div id="color-palettes" class="section level2 hasAnchor" number="11.15">
<h2><span class="header-section-number">11.15</span> Color Palettes<a href="visualization-5.html#color-palettes" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="wes-anderson-palettes" class="section level3 hasAnchor" number="11.15.1">
<h3><span class="header-section-number">11.15.1</span> Wes Anderson Palettes<a href="visualization-5.html#wes-anderson-palettes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><a href="https://github.com/karthik/wesanderson">Wes Anderson Palettes</a> is from <strong>wesanderson</strong> package and we have integrated it into <strong>XMAS2.0</strong>.</p>
<div class="sourceCode" id="cb456"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb456-1"><a href="visualization-5.html#cb456-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;amplicon_ps&quot;</span>)</span>
<span id="cb456-2"><a href="visualization-5.html#cb456-2" tabindex="-1"></a>dat_alpha <span class="ot">&lt;-</span> <span class="fu">run_alpha_diversity</span>(<span class="at">ps =</span> amplicon_ps, <span class="at">measures =</span> <span class="fu">c</span>(<span class="st">&quot;Shannon&quot;</span>, <span class="st">&quot;Chao1&quot;</span>))</span>
<span id="cb456-3"><a href="visualization-5.html#cb456-3" tabindex="-1"></a></span>
<span id="cb456-4"><a href="visualization-5.html#cb456-4" tabindex="-1"></a><span class="co"># origin</span></span>
<span id="cb456-5"><a href="visualization-5.html#cb456-5" tabindex="-1"></a>pl_origin <span class="ot">&lt;-</span> <span class="fu">plot_boxplot</span>(<span class="at">data =</span> dat_alpha,</span>
<span id="cb456-6"><a href="visualization-5.html#cb456-6" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="st">&quot;Shannon&quot;</span>,</span>
<span id="cb456-7"><a href="visualization-5.html#cb456-7" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb456-8"><a href="visualization-5.html#cb456-8" tabindex="-1"></a>             <span class="at">do_test =</span> <span class="cn">TRUE</span>,</span>
<span id="cb456-9"><a href="visualization-5.html#cb456-9" tabindex="-1"></a>             <span class="at">cmp_list =</span> <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">&quot;gut&quot;</span>, <span class="st">&quot;right palm&quot;</span>), <span class="fu">c</span>(<span class="st">&quot;gut&quot;</span>, <span class="st">&quot;left palm&quot;</span>)),</span>
<span id="cb456-10"><a href="visualization-5.html#cb456-10" tabindex="-1"></a>             <span class="at">method =</span> <span class="st">&quot;wilcox.test&quot;</span>)</span>
<span id="cb456-11"><a href="visualization-5.html#cb456-11" tabindex="-1"></a></span>
<span id="cb456-12"><a href="visualization-5.html#cb456-12" tabindex="-1"></a><span class="co"># Wes Anderson Palettes</span></span>
<span id="cb456-13"><a href="visualization-5.html#cb456-13" tabindex="-1"></a>pal <span class="ot">&lt;-</span> <span class="fu">wes_palette</span>(<span class="at">name =</span> <span class="st">&quot;GrandBudapest1&quot;</span>, <span class="dv">4</span>, <span class="at">type =</span> <span class="st">&quot;discrete&quot;</span>)</span>
<span id="cb456-14"><a href="visualization-5.html#cb456-14" tabindex="-1"></a>pl_wes <span class="ot">&lt;-</span> <span class="fu">plot_boxplot</span>(<span class="at">data =</span> dat_alpha,</span>
<span id="cb456-15"><a href="visualization-5.html#cb456-15" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="st">&quot;Shannon&quot;</span>,</span>
<span id="cb456-16"><a href="visualization-5.html#cb456-16" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb456-17"><a href="visualization-5.html#cb456-17" tabindex="-1"></a>             <span class="at">group_color =</span> pal,</span>
<span id="cb456-18"><a href="visualization-5.html#cb456-18" tabindex="-1"></a>             <span class="at">do_test =</span> <span class="cn">TRUE</span>,</span>
<span id="cb456-19"><a href="visualization-5.html#cb456-19" tabindex="-1"></a>             <span class="at">cmp_list =</span> <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">&quot;gut&quot;</span>, <span class="st">&quot;right palm&quot;</span>), <span class="fu">c</span>(<span class="st">&quot;gut&quot;</span>, <span class="st">&quot;left palm&quot;</span>)),</span>
<span id="cb456-20"><a href="visualization-5.html#cb456-20" tabindex="-1"></a>             <span class="at">method =</span> <span class="st">&quot;wilcox.test&quot;</span>)</span>
<span id="cb456-21"><a href="visualization-5.html#cb456-21" tabindex="-1"></a></span>
<span id="cb456-22"><a href="visualization-5.html#cb456-22" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(pl_origin, pl_wes,</span>
<span id="cb456-23"><a href="visualization-5.html#cb456-23" tabindex="-1"></a>                   <span class="at">align =</span> <span class="st">&quot;hv&quot;</span>,</span>
<span id="cb456-24"><a href="visualization-5.html#cb456-24" tabindex="-1"></a>                   <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;Origin&quot;</span>, <span class="st">&quot;Wes Anderson&quot;</span>))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-248"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-248-1.png" alt="Wes Anderson Palettes" width="864" />
<p class="caption">
Figure 11.50: Wes Anderson Palettes
</p>
</div>
</div>
<div id="usemycol" class="section level3 hasAnchor" number="11.15.2">
<h3><span class="header-section-number">11.15.2</span> useMyCol<a href="visualization-5.html#usemycol" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb457"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb457-1"><a href="visualization-5.html#cb457-1" tabindex="-1"></a>pal <span class="ot">&lt;-</span> <span class="fu">useMyCol</span>(<span class="at">platte =</span> <span class="st">&quot;stallion&quot;</span>, <span class="at">n =</span> <span class="fu">length</span>(<span class="fu">unique</span>(dat_alpha<span class="sc">$</span>SampleType)))</span></code></pre></div>
<pre><code>## [1] &quot;This palatte have 20 colors!&quot;</code></pre>
<div class="sourceCode" id="cb459"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb459-1"><a href="visualization-5.html#cb459-1" tabindex="-1"></a><span class="fu">plot_boxplot</span>(<span class="at">data =</span> dat_alpha,</span>
<span id="cb459-2"><a href="visualization-5.html#cb459-2" tabindex="-1"></a>             <span class="at">y_index =</span> <span class="st">&quot;Shannon&quot;</span>,</span>
<span id="cb459-3"><a href="visualization-5.html#cb459-3" tabindex="-1"></a>             <span class="at">group =</span> <span class="st">&quot;SampleType&quot;</span>,</span>
<span id="cb459-4"><a href="visualization-5.html#cb459-4" tabindex="-1"></a>             <span class="at">do_test =</span> <span class="cn">TRUE</span>,</span>
<span id="cb459-5"><a href="visualization-5.html#cb459-5" tabindex="-1"></a>             <span class="at">group_color =</span> pal,</span>
<span id="cb459-6"><a href="visualization-5.html#cb459-6" tabindex="-1"></a>             <span class="at">cmp_list =</span> <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">&quot;gut&quot;</span>, <span class="st">&quot;right palm&quot;</span>), <span class="fu">c</span>(<span class="st">&quot;gut&quot;</span>, <span class="st">&quot;left palm&quot;</span>)),</span>
<span id="cb459-7"><a href="visualization-5.html#cb459-7" tabindex="-1"></a>             <span class="at">method =</span> <span class="st">&quot;wilcox.test&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-249"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-249-1.png" alt="useMyCol Palettes" width="864" />
<p class="caption">
Figure 11.51: useMyCol Palettes
</p>
</div>
</div>
</div>
<div id="ordination-plots-with-ggplot2" class="section level2 hasAnchor" number="11.16">
<h2><span class="header-section-number">11.16</span> Ordination plots with ggplot2<a href="visualization-5.html#ordination-plots-with-ggplot2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="principal-components-analysis-with-the-iris-data-set" class="section level3 hasAnchor" number="11.16.1">
<h3><span class="header-section-number">11.16.1</span> principal components analysis with the iris data set<a href="visualization-5.html#principal-components-analysis-with-the-iris-data-set" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb460"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb460-1"><a href="visualization-5.html#cb460-1" tabindex="-1"></a>ord <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(iris[, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>])</span>
<span id="cb460-2"><a href="visualization-5.html#cb460-2" tabindex="-1"></a></span>
<span id="cb460-3"><a href="visualization-5.html#cb460-3" tabindex="-1"></a><span class="fu">ggord</span>(ord, iris<span class="sc">$</span>Species, <span class="at">cols =</span> <span class="fu">c</span>(<span class="st">&#39;purple&#39;</span>, <span class="st">&#39;orange&#39;</span>, <span class="st">&#39;blue&#39;</span>)) <span class="sc">+</span> </span>
<span id="cb460-4"><a href="visualization-5.html#cb460-4" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="st">&#39;Groups&#39;</span>, <span class="at">values =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>)) <span class="sc">+</span> </span>
<span id="cb460-5"><a href="visualization-5.html#cb460-5" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span> </span>
<span id="cb460-6"><a href="visualization-5.html#cb460-6" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&#39;top&#39;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-250"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-250-1.png" alt="principal components analysis" width="672" />
<p class="caption">
Figure 11.52: principal components analysis
</p>
</div>
</div>
<div id="multiple-correspondence-analysis-with-the-tea-dataset" class="section level3 hasAnchor" number="11.16.2">
<h3><span class="header-section-number">11.16.2</span> multiple correspondence analysis with the tea dataset<a href="visualization-5.html#multiple-correspondence-analysis-with-the-tea-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb461"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb461-1"><a href="visualization-5.html#cb461-1" tabindex="-1"></a><span class="fu">data</span>(tea, <span class="at">package =</span> <span class="st">&#39;FactoMineR&#39;</span>)</span>
<span id="cb461-2"><a href="visualization-5.html#cb461-2" tabindex="-1"></a>tea <span class="ot">&lt;-</span> tea[, <span class="fu">c</span>(<span class="st">&#39;Tea&#39;</span>, <span class="st">&#39;sugar&#39;</span>, <span class="st">&#39;price&#39;</span>, <span class="st">&#39;age_Q&#39;</span>, <span class="st">&#39;sex&#39;</span>)]</span>
<span id="cb461-3"><a href="visualization-5.html#cb461-3" tabindex="-1"></a></span>
<span id="cb461-4"><a href="visualization-5.html#cb461-4" tabindex="-1"></a>ord <span class="ot">&lt;-</span> FactoMineR<span class="sc">::</span><span class="fu">MCA</span>(tea[, <span class="sc">-</span><span class="dv">1</span>], <span class="at">graph =</span> <span class="cn">FALSE</span>)</span>
<span id="cb461-5"><a href="visualization-5.html#cb461-5" tabindex="-1"></a></span>
<span id="cb461-6"><a href="visualization-5.html#cb461-6" tabindex="-1"></a><span class="fu">ggord</span>(ord, tea<span class="sc">$</span>Tea, <span class="at">parse =</span> <span class="cn">FALSE</span>) <span class="co"># use parse = FALSE for labels with non alphanumeric characters</span></span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-251"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-251-1.png" alt="multiple correspondence analysis" width="672" />
<p class="caption">
Figure 11.53: multiple correspondence analysis
</p>
</div>
</div>
<div id="linear-discriminant-analysis" class="section level3 hasAnchor" number="11.16.3">
<h3><span class="header-section-number">11.16.3</span> linear discriminant analysis<a href="visualization-5.html#linear-discriminant-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb462"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb462-1"><a href="visualization-5.html#cb462-1" tabindex="-1"></a>ord <span class="ot">&lt;-</span> MASS<span class="sc">::</span><span class="fu">lda</span>(Species <span class="sc">~</span> ., iris, <span class="at">prior =</span> <span class="fu">rep</span>(<span class="dv">1</span>, <span class="dv">3</span>)<span class="sc">/</span><span class="dv">3</span>)</span>
<span id="cb462-2"><a href="visualization-5.html#cb462-2" tabindex="-1"></a></span>
<span id="cb462-3"><a href="visualization-5.html#cb462-3" tabindex="-1"></a><span class="fu">ggord</span>(ord, iris<span class="sc">$</span>Species)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-252"></span>
<img src="XMAS-2.0-Tutorial_files/figure-html/unnamed-chunk-252-1.png" alt="linear discriminant analysis" width="672" />
<p class="caption">
Figure 11.54: linear discriminant analysis
</p>
</div>
</div>
</div>
<div id="systematic-information-8" class="section level2 hasAnchor" number="11.17">
<h2><span class="header-section-number">11.17</span> Systematic Information<a href="visualization-5.html#systematic-information-8" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb463"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb463-1"><a href="visualization-5.html#cb463-1" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 4.1.3 (2022-03-10)
##  os       macOS Monterey 12.2.1
##  system   x86_64, darwin17.0
##  ui       RStudio
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       Asia/Shanghai
##  date     2023-11-30
##  rstudio  2023.09.0+463 Desert Sunflower (desktop)
##  pandoc   3.1.1 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/ (via rmarkdown)
## 
## ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package              * version   date (UTC) lib source
##  abind                  1.4-5     2016-07-21 [2] CRAN (R 4.1.0)
##  ade4                   1.7-22    2023-02-06 [2] CRAN (R 4.1.2)
##  ALDEx2                 1.30.0    2022-11-01 [2] Bioconductor
##  annotate               1.72.0    2021-10-26 [2] Bioconductor
##  AnnotationDbi          1.60.2    2023-03-10 [2] Bioconductor
##  ape                  * 5.7-1     2023-03-13 [2] CRAN (R 4.1.2)
##  askpass                1.1       2019-01-13 [2] CRAN (R 4.1.0)
##  backports              1.4.1     2021-12-13 [2] CRAN (R 4.1.0)
##  base64enc              0.1-3     2015-07-28 [2] CRAN (R 4.1.0)
##  bayesm                 3.1-5     2022-12-02 [2] CRAN (R 4.1.2)
##  Biobase                2.54.0    2021-10-26 [2] Bioconductor
##  BiocGenerics           0.40.0    2021-10-26 [2] Bioconductor
##  BiocParallel           1.28.3    2021-12-09 [2] Bioconductor
##  biomformat             1.22.0    2021-10-26 [2] Bioconductor
##  Biostrings             2.62.0    2021-10-26 [2] Bioconductor
##  bit                    4.0.5     2022-11-15 [2] CRAN (R 4.1.2)
##  bit64                  4.0.5     2020-08-30 [2] CRAN (R 4.1.0)
##  bitops                 1.0-7     2021-04-24 [2] CRAN (R 4.1.0)
##  blob                   1.2.4     2023-03-17 [2] CRAN (R 4.1.2)
##  bookdown               0.34      2023-05-09 [2] CRAN (R 4.1.2)
##  broom                  1.0.5     2023-06-09 [2] CRAN (R 4.1.3)
##  bslib                  0.6.0     2023-11-21 [1] CRAN (R 4.1.3)
##  cachem                 1.0.8     2023-05-01 [2] CRAN (R 4.1.2)
##  callr                  3.7.3     2022-11-02 [2] CRAN (R 4.1.2)
##  car                    3.1-2     2023-03-30 [2] CRAN (R 4.1.2)
##  carData                3.0-5     2022-01-06 [2] CRAN (R 4.1.2)
##  caTools                1.18.2    2021-03-28 [2] CRAN (R 4.1.0)
##  checkmate              2.2.0     2023-04-27 [2] CRAN (R 4.1.2)
##  class                  7.3-22    2023-05-03 [2] CRAN (R 4.1.2)
##  classInt               0.4-9     2023-02-28 [2] CRAN (R 4.1.2)
##  cli                    3.6.1     2023-03-23 [2] CRAN (R 4.1.2)
##  cluster                2.1.4     2022-08-22 [2] CRAN (R 4.1.2)
##  coda                   0.19-4    2020-09-30 [2] CRAN (R 4.1.0)
##  codetools              0.2-19    2023-02-01 [2] CRAN (R 4.1.2)
##  coin                   1.4-2     2021-10-08 [2] CRAN (R 4.1.0)
##  colorspace             2.1-0     2023-01-23 [2] CRAN (R 4.1.2)
##  compositions           2.0-6     2023-04-13 [2] CRAN (R 4.1.2)
##  conflicted           * 1.2.0     2023-02-01 [2] CRAN (R 4.1.2)
##  corrplot               0.92      2021-11-18 [2] CRAN (R 4.1.0)
##  cowplot                1.1.1     2020-12-30 [2] CRAN (R 4.1.0)
##  crayon                 1.5.2     2022-09-29 [2] CRAN (R 4.1.2)
##  crosstalk              1.2.0     2021-11-04 [2] CRAN (R 4.1.0)
##  data.table             1.14.8    2023-02-17 [2] CRAN (R 4.1.2)
##  DBI                    1.1.3     2022-06-18 [2] CRAN (R 4.1.2)
##  DelayedArray           0.20.0    2021-10-26 [2] Bioconductor
##  DEoptimR               1.0-14    2023-06-09 [2] CRAN (R 4.1.3)
##  DESeq2                 1.34.0    2021-10-26 [2] Bioconductor
##  devtools               2.4.5     2022-10-11 [2] CRAN (R 4.1.2)
##  digest                 0.6.33    2023-07-07 [1] CRAN (R 4.1.3)
##  dplyr                * 1.1.2     2023-04-20 [2] CRAN (R 4.1.2)
##  DT                     0.28      2023-05-18 [2] CRAN (R 4.1.3)
##  e1071                  1.7-13    2023-02-01 [2] CRAN (R 4.1.2)
##  edgeR                  3.36.0    2021-10-26 [2] Bioconductor
##  ellipsis               0.3.2     2021-04-29 [2] CRAN (R 4.1.0)
##  emmeans                1.8.7     2023-06-23 [1] CRAN (R 4.1.3)
##  estimability           1.4.1     2022-08-05 [2] CRAN (R 4.1.2)
##  evaluate               0.21      2023-05-05 [2] CRAN (R 4.1.2)
##  FactoMineR             2.8       2023-03-27 [2] CRAN (R 4.1.2)
##  fansi                  1.0.4     2023-01-22 [2] CRAN (R 4.1.2)
##  farver                 2.1.1     2022-07-06 [2] CRAN (R 4.1.2)
##  fastmap                1.1.1     2023-02-24 [2] CRAN (R 4.1.2)
##  flashClust             1.01-2    2012-08-21 [2] CRAN (R 4.1.0)
##  foreach                1.5.2     2022-02-02 [2] CRAN (R 4.1.2)
##  foreign                0.8-84    2022-12-06 [2] CRAN (R 4.1.2)
##  Formula                1.2-5     2023-02-24 [2] CRAN (R 4.1.2)
##  fs                     1.6.2     2023-04-25 [2] CRAN (R 4.1.2)
##  genefilter             1.76.0    2021-10-26 [2] Bioconductor
##  geneplotter            1.72.0    2021-10-26 [2] Bioconductor
##  generics               0.1.3     2022-07-05 [2] CRAN (R 4.1.2)
##  GenomeInfoDb           1.30.1    2022-01-30 [2] Bioconductor
##  GenomeInfoDbData       1.2.7     2022-03-09 [2] Bioconductor
##  GenomicRanges          1.46.1    2021-11-18 [2] Bioconductor
##  ggiraph                0.8.7     2023-03-17 [2] CRAN (R 4.1.2)
##  ggiraphExtra           0.3.0     2020-10-06 [2] CRAN (R 4.1.2)
##  ggplot2              * 3.4.2     2023-04-03 [2] CRAN (R 4.1.2)
##  ggpubr               * 0.6.0     2023-02-10 [2] CRAN (R 4.1.2)
##  ggrepel                0.9.3     2023-02-03 [2] CRAN (R 4.1.2)
##  ggsci                  3.0.0     2023-03-08 [2] CRAN (R 4.1.2)
##  ggsignif               0.6.4     2022-10-13 [2] CRAN (R 4.1.2)
##  ggVennDiagram          1.2.2     2022-09-08 [2] CRAN (R 4.1.2)
##  glmnet                 4.1-7     2023-03-23 [2] CRAN (R 4.1.2)
##  glue                   1.6.2     2022-02-24 [2] CRAN (R 4.1.2)
##  gplots                 3.1.3     2022-04-25 [2] CRAN (R 4.1.2)
##  gridExtra              2.3       2017-09-09 [2] CRAN (R 4.1.0)
##  gtable                 0.3.3     2023-03-21 [2] CRAN (R 4.1.2)
##  gtools                 3.9.4     2022-11-27 [2] CRAN (R 4.1.2)
##  highr                  0.10      2022-12-22 [2] CRAN (R 4.1.2)
##  Hmisc                  5.1-0     2023-05-08 [2] CRAN (R 4.1.2)
##  htmlTable              2.4.1     2022-07-07 [2] CRAN (R 4.1.2)
##  htmltools              0.5.7     2023-11-03 [1] CRAN (R 4.1.3)
##  htmlwidgets            1.6.2     2023-03-17 [2] CRAN (R 4.1.2)
##  httpuv                 1.6.11    2023-05-11 [2] CRAN (R 4.1.3)
##  httr                   1.4.6     2023-05-08 [2] CRAN (R 4.1.2)
##  igraph                 1.5.0     2023-06-16 [1] CRAN (R 4.1.3)
##  insight                0.19.3    2023-06-29 [2] CRAN (R 4.1.3)
##  IRanges                2.28.0    2021-10-26 [2] Bioconductor
##  iterators              1.0.14    2022-02-05 [2] CRAN (R 4.1.2)
##  jquerylib              0.1.4     2021-04-26 [2] CRAN (R 4.1.0)
##  jsonlite               1.8.7     2023-06-29 [2] CRAN (R 4.1.3)
##  kableExtra             1.3.4     2021-02-20 [2] CRAN (R 4.1.2)
##  KEGGREST               1.34.0    2021-10-26 [2] Bioconductor
##  KernSmooth             2.23-22   2023-07-10 [2] CRAN (R 4.1.3)
##  knitr                  1.43      2023-05-25 [2] CRAN (R 4.1.3)
##  labeling               0.4.2     2020-10-20 [2] CRAN (R 4.1.0)
##  later                  1.3.1     2023-05-02 [2] CRAN (R 4.1.2)
##  lattice              * 0.21-8    2023-04-05 [2] CRAN (R 4.1.2)
##  leaps                  3.1       2020-01-16 [2] CRAN (R 4.1.0)
##  libcoin                1.0-9     2021-09-27 [2] CRAN (R 4.1.0)
##  lifecycle              1.0.3     2022-10-07 [2] CRAN (R 4.1.2)
##  limma                  3.50.3    2022-04-07 [2] Bioconductor
##  locfit                 1.5-9.8   2023-06-11 [2] CRAN (R 4.1.3)
##  LOCOM                  1.1       2022-08-05 [2] Github (yijuanhu/LOCOM@c181e0f)
##  magrittr               2.0.3     2022-03-30 [2] CRAN (R 4.1.2)
##  MASS                   7.3-60    2023-05-04 [2] CRAN (R 4.1.2)
##  Matrix                 1.6-0     2023-07-08 [2] CRAN (R 4.1.3)
##  MatrixGenerics         1.6.0     2021-10-26 [2] Bioconductor
##  matrixStats            1.0.0     2023-06-02 [2] CRAN (R 4.1.3)
##  mbzinb                 0.2       2022-03-16 [2] local
##  memoise                2.0.1     2021-11-26 [2] CRAN (R 4.1.0)
##  metagenomeSeq          1.36.0    2021-10-26 [2] Bioconductor
##  mgcv                   1.8-42    2023-03-02 [2] CRAN (R 4.1.2)
##  microbiome             1.16.0    2021-10-26 [2] Bioconductor
##  mime                   0.12      2021-09-28 [2] CRAN (R 4.1.0)
##  miniUI                 0.1.1.1   2018-05-18 [2] CRAN (R 4.1.0)
##  modeltools             0.2-23    2020-03-05 [2] CRAN (R 4.1.0)
##  multcomp               1.4-25    2023-06-20 [2] CRAN (R 4.1.3)
##  multcompView           0.1-9     2023-04-09 [2] CRAN (R 4.1.2)
##  multtest               2.50.0    2021-10-26 [2] Bioconductor
##  munsell                0.5.0     2018-06-12 [2] CRAN (R 4.1.0)
##  mvtnorm                1.2-2     2023-06-08 [2] CRAN (R 4.1.3)
##  mycor                  0.1.1     2018-04-10 [2] CRAN (R 4.1.0)
##  NADA                   1.6-1.1   2020-03-22 [2] CRAN (R 4.1.0)
##  nlme                 * 3.1-162   2023-01-31 [2] CRAN (R 4.1.2)
##  nnet                   7.3-19    2023-05-03 [2] CRAN (R 4.1.2)
##  openssl                2.0.6     2023-03-09 [2] CRAN (R 4.1.2)
##  permute              * 0.9-7     2022-01-27 [2] CRAN (R 4.1.2)
##  pheatmap               1.0.12    2019-01-04 [2] CRAN (R 4.1.0)
##  phyloseq             * 1.38.0    2021-10-26 [2] Bioconductor
##  picante              * 1.8.2     2020-06-10 [2] CRAN (R 4.1.0)
##  pillar                 1.9.0     2023-03-22 [2] CRAN (R 4.1.2)
##  pkgbuild               1.4.2     2023-06-26 [2] CRAN (R 4.1.3)
##  pkgconfig              2.0.3     2019-09-22 [2] CRAN (R 4.1.0)
##  pkgload                1.3.2.1   2023-07-08 [2] CRAN (R 4.1.3)
##  plyr                   1.8.8     2022-11-11 [2] CRAN (R 4.1.2)
##  png                    0.1-8     2022-11-29 [2] CRAN (R 4.1.2)
##  ppcor                  1.1       2015-12-03 [2] CRAN (R 4.1.0)
##  prettyunits            1.1.1     2020-01-24 [2] CRAN (R 4.1.0)
##  processx               3.8.2     2023-06-30 [2] CRAN (R 4.1.3)
##  profvis                0.3.8     2023-05-02 [2] CRAN (R 4.1.2)
##  promises               1.2.0.1   2021-02-11 [2] CRAN (R 4.1.0)
##  protoclust             1.6.4     2022-04-01 [2] CRAN (R 4.1.2)
##  proxy                  0.4-27    2022-06-09 [2] CRAN (R 4.1.2)
##  ps                     1.7.5     2023-04-18 [2] CRAN (R 4.1.2)
##  pscl                   1.5.5.1   2023-05-10 [2] CRAN (R 4.1.2)
##  purrr                  1.0.1     2023-01-10 [2] CRAN (R 4.1.2)
##  qvalue                 2.26.0    2021-10-26 [2] Bioconductor
##  R6                     2.5.1     2021-08-19 [2] CRAN (R 4.1.0)
##  RAIDA                  1.0       2022-03-14 [2] local
##  RColorBrewer         * 1.1-3     2022-04-03 [2] CRAN (R 4.1.2)
##  Rcpp                   1.0.11    2023-07-06 [1] CRAN (R 4.1.3)
##  RcppZiggurat           0.1.6     2020-10-20 [2] CRAN (R 4.1.0)
##  RCurl                  1.98-1.12 2023-03-27 [2] CRAN (R 4.1.2)
##  remotes                2.4.2     2021-11-30 [2] CRAN (R 4.1.0)
##  reshape2               1.4.4     2020-04-09 [2] CRAN (R 4.1.0)
##  reticulate             1.30      2023-06-09 [2] CRAN (R 4.1.3)
##  Rfast                  2.0.8     2023-07-03 [2] CRAN (R 4.1.3)
##  rhdf5                  2.38.1    2022-03-10 [2] Bioconductor
##  rhdf5filters           1.6.0     2021-10-26 [2] Bioconductor
##  Rhdf5lib               1.16.0    2021-10-26 [2] Bioconductor
##  rlang                  1.1.1     2023-04-28 [1] CRAN (R 4.1.2)
##  rmarkdown              2.23      2023-07-01 [2] CRAN (R 4.1.3)
##  robustbase             0.99-0    2023-06-16 [2] CRAN (R 4.1.3)
##  rpart                  4.1.19    2022-10-21 [2] CRAN (R 4.1.2)
##  RSpectra               0.16-1    2022-04-24 [2] CRAN (R 4.1.2)
##  RSQLite                2.3.1     2023-04-03 [2] CRAN (R 4.1.2)
##  rstatix                0.7.2     2023-02-01 [2] CRAN (R 4.1.2)
##  rstudioapi             0.15.0    2023-07-07 [2] CRAN (R 4.1.3)
##  Rtsne                  0.16      2022-04-17 [2] CRAN (R 4.1.2)
##  RVenn                  1.1.0     2019-07-18 [2] CRAN (R 4.1.0)
##  rvest                  1.0.3     2022-08-19 [2] CRAN (R 4.1.2)
##  S4Vectors              0.32.4    2022-03-29 [2] Bioconductor
##  sandwich               3.0-2     2022-06-15 [2] CRAN (R 4.1.2)
##  sass                   0.4.6     2023-05-03 [2] CRAN (R 4.1.2)
##  scales                 1.2.1     2022-08-20 [2] CRAN (R 4.1.2)
##  scatterplot3d          0.3-44    2023-05-05 [2] CRAN (R 4.1.2)
##  sessioninfo            1.2.2     2021-12-06 [2] CRAN (R 4.1.0)
##  sf                     1.0-7     2022-03-07 [2] CRAN (R 4.1.2)
##  shape                  1.4.6     2021-05-19 [2] CRAN (R 4.1.0)
##  shiny                  1.7.4.1   2023-07-06 [2] CRAN (R 4.1.3)
##  sjlabelled             1.2.0     2022-04-10 [2] CRAN (R 4.1.2)
##  sjmisc                 2.8.9     2021-12-03 [2] CRAN (R 4.1.0)
##  stringi                1.7.12    2023-01-11 [2] CRAN (R 4.1.2)
##  stringr                1.5.0     2022-12-02 [2] CRAN (R 4.1.2)
##  SummarizedExperiment   1.24.0    2021-10-26 [2] Bioconductor
##  survival               3.5-5     2023-03-12 [2] CRAN (R 4.1.2)
##  svglite                2.1.1     2023-01-10 [2] CRAN (R 4.1.2)
##  systemfonts            1.0.4     2022-02-11 [2] CRAN (R 4.1.2)
##  tensorA                0.36.2    2020-11-19 [2] CRAN (R 4.1.0)
##  TH.data                1.1-2     2023-04-17 [2] CRAN (R 4.1.2)
##  tibble               * 3.2.1     2023-03-20 [2] CRAN (R 4.1.2)
##  tidyr                  1.3.0     2023-01-24 [2] CRAN (R 4.1.2)
##  tidyselect             1.2.0     2022-10-10 [2] CRAN (R 4.1.2)
##  truncnorm              1.0-9     2023-03-20 [2] CRAN (R 4.1.2)
##  umap                   0.2.10.0  2023-02-01 [2] CRAN (R 4.1.2)
##  units                  0.8-2     2023-04-27 [2] CRAN (R 4.1.2)
##  urlchecker             1.0.1     2021-11-30 [2] CRAN (R 4.1.0)
##  usethis                2.2.2     2023-07-06 [2] CRAN (R 4.1.3)
##  utf8                   1.2.3     2023-01-31 [2] CRAN (R 4.1.2)
##  uuid                   1.1-0     2022-04-19 [2] CRAN (R 4.1.2)
##  vctrs                  0.6.3     2023-06-14 [1] CRAN (R 4.1.3)
##  vegan                * 2.6-4     2022-10-11 [2] CRAN (R 4.1.2)
##  viridis              * 0.6.3     2023-05-03 [2] CRAN (R 4.1.2)
##  viridisLite          * 0.4.2     2023-05-02 [2] CRAN (R 4.1.2)
##  webshot                0.5.5     2023-06-26 [2] CRAN (R 4.1.3)
##  withr                  2.5.0     2022-03-03 [2] CRAN (R 4.1.2)
##  Wrench                 1.12.0    2021-10-26 [2] Bioconductor
##  xfun                   0.40      2023-08-09 [1] CRAN (R 4.1.3)
##  XMAS2                * 2.2.0     2023-11-30 [1] local
##  XML                    3.99-0.14 2023-03-19 [2] CRAN (R 4.1.2)
##  xml2                   1.3.5     2023-07-06 [2] CRAN (R 4.1.3)
##  xtable                 1.8-4     2019-04-21 [2] CRAN (R 4.1.0)
##  XVector                0.34.0    2021-10-26 [2] Bioconductor
##  yaml                   2.3.7     2023-01-23 [2] CRAN (R 4.1.2)
##  zCompositions          1.4.0-1   2022-03-26 [2] CRAN (R 4.1.2)
##  zlibbioc               1.40.0    2021-10-26 [2] Bioconductor
##  zoo                    1.8-12    2023-04-13 [2] CRAN (R 4.1.2)
## 
##  [1] /Users/zouhua/Library/R/x86_64/4.1/library
##  [2] /Library/Frameworks/R.framework/Versions/4.1/Resources/library
## 
## ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="differential-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="association-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/XMAS_tutorial/blob/main/10-Visualization.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["XMAS 2.0 Tutorial.pdf", "XMAS 2.0 Tutorial.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true,
"toc_depth": 3
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
